
/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
DROP TABLE IF EXISTS `domain`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `domain` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `hit_id` varchar(25) NOT NULL,
  `description` blob,
  `DomainID` varchar(10) DEFAULT NULL,
  `Name` varchar(25) DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `hit_id` (`hit_id`)
) ENGINE=InnoDB AUTO_INCREMENT=1843545 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `domain` WRITE;
/*!40000 ALTER TABLE `domain` DISABLE KEYS */;
INSERT INTO `domain` VALUES (955386,'gnl|CDD|273690',_binary 'phage major capsid protein, HK97 family.  This model family represents the major capsid protein component of the heads (capsids) of bacteriophage HK97, phi-105, P27, and related phage. This model represents one of several analogous families lacking detectable sequence similarity. The gene encoding this component is typically located in an operon encoding the small and large terminase subunits, the portal protein and the prohead or maturation protease. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01554','prophage_Lp3_protein_18'),(955387,'gnl|CDD|309973',_binary 'Phage capsid family.  Family of bacteriophage hypothetical proteins and capsid proteins.','pfam05065','Phage_capsid'),(955388,'gnl|CDD|223730',_binary 'Esterase/lipase [Lipid metabolism].','COG0657','Aes'),(955389,'gnl|CDD|311692',_binary 'alpha/beta hydrolase fold.  This catalytic domain is found in a very wide range of enzymes.','pfam07859','Abhydrolase_3'),(955390,'gnl|CDD|224423',_binary 'Dipeptidyl aminopeptidases/acylaminoacyl-peptidases [Amino acid transport and metabolism].','COG1506','DAP2'),(955391,'gnl|CDD|278741',_binary 'Prolyl oligopeptidase family.','pfam00326','Peptidase_S9'),(955392,'gnl|CDD|225181',_binary 'Carboxylesterase type B [Lipid metabolism].','COG2272','PnbA'),(955393,'gnl|CDD|240798',_binary 'RNA recognition motif 1 in granule-associated RNA binding proteins p40-TIA-1 and TIAR.  This subfamily corresponds to the RRM1 of nucleolysin TIA-1 isoform p40 (p40-TIA-1 or TIA-1) and nucleolysin TIA-1-related protein (TIAR), both of which are granule-associated RNA binding proteins involved in inducing apoptosis in cytotoxic lymphocyte (CTL) target cells. TIA-1 and TIAR share high sequence similarity. They are expressed in a wide variety of cell types. TIA-1 can be phosphorylated by a serine/threonine kinase that is activated during Fas-mediated apoptosis.TIAR is mainly localized in the nucleus of hematopoietic and nonhematopoietic cells. It is translocated from the nucleus to the cytoplasm in response to exogenous triggers of apoptosis. Both, TIA-1 and TIAR, bind specifically to poly(A) but not to poly(C) homopolymers. They are composed of three N-terminal highly homologous RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glutamine-rich C-terminal auxiliary domain containing a lysosome-targeting motif. TIA-1 and TIAR interact with RNAs containing short stretches of uridylates and their RRM2 can mediate the specific binding to uridylate-rich RNAs. The C-terminal auxiliary domain may be responsible for interacting with other proteins. In addition, TIA-1 and TIAR share a potential serine protease-cleavage site (Phe-Val-Arg) localized at the junction between their RNA binding domains and their C-terminal auxiliary domains.','cd12352','RRM1_TIA1_like'),(955394,'gnl|CDD|270700',_binary 'Choline Kinase and similar proteins.  This subfamily is composed of bacterial and eukaryotic choline kinases, as well as eukaryotic ethanolamine kinase. ChoK catalyzes the transfer of the gamma-phosphoryl group from ATP (or CTP) to its substrate, choline, producing phosphorylcholine (PCho), a precursor to the biosynthesis of two major membrane phospholipids, phosphatidylcholine (PC), and sphingomyelin (SM). Although choline is the preferred substrate, ChoK also shows substantial activity towards ethanolamine and its N-methylated derivatives. Bacterial ChoK is also referred to as licA protein. ETNK catalyzes the transfer of the gamma-phosphoryl group from CTP to ethanolamine (Etn), the first step in the CDP-Etn pathway for the formation of the major phospholipid, phosphatidylethanolamine (PtdEtn). Unlike ChoK, ETNK shows specific activity for its substrate and displays negligible activity towards N-methylated derivatives of Etn. ChoK plays an important role in cell signaling pathways and the regulation of cell growth. The ChoK subfamily is part of a larger superfamily that includes the catalytic domains of other kinases, such as the typical serine/threonine/tyrosine protein kinases (PKs), RIO kinases, actin-fragmin kinase (AFK), and phosphoinositide 3-kinase (PI3K).','cd05151','ChoK-like'),(955395,'gnl|CDD|274749',_binary 'Kae1-associated kinase Bud32.  Members of this protein family are the Bud32 protein associated with Kae1 (kinase-associated endopeptidase 1) in the Archaea. In many Archaeal genomes, Kae1 and Bud32 are fused. The complex is homologous to the Kae1 and Bud32 subunits of the eukaryotic KEOPS complex, an apparently ancient protein kinase-containing molecular machine. [Unknown function, General].','TIGR03724','Probable_serine/threonine'),(955396,'gnl|CDD|270690',_binary 'Aminoglycoside 3\'-phosphotransferase and Choline Kinase family.  This family is composed of APH, ChoK, ethanolamine kinase (ETNK), macrolide 2\'-phosphotransferase (MPH2\'), an unusual homoserine kinase, and uncharacterized proteins with similarity to the N-terminal domain of acyl-CoA dehydrogenase 10 (ACAD10). The members of this family catalyze the transfer of the gamma-phosphoryl group from ATP (or CTP) to small molecule substrates such as aminoglycosides, macrolides, choline, ethanolamine, and homoserine. Phosphorylation of the antibiotics, aminoglycosides and macrolides, leads to their inactivation and to bacterial antibiotic resistance. Phosphorylation of choline, ethanolamine, and homoserine serves as precursors to the synthesis of important biological compounds, such as the major phospholipids, phosphatidylcholine and phosphatidylethanolamine and the amino acids, threonine, methionine, and isoleucine. The APH/ChoK family is part of a larger superfamily that includes the catalytic domains of other kinases, such as the typical serine/threonine/tyrosine protein kinases (PKs), RIO kinases, actin-fragmin kinase (AFK), and phosphoinositide 3-kinase (PI3K).','cd05120','APH_ChoK_like'),(955397,'gnl|CDD|236586',_binary 'bifunctional UGMP family protein/serine/threonine protein kinase; Validated.','PRK09605','PRK09605'),(955398,'gnl|CDD|237847',_binary 'serine/threonine protein kinase; Provisional.','PRK14879','PRK14879'),(955399,'gnl|CDD|270691',_binary 'Activator of bc1 complex (ABC1) kinases (also called aarF domain containing kinase 3) and similar proteins.  This family is composed of the atypical yeast protein kinase Abc1p, its human homolog ADCK3 (also called CABC1), and similar proteins. Abc1p (also called Coq8p) is required for the biosynthesis of Coenzyme Q (ubiquinone or Q), which is an essential lipid component in respiratory electron and proton transport. It is necessary for the formation of a multi-subunit Q-biosynthetic complex and may also function in the regulation of Q synthesis. Human ADCK3 is able to rescue defects in Q synthesis and the phosphorylation state of Coq proteins in yeast Abc1 (or Coq8) mutants. Mutations in ADCK3 cause progressive cerebellar ataxia and atrophy due to Q10 deficiency. Eukaryotes contain at least two more ABC1/ADCK3-like proteins: in humans, these are the putative atypical protein kinases named ADCK1 and ADCK2. In algae and higher plants, ABC1 kinases have proliferated to more than 15 subfamilies, most of which are located in plastids or mitochondria. Eight of these plant ABC1 kinase subfamilies (ABC1K1-8) are specific for photosynthetic organisms. ABC1 kinases are not related to the ATP-binding cassette (ABC) membrane transporter family.','cd05121','ABC1_ADCK3-like'),(955400,'gnl|CDD|274298',_binary 'double-strand break repair helicase AddA, alphaproteobacterial type.  AddAB, also called RexAB, substitutes for RecBCD in several bacterial lineages. These DNA recombination proteins act before synapse and are particularly important for DNA repair of double-stranded breaks by homologous recombination. The term AddAB is used broadly, with AddA homologous between the alphaproteobacteria (as modeled here) and the Firmicutes, while the partner AddB proteins show no strong homology across the two groups of species. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02784','ATP-dependent_nuclease_su'),(955401,'gnl|CDD|270916',_binary 'Catalytic domain of bacterial Serine/Threonine kinases, PknB and similar proteins.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This subfamily includes many bacterial eukaryotic-type STKs including Staphylococcus aureus PknB (also called PrkC or Stk1), Bacillus subtilis PrkC, and Mycobacterium tuberculosis Pkn proteins (PknB, PknD, PknE, PknF, PknL, and PknH), among others. S. aureus PknB is the only eukaryotic-type STK present in this species, although many microorganisms encode for several such proteins. It is important for the survival and pathogenesis of S. aureus as it is involved in the regulation of purine and pyrimidine biosynthesis, cell wall metabolism, autolysis, virulence, and antibiotic resistance. M. tuberculosis PknB is essential for growth and it acts on diverse substrates including proteins involved in peptidoglycan synthesis, cell division, transcription, stress responses, and metabolic regulation. B. subtilis PrkC is located at the inner membrane of endospores and functions to trigger spore germination. Bacterial STKs in this subfamily show varied domain architectures. The well-characterized members such as S. aureus and M. tuberculosis PknB, and B. subtilis PrkC, contain an N-terminal cytosolic  kinase domain, a transmembrane (TM) segment, and mutliple C-terminal extracellular PASTA domains. The PknB subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14014','STKc_PknB_like'),(955402,'gnl|CDD|226168',_binary 'Mn2+-dependent serine/threonine protein kinase [Signal transduction mechanisms].','COG3642','COG3642'),(955403,'gnl|CDD|279908',_binary 'Phosphotransferase enzyme family.  This family consists of bacterial antibiotic resistance proteins, which confer resistance to various aminoglycosides they include: aminoglycoside 3\'-phosphotransferase or kanamycin kinase / neomycin-kanamycin phosphotransferase and streptomycin 3\'\'-kinase or streptomycin 3\'\'-phosphotransferase. The aminoglycoside phosphotransferases inactivate aminoglycoside antibiotics via phosphorylation. This family also includes homoserine kinase. This family is related to fructosamine kinase pfam03881.','pfam01636','APH'),(955404,'gnl|CDD|306557',_binary 'Protein kinase domain.','pfam00069','Pkinase'),(955405,'gnl|CDD|270993',_binary 'C-terminal catalytic domain of the Serine/Threonine Kinases, Ribosomal S6 kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. RSKs contain an N-terminal kinase domain (NTD) from the AGC family and a C-terminal kinase domain (CTD) from the CAMK family. They are activated by signaling inputs from extracellular regulated kinase (ERK) and phosphoinositide dependent kinase 1 (PDK1). ERK phosphorylates and activates the CTD of RSK, serving as a docking site for PDK1, which phosphorylates and activates the NTD, which in turn phosphorylates all known RSK substrates. RSKs act as downstream effectors of mitogen-activated protein kinase (MAPK) and play key roles in mitogen-activated cell growth, differentiation, and survival. Mammals possess four RSK isoforms (RSK1-4) from distinct genes. RSK proteins are also referred to as MAP kinase-activated protein kinases (MAPKAPKs), 90 kDa ribosomal protein S6 kinases (p90-RSKs), or p90S6Ks. The RSK subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14091','STKc_RSK_C'),(955406,'gnl|CDD|214567',_binary 'Serine/Threonine protein kinases, catalytic domain.  Phosphotransferases. Serine or threonine-specific kinase subfamily.','smart00220','S_TKc'),(955407,'gnl|CDD|270622',_binary 'Catalytic domain of Protein Kinases.  PKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine or tyrosine residues on protein substrates. PKs make up a large family of serine/threonine kinases (STKs), protein tyrosine kinases (PTKs), and dual-specificity PKs that phosphorylate both serine/threonine and tyrosine residues of target proteins. Majority of protein phosphorylation occurs on serine residues while only 1% occurs on tyrosine residues. Protein phosphorylation is a mechanism by which a wide variety of cellular proteins, such as enzymes and membrane channels, are reversibly regulated in response to certain stimuli. PKs often function as components of signal transduction pathways in which one kinase activates a second kinase, which in turn, may act on other kinases; this sequential action transmits a signal from the cell surface to target proteins, which results in cellular responses. The PK family is one of the largest known protein families with more than 100 homologous yeast enzymes and more than 500 human proteins. A fraction of PK family members are pseudokinases that lack crucial residues for catalytic activity. The mutiplicity of kinases allows for specific regulation according to substrate, tissue distribution, and cellular localization. PKs regulate many cellular processes including proliferation, division, differentiation, motility, survival, metabolism, cell-cycle progression, cytoskeletal rearrangement, immunity, and neuronal functions. Many kinases are implicated in the development of various human diseases including different types of cancer. The PK family is part of a larger superfamily that includes the catalytic domains of RIO kinases, aminoglycoside phosphotransferase, choline kinase, phosphoinositide 3-kinase (PI3K), and actin-fragmin kinase.','cd00180','PKc'),(955408,'gnl|CDD|270687',_binary 'The catalytic domain of CAMK family Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. CaMKs are multifunctional calcium and calmodulin (CaM) stimulated STKs involved in cell cycle regulation. There are several types of CaMKs including CaMKI, CaMKII, and CaMKIV. CaMKI proteins are monomeric and they play pivotal roles in the nervous system, including long-term potentiation, dendritic arborization, neurite outgrowth, and the formation of spines, synapses, and axons. CaMKII is a signaling molecule that translates upstream calcium and reactive oxygen species (ROS) signals into downstream responses that play important roles in synaptic function and cardiovascular physiology. CAMKIV is implicated in regulating several transcription factors like CREB, MEF2, and retinoid orphan receptors, as well as in T-cell development and signaling. The CAMK family also consists of other related kinases including the Phosphorylase kinase Gamma subunit (PhKG), the C-terminal kinase domains of Ribosomal S6 kinase (RSK) and Mitogen and stress-activated kinase (MSK), Doublecortin-like kinase (DCKL), and the MAPK-activated protein kinases MK2, MK3, and MK5, among others. The CAMK family is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05117','STKc_CAMK'),(955409,'gnl|CDD|271084',_binary 'Catalytic domain of the Serine/Threonine Kinase, Phosphorylase kinase Gamma 1 subunit.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Phosphorylase kinase (PhK) catalyzes the phosphorylation of inactive phosphorylase b to form the active phosphorylase a. It coordinates hormonal, metabolic, and neuronal signals to initiate the breakdown of glycogen stores, which enables the maintenance of blood-glucose homeostasis during fasting, and is also used as a source of energy for muscle contraction. PhK is one of the largest and most complex protein kinases, composed of a heterotetramer containing four molecules each of four subunit types: one catalytic (gamma) and three regulatory (alpha, beta, and delta). The gamma 1 subunit (PhKG1) is also referred to as the muscle gamma isoform. The gamma subunit, when isolated, is constitutively active and does not require phosphorylation of the A-loop for activity. The regulatory subunits restrain this kinase activity until signals are received to relieve this inhibition. For example, the kinase is activated in response to hormonal stimulation, after autophosphorylation or phosphorylation by cAMP-dependent kinase of the alpha and beta subunits. The high-affinity binding of ADP to the beta subunit also stimulates kinase activity, whereas calcium relieves inhibition by binding to the delta (calmodulin) subunit. The PhKG1 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14182','STKc_PhKG1'),(955410,'gnl|CDD|270692',_binary 'Catalytic domain of STE family Protein Kinases.  PKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine (ST) or tyrosine residues on protein substrates. This family is composed of STKs, and some dual-specificity PKs that phosphorylate both threonine and tyrosine residues of target proteins. Most members are kinases involved in mitogen-activated protein kinase (MAPK) signaling cascades, acting as MAPK kinases (MAPKKs), MAPKK kinases (MAPKKKs), or MAPKKK kinases (MAP4Ks). The MAPK signaling pathways are important mediators of cellular responses to extracellular signals. The pathways involve a triple kinase core cascade comprising of the MAPK, which is phosphorylated and activated by a MAPKK, which itself is phosphorylated and activated by a MAPKKK. Each MAPK cascade is activated either by a small GTP-binding protein or by an adaptor protein, which transmits the signal either directly to a MAPKKK to start the triple kinase core cascade or indirectly through a mediator kinase, a MAP4K. Other STE family members include p21-activated kinases (PAKs) and class III myosins, among others. PAKs are Rho family GTPase-regulated kinases that serve as important mediators in the function of Cdc42 (cell division cycle 42) and Rac. Class III myosins are motor proteins containing an N-terminal kinase catalytic domain and a C-terminal actin-binding domain, which can phosphorylate several cytoskeletal proteins, conventional myosin regulatory light chains, as well as autophosphorylate the C-terminal motor domain. They play an important role in maintaining the structural integrity of photoreceptor cell microvilli. The STE family is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05122','PKc_STE'),(955411,'gnl|CDD|223589',_binary 'Serine/threonine protein kinase [General function prediction only / Signal transduction mechanisms / Transcription / DNA replication, recombination, and repair].','COG0515','SPS1'),(955412,'gnl|CDD|270796',_binary 'Catalytic domain of the Protein Serine/Threonine Kinase, Mitogen-Activated Protein (MAP)/Extracellular signal-Regulated Kinase (ERK) Kinase Kinase 4.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. MEKK4 is a MAPK kinase kinase that phosphorylates and activates the c-Jun N-terminal kinase (JNK) and p38 MAPK signaling pathways by directly activating their respective MAPKKs, MKK4/MKK7 and MKK3/MKK6. JNK and p38 are collectively known as stress-activated MAPKs, as they are activated in response to a variety of environmental stresses and pro-inflammatory cytokines. MEKK4 also plays roles in the re-polarization of the actin cytoskeleton in response to osmotic stress, in the proper closure of the neural tube, in cardiovascular development, and in immune responses. The MEKK4 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd06626','STKc_MEKK4'),(955413,'gnl|CDD|214702',_binary 'HNH nucleases.','smart00507','HNHc'),(955414,'gnl|CDD|238038',_binary 'HNH nucleases; HNH endonuclease signature which is found in viral, prokaryotic, and eukaryotic proteins. The alignment includes members of the large group of homing endonucleases, yeast intron 1 protein, MutS, as well as bacterial colicins, pyocins, and anaredoxins.','cd00085','HNHc'),(955415,'gnl|CDD|280088',_binary 'HNH endonuclease.','pfam01844','HNH'),(955416,'gnl|CDD|223348',_binary 'Site-specific DNA methylase [DNA replication, recombination, and repair].','COG0270','Dcm'),(955417,'gnl|CDD|238192',_binary 'Cytosine-C5 specific DNA methylases; Methyl transfer reactions play an important role in many aspects of biology. Cytosine-specific DNA methylases are found both in prokaryotes and eukaryotes. DNA methylation, or the covalent addition of a methyl group to cytosine within the context of the CpG dinucleotide, has profound effects on the mammalian genome. These effects include transcriptional repression via inhibition of transcription factor binding or the recruitment of methyl-binding proteins and their associated chromatin remodeling factors, X chromosome inactivation, imprinting and the suppression of parasitic DNA sequences. DNA methylation is also essential for proper embryonic development and is an important player in both DNA repair and genome stability.','cd00315','Cyt_C5_DNA_methylase'),(955418,'gnl|CDD|273211',_binary 'DNA-methyltransferase (dcm).  All proteins in this family for which functions are known are DNA-cytosine methyltransferases. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00675','Modification_methylase'),(955419,'gnl|CDD|306622',_binary 'C-5 cytosine-specific DNA methylase.','pfam00145','DNA_methylase'),(955420,'gnl|CDD|100107',_binary 'S-adenosylmethionine-dependent methyltransferases (SAM or AdoMet-MTase), class I;  AdoMet-MTases are enzymes that use S-adenosyl-L-methionine (SAM or AdoMet) as a substrate for methyltransfer, creating the product S-adenosyl-L-homocysteine (AdoHcy). There are at least five structurally distinct families of AdoMet-MTases, class I being the largest and most diverse. Within this class enzymes can be classified by different substrate specificities (small molecules, lipids, nucleic acids, etc.) and different target atoms for methylation (nitrogen, oxygen, carbon, sulfur, etc.).','cd02440','AdoMet_MTases'),(955421,'gnl|CDD|312819',_binary 'RNA cap guanine-N2 methyltransferase.  RNA cap guanine-N2 methyltransferases such as Schizosaccharomyces pombe Tgs1 and Giardia lamblia Tgs2 catalyze methylation of the exocyclic N2 amine of 7-methylguanosine.','pfam09445','Methyltransf_15'),(955422,'gnl|CDD|308210',_binary 'Transcription factor WhiB.  WhiB is a putative transcription factor in Actinobacteria, required for differentiation and sporulation.','pfam02467','Whib'),(955423,'gnl|CDD|316196',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13649','Methyltransf_25'),(955424,'gnl|CDD|311936',_binary 'Methyltransferase domain.  Members of this family are SAM dependent methyltransferases.','pfam08242','Methyltransf_12'),(955425,'gnl|CDD|311935',_binary 'Methyltransferase domain.  Members of this family are SAM dependent methyltransferases.','pfam08241','Methyltransf_11'),(955426,'gnl|CDD|225137',_binary '2-polyprenyl-3-methyl-5-hydroxy-6-metoxy-1,4-benzoquinol methylase [Coenzyme metabolism].','COG2227','UbiG'),(955427,'gnl|CDD|316049',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13489','Methyltransf_23'),(955428,'gnl|CDD|308543',_binary 'Recombination endonuclease VII.','pfam02945','Endonuclease_7'),(955429,'gnl|CDD|177395',_binary 'recombination endonuclease VII; Provisional.','PHA02565','49'),(955430,'gnl|CDD|222856',_binary 'polynucleotide kinase; Provisional.','PHA02530','pseT'),(955431,'gnl|CDD|319805',_binary 'C-terminal phosphatase domain of T4 polynucleotide kinase/phosphatase (PNKP) and related phosphatases.  This family includes the C-terminal domain of the bifunctional enzyme T4 polynucleotide kinase/phosphatase, PNKP. The PNKP phosphatase domain can catalyze the hydrolytic removal of the 3\'-phosphoryl of DNA, RNA and deoxynucleoside 3\'-monophosphates. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd07502','HAD_PNKP-C'),(955432,'gnl|CDD|316215',_binary 'AAA domain.  This family of domains contain only a P-loop motif, that is characteristic of the AAA superfamily. Many of the proteins in this family are just short fragments so there is no Walker B motif.','pfam13671','AAA_33'),(955433,'gnl|CDD|226986',_binary 'Predicted kinase [General function prediction only].','COG4639','COG4639'),(955434,'gnl|CDD|223718',_binary 'Predicted kinase [General function prediction only].','COG0645','COG0645'),(955435,'gnl|CDD|310778',_binary 'Zeta toxin.  This family consists of several bacterial zeta toxin proteins. Zeta toxin is thought to be part of a postregulational killing system in bacteria. It relies on antitoxin/toxin systems that secure stable inheritance of low and medium copy number plasmids during cell division and kill cells that have lost the plasmid.','pfam06414','Zeta_toxin'),(955436,'gnl|CDD|226573',_binary 'Predicted nucleotide kinase [Nucleotide transport and metabolism].','COG4088','COG4088'),(955437,'gnl|CDD|132613',_binary 'L-seryl-tRNA(Sec) kinase, archaeal.  Members of this protein are L-seryl-tRNA(Sec) kinase. This enzyme is part of a two-step pathway in Eukaryota and Archaea for performing selenocysteine biosynthesis by changing serine misacylated on selenocysteine-tRNA to selenocysteine. This enzyme performs the first step, phosphorylation of the OH group of the serine side chain. This family represents archaeal proteins with this activity. [Protein synthesis, tRNA aminoacylation].','TIGR03574','L-seryl-tRNASec_kinase'),(955438,'gnl|CDD|312064',_binary 'Chromatin associated protein KTI12.  This is a family of chromatin associated proteins which interact with the Elongator complex, a component of the elongating form of RNA polymerase II. The Elongator complex has histone acetyltransferase activity.','pfam08433','KTI12'),(955439,'gnl|CDD|315820',_binary 'AAA domain.','pfam13238','AAA_18'),(955440,'gnl|CDD|130724',_binary 'polynucleotide 5\'-kinase 3\'-phosphatase.  This model represents the metazoan 5\'-polynucleotide-kinase-3\'-phosphatase, PNKP, which is believed to be involved in repair of oxidative DNA damage. Removal of 3\' phosphates is essential for the further processing of the break by DNA polymerases. The central phosphatase domain is a member of the IIIA subfamily (TIGR01662) of the haloacid dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. As is common in this superfamily, the enzyme is magnesium dependent. A difference between this enzyme and other HAD-superfamily phosphatases is in the third conserved catalytic motif which usually contains two conserved aspartate residues believed to be involved in binding the magnesium ion. Here, the second aspartate is replaced by a conserved arginine residue which may indicate an interaction with the phosphate backbone of the substrate. Very close relatives of this domain are also found separate from the N- and C-terminal domains seen here, as in the 3\'-phosphatase found in plants. The larger family of these domains is described by TIGR01664. Outside of the phosphatase domain is a P-loop ATP-binding motif associated with the kinase activity. The entry for the mouse homolog appears to be missing a large piece of sequence corresponding to the first conserved catalytic motif of the phosphatase domain as well as the conserved threonine of the second motif. Either this is a sequencing artifact or this may represent a pseudo- or non-functional gene. Note that the EC number for the kinase function is: 2.7.1.78.','TIGR01663','polynucleotide_kinase-3\'-'),(955441,'gnl|CDD|315948',_binary 'Homeodomain-like domain.','pfam13384','HTH_23'),(955442,'gnl|CDD|259851',_binary 'Helix-turn-helix domain of Hin and related proteins.  This domain model summarizes a family of DNA-binding domains unique to bacteria and represented by the Hin protein of Salmonella. The basic HTH domain is a simple fold comprised of three core helices that form a right-handed helical bundle. The principal DNA-protein interface is formed by the third helix, the recognition helix, inserting itself into the major groove of the DNA. A diverse array of HTH domains participate in a variety of functions that depend on their DNA-binding properties. HTH_Hin represents one of the simplest versions of the HTH domains; the characterization of homologous relationships between various sequence-diverse HTH domain families remains difficult. The Hin recombinase induces the site-specific inversion of a chromosomal DNA segment containing a promoter, which controls the alternate expression of two genes by reversibly switching orientation. The Hin recombinase consists of a single polypeptide chain containing a C-terminal DNA-binding domain (HTH_Hin) and a catalytic domain.','cd00569','HTH_Hin_like'),(955443,'gnl|CDD|312675',_binary 'Bifunctional DNA primase/polymerase, N-terminal.  Members of this family adopt a structure consisting of a core of antiparallel beta sheets. They are found in various bacterial hypothetical proteins, and have been shown to harbour both primase and polymerase activities.','pfam09250','Prim-Pol'),(955444,'gnl|CDD|214927',_binary 'Bifunctional DNA primase/polymerase, N-terminal.  Members of this family adopt a structure consisting of a core of antiparallel beta sheets. They are found in various bacterial hypothetical proteins, and have been shown to harbour both primase and polymerase activities.','smart00943','Prim-Pol'),(955445,'gnl|CDD|240129',_binary 'Prim_Pol: Primase-polymerase (primpol) domain of the type found in bifunctional replicases from archaeal plasmids, including ORF904 protein of the crenarchaeal plasmid pRN1 from Sulfolobus islandicus (pRN1 primpol). These primpol domains belong to the archaeal/eukaryal primase (AEP) superfamily. This group includes archaeal plasmids and bacteriophage AEPs. The ORF904 protein is a multifunctional protein having ATPase, primase and DNA polymerase activity, and may play a role in the replication of the archaeal plasmid. The pRN1 primpol domain exhibits DNA polymerase and primase activities; a cluster of active site residues (three acidic residues, and a histidine) is required for both these activities. For pRN1 primpol, the primase activity prefers dNTPs to rNTPs; incorporation of dNTPs requires rNTP as cofactor. The pRN1 primpol contains an unusual zinc-binding stem, which is not conserved in other members of this group.','cd04859','Prim_Pol'),(955446,'gnl|CDD|132997',_binary 'Glycosyltransferase family A (GT-A) includes diverse families of glycosyl transferases with a common GT-A type structural fold.  Glycosyltransferases (GTs) are enzymes that synthesize oligosaccharides, polysaccharides, and glycoconjugates by transferring the sugar moiety from an activated nucleotide-sugar donor to an acceptor molecule, which may be a growing oligosaccharide, a lipid, or a protein.  Based on the stereochemistry of the donor and acceptor molecules, GTs are classified as either retaining or inverting enzymes. To date, all GT structures adopt one of two possible folds, termed GT-A fold and GT-B fold.  This hierarchy includes diverse families of glycosyl transferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. The majority of the proteins in this superfamily are Glycosyltransferase family 2 (GT-2) proteins. But it also includes families GT-43, GT-6, GT-8, GT13 and GT-7; which are evolutionarily related to GT-2 and share structure similarities.','cd00761','Glyco_tranf_GTA_type'),(955447,'gnl|CDD|278936',_binary 'Glycosyl transferase family 2.  Diverse family, transferring sugar from UDP-glucose, UDP-N-acetyl- galactosamine, GDP-mannose or CDP-abequose, to a range of substrates including cellulose, dolichol phosphate and teichoic acids.','pfam00535','Glycos_transf_2'),(955448,'gnl|CDD|223539',_binary 'Glycosyltransferases involved in cell wall biogenesis [Cell envelope biogenesis, outer membrane].','COG0463','WcaA'),(955449,'gnl|CDD|133039',_binary 'Subfamily of Glycosyltransferase Family GT2 of unknown function.  GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd04196','GT_2_like_d'),(955450,'gnl|CDD|133005',_binary 'UDP-glucose LOS-beta-1,4 glucosyltransferase is required for biosynthesis of lipooligosaccharide.  UDP-glucose: lipooligosaccharide (LOS)  beta-1-4-glucosyltransferase catalyzes the addition of the first residue, glucose, of the lacto-N-neotetrase structure to HepI of the LOS inner core.  LOS is the major constituent of the outer leaflet of the outer membrane of gram-positive bacteria. It consists of a short oligosaccharide chain of variable composition (alpha chain) attached to a branched inner core which is lined in turn to lipid A. Beta 1,4 glucosyltransferase is required to attach the alpha chain to the inner core.','cd02511','Beta4Glucosyltransferase'),(955451,'gnl|CDD|133055',_binary 'WfgS and WfeV are involved in O-antigen biosynthesis.  Escherichia coli WfgS and Shigella dysenteriae WfeV are glycosyltransferase 2 family enzymes involved in O-antigen biosynthesis. GT-2 enzymes have GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd06433','GT_2_WfgS_like'),(955452,'gnl|CDD|182223',_binary 'putative glycosyl transferase; Provisional.','PRK10073','PRK10073'),(955453,'gnl|CDD|133038',_binary 'GT2_AmsE_like is involved in exopolysaccharide amylovora biosynthesis.  AmsE is a glycosyltransferase involved in exopolysaccharide amylovora biosynthesis in Erwinia amylovora. Amylovara is one of the three exopolysaccharide produced by E. amylovora. Amylovara-deficient mutants are non-pathogenic. It is a subfamily of Glycosyltransferase Family GT2, which includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds.','cd04195','GT2_AmsE_like'),(955454,'gnl|CDD|133027',_binary 'Myxococcus xanthus RfbC like proteins are required for O-antigen biosynthesis.  The rfbC gene encodes a predicted protein of 1,276 amino acids, which is required for O-antigen biosynthesis in Myxococcus xanthus. It is a subfamily of Glycosyltransferase Family GT2, which includes diverse families of glycosyl transferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds.','cd04184','GT2_RfbC_Mx_like'),(955456,'gnl|CDD|197715',_binary 'helix_turn_helix, Lux Regulon.  lux regulon (activates the bioluminescence operon.','smart00421','HTH_LUXR'),(955457,'gnl|CDD|99777',_binary 'C-terminal DNA-binding domain of LuxR-like proteins. This domain contains a helix-turn-helix motif and binds DNA. Proteins belonging to this group are response regulators; some act as transcriptional activators, others as transcriptional repressors. Many are active as homodimers. Many are two domain proteins in which the DNA binding property of the C-terminal DNA binding domain is modulated by modifications of the N-terminal domain.  For example in the case of Lux R which participates in the regulation of gene expression in response to fluctuations in cell-population density (quorum-sensing), a signaling molecule, the pheromone Acyl HSL (N-acyl derivatives of homoserine lactone), binds to the N-terminal domain and leads to LuxR dimerization.  For others phophorylation of the N-terminal domain leads to multimerization, for example Escherichia coli NarL and Sinorhizobium melilot FixJ. NarL controls gene expression of many respiratory-related operons when environmental nitrate or nitrite is present under anerobic conditions. FixJ is involved in the transcriptional activation of nitrogen fixation genes. The group also includes small proteins which lack an N-terminal signaling domain, such as Bacillus subtilis GerE.  GerE is dimeric and acts in conjunction with sigmaK as an activator or a repressor modulating the expression of various genes in particular those encoding the spore-coat. These LuxR family regulators may share a similar organization of their target binding sites. For example the LuxR dimer binds the lux box, a 20bp inverted repeat, GerE dimers bind two 12bp consensus sequences in inverted orientation having the central four bases overlap, and the NarL dimer binds two 7bp inverted repeats separated by 2 bp.','cd06170','LuxR_C_like'),(955458,'gnl|CDD|225107',_binary 'Response regulator containing a CheY-like receiver domain and an HTH DNA-binding domain [Signal transduction mechanisms / Transcription].','COG2197','CitB'),(955459,'gnl|CDD|311418',_binary 'NUMOD4 motif.  NUMOD4 is a putative DNA-binding motif found in homing endonucleases and related proteins.','pfam07463','NUMOD4'),(955460,'gnl|CDD|316042',_binary 'AAA domain.  This AAA domain is found in a wide variety of presumed DNA repair proteins.','pfam13481','AAA_25'),(955461,'gnl|CDD|238545',_binary 'Hexameric Replicative Helicase RepA.  RepA is encoded by a plasmid, which is found in most Gram negative bacteria. RepA is a 5\'-3\' DNA helicase which can utilize ATP, GTP and CTP to a lesser extent.','cd01125','repA'),(955462,'gnl|CDD|226126',_binary 'RecA-family ATPase [DNA replication, recombination, and repair].','COG3598','RepA'),(955463,'gnl|CDD|223931',_binary 'DNA modification methylase [DNA replication, recombination, and repair].','COG0863','COG0863'),(955465,'gnl|CDD|307613',_binary 'DNA methylase.  Members of this family are DNA methylases. The family contains both N-4 cytosine-specific DNA methylases and N-6 Adenine-specific DNA methylases.','pfam01555','N6_N4_Mtase'),(955467,'gnl|CDD|184255',_binary 'putative methylase; Provisional.','PRK13699','PRK13699'),(955469,'gnl|CDD|183177',_binary 'putative methyltransferase; Provisional.','PRK11524','PRK11524'),(955471,'gnl|CDD|225319',_binary 'Predicted archaeal methyltransferase [General function prediction only].','COG2521','COG2521'),(955472,'gnl|CDD|316771',_binary 'HNH endonuclease.  This domain is related to other HNH domain families such as pfam01844. Suggesting that these proteins have a nucleic acid cleaving function.','pfam14279','HNH_5'),(955473,'gnl|CDD|132927',_binary 'Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease.  Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. This subfamily only contains bacterial sequences. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1\' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.','cd07016','S14_ClpP_1'),(955474,'gnl|CDD|223811',_binary 'Protease subunit of ATP-dependent Clp proteases [Posttranslational modification, protein turnover, chaperones / Intracellular trafficking and secretion].','COG0740','ClpP'),(955475,'gnl|CDD|132928',_binary 'Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease.  Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1\' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.','cd07017','S14_ClpP_2'),(955476,'gnl|CDD|306945',_binary 'Clp protease.  The Clp protease has an active site catalytic triad. In E. coli Clp protease, ser-111, his-136 and asp-185 form the catalytic triad. Some members have lost active site residues and are therefore inactive, some contain one or two large insertions.','pfam00574','CLP_protease'),(955477,'gnl|CDD|214685',_binary 'exonuclease domain in DNA-polymerase alpha and epsilon chain, ribonuclease T and other exonucleases.','smart00479','EXOIII'),(955478,'gnl|CDD|237133',_binary 'ATP-dependent Clp protease proteolytic subunit; Reviewed.','PRK12553','PRK12553'),(955479,'gnl|CDD|176648',_binary 'DEDDh 3\'-5\' exonuclease domain family.  DEDDh exonucleases, part of the DnaQ-like (or DEDD) exonuclease superfamily, catalyze the excision of nucleoside monophosphates at the DNA or RNA termini in the 3\'-5\' direction. These proteins contain four invariant acidic residues in three conserved sequence motifs termed ExoI, ExoII and ExoIII. DEDDh exonucleases are classified as such because of the presence of specific Hx(4)D conserved pattern at the ExoIII motif. The four conserved acidic residues are clustered around the active site and serve as ligands for the two metal ions required for catalysis. Most DEDDh exonucleases are the proofreading subunits (epsilon) or domains of bacterial DNA polymerase III, the main replicating enzyme in bacteria, which functions as the chromosomal replicase. Other members include other DNA and RNA exonucleases such as RNase T, Oligoribonuclease, and RNA exonuclease (REX), among others.','cd06127','DEDDh'),(955480,'gnl|CDD|237742',_binary 'ATP-dependent Clp protease proteolytic subunit; Provisional.','PRK14513','PRK14513'),(955481,'gnl|CDD|236281',_binary 'DNA polymerase III subunit epsilon; Provisional.','PRK08517','PRK08517'),(955482,'gnl|CDD|132924',_binary 'Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease.  Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. Additionally, they are implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1\' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.','cd07013','S14_ClpP'),(955483,'gnl|CDD|273601',_binary 'DNA polymerase III, alpha chain, Gram-positive type.  This model describes a polypeptide chain of DNA polymerase III. Full-length homologs of this protein are restricted to the Gram-positive lineages, including the Mycoplasmas. This protein is designated alpha chain and given the gene symbol polC, but is not a full-length homolog of other polC genes. The N-terminal region of about 200 amino acids is rich in low-complexity sequence, poorly alignable, and not included n this model. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01405','DNA_polymerase_III_PolC-t'),(955484,'gnl|CDD|188055',_binary 'ATP-dependent Clp endopeptidase, proteolytic subunit ClpP.  This model for the proteolytic subunit ClpP has been rebuilt to a higher stringency. In every bacterial genome with the ClpXP machine, a ClpP protein will be found that scores well with this model. In general, this ClpP member will be encoded adjacent to the clpX gene, as were all examples used in the seed alignment. A large fraction of genomes have one or more additional ClpP paralogs, sometimes encoded nearby and sometimes elsewhere. The stringency of the trusted cutoff used here excludes the more divergent ClpP paralogs from being called authentic ClpP by this model. [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00493','ATP-dependent_Clp_proteas'),(955485,'gnl|CDD|279297',_binary 'Exonuclease.  This family includes a variety of exonuclease proteins, such as ribonuclease T and the epsilon subunit of DNA polymerase III.;.','pfam00929','RNase_T'),(955486,'gnl|CDD|237741',_binary 'ATP-dependent Clp protease proteolytic subunit; Provisional.','PRK14512','PRK14512'),(955487,'gnl|CDD|273602',_binary 'DnaQ family exonuclease/DinG family helicase, putative.  This model represents a family of proteins in Gram-positive bacteria. The N-terminal region of about 200 amino acids resembles the epsilon subunit of E. coli DNA polymerase III and the homologous region of the Gram-positive type DNA polymerase III alpha subunit. The epsilon subunit contains an exonuclease domain. The remainder of this protein family resembles a predicted ATP-dependent helicase, the DNA damage-inducible protein DinG of E. coli. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01407','dinG_rel'),(955488,'gnl|CDD|178955',_binary 'ATP-dependent Clp protease proteolytic subunit; Reviewed.','PRK00277','clpP'),(955489,'gnl|CDD|223916',_binary 'DNA polymerase III, epsilon subunit and related 3\'-5\' exonucleases [DNA replication, recombination, and repair].','COG0847','DnaQ'),(955490,'gnl|CDD|214340',_binary 'ATP-dependent Clp protease proteolytic subunit.','CHL00028','clpP'),(955491,'gnl|CDD|99837',_binary 'DEDDh 3\'-5\' exonuclease domain of RNase T.  RNase T is a DEDDh-type DnaQ-like 3\'-5\' exoribonuclease E implicated in the 3\' maturation of small stable RNAs and 23srRNA, and in the end turnover of tRNA. It contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. RNase T is related to the proofreading domain of DNA polymerase III. Despite its important role, RNase T is mainly found only in gammaproteobacteria. It is speculated that it might have originated from DNA polymerase III at the time the gamma division of proteobacteria diverged from other bacteria. RNase T is a homodimer with the catalytic residues of one monomer contacting a large basic patch on the other monomer to form a functional active site.','cd06134','RNaseT'),(955492,'gnl|CDD|132923',_binary 'Caseinolytic protease (ClpP) is an ATP-dependent protease.  Clp protease (caseinolytic protease; ClpP; endopeptidase Clp; Peptidase S14; ATP-dependent protease, ClpAP)-like enzymes are highly conserved serine proteases and belong to the ClpP/Crotonase superfamily. Included in this family are Clp proteases that are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. The functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP. Active site consists of the triad Ser, His and Asp, preferring hydrophobic or non-polar residues at P1 or P1\' positions. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function. Another family included in this class of enzymes is the signal peptide peptidase A (SppA; S49) which is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Mutagenesis studies suggest that the catalytic center of SppA comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. In addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members, the E. coli SppA contains an amino-terminal domain. Others, including sohB peptidase, protein C, protein 1510-N and archaeal signal peptide peptidase, do not contain the amino-terminal domain. The third family included in this hierarchy is nodulation formation efficiency D (NfeD) which is a membrane-bound Clp-class protease and only found in bacteria and archaea. Majority of the NfeD genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named stomatin operon partner protein (STOPP). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 from Pyrococcus horikoshii has been shown to possess serine protease activity having a Ser-Lys catalytic dyad.','cd00394','Clp_protease_like'),(955493,'gnl|CDD|236148',_binary 'bifunctional ATP-dependent DNA helicase/DNA polymerase III subunit epsilon; Validated.','PRK08074','PRK08074'),(955494,'gnl|CDD|184722',_binary 'ATP-dependent Clp protease proteolytic subunit; Provisional.','PRK14514','PRK14514'),(955495,'gnl|CDD|234767',_binary 'DNA polymerase III PolC; Validated.','PRK00448','polC'),(955496,'gnl|CDD|139060',_binary 'ATP-dependent Clp protease proteolytic subunit; Reviewed.','PRK12551','PRK12551'),(955497,'gnl|CDD|180905',_binary 'bifunctional ATP-dependent DNA helicase/DNA polymerase III subunit epsilon; Validated.','PRK07246','PRK07246'),(955498,'gnl|CDD|188129',_binary 'ribonuclease T.  This model describes ribonuclease T, an enzyme found so far only in gamma-subdivision Proteobacteria such as Escherichia coli and Xylella fastidiosa. Ribonuclease T is homologous to the DNA polymerase III alpha chain. It can liberate AMP from the common C-C-A terminus of uncharged tRNA. It appears also to be involved in RNA maturation. It also acts as a 3\' to 5\' single-strand DNA-specific exonuclease; it is distinctive for its ability to remove residues near a double-stranded stem. Ribonuclease T is a high copy suppressor in E. coli of a uv-repair defect caused by deletion of three other single-stranded DNA exonucleases. [Transcription, RNA processing].','TIGR01298','Ribonuclease_T'),(955499,'gnl|CDD|99834',_binary 'an uncharacterized bacterial subgroup of the DEDDh 3\'-5\' exonuclease domain family with similarity to the epsilon subunit of DNA polymerase III.  This subfamily is composed of uncharacterized bacterial proteins with similarity to the epsilon subunit of DNA polymerase III (Pol III), a multisubunit polymerase which is the main DNA replicating enzyme in bacteria, functioning as the chromosomal replicase. The Pol III holoenzyme is a complex of ten different subunits, three of which (alpha, epsilon, and theta) compose the catalytic core. The Pol III epsilon subunit, encoded by the dnaQ gene, is a DEDDh-type 3\'-5\' exonuclease which is responsible for the proofreading activity of the polymerase, increasing the fidelity of DNA synthesis. It contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The epsilon subunit of Pol III also functions as a stabilizer of the holoenzyme complex.','cd06130','DNA_pol_III_epsilon_like'),(955500,'gnl|CDD|179951',_binary 'ribonuclease T; Provisional.','PRK05168','PRK05168'),(955501,'gnl|CDD|291057',_binary 'Domain of unknown function (DUF4406).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 98 and 145 amino acids in length.','pfam14359','DUF4406'),(955502,'gnl|CDD|212151',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in Deinococcus radiodurans DR2231 protein and its bacterial homologs.  This family corresponds to the DR2231 protein, a MazG-like NTP-PPase from Deinococcus radiodurans, and its bacterial homologs. All family members contain a well-conserved divalent ion binding motif, EXXEX(12-28)EXXD, which is the identity signature for all-alpha-helical NTP-PPase superfamily. DR2231 shows significant structural resemblance to MazG proteins, but is functionally related to the dimeric dUTPases. It might be an evolutionary precursor of dimeric dUTPases with  very high specificity in hydrolyzing dUTP into dUMP, but an inability to hydrolyze dTTP, a typical feature of dUTPases. Moreover, unlike the dUPase monomer containing a single active site, the DR2231 protein dimer holds two putative active sites.','cd11544','NTP-PPase_DR2231'),(955503,'gnl|CDD|212137',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in Deinococcus radiodurans DR2231 protein and its bacterial homologs.  This family includes a MazG-like NTP-PPase from Deinococcus radiodurans (DR2231), a putative NTP-PPase YP_001813558.1 from Exiguobacterium sibiricum and their bacterial homologs. DR2231 shows significant structural resemblance to MazG proteins, but is functionally related to the dimeric dUTPases. It can hydrolyze dUTP into dUMP. DR2231-like proteins contain a well conserved divalent ion binding motif, EXXEX(12-28)EXXD, which is the identity signature for the all-alpha-helical NTP-PPase superfamily. Unlike normal dimeric dUTPase-like proteins with a central four-helix bundle forming the active site, YP_001813558.1 displays a very unusual interlaced segment-swapped dimer. It potentially prefers to hydrolyze dCTPs or its derivatives. YP_001813558.1-like proteins contain a variant divalent ion binding motif, EXXEX(12-28)AXXD.','cd11530','NTP-PPase_DR2231_like'),(955504,'gnl|CDD|212152',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in Exiguobacterium sibiricum YP_001813558.1 protein and its bacterial homologs.  This family contains a putative NTP_PPase (YP_001813558.1) from Exiguobacterium sibiricum and its bacterial homologs. Unlike normal dimeric dUTPase-like proteins with a central four-helix bundle forming the active site, YP_001813558.1 displays a very unusual interlaced segment-swapped dimer that might be important for it to adapt to an extremely cold environment. Moreover, structural analysis and comparisons indicate that YP_001813558.1 potentially prefers to hydrolyze dCTPs or its derivatives.','cd11545','NTP-PPase_YP_001813558'),(955505,'gnl|CDD|307585',_binary 'Phosphoribosyl-ATP pyrophosphohydrolase.  This enzyme catalyzes the second step in the histidine biosynthetic pathway.','pfam01503','PRA-PH'),(955506,'gnl|CDD|225040',_binary 'Predicted phosphoesterases, related to the Icc protein [General function prediction only].','COG2129','COG2129'),(955507,'gnl|CDD|238606',_binary 'FHIT (fragile histidine family): FHIT proteins, related to the HIT family carry a motif HxHxH/Qxx (x, is a hydrophobic amino acid), On the basis of sequence, substrate specificity, structure, evolution and mechanism, HIT proteins are classified into three  branches: the Hint branch, which consists of adenosine 5\' -monophosphoramide hydrolases, the Fhit branch, that consists of diadenosine polyphosphate hydrolases, and the GalT branch consisting of specific nucloside monophosphate transferases. Fhit plays a very important role in the development of tumours. Infact, Fhit deletions are among the earliest and most frequent genetic alterations in the development of tumours.','cd01275','FHIT'),(955508,'gnl|CDD|238608',_binary 'HINT (histidine triad nucleotide-binding protein) subgroup: Members of this CD belong to the superfamily of histidine triad hydrolases that act on alpha-phosphate of ribonucleotides. This subgroup includes members from all three forms of cellular life. Although the biochemical function has not been characterised for many of the members of this subgroup, the proteins from Yeast have been shown to be involved in secretion, peroxisome formation and gene expression.','cd01277','HINT_subgroup'),(955509,'gnl|CDD|223611',_binary 'Diadenosine tetraphosphate (Ap4A) hydrolase and other HIT family hydrolases [Nucleotide transport and metabolism / Carbohydrate transport and metabolism / General function prediction only].','COG0537','Hit'),(955510,'gnl|CDD|307405',_binary 'HIT domain.','pfam01230','HIT'),(955511,'gnl|CDD|316861',_binary 'Superinfection immunity protein.  This family includes the E. coli bacteriophage T4 superinfection immunity (imm) protein. When E. coli is sequentially infected with two T-even type bacteriophage the DNA of the superinfecting phage is excluded from the host, into the periplasmic space. The immunity protein plays a role in this process.','pfam14373','Imm_superinfect'),(955513,'gnl|CDD|224588',_binary 'DNA segregation ATPase FtsK/SpoIIIE and related proteins [Cell division and chromosome partitioning].','COG1674','FtsK'),(955515,'gnl|CDD|236669',_binary 'DNA translocase FtsK; Provisional.','PRK10263','PRK10263'),(955516,'gnl|CDD|279863',_binary 'FtsK/SpoIIIE family.  FtsK has extensive sequence similarity to wide variety of proteins from prokaryotes and plasmids, termed the FtsK/SpoIIIE family. This domain contains a putative ATP binding P-loop motif. It is found in the FtsK cell division protein from E. coli and the stage III sporulation protein E SpoIIIE which has roles in regulation of prespore specific gene expression in B. subtilis. A mutation in FtsK causes a temperature sensitive block in cell division and it is involved in peptidoglycan synthesis or modification. The SpoIIIE protein is implicated in intercellular chromosomal DNA transfer.','pfam01580','FtsK_SpoIIIE'),(955517,'gnl|CDD|274860',_binary 'type VII secretion protein EssC, C-terminal domain.  This model describes the C-terminal domain, or longer subunit, of the Firmicutes type VII secretion protein EssC. This protein (homologous to EccC in Actinobacteria) and the WXG100 target proteins are the only homologous parts of type VII secretion between Firmicutes and Actinobacteria. [Protein fate, Protein and peptide secretion and trafficking].','TIGR03928','T7_EssCb_Firm'),(955520,'gnl|CDD|274858',_binary 'type VII secretion protein EccCa.  This model represents the N-terminal domain or EccCa subunit of the type VII secretion protein EccC as found in the Actinobacteria. Type VII secretion is defined more broadly as including secretion systems for ESAT-6-like proteins in the Firmicutes as well as in the Actinobacteria, but this family does not show close homologs in the Firmicutes. [Protein fate, Protein and peptide secretion and trafficking].','TIGR03924','T7SS_EccC_a'),(955521,'gnl|CDD|274859',_binary 'type VII secretion protein EccCb.  This model represents the C-terminal domain or EccCb subunit of the type VII secretion protein EccC as found in the Actinobacteria. Type VII secretion is defined more broadly as including secretion systems for ESAT-6-like proteins in the Firmicutes as well as in the Actinobacteria, but this family does not show close homologs in the Firmicutes. [Protein fate, Protein and peptide secretion and trafficking].','TIGR03925','T7SS_EccC_b'),(955523,'gnl|CDD|238408',_binary 'Ku-core domain, Ku-like subfamily; composed of prokaryotic homologs of the eukaryotic DNA binding protein Ku. The alignment includes the core domain shared by the prokaryotic YkoV-like proteins and the eukaryotic Ku70 and Ku80. The prokaryotic Ku homologs are predicted to form homodimers. It is proposed that the Ku homologs are functionally associated with ATP-dependent DNA ligase and the eukaryotic-type primase, probably as components of a double-strand break repair system.','cd00789','KU_like'),(955524,'gnl|CDD|274290',_binary 'Ku protein, prokaryotic.  Members of this protein family are Ku proteins of non-homologous end joining (NHEJ) DNA repair in bacteria and in at least one member of the archaea (Archaeoglobus fulgidus). Most members are encoded by a gene adjacent to the gene for the DNA ligase that completes the repair. The NHEJ system is broadly but rather sparsely distributed, being present in about one fifth of the first 250 completed prokarytotic genomes. A few species (e.g. Archaeoglobus fulgidus and Bradyrhizobium japonicum) have multiple copies that appear to represent recent paralogous family expansion. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02772','conserved_hypothetical_pr'),(955525,'gnl|CDD|224192',_binary 'Ku-homolog [Replication, recombination, and repair].','COG1273','COG1273'),(955526,'gnl|CDD|238334',_binary 'Ku-core domain; includes the central DNA-binding beta-barrels, polypeptide rings, and the C-terminal arm of Ku proteins. The Ku protein consists of two tightly associated homologous subunits, Ku70 and Ku80, and was originally identified as an autoantigen recognized by the sera of patients with an autoimmunity disease. In eukaryotes, the Ku heterodimer contributes to genomic integrity through its ability to bind DNA double-strand breaks and facilitate repair by non-homologous end-joining. The bacterial Ku homologs does not contain the conserved N-terminal extension that is present in the eukaryotic Ku protein.','cd00594','KU'),(955527,'gnl|CDD|308390',_binary 'Ku70/Ku80 beta-barrel domain.  The Ku heterodimer (composed of Ku70 and Ku80) contributes to genomic integrity through its ability to bind DNA double-strand breaks and facilitate repair by the non-homologous end-joining pathway. This is the central DNA-binding beta-barrel domain. This domain is found in both the Ku70 and Ku80 proteins that form a DNA binding heterodimer.','pfam02735','Ku'),(955528,'gnl|CDD|128831',_binary 'Ku70 and Ku80 are 70kDa and 80kDa subunits of the Lupus Ku autoantigen.  This is a single stranded DNA- and ATP-depedent helicase that has a role in chromosome translocation. This is a domain of unknown function C-terminal to its von Willebrand factor A domain, that also occurs in bacterial hypothetical proteins.','smart00559','Ku78'),(955529,'gnl|CDD|238407',_binary 'Ku-core domain, Ku70 subfamily; Ku70 is a subunit of the Ku protein, which plays a key role in multiple nuclear processes such as DNA repair, chromosome maintenance, transcription regulation, and V(D)J recombination. The mechanism underlying the regulation of all the diverse functions of Ku is still unclear, although it seems that Ku is a multifunctional protein that works in the nuclei. In mammalian cells, the Ku heterodimer recruits the catalytic subunit of DNA-dependent protein kinase (DNA-PK), which is dependent on its association with the Ku70/80 heterodimer bound to DNA for its protein kinase activity.','cd00788','KU70'),(955530,'gnl|CDD|314613',_binary 'Lsr2.  Lsr2 is a small, basic DNA-bridging protein present in Mycobacterium and related actinomycetes. It is a functional homolog of the H-NS-like proteins. H-NS proteins play a role in nucleoid organisation and also function as a pleiotropic regulator of gene expression.','pfam11774','Lsr2'),(955531,'gnl|CDD|238445',_binary 'Ku-core domain, Ku80 subfamily; Ku80 is a subunit of the Ku protein, which plays a key role in multiple nuclear processes such as DNA repair, chromosome maintenance, transcription regulation, and V(D)J recombination. The mechanism underlying the regulation of all the diverse functions of Ku is still unclear, although it seems that Ku is a multifunctional protein that works in nuclei. In mammalian cells, the Ku heterodimer recruits the catalytic subunit of DNA-dependent protein kinase (DNA-PK), which is dependent on its association with the Ku70/80 heterodimer bound to DNA for its protein kinase activity.','cd00873','KU80'),(955532,'gnl|CDD|306621',_binary 'Beta-lactamase.  This family appears to be distantly related to pfam00905 and PF00768 D-alanyl-D-alanine carboxypeptidase.','pfam00144','Beta-lactamase'),(955533,'gnl|CDD|224594',_binary 'Beta-lactamase class C and other penicillin binding proteins [Defense mechanisms].','COG1680','AmpC'),(955534,'gnl|CDD|236894',_binary 'beta-lactamase/D-alanine carboxypeptidase; Provisional.','PRK11289','ampC'),(955535,'gnl|CDD|236732',_binary 'beta-lactam binding protein AmpH; Provisional.','PRK10662','PRK10662'),(955536,'gnl|CDD|179620',_binary 'putative periplasmic esterase; Provisional.','PRK03642','PRK03642'),(955537,'gnl|CDD|171868',_binary 'D-aminopeptidase; Reviewed.','PRK13128','PRK13128'),(955538,'gnl|CDD|238020',_binary 'Fibronectin type 3 domain; One of three types of internal repeats found in the plasma protein fibronectin. Its tenth fibronectin type III repeat contains an RGD cell recognition sequence in a flexible loop between 2 strands. Approximately 2% of all animal proteins contain the FN3 repeat; including extracellular and intracellular proteins, membrane spanning cytokine receptors, growth hormone receptors, tyrosine phosphatase receptors, and adhesion molecules. FN3-like domains are also found in bacterial glycosyl hydrolases.','cd00063','FN3'),(955539,'gnl|CDD|315949',_binary 'Concanavalin A-like lectin/glucanases superfamily.  This domain belongs to the Concanavalin A-like lectin/glucanases superfamily.','pfam13385','Laminin_G_3'),(955544,'gnl|CDD|224321',_binary 'Restriction endonuclease [Defense mechanisms].','COG1403','McrA'),(955545,'gnl|CDD|119373',_binary 'Endo-N-acetylmuramidases (muramidases) are lysozymes (also referred to as peptidoglycan hydrolases) that degrade bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.  This family of muramidases contains a glycosyl hydrolase family 25 (GH25) catalytic domain and is found in bacteria, fungi, slime molds, round worms, protozoans and bacteriophages.  The bacteriophage members are referred to as endolysins which are involved in lysing the host cell at the end of the replication cycle to allow release of mature phage particles.  Endolysins are typically modular enzymes consisting of a catalytically active domain that hydrolyzes the peptidoglycan cell wall and a cell wall-binding domain that anchors the protein to the cell wall.  Endolysins generally have narrow substrate specificities with either intra-species or intra-genus bacteriolytic activity.','cd00599','GH25_muramidase'),(955546,'gnl|CDD|119385',_binary 'Lyc muramidase is an autolytic lysozyme (autolysin) from Clostridium acetobutylicum encoded by the lyc gene.  Lyc has a glycosyl hydrolase family 25 (GH25) catalytic domain.  Endo-N-acetylmuramidases are lysozymes (also referred to as peptidoglycan hydrolases) that degrade bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.','cd06525','GH25_Lyc-like'),(955547,'gnl|CDD|119384',_binary 'YegX is an uncharacterized bacterial protein with a glycosyl hydrolase family 25 (GH25) catalytic domain that is similar in sequence to the CH-type (Chalaropsis-type) lysozymes of the GH25 family of endolysins.','cd06524','GH25_YegX-like'),(955548,'gnl|CDD|119374',_binary 'CH-type (Chalaropsis-type) lysozymes represent one of four functionally-defined classes of peptidoglycan hydrolases (also referred to as endo-N-acetylmuramidases) that cleave bacterial cell wall peptidoglycans.  CH-type lysozymes exhibit both lysozyme (acetylmuramidase) and diacetylmuramidase activity. The first member of this family to be described was a muramidase from the fungus Chalaropsis.  However, a majority of the CH-type lysozymes are found in bacteriophages and Gram-positive bacteria such as Streptomyces and Clostridium.  CH-type lysozymes have a single glycosyl hydrolase family 25 (GH25) domain with an unusual beta/alpha-barrel fold in which the last strand of the barrel is antiparallel to strands beta7 and beta1.  Most CH-type lysozymes appear to lack the cell wall-binding domain found in other GH25 muramidases.','cd06412','GH25_CH-type'),(955549,'gnl|CDD|279518',_binary 'Glycosyl hydrolases family 25.','pfam01183','Glyco_hydro_25'),(955550,'gnl|CDD|119375',_binary 'Uncharacterized bacterial muramidase containing a glycosyl hydrolase family 25 (GH25) catalytic domain.  Endo-N-acetylmuramidases are lysozymes (also referred to as peptidoglycan hydrolases) that degrade bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.','cd06413','GH25_muramidase_1'),(955551,'gnl|CDD|239274',_binary 'NrdH-redoxin (NrdH) family; NrdH is a small monomeric protein with a conserved redox active CXXC motif within a TRX fold, characterized by a glutaredoxin (GRX)-like sequence and TRX-like activity profile. In vitro, it displays protein disulfide reductase activity that is dependent on TRX reductase, not glutathione (GSH). It is part of the NrdHIEF operon, where NrdEF codes for class Ib ribonucleotide reductase (RNR-Ib), an efficient enzyme at low oxygen levels. Under these conditions when GSH is mostly conjugated to spermidine, NrdH can still function and act as a hydrogen donor for RNR-Ib. It has been suggested that the NrdHEF system may be the oldest RNR reducing system, capable of functioning in a microaerophilic environment, where GSH was not yet available. NrdH from Corynebacterium ammoniagenes can form domain-swapped dimers, although it is unknown if this happens in vivo. Domain-swapped dimerization, which results in the blocking of the TRX reductase binding site, could be a mechanism for regulating the oxidation state of the protein.','cd02976','NrdH'),(955552,'gnl|CDD|119379',_binary 'LysA is a cell wall endolysin produced by Lactobacillus fermentum, which degrades bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.  The N-terminal glycosyl hydrolase family 25 (GH25) domain of LysA has sequence similarity with other murein hydrolase catalytic domains while the C-terminal domain has sequence similarity with putative bacterial cell wall-binding SH3b domains.  This domain family also includes LysL of Lactococcus lactis.','cd06417','GH25_LysA-like'),(955553,'gnl|CDD|223767',_binary 'Glutaredoxin and related proteins [Posttranslational modification, protein turnover, chaperones].','COG0695','GrxC'),(955554,'gnl|CDD|119377',_binary 'Cpl-1 lysin (also known as Cpl-9 lysozyme / muramidase) is a bacterial cell wall endolysin encoded by the pneumococcal bacteriophage Cp-1, which cleaves the glycosidic N-acetylmuramoyl-(beta1,4)-N-acetylglucosamine bonds of the pneumococcal glycan chain, thus acting as an enzymatic antimicrobial agent (an enzybiotic) against streptococcal infections. Cpl-1 belongs to the CP family of lysozymes (CPL lysozymes) which includes the Cpl-7 lysin.  Cpl-1 has a glycosyl hydrolase family 25 (GH25) catalytic domain with an irregular (beta/alpha)5-beta3 barrel and a C-terminal cell wall-anchoring module formed by six similar choline-binding repeats (ChBr\'s). The ChBr\'s facilitate the anchoring of Cpl-1 to the choline-containing teichoic acid of the pneumococcal cell wall. Other members of this domain family have an N-terminal CHAP (cysteine, histidine-dependent amidohydrolases/peptidases) domain similar to that of the firmicute CHAP lysins and associated with endopeptidase activity.  The Cpl-7 lysin is also included here as is LysB of Lactococcus phage, and the Mur lysin of Lactobacillus phage.','cd06415','GH25_Cpl1-like'),(955555,'gnl|CDD|131249',_binary 'Glutaredoxin-like protein NrdH.  NrdH-redoxin is a representative of a class of small redox proteins that contain a conserved CXXC motif and are characterized by a glutaredoxin-like amino acid sequence and thioredoxin-like activity profile. Unlike other the glutaredoxins to which it is most closely related, NrdH aparrently does not interact with glutathione/glutathione reductase, but rather with thioredoxin reductase to catalyze the reduction of ribonucleotide reductase.','TIGR02194','Glutaredoxin-like_protein'),(955556,'gnl|CDD|226280',_binary 'Lyzozyme M1 (1,4-beta-N-acetylmuramidase) [Cell envelope biogenesis, outer membrane].','COG3757','Acm'),(955557,'gnl|CDD|182381',_binary 'glutaredoxin-like protein; Provisional.','PRK10329','PRK10329'),(955558,'gnl|CDD|128889',_binary 'Glycosyl hydrolases family 25.','smart00641','Glyco_25'),(955559,'gnl|CDD|306872',_binary 'Glutaredoxin.','pfam00462','Glutaredoxin'),(955560,'gnl|CDD|119376',_binary 'The LytC lysozyme of Streptococcus pneumoniae is a bacterial cell wall hydrolase that cleaves the beta1-4-glycosydic bond located between the N-acetylmuramoyl-N-glucosaminyl residues of the cell wall polysaccharide chains.   LytC is composed of a C-terminal glycosyl hydrolase family 25 (GH25) domain and an N-terminal choline-binding module (CBM) consisting of eleven homologous repeats that specifically recognizes the choline residues of pneumococcal lipoteichoic and teichoic acids. This domain arrangement is the reverse of the major pneumococcal autolysin, LytA, and the CPL-1-like lytic enzymes of the pneumococcal bacteriophages, in which the CBM (consisting of six repeats) is at the C-terminus. This model represents the C-terminal catalytic domain of the LytC-like enzymes.','cd06414','GH25_LytC-like'),(955561,'gnl|CDD|274027',_binary 'Glutaredoxin-like protein, YruB-family.  This glutaredoxin-like protein family contains the conserved CxxC motif and includes the Clostridium pasteurianum protein YruB which has been cloned from a rubredoxin operon. Somewhat related to NrdH, it is unknown whether this protein actually interacts with glutathione/glutathione reducatase, or, like NrdH, some other reductant system.','TIGR02196','Gene_56_protein'),(955562,'gnl|CDD|119382',_binary 'AtlA is an autolysin found in Gram-positive lactic acid bacteria that degrades bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.  This family includes the AtlA and Aml autolysins from Streptococcus mutans which have a C-terminal glycosyl hydrolase family 25 (GH25) catalytic domain as well as six tandem N-terminal repeats of the GBS (group B Streptococcus) Bsp-like peptidoglycan-binding domain.  Other members of this family have one or more C-terminal peptidoglycan-binding domain(s) (SH3 or LysM) in addition to the GH25 domain.','cd06522','GH25_AtlA-like'),(955563,'gnl|CDD|119381',_binary 'Uncharacterized bacterial muramidase containing a glycosyl hydrolase family 25 (GH25) catalytic domain.  Endo-N-acetylmuramidases are lysozymes (also referred to as peptidoglycan hydrolases) that degrade bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.','cd06419','GH25_muramidase_2'),(955564,'gnl|CDD|119383',_binary 'PlyB is a bacteriophage endolysin that displays potent lytic activity toward Bacillus anthracis.  PlyB has an N-terminal glycosyl hydrolase family 25 (GH25) catalytic domain and a C-terminal bacterial SH3-like domain, SH3b.  Both domains are required for effective catalytic activity.  Endolysins are produced by bacteriophages at the end of their life cycle and participate in lysing the bacterial cell in order to release the newly formed progeny.  Endolysins (also referred to as endo-N-acetylmuramidases or peptidoglycan hydrolases) degrade bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.','cd06523','GH25_PlyB-like'),(955566,'gnl|CDD|310071',_binary 'Hom_end-associated Hint.  Homing endonucleases are encoded by mobile DNA elements that are found inserted within host genes in all domains of life. The crystal structure of the homing nuclease PI-Sce revealed two domains: an endonucleolytic centre resembling the C-terminal domain of Drosophila melanogaster Hedgehog protein, and a second domain containing the protein-splicing active site. This Domain corresponds to the latter protein-splicing domain.','pfam05203','Hom_end_hint'),(955567,'gnl|CDD|310072',_binary 'Homing endonuclease.  Homing endonucleases are encoded by mobile DNA elements that are found inserted within host genes in all domains of life.','pfam05204','Hom_end'),(955568,'gnl|CDD|316994',_binary 'LAGLIDADG-like domain.  This domain is part of the LAGLIDADG superfamily.','pfam14528','LAGLIDADG_3'),(955569,'gnl|CDD|197642',_binary 'Hint (Hedgehog/Intein) domain N-terminal region.  Hedgehog/Intein domain, N-terminal region. Domain has been split to accommodate large insertions of endonucleases.','smart00306','HintN'),(955570,'gnl|CDD|239091',_binary 'Peptidyl-tRNA hydrolase, type 2 (PTH2)_like . Peptidyl-tRNA hydrolase activity releases tRNA from the premature translation termination product peptidyl-tRNA. Two structurally different enzymes have been reported to encode such activity, Pth present in bacteria and eukaryotes and Pth2 present in archaea and eukaryotes.','cd02407','PTH2_family'),(955571,'gnl|CDD|239108',_binary 'Peptidyl-tRNA hydrolase, type 2 (PTH2). Peptidyl-tRNA hydrolase (PTH) activity releases tRNA from the premature translation termination product peptidyl-tRNA, therefore allowing the tRNA and peptide to be reused in protein synthesis. PTH2 is present in archaea and eukaryotes.','cd02430','PTH2'),(955572,'gnl|CDD|307896',_binary 'Peptidyl-tRNA hydrolase PTH2.  Peptidyl-tRNA hydrolases are enzymes that release tRNAs from peptidyl-tRNA during translation.','pfam01981','PTH2'),(955573,'gnl|CDD|235280',_binary 'peptidyl-tRNA hydrolase; Provisional.','PRK04322','PRK04322'),(955574,'gnl|CDD|224901',_binary 'Peptidyl-tRNA hydrolase [Translation, ribosomal structure and biogenesis].','COG1990','pth2'),(955575,'gnl|CDD|161803',_binary 'peptidyl-tRNA hydrolase.  This model describes an archaeal/eukaryotic form of peptidyl-tRNA hydrolase. Most bacterial forms are described by TIGR00447. [Protein synthesis, Other].','TIGR00283','Peptidyl-tRNA_hydrolase'),(955576,'gnl|CDD|239107',_binary 'Peptidyl-tRNA hydrolase, type 2 (PTH2)_like . Peptidyl-tRNA hydrolase activity releases tRNA from the premature translation termination product peptidyl-tRNA. Two structurally different enzymes have been reported  to encode such activity, Pth present in bacteria and eukaryotes and  Pth2 present in archaea and eukaryotes. There is no functional information for this eukaryote-specific subgroup.','cd02429','PTH2_like'),(955577,'gnl|CDD|309822',_binary 'Phage portal protein.  Bacteriophage portal proteins form a dodecamer and is located at a five-fold vertex of the viral capsid. The portal complex forms a channel through which the viral DNA is packaged into the capsid, and exits during infection. The portal protein is though to rotate during DNA packaging. Portal proteins from different phage show little sequence homology, so this family does not represent all portal proteins.','pfam04860','Phage_portal'),(955578,'gnl|CDD|273677',_binary 'phage portal protein, HK97 family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01537','portal_protein'),(955579,'gnl|CDD|227039',_binary 'Phage-related protein [Function unknown].','COG4695','COG4695'),(955582,'gnl|CDD|271189',_binary 'C-terminal catalytic domain of integrases from bacterial phages and conjugate transposons.  This family of tyrosine based site-specific integrases is has origins in bacterial phages and conjugate transposons. One member is the integrase from Bacillus subtilis conjugative transposon ICEBs1. ICEBs1 can be excised and transfered to various recipients in response to DNA damage or high concentrations of potential mating partners. The family belongs to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd01189','INT_ICEBs1_C_like'),(955583,'gnl|CDD|271177',_binary 'Shufflon-specific DNA recombinase Rci and Bacteriophage Hp1_like integrase, C-terminal catalytic domain.  Rci protein is a tyrosine recombinase specifically involved in Shufflon type of DNA rearrangement in bacteria. The shufflon of plasmid R64 consists of four invertible DNA segments which are separated and flanked by seven 19-bp repeat sequences. RCI recombinase facilitates the site-specific recombination between any inverted repeats results in an inversion of the DNA segment(s) either independently or in groups. HP1 integrase promotes site-specific recombination of the HP1 genome into that of Haemophilus influenza. Bacteriophage Hp1_like integrases are tyrosine based site specific recombinases. They belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism is essentially identical and involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd00796','INT_Rci_Hp1_C'),(955584,'gnl|CDD|234698',_binary 'site-specific tyrosine recombinase XerC; Reviewed.','PRK00236','xerC'),(955585,'gnl|CDD|274042',_binary 'tyrosine recombinase XerC.  The phage integrase family describes a number of recombinases with tyrosine active sites that transiently bind covalently to DNA. Many are associated with mobile DNA elements, including phage, transposons, and phase variation loci. This model represents XerC, one of two closely related chromosomal proteins along with XerD (TIGR02225). XerC and XerD are site-specific recombinases which help resolve chromosome dimers to monomers for cell division after DNA replication. In species with a large chromosome and homologs of XerC on other replicons, the chomosomal copy was preferred for building this model. This model does not detect all XerC, as some apparent XerC examples score in the gray zone between trusted (450) and noise (410) cutoffs, along with some XerD examples. XerC and XerD interact with cell division protein FtsK. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02224','Tyrosine_recombinase_XerC'),(955586,'gnl|CDD|271175',_binary 'DNA breaking-rejoining enzymes, C-terminal catalytic domain.  The DNA breaking-rejoining enzyme superfamily includes type IB topoisomerases and tyrosine based site-specific recombinases (integrases) that share the same fold in their catalytic domain containing conserved active site residues. The best-studied members of this diverse superfamily include Human topoisomerase I, the bacteriophage lambda integrase, the bacteriophage P1 Cre recombinase, the yeast Flp recombinase, and the bacterial XerD/C recombinases. Their overall reaction mechanism is essentially identical and involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. The enzymes differ in that topoisomerases cleave and then rejoin the same 5\' and 3\' termini, whereas a site-specific recombinase transfers a 5\' hydroxyl generated by recombinase cleavage to a new 3\' phosphate partner located in a different duplex region. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd00397','DNA_BRE_C'),(955587,'gnl|CDD|274043',_binary 'tyrosine recombinase XerD.  The phage integrase family describes a number of recombinases with tyrosine active sites that transiently bind covalently to DNA. Many are associated with mobile DNA elements, including phage, transposons, and phase variation loci. This model represents XerD, one of two closely related chromosomal proteins along with XerC (TIGR02224). XerC and XerD are site-specific recombinases which help resolve chromosome dimers to monomers for cell division after DNA replication. In species with a large chromosome and with homologs of XerD on other replicons, the chomosomal copy was preferred for building this model. This model does not detect all XerD, as some apparent XerD examples score below the trusted and noise cutoff scores. XerC and XerD interact with cell division protein FtsK. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02225','Tyrosine_recombinase_XerD'),(955588,'gnl|CDD|271186',_binary 'Putative Transposase A from transposon Tn554, C-terminal catalytic domain.  This family includes putative Transposase A from transposon Tn554. It belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01186','INT_tnpA_C_Tn554'),(955589,'gnl|CDD|271179',_binary 'XerD and XerC integrases, C-terminal catalytic domains.  XerDC-like integrases are involved in the site-specific integration and excision of lysogenic bacteriophage genomes, transposition of conjugative transposons, termination of chromosomal replication, and stable plasmid inheritance. They share the same fold in their catalytic domain containing six conserved active site residues and the overall reaction mechanism with the DNA breaking-rejoining enzyme superfamily. In Escherichia coli, the Xer site-specific recombination system acts to convert dimeric chromosomes, which are formed by homologous recombination to monomers. Two related recombinases, XerC and XerD, bind cooperatively to a recombination site present in the E. coli chromosome. Each recombinase catalyzes the exchange of one pair of DNA strand in a reaction that proceeds through a Holliday junction intermediate. These enzymes can bridge two different and well-separated DNA sequences called arm- and core-sites. The C-terminal domain binds, cleaves, and re-ligates DNA strands at the core-sites, while the N-terminal domain is largely responsible for high-affinity binding to the arm-type sites.','cd00798','INT_XerDC_C'),(955590,'gnl|CDD|278986',_binary 'Phage integrase family.  Members of this family cleave DNA substrates by a series of staggered cuts, during which the protein becomes covalently linked to the DNA through a catalytic tyrosine residue at the carboxy end of the alignment. The catalytic site residues in CRE recombinase are Arg-173, His-289, Arg-292 and Tyr-324.','pfam00589','Phage_integrase'),(955591,'gnl|CDD|227308',_binary 'Site-specific recombinase XerD [DNA replication, recombination, and repair].','COG4974','XerD'),(955592,'gnl|CDD|223655',_binary 'Integrase [DNA replication, recombination, and repair].','COG0582','XerC'),(955593,'gnl|CDD|271185',_binary 'Integrase IntN1 of Bacteroides mobilizable transposon NBU1 and similar proteins, C-terminal catalytic domain.  IntN1 is a tyrosine recombinase for the integration and excision of Bacteroides mobilizable transposon NBU1 from the host chromosome. IntN1 does not require strict homology between the recombining sites seen with other tyrosine recombinases. This family belongs to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd01185','INTN1_C_like'),(955594,'gnl|CDD|222904',_binary 'integrase; Provisional.','PHA02601','int'),(955595,'gnl|CDD|271194',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01194','INT_C_like_4'),(955596,'gnl|CDD|131303',_binary 'integron integrase.  Members of this family are integrases associated with integrons (and super-integrons), which are systems for incorporating and expressing cassettes of laterally transferred DNA. Incorporation occurs at an attI site. A super-integron, as in Vibrio sp., may include over 100 cassettes. This family belongs to the phage integrase family (pfam00589) that also includes recombinases XerC (TIGR02224) and XerD (TIGR02225), which are bacterial housekeeping proteins. Within this family of integron integrases, some are designated by class, e.g. IntI4, a class 4 integron integrase from Vibrio cholerae N16961. [DNA metabolism, DNA replication, recombination, and repair, Mobile and extrachromosomal element functions, Other].','TIGR02249','Integrase/recombinase_E2_'),(955597,'gnl|CDD|227307',_binary 'Site-specific recombinase XerC [DNA replication, recombination, and repair].','COG4973','XerC'),(955598,'gnl|CDD|271188',_binary 'C-terminal catalytic domain of recombinase RitA, a component of the recombinase trio.  Recombinases RitA (also known as pAE1), RitB, and RitC are encoded by three adjacent and overlapping genes. Collectively they are known as the Recombinase in Trio (RIT). This RitA family includes various bacterial integrases and integrases from the deletion-prone region of plasmid pAE1 of Alcaligenes eutrophus H1. All three integrases contain a catalytic motif, suggesting that they are all active enzymes. However, their specific roles are not fully understood. All three families belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism is essentially identical and involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd01188','INT_RitA_C_like'),(955599,'gnl|CDD|234713',_binary 'site-specific tyrosine recombinase XerD; Reviewed.','PRK00283','xerD'),(955600,'gnl|CDD|271193',_binary 'Integron integrase and similar protiens, C-terminal catalytic domain.  Integron integrases mediate site-specific DNA recombination between a proximal primary site (attI) and a secondary target site (attC) found within mobile gene cassettes encoding resistance or virulence factors. Unlike other site specific recombinases, the attC sites lack sequence conservation. Integron integrase exhibits broader DNA specificity by recognizing the non-conserved attC sites. The structure shows that DNA target site recognition are not dependent on canonical DNA but on the position of two flipped-out bases that interact in cis and in trans with the integrase. Integron-integrases are present in many natural occurring mobile elements, including transposons and conjugative plasmids. Vibrio, Shewanella, Xanthomonas, and Pseudomonas species harbor chromosomal super-integrons. All integron-integrases carry large inserts unlike the TnpF ermF-like proteins also seen in this group.','cd01193','INT_IntI_C'),(955601,'gnl|CDD|235339',_binary 'site-specific tyrosine recombinase XerS; Reviewed.','PRK05084','xerS'),(955602,'gnl|CDD|185315',_binary 'integrase/recombinase; Provisional.','PRK15417','PRK15417'),(955603,'gnl|CDD|317104',_binary 'Phage integrase, N-terminal SAM-like domain.  This domain is found in a variety of phage integrase proteins.','pfam14659','Phage_int_SAM_3'),(955604,'gnl|CDD|271182',_binary 'Bacteriophage P4 integrase, C-terminal catalytic domain.  P4-like integrases are found in temperate bacteriophages, integrative plasmids, pathogenicity and symbiosis islands, and other mobile genetic elements. The P4 integrase mediates integrative and excisive site-specific recombination between two sites, called attachment sites, located on the phage genome and the bacterial chromosome. The phage attachment site is often found adjacent to the integrase gene, while the host attachment sites are typically situated near tRNA genes. This family belongs to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd00801','INT_P4_C'),(955605,'gnl|CDD|271187',_binary 'Putative Transposase B from transposon Tn554, C-terminal catalytic domain.  This family includes putative Transposase B from transposon Tn554. It belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain containing six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01187','INT_tnpB_C_Tn554'),(955606,'gnl|CDD|309640',_binary 'Caudovirus prohead serine protease.  Family of Caudovirus prohead serine proteases also found in a number of bacteria possibly as the result of horizontal transfer.','pfam04586','Peptidase_S78'),(955607,'gnl|CDD|226263',_binary 'Phage head maturation protease [General function prediction only].','COG3740','COG3740'),(955608,'gnl|CDD|273682',_binary 'phage prohead protease, HK97 family.  This model describes the prohead protease of HK97 and related phage. It is generally encoded next to the gene for the capsid protein that it processes, and in some cases may be fused to it. This family does not show similarity to the prohead protease of phage T4 (see pfam03420). [Mobile and extrachromosomal element functions, Prophage functions, Protein fate, Other].','TIGR01543','Putative_prohead_protease'),(955609,'gnl|CDD|316874',_binary 'Domain of unknown function (DUF4417).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 220 and 340 amino acids in length. There is a single completely conserved residue G that may be functionally important.','pfam14386','DUF4417'),(955610,'gnl|CDD|292699',_binary 'Mycobacterial 2 TMS Phage Holin (M2 Hol) Family.  Phage_holin_8_1 is a family of two transmembrane mycobacteriophage holins, small hydrophobic proteins that effect lysis of host mycobacterial cells in conjunction with a mycobacteria-specific lysin, lysB. The product of lysB gene targets the mycobacteria outer membrane, the last barrier to bacteriophage release.','pfam16081','Phage_holin_7_1'),(955611,'gnl|CDD|310975',_binary 'Transglycosylase-like domain.  This family of proteins are very likely to act as transglycosylase enzymes related to pfam00062 and pfam01464. These other families are weakly matched by this family, and include the known active site residues.','pfam06737','Transglycosylas'),(955612,'gnl|CDD|316097',_binary 'D-alanyl-D-alanine carboxypeptidase.  This family resembles VanY, pfam02557, which is part of the peptidase M15 family.','pfam13539','Peptidase_M15_4'),(955613,'gnl|CDD|310019',_binary 'Phage portal protein, SPP1 Gp6-like.  This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. During SPP1 morphogenesis, Gp6 participates in the procapsid assembly reaction. This family also includes the old Pfam family Phage_min_cap (PF05126).','pfam05133','Phage_prot_Gp6'),(955614,'gnl|CDD|226974',_binary 'Phage terminase-like protein, large subunit [General function prediction only].','COG4626','COG4626'),(955615,'gnl|CDD|281364',_binary 'Phage Terminase.  The majority of the members of this family are bacteriophage proteins, several of which are thought to be terminase large subunit proteins. There are also a number of bacterial proteins of unknown function.','pfam03354','Terminase_1'),(955618,'gnl|CDD|286444',_binary 'Phage protein Gp19/Gp15/Gp42.  This family of proteins are functionally uncharacterized. They are found in a variety of bacteriophage.','pfam09355','Phage_Gp19'),(955619,'gnl|CDD|287830',_binary 'Protein of unknown function (DUF2744).  This is a viral family of proteins with unknown function.','pfam10910','DUF2744'),(955620,'gnl|CDD|227699',_binary 'Phage-related protein [Function unknown].','COG5412','COG5412'),(955621,'gnl|CDD|313949',_binary 'Protein of unknown function (DUF2746).  This family of proteins has no known function.','pfam10874','DUF2746'),(955624,'gnl|CDD|222854',_binary 'ParA-like protein; Provisional.','PHA02518','PHA02518'),(955625,'gnl|CDD|224113',_binary 'ATPases involved in chromosome partitioning [Cell division and chromosome partitioning].','COG1192','Soj'),(955626,'gnl|CDD|238997',_binary 'ParA and ParB of Caulobacter crescentus belong to a conserved family of bacterial proteins implicated in chromosome segregation. ParB binds to DNA sequences adjacent to the origin of replication and localizes to opposite cell poles shortly following the initiation of DNA replication. ParB regulates the ParA ATPase activity by promoting nucleotide exchange in a fashion reminiscent of the exchange factors of eukaryotic G proteins. ADP-bound ParA binds single-stranded DNA, whereas the ATP-bound form dissociates ParB from its DNA binding sites. Increasing the fraction of ParA-ADP in the cell inhibits cell division, suggesting that this simple nucleotide switch may regulate cytokinesis. ParA shares sequence similarity to a conserved and widespread family of ATPases which includes the repA protein of the repABC operon in R. etli Sym plasmid. This operon is involved in the plasmid replication and partition.','cd02042','ParA'),(955627,'gnl|CDD|148565',_binary 'VirC1 protein.  This family consists of several bacterial VirC1 proteins. In Agrobacterium tumefaciens, a cis-active 24-base-pair sequence adjacent to the right border of the T-DNA, called overdrive, stimulates tumor formation by increasing the level of T-DNA processing. It is thought that the virC operon which enhances T-DNA processing probably does so because the VirC1 protein interacts with overdrive. It has now been shown that the virC1 gene product binds to overdrive but not to the right border of T-DNA.','pfam07015','VirC1'),(955628,'gnl|CDD|238613',_binary 'Deoxycytidylate deaminase domain. Deoxycytidylate deaminase catalyzes the deamination of dCMP to dUMP,  providing the nucleotide substrate for thymidylate synthase. The enzyme binds Zn++, which is required for catalytic activity. The activity of the enzyme is allosterically regulated by the ratio of dCTP to dTTP not only in eukaryotic cells but also in T-even phage-infected Escherichia coli, with dCTP acting as an activator and dTTP as an inhibitor.','cd01286','deoxycytidylate_deaminase'),(955629,'gnl|CDD|225042',_binary 'Deoxycytidylate deaminase [Nucleotide transport and metabolism].','COG2131','ComEB'),(955630,'gnl|CDD|131622',_binary 'ComE operon protein 2.  This protein is found in the ComE operon for \'late competence\' as characterized in B. subtilis. Proteins in this family contain homology to a cytidine/deoxycytidine deaminase domain family (pfam00383), and may carry out this activity.','TIGR02571','putative_dCMP_deaminase'),(955631,'gnl|CDD|222894',_binary 'deoxycytidylate deaminase; Provisional.','PHA02588','cd'),(955632,'gnl|CDD|278795',_binary 'Cytidine and deoxycytidylate deaminase zinc-binding region.','pfam00383','dCMP_cyt_deam_1'),(955633,'gnl|CDD|316921',_binary 'MafB19-like deaminase.  A member of the nucleic acid/nucleotide deaminase superfamily prototyped by Neisseria MafB19. Members of this family are present in a wide phyletic range of bacteria and are predicted to function as toxins in bacterial polymorphic toxin systems.','pfam14437','MafB19-deam'),(955634,'gnl|CDD|223663',_binary 'Cytosine/adenosine deaminases [Nucleotide transport and metabolism / Translation, ribosomal structure and biogenesis].','COG0590','CumB'),(955635,'gnl|CDD|223195',_binary 'Pyrimidine deaminase [Coenzyme metabolism].','COG0117','RibD'),(955636,'gnl|CDD|238611',_binary 'Riboflavin-specific deaminase. Riboflavin biosynthesis protein RibD (Diaminohydroxyphosphoribosylaminopyrimidine deaminase) catalyzes the deamination of 2,5-diamino-6-ribosylamino-4(3H)-pyrimidinone 5\'-phosphate, which is an intermediate step in the biosynthesis of riboflavin.The ribG gene of Bacillus subtilis and the ribD gene of E. coli are bifunctional and contain this deaminase domain and a reductase domain which catalyzes the subsequent reduction of the ribosyl side chain.','cd01284','Riboflavin_deaminase-redu'),(955637,'gnl|CDD|182786',_binary 'tRNA-specific adenosine deaminase; Provisional.','PRK10860','PRK10860'),(955638,'gnl|CDD|238406',_binary 'Cytidine and deoxycytidylate deaminase zinc-binding region. The family contains cytidine deaminases, nucleoside deaminases, deoxycytidylate deaminases and riboflavin deaminases. Also included are the apoBec family of mRNA editing enzymes.  All members are Zn dependent. The zinc ion in the active site plays a central role in the proposed catalytic mechanism, activating a water molecule to form a hydroxide ion that performs a nucleophilic attack on the substrate.','cd00786','cytidine_deaminase-like'),(955639,'gnl|CDD|273015',_binary 'riboflavin biosynthesis protein RibD.  This model describes the ribD protein as found in Escherichia coli. The N-terminal domain includes the conserved zinc-binding site region captured in the model dCMP_cyt_deam and shared by proteins such as cytosine deaminase, mammalian apolipoprotein B mRNA editing protein, blasticidin-S deaminase, and Bacillus subtilis competence protein comEB. The C-terminal domain is homologous to the full length of yeast HTP reductase, a protein required for riboflavin biosynthesis. A number of archaeal proteins believed related to riboflavin biosynthesis contain only this C-terminal domain and are not found as full-length matches to this model. [Biosynthesis of cofactors, prosthetic groups, and carriers, Riboflavin, FMN, and FAD].','TIGR00326','eubact_ribD'),(955640,'gnl|CDD|238612',_binary 'Nucleoside deaminases include adenosine, guanine and cytosine deaminases. These enzymes are Zn dependent and catalyze the deamination of nucleosides. The zinc ion in the active site plays a central role in the proposed catalytic mechanism, activating a water molecule to form a hydroxide ion that performs a nucleophilic attack on the substrate. The functional enzyme is a homodimer. Cytosine deaminase catalyzes the deamination of cytosine to uracil and ammonia and is a member of the pyrimidine salvage pathway. Cytosine deaminase is found in bacteria and fungi but is not present in mammals; for this reason, the enzyme is currently of interest for antimicrobial drug design and gene therapy applications against tumors. Some members of this family are tRNA-specific adenosine deaminases that generate inosine at the first position of their anticodon (position 34) of specific tRNAs; this modification is thought to enlarge the codon recognition capacity during protein synthesis. Other members of the family are guanine deaminases which deaminate guanine to xanthine as part of the utilization of guanine as a nitrogen source.','cd01285','nucleoside_deaminase'),(955642,'gnl|CDD|308231',_binary 'Thymidylate synthase complementing protein.  Thymidylate synthase complementing protein (Thy1) complements the thymidine growth requirement of the organisms in which it is found, but shows no homology to thymidylate synthase. The bacterial members of this family at least are flavin-dependent thymidylate synthases.','pfam02511','Thy1'),(955643,'gnl|CDD|214687',_binary 'DNA polymerase A domain.','smart00482','POLAc'),(955644,'gnl|CDD|234847',_binary 'FAD-dependent thymidylate synthase; Reviewed.','PRK00847','thyX'),(955645,'gnl|CDD|306881',_binary 'DNA polymerase family A.','pfam00476','DNA_pol_A'),(955646,'gnl|CDD|274010',_binary 'thymidylate synthase, flavin-dependent.  Two forms of microbial thymidylate synthase are known: ThyA (2.1.1.45) and ThyX (2.1.1.148). This model describes ThyX, a homotetrameric flavoprotein. Both enzymes convert dUMP to dTMP. Under oxygen-limiting conditions, thyX can complement a thyA mutation. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02170','Thymidylate_synthase_ThyX'),(955647,'gnl|CDD|223820',_binary 'DNA polymerase I - 3\'-5\' exonuclease and polymerase domains [DNA replication, recombination, and repair].','COG0749','PolA'),(955648,'gnl|CDD|224270',_binary 'Predicted alternative thymidylate synthase [Nucleotide transport and metabolism].','COG1351','THY1'),(955649,'gnl|CDD|235591',_binary 'DNA polymerase I; Provisional.','PRK05755','PRK05755'),(955650,'gnl|CDD|176474',_binary 'Polymerase I functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication.  Family A polymerase (polymerase I) functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I (pol I) ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5\'-3\' polymerase and 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used to search for protein signatures. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.','cd08637','DNA_pol_A_pol_I_C'),(955651,'gnl|CDD|273160',_binary 'DNA polymerase I.  All proteins in this family for which functions are known are DNA polymerases Many also have an exonuclease motif. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00593','DNA_polymerase_I'),(955652,'gnl|CDD|237876',_binary 'bifunctional 3\'-5\' exonuclease/DNA polymerase; Provisional.','PRK14975','PRK14975'),(955653,'gnl|CDD|176476',_binary 'Phylum Aquificae Pol A is different from Escherichia coli  Pol A by three signature sequences.  Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5\'-3\' polymerase and 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used for phylogenetic anaylsis of bacteria. Species of the phylum Aquificae grow in extreme thermophilic environments. The Aquificae are non-spore-forming, Gram-negative rods and strictly thermophilic. Phylum Aquificae Pol A is different from E. coli Pol I by three signature sequences consisting of a 2 amino acids (aa) insert, a 5-6 aa insert and a 6 aa deletion. These signature sequences may provide a molecular marker for the family Aquificaceae and related species.','cd08639','DNA_pol_A_Aquificae_like'),(955654,'gnl|CDD|176475',_binary 'DNA polymerase theta is a low-fidelity family A enzyme implicated in translesion synthesis and in somatic hypermutation.  DNA polymerase theta is a low-fidelity family A enzyme implicated in translesion synthesis (TLS) and in somatic hypermutation (SHM). DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. Pol theta is an exception among family A polymerases and generates processive single base substitutions. Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I (pol I) ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5\'-3\' polymerase and 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. Polymerase theta mostly has amino-terminal helicase domain, a carboxy-terminal polymerase domain and an intervening space region.','cd08638','DNA_pol_A_theta'),(955655,'gnl|CDD|176477',_binary 'DNA polymerase A type from plastids of higher plants possibly involve in DNA replication or in the repair of errors occurring during replication.  DNA polymerase A type from plastids of higher plants possibly involve in DNA replication or in the repair of errors occurring during replication. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7).   The three-dimensional structure of plastid DNA polymerase has substantial similarity to Pol I. The structure of Pol I resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.','cd08640','DNA_pol_A_plastid_like'),(955656,'gnl|CDD|176473',_binary 'Family A polymerase primarily fills DNA gaps that arise during DNA repair, recombination and replication.  DNA polymerase family A, 5\'-3\' polymerase domain. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified into six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaeota polymerase II (class D), human polymerase  beta (class X), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerases are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I, mitochondrial polymerase gamma, and several bacteriophage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic polymerase I (pol I) has two functional domains located on the same polypeptide; a 5\'-3\' polymerase and a 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and the DNA polymerase activity to fill in the resulting gap. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.','cd06444','DNA_pol_A'),(955657,'gnl|CDD|214681',_binary '3\'-5\' exonuclease.  3\' -5\' exonuclease proofreading domain present in DNA polymerase I, Werner syndrome helicase, RNase D and other enzymes.','smart00474','35EXOc'),(955658,'gnl|CDD|176480',_binary 'Polymerase I functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication.  Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5\'-3\' polymerase and 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used to search for protein signatures. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.','cd08643','DNA_pol_A_pol_I_B'),(955659,'gnl|CDD|307647',_binary '3\'-5\' exonuclease.  This domain is responsible for the 3\'-5\' exonuclease proofreading activity of E. coli DNA polymerase I (polI) and other enzymes, it catalyzes the hydrolysis of unpaired or mismatched nucleotides. This domain consists of the amino-terminal half of the Klenow fragment in E. coli polI it is also found in the Werner syndrome helicase (WRN), focus forming activity 1 protein (FFA-1) and ribonuclease D (RNase D). Werner syndrome is a human genetic disorder causing premature aging; the WRN protein has helicase activity in the 3\'-5\' direction. The FFA-1 protein is required for formation of a replication foci and also has helicase activity; it is a homolog of the WRN protein. RNase D is a 3\'-5\' exonuclease involved in tRNA processing. Also found in this family is the autoantigen PM/Scl thought to be involved in polymyositis-scleroderma overlap syndrome.','pfam01612','DNA_pol_A_exo1'),(955660,'gnl|CDD|176651',_binary 'DEDDy 3\'-5\' exonuclease domain of Escherichia coli DNA polymerase I and similar bacterial family-A DNA polymerases.  Escherichia coli-like Polymerase I (Pol I), a subgroup of family-A DNA polymerases, contains a DEDDy-type DnaQ-like 3\'-5\' exonuclease domain in the same polypeptide chain as the polymerase domain. The exonuclease domain contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The 3\'-5\' exonuclease domain of DNA polymerases has a fundamental role in reducing polymerase errors and is involved in proofreading activity. E. coli DNA Pol I is involved in genome replication but is not the main replicating enzyme. It is also implicated in DNA repair.','cd06139','DNA_polA_I_Ecoli_like_exo'),(955661,'gnl|CDD|176479',_binary 'Polymerase I functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication.  Family A polymerase (polymerase I) functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5\'-3\' polymerase and 5\'-3\' exonuclease. Pol I uses its 5\' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used to search for protein signatures. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.','cd08642','DNA_pol_A_pol_I_A'),(955662,'gnl|CDD|176654',_binary 'DEDDy 3\'-5\' exonuclease domain of Ribonuclease D and similar proteins.  Ribonuclease (RNase) D is a bacterial enzyme involved in the maturation of small stable RNAs and the 3\' maturation of tRNA. It contains a DEDDy-type DnaQ-like 3\'-5\' exonuclease domain possessing three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. In vivo, RNase D only becomes essential upon removal of other ribonucleases. Eukaryotic RNase D homologs include yeast Rrp6p, human PM/Scl-100, and the Drosophila melanogaster egalitarian protein.','cd06142','RNaseD_exo'),(955663,'gnl|CDD|223426',_binary 'Ribonuclease D [Translation, ribosomal structure and biogenesis].','COG0349','Rnd'),(955664,'gnl|CDD|176478',_binary 'Pol gammaA is a family A polymerase that is responsible for DNA replication and repair in mitochondria.  DNA polymerase gamma (Pol gamma), 5\'-3\' polymerase domain (Pol gammaA). Pol gammaA is a family A polymerase that is responsible for DNA replication and repair in mitochondria. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified into six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaeota polymerase II (class D), human polymerase beta (class X), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerases are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I, mitochondrial polymerase gammaA, and several bacteriophage polymerases including those from odd-numbered phage (T3, T5, and T7).   The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains. Pol gammaA has also the right hand configuration. Pol gammaA has both polymerase and proofreading exonuclease activities separated by a spacer. Pol gamma holoenzyme is a heterotrimer containing one Pol gammaA subunit and a dimeric Pol gammaB subunit. Pol gamma is important for mitochondria DNA maintenance and mutation of the catalytic subunit of Pol gamma is implicated in more than 30 human diseases.','cd08641','DNA_pol_gammaA'),(955665,'gnl|CDD|274169',_binary 'ribonucleoside-triphosphate reductase, adenosylcobalamin-dependent.  This model represents a group of adenosylcobalamin(B12)-dependent ribonucleotide reductases (RNR) related to the characterized species from Lactococcus leichmannii. RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. Thus model identifies NrdJ enzymes only in cyanobacteria, lactococcus and certain bacteriophage. A separate model (TIGR02504) identifies a larger group of divergent B12-dependent RNR\'s. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02505','RTPR'),(955666,'gnl|CDD|153085',_binary 'Class II ribonucleotide reductase, monomeric form.  Ribonucleotide reductase (RNR) catalyzes the reductive synthesis of deoxyribonucleotides from their corresponding ribonucleotides. It provides the precursors necessary for DNA synthesis. RNRs are separated into three classes based on their metallocofactor usage. Class I RNRs, found in eukaryotes, bacteria, and bacteriophage, use a diiron-tyrosyl radical. Class II RNRs, found in bacteria, bacteriophage, algae and archaea, use coenzyme B12 (adenosylcobalamin, AdoCbl). Class III RNRs, found in anaerobic bacteria, bacteriophage, and archaea, use an FeS cluster and S-adenosylmethionine to generate a glycyl radical. Many organisms have more than one class of RNR present in their genomes. All three RNRs have a ten-stranded alpha-beta barrel domain that is structurally similar to  the domain of PFL (pyruvate formate lyase). Class II RNRs are found in bacteria that can live under both aerobic and anaerobic conditions. Many, but not all members of this class, are found to be homodimers. This particular subfamily is found to be active as a monomer. Adenosylcobalamin interacts directly with an active site cysteine to form the reactive cysteine radical.','cd01676','RNR_II_monomer'),(955667,'gnl|CDD|274168',_binary 'ribonucleoside-diphosphate reductase, adenosylcobalamin-dependent.  This model represents a group of adenosylcobalamin(B12)-dependent ribonucleotide reductases (Class II RNRs) related to the characterized species from Pyrococcus, Thermoplasma, Corynebacterium, and Deinococcus. RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. This model identifies genes in a wide range of deeply branching bacteria. All are structurally related to the class I (non-heme iron dependent) RNRs. In most species this gene is known as NrdJ, while in mycobacteria it is called NrdZ. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02504','ribonucleotide_reductase'),(955669,'gnl|CDD|153089',_binary 'Class II ribonucleotide reductase, dimeric form.  Ribonucleotide reductase (RNR) catalyzes the reductive synthesis of deoxyribonucleotides from their corresponding ribonucleotides. It provides the precursors necessary for DNA synthesis. RNRs are separated into three classes based on their metallocofactor usage. Class I RNRs, found in eukaryotes, bacteria, and bacteriophage, use a diiron-tyrosyl radical. Class II RNRs, found in bacteria, bacteriophage, algae and archaea, use coenzyme B12 (adenosylcobalamin, AdoCbl). Class III RNRs, found in anaerobic bacteria, bacteriophage, and archaea, use an FeS cluster and S-adenosylmethionine to generate a glycyl radical. Many organisms have more than one class of RNR present in their genomes. All three RNRs have a ten-stranded alpha-beta barrel domain that is structurally similar to the domain of PFL (pyruvate formate lyase). Class II RNRs are found in bacteria that can live under both aerobic and anaerobic conditions. Many, but not all members of this class are found to be homodimers. Adenosylcobalamin interacts directly with an active site cysteine to form the reactive cysteine radical.','cd02888','RNR_II_dimer'),(955670,'gnl|CDD|308485',_binary 'Ribonucleotide reductase, barrel domain.','pfam02867','Ribonuc_red_lgC'),(955671,'gnl|CDD|236330',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK08665','PRK08665'),(955672,'gnl|CDD|235796',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK06406','PRK06406'),(955673,'gnl|CDD|153088',_binary 'Class I ribonucleotide reductase.  Ribonucleotide reductase (RNR) catalyzes the reductive synthesis of deoxyribonucleotides from their corresponding ribonucleotides. It provides the precursors necessary for DNA synthesis. RNRs are separated into three classes based on their metallocofactor usage. Class I RNRs, found in eukaryotes, bacteria, and many viruses, use a diiron-tyrosyl radical. Class II RNRs, found in bacteria, bacteriophage, algae and archaea, use coenzyme B12 (adenosylcobalamin, AdoCbl). Class III RNRs, found in anaerobic bacteria, bacteriophages, and archaea, use an FeS cluster and S-adenosylmethionine to generate a glycyl radical. Many organisms have more than one class of RNR present in their genomes. All three RNRs have a ten-stranded alpha-beta barrel domain that is structurally similar to  the domain of PFL (pyruvate formate lyase). Class I RNR is oxygen-dependent and can be subdivided into classes Ia (eukaryotes, prokaryotes, viruses and phages) and Ib (which is found in prokaryotes only). It is a tetrameric enzyme of two alpha and two beta subunits; this model covers the major part of the alpha or large subunit, called R1 in class Ia and R1E in class Ib.','cd01679','RNR_I'),(955674,'gnl|CDD|223287',_binary 'Ribonucleotide reductase, alpha subunit [Nucleotide transport and metabolism].','COG0209','NrdA'),(955676,'gnl|CDD|274170',_binary 'ribonucleoside-diphosphate reductase, alpha subunit.  This model represents the alpha (large) chain of the class I ribonucleotide reductase (RNR). RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. Class I RNR\'s generate the required radical (on tyrosine) via a \'non-heme\' iron cofactor which resides in the beta (small) subunit. The alpha subunit contains the catalytic and allosteric regulatory sites. The mechanism of this enzyme requires molecular oxygen. E. Coli contains two versions of this enzyme which are regulated independently (NrdAB and NrdEF, where NrdA and NrdE are the large chains). Most organisms contain only one, but the application of the gene symbols NrdA and NrdE are somewhat arbitrary. This model identifies RNR\'s in diverse clades of bacteria, eukaryotes as well as numerous DNA viruses and phage. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02506','Ribonucleoside-diphosphat'),(955677,'gnl|CDD|180772',_binary 'ribonucleotide reductase-like protein; Provisional.','PRK06948','PRK06948'),(955678,'gnl|CDD|236152',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK08115','PRK08115'),(955680,'gnl|CDD|235828',_binary 'vitamin B12-dependent ribonucleotide reductase; Validated.','PRK06556','PRK06556'),(955681,'gnl|CDD|153083',_binary 'Ribonucleotide reductase and Pyruvate formate lyase.  Ribonucleotide reductase (RNR) and pyruvate formate lyase (PFL) are believed to have diverged from a common ancestor. They have a structurally similar ten-stranded alpha-beta barrel domain that hosts the active site, and are radical enzymes. RNRs are found in all organisms and provide the only mechanism by which nucleotides are converted to deoxynucleotides. RNRs are separated into three classes based on their metallocofactor usage. Class I RNRs use a diiron-tyrosyl radical while Class II RNRs use coenzyme B12 (adenosylcobalamin, AdoCbl). Class III RNRs use an FeS cluster and S-adenosylmethionine to generate a glycyl radical. PFL, an essential enzyme in anaerobic bacteria, catalyzes the conversion of pyruvate and CoA to acteylCoA and formate in a mechanism that uses a glycyl radical.','cd00576','RNR_PFL'),(955682,'gnl|CDD|181392',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK08332','PRK08332'),(955684,'gnl|CDD|188230',_binary 'ribonucleoside-diphosphate reductase, alpha chain.  This model represents a small clade of ribonucleoside-diphosphate reductase, alpha chains which are sufficiently divergent from the usual Class I RNR alpha chains (NrdE or NrdA, TIGR02506) as to warrant their own model. The genes from Thermus thermophilus, Dichelobacter and Salinibacter are adjacent to the usual RNR beta chain. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02510','NrdE-prime'),(955685,'gnl|CDD|223435',_binary 'DNA primase (bacterial type) [DNA replication, recombination, and repair].','COG0358','DnaG'),(955686,'gnl|CDD|273595',_binary 'DNA primase, catalytic core.  Members of this family are DNA primase, a ubiquitous bacteria protein. Most members of this family contain nearly two hundred additional residues C-terminal to the region represented here, but conservation between species is poor and the C-terminal region was not included in the seed alignment. This protein contains a CHC2 zinc finger (pfam01807) and a Toprim domain (pfam01751). [DNA metabolism, DNA replication, recombination, and repair].','TIGR01391','DNA_primase'),(955687,'gnl|CDD|173779',_binary 'TOPRIM_primases: The topoisomerase-primase (TORPIM) nucleotidyl transferase/hydrolase domain found in the active site regions of bacterial DnaG-type primases and their homologs. Primases synthesize RNA primers for the initiation of DNA replication. DnaG type primases are often closely associated with DNA helicases in primosome assemblies.  The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). This glutamate and two aspartates, cluster together to form a highly acid surface patch. The conserved glutamate may act as a general base in nucleotide polymerization by primases. The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function. The prototypical bacterial primase. Escherichia coli DnaG is a single subunit enzyme.','cd01029','TOPRIM_primases'),(955688,'gnl|CDD|214695',_binary 'topoisomerases, DnaG-type primases, OLD family nucleases and RecR proteins.','smart00493','TOPRIM'),(955689,'gnl|CDD|235551',_binary 'DNA primase; Validated.','PRK05667','dnaG'),(955690,'gnl|CDD|315749',_binary 'Toprim-like.  This is a family or Toprim-like proteins.','pfam13155','Toprim_2'),(955691,'gnl|CDD|173784',_binary 'TOPRIM_DnaG_primases: The topoisomerase-primase (TORPIM) nucleotidyl transferase/hydrolase domain found in the active site regions of proteins similar to Escherichia coli DnaG. Primases synthesize RNA primers for the initiation of DNA replication. DnaG type primases are often closely associated with DNA helicases in primosome assemblies.  The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). This glutamate and two aspartates, cluster together to form a highly acid surface patch. The conserved glutamate may act as a general base in nucleotide polymerization by primases. The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.  E. coli DnaG is a single subunit enzyme.','cd03364','TOPRIM_DnaG_primases'),(955692,'gnl|CDD|128681',_binary 'zinc finger.','smart00400','ZnF_CHCC'),(955694,'gnl|CDD|280055',_binary 'CHC2 zinc finger.  This domain is principally involved in DNA binding in DNA primases.','pfam01807','zf-CHC2'),(955699,'gnl|CDD|238484',_binary 'DnaB helicase C terminal domain. The hexameric helicase DnaB unwinds the DNA duplex at the  chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerization of the N-terminal domain has been observed and may occur during the enzymatic cycle. This C-terminal domain contains an ATP-binding site and is therefore probably the site of ATP hydrolysis.','cd00984','DnaB_C'),(955700,'gnl|CDD|238542',_binary 'GP4d_helicase is a homohexameric 5\'-3\' helicases. Helicases couple NTP hydrolysis to the unwinding of nucleic acid duplexes into their component strands.','cd01122','GP4d_helicase'),(955702,'gnl|CDD|168658',_binary 'replicative DNA helicase; Provisional.','PRK06749','PRK06749'),(955703,'gnl|CDD|223382',_binary 'Replicative DNA helicase [DNA replication, recombination, and repair].','COG0305','DnaB'),(955704,'gnl|CDD|223543',_binary 'RecA-superfamily ATPases implicated in signal transduction [Signal transduction mechanisms].','COG0467','RAD55'),(955705,'gnl|CDD|238540',_binary 'RecA-like NTPases. This family includes the NTP binding domain of F1 and V1 H+ATPases, DnaB and related helicases as well as bacterial RecA and related eukaryotic and archaeal recombinases. This group also includes bacterial conjugation proteins and related DNA transfer proteins involved in type II and type IV secretion.','cd01120','RecA-like_NTPases'),(955706,'gnl|CDD|309065',_binary 'DnaB-like helicase C terminal domain.  The hexameric helicase DnaB unwinds the DNA duplex at the Escherichia coli chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerisation of the N-terminal domain has been observed and may occur during the enzymatic cycle. This C-terminal domain contains an ATP-binding site and is therefore probably the site of ATP hydrolysis.','pfam03796','DnaB_C'),(955707,'gnl|CDD|274672',_binary 'phage replicative helicase, DnaB family, HK022 subfamily.  Members of this family are phage (or prophage-region) homologs of the bacterial homohexameric replicative helicase DnaB. Some phage may rely on host DnaB, while others encode their own verions. This model describes the largest phage-specific clade among the close homologs of DnaB, but there are, or course, other DnaB homologs from phage that fall outside the scope of this model. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR03600','phage_DnaB'),(955708,'gnl|CDD|315390',_binary 'PD-(D/E)XK nuclease superfamily.  Members of this family belong to the PD-(D/E)XK nuclease superfamily.','pfam12705','PDDEXK_1'),(955709,'gnl|CDD|225441',_binary 'RecB family exonuclease [DNA replication, recombination, and repair].','COG2887','COG2887'),(955710,'gnl|CDD|187768',_binary 'CRISPR/Cas system-associated protein Cas4.  CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) and associated Cas proteins comprise a system for heritable host defense by prokaryotic cells against phage and other foreign DNA; Cas4 is RecB-like nuclease with three-cysteine C-terminal cluster.','cd09637','Cas4_I-A_I-B_I-C_I-D_II-B'),(955711,'gnl|CDD|225320',_binary 'Predicted transcriptional regulator [General function prediction only].','COG2522','COG2522'),(955712,'gnl|CDD|214794',_binary 'SprT homologues.  Predicted to have roles in transcription elongation. Contains a conserved HExxH motif, indicating a metalloprotease function.','smart00731','SprT'),(955713,'gnl|CDD|313489',_binary 'SprT-like family.  This family represents a domain found in eukaryotes and prokaryotes. The domain contains a characteristic motif of the zinc metallopeptidases. This family includes the bacterial SprT protein.','pfam10263','SprT-like'),(955714,'gnl|CDD|238124',_binary 'Zinc-dependent metalloprotease. This super-family of metalloproteases contains two major branches, the astacin-like proteases and the adamalysin/reprolysin-like proteases. Both branches have wide phylogenetic distribution, and contain sub-families, which are involved in vertebrate development and disease.','cd00203','ZnMc'),(955719,'gnl|CDD|236696',_binary 'DNA cytosine methylase; Provisional.','PRK10458','PRK10458'),(955720,'gnl|CDD|309304',_binary 'Low affinity iron permease.','pfam04120','Iron_permease'),(955721,'gnl|CDD|227765',_binary 'Predicted small integral membrane protein [Function unknown].','COG5478','COG5478'),(955722,'gnl|CDD|307155',_binary 'NlpC/P60 family.  The function of this domain is unknown. It is found in several lipoproteins.','pfam00877','NLPC_P60'),(955723,'gnl|CDD|223862',_binary 'Cell wall-associated hydrolases (invasion-associated proteins) [Cell envelope biogenesis, outer membrane].','COG0791','Spr'),(955724,'gnl|CDD|237555',_binary 'invasion associated secreted endopeptidase; Provisional.','PRK13914','PRK13914'),(955725,'gnl|CDD|236773',_binary 'outer membrane lipoprotein; Provisional.','PRK10838','spr'),(955726,'gnl|CDD|309825',_binary 'Baseplate J-like protein.  The P2 bacteriophage J protein lies at the edge of the baseplate. This family also includes a number of bacterial homologs, which are thought to have been horizontally transferred.','pfam04865','Baseplate_J'),(955732,'gnl|CDD|224600',_binary 'D-alanyl-D-alanine carboxypeptidase [Cell envelope biogenesis, outer membrane].','COG1686','DacC'),(955734,'gnl|CDD|306726',_binary 'Helicase conserved C-terminal domain.  The Prosite family is restricted to DEAD/H helicases, whereas this domain family is found in a wide variety of helicases and helicase related proteins. It may be that this is not an autonomously folding unit, but an integral part of the helicase.','pfam00271','Helicase_C'),(955735,'gnl|CDD|238034',_binary 'Helicase superfamily c-terminal domain; associated with DEXDc-, DEAD-, and DEAH-box proteins, yeast initiation factor 4A, Ski2p, and Hepatitis C virus NS3 helicases; this domain is found in a wide variety of helicases and helicase related proteins; may not be an autonomously folding unit, but an integral part of the helicase; 4 helicase superfamilies at present according to the organization of their signature motifs; all helicases share the ability to unwind nucleic acid duplexes with a distinct directional polarity; they utilize the free energy from nucleoside triphosphate hydrolysis to fuel their translocation along DNA, unwinding the duplex in the process.','cd00079','HELICc'),(955737,'gnl|CDD|223627',_binary 'Superfamily II DNA/RNA helicases, SNF2 family [Transcription / DNA replication, recombination, and repair].','COG0553','HepA'),(955738,'gnl|CDD|197757',_binary 'helicase superfamily c-terminal domain.','smart00490','HELICc'),(955740,'gnl|CDD|317315',_binary 'Intein splicing domain.  Inteins are segments of protein which excise themselves from a precursor protein and mediate the rejoining of the remainder of the precursor (the extein). Most inteins consist of a splicing domain which is split into two segments by a homing endonuclease domain. This domain represents the splicing domain.','pfam14890','Intein_splicing'),(955741,'gnl|CDD|227325',_binary 'Ornithine/acetylornithine aminotransferase [Amino acid transport and metabolism].','COG4992','ArgD'),(955742,'gnl|CDD|99735',_binary 'Acetyl ornithine aminotransferase family. This family belongs to pyridoxal phosphate (PLP)-dependent aspartate aminotransferase superfamily (fold I). The major groups in this CD correspond to ornithine aminotransferase, acetylornithine aminotransferase, alanine-glyoxylate aminotransferase, dialkylglycine decarboxylase, 4-aminobutyrate aminotransferase, beta-alanine-pyruvate aminotransferase, adenosylmethionine-8-amino-7-oxononanoate aminotransferase, and glutamate-1-semialdehyde 2,1-aminomutase. All the enzymes belonging to this family act on basic amino acids and their derivatives are involved in transamination or decarboxylation.','cd00610','OAT_like'),(955744,'gnl|CDD|235687',_binary 'hypothetical protein; Provisional.','PRK06062','PRK06062'),(955745,'gnl|CDD|223238',_binary '4-aminobutyrate aminotransferase and related aminotransferases [Amino acid transport and metabolism].','COG0160','GabT'),(955746,'gnl|CDD|236437',_binary 'diaminobutyrate--2-oxoglutarate aminotransferase; Validated.','PRK09264','PRK09264'),(955747,'gnl|CDD|179270',_binary 'acetylornithine transaminase protein; Provisional.','PRK01278','argD'),(955748,'gnl|CDD|235599',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK05769','PRK05769'),(955749,'gnl|CDD|223239',_binary 'Adenosylmethionine-8-amino-7-oxononanoate aminotransferase [Coenzyme metabolism].','COG0161','BioA'),(955750,'gnl|CDD|274117',_binary 'diaminobutyrate--2-oxoglutarate aminotransferase.  Members of this family of class III pyridoxal-phosphate-dependent aminotransferases are diaminobutyrate--2-oxoglutarate aminotransferase (EC 2.6.1.76) that catalyze the first step in ectoine biosynthesis from L-aspartate beta-semialdehyde. This family is readily separated phylogenetically from enzymes with the same substrate and product but involved in other process such as siderophore (SP|Q9Z3R2) or 1,3-diaminopropane (SP|P44951) biosynthesis. The family TIGR00709 previously included both groups but has now been revised to exclude the ectoine biosynthesis proteins of this family. Ectoine is a compatible solute particularly effective in conferring salt tolerance. [Cellular processes, Adaptations to atypical conditions].','TIGR02407','Diaminobutyrate--2-oxoglu'),(955751,'gnl|CDD|273228',_binary 'transaminase, acetylornithine/succinylornithine family.  This family of proteins, for which ornithine aminotransferases form an outgroup, consists mostly of proteins designated acetylornithine aminotransferase. However, the two very closely related members from E. coli are assigned different enzymatic activities. One is acetylornithine aminotransferase (EC 2.6.1.11), ArgD, an enzyme of arginine biosynthesis, while another is succinylornithine aminotransferase, an enzyme of the arginine succinyltransferase pathway, an ammonia-generating pathway of arginine catabolism (See MEDLINE:98361920). Members of this family may also act on ornithine, like ornithine aminotransferase (EC 2.6.1.13) (see MEDLINE:90337349) and on succinyldiaminopimelate, like N-succinyldiaminopmelate-aminotransferase (EC 2.6.1.17, DapC, an enzyme of lysine biosynthesis) (see MEDLINE:99175097).','TIGR00707','Acetylornithine_aminotran'),(955752,'gnl|CDD|236153',_binary 'hypothetical protein; Validated.','PRK08116','PRK08116'),(955753,'gnl|CDD|224401',_binary 'DNA replication protein [DNA replication, recombination, and repair].','COG1484','DnaC'),(955754,'gnl|CDD|235871',_binary 'DNA replication protein DnaC; Validated.','PRK06835','PRK06835'),(955755,'gnl|CDD|180750',_binary 'hypothetical protein; Provisional.','PRK06921','PRK06921'),(955756,'gnl|CDD|181180',_binary 'DNA replication protein DnaC; Validated.','PRK07952','PRK07952'),(955757,'gnl|CDD|99707',_binary 'The AAA+ (ATPases Associated with a wide variety of cellular Activities) superfamily represents an ancient group of ATPases belonging to the ASCE (for additional strand, catalytic E) division of the P-loop NTPase fold. The ASCE division also includes ABC, RecA-like, VirD4-like, PilT-like, and SF1/2 helicases. Members of the AAA+ ATPases function as molecular chaperons, ATPase subunits of proteases, helicases, or nucleic-acid stimulated ATPases. The AAA+ proteins contain several distinct features in addition to the conserved alpha-beta-alpha core domain structure and the Walker A and B motifs of the P-loop NTPases.','cd00009','AAA'),(955758,'gnl|CDD|183482',_binary 'putative replication protein; Provisional.','PRK12377','PRK12377'),(955763,'gnl|CDD|235525',_binary 'replicative DNA helicase; Provisional.','PRK05595','PRK05595'),(955764,'gnl|CDD|236278',_binary 'replicative DNA helicase; Provisional.','PRK08506','PRK08506'),(955765,'gnl|CDD|236093',_binary 'replicative DNA helicase; Validated.','PRK07773','PRK07773'),(955766,'gnl|CDD|273206',_binary 'replicative DNA helicase.  This model describes the helicase DnaB, a homohexameric protein required for DNA replication. The homohexamer can form a ring around a single strand of DNA near a replication fork. An intein of > 400 residues is found at a conserved location in DnaB of Synechocystis PCC6803, Rhodothermus marinus (both experimentally confirmed), and Mycobacterium tuberculosis. The intein removes itself by a self-splicing reaction. The seed alignment contains inteins so that the model built from the seed alignment will model a low cost at common intein insertion sites. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00665','DnaB'),(955767,'gnl|CDD|180177',_binary 'replicative DNA helicase; Provisional.','PRK05636','PRK05636'),(955768,'gnl|CDD|181547',_binary 'replicative DNA helicase; Provisional.','PRK08760','PRK08760'),(955774,'gnl|CDD|222844',_binary 'putative DnaG-like primase.','PHA02031','PHA02031'),(955776,'gnl|CDD|237666',_binary 'chaperone protein DnaJ; Provisional.','PRK14296','PRK14296'),(955777,'gnl|CDD|237659',_binary 'chaperone protein DnaJ; Provisional.','PRK14287','PRK14287'),(955778,'gnl|CDD|237656',_binary 'chaperone protein DnaJ; Provisional.','PRK14280','PRK14280'),(955779,'gnl|CDD|182347',_binary 'curved DNA-binding protein CbpA; Provisional.','PRK10266','PRK10266'),(955780,'gnl|CDD|237667',_binary 'chaperone protein DnaJ; Provisional.','PRK14299','PRK14299'),(955781,'gnl|CDD|184612',_binary 'chaperone protein DnaJ; Provisional.','PRK14298','PRK14298'),(955782,'gnl|CDD|184604',_binary 'chaperone protein DnaJ; Provisional.','PRK14283','PRK14283'),(955783,'gnl|CDD|237655',_binary 'chaperone protein DnaJ; Provisional.','PRK14279','PRK14279'),(955784,'gnl|CDD|237662',_binary 'chaperone protein DnaJ; Provisional.','PRK14292','PRK14292'),(955785,'gnl|CDD|237653',_binary 'chaperone protein DnaJ; Provisional.','PRK14276','PRK14276'),(955786,'gnl|CDD|237663',_binary 'chaperone protein DnaJ; Provisional.','PRK14293','PRK14293'),(955787,'gnl|CDD|237661',_binary 'chaperone protein DnaJ; Provisional.','PRK14291','PRK14291'),(955788,'gnl|CDD|172778',_binary 'chaperone protein DnaJ; Provisional.','PRK14290','PRK14290'),(955789,'gnl|CDD|306689',_binary 'DnaJ domain.  DnaJ domains (J-domains) are associated with hsp70 heat-shock system and it is thought that this domain mediates the interaction. DnaJ-domain is therefore part of a chaperone (protein folding) system. The T-antigens, although not in Prosite are confirmed as DnaJ containing domains from literature.','pfam00226','DnaJ'),(955790,'gnl|CDD|306878',_binary 'RF-1 domain.  This domain is found in peptide chain release factors such as RF-1 and RF-2, and a number of smaller proteins of unknown function. This domain contains the peptidyl-tRNA hydrolase activity. The domain contains a highly conserved motif GGQ, where the glutamine is thought to coordinate the water that mediates the hydrolysis.','pfam00472','RF-1'),(955791,'gnl|CDD|223294',_binary 'Protein chain release factor A [Translation, ribosomal structure and biogenesis].','COG0216','PrfA'),(955792,'gnl|CDD|234801',_binary 'peptide chain release factor 1; Validated.','PRK00591','prfA'),(955793,'gnl|CDD|234799',_binary 'peptide chain release factor 2; Validated.','PRK00578','prfB'),(955794,'gnl|CDD|224107',_binary 'Protein chain release factor B [Translation, ribosomal structure and biogenesis].','COG1186','PrfB'),(955795,'gnl|CDD|129130',_binary 'peptide chain release factor 1.  This model describes peptide chain release factor 1 (PrfA, RF-1), and excludes the related peptide chain release factor 2 (PrfB, RF-2). RF-1 helps recognize and terminate translation at UAA and UAG stop codons. The mitochondrial release factors are prfA-like, although not included above the trusted cutoff for this model. RF-1 does not have a translational frameshift. [Protein synthesis, Translation factors].','TIGR00019','Peptide_chain_release_fac'),(955796,'gnl|CDD|272858',_binary 'peptide chain release factor 2.  In many but not all taxa, there is a conserved real translational frameshift at a TGA codon. RF-2 helps terminate translation at TGA codons and can therefore regulate its own production by readthrough when RF-2 is insufficient. There is a Pfam model called \'RF-1\' for the superfamily of RF-1, RF-2, mitochondrial, RF-H, etc. [Protein synthesis, Translation factors].','TIGR00020','Peptide_chain_release_fac'),(955797,'gnl|CDD|213768',_binary 'putative peptide chain release factor H.  Members of this protein family are bacterial proteins homologous to peptide chain release factors 1 (RF-1, product of the prfA gene), and 2 (RF-2, product of the prfB gene). The member from Escherichia coli K-12, designated prfH, appears to be a pseudogene. This class I release factor is always found as the downstream gene of a two-gene operon. [Protein synthesis, Translation factors].','TIGR03072','release_prfH'),(955798,'gnl|CDD|75726',_binary 'peptide chain release factor 2; Provisional.','PRK06746','PRK06746'),(955799,'gnl|CDD|235997',_binary 'peptide chain release factor 2; Provisional.','PRK07342','PRK07342'),(955800,'gnl|CDD|235520',_binary 'peptide chain release factor 2; Provisional.','PRK05589','PRK05589'),(955801,'gnl|CDD|181271',_binary 'peptide chain release factor-like protein; Reviewed.','PRK08179','prfH'),(955802,'gnl|CDD|136970',_binary 'peptide chain release factor 2; Provisional.','PRK08787','PRK08787'),(955803,'gnl|CDD|238483',_binary 'RecA is a  bacterial enzyme which has roles in homologous recombination, DNA repair, and the induction of the SOS response.  RecA couples ATP hydrolysis to DNA strand exchange.','cd00983','recA'),(955804,'gnl|CDD|162659',_binary 'protein RecA.  This model describes orthologs of the recA protein. RecA promotes hybridization of homolgous regions of DNA. A segment of ssDNA can be hybridized to another ssDNA region, or to a dsDNA region. ATP is hydrolyzed in the process. Part of the SOS respones, it is regulated by LexA via autocatalytic cleavage. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02012','Protein_RecA'),(955805,'gnl|CDD|306628',_binary 'recA bacterial DNA recombination protein.  RecA is a DNA-dependent ATPase and functions in DNA repair systems. RecA protein catalyzes an ATP-dependent DNA strand-exchange reaction that is the central step in the repair of dsDNA breaks by homologous recombination.','pfam00154','RecA'),(955806,'gnl|CDD|236476',_binary 'recombinase A; Provisional.','PRK09354','recA'),(955807,'gnl|CDD|223544',_binary 'RecA/RadA recombinase [DNA replication, recombination, and repair].','COG0468','RecA'),(955808,'gnl|CDD|77219',_binary 'DNA recombination protein RecA; Reviewed.','PRK09519','recA'),(955809,'gnl|CDD|238687',_binary 'RecA is a  bacterial enzyme which has roles in homologous recombination, DNA repair, and the induction of the SOS response.  RecA couples ATP hydrolysis to DNA strand exchange. While prokaryotes have a single RecA protein, eukaryotes have multiple RecA homologs such as Rad51, DMC1 and Rad55/57.  Archaea have the RecA-like homologs radA and radB.','cd01393','recA_like'),(955810,'gnl|CDD|310450',_binary 'Endodeoxyribonuclease RusA.  This family consists of several bacterial and phage Holliday junction resolvase (RusA) like proteins. The RusA protein of Escherichia coli is an endonuclease that can resolve Holliday intermediates and correct the defects in genetic recombination and DNA repair associated with inactivation of RuvAB or RuvC.','pfam05866','RusA'),(955811,'gnl|CDD|223660',_binary 'DNA polymerase III, alpha subunit [DNA replication, recombination, and repair].','COG0587','DnaE'),(955812,'gnl|CDD|235554',_binary 'DNA polymerase III subunit alpha; Validated.','PRK05673','dnaE'),(955813,'gnl|CDD|273161',_binary 'DNA-directed DNA polymerase III (polc).  All proteins in this family for which functions are known are DNA polymerases. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00594','DNA_polymerase_III_subuni'),(955814,'gnl|CDD|235868',_binary 'DNA polymerase III DnaE; Reviewed.','PRK06826','dnaE'),(955815,'gnl|CDD|168927',_binary 'DNA polymerase III subunit alpha; Validated.','PRK07374','dnaE'),(955816,'gnl|CDD|180749',_binary 'DNA polymerase III DnaE; Reviewed.','PRK06920','dnaE'),(955817,'gnl|CDD|181933',_binary 'DNA polymerase III subunit alpha; Reviewed.','PRK09532','PRK09532'),(955818,'gnl|CDD|235553',_binary 'error-prone DNA polymerase; Validated.','PRK05672','dnaE2'),(955819,'gnl|CDD|135648',_binary 'DNA polymerase III DnaE; Validated.','PRK05898','dnaE'),(955820,'gnl|CDD|311600',_binary 'Bacterial DNA polymerase III alpha subunit.','pfam07733','DNA_pol3_alpha'),(955821,'gnl|CDD|235944',_binary 'DNA polymerase III DnaE; Validated.','PRK07135','dnaE'),(955823,'gnl|CDD|180917',_binary 'DNA polymerase III DnaE; Reviewed.','PRK07279','dnaE'),(955825,'gnl|CDD|213997',_binary 'Polymerase and Histidinol Phosphatase domain of alpha-subunit of bacterial polymerase III DnaE3.  PolIIIAs that contain an N-terminal PHP domain have been classified into four basic groups based on genome composition, phylogenetic, and domain structural analysis: polC, dnaE1, dnaE2, and dnaE3. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. DNA polymerase III holoenzyme is one of the five eubacterial DNA polymerases that is responsible for the replication of the DNA duplex. The alpha subunit of DNA polymerase III core enzyme catalyzes the reaction for polymerizing both DNA strands. The PolIIIA PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination, and like other PHP structures, the PolIIIA PHP exhibits a distorted (beta/alpha) 7 barrel and coordinates up to 3 metals. Initially, it was proposed that PHP region might be involved in pyrophosphate hydrolysis, but such an activity has not been found. It has been shown that the PHP of PolIIIA has a trinuclear metal complex and is capable of proofreading activity. Bacterial genome replication and DNA repair mechanisms is related to the GC content of its genomes. There is a correlation between GC content variations and the dimeric combinations of PolIIIA subunits. Eubacteria can be grouped into different GC variable groups: the full-spectrum or dnaE1 group, the high-GC or dnaE2-dnaE1 group, and the low GC or polC-dnaE3 group.','cd12113','PHP_PolIIIA_DnaE3'),(955826,'gnl|CDD|308452',_binary 'PHP domain.  The PHP (Polymerase and Histidinol Phosphatase) domain is a putative phosphoesterase domain.','pfam02811','PHP'),(955827,'gnl|CDD|213988',_binary 'Polymerase and Histidinol Phosphatase domain of alpha-subunit of bacterial polymerase III DnaE1.  PolIIIAs that contain an N-terminal PHP domain have been classified into four basic groups based on genome composition, phylogenetic, and domain structural analysis: polC, dnaE1, dnaE2, and dnaE3. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. DNA polymerase III holoenzyme is one of the five eubacterial DNA polymerases that are responsible for the replication of the DNA duplex. PolIIIA core enzyme catalyzes the reaction for polymerizing both DNA strands. dnaE1 is the longest compared to dnaE2 and dnaE3. A unique motif was also identified in dnaE1 and dnaE3 genes.','cd07433','PHP_PolIIIA_DnaE1'),(955828,'gnl|CDD|213986',_binary 'Polymerase and Histidinol Phosphatase domain of alpha-subunit of bacterial polymerase III.  PolIIIAs that contain an N-terminal PHP domain have been classified into four basic groups based on genome composition, phylogenetic, and domain structural analysis: polC, dnaE1, dnaE2, and dnaE3. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. DNA polymerase III holoenzyme is one of the five eubacterial DNA polymerases that is responsible for the replication of the DNA duplex. The alpha subunit of DNA polymerase III core enzyme catalyzes the reaction for polymerizing both DNA strands. The PolIIIA PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination, and like other PHP structures, exhibits a distorted (beta/alpha) 7 barrel and coordinates up to 3 metals. Initially, it was proposed that PHP region might be involved in pyrophosphate hydrolysis, but such activity has not been found. It has been shown that the PHP domain of PolIIIA has a trinuclear metal complex and is capable of proofreading activity.','cd07431','PHP_PolIIIA'),(955829,'gnl|CDD|317037',_binary 'Helix-hairpin-helix motif.  The HHH domain is a short DNA-binding domain.','pfam14579','HHH_6'),(955830,'gnl|CDD|213990',_binary 'Polymerase and Histidinol Phosphatase domain of alpha-subunit of bacterial polymerase III at PolC gene.  DNA polymerase III alphas (PolIIIAs) that contain a PHP domain have been classified into four basic groups based on phylogenetic and domain structural analyses: polC, dnaE1, dnaE2, and dnaE3. The PolC group is distinct from the other three and is clustered together. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. DNA polymerase III holoenzyme is one of the five eubacterial DNA polymerases that are responsible for the replication of the DNA duplex. The alpha subunit of DNA polymerase III core enzyme catalyzes the reaction for polymerizing both DNA strands. PolC PHP is located in different location compare to dnaE1, 2, and 3. The PHP domain has four conserved sequence motifs and and contains an invariant histidine that is involved in metal ion coordination.The PHP domain of PolC is structurally homologous to other members of the PHP family that have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel. PHP domains found in dnaEs of thermophilic origin exhibit 3\'-5\' exonuclease activity. In contrast, PolC PHP lacks detectable nuclease activity.','cd07435','PHP_PolIIIA_POLC'),(955831,'gnl|CDD|236003',_binary 'DNA polymerase III subunit alpha; Reviewed.','PRK07373','PRK07373'),(955832,'gnl|CDD|197753',_binary 'DNA polymerase alpha chain like domain.  DNA polymerase alpha chain like domain, incl. family of hypothetical proteins.','smart00481','POLIIIAc'),(955833,'gnl|CDD|213989',_binary 'Polymerase and Histidinol Phosphatase domain of alpha-subunit of bacterial polymerase III at DnaE2 gene.  PolIIIA DnaE2 plays a role in SOS mutagenesis/translesion synthesis and has dominant effects in determining GC variability in the bacterial genome. PolIIIAs that contain an N-terminal PHP domain have been classified into four basic groups based on genome composition, phylogenetic, and domain structural analysis: polC, dnaE1, dnaE2, and dnaE3. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. DNA polymerase III holoenzyme is one of the five eubacterial DNA polymerases that are responsible for the replication of the DNA duplex. PolIIIA core enzyme catalyzes the reaction for polymerizing both DNA strands. PolC PHP is located in a different location compared to dnaE1, 2, and 3. dnaE1 is the longest compared to dnaE2 and dnaE3. A unique motif was also identified in dnaE1 and dnaE3 genes. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. PHP domains found in DnaEs of thermophilic origin exhibit 3\'-5\' exonuclease activity.','cd07434','PHP_PolIIIA_DnaE2'),(955836,'gnl|CDD|213993',_binary 'Polymerase and Histidinol Phosphatase domain of Histidinol phosphate phosphatase (HisPPase) AMP bound.  The PHP domain of this HisPPase family has an unknown function. It has a second domain inserted in the middle that binds adenosine 5-monophosphate (AMP). The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. HisPPase catalyzes the eighth step of histidine biosynthesis, in which L-histidinol phosphate undergoes dephosphorylation to give histidinol. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. The PHP domain of HisPPase is structurally homologous to the other members of the PHP family that have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd07438','PHP_HisPPase_AMP'),(955839,'gnl|CDD|239931',_binary 'DnaE_OBF: A subfamily of OB folds corresponding to the C-terminal OB-fold nucleic acid binding domain of Thermus aquaticus and Escherichia coli type C replicative DNA polymerase III alpha subunit (DnaE). The DNA polymerase holoenzyme of E. coli contains two copies of this replicative polymerase, each of which copies a different DNA strand. This group also contains Bacillus subtilis DnaE. Replication in B. subtilis and Staphylococcus aureus requires two different type C polymerases, polC and DnaE, both of which are thought to be included in the DNA polymerase holoenzyme. At the B. subtilis replication fork, polC appears to be involved in leading strand synthesis and DnaE in lagging strand synthesis.','cd04485','DnaE_OBF'),(955840,'gnl|CDD|213987',_binary 'Polymerase and Histidinol Phosphatase domain of Histidinol phosphate phosphatase.  HisPPase catalyzes the eighth step of histidine biosynthesis, in which L-histidinol phosphate undergoes dephosphorylation to produce histidinol. HisPPase can be classified into two types: the bifunctional HisPPase found in proteobacteria that belongs to the DDDD superfamily and the monofunctional Bacillus subtilis type that is a member of the PHP family. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. The PHP domain of HisPPase is structurally homologous to other members of the PHP family that have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd07432','PHP_HisPPase'),(955841,'gnl|CDD|225087',_binary 'DNA polymerase III, alpha subunit (gram-positive type) [DNA replication, recombination, and repair].','COG2176','PolC'),(955842,'gnl|CDD|213985',_binary 'Polymerase and Histidinol Phosphatase domain.  The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. PHP in polymerases has trinuclear zinc/magnesium dependent proofreading activity. It has also been shown that the PHP domain functions in DNA repair. The PHP structures have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd07309','PHP'),(955843,'gnl|CDD|213991',_binary 'Polymerase and Histidinol Phosphatase domain of bacterial polymerase X.  The bacterial/archaeal X-family DNA polymerases (PolXs) have a PHP domain at their C-terminus. The bacterial/archaeal PolX core domain and PHP domain interact with each other and together are involved in metal dependent 3\'-5\' exonuclease activity. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. PolX is found in all kingdoms, however bacterial PolXs have a completely different domain structure from eukaryotic PolXs. Bacterial PolX has an extended conformation in contrast to the common closed \'right hand\' conformation for DNA polymerases. This extended conformation is stabilized by the PHP domain. The PHP domain of PolX is structurally homologous to other members of the PHP family that has a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd07436','PHP_PolX'),(955844,'gnl|CDD|310383',_binary 'TROVE domain.  This presumed domain is found in TEP1 and Ro60 proteins, that are RNA-binding components of Telomerase, Ro and Vault RNPs. This domain has been named TROVE, (after Telomerase, Ro and Vault). This domain is probably RNA-binding.','pfam05731','TROVE'),(955845,'gnl|CDD|133017',_binary 'RfbF is a putative dTDP-rhamnosyl transferase.  Shigella flexneri RfbF protein is a putative dTDP-rhamnosyl transferase. dTDP rhamnosyl  transferases of Shigella flexneri  add rhamnose sugars to N-acetyl-glucosamine in the O-antigen tetrasaccharide repeat. Lipopolysaccharide O antigens are important virulence determinants for many bacteria. The variations of sugar composition, the sequence of the sugars and the linkages in the O antigen provide structural diversity of the O antigen.','cd02526','GT2_RfbF_like'),(955846,'gnl|CDD|133029',_binary 'Subfamily of Glycosyltransferase Family GT2 of unknown function.  GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd04186','GT_2_like_c'),(955847,'gnl|CDD|307591',_binary 'N-acetylmuramoyl-L-alanine amidase.  This family includes zinc amidases that have N-acetylmuramoyl-L-alanine amidase activity EC:3.5.1.28. This enzyme domain cleaves the amide bond between N-acetylmuramoyl and L-amino acids in bacterial cell walls (preferentially: D-lactyl-L-Ala). The structure is known for the bacteriophage T7 structure and shows that two of the conserved histidines are zinc binding.','pfam01510','Amidase_2'),(955848,'gnl|CDD|214760',_binary 'Ami_2 domain.','smart00644','Ami_2'),(955849,'gnl|CDD|133475',_binary 'Peptidoglycan recognition proteins (PGRPs) are pattern recognition receptors that bind, and in certain cases, hydrolyze peptidoglycans (PGNs) of bacterial cell walls. PGRPs have been divided into three classes: short PGRPs (PGRP-S), that are small (20 kDa) extracellular proteins; intermediate PGRPs (PGRP-I) that are 40-45 kDa and are predicted to be transmembrane proteins; and long PGRPs (PGRP-L), up to 90 kDa, which may be either intracellular or transmembrane. Several structures of PGRPs are known in insects and mammals, some bound with substrates like Muramyl Tripeptide (MTP) or Tracheal Cytotoxin (TCT). The substrate binding site is conserved in PGRP-LCx, PGRP-LE, and PGRP-Ialpha proteins. This family includes Zn-dependent N-Acetylmuramoyl-L-alanine Amidase, EC:3.5.1.28. This enzyme cleaves the amide bond between N-acetylmuramoyl and L-amino acids, preferentially D-lactyl-L-Ala, in bacterial cell walls. The structure for the bacteriophage T7 lysozyme shows that two of the conserved histidines and a cysteine are zinc binding residues. Site-directed mutagenesis of T7 lysozyme indicates that two conserved residues, a Tyr and a Lys, are important for amidase activity.','cd06583','PGRP'),(955850,'gnl|CDD|307341',_binary 'SPFH domain / Band 7 family.  This family has been called SPFH, Band 7 or PHB domain. Recent phylogenetic analysis has shown this domain to be a slipin or Stomatin-like integral membrane domain conserved from protozoa to mammals.','pfam01145','Band_7'),(955851,'gnl|CDD|259799',_binary 'Prohibitin family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model characterizes proteins similar to prohibitin (a lipid raft-associated integral membrane protein). Individual proteins of the SPFH (band 7) domain superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. These microdomains, in addition to being stable scaffolds, may also be dynamic units with their own regulatory functions. Prohibitin is a mitochondrial inner-membrane protein which may act as a chaperone for the stabilization of mitochondrial proteins. Human prohibitin forms a hetero-oligomeric complex with Bap-37 (prohibitin 2, an SPFH domain carrying homolog). This complex may protect non-assembled membrane proteins against proteolysis by the m-AAA protease. Prohibitin and Bap-37 yeast homologs have been implicated in yeast longevity and in the maintenance of mitochondrial morphology.','cd03401','SPFH_prohibitin'),(955852,'gnl|CDD|259802',_binary 'High frequency of lysogenization K (HflK) family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model characterizes proteins similar to prokaryotic HflK (High frequency of lysogenization K). Although many members of the SPFH (or band 7) superfamily are lipid raft associated, prokaryote plasma membranes lack cholesterol and are unlikely to have lipid raft domains. Individual proteins of this SPFH domain superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Escherichia coli HflK is an integral membrane protein which may localize to the plasma membrane. HflK associates with another SPFH superfamily member (HflC) to form an HflKC complex. HflKC interacts with FtsH in a large complex termed the FtsH holo-enzyme. FtsH is an AAA ATP-dependent protease which exerts progressive proteolysis against membrane-embedded and soluble substrate proteins. HflKC can modulate the activity of FtsH. HflKC plays a role in the decision between lysogenic and lytic cycle growth during lambda phage infection.','cd03404','SPFH_HflK'),(955853,'gnl|CDD|223407',_binary 'Membrane protease subunits, stomatin/prohibitin homologs [Posttranslational modification, protein turnover, chaperones].','COG0330','HflC'),(955854,'gnl|CDD|259803',_binary 'High frequency of lysogenization C (HflC) family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model characterizes proteins similar to prokaryotic HflC (High frequency of lysogenization C). Although many members of the SPFH (or band 7) superfamily are lipid raft associated, prokaryote plasma membranes lack cholesterol and are unlikely to have lipid raft domains. Individual proteins of this SPFH domain superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Escherichia coli HflC is an integral membrane protein which may localize to the plasma membrane. HflC associates with another SPFH superfamily member (HflK) to form an HflKC complex. HflKC interacts with FtsH in a large complex termed the FtsH holo-enzyme. FtsH is an AAA ATP-dependent protease which exerts progressive proteolysis against membrane-embedded and soluble substrate proteins. HflKC can modulate the activity of FtsH. HflKC plays a role in the decision between lysogenic and lytic cycle growth during lambda phage infection.','cd03405','SPFH_HflC'),(955858,'gnl|CDD|238035',_binary 'Hedgehog/Intein domain, found in Hedgehog proteins as well as proteins which contain inteins and undergo protein splicing (e.g. DnaB, RIR1-2, GyrA and Pol). In protein splicing an intervening polypeptide sequence - the intein - is excised from a protein, and the flanking polypeptide sequences - the exteins - are joined by a peptide bond. In addition to the autocatalytic splicing domain, many inteins contain an inserted endonuclease domain, which plays a role in spreading inteins. Hedgehog proteins are a major class of intercellular signaling molecules, which control inductive interactions during animal development. The mature signaling forms of hedgehog proteins are the N-terminal fragments, which are covalently linked to cholesterol at their C-termini. This modification is the result of an autoprocessing step catalyzed by the C-terminal fragments, which are aligned here.','cd00081','Hint'),(955862,'gnl|CDD|270688',_binary 'Catalytic domain of CMGC family Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The CMGC family consists of Cyclin-Dependent protein Kinases (CDKs), Mitogen-activated protein kinases (MAPKs) such as Extracellular signal-regulated kinase (ERKs), c-Jun N-terminal kinases (JNKs), and p38, and other kinases. CDKs belong to a large subfamily of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. MAPKs serve as important mediators of cellular responses to extracellular signals. They control critical cellular functions including differentiation, proliferation, migration, and apoptosis. They are also implicated in the pathogenesis of many diseases including multiple types of cancer, stroke, diabetes, and chronic inflammation. Other members of the CMGC family include casein kinase 2 (CK2), Dual-specificity tYrosine-phosphorylated and -Regulated Kinase (DYRK), Glycogen Synthase Kinase 3 (GSK3), among many others. The CMGC family is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05118','STKc_CMGC'),(955865,'gnl|CDD|271034',_binary 'Catalytic subunit (alpha) of the Serine/Threonine Kinase, Casein Kinase 2.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. CK2 is a tetrameric protein with two catalytic (alpha) and two regulatory (beta) subunits. It is constitutively active and ubiquitously expressed, and is found in the cytoplasm, nucleus, as well as in the plasma membrane. It phosphorylates a wide variety of substrates including gylcogen synthase, cell cycle proteins, nuclear proteins (e.g. DNA topoisomerase II), and ion channels (e.g. ENaC), among others. It may be considered a master kinase controlling the activity or lifespan of many other kinases and exerting its effect over cell fate, gene expression, protein synthesis and degradation, and viral infection. CK2 is implicated in every stage of the cell cycle and is required for cell cycle progression. It plays crucial roles in cell differentiation, proliferation, and survival, and is thus implicated in cancer. CK2 is not an oncogene by itself but elevated CK2 levels create an environment that enhances the survival of tumor cells. The CK2 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14132','STKc_CK2_alpha'),(955868,'gnl|CDD|270875',_binary 'Pseudokinase domain of the peptidoglycan biosynthetic protein MviN.  The pseudokinase domain shows similarity to protein kinases but lacks crucial residues for catalytic activity. This family is composed of the mycobacterial protein MviN and similar proteins. MviN is an integral membrane protein that is essential for growth and is required for cell wall integrity and peptidogylcan (PG) biosynthesis. It comprises of 14 predicted transmembrane (TM) helices at the N-terminus, followed by an intracellular pseudokinase domain linked through a single TM helix to a carbohydrate binding extracellular domain. Phosphorylation of the MviN pseudokinase domain by the PG-sensitive serine/threonine protein kinase PknB recruits a forkhead associated (FHA) domain protein FhaA, which modulates local PG synthesis at cell poles and the septum. The MviN pseudokinase forms a canonical receptor kinase dimer.','cd13973','PK_MviN-like'),(955869,'gnl|CDD|270856',_binary 'Pseudokinase domain of STE20-related kinase adapter protein.  The pseudokinase domain shows similarity to protein kinases but lacks crucial residues for catalytic activity. STRAD forms a complex with the scaffolding protein MO25, and the serine/threonine kinase (STK), LKB1, resulting in the activation of the kinase. In the complex, LKB1 phosphorylates and activates adenosine monophosphate-activated protein kinases (AMPKs), which regulate cell energy metabolism and cell polarity. LKB1 is a tumor suppressor linked to the rare inherited disease, Peutz-Jeghers syndrome, which is characterized by a predisposition to benign polyps and hyperpigmentation of the buccal mucosa. There are two forms of STRAD, alpha and beta, that complex with LKB1 and MO25. The structure of STRAD-alpha is available and shows that this protein binds ATP, has an ordered activation loop, and adopts a closed conformation typical of fully active protein kinases. It does not possess activity due to nonconservative substitutions of essential catalytic residues. ATP binding enhances the affinity of STRAD for MO25. The conformation of STRAD-alpha stabilized through ATP and MO25 may be needed to activate LKB1. The STRAD subfamily is part of a larger superfamily that includes the catalytic domains of STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd08216','PK_STRAD'),(955870,'gnl|CDD|270905',_binary 'Catalytic domain of AMP-activated protein kinase-like Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The AMPK-like subfamily is composed of AMPK, MARK, BRSK, NUAK, MELK, SNRK, TSSK, and SIK, among others. LKB1 serves as a master upstream kinase that activates AMPK and most AMPK-like kinases. AMPK, also called SNF1 (sucrose non-fermenting1) in yeasts and SnRK1 (SNF1-related kinase1) in plants, is a heterotrimeric enzyme composed of a catalytic alpha subunit and two regulatory subunits, beta and gamma. It is a stress-activated kinase that serves as master regulator of glucose and lipid metabolism by monitoring carbon and energy supplies, via sensing the cell\'s AMP:ATP ratio. MARKs phosphorylate tau and related microtubule-associated proteins (MAPs), and regulates microtubule-based intracellular transport. They are involved in embryogenesis, epithelial cell polarization, cell signaling, and neuronal differentiation. BRSKs play important roles in establishing neuronal polarity. TSSK proteins are almost exclusively expressed postmeiotically in the testis and play important roles in spermatogenesis and/or spermiogenesis. The AMPK-like subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14003','STKc_AMPK-like'),(955871,'gnl|CDD|270733',_binary 'Catalytic domain of the Serine/Threonine Kinase, Phosphoinositide-dependent kinase 1.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. PDK1 carries an N-terminal catalytic domain and a C-terminal pleckstrin homology (PH) domain that binds phosphoinositides. It phosphorylates the activation loop of AGC kinases that are regulated by PI3K such as PKB, SGK, and PKC, among others, and is crucial for their activation. Thus, it contributes in regulating many processes including metabolism, growth, proliferation, and survival. PDK1 also has the ability to autophosphorylate and is constitutively active in mammalian cells. It is essential for normal embryo development and is important in regulating cell volume. The PDK1 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05581','STKc_PDK1'),(955872,'gnl|CDD|270921',_binary 'Catalytic domain of the Serine/Threonine Kinase, Cell Division Cycle 7 kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Cdc7 kinase (or Hsk1 in fission yeast) is a critical regulator in the initiation of DNA replication. It forms a complex with a Dbf4-related regulatory subunit, a cyclin-like molecule that activates the kinase in late G1 phase, and is also referred to as Dbf4-dependent kinase (DDK). Its main targets are mini-chromosome maintenance (MCM) proteins. Cdc7 kinase may also have additional roles in meiosis, checkpoint responses, the maintenance and repair of chromosome structures, and cancer progression. The Cdc7 kinase subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14019','STKc_Cdc7'),(955873,'gnl|CDD|270731',_binary 'Catalytic domain of Microtubule-associated serine/threonine (MAST) kinase-like proteins.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This subfamily includes MAST kinases, MAST-like (MASTL) kinases (also called greatwall kinase or Gwl), and fungal kinases with similarity to Saccharomyces cerevisiae Rim15 and Schizosaccharomyces pombe cek1. MAST kinases contain an N-terminal domain of unknown function, a central catalytic domain, and a C-terminal PDZ domain that mediates protein-protein interactions. MASTL kinases carry only a catalytic domain which contains a long insert relative to other kinases. The fungal kinases in this subfamily harbor other domains in addition to a central catalytic domain, which like in MASTL, also contains an insert relative to MAST kinases. Rim15 contains a C-terminal signal receiver (REC) domain while cek1 contains an N-terminal PAS domain. MAST kinases are cytoskeletal associated kinases of unknown function that are also expressed at neuromuscular junctions and postsynaptic densities. MASTL/Gwl is involved in the regulation of mitotic entry, mRNA stabilization, and DNA checkpoint recovery. The fungal proteins Rim15 and cek1 are involved in the regulation of meiosis and mitosis, respectively. The MAST-like kinase subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05579','STKc_MAST_like'),(955874,'gnl|CDD|271009',_binary 'Catalytic kinase domain, first repeat, of the Giant Serine/Threonine Kinase Obscurin.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Obscurin, approximately 800 kDa in size, is one of three giant proteins expressed in vetebrate striated muscle, together with titin and nebulin. It is a multidomain protein composed of tandem adhesion and signaling domains, including 49 immunoglobulin (Ig) and 2 fibronectin type III (FN3) domains at the N-terminus followed by a more complex region containing more Ig domains, a conserved SH3 domain near a RhoGEF and PH domains, non-modular regions, as well as IQ and phosphorylation motifs. The obscurin gene also encode two kinase domains, which are not expressed as part of the 800 kDa protein, but as a smaller, alternatively spliced product present mainly in the heart muscle, also called obscurin-MLCK. Obscurin is localized at the peripheries of Z-disks and M-lines, where it is able to communicate with the surrounding myoplasm. It interacts with diverse proteins including sAnk1, myosin, titin, and MyBP-C. It may act as a scaffold for the assembly of elements of the contractile apparatus. The obscurin subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14107','STKc_obscurin_rpt1'),(955875,'gnl|CDD|270948',_binary 'Catalytic domain, repeat 2, of the Serine/Threonine kinase, eukaryotic translation Initiation Factor 2-Alpha Kinase 4 or General Control Non-derepressible-2.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. GCN2 (or EIF2AK4) is activated by amino acid or serum starvation and UV irradiation. It induces GCN4, a transcriptional activator of amino acid biosynthetic genes, leading to increased production of amino acids under amino acid-deficient conditions. In serum-starved cells, GCN2 activation induces translation of the stress-responsive transcription factor ATF4, while under UV stress, GCN2 triggers transcriptional rescue via NF-kB signaling. GCN2 contains an N-terminal RWD, a degenerate kinase-like (repeat 1), the catalytic kinase (repeat 2), a histidyl-tRNA synthetase (HisRS)-like, and a C-terminal ribosome-binding and dimerization (RB/DD) domains. Its kinase domain is activated via conformational changes as a result of the binding of uncharged tRNA to the HisRS-like domain. EIF2AKs phosphorylate the alpha subunit of eIF-2, resulting in the overall downregulation of protein synthesis. The GCN2 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14046','STKc_EIF2AK4_GCN2_rpt2'),(955876,'gnl|CDD|270827',_binary 'Catalytic domain of Cyclin-Dependent protein Kinase Like Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This subfamily is composed of CDKL1-5 and similar proteins. Some CDKLs, like CDKL1 and CDKL3, may be implicated in transformation and others, like CDKL3 and CDKL5, are associated with mental retardation when impaired. CDKL2 plays a role in learning and memory. CDKs belong to a large family of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. The CDKL subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd07833','STKc_CDKL'),(955877,'gnl|CDD|270762',_binary 'Catalytic domain of fungal Rim15-like Protein Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Members of this group include Saccharomyces cerevisiae Rim15, Schizosaccharomyces pombe cek1, and similar fungal proteins. They contain a central catalytic domain, which contains an insert relative to MAST kinases. In addition, Rim15 contains a C-terminal signal receiver (REC) domain while cek1 contains an N-terminal PAS domain. Rim15 (or Rim15p) functions as a regulator of meiosis. It acts as a downstream effector of PKA and regulates entry into stationary phase (G0). Thus, it plays a crucial role in regulating yeast proliferation, differentiation, and aging. Cek1 may facilitate progression of mitotic anaphase. The Rim15-like subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd05611','STKc_Rim15_like'),(955878,'gnl|CDD|270952',_binary 'Catalytic domain of the Dual-specificity protein kinase, Myt1.  Dual-specificity PKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine as well as tyrosine residues on protein substrates. Myt1 is a cytoplasmic cell cycle checkpoint kinase that can keep the cyclin-dependent kinase CDK1 in an inactive state through phosphorylation of N-terminal thr (T14) and tyr (Y15) residues, leading to the delay of meiosis I entry. Meiotic progression is ensured by a two-step inhibition and downregulation of Myt1 by CDK1/XRINGO and p90Rsk during oocyte maturation. In addition, Myt1 targets cyclin B1/B2 and is essential for Golgi and ER assembly during telophase. In Drosophila, Myt1 may be a downstream target of Notch during eye development. The Myt1 subfamily is part of a larger superfamily that includes the catalytic domains of other protein serine/threonine PKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14050','PKc_Myt1'),(955879,'gnl|CDD|270895',_binary 'Catalytic domain of Fungal Pat1-like Serine/Threonine kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This subfamily is composed of Schizosaccharomyces pombe Pat1 (also called Ran1), Saccharomyces cerevisiae VHS1 and KSP1, and similar fungal STKs. Pat1 blocks Mei2, an RNA-binding protein which is indispensable in the initiation of meiosis. Pat1 is inactivated and Mei2 activated, which initiates meiosis, under nutrient-deprived conditions through a signaling cascade involving Ste11. Meiosis induced by Pat1 inactivation may show different characteristics than normal meiosis including aberrant positioning of centromeres. VHS1 was identified in a screen for suppressors of cell cycle arrest at the G1/S transition, while KSP1 may be involved in regulating PRP20, which is required for mRNA export and maintenance of nuclear structure. The Pat1-like subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd13993','STKc_Pat1_like'),(955880,'gnl|CDD|270910',_binary 'Catalytic domain of the Serine/Threonine kinases, Liver Kinase B1, Calmodulin Dependent Protein Kinase Kinase, and similar proteins.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Both LKB1 and CaMKKs can phosphorylate and activate AMP-activated protein kinase (AMPK). LKB1, also called STK11, serves as a master upstream kinase that activates AMPK and most AMPK-like kinases. LKB1 and AMPK are part of an energy-sensing pathway that links cell energy to metabolism and cell growth. They play critical roles in the establishment and maintenance of cell polarity, cell proliferation, cytoskeletal organization, as well as T-cell metabolism, including T-cell development, homeostasis, and effector function. CaMKKs are upstream kinases of the CaM kinase cascade that phosphorylate and activate CaMKI and CamKIV. They may also phosphorylate other substrates including PKB and AMPK. Vertebrates contain two CaMKKs, CaMKK1 (or alpha) and CaMKK2 (or beta). CaMKK1 is involved in the regulation of glucose uptake in skeletal muscles. CaMKK2 is involved in regulating energy balance, glucose metabolism, adiposity, hematopoiesis, inflammation, and cancer. The LKB1/CaMKK subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14008','STKc_LKB1_CaMKK'),(955883,'gnl|CDD|273629',_binary 'intein N-terminal splicing region.  This model is based on interated search results, starting with a curated collection of intein N-terminal splicing regions from InBase, the New England Biolabs Intein Database, as presented on its web site. It is designed to recognize inteins but not the related region of the sonic hedgehog protein.','TIGR01445','Probable_translation_init'),(955885,'gnl|CDD|307037',_binary 'Adenylosuccinate synthetase.','pfam00709','Adenylsucc_synt'),(955886,'gnl|CDD|197875',_binary 'Adenylosuccinate synthetase.  Adenylosuccinate synthetase plays an important role in purine biosynthesis, by catalyzing the GTP-dependent conversion of IMP and aspartic acid to AMP. Adenylosuccinate synthetase has been characterized from various sources ranging from Escherichia coli (gene purA) to vertebrate tissues. In vertebrates, two isozymes are present - one involved in purine biosynthesis and the other in the purine nucleotide cycle. The crystal structure of adenylosuccinate synthetase from E. coli reveals that the dominant structural element of each monomer of the homodimer is a central beta-sheet of 10 strands. The first nine strands of the sheet are mutually parallel with right-handed crossover connections between the strands. The 10th strand is antiparallel with respect to the first nine strands. In addition, the enzyme has two antiparallel beta-sheets, comprised of two strands and three strands each, 11 alpha-helices and two short 3/10-helices. Further, it has been suggested that the similarities in the GTP-binding domains of the synthetase and the p21ras protein are an example of convergent evolution of two distinct families of GTP-binding proteins. Structures of adenylosuccinate synthetase from Triticum aestivum and Arabidopsis thaliana when compared with the known structures from E. coli reveals that the overall fold is very similar to that of the E. coli protein.','smart00788','Adenylsucc_synt'),(955887,'gnl|CDD|223182',_binary 'Adenylosuccinate synthase [Nucleotide transport and metabolism].','COG0104','PurA'),(955888,'gnl|CDD|272949',_binary 'adenylosuccinate synthase.  Alternate name IMP--aspartate ligase. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR00184','Adenylosuccinate_syntheta'),(955889,'gnl|CDD|234904',_binary 'adenylosuccinate synthetase; Provisional.','PRK01117','PRK01117'),(955890,'gnl|CDD|184326',_binary 'adenylosuccinate synthetase; Provisional.','PRK13788','PRK13788'),(955891,'gnl|CDD|237506',_binary 'adenylosuccinate synthetase; Provisional.','PRK13785','PRK13785'),(955892,'gnl|CDD|237505',_binary 'adenylosuccinate synthetase; Provisional.','PRK13783','PRK13783'),(955893,'gnl|CDD|172322',_binary 'adenylosuccinate synthetase; Provisional.','PRK13784','PRK13784'),(955894,'gnl|CDD|240376',_binary 'adenylosuccinate synthetase; Provisional.','PTZ00350','PTZ00350'),(955895,'gnl|CDD|184325',_binary 'adenylosuccinate synthetase; Provisional.','PRK13786','PRK13786'),(955896,'gnl|CDD|178129',_binary 'adenylosuccinate synthase.','PLN02513','PLN02513'),(955897,'gnl|CDD|239382',_binary 'Adenylosuccinate synthetase (AdSS) catalyzes the first step in the de novo biosynthesis of AMP. IMP and L-aspartate are conjugated in a two-step reaction accompanied by the hydrolysis of GTP to GDP in the presence of Mg2+. In the first step, the r-phosphate group of GTP is transferred to the 6-oxygen atom of IMP. An aspartate then displaces this 6-phosphate group to form the product adenylosuccinate. Because of its critical role in purine biosynthesis, AdSS is a target of antibiotics, herbicides and antitumor drugs.','cd03108','AdSS'),(955899,'gnl|CDD|172324',_binary 'adenylosuccinate synthetase; Provisional.','PRK13787','PRK13787'),(955900,'gnl|CDD|179812',_binary 'adenylosuccinate synthetase; Provisional.','PRK04293','PRK04293'),(955901,'gnl|CDD|173544',_binary 'adenylosuccinate synthetase; Provisional.','PTZ00351','PTZ00351'),(955902,'gnl|CDD|313370',_binary 'Predicted nucleotidyltransferase.  Members of this family of bacterial proteins catalyze the transfer of nucleotide residues from nucleoside diphosphates or triphosphates into dimer or polymer forms.','pfam10127','Nuc-transf'),(955904,'gnl|CDD|222905',_binary 'hypothetical protein; Provisional.','PHA02603','nrdC.11'),(955905,'gnl|CDD|316281',_binary 'Helix-turn-helix domain.  Members of this family contains a DNA-binding helix-turn-helix domain.','pfam13744','HTH_37'),(955906,'gnl|CDD|197775',_binary 'Helix-turn-helix XRE-family like proteins.','smart00530','HTH_XRE'),(955907,'gnl|CDD|238045',_binary 'Helix-turn-helix XRE-family like proteins. Prokaryotic DNA binding proteins belonging to the xenobiotic response element family of transcriptional regulators.','cd00093','HTH_XRE'),(955908,'gnl|CDD|316117',_binary 'Helix-turn-helix domain.  This domain is a helix-turn-helix domain that probably binds to DNA.','pfam13560','HTH_31'),(955909,'gnl|CDD|226147',_binary 'Predicted transcriptional regulator with C-terminal CBS domains [Transcription].','COG3620','COG3620'),(955910,'gnl|CDD|307507',_binary 'Helix-turn-helix.  This large family of DNA binding helix-turn helix proteins includes Cro and CI. Within Neisseria gonorrhoeae NGO_0477, the full protein fold incorporates a helix-turn-helix motif, but the function of this member is unlikely to be that of a DNA-binding regulator, the function of most other members, so is not necessarily characteristic of the whole family.','pfam01381','HTH_3'),(955911,'gnl|CDD|223876',_binary 'Sec-independent protein secretion pathway component TatC [Intracellular trafficking and secretion].','COG0805','TatC'),(955912,'gnl|CDD|319261',_binary 'pNOB8 ParB-like N-terminal domain, plasmid partitioning system protein domain.  archaeal pNOB8 ParB acts in a plasmid partitioning system made up of 3 parts: AspA, ParA motor protein, and ParB, which links ParA to the protein-DNA superhelix. As demonstrated in Sulfolobus, AspA spreads along DNA, which allows ParB binding, and links to the Walker-motif containing ParA motor protein. The Sulfolobus ParB C-terminal domain resembles eukaryotic segregation protein CenpA, and other histones. This family is related to the N-terminal domain of ParB (Spo0J in Bacillus subtilis), a DNA-binding component of the prokaryotic parABS partitioning system and related proteins.','cd16404','pNOB8_ParB_N_like'),(955913,'gnl|CDD|214678',_binary 'ParB-like nuclease domain.  Plasmid RK2 ParB preferentially cleaves single-stranded DNA. ParB also nicks supercoiled plasmid DNA preferably at sites with potential single-stranded character, like AT-rich regions and sequences that can form cruciform structures. ParB also exhibits 5-->3 exonuclease activity.','smart00470','ParB'),(955918,'gnl|CDD|197609',_binary 'Lysin motif.','smart00257','LysM'),(955919,'gnl|CDD|212030',_binary 'Lysine Motif is a small domain involved in binding peptidoglycan.  LysM, a small globular domain with approximately 40 amino acids, is a widespread protein module involved in binding peptidoglycan in bacteria and chitin in eukaryotes. The domain was originally identified in enzymes that degrade bacterial cell walls, but proteins involved in many other biological functions also contain this domain. It has been reported that the LysM domain functions as a signal for specific plant-bacteria recognition in bacterial pathogenesis. Many of these enzymes are modular and are composed of catalytic units linked to one or several repeats of LysM domains. LysM domains are found in bacteria and eukaryotes.','cd00118','LysM'),(955920,'gnl|CDD|307567',_binary 'LysM domain.  The LysM (lysin motif) domain is about 40 residues long. It is found in a variety of enzymes involved in bacterial cell wall degradation. This domain may have a general peptidoglycan binding function. The structure of this domain is known.','pfam01476','LysM'),(955921,'gnl|CDD|315517',_binary 'Oxygenase domain of the 2OGFeDO superfamily.  A double-stranded beta helix (DSBH) fold domain of the 2-oxoglutarate (2OG)-Fe(II)-dependent dioxygenase (2OGFeDO) superfamily found in various eukaryotes, bacteria and bacteriophages. Members of this family catalyze nucleic acid modifications, such as thymidine hydroxylation during base J synthesis in kinetoplastids, and the conversion of 5 methyl-cytosine (5-mC) to 5-hydroxymethyl-cytosine (hmC), or further oxidation to 5-formylcytosine (5fC) and 5-carboxylcytosine (5caC). Metazoan TET proteins contain a cysteine-rich region inserted into the core of the DSBH fold. Vertebrate TET proteins are oncogenes that are mutated in various myeloid cancers. Fungal and algal versions of this family are linked to a predicted transposase and show lineage-specific expansions.','pfam12851','Tet_JBP'),(955922,'gnl|CDD|227697',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG5410','COG5410'),(955923,'gnl|CDD|308715',_binary 'Terminase-like family.  This family represents a group of terminase proteins.','pfam03237','Terminase_6'),(955926,'gnl|CDD|225567',_binary 'N-acetyl-anhydromuramyl-L-alanine amidase [Cell envelope biogenesis, outer membrane].','COG3023','ampD'),(955927,'gnl|CDD|223989',_binary 'DNA or RNA helicases of superfamily II [Transcription / DNA replication, recombination, and repair].','COG1061','SSL2'),(955928,'gnl|CDD|309815',_binary 'Type III restriction enzyme, res subunit.','pfam04851','ResIII'),(955929,'gnl|CDD|226581',_binary 'Type I site-specific restriction-modification system, R (restriction) subunit and related helicases [Defense mechanisms].','COG4096','HsdR'),(955931,'gnl|CDD|214692',_binary 'DEAD-like helicases superfamily.','smart00487','DEXDc'),(955934,'gnl|CDD|236912',_binary 'type I restriction enzyme EcoKI subunit R; Provisional.','PRK11448','hsdR'),(955935,'gnl|CDD|223587',_binary 'Superfamily II DNA and RNA helicases [DNA replication, recombination, and repair / Transcription / Translation, ribosomal structure and biogenesis].','COG0513','SrmB'),(955951,'gnl|CDD|223887',_binary 'Holliday junction resolvasome, endonuclease subunit [DNA replication, recombination, and repair].','COG0817','RuvC'),(955952,'gnl|CDD|238294',_binary 'Holliday junction resolvases (HJRs) are endonucleases that specifically resolve Holliday junction DNA intermediates during homologous recombination.  HJR\'s occur in archaea, bacteria, and in the mitochondria of certain fungi, however this CD includes only the bacterial and mitochondrial HJR\'s.  These are referred to as the RuvC family of Holliday junction resolvases, RuvC being the E.coli HJR.  RuvC and its orthologs are homodimers and are structurely similar to RNase H and Hsp70.','cd00529','RuvC_resolvase'),(955953,'gnl|CDD|311201',_binary 'Protein of unknown function (DUF1360).  This family consists of several bacterial proteins of around 115 residues in length. Members of this family are found in Bacillus species and Streptomyces coelicolor, the function of the family is unknown.','pfam07098','DUF1360'),(955954,'gnl|CDD|310569',_binary 'Domain of unknown function (DUF932).  Family of prokaryotic proteins with unknown function. Contains a number of highly conserved polar residues that could suggest an enzymatic activity.','pfam06067','DUF932'),(955955,'gnl|CDD|274510',_binary 'phage/plasmid-like protein TIGR03299.  Members of this uncharacterized protein family are found in various Mycobacterium phage genomes, in Streptomyces coelicolor plasmid SCP1, and in bacterial genomes near various markers that suggest lateral gene transfer. The function is unknown. [Mobile and extrachromosomal element functions, Other].','TIGR03299','LGT_TIGR03299'),(955966,'gnl|CDD|238607',_binary 'Protein Kinase C Interacting protein related (PKCI): PKCI and related proteins belong to the ubiquitous HIT family of hydrolases that act on alpha-phosphates of ribonucleotides. The members of this subgroup have a conserved HxHxHxx motif (x is a hydrophobic residue) that is a signature for this family. No enzymatic activity has been reported however, for PKCI and its related members.','cd01276','PKCI_related'),(955968,'gnl|CDD|238263',_binary 'HIT family: HIT (Histidine triad) proteins, named for a motif related to the sequence HxHxH/Qxx (x, a hydrophobic amino acid), are a superfamily of nucleotide hydrolases and transferases, which act on the alpha-phosphate of ribonucleotides. On the basis of sequence, substrate specificity, structure, evolution and mechanism, HIT proteins are classified in the literacture into three major branches: the Hint branch, which consists of adenosine 5\' -monophosphoramide hydrolases, the Fhit branch, that consists of diadenosine polyphosphate hydrolases, and the GalT branch consisting of specific nucloside monophosphate transferases. Further sequence analysis reveals several new closely related, yet uncharacterized subgroups.','cd00468','HIT_like'),(955991,'gnl|CDD|306669',_binary 'Aminotransferase class-III.','pfam00202','Aminotran_3'),(955992,'gnl|CDD|235056',_binary 'acetylornithine aminotransferase; Provisional.','PRK02627','PRK02627'),(955994,'gnl|CDD|177748',_binary 'acetylornithine transaminase.','PLN00144','PLN00144'),(955995,'gnl|CDD|180994',_binary 'putative aminotransferase; Validated.','PRK07480','PRK07480'),(955996,'gnl|CDD|180426',_binary 'hypothetical protein; Provisional.','PRK06148','PRK06148'),(955997,'gnl|CDD|180401',_binary 'aminotransferase; Provisional.','PRK06105','PRK06105'),(956094,'gnl|CDD|315955',_binary 'HNH endonuclease.  This is a zinc-binding loop of Fold group 7 as found in endo-deoxy-ribonucleases and HNH nucleases.','pfam13392','HNH_3'),(956100,'gnl|CDD|224137',_binary 'Predicted glycosyltransferases [General function prediction only].','COG1216','COG1216'),(956102,'gnl|CDD|133028',_binary 'Subfamily of Glycosyltransferase Family GT2 of unknown function.  GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd04185','GT_2_like_b'),(956103,'gnl|CDD|313356',_binary 'Glycosyltransferase like family 2.  Members of this family of prokaryotic proteins include putative glucosyltransferase, which are involved in bacterial capsule biosynthesis.','pfam10111','Glyco_tranf_2_2'),(956166,'gnl|CDD|260122',_binary '4-Hydroxybenzoate polyprenyltransferase.  4-Hydroxybenzoate polyprenyltransferase, also known as Coq2, catalyzes the prenylation of p-hydroxybenzoate with an all-trans polyprenyl group, an important step in ubiquinone (CoQ) biosynthesis. Prenyltransferases (PTs) catalyze the regioselective transfer of prenyl moieties onto a wide variety of substrates and play an important role in many biosynthetic pathways.','cd13959','PT_UbiA_COQ2'),(956169,'gnl|CDD|319258',_binary 'ParB N-terminal-like domain, some attached to C-terminal S-adenosylmethionine-dependent methyltransferase domain.  This family represents domains related to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, fused to a variety of C-terminal domains, including S-adenosylmethionine-dependent methyltransferase-like domains. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16401','ParB_N_like_MT'),(956178,'gnl|CDD|225720',_binary 'Predicted chitinase [General function prediction only].','COG3179','COG3179'),(956179,'gnl|CDD|307564',_binary 'Putative peptidoglycan binding domain.  This domain is composed of three alpha helices. This domain is found at the N or C terminus of a variety of enzymes involved in bacterial cell wall degradation. This domain may have a general peptidoglycan binding function. This family is found N-terminal to the catalytic domain of matrixins. The domain is found to bind peptidoglycan experimentally.','pfam01471','PG_binding_1'),(956180,'gnl|CDD|225943',_binary 'Putative peptidoglycan-binding domain-containing protein [Cell envelope biogenesis, outer membrane].','COG3409','COG3409'),(956181,'gnl|CDD|238199',_binary 'Glycoside hydrolase family 19 chitinase domain. Chitinases are enzymes that catalyze the hydrolysis of the beta-1,4-N-acetyl-D-glucosamine linkages in chitin polymers. Family 19 chitinases are found primarily in plants (classes I, III, and IV), but some are found in bacteria. Class I and II chitinases are similar in their catalytic domains. Class I chitinases have an N-terminal cysteine-rich, chitin-binding domain which is separated from the catalytic domain by a proline and glycine-rich hinge region. Class II chitinases lack both the chitin-binding domain and the hinge region. Class IV chitinases are similar to class I chitinases but they are smaller in size due to certain deletions. Despite any significant sequence homology with lysozymes, structural analysis reveals that family 19 chitinases, together with family 46 chitosanases, are similar to several lysozymes including those from T4-phage and from goose. The structures reveal that the different enzyme groups arose from a common ancestor glycohydrolase antecedent to the procaryotic/eucaryotic divergence.','cd00325','chitinase_glyco_hydro_19'),(956189,'gnl|CDD|227605',_binary 'Phage-related minor tail protein [Function unknown].','COG5280','COG5280'),(956195,'gnl|CDD|316165',_binary 'AAA domain.  This family includes a wide variety of AAA domains including some that have lost essential nucleotide binding residues in the P-loop.','pfam13614','AAA_31'),(956197,'gnl|CDD|139909',_binary 'putative crown gall tumor protein VirC1; Provisional.','PRK13849','PRK13849'),(956198,'gnl|CDD|238994',_binary 'MRP (Multiple Resistance and pH adaptation) is a homologue of the Fer4_NifH superfamily. Like the other members of the superfamily, MRP contains a ATP-binding domain at the N-termini. It is found in bacteria as a membrane-spanning protein and functions as a Na+/H+ antiporter.','cd02037','MRP-like'),(956199,'gnl|CDD|307676',_binary 'CobQ/CobB/MinD/ParA nucleotide binding domain.  This family consists of various cobyrinic acid a,c-diamide synthases. These include CbiA and CbiP from S.typhimurium, and CobQ from R. capsulatus. These amidases catalyze amidations to various side chains of hydrogenobyrinic acid or cobyrinic acid a,c-diamide in the biosynthesis of cobalamin (vitamin B12) from uroporphyrinogen III. Vitamin B12 is an important cofactor and an essential nutrient for many plants and animals and is primarily produced by bacteria. The family also contains dethiobiotin synthetases as well as the plasmid partitioning proteins of the MinD/ParA family.','pfam01656','CbiA'),(956200,'gnl|CDD|223563',_binary 'ATPases involved in chromosome partitioning [Cell division and chromosome partitioning].','COG0489','Mrp'),(956201,'gnl|CDD|238993',_binary 'Bacterial cell division requires the formation of a septum at mid-cell. The site is determined by the min operon products MinC, MinD and MinE. MinC is a nonspecific inhibitor of the septum protein FtsZ. MinE is the supressor of MinC. MinD plays a pivotal role, selecting the mid-cell over other sites through the activation and regulation of MinC and MinE. MinD is a membrane-associated ATPase, related to nitrogenase iron protein. More distantly related proteins include flagellar biosynthesis proteins and ParA chromosome partitioning proteins. MinD is a monomer.','cd02036','MinD'),(956202,'gnl|CDD|238941',_binary 'The Fer4_NifH superfamily contains a variety of proteins which share a common ATP-binding domain. Functionally, proteins in this superfamily use the energy from hydrolysis of NTP to transfer electron or ion.','cd01983','Fer4_NifH'),(956203,'gnl|CDD|238995',_binary 'FleN is a member of the Fer4_NifH superfamily. It shares the common function as an ATPase, with the ATP-binding domain at the N-terminus. In Pseudomonas aeruginosa, FleN gene is involved in regulating the number of flagella and chemotactic motility by influencing FleQ activity.','cd02038','FleN-like'),(956204,'gnl|CDD|225447',_binary 'Septum formation inhibitor-activating ATPase [Cell division and chromosome partitioning].','COG2894','MinD'),(956205,'gnl|CDD|274549',_binary 'cellulose synthase operon protein YhjQ.  Members of this family are the YhjQ protein, found immediately upsteam of bacterial cellulose synthase (bcs) genes in a broad range of bacteria, including both copies of the bcs locus in Klebsiella pneumoniae. In several species it is seen clearly as part of the bcs operon. It is identified as a probable component of the bacterial cellulose metabolic process not only by gene location, but also by partial phylogenetic profiling, or Haft-Selengut algorithm (), based on a bacterial cellulose biosynthesis genome property profile. Cellulose plays an important role in biofilm formation and structural integrity in some bacteria. Mutants in yhjQ in Escherichia coli, show altered morphology an growth, but the function of YhjQ has not yet been determined. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR03371','cellulose_yhjQ'),(956206,'gnl|CDD|274585',_binary 'plasmid partitioning protein RepA.  Members of this family are the RepA (or ParA) protein involved in replicon partitioning. All known examples occur in bacterial species with two or more replicons, on a plasmid or the smaller chromosome. Note that an apparent exception may be seen as a pseudomolecule from assembly of an incompletely sequenced genome. Members of this family belong to a larger family that also includes the enzyme cobyrinic acid a,c-diamide synthase, but assignment of that name to members of this family would be in error. [Mobile and extrachromosomal element functions, Plasmid functions].','TIGR03453','partition_RepA'),(956207,'gnl|CDD|238992',_binary 'ArsA ATPase functionas as an efflux pump located on the inner membrane of the cell. This ATP-driven oxyanion pump catalyzes the extrusion of arsenite, antimonite and arsenate. Maintenance of a low intracellular concentration of oxyanion produces resistance to the toxic agents. The pump is composed of two subunits, the catalytic ArsA subunit and the membrane subunit ArsB, which are encoded by arsA and arsB genes respectively. Arsenic efflux in bacteria is catalyzed by either ArsB alone or by ArsAB complex. The ATP-coupled pump, however, is more efficient. ArsA is composed of two homologous halves, A1 and A2, connected by a short linker sequence.','cd02035','ArsA'),(956208,'gnl|CDD|182756',_binary 'cell division inhibitor MinD; Provisional.','PRK10818','PRK10818'),(956209,'gnl|CDD|214385',_binary 'septum-site determining protein; Validated.','CHL00175','minD'),(956210,'gnl|CDD|223531',_binary 'Antiactivator of flagellar biosynthesis FleN, an ATPase [Cell motility].','COG0455','flhG'),(956211,'gnl|CDD|223082',_binary 'Predicted ATPase involved in chromosome partitioning [Cell division and chromosome partitioning].','COG0003','ArsA'),(956212,'gnl|CDD|238991',_binary 'The accessory protein CooC, which contains a nucleotide-binding domain (P-loop) near the N-terminus, participates in the maturation of the nickel center of carbon monoxide dehydrogenase (CODH). CODH from Rhodospirillum rubrum catalyzes the reversible oxidation of CO to CO2. CODH contains a nickel-iron-sulfur cluster (C-center) and an iron-sulfur cluster (B-center). CO oxidation occurs at the C-center. Three accessory proteins encoded by cooCTJ genes are involved in nickel incorporation into a nickel site. CooC functions as a nickel insertase that mobilizes nickel to apoCODH using energy released from ATP hydrolysis. CooC is a homodimer and has NTPase activities. Mutation at the P-loop abolishs its function.','cd02034','CooC'),(956213,'gnl|CDD|131023',_binary 'septum site-determining protein MinD.  This model describes the bacterial and chloroplast form of MinD, a multifunctional cell division protein that guides correct placement of the septum. The homologous archaeal MinD proteins, with many archaeal genomes having two or more forms, are described by a separate model. [Cellular processes, Cell division].','TIGR01968','Septum_site-determining_p'),(956214,'gnl|CDD|226166',_binary 'CO dehydrogenase maturation factor [Cell division and chromosome partitioning].','COG3640','CooC'),(956215,'gnl|CDD|313763',_binary 'NUBPL iron-transfer P-loop NTPase.  This family contains ATPases involved in plasmid partitioning. It also contains the cytosolic Fe-S cluster assembling factor NBP35 which is required for biogenesis and export of both ribosomal subunits.','pfam10609','ParA'),(956216,'gnl|CDD|312607',_binary 'ATPase MipZ.  MipZ is an ATPase that forms a complex with the chromosome partitioning protein ParB near the chromosomal origin of replication. It is responsible for the temporal and spatial regulation of FtsZ ring formation.','pfam09140','MipZ'),(956217,'gnl|CDD|131024',_binary 'cell division ATPase MinD, archaeal.  This model represents the archaeal branch of the MinD family. MinD, a weak ATPase, works in bacteria with MinC as a generalized cell division inhibitor and, through interaction with MinE, prevents septum placement inappropriate sites. Often several members of this family are found in archaeal genomes, and the function is uncharacterized. More distantly related proteins ParA chromosome partitioning proteins. The exact roles of the various archaeal MinD homologs are unknown.','TIGR01969','Uncharacterized_ATP-bindi'),(956254,'gnl|CDD|277320',_binary 'Escherichia coli YfcE and related proteins, metallophosphatase domain.  YfcE is a manganase-dependent metallophosphatase, found in bacteria and archaea, that cleaves bis-p-nitrophenyl phosphate, thymidine 5\'-monophosphate-p-nitrophenyl ester, and p-nitrophenyl phosphorylcholine, but is unable to hydrolyze 2\',3 \' or 3\',5\' cyclic nucleic phosphodiesters, and various phosphomonoesters, including p-nitrophenyl phosphate. This family also includes the Bacilus subtilis YsnB and Methanococcus jannaschii MJ0936 proteins.  This domain family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd00841','MPP_YfcE'),(956256,'gnl|CDD|315516',_binary 'Calcineurin-like phosphoesterase superfamily domain.  Members of this family are part of the Calcineurin-like phosphoesterase superfamily.','pfam12850','Metallophos_2'),(956258,'gnl|CDD|306625',_binary 'Calcineurin-like phosphoesterase.  This family includes a diverse range of phosphoesterases, including protein phosphoserine phosphatases, nucleotidases, sphingomyelin phosphodiesterases and 2\'-3\' cAMP phosphodiesterases as well as nucleases such as bacterial SbcD or yeast MRE11. The most conserved regions in this superfamily centre around the metal chelating residues.','pfam00149','Metallophos'),(956265,'gnl|CDD|131255',_binary 'Glutaredoxin-like protein.  This family of glutaredoxin-like proteins is limited to the Actinobacteria and contains the conserved CxxC motif.','TIGR02200','conserved_hypothetical_pr'),(956266,'gnl|CDD|239017',_binary 'Glutaredoxin (GRX) family; composed of GRX, approximately 10 kDa in size, and proteins containing a GRX or GRX-like domain. GRX is a glutathione (GSH) dependent reductase, catalyzing the disulfide reduction of target proteins such as ribonucleotide reductase. It contains a redox active CXXC motif in a TRX fold and uses a similar dithiol mechanism employed by TRXs for intramolecular disulfide bond reduction of protein substrates. Unlike TRX, GRX has preference for mixed GSH disulfide substrates, in which it uses a monothiol mechanism where only the N-terminal cysteine is required. The flow of reducing equivalents in the GRX system goes from NADPH -> GSH reductase -> GSH -> GRX -> protein substrates. By altering the redox state of target proteins, GRX is involved in many cellular functions including DNA synthesis, signal transduction and the defense against oxidative stress. Different classes are known including human GRX1 and GRX2, as well as E. coli GRX1 and GRX3, which are members of this family. E. coli GRX2, however, is a 24-kDa protein that belongs to the GSH S-transferase (GST) family.','cd02066','GRX_family'),(956267,'gnl|CDD|239327',_binary 'Glutaredoxin (GRX) family, PRX5 hybrid subfamily; composed of hybrid proteins containing peroxiredoxin (PRX) and GRX domains, which is found in some pathogenic bacteria and cyanobacteria. PRXs are thiol-specific antioxidant (TSA) proteins that confer a protective antioxidant role in cells through their peroxidase activity in which hydrogen peroxide, peroxynitrate, and organic hydroperoxides are reduced and detoxified using reducing equivalents derived from either thioredoxin, glutathione, trypanothione and AhpF. GRX is a glutathione (GSH) dependent reductase, catalyzing the disulfide reduction of target proteins. PRX-GRX hybrid proteins from Haemophilus influenza and Neisseria meningitis exhibit GSH-dependent peroxidase activity. The flow of reducing equivalents in the catalytic cycle of the hybrid protein goes from NADPH -> GSH reductase -> GSH -> GRX domain of hybrid -> PRX domain of hybrid -> peroxide substrate.','cd03029','GRX_hybridPRX5'),(956268,'gnl|CDD|239510',_binary 'Glutaredoxin (GRX) family, GRX bacterial class 1 and 3 (b_1_3)-like subfamily; composed of bacterial GRXs, approximately 10 kDa in size, and proteins containing a GRX or GRX-like domain. GRX is a glutathione (GSH) dependent reductase, catalyzing the disulfide reduction of target proteins such as ribonucleotide reductase. It contains a redox active CXXC motif in a TRX fold and uses a similar dithiol mechanism employed by TRXs for intramolecular disulfide bond reduction of protein substrates. Unlike TRX, GRX has preference for mixed GSH disulfide substrates, in which it uses a monothiol mechanism where only the N-terminal cysteine is required. The flow of reducing equivalents in the GRX system goes from NADPH -> GSH reductase -> GSH -> GRX -> protein substrates. By altering the redox state of target proteins, GRX is involved in many cellular functions including DNA synthesis, signal transduction and the defense against oxidative stress. Different classes are known including  E. coli GRX1 and GRX3, which are members of this subfamily.','cd03418','GRX_GRXb_1_3_like'),(956269,'gnl|CDD|274017',_binary 'Glutaredoxin, GrxC family.  Glutaredoxins are thioltransferases (disulfide reductases) which utilize glutathione and NADPH as cofactors. Oxidized glutathione is regenerated by glutathione reductase. Together these components compose the glutathione system. Glutaredoxins utilize the CXXC motif common to thioredoxins and are involved in multiple cellular processes including protection from redox stress, reduction of critical enzymes such as ribonucleotide reductase and the generation of reduced sulfur for iron sulfur cluster formation. Glutaredoxins are capable of reduction of mixed disulfides of glutathione as well as the formation of glutathione mixed disulfides. This family of glutaredoxins includes the E. coli protein GrxC (Grx3) which appears to have a secondary role in reducing ribonucleotide reductase (in the absence of GrxA) possibly indicating a role in the reduction of other protein disulfides. [Energy metabolism, Electron transport].','TIGR02181','GRX_bact'),(956270,'gnl|CDD|131245',_binary 'Glutaredoxin-family domain.  This C-terminal domain with homology to glutaredoxin is fused to an N-terminal peroxiredoxin-like domain.','TIGR02190','probable_peroxiredoxin/gl'),(956271,'gnl|CDD|239275',_binary 'Arsenate Reductase (ArsC) family; composed of TRX-fold arsenic reductases and similar proteins including the transcriptional regulator, Spx. ArsC catalyzes the reduction of arsenate [As(V)] to arsenite [As(III)], using reducing equivalents derived from glutathione (GSH) via glutaredoxin (GRX), through a single catalytic cysteine. This family of predominantly bacterial enzymes is unrelated to two other families of arsenate reductases which show similarity to low-molecular-weight acid phosphatases and phosphotyrosyl phosphatases. Spx is a general regulator that exerts negative and positive control over transcription initiation by binding to the C-terminal domain of the alpha subunit of RNA polymerase.','cd02977','ArsC_family'),(956273,'gnl|CDD|238319',_binary 'Glutathione S-transferase (GST) family, N-terminal domain; a large, diverse group of cytosolic dimeric proteins involved in cellular detoxification by catalyzing the conjugation of  glutathione (GSH) with a wide range of endogenous and xenobiotic alkylating agents, including carcinogens, therapeutic drugs, environmental toxins and products of oxidative stress. In addition, GSTs also show GSH peroxidase activity and are involved in the synthesis of prostaglandins and leukotrienes. This family, also referred to as soluble GSTs, is the largest family of GSH transferases and is only distantly related to the mitochondrial GSTs (GSTK subfamily, a member of the DsbA family). Soluble GSTs bear no structural similarity to microsomal GSTs (MAPEG family) and display additional activities unique to their group, such as catalyzing thiolysis, reduction  and isomerization of certain compounds. The GST fold contains an N-terminal TRX-fold domain and a C-terminal alpha helical domain, with an active site located in a cleft between the two domains. Based on sequence similarity, different classes of GSTs have been identified, which display varying tissue distribution, substrate specificities and additional specific activities. In humans, GSTs display polymorphisms which may influence individual susceptibility to diseases such as cancer, arthritis, allergy and sclerosis. Some GST family members with non-GST functions include glutaredoxin 2, the CLIC subfamily of anion channels, prion protein Ure2p, crystallins, metaxin 2 and stringent starvation protein A.','cd00570','GST_N_family'),(956304,'gnl|CDD|315411',_binary 'Helix-turn-helix domain.  This domain is a DNA-binding helix-turn-helix domain.','pfam12728','HTH_17'),(956306,'gnl|CDD|133390',_binary 'Helix-Turn-Helix DNA binding domain of truncated MerR-like proteins.  Proteins in this family mostly have a truncated helix-turn-helix (HTH) MerR-like domain. They lack a portion of the C-terminal region, called Wing 2 and the long dimerization helix that is typically present in MerR-like proteins. These truncated domains are found in response regulator receiver (REC) domain proteins (i.e., CheY), cytosine-C5 specific DNA methylases, IS607 transposase-like proteins, and RacA, a bacterial protein that anchors chromosomes to cell poles.','cd04762','HTH_MerR-trunc'),(956308,'gnl|CDD|200128',_binary 'DNA binding domain, excisionase family.  An excisionase, or Xis protein, is a small protein that binds and promotes excisive recombination; it is not enzymatically active. This model represents a number of putative excisionases and related proteins from temperate phage, plasmids, and transposons, as well as DNA binding domains of other proteins, such as a DNA modification methylase. This model identifies mostly small proteins and N-terminal regions of large proteins, but some proteins appear to have two copies. This domain appears similar, in both sequence and predicted secondary structure (PSIPRED) to the MerR family of transcriptional regulators (pfam00376). [Unknown function, General].','TIGR01764','Probable_excisionase'),(956310,'gnl|CDD|306808',_binary 'MerR family regulatory protein.','pfam00376','MerR'),(956311,'gnl|CDD|271184',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain containing six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01184','INT_C_like_1'),(956312,'gnl|CDD|226234',_binary 'Transcriptional antiterminator [Transcription].','COG3711','BglG'),(956327,'gnl|CDD|182729',_binary 'bifunctional diaminohydroxyphosphoribosylaminopyrimidine deaminase/5-amino-6-(5-phosphoribosylamino)uracil reductase; Provisional.','PRK10786','ribD'),(956366,'gnl|CDD|236378',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK09102','PRK09102'),(956373,'gnl|CDD|173773',_binary 'Topoisomerase-primase domain. This is a nucleotidyl transferase/hydrolase domain found in type IA, type IIA and type IIB topoisomerases, bacterial DnaG-type primases, small primase-like proteins from bacteria and archaea, OLD family nucleases from bacterial and archaea, and bacterial DNA repair proteins of the RecR/M family. This domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). This glutamate and two aspartates, cluster together to form a highly acid surface patch. The conserved glutamate may act as a general base in nucleotide polymerization by primases and in strand joining in topoisomerases and, as a general acid in strand cleavage by topisomerases and nucleases. The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.','cd00188','TOPRIM'),(956376,'gnl|CDD|313820',_binary 'Protein of unknown function (DUF2493).  Members of this family are all Proteobacteria. The function is not known.','pfam10686','DUF2493'),(956377,'gnl|CDD|306935',_binary 'alpha/beta hydrolase fold.  This catalytic domain is found in a very wide range of enzymes.','pfam00561','Abhydrolase_1'),(956378,'gnl|CDD|223669',_binary 'Predicted hydrolases or acyltransferases (alpha/beta hydrolase superfamily) [General function prediction only].','COG0596','MhpC'),(956380,'gnl|CDD|184875',_binary 'acetoin dehydrogenase E2 subunit dihydrolipoyllysine-residue acetyltransferase; Provisional.','PRK14875','PRK14875'),(956382,'gnl|CDD|274729',_binary '2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase.  This protein catalyzes the formation of SHCHC, or (1 R,6 R)-2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate, by elmination of pyruvate from 2-succinyl-5-enolpyruvyl-6-hydroxy-3-cyclohexene-1-carboxylate (SEPHCHC). Note that SHCHC synthase activity previously was attributed to MenD, which in fact is SEPHCHC synthase. [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR03695','menH_SHCHC'),(956384,'gnl|CDD|314939',_binary 'Serine aminopeptidase, S33.  This domain is found in bacteria and eukaryotes and is approximately 110 amino acids in length. It is found in association with pfam00561. The majority of the members in this family carry the exopeptidase active-site residues of Ser-122, Asp-239 and His-269 as in UniProtKB:Q7ZWC2.','pfam12146','Hydrolase_4'),(956386,'gnl|CDD|225176',_binary 'Lysophospholipase [Lipid metabolism].','COG2267','PldB'),(956388,'gnl|CDD|131480',_binary '3-oxoadipate enol-lactonase.  Members of this family are 3-oxoadipate enol-lactonase. Note that the substrate is known as 3-oxoadipate enol-lactone, 2-oxo-2,3-dihydrofuran-5-acetate, 4,5-Dihydro-5-oxofuran-2-acetate, and 5-oxo-4,5-dihydrofuran-2-acetate. The enzyme the catalyzes the fourth step in the protocatechuate degradation to beta-ketoadipate and then to succinyl-CoA and acetyl-CoA. 4-hydroxybenzoate, 3-hydroxybenzoate, and vanillate all can be converted in one step to protocatechuate. This enzyme also acts in catechol degradation. In genomes that catabolize both catechol and protocatechuate, two forms of this enzyme may be found. All members of the seed alignment for this model were chosen from within protocatechuate degradation operons of at least three genes of the pathway, from genomes with the complete pathway through beta-ketoadipate. [Energy metabolism, Other].','TIGR02427','b-ketoadipate_enol-lacton'),(956390,'gnl|CDD|315385',_binary 'Alpha/beta hydrolase family.  This family contains alpha/beta hydrolase enzymes of diverse specificity.','pfam12697','Abhydrolase_6'),(956392,'gnl|CDD|132100',_binary 'putative magnesium chelatase accessory protein.  Members of this family belong to the alpha/beta fold family hydrolases (pfam00561). Members are found in bacterial genomes if and only if they encoded for anoxygenic photosynthetic systems similar to that of Rhodobacter capsulatus and other alpha-Proteobacteria. Members often are encoded in the same operon as subunits of the protoporphyrin IX magnesium chelatase, and were once designated BchO. No literature supports a role as an actual subunit of magnesium chelatase, but an accessory role is possible, as suggested by placement by its probable hydrolase activity. [Energy metabolism, Photosynthesis].','TIGR03056','bchO_mg_che_rel'),(956394,'gnl|CDD|182637',_binary 'acyl-CoA esterase; Provisional.','PRK10673','PRK10673'),(956395,'gnl|CDD|284219',_binary 'KaiC.  This family is in the P-loop NTPase superfamily and is found in archaea, bacteria and eukaryotes. More than one copy is sometimes found in each protein. This family includes KaiC, which is one of the Kai proteins among which direct protein-protein association may be a critical process in the generation of circadian rhythms in cyanobacteria.','pfam06745','ATPase'),(956396,'gnl|CDD|235135',_binary 'haloalkane dehalogenase; Provisional.','PRK03592','PRK03592'),(956397,'gnl|CDD|211851',_binary 'pyrimidine utilization protein D.  This protein is observed in operons extremely similar to that characterized in E. coli K-12 responsible for the import and catabolism of pyrimidines, primarily uracil. This protein is a member of the hydrolase, alpha/beta fold family defined by pfam00067.','TIGR03611','RutD'),(956398,'gnl|CDD|132386',_binary '2-hydroxy-6-oxo-6-phenylhexa-2,4-dienoate hydrolase.  Members of this family are 2-hydroxy-6-oxo-6-phenylhexa-2,4-dienoate hydrolase, or HOPD hydrolase, the BphD protein of biphenyl degradation. BphD acts on the product of ring meta-cleavage by BphC. Many species carrying bphC and bphD are capable of degrading polychlorinated biphenyls as well as biphenyl itself.','TIGR03343','2-hydroxy-6-oxo-6-phenylh'),(956399,'gnl|CDD|131294',_binary 'poly(3-hydroxyalkanoate) depolymerase.  This family consists of the polyhydroxyalkanoic acid (PHA) depolymerase of Pseudomonas oleovorans, Pseudomonas putida BM01, and related species. This enzyme is part of polyester storage and mobilization system as in many bacteria. However, species containing this enzyme are unusual in their capacity to produce aromatic polyesters when grown on carbon sources such as benzoic acid or phenylacetic acid. [Energy metabolism, Other].','TIGR02240','Poly3-hydroxyalkanoate_de'),(956400,'gnl|CDD|236855',_binary '2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase; Provisional.','PRK11126','PRK11126'),(956401,'gnl|CDD|215484',_binary 'hydrolase, alpha/beta fold family protein.','PLN02894','PLN02894'),(956402,'gnl|CDD|215530',_binary '2-oxoglutarate decarboxylase/ hydro-lyase/ magnesium ion binding  / thiamin pyrophosphate binding.','PLN02980','PLN02980'),(956403,'gnl|CDD|254111',_binary 'Chlorophyllase.  This family consists of several plant specific Chlorophyllase proteins (EC:3.1.1.14). Chlorophyllase (Chlase) is the first enzyme involved in chlorophyll (Chl) degradation and catalyzes the hydrolysis of ester bond to yield chlorophyllide and phytol.','pfam07224','Chlorophyllase'),(956404,'gnl|CDD|215315',_binary 'hydrolase.','PLN02578','PLN02578'),(956405,'gnl|CDD|315422',_binary 'Chlorophyllase enzyme.  This family consists of several chlorophyllase and chlorophyllase-2 (EC:3.1.1.14) enzymes. Chlorophyllase (Chlase) is the first enzyme involved in chlorophyll (Chl) degradation and catalyzes the hydrolysis of an ester bond to yield chlorophyllide and phytol. The family includes both plant and Amphioxus members.','pfam12740','Chlorophyllase2'),(956406,'gnl|CDD|273596',_binary 'homoserine O-acetyltransferase.  This family describes homoserine-O-acetyltransferase, an enzyme of methionine biosynthesis. This model has been rebuilt to identify sequences more broadly, including a number of sequences suggested to be homoserine O-acetyltransferase based on proximity to other Met biosynthesis genes. [Amino acid biosynthesis, Aspartate family].','TIGR01392','Homoserine_O-acetyltransf'),(956407,'gnl|CDD|178633',_binary 'alpha/beta hydrolase fold protein; Provisional.','PLN03084','PLN03084'),(956408,'gnl|CDD|234678',_binary 'homoserine O-acetyltransferase; Provisional.','PRK00175','metX'),(956409,'gnl|CDD|179147',_binary 'haloalkane dehalogenase; Provisional.','PRK00870','PRK00870'),(956410,'gnl|CDD|182555',_binary 'esterase; Provisional.','PRK10566','PRK10566'),(956411,'gnl|CDD|179554',_binary 'haloalkane dehalogenase; Provisional.','PRK03204','PRK03204'),(956412,'gnl|CDD|177659',_binary 'chlorophyllase.','PLN00021','PLN00021'),(956413,'gnl|CDD|224932',_binary 'Homoserine acetyltransferase [Amino acid transport and metabolism].','COG2021','MET2'),(956414,'gnl|CDD|223477',_binary 'Predicted esterase [General function prediction only].','COG0400','COG0400'),(956415,'gnl|CDD|273783',_binary 'pimelyl-[acyl-carrier protein] methyl ester esterase.  This CoA-binding enzyme is required for the production of pimeloyl-coenzyme A, the substrate of the BioF protein early in the biosynthesis of biotin. Its exact function is unknown, but is proposed in ref 2. This enzyme belongs to the alpha/beta hydrolase fold family (pfam00561). Members of this family are restricted to the Proteobacteria. [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin].','TIGR01738','Pimeloyl-_methyl_ester_es'),(956416,'gnl|CDD|215352',_binary 'hydrolase; alpha/beta fold family protein.','PLN02652','PLN02652'),(956417,'gnl|CDD|315383',_binary 'Alpha/beta hydrolase family.  This family contains a diverse range of alpha/beta hydrolase enzymes.','pfam12695','Abhydrolase_5'),(956420,'gnl|CDD|316087',_binary 'Peptidase_C39 like family.','pfam13529','Peptidase_C39_2'),(956421,'gnl|CDD|310314',_binary 'Protein of unknown function (DUF790).  This family consists of several hypothetical archaeal proteins of unknown function.','pfam05626','DUF790'),(956615,'gnl|CDD|134010',_binary 'hypothetical protein.','PHA02450','PHA02450'),(956631,'gnl|CDD|307612',_binary 'Peptidase family M23.  Members of this family are zinc metallopeptidases with a range of specificities. The peptidase family M23 is included in this family, these are Gly-Gly endopeptidases. Peptidase family M23 are also endopeptidases. This family also includes some bacterial lipoproteins for which no proteolytic activity has been demonstrated. This family also includes leukocyte cell-derived chemotaxin 2 (LECT2) proteins. LECT2 is a liver-specific protein which is thought to be linked to hepatocyte growth although the exact function of this protein is unknown.','pfam01551','Peptidase_M23'),(956632,'gnl|CDD|223810',_binary 'Membrane proteins related to metalloendopeptidases [Cell envelope biogenesis, outer membrane].','COG0739','NlpD'),(956633,'gnl|CDD|227278',_binary 'Membrane-bound metallopeptidase [Cell division and chromosome partitioning].','COG4942','COG4942'),(956635,'gnl|CDD|236942',_binary 'AmiB activator; Provisional.','PRK11637','PRK11637'),(956636,'gnl|CDD|236946',_binary 'putative peptidase; Provisional.','PRK11649','PRK11649'),(956641,'gnl|CDD|213747',_binary 'spore cortex-lytic enzyme.  Members of this protein family are the spore cortex-lytic enzyme SleB from Bacillus subtilis and other Gram-positive, endospore-forming bacterial species. This protein is stored in an inactive form in the spore and activated during germination. [Cellular processes, Sporulation and germination].','TIGR02869','Spore_cortex-lytic_enzyme'),(956642,'gnl|CDD|236782',_binary 'lipoprotein NlpD; Provisional.','PRK10871','nlpD'),(956649,'gnl|CDD|224314',_binary 'Predicted transcriptional regulators [Transcription].','COG1396','HipB'),(956651,'gnl|CDD|237361',_binary 'bifunctional HTH-domain containing protein/aminotransferase; Provisional.','PRK13355','PRK13355'),(956653,'gnl|CDD|315510',_binary 'Helix-turn-helix domain.  Members of this family contains a DNA-binding helix-turn-helix domain. This family contains many example antitoxins from bacterial toxin-antitoxin systems. These antitoxins are likely to be DNA-binding domains.','pfam12844','HTH_19'),(956655,'gnl|CDD|311152',_binary 'Bacteriophage CI repressor helix-turn-helix domain.  This family consists of several phage CI repressor proteins and related bacterial sequences. The CI repressor is known to function as a transcriptional switch, determining whether transcription is lytic or lysogenic.','pfam07022','Phage_CI_repr'),(956657,'gnl|CDD|213767',_binary 'transcriptional regulator, y4mF family.  Members of this family belong to a clade of helix-turn-helix DNA-binding proteins, among the larger family pfam01381 (HTH_3; Helix-turn-helix). Members are similar in sequence to the HipB protein of E. coli. Genes for members of the seed alignment for this protein family were found to be closely linked to genes encoding proteins related to HipA. The HibBA operon appears to have some features in common with toxin-antitoxin post-segregational killing systems. [Regulatory functions, DNA interactions].','TIGR03070','couple_hipB'),(956668,'gnl|CDD|271183',_binary 'C-terminal catalytic domain of recombinase RitC, a component of the recombinase trio.  Recombinases belonging to the RitA (also known as pAE1 due to its presence in the deletion prone region of plasmid pAE1 of Alcaligenes eutrophus H1), RitB, and RitC families are associated in a complex referred to as a Recombinase in Trio (RIT) element.  These RIT elements consist of three adjacent and unidirectional overlapping genes, one from each family (ritABC in order of transcription).  All three integrases contain a catalytic motif, suggesting that they are all active enzymes.  However, their specific roles are not yet fully understood.  All three families belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism.','cd01182','INT_RitC_C_like'),(956669,'gnl|CDD|234935',_binary 'site-specific tyrosine recombinase XerC; Reviewed.','PRK01287','xerC'),(956674,'gnl|CDD|271181',_binary 'C-terminal catalytic domain of Lambda integrase, a tyrosine-based site-specific recombinase.  Lambda-type integrases catalyze site-specific integration and excision of temperate bacteriophages and other mobile genetic elements to and from the bacterial host chromosome. They are tyrosine-based site-specific recombinase and belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The phage lambda integrase can bridge two different and well-separated DNA sequences called arm- and core-sites. The C-terminal domain binds, cleaves and re-ligates DNA strands at the core-sites, while the N-terminal domain is largely responsible for high-affinity binding to the arm-type sites.','cd00800','INT_Lambda_C'),(956676,'gnl|CDD|271192',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01192','INT_C_like_3'),(956679,'gnl|CDD|271178',_binary 'C-terminal catalytic domain of recombinase RitB, a component of the recombinase trio.  Recombinases belonging to the RitA (also known as pAE1 due to its presence in the deletion prone region of plasmid pAE1 of Alcaligenes eutrophus H1), RitB, and RitC families are associated in a complex referred to as a Recombinase in Trio (RIT) element.  These RIT elements consist of three adjacent and unidirectional overlapping genes, one from each family (ritABC in order of transcription).  All three integrases contain a catalytic motif, suggesting that they are all active enzymes.  However, their specific roles are not yet fully understood.  All three families belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism.','cd00797','INT_RitB_C_like'),(956680,'gnl|CDD|315611',_binary 'Domain of Unknown Function with PDB structure (DUF3850).  The search results from NCBI sequence alignment indicates a conserved domain belonging to ASCH superfamily. Dali searching results show that the protein is a structurally similar to the PUA domain, suggesting it may be involved in RNA recognition. It has been reported that the deletion of PUA genes results in impaired growth (RluD) and competitive disadvantage (TruB) in Escherichia coli. Suggestions have been put forward that, apart from their usual catalytic role, certain PUS enzymes (e.g. TruB) may also act as chaperones for RNA folding. The interface interaction indicates that the biomolecule of protein NP_809782.1 should be a dimer.','pfam12961','DUF3850'),(956681,'gnl|CDD|315374',_binary 'PDDEXK-like domain of unknown function (DUF3799).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 265 and 420 amino acids in length. It appears that these proteins are distantly related to the PDDEXK superfamily and so these domains are likely to be nucleases. This family has a C-terminal cysteine cluster similar to that found in pfam01930.','pfam12684','DUF3799'),(956682,'gnl|CDD|236615',_binary 'exonuclease VIII; Reviewed.','PRK09709','PRK09709'),(956683,'gnl|CDD|310133',_binary 'Protein of unknown function (DUF732).  This family consists of several uncharacterized Mycobacterium tuberculosis and leprae proteins of unknown function.','pfam05305','DUF732'),(956685,'gnl|CDD|315958',_binary 'HNH endonuclease.  This HNH nuclease domain is found in CRISPR-related proteins.','pfam13395','HNH_4'),(956848,'gnl|CDD|271035',_binary 'Catalytic domain of Dual-specificity tYrosine-phosphorylated and -Regulated Kinase-like protein kinases.  Dual-specificity PKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine (S/T) as well as tyrosine residues on protein substrates. This subfamily is composed of the dual-specificity DYRKs and YAK1, as well as the S/T kinases (STKs), HIPKs. DYRKs and YAK1 autophosphorylate themselves on tyrosine residues and phosphorylate their substrates exclusively on S/T residues. Proteins in this subfamily play important roles in cell proliferation, differentiation, survival, growth, and development. The DYRK-like subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14133','PKc_DYRK_like'),(956851,'gnl|CDD|270918',_binary 'Catalytic domain of the Serine/Threonine protein kinase, Casein Kinase 1.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. CK1 phosphorylates a variety of substrates including enzymes, transcription and splice factors, cytoskeletal proteins, viral oncogenes, receptors, and membrane-associated proteins. There are mutliple isoforms of CK1 and in mammals, seven isoforms (alpha, beta, gamma1-3, delta, and epsilon) have been characterized. These isoforms differ mainly in the length and structure of their C-terminal non-catalytic region. Some isoforms have several splice variants such as the long (L) and short (S) variants of CK1alpha. CK1 proteins are involved in the regulation of many cellular processes including membrane transport processes, circadian rhythm, cell division, apoptosis, and the development of cancer and neurodegenerative diseases. The CK1 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14016','STKc_CK1'),(956852,'gnl|CDD|270881',_binary 'Catalytic domain of the Serine/Threonine kinase, Oocyte maturation factor Mos.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Mos (or c-Mos) is a germ-cell specific kinase that plays roles in both the release of primary arrest and the induction of secondary arrest in oocytes. It is expressed towards the end of meiosis I and is quickly degraded upon fertilization. It is a component of the cytostatic factor (CSF), which is responsible for metaphase II arrest. In addition, Mos activates a phoshorylation cascade that leads to the activation of the p34 subunit of MPF (mitosis-promoting factor or maturation promoting factor), a cyclin-dependent kinase that is responsible for the release of primary arrest in meiosis I. The Mos subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd13979','STKc_Mos'),(956855,'gnl|CDD|270914',_binary 'Pseudokinase domain, repeat 1, of eukaryotic translation Initiation Factor 2-Alpha Kinase 4 or General Control Non-derepressible-2.  The pseudokinase domain shows similarity to protein kinases but lacks crucial residues for catalytic activity. EIF2AKs phosphorylate the alpha subunit of eIF-2, resulting in the overall downregulation of protein synthesis. eIF-2 phosphorylation is induced in response to cellular stresses including virus infection, heat shock, nutrient deficiency, and the accummulation of unfolded proteins, among others. There are four distinct kinases that phosphorylate eIF-2 and control protein synthesis under different stress conditions: GCN2, protein kinase regulated by RNA (PKR), heme-regulated inhibitor kinase (HRI), and PKR-like endoplasmic reticulum kinase (PERK). GCN2 is activated by amino acid or serum starvation and UV irradiation. It induces GCN4, a transcriptional activator of amino acid biosynthetic genes, leading to increased production of amino acids under amino acid-deficient conditions. In serum-starved cells, GCN2 activation induces translation of the stress-responsive transcription factor ATF4, while under UV stress, GCN2 triggers transcriptional rescue via NF-kappaB signaling. GCN2 contains an N-terminal RWD, a degenerate kinase-like (repeat 1), the catalytic kinase (repeat 2), a histidyl-tRNA synthetase (HisRS)-like, and a C-terminal ribosome-binding and dimerization (RB/DD) domains. The degenerate pseudokinase domain of GCN2 may function as a regulatory domain. The GCN2 subfamily is part of a larger superfamily that includes the catalytic domains of serine/threonine kinases, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14012','PK_eIF2AK_GCN2_rpt1'),(956858,'gnl|CDD|270951',_binary 'Catalytic domain of the Serine/Threonine kinase, eukaryotic translation Initiation Factor 2-Alpha Kinase 2 or Heme-Regulated Inhibitor kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. HRI (or EIF2AK1) contains an N-terminal regulatory heme-binding domain and a C-terminal catalytic kinase domain. It is suppressed under normal conditions by binding of the heme iron, and is activated during heme deficiency. It functions as a critical regulator that ensures balanced synthesis of globins and heme, in order to form stable hemoglobin during erythroid differentiation and maturation. HRI also protects cells and enhances survival under iron-deficient conditions. EIF2AKs phosphorylate the alpha subunit of eIF-2, resulting in the downregulation of protein synthesis. The HRI subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14049','STKc_EIF2AK1_HRI'),(956917,'gnl|CDD|235595',_binary 'DNA polymerase II; Reviewed.','PRK05762','PRK05762'),(956918,'gnl|CDD|99826',_binary 'DEDDy 3\'-5\' exonuclease domain of Sulfolobus solfataricus DNA polymerase B1 and similar archaeal family-B DNA polymerases.  The 3\'-5\' exonuclease domain of Sulfolobus solfataricus DNA polymerase B1 and similar archaeal proteins. B1 is a family-B DNA polymerase. Family-B DNA polymerases contain an N-terminal DEDDy DnaQ-like exonuclease domain in the same polypeptide chain as the polymerase domain, similar to family-A DNA polymerases. B1displays thermostable polymerase and 3\'-5\' exonuclease activities. This exonuclease domain contains three sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and are involved in metal binding and catalysis. The exonuclease domain of family-B polymerases also contains a beta hairpin structure that plays an important role in active site switching in the event of nucleotide misincorporation. Family-B DNA polymerases from thermophilic archaea are unique in that they are able to recognize the presence of uracil in the template strand, leading to the stalling of DNA synthesis. This is an additional safeguard mechanism against increased levels of deaminated bases during genome duplication at high temperatures. S. solfataricus B1 also interacts with DNA polymerase Y and may contribute to genome stability mechanisms.','cd05783','DNA_polB_B1_exo'),(956927,'gnl|CDD|238543',_binary 'Rad51_DMC1_radA,B. This group of recombinases includes the eukaryotic proteins RAD51, RAD55/57 and the meiosis-specific protein DMC1, and the archaeal proteins radA and radB. They are closely related to the bacterial RecA group. Rad51 proteins catalyze a similiar recombination reaction as RecA, using ATP-dependent DNA binding activity and a DNA-dependent ATPase. However, this reaction is less efficient and requires accessory proteins such as RAD55/57 .','cd01123','Rad51_DMC1_radA'),(956928,'gnl|CDD|274047',_binary 'DNA repair and recombination protein RadB.  This family consists exclusively of archaeal RadB protein, a homolog of bacterial RecA (TIGR02012), eukaryotic RAD51 (TIGR02239) and DMC1 (TIGR02238), and archaeal RadA (TIGR02236).','TIGR02237','DNA_repair_and_recombinat'),(956929,'gnl|CDD|236482',_binary 'DNA repair and recombination protein RadB; Provisional.','PRK09361','radB'),(956931,'gnl|CDD|238688',_binary 'RadB. The archaeal protein radB shares similarity radA, the archaeal functional homologue to the bacterial RecA. The precise function of radB is unclear.','cd01394','radB'),(956932,'gnl|CDD|131292',_binary 'meiotic recombinase Dmc1.  This model describes DMC1, a subfamily of a larger family of DNA repair and recombination proteins. It is eukaryotic only and most closely related to eukaryotic RAD51. It also resembles archaeal RadA (TIGR02236) and RadB (TIGR02237) and bacterial RecA (TIGR02012). It has been characterized for human as a recombinase active only in meiosis.','TIGR02238','Meiotic_recombination_pro'),(956933,'gnl|CDD|312056',_binary 'Rad51.  Rad51 is a DNA repair and recombination protein and is a homolog of the bacterial ATPase RecA protein.','pfam08423','Rad51'),(956934,'gnl|CDD|235273',_binary 'DNA repair and recombination protein RadA; Validated.','PRK04301','radA'),(956935,'gnl|CDD|185407',_binary 'Rad51 protein; Provisional.','PTZ00035','PTZ00035'),(956937,'gnl|CDD|215620',_binary 'meiotic recombination protein DMC1 homolog; Provisional.','PLN03187','PLN03187'),(956938,'gnl|CDD|178728',_binary 'DNA repair protein RAD51 homolog; Provisional.','PLN03186','PLN03186'),(956940,'gnl|CDD|131290',_binary 'DNA repair and recombination protein RadA.  This family consists exclusively of archaeal RadA protein, a homolog of bacterial RecA (TIGR02012), eukaryotic RAD51 (TIGR02239), and archaeal RadB (TIGR02237). This protein is involved in DNA repair and recombination. The member from Pyrococcus horikoshii contains an intein. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02236','DNA_repair_and_recombinat'),(956941,'gnl|CDD|163593',_binary 'KaiC domain protein, PAE1156 family.  Members of this protein family are archaeal single-domain KaiC_related proteins, homologous to the Cyanobacterial circadian clock cycle protein KaiC, an autokinase/autophosphorylase that has two copies of the domain.','TIGR03881','KaiC_arch_4'),(956942,'gnl|CDD|214640',_binary 'ATPases associated with a variety of cellular activities.  AAA - ATPases associated with a variety of cellular activities. This profile/alignment only detects a fraction of this vast family. The poorly conserved N-terminal helix is missing from the alignment.','smart00382','AAA'),(956943,'gnl|CDD|274048',_binary 'DNA repair protein RAD51.  This eukaryotic sequence family consists of RAD51, a protein involved in DNA homologous recombination and repair. It is similar in sequence the exclusively meiotic recombinase DMC1 (TIGR02238), to archaeal families RadA (TIGR02236) and RadB (TIGR02237), and to bacterial RecA (TIGR02012).','TIGR02239','DNA_repair_protein_RAD51'),(956944,'gnl|CDD|238541',_binary 'Sms (bacterial radA) DNA repair protein. This protein is not related to archael radA any more than is to other RecA-like NTPases. Sms has a role in recombination and recombinational repair and is responsible for the stabilization or processing of branched DNA molecules.','cd01121','Sms'),(956945,'gnl|CDD|236994',_binary 'DNA repair protein RadA; Provisional.','PRK11823','PRK11823'),(956946,'gnl|CDD|163592',_binary 'KaiC domain protein, AF_0351 family.  This model represents a rather narrowly distributed archaeal protein family in which members have a single copy of the KaiC domain. This stands in contrast to the circadian clock protein KaiC itself, with two copies of the domain. Members are expected to have weak ATPase activity, by homology to the autokinase/autophosphorylase KaiC itself.','TIGR03880','KaiC_arch_3'),(956947,'gnl|CDD|223993',_binary 'Predicted ATP-dependent serine protease [Posttranslational modification, protein turnover, chaperones].','COG1066','Sms'),(956948,'gnl|CDD|316040',_binary 'AAA domain.  This AAA domain is found in a wide variety of presumed phage proteins.','pfam13479','AAA_24'),(956949,'gnl|CDD|307662',_binary 'ATPase domain predominantly from Archaea.  This family contain a conserved P-loop motif that is involved in binding ATP. There are eukaryote members as well as archaeal members in this family.','pfam01637','ATPase_2'),(956950,'gnl|CDD|225429',_binary 'Predicted ATPases involved in biogenesis of archaeal flagella [Cell motility and secretion / Intracellular trafficking and secretion].','COG2874','FlaH'),(956951,'gnl|CDD|236461',_binary 'circadian clock protein KaiC; Reviewed.','PRK09302','PRK09302'),(956954,'gnl|CDD|313385',_binary 'Phage-related minor tail protein.  Members of this family are found in putative phage tail tape measure proteins.','pfam10145','PhageMin_Tail'),(956955,'gnl|CDD|273790',_binary 'phage tail tape measure protein, TP901 family, core region.  This model represents a reasonably well conserved core region of a family of phage tail proteins. The member from phage TP901-1 was characterized as a tail length tape measure protein in that a shortened form of the protein leads to phage with proportionately shorter tails. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01760','COG5283:_Phage-related_ta'),(956961,'gnl|CDD|307296',_binary 'Cutinase.','pfam01083','Cutinase'),(956962,'gnl|CDD|311933',_binary 'PE-PPE domain.  This domain is found C terminal to the PE (pfam00934) and PPE (pfam00823) domains. The secondary structure of this domain is predicted to be a mixture of alpha helices and beta strands.','pfam08237','PE-PPE'),(956975,'gnl|CDD|238829',_binary 'Thioredoxin (TRX) superfamily; a large, diverse group of proteins containing a TRX-fold. Many members contain a classic TRX domain with a redox active CXXC motif. They function as protein disulfide oxidoreductases (PDOs), altering the redox state of target proteins via the reversible oxidation of their active site dithiol. The PDO members of this superfamily include TRX, protein disulfide isomerase (PDI), tlpA-like, glutaredoxin, NrdH redoxin, and the bacterial Dsb (DsbA, DsbC, DsbG, DsbE, DsbDgamma) protein families. Members of the superfamily that do not function as PDOs but contain a TRX-fold domain include phosducins, peroxiredoxins and glutathione (GSH) peroxidases, SCO proteins, GSH transferases (GST, N-terminal domain), arsenic reductases, TRX-like ferredoxins and calsequestrin, among others.','cd01659','TRX_superfamily'),(956977,'gnl|CDD|224311',_binary 'Arsenate reductase and related proteins, glutaredoxin family [Inorganic ion transport and metabolism].','COG1393','ArsC'),(956978,'gnl|CDD|239325',_binary 'Glutaredoxin (GRX) family, Dishevelled, Egl-10, and Pleckstrin (DEP) subfamily; composed of uncharacterized proteins containing a GRX domain and additional domains DEP and DUF547, both of which have unknown functions.  GRX is a glutathione (GSH) dependent reductase containing a redox active CXXC motif in a TRX fold. It has preference for mixed GSH disulfide substrates, in which it uses a monothiol mechanism where only the N-terminal cysteine is required. By altering the redox state of target proteins, GRX is involved in many cellular functions.','cd03027','GRX_DEP'),(956979,'gnl|CDD|310006',_binary 'Phage terminase, small subunit.','pfam05119','Terminase_4'),(956980,'gnl|CDD|226270',_binary 'Phage terminase, small subunit [DNA replication, recombination, and repair].','COG3747','COG3747'),(956981,'gnl|CDD|273691',_binary 'phage terminase, small subunit, putative, P27 family.  This model describes a distinct family of phage (and integrated prophage) putative terminase small subunit. Members tend to be adjacent to the phage terminase large subunit gene. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01558','hypothetical_protein'),(956992,'gnl|CDD|271195',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01195','INT_C_like_5'),(956993,'gnl|CDD|271180',_binary 'C-terminal catalytic domain of Cre recombinase (also called integrase).  Cre-like recombinases are tyrosine based site specific recombinases. They belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The bacteriophage P1 Cre recombinase maintains the circular phage replicon in a monomeric state by catalyzing a site-specific recombination between two loxP sites. The catalytic core domain of Cre recombinase is linked to a more divergent helical N-terminal domain, which interacts primarily with the DNA major groove proximal to the crossover region.','cd00799','INT_Cre_C'),(957003,'gnl|CDD|316055',_binary 'Phage integrase, N-terminal SAM-like domain.','pfam13495','Phage_int_SAM_4'),(957004,'gnl|CDD|224276',_binary 'Pentapeptide repeats containing protein [Function unknown].','COG1357','COG1357'),(957009,'gnl|CDD|185118',_binary 'secreted effector protein PipB2; Provisional.','PRK15196','PRK15196'),(957013,'gnl|CDD|185119',_binary 'secreted effector protein PipB; Provisional.','PRK15197','PRK15197'),(957019,'gnl|CDD|316151',_binary 'Pentapeptide repeats (9 copies).','pfam13599','Pentapeptide_4'),(957024,'gnl|CDD|307102',_binary 'Pentapeptide repeats (8 copies).  These repeats are found in many cyanobacterial proteins. The repeats were first identified in hglK. The function of these repeats is unknown. The structure of this repeat has been predicted to be a beta-helix. The repeat can be approximately described as A(D/N)LXX, where X can be any amino acid.','pfam00805','Pentapeptide'),(957036,'gnl|CDD|309515',_binary 'ERF superfamily.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to ERF.','pfam04404','ERF'),(957044,'gnl|CDD|180232',_binary 'replicative DNA helicase; Provisional.','PRK05748','PRK05748'),(957047,'gnl|CDD|235907',_binary 'replicative DNA helicase; Provisional.','PRK07004','PRK07004'),(957050,'gnl|CDD|181562',_binary 'replicative DNA helicase; Provisional.','PRK08840','PRK08840'),(957051,'gnl|CDD|136106',_binary 'replicative DNA helicase; Validated.','PRK06904','PRK06904'),(957052,'gnl|CDD|181193',_binary 'replicative DNA helicase; Provisional.','PRK08006','PRK08006'),(957053,'gnl|CDD|181676',_binary 'replicative DNA helicase; Provisional.','PRK09165','PRK09165'),(957054,'gnl|CDD|307081',_binary 'DnaB-like helicase N terminal domain.  The hexameric helicase DnaB unwinds the DNA duplex at the Escherichia coli chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerisation of the N-terminal domain has been observed and may occur during the enzymatic cycle. This N-terminal domain is required both for interaction with other proteins in the primosome and for DnaB helicase activity.','pfam00772','DnaB'),(957056,'gnl|CDD|180528',_binary 'replicative DNA helicase; Provisional.','PRK06321','PRK06321'),(957057,'gnl|CDD|168322',_binary 'replicative DNA helicase; Provisional.','PRK05973','PRK05973'),(957060,'gnl|CDD|273067',_binary 'DNA repair protein RadA.  The gene protuct codes for a probable ATP-dependent protease involved in both DNA repair and degradation of proteins, peptides, glycopeptides. Also known as sms. Residues 11-28 of the SEED alignment contain a putative Zn binding domain. Residues 110-117 of the seed contain a putative ATP binding site both documented in Haemophilus (SP:P45266) and in Listeria monocytogenes (SP:Q48761) . for E.coli see ( J. BACTERIOL. 178:5045-5048(1996)). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00416','DNA_repair_protein_RadA_h'),(957074,'gnl|CDD|238979',_binary 'Gluconate kinase (GntK) catalyzes the phosphoryl transfer from ATP to gluconate. The resulting product gluconate-6-phoshate is an important precursor of gluconate metabolism. GntK acts as a dimmer composed of two identical subunits.','cd02021','GntK'),(957075,'gnl|CDD|235858',_binary 'hypothetical protein; Provisional.','PRK06762','PRK06762'),(957077,'gnl|CDD|274075',_binary 'RNA ligase, DRB0094 family.  The member of this family from Deinococcus radiodurans, a species that withstands and recovers from extensive radiation or dessication damage, is an apparent RNA ligase. It repairs RNA stand breaks in nicked DNA:RNA and RNA:RNA but not DNA:DNA duplexes. It has adenylyltransferase activity associated with the C-terminal domain. Related proteins also in this family are found in Streptomyces avermitilis MA-4680 and in bacteriophage 44RR2.8t. The phage example is unsurprising since one mechanism of host cell defense against phage is cleavage and inactivation of certain tRNA molecules. A fungal sequence from Neurospora crassa scores between trusted and noise cutofffs and may be similar in function.','TIGR02306','hypothetical_protein_44RR'),(957078,'gnl|CDD|274076',_binary 'RNA ligase, Rnl2 family.  Members of this family ligate (seal breaks in) RNA. Members so far include phage proteins that can counteract a host defense of cleavage of specific tRNA molecules, trypanosome ligases involved in RNA editing, but no prokaryotic host proteins . [Mobile and extrachromosomal element functions, Prophage functions, Transcription, RNA processing].','TIGR02307','RNA-editing_ligase_2_mito'),(957079,'gnl|CDD|236092',_binary 'single-stranded DNA-binding protein; Provisional.','PRK07772','PRK07772'),(957080,'gnl|CDD|239942',_binary 'SSB_OBF: A subfamily of OB folds similar to the OB fold of ssDNA-binding protein (SSB). SSBs bind with high affinity to ssDNA. They bind to and protect ssDNA intermediates during DNA metabolic pathways. All bacterial and eukaryotic SSBs studied to date oligomerize to bring together four OB folds in their active state. The majority (e.g. Escherichia coli SSB) have a single OB fold per monomer, which oligomerize to form a homotetramer. However, Deinococcus and Thermus SSB proteins have two OB folds per monomer, which oligomerize to form a homodimer. Mycobacterium tuberculosis SSB varies in quaternary structure from E. coli SSB. It forms a dimer of dimers having a unique dimer interface, which lends the protein greater stability. Included in this group are OB folds similar to Escherichia coli PriB. E.coli PriB is homodimeric with each monomer having a single OB fold. It does not appear to form higher order oligomers. PriB is an essential protein for the replication restart at forks that have stalled at sites of DNA damage. It also plays a role in the assembly of primosome during replication initiation at the bacteriophage phiX174 origin. PriB physically interacts with SSB and binds ssDNA with high affinity.','cd04496','SSB_OBF'),(957081,'gnl|CDD|223702',_binary 'Single-stranded DNA-binding protein [DNA replication, recombination, and repair].','COG0629','Ssb'),(957082,'gnl|CDD|273179',_binary 'single stranded DNA-binding protein (ssb).  All proteins in this family for which functions are known are single-stranded DNA binding proteins that function in many processes including transcription, repair, replication and recombination. Members encoded between genes for ribosomal proteins S6 and S18 should be annotated as primosomal protein N (PriB). Forms in gamma-protoeobacteria are much shorter and poorly recognized by this model. Additional members of this family include phage proteins. Eukaryotic members are organellar proteins. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00621','Single-stranded_DNA-bindi'),(957083,'gnl|CDD|306854',_binary 'Single-strand binding protein family.  This family includes single stranded binding proteins and also the primosomal replication protein N (PriB). PriB forms a complex with PriA, PriC and ssDNA.','pfam00436','SSB'),(957084,'gnl|CDD|235626',_binary 'hypothetical protein; Validated.','PRK05853','PRK05853'),(957085,'gnl|CDD|168659',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06751','PRK06751'),(957086,'gnl|CDD|180915',_binary 'single-stranded DNA-binding protein; Provisional.','PRK07275','PRK07275'),(957087,'gnl|CDD|168660',_binary 'single-stranded DNA-binding protein; Validated.','PRK06752','PRK06752'),(957088,'gnl|CDD|180517',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06293','PRK06293'),(957089,'gnl|CDD|180777',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06958','PRK06958'),(957090,'gnl|CDD|184284',_binary 'single-stranded DNA-binding protein; Provisional.','PRK13732','PRK13732'),(957091,'gnl|CDD|236362',_binary 'single-stranded DNA-binding protein; Provisional.','PRK09010','PRK09010'),(957092,'gnl|CDD|235585',_binary 'single-stranded DNA-binding protein; Provisional.','PRK05733','PRK05733'),(957093,'gnl|CDD|136097',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06863','PRK06863'),(957094,'gnl|CDD|180986',_binary 'single-stranded DNA-binding protein; Provisional.','PRK07459','PRK07459'),(957095,'gnl|CDD|180914',_binary 'single-stranded DNA-binding protein; Provisional.','PRK07274','PRK07274'),(957096,'gnl|CDD|181549',_binary 'single-stranded DNA-binding protein; Provisional.','PRK08763','PRK08763'),(957097,'gnl|CDD|236275',_binary 'single-stranded DNA-binding protein; Provisional.','PRK08486','PRK08486'),(957098,'gnl|CDD|235782',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06341','PRK06341'),(957099,'gnl|CDD|239935',_binary 'ExoVII_LU_OBF: A subfamily of OB folds corresponding to the N-terminal OB-fold domain of Escherichia coli exodeoxyribonuclease VII (ExoVII) large subunit. E. coli ExoVII is composed of two non-identical subunits. E. coli ExoVII is a single-strand-specific exonuclease which degrades ssDNA from both 3-prime and 5-prime ends. ExoVII plays a role in methyl-directed mismatch repair in vivo. ExoVII may also guard the genome from mutagenesis by removing excess ssDNA, since the build up of ssDNA would lead to SOS induction and PolIV-dependent mutagenesis.','cd04489','ExoVII_LU_OBF'),(957100,'gnl|CDD|169261',_binary 'single-stranded DNA-binding protein; Provisional.','PRK08182','PRK08182'),(957101,'gnl|CDD|316279',_binary 'OB-fold nucleic acid binding domain.  This family contains OB-fold domains that bind to nucleic acids.','pfam13742','tRNA_anti_2'),(957102,'gnl|CDD|272979',_binary 'exodeoxyribonuclease VII, large subunit.  This family consist of exodeoxyribonuclease VII, large subunit XseA which catalyses exonucleolytic cleavage in either the 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. Exonuclease VII consists of one large subunit and four small subunits. [DNA metabolism, Degradation of DNA].','TIGR00237','exodeoxyribonuclease_VII_'),(957103,'gnl|CDD|135984',_binary 'single-stranded DNA-binding protein; Provisional.','PRK06642','PRK06642'),(957104,'gnl|CDD|234714',_binary 'exodeoxyribonuclease VII large subunit; Reviewed.','PRK00286','xseA'),(957327,'gnl|CDD|212149',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from bacteria and archaea.  This family corresponds to a group of uncharacterized hypothetical proteins from bacteria, showing a high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain that contains a well conserved divalent ion-binding motif EXX[E/D].','cd11542','NTP-PPase_u5'),(957328,'gnl|CDD|273716',_binary 'phage/plasmid primase, P4 family, C-terminal domain.  This model represents a clade within a larger family of proteins from viruses of bacteria and animals. Members of this family are found in phage and plasmids of bacteria and archaea only. The model describes a domain of about 300 residues, found generally toward the protein C-terminus. [DNA metabolism, DNA replication, recombination, and repair, Mobile and extrachromosomal element functions, Prophage functions].','TIGR01613','putative_primase'),(957329,'gnl|CDD|225913',_binary 'Phage associated DNA primase [General function prediction only].','COG3378','COG3378'),(957330,'gnl|CDD|173172',_binary 'hypothetical protein; Provisional.','PRK14709','PRK14709'),(957331,'gnl|CDD|235927',_binary 'hypothetical protein; Validated.','PRK07078','PRK07078'),(957334,'gnl|CDD|312290',_binary 'D5 N terminal like.  This domain is found in D5 proteins of DNA viruses and bacteriophage P4 DNA primases phages.','pfam08706','D5_N'),(957336,'gnl|CDD|197953',_binary 'D5 N terminal like.  This domain is found in D5 proteins of DNA viruses and bacteriophage P4 DNA primases phages.','smart00885','D5_N'),(957337,'gnl|CDD|163701',_binary 'Mpr1p, Pad1p N-terminal (MPN) domains with catalytic isopeptidase activity (metal-binding).  This family contains archaeal and bacterial MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+)-like domains. These domains contain the signature JAB1/MPN/Mov34 metalloenzyme (JAMM) motif, EXnHS/THX7SXXD, which is involved in zinc ion coordination and provides the active site for isopeptidase activity for the release of ubiquitin from ubiquitinated proteins (thus having deubiquitinating (DUB) activity) that are tagged for degradation.  The JAMM proteins likely hydrolyze ubiquitin conjugates in a manner similar to thermolysin, in which the zinc-polarized aqua ligand serves as the nucleophile, compared with the classical DUBs that do so with a cysteine residue in the active site.','cd08070','MPN_like'),(957338,'gnl|CDD|163704',_binary 'Mpr1p, Pad1p N-terminal (MPN) domains with catalytic isopeptidase activity (metal-binding) found in proteins also containing NlpC/P60 domains.  This family contains bacterial MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+)-like domains at the N-terminus of NlpC/P60 phage tail protein domains. These domains contain the signature JAB1/MPN/Mov34 metalloenzyme (JAMM) motif, EXnHS/THX7SXXD, which is involved in zinc ion coordination and provides the active site for isopeptidase activity for the release of ubiquitin from ubiquitinated proteins (thus having deubiquitinating (DUB) activity) that are tagged for degradation.  The JAMM proteins likely hydrolyze ubiquitin conjugates in a manner similar to thermolysin, in which the zinc-polarized aqua ligand serves as the nucleophile, compared with the classical DUBs that do so with a cysteine residue in the active site.','cd08073','MPN_NLPC_P60'),(957339,'gnl|CDD|224229',_binary 'Predicted metal-dependent protease of the PAD1/JAB1 superfamily [General function prediction only].','COG1310','COG1310'),(957340,'gnl|CDD|316939',_binary 'Prokaryotic homologs of the JAB domain.  These are metalloenzymes that function as the ubiquitin isopeptidase/ deubiquitinase in the ubiquitin-based signaling and protein turnover pathways in eukaryotes. Prokaryotic JAB domains are predicted to have a similar role in their cognates of the ubiquitin modification pathway. The domain is widely found in bacteria, archaea and phages where they are present in several gene contexts in addition to those that correspond to the prokaryotic cognates of the eukaryotic Ub pathway. Other contexts in which JAB domains are present include gene neighbor associations with ubiquitin fold domains in cysteine and siderophore biosynthesis, and phage tail morphogenesis, where they are shown or predicted to process the associated ubiquitin. A distinct family, the RadC-like JAB domains are widespread in bacteria and are predicted to function as nucleases. In halophilic archaea the JAB domain shows strong gene-neighborhood associations with a nucleotidyltransferase suggesting a role in nucleotide metabolism.','pfam14464','Prok-JAB'),(957341,'gnl|CDD|213641',_binary 'phage putative head morphogenesis protein, SPP1 gp7 family.  This model describes a region of about 110 amino acids found exclusively in phage-related proteins, internally or toward the C-terminus. One member, gp7 of phage SPP1, appears involved in head morphogenesis. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01641','Uncharacterized_protein_M'),(957342,'gnl|CDD|309383',_binary 'Phage Mu protein F like protein.  Members of this family are found in double-stranded DNA bacteriophages, and in some bacteria. A member of this family is required for viral head morphogenesis in bacteriophage SPP1. This family is possibly a minor head protein. This family may be related to the family TT_ORF1 (pfam02956).','pfam04233','Phage_Mu_F'),(957345,'gnl|CDD|316690',_binary 'YodL-like.  The YodL-like protein family includes the B. subtilis YodL protein, which is functionally uncharacterized. This domain family is found in bacteria, and is approximately 100 amino acids in length. There are two completely conserved residues (Y and D) that may be functionally important.','pfam14191','YodL'),(957350,'gnl|CDD|306645',_binary 'SNF2 family N-terminal domain.  This domain is found in proteins involved in a variety of processes including transcription regulation (e.g., SNF2, STH1, brahma, MOT1), DNA repair (e.g., ERCC6, RAD16, RAD5), DNA recombination (e.g., RAD54), and chromatin unwinding (e.g., ISWI) as well as a variety of other proteins with little functional information (e.g., lodestar, ETL1).','pfam00176','SNF2_N'),(957352,'gnl|CDD|215601',_binary 'Probable chromatin-remodeling complex ATPase chain; Provisional.','PLN03142','PLN03142'),(957357,'gnl|CDD|238005',_binary 'DEAD-like helicases superfamily. A diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.','cd00046','DEXDc'),(957359,'gnl|CDD|130679',_binary 'phage nucleotide-binding protein.  This model represents an uncharacterized family of proteins from a number of phage of Gram-positive bacteria. This protein contains a P-loop motif, G/A-X-X-G-X-G-K-T near its amino end. The function of this protein is unknown. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01618','phage_P_loop'),(957389,'gnl|CDD|224305',_binary 'Histidinol phosphatase and related hydrolases of the PHP family [Amino acid transport and metabolism / General function prediction only].','COG1387','HIS2'),(957400,'gnl|CDD|310573',_binary 'Protein of unknown function (DUF935).  This family consists of several bacterial proteins of unknown function as well as the Bacteriophage Mu gp29 protein.','pfam06074','DUF935'),(957426,'gnl|CDD|271197',_binary 'FimB and FimE and related proteins, integrase/recombinases.  This CD includes proteins similar to E.coli FimE and FimB and Proteus mirabilis MrpI. FimB and FimE are the regulatory proteins during expression of type 1 fimbriae in Escherichia coli. The fimB and fimE proteins direct the phase switch into the \'on\' and \'off\' position. MrpI is the regulatory protein of proteus mirabilis fimbriae expression. This family belongs to the integrase/recombinase superfamily.','cd01197','INT_FimBE_like'),(957433,'gnl|CDD|182126',_binary 'tyrosine recombinase; Provisional.','PRK09871','PRK09871'),(957434,'gnl|CDD|226144',_binary 'Prophage antirepressor [Transcription].','COG3617','COG3617'),(957435,'gnl|CDD|214989',_binary 'BRO family, N-terminal domain.  This family includes the N-terminus of baculovirus BRO and ALI motif proteins. The function of BRO proteins is unknown. It has been suggested that BRO-A and BRO-C are DNA binding proteins that influence host DNA replication and/or transcription. This Pfam domain does not include the characteristic invariant alanine, leucine, isoleucine motif of the ALI proteins.','smart01040','Bro-N'),(957436,'gnl|CDD|308223',_binary 'BRO family, N-terminal domain.  This family includes the N-terminus of baculovirus BRO and ALI motif proteins. The function of BRO proteins is unknown. It has been suggested that BRO-A and BRO-C are DNA binding proteins that influence host DNA replication and/or transcription. This Pfam domain does not include the characteristic invariant alanine, leucine, isoleucine motif of the ALI proteins.','pfam02498','Bro-N'),(957437,'gnl|CDD|308789',_binary 'Phage antirepressor protein KilAC domain.  This domain was called the KilAC domain by Iyer and colleagues.','pfam03374','ANT'),(957438,'gnl|CDD|226171',_binary 'Uncharacterized phage-encoded protein [Function unknown].','COG3645','COG3645'),(957452,'gnl|CDD|132934',_binary 'Signal peptide peptidase A (SppA), a serine protease, has catalytic Ser-Lys dyad.  Signal peptide peptidase A (SppA; Peptidase S49; Protease IV): SppA is found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. This subfamily contains members with either a single domain (sometimes referred to as 36K type), such as sohB peptidase, protein C and archaeal signal peptide peptidase, or an amino-terminal domain in addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members (sometimes referred to as 67K type), similar to E. coli and Arabidopsis thaliana SppA peptidases. Site-directed mutagenesis and sequence analysis have shown these SppAs to be serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. Interestingly, the single membrane spanning E. coli SppA carries out catalysis using a Ser-Lys dyad with the serine located in the conserved carboxy-terminal protease domain and the lysine in the non-conserved amino-terminal domain.','cd07023','S49_Sppa_N_C'),(957453,'gnl|CDD|132932',_binary 'Nodulation formation efficiency D (NfeD) is a membrane-bound ClpP-class protease.  Nodulation formation efficiency D (NfeD; stomatin operon partner protein, STOPP; DUF107) is a member of membrane-anchored ClpP-class proteases. Currently, more than 300 NfeD homologs have been identified - all of which are bacterial or archaeal in origin. Majority of these genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named STOPP (stomatin operon partner protein). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 (1510-N or PH1510-N) from Pyrococcus horikoshii has been shown to possess serine protease activity and has a Ser-Lys catalytic dyad, preferentially cleaving hydrophobic substrates. Difference in oligomeric form and catalytic residues between 1510-N (forming a dimer) and ClpP (forming a tetradecamer) shows a possible functional difference: 1510-N is likely to have a regulatory function while ClpP is involved in protein quality control.','cd07021','Clp_protease_NfeD_like'),(957455,'gnl|CDD|223689',_binary 'Periplasmic serine proteases (ClpP class) [Posttranslational modification, protein turnover, chaperones / Intracellular trafficking and secretion].','COG0616','SppA'),(957457,'gnl|CDD|273227',_binary 'signal peptide peptidase SppA, 36K type.  The related but duplicated, double-length protein SppA (protease IV) of E. coli was shown experimentally to degrade signal peptides as are released by protein processing and secretion. This protein shows stronger homology to the C-terminal region of SppA than to the N-terminal domain or to the related putative protease SuhB. The member of this family from Bacillus subtilis was shown to have properties consistent with a role in degrading signal peptides after cleavage from precursor proteins, although it was not demonstrated conclusively. [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00706','Putative_signal_peptide_p'),(957458,'gnl|CDD|110924',_binary 'Serine dehydrogenase proteinase.  This family of archaebacterial proteins, formerly known as DUF114, has been found to be a serine dehydrogenase proteinase distantly related to ClpP proteinases that belong to the serine proteinase superfamily. The family has a catalytic triad of Ser, Asp, His residues, which shows an altered residue ordering compared with the ClpP proteinases but similar to that of the carboxypeptidase clan.','pfam01972','SDH_sah'),(957459,'gnl|CDD|183588',_binary 'ATP-dependent Clp protease-like protein; Reviewed.','PRK12552','PRK12552'),(957460,'gnl|CDD|236628',_binary 'recombination and repair protein RecT; Reviewed.','PRK09846','recT'),(957462,'gnl|CDD|309097',_binary 'RecT family.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to RecT.','pfam03837','RecT'),(957464,'gnl|CDD|129703',_binary 'recombinase, phage RecT family.  All proteins in this family for which functions are known bind ssDNA and are involved in the the pairing of homologous DNA This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). RecT and homologs are found in prophage regions of bacterial genomes. RecT works with a partner protein, RecE. [DNA metabolism, DNA replication, recombination, and repair, Mobile and extrachromosomal element functions, Prophage functions].','TIGR00616','Uncharacterized_protein_Y'),(957466,'gnl|CDD|226246',_binary 'Recombinational DNA repair protein (RecE pathway) [DNA replication, recombination, and repair].','COG3723','RecT'),(957468,'gnl|CDD|312784',_binary 'Ftsk gamma domain.  This domain directs oriented DNA translocation and forms a winged helix structure. Mutated proteins with substitutions in the FtsK gamma DNA-recognition helix are impaired in DNA binding.','pfam09397','Ftsk_gamma'),(957469,'gnl|CDD|197911',_binary 'This domain directs oriented DNA translocation and forms a winged helix structure.  Mutated proteins with substitutions in the FtsK gamma DNA-recognition helix are impaired in DNA binding.','smart00843','Ftsk_gamma'),(957480,'gnl|CDD|283256',_binary 'Zinc-binding loop region of homing endonuclease.  This domain is the short zinc-binding loops region of a number of much longer chain homing endonucleases. Such loops are probably stabilized by the zinc and may be viewed as small but separate domains. The common structural feature of these domains is that at least three zinc ligands lie very close to each other in the sequence and are not incorporated into regular secondary structural elements. The biological roles played by these small zinc-binding domains are presently unknown.','pfam05551','zf-His_Me_endon'),(957487,'gnl|CDD|224343',_binary 'Predicted transcriptional regulator contains Xre-like HTH domain [Function unknown].','COG1426','COG1426'),(957489,'gnl|CDD|224393',_binary 'Predicted transcriptional regulators [Transcription].','COG1476','COG1476'),(957510,'gnl|CDD|222875',_binary 'UvsW helicase; Provisional.','PHA02558','uvsW'),(957515,'gnl|CDD|306725',_binary 'DEAD/DEAH box helicase.  Members of this family include the DEAD and DEAH box helicases. Helicases are involved in unwinding nucleic acids. The DEAD box helicases are involved in various aspects of RNA metabolism, including nuclear transcription, pre mRNA splicing, ribosome biogenesis, nucleocytoplasmic transport, translation, RNA decay and organellar gene expression.','pfam00270','DEAD'),(957536,'gnl|CDD|99835',_binary 'DEDDh 3\'-5\' exonuclease domain of the epsilon subunit of Escherichia coli DNA polymerase III and similar proteins.  This subfamily is composed of the epsilon subunit of Escherichia coli DNA polymerase III (Pol III) and similar proteins. Pol III is the main DNA replicating enzyme in bacteria, functioning as the chromosomal replicase. It is a holoenzyme complex of ten different subunits, three of which (alpha, epsilon, and theta) compose the catalytic core. The Pol III epsilon subunit, encoded by the dnaQ gene, is a DEDDh-type 3\'-5\' exonuclease which is responsible for the proofreading activity of the polymerase, increasing the fidelity of DNA synthesis. It contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The epsilon subunit of Pol III also functions as a stabilizer of the holoenzyme complex.','cd06131','DNA_pol_III_epsilon_Ecoli'),(957538,'gnl|CDD|153095',_binary 'PLA2_plant: Plant-specific sub-family of  Phospholipase A2, a super-family of secretory and cytosolic enzymes; the latter are either Ca dependent or Ca independent. Enzymatically active PLA2 cleaves the sn-2 position of the glycerol backbone of phospholipids; secreted PLA2s have also been found to specifically bind to a variety of soluble and membrane proteins in mammals, including receptors. As a toxin, PLA2 is a potent presynaptic neurotoxin which blocks nerve terminals by binding to the nerve membrane and hydrolyzing stable membrane lipids. The products of the hydrolysis cannot form bilayers leading to a change in membrane conformation and ultimately to a block in the release of neurotransmitters. PLA2 may form dimers or oligomers. This sub-family does not appear to have a conserved active site and metal-binding loop.','cd04706','PLA2_plant'),(957547,'gnl|CDD|317107',_binary 'Coiled-coil region of CCDC155 or KASH.  This coiled-coil region is found in the central part of KASH or Klarsicht/ANC-1/Syne/homology proteins. KASH are a meiosis-specific proteins that localize at telomeres and interact with SUN1, thus being implicated in meiotic chromosome dynamics and homolog pairing.','pfam14662','KASH_CCD'),(957551,'gnl|CDD|214861',_binary 'Resolvase, N terminal domain.  The N-terminal domain of the resolvase family contains the active site and the dimer interface. The extended arm at the C-terminus of this domain connects to the C-terminal helix-turn-helix domain of resolvase.','smart00857','Resolvase'),(957552,'gnl|CDD|238206',_binary 'Serine Recombinase family, catalytic domain; a DNA binding domain may be present either N- or C-terminal to the catalytic domain. These enzymes perform site-specific recombination of DNA molecules by a concerted, four-strand cleavage and rejoining mechanism which involves a transient phosphoserine linkage between DNA and serine recombinase. Serine recombinases demonstrate functional versatility and include resolvases, invertases, integrases, and transposases. Resolvases and invertases (i.e. Tn3, gamma-delta, Tn5044 resolvases, Gin and Hin invertases) in this family contain a C-terminal DNA binding domain and comprise a major phylogenic group. Also included are phage- and bacterial-encoded recombinases such as phiC31 integrase, SpoIVCA excisionase, and Tn4451 TnpX transposase. These integrases and transposases have larger C-terminal domains compared to resolvases/invertases and are referred to as large serine recombinases. Also belonging to this family are proteins with N-terminal DNA binding domains similar to IS607- and IS1535-transposases from Helicobacter and Mycobacterium.','cd00338','Ser_Recombinase'),(957553,'gnl|CDD|306701',_binary 'Resolvase, N terminal domain.  The N-terminal domain of the resolvase family (this family) contains the active site and the dimer interface. The extended arm at the C-terminus of this domain connects to the C-terminal helix-turn-helix domain of resolvase - see pfam02796.','pfam00239','Resolvase'),(957554,'gnl|CDD|224872',_binary 'Site-specific recombinases, DNA invertase Pin homologs [DNA replication, recombination, and repair].','COG1961','PinR'),(957555,'gnl|CDD|239737',_binary 'Serine Recombinase (SR) family, Resolvase and Invertase subfamily, catalytic domain; members contain a C-terminal DNA binding domain. Serine recombinases catalyze site-specific recombination of DNA molecules by a concerted, four-strand cleavage and rejoining mechanism which involves a transient phosphoserine linkage between DNA and the enzyme. They are functionally versatile and include resolvases, invertases, integrases, and transposases. Resolvases and invertases affect resolution or inversion and comprise a major phylogenic group. Resolvases (e.g. Tn3, gamma-delta, and Tn5044) normally recombine two sites in direct repeat causing deletion of the DNA between the sites. Invertases (e.g. Gin and Hin) recombine sites in inverted repeat to invert the DNA between the sites. Cointegrate resolution with gamma-delta resolvase requires the formation of a synaptosome of three resolvase dimers bound to each of two res sites on the DNA. Also included in this subfamily are some putative integrases including a sequence from bacteriophage phi-FC1.','cd03768','SR_ResInv'),(957601,'gnl|CDD|318635',_binary '3\' exoribonuclease, RNase T-like.  This is a highly divergent 3\' exoribonuclease family. The proteins constitute a typical RNase fold, where the active site residues form a magnesium catalytic centre. The protein of the solved structure readily cleaves 3### overhangs in a time-dependent manner. It is similar to DEDD-type RNases and is an unusual ATP-binding protein that binds ATP and dATP. It forms a dimer in solution and both protomers in the asymmetric unit bind a magnesium ion through Asp-6 in UniProtKB:P9WJ73.','pfam16473','DUF5051'),(957602,'gnl|CDD|307736',_binary 'Glycosyltransferase family 25 (LPS biosynthesis protein).  Members of this family belong to Glycosyltransferase family 25 This is a family of glycosyltransferases involved in lipopolysaccharide (LPS) biosynthesis. These enzymes catalyze the transfer of various sugars onto the growing LPS chain during its biosynthesis.','pfam01755','Glyco_transf_25'),(957603,'gnl|CDD|133474',_binary 'Glycosyltransferase family 25 [lipooligosaccharide (LOS) biosynthesis protein] is a family of glycosyltransferases involved in LOS biosynthesis. The members include the beta(1,4) galactosyltransferases: Lgt2 of Moraxella catarrhalis, LgtB and LgtE of Neisseria gonorrhoeae and Lic2A of Haemophilus influenzae. M. catarrhalis Lgt2 catalyzes the addition of galactose (Gal) to the growing chain of LOS on the cell surface. N. gonorrhoeae LgtB and LgtE link Gal-beta(1,4)  to GlcNAc (N-acetylglucosamine) and Glc (glucose), respectively. The genes encoding LgtB and LgtE are two genes of a five gene locus involved in the synthesis of gonococcal LOS. LgtE is believed to perform the first step in LOS biosynthesis.','cd06532','Glyco_transf_25'),(957604,'gnl|CDD|316134',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13578','Methyltransf_24'),(957605,'gnl|CDD|310370',_binary 'Macrocin-O-methyltransferase (TylF).  This family consists of bacterial macrocin O-methyltransferase (TylF) proteins. TylF is responsible for the methylation of macrocin to produce tylosin. Tylosin is a macrolide antibiotic used in veterinary medicine to treat infections caused by Gram-positive bacteria and as an animal growth promoter in the swine industry. It is produced by several Streptomyces species. As with other macrolides, the antibiotic activity of tylosin is due to the inhibition of protein biosynthesis by a mechanism that involves the binding of tylosin to the ribosome, preventing the formation of the mRNA-aminoacyl-tRNA-ribosome complex. The structure of one representative sequence from this family, NovP, shows it to be an S-adenosyl-l-methionine-dependent O-methyltransferase that catalyzes the penultimate step in the biosynthesis of the aminocoumarin antibiotic novobiocin. Specifically, it methylates at 4-OH of the noviose moiety, and the resultant methoxy group is important for the potency of the mature antibiotic. It is likely that the key structural features of NovP are common to the rest of the family and include: a helical \'lid\' region that gates access to the co-substrate binding pocket and an active centre that contains a 3-Asp putative metal binding site. A further conserved Asp probably acts as the general base that initiates the reaction by de-protonating the 4-OH group of the noviose unit.','pfam05711','TylF'),(957613,'gnl|CDD|226450',_binary 'Mu-like prophage protein [General function prediction only].','COG3941','COG3941'),(957627,'gnl|CDD|183178',_binary 'DNA-damage-inducible protein D; Provisional.','PRK11525','dinD'),(957654,'gnl|CDD|309835',_binary 'Bacteriophage HK97-gp10, putative tail-component.  This family of proteins is found in the caudovirales. It may be a tail component.','pfam04883','HK97-gp10_like'),(957854,'gnl|CDD|134000',_binary 'putative RNA ligase.','PHA02142','PHA02142'),(957865,'gnl|CDD|313836',_binary 'Protein of unknown function (DUF2510).  This is family of proteins conserved in Actinobacteria. Many members are annotated as putative membrane proteins but this could not be confirmed.','pfam10708','DUF2510'),(957870,'gnl|CDD|311451',_binary 'Recombinase.  This domain is usually found associated with pfam00239 in putative integrases/recombinases of mobile genetic elements of diverse bacteria and phages.','pfam07508','Recombinase'),(957883,'gnl|CDD|316458',_binary 'Helix-turn-helix domain.  This helix-turn-helix domain is often found in transferases and is likely to be DNA-binding.','pfam13936','HTH_38'),(957884,'gnl|CDD|225635',_binary 'Plasmid maintenance system antidote protein [General function prediction only].','COG3093','VapI'),(957886,'gnl|CDD|316164',_binary 'Helix-turn-helix of DDE superfamily endonuclease.  This domain is the probable DNA-binding region of transposase enzymes, necessary for efficient DNA transposition. Most of the members derive from the IS superfamily IS5 and rather fewer from IS4.','pfam13613','HTH_Tnp_4'),(957952,'gnl|CDD|214007',_binary 'Uncharacterized enzymes similar to novel bacterial esterase that cleaves esters on halogenated cyclic compounds.  This family contains uncharacterized proteins similar to a novel bacterial esterase (Alcaligenes esterase 713) with the alpha/beta hydrolase fold but does not contain the GXSXXG pentapeptide around the active site serine residue as commonly seen in other enzymes of this class. Esterase 713 shows negligible sequence homology to other esterase and lipase enzymes. It is active as a dimer and cleaves esters on halogenated cyclic compounds though its natural substrate is unknown.','cd12808','Esterase_713_like-1'),(957962,'gnl|CDD|274300',_binary 'double-strand break repair protein AddB, alphaproteobacterial type.  AddAB is a system well described in the Firmicutes as a replacement for RecBCD in many prokaryotes for the repair of double stranded break DNA damage. More recently, a distantly related gene pair conserved in many alphaproteobacteria was shown also to function in double-stranded break repair in Rhizobium etli. This family consists of AddB proteins of the alphaproteobacteial type. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02786','conserved_hypothetical_pr'),(957964,'gnl|CDD|314021',_binary 'Protein of unknown function (DUF2786).  This family of proteins has no known function.','pfam10979','DUF2786'),(957974,'gnl|CDD|224638',_binary 'Predicted RNA binding protein (dsRBD-like fold), HicA family    [General function prediction only].','COG1724','COG1724'),(957975,'gnl|CDD|311743',_binary 'HicA toxin of bacterial toxin-antitoxin,.  HicA_toxin is a bacterial family of toxins that act as mRNA interferases. The antitoxin that neutralizes this is family HicB, pfam15919.','pfam07927','HicA_toxin'),(957976,'gnl|CDD|225412',_binary 'Predicted Zn peptidase [Amino acid transport and metabolism].','COG2856','COG2856'),(957977,'gnl|CDD|310599',_binary 'Domain of unknown function (DUF955).  Family of bacterial and viral proteins with undetermined function. A conserved H-E-X-X-H motif is suggestive of a catalytic active site and shows similarity to pfam01435.','pfam06114','DUF955'),(957978,'gnl|CDD|317953',_binary 'Restriction endonuclease fold toxin 5.  A predicted toxin of the restriction endonuclease fold present in bacterial polymorphic toxin systems. In bacterial polymorphic toxin systems, the toxin is exported by the type 2, type 5, type 6, or PrsW-peptidase dependent secretion system. Versions of this domain are also found in caudoviruses.','pfam15648','Tox-REase-5'),(957979,'gnl|CDD|310034',_binary 'TM2 domain.  This family is composed of a pair of transmembrane alpha helices connected by a short linker. The function of this domain is unknown, however it occurs in a wide range or protein contexts.','pfam05154','TM2'),(957981,'gnl|CDD|314518',_binary 'Protein of unknown function (DUF3263).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11662','DUF3263'),(958023,'gnl|CDD|227000',_binary 'Predicted phage phi-C31 gp36 major capsid-like protein [General function prediction only].','COG4653','COG4653'),(958024,'gnl|CDD|119346',_binary 'ASC-1 homology domain, ASC-1-like subfamily. The ASCH domain, a small beta-barrel domain found in all three kingdoms of life, resembles the RNA-binding PUA domain and may also interact with RNA. ASCH has been proposed to function as an RNA-binding domain during coactivation, RNA-processing and the regulation of prokaryotic translation. The domain has been named after the ASC-1 protein, the activating signal cointegrator 1 or thyroid hormone receptor interactor protein 4 (TRIP4). ASC-1 is conserved in many eukaryotes and has been suggested to participate in a protein complex that interacts with RNA. It has been shown that ASC-1 mediates the interaction between various transciption factors and the basal transcriptional machinery.','cd06554','ASCH_ASC-1_like'),(958031,'gnl|CDD|311448',_binary 'Protein of unknown function (DUF5131).  This is a family of bacterial and phage proteins of unknown function. There are three highly conserved cysteine residues in the disposition, Cx6Cxxc, amongst many highly conserved residues.','pfam07505','DUF5131'),(958033,'gnl|CDD|226846',_binary 'Bacteriophage protein gp37 [Function unknown].','COG4422','COG4422'),(958035,'gnl|CDD|275264',_binary 'radical SAM mobile pair protein A.  Members of this family are the upstream member (A) of a pair of tandem-encoded radical SAM enzymes. Most of these radical SAM gene pairs have an additional upstream regulatory gene in the MarR family. Examples of high sequence identity (over 96 percent) from cassettes in several Treponema species of the oral cavity to those in multiple Firmicutes in the gut microbiome suggest recent lateral gene transfer, as might be expected for antibiotic resistance genes. The function is unknown.','TIGR04471','hypothetical_protein_ALIP'),(958040,'gnl|CDD|223676',_binary 'Predicted PP-loop superfamily ATPase [General function prediction only].','COG0603','COG0603'),(958041,'gnl|CDD|238953',_binary 'ExsB is a transcription regulator related protein. It is a subfamily of a Adenosine nucleotide binding superfamily of proteins. This protein family is represented by a single member in nearly every completed large (> 1000 genes) prokaryotic genome. In Rhizobium meliloti, a species in which the exo genes make succinoglycan, a symbiotically important exopolysaccharide, exsB is located nearby and affects succinoglycan levels, probably through polar effects on exsA expression or the same polycistronic mRNA. In Arthrobacter viscosus, the homologous gene is designated ALU1 and is associated with an aluminum tolerance phenotype. The function is unknown.','cd01995','ExsB'),(958042,'gnl|CDD|284031',_binary 'Queuosine biosynthesis protein QueC.  This family of proteins participate in the biosynthesis of 7-carboxy-7-deazaguanine. They catalyze the conversion of 7-deaza-7-carboxyguanine to preQ0.','pfam06508','QueC'),(958043,'gnl|CDD|129461',_binary 'queuosine biosynthesis protein QueC.  Members of this protein family are QueC, involved in synthesizing pre-Q0 from GTP en route to tRNA modification with queuosine. This protein family is represented by a single member in nearly every completed large (> 1000 genes) prokaryotic genome. In Rhizobium meliloti, the gene was designated exsB, possibly because of polar effects on exsA expression in a shared polycistronic mRNA. In Arthrobacter viscosus, the homologous gene was designated ALU1 and was associated with an aluminum tolerance phenotype. [Unknown function, General].','TIGR00364','7-cyano-7-deazaguanine_sy'),(958044,'gnl|CDD|182967',_binary 'queuosine biosynthesis protein QueC; Provisional.','PRK11106','PRK11106'),(958045,'gnl|CDD|307413',_binary '6-pyruvoyl tetrahydropterin synthase.  6-Pyruvoyl tetrahydrobiopterin synthase catalyzes the conversion of dihydroneopterin triphosphate to 6-pyruvoyl tetrahydropterin, the second of three enzymatic steps in the synthesis of tetrahydrobiopterin from GTP. The functional enzyme is a hexamer of identical subunits.','pfam01242','PTPS'),(958046,'gnl|CDD|223792',_binary '6-pyruvoyl-tetrahydropterin synthase [Coenzyme metabolism].','COG0720','COG0720'),(958047,'gnl|CDD|274547',_binary 'queuosine biosynthesis protein QueD.  Members of this protein family, closely related to eukaryotic 6-pyruvoyl tetrahydrobiopterin synthase enzymes, are the QueD protein of queuosine biosynthesis. Queuosine is a hypermodified base in the wobble position of tRNAs for Tyr, His, Asp, and Asn in many species. This modification, although widespread, appears not to be important for viability. The queuosine precursor made by this enzyme may be converted instead to archeaosine as in some Archaea. [Protein synthesis, tRNA and rRNA base modification].','TIGR03367','queuosine_QueD'),(958048,'gnl|CDD|272868',_binary '6-pyruvoyl tetrahydropterin synthase/QueD family protein.  This model has been downgraded from hypothetical_equivalog to subfamily. The animal enzymes are known to be 6-pyruvoyl tetrahydropterin synthase. The function of the bacterial branch of the sequence lineage had been thought to be the same, but many are now taken to be QueD, and enzyme of queuosine biosynthesis. Queuosine is a hypermodified base in the wobble position of some tRNAs in most species. A new model is built to be the QueD equivalog model. [Protein synthesis, tRNA and rRNA base modification].','TIGR00039','Putative_6-carboxy-5678-t'),(958051,'gnl|CDD|223675',_binary 'Organic radical activating enzymes [Posttranslational modification, protein turnover, chaperones].','COG0602','NrdG'),(958052,'gnl|CDD|274545',_binary '7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE.  This uncharacterized enzyme, designated QueE, participates in the biosynthesis, from GTP, of 7-cyano-7-deazaguanosine, also called preQ0 because in many species it is a precursor of queuosine. In most Archaea, it is instead the precursor of a different tRNA modified base, archaeosine. [Protein synthesis, tRNA and rRNA base modification].','TIGR03365','Bsubt_queE'),(958053,'gnl|CDD|188478',_binary 'putative 7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE, clostridial.  Members of this radical SAM domain protein family appear to be the Clostridial form of the queuosine biosynthesis protein QueE. QueE is involved in making preQ0 (7-cyano-7-deazaquanine), a precursor of both the bacterial/eukaryotic modified tRNA base queuosine and the archaeal modified base archaeosine. Members occur in preQ0 operons species that lack members of related protein family TIGR03365. [Protein synthesis, tRNA and rRNA base modification].','TIGR03963','rSAM_QueE_Clost'),(958054,'gnl|CDD|275126',_binary 'putative 7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE.  Members of this radical SAM domain protein family appear to be the E. coli form of the queuosine biosynthesis protein QueE. QueE is involved in making preQ0 (7-cyano-7-deazaquanine), a precursor of both the bacterial/eukaryotic modified tRNA base queuosine and the archaeal modified base archaeosine. Members occur in species that lack known forms of QueE but usually are not found in queuosine biosynthesis operons. Members of this family tend to form bi-directional best hit matches to members of known (TIGR03365) and putative (TIGR03963) QueE families from other lineages.','TIGR04322','organic_radical_activatin'),(958055,'gnl|CDD|275301',_binary '7-carboxy-7-deazaguanine synthase, Cx14CxxC type.  In the pathway of 7-cyano-7-deazaquanine (preQ0) biosynthesis, the radical SAM enzyme QueE is quite variable. This model describes a variant form in which the three-Cys motif that binds the signature 4Fe-4S cluster takes the form Cx14CxxC, as in Burkholderia multivorans ATCC 17616. The crystal structure is known.','TIGR04508','queE_Cx14CxxC'),(958056,'gnl|CDD|315957',_binary '4Fe-4S single cluster domain.','pfam13394','Fer4_14'),(958057,'gnl|CDD|275145',_binary 'putative 7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE, gammaproteobacterial type.  Members of this radical SAM domain protein family appear to be a form of the queuosine biosynthesis protein QueE. QueE is involved in making preQ0 (7-cyano-7-deazaquanine), a precursor of both the bacterial/eukaryotic modified tRNA base queuosine and the archaeal modified base archaeosine. Members occur in preQ0 operons species that lack members of related protein family TIGR03365.','TIGR04349','Radical_SAM_domain_protei'),(958058,'gnl|CDD|315919',_binary '4Fe-4S single cluster domain.  This family includes proteins containing domains which bind to iron-sulfur clusters. Members include bacterial ferredoxins, various dehydrogenases, and various reductases. The structure of the domain is an alpha-antiparallel beta sandwich.','pfam13353','Fer4_12'),(958059,'gnl|CDD|281979',_binary 'Radical SAM superfamily.  Radical SAM proteins catalyze diverse reactions, including unusual methylations, isomerisation, sulphur insertion, ring formation, anaerobic oxidation and protein radical formation.','pfam04055','Radical_SAM'),(958060,'gnl|CDD|224101',_binary 'Pyruvate-formate lyase-activating enzyme [Posttranslational modification, protein turnover, chaperones].','COG1180','PflA'),(958061,'gnl|CDD|223609',_binary 'Predicted Fe-S oxidoreductases [General function prediction only].','COG0535','COG0535'),(958062,'gnl|CDD|274164',_binary 'anaerobic ribonucleoside-triphosphate reductase activating protein.  This enzyme is a member of the radical-SAM family (pfam04055). It is often gene clustered with the class III (anaerobic) ribonucleotide triphosphate reductase (NrdD, TIGR02487) and presumably fulfills the identical function as NrdG, which utilizes S-adenosyl methionine, an iron-sulfur cluster and a reductant (dihydroflavodoxin) to produce a glycine-centered radical in NrdD. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism, Protein fate, Protein modification and repair].','TIGR02495','RADICAL-ACTIVATING_ENZYME'),(958063,'gnl|CDD|275125',_binary 'spiro-SPASM protein.  This three-domain protein is restricted to the spirochetes and widely distributed (excepting Borrelia). It has a conserved C-terminal SPASM domain, a 4Fe-4S binding domain shared by a number of peptide-modifying and heme-modifying radical SAM proteins. It has a central radical SAM domain, although half the members have lost the signature 4Fe-4S-binding Cys residues, fail to register with the radical SAM domain definition of pfam04055, and must be considered pseudo-SAM proteins. PSI-BLAST shows a relationship between the N-terminal domain and various predicted glycosyltransferases (e.g. Bacillus subtilis SpsF) and cytidyltransferases. In some Treponema species, this protein appears to split into two tandem genes.','TIGR04321','hypothetical_protein_Tres'),(958064,'gnl|CDD|274946',_binary 'putative heme d1 biosynthesis radical SAM protein NirJ1.  Members of this radical SAM protein subfamily, designated NirJ1, occur in genomic contexts with a paralog NirJ2 and with other nitrite reductase operon genes associated with heme d1 biosynthesis, as in Heliobacillus mobilis and Heliophilum fasciatum. NirJ1 is presumed by bioinformatics analysis (Xiong, et al.) to be a heme d1 biosynthesis protein by context, perhaps involved in conversions of acetate groups to methyl groups in conversion from uroporphyrinogen III. A very closely related protein, involved in alternative heme b biosynthesis, occurs in Desulfovibrio and in methanogens. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR04054','rSAM_NirJ1'),(958073,'gnl|CDD|224123',_binary 'Superfamily II helicase, archaea-specific [General function prediction only].','COG1202','COG1202'),(958075,'gnl|CDD|274978',_binary 'DNA phosphorothioation system restriction enzyme.  The DNA phosphorothioate modification system dnd (DNA instability during electrophoresis) recently has been shown to provide a modification essential to a restriction system. This protein family was detected by Partial Phylogenetic Profiling as linked to dnd, and its members usually are clustered with the dndABCDE genes.','TIGR04095','type_III_restriction_prot'),(958076,'gnl|CDD|224036',_binary 'ERCC4-like helicases [DNA replication, recombination, and repair].','COG1111','MPH1'),(958077,'gnl|CDD|223588',_binary 'Superfamily II DNA helicase [DNA replication, recombination, and repair].','COG0514','RecQ'),(958092,'gnl|CDD|307403',_binary 'GTP cyclohydrolase I.  This family includes GTP cyclohydrolase enzymes and a family of related bacterial proteins.','pfam01227','GTP_cyclohydroI'),(958094,'gnl|CDD|236472',_binary 'GTP cyclohydrolase I; Provisional.','PRK09347','folE'),(958097,'gnl|CDD|223379',_binary 'GTP cyclohydrolase I [Coenzyme metabolism].','COG0302','FolE'),(958099,'gnl|CDD|238349',_binary 'GTP cyclohydrolase I (GTP-CH-I) catalyzes the conversion of GTP into dihydroneopterin triphosphate.  The enzyme product is the precursor of tetrahydrofolate in eubacteria, fungi, and plants and of the folate analogs in methanogenic bacteria.  In vertebrates and insects it is the biosynthtic precursor of tetrahydrobiopterin (BH4) which is involved in the formation of catacholamines, nitric oxide, and the stimulation of T lymphocytes. The biosynthetic reaction of BH4 is controlled by a regulatory protein GFRP which mediates feedback inhibition of GTP-CH-I by BH4.  This inhibition is reversed by phenylalanine. The decameric GTP-CH-I forms a complex with two pentameric GFRP in the presence of phenylalanine or a combination of GTP and BH4, respectively.','cd00642','GTP_cyclohydro1'),(958101,'gnl|CDD|237149',_binary 'GTP cyclohydrolase I; Reviewed.','PRK12606','PRK12606'),(958103,'gnl|CDD|129173',_binary 'GTP cyclohydrolase I.  alternate names: Punch (Drosophila),GTP cyclohydrolase I (EC 3.5.4.16) catalyzes the biosynthesis of formic acid and dihydroneopterin triphosphate from GTP. This reaction is the first step in the biosynthesis of tetrahydrofolate in prokaryotes, of tetrahydrobiopterin in vertebrates, and of pteridine-containing pigments in insects. [Biosynthesis of cofactors, prosthetic groups, and carriers, Folic acid].','TIGR00063','GTP_cyclohydrolase_1'),(958105,'gnl|CDD|215549',_binary 'GTP cyclohydrolase I; Provisional.','PLN03044','PLN03044'),(958107,'gnl|CDD|240434',_binary 'GTP cyclohydrolase I; Provisional.','PTZ00484','PTZ00484'),(958109,'gnl|CDD|215290',_binary 'GTP cyclohydrolase I.','PLN02531','PLN02531'),(958116,'gnl|CDD|129331',_binary 'crossover junction endodeoxyribonuclease RuvC.  Endonuclease that resolves Holliday junction intermediates in genetic recombination. The active form of the protein is a dimer. Structure studies reveals that the catalytic center, comprised of four acidic residues, lies at the bottom of a cleft that fits a DNA duplex. The model hits a single Synechocystis PCC6803 protein at a score of 30, below the trusted cutoff, that appears orthologous and may act as authentic RuvC. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00228','Crossover_junction_endode'),(958128,'gnl|CDD|225843',_binary 'Glycosyltransferase involved in LPS biosynthesis [Cell envelope biogenesis, outer membrane].','COG3306','COG3306'),(958133,'gnl|CDD|197689',_binary 'DNA-binding domain in plant proteins such as APETALA2 and EREBPs.','smart00380','AP2'),(958134,'gnl|CDD|310369',_binary 'Phage tail protein.  This family consists of several Siphovirus and other phage tail component proteins as well as some bacterial proteins of unknown function.','pfam05709','Sipho_tail'),(958145,'gnl|CDD|238082',_binary 'Beta clamp domain.  The beta subunit (processivity factor) of DNA polymerase III holoenzyme, refered to as the beta clamp, forms a ring shaped dimer that encircles dsDNA (sliding clamp) in bacteria.  The beta-clamp is structurally similar to the trimeric ring formed by PCNA (found in eukaryotes and archaea) and the processivity factor (found in bacteriophages T4 and RB69).  This structural correspondence further substantiates the mechanistic connection between eukaryotic and prokaryotic DNA replication that has been suggested on biochemical grounds. .','cd00140','beta_clamp'),(958146,'gnl|CDD|235541',_binary 'DNA polymerase III subunit beta; Validated.','PRK05643','PRK05643'),(958147,'gnl|CDD|214686',_binary 'DNA polymerase III beta subunit.','smart00480','POL3Bc'),(958148,'gnl|CDD|236089',_binary 'DNA polymerase III subunit beta; Validated.','PRK07761','PRK07761'),(958149,'gnl|CDD|273204',_binary 'DNA polymerase III, beta subunit.  All proteins in this family for which functions are known are components of the DNA polymerase III complex (beta subunit). This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00663','DNA_polymerase_III_subuni'),(958150,'gnl|CDD|223665',_binary 'DNA polymerase sliding clamp subunit (PCNA homolog) [DNA replication, recombination, and repair].','COG0592','DnaN'),(958151,'gnl|CDD|184904',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14940','PRK14940'),(958152,'gnl|CDD|184906',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14942','PRK14942'),(958153,'gnl|CDD|135998',_binary 'DNA polymerase III subunit beta; Validated.','PRK06673','PRK06673'),(958154,'gnl|CDD|184909',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14945','PRK14945'),(958162,'gnl|CDD|308032',_binary 'ParB-like nuclease domain.','pfam02195','ParBc'),(958163,'gnl|CDD|319246',_binary 'ParB N-terminal domain and sulfiredoxin protein-related families.  The ParB N-terminal domain/Sulfiredoxin (Srx) superfamily contains proteins with diverse activities. Many of the families are involved in segregation and competition between plasmids and chromosomes. Several families share similar activities with the N-terminal domain of ParB (Spo0J in Bacillus subtilis), a DNA-binding component of the prokaryotic parABS partitioning system. Also within this superfamily is sulfiredoxin (Srx; reactivator of oxidatively inactivated 2-cys peroxiredoxins), RepB N-terminal domain (plasmid segregation replication protein B like protein), nucleoid occlusion protein, KorB N-terminal domain partition protein of low copy number plasmid RK2, irbB (immunoglobulin-binding regulator that activates eib genes), N-terminal domain of sopB protein (promotes proper partitioning of F1 plasmid), fertility inhibition factors OSA and FiwA,DNA sulfur modification protein DndB, and a ParB-like toxin domain. Other activities includes a StrR (regulator in the streptomycin biosynthetic gene cluster), and a family containing a Pyrococcus furiosus nuclease and putative transcriptional regulators sbnI (Staphylococcus aureus siderophore biosynthetic gene cluster ). Nuclease activity has also been reported in Arabidopsis Srx.','cd16387','ParB_N_Srx'),(958165,'gnl|CDD|319251',_binary 'Thermus thermophilus stage 0 sporulation protein J-like N-terminal domain, ParB family member.  Spo0J (stage 0 sporulation protein J) is a ParB family member, a critical component of the ParABS-type bacterial chromosome segregation system. The Spo0J N-terminal region acts in protein-protein interaction and is adjacent to the DNA-binding domain that binds to parS sites. Two Spo0J bind per parS site, and Spo0J interacts with neighbors via the N-terminal domain to form oligomers via an Arginine-rich patch (RRXR).  This superfamily represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16393','SPO0J_N'),(958166,'gnl|CDD|272946',_binary 'ParB/RepB/Spo0J family partition protein.  This model represents the most well-conserved core of a set of chromosomal and plasmid partition proteins related to ParB, including Spo0J, RepB, and SopB. Spo0J has been shown to bind a specific DNA sequence that, when introduced into a plasmid, can serve as partition site. Study of RepB, which has nicking-closing activity, suggests that it forms a transient protein-DNA covalent intermediate during the strand transfer reaction.','TIGR00180','Probable_chromosome-parti'),(958167,'gnl|CDD|319264',_binary 'ParB N-terminal, parA-binding, -like domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16407','ParB_N_like'),(958168,'gnl|CDD|224392',_binary 'Stage 0 sporulation protein J (antagonist of Soj) containing ParB-like nuclease domain [Transcription].','COG1475','Spo0J'),(958169,'gnl|CDD|319265',_binary 'ParB N-terminal, parA -binding, -like domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16408','ParB_N_like'),(958170,'gnl|CDD|319256',_binary 'ParB-like partition protein of low copy number plasmid RK2, N-terminal domain and related domains.  KorB, a member of the ParB like family, is present on the low copy number, broad host range plasmid RK2. KorB encodes a gene product involved in segregation of RK2 and acts as a transcriptional regulator, down-regulating at least 6 RK2 operons. KorB binds RNA polymerase and acts cooperatively with several co-repressors in modulating transcription. KorB is comprised of 3 domains, including a beta-strand C-terminal domain similar to SH3 domains and an alpha helical central domain that interacts with operator DNA. In ParB of P1 and SopB of F, the N-terminal region is responsible for interaction with the parA component. However, korB interaction with the RK2 parA-equivalent IncC has been mapped to the central HTH motif.  This family is related to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16398','KorB_N_like'),(958171,'gnl|CDD|275105',_binary 'nucleoid occlusion protein.  This model describes nucleoid occlusion protein, a close homolog to ParB chromosome partitioning proteins including Spo0J in Bacillus subtilis. Its gene often is located near the gene for the Spo0J ortholog. This protein bind a specific DNA sequence and blocks cytokinesis from happening until chromosome segregation is complete.','TIGR04285','parB-like_partition_prote'),(958172,'gnl|CDD|319252',_binary 'N-terminal domain of sopB protein, which promotes proper partitioning of F1 plasmid.  Escherichia coli SopB acts in the equitable partitioning of the F plasmid in the SopABC system. SopA binds to the sopAB promoter, while SopB binds SopC and helps stimulate polymerization of SopA in the presence of ATP and Mg(II). Mutation of SopA inhibits proper plasmid segregation. This N-terminal domain is related to the ParB N-terminal domain of bacterial and plasmid parABS partitioning systems, which binds parA.','cd16394','sopB_N'),(958173,'gnl|CDD|319260',_binary 'ParB N-terminal-like domain, some attached to C-terminal S-adenosylmethionine-dependent methyltransferase.  This family represents domains related to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, fused to a variety of C-terminal domains, including S-adenosylmethionine-dependent methyltransferase-like domains. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16403','ParB_N_like_MT'),(958174,'gnl|CDD|319267',_binary 'ParB N-terminal, parA-binding, -like domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16410','ParB_N_like'),(958175,'gnl|CDD|319254',_binary 'nucleoid occlusion protein, N-terminal domain, and related domains of the ParB partitioning protein family.  Nucleoid occlusion protein has been shown in Bacillus subtilis to bind to specific DNA sequences on the chromosome (Noc-binding DNA sequences, NBS), inhibiting cell division near the nucleoid and thereby protecting the chromosome. This N-terminal domain is related to the N-terminal domain of ParB/repB partitioning system proteins.','cd16396','Noc_N'),(958176,'gnl|CDD|184353',_binary 'plasmid partitioning protein; Provisional.','PRK13832','PRK13832'),(958178,'gnl|CDD|319266',_binary 'ParB N-terminal-like domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16409','ParB_N_like'),(958181,'gnl|CDD|177094',_binary 'Ycf46; Provisional.','CHL00195','ycf46'),(958182,'gnl|CDD|223540',_binary 'ATPases of the AAA+ class [Posttranslational modification, protein turnover, chaperones].','COG0464','SpoVK'),(958184,'gnl|CDD|306510',_binary 'ATPase family associated with various cellular activities (AAA).  AAA family proteins often perform chaperone-like functions that assist in the assembly, operation, or disassembly of protein complexes.','pfam00004','AAA'),(958185,'gnl|CDD|130309',_binary '26S proteasome subunit P45 family.  Many proteins may score above the trusted cutoff because an internal.','TIGR01242','Proteasome-activating_nuc'),(958186,'gnl|CDD|273521',_binary 'AAA family ATPase, CDC48 subfamily.  This subfamily of the AAA family ATPases includes two members each from three archaeal species. It also includes yeast CDC48 (cell division control protein 48) and the human ortholog, transitional endoplasmic reticulum ATPase (valosin-containing protein). These proteins in eukaryotes are involved in the budding and transfer of membrane from the transitional endoplasmic reticulum to the Golgi apparatus.','TIGR01243','Cell_division_cycle_prote'),(958188,'gnl|CDD|224143',_binary 'ATP-dependent 26S proteasome regulatory subunit [Posttranslational modification, protein turnover, chaperones].','COG1222','RPT1'),(958189,'gnl|CDD|182683',_binary 'ATP-dependent metalloprotease; Reviewed.','PRK10733','hflB'),(958190,'gnl|CDD|273520',_binary 'ATP-dependent metalloprotease FtsH.  HflB(FtsH) is a pleiotropic protein required for correct cell division in bacteria. It has ATP-dependent zinc metalloprotease activity. It was formerly designated cell division protein FtsH. [Cellular processes, Cell division, Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR01241','ATP-dependent_zinc_metall'),(958191,'gnl|CDD|224144',_binary 'Predicted ATPase (AAA+ superfamily) [General function prediction only].','COG1223','COG1223'),(958192,'gnl|CDD|214386',_binary 'cell division protein; Validated.','CHL00176','ftsH'),(958193,'gnl|CDD|179699',_binary 'proteasome-activating nucleotidase; Provisional.','PRK03992','PRK03992'),(958194,'gnl|CDD|200312',_binary 'proteasome ATPase.  In the Actinobacteria, as shown for Mycobacterium tuberculosis, some proteins are modified by ligation between an epsilon-amino group of a lysine side chain and the C-terminal carboxylate of the ubiquitin-like protein Pup. This modification leads to protein degradation by the archaeal-like proteasome found in the Actinobacteria. Members of this protein family belong to the AAA family of ATPases and tend to be clustered with the genes for Pup, the Pup ligase PafA, and structural components of the proteasome. This protein forms hexameric rings with ATPase activity. [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR03689','pup_AAA'),(958195,'gnl|CDD|223541',_binary 'ATP-dependent Zn proteases [Posttranslational modification, protein turnover, chaperones].','COG0465','HflB'),(958196,'gnl|CDD|240423',_binary '26S protease regulatory subunit 6B-like protein; Provisional.','PTZ00454','PTZ00454'),(958199,'gnl|CDD|185575',_binary '26 proteosome regulatory subunit 4-like protein; Provisional.','PTZ00361','PTZ00361'),(958205,'gnl|CDD|289462',_binary 'S-adenosyl-L-methionine methyltransferase.  This domain is found in bacterial proteins. The structure of the proteins in this family suggest that they function as a methyltransferase.','pfam12692','Methyltransf_17'),(958206,'gnl|CDD|133045',_binary 'CESA_like is  the cellulose synthase superfamily.  The cellulose synthase (CESA) superfamily includes a wide variety of glycosyltransferase family 2 enzymes that share the common characteristic of catalyzing the elongation of polysaccharide chains. The members include cellulose synthase catalytic subunit, chitin synthase, glucan biosynthesis protein and other families of CESA-like proteins. Cellulose synthase catalyzes the polymerization reaction of cellulose, an aggregate of unbranched polymers of beta-1,4-linked glucose residues in  plants, most algae, some bacteria and fungi, and even some animals. In bacteria, algae and lower eukaryotes, there is a second unrelated type of cellulose synthase (Type II), which produces acylated cellulose, a derivative of cellulose. Chitin synthase catalyzes the incorporation of GlcNAc from substrate UDP-GlcNAc into chitin, which is a linear homopolymer of beta-(1,4)-linked GlcNAc residues and Glucan Biosynthesis protein catalyzes the elongation of beta-1,2 polyglucose chains of Glucan.','cd06423','CESA_like'),(958209,'gnl|CDD|225848',_binary 'Predicted transcriptional regulator [Transcription].','COG3311','AlpA'),(958214,'gnl|CDD|293550',_binary 'Putative lactococcus lactis phage r1t holin.  Phage_r1t_holin is a family of putative phage r1t holins from lactococcus. these holins carry two hydrophobic putative TMs separated by a short beta-turn region.','pfam16945','Phage_r1t_holin'),(958220,'gnl|CDD|306651',_binary 'Chitinase class I.','pfam00182','Glyco_hydro_19'),(958223,'gnl|CDD|316006',_binary 'Cro/C1-type HTH DNA-binding domain.  This is a helix-turn-helix domain that probably binds to DNA.','pfam13443','HTH_26'),(958256,'gnl|CDD|311386',_binary 'Domain of unknown function (DUF1508).  This family represents a series of bacterial domains of unknown function of around 50 residues in length. Members of this family are often found as tandem repeats and in some cases represent the whole protein. All member proteins are described as being hypothetical.','pfam07411','DUF1508'),(958262,'gnl|CDD|223971',_binary 'Predicted DNA modification methylase [DNA replication, recombination, and repair].','COG1041','COG1041'),(958408,'gnl|CDD|183880',_binary 'serine/threonine-protein kinase; Reviewed.','PRK13184','pknD'),(958457,'gnl|CDD|306613',_binary 'Carboxylesterase family.','pfam00135','COesterase'),(958458,'gnl|CDD|236660',_binary 'acetyl esterase; Provisional.','PRK10162','PRK10162'),(958459,'gnl|CDD|238191',_binary 'Esterases and lipases (includes fungal lipases, cholinesterases, etc.)  These enzymes act on carboxylic esters (EC: 3.1.1.-). The catalytic apparatus involves three residues (catalytic triad): a serine, a glutamate or aspartate and a histidine.These catalytic residues are responsible for the nucleophilic attack on the carbonyl carbon atom of the ester bond. In contrast with other alpha/beta hydrolase fold family members, p-nitrobenzyl esterase and acetylcholine esterase have a Glu instead of Asp at the active site carboxylate.','cd00312','Esterase_lipase'),(958466,'gnl|CDD|224291',_binary 'Intein/homing endonuclease [DNA replication, recombination, and repair].','COG1372','COG1372'),(958494,'gnl|CDD|225949',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG3415','COG3415'),(958499,'gnl|CDD|238291',_binary 'AE_Prim_S_like: primase domain similar to that found in the small subunit of archaeal and eukaryotic (A/E) DNA primases. The replication machineries of A/Es are distinct from that of bacteria. Primases are DNA-dependent RNA polymerases which synthesis the short RNA primers required for DNA replication. In eukaryotes, this small catalytically active primase subunit (p50) and a larger primase subunit (p60), referred to jointly as the core primase, associate with the B subunit and the DNA polymerase alpha subunit in a complex, called Pol alpha-pri. In addition to its catalytic role in replication, eukaryotic DNA primase may play a role in coupling replication to DNA damage repair and in checkpoint control during S phase. Pfu41 and Pfu46 comprise the primase complex of the archaea Pyrococcus furiosus; these proteins have sequence identity to the eukaryotic p50 and p60 primase proteins respectively. Pfu41 preferentially uses dNTPs as substrate. Pfu46 regulates the primase activity of Pfu41. Also found in this group is the primase-polymerase (primpol) domain of replicases from archaeal plasmids including the ORF904 protein of pRN1 from Sulfolobus islandicus (pRN1 primpol). The pRN1 primpol domain exhibits DNA polymerase and primase activities; a cluster of active site residues (three acidic residues, and a histidine) is required for both these activities. The pRN1 primpol primase activity prefers dNTPs to rNTPs; however incorporation of dNTPs requires rNTP as cofactor. This group also includes the Pol domain of bacterial LigD proteins such Mycobacterium tuberculosis (Mt)LigD. MtLigD contains an N-terminal Pol domain, a central phosphoesterase module, and a C-terminal ligase domain. LigD Pol plays a role in non-homologous end joining (NHEJ)-mediated repair of DNA double-strand breaks (DSB) in vivo, perhaps by filling in short 5\'-overhangs with ribonucleotides; the filled in termini would be sealed by the associated LigD ligase domain. The MtLigD Pol domain is stimulated by manganese, is error-prone, and prefers adding rNTPs to dNTPs in vitro.','cd00525','AE_Prim_S_like'),(958520,'gnl|CDD|224726',_binary 'Predicted transcription factor, homolog of eukaryotic MBF1 [Transcription].','COG1813','COG1813'),(958523,'gnl|CDD|225495',_binary 'Predicted transcriptional regulator [Transcription].','COG2944','COG2944'),(958529,'gnl|CDD|133016',_binary 'ExoA is involved in the biosynthesis of succinoglycan.  Succinoglycan Biosynthesis Protein ExoA catalyzes the formation of a beta-1,3 linkage of the second sugar (glucose) of the succinoglycan with the galactose on the lipid carrie. Succinoglycan is an acidic exopolysaccharide that is important for invasion of the nodules. Succinoglycan is a high-molecular-weight polymer composed of repeating octasaccharide units. These units are synthesized on membrane-bound isoprenoid lipid carriers, beginning with galactose followed by seven glucose molecules, and modified by the addition of acetate, succinate, and pyruvate. ExoA is a membrane protein with a transmembrance domain at c-terminus.','cd02525','Succinoglycan_BP_ExoA'),(958560,'gnl|CDD|316372',_binary 'Methyltransferase domain.  This family appears to have methyltransferase activity.','pfam13847','Methyltransf_31'),(958588,'gnl|CDD|173412',_binary 'MAEBL; Provisional.','PTZ00121','PTZ00121'),(958596,'gnl|CDD|224117',_binary 'Chromosome segregation ATPases [Cell division and chromosome partitioning].','COG1196','Smc'),(958602,'gnl|CDD|274008',_binary 'chromosome segregation protein SMC, common bacterial type.  SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins].','TIGR02168','Chromosome_partition_prot'),(958607,'gnl|CDD|223496',_binary 'ATPase involved in DNA repair [DNA replication, recombination, and repair].','COG0419','SbcC'),(958612,'gnl|CDD|274009',_binary 'chromosome segregation protein SMC, primarily archaeal type.  SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins].','TIGR02169','chromosome_segregation_pr'),(958614,'gnl|CDD|235175',_binary 'chromosome segregation protein; Provisional.','PRK03918','PRK03918'),(958622,'gnl|CDD|179385',_binary 'chromosome segregation protein; Provisional.','PRK02224','PRK02224'),(958630,'gnl|CDD|224259',_binary 'Uncharacterized archaeal coiled-coil protein [Function unknown].','COG1340','COG1340'),(958631,'gnl|CDD|310534',_binary 'Laminin Domain I.  coiled-coil structure. It has been suggested that the domains I and II from laminin A, B1 and B2 may come together to form a triple helical coiled-coil structure.','pfam06008','Laminin_I'),(958632,'gnl|CDD|308206',_binary 'RecF/RecN/SMC N terminal domain.  This domain is found at the N terminus of SMC proteins. The SMC (structural maintenance of chromosomes) superfamily proteins have ATP-binding domains at the N- and C-termini, and two extended coiled-coil domains separated by a hinge in the middle. The eukaryotic SMC proteins form two kind of heterodimers: the SMC1/SMC3 and the SMC2/SMC4 types. These heterodimers constitute an essential part of higher order complexes, which are involved in chromatin and DNA dynamics. This family also includes the RecF and RecN proteins that are involved in DNA metabolism and recombination.','pfam02463','SMC_N'),(958633,'gnl|CDD|316391',_binary 'Trichohyalin-plectin-homology domain.  This family is a mixtrue of two different families of eukaryotic proteins. Trichoplein or mitostatin, was first defined as a meiosis-specific nuclear structural protein. It has since been linked with mitochondrial movement. It is associated with the mitochondrial outer membrane, and over-expression leads to reduction in mitochondrial motility whereas lack of it enhances mitochondrial movement. The activity appears to be mediated through binding the mitochondria to the actin intermediate filaments (IFs). The family is in the trichohyalin-plectin-homology domain.','pfam13868','TPH'),(958634,'gnl|CDD|318039',_binary 'Domain of unknown function (DUF4686).  This family of proteins is found in eukaryotes. Proteins in this family are typically between 498 and 775 amino acids in length. There is a conserved DLK sequence motif.','pfam15742','DUF4686'),(958635,'gnl|CDD|224495',_binary 'Zn-ribbon protein, possibly nucleic acid-binding [General function prediction only].','COG1579','COG1579'),(958636,'gnl|CDD|317931',_binary 'Ciliary protein causing Leber congenital amaurosis disease.  Lebercilin is a family of eukaryotic ciliary proteins. Mutations in the gene, LCA5, are implicated in the disease Leber congenital amaurosis. In photoreceptors, lebercilin is uniquely localized at the cilium that bridges the inner and outer segments. Lebercilin functions as an integral element of selective protein transport through photoreceptor cilia. Lebercilin specifically interacts with the intraflagellar transport (IFT), and disruption of IFT can lead to Leber congenital amaurosis.','pfam15619','Lebercilin'),(958637,'gnl|CDD|237177',_binary 'phosphodiesterase; Provisional.','PRK12704','PRK12704'),(958638,'gnl|CDD|227355',_binary 'Myosin heavy chain [Cytoskeleton].','COG5022','COG5022'),(958639,'gnl|CDD|314924',_binary 'Protein of unknown function (DUF3584).  This protein is found in bacteria and eukaryotes. Proteins in this family are typically between 943 to 1234 amino acids in length. This family contains a P-loop motif suggesting it is a nucleotide binding protein. It may be involved in replication.','pfam12128','DUF3584'),(958640,'gnl|CDD|310341',_binary 'Protein of unknown function (DUF812).  This family consists of several eukaryotic proteins of unknown function.','pfam05667','DUF812'),(958641,'gnl|CDD|307627',_binary 'Myosin tail.  The myosin molecule is a multi-subunit complex made up of two heavy chains and four light chains it is a fundamental contractile protein found in all eukaryote cell types. This family consists of the coiled-coil myosin heavy chain tail region. The coiled-coil is composed of the tail from two molecules of myosin. These can then assemble into the macromolecular thick filament. The coiled-coil region provides the structural backbone the thick filament.','pfam01576','Myosin_tail_1'),(958659,'gnl|CDD|274765',_binary 'integrating conjugative element protein, PFL_4705 family.  Members of this protein family are found occasionally on plasmids such as the Pseudomonas putida toluene catabolic TOL plasmid pWWO_p085. Usually, however, they are found on the bacterial main chromosome in regions flanked by markers of conjugative transfer and/or transposition. [Mobile and extrachromosomal element functions, Plasmid functions].','TIGR03752','conj_TIGR03752'),(958695,'gnl|CDD|131309',_binary 'integrative and conjugative element protein, VC0181 family.  This uncharacterized protein is found in several Proteobacteria, among them Rhizobium sp. NGR234, Vibrio cholerae, Myxococcus xanthus, and E. coli strain ECOR31. In the latter, it is part of an integrative and conjugative element that is readily induced to excise and circularize.','TIGR02256','Uncharacterized_protein_y'),(958707,'gnl|CDD|311971',_binary 'LGFP repeat.  This 54 amino acid repeat is found in many hypothetical proteins. Several hypothetical proteins from C.glutamicum and C.efficiens along with PS1 protein contain this repeat region. The N-terminus region of PS1 contains an esterase domain which transfers corynomycolic acid. The C-terminus region consists of 4 tandem LGFP repeats. It is hypothesized that the PS1 proteins in Corynebacterium, when associated with the cell wall, may be anchored via the LGFP tandem repeats that may be important for maintaining cell wall integrity [Adindla et al. Comparative and Functional Genomics 2004; 5:2-16]. Deletion of Corynebacterium glutamicum csp1 protein results in a 10-fold increase in the cell volume of the organism and infers the corresponding proteins involvement in the cell shape formation. The secondary structure of each repeat is predicted to comprise two beta-strands and one alpha-helix [Adindla et al. 2004].','pfam08310','LGFP'),(958709,'gnl|CDD|227766',_binary 'Uncharacterized protein potentially involved in peptidoglycan biosynthesis [Cell envelope biogenesis, outer membrane].','COG5479','COG5479'),(958717,'gnl|CDD|130402',_binary 'photosystem I core protein PsaA.  The core proteins of photosystem I are PsaA and PsaB, homologous integral membrane proteins that form a heterodimer. The heterodimer binds the electron-donating chlorophyll dimer P700, as well as chlorophyll, phylloquinone, and 4FE-4S electron acceptors. This model describes PsaA only. [Energy metabolism, Photosynthesis].','TIGR01335','Photosystem_I_P700_chloro'),(958768,'gnl|CDD|308872',_binary 'ADP-ribosyltransferase exoenzyme.  This is a family of bacterial and viral bi-glutamic acid ADP-ribosyltransferases, where, in Aeromonas salmonicida AexT, E403 is the catalytic residue and E401 contributes to the transfer of ADP-ribose to the target protein. In clostridial species it is actin that is being ADP-ribosylated; this result is lethal and dermonecrotic in infected mammals.','pfam03496','ADPrib_exo_Tox'),(958769,'gnl|CDD|238144',_binary 'VIP2; A family of actin-ADP-ribosylating toxin. A member of the Bacillus-prodiced vegetative insecticidal proteins (VIPs) possesses high specificity against the major insect pest, corn rootworms, and belongs to a classs of binary toxins and regulators of biological pathways distinct from classical A-B toxins. A novel family of insecticidal ADP-ribosyltransferses were isolated from Bacillus cereus during vegetative growth, where VIP1 likely targets insect cells and VIP2 ribosylates actin. VIP2 shares significant sequence similarity with enzymatic components of other binary toxins, Clostridium botulinum C2 toxin, C. perfringens iota toxin, C. piroforme toxin, C. piroforme toxin and C. difficile toxin.','cd00233','VIP2'),(958772,'gnl|CDD|107029',_binary 'putative minor structural protein.','PHA01351','PHA01351'),(958784,'gnl|CDD|239385',_binary 'This protein family consists of proteins similar to the cpaE protein of the Caulobacter pilus assembly and the orf4 protein of Actinobacillus pilus formation gene cluster. The function of these proteins are unkown. The Caulobacter pilus assembly contains 7 genes: pilA, cpaA, cpaB, cpaC, cpaD, cpaE and cpaF. These genes are clustered together on chromosome.','cd03111','CpaE_like'),(958836,'gnl|CDD|306760',_binary 'Ribonucleotide reductase, all-alpha domain.','pfam00317','Ribonuc_red_lgN'),(958851,'gnl|CDD|307732',_binary 'Toprim domain.  This is a conserved region from DNA primase. This corresponds to the Toprim domain common to DnaG primases, topoisomerases, OLD family nucleases and RecR proteins. Both DnaG motifs IV and V are present in the alignment, the DxD (V) motif may be involved in Mg2+ binding and mutations to the conserved glutamate (IV) completely abolish DnaG type primase activity. DNA primase EC:2.7.7.6 is a nucleotidyltransferase it synthesizes the oligoribonucleotide primers required for DNA replication on the lagging strand of the replication fork; it can also prime the leading stand and has been implicated in cell division. This family also includes the atypical archaeal A subunit from type II DNA topoisomerases. Type II DNA topoisomerases catalyze the relaxation of DNA supercoiling by causing transient double strand breaks.','pfam01751','Toprim'),(958876,'gnl|CDD|273040',_binary 'CRISPR-associated protein Cas4.  This model represents a family of proteins associated with CRISPR repeats in a wide set of prokaryotic genomes. This scope of this model has been broadened since it was first built to describe an archaeal subset only. The function of the protein is undefined. Distantly related proteins, excluded from this model, include ORFs from Mycobacteriophage D29 and Sulfolobus islandicus filamentous virus and a region of the Schizosaccharomyces pombe DNA replication helicase Dna2p.','TIGR00372','conserved_hypothetical_pr'),(958877,'gnl|CDD|224385',_binary 'CRISPR-associated protein Cas4 (RecB family exonuclease) [Defense    mechanisms].','COG1468','COG1468'),(958919,'gnl|CDD|280281',_binary 'Crossover junction endodeoxyribonuclease RuvC.','pfam02075','RuvC'),(959011,'gnl|CDD|224017',_binary 'Predicted SAM-dependent methyltransferases [General function prediction only].','COG1092','COG1092'),(959066,'gnl|CDD|177364',_binary 'hypothetical protein.','PHA02451','PHA02451'),(959081,'gnl|CDD|315057',_binary 'Protein of unknown function (DUF3626).  This family of proteins is found in bacteria. Proteins in this family are typically between 294 and 374 amino acids in length.','pfam12294','DUF3626'),(959162,'gnl|CDD|239330',_binary 'Arsenate Reductase (ArsC) family, Spx subfamily; Spx is a unique RNA polymerase (RNAP)-binding protein present in bacilli and some mollicutes. It inhibits transcription by binding to the C-terminal domain of the alpha subunit of RNAP, disrupting complex formation between RNAP and certain transcriptional activator proteins like ResD and ComA. In response to oxidative stress, Spx can also activate transcription, making it a general regulator that exerts both positive and negative control over transcription initiation. Spx has been shown to exert redox-sensitive transcriptional control over genes like trxA (TRX) and trxB (TRX reductase), genes that function in thiol homeostasis. This redox-sensitive activity is dependent on the presence of a CXXC motif, present in some members of the Spx subfamily, that acts as a thiol/disulfide switch. Spx has also been shown to repress genes in a sulfate-dependent manner independent of the presence of the CXXC motif.','cd03032','ArsC_Spx'),(959165,'gnl|CDD|239334',_binary 'Arsenate Reductase (ArsC) family, unknown subfamily; uncharacterized proteins containing a CXXC motif with similarity to thioredoxin (TRX)-fold arsenic reductases, ArsC. Proteins containing a redox active CXXC motif like TRX and glutaredoxin (GRX) function as protein disulfide oxidoreductases, altering the redox state of target proteins via the reversible oxidation of the active site dithiol. ArsC catalyzes the reduction of arsenate [As(V)] to arsenite [As(III)], using reducing equivalents derived from glutathione via GRX, through a single catalytic cysteine.','cd03036','ArsC_like'),(959171,'gnl|CDD|273720',_binary 'transcriptional regulator, Spx/MgsR family.  This model represents a portion of the proteins within the larger set covered by pfam03960. That larger family includes a glutaredoxin-dependent arsenate reductase (TIGR00014). Characterized members of this family include Spx and MgsR from Bacillus subtili. Spx is a global regulator for response to thiol-specific oxidative stress. It interacts with RNA polymerase. MgsR (modulator of the general stress response, also called YqgZ) provides a second level of regulation for more than a third of the proteins in the B. subtilis general stress regulon controlled by Sigma-B. [Regulatory functions, DNA interactions].','TIGR01617','Uncharacterized_protein_U'),(959173,'gnl|CDD|239511',_binary 'Glutaredoxin (GRX) family, GRX human class 1 and 2 (h_1_2)-like subfamily; composed of proteins similar to human GRXs, approximately 10 kDa in size, and proteins containing a GRX or GRX-like domain. GRX is a glutathione (GSH) dependent reductase, catalyzing the disulfide reduction of target proteins such as ribonucleotide reductase. It contains a redox active CXXC motif in a TRX fold and uses a similar dithiol mechanism employed by TRXs for intramolecular disulfide bond reduction of protein substrates. Unlike TRX, GRX has preference for mixed GSH disulfide substrates, in which it uses a monothiol mechanism where only the N-terminal cysteine is required. The flow of reducing equivalents in the GRX system goes from NADPH -> GSH reductase -> GSH -> GRX -> protein substrates. By altering the redox state of target proteins, GRX is involved in many cellular functions including DNA synthesis, signal transduction and the defense against oxidative stress. Different classes are known including human GRX1 and GRX2, which are members of this subfamily. Also included in this subfamily are the N-terminal GRX domains of proteins similar to human thioredoxin reductase 1 and 3.','cd03419','GRX_GRXh_1_2_like'),(959199,'gnl|CDD|226649',_binary 'Predicted phosphoesterase or phosphohydrolase [General function prediction only].','COG4186','COG4186'),(959200,'gnl|CDD|277336',_binary 'Aquifex aeolicus AQ1575 and related proteins, metallophosphatase domain.  This family includes bacterial and archeal proteins homologous to AQ1575, an uncharacterized Aquifex aeolicus protein.  AQ1575 may play an accessory role in DNA repair, based on the close proximity of its gene to Holliday junction resolvasome genes.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07390','MPP_AQ1575'),(959201,'gnl|CDD|277339',_binary 'Deinococcus radiodurans DR1119 and related proteins, metallophosphatase domain.  DR1119 is an uncharacterized Deinococcus radiodurans protein with a metallophosphatase domain.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07393','MPP_DR1119'),(959202,'gnl|CDD|224682',_binary 'Predicted phosphohydrolase [General function prediction only].','COG1768','COG1768'),(959221,'gnl|CDD|226462',_binary 'SLT domain proteins [General function prediction only].','COG3953','COG3953'),(959232,'gnl|CDD|316235',_binary 'Winged helix-turn-helix DNA-binding.','pfam13693','HTH_35'),(959252,'gnl|CDD|182125',_binary 'tyrosine recombinase; Provisional.','PRK09870','PRK09870'),(959264,'gnl|CDD|271191',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01191','INT_C_like_2'),(959267,'gnl|CDD|271196',_binary 'Uncharacterized site-specific tyrosine recombinase, C-terminal catalytic domain.  Tyrosine recombinase (integrase) belongs to a DNA breaking-rejoining enzyme superfamily. The catalytic domain contains six conserved active site residues. The recombination reaction involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd01196','INT_C_like_6'),(959270,'gnl|CDD|200235',_binary 'putative phage-type endonuclease.  Members of this protein family are found often in phage genomes and in prokaryotic genomes in uncharacterized regions that resemble integrated prophage regions.','TIGR03033','hypothetical_protein'),(959272,'gnl|CDD|273871',_binary 'phage recombination protein Bet.  This model represents the phage recombination protein Bet from a number of phage, including phage lambda. All members of this family are found in phage genomes or in putative prophage regions of bacterial genomes. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01913','Uncharacterized_protein_U'),(959568,'gnl|CDD|226936',_binary 'Holliday junction resolvase [DNA replication, recombination, and repair].','COG4570','Rus'),(959705,'gnl|CDD|212148',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from bacteria and archaea.  This family corresponds to a group of uncharacterized hypothetical proteins from bacteria, showing a high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain that contains a well conserved divalent ion-binding motif EXX[E/D].','cd11541','NTP-PPase_u4'),(959706,'gnl|CDD|212133',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain superfamily.  This superfamily contains enzymes that hydrolyze the alpha-beta phosphodiester bond of all canonical NTPs into monophosphate derivatives and pyrophosphate (PPi). Divalent ions, such as Mg2+ ion(s), are essential to activate a proposed water nucleophile and stabilize the charged intermediates to facilitate catalysis. These enzymes share a conserved divalent ion-binding motif EXX[E/D] in their active sites. They also share a highly conserved four-helix bundle, where one face forms the active site, while the other participates in oligomer assembly. The four-helix bundle consists of two central antiparallel alpha-helices that can be contained within a single protomer or form upon dimerization. The superfamily members include dimeric dUTP pyrophosphatases (dUTPases; EC 3.6.1.23), the nonspecific NTP-PPase MazG proteins, HisE-encoded phosphoribosyl ATP pyrophosphohydolase (PRA-PH), fungal histidine biosynthesis trifunctional proteins, and several uncharacterized protein families.','cd11523','NTP-PPase'),(959707,'gnl|CDD|224608',_binary 'Predicted pyrophosphatase [General function prediction only].','COG1694','MazG'),(959708,'gnl|CDD|281770',_binary 'MazG nucleotide pyrophosphohydrolase domain.  This domain is about 100 amino acid residues in length. It is found in the MazG protein from E. coli. It contains four conserved negatively charged residues that probably form an active site or metal binding site. This domain is found in isolation in some proteins as well as associated with pfam00590. This domain is clearly related to pfam01503 another pyrophosphohydrolase involved in histidine biosynthesis. This family may be structurally related to the NUDIX domain pfam00293 (Bateman A pers. obs.).','pfam03819','MazG'),(959709,'gnl|CDD|212138',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain putative pyrophosphatase YpjD from Bacillus subtilis and its bacterial homologs.  This family includes a putative pyrophosphatase Ypjd from Bacillus subtilis (BsYpjD) and its homologs. Although its biological role has not been described in detail, BsYpjD shows significant sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, BsYpjD contains a single MazG-like domain.','cd11531','NTP-PPase_BsYpjD'),(959710,'gnl|CDD|212144',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in mouse RS21-C6 protein and its homologs.  RS21-C6 proteins, highly expressed in all vertebrate genomes and green plants, act as house-cleaning enzymes, removing 5-methyl dCTP (m5dCTP) in order to prevent gene silencing. They show significant sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, RS21-C6 contains a single MazG-like domain and functions as a tetramer (a dimer of dimers) with a typical four-helical bundle. Divalent ions, such as Mg2+, are required for its pyrophosphatase activity. This family also includes a pyrophosphatase from Archaeoglobus fulgidus (Af1178). Although its biological role remains unclear, Af1178 shows significant sequence similarity to the mouse RS21-C6 protein.','cd11537','NTP-PPase_RS21-C6_like'),(959763,'gnl|CDD|283436',_binary 'Glutaredoxin-like domain (DUF836).  These proteins are related to the pfam00462 family.','pfam05768','DUF836'),(959779,'gnl|CDD|316207',_binary 'Toprim domain.  The toprim domain is found in a wide variety of enzymes involved in nucleic acid manipulation.','pfam13662','Toprim_4'),(959815,'gnl|CDD|227226',_binary 'Predicted helicase [General function prediction only].','COG4889','COG4889'),(959908,'gnl|CDD|131723',_binary 'tape measure domain.  Proteins containing this domain are strictly bacterial, including bacteriophage and prophage regions of bacterial genomes. Most members are 800 to 1800 amino acids long, making them among the longest predicted proteins of their respective phage genomes, where they are encoded in tail protein regions. This roughly 80-residue domain described here usually begins between residue 100 and 250. Many members are known or predicted to act as phage tail tape measure proteins, a minor tail component that regulates tail length.','TIGR02675','Mu-like_prophage_FluMu_pr'),(959929,'gnl|CDD|237547',_binary 'conjugal transfer protein TrbA; Provisional.','PRK13890','PRK13890'),(959932,'gnl|CDD|236167',_binary 'anaerobic benzoate catabolism transcriptional regulator; Reviewed.','PRK08154','PRK08154'),(959954,'gnl|CDD|177363',_binary 'hypothetical protein.','PHA02446','PHA02446'),(959957,'gnl|CDD|164916',_binary 'hypothetical protein.','PHA02447','PHA02447'),(959958,'gnl|CDD|107186',_binary 'hypothetical protein.','PHA02448','PHA02448'),(959961,'gnl|CDD|198380',_binary 'GIY-YIG nuclease domain superfamily.  The GIY-YIG nuclease domain superfamily includes a large and diverse group of proteins involved in many cellular processes, such as class I homing GIY-YIG family endonucleases, prokaryotic nucleotide excision repair proteins UvrC and Cho, type II restriction enzymes, the endonuclease/reverse transcriptase of eukaryotic retrotransposable elements, and a family of eukaryotic enzymes that repair stalled replication forks. All of these members contain a conserved GIY-YIG nuclease domain that may serve as a scaffold for the coordination of a divalent metal ion required for catalysis of the phosphodiester bond cleavage. By combining with different specificity, targeting, or other domains, the GIY-YIG nucleases may perform different functions.','cd00719','GIY-YIG_SF'),(959962,'gnl|CDD|279831',_binary 'GIY-YIG catalytic domain.  This domain called GIY-YIG is found in the amino terminal region of excinuclease abc subunit c (uvrC), bacteriophage T4 endonucleases segA, segB, segC, segD and segE; it is also found in putative endonucleases encoded by group I introns of fungi and phage. The structure of I-TevI a GIY-YIG endonuclease, reveals a novel alpha/beta-fold with a central three-stranded antiparallel beta-sheet flanked by three helices. The most conserved and putative catalytic residues are located on a shallow, concave surface and include a metal coordination site.','pfam01541','GIY-YIG'),(959964,'gnl|CDD|223747',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG0675','COG0675'),(959965,'gnl|CDD|284650',_binary 'Putative transposase DNA-binding domain.  This putative domain is found at the C-terminus of a large number of transposase proteins. This domain contains four conserved cysteines suggestive of a zinc binding domain. Given the need for transposases to bind DNA as well as the large number of DNA-binding zinc fingers we hypothesize this domain is DNA-binding.','pfam07282','OrfB_Zn_ribbon'),(959966,'gnl|CDD|307511',_binary 'Probable transposase.  This family includes IS891, IS1136 and IS1341. DUF1225, pfam06774, has now been merged into this family.','pfam01385','OrfB_IS605'),(959967,'gnl|CDD|315082',_binary 'Helix-turn-helix domain.  This is the N terminal helix-turn-helix domain of Transposase_2 pfam01385.','pfam12323','HTH_OrfB_IS605'),(959968,'gnl|CDD|273793',_binary 'transposase, IS605 OrfB family, central region.  This model represents a region of a sequence similarity between a family of putative transposases of Thermoanaerobacter tengcongensis, smaller related proteins from Bacillus anthracis, putative transposes described by pfam01385, and other proteins. [Mobile and extrachromosomal element functions, Transposon functions].','TIGR01766','Putative_transposase_MJ07'),(959970,'gnl|CDD|316017',_binary 'Meiotically up-regulated gene 113.  This is a family of fungal proteins found to be up-regulated in meiosis.','pfam13455','MUG113'),(960042,'gnl|CDD|214573',_binary 'JAB/MPN domain.  Domain in Jun kinase activation domain binding protein and proteasomal subunits. Domain at Mpr1p and Pad1p N-termini. Domain of unknown function.','smart00232','JAB_MPN'),(960044,'gnl|CDD|163690',_binary 'Mpr1p, Pad1p N-terminal (MPN) domains with catalytic isopeptidase activity (metal-binding); prokaryotic.  This family contains bacterial and archaeal MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+)-like domains. These catalytically active domains contain the signature JAB1/MPN/Mov34 metalloenzyme (JAMM) motif, EXnHS/THX7SXXD, which is involved in zinc ion coordination and provides the active site for isopeptidase activity for the release of ubiquitin from ubiquitinated proteins (thus having deubiquitinating (DUB) activity) that are tagged for degradation.  The JAMM proteins likely hydrolyze ubiquitin conjugates in a manner similar to thermolysin, in which the zinc-polarized aqua ligand serves as the nucleophile, compared with the classical DUBs that do so with a cysteine residue in the active site.','cd08059','MPN_prok_mb'),(960047,'gnl|CDD|235569',_binary 'flagellar biosynthesis protein FlhB; Reviewed.','PRK05702','flhB'),(960050,'gnl|CDD|238157',_binary 'Lytic Transglycosylase (LT)  and Goose Egg White Lysozyme (GEWL) domain. Members include the soluble and  insoluble membrane-bound LTs in bacteria, the LTs in bacteriophage lambda, as well as, the eukaryotic \'goose-type\' lysozymes (GEWL).  LTs catalyze the cleavage of the beta-1,4-glycosidic bond between N-acetylmuramic acid (MurNAc) and N-acetyl-D-glucosamine (GlcNAc), as do \'goose-type\' lysozymes. However, in addition to this, they also make a new glycosidic bond with the C6 hydroxyl group of the same muramic acid residue.','cd00254','LT_GEWL'),(960056,'gnl|CDD|225956',_binary 'Uncharacterized conserved protein [Function unknown].','COG3422','COG3422'),(960070,'gnl|CDD|319786',_binary '5\'(3\')-deoxyribonucleotidase.  This family includes cytosolic 5\'(3\')-deoxyribonucleotidase (cdN) and mitochondrial 5\'(3\')-deoxyribonucleotidase (mdN). cdN and mdN specifically dephosphorylate the deoxyribo form of nucleoside monophosphates helps maintain homeostasis of deoxynucleosides required for mitochondrial DNA synthesis. Their preferred substrates are dUMP and dTMP. cdN also dephosphorylates dGMP and dIMP efficiently. They can also dephosphorylate the 5\'- or 3\'-phosphates of pyrimidine ribonucleotides.  This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd02587','HAD_5-3dNT'),(960071,'gnl|CDD|284381',_binary '5\' nucleotidase, deoxy (Pyrimidine), cytosolic type C protein (NT5C).  This family consists of several 5\' nucleotidase, deoxy (Pyrimidine), cytosolic type C (NT5C) proteins. 5\'(3\')-Deoxyribonucleotidase is a ubiquitous enzyme in mammalian cells whose physiological function is not known.','pfam06941','NT5C'),(960073,'gnl|CDD|223620',_binary 'Predicted phosphatases [General function prediction only].','COG0546','Gph'),(960074,'gnl|CDD|181382',_binary 'hypothetical protein; Provisional.','PRK08317','PRK08317'),(960076,'gnl|CDD|212491',_binary 'classical (c) SDRs.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05233','SDR_c'),(960077,'gnl|CDD|223959',_binary 'Dehydrogenases with different specificities (related to short-chain alcohol dehydrogenases) [Secondary metabolites biosynthesis, transport, and catabolism / General function prediction only].','COG1028','FabG'),(960078,'gnl|CDD|316118',_binary 'Enoyl-(Acyl carrier protein) reductase.','pfam13561','adh_short_C2'),(960079,'gnl|CDD|237218',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK12825','fabG'),(960080,'gnl|CDD|306588',_binary 'short chain dehydrogenase.  This family contains a wide variety of dehydrogenases.','pfam00106','adh_short'),(960081,'gnl|CDD|235546',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Validated.','PRK05653','fabG'),(960082,'gnl|CDD|183775',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Reviewed.','PRK12826','PRK12826'),(960083,'gnl|CDD|180817',_binary 'short chain dehydrogenase; Provisional.','PRK07060','PRK07060'),(960084,'gnl|CDD|235500',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Validated.','PRK05557','fabG'),(960085,'gnl|CDD|187592',_binary '2,3 dihydro-2,3 dihydrozybenzoate dehydrogenases, classical (c) SDRs.  2,3 dihydro-2,3 dihydrozybenzoate dehydrogenase shares the characteristics of the classical SDRs. This subgroup includes Escherichai coli EntA which catalyzes the NAD+-dependent oxidation of 2,3-dihydro-2,3-dihydroxybenzoate to 2,3-dihydroxybenzoate during biosynthesis of the siderophore Enterobactin. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05331','DH-DHB-DH_SDR_c'),(960086,'gnl|CDD|187594',_binary 'beta-Keto acyl carrier protein reductase (BKR), involved in Type II FAS, classical (c) SDRs.  This subgroup includes the Escherichai coli K12 BKR, FabG. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet) NAD(P)(H) binding region and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues.   Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD binding motif and characteristic NAD-binding and catalytic sequence patterns.  These enzymes have a 3-glycine N-terminal NAD(P)(H) binding pattern: TGxxxGxG in classical SDRs.  Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif.  Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P) binding motif and  an altered active site motif (YXXXN).  Fungal type type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.  Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P) binding motif and missing or unusual active site residues.  Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site.  Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr-151 and Lys-155, and well as Asn-111 (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05333','BKR_SDR_c'),(960087,'gnl|CDD|226674',_binary 'Short-chain alcohol dehydrogenase of unknown specificity [General function prediction only].','COG4221','COG4221'),(960088,'gnl|CDD|237219',_binary 'short chain dehydrogenase; Provisional.','PRK12827','PRK12827'),(960089,'gnl|CDD|275120',_binary '2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase.  Members of this family are 2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase (EC 1.3.1.28), the third enzyme in the biosynthesis of 2,3-dihydroxybenzoic acid (DHB) from chorismate. The first two enzymes are isochorismate synthase (EC 5.4.4.2) and isochorismatase (EC 3.3.2.1). Synthesis is often followed by adenylation by the enzyme DHBA-AMP ligase (EC 2.7.7.58) to activate (DHB) for a non-ribosomal peptide synthetase.','TIGR04316','23-dihydro-23-dihydroxybe'),(960090,'gnl|CDD|235506',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK05565','fabG'),(960091,'gnl|CDD|187609',_binary 'xylulose reductase-like, classical (c) SDRs.  Members of this subgroup include proteins identified as L-xylulose reductase (XR) and carbonyl reductase; they are members of the SDR family. XR, catalyzes the NADP-dependent reduction of L-xyulose and other sugars. Tetrameric mouse carbonyl reductase is involved in the metabolism of biogenic and xenobiotic carbonyl compounds. This subgroup also includes tetrameric chicken liver D-erythrulose reductase, which catalyzes the reduction of D-erythrulose to D-threitol. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser).','cd05351','XR_like_SDR_c'),(960092,'gnl|CDD|187622',_binary 'classical (c) SDR, subgroup 11.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05364','SDR_c11'),(960093,'gnl|CDD|181668',_binary 'pteridine reductase; Provisional.','PRK09135','PRK09135'),(960094,'gnl|CDD|235975',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK07231','fabG'),(960095,'gnl|CDD|187584',_binary 'insect type alcohol dehydrogenase (ADH)-like, classical (c) SDRs.  This subgroup contains insect type ADH, and 15-hydroxyprostaglandin dehydrogenase (15-PGDH) type I; these proteins are classical SDRs. ADH catalyzes the NAD+-dependent oxidation of alcohols to aldehydes/ketones. This subgroup is distinct from the zinc-dependent alcohol dehydrogenases of the medium chain dehydrogenase/reductase family, and evolved in fruit flies to allow the digestion of fermenting fruit. 15-PGDH catalyzes the NAD-dependent interconversion of (5Z,13E)-(15S)-11alpha,15-dihydroxy-9-oxoprost-13-enoate and (5Z,13E)-11alpha-hydroxy-9,15-dioxoprost-13-enoate, and has a typical SDR glycine-rich NAD-binding motif, which is not fully present in ADH.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05323','ADH_SDR_c_like'),(960096,'gnl|CDD|274889',_binary 'SDR family mycofactocin-dependent oxidoreductase.  Members of this protein subfamily are putative oxidoreductases belonging to the larger SDR family. All members occur in genomes that encode a cassette for the biosynthesis of mycofactocin, a proposed electron carrier of a novel redox pool. Characterized members of this family are described as NDMA-dependent, meaning that a blue aniline dye serving as an artificial electron acceptor is required for members of this family to cycle in vitro, since the bound NAD residue is not exchangeable. See EC 1.1.99.36. [Unknown function, Enzymes of unknown specificity].','TIGR03971','short-chain_dehydrogenase'),(960097,'gnl|CDD|187586',_binary 'carbonyl reductase sniffer-like, classical (c) SDRs.  Sniffer is an NADPH-dependent carbonyl reductase of the classical SDR family. Studies in Drosophila melanogaster implicate Sniffer in the prevention of neurodegeneration due to aging and oxidative-stress. This subgroup also includes Rhodococcus sp. AD45 IsoH, which is an NAD-dependent 1-hydroxy-2-glutathionyl-2-methyl-3-butene dehydrogenase involved in isoprene metabolism, Aspergillus nidulans StcE encoded by a gene which is part of a proposed sterigmatocystin biosynthesis gene cluster, Bacillus circulans SANK 72073 BtrF encoded by a gene found in the butirosin biosynthesis gene cluster, and Aspergillus parasiticus nor-1 involved in the biosynthesis of aflatoxins. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05325','carb_red_sniffer_like_SDR'),(960098,'gnl|CDD|236190',_binary '2,3-dihydroxybenzoate-2,3-dehydrogenase; Validated.','PRK08220','PRK08220'),(960099,'gnl|CDD|212493',_binary 'HetN oxidoreductase-like, classical (c) SDR.  This subgroup includes Anabaena sp. strain PCC 7120 HetN, a putative oxidoreductase involved in heterocyst differentiation, and related proteins.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08932','HetN_like_SDR_c'),(960100,'gnl|CDD|187625',_binary 'sepiapterin reductase (SPR)-like, classical (c) SDRs.  Human SPR, a member of the SDR family, catalyzes the NADP-dependent reduction of sepiaptern to 7,8-dihydrobiopterin (BH2). In addition to SPRs, this subgroup also contains Bacillus cereus yueD, a benzil reductase, which catalyzes the stereospecific reduction of benzil to (S)-benzoin. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05367','SPR-like_SDR_c'),(960101,'gnl|CDD|236241',_binary 'short chain dehydrogenase; Validated.','PRK08324','PRK08324'),(960103,'gnl|CDD|273824',_binary '3-oxoacyl-(acyl-carrier-protein) reductase.  This model represents 3-oxoacyl-[ACP] reductase, also called 3-ketoacyl-acyl carrier protein reductase, an enzyme of fatty acid biosynthesis. [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR01830','3-oxoacyl-_reductase_FabG'),(960105,'gnl|CDD|235816',_binary 'short chain dehydrogenase; Provisional.','PRK06500','PRK06500'),(960107,'gnl|CDD|237187',_binary 'oxidoreductase; Provisional.','PRK12743','PRK12743'),(960109,'gnl|CDD|236056',_binary 'short chain dehydrogenase; Provisional.','PRK07576','PRK07576'),(960110,'gnl|CDD|187602',_binary 'putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR)-like, SDR.  This subgroup resembles the SDR family, but does not have a perfect match to the NAD-binding motif or the catalytic tetrad characteristic of the SDRs. It includes the SDRs, Q9HYA2 from Pseudomonas aeruginosa PAO1 and APE0912 from Aeropyrum pernix K1. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05344','BKR_like_SDR_like'),(960111,'gnl|CDD|168574',_binary 'short chain dehydrogenase; Validated.','PRK06484','PRK06484'),(960113,'gnl|CDD|187627',_binary 'Trans-2-enoyl-CoA reductase (TER) and 2,4-dienoyl-CoA reductase (DECR), atypical (a) SDR.  TTER is a peroxisomal protein with a proposed role in fatty acid elongation. Fatty acid synthesis is known to occur in the both endoplasmic reticulum and mitochondria; peroxisomal TER has been proposed as an additional fatty acid elongation system, it reduces the double bond at C-2 as the last step of elongation.  This system resembles the mitochondrial system in that acetyl-CoA is used as a carbon donor. TER may also function in phytol metabolism, reducting phytenoyl-CoA to phytanoyl-CoA in peroxisomes. DECR processes double bonds in fatty acids to increase their utility in fatty acid metabolism; it reduces 2,4-dienoyl-CoA to an enoyl-CoA. DECR is active in mitochondria and peroxisomes. This subgroup has the Gly-rich NAD-binding motif of the classical SDR family, but does not display strong identity to the canonical active site tetrad, and lacks the characteristic Tyr at the usual position. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05369','TER_DECR_SDR_a'),(960114,'gnl|CDD|187605',_binary 'gluconate 5-dehydrogenase (Ga5DH)-like, classical (c) SDRs.  Ga5DH catalyzes the NADP-dependent conversion of carbon source D-gluconate and 5-keto-D-gluconate. This SDR subgroup has a classical Gly-rich NAD(P)-binding motif and a conserved active site tetrad pattern. However, it has been proposed that Arg104 (Streptococcus suis Ga5DH numbering), as well as an active site Ca2+, play a critical role in catalysis. In addition to Ga5DHs this subgroup contains Erwinia chrysanthemi KduD which is involved in pectin degradation, and is a putative 2,5-diketo-3-deoxygluconate dehydrogenase. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107,15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05347','Ga5DH-like_SDR_c'),(960115,'gnl|CDD|187615',_binary 'pteridine reductase (PR), classical (c) SDRs.  Pteridine reductases (PRs), members of the SDR family, catalyzes the NAD-dependent reduction of folic acid, dihydrofolate and related compounds. In Leishmania, pteridine reductase (PTR1) acts to circumvent the anti-protozoan drugs that attack dihydrofolate reductase activity. Proteins in this subgroup have an N-terminal NAD-binding motif and a YxxxK active site motif, but have an Asp instead of the usual upstream catalytic Ser. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05357','PR_SDR_c'),(960116,'gnl|CDD|212497',_binary 'Lin1944 and related proteins, classical (c) SDRs.  Lin1944 protein from Listeria Innocua is a classical SDR, it contains a glycine-rich motif similar to the canonical motif of the SDR NAD(P)-binding site. However, the typical SDR active site residues are absent in this subgroup of proteins of undetermined function. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd11731','Lin1944_like_SDR_c'),(960117,'gnl|CDD|180604',_binary 'short chain dehydrogenase; Provisional.','PRK06523','PRK06523'),(960118,'gnl|CDD|131468',_binary 'acetoin reductases.  One member of this family, as characterized in Klebsiella terrigena, is described as able to interconvert acetoin + NADH with meso-2,3-butanediol + NAD(+). It is also called capable of irreversible reduction of diacetyl with NADH to acetoin. Blomqvist, et al. decline to specify either EC 1.1.1.4 which is (R,R)-butanediol dehydrogenase, or EC 1.1.1.5, which is acetoin dehydrogenase without a specified stereochemistry, for this enzyme. This enzyme is a homotetramer in the family of short chain dehydrogenases (pfam00106). Another member of this family, from Corynebacterium glutamicum, is called L-2,3-butanediol dehydrogenase (). [Energy metabolism, Fermentation].','TIGR02415','Diacetyl_reductase'),(960119,'gnl|CDD|187616',_binary 'glucose 1 dehydrogenase (GlcDH), classical (c) SDRs.  GlcDH, is a tetrameric member of the SDR family, it catalyzes the NAD(P)-dependent oxidation of beta-D-glucose to D-glucono-delta-lactone. GlcDH has a typical NAD-binding site glycine-rich pattern as well as the canonical active site tetrad (YXXXK motif plus upstream Ser and Asn). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05358','GlcDH_SDR_c'),(960120,'gnl|CDD|187629',_binary '17hydroxysteroid dehydrogenase type 10 (HSD10)-like, classical (c) SDRs.  HSD10, also known as amyloid-peptide-binding alcohol dehydrogenase (ABAD), was previously identified as a L-3-hydroxyacyl-CoA dehydrogenase, HADH2. In fatty acid metabolism, HADH2 catalyzes the third step of beta-oxidation, the conversion of a hydroxyl to a keto group in the NAD-dependent oxidation of L-3-hydroxyacyl CoA. In addition to alcohol dehydrogenase and HADH2 activites, HSD10 has steroid dehydrogenase activity. Although the mechanism is unclear, HSD10 is implicated in the formation of amyloid beta-petide in the brain (which is linked to the development of Alzheimer\'s disease). Although HSD10 is normally concentrated in the mitochondria, in the presence of amyloid beta-peptide it translocates into the plasma membrane, where it\'s action may generate cytotoxic aldehydes and may lower estrogen levels through its use of 17-beta-estradiol as a substrate. HSD10 is a member of the SRD family, but differs from other SDRs by the presence of two insertions of unknown function. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05371','HSD10-like_SDR_c'),(960121,'gnl|CDD|181159',_binary 'short chain dehydrogenase; Provisional.','PRK07890','PRK07890'),(960122,'gnl|CDD|187634',_binary 'classical (c) SDR, subgroup 4.  This subgroup has a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08929','SDR_c4'),(960123,'gnl|CDD|171821',_binary 'short chain dehydrogenase; Provisional.','PRK12937','PRK12937'),(960124,'gnl|CDD|187624',_binary 'meso-2,3-butanediol dehydrogenase-like, classical (c) SDRs.  2,3-butanediol dehydrogenases (BDHs) catalyze the NAD+ dependent conversion of 2,3-butanediol to acetonin; BDHs are classified into types according to their stereospecificity as to substrates and products. Included in this subgroup are Klebsiella pneumonia meso-BDH which catalyzes meso-2,3-butanediol to D(-)-acetonin, and Corynebacterium glutamicum L-BDH which catalyzes lX+)-2,3-butanediol to L(+)-acetonin. This subgroup is comprised of classical SDRs with the characteristic catalytic triad and NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05366','meso-BDH-like_SDR_c'),(960125,'gnl|CDD|180753',_binary 'short chain dehydrogenase; Provisional.','PRK06924','PRK06924'),(960126,'gnl|CDD|187632',_binary '17beta hydroxysteroid dehydrogenase-like, classical (c) SDRs.  17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05374','17beta-HSD-like_SDR_c'),(960127,'gnl|CDD|183833',_binary 'short chain dehydrogenase; Provisional.','PRK12939','PRK12939'),(960128,'gnl|CDD|237188',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK12745','PRK12745'),(960129,'gnl|CDD|187603',_binary 'putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR), subgroup 3, classical (c) SDR.  This subgroup includes the putative Brucella melitensis biovar Abortus 2308 BKR, FabG, Mesorhizobium loti MAFF303099 FabG, and other classical SDRs. BKR, a member of the SDR family, catalyzes the NADPH-dependent reduction of acyl carrier protein in the first reductive step of de novo fatty acid synthesis (FAS).  FAS consists of 4 elongation steps, which are repeated to extend the fatty acid chain thru the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I Fas utilizes one or 2 multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05345','BKR_3_SDR_c'),(960130,'gnl|CDD|187610',_binary 'mannitol dehydrogenase (MDH)-like, classical (c) SDRs.  NADP-mannitol dehydrogenase catalyzes the conversion of fructose to mannitol, an acyclic 6-carbon sugar. MDH is a tetrameric member of the SDR family. This subgroup also includes various other tetrameric SDRs, including Pichia stipitis D-arabinitol dehydrogenase (aka polyol dehydrogenase), Candida albicans Sou1p, a sorbose reductase, and Candida parapsilosis (S)-specific carbonyl reductase (SCR, aka S-specific alcohol dehydrogenase) which catalyzes the enantioselective reduction of 2-hydroxyacetophenone into (S)-1-phenyl-1,2-ethanediol. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser).','cd05352','MDH-like_SDR_c'),(960131,'gnl|CDD|131680',_binary 'rhamnulose-1-phosphate aldolase/alcohol dehydrogenase.','TIGR02632','conserved_hypothetical_pr'),(960132,'gnl|CDD|187535',_binary 'Short-chain dehydrogenases/reductases (SDR).  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase (KR) domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type KRs have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd02266','SDR'),(960133,'gnl|CDD|187600',_binary '3beta17beta hydroxysteroid dehydrogenase-like, classical (c) SDRs.  This subgroup includes members identified as 3beta17beta hydroxysteroid dehydrogenase, 20beta hydroxysteroid dehydrogenase, and R-alcohol dehydrogenase. These proteins exhibit the canonical active site tetrad and glycine rich NAD(P)-binding motif of the classical SDRs. 17beta-dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens, and include members of the SDR family. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05341','3beta-17beta-HSD_like_SDR'),(960134,'gnl|CDD|235608',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK05786','fabG'),(960135,'gnl|CDD|235990',_binary 'short chain dehydrogenase; Provisional.','PRK07326','PRK07326'),(960136,'gnl|CDD|180723',_binary 'short chain dehydrogenase; Provisional.','PRK06841','PRK06841'),(960137,'gnl|CDD|187617',_binary '1-cyclohexenylcarbonyl_coenzyme A_reductase (ChcA)_like, classical (c) SDRs.  This subgroup contains classical SDR proteins, including members identified as 1-cyclohexenylcarbonyl coenzyme A reductase. ChcA of Streptomyces collinus is implicated in the final reduction step of shikimic acid to ansatrienin. ChcA shows sequence similarity to the SDR family of NAD-binding proteins, but it lacks the conserved Tyr of the characteristic catalytic site. This subgroup also contains the NADH-dependent enoyl-[acyl-carrier-protein(ACP)] reductase FabL from Bacillus subtilis. This enzyme participates in bacterial fatty acid synthesis, in type II fatty-acid synthases and catalyzes the last step in each elongation cycle. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05359','ChcA_like_SDR_c'),(960138,'gnl|CDD|187585',_binary 'Porcine testicular carbonyl reductase (PTCR)-like, classical (c) SDRs.  PTCR is a classical SDR which catalyzes the NADPH-dependent reduction of ketones on steroids and prostaglandins. Unlike most SDRs, PTCR functions as a monomer. This subgroup also includes human carbonyl reductase 1 (CBR1) and CBR3. CBR1 is an NADPH-dependent SDR with broad substrate specificity and may be responsible for the in vivo reduction of quinones, prostaglandins, and other carbonyl-containing compounds. In addition it includes poppy NADPH-dependent salutaridine reductase which catalyzes the stereospecific reduction of salutaridine to 7(S)-salutaridinol in the biosynthesis of morphine, and Arabidopsis SDR1,a menthone reductase, which catalyzes the reduction of menthone to neomenthol, a compound with antimicrobial activity; SDR1  can also carry out neomenthol oxidation. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05324','carb_red_PTCR-like_SDR_c'),(960139,'gnl|CDD|187620',_binary 'tetrahydroxynaphthalene/trihydroxynaphthalene reductase-like, classical (c) SDRs.  1,3,6,8-tetrahydroxynaphthalene reductase (4HNR) of Magnaporthe grisea and the related 1,3,8-trihydroxynaphthalene reductase (3HNR) are typical members of the SDR family containing the canonical glycine rich NAD(P)-binding site and active site tetrad, and function in fungal melanin biosynthesis. This subgroup also includes an SDR from Norway spruce that may function to protect against both biotic and abitoic stress. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05362','THN_reductase-like_SDR_c'),(960140,'gnl|CDD|183778',_binary 'short chain dehydrogenase; Provisional.','PRK12829','PRK12829'),(960141,'gnl|CDD|183773',_binary 'acetoacetyl-CoA reductase; Provisional.','PRK12824','PRK12824'),(960142,'gnl|CDD|187665',_binary 'human 17beta-hydroxysteroid dehydrogenase type 2 (type 2 17beta-HSD)-like, classical (c) SDRs.  17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. This classical-SDR subgroup includes the human proteins: type 2 17beta-HSD, type 6 17beta-HSD,  type 2 11beta-HSD, dehydrogenase/reductase SDR family member 9,  short-chain dehydrogenase/reductase family 9C member 7, 3-hydroxybutyrate dehydrogenase type 1, and retinol dehydrogenase 5. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd09805','type2_17beta_HSD-like_SDR'),(960143,'gnl|CDD|187649',_binary 'Polyketide ketoreductase, classical (c) SDR.  Polyketide ketoreductase (KR) is a classical SDR with a characteristic NAD-binding pattern and active site tetrad.  Aromatic polyketides include various aromatic compounds of pharmaceutical interest. Polyketide KR, part of the type II polyketide synthase (PKS) complex, is comprised of stand-alone domains that resemble the domains found in fatty acid synthase and multidomain type I PKS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd08945','PKR_SDR_c'),(960144,'gnl|CDD|181585',_binary 'glucose-1-dehydrogenase; Provisional.','PRK08936','PRK08936'),(960145,'gnl|CDD|187591',_binary 'cyclohexanol reductases, including levodione reductase, classical (c) SDRs.  Cyloclohexanol reductases,including (6R)-2,2,6-trimethyl-1,4-cyclohexanedione (levodione) reductase of Corynebacterium aquaticum, catalyze the reversible oxidoreduction of hydroxycyclohexanone derivatives. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05330','cyclohexanol_reductase_SD'),(960146,'gnl|CDD|275297',_binary 'SDR family mycofactocin-dependent oxidoreductase.  Members of this protein subfamily are putative oxidoreductases belonging to the larger SDR family. All members occur in genomes that encode a cassette for the biosynthesis of mycofactocin, a proposed electron carrier of a novel redox pool. Characterized members of this family are described as NDMA-dependent, meaning that a blue aniline dye serving as an artificial electron acceptor is required for members of this family to cycle in vitro, since the bound NAD residue is not exchangeable. This family resembles TIGR03971 most closely in the N-terminal region, consistent with the published hypothesis of NAD interaction with mycofactocin. See EC 1.1.99.36. [Unknown function, Enzymes of unknown specificity].','TIGR04504','SDR_subfam_2'),(960147,'gnl|CDD|236216',_binary 'D-mannonate oxidoreductase; Provisional.','PRK08277','PRK08277'),(960148,'gnl|CDD|187590',_binary 'tropinone reductase-I and II (TR-1, and TR-II)-like, classical (c) SDRs.  This subgroup includes TR-I and TR-II; these proteins are members of the SDR family. TRs catalyze the NADPH-dependent reductions of the 3-carbonyl group of tropinone, to a beta-hydroxyl group. TR-I and TR-II produce different stereoisomers from tropinone, TR-I produces tropine (3alpha-hydroxytropane), and TR-II, produces pseudotropine (sigma-tropine, 3beta-hydroxytropane).  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05329','TR_SDR_c'),(960149,'gnl|CDD|187587',_binary 'secoisolariciresinol dehydrogenase (secoisolariciresinol-DH)-like, classical (c) SDRs.  Podophyllum secoisolariciresinol-DH is a homo tetrameric, classical SDR that catalyzes the NAD-dependent conversion of (-)-secoisolariciresinol to (-)-matairesinol via a (-)-lactol intermediate. (-)-Matairesinol is an intermediate to various 8\'-lignans, including the cancer-preventive mammalian lignan, and those involved in vascular plant defense. This subgroup also includes rice momilactone A synthase which catalyzes the conversion of 3beta-hydroxy-9betaH-pimara-7,15-dien-19,6beta-olide into momilactone A, Arabidopsis ABA2 which during abscisic acid (ABA) biosynthesis, catalyzes the conversion of xanthoxin to abscisic aldehyde and, maize Tasselseed2 which participate in the maize sex determination pathway. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05326','secoisolariciresinol-DH_l'),(960150,'gnl|CDD|187647',_binary 'rhamnulose-1-phosphate aldolase/alcohol dehydrogenase, classical (c) SDRs.  This family has bifunctional proteins with an N-terminal aldolase and a C-terminal classical SDR domain. One member is identified as a rhamnulose-1-phosphate aldolase/alcohol dehydrogenase. The SDR domain has a canonical SDR glycine-rich NAD(P) binding motif and a match to the characteristic active site triad. However, it lacks an upstream active site Asn typical of SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08943','R1PA_ADH_SDR_c'),(960151,'gnl|CDD|180586',_binary 'dihydromonapterin reductase; Provisional.','PRK06483','PRK06483'),(960152,'gnl|CDD|181721',_binary 'tropinone reductase; Provisional.','PRK09242','PRK09242'),(960153,'gnl|CDD|180771',_binary 'glucose-1-dehydrogenase; Provisional.','PRK06947','PRK06947'),(960155,'gnl|CDD|187648',_binary 'classical (c) SDR, subgroup 12.  These are classical SDRs, with the canonical active site tetrad and glycine-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08944','SDR_c12'),(960157,'gnl|CDD|187640',_binary 'putative D-mannonate oxidoreductase, classical (c) SDR.  D-mannonate oxidoreductase catalyzes the NAD-dependent interconversion of D-mannonate and D-fructuronate. This subgroup includes Bacillus subtitils UxuB/YjmF, a putative D-mannonate oxidoreductase; the B. subtilis UxuB gene is part of a putative ten-gene operon (the Yjm operon) involved in hexuronate catabolism. Escherichia coli UxuB does not belong to this subgroup. This subgroup has a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08935','mannonate_red_SDR_c'),(960159,'gnl|CDD|187596',_binary 'putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR), subgroup 1, classical (c) SDR.  This subgroup includes Escherichia coli CFT073 FabG. The Escherichai coli K12 BKR, FabG, belongs to a different subgroup. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet) NAD(P)(H) binding region and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues.   Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD binding motif and characteristic NAD-binding and catalytic sequence patterns.  These enzymes have a 3-glycine N-terminal NAD(P)(H) binding pattern: TGxxxGxG in classical SDRs. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif.  Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P) binding motif and  an altered active site motif (YXXXN).  Fungal type type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.  Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P) binding motif and missing or unusual active site residues.  Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site.  Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr-151 and Lys-155, and well as Asn-111 (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05337','BKR_1_SDR_c'),(960161,'gnl|CDD|223696',_binary 'Enoyl-[acyl-carrier-protein].','COG0623','FabI'),(960163,'gnl|CDD|180796',_binary 'short chain dehydrogenase; Provisional.','PRK07023','PRK07023'),(960165,'gnl|CDD|180300',_binary 'short chain dehydrogenase; Provisional.','PRK05875','PRK05875'),(960168,'gnl|CDD|180949',_binary 'enoyl-(acyl carrier protein) reductase; Validated.','PRK07370','PRK07370'),(960171,'gnl|CDD|187644',_binary 'd-3-hydroxybutyrate dehydrogenase (HBDH), classical (c) SDRs.  DHBDH, an NAD+ -dependent enzyme, catalyzes the interconversion of D-3-hydroxybutyrate and acetoacetate. It is a classical SDR, with the canonical NAD-binding motif and active site tetrad. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08940','HBDH_SDR_c'),(960173,'gnl|CDD|187607',_binary 'putative beta-ketoacyl acyl carrier protein [ACP]reductase (BKR), subgroup 2, classical (c) SDR.  This subgroup includes Rhizobium sp. NGR234 FabG1. The Escherichai coli K12 BKR, FabG, belongs to a different subgroup. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05349','BKR_2_SDR_c'),(960175,'gnl|CDD|187597',_binary 'human dehydrogenase/reductase (SDR family) member 1 (DHRS1) and human hydroxysteroid dehydrogenase-like protein 2 (HSDL2), classical (c) SDRs.  This subgroup includes human DHRS1 and human HSDL2 and related proteins. These are members of the classical SDR family, with a canonical Gly-rich  NAD-binding motif and the typical YXXXK active site motif. However, the rest of the catalytic tetrad is not strongly conserved. DHRS1 mRNA has been detected in many tissues, liver, heart, skeletal muscle, kidney and pancreas; a longer transcript is predominantly expressed in the liver , a shorter one in the heart. HSDL2 may play a part in fatty acid metabolism, as it is found in peroxisomes. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05338','DHRS1_HSDL2-like_SDR_c'),(960177,'gnl|CDD|180439',_binary 'sorbitol-6-phosphate 2-dehydrogenase; Provisional.','PRK06171','PRK06171'),(960179,'gnl|CDD|180576',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK06463','fabG'),(960181,'gnl|CDD|236207',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK08261','fabG'),(960183,'gnl|CDD|236057',_binary 'short chain dehydrogenase; Provisional.','PRK07578','PRK07578'),(960185,'gnl|CDD|236094',_binary 'short chain dehydrogenase; Provisional.','PRK07774','PRK07774'),(960187,'gnl|CDD|181297',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK08217','fabG'),(960189,'gnl|CDD|135631',_binary 'short chain dehydrogenase; Provisional.','PRK05867','PRK05867'),(960191,'gnl|CDD|180993',_binary 'short chain dehydrogenase; Provisional.','PRK07478','PRK07478'),(960193,'gnl|CDD|180440',_binary 'short chain dehydrogenase; Provisional.','PRK06172','PRK06172'),(960195,'gnl|CDD|187642',_binary '1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase (DHB DH)-like, classical (c) SDR.  DHB DH (aka 1,2-dihydroxycyclohexa-3,5-diene-1-carboxylate dehydrogenase) catalyzes the NAD-dependent conversion of 1,2-dihydroxycyclohexa-3,4-diene carboxylate to a catechol. This subgroup also contains Pseudomonas putida F1 CmtB, 2,3-dihydroxy-2,3-dihydro-p-cumate dehydrogenase, the second enzyme in  the pathway for catabolism of p-cumate catabolism. This subgroup shares the glycine-rich NAD-binding motif of the classical SDRs and shares the same catalytic triad; however, the upstream Asn implicated in cofactor binding or catalysis in other SDRs is generally substituted by a Ser. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08937','DHB_DH-like_SDR_c'),(960197,'gnl|CDD|181518',_binary 'acetoin reductase; Validated.','PRK08643','PRK08643'),(960199,'gnl|CDD|236145',_binary 'enoyl-(acyl carrier protein) reductase; Provisional.','PRK08063','PRK08063'),(960202,'gnl|CDD|236116',_binary 'short chain dehydrogenase; Provisional.','PRK07856','PRK07856'),(960204,'gnl|CDD|187635',_binary 'classical (c) SDR, subgroup 8.  This subgroup has a fairly well conserved active site tetrad and domain size of the classical SDRs, but has an atypical NAD-binding motif ([ST]G[GA]XGXXG). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08930','SDR_c8'),(960206,'gnl|CDD|180802',_binary 'short chain dehydrogenase; Provisional.','PRK07035','PRK07035'),(960208,'gnl|CDD|181131',_binary 'short chain dehydrogenase; Provisional.','PRK07814','PRK07814'),(960210,'gnl|CDD|131732',_binary 'pteridine reductase.  Pteridine reductase is an enzyme used by trypanosomatids (including Trypanosoma cruzi and Leishmania major) to obtain reduced pteridines by salvage rather than biosynthetic pathways. Enzymes in T. cruzi described as pteridine reductase 1 (PTR1) and pteridine reductase 2 (PTR2) have different activity profiles. PTR1 is more active with with fully oxidized biopterin and folate than with reduced forms, while PTR2 reduces dihydrobiopterin and dihydrofolate but not oxidized pteridines. T. cruzi PTR1 and PTR2 are more similar to each other in sequence than either is to the pteridine reductase of Leishmania major, and all are included in this family.','TIGR02685','Pteridine_reductase_1'),(960212,'gnl|CDD|180984',_binary 'short chain dehydrogenase; Provisional.','PRK07454','PRK07454'),(960214,'gnl|CDD|168204',_binary 'oxidoreductase; Validated.','PRK05717','PRK05717'),(960216,'gnl|CDD|235794',_binary 'aldose dehydrogenase; Validated.','PRK06398','PRK06398'),(960217,'gnl|CDD|315503',_binary 'Helix-hairpin-helix motif.  The HhH domain is a short DNA-binding domain.','pfam12836','HHH_3'),(960218,'gnl|CDD|273825',_binary '3-oxoacyl-(acyl-carrier-protein) reductase, putative.  This model represents a small, very well conserved family of proteins closely related to the FabG family, TIGR01830, and possibly equal in function. In all completed genomes with a member of this family, a FabG in TIGR01830 is also found. [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR01831','fabG_rel'),(960220,'gnl|CDD|187613',_binary 'classical (c) SDR, subgroup 1.  These proteins are members of the classical SDR family, with a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05355','SDR_c1'),(960221,'gnl|CDD|180543',_binary 'hypothetical protein; Provisional.','PRK06361','PRK06361'),(960222,'gnl|CDD|181508',_binary 'short chain dehydrogenase; Provisional.','PRK08628','PRK08628'),(960224,'gnl|CDD|273823',_binary 'acetoacetyl-CoA reductase.  This model represent acetoacetyl-CoA reductase, a member of the family short-chain-alcohol dehydrogenases. Note that, despite the precision implied by the enzyme name, the reaction of EC 1.1.1.36 is defined more generally as (R)-3-hydroxyacyl-CoA + NADP+ = 3-oxoacyl-CoA + NADPH. Members of this family may act in the biosynthesis of poly-beta-hydroxybutyrate (e.g. Rhizobium meliloti) and related poly-beta-hydroxyalkanoates. Note that the member of this family from Azospirillum brasilense, designated NodG, appears to lack acetoacetyl-CoA reductase activity and to act instead in the production of nodulation factor. This family is downgraded to subfamily for this NodG. Other proteins designated NodG, as from Rhizobium, belong to related but distinct protein families.','TIGR01829','Acetoacetyl-CoA_reductase'),(960225,'gnl|CDD|316987',_binary 'Helix-hairpin-helix domain.','pfam14520','HHH_5'),(960226,'gnl|CDD|183718',_binary 'short chain dehydrogenase; Provisional.','PRK12746','PRK12746'),(960227,'gnl|CDD|224472',_binary 'DNA uptake protein and related DNA-binding proteins [DNA replication, recombination, and repair].','COG1555','ComEA'),(960228,'gnl|CDD|237189',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK12748','PRK12748'),(960229,'gnl|CDD|213597',_binary 'comEA protein.  This model describes the ComEA protein in bacteria. The com E locus is obligatory for bacterial cell competence - the process of internalizing the exogenous added DNA. Lesions in the loci has been variously described for the appearance of competence-related pheonotypes and impairment of competence, suggesting their intimate functional role in bacterial transformation. [Cellular processes, DNA transformation].','TIGR01259','ComE_operon_protein_1'),(960327,'gnl|CDD|187608',_binary 'classical (c) SDR, subgroup 6.  These proteins are members of the classical SDR family, with a canonical active site tetrad  and a fairly well conserved typical Gly-rich  NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05350','SDR_c6'),(960336,'gnl|CDD|223377',_binary 'Short-chain dehydrogenases of various substrate specificities [General function prediction only].','COG0300','DltE'),(960352,'gnl|CDD|187643',_binary '3-ketodihydrosphingosine reductase (KDSR) and related proteins, classical (c) SDR.  These proteins include members identified as KDSR, ribitol type dehydrogenase, and others. The group shows strong conservation of the active site tetrad and glycine rich NAD-binding motif of the classical SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08939','KDSR-like_SDR_c'),(960360,'gnl|CDD|235702',_binary 'gluconate 5-dehydrogenase; Provisional.','PRK06124','PRK06124'),(960386,'gnl|CDD|181139',_binary 'short chain dehydrogenase; Provisional.','PRK07832','PRK07832'),(960388,'gnl|CDD|181020',_binary 'enoyl-(acyl carrier protein) reductase; Provisional.','PRK07533','PRK07533'),(960392,'gnl|CDD|181416',_binary 'enoyl-(acyl carrier protein) reductase; Provisional.','PRK08415','PRK08415'),(960394,'gnl|CDD|236110',_binary 'short chain dehydrogenase; Provisional.','PRK07831','PRK07831'),(960459,'gnl|CDD|315976',_binary 'Helix-turn-helix domain.  This domain is a helix-turn-helix domain that probably binds to DNA.','pfam13413','HTH_25'),(960500,'gnl|CDD|223860',_binary 'Predicted transcriptional regulators [Transcription].','COG0789','SoxR'),(960501,'gnl|CDD|315974',_binary 'MerR HTH family regulatory protein.','pfam13411','MerR_1'),(960503,'gnl|CDD|131098',_binary 'Zn(II)-responsive transcriptional regulator.  This model represents the zinc and cadmium (II) responsive transcriptional activator of the gamma proteobacterial zinc efflux system. This protein is a member of the MerR family of transcriptional activators (pfam00376) and contains a distinctive pattern of cysteine residues in its metal binding loop, Cys-Cys-X(8-9)-Cys, as well as a conserved and critical cysteine at the N-terminal end of the dimerization helix. [Regulatory functions, DNA interactions].','TIGR02043','transcriptional_regulator'),(960508,'gnl|CDD|226478',_binary 'Predicted phosphoadenosine phosphosulfate sulfotransferase [General function prediction only].','COG3969','COG3969'),(960509,'gnl|CDD|238846',_binary 'This domain is found in phosphoadenosine phosphosulphate (PAPS) reductase enzymes or PAPS sulphotransferase. PAPS reductase is part of the adenine nucleotide alpha hydrolases superfamily also including N type ATP PPases and ATP sulphurylases. A highly modified version of the P loop, the fingerprint peptide of mononucleotide-binding proteins, is present in the active site of the protein, which appears to be a positively charged cleft containing a number of conserved arginine and lysine residues. Although PAPS reductase has no ATPase activity, it shows a striking similarity to the structure of the ATP pyrophosphatase (ATP PPase) domain of GMP synthetase, indicating that both enzyme families have evolved from a common ancestral nucleotide-binding fold.   The enzyme uses thioredoxin as an electron donor for the reduction of PAPS to phospho-adenosine-phosphate (PAP) . It is also found in NodP nodulation protein P from Rhizobium meliloti which has ATP sulphurylase activity (sulphate adenylate transferase) .','cd01713','PAPS_reductase'),(960510,'gnl|CDD|223253',_binary '3\'-phosphoadenosine 5\'-phosphosulfate sulfotransferase (PAPS reductase)/FAD synthetase and related enzymes [Amino acid transport and metabolism / Coenzyme metabolism].','COG0175','CysH'),(960511,'gnl|CDD|307589',_binary 'Phosphoadenosine phosphosulfate reductase family.  This domain is found in phosphoadenosine phosphosulfate (PAPS) reductase enzymes or PAPS sulfotransferase. PAPS reductase is part of the adenine nucleotide alpha hydrolases superfamily also including N type ATP PPases and ATP sulphurylases. The enzyme uses thioredoxin as an electron donor for the reduction of PAPS to phospho-adenosine-phosphate (PAP). It is also found in NodP nodulation protein P from Rhizobium which has ATP sulfurylase activity (sulfate adenylate transferase).','pfam01507','PAPS_reduct'),(960512,'gnl|CDD|181465',_binary 'hypothetical protein; Provisional.','PRK08557','PRK08557'),(960514,'gnl|CDD|319255',_binary 'immunoglobulin-binding regulator IbrB activates eib genes.  IbrB (along with IbrA) activates immunoglobulin-binding eib genes in Escherichia coli. IbrB is related to the ParB N-terminal domain family, which includes DNA-binding proteins involved in chromosomal/plasmid segregation and transcriptional regulation, consistent with a possible mechanism for IbrB in DNA binding-related regulation of eib expression. The ParB like family is a diverse domain superfamily with structural and sequence similarity to ParB of bacterial chromosomes/plasmid parABS partitioning system and Sulfiredoxin (Srx), which is a reactivator of oxidatively inactivated 2-cys peroxiredoxins. Other families includes proteins related to StrR, a putative regulator in the biosynthetic gene cluster and a family containing a Pyrococcus furiosus nuclease and putative transcriptional regulators SbnI (Staphylococcus aureus siderophore biosynthetic gene cluster ) and EdeB (Brevibacillus brevis antimicrobial peptide edeine biosynthetic cluster). Nuclease activity has also been reported in arabidopsis Srx.','cd16397','IbrB_like'),(960527,'gnl|CDD|226947',_binary 'Superfamily II RNA helicase [DNA replication, recombination, and repair].','COG4581','COG4581'),(960532,'gnl|CDD|237496',_binary 'Hef nuclease; Provisional.','PRK13766','PRK13766'),(960560,'gnl|CDD|197641',_binary 'Hint (Hedgehog/Intein) domain C-terminal region.  Hedgehog/Intein domain, C-terminal region. Domain has been split to accommodate large insertions of endonucleases.','smart00305','HintC'),(960566,'gnl|CDD|213622',_binary 'intein C-terminal splicing region.  This model represents the well-conserved C-terminal region of a large number of inteins. It is based on interated search results, starting with a curated collection of intein N-terminal splicing regions from InBase, the New England Biolabs Intein Database, as presented on its web site. Inteins are regions encoded within proteins from which they remove themselves after translation in a self-splicing reaction, leaving the remainder of the coding region to form a complete, functional protein as if the intein were never there. Proteins with inteins include RecA, GyrA, ribonucleotide reductase, and others. Most inteins have a central region with putative endonuclease activity.','TIGR01443','DNA-directed_RNA_polymera'),(960664,'gnl|CDD|273028',_binary 'type I site-specific deoxyribonuclease, HsdR family.  This gene is part of the type I restriction and modification system which is composed of three polypeptides R (restriction endonuclease), M (modification) and S (specificity). This group of enzymes recognize specific short DNA sequences and have an absolute requirement for ATP (or dATP) and S-adenosyl-L-methionine. They also catalyse the reactions of EC 2.1.1.72 and EC 2.1.1.73, with similar site specificity.(J. Mol. Biol. 271 (3), 342-348 (1997)). Members of this family are assumed to differ from each other in DNA site specificity. [DNA metabolism, Restriction/modification].','TIGR00348','R_protein'),(960669,'gnl|CDD|226583',_binary 'Superfamily II DNA/RNA helicase required for DNA uptake (late competence protein) [DNA replication, recombination, and repair].','COG4098','comFA'),(960672,'gnl|CDD|234587',_binary 'Holliday junction resolvase; Reviewed.','PRK00039','ruvC'),(960807,'gnl|CDD|189968',_binary 'Collagen triple helix repeat (20 copies).  Members of this family belong to the collagen superfamily. Collagens are generally extracellular structural proteins involved in formation of connective tissue structure. The alignment contains 20 copies of the G-X-Y repeat that forms a triple helix. The first position of the repeat is glycine, the second and third positions can be any residue but are frequently proline and hydroxy-proline. Collagens are post translationally modified by proline hydroxylase to form the hydroxy-proline residues. Defective hydroxylation is the cause of scurvy. Some members of the collagen superfamily are not involved in connective tissue structure but share the same triple helical structure. The family includes bacterial collagen-like triple-helix repeat proteins.','pfam01391','Collagen'),(960871,'gnl|CDD|274268',_binary 'thiamine kinase.  Members of this family are the ycfN gene product of Escherichia coli, now identified as the salvage enzyme thiamine kinase (thiK), and additional proteobacterial homologs taken to be orthologs with equivalent function. [Biosynthesis of cofactors, prosthetic groups, and carriers, Thiamine].','TIGR02721','Thiamine_kinase'),(961290,'gnl|CDD|106901',_binary 'putative NHN endonuclease.','PHA00280','PHA00280'),(961302,'gnl|CDD|314756',_binary 'Protein of unknown function (DUF3450).  This family of proteins are functionally uncharacterized. This protein is found in bacteria and eukaryotes. Proteins in this family are about 260 amino acids in length.','pfam11932','DUF3450'),(961356,'gnl|CDD|224000',_binary 'ATP-dependent exoDNAse (exonuclease V) beta subunit (contains helicase and exonuclease domains) [DNA replication, recombination, and repair].','COG1074','RecB'),(961414,'gnl|CDD|309183',_binary 'ArsC family.  This family is related to glutaredoxins pfam00462.','pfam03960','ArsC'),(961423,'gnl|CDD|279396',_binary 'MarR family.  The Mar proteins are involved in the multiple antibiotic resistance, a non-specific resistance system. The expression of the mar operon is controlled by a repressor, MarR. A large number of compounds induce transcription of the mar operon. This is thought to be due to the compound binding to MarR, and the resulting complex stops MarR binding to the DNA. With the MarR repression lost, transcription of the operon proceeds. The structure of MarR is known and shows MarR as a dimer with each subunit containing a winged-helix DNA binding motif.','pfam01047','MarR'),(961424,'gnl|CDD|238042',_binary 'Arsenical Resistance Operon Repressor and similar prokaryotic, metal regulated homodimeric repressors. ARSR subfamily of helix-turn-helix bacterial transcription regulatory proteins (winged helix topology). Includes several proteins that appear to dissociate from DNA in the presence of metal ions.','cd00090','HTH_ARSR'),(961425,'gnl|CDD|197713',_binary 'helix_turn_helix, Arsenical Resistance Operon Repressor.','smart00418','HTH_ARSR'),(961514,'gnl|CDD|133949',_binary 'membrane protein P6.','PHA01399','PHA01399'),(961515,'gnl|CDD|313985',_binary 'Protein of unknown function (DUF2637).  This family of proteins has no known function.','pfam10935','DUF2637'),(961622,'gnl|CDD|238260',_binary 'Shikimate kinase (SK) is the fifth enzyme in the shikimate pathway, a seven-step biosynthetic pathway which converts erythrose-4-phosphate to chorismic acid, found in bacteria, fungi and plants. Chorismic acid is a important intermediate in the synthesis of aromatic compounds, such as aromatic amino acids, p-aminobenzoic acid, folate and ubiquinone. Shikimate kinase catalyses the phosphorylation of the 3-hydroxyl group of shikimic acid using ATP.','cd00464','SK'),(961660,'gnl|CDD|182592',_binary 'histidine/lysine/arginine/ornithine transporter subunit; Provisional.','PRK10619','PRK10619'),(961669,'gnl|CDD|163689',_binary 'Mpr1p, Pad1p N-terminal (MPN) domains with catalytic isopeptidase activity (metal-binding); eukaryotic.  This family contains eukaryotic MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+) domains found in proteins with a variety of functions, including AMSH (associated molecule with the Src homology 3 domain (SH3) of STAM), H2A-DUB (histone H2A deubiquitinase), BRCC36 (BRCA1/BRCA2-containing complex subunit 36), as well as Rpn11 (regulatory particle number 11) and CSN5 (COP9 signalosome complex subunit 5). These domains contain the signature JAB1/MPN/Mov34 metalloenzyme (JAMM) motif, EXnHS/THX7SXXD, which is involved in zinc ion coordination and provides the active site for isopeptidase activity. Rpn11 is responsible for substrate deubiquitination during proteasomal degradation. It is essential for maintaining a correct cell cycle and normal mitochondrial morphology and physiology. CSN5 is critical for nuclear export and the degradation of several tumor suppressor proteins, including p53, p27, and Smad4. Over-expression of CSN5 has been implicated in cancer initiation and progression. AMSH specifically cleaves Lys 63 and not Lys48-linked polyubiquitin (poly-Ub) chains, thus facilitating the recycling and subsequent trafficking of receptors to the cell surface. It is involved in the degradation of EGF receptor (EGFR) and possibly other ubiquitinated endocytosed proteins. BRCC36 is part of the BRCA1/BRCA2/BARD1-containing nuclear complex that displays an E3 ubiquitin ligase activity; it is targeted to DNA damage foci after irradiation. 2A-DUB is specific for monoubiquitinated H2A (uH2A), regulating transcription by coordinating histone acetylation and deubiquitination, and destabilizing the association of linker histone H1 with nucleosomes. It is a positive regulator of androgen receptor (AR) transactivation activity on a reporter gene and serves as a marker in prostate tumors.','cd08058','MPN_euk_mb'),(961672,'gnl|CDD|279706',_binary 'JAB1/Mov34/MPN/PAD-1 ubiquitin protease.  Members of this family are found in proteasome regulatory subunits, eukaryotic initiation factor 3 (eIF3) subunits and regulators of transcription factors. This family is also known as the MPN domain and PAD-1-like domain, JABP1 domain or JAMM domain. These are metalloenzymes that function as the ubiquitin isopeptidase/ deubiquitinase in the ubiquitin-based signalling and protein turnover pathways in eukaryotes. Versions of the domain in prokaryotic cognates of the ubiquitin-modification pathway are shown to have a similar role, and the archael protein from Haloferax volcanii is found to cleave ubiquitin-like small archaeal modifier proteins (SAMP1/2) from protein conjugates.','pfam01398','JAB'),(961674,'gnl|CDD|214495',_binary 'Fibronectin type 3 domain.  One of three types of internal repeat within the plasma protein, fibronectin. The tenth fibronectin type III repeat contains a RGD cell recognition sequence in a flexible loop between 2 strands. Type III modules are present in both extracellular and intracellular proteins.','smart00060','FN3'),(961855,'gnl|CDD|309550',_binary 'Protein of unknown function (DUF550).  This family is found in a range of Proteobacteria and a few P-22 dsDNA virus particles. The function is currently not known.','pfam04447','DUF550'),(961979,'gnl|CDD|185275',_binary 'E3 ubiquitin-protein ligase SopA; Provisional.','PRK15377','PRK15377'),(962053,'gnl|CDD|227633',_binary 'Uncharacterized conserved protein [Function unknown].','COG5323','COG5323'),(962054,'gnl|CDD|314594',_binary 'Protein of unknwon function (DUF3310).  This is a family of conserved bacteriophage proteins of unknown function.','pfam11753','DUF3310'),(962485,'gnl|CDD|314791',_binary 'Scavenger mRNA decapping enzyme C-term binding.  This family consists of several scavenger mRNA decapping enzymes (DcpS) and is the C-terminal region. DcpS is a scavenger pyrophosphatase that hydrolyses the residual cap structure following 3\' to 5\' decay of an mRNA. The association of DcpS with 3\' to 5\' exonuclease exosome components suggests that these two activities are linked and there is a coupled exonucleolytic decay-dependent decapping pathway. The C-terminal domain contains a histidine triad (HIT) sequence with three histidines separated by hydrophobic residues. The central histidine within the DcpS HIT motif is critical for decapping activity and defines the HIT motif as a new mRNA decapping domain, making DcpS the first member of the HIT family of proteins with a defined biological function.','pfam11969','DcpS_C'),(962487,'gnl|CDD|309906',_binary 'Phage tail sheath protein.  This family includes a variety of phage tail sheath proteins.','pfam04984','Phage_sheath_1'),(962488,'gnl|CDD|282340',_binary 'Phage terminase large subunit.  Initiation of packaging of double-stranded viral DNA involves the specific interaction of the prohead with viral DNA in a process mediated by a phage-encoded terminase protein. The terminase enzymes are usually hetero-oligomers composed of a small and a large subunit. This region is found on the large subunit and possess an endonuclease and ATPase activity that require Mg2+ and a neutral or slightly basic reaction. This region is also found in bacterial sequences.','pfam04466','Terminase_3'),(962494,'gnl|CDD|225836',_binary 'Uncharacterized homolog of phage Mu protein gp47 [Function unknown].','COG3299','XkdT'),(962495,'gnl|CDD|309184',_binary 'Protein of unknown function (DUF342).  This family of bacterial proteins has no known function. The proteins are in the region of 500-600 amino acid residues in length.','pfam03961','DUF342'),(962504,'gnl|CDD|238119',_binary 'Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses  In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains.','cd00198','vWFA'),(962505,'gnl|CDD|214621',_binary 'von Willebrand factor (vWF) type A domain.  VWA domains in extracellular eukaryotic proteins mediate adhesion via metal ion-dependent adhesion sites (MIDAS). Intracellular VWA domains and homologues in prokaryotes have recently been identified. The proposed VWA domains in integrin beta subunits have recently been substantiated using sequence-based methods.','smart00327','VWA'),(962512,'gnl|CDD|235112',_binary 'acetylornithine aminotransferase; Provisional.','PRK03244','argD'),(962521,'gnl|CDD|240281',_binary 'ornithine aminotransferase-like protein; Provisional.','PTZ00125','PTZ00125'),(962522,'gnl|CDD|236353',_binary 'primosomal protein DnaI; Reviewed.','PRK08939','PRK08939'),(962525,'gnl|CDD|180390',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK06082','PRK06082'),(962528,'gnl|CDD|129783',_binary '4-aminobutyrate aminotransferase, prokaryotic type.  This enzyme is a class III pyridoxal-phosphate-dependent aminotransferase. This model describes known bacterial examples of the enzyme. The best archaeal matches are presumed but not trusted to have the equivalent function. The degree of sequence difference between this set and known eukaryotic (mitochondrial) examples is greater than the distance to some proteins known to have different functions, and so separate models are built for prokaryotic and eukaryotic sets. E. coli has two isozymes. Alternate names include GABA transaminase, gamma-amino-N-butyrate transaminase, and beta-alanine--oxoglutarate aminotransferase. [Central intermediary metabolism, Other].','TIGR00700','semialdehyde_transaminase'),(962529,'gnl|CDD|168967',_binary 'hypothetical protein; Provisional.','PRK07481','PRK07481'),(962530,'gnl|CDD|179505',_binary 'acetylornithine aminotransferase; Provisional.','PRK02936','argD'),(962531,'gnl|CDD|179803',_binary 'acetylornithine aminotransferase; Provisional.','PRK04260','PRK04260'),(962532,'gnl|CDD|181234',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK08117','PRK08117'),(962533,'gnl|CDD|236026',_binary 'hypothetical protein; Provisional.','PRK07482','PRK07482'),(962534,'gnl|CDD|274490',_binary 'succinylornithine transaminase family.  Members of the seed alignment for this protein family are the enzyme succinylornithine transaminase (EC 2.6.1.81), which catalyzes the third of five steps in arginine succinyltransferase (AST) pathway, an ammonia-releasing pathway of arginine degradation. All seed alignment sequences are found within arginine succinyltransferase operons, and all proteins that score above 820.0 bits should function as succinylornithine transaminase. However, a number of sequences extremely closely related in sequence, found in different genomic contexts, are likely to act in different biological processes and may act on different substrates. This model is desigated subfamily rather than equivalog, pending further consideration, for this reason. [Energy metabolism, Amino acids and amines].','TIGR03246','arg_catab_astC'),(962535,'gnl|CDD|235889',_binary 'diaminobutyrate--2-oxoglutarate aminotransferase; Provisional.','PRK06931','PRK06931'),(962536,'gnl|CDD|179636',_binary 'acetylornithine transaminase protein; Provisional.','PRK03715','argD'),(962537,'gnl|CDD|274492',_binary 'L-lysine 6-transaminase.  Characterized members of this protein family are L-lysine 6-transaminase, also called lysine epsilon-aminotransferase (LAT). The immediate product of the reaction of this enzyme on lysine, 2-aminoadipate 6-semialdehyde, becomes 1-piperideine 6-carboxylate, or P6C. This product may be converted subsequently to pipecolate or alpha-aminoadipate, lysine catabolites that may be precursors of certain seconary metabolites.','TIGR03251','LAT_fam'),(962538,'gnl|CDD|235885',_binary '4-aminobutyrate aminotransferase; Reviewed.','PRK06918','PRK06918'),(962539,'gnl|CDD|235913',_binary 'hypothetical protein; Provisional.','PRK07036','PRK07036'),(962542,'gnl|CDD|181078',_binary 'aminotransferase; Validated.','PRK07678','PRK07678'),(962544,'gnl|CDD|236224',_binary 'L-lysine aminotransferase; Provisional.','PRK08297','PRK08297'),(962547,'gnl|CDD|236149',_binary '4-aminobutyrate aminotransferase; Validated.','PRK08088','PRK08088'),(962552,'gnl|CDD|129792',_binary '2,4-diaminobutyrate 4-transaminases.  This family consists of L-diaminobutyric acid transaminases. This general designation covers both 2.6.1.76 (diaminobutyrate-2-oxoglutarate transaminase, which uses glutamate as the amino donor in DABA biosynthesis), and 2.6.1.46 (diaminobutyrate--pyruvate transaminase, which uses alanine as the amino donor). Most members with known function are 2.6.1.76, and at least some annotations as 2.6.1.46 in current databases at time of model revision are incorrect. A distinct branch of this family contains examples of 2.6.1.76 nearly all of which are involved in ectoine biosynthesis. A related enzyme is 4-aminobutyrate aminotransferase (EC 2.6.1.19), also called GABA transaminase. These enzymes all are pyridoxal phosphate-containing class III aminotransferase. [Central intermediary metabolism, Other].','TIGR00709','Diaminobutyrate--2-oxoglu'),(962558,'gnl|CDD|183175',_binary 'putrescine--2-oxoglutarate aminotransferase; Provisional.','PRK11522','PRK11522'),(962560,'gnl|CDD|132415',_binary 'putrescine aminotransferase.  Members of this family are putrescine aminotransferase, as found in Escherichia coli, Erwinia carotovora subsp. atroseptica, and closely related species. This pyridoxal phosphate enzyme, as characterized in E. coli, can act also on cadaverine and, more weakly, spermidine. [Central intermediary metabolism, Polyamine biosynthesis].','TIGR03372','Putrescine_aminotransfera'),(962562,'gnl|CDD|183486',_binary 'bifunctional succinylornithine transaminase/acetylornithine transaminase; Provisional.','PRK12381','PRK12381'),(962564,'gnl|CDD|273109',_binary 'adenosylmethionine-8-amino-7-oxononanoate transaminase.  All members of the seed alignment have been demonstrated experimentally to act as EC 2.6.1.62, an enzyme in the biotin biosynthetic pathway. Alternate names include 7,8-diaminopelargonic acid aminotransferase, DAPA aminotransferase, and adenosylmethionine-8-amino-7-oxononanoate aminotransferase. The gene symbol is bioA in E. coli and BIO3 in S. cerevisiae. [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin].','TIGR00508','hypothetical_protein_DESP'),(962566,'gnl|CDD|235884',_binary 'hypothetical protein; Provisional.','PRK06917','PRK06917'),(962568,'gnl|CDD|215335',_binary 'ornithine-delta-aminotransferase.','PLN02624','PLN02624'),(962570,'gnl|CDD|181401',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK08360','PRK08360'),(962572,'gnl|CDD|168145',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK05639','PRK05639'),(962573,'gnl|CDD|222864',_binary '41 helicase; Provisional.','PHA02542','41'),(962574,'gnl|CDD|273853',_binary 'ornithine aminotransferase.  This model describes the final step in the biosynthesis of ornithine from glutamate via the non-acetylated pathway. Ornithine amino transferase takes L-glutamate 5-semialdehyde and makes it into ornithine, which is used in the urea cycle, as well as in the biosynthesis of arginine. This model includes low-GC bacteria and eukaryotic species. The genes from two species are annotated as putative acetylornithine aminotransferases - one from Porphyromonas gingivalis (OMNI|PG1271), and the other from Staphylococcus aureus (OMNI|SA0170). After homology searching using BLAST it was determined that these two sequences were most closely related to ornithine aminotransferases. This model\'s seed includes one characterized hit, from Bacillus subtilis (SP|P38021).','TIGR01885','Ornithine_aminotransferas'),(962576,'gnl|CDD|181493',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK08593','PRK08593'),(962578,'gnl|CDD|236027',_binary 'hypothetical protein; Provisional.','PRK07483','PRK07483'),(962580,'gnl|CDD|235685',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK06058','PRK06058'),(962581,'gnl|CDD|236032',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK07495','PRK07495'),(962583,'gnl|CDD|180746',_binary 'adenosylmethionine--8-amino-7-oxononanoate transaminase; Provisional.','PRK06916','PRK06916'),(962584,'gnl|CDD|180690',_binary '4-aminobutyrate aminotransferase; Provisional.','PRK06777','PRK06777'),(962585,'gnl|CDD|129782',_binary '4-aminobutyrate aminotransferase, eukaryotic type.  This enzyme is a class III pyridoxal-phosphate-dependent aminotransferase. This model describes known eukaryotic examples of the enzyme. The degree of sequence difference between this set and known bacterial examples is greater than the distance between either set the most similar enzyme with distinct function, and so separate models are built for prokaryotic and eukaryotic sets. Alternate names include GABA transaminase, gamma-amino-N-butyrate transaminase, and beta-alanine--oxoglutarate aminotransferase. [Central intermediary metabolism, Other].','TIGR00699','4-aminobutyrate_aminotran'),(962586,'gnl|CDD|234848',_binary 'ornithine--oxo-acid transaminase; Reviewed.','PRK00854','rocD'),(962587,'gnl|CDD|183999',_binary 'omega amino acid--pyruvate transaminase; Provisional.','PRK13360','PRK13360'),(962588,'gnl|CDD|179868',_binary 'acetylornithine transaminase protein; Provisional.','PRK04612','argD'),(962589,'gnl|CDD|179736',_binary 'ornithine--oxo-acid transaminase; Provisional.','PRK04073','rocD'),(962590,'gnl|CDD|235892',_binary 'diaminobutyrate--2-oxoglutarate aminotransferase; Provisional.','PRK06938','PRK06938'),(962591,'gnl|CDD|274090',_binary 'chaperone protein DnaJ.  This model represents bacterial forms of DnaJ, part of the DnaK-DnaJ-GrpE chaperone system. The three components typically are encoded by consecutive genes. DnaJ homologs occur in many genomes, typically not near DnaK and GrpE-like genes; most such genes are not included by this family. Eukaryotic (mitochondrial and chloroplast) forms are not included in the scope of this family.','TIGR02349','Chaperone_protein_DnaJ'),(962592,'gnl|CDD|235716',_binary 'hypothetical protein; Provisional.','PRK06149','PRK06149'),(962594,'gnl|CDD|235656',_binary 'adenosylmethionine--8-amino-7-oxononanoate transaminase; Provisional.','PRK05964','PRK05964'),(962596,'gnl|CDD|182078',_binary '4-aminobutyrate transaminase; Provisional.','PRK09792','PRK09792'),(962597,'gnl|CDD|223560',_binary 'DnaJ-class molecular chaperone with C-terminal Zn finger domain [Posttranslational modification, protein turnover, chaperones].','COG0484','DnaJ'),(962598,'gnl|CDD|235823',_binary 'hypothetical protein; Provisional.','PRK06541','PRK06541'),(962600,'gnl|CDD|180330',_binary 'hypothetical protein; Provisional.','PRK05965','PRK05965'),(962602,'gnl|CDD|171472',_binary 'putative aminotransferase; Provisional.','PRK12403','PRK12403'),(962603,'gnl|CDD|237654',_binary 'chaperone protein DnaJ; Provisional.','PRK14278','PRK14278'),(962604,'gnl|CDD|273230',_binary 'glutamate-1-semialdehyde-2,1-aminomutase.  This enzyme, glutamate-1-semialdehyde-2,1-aminomutase (glutamate-1-semialdehyde aminotransferase, GSA aminotransferase), contains a pyridoxal phosphate attached at a Lys residue at position 283 of the seed alignment. It is in the family of class III aminotransferases. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR00713','Glutamate-1-semialdehyde_'),(962611,'gnl|CDD|197617',_binary 'DnaJ molecular chaperone homology domain.','smart00271','DnaJ'),(962612,'gnl|CDD|99751',_binary 'DnaJ domain or J-domain.  DnaJ/Hsp40 (heat shock protein 40) proteins are highly conserved and play crucial roles in protein translation, folding, unfolding, translocation, and degradation. They act primarily by stimulating the ATPase activity of Hsp70s, an important chaperonine family. Hsp40 proteins are characterized by the presence of a J domain, which mediates the interaction with Hsp70. They may contain other domains as well, and the architectures provide a means of classification.','cd06257','DnaJ'),(962613,'gnl|CDD|184611',_binary 'chaperone protein DnaJ; Provisional.','PRK14297','PRK14297'),(962614,'gnl|CDD|237665',_binary 'chaperone protein DnaJ; Provisional.','PRK14295','PRK14295'),(962615,'gnl|CDD|184599',_binary 'chaperone protein DnaJ; Provisional.','PRK14277','PRK14277'),(962616,'gnl|CDD|236757',_binary 'chaperone protein DnaJ; Provisional.','PRK10767','PRK10767'),(962617,'gnl|CDD|237658',_binary 'chaperone protein DnaJ; Provisional.','PRK14284','PRK14284'),(962619,'gnl|CDD|172788',_binary 'chaperone protein DnaJ; Provisional.','PRK14300','PRK14300'),(962621,'gnl|CDD|172773',_binary 'chaperone protein DnaJ; Provisional.','PRK14285','PRK14285'),(962622,'gnl|CDD|237668',_binary 'chaperone protein DnaJ; Provisional.','PRK14301','PRK14301'),(962623,'gnl|CDD|274808',_binary 'terminal organelle assembly protein TopJ.  This model describes TopJ (MG_200, CbpA), a DnaJ homolog and probable assembly protein of the Mycoplasma terminal organelle. The terminal organelle is involved in both cytadherence and gliding motility. [Cellular processes, Chemotaxis and motility].','TIGR03835','DnaJ-like_protein_MG200_h'),(962624,'gnl|CDD|237657',_binary 'chaperone protein DnaJ; Provisional.','PRK14281','PRK14281'),(962625,'gnl|CDD|184603',_binary 'chaperone protein DnaJ; Provisional.','PRK14282','PRK14282'),(962626,'gnl|CDD|172774',_binary 'chaperone protein DnaJ; Provisional.','PRK14286','PRK14286'),(962627,'gnl|CDD|240236',_binary 'DnaJ_C chaperone protein; Provisional.','PTZ00037','PTZ00037'),(962628,'gnl|CDD|237664',_binary 'chaperone protein DnaJ; Provisional.','PRK14294','PRK14294'),(962629,'gnl|CDD|225124',_binary 'DnaJ-class molecular chaperone [Posttranslational modification, protein turnover, chaperones].','COG2214','CbpA'),(962631,'gnl|CDD|237660',_binary 'chaperone protein DnaJ; Provisional.','PRK14289','PRK14289'),(962633,'gnl|CDD|227694',_binary 'Preprotein translocase subunit Sec63 [Intracellular trafficking and secretion].','COG5407','SEC63'),(962684,'gnl|CDD|238741',_binary 'VWA subfamily: Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses  In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains. Members of this subgroup have no assigned function. This subfamily is typified by the presence of a conserved MIDAS motif.','cd01464','vWA_subfamily'),(962690,'gnl|CDD|314591',_binary 'Protein of unknown function (DUF3307).  This family of bacterial proteins has no known function.','pfam11750','DUF3307'),(962697,'gnl|CDD|316340',_binary 'Domain of unknown function (DUF4185).','pfam13810','DUF4185'),(962701,'gnl|CDD|214361',_binary 'Clp protease ATP binding subunit.','CHL00095','clpC'),(962702,'gnl|CDD|223616',_binary 'ATP-binding subunits of Clp protease and DnaK/DnaJ chaperones [Posttranslational modification, protein turnover, chaperones].','COG0542','clpA'),(962703,'gnl|CDD|308480',_binary 'Clp amino terminal domain, pathogenicity island component.  This short domain is found in one or two copies at the amino terminus of ClpA and ClpB proteins from bacteria and eukaryotes. The function of these domains is uncertain but they may form a protein binding site. In many bacterial species, including E.coli, this region represents the N-terminus of one of the key components of the pathogenicity island complex that injects toxin from one bacterium into another.','pfam02861','Clp_N'),(962705,'gnl|CDD|274529',_binary 'ATP-dependent chaperone ClpB.  Members of this protein family are the bacterial ATP-dependent chaperone ClpB. This protein belongs to the AAA family, ATPases associated with various cellular activities (pfam00004). This molecular chaperone does not act as a protease, but rather serves to disaggregate misfolded and aggregated proteins. [Protein fate, Protein folding and stabilization].','TIGR03346','chaperone_ClpB'),(962707,'gnl|CDD|316737',_binary 'Domain of unknown function (DUF4343).  This domain family is found in bacteria, eukaryotes and viruses, and is typically between 127 and 142 amino acids in length.','pfam14243','DUF4343'),(962708,'gnl|CDD|275027',_binary 'ribonucleoside-diphosphate reductase, class 1b, beta subunit.  Members of this family are NrdF, the beta subunit of class 1b ribonucleotide reductase. This form uses a dimanganese moiety associated with a tyrosine radical to reduce the cellular requirement for iron. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR04171','ribonucleotide-diphosphat'),(962709,'gnl|CDD|184425',_binary 'ribonucleotide-diphosphate reductase subunit beta; Provisional.','PRK13965','PRK13965'),(962710,'gnl|CDD|140022',_binary 'ribonucleotide-diphosphate reductase subunit beta; Provisional.','PRK13966','nrdF2'),(962711,'gnl|CDD|236591',_binary 'ribonucleotide-diphosphate reductase subunit beta; Reviewed.','PRK09614','nrdF'),(962713,'gnl|CDD|140023',_binary 'ribonucleotide-diphosphate reductase subunit beta; Provisional.','PRK13967','nrdF1'),(962715,'gnl|CDD|223286',_binary 'Ribonucleotide reductase, beta subunit [Nucleotide transport and metabolism].','COG0208','NrdF'),(962716,'gnl|CDD|306723',_binary 'Ribonucleotide reductase, small chain.','pfam00268','Ribonuc_red_sm'),(962717,'gnl|CDD|153108',_binary 'Ribonucleotide Reductase, R2/beta subunit, ferritin-like diiron-binding domain.  Ribonucleotide Reductase, R2/beta subunit (RNRR2) is a member of a broad superfamily of ferritin-like diiron-carboxylate proteins. The RNR protein catalyzes the conversion of ribonucleotides to deoxyribonucleotides and is found in all eukaryotes, many prokaryotes, several viruses, and few archaea. The catalytically active form of RNR is a proposed alpha2-beta2 tetramer. The homodimeric alpha subunit (R1) contains the active site and redox active cysteines as well as the allosteric binding sites. The beta subunit (R2) contains a diiron cluster that, in its reduced state, reacts with dioxygen to form a stable tyrosyl radical and a diiron(III) cluster. This essential tyrosyl radical is proposed to generate a thiyl radical, located on a cysteine residue in the R1 active site that initiates ribonucleotide reduction. The beta subunit is composed of 10-13 helices, the 8 longest helices form an alpha-helical bundle; some have 2 addition beta strands. Yeast is unique in that it assembles both homodimers and heterodimers of RNRR2. The yeast heterodimer, Y2Y4, contains R2 (Y2) and a R2 homolog (Y4) that lacks the diiron center and is proposed to only assist in cofactor assembly, and perhaps stabilize R1 (Y1) in its active conformation.','cd01049','RNRR2'),(962718,'gnl|CDD|181647',_binary 'ribonucleotide-diphosphate reductase subunit beta; Reviewed.','PRK09101','nrdB'),(962719,'gnl|CDD|235968',_binary 'ribonucleotide-diphosphate reductase subunit beta; Validated.','PRK07209','PRK07209'),(962720,'gnl|CDD|240315',_binary 'ribonucleoside-diphosphate reductase small subunit; Provisional.','PTZ00211','PTZ00211'),(962721,'gnl|CDD|215272',_binary 'ribonucleoside-diphosphate reductase.','PLN02492','PLN02492'),(962722,'gnl|CDD|139206',_binary 'bifunctional gluaredoxin/ribonucleoside-diphosphate reductase subunit beta; Provisional.','PRK12759','PRK12759'),(962723,'gnl|CDD|311712',_binary 'Ion channel.  This family includes the two membrane helix type ion channels found in bacteria.','pfam07885','Ion_trans_2'),(962724,'gnl|CDD|236711',_binary 'voltage-gated potassium channel; Provisional.','PRK10537','PRK10537'),(962725,'gnl|CDD|223538',_binary 'Phosphoribosylpyrophosphate synthetase [Nucleotide transport and metabolism / Amino acid transport and metabolism].','COG0462','PrsA'),(962726,'gnl|CDD|273523',_binary 'ribose-phosphate pyrophosphokinase.  Alternate name: phosphoribosylpyrophosphate synthetase In some systems, close homologs lacking enzymatic activity exist and perform regulatory functions. The model is designated subfamily rather than equivalog for this reason. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01251','Ribose-phosphate_pyrophos'),(962727,'gnl|CDD|234868',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK00934','PRK00934'),(962728,'gnl|CDD|234929',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK01259','PRK01259'),(962729,'gnl|CDD|206754',_binary 'Phosphoribosyl transferase (PRT)-type I domain.  Phosphoribosyl transferase (PRT) domain. The type I PRTases are identified by a conserved PRPP binding motif which features two adjacent acidic residues surrounded by one or more hydrophobic residue. PRTases catalyze the displacement of the alpha-1\'-pyrophosphate of 5-phosphoribosyl-alpha1-pyrpphosphate (PRPP) by a nitrogen-containing nucleophile. The reaction products are an alpha-1 substituted ribose-5\'-phosphate and a free pyrophosphate (PP).  PRPP, an activated form of ribose-5-phosphate, is a key metabolite connecting nucleotide synthesis and salvage pathways. The type I PRTase family includes a range of diverse phosphoribosyl transferase enzymes and regulatory proteins of the nucleotide synthesis and salvage pathways, including adenine phosphoribosyltransferase EC:2.4.2.7., hypoxanthine-guanine-xanthine phosphoribosyltransferase, hypoxanthine phosphoribosyltransferase EC:2.4.2.8., ribose-phosphate pyrophosphokinase EC:2.7.6.1., amidophosphoribosyltransferase EC:2.4.2.14., orotate phosphoribosyltransferase EC:2.4.2.10., uracil phosphoribosyltransferase EC:2.4.2.9., and xanthine-guanine phosphoribosyltransferase EC:2.4.2.22.','cd06223','PRTases_typeI'),(962730,'gnl|CDD|167353',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK02269','PRK02269'),(962732,'gnl|CDD|215209',_binary 'ribose-phosphate pyrophosphokinase.','PLN02369','PLN02369'),(962734,'gnl|CDD|179062',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK00553','PRK00553'),(962736,'gnl|CDD|235960',_binary 'phosphoribosylpyrophosphate synthetase; Provisional.','PRK07199','PRK07199'),(962737,'gnl|CDD|310381',_binary 'NACHT domain.  This NTPase domain is found in apoptosis proteins as well as those involved in MHC transcription activation. This family is closely related to pfam00931.','pfam05729','NACHT'),(962738,'gnl|CDD|180714',_binary 'phosphoribosylpyrophosphate synthetase; Provisional.','PRK06827','PRK06827'),(962739,'gnl|CDD|235072',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK02812','PRK02812'),(962740,'gnl|CDD|179893',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK04923','PRK04923'),(962741,'gnl|CDD|240290',_binary 'phosphoribosylpyrophosphate synthetase; Provisional.','PTZ00145','PTZ00145'),(962742,'gnl|CDD|177934',_binary 'ribose-phosphate pyrophosphokinase.','PLN02297','PLN02297'),(962743,'gnl|CDD|235039',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK02458','PRK02458'),(962744,'gnl|CDD|222898',_binary 'nicotinamide phosphoribosyl transferase; Provisional.','PHA02594','nadV'),(962745,'gnl|CDD|306630',_binary 'Phosphoribosyl transferase domain.  This family includes a range of diverse phosphoribosyl transferase enzymes. This family includes: Adenine phosphoribosyl-transferase EC:2.4.2.7, Hypoxanthine-guanine-xanthine phosphoribosyl-transferase, Hypoxanthine phosphoribosyl-transferase EC:2.4.2.8. Ribose-phosphate pyrophosphokinase i EC:2.7.6.1. Amidophosphoribosyltransferase EC:2.4.2.14. Orotate phosphoribosyl-transferase EC:2.4.2.10, Uracil phosphoribosyl-transferase EC:2.4.2.9, Xanthine-guanine phosphoribosyl-transferase EC:2.4.2.22. In Arabidopsis, at the very N-terminus of this domain is the P-Loop NTPase domain.','pfam00156','Pribosyltran'),(962746,'gnl|CDD|236407',_binary 'putative nicotinate phosphoribosyltransferase; Provisional.','PRK09198','PRK09198'),(962747,'gnl|CDD|238803',_binary 'pre-B-cell colony-enhancing factor (PBEF)-like. The mammalian members of this group of nicotinate phosphoribosyltransferases (NAPRTases) were originally identified as genes whose expression is upregulated upon activation in lymphoid cells. In general, nicotinate phosphoribosyltransferase catalyses the formation of NAMN and PPi from 5-phosphoribosy -1-pyrophosphate (PRPP) and nicotinic acid, this is the first, and also rate limiting, reaction in the NAD salvage synthesis.','cd01569','PBEF_like'),(962748,'gnl|CDD|238801',_binary 'Nicotinate phosphoribosyltransferase (NAPRTase) family. Nicotinate phosphoribosyltransferase catalyses the formation of NAMN and PPi from 5-phosphoribosy -1-pyrophosphate (PRPP) and nicotinic acid, this is the first, and also rate limiting, reaction in the NAD salvage synthesis. This salvage pathway serves to recycle NAD degradation products.','cd01567','NAPRTase_PncB'),(962749,'gnl|CDD|282015',_binary 'Nicotinate phosphoribosyltransferase (NAPRTase) family.  Nicotinate phosphoribosyltransferase (EC:2.4.2.11) is the rate limiting enzyme that catalyzes the first reaction in the NAD salvage synthesis. This family also includes Pre-B cell enhancing factor that is a cytokine (NAMPT). This family is related to Quinolinate phosphoribosyltransferase pfam01729.','pfam04095','NAPRTase'),(962750,'gnl|CDD|224405',_binary 'Nicotinic acid phosphoribosyltransferase [Coenzyme metabolism].','COG1488','PncB'),(962751,'gnl|CDD|238286',_binary 'Phosphoribosyltransferase (PRTase) type II; This family contains two enzymes that play an important role in NAD production by either allowing quinolinic acid (QA) , quinolinate phosphoribosyl transferase (QAPRTase), or nicotinic acid (NA), nicotinate phosphoribosyltransferase (NAPRTase), to be used in the synthesis of NAD. QAPRTase catalyses the reaction of quinolinic acid (QA) with 5-phosphoribosyl-1-pyrophosphate (PRPP) in the presence of Mg2+ to produce nicotinic acid mononucleotide (NAMN), pyrophosphate and carbon dioxide, an important step in the de novo synthesis of NAD. NAPRTase catalyses a similar reaction leading to NAMN and pyrophosphate, using nicotinic acid an PPRP as substrates, used in the NAD salvage pathway.','cd00516','PRTase_typeII'),(962755,'gnl|CDD|131739',_binary 'tRNA adenylyltransferase.  The enzyme tRNA adenylyltransferase, also called tRNA-nucleotidyltransferase and CCA-adding enzyme, can add or repair the required CCA triplet at the 3\'-end of tRNA molecules. Genes encoding tRNA include the CCA tail in some but not all bacteria, and this enzyme may be required for viability. Members of this family represent a distinct clade within the larger family pfam01743 (tRNA nucleotidyltransferase/poly(A) polymerase family protein). The example from Streptomyces coelicolor was shown to act as a CCA-adding enzyme and not as a poly(A) polymerase. [Protein synthesis, tRNA and rRNA base modification].','TIGR02692','putative_tRNA_nucleotidyl'),(962756,'gnl|CDD|223690',_binary 'tRNA nucleotidyltransferase/poly(A) polymerase [Translation, ribosomal structure and biogenesis].','COG0617','PcnB'),(962757,'gnl|CDD|237339',_binary 'tRNA CCA-pyrophosphorylase; Provisional.','PRK13299','PRK13299'),(962758,'gnl|CDD|280000',_binary 'Poly A polymerase head domain.  This family includes nucleic acid independent RNA polymerases, such as Poly(A) polymerase, which adds the poly (A) tail to mRNA EC:2.7.7.19. This family also includes the tRNA nucleotidyltransferase that adds the CCA to the 3\' of the tRNA EC:2.7.7.25. This family is part of the nucleotidyltransferase superfamily.','pfam01743','PolyA_pol'),(962759,'gnl|CDD|143388',_binary 'Nucleotidyltransferase (NT) domain of ClassII CCA-adding enzymes.  CCA-adding enzymes add the sequence [cytidine(C)-cytidine-adenosine (A)], one nucleotide at a time, onto the 3\' end of tRNA, in a template-independent reaction. This Class II group is comprised mainly of eubacterial and eukaryotic enzymes and includes Bacillus stearothermophilus CCAase, Escherichia coli poly(A) polymerase I, human mitochondrial CCAase, and Saccharomyces cerevisiae CCAase (CCA1). CCA-adding enzymes have a single catalytic pocket, which recognizes both ATP and CTP substrates. Included in this subgroup are CC- and A-adding enzymes from various ancient species of bacteria such as Aquifex aeolicus; these enzymes collaborate to add CCA to tRNAs. This family belongs to the Pol beta-like NT superfamily. In the majority of enzymes in this superfamily, two carboxylates, Dx[D/E], together with a third more distal carboxylate, coordinate two divalent metal cations involved in a two-metal ion mechanism of nucleotide addition. These carboxylate residues are fairly well conserved in this family. Escherichia coli CCAase is related to this group but has not been included in this alignment as this enzyme lacks the N-terminal helix conserved in the remainder of the NT superfamily.','cd05398','NT_ClassII-CCAase'),(962760,'gnl|CDD|130997',_binary 'poly(A) polymerase.  This model describes the pcnB family of poly(A) polymerases (also known as plasmid copy number protein). These enzymes sequentially add adenosine nucleotides to the 3\' end of RNAs, targeting them for degradation by the cell. This was originally described for anti-sense RNAs, but was later demonstrated for mRNAs as well. Members of this family are as yet limited to the gamma- and beta-proteobacteria, with putative members in the Chlamydiacae and spirochetes. This family has homology to tRNA nucleotidyltransferase (cca).','TIGR01942','pcnB'),(962761,'gnl|CDD|236939',_binary 'poly(A) polymerase I; Provisional.','PRK11623','pcnB'),(962762,'gnl|CDD|237338',_binary 'tRNA CCA-pyrophosphorylase; Provisional.','PRK13298','PRK13298'),(962763,'gnl|CDD|182810',_binary 'multifunctional tRNA nucleotidyl transferase/2\'3\'-cyclic phosphodiesterase/2\'nucleotidase/phosphatase; Reviewed.','PRK10885','cca'),(962764,'gnl|CDD|106256',_binary 'tRNA CCA-pyrophosphorylase; Provisional.','PRK13296','PRK13296'),(962765,'gnl|CDD|139469',_binary 'tRNA CCA-pyrophosphorylase; Provisional.','PRK13297','PRK13297'),(962766,'gnl|CDD|316533',_binary 'Protein of unknown function (DUF4236).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 69 and 402 amino acids in length.','pfam14020','DUF4236'),(962767,'gnl|CDD|315326',_binary 'Probable RNA and SrmB- binding site of polymerase A.  This region encompasses much of the RNA and SrmB binding motifs on polymerase A.','pfam12627','PolyA_pol_RNAbd'),(962793,'gnl|CDD|316637',_binary 'Domain of unknown function (DUF4294).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria. Proteins in this family are typically between 192 and 226 amino acids in length.','pfam14127','DUF4294'),(962798,'gnl|CDD|316120',_binary '2\'-5\' RNA ligase superfamily.  This family contains proteins related to pfam02834. These proteins are likely to be enzymes, but they may not share the RNA ligase activity.','pfam13563','2_5_RNA_ligase2'),(963470,'gnl|CDD|225536',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG2989','COG2989'),(963474,'gnl|CDD|237292',_binary 'polar chromosome segregation protein; Reviewed.','PRK13182','racA'),(963480,'gnl|CDD|197716',_binary 'helix_turn_helix, mercury resistance.','smart00422','HTH_MERR'),(963483,'gnl|CDD|225296',_binary 'Predicted site-specific integrase-resolvase [DNA replication, recombination, and repair].','COG2452','COG2452'),(963485,'gnl|CDD|133400',_binary 'Second Helix-Turn-Helix DNA binding domain of the regulatory protein TioE.  Putative helix-turn-helix (HTH) regulatory protein, TioE, and related proteins. TioE is part of the thiocoraline gene cluster, which is involved in the biosynthesis of the antitumor thiocoraline from the marine actinomycete, Micromonospora. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Proteins in this family are unique within the MerR superfamily in that they are composed of just two adjacent MerR-like N-terminal domains; this CD mainly contains the C-terminal or second repeat (rpt2) of these tandem MerR-like domain proteins.','cd04773','HTH_TioE_rpt2'),(963488,'gnl|CDD|133378',_binary 'Helix-Turn-Helix DNA binding domain of MerR-like transcription regulators.  Helix-turn-helix (HTH) MerR-like transcription regulator, N-terminal domain. The MerR family transcription regulators have been shown to mediate responses to stress including exposure to heavy metals, drugs, or oxygen radicals in eubacterial and some archaeal species. They regulate transcription of multidrug/metal ion transporter genes and oxidative stress regulons by reconfiguring the spacer between the -35 and -10 promoter elements.  A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd00592','HTH_MerR-like'),(963491,'gnl|CDD|133389',_binary 'Helix-Turn-Helix DNA binding domain of transcription regulators from the MerR superfamily.  Helix-turn-helix (HTH) transcription regulator MerR superfamily, N-terminal domain. The MerR family transcription regulators have been shown to mediate responses to stress including exposure to heavy metals, drugs, or oxygen radicals in eubacterial and some archaeal species. They regulate transcription of multidrug/metal ion transporter genes and oxidative stress regulons by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04761','HTH_MerR-SF'),(963521,'gnl|CDD|227668',_binary 'Phage-related protein, predicted endonuclease [DNA replication, recombination, and repair].','COG5377','COG5377'),(963524,'gnl|CDD|312929',_binary 'YqaJ-like viral recombinase domain.  This protein family is found in many different bacterial species but is of viral origin. The protein forms an oligomer and functions as a processive alkaline exonuclease that digests linear double-stranded DNA in a Mg(2+)-dependent reaction, It has a preference for 5\'-phosphorylated DNA ends. It thus forms part of the two-component SynExo viral recombinase functional unit.','pfam09588','YqaJ'),(963533,'gnl|CDD|316712',_binary 'Domain of unknown function (DUF4326).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea, eukaryotes and viruses. Proteins in this family are typically between 100 and 162 amino acids in length. There are two completely conserved residues (P and C) that may be functionally important.','pfam14216','DUF4326'),(963748,'gnl|CDD|306538',_binary 'Fibronectin type III domain.','pfam00041','fn3'),(964157,'gnl|CDD|224937',_binary 'Cytotoxic translational repressor of toxin-antitoxin stability system [Translation, ribosomal structure and biogenesis / Cell division and chromosome partitioning].','COG2026','RelE'),(964158,'gnl|CDD|309933',_binary 'ParE toxin of type II toxin-antitoxin system, parDE.  ParE is the toxin family of a type II toxin-antitoxin family. It is toxic towards DNA gyrase, but is neutralized by the antitoxin ParD. The family also encompasses RelE/ParE described in.','pfam05016','ParE_toxin'),(964159,'gnl|CDD|273688',_binary 'prevent-host-death family protein.  This model recognizes a region of about 55 amino acids toward the N-terminal end of bacterial proteins of about 85 amino acids in length. The best-characterized member is prevent-host-death (phd) of bacteriophage P1, the antidote partner of death-on-curing (doc) (TIGR01550) in an addiction module. Addiction modules prevent plasmid curing by killing the host cell as the longer-lived killing protein persists while the gene for the shorter-lived antidote is lost. Note, however, that relatively few members of this family appear to be plasmid or phage-encoded. Also, there is little overlap, except for phage P1 itself, of species with this family and with the doc family. [Cellular processes, Toxin production and resistance, Mobile and extrachromosomal element functions, Other].','TIGR01552','Hypothetical_protein_Rv33'),(964160,'gnl|CDD|308299',_binary 'Antitoxin Phd_YefM, type II toxin-antitoxin system.  Members of this family act as antitoxins in type II toxin-antitoxin systems. When bound to their toxin partners, they can bind DNA via the N-terminus and repress the expression of operons containing genes encoding the toxin and the antitoxin. This domain complexes with Txe toxins containing pfam06769, Fic/DOC toxins containing pfam02661 and YafO toxins containing pfam13957.','pfam02604','PhdYeFM_antitox'),(964223,'gnl|CDD|163703',_binary 'Mov34/MPN/PAD-1 family: archaeal JAB1/MPN/Mov34 metalloenzyme.  This family contains only archaeal MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+)-like domains. These domains contain the signature JAB1/MPN/Mov34 metalloenzyme (JAMM) motif, EXnHS/THX7SXXD, which is involved in zinc ion coordination and provides the active site for isopeptidase activity for the release of ubiquitin from ubiquitinated proteins (thus having deubiquitinating (DUB) activity) that are tagged for degradation.  The JAMM proteins likely hydrolyze ubiquitin conjugates in a manner similar to thermolysin, in which the zinc-polarized aqua ligand serves as the nucleophile, compared with the classical DUBs that do so with a cysteine residue in the active site.','cd08072','MPN_archaeal'),(964348,'gnl|CDD|237854',_binary 'DNA-directed RNA polymerase subunit A\'\'; Provisional.','PRK14898','PRK14898'),(964352,'gnl|CDD|315966',_binary 'Hint domain.  This domain is found in inteins.','pfam13403','Hint_2'),(964355,'gnl|CDD|198381',_binary 'Catalytic GIY-YIG domain of nucleotide excision repair endonucleases UvrC, Cho, and similar proteins.  UvrC is essential for nucleotide excision repair (NER). The N-terminal catalytic GIY-YIG domain of UvrC (also known as Uri domain) is responsible for the 3\' incision reaction and the C-terminal half of UvrC, consisting of an UvrB-binding domain (UvrBb), EndoV-like nuclease domain and a helix-hairpin-helix (HhH) DNA-binding domain, contains the residues involved in 5\' incision. The N- and C-terminal regions are joined by a common Cys-rich domain containing four conserved Cys residues. Besides UvrC, protein Cho (UvrC homolog) serves as a second endonuclease  in E. coli NER. Cho contains GIY-YIG motif followed by a Cys-rich region and shares sequence homology with the N-terminal half of UvrC. It is capable of incising the DNA at the 3\' side of a lesion in the presence of the UvrA and UvrB proteins during NER. The C-terminal half of Cho is a unique uncharacterized domain, which is distinct from that of UvrC. Moreover, unlike UvrC, Cho does not require the UvrC-binding domain of UvrB for the 3\' incision reaction, which might cause the shift in incision position and the difference in incision efficiencies between Cho and UvrC on different damaged substrates. Due to this, the range of NER in E. coli can be broadened by combining action of Cho and UvrC. This family also includes many uncharacterized epsilon proofreading subunits of DNA polymerase III, which have an additional N-terminal ExoIII domain and  a 3\'-5\' exonuclease domain homolog, fused to an UvrC-like region or a Cho-like region. The UvrC-like region includes a GIY-YIG motif, followed by a Cys-rich region, and an UvrB-binding domain (UvrBb), but lacks the EndoV-like nuclease domain and the helix-hairpin-helix (HhH) DNA-binding domain. The Cho-like region consists of a GIY-YIG motif, followed by the Cys-rich region, and the unique uncharacterized domain presenting in the C-terminal half of Cho. Some family members may not carry the Cys-rich region. This family also includes a specific Cho-like protein from G. violaceus, which possesses only UvrBb domain at the C-terminus, but lacks the additional N-terminal ExoIII domain. The oother two remote homologs of UvrC, Bacillus-I and -II, are included in this family as well. Both of them contain a GIY-YIG domain, but no Cys-rich region. Moreover, the whole C-terminal region of Bacillus-I is replaces by an unknown domain, and Bacillus-II possesses another unknown N-terminal extension.','cd10434','GIY-YIG_UvrC_Cho'),(964356,'gnl|CDD|214677',_binary 'GIY-YIG type nucleases (URI domain).','smart00465','GIYc'),(964358,'gnl|CDD|198394',_binary 'GIY-YIG domain of uncharacterized hypothetical protein found in bacteria and archaea.  The family includes a group of uncharacterized hypothetical proteins, mainly found in bacteria and a few found in archaea, with a GIY-YIG domain that shows statistically significant similarity to the N-terminal catalytic domains of GIY-YIG family of intron-encoded homing endonuclease I-TevI and catalytic GIY-YIG domain of nucleotide excision repair endonuclease UvrC.','cd10447','GIY-YIG_unchar_2'),(964407,'gnl|CDD|319270',_binary 'DGQHR motif containing domain.  Uncharacterized diverse domain family with conserved DGQHR motif, in addition to QR and FXXXN motifs. Some proteins have been identified as parts of DNA phosphorothioation systems. Related to dndB, which acts in the regulation of DNA modifications, including DNA phosphorothioation. These proteins show similarity to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, and other members of the ParB/Srx superfamily.','cd16413','DGQHR_domain'),(964806,'gnl|CDD|309899',_binary 'Nicotinamide mononucleotide transporter.  Members of this family are integral membrane proteins that are involved in transport of nicotinamide mononucleotide.','pfam04973','NMN_transporter'),(964888,'gnl|CDD|284915',_binary 'Pretoxin HINT domain.  A member of the HINT superfamily of proteases that is usually found N-terminal to the toxin module in polymorphic toxin systems. The domain is predicted to function in releasing the toxin domain by autoproteolysis.','pfam07591','PT-HINT'),(964891,'gnl|CDD|237799',_binary 'DNA polymerase II large subunit; Provisional.','PRK14715','PRK14715'),(965187,'gnl|CDD|315971',_binary 'Recombinase zinc beta ribbon domain.  This short bacterial protein contains a zinc ribbon domain that is likely to be DNA-binding. This domain is found in site specific recombinase proteins. This family appears most closely related to pfam04606.','pfam13408','Zn_ribbon_recom'),(965236,'gnl|CDD|235966',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07207','PRK07207'),(965237,'gnl|CDD|180610',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK06539','PRK06539'),(965256,'gnl|CDD|165193',_binary 'monoglyceride lipase; Provisional.','PHA02857','PHA02857'),(965258,'gnl|CDD|214008',_binary 'Uncharacterized enzymes similar to novel bacterial esterase that cleaves esters on halogenated cyclic compounds.  This family contains uncharacterized proteins similar to a novel bacterial esterase (Alcaligenes esterase 713) with the alpha/beta hydrolase fold but does not contain the GXSXXG pentapeptide around the active site serine residue as commonly seen in other enzymes of this class. Esterase 713 shows negligible sequence homology to other esterase and lipase enzymes. It is active as a dimer and cleaves esters on halogenated cyclic compounds though its natural substrate is unknown.','cd12809','Esterase_713_like-2'),(965259,'gnl|CDD|307221',_binary 'Thioesterase domain.  Peptide synthetases are involved in the non-ribosomal synthesis of peptide antibiotics. Next to the operons encoding these enzymes, in almost all cases, are genes that encode proteins that have similarity to the type II fatty acid thioesterases of vertebrates. There are also modules within the peptide synthetases that also share this similarity. With respect to antibiotic production, thioesterases are required for the addition of the last amino acid to the peptide antibiotic, thereby forming a cyclic antibiotic. Thioesterases (non-integrated) have molecular masses of 25-29 kDa.','pfam00975','Thioesterase'),(965263,'gnl|CDD|181553',_binary 'homoserine O-acetyltransferase; Provisional.','PRK08775','PRK08775'),(965265,'gnl|CDD|234771',_binary 'orotate phosphoribosyltransferase; Validated.','PRK00455','pyrE'),(965266,'gnl|CDD|235023',_binary 'orotate phosphoribosyltransferase-like protein; Provisional.','PRK02277','PRK02277'),(965267,'gnl|CDD|223537',_binary 'Orotate phosphoribosyltransferase [Nucleotide transport and metabolism].','COG0461','PyrE'),(965268,'gnl|CDD|223925',_binary 'Orotate phosphoribosyltransferase homologs [Nucleotide transport and metabolism].','COG0856','COG0856'),(965281,'gnl|CDD|274299',_binary 'helicase-exonuclease AddAB, AddA subunit, Firmicutes type.  AddAB, also called RexAB, substitutes for RecBCD in several bacterial lineages. These DNA recombination proteins act before synapse and are particularly important for DNA repair of double-stranded breaks by homologous recombination. The term AddAB is used broadly, with AddA homologous between the Firmicutes (as modeled here) and the alphaproteobacteria, while the partner AddB proteins show no strong homology across the two groups of species. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02785','ATP-dependent_helicase/nu'),(965896,'gnl|CDD|184762',_binary 'Holliday junction DNA helicase RuvA; Provisional.','PRK14606','ruvA'),(966036,'gnl|CDD|307213',_binary 'LAGLIDADG endonuclease.','pfam00961','LAGLIDADG_1'),(966052,'gnl|CDD|293075',_binary 'Domain of unknown function (DUF5047).  This family consists of N-terminal of several uncharacterized proteins and peptidases around 360 residues in length and is mainly found in various Streptomyces species. The function of this family is unknown.','pfam16466','DUF5047'),(966058,'gnl|CDD|222885',_binary 'deoxynucleoside monophosphate kinase; Provisional.','PHA02575','1'),(966060,'gnl|CDD|274153',_binary 'RNA polymerase sigma factor, FliA/WhiG family.  Most members of this family are the flagellar operon sigma factor FliA, controlling transcription of bacterial flagellar genes by RNA polymerase. An exception is the sigma factor WhiG in the genus Streptomyces, involved in the production of sporulating aerial mycelium.','TIGR02479','RNA_polymerase_sigma_fact'),(966085,'gnl|CDD|316137',_binary 'Histidine kinase-like ATPase domain.','pfam13581','HATPase_c_2'),(966086,'gnl|CDD|308236',_binary 'Histidine kinase-, DNA gyrase B-, and HSP90-like ATPase.  This family represents the structurally related ATPase domains of histidine kinase, DNA gyrase B and HSP90.','pfam02518','HATPase_c'),(966094,'gnl|CDD|239739',_binary 'Serine Recombinase (SR) family, TndX-like transposase subfamily, catalytic domain; composed of large serine recombinases similar to Clostridium TndX and TnpX transposases. Serine recombinases catalyze site-specific recombination of DNA molecules by a concerted, four-strand cleavage and rejoining mechanism which involves a transient phosphoserine linkage between DNA and the enzyme. They are functionally versatile and include resolvases, invertases, integrases, and transposases. TndX mediates the excision and circularization of the conjugative transposon Tn5397 from Clostridium difficile. TnpX is responsible for the movement of the nonconjugative chloramphenicol resistance elements of the Tn4451/3 family. Mobile genetic elements such as transposons are important vehicles for the transmission of virulence and antibiotic resistance in many microorganisms.','cd03770','SR_TndX_transposase'),(966124,'gnl|CDD|162605',_binary 'O-succinylbenzoate-CoA ligase.  This model represents an enzyme, O-succinylbenzoate-CoA ligase, which is involved in the fourth step of the menaquinone biosynthesis pathway. O-succinylbenzoate-CoA ligase, together with menB - naphtoate synthase, take 2-succinylbenzoate and convert it into 1,4-di-hydroxy-2- naphtoate. [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR01923','2-succinylbenzoate--CoA_l'),(966187,'gnl|CDD|226653',_binary 'Predicted transcriptional regulator [Transcription].','COG4190','COG4190'),(966295,'gnl|CDD|188121',_binary 'proline-specific peptidase, Bacillus coagulans-type subfamily.  This model describes a subfamily of the alpha/beta fold family of hydrolases. Characterized members include prolinases (Pro-Xaa dipeptidase, EC 3.4.13.8), prolyl aminopeptidases (EC 3.4.11.5), and a leucyl aminopeptidase.','TIGR01250','Proline_iminopeptidase'),(966547,'gnl|CDD|224431',_binary '2\'-5\' RNA ligase [Translation, ribosomal structure and biogenesis].','COG1514','LigT'),(966552,'gnl|CDD|273684',_binary 'phage terminase, large subunit, PBSX family.  This model detects members of a highly divergent family of the large subunit of phage terminase. All members are encoded by phage genomes or within prophage regions of bacterial genomes. This is a distinct family from pfam03354. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01547','Uncharacterized_protein_Y'),(966573,'gnl|CDD|235450',_binary 'glmZ(sRNA)-inactivating NTPase; Provisional.','PRK05416','PRK05416'),(966575,'gnl|CDD|224574',_binary 'Predicted P-loop-containing kinase [General function prediction only].','COG1660','COG1660'),(966577,'gnl|CDD|112483',_binary 'P-loop ATPase protein family.  This family contains an ATP-binding site and could be an ATPase (personal obs:C Yeats).','pfam03668','ATP_bind_2'),(966594,'gnl|CDD|273168',_binary 'DNA repair helicase rad25.  All proteins in this family for which functions are known are DNA-DNA helicases used for the initiation of nucleotide excision repair and transacription as part of the TFIIH complex.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00603','repair_helicase_RAD25'),(966597,'gnl|CDD|215103',_binary 'DEAD-box ATP-dependent RNA helicase; Provisional.','PLN00206','PLN00206'),(966601,'gnl|CDD|315975',_binary 'Winged helix-turn-helix DNA-binding.','pfam13412','HTH_24'),(966602,'gnl|CDD|309331',_binary 'EAP30/Vps36 family.  This family includes EAP30 as well as the Vps36 protein. Vps36 is involved in Golgi to endosome trafficking. EAP30 is a subunit of the ELL complex. The ELL is an 80-kDa RNA polymerase II transcription factor. ELL interacts with three other proteins to form the complex known as ELL complex. The ELL complex is capable of increasing that catalytic rate of transcription elongation, but is unable to repress initiation of transcription by RNA polymerase II as is the case of ELL. EAP30 is thought to lead to the derepression of ELL\'s transcriptional inhibitory activity.','pfam04157','EAP30'),(966665,'gnl|CDD|225031',_binary 'Uncharacterized proteins, LmbE homologs [Function unknown].','COG2120','COG2120'),(966666,'gnl|CDD|308281',_binary 'GlcNAc-PI de-N-acetylase.  Members of this family are related to PIG-L an N-acetylglucosaminylphosphatidylinositol de-N-acetylase (EC:3.5.1.89) that catalyzes the second step in GPI biosynthesis.','pfam02585','PIG-L'),(966667,'gnl|CDD|274915',_binary 'bacillithiol biosynthesis deacetylase BshB1.  Members of this protein family are BshB1 (YpjG), an enzyme of bacillithiol biosynthesis; either BshB1 or BshB2 (YojG) must be present, and often both are present. Bacillithiol is a low-molecular-weight thiol, an analog of glutathione and mycothiol, and is found largely in the Firmicutes. [Biosynthesis of cofactors, prosthetic groups, and carriers, Glutathione and analogs].','TIGR04001','thiol_BshB1'),(966675,'gnl|CDD|239109',_binary 'A sub-family of peptidase family C39. Peptidase family C39 mostly contains bacteriocin-processing endopeptidases from bacteria. The cysteine peptidases in family C39 cleave the \'double-glycine\' leader peptides from the precursors of various bacteriocins (mostly non-lantibiotic). The cleavage is mediated by the transporter as part of the secretion process. Bacteriocins are antibiotic proteins secreted by some species of bacteria that inhibit the growth of other bacterial species. The bacteriocin is synthesized as a precursor with an N-terminal leader peptide, and processing involves removal of the leader peptide by cleavage at a Gly-Gly bond, followed by translocation of the mature bacteriocin across the cytoplasmic membrane. Most endopeptidases of family C39 are N-terminal domains in larger proteins (ABC transporters) that serve both functions. The proposed protease active site is conserved in this sub-family of proteins with a single peptidase domain, which are lacking the nucleotide-binding transporter signature or have different domain architectures.','cd02549','Peptidase_C39A'),(966692,'gnl|CDD|223683',_binary 'Type I site-specific restriction-modification system, R (restriction) subunit and related helicases [Defense mechanisms].','COG0610','COG0610'),(966715,'gnl|CDD|235574',_binary 'DNA polymerase III subunit epsilon; Provisional.','PRK05711','PRK05711'),(966832,'gnl|CDD|226375',_binary 'ATP-dependent nuclease, subunit B [DNA replication, recombination, and repair].','COG3857','AddB'),(966837,'gnl|CDD|132031',_binary 'type II restriction endonuclease, Alw26I/Eco31I/Esp3I family.  Members of this family are type II restriction endonucleases of the Alw26I/Eco31I/Esp3I family. Characterized specificities of three members are GGTCTC, CGTCTC, and the shared subsequence GTCTC. [DNA metabolism, Restriction/modification].','TIGR02986','restriction_endonuclease'),(966839,'gnl|CDD|187774',_binary 'CRISPR/Cas system-associated protein Cas9.  CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) and associated Cas proteins comprise a system for heritable host defense by prokaryotic cells against phage and other foreign DNA; Very large protein containing McrA/HNH-nuclease related domain and a RuvC-like nuclease domain; signature gene for type II.','cd09643','Csn1'),(966840,'gnl|CDD|273840',_binary 'CRISPR subtype II/NMENI RNA-guided endonuclease Cas9/Csn1.  CRISPR loci appear to be mobile elements with a wide host range. This model represents a protein found only in CRISPR-containing species, near other CRISPR-associated proteins (cas), as part of the NMENI subtype of CRISPR/Cas locus. The species range so far for this protein is animal pathogens and commensals only.','TIGR01865','conserved_hypothetical_pr'),(966860,'gnl|CDD|239736',_binary 'Serine recombinase (SR) family, Partitioning (par)-Resolvase subfamily, catalytic domain; Serine recombinases catalyze site-specific recombination of DNA molecules by a concerted, four-strand cleavage and rejoining mechanism which involves a transient phosphoserine linkage between DNA and the enzyme. They are functionally versatile and include resolvases, invertases, integrases, and transposases. This subgroup is composed of proteins similar to the E. coli resolvase found in the par region of the RP4 plasmid, which encodes a highly efficient partitioning system. This protein is part of a complex stabilization system involved in the resolution of plasmid dimers during cell division. Similar to Tn3 and other resolvases, members of this family may contain a C-terminal DNA binding domain.','cd03767','SR_Res_par'),(966920,'gnl|CDD|315607',_binary 'Domain of unknown function (DUF3846).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences. This domain is found associated with an pfam07275 like domain. This suggests that this family may also be involved in evading host restriction.','pfam12957','DUF3846'),(966921,'gnl|CDD|107157',_binary 'ATPase-like protein.','PHA02244','PHA02244'),(966922,'gnl|CDD|223786',_binary 'MoxR-like ATPases [General function prediction only].','COG0714','COG0714'),(966923,'gnl|CDD|311595',_binary 'AAA domain (dynein-related subfamily).  This Pfam entry includes some of the AAA proteins not detected by the pfam00004 model.','pfam07728','AAA_5'),(966924,'gnl|CDD|227596',_binary 'AAA ATPase containing von Willebrand factor type A (vWA) domain [General function prediction only].','COG5271','MDN1'),(966929,'gnl|CDD|131688',_binary 'gas vesicle protein GvpN.  Members of this family are the GvpN protein associated with the production of gas vesicles produced in some prokaryotes to give cells buoyancy. This family belongs to a larger family of ATPases (pfam07728). [Cellular processes, Other].','TIGR02640','Gas_vesicle_protein_GvpN'),(966930,'gnl|CDD|223542',_binary 'ATP-dependent Lon protease, bacterial type [Posttranslational modification, protein turnover, chaperones].','COG0466','Lon'),(966931,'gnl|CDD|308144',_binary 'Bacterial Ig-like domain (group 2).  This family consists of bacterial domains with an Ig-like fold. Members of this family are found in bacterial and phage surface proteins such as intimins.','pfam02368','Big_2'),(966932,'gnl|CDD|227779',_binary 'Bacterial surface proteins containing Ig-like domains [Cell motility and secretion].','COG5492','COG5492'),(966933,'gnl|CDD|214752',_binary 'Bacterial Ig-like domain 2.','smart00635','BID_2'),(966934,'gnl|CDD|307919',_binary 'Carbohydrate binding domain.  This family includes diverse carbohydrate binding domains.','pfam02018','CBM_4_9'),(966939,'gnl|CDD|236892',_binary 'hypothetical protein; Provisional.','PRK11281','PRK11281'),(966940,'gnl|CDD|224428',_binary 'Predicted membrane protein [Function unknown].','COG1511','COG1511'),(966956,'gnl|CDD|226071',_binary 'Predicted nucleotidyltransferase [General function prediction only].','COG3541','COG3541'),(966962,'gnl|CDD|285479',_binary 'HTH domain.  This family includes helix-turn-helix domains in a wide variety of proteins.','pfam08279','HTH_11'),(966982,'gnl|CDD|181869',_binary 'phosphodiesterase; Provisional.','PRK09453','PRK09453'),(966984,'gnl|CDD|314381',_binary 'Domain of unknown function (DUF2828).  This is a uncharacterized domain found in eukaryotes and viruses.','pfam11443','DUF2828'),(967026,'gnl|CDD|238544',_binary 'KaiC is a circadian clock protein primarily found in cyanobacteria KaiC is a RecA-like ATPase, having both Walker A and Walker B motifs. A related protein is found in archaea.','cd01124','KaiC'),(967033,'gnl|CDD|163589',_binary 'KaiC domain protein, Ph0284 family.  Members of this family contain a single copy of the KaiC domain (pfam06745) that occurs in two copies of the circadian clock protein kinase KaiC itself. Members occur primarily in thermophilic archaea and in Thermotoga.','TIGR03877','thermo_KaiC_1'),(967042,'gnl|CDD|273666',_binary 'putative nicotinate phosphoribosyltransferase.  A deep split separates two related families of proteins, one of which includes experimentally characterized examples of nicotinate phosphoribosyltransferase, an the first enzyme of NAD salvage biosynthesis. This model represents the other family. Members have a different (longer) spacing of several key motifs and have an additional C-terminal domain of up to 100 residues. One argument suggesting that this family represents the same enzyme is that no species has a member of both families. Another is that the gene encoding this protein is located near other NAD salvage biosynthesis genes in Nostoc and in at least four different Gram-positive bacteria. NAD and NADP are ubiquitous in life. Most members of this family are Gram-positive bacteria. An additional set of mutually closely related archaeal sequences score between the trusted and noise cutoffs. [Biosynthesis of cofactors, prosthetic groups, and carriers, Pyridine nucleotides].','TIGR01513','Nicotinate_phosphoribosyl'),(967052,'gnl|CDD|274963',_binary 'polynucleotide kinase-phosphatase.  Members of this protein family are the bacterial polynucleotide kinase-phosphatase (Pnkp) whose genes occur paired with genes for the 3\' terminal RNA ribose 2\'-O-methyltransferase Hen1. All members of the seed alignment belong to a cassette with the Hen1. The pair acts in bacterial RNA repair. This enzyme performs end-healing reactions on broken RNA, preparing from the RNA ligase to close the break. The working hypothesis is that the combination of Pnkp (RNA repair) and Hen1 (RNA modification) serves to first repair RNA damage from ribotoxins and then perform a modification that prevents the damage from recurring. [Transcription, RNA processing].','TIGR04075','Ser/Thr_phosphatase_famil'),(967078,'gnl|CDD|314507',_binary 'P22 coat protein - gene protein 5.  This family of proteins represents gene product 5 from bacteriophage P22. This protein is involved in the formation of the pro-capsid shells in the bacteriophage. In total, there are 415 molecules of the coat protein which are arranged in an icosahedral shell.','pfam11651','P22_CoatProtein'),(967083,'gnl|CDD|176646',_binary 'DEDDy 3\'-5\' exonuclease domain of family-B DNA polymerases.  The 3\'-5\' exonuclease domain of family-B DNA polymerases. This domain has a fundamental role in reducing polymerase errors and is involved in proofreading activity. Family-B DNA polymerases contain an N-terminal DEDDy DnaQ-like exonuclease domain in the same polypeptide chain as the polymerase domain, similar to family-A DNA polymerases. This domain contains three sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The exonuclease domain of family B polymerase also contains a beta hairpin structure that plays an important role in active site switching in the event of nucleotide misincorporation. Members include Escherichia coli DNA polymerase II, some eubacterial phage DNA polymerases, nuclear replicative DNA polymerases (alpha, delta, epsilon and zeta), and eukaryotic viral and plasmid-borne enzymes. Nuclear DNA polymerases alpha and zeta lack the four conserved acidic metal-binding residues. Family-B DNA polymerases are predominantly involved in DNA replication and DNA repair.','cd05160','DEDDy_DNA_polB_exo'),(967084,'gnl|CDD|235594',_binary 'DNA polymerase I; Reviewed.','PRK05761','PRK05761'),(967129,'gnl|CDD|182075',_binary 'endodeoxyribonuclease RUS; Reviewed.','PRK09786','PRK09786'),(967140,'gnl|CDD|307338',_binary 'tRNA-splicing ligase RtcB.  This family of RNA ligases (EC:6.5.1.3) join 2\',3\'-cyclic phosphate and 5\'-OH ends. They catalyze the splicing of tRNA and may also participate in tRNA repair and recovery from stress-induced RNA damage.','pfam01139','RtcB'),(967141,'gnl|CDD|224604',_binary 'Uncharacterized conserved protein [Function unknown].','COG1690','RtcB'),(967142,'gnl|CDD|181972',_binary 'hypothetical protein; Reviewed.','PRK09588','PRK09588'),(967143,'gnl|CDD|274417',_binary 'release factor H-coupled RctB family protein.  Members of this family are related to RctB. RctB a protein of known structure but unknown function that often is encoded near RNA cyclase and therefore is suggested to be a tRNA or mRNA processing enzyme. This family of RctB-like proteins in encoded upstream of, and apparently is translationally coupled to, the putative peptide chain release factor RF-H (TIGR03072), product of the prfH gene. Note that a large deletion at the junction between this gene and the prfH gene in Escherichia coli K-12 marks both as probable pseudogenes. [Protein synthesis, Other].','TIGR03073','release_rtcB'),(967220,'gnl|CDD|310109',_binary 'CHAP domain.  This domain corresponds to an amidase function. Many of these proteins are involved in cell wall metabolism of bacteria. This domain is found at the N-terminus of Escherichia coli gss, where it functions as a glutathionylspermidine amidase EC:3.5.1.78. This domain is found to be the catalytic domain of PlyCA. CHAP is the amidase domain of bifunctional Escherichia coli glutathionylspermidine synthetase/amidase, and it catalyzes the hydrolysis of Gsp (glutathionylspermidine) into glutathione and spermidine.','pfam05257','CHAP'),(967250,'gnl|CDD|235245',_binary 'glutamyl-tRNA(Gln) amidotransferase subunit D; Validated.','PRK04183','PRK04183'),(967296,'gnl|CDD|236391',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK09145','PRK09145'),(967327,'gnl|CDD|309113',_binary 'Phage major capsid protein E.  Major capsid protein E is involved with the stabilisation of the condensed form of the DNA molecule in phage heads.','pfam03864','Phage_cap_E'),(967359,'gnl|CDD|99838',_binary 'DEDDh 3\'-5\' exonuclease domain of oligoribonuclease and similar proteins.  Oligoribonuclease (Orn) is a DEDDh-type DnaQ-like 3\'-5\' exoribonuclease that is responsible for degrading small oligoribonucleotides to mononucleotides. It contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. Orn is essential for Escherichia coli survival. The human homolog, also called Sfn (small fragment nuclease), is able to hydrolyze short single-stranded RNA and DNA oligomers. It plays a role in cellular nucleotide recycling.','cd06135','Orn'),(967360,'gnl|CDD|235429',_binary 'oligoribonuclease; Provisional.','PRK05359','PRK05359'),(967361,'gnl|CDD|224860',_binary 'Oligoribonuclease (3\'->5\' exoribonuclease) [RNA processing and modification].','COG1949','Orn'),(967366,'gnl|CDD|236397',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK09182','PRK09182'),(967384,'gnl|CDD|238731',_binary 'norD type: Denitrifying bacteria contain both membrane bound and periplasmic nitrate reductases. Denitrification plays a major role  in completing the nitrogen cycle by converting nitrate or nitrite to nitrogen gas. The pathway for microbial denitrification has been established as NO3-  ------> NO2- ------> NO -------> N2O ---------> N2. This reaction generally occurs under oxygen limiting conditions. Genetic and biochemical studies have shown that the first srep of the biochemical pathway is catalyzed by periplasmic nitrate reductases. This family is widely present in proteobacteria and firmicutes. This version of the domain is also present in some archaeal members. The function of the vWA domain in this sub-group is not known. Members of this subgroup have a conserved MIDAS motif.','cd01454','vWA_norD_type'),(967385,'gnl|CDD|288608',_binary 'Cobalamin biosynthesis protein CobT VWA domain.  This family consists of several bacterial cobalamin biosynthesis (CobT) proteins. CobT is involved in the transformation of precorrin-3 into cobyrinic acid.','pfam11775','CobT_C'),(967386,'gnl|CDD|226920',_binary 'Cobalamin biosynthesis protein CobT (nicotinate-mononucleotide:5, 6-dimethylbenzimidazole phosphoribosyltransferase) [Coenzyme metabolism].','COG4547','CobT'),(967388,'gnl|CDD|130712',_binary 'cobaltochelatase, CobT subunit.  This model describes Pseudomonas denitrificans CobT gene product, which is a cobalt chelatase subunit that functions in cobalamin biosynthesis. Cobalamin (vitamin B12) can be synthesized via several pathways, including an aerobic pathway (found in Pseudomonas denitrificans) and an anaerobic pathway (found in P. shermanii and Salmonella typhimurium). These pathways differ in the point of cobalt insertion during corrin ring formation. There are apparently a number of variations on these two pathways, where the major differences seem to be concerned with the process of ring contraction. Confusion regarding the functions of enzymes found in the aerobic vs. anaerobic pathways has arisen because nonhomologous genes in these different pathways were given the same gene symbols. Thus, cobT in the aerobic pathway (P. denitrificans) is not a homolog of cobT in the anaerobic pathway (S. typhimurium). It should be noted that E. coli synthesizes cobalamin only when it is supplied with the precursor cobinamide, which is a complex intermediate. Additionally, all E. coli cobalamin synthesis genes (cobU, cobS and cobT) were named after their Salmonella typhimurium homologs which function in the anaerobic cobalamin synthesis pathway. This model describes the aerobic cobalamin pathway Pseudomonas denitrificans CobT gene product, which is a cobalt chelatase subunit, with a MW ~70 kDa. The aerobic pathway cobalt chelatase is a heterotrimeric, ATP-dependent enzyme that catalyzes cobalt insertion during cobalamin biosynthesis. The other two subunits are the P. denitrificans CobS (TIGR01650) and CobN (pfam02514 CobN/Magnesium Chelatase) proteins. To avoid potential confusion with the nonhomologous Salmonella typhimurium/E.coli cobT gene product, the P. denitrificans gene symbol is not used in the name of this model. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR01651','CobT'),(967753,'gnl|CDD|270908',_binary 'Catalytic kinase domain of Myosin Light Chain Kinase-like Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This family is composed of MLCKs and related MLCK-like kinase domains from giant STKs such as titin, obscurin, SPEG, Unc-89, Trio, kalirin, and Twitchin. Also included in this family are Death-Associated Protein Kinases (DAPKs) and Death-associated protein kinase-Related Apoptosis-inducing protein Kinase (DRAKs). MLCK phosphorylates myosin regulatory light chain and controls the contraction of all muscle types. Titin, obscurin, Twitchin, and SPEG are muscle proteins involved in the contractile apparatus. The giant STKs are multidomain proteins containing immunoglobulin (Ig), fibronectin type III (FN3), SH3, RhoGEF, PH and kinase domains. Titin, obscurin, Twitchin, and SPEG contain many Ig domain repeats at the N-terminus, while Trio and Kalirin contain spectrin-like repeats. The MLCK-like family is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14006','STKc_MLCK-like'),(967756,'gnl|CDD|270971',_binary 'Catalytic domain of the Serine/Threonine kinase, Checkpoint kinase 1.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Chk1 is implicated in many major checkpoints of the cell cycle, providing a link between upstream sensors and the cell cycle engine. It plays an important role in DNA damage response and maintaining genomic stability. Chk1 acts as an effector of the sensor kinase, ATR (ATM and Rad3-related), a member of the PI3K family, which is activated upon DNA replication stress. Chk1 delays mitotic entry in response to replication blocks by inhibiting cyclin dependent kinase (Cdk) activity. In addition, Chk1 contributes to the function of centrosome and spindle-based checkpoints, inhibits firing of origins of DNA replication (Ori), and represses transcription of cell cycle proteins including cyclin B and Cdk1. The Chk1 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14069','STKc_Chk1'),(967757,'gnl|CDD|270907',_binary 'Catalytic domain of the Serine/Threonine kinase, Proviral Integration Moloney virus (PIM) kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The PIM gene locus was discovered as a result of the cloning of retroviral intergration sites in murine Moloney leukemia virus, leading to the identification of PIM kinases. They are constitutively active STKs with a broad range of cellular targets and are overexpressed in many haematopoietic malignancies and solid cancers. Vertebrates contain three distinct PIM kinase genes (PIM1-3); each gene may result in mutliple protein isoforms. There are two PIM1 and three PIM2 isoforms as a result of alternative translation initiation sites, while there is only one PIM3 protein. Compound knockout mice deficient of all three PIM kinases that survive the perinatal period show a profound reduction in body size, indicating that PIMs are important for body growth. The PIM subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd14005','STKc_PIM'),(968180,'gnl|CDD|314926',_binary 'Protein of unknown function (DUF3585).  This domain is found in eukaryotes. This domain is typically between 135 and 149 amino acids in length and is found associated with pfam00307.','pfam12130','DUF3585'),(968191,'gnl|CDD|313442',_binary 'Axonemal dynein light chain.  Axonemal dynein light chain proteins play a dynamic role in flagellar and cilia motility. Eukaryotic cilia and flagella are complex organelles consisting of a core structure, the axoneme, which is composed of nine microtubule doublets forming a cylinder that surrounds a pair of central singlet microtubules. This ultra-structural arrangement seems to be one of the most stable micro-tubular assemblies known and is responsible for the flagellar and ciliary movement of a large number of organisms ranging from protozoan to mammals. This light chain interacts directly with the N-terminal half of the heavy chains.','pfam10211','Ax_dynein_light'),(968192,'gnl|CDD|314095',_binary 'YlqD protein.  The structure of a representative of this family has been solved (Structure 4dci) and found to form a tetrameric structure of prefoldin-like architecture with the beta-barrel core and helical coiled coil tentacles. This suggests that this family may act as molecular chaperones.','pfam11068','YlqD'),(968317,'gnl|CDD|310150',_binary 'Putative Ig domain.  This alignment represents the conserved core region of ~90 residue repeat found in several haemagglutinins and other cell surface proteins. Sequence similarities to (pfam02494) and (pfam00801) suggest an Ig-like fold (personal obs:C. Yeats). So this family may be similar in function to the (pfam02639) and (pfam02638) domains. This domain is also found in the WisP family of proteins of Tropheryma whipplei.','pfam05345','He_PIG'),(968323,'gnl|CDD|274256',_binary 'TIGR02680 family protein.  Members of this protein family belong to a conserved gene four-gene neighborhood found sporadically in a phylogenetically broad range of bacteria: Nocardia farcinica, Symbiobacterium thermophilum, and Streptomyces avermitilis (Actinobacteria), Geobacillus kaustophilus (Firmicutes), Azoarcus sp. EbN1 and Ralstonia solanacearum (Betaproteobacteria). Proteins in this family average over 1400 amino acids in length. [Hypothetical proteins, Conserved].','TIGR02680','conserved_hypothetical_pr'),(968327,'gnl|CDD|273539',_binary 'nitrogenase cofactor biosynthesis protein NifB.  This model describes NifB, a protein required for the biosynthesis of the iron-molybdenum (or iron-vanadium) cofactor used by the nitrogen-fixing enzyme nitrogenase. NifB belongs to the radical SAM family, and the FeMo cluster biosynthesis process requires S-adenosylmethionine. Archaeal homologs lack the most C-terminal region and score between the trusted and noise cutoffs of this model. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other, Central intermediary metabolism, Nitrogen fixation].','TIGR01290','FeMo_cofactor_biosynthesi'),(968328,'gnl|CDD|214959',_binary 'This model contains proteins with the VRR-NUC domain.  It is associated with members of the PD-(D/E)XK nuclease superfamily, which include the type III restriction modification enzymes, for example StyLTI.','smart00990','VRR_NUC'),(968329,'gnl|CDD|312345',_binary 'VRR-NUC domain.','pfam08774','VRR_NUC'),(968330,'gnl|CDD|307811',_binary 'Archaeal holliday junction resolvase (hjc).  This family of archaebacterial proteins are holliday junction resolvases (hjc gene). The Holliday junction is an essential intermediate of homologous recombination. This protein is the archaeal equivalent of RuvC but is not sequence similar.','pfam01870','Hjc'),(968348,'gnl|CDD|206605',_binary 'Bd3614-like deaminase.  A member of the nucleic acid/nucleotide deaminase superfamily prototyped by Bdellovibrio Bd3614. They are typified by a distinct N-terminal globular domain. The Bdellovibrio version occurs in a predicted operon with a 23S rRNA G2445-modifying methylase suggesting that it might be involved in RNA editing.','pfam14439','Bd3614-deam'),(968350,'gnl|CDD|238211',_binary 'Thymidylate synthase and pyrimidine hydroxymethylase: Thymidylate synthase (TS) and deoxycytidylate hydroxymethylase (dCMP-HMase) are homologs that catalyze analogous alkylation of C5 of pyrimidine nucleotides. Both enzymes are involved in the biosynthesis of DNA precursors and are active as homodimers. However, they exhibit distinct pyrimidine base specificities and differ in the details of their catalyzed reactions. TS is biologically ubiquitous and catalyzes the conversion of dUMP and methylene-tetrahydrofolate (CH2THF) to dTMP and dihydrofolate (DHF). It also acts as a regulator of its own expression by binding and inactivating its own RNA. Due to its key role in the de novo pathway for thymidylate synthesis and, hence, DNA synthesis, it is one of the most conserved enzymes across species and phyla. TS is a well-recognized target for anticancer chemotherapy, as well as a valuable new target against infectious diseases. Interestingly, in several protozoa, a single polypeptide chain codes for both, dihydrofolate reductase (DHFR) and thymidylate synthase (TS), forming a bifunctional enzyme (DHFR-TS), possibly through gene fusion at a single evolutionary point. DHFR-TS is also active as a dimer. Virus encoded dCMP-HMase catalyzes the reversible conversion of dCMP and CH2THF to hydroxymethyl-dCMP and THF. This family also includes dUMP hydroxymethylase, which is encoded by several bacteriophages that infect Bacillus subtilis, for their own protection against the host restriction system,  and contain hydroxymethyl-dUMP instead of dTMP in their DNA.','cd00351','TS_Pyrimidine_HMase'),(968351,'gnl|CDD|223285',_binary 'Thymidylate synthase [Nucleotide transport and metabolism].','COG0207','ThyA'),(968368,'gnl|CDD|198428',_binary 'Family 4 Uracil-DNA glycosylase (UDG), found exclusively in thermophilic organisms.  The enzymes of Family 4 Uracil-DNA glycosylase (UDG), found only in thermophilic organisms, are thermostable enzymes. Uracil-DNA glycosylases (UDGs) are DNA repair enzymes that catalyze the removal of mismatched uracil from DNA to initiate DNA base excision repair pathway. The Thermus thermophilus enzyme TTUDGA removes uracil from both, ssDNA and dsDNA, but not thymine from a G:T mismatch. These details suggest that the mechanism by which Family 4 UDGs remove uracils from DNA is similar to that of Family 1 enzymes. The thermostability of the enzyme may be linked to the presence of an iron-sulfur cluster, salt-bridges and ion pairs on the molecular surface as well as prolines on loops and turns, as commonly found in the Family 4 enzymes. Uracil in DNA can arise as a result of mis-incorporation of dUMP residues by DNA polymerase or deamination of cytosine. Uracil mispaired with guanine in DNA is one of the major pro-mutagenic events, causing G:C->A:T mutations.','cd10030','UDG_F4_TTUDGA_like'),(968370,'gnl|CDD|177427',_binary 'dCTP pyrophosphatase; Provisional.','PHA02602','56'),(968371,'gnl|CDD|212134',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in dimeric 2-Deoxyuridine 5\'-triphosphate nucleotidohydrolase and similar proteins.  dUTPase (dUTP pyrophosphatase; EC 3.6.1.23) catalyzes the hydrolysis of dUTP to dUMP and pyrophosphate. It acts to ensure chromosomal integrity by reducing the effective ratio of dUTP/dTTP. Members in this family are dimeric dUTPases, such as those from Leishmania major, Trypanosoma cruzi, and Campylobacter jejuni, which differ from the monomeric and trimeric forms and adopt an all-alpha topology. A central four-helix bundle, consisting of two alpha-helices from the rigid domain and two helices from the mobile domain and connecting loops, form the active site in dimeric dUTPase-like proteins, requiring the presence of metal ion cofactors to hydrolyze both dUTP and dUDP.','cd11527','NTP-PPase_dUTPase'),(968372,'gnl|CDD|234719',_binary 'guanylate kinase; Provisional.','PRK00300','gmk'),(968373,'gnl|CDD|213788',_binary 'guanylate kinase.  Members of this family are the enzyme guanylate kinase, also called GMP kinase. This enzyme transfers a phosphate from ATP to GMP, yielding ADP and GDP. [Purines, pyrimidines, nucleosides, and nucleotides, Nucleotide and nucleoside interconversions].','TIGR03263','Guanylate_kinase'),(968384,'gnl|CDD|318902',_binary 'RepB DNA-primase from phage plasmid.  RepB_primase is a DNA-primase produced by P4-like phages. It is a zinc-independent primase unlike Pri-type primases. It takes up a dumbbell shaped consisting of an N-terminal catalytic domain separated by a long alpha-helix plus tether and a C-terminal helical-bundle domain. Primases are necessary for phage replication. RepBprime primases such as in this family recognize both ssiA and ssiB, ie only 1 single-stranded primase initiation site on each strand, independently of each other and then synthesize primers that are elongated by DNA polymerase III. The phage is thus replicated exclusively in leading strand mode.','pfam16793','RepB_primase'),(968400,'gnl|CDD|184041',_binary 'multiple promoter invertase; Provisional.','PRK13413','mpi'),(968458,'gnl|CDD|315927',_binary 'Toprim domain.  The toprim domain is found in a wide variety of enzymes involved in nucleic acid manipulation.','pfam13362','Toprim_3'),(968468,'gnl|CDD|236723',_binary 'murein L,D-transpeptidase; Provisional.','PRK10594','PRK10594'),(968474,'gnl|CDD|311301',_binary 'Antirestriction protein (ArdA).  This family consists of several bacterial antirestriction (ArdA) proteins. ArdA functions in bacterial conjugation to allow an unmodified plasmid to evade restriction in the recipient bacterium and yet acquire cognate modification.','pfam07275','ArdA'),(968475,'gnl|CDD|227077',_binary 'Antirestriction protein [General function prediction only].','COG4734','ArdA'),(968480,'gnl|CDD|273775',_binary 'phage protein, HK97 gp10 family.  This model represents an uncharacterized, highly divergent bacteriophage family. The family includes gp10 from HK022 and HK97. It appears related to TIGR01635, a phage morphogenesis family believed to be involved in tail completion. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01725','hypothetical_protein'),(968487,'gnl|CDD|285830',_binary 'BetR domain.  This family includes an N-terminal helix-turn-helix domain.','pfam08667','BetR'),(968488,'gnl|CDD|283090',_binary 'Protein of unknown function (DUF739).  This family contains several bacteriophage proteins. Some of the proteins in this family have been labeled putative cro repressor proteins.','pfam05339','DUF739'),(968522,'gnl|CDD|271190',_binary 'Putative XerD in Streptococcus pneumonia and similar proteins, C-terminal catalytic domain.  This family includes a putative XerD recombinase in Streptococcus pneumonia and similar tyrosine recombinases. However, the members of this family contain unusual active site motifs from the XerD from Escherichia coli. E. coli XerD and homologous enzymes show four conserved amino acids R-H-R-H that are spaced along the C-terminal domain. The putative S. pneumoniae XerD contains three unique replacements at the conserved positions resulting in L-Q-R-L. Severe growth defects in a loss-of-function xerD mutant demonstrate an important in vivo function of the S. pneumoniae XerD protein. This family belongs to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The catalytic domain contains six conserved active site residues. Their overall reaction mechanism involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA.','cd01190','INT_StrepXerD_C_like'),(968652,'gnl|CDD|311487',_binary 'Host cell surface-exposed lipoprotein.  This is a family of lipoproteins that is involved in superinfection exclusion. Proteins in this family have been shown to act at the stage of DNA release from the phage head into the cell.','pfam07553','Lipoprotein_Ltp'),(968664,'gnl|CDD|312473',_binary 'Domain of unknown function (DUF1906).  This domain is found in a set of uncharacterized hypothetical bacterial proteins.','pfam08924','DUF1906'),(968665,'gnl|CDD|119380',_binary 'BacA is a bacterial lysin from Enterococcus faecalis that degrades bacterial cell walls by catalyzing the hydrolysis of 1,4-beta-linkages between N-acetylmuramic acid and N-acetyl-D-glucosamine residues.  BacA is homologous to the YbfG and YkuG lysins of Bacillus subtilis. BacA has a C-terminal catalytic glycosyl hydrolase family 25 (GH25) domain and an N-terminal peptidoglycan-binding domain comprised of three alpha helices which is similar to a domain found in matrixins.','cd06418','GH25_BacA-like'),(968682,'gnl|CDD|225100',_binary 'Adenine specific DNA methylase Mod [DNA replication, recombination, and repair].','COG2189','COG2189'),(968714,'gnl|CDD|307502',_binary 'Trp repressor protein.  This protein binds to tryptophan and represses transcription of the Trp operon.','pfam01371','Trp_repressor'),(968716,'gnl|CDD|308441',_binary 'Helix-turn-helix domain of resolvase.','pfam02796','HTH_7'),(968755,'gnl|CDD|223695',_binary 'Predicted phosphoesterase [General function prediction only].','COG0622','COG0622'),(968757,'gnl|CDD|224325',_binary 'Predicted ICC-like phosphoesterases [General function prediction only].','COG1407','COG1407'),(968852,'gnl|CDD|317048',_binary 'Siphovirus ReqiPepy6 Gp37-like protein.  This family includes numerous phage proteins from Siphoviruses. The function of this protein is uncertain, but it is related to pfam06605. In Rhodococcus phage ReqiPepy6 this protein is called Gp37.','pfam14594','Sipho_Gp37'),(968865,'gnl|CDD|99912',_binary 'DNA polymerase type-B family catalytic domain. DNA-directed DNA polymerases elongate DNA by adding nucleotide triphosphate (dNTP) residues to the 5\'-end of the growing chain of DNA. DNA-directed DNA polymerases are multifunctional with both synthetic (polymerase) and degradative modes (exonucleases) and play roles in the processes of DNA replication, repair, and recombination. DNA-dependent DNA polymerases can be classified in six main groups based upon their phylogenetic relationships with E. coli polymerase I (class A), E. coli polymerase II (class B), E. coli polymerase III (class C), euryarchaeota polymerase II (class D), human polymerase beta (class x), E. coli UmuC/DinB, and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y).  Family B DNA polymerases include E. coli DNA polymerase II, some eubacterial phage DNA polymerases, nuclear replicative DNA polymerases (alpha, delta, epsilon, and zeta), and eukaryotic viral and plasmid-borne enzymes. DNA polymerase is made up of distinct domains and sub-domains. The polymerase domain of DNA polymerase type B (Pol domain) is responsible for the template-directed polymerization of dNTPs onto the growing primer strand of duplex DNA that is usually magnesium dependent. In general, the architecture of the Pol domain has been likened to a right hand with fingers, thumb, and palm sub-domains with a deep groove to accommodate the nucleic acid substrate. There are a few conserved motifs in the Pol domain of family B DNA polymerases. The conserved aspartic acid residues in the DTDS motifs of the palm sub-domain is crucial for binding to divalent metal ion and is suggested to be important for polymerase catalysis.','cd00145','POLBc'),(968866,'gnl|CDD|223494',_binary 'DNA polymerase elongation subunit (family B) [DNA replication, recombination, and repair].','COG0417','PolB'),(968867,'gnl|CDD|214691',_binary 'DNA polymerase type-B family.  DNA polymerase alpha, delta, epsilon and zeta chain (eukaryota), DNA polymerases in archaea, DNA polymerase II in e. coli, mitochondrial DNA polymerases and and virus DNA polymerases.','smart00486','POLBc'),(968870,'gnl|CDD|99914',_binary 'DNA polymerase type-B B2 subfamily catalytic domain. Archaeal proteins that are involved in DNA replication are similar to those from eukaryotes. Some archaeal members also possess multiple family B DNA polymerases (B1, B2 and B3). So far there is no specific function(s) has been assigned for different members of the archaea type B DNA polymerases. Phylogenetic analyses of eubacterial, archaeal, and eukaryotic family B DNA polymerases are support independent gene duplications during the evolution of archaeal and eukaryotic family B DNA polymerases.','cd05531','POLBc_B2'),(968872,'gnl|CDD|99919',_binary 'DNA polymerase type-B B3 subfamily catalytic domain. Archaeal proteins that are involved in DNA replication are similar to those from eukaryotes. Some members of the archaea also possess multiple family B DNA polymerases (B1, B2 and B3). So far there is no specific function(s) has been assigned for different members of the archaea type B DNA polymerases. Phylogenetic analyses of eubacterial, archaeal, and eukaryotic family B DNA polymerases are support independent gene duplications during the evolution of archaeal and eukaryotic family B DNA polymerases. Structural comparison of the thermostable DNA polymerase type B to its mesostable homolog suggests several adaptations to high temperature such as shorter loops, disulfide bridges, and increasing electrostatic interaction at subdomain interfaces.','cd05536','POLBc_B3'),(968873,'gnl|CDD|313979',_binary 'Protein of unknown function (DUF2800).  This is a family of uncharacterized proteins found in bacteria and viruses. Some members of this family are annotated as being Phi APSE P51-like proteins.','pfam10926','DUF2800'),(969180,'gnl|CDD|206733',_binary 'Elongation factor G (EF-G) family.  Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains only bacterial members.','cd04170','EF-G_bact'),(969324,'gnl|CDD|239247',_binary 'TRX domain, novel NADPH thioredoxin reductase (NTR) family; composed of fusion proteins found only in oxygenic photosynthetic organisms containing both TRX and NTR domains. The TRX domain functions as a protein disulfide reductase via the reversible oxidation of an active center dithiol present in a CXXC motif, while the NTR domain functions as a reductant to oxidized TRX. The fusion protein is  bifunctional, showing both TRX and NTR activities, but it is not an independent NTR/TRX system. In plants, the protein is found exclusively in shoots and mature leaves and is localized in the chloroplast. It is involved in plant protection against oxidative stress.','cd02949','TRX_NTR'),(969385,'gnl|CDD|213177',_binary 'chitin-binding domain of Chi A1-like proteins.  This group contains proteins related to the chitin binding domain of chitinase A1 (ChiA1) of Bacillus circulans WL-12. Glycosidase ChiA1 hydrolyzes chitin and is comprised of several domains: the C-terminal chitin binding domain, an N-terminal and catalytic domain, and 2 fibronectin type III-like domains. Chitinases function in invertebrates in the degradation of old exoskeletons, in fungi to utilize chitin in cell walls, and in bacteria which use chitin as an energy source. Bacillus circulans WL-12 ChiA1 facilitates invasion of fungal cell walls. The ChiAi chitin binding domain is required for the specific recognition of insoluble chitin. although topologically and structurally related, ChiA1 lacks the characteristic aromatic residues of Erwinia chrysanthemi endoglucanase Z (CBD(EGZ)).','cd12214','ChiA1_BD'),(969387,'gnl|CDD|316407',_binary 'Chaperone of endosialidase.  This is the very C-terminal, chaperone, domain of the bacteriophage protein endosialidase. It releases itself, via the serine-lysine dyad at the N-terminus, from the remainder of the end-tail-spike. Cleavage occurs after the threonine which is the final residue of the End-tail-spike family, pfam12219. The endosialidase protein forms homotrimeric molecules in bacteriophages. The catalytic dyad allows this portion of the molecule to be cleaved from the more N-terminal region such that the latter can fold and presumably bind to DNA.','pfam13884','Peptidase_S74'),(969389,'gnl|CDD|131643',_binary 'TIGR02594 family protein.  Members of this protein family known so far are restricted to the bacteria, and for the most to the proteobacteria. The function is unknown.','TIGR02594','hypothetical_protein'),(969392,'gnl|CDD|180120',_binary 'bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase protein; Provisional.','PRK05506','PRK05506'),(969434,'gnl|CDD|171440',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Provisional.','PRK12365','PRK12365'),(969439,'gnl|CDD|239738',_binary 'Serine Recombinase (SR) family, IS607-like transposase subfamily, catalytic domain; members contain a DNA binding domain with homology to MerR/SoxR located N-terminal to the catalytic domain. Serine recombinases catalyze site-specific recombination of DNA molecules by a concerted, four-strand cleavage and rejoining mechanism which involves a transient phosphoserine linkage between DNA and the enzyme. They are functionally versatile and include resolvases, invertases, integrases, and transposases. This subfamily is composed of proteins that catalyze the transposition of insertion sequence (IS) elements such as IS607 from Helicobacter and IS1535 from Mycobacterium, and similar proteins from other bacteria and several archaeal species. IS elements are DNA segments that move to new sites in prokaryotic and eukaryotic genomes causing insertion mutations and gene rearrangements.','cd03769','SR_IS607_transposase_like'),(969442,'gnl|CDD|279766',_binary 'Transglycosylase SLT domain.  This family is distantly related to pfam00062. Members are found in phages, type II, type III and type IV secretion systems.','pfam01464','SLT'),(969444,'gnl|CDD|235562',_binary 'flagellar assembly protein H; Validated.','PRK05687','fliH'),(969445,'gnl|CDD|237372',_binary 'flagellar assembly protein H; Provisional.','PRK13386','fliH'),(969585,'gnl|CDD|100119',_binary 'Sigma70, region (SR) 4 refers to the most C-terminal of four conserved domains found in Escherichia coli (Ec) sigma70, the main housekeeping sigma, and related sigma-factors (SFs). A SF is a dissociable subunit of RNA polymerase, it directs bacterial or plastid core RNA polymerase to specific promoter elements located upstream of transcription initiation points. The SR4 of Ec sigma70 and other essential primary SFs contact promoter sequences located 35 base-pairs upstream of the initiation point, recognizing a 6-base-pair -35 consensus TTGACA.  Sigma70 related SFs also include SFs which are dispensable for bacterial cell growth for example Ec sigmaS, SFs which activate regulons in response to a specific signal for example heat-shock Ec sigmaH, and a group of SFs which includes the extracytoplasmic function (ECF) SFs and is typified by Ec sigmaE which contains SR2 and -4 only. ECF SFs direct the transcription of genes that regulate various responses including periplasmic stress and pathogenesis.   Ec sigmaE SR4 also contacts the -35 element, but recognizes a different consensus (a 7-base-pair GGAACTT).  Plant SFs recognize sigma70 type promoters and direct transcription of the major plastid RNA polymerase, plastid-encoded RNA polymerase (PEP).','cd06171','Sigma70_r4'),(969586,'gnl|CDD|224511',_binary 'DNA-directed RNA polymerase specialized sigma subunit, sigma24 homolog [Transcription].','COG1595','RpoE'),(969587,'gnl|CDD|311954',_binary 'Sigma-70, region 4.  Region 4 of sigma-70 like sigma-factors are involved in binding to the -35 promoter element via a helix-turn-helix motif.','pfam08281','Sigma70_r4_2'),(969588,'gnl|CDD|274357',_binary 'RNA polymerase sigma factor, sigma-70 family.  This model encompasses all varieties of the sigma-70 type sigma factors including the ECF subfamily. A number of sigma factors have names with a different number than 70 (i.e. sigma-38), but in fact, all except for the Sigma-54 family (TIGR02395) are included within this family. Several Pfam models hit segments of these sequences including Sigma-70 region 2 (pfam04542) and Sigma-70, region 4 (pfam04545), but not always above their respective trusted cutoffs.','TIGR02937','RNA_polymerase_sigma_fact'),(969589,'gnl|CDD|309608',_binary 'Sigma-70, region 4.  Region 4 of sigma-70 like sigma-factors are involved in binding to the -35 promoter element via a helix-turn-helix motif. Due to the way Pfam works, the threshold has been set artificially high to prevent overlaps with other helix-turn-helix families. Therefore there are many false negatives.','pfam04545','Sigma70_r4'),(969590,'gnl|CDD|132028',_binary 'RNA polymerase sigma-70 factor, sigma-E family.  This group of similar sigma-70 factors includes the sigE factor from Streptomyces coelicolor. The family appears to include a paralagous expansion in the Streptomycetes lineage, while related Actinomycetales have at most two representatives.','TIGR02983','putative_RNA_polymerase_E'),(969591,'gnl|CDD|180757',_binary 'positive control sigma-like factor; Validated.','PRK06930','PRK06930'),(969592,'gnl|CDD|254323',_binary 'ECF sigma factor.  These proteins are probably RNA polymerase sigma factors belonging to the extra-cytoplasmic function (ECF) subfamily and show sequence similarity to pfam04542 and pfam04545.','pfam07638','Sigma70_ECF'),(969598,'gnl|CDD|273953',_binary 'malonyl-acyl carrier protein O-methyltransferase BioC.  This enzyme, which is found in biotin biosynthetic gene clusters in proteobacteria, firmicutes, green-sulfur bacteria, fusobacterium and bacteroides, carries out an enzymatic step prior to the formation of pimeloyl-CoA, namely O-methylation of the malonyl group preferentially while on acyl carrier protein. The enzyme is recognizable as a methyltransferase by homology. [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin].','TIGR02072','Malonyl-_O-methyltransfer'),(969600,'gnl|CDD|223574',_binary 'SAM-dependent methyltransferases [Secondary metabolites biosynthesis, transport, and catabolism / General function prediction only].','COG0500','SmtA'),(969601,'gnl|CDD|225139',_binary 'Cyclopropane fatty acid synthase and related methyltransferases [Cell envelope biogenesis, outer membrane].','COG2230','Cfa'),(969602,'gnl|CDD|225136',_binary 'Methylase involved in ubiquinone/menaquinone biosynthesis [Coenzyme metabolism].','COG2226','UbiE'),(969604,'gnl|CDD|273930',_binary 'magnesium protoporphyrin O-methyltransferase.  This model represents the S-adenosylmethionine-dependent O-methyltransferase responsible for methylation of magnesium protoporphyrin IX. This step is essentiasl for the biosynthesis of both chlorophyll and bacteriochlorophyll. This model encompasses two closely related clades, from cyanobacteria (and plants) where it is called ChlM and other photosynthetic bacteria where it is known as BchM. [Biosynthesis of cofactors, prosthetic groups, and carriers, Chlorophyll and bacteriochlorphyll].','TIGR02021','magnesium-protoporphyrin_'),(969605,'gnl|CDD|237872',_binary 'putative methyltransferase; Provisional.','PRK14968','PRK14968'),(969606,'gnl|CDD|226591',_binary 'Trans-aconitate methyltransferase [General function prediction only].','COG4106','Tam'),(969607,'gnl|CDD|273884',_binary 'ubiquinone/menaquinone biosynthesis methyltransferases.  This model represents a family of methyltransferases involved in the biosynthesis of menaquinone and ubiqinone. Some members such as the UbiE enzyme from E. coli are believed to act in both pathways, while others may act in only the menaquinone pathway. These methyltransferases are members of the UbiE/CoQ family of methyltransferases (pfam01209) which also contains ubiquinone methyltransferases and other methyltransferases. Members of this clade include a wide distribution of bacteria and eukaryotes, but no archaea. An outgroup for this clade is provided by the phosphatidylethanolamine methyltransferase (EC 2.1.1.17) from Rhodobacter sphaeroides. Note that a number of non-orthologous genes which are members of pfam03737 have been erroneously annotated as MenG methyltransferases. [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR01934','methylase_mitochondrial'),(969608,'gnl|CDD|227310',_binary 'Predicted methyltransferase (contains TPR repeat) [General function prediction only].','COG4976','COG4976'),(969609,'gnl|CDD|236059',_binary 'Mg-protoporphyrin IX methyl transferase; Validated.','PRK07580','PRK07580'),(969610,'gnl|CDD|279542',_binary 'ubiE/COQ5 methyltransferase family.','pfam01209','Ubie_methyltran'),(969611,'gnl|CDD|183282',_binary 'cyclopropane fatty acyl phospholipid synthase; Provisional.','PRK11705','PRK11705'),(969612,'gnl|CDD|280508',_binary 'Mycolic acid cyclopropane synthetase.  This family consist of Cyclopropane-fatty-acyl-phospholipid synthase or CFA synthase EC:2.1.1.79 this enzyme catalyze the reaction: S-adenosyl-L-methionine + phospholipid olefinic fatty acid <=> S-adenosyl-L-homocysteine + phospholipid cyclopropane fatty acid.','pfam02353','CMAS'),(969613,'gnl|CDD|225443',_binary 'Methylase of polypeptide chain release factors [Translation, ribosomal structure and biogenesis].','COG2890','HemK'),(969614,'gnl|CDD|226608',_binary 'Predicted O-methyltransferase [General function prediction only].','COG4123','COG4123'),(969616,'gnl|CDD|225316',_binary 'Protein-L-isoaspartate carboxylmethyltransferase [Posttranslational modification, protein turnover, chaperones].','COG2518','Pcm'),(969617,'gnl|CDD|234970',_binary 'trans-aconitate 2-methyltransferase; Provisional.','PRK01683','PRK01683'),(969618,'gnl|CDD|237007',_binary 'arsenite S-adenosylmethyltransferase; Reviewed.','PRK11873','arsM'),(969619,'gnl|CDD|234689',_binary 'ubiquinone/menaquinone biosynthesis methyltransferase; Reviewed.','PRK00216','ubiE'),(969620,'gnl|CDD|273958',_binary 'methionine biosynthesis protein MetW.  This protein is found alongside MetX, of the enzyme that acylates homoserine as a first step toward methionine biosynthesis, in many species. It appears to act in methionine biosynthesis but is not fully characterized. [Amino acid biosynthesis, Aspartate family].','TIGR02081','conserved_hypothetical_pr'),(969621,'gnl|CDD|182340',_binary 'biotin biosynthesis protein BioC; Provisional.','PRK10258','PRK10258'),(969623,'gnl|CDD|214839',_binary 'Methyltransferase in polyketide synthase (PKS) enzymes.','smart00828','PKS_MT'),(969625,'gnl|CDD|254024',_binary 'Methionine biosynthesis protein MetW.  This family consists of several bacterial and one archaeal methionine biosynthesis MetW proteins. Biosynthesis of methionine from homoserine in Pseudomonas putida takes place in three steps. The first step is the acylation of homoserine to yield an acyl-L-homoserine. This reaction is catalyzed by the products of the metXW genes and is equivalent to the first step in enterobacteria, gram-positive bacteria and fungi, except that in these microorganisms the reaction is catalyzed by a single polypeptide (the product of the metA gene in Escherichia coli and the met5 gene product in Neurospora crassa). In Pseudomonas putida, as in gram-positive bacteria and certain fungi, the second and third steps are a direct sulfhydrylation that converts the O-acyl-L-homoserine into homocysteine and further methylation to yield methionine. The latter reaction can be mediated by either of the two methionine synthetases present in the cells.','pfam07021','MetW'),(969657,'gnl|CDD|239245',_binary 'TRX family; composed of two groups: Group I, which includes proteins that exclusively encode a TRX domain; and Group II, which are composed of fusion proteins of TRX and additional domains. Group I TRX is a small ancient protein that alter the redox state of target proteins via the reversible oxidation of an active site dithiol, present in a CXXC motif, partially exposed at the protein\'s surface. TRX reduces protein disulfide bonds, resulting in a disulfide bond at its active site. Oxidized TRX is converted to the active form by TRX reductase, using reducing equivalents derived from either NADPH or ferredoxins. By altering their redox state, TRX regulates the functions of at least 30 target proteins, some of which are enzymes and transcription factors. It also plays an important role in the defense against oxidative stress by directly reducing hydrogen peroxide and certain radicals, and by serving as a reductant for peroxiredoxins. At least two major types of functional TRXs have been reported in most organisms; in eukaryotes, they are located in the cytoplasm and the mitochondria. Higher plants contain more types (at least 20 TRX genes have been detected in the genome of Arabidopsis thaliana), two of which (types f amd m) are located in the same compartment, the chloroplast. Also included in the alignment are TRX-like domains which show sequence homology to TRX but do not contain the redox active CXXC motif. Group II proteins, in addition to either a redox active TRX or a TRX-like domain, also contain additional domains, which may or may not possess homology to known proteins.','cd02947','TRX_family'),(969685,'gnl|CDD|132002',_binary 'RNA polymerase sigma-70 factor, TIGR02957 family.  This group of sigma factors are members of the sigma-70 family (TIGR02937). They and appear by homology, tree building and bidirectional best hits, to represent a conserved family. This family is found in a limited number of bacterial lineages. This family includes apparent paralogous expansion in Streptomyces coelicolor A3(2), and multiple copies in Mycobacterium smegmatis MC2, Streptomyces avermitilis MA-4680 and Nocardia farcinica IFM10152.','TIGR02957','putative_sigma_factor'),(969687,'gnl|CDD|200072',_binary 'thioredoxin.  Several proteins, such as protein disulfide isomerase, have two or more copies of a domain closely related to thioredoxin. This model is designed to recognize authentic thioredoxin, a small protein that should be hit exactly once by this model. Any protein that hits once with a score greater than the second (per domain) trusted cutoff may be taken as thioredoxin. [Energy metabolism, Electron transport].','TIGR01068','Thioredoxin-like_protein_'),(969688,'gnl|CDD|182889',_binary 'thioredoxin 2; Provisional.','PRK10996','PRK10996'),(969689,'gnl|CDD|278513',_binary 'Thioredoxin.  Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond. Some members with only the active site are not separated from the noise.','pfam00085','Thioredoxin'),(969690,'gnl|CDD|239282',_binary 'TRX domain, PICOT (for PKC-interacting cousin of TRX) subfamily; PICOT is a protein that interacts with protein kinase C (PKC) theta, a calcium independent PKC isoform selectively expressed in skeletal muscle and T lymphocytes. PICOT contains an N-terminal TRX-like domain, which does not contain the catalytic CXXC motif, followed by one to three glutaredoxin domains. The TRX-like domain is required for interaction with PKC theta. PICOT inhibits the activation of c-Jun N-terminal kinase and the transcription factors, AP-1 and NF-kB, induced by PKC theta or T-cell activating stimuli.','cd02984','TRX_PICOT'),(969691,'gnl|CDD|315783',_binary 'Thioredoxin domain.','pfam13192','Thioredoxin_3'),(969692,'gnl|CDD|223600',_binary 'Thiol-disulfide isomerase and thioredoxins [Posttranslational modification, protein turnover, chaperones / Energy production and conversion].','COG0526','TrxA'),(969695,'gnl|CDD|315710',_binary 'Thioredoxin-like domain.','pfam13098','Thioredoxin_2'),(969696,'gnl|CDD|239262',_binary 'Tryparedoxin (TryX)-like family; composed of TryX and related proteins including nucleoredoxin (NRX), rod-derived cone viability factor (RdCVF) and the nematode homolog described as a 16-kD class of TRX. Most members of this family, except RdCVF, are protein disulfide oxidoreductases containing an active site CXXC motif, similar to TRX.','cd02964','TryX_like_family'),(969697,'gnl|CDD|129505',_binary 'small redox-active disulfide protein 1.  This protein is homologous to a family of proteins that includes thioredoxins, glutaredoxins, protein-disulfide isomerases, and others, some of which have several such domains. The sequence of this protein at the redox-active disufide site, CPYC, matches glutaredoxins rather than thioredoxins, although its overall sequence seems closer to thioredoxins. It is suggested to be a ribonucleotide-reducing system component distinct from thioredoxin or glutaredoxin. [Unknown function, General].','TIGR00411','Probable_Thioredoxin'),(969738,'gnl|CDD|238141',_binary 'SGNH_hydrolase, or GDSL_hydrolase, is a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the typical Ser-His-Asp(Glu) triad from other serine hydrolases, but may lack the carboxlic acid.','cd00229','SGNH_hydrolase'),(969739,'gnl|CDD|316033',_binary 'GDSL-like Lipase/Acylhydrolase family.  This family of presumed lipases and related enzymes are similar to pfam00657.','pfam13472','Lipase_GDSL_2'),(969740,'gnl|CDD|307002',_binary 'GDSL-like Lipase/Acylhydrolase.','pfam00657','Lipase_GDSL'),(969741,'gnl|CDD|238870',_binary 'Members of the SGNH-hydrolase superfamily, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad from other serine hydrolases, but may lack the carboxlic acid. Myxobacterial members of this subfamily have been reported to be involved in adventurous gliding motility.','cd01832','SGNH_hydrolase_like_1'),(969742,'gnl|CDD|238871',_binary 'SGNH_hydrolase subfamily, similar to Ruminococcus flavefaciens XynB. Most likely a secreted hydrolase with xylanase activity. SGNH hydrolases are a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01833','XynB_like'),(969743,'gnl|CDD|238863',_binary 'SGNH_peri1; putative periplasmic member of the SGNH-family of hydrolases, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01825','SGNH_hydrolase_peri1'),(969744,'gnl|CDD|313223',_binary 'Uncharacterized conserved protein (DUF2190).  This domain, found in various hypothetical prokaryotic proteins, as well as in some putative RecA/RadA recombinases, has no known function.','pfam09956','DUF2190'),(969745,'gnl|CDD|312966',_binary 'Gp37 protein.  This protein of 154 residues consists of a unit of helices and beta sheets that crystallizes into a beautiful asymmetrical dodecameric barrel-structure, of two six-membered rings one on top of the other. It is expressed in bacteria but is of viral origin as it is found in phage BcepMu and is probably a pathogenesis factor.','pfam09646','Gp37'),(969752,'gnl|CDD|227919',_binary 'N-acetylmuramoyl-L-alanine amidase [Cell envelope biogenesis, outer membrane].','COG5632','COG5632'),(969753,'gnl|CDD|316043',_binary 'RNase_H superfamily.','pfam13482','RNase_H_2'),(969755,'gnl|CDD|99824',_binary 'DEDDy 3\'-5\' exonuclease domain of Sulfurisphaera ohwakuensis DNA polymerase B3 and similar archaeal family-B DNA polymerases.  The 3\'-5\' exonuclease domain of archaeal proteins with similarity to Sulfurisphaera ohwakuensis DNA polymerase B3. B3 is a family-B DNA polymerase. Family-B DNA polymerases contain an N-terminal DEDDy DnaQ-like exonuclease domain in the same polypeptide chain as the polymerase domain, similar to family-A DNA polymerases. B3 exhibits both polymerase and 3\'-5\' exonuclease activities. This exonuclease domain contains three sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and are involved in metal binding and catalysis. The exonuclease domain of family B polymerases also contains a beta hairpin structure that plays an important role in active site switching in the event of nucleotide misincorporation. Archaeal proteins that are involved in DNA replication are similar to those from eukaryotes. Some archaea possess multiple family-B DNA polymerases. B3 is mainly found in crenarchaea. Phylogenetic analyses of eubacterial, archaeal, and eukaryotic family B-DNA polymerases support independent gene duplications during the evolution of archaeal and eukaryotic family-B DNA polymerases.','cd05781','DNA_polB_B3_exo'),(969771,'gnl|CDD|315744',_binary 'Protein of unknown function (DUF3987).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences.','pfam13148','DUF3987'),(969780,'gnl|CDD|319763',_binary 'Haloacid dehalogenase-like hydrolases.  The haloacid dehalogenase-like (HAD) superfamily includes L-2-haloacid dehalogenase, epoxide hydrolase, phosphoserine phosphatase, phosphomannomutase, phosphoglycolate phosphatase, P-type ATPase, and many others. This superfamily includes a variety of enzymes that catalyze the cleavage of substrate C-Cl, P-C, and P-OP bonds via nucleophilic substitution pathways. All of which use a nucleophilic aspartate in their phosphoryl transfer reaction. They catalyze nucleophilic substitution reactions at phosphorus or carbon centers, using a conserved Asp carboxylate in covalent catalysis. All members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. Members of this superfamily are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd01427','HAD_like'),(969781,'gnl|CDD|222895',_binary 'RNA ligase A; Provisional.','PHA02589','rnlA'),(969782,'gnl|CDD|213699',_binary 'RNA ligase, T4 RnlA family.  Members of this family are phage proteins with ATP-dependent RNA ligase activity. Host defense to phage may include cleavage and inactivation of specific tRNA molecules; members of this family act to reverse this RNA damage. The enzyme is adenylated, transiently, on a Lys residue in a motif KXDGSL. [Mobile and extrachromosomal element functions, Prophage functions, Transcription, RNA processing].','TIGR02308','T4_RNA_ligase_1'),(969783,'gnl|CDD|312872',_binary 'RNA ligase.  Members of this family include T4 phage proteins with ATP-dependent RNA ligase activity. Host defense to phage may include cleavage and inactivation of specific tRNA molecules; members of this family act to reverse this RNA damage. The enzyme is adenylated, transiently, on a Lys residue in a motif KXDGSL. This family also includes fungal tRNA ligases that have adenylyltransferase activity. tRNA ligases are enzymes required for the splicing of precursor tRNA molecules containing introns.i.','pfam09511','RNA_lig_T4_1'),(969819,'gnl|CDD|275316',_binary 'helix-rich Mycoplasma protein.  Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.','TIGR04523','conserved_hypothetical_pr'),(969824,'gnl|CDD|238103',_binary 'Spectrin repeats, found in several proteins involved in cytoskeletal structure; family members include spectrin, alpha-actinin and dystrophin; the spectrin repeat forms a three helix bundle with the second helix interrupted by proline in some sequences; the repeats are independent folding units; tandem repeats are found in differing numbers and arrange in an antiparallel manner to form dimers; the repeats are defined by a characteristic tryptophan (W) residue in helix A and a leucine (L) at the carboxyl end of helix C and separated by a linker of 5 residues; two copies of the repeat are present here.','cd00176','SPEC'),(969831,'gnl|CDD|227287',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4951','COG4951'),(969846,'gnl|CDD|99836',_binary 'DEDDh 3\'-5\' exonuclease domain of Caenorhabditis elegans ERI-1, human 3\' exonuclease, and similar proteins.  This subfamily is composed of Caenorhabditis elegans ERI-1, human 3\' exonuclease (3\'hExo), Drosophila exonuclease snipper (snp), and similar proteins from eukaryotes and bacteria. These are DEDDh-type DnaQ-like 3\'-5\' exonucleases containing three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. ERI-1 has been implicated in the degradation of small interfering RNAs (RNAi). 3\'hExo participates in the degradation of histone mRNAs. Snp is a non-essential exonuclease that efficiently degrades structured RNA and DNA substrates as long as there is a minimum of 2 nucleotides in the 3\' overhang to initiate degradation. Snp is not a functional homolog of either ERI-1 or 3\'hExo.','cd06133','ERI-1_3\'hExo_like'),(969868,'gnl|CDD|130711',_binary 'cobaltochelatase, CobS subunit.  This model describes Pseudomonas denitrificans CobS gene product, which is a cobalt chelatase subunit that functions in cobalamin biosynthesis. Cobalamin (vitamin B12) can be synthesized via several pathways, including an aerobic pathway (found in Pseudomonas denitrificans) and an anaerobic pathway (found in P. shermanii and Salmonella typhimurium). These pathways differ in the point of cobalt insertion during corrin ring formation. There are apparently a number of variations on these two pathways, where the major differences seem to be concerned with the process of ring contraction. Confusion regarding the functions of enzymes found in the aerobic vs. anaerobic pathways has arisen because nonhomologous genes in these different pathways were given the same gene symbols. Thus, cobS in the aerobic pathway (P. denitrificans) is not a homolog of cobS in the anaerobic pathway (S. typhimurium). It should be noted that E. coli synthesizes cobalamin only when it is supplied with the precursor cobinamide, which is a complex intermediate. Additionally, all E. coli cobalamin synthesis genes (cobU, cobS and cobT) were named after their Salmonella typhimurium homologs which function in the anaerobic cobalamin synthesis pathway. This model describes the aerobic cobalamin pathway Pseudomonas denitrificans CobS gene product, which is a cobalt chelatase subunit, with a MW ~37 kDa. The aerobic pathway cobalt chelatase is a heterotrimeric, ATP-dependent enzyme that catalyzes cobalt insertion during cobalamin biosynthesis. The other two subunits are the P. denitrificans CobT (TIGR01651) and CobN (pfam02514 CobN/Magnesium Chelatase) proteins. To avoid potential confusion with the nonhomologous Salmonella typhimurium/E.coli cobS gene product, the P. denitrificans gene symbol is not used in the name of this model. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR01650','PD_CobS'),(969929,'gnl|CDD|212142',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in Sulfolobus solfataricus (Ss) and its homologs from archaea and bacteria.  This family includes a MazG-like protein from Sulfolobus solfataricus (SsMazG) and its homologs from archaea and bacteria. Although its biological roles remain still unclear, SsMazG shows significant sequence similarity to the NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, SsMazG contains a single MazG-like domain. It is predicted that SsMazG might participate in house-cleaning by preventing incorporation of the oxidation product 2-oxo-(d)ATP (iso-dGTP), a mutagenic derivative of ATP, into DNA.','cd11535','NTP-PPase_SsMazG'),(969936,'gnl|CDD|185724',_binary 'Glycosyl hydrolase family 43.  This glycosyl hydrolase family 43 (GH43) includes enzymes with beta-1,4-xylosidase (xylan 1,4-beta-xylosidase; EC 3.2.1.37), beta-1,3-xylosidase (EC 3.2.1.-), alpha-L-arabinofuranosidase (EC 3.2.1.55), arabinanase (EC 3.2.1.99), xylanase (EC 3.2.1.8), endo-alpha-L-arabinanase and galactan 1,3-beta-galactosidase (EC 3.2.1.145) activities. These are inverting enzymes (i.e. they invert the stereochemistry of the anomeric carbon atom of the substrate) that have an aspartate as the catalytic general base, a glutamate as the catalytic general acid and another aspartate that is responsible for pKa modulation and orienting the catalytic acid. Many of the enzymes in this family display both alpha-L-arabinofuranosidase and beta-D-xylosidase activity using aryl-glycosides as substrates. A common structural feature of GH43 enzymes is a 5-bladed beta-propeller domain that contains the catalytic acid and catalytic base. A long V-shaped groove, partially enclosed at one end, forms a single extended substrate-binding surface across the face of the propeller.','cd08983','GH43_4'),(969946,'gnl|CDD|224049',_binary 'ABC-type dipeptide/oligopeptide/nickel transport system, ATPase component [Amino acid transport and metabolism / Inorganic ion transport and metabolism].','COG1124','DppF'),(969957,'gnl|CDD|313717',_binary 'P22_AR N-terminal domain.  This domain was identified by Iyer and colleagues.','pfam10547','P22_AR_N'),(969992,'gnl|CDD|235616',_binary 'single-stranded DNA-binding protein; Provisional.','PRK05813','PRK05813'),(970003,'gnl|CDD|177331',_binary 'capsid protein.','PHA02004','PHA02004'),(970004,'gnl|CDD|223910',_binary 'Methyl-accepting chemotaxis protein [Cell motility and secretion / Signal transduction mechanisms].','COG0840','Tar'),(970005,'gnl|CDD|214599',_binary 'Methyl-accepting chemotaxis-like domains (chemotaxis sensory transducer).  Thought to undergo reversible methylation in response to attractants or repellants during bacterial chemotaxis.','smart00283','MA'),(970065,'gnl|CDD|319765',_binary 'vegetative storage proteins and related proteins, similar to soybean VSPalpha and VSPbeta proteins; belongs to the haloacid dehalogenase-like superfamily.  Soybean [Glycine max (L.) Merr.] vegetative storage protein VSPalpha and VSPbeta levels were identified as storage proteins due to their abundance and pattern of expression in plant tissues, they accumulate to almost one-half the amount of soluble leaf protein when soybean plants are continually depodded. They possess acid phosphatase activity which appears to be low compared to several other plant acid phosphatases; it increases in the leaves of depodded soybean plants, but to no more than 0.1% of the total acid phosphatase activity in these leaves. This acid phosphatase activity has maximal activity at pH 5.0 - 5.5, and can liberate Pi from different substrates such as napthyl acid phosphate, carboxyphenyl phosphate, sugar-phosphates, glyceraldehyde 3-phosphate, dihydroxyacetone phosphate, phosphoenolpyruvate, ATP, ADP, PPi, and short chain polyphosphates; they cleave phosphoenolpyruvate, ATP, ADP, PPI, and polyphosphates most efficiently. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Soybean VSPalpha and VSPbeta lack this active site aspartate, other members of this family have this aspartate and may be more active.  Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd01624','HAD_VSP_like'),(970146,'gnl|CDD|182607',_binary 'glutaredoxin 3; Provisional.','PRK10638','PRK10638'),(970160,'gnl|CDD|237833',_binary 'translation initiation factor IF-2; Provisional.','PRK14845','PRK14845'),(970177,'gnl|CDD|307293',_binary 'Hint module.  This is an alignment of the Hint module in the Hedgehog proteins. It does not include any Inteins which also possess the Hint module.','pfam01079','Hint'),(970178,'gnl|CDD|317072',_binary 'Hint-domain.  This short domain is a conserved region of intein-containing proteins from lower eukaryotes.','pfam14623','Vint'),(970179,'gnl|CDD|235223',_binary 'replication factor C small subunit; Provisional.','PRK04132','PRK04132'),(970180,'gnl|CDD|213027',_binary 'Putative catalytic NodB homology domain of uncharacterized prokaryotic polysaccharide deacetylases which consist of a 6-stranded beta/alpha barrel.  This family contains uncharacterized prokaryotic polysaccharide deacetylases. Although their biological functions remain unknown, all members of the family contain a conserved domain with a 6-stranded beta/alpha barrel, which is similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, belonging to the larger carbohydrate esterase 4 (CE4) superfamily.','cd10970','CE4_DAC_u1_6s'),(970181,'gnl|CDD|200589',_binary 'Putative catalytic NodB homology domain of gellan lyase and similar proteins.  This family is represented by the extracellular polysaccharide-degrading enzyme, gellan lyase (gellanase, EC 4.2.2.-), from Bacillus sp. The enzyme acts on gellan exolytically and releases a tetrasaccharide of glucuronyl-glucosyl-rhamnosyl-glucose with unsaturated glucuronic acid at the nonreducing terminus. The family also includes many uncharacterized prokaryotic polysaccharide deacetylases, which show high sequence similarity to Bacillus sp. gellan lyase. Although their biological functions remain unknown, all members of the family contain a conserved domain with a 6-stranded beta/alpha barrel, which is similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, belonging to the larger carbohydrate esterase 4 (CE4) superfamily.','cd10967','CE4_GLA_like_6s'),(970183,'gnl|CDD|213023',_binary 'Putative catalytic NodB homology domain of PgaB, IcaB, and similar proteins which consist of a deformed (beta/alpha)8 barrel fold with 5- or 6-strands.  This family belongs to the large and functionally diverse carbohydrate esterase 4 (CE4) superfamily, whose members show strong sequence similarity with some variability due to their distinct carbohydrate substrates. It includes bacterial poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase PgaB, hemin storage system HmsF protein in gram-negative species, intercellular adhesion proteins IcaB, and many uncharacterized prokaryotic polysaccharide deacetylases. It also includes a putative polysaccharide deacetylase YxkH encoded by the Bacillus subtilis yxkH gene, which is one of six polysaccharide deacetylase gene homologs present in the Bacillus subtilis genome. Sequence comparison shows all family members contain a conserved domain similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, which consists of a deformed (beta/alpha)8 barrel fold with 6 or 7 strands. However, in this family, most proteins have 5 strands and some have 6 strands. Moreover, long insertions are found in many family members, whose function remains unknown.','cd10918','CE4_NodB_like_5s_6s'),(970184,'gnl|CDD|213024',_binary 'Putative catalytic polysaccharide deacetylase domain of uncharacterized protein yadE and similar proteins.  This family contains an uncharacterized protein yadE from Escherichia coli and its bacterial homologs. Although its molecular function remains unknown, yadE shows high sequence similarity with the catalytic NodB homology domain of outer membrane lipoprotein PgaB and the surface-attached protein intercellular adhesion protein IcaB. Both PgaB and IcaB are essential in bacterial biofilm formation.','cd10966','CE4_yadE_5s'),(970204,'gnl|CDD|310452',_binary 'DNA N-6-adenine-methyltransferase (Dam).  This family consists of several bacterial and phage DNA N-6-adenine-methyltransferase (Dam) like sequences.','pfam05869','Dam'),(970205,'gnl|CDD|273769',_binary 'phage N-6-adenine-methyltransferase.  This model is a fragment-mode model for a phage-borne DNA N-6-adenine-methyltransferase. [Mobile and extrachromosomal element functions, Prophage functions, DNA metabolism, Restriction/modification].','TIGR01712','DNA_N-6-adenine-methyltra'),(970280,'gnl|CDD|273678',_binary 'phage portal protein, SPP1 family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01538','Portal_protein'),(970314,'gnl|CDD|275267',_binary 'three-Cys-motif partner protein.  Members of this family occur regularly as a partner to as a member of family pfam07505, which has been called a phage protein but which seems to occur also in other contexts. Members average about 400 residues in length, but the conserved region covered by the model averages 260 residues and excludes the C-terminus. Conserved motifs suggest enzymatic activity. Note that its frequent partner protein (see pfam07505) has a three-cysteine motif that resembles the Cx3CxxC motif of radical SAM proteins, and that in one branch (see TIGR04471) actually becomes Cx3CxxC. We suggest the name three-Cys-motif partner protein (tcmP), and renaming pfam07505 to three-Cys-motif family protein.','TIGR04474','conserved_hypothetical_pr'),(970319,'gnl|CDD|313316',_binary 'Uncharacterized conserved protein (DUF2303).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10065','DUF2303'),(970395,'gnl|CDD|313378',_binary 'vWA found in TerF C terminus.  vWA domain fused to TerD domain typified by the TerF protein. Some times found as solos.','pfam10138','vWA-TerF-like'),(970396,'gnl|CDD|238734',_binary 'VWA ORF176 type: Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses. In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains. The members of this subgroup are Eubacterial in origin and have a conserved MIDAS motif. Not much is known about the biochemistry of these.','cd01457','vWA_ORF176_type'),(970402,'gnl|CDD|227634',_binary 'Uncharacterized conserved protein [Function unknown].','COG5324','COG5324'),(970408,'gnl|CDD|309043',_binary 'HAD superfamily, subfamily IIIB (Acid phosphatase).  This family proteins includes acid phosphatases and a number of vegetative storage proteins.','pfam03767','Acid_phosphat_B'),(970411,'gnl|CDD|238698',_binary 'SIR2 family of proteins includes silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation, where the acetyl group from the lysine epsilon-amino group is transferred to the ADP-ribose moiety of NAD+, producing nicotinamide and the novel metabolite O-acetyl-ADP-ribose. Sir2 proteins, also known as sirtuins, are found in all eukaryotes and many archaea and prokaryotes and have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span. The most-studied function, gene silencing, involves the inactivation of chromosome domains containing key regulatory genes by packaging them into a specialized chromatin structure that is inaccessible to DNA-binding proteins. The oligomerization state of Sir2 appears to be organism-dependent, sometimes occurring as a monomer and sometimes as a multimer.','cd01407','SIR2-fam'),(970412,'gnl|CDD|223915',_binary 'NAD-dependent protein deacetylases, SIR2 family [Transcription].','COG0846','SIR2'),(970413,'gnl|CDD|239230',_binary 'Macro domain, Poa1p_like family. The macro domain is a high-affinity ADP-ribose binding module found in a variety of proteins as a stand-alone domain or in combination with other domains like in histone macroH2A and some PARPs (poly ADP-ribose polymerases). Some macro domains recognize poly ADP-ribose as a ligand. Previously identified as displaying an Appr-1\'-p (ADP-ribose-1\'-monophosphate) processing activity, the macro domain may play roles in distinct ADP-ribose pathways, such as the ADP-ribosylation of proteins, an important post-translational modification which occurs in DNA repair, transcription, chromatin biology, and long-term memory formation, among other processes. Members of this family show similarity to the yeast protein Poa1p, reported to be a phosphatase specific for Appr-1\'-p, a tRNA splicing metabolite. Poa1p may play a role in tRNA splicing regulation.','cd02901','Macro_Poa1p_like'),(970414,'gnl|CDD|234777',_binary 'NAD-dependent deacetylase; Provisional.','PRK00481','PRK00481'),(970415,'gnl|CDD|239447',_binary 'Macro domain, Poa1p_like family, SNF2 subfamily. The macro domain is a high-affinity ADP-ribose binding module found in a variety of proteins as a stand-alone domain or in combination with other domains like in histone macroH2A and some PARPs (poly ADP-ribose polymerases). Some macro domains recognize poly ADP-ribose as a ligand. Previously identified as displaying an Appr-1\'-p (ADP-ribose-1\'-monophosphate) processing activity, the macro domain may play roles in distinct ADP-ribose pathways, such as the ADP-ribosylation of proteins, an important post-translational modification which occurs in DNA repair, transcription, chromatin biology, and long-term memory formation, among other processes. Members of this subfamily contain a C-terminal macro domain that show similarity to the yeast protein Poa1p, reported to be a phosphatase specific for Appr-1\'-p, a tRNA splicing metabolite. In addition, they also contain an SNF2 domain, defined by the presence of seven motifs with sequence similarity to DNA helicases. SNF2 proteins have the capacity to use the energy released by their DNA-dependent ATPase activity to stabilize or perturb protein-DNA interactions and play important roles in transcriptional regulation, maintenance of chromosome integrity and DNA repair.','cd03331','Macro_Poa1p_like_SNF2'),(970416,'gnl|CDD|238184',_binary 'SIR2 superfamily of proteins includes silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation, where the acetyl group from the lysine epsilon-amino group is transferred to the ADP-ribose moiety of NAD+, producing nicotinamide and the novel metabolite O-acetyl-ADP-ribose. Sir2 proteins, also known as sirtuins, are found in all eukaryotes and many archaea and prokaryotes and have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span. The most-studied function, gene silencing, involves the inactivation of chromosome domains containing key regulatory genes by packaging them into a specialized chromatin structure that is inaccessible to DNA-binding proteins. The oligomerization state of Sir2 appears to be organism-dependent, sometimes occurring as a monomer and sometimes as a multimer. Also included in this superfamily is a group of uncharacterized Sir2-like proteins which lack certain key catalytic residues and conserved zinc binding cysteines.','cd00296','SIR2'),(970417,'gnl|CDD|214701',_binary 'Appr-1\'-p processing enzyme.  Function determined by Martzen et al. Extended family detected by reciprocal PSI-BLAST searches (unpublished results, and Pehrson & Fuji).','smart00506','A1pp'),(970418,'gnl|CDD|238703',_binary 'SIRT5_Af1_CobB: Eukaryotic, archaeal and prokaryotic group (class3) which includes human sirtuin SIRT5, Archaeoglobus fulgidus Sir2-Af1, and E. coli CobB; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span. CobB is a bacterial sirtuin that deacetylates acetyl-CoA synthetase at an active site lysine to stimulate its enzymatic activity. .','cd01412','SIRT5_Af1_CobB'),(970419,'gnl|CDD|222899',_binary 'hypothetical protein; Provisional.','PHA02595','tk.4'),(970420,'gnl|CDD|238704',_binary 'SIR2_Af2: Archaeal and prokaryotic group which includes Archaeoglobus fulgidus Sir2-Af2, Sulfolobus solfataricus ssSir2, and several bacterial homologs; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span. The Sir2 homolog from the archaea Sulfolobus solftaricus deacetylates the non-specific DNA protein Alba to mediate transcription repression.','cd01413','SIR2_Af2'),(970421,'gnl|CDD|239150',_binary 'Macro domain, a high-affinity ADP-ribose binding module found in a variety of proteins as a stand-alone domain or in combination with other domains like in histone macroH2A and some PARPs (poly ADP-ribose polymerases). Some macro domains recognize poly ADP-ribose as a ligand. Previously identified as displaying an Appr-1\'-p (ADP-ribose-1\'-monophosphate) processing activity, the macro domain may play roles in distinct ADP-ribose pathways, such as the ADP-ribosylation of proteins, an important post-translational modification which occurs in DNA repair, transcription, chromatin biology, and long-term memory formation, among other processes.','cd02749','Macro'),(970422,'gnl|CDD|240405',_binary 'NAD-dependent deacetylase; Provisional.','PTZ00408','PTZ00408'),(970423,'gnl|CDD|280333',_binary 'Sir2 family.  This region is characteristic of Silent information regulator 2 (Sir2) proteins, or sirtuins. These are protein deacetylases that depend on nicotine adenine dinucleotide (NAD). They are found in many subcellular locations, including the nucleus, cytoplasm and mitochondria. Eukaryotic forms play in important role in the regulation of transcriptional repression. Moreover, they are involved in microtubule organisation and DNA damage repair processes.i.','pfam02146','SIR2'),(970424,'gnl|CDD|238701',_binary 'SIRT7: Eukaryotic and prokaryotic group (class4) which includes human sirtuin SIRT6, SIRT7, and several bacterial homologs; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span.','cd01410','SIRT7'),(970425,'gnl|CDD|238702',_binary 'SIR2H: Uncharacterized prokaryotic Sir2 homologs from several gram positive bacterial species and Fusobacteria; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span.','cd01411','SIR2H'),(970426,'gnl|CDD|238700',_binary 'SIRT4: Eukaryotic and prokaryotic group (class2) which includes human sirtuin SIRT4 and several bacterial homologs; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, metabolic enzymes, and life span.','cd01409','SIRT4'),(970427,'gnl|CDD|172627',_binary 'NAD-dependent deacetylase; Provisional.','PRK14138','PRK14138'),(970428,'gnl|CDD|235415',_binary 'NAD-dependent deacetylase; Provisional.','PRK05333','PRK05333'),(970429,'gnl|CDD|238699',_binary 'SIRT1: Eukaryotic group (class1) which includes human sirtuins SIRT1-3 and yeast Hst1-4; and are members of the SIR2 family of proteins, silent information regulator 2 (Sir2) enzymes which catalyze NAD+-dependent protein/histone deacetylation. Sir2 proteins have been shown to regulate gene silencing, DNA repair, and life span. The most-studied function, gene silencing, involves the inactivation of chromosome domains containing key regulatory genes by packaging them into a specialized chromatin structure that is inaccessible to DNA-binding proteins. The nuclear SIRT1 has been shown to target the p53 tumor suppressor protein for deacetylation to suppress DNA damage, and the cytoplasmic SIRT2 homolog has been shown to target alpha-tubulin for deacetylation for the maintenance of cell integrity.','cd01408','SIRT1'),(970430,'gnl|CDD|173599',_binary 'Sir2 (Silent Information Regulator) protein; Provisional.','PTZ00409','PTZ00409'),(970431,'gnl|CDD|185600',_binary 'NAD-dependent SIR2; Provisional.','PTZ00410','PTZ00410'),(970432,'gnl|CDD|237603',_binary 'HSP90 family protein; Provisional.','PRK14083','PRK14083'),(970435,'gnl|CDD|223515',_binary 'Glycosyltransferase [Cell envelope biogenesis, outer membrane].','COG0438','RfaG'),(970436,'gnl|CDD|99981',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. mtfB (mannosyltransferase B) in E. coli has been shown to direct the growth of the O9-specific polysaccharide chain. It transfers two mannoses into the position 3 of the previously synthesized polysaccharide.','cd03809','GT1_mtfB_like'),(970437,'gnl|CDD|99974',_binary 'This family is most closely related to the GT1 family of glycosyltransferases and named after YqgM in Bacillus licheniformis about which little is known. Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found mainly in certain bacteria and archaea.','cd03801','GT1_YqgM_like'),(970438,'gnl|CDD|274943',_binary 'glycosyltransferase, MSMEG_0565 family.  A conserved gene cluster found sporadically from Actinobacteria to Proteobacteria to Cyanobacteria features a radical SAM protein, an N-acetyltransferase, an oxidoreductase, and two additional proteins whose functional classes are unclear. The metabolic role of the cluster is probably biosynthetic. This glycosyltransferase, named from member MSMEG_0565 from Mycobacterium smegmatis, occurs in most but not all instances of the cluster. [Unknown function, Enzymes of unknown specificity].','TIGR04047','MSMEG_0565_glyc'),(970439,'gnl|CDD|278935',_binary 'Glycosyl transferases group 1.  Mutations in this domain of PIGA lead to disease (Paroxysmal Nocturnal haemoglobinuria). Members of this family transfer activated sugars to a variety of substrates, including glycogen, Fructose-6-phosphate and lipopolysaccharides. Members of this family transfer UDP, ADP, GDP or CMP linked sugars. The eukaryotic glycogen synthases may be distant members of this family.','pfam00534','Glycos_transf_1'),(970440,'gnl|CDD|164861',_binary 'putative group 1 glycosyl transferase.','PHA01630','PHA01630'),(970441,'gnl|CDD|99993',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. ExpE7 in Sinorhizobium meliloti has been shown to be involved in the biosynthesis of galactoglucans (exopolysaccharide II).','cd03823','GT1_ExpE7_like'),(970442,'gnl|CDD|316234',_binary 'Glycosyl transferases group 1.','pfam13692','Glyco_trans_1_4'),(970443,'gnl|CDD|99959',_binary 'Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. The structures of the formed glycoconjugates are extremely diverse, reflecting a wide range of biological functions. The members of this family share a common GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility.','cd01635','Glycosyltransferase_GTB_t'),(970444,'gnl|CDD|99980',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. cap1E in Streptococcus pneumoniae is required for the synthesis of type 1 capsular polysaccharides.','cd03808','GT1_cap1E_like'),(970445,'gnl|CDD|99973',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. The sucrose-phosphate synthases in this family may be unique to plants and photosynthetic bacteria. This enzyme catalyzes the synthesis of sucrose 6-phosphate from fructose 6-phosphate and uridine 5\'-diphosphate-glucose, a key regulatory step of sucrose metabolism. The activity of this enzyme is regulated by phosphorylation and moderated by the concentration of various metabolites and light.','cd03800','GT1_Sucrose_synthase'),(970446,'gnl|CDD|99991',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. Bme6 in Brucella melitensis has been shown to be involved in the biosynthesis of a polysaccharide.','cd03821','GT1_Bme6_like'),(970450,'gnl|CDD|239259',_binary 'Protein Disulfide Isomerase (PDIa) family, redox active TRX domains; composed of eukaryotic proteins involved in oxidative protein folding in the endoplasmic reticulum (ER) by acting as catalysts and folding assistants. Members of this family include PDI and PDI-related proteins like ERp72, ERp57 (or ERp60), ERp44, P5, PDIR, ERp46 and the transmembrane PDIs. PDI, ERp57, ERp72, P5, PDIR and ERp46 are all oxidases, catalyzing the formation of disulfide bonds of newly synthesized polypeptides in the ER. They also exhibit reductase activity in acting as isomerases to correct any non-native disulfide bonds, as well as chaperone activity to prevent protein aggregation and facilitate the folding of newly synthesized proteins. These proteins usually contain multiple copies of a redox active TRX (a) domain containing a CXXC motif, and may also contain one or more redox inactive TRX-like (b) domains. Only one a domain is required for the oxidase function but multiple copies are necessary for the isomerase function. The different types of PDIs may show different substrate specificities and tissue-specific expression, or may be induced by stress. PDIs are in their reduced form at steady state and are oxidized to the active form by Ero1, which is localized in the ER through ERp44. Some members of this family also contain a DnaJ domain in addition to the redox active a domains; examples are ERdj5 and Pfj2. Also included in the family is the redox inactive N-terminal TRX-like domain of ERp29.','cd02961','PDI_a_family'),(970451,'gnl|CDD|240266',_binary 'disulphide isomerase; Provisional.','PTZ00102','PTZ00102'),(970452,'gnl|CDD|173347',_binary 'thioredoxin; Provisional.','PTZ00051','PTZ00051'),(970454,'gnl|CDD|273454',_binary 'protein disulfide-isomerase domain.  This model describes a domain of eukaryotic protein disulfide isomerases, generally found in two copies. The high cutoff for total score reflects the expectation of finding both copies. The domain is similar to thioredoxin but the redox-active disulfide region motif is APWCGHCK. [Protein fate, Protein folding and stabilization].','TIGR01126','Putative_protein_disulfid'),(970455,'gnl|CDD|239299',_binary 'PDIa family, P5 subfamily; composed of eukaryotic proteins similar to human P5, a PDI-related protein with a domain structure of aa\'b (where a and a\' are redox active TRX domains and b is a redox inactive TRX-like domain). Like PDI, P5 is located in the endoplasmic reticulum (ER) and displays both isomerase and chaperone activities, which are independent of each other. Compared to PDI, the isomerase and chaperone activities of P5 are lower. The first cysteine in the CXXC motif of both redox active domains in P5 is necessary for isomerase activity. The P5 gene was first isolated as an amplified gene from a hydroxyurea-resistant hamster cell line. The zebrafish P5 homolog has been implicated to play a critical role in establishing left/right asymmetries in the embryonic midline. Some members of this subfamily are P5-like proteins containing only one redox active TRX domain.','cd03001','PDI_a_P5'),(970456,'gnl|CDD|273457',_binary 'protein disulfide isomerase, eukaryotic.  This model represents eukaryotic protein disulfide isomerases retained in the endoplasmic reticulum (ER) and closely related forms. Some members have been assigned alternative or additional functions such as prolyl 4-hydroxylase and dolichyl-diphosphooligosaccharide-protein glycotransferase. Members of this family have at least two protein-disulfide domains, each similar to thioredoxin but with the redox-active disulfide in the motif PWCGHCK, and an ER retention signal at the extreme C-terminus (KDEL, HDEL, and similar motifs).','TIGR01130','Protein_disulfide-isomera'),(970459,'gnl|CDD|309753',_binary 'OST3 / OST6 family, transporter family.  The proteins in this family are part of a complex of eight ER proteins that transfers core oligosaccharide from dolichol carrier to Asn-X-Ser/Thr motifs. This family includes both OST3 and OST6, each of which contains four predicted transmembrane helices. Disruption of OST3 and OST6 leads to a defect in the assembly of the complex. Hence, the function of these genes seems to be essential for recruiting a fully active complex necessary for efficient N-glycosylation. These proteins are also thought to be novel Mg2+ transporters.','pfam04756','OST3_OST6'),(970460,'gnl|CDD|181812',_binary 'thioredoxin; Provisional.','PRK09381','trxA'),(970461,'gnl|CDD|239296',_binary 'PDIa family, endoplasmic reticulum protein 38 (ERp38) subfamily; composed of proteins similar to the P5-like protein first isolated from alfalfa, which contains two redox active TRX (a) domains at the N-terminus, like human P5, and a C-terminal domain with homology to the C-terminal domain of ERp29, unlike human P5. The cDNA clone of this protein (named G1) was isolated from an alfalfa cDNA library by screening with human protein disulfide isomerase (PDI) cDNA. The G1 protein is constitutively expressed in all major organs of the plant and its expression is induced by treatment with tunicamycin, indicating that it may be a glucose-regulated protein. The G1 homolog in the eukaryotic social amoeba Dictyostelium discoideum is also described as a P5-like protein, which is located in the endoplasmic reticulum (ER) despite the absence of an ER-retrieval signal. G1 homologs from Aspergillus niger and Neurospora crassa have also been characterized, and are named TIGA and ERp38, respectively. Also included in the alignment is an atypical PDI from Leishmania donovani containing a single a domain, and the C-terminal a domain of a P5-like protein from Entamoeba histolytica.','cd02998','PDI_a_ERp38'),(970462,'gnl|CDD|225660',_binary 'Thioredoxin domain-containing protein [Posttranslational modification, protein turnover, chaperones].','COG3118','COG3118'),(970463,'gnl|CDD|239295',_binary 'PDIa family, PDIR subfamily; composed of proteins similar to human PDIR (for Protein Disulfide Isomerase Related). PDIR is composed of three redox active TRX (a) domains and an N-terminal redox inactive TRX-like (b) domain. Similar to PDI, it is involved in oxidative protein folding in the endoplasmic reticulum (ER) through its isomerase and chaperone activities. These activities are lower compared to PDI, probably due to PDIR acting only on a subset of proteins. PDIR is preferentially expressed in cells actively secreting proteins and its expression is induced by stress. Similar to PDI, the isomerase and chaperone activities of PDIR are independent; CXXC mutants lacking isomerase activity retain chaperone activity.','cd02997','PDI_a_PDIR'),(970466,'gnl|CDD|239254',_binary 'ybbN protein family; ybbN is a hypothetical protein containing a redox-inactive TRX-like domain. Its gene has been sequenced from several gammaproteobacteria and actinobacteria.','cd02956','ybbN'),(970467,'gnl|CDD|239300',_binary 'PDI family, MPD1-like subfamily; composed of eukaryotic proteins similar to Saccharomyces cerevisiae MPD1 protein, which contains a single redox active TRX domain located at the N-terminus, and an ER retention signal at the C-terminus indicative of an ER-resident protein. MPD1 has been shown to suppress the maturation defect of carboxypeptidase Y caused by deletion of the yeast PDI1 gene. Other characterized members of this subfamily include the Aspergillus niger prpA protein and Giardia PDI-1. PrpA is non-essential to strain viability, however, its transcript level is induced by heterologous protein expression suggesting a possible role in oxidative protein folding during high protein production. Giardia PDI-1 has the ability to refold scrambled RNase and exhibits transglutaminase activity.','cd03002','PDI_a_MPD1_like'),(970468,'gnl|CDD|239303',_binary 'PDIa family, endoplasmic reticulum protein 46 (ERp46) subfamily; ERp46 is an ER-resident protein containing three redox active TRX domains. Yeast complementation studies show that ERp46 can substitute for protein disulfide isomerase (PDI) function in vivo. It has been detected in many tissues, however, transcript and protein levels do not correlate in all tissues, suggesting regulation at a posttranscriptional level. An identical protein, named endoPDI, has been identified as an endothelial PDI that is highly expressed in the endothelium of tumors and hypoxic lesions. It has a protective effect on cells exposed to hypoxia.','cd03005','PDI_a_ERp46'),(970480,'gnl|CDD|99971',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. wlbH in Bordetella parapertussis has been shown to be required for the biosynthesis of a trisaccharide that, when attached to the B. pertussis lipopolysaccharide (LPS) core (band B), generates band A LPS.','cd03798','GT1_wlbH_like'),(970481,'gnl|CDD|132490',_binary 'D-inositol-3-phosphate glycosyltransferase.  Members of this protein family, found exclusively in the Actinobacteria, are MshA, the glycosyltransferase of mycothiol biosynthesis. Mycothiol replaces glutathione in these species.','TIGR03449','mycothiol_MshA'),(970482,'gnl|CDD|215469',_binary 'UDP-sulfoquinovose:DAG sulfoquinovosyltransferase.','PLN02871','PLN02871'),(970483,'gnl|CDD|271319',_binary 'Escherichia coli swarming motility protein YbiA and related proteins.  This family of uncharacterized domains was initially classified as Domain of Unknown Function (DUF1768). It contains members such as the E. coli swarming motility protein YbiA. Mutations in YbiA cause defects in Escherichia coli swarming, but not necessarily in motility. More recently, this family has been predicted to be involved in NAD-utilizing pathways, likely to act on ADP-ribose derivatives, and has been named the NADAR (NAD and ADP-ribose) superfamily.','cd15457','NADAR'),(970484,'gnl|CDD|312300',_binary 'Domain of unknown function (DUF1768).  This is a domain of unknown function. It is alpha helical in structure. The GO annotation for this protein suggests it is involved in nematode larval development and has a positive regulation on growth rate.','pfam08719','DUF1768'),(970485,'gnl|CDD|225776',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3236','COG3236'),(970486,'gnl|CDD|274144',_binary 'conserved hypothetical protein, ribA/ribD-fused.  This model describes a sequence region that occurs in at least three different polypeptide contexts. It is found fused to GTP cyclohydrolase II, the RibA of riboflavin biosynthesis (TIGR00505), as in Vibrio vulnificus. It is found fused to riboflavin biosynthesis protein RibD (TIGR00326) in rice and Arabidopsis. It occurs as a standalone protein in a number of bacterial species in varied contexts, including single gene operons and bacteriophage genomes. The member from E. coli currently is named YbiA. The function(s) of members of this family is unknown.','TIGR02464','Swarming_motility_protein'),(970488,'gnl|CDD|309706',_binary 'Streptomyces sporulation and cell division protein, SsgA.  The precise function of SsgA is unknown. It has been found to be essential for spore formation, and to stimulate cell division.','pfam04686','SsgA'),(970498,'gnl|CDD|224306',_binary 'FOG: LysM repeat [Cell envelope biogenesis, outer membrane].','COG1388','LytE'),(970499,'gnl|CDD|180536',_binary 'autolysin; Reviewed.','PRK06347','PRK06347'),(970504,'gnl|CDD|182727',_binary 'membrane-bound lytic murein transglycosylase D; Provisional.','PRK10783','mltD'),(970518,'gnl|CDD|273680',_binary 'phage portal protein, PBSX family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa. This family shows clear homology to TIGR01537. The alignment for this group was trimmed of poorly alignable N-terminal sequence of about 50 residues and of C-terminal regions present in some but not all members of up 180 residues. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01540','Phage-like_element_PBSX_p'),(970520,'gnl|CDD|227805',_binary 'Bacteriophage capsid portal protein [General function prediction only].','COG5518','COG5518'),(970521,'gnl|CDD|222860',_binary 'portal vertex protein; Provisional.','PHA02536','Q'),(970524,'gnl|CDD|317012',_binary 'Putative phage serine protease XkdF.  This domain is largely found on phage proteins. In a number of cases the domain is associated with a SAM-dependent methyltransferase. Members are serine peptidases.','pfam14550','Peptidase_S78_2'),(970525,'gnl|CDD|310797',_binary 'Domain of Unknown Function (DUF1080).  This family has structural similarity to an endo-1,3-1,4-beta glucanase belonging to glycoside hydrolase family 16. However, the structure surrounding the active site differs from that of the endo-1,3-1,4-beta glucanase.','pfam06439','DUF1080'),(970573,'gnl|CDD|234802',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase; Provisional.','PRK00601','dut'),(970574,'gnl|CDD|273149',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase (dut).  The main function of these proteins is in maintaining the levels of dUTP in the cell to prevent dUTP incorporation into DNA during DNA replication. Pol proteins in viruses are very similar to this protein family. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). Changed role from 132 to 123. RTD [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR00576','Deoxyuridine_5\'-triphosph'),(970575,'gnl|CDD|223827',_binary 'dUTPase [Nucleotide transport and metabolism].','COG0756','Dut'),(970576,'gnl|CDD|279082',_binary 'dUTPase.  dUTPase hydrolyses dUTP to dUMP and pyrophosphate.','pfam00692','dUTPase'),(970577,'gnl|CDD|143638',_binary 'Trimeric dUTP diphosphatases.  Trimeric dUTP diphosphatases, or dUTPases, are the most common family of dUTPase, found in bacteria, eukaryotes, and archaea. They catalyze the hydrolysis of the dUTP-Mg complex (dUTP-Mg) into dUMP and pyrophosphate. This reaction is crucial for the preservation of chromosomal integrity as it removes dUTP and therefore reduces the cellular dUTP/dTTP ratio, and prevents dUTP from being incorporated into DNA.  It also provides dUMP as the precursor for dTTP synthesis via the thymidylate synthase pathway. dUTPases are homotrimeric, except some monomeric viral dUTPases, which have been shown to mimic a trimer. Active sites are located at the subunit interface.','cd07557','trimeric_dUTPase'),(970578,'gnl|CDD|215302',_binary 'dUTP pyrophosphatase.','PLN02547','PLN02547'),(970579,'gnl|CDD|165376',_binary 'dUTPase; Provisional.','PHA03094','PHA03094'),(970580,'gnl|CDD|165079',_binary 'ORF007 dUTPase; Provisional.','PHA02703','PHA02703'),(970581,'gnl|CDD|240288',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase; Provisional.','PTZ00143','PTZ00143'),(970582,'gnl|CDD|223789',_binary 'Deoxycytidine deaminase [Nucleotide transport and metabolism].','COG0717','Dcd'),(970583,'gnl|CDD|274062',_binary 'deoxycytidine triphosphate deaminase.  Members of this family include the Escherichia coli monofunctional deoxycytidine triphosphate deaminase (dCTP deaminase) and a Methanocaldococcus jannaschii bifunctional dCTP deaminase (3.5.4.13)/dUTP diphosphatase (EC 3.6.1.23), which has the EC number 3.5.4.30 for the overall operation. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02274','Probable_deoxycytidine_tr'),(970584,'gnl|CDD|184417',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase; Provisional.','PRK13956','dut'),(970585,'gnl|CDD|179395',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase; Provisional.','PRK02253','PRK02253'),(970586,'gnl|CDD|234753',_binary 'deoxycytidine triphosphate deaminase; Reviewed.','PRK00416','dcd'),(970587,'gnl|CDD|165396',_binary 'dUTPase; Provisional.','PHA03124','PHA03124'),(970588,'gnl|CDD|222993',_binary 'dUTPase; Provisional.','PHA03127','PHA03127'),(970589,'gnl|CDD|107053',_binary '2\'-deoxyuridine 5\'-triphosphatase.','PHA01707','dut'),(970591,'gnl|CDD|222996',_binary 'dUTPase; Provisional.','PHA03131','PHA03131'),(970646,'gnl|CDD|307860',_binary 'Domain of unknown function DUF83.  This domain has no known function. The domain contains three conserved cysteines at its C terminus.','pfam01930','Cas_Cas4'),(970648,'gnl|CDD|164858',_binary 'CRISPR-associated Cas4-like protein.','PHA01622','PHA01622'),(970652,'gnl|CDD|197205',_binary 'Putative catalytic domain, repeat 1, of vertebrate phospholipases, PLD3, PLD4 and PLD5, viral envelope proteins K4 and p37, and similar proteins.  Putative catalytic domain, repeat 1, of vertebrate phospholipases D, PLD3, PLD4, and PLD5 (EC 3.1.4.4), viral envelope proteins (vaccinia virus proteins K4 and p37), and similar proteins. Most family members contain two copies of the HKD motifs (H-x-K-x(4)-D, where x represents any amino acid residue), and have been classified into the phospholipase D (PLD) superfamily. Proteins in this subfamily are associated with Golgi membranes, altering their lipid content by the conversion of phospholipids into phosphatidic acid, which is thought to be involved in the regulation of lipid movement. ADP ribosylation factor (ARF), a small guanosine triphosphate binding protein, might be required activity. The vaccinia virus p37 protein, encoded by the F13L gene, is also associated with Golgi membranes and is required for the envelopment and spread of the extracellular enveloped virus (EEV). The vaccinia virus protein K4, encoded by the HindIII K4L gene, remains to be characterized. Sequence analysis indicates that the vaccinia virus proteins K4 and p37 might have evolved from one or more captured eukaryotic genes involved in cellular lipid metabolism. Up to date, no catalytic activity of PLD3 has been shown. Furthermore, due to the lack of functional important histidine and lysine residues in the HKD motif, mammalian PLD5 has been characterized as an inactive PLD. The poxvirus p37 proteins may also lack PLD enzymatic activity, since they contain only one partially conserved HKD motif (N-x-K-x(4)-D).','cd09106','PLDc_vPLD3_4_5_like_1'),(970653,'gnl|CDD|311081',_binary 'Protein of unknown function (DUF1273).  This family consists of several hypothetical bacterial proteins of around 180 residues in length. The function of this family is unknown.','pfam06908','DUF1273'),(970654,'gnl|CDD|226880',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4474','COG4474'),(970655,'gnl|CDD|173922',_binary 'glycerol-3-phosphate cytidylyltransferase.  Glycerol-3-phosphate cytidylyltransferase,(CDP-glycerol pyrophosphorylase). Glycerol-3-phosphate cytidyltransferase acts in pathways of teichoic acid biosynthesis. Teichoic acids are substituted polymers, linked by phosphodiester bonds, of glycerol, ribitol, etc. An example is poly(glycerol phosphate), the major teichoic acid of the Bacillus subtilis cell wall. Most, but not all, species encoding proteins in this family are Gram-positive bacteria.  A closely related protein assigned a different function experimentally is a human ethanolamine-phosphate cytidylyltransferase.','cd02171','G3P_Cytidylyltransferase'),(970656,'gnl|CDD|223688',_binary 'Cytidylyltransferase [Cell envelope biogenesis, outer membrane / Lipid metabolism].','COG0615','TagD'),(970657,'gnl|CDD|130581',_binary 'glycerol-3-phosphate cytidylyltransferase.  This model describes glycerol-3-phosphate cytidyltransferase, also called CDP-glycerol pyrophosphorylase. A closely related protein assigned a different function experimentally is a human ethanolamine-phosphate cytidylyltransferase (EC 2.7.7.14). Glycerol-3-phosphate cytidyltransferase acts in pathways of teichoic acid biosynthesis. Teichoic acids are substituted polymers, linked by phosphodiester bonds, of glycerol, ribitol, etc. An example is poly(glycerol phosphate), the major teichoic acid of the Bacillus subtilis cell wall. Most but not all species encoding proteins in this family are Gram-positive bacteria. [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan].','TIGR01518','Glycerol-3-phosphate_cyti'),(970658,'gnl|CDD|173921',_binary 'cytidylyltransferase.  The cytidylyltransferase family includes cholinephosphate cytidylyltransferase (CCT), glycerol-3-phosphate cytidylyltransferase, RafE and  phosphoethanolamine cytidylyltransferase (ECT). All enzymes catalyze the transfer of a cytidylyl group from CTP to various substrates.','cd02170','cytidylyltransferase'),(970659,'gnl|CDD|173925',_binary 'CTP:phosphocholine cytidylyltransferase.  CTP:phosphocholine cytidylyltransferase (CCT) catalyzes the condensation of CTP and phosphocholine to form CDP-choline as the rate-limiting and regulatory step in the CDP-choline pathway. CCT is unique in that its enzymatic activity is regulated by the extent of its association with membrane structures. A current model posts that the elastic stress of the bilayer curvature is sensed by CCT and this governs the degree of membrane association, thus providing a mechanism for both positive and negative regulation of activity.','cd02174','CCT'),(970660,'gnl|CDD|307563',_binary 'Cytidylyltransferase-like.  This family includes: Cholinephosphate cytidylyltransferase; glycerol-3-phosphate cytidylyltransferase. It also includes putative adenylyltransferases, and FAD synthases.','pfam01467','CTP_transf_like'),(970661,'gnl|CDD|173924',_binary 'CTP:phosphoethanolamine cytidylyltransferase (ECT).  CTP:phosphoethanolamine cytidylyltransferase (ECT) catalyzes the conversion of phosphoethanolamine to CDP-ethanolamine as part of the CDP-ethanolamine biosynthesis pathway.  ECT expression in hepatocytes is localized predominantly to areas of the cytoplasm that are rich in rough endoplasmic reticulum. Several ECTs, including yeast and human ECT, have large repetitive sequences located within their N- and C-termini.','cd02173','ECT'),(970662,'gnl|CDD|131254',_binary 'rfaE bifunctional protein, domain II.  RfaE is a protein involved in the biosynthesis of ADP-L-glycero-D-manno-heptose, a precursor for LPS inner core biosynthesis. RfaE is a bifunctional protein in E. coli, and separate proteins in some other genome. Domain I (TIGR02198) is suggested to act in D-glycero-D-manno-heptose 1-phosphate biosynthesis, while domain II (this family) adds ADP to yield ADP-D-glycero-D-manno-heptose. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR02199','rfaE_dom_II'),(970663,'gnl|CDD|225425',_binary 'ADP-heptose synthase, bifunctional sugar kinase/adenylyltransferase [Cell envelope biogenesis, outer membrane].','COG2870','RfaE'),(970664,'gnl|CDD|272920',_binary 'cytidyltransferase-like domain.  Protein families that contain at least one copy of this domain include citrate lyase ligase, pantoate-beta-alanine ligase, glycerol-3-phosphate cytidyltransferase, ADP-heptose synthase, phosphocholine cytidylyltransferase, lipopolysaccharide core biosynthesis protein KdtB, the bifunctional protein NadR, and a number whose function is unknown. Many of these proteins are known to use CTP or ATP and release pyrophosphate.','TIGR00125','Phosphopantetheine_adenyl'),(970665,'gnl|CDD|140329',_binary 'ethanolamine-phosphate cytidylyltransferase; Provisional.','PTZ00308','PTZ00308'),(970667,'gnl|CDD|183085',_binary 'bifunctional heptose 7-phosphate kinase/heptose 1-phosphate adenyltransferase; Provisional.','PRK11316','PRK11316'),(970668,'gnl|CDD|173923',_binary 'N-terminal domain of RfaE.  RfaE is a protein involved in the biosynthesis of ADP-L-glycero-D-manno-heptose, a precursor for LPS inner core biosynthesis. RfaE is a bifunctional protein in Escherichia coli, and separate proteins in other organisms. Domain I  is suggested to act in D-glycero-D-manno-heptose 1-phosphate biosynthesis, while domain II (this family) adds ADP to yield ADP-D-glycero-D-manno-heptose .','cd02172','RfaE_N'),(970669,'gnl|CDD|215227',_binary 'ethanolamine-phosphate cytidylyltransferase.','PLN02406','PLN02406'),(970671,'gnl|CDD|185678',_binary 'Cytidylyltransferase-like domain.  Cytidylyltransferase-like domain. Many of these proteins are known to use CTP or ATP and release pyrophosphate. Protein families that contain at least one copy of this domain include citrate lyase ligase, pantoate-beta-alanine ligase, glycerol-3-phosphate cytidyltransferase, ADP-heptose synthase, phosphocholine cytidylyltransferase, lipopolysaccharide core biosynthesis protein KdtB, the bifunctional protein NadR, and a number whose function is unknown.','cd02039','cytidylyltransferase_like'),(970672,'gnl|CDD|234834',_binary 'phosphopantetheine adenylyltransferase; Provisional.','PRK00777','PRK00777'),(970673,'gnl|CDD|215229',_binary 'choline-phosphate cytidylyltransferase.','PLN02413','PLN02413'),(970674,'gnl|CDD|223950',_binary 'Predicted nucleotidyltransferase [General function prediction only].','COG1019','COG1019'),(970675,'gnl|CDD|173915',_binary 'phosphopantetheine adenylyltransferase domain of eukaryotic and archaeal bifunctional enzymes.  The PPAT domain of the bifunctional enzyme with PPAT and DPCK functions. The final two steps of the CoA biosynthesis pathway are catalyzed by phosphopantetheine adenylyltransferase (PPAT) and dephospho-CoA (dPCoA) kinase (DPCK). The PPAT reaction involves the reversible adenylation of 4\'-phosphopantetheine to form 3\'-dPCoA and PPi, and DPCK catalyses phosphorylation of the 3\'-hydroxy group of the ribose moiety of dPCoA.  In eukaryotes the two enzymes are part of a large multienzyme complex . Studies in Corynebacterium ammoniagenes suggested that separate enzymes were present, and this was confirmed through identification of the bacterial PPAT/CoAD.','cd02164','PPAT_CoAS'),(970679,'gnl|CDD|235864',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK06807','PRK06807'),(970683,'gnl|CDD|236123',_binary 'hypothetical protein; Validated.','PRK07883','PRK07883'),(970689,'gnl|CDD|169400',_binary 'hypothetical protein; Provisional.','PRK08356','PRK08356'),(970696,'gnl|CDD|197903',_binary 'Putative regulatory protein.  CxxC_CXXC_SSSS represents a region of about 41 amino acids found in a number of small proteins in a wide range of bacteria. The region usually begins with the initiator Met and contains two CxxC motifs separated by 17 amino acids. One protein in this entry has been noted as a putative regulatory protein, designated FmdB. Most proteins in this entry have a C-terminal region containing highly degenerate sequence.','smart00834','CxxC_CXXC_SSSS'),(970697,'gnl|CDD|274226',_binary 'putative regulatory protein, FmdB family.  This model represents a region of about 50 amino acids found in a number of small proteins in a wide range of bacteria. The region begins usually with the initiator Met and contains two CxxC motifs separated by 17 amino acids. One member of this family is has been noted as a putative regulatory protein, designated FmdB (SP:Q50229, ). Most members of this family have a C-terminal region containing highly degenerate sequence, such as SSTSESTKSSGSSGSSGSSESKASGSTEKSTSSTTAAAAV in Mycobacterium tuberculosis and VAVGGSAPAPSPAPRAGGGGGGCCGGGCCG in Streptomyces avermitilis. These low complexity regions, which are not included in the model, resemble low-complexity C-terminal regions of some heterocycle-containing bacteriocin precursors. [Regulatory functions, DNA interactions].','TIGR02605','CxxC_CxxC_SSSS'),(970698,'gnl|CDD|225210',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG2331','COG2331'),(970753,'gnl|CDD|235745',_binary 'sulfite reductase; Provisional.','PRK06214','PRK06214'),(970804,'gnl|CDD|311022',_binary 'Phage minor structural protein GP20.  This family consists of several phage minor structural protein GP20 sequences of around 180 residues in length. The function of this family is unknown.','pfam06810','Phage_GP20'),(970805,'gnl|CDD|307935',_binary 'SAP domain.  The SAP (after SAF-A/B, Acinus and PIAS) motif is a putative DNA/RNA binding domain found in diverse nuclear and cytoplasmic proteins.','pfam02037','SAP'),(970806,'gnl|CDD|128789',_binary 'Putative DNA-binding (bihelical) motif predicted to be involved in chromosomal organisation.','smart00513','SAP'),(970810,'gnl|CDD|222888',_binary 'baseplate wedge subunit and tail pin; Provisional.','PHA02582','10'),(970885,'gnl|CDD|143386',_binary 'Nucleotidyltransferase (NT) domain of family X DNA Polymerases.  X family polymerases fill in short gaps during DNA repair. They are relatively inaccurate enzymes and play roles in base excision repair, in non-homologous end joining (NHEJ) which acts mainly to repair damage due to ionizing radiation, and in V(D)J recombination. This family includes eukaryotic Pol beta, Pol lambda, Pol mu, and terminal deoxyribonucleotidyl transferase (TdT). Pol beta and Pol lambda are primarily DNA template-dependent polymerases. TdT is a DNA template-independent polymerase. Pol mu has both template dependent and template independent activities. This subgroup belongs to the Pol beta-like NT superfamily. In the majority of enzymes in this superfamily, two carboxylates, Dx[D/E], together with a third more distal carboxylate, coordinate two divalent metal cations involved in a two-metal ion mechanism of nucleotide addition. These three carboxylate residues are fairly well conserved in this family.','cd00141','NT_POLXc'),(971085,'gnl|CDD|308443',_binary 'Glutathione S-transferase, N-terminal domain.  Function: conjugation of reduced glutathione to a variety of targets. Also included in the alignment, but not GSTs: S-crystallins from squid (similarity to GST previously noted); eukaryotic elongation factors 1-gamma (not known to have GST activity and similarity not previously recognized); HSP26 family of stress-related proteins including auxin-regulated proteins in plants and stringent starvation proteins in E. coli (not known to have GST activity and similarity not previously recognized). The glutathione molecule binds in a cleft between the N- and C-terminal domains - the catalytically important residues are proposed to reside in the N-terminal domain.','pfam02798','GST_N'),(971118,'gnl|CDD|317950',_binary 'Papain fold toxin 1, glutamine deamidase.  A papain fold toxin domain found in bacterial polymorphic toxin systems. In these systems they might function either as a releasing peptidase or toxin. In Shigella flexneri, UniProtKB:Q8VSD5, this protein is expressed from a plasmid, and delivered into the host via the type III secretion system where it deamidates the glutamine residue at position 100 in ubiquitin-activating enzyme E2, UBC13, to a glutamic acid residue. Invasion of host cells by pathogens normally invokes an acute inflammatory response through activating the TRAF6-mediated signalling pathway. UBC13 helps to activate TRAF6. Thus deamidation of UBC13 results in the dampening of the inflammatory response. The key glutaminase deamidase activity is mediated by a cys-his-glu triad, present in all members of the family.','pfam15644','Gln_amidase'),(971128,'gnl|CDD|273636',_binary 'group I intron endonuclease.  This model represents one subfamily of endonucleases containing the endo/excinuclease amino terminal domain, pfam01541 at its amino end. A distinct subfamily includes excinuclease abc subunit c (uvrC). Members of pfam01541 are often termed GIY-YIG endonucleases after conserved motifs near the amino end. This subfamily in this model is found in open reading frames of group I introns in both phage and mitochondria. The closely related endonucleases of phage T4: segA, segB, segC, segD and segE, score below the trusted cutoff for the family.','TIGR01453','unnamed_protein_product_p'),(971218,'gnl|CDD|227660',_binary 'Phage-related terminase [General function prediction only].','COG5362','COG5362'),(971220,'gnl|CDD|130691',_binary 'phage uncharacterized protein (putative large terminase), C-terminal domain.  This model represents the C-terminal region of a set of phage proteins typically about 400-500 amino acids in length, although some members are considerably shorter. An article on Methanobacterium phage Psi-M2 ( calls the member from that phage, ORF9, a putative large terminase subunit, and ORF8 a candidate terminase small subunit. Most proteins in this family have an apparent P-loop nucleotide-binding sequence toward the N-terminus. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01630','psiM2_ORF9'),(971221,'gnl|CDD|310893',_binary 'Prophage endopeptidase tail.  This family is of prophage tail proteins that are probably acting as endopeptidases.','pfam06605','Prophage_tail'),(971222,'gnl|CDD|273743',_binary 'phage minor structural protein, N-terminal region.  This model represents the conserved N-terminal region, typically from about residue 25 to about residue 350, of a family of uncharacterized phage proteins 500 to 1700 residues in length. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01665','structural_protein'),(971223,'gnl|CDD|227263',_binary 'Phage-related protein [Function unknown].','COG4926','PblB'),(971227,'gnl|CDD|119353',_binary 'Cortical fragment-lytic enzyme (CFLE) is a peptidoglycan hydrolase involved in  bacterial endospore germination.  CFLE is expressed as an inactive preprotein (called SleB) in the forespore compartment of sporulating cells.  SleB translocates across the forespore inner membrane and is deposited as a mature enzyme in the cortex layer of the spore.  As part of a sensory mechanism capable of initiating germination, CFLE degrades a spore-specific peptidoglycan constituent called muramic-acid delta-lactam that comprises the outer cortex.  CFLE has a C-terminal glycosyl hydrolase family 18 (GH18) catalytic domain as well as two N-terminal LysM peptidoglycan-binding domains.  In addition to SleB, this family includes YaaH, YdhD, and YvbX from Bacillus subtilis.','cd02874','GH18_CFLE_spore_hydrolase'),(971228,'gnl|CDD|226376',_binary 'Predicted glycosyl hydrolase [General function prediction only].','COG3858','COG3858'),(971229,'gnl|CDD|307034',_binary 'Glycosyl hydrolases family 18.','pfam00704','Glyco_hydro_18'),(971230,'gnl|CDD|214753',_binary 'Glyco_18 domain.','smart00636','Glyco_18'),(971231,'gnl|CDD|119366',_binary 'GH18 domain of an uncharacterized family of bacterial proteins, which share a common three-domain architecture: an N-terminal glycosyl hydrolase family 18 (GH18) domain, a glycosyl transferase family 2 domain, and a C-terminal polysaccharide deacetylase domain.','cd06549','GH18_trifunctional'),(971232,'gnl|CDD|119365',_binary 'The GH18 (glycosyl hydrolases, family 18) type II chitinases hydrolyze chitin, an abundant polymer of N-acetylglucosamine and have been identified in bacteria, fungi, insects, plants, viruses, and protozoan parasites.  The structure of this domain is an eight-stranded alpha/beta barrel with a pronounced active-site cleft at the C-terminal end of the beta-barrel.','cd06548','GH18_chitinase'),(971233,'gnl|CDD|119355',_binary 'Stabilin-1 interacting chitinase-like protein (SI-CLP) is a eukaryotic chitinase-like protein of unknown function that interacts with the endocytic/sorting transmembrane receptor stabilin-1 and is secreted from the lysosome.  SI-CLP has a glycosyl hydrolase family 18 (GH18) domain but lacks a chitin-binding domain. The catalytic amino acids of the GH18 domain are not conserved in SI-CLP, similar to the chitolectins YKL-39, YKL-40, and YM1/2.  Human SI-CLP is sorted to late endosomes and secretory lysosomes in alternatively activated macrophages.','cd02876','GH18_SI-CLP'),(971234,'gnl|CDD|119362',_binary 'The Bacteroides thetaiotaomicron protein represented by pdb structure 3CO4 is an uncharacterized bacterial member of the family 18 glycosyl hydrolases with homologs found in Flavobacterium, Stigmatella, and Pseudomonas.','cd06545','GH18_3CO4_chitinase'),(971235,'gnl|CDD|225862',_binary 'Chitinase [Carbohydrate transport and metabolism].','COG3325','ChiA'),(971236,'gnl|CDD|119358',_binary 'The class V plant chitinases have a glycosyl hydrolase family 18 (GH18) domain, but lack the chitin-binding domain present in other GH18 enzymes.  The GH18 domain of the class V chitinases has endochitinase activity in some cases and no catalytic activity in others.  Included in this family is a lectin found in black locust (Robinia pseudoacacia) bark, which binds chitin but lacks chitinase activity.  Also included is a chitinase-related receptor-like kinase (CHRK1) from tobacco (Nicotiana tabacum), with an N-terminal GH18 domain and a C-terminal kinase domain, which is thought to be part of a plant signaling pathway.  The GH18 domain of CHRK1 is expressed extracellularly where it binds chitin but lacks chitinase activity.','cd02879','GH18_plant_chitinase_clas'),(971271,'gnl|CDD|227040',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4696','COG4696'),(971273,'gnl|CDD|315822',_binary 'zinc-ribbon domain.  This family consists of a single zinc ribbon domain, ie half of a pair as in family DZR. pfam12773.','pfam13240','zinc_ribbon_2'),(971465,'gnl|CDD|238748',_binary 'Micronemal proteins: The Toxoplasma lytic cycle begins when the parasite actively invades a target cell. In association with invasion, T. gondii sequentially discharges three sets of secretory organelles beginning with the micronemes, which contain adhesive proteins involved in parasite attachment to a host cell. Deployed as protein complexes, several micronemal proteins possess vertebrate-derived adhesive sequences that function in binding receptors. The VWA domain likely mediates the protein-protein interactions of these with their interacting partners.','cd01471','vWA_micronemal_protein'),(971466,'gnl|CDD|316077',_binary 'von Willebrand factor type A domain.','pfam13519','VWA_2'),(971546,'gnl|CDD|308527',_binary 'Bacteriophage lambda head decoration protein D.','pfam02924','HDPD'),(971551,'gnl|CDD|239103',_binary 'A sub-family of peptidase family C39. Peptidase family C39 mostly contains bacteriocin-processing endopeptidases from bacteria. The cysteine peptidases in family C39 cleave the \'double-glycine\' leader peptides from the precursors of various bacteriocins (mostly non-lantibiotic). The cleavage is mediated by the transporter as part of the secretion process. Bacteriocins are antibiotic proteins secreted by some species of bacteria that inhibit the growth of other bacterial species. The bacteriocin is synthesized as a precursor with an N-terminal leader peptide, and processing involves removal of the leader peptide by cleavage at a Gly-Gly bond, followed by translocation of the mature bacteriocin across the cytoplasmic membrane. Most endopeptidases of family C39 are N-terminal domains in larger proteins (ABC transporters) that serve both functions. The proposed protease active site is conserved in this sub-family of proteins with a single peptidase domain, which are lacking the nucleotide-binding transporter signature.','cd02423','Peptidase_C39G'),(971561,'gnl|CDD|293882',_binary 'OCRE domain found in RNA-binding protein 6 (RBM6) and similar proteins.  RBM6, also called lung cancer antigen NY-LU-12, or protein G16, or RNA-binding protein DEF-3, has been predicted to be a nuclear factor based on its nuclear localization signal. It shows high sequence similarity to RNA-binding protein 5 (RBM5 or LUCA15 or NY-REN-9). Both specifically binds poly(G) RNA. RBM6 contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), an OCtamer REpeat (OCRE) domain, two C2H2-type zinc fingers, a nuclear localization signal, and a G-patch/D111 domain. In contrast to RBM5, RBM6 has an additional unique domain, the POZ (poxvirus and zinc finger) domain, which may be involved in protein-protein interactions and inhibit binding of target sequences by zinc fingers.','cd16163','OCRE_RBM6'),(971563,'gnl|CDD|224636',_binary 'Exonuclease VII small subunit [DNA replication, recombination, and repair].','COG1722','XseB'),(971564,'gnl|CDD|179195',_binary 'exodeoxyribonuclease VII small subunit; Provisional.','PRK00977','PRK00977'),(971565,'gnl|CDD|273536',_binary 'exodeoxyribonuclease VII, small subunit.  This protein is the small subunit for exodeoxyribonuclease VII. Exodeoxyribonuclease VII is made of a complex of four small subunits to one large subunit. The complex degrades single-stranded DNA into large acid-insoluble oligonucleotides. These nucleotides are then degraded further into acid-soluble oligonucleotides. [DNA metabolism, Degradation of DNA].','TIGR01280','xseB'),(971566,'gnl|CDD|308303',_binary 'Exonuclease VII small subunit.  This family consist of exonuclease VII, small subunit EC:3.1.11.6 This enzyme catalyzes exonucleolytic cleavage in either 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. This exonuclease VII enzyme is composed of one large subunit and 4 small ones.','pfam02609','Exonuc_VII_S'),(971567,'gnl|CDD|172559',_binary 'exodeoxyribonuclease VII small subunit; Provisional.','PRK14069','PRK14069'),(971568,'gnl|CDD|172554',_binary 'exodeoxyribonuclease VII small subunit; Provisional.','PRK14064','PRK14064'),(971571,'gnl|CDD|224486',_binary 'Exonuclease VII, large subunit [DNA replication, recombination, and repair].','COG1570','XseA'),(971572,'gnl|CDD|308296',_binary 'Exonuclease VII, large subunit.  This family consist of exonuclease VII, large subunit EC:3.1.11.6 This enzyme catalyzes exonucleolytic cleavage in either 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. This exonuclease VII enzyme is composed of one large subunit and 4 small ones.','pfam02601','Exonuc_VII_L'),(971575,'gnl|CDD|307482',_binary 'OB-fold nucleic acid binding domain.  This family contains OB-fold domains that bind to nucleic acids. The family includes the anti-codon binding domain of lysyl, aspartyl, and asparaginyl -tRNA synthetases (see pfam00152). Aminoacyl-tRNA synthetases catalyze the addition of an amino acid to the appropriate tRNA molecule EC:6.1.1.-. This family also includes part of RecG helicase involved in DNA repair. Replication factor A is a hetero-trimeric complex, that contains a subunit in this family. This domain is also found at the C-terminus of bacterial DNA polymerase III alpha chain.','pfam01336','tRNA_anti-codon'),(971580,'gnl|CDD|277337',_binary 'Pyrococcus furiosus PF1019 and related proteins, metallophosphatase domain.  This family includes bacterial and archeal proteins homologous to PF1019, an uncharacterized Pyrococcus furiosus protein.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07391','MPP_PF1019'),(971586,'gnl|CDD|314229',_binary 'Protein of unknown function (DUF3039).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11238','DUF3039'),(971623,'gnl|CDD|275180',_binary 'major capsid protein, N4-gp56 family.  Members of this family are phage major capsid proteins as found in phage N4 (a double-stranded DNA virus) plus many additional lytic phage and integrated prophage regions. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR04387','hypothetical_protein'),(971671,'gnl|CDD|319259',_binary 'ParB N-terminal-like domain, some attached to C-terminal S-adenosylmethionine-dependent methyltransferase domain.  This family represents domains related to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, fused to a variety of C-terminal domains, including S-adenosylmethionine-dependent methyltransferase-like domains and DUF4417. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16402','ParB_N_like_MT'),(971697,'gnl|CDD|224697',_binary 'Phage terminase large subunit [General function prediction only].','COG1783','XtmB'),(971698,'gnl|CDD|310619',_binary 'Phage minor capsid protein 2.  Family of related phage minor capsid proteins.','pfam06152','Phage_min_cap2'),(971720,'gnl|CDD|238868',_binary 'SGNH_hydrolase subfamily, similar to the putative arylesterase/acylhydrolase from the rumen anaerobe Prevotella bryantii XynE. The P. bryantii XynE gene is located in a xylanase gene cluster. SGNH hydrolases are a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01830','XynE_like'),(971751,'gnl|CDD|163426',_binary 'accessory Sec system translocase SecA2.  Members of this protein family are homologous to SecA and part of the accessory Sec system. This system, including both five core proteins for export and a variable number of proteins for glycosylation, operates in certain Gram-positive pathogens for the maturation and delivery of serine-rich glycoproteins such as the cell surface glycoprotein GspB in Streptococcus gordonii. [Protein fate, Protein and peptide secretion and trafficking].','TIGR03714','Protein_translocase_subun'),(971752,'gnl|CDD|312982',_binary 'Phage regulatory protein Rha (Phage_pRha).  Members of this protein family are found in temperate phage and bacterial prophage regions. Members include the product of the rha gene of the lambdoid phage phi-80, a late operon gene. The presence of this gene interferes with infection of bacterial strains that lack integration host factor (IHF), which regulates the rha gene. It is suggested that Rha is a phage regulatory protein.','pfam09669','Phage_pRha'),(971754,'gnl|CDD|226172',_binary 'Uncharacterized phage-encoded protein [Function unknown].','COG3646','COG3646'),(971763,'gnl|CDD|272953',_binary 'excinuclease ABC, C subunit.  This family consists of the DNA repair enzyme UvrC, an ABC excinuclease subunit which interacts with the UvrA/UvrB complex to excise UV-damaged nucleotide segments. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00194','UvrABC_system_protein_C'),(971765,'gnl|CDD|223399',_binary 'Nuclease subunit of the excinuclease complex [DNA replication, recombination, and repair].','COG0322','UvrC'),(971769,'gnl|CDD|312831',_binary 'H-type lectin domain.  The H-type lectin domain is a unit of six beta chains, combined into a homo-hexamer. It is involved in self/non-self recognition of cells, through binding with carbohydrates. It is sometimes found in association with the F5_F8_type_C domain pfam00754.','pfam09458','H_lectin'),(971773,'gnl|CDD|223048',_binary 'putative ATPase subunit of terminase; Provisional.','PHA03333','PHA03333'),(971778,'gnl|CDD|143510',_binary 'Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme PvsE.  This subfamily is composed of PvsE from Vibrio parahaemolyticus and similar proteins. PvsE is a vibrioferrin biosynthesis protein which is homologous to eukaryotic ornithine decarboxylase (ODC) and diaminopimelate decarboxylase (DapDC). ODC and DapDC are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. It has been suggested that PvsE may be involved in the biosynthesis of the polycarboxylate siderophore vibrioferrin. It may catalyze the decarboxylation of serine to yield ethanolamine. PvsE may require homodimer formation and the presence of the PLP cofactor for activity.','cd06843','PLPDE_III_PvsE_like'),(972350,'gnl|CDD|198386',_binary 'GIY-YIG domain of uncharacterized bacterial protein structurally related to COG3410.  This family contains a group of uncharacterized bacterial proteins. Although their function roles have not been recognized, these proteins contain a putative GIY-YIG domain in their N-terminus. Moreover, a  conserved domain COG3410 with unknown function has been found in the C-terminus of most family members.','cd10439','GIY-YIG_COG3410'),(972380,'gnl|CDD|313144',_binary 'Protein of unknown function (DUF2089).  This domain, found in various hypothetical prokaryotic proteins, has no known function. This domain is a zinc-ribbon.','pfam09862','DUF2089'),(972843,'gnl|CDD|137836',_binary 'carboxylesterase BioH; Provisional.','PRK10349','PRK10349'),(972881,'gnl|CDD|227540',_binary 'Karyopherin (importin) beta [Intracellular trafficking and secretion].','COG5215','KAP95'),(972992,'gnl|CDD|270121',_binary 'Slm1 Pleckstrin homology (PH) domain.  Slm1 is a component of the target of rapamycin complex 2 (TORC2) signaling pathway. It plays a role in the regulation of actin organization and is a target of sphingolipid signaling during the heat shock response. Slm1 contains a single PH domain that binds PtdIns(4,5)P2, PtdIns(4)P, and dihydrosphingosine 1-phosphate (DHS-1P). Slm1 possesses two binding sites for anionic lipids. The non-canonical binding site of the PH domain of Slm1 is used for ligand binding, and it is proposed that beta-spectrin, Tiam1 and ArhGAP9 also have this type of phosphoinositide binding site. PH domains have diverse functions, but in general are involved in targeting proteins to the appropriate cellular location or in the interaction with a binding partner. They share little sequence conservation, but all have a common fold, which is electrostatically polarized. Less than 10% of PH domains bind phosphoinositide phosphates (PIPs) with high affinity and specificity. PH domains are distinguished from other PIP-binding domains by their specific high-affinity binding to PIPs with two vicinal phosphate groups: PtdIns(3,4)P2, PtdIns(4,5)P2 or PtdIns(3,4,5)P3 which results in targeting some PH domain proteins to the plasma membrane. A few display strong specificity in lipid binding. Any specificity is usually determined by loop regions or insertions in the N-terminus of the domain, which are not conserved across all PH domains. PH domains are found in cellular signaling proteins such as serine/threonine kinase, tyrosine kinases, regulators of G-proteins, endocytotic GTPases, adaptors, as well as cytoskeletal associated molecules and in lipid associated enzymes.','cd13311','PH_Slm1'),(973009,'gnl|CDD|273681',_binary 'phage tail tape measure protein, lambda family.  This model represents a relatively well-conserved region near the C-terminus of the tape measure protein of a lambda and related phage. This protein, which controls phage tail length, is typically about 1000 residues in length. Both low-complexity sequence and insertion/deletion events appear common in this family. Mutational studies suggest a ruler or template role in the determination of phage tail length. Similar behavior is attributed to proteins from distantly related or unrelated families in other phage. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01541','tail_length_tape'),(973126,'gnl|CDD|225222',_binary 'Predicted transcriptional regulator [Transcription].','COG2345','COG2345'),(973128,'gnl|CDD|153418',_binary 'Winged helix-turn-helix (WHTH) DNA-binding domain of the GntR family of transcriptional regulators.  This CD represents the winged HTH DNA-binding domain of the GntR (named after the gluconate operon repressor in Bacillus subtilis) family of bacterial transcriptional regulators and their putative homologs found in eukaryota and archaea. The GntR family has over 6000 members distributed among almost all bacterial species, which is comprised of FadR, HutC, MocR, YtrA, AraR, PlmA, and other subfamilies for the regulation of the most varied biological process. The monomeric proteins of the GntR family are characterized by two function domains: a small highly conserved winged helix-turn-helix prokaryotic DNA binding domain in the N-terminus, and a very diverse regulatory ligand-binding domain in the C-terminus for effector-binding/oligomerization, which provides the basis for the subfamily classifications.  Binding of the effector to GntR-like transcriptional regulators is presumed to result in a conformational change that regulates the DNA-binding affinity of the repressor. The GntR-like proteins bind as dimers, where each monomer recognizes a half-site of 2-fold symmetric DNA sequences.','cd07377','WHTH_GntR'),(973129,'gnl|CDD|177267',_binary 'endonuclease I.','PHA00159','PHA00159'),(973160,'gnl|CDD|225353',_binary 'Lysophospholipase L1 and related esterases [Amino acid transport and metabolism].','COG2755','TesA'),(973161,'gnl|CDD|238872',_binary 'SGNH_hydrolase subfamily. SGNH hydrolases are a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01834','SGNH_hydrolase_like_2'),(973162,'gnl|CDD|238860',_binary 'Lysophospholipase L1-like subgroup of SGNH-hydrolases. The best characterized member in this family is TesA, an E. coli periplasmic protein with thioesterase, esterase, arylesterase, protease and lysophospholipase activity.','cd01822','Lysophospholipase_L1_like'),(973163,'gnl|CDD|238876',_binary 'Isoamyl-acetate hydrolyzing esterase-like proteins. SGNH_hydrolase subfamily similar to the Saccharomyces cerevisiae IAH1. IAH1 may be the major esterase that hydrolyses isoamyl acetate in sake mash.  The SGNH-family of hydrolases is a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01838','Isoamyl_acetate_hydrolase'),(973370,'gnl|CDD|309566',_binary 'Restriction endonuclease.  Prokaryotic family found in type II restriction enzymes containing the hallmark (D/E)-(D/E)XK active site. Presence of catalytic residues implicates this region in the enzymatic cleavage of DNA.','pfam04471','Mrr_cat'),(973397,'gnl|CDD|239292',_binary 'PDIa family, TMX subfamily; composed of proteins similar to the TRX-related human transmembrane protein, TMX. TMX is a type I integral membrane protein; the N-terminal redox active TRX domain is present in the endoplasmic reticulum (ER) lumen while the C-terminus is oriented towards the cytoplasm. It is expressed in many cell types and its active site motif (CPAC) is unique. In vitro, TMX reduces interchain disulfides of insulin and renatures inactive RNase containing incorrect disulfide bonds. The C. elegans homolog, DPY-11, is expressed only in the hypodermis and resides in the cytoplasm. It is required for body and sensory organ morphogeneis. Another uncharacterized TRX-related transmembrane protein, human TMX4, is included in the alignment. The active site sequence of TMX4 is CPSC.','cd02994','PDI_a_TMX'),(973445,'gnl|CDD|239938',_binary 'YhaM_OBF_like: A subfamily of OB folds similar to that found in Bacillus subtilis YhaM and Staphylococcus aureus cmp-binding factor-1 (SaCBF1). Both these proteins are 3\'-to-5\'exoribonucleases. YhaM requires Mn2+ or Co2+ for activity and is inactive in the presence of Mg2+. YhaM also has a Mn2+ dependent 3\'-to-5\'single-stranded DNA exonuclease activity. SaCBF is also a double-stranded DNA binding protein, binding specifically to cmp, the replication enhancer found in S. aureus plasmid pT181. Proteins in this group combine an N-terminal OB fold with a C-terminal HD domain. The HD domain is found in metal-dependent phosphohydrolases.','cd04492','YhaM_OBF_like'),(973683,'gnl|CDD|205071',_binary 'Phage tail repeat like.  This family largely contains proteins from the eukaryote Trichomonas vaginalis. These proteins contain multiple HXH repeats. Some proteins in this family are annotated as having phage tail repeats. The function of this family is unknown.','pfam12789','PTR'),(974252,'gnl|CDD|225099',_binary 'Transcriptional regulators [Transcription].','COG2188','PhnF'),(974253,'gnl|CDD|224112',_binary 'DNA-directed RNA polymerase specialized sigma subunit [Transcription].','COG1191','FliA'),(974255,'gnl|CDD|223642',_binary 'DNA-directed RNA polymerase, sigma subunit (sigma70/sigma32) [Transcription].','COG0568','RpoD'),(974256,'gnl|CDD|315341',_binary 'Helix-turn-helix domain.  This domain appears to be a helix-turn-helix domain suggesting that this might be a transcriptional regulatory protein. Some members of this family are annotated as conjugative transposon domains.','pfam12645','HTH_16'),(974257,'gnl|CDD|309607',_binary 'Sigma-70 region 2.  Region 2 of sigma-70 is the most conserved region of the entire protein. All members of this class of sigma-factor contain region 2. The high conservation is due to region 2 containing both the -10 promoter recognition helix and the primary core RNA polymerase binding determinant. The core binding helix, interacts with the clamp domain of the largest polymerase subunit, beta prime. The aromatic residues of the recognition helix, found at the C-terminus of this domain are though to mediate strand separation, thereby allowing transcription initiation.','pfam04542','Sigma70_r2'),(974258,'gnl|CDD|235901',_binary 'flagellar biosynthesis sigma factor; Validated.','PRK06986','fliA'),(974259,'gnl|CDD|181075',_binary 'RNA polymerase sigma factor SigD; Validated.','PRK07670','PRK07670'),(974260,'gnl|CDD|131931',_binary 'RNA polymerase sigma-F factor.  Members of this protein family are the RNA polymerase sigma factor F. Sigma-F is specifically and universally a component of the Firmicutes lineage endospore formation program, and is expressed in the forespore to turn on expression of dozens of genes. It is closely homologous to sigma-G, which is also expressed in the forespore. [Transcription, Transcription factors, Cellular processes, Sporulation and germination].','TIGR02885','RNA_polymerase_sigma-F_fa'),(974261,'gnl|CDD|181361',_binary 'RNA polymerase factor sigma-70; Validated.','PRK08295','PRK08295'),(974262,'gnl|CDD|131447',_binary 'RNA polymerase sigma factor RpoS.  A sigma factor is a DNA-binding protein protein that binds to the DNA-directed RNA polymerase core to produce the holoenzyme capable of initiating transcription at specific sites. Different sigma factors act in vegetative growth, heat shock, extracytoplasmic functions (ECF), etc. This model represents the clade of sigma factors called RpoS (also called sigma-38, KatF, etc.), found only in Proteobacteria. This sigma factor is induced in stationary phase (in response to the stress of nutrient limitation) and becomes the second prinicipal sigma factor at that time. RpoS is a member of the larger Sigma-70 subfamily (TIGR02937) and most closely related to RpoD (TIGR02393). [Cellular processes, Adaptations to atypical conditions, Transcription, Transcription factors].','TIGR02394','RNA_polymerase_sigma_fact'),(974286,'gnl|CDD|308746',_binary 'Poxvirus D5 protein-like.  This family includes D5 from Poxviruses which is necessary for viral DNA replication, and is a nucleic acid independent nucleoside triphosphatase. Members of this family are also found outside of poxviruses. This domain is a DNA-binding winged HTH domain.','pfam03288','Pox_D5'),(974681,'gnl|CDD|275041',_binary 'methyltransferase, ATP-grasp peptide maturase system.  Members of this protein family are predicted SAM-dependent methyltransferases that regularly occur in the context of a putative peptide modification ATP-grasp enzyme (TIGR04187, related to enzymes of microviridin maturation) and a putative ribosomal peptide modification target (TIGR04186).','TIGR04188','conserved_hypothetical_pr'),(974683,'gnl|CDD|250389',_binary 'Protein-L-isoaspartate(D-aspartate) O-methyltransferase (PCMT).','pfam01135','PCMT'),(974684,'gnl|CDD|178974',_binary 'protein-L-isoaspartate O-methyltransferase; Reviewed.','PRK00312','pcm'),(974685,'gnl|CDD|275158',_binary 'methyltransferase, FxLD system.  Members of this family resemble occur regularly in the vicinity of lantibiotic biosynthesis enzymes and their probable target, the FxLD family of putative ribosomal natural product precursor (TIGR04363). Members resemble protein-L-isoaspartate O-methyltransferase (TIGR00080) and a predicted methyltranserase, TIGR04188, of another putative peptide modification system.','TIGR04364','methyltran_FxLD'),(974686,'gnl|CDD|184409',_binary 'protein-L-isoaspartate O-methyltransferase; Provisional.','PRK13942','PRK13942'),(974687,'gnl|CDD|272896',_binary 'protein-L-isoaspartate(D-aspartate) O-methyltransferase.  This is an all-kingdom (but not all species) full-length ortholog enzyme for repairing aging proteins. Among the prokaryotes, the gene name is pcm. Among eukaryotes, pimt. [Protein fate, Protein modification and repair].','TIGR00080','Protein-L-isoaspartate_O-'),(974688,'gnl|CDD|237568',_binary 'protein-L-isoaspartate O-methyltransferase; Provisional.','PRK13943','PRK13943'),(974689,'gnl|CDD|140001',_binary 'protein-L-isoaspartate O-methyltransferase; Provisional.','PRK13944','PRK13944'),(974691,'gnl|CDD|226472',_binary 'Phospholipid N-methyltransferase [Lipid metabolism].','COG3963','COG3963'),(974696,'gnl|CDD|102369',_binary 'transcription factor; Provisional.','PRK06424','PRK06424'),(974699,'gnl|CDD|309325',_binary 'Domain of unknown function (DUF397).  The function of this family is unknown.','pfam04149','DUF397'),(974700,'gnl|CDD|130605',_binary 'phage portal protein, putative, A118 family.  This model represents a family of phage minor structural proteins. The protein is suggested to be the head-tail connector, or portal protein, on the basis of its position in the phage gene order, its presence in mature phage, its size, and its conservation across a number of complete genomes of tailed phage that lack other candidate portal proteins. Several other known portal protein families lack clear homology to this family and to each other. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01542','minor_structural_protein_'),(974720,'gnl|CDD|225382',_binary 'Transposase and inactivated derivatives, IS30 family [DNA replication, recombination, and repair].','COG2826','Tra8'),(974996,'gnl|CDD|222867',_binary 'endonuclease subunit; Provisional.','PHA02546','47'),(975184,'gnl|CDD|314483',_binary 'Domain of unknown function (DUF4352).  Members of these family are putative lipoproteins that fall into the Antigen MPT63/MPB63 (immunoprotective extracellular protein) superfamily.','pfam11611','DUF4352'),(975275,'gnl|CDD|273579',_binary 'orotate phosphoribosyltransferase, Thermus family.  This model represents a distinct clade of orotate phosphoribosyltransferases. Members include the experimentally determined example from Thermus aquaticus and additional examples from Caulobacter crescentus, Helicobacter pylori, Mesorhizobium loti, and related species. [Purines, pyrimidines, nucleosides, and nucleotides, Pyrimidine ribonucleotide biosynthesis].','TIGR01367','Orotate_phosphoribosyltra'),(975435,'gnl|CDD|119351',_binary 'This conserved domain family includes a large number of catalytically inactive chitinase-like lectins (chitolectins) including YKL-39, YKL-40 (HCGP39), YM1, oviductin, and AMCase (acidic mammalian chitinase), as well as catalytically active chitotriosidases.  The conserved domain is an eight-stranded alpha/beta barrel fold belonging to the family 18 glycosyl hydrolases.  The fold has a pronounced active-site cleft at the C-terminal end of the beta-barrel.  The chitolectins lack a key active site glutamate (the proton donor required for hydrolytic activity) but retain highly conserved residues involved in oligosaccharide binding.  Chitotriosidase is a chitinolytic enzyme expressed in maturing macrophages, which suggests that it plays a part in antimicrobial defense.  Chitotriosidase hydrolyzes chitotriose, as well as colloidal chitin to yield chitobiose and is therefore considered an exochitinase. Chitotriosidase occurs in two major forms, the large form being converted to the small form by either RNA or post-translational processing.  Although the small form, containing the chitinase domain alone, is sufficient for the chitinolytic activity, the additional C-terminal chitin-binding domain of the large form plays a role in processing colloidal chitin. The chitotriosidase gene is nonessential in humans, as about 35% of the population are heterozygous and 6% homozygous for an inactivated form of the gene.  HCGP39 is a 39-kDa human cartilage glycoprotein thought to play a role in connective tissue remodeling and defense against pathogens.','cd02872','GH18_chitolectin_chitotri'),(975496,'gnl|CDD|133401',_binary 'Helix-Turn-Helix DNA binding domain of the YfmP transcription regulator.  Helix-turn-helix (HTH) transcription regulator, YfmP, and related proteins; N-terminal domain. YfmP regulates the multidrug efflux protein, YfmO, and indirectly regulates the expression of the Bacillus subtilis copZA operon encoding a metallochaperone, CopZ, and a CPx-type ATPase efflux protein, CopA. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.','cd04774','HTH_YfmP'),(975497,'gnl|CDD|133410',_binary 'Helix-Turn-Helix DNA binding domain of the MerR1 transcription regulator.  Helix-turn-helix (HTH) transcription regulator MerR1. MerR1 transcription regulators, such as Tn21 MerR and Tn501 MerR, mediate response to mercury exposure in eubacteria. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines that define a mercury binding site. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.','cd04783','HTH_MerR1'),(975500,'gnl|CDD|133394',_binary 'Helix-Turn-Helix DNA binding domain of the HspR transcription regulator.  Helix-turn-helix (HTH) transcription regulator HspR, N-terminal domain. Heat shock protein regulators (HspR) have been shown to regulate expression of specific regulons in response to high temperature or high osmolarity in Streptomyces and Helicobacter, respectively. These proteins share the N-terminal DNA binding domain  with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.  A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.','cd04766','HTH_HspR'),(975501,'gnl|CDD|316269',_binary 'Helix-turn-helix domain.','pfam13730','HTH_36'),(975502,'gnl|CDD|315472',_binary 'MarR family.  The Mar proteins are involved in the multiple antibiotic resistance, a non-specific resistance system. The expression of the mar operon is controlled by a repressor, MarR. A large number of compounds induce transcription of the mar operon. This is thought to be due to the compound binding to MarR, and the resulting complex stops MarR binding to the DNA. With the MarR repression lost, transcription of the operon proceeds. The structure of MarR is known and shows MarR as a dimer with each subunit containing a winged-helix DNA binding motif.','pfam12802','MarR_2'),(975512,'gnl|CDD|227316',_binary 'Uncharacterized conserved protein [Function unknown].','COG4983','COG4983'),(975532,'gnl|CDD|119378',_binary 'Lys-1 is a lysozyme encoded by the Caenorhabditis elegans lys-1 gene. This gene is one of a several lysozyme genes upregulated upon infection by the Gram-negative bacterial pathogen Serratia marcescens.  Lys-1 contains a glycosyl hydrolase family 25 (GH25) catalytic domain.  This family also includes Lys-5 from Caenorhabditis elegans.','cd06416','GH25_Lys1-like'),(975547,'gnl|CDD|281020',_binary 'Bacterial regulatory protein, Fis family.','pfam02954','HTH_8'),(975553,'gnl|CDD|225114',_binary 'Response regulator containing CheY-like receiver, AAA-type ATPase, and DNA-binding domains [Signal transduction mechanisms].','COG2204','AtoC'),(975694,'gnl|CDD|293543',_binary 'Putative phage holin Dp-1.  Phage_holin_Dp1 is a family of putative phage-holins from Gram-positive bacteria, largely Firmicutes, with two probable TMSs. The family shows lytic activity.','pfam16938','Phage_holin_Dp1'),(975711,'gnl|CDD|311927',_binary 'Cpl-7 lysozyme C-terminal domain.  This domain was originally found in the C-terminal moiety of the Cpl-7 lysozyme encoded by the Streptococcus pneumoniae bacteriophage Cp-7. It is assumed that these repeats represent cell wall binding motifs although no direct evidence has been obtained so far.','pfam08230','Cpl-7'),(975715,'gnl|CDD|275124',_binary 'SEC10/PgrA surface exclusion domain.  This model describes a conserved domain found in surface proteins of a number of Firmutes. Many members have LPXTG C-terminal anchoring motifs and a substantial number have the KxYKxGKxW putative sorting signal at the N-terminus. The tetracycline resistance plasmid pCF10 in Enterococcus faecalis promotes conjugal plasmid transfer in response to sex pheromones, but PgrA/Sec10 encoded by that plasmid, a member of this family, specifically inhibits the ability of cells to receive homologous plasmids. The phenomenon is called surface exclusion.','TIGR04320','hypothetical_protein'),(975721,'gnl|CDD|227076',_binary 'Phage-related protein, tail component [Function unknown].','COG4733','COG4733'),(975736,'gnl|CDD|198163',_binary 'Cpl-7 lysozyme C-terminal domain.  This domain was originally found in the C-terminal moiety of the Cpl-7 lysozyme encoded by the Streptococcus pneumoniae bacteriophage Cp-7. It is assumed that these repeats represent cell wall binding motifs although no direct evidence has been obtained so far.','smart01095','Cpl-7'),(975737,'gnl|CDD|274102',_binary 'addiction module antitoxin, RelB/DinJ family.  Plasmids may be maintained stably in bacterial populations through the action of addiction modules, in which a toxin and antidote are encoded in a cassette on the plasmid. In any daughter cell that lacks the plasmid, the toxin persists and is lethal after the antidote protein is depleted. Toxin/antitoxin pairs are also found on main chromosomes, and likely represent selfish DNA. Sequences in the seed for this alignment all were found adjacent to toxin genes. The resulting model appears to describe a narrower set of proteins than pfam04221, although many in the scope of this model are not obviously paired with toxin proteins. Several toxin/antitoxin pairs may occur in a single species. [Cellular processes, Toxin production and resistance, Mobile and extrachromosomal element functions, Other].','TIGR02384','Putative_antitoxin_RelB'),(975738,'gnl|CDD|225619',_binary 'DNA-damage-inducible protein J [DNA replication, recombination, and repair].','COG3077','RelB'),(975739,'gnl|CDD|282125',_binary 'RelB antitoxin.  RelE and RelB form a toxin-antitoxin system. RelE represses translation, probably through binding ribosomes. RelB stably binds RelE, presumably deactivating it.','pfam04221','RelB'),(975740,'gnl|CDD|183047',_binary 'bifunctional antitoxin/transcriptional repressor RelB; Provisional.','PRK11235','PRK11235'),(975741,'gnl|CDD|308199',_binary 'PemK-like, MazF-like toxin of type II toxin-antitoxin system.  PemK is a growth inhibitor in E. coli known to bind to the promoter region of the Pem operon, auto-regulating synthesis. This family represents the toxin molecule of a typical bacterial toxin-antitoxin system pairing. The family includes a number of different toxins, such as MazF, Kid, PemK, ChpA, ChpB and ChpAK.','pfam02452','PemK_toxin'),(975742,'gnl|CDD|316099',_binary 'Subunit ChlI of Mg-chelatase.','pfam13541','ChlI'),(975743,'gnl|CDD|223679',_binary 'Predicted ATPase with chaperone activity [Posttranslational modification, protein turnover, chaperones].','COG0606','COG0606'),(975744,'gnl|CDD|129465',_binary 'Mg chelatase-related protein.  The N-terminal end matches very strongly a pfam Mg_chelatase domain. [Unknown function, General].','TIGR00368','Uncharacterized_protein_Y'),(975745,'gnl|CDD|182120',_binary 'putative ATP-dependent protease; Provisional.','PRK09862','PRK09862'),(975757,'gnl|CDD|309994',_binary 'DivIVA protein.  The Bacillus subtilis divIVA1 mutation causes misplacement of the septum during cell division, resulting in the formation of small, circular, anucleate mini-cells. Inactivation of divIVA produces a mini-cell phenotype, whereas overproduction of DivIVA results in a filamentation phenotype. These proteins appear to contain coiled-coils.','pfam05103','DivIVA'),(975759,'gnl|CDD|274639',_binary 'DivIVA domain.  This model describes a domain found in Bacillus subtilis cell division initiation protein DivIVA, and homologs, toward the N-terminus. It is also found as a repeated domain in certain other proteins, including family TIGR03543.','TIGR03544','cell_division_initiation_'),(975763,'gnl|CDD|226128',_binary 'Uncharacterized phage-associated protein [Function unknown].','COG3600','GepA'),(975764,'gnl|CDD|315855',_binary 'Protein of unknown function (DUF4065).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea and viruses. Proteins in this family are typically between 155 and 202 amino acids in length.','pfam13274','DUF4065'),(976064,'gnl|CDD|225196',_binary 'Predicted membrane protein [Function unknown].','COG2314','XynA'),(976318,'gnl|CDD|239331',_binary 'Arsenate Reductase (ArsC) family, 15kD protein subfamily; composed of proteins of unknown function with similarity to thioredoxin-fold arsenic reductases, ArsC. It is encoded by an ORF present in a gene cluster associated with nitrogen fixation that also encodes dinitrogenase reductase ADP-ribosyltransferase (DRAT) and dinitrogenase reductase activating glycohydrolase (DRAG). ArsC catalyzes the reduction of arsenate [As(V)] to arsenite [As(III)], using reducing equivalents derived from glutathione via glutaredoxin, through a single catalytic cysteine.','cd03033','ArsC_15kD'),(977431,'gnl|CDD|319808',_binary 'beta-phosphoglucomutase-like family of the haloacid dehalogenase-like (HAD) hydrolase superfamily.  This family represents the beta-phosphoglucomutase-like family of the haloacid dehalogenase-like (HAD) hydrolase superfamily. Family members include Lactococcus lactis beta-PGM, a mutase which catalyzes the interconversion of beta-D-glucose 1-phosphate (G1P) and D-glucose 6-phosphate (G6P), Saccharomyces cerevisiae phosphatases GPP1 and GPP2 that dephosphorylate DL-glycerol-3-phosphate and DOG1 and DOG2 that dephosphorylate 2-deoxyglucose-6-phosphate, and Escherichia coli 6-phosphogluconate phosphatase YieH. It belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd07505','HAD_BPGM-like'),(977435,'gnl|CDD|213655',_binary '3-hydroxy-3-methylglutaryl CoA synthase, prokaryotic clade.  This clade of hydroxymethylglutaryl-CoA (HMG-CoA) synthases is found in a limited spectrum of mostly gram-positive bacteria which make isopentenyl pyrophosphate (IPP) via the mevalonate pathway. This pathway is found primarily in eukaryotes and archaea, but the bacterial homologs are distinct, having aparrently diverged after being laterally transferred from an early eukaryote. HMG-CoA synthase is the first step in the pathway and joins acetyl-CoA with acetoacetyl-CoA with the release of one molecule of CoA. The Borellia sequence may have resulted from a separate lateral transfer event.','TIGR01835','3-hydroxy-3-methylglutary'),(977503,'gnl|CDD|319272',_binary 'ParB N-terminal-like domain, some attached to C-terminal S-adenosylmethionine-dependent methyltransferase domain.  This family represents domains related to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, fused to a variety of C-terminal domains, including S-adenosylmethionine-dependent methyltransferase-like domains and DUF4417. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16844','ParB_N_like_MT'),(977512,'gnl|CDD|313716',_binary 'P63C domain.  This domain was identified by Iyer and colleagues.','pfam10546','P63C'),(977531,'gnl|CDD|182039',_binary 'transcriptional repressor DicA; Reviewed.','PRK09706','PRK09706'),(977635,'gnl|CDD|309976',_binary 'Phage virion morphogenesis family.  Protein S of phage P2 is thought to be involved in tail completion and stable head joining.','pfam05069','Phage_tail_S'),(977636,'gnl|CDD|227338',_binary 'Mu-like prophage protein gpG [General function prediction only].','COG5005','COG5005'),(977638,'gnl|CDD|224566',_binary 'Uncharacterized protein containing LysM domain [Function unknown].','COG1652','XkdP'),(977639,'gnl|CDD|236880',_binary 'LysM domain/BON superfamily protein; Provisional.','PRK11198','PRK11198'),(977643,'gnl|CDD|131945',_binary 'spore coat assembly protein SafA.  SafA (YrbB) (SafA) of Bacillus subtilis is a protein found at the interface of the spore cortex and spore coat, and is dependent on SpoVID for its localization. This model is based on the N-terminal LysM (lysin motif) domain (see pfamAM model pfam01476) of SafA and, from several other spore-forming species, the protein with the most similar N-terminal region. However, this set of proteins differs greatly in C-terminal of the LysM domaim; blocks of 12-residue and 13-residue repeats are found in the Bacillus cereus group, tandem LysM domains in Thermoanaerobacter tengcongensis MB4, etc. in which one of which is found in most examples of endospore-forming bacteria. [Cellular processes, Sporulation and germination].','TIGR02899','hypothetical_conserved_pr'),(977649,'gnl|CDD|309894',_binary 'Gene 25-like lysozyme.  This family includes the phage protein Gene 25 from T4 which is a structural component of the outer wedge of the baseplate that has acidic lysozyme activity. The family also includes relatives from bacteria that are also presumably lysozymes.','pfam04965','GPW_gp25'),(977654,'gnl|CDD|274051',_binary 'putative baseplate assembly protein.  This family consists of a large, conserved hypothetical protein in phage tail-like regions of at least six bacterial genomes: Gloeobacter violaceus PCC 7421, Geobacter sulfurreducens PCA, Streptomyces coelicolor A3(2), Streptomyces avermitilis MA-4680, Mesorhizobium loti, and Myxococcus xanthus. The C-terminal region is identified by the broader model pfam04865 as related to baseplate protein J from phage P2, but that relationship is not observed directly. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR02243','hypothetical_protein_SCD8'),(977655,'gnl|CDD|222871',_binary 'baseplate wedge subunit; Provisional.','PHA02553','6'),(977685,'gnl|CDD|239601',_binary 'RPA2_OBF_family: A family of oligonucleotide binding (OB) folds with similarity to the OB fold of the single strand (ss) DNA-binding domain (DBD)-D of human RPA2 (also called RPA32). RPA2 is a subunit of Replication protein A (RPA). RPA is a nuclear ssDNA-binding protein (SSB) which appears to be involved in all aspects of DNA metabolism including replication, recombination, and repair. RPA also mediates specific interactions of various nuclear proteins. In animals, plants, and fungi, RPA is a heterotrimer with subunits of 70KDa (RPA1), 32kDa (RPA2), and 14 KDa (RPA3). RPA contains six OB folds, which are involved in ssDNA binding and in trimerization. The ssDNA binding mechanism is believed to be multistep and to involve conformational change. This family also includes OB folds similar to those found in Escherichia coli SSB, the wedge domain of E. coli RecG (a branched-DNA-specific helicase), E. coli ssDNA specific exodeoxyribonuclease VII large subunit, Pyrococcus abyssi DNA polymerase II (Pol II) small subunit, Sulfolobus solfataricus SSB, and Bacillus subtilis YhaM (a 3\'-to-5\'exoribonuclease). It also includes the OB folds of breast cancer susceptibility gene 2 protein (BRCA2), Oxytricha nova telomere end binding protein (TEBP), Saccharomyces cerevisiae telomere-binding protein (Cdc13), and human protection of telomeres 1 protein (POT1).','cd03524','RPA2_OBF_family'),(977697,'gnl|CDD|319257',_binary 'ParB/Srx_like nuclease and putative transcriptional regulators related to SbnI.  This family contains a Pyrococcus Furiosus enzyme reported to have DNA nuclease activity and resembles the N-terminal domain of ParB proteins of the parABS bacterial chromosome partitioning system. This sub-family also includes siderophore staphylobactin biosynthesis protein SbnI. 60% of the P. furiosus nuclease activity was retained at 90 degree C, suggesting a physiological role in the organism, which can grow in temperatures as high as 100 degrees Celsius. The protein has endo- and exo-nuclease activity vs. single- and double-stranded DNA, and nuclease activity was lost in methylated proteins prepared for structure solution. This family has a fairly well-conserved DGHHR motif that corresponds to the same structural position as the phosphorylation site (a portion of the ATP-Mg-binding site) of sulfiredoxin and the arginine-rich ParB BoxII of ParB.','cd16400','ParB_Srx_like_nuclease'),(977712,'gnl|CDD|184907',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14943','PRK14943'),(977713,'gnl|CDD|184908',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14944','PRK14944'),(977715,'gnl|CDD|308417',_binary 'DNA polymerase III beta subunit, central domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam02767','DNA_pol3_beta_2'),(977716,'gnl|CDD|280863',_binary 'DNA polymerase III beta subunit, C-terminal domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam02768','DNA_pol3_beta_3'),(977753,'gnl|CDD|277319',_binary 'Mre11 nuclease, N-terminal metallophosphatase domain.  Mre11 (also known as SbcD in Escherichia coli) is a subunit of the MRX protein complex. This complex includes: Mre11, Rad50, and Xrs2/Nbs1, and plays a vital role in several nuclear processes including DNA double-strand break repair, telomere length maintenance, cell cycle checkpoint control, and meiotic recombination, in eukaryotes. During double-strand break repair, the MRX complex is required to hold the two ends of a broken chromosome together. In vitro studies show that Mre11 has 3\'-5\' exonuclease activity on dsDNA templates and endonuclease activity on dsDNA and ssDNA templates. In addition to the N-terminal phosphatase domain, the eukaryotic MRE11 members of this family have a C-terminal DNA binding domain (not included in this alignment model). MRE11-like proteins are found in prokaryotes and archaea was well as in eukaryotes. Mre11 belongs to the metallophosphatase (MPP) superfamily. MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases). The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd00840','MPP_Mre11_N'),(977785,'gnl|CDD|143480',_binary 'Iron-sulfur cluster scaffold-like proteins.  IscU_like and NifU_like proteins. IscU and NifU function as a scaffold for the assembly of [2Fe-2S] clusters before they are transferred to apo target proteins. They are highly conserved and play vital roles in the ISC and NIF systems of Fe-S protein maturation. NIF genes participate in nitrogen fixation in several isolated bacterial species. The NifU domain, however, is also found in bacteria that do not fix nitrogen, so it may have wider significance in the cell. Human IscU interacts with frataxin, the Friedreich ataxia gene product, and incorrectly spliced IscU has been shown to disrupt iron homeostasis in skeletal muscle and cause myopathy.','cd06664','IscU_like'),(977786,'gnl|CDD|273918',_binary 'SUF system FeS assembly protein, NifU family.  Three iron-sulfur cluster assembly systems are known so far. ISC is broadly distributed while NIF tends to be associated with nitrogenase in nitrogen-fixing bacteria. The most recently described is SUF, believed to be important to maintain the function during aerobic stress of enzymes with labile Fe-S clusters. It is fairly widely distributed. This family represents one of two different proteins proposed to act as a scaffold on which the Fe-S cluster is built and from which it is transferred. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other].','TIGR01994','Iron-sulfur_cluster_assem'),(977787,'gnl|CDD|223892',_binary 'NifU homolog involved in Fe-S cluster formation [Energy production and conversion].','COG0822','IscU'),(977788,'gnl|CDD|307635',_binary 'NifU-like N terminal domain.  This domain is found in NifU in combination with pfam01106. This domain is found on isolated in several bacterial species. The nif genes are responsible for nitrogen fixation. However this domain is found in bacteria that do not fix nitrogen, so it may have a broader significance in the cell than nitrogen fixation. These proteins appear to be scaffold proteins for iron-sulfur clusters.','pfam01592','NifU_N'),(977789,'gnl|CDD|132460',_binary 'FeS cluster assembly scaffold protein NifU, Clostridium type.  NifU and NifS form a pair of iron-sulfur (FeS) cluster biosynthesis proteins much simpler than the ISC and SUF systems. Members of this protein family are a distinct group of NifU-like proteins, found always to a NifS-like protein and restricted to species that lack a SUF system. Typically, NIF systems service a smaller number of FeS-containing proteins than do ISC or SUF. Members of this particular branch typically are found, almost half the time, near the mnmA gene, involved in the carboxymethylaminomethyl modification of U34 in some tRNAs (see GenProp0704). While other NifU proteins are associated with nitrogen fixation, this family is not. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other].','TIGR03419','NifU_clost'),(977790,'gnl|CDD|183087',_binary 'scaffold protein; Provisional.','PRK11325','PRK11325'),(977839,'gnl|CDD|180831',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07088','PRK07088'),(978244,'gnl|CDD|273486',_binary 'putative methyltransferase, TIGR01177 family.  This family of probable methyltransferases is found exclusively in the Archaea. [Hypothetical proteins, Conserved].','TIGR01177','tRNA_guanine10-N2-dimethy'),(978313,'gnl|CDD|225244',_binary 'Uncharacterized protein, homolog of phage Mu protein gp30 [Function unknown].','COG2369','COG2369'),(978314,'gnl|CDD|274058',_binary '2\'-5\' RNA ligase.  This protein family consists of bacterial and archaeal proteins with two tandem copies of Pfam domain pfam02834. Members for which activity has been measured perform a reversible, ATP-independent 2\'-5\'-ligation of what is presumably a non-phyiological substrate: half-tRNA splice intermediates from an intron-containing yeast tRNA. The physiological substrate(s) in prokaryotes may include small 2\'-5\'-link-containing oligonucleotides, perhaps with regulatory or biosynthetic roles. [Transcription, RNA processing].','TIGR02258','UPF0097_protein_AF_2157'),(978341,'gnl|CDD|315614',_binary 'Domain of unknown function (DUF3854).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences. This domain is likely to be related to the Toprim domain.','pfam12965','DUF3854'),(978364,'gnl|CDD|309707',_binary 'SPP1 phage holin.  This family constitutes holin proteins from the dsDNA Siphidoviridae group bacteriophages with two transmembrane segments. Most bacteriophages require an endolysin and a holin for host lysis. During late gene expression, holins accumulate and oligomerise in the host cell membrane. They then suddenly trigger to permeablise the membrane, which causes lysis by allowing endolysin to attach the peptidoglycan. There are thought to be at least 35 different families of holin genes.','pfam04688','Holin_SPP1'),(978365,'gnl|CDD|130653',_binary 'holin, SPP1 family.  This model represents one of more than 30 families of phage proteins, all lacking detectable homology with each other, known or believed to act as holins. Holins act in cell lysis by bacteriophage. Members of this family are found in phage PBSX and phage SPP1, among others. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01592','Uncharacterized_9.5_kDa_p'),(978368,'gnl|CDD|224439',_binary 'Transcriptional regulators [Transcription].','COG1522','Lrp'),(978370,'gnl|CDD|214628',_binary 'helix_turn_helix ASNC type.  AsnC: an autogenously regulated activator of asparagine synthetase A transcription in Escherichia coli).','smart00344','HTH_ASNC'),(978371,'gnl|CDD|197670',_binary 'helix_turn_helix multiple antibiotic resistance protein.','smart00347','HTH_MARR'),(978381,'gnl|CDD|316920',_binary 'Bacterial EndoU nuclease.  This is a bacterial virion of EndoU nuclease. It is found at C-terminal region of polymorphic toxin proteins.','pfam14436','EndoU_bacteria'),(978384,'gnl|CDD|188241',_binary 'probable addiction module antidote protein.  Members of this strictly bacterial protein family are small, at roughly 100 amino acids. The gene is almost invariably the downstream member of a gene pair. It is a predicted DNA-binding protein from a clade within Pfam helix-turn-helix family pfam01381. These gene pairs, when found on the bacterial chromosome, are located often with prophage regions, but also both in integrated plasmid regions and in housekeeping gene regions. Analysis suggests that the gene pair may serve as an addiction module. [Mobile and extrachromosomal element functions, Other].','TIGR02684','conserved_hypothetical_pr'),(978385,'gnl|CDD|226162',_binary 'Predicted transcriptional regulator [Transcription].','COG3636','COG3636'),(978388,'gnl|CDD|318191',_binary 'HicB_like antitoxin of bacterial toxin-antitoxin system.  This is a family of HicB-like antitoxins.','pfam15919','HicB_lk_antitox'),(978389,'gnl|CDD|224514',_binary 'Predicted nuclease of the RNAse H fold, HicB family [General    function prediction only].','COG1598','COG1598'),(978396,'gnl|CDD|223736',_binary 'cAMP-binding proteins - catabolite gene activator and regulatory subunit of cAMP-dependent protein kinases [Signal transduction mechanisms].','COG0664','Crp'),(978398,'gnl|CDD|128696',_binary 'helix_turn_helix, cAMP Regulatory protein.','smart00419','HTH_CRP'),(978424,'gnl|CDD|274755',_binary 'PRTRC system ParB family protein.  A novel genetic system characterized by six major proteins, included a ParB homolog and a ThiF homolog, is designated PRTRC, or ParB-Related,ThiF-Related Cassette. It is often found on plasmids. This protein family the member related to ParB, and is designated PRTRC system ParB family protein.','TIGR03734','PRTRC_parB'),(978432,'gnl|CDD|274586',_binary 'plasmid partitioning protein RepB.  Members of this family are the RepB protein involved in replicon partitioning. RepB is found, in general, as part of a repABC operon in plasmids and small chromosomes, separate from the main chromosome, in various bacteria. This model describes a rather narrow clade of proteins; it should be noted that additional homologs scoring below the trusted cutoff have very similar functions, although they may be named differently. [Mobile and extrachromosomal element functions, Plasmid functions].','TIGR03454','partition_RepB'),(978434,'gnl|CDD|319263',_binary 'ParB N-terminal, parA-binding, -like domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16406','ParB_N_like'),(978437,'gnl|CDD|319262',_binary 'plasmid segregation replication protein B like protein, N-terminal domain.  RepB, found on plasmids and secondary chromosomes, works along with repA in directing plasmid segregation, and has been shown in Rhizobium etli to require the parS centromere-like sequence for full transcriptional repression of the repABC operon, inducing plasmid incompatibility. RepA is a Walker-type ATPase that complexes with RepB to form DNA-protein complexes in the presence of ATP/ADP. RepC is an initiator protein for the plasmid. repA and repB are homologous to the parA and ParB genes of the parABS partitioning system found on primary chromosomes.','cd16405','RepB_like_N'),(978439,'gnl|CDD|319268',_binary 'ParB N-terminal, parA -binding, domain of bacterial and plasmid parABS partitioning systems.  This family represents the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system. parABS contributes to the efficient segregation of chromosomes and low-copy number plasmids to daughter cells during prokaryotic cell division. The process includes the parA (Walker box) ATPase, the ParB DNA-binding protein and a parS cis-acting DNA sites. Binding of ParB to centromere-like parS sites is followed by non-specific binding to DNA (\'spreading\', which has been implicated in gene silencing in plasmid P1) and oligomerization of additional ParB molecules near the parS sites. It has been proposed that ParB-ParB cross-linking compacts the DNA, binds to parA via the N-terminal region, and leads to parA separating the ParB-parS complexes and the recruitment of the SMC (structural maintenance of chromosomes) complexes. The ParB N-terminal domain of Bacillus subtilis and other species contains a Arginine-rich ParB Box II with residues essential for bridging of the ParB-parS complexes. The arginine-rich ParB Box II consensus (I[VIL]AGERR[FYW]RA[AS] identified in several species is partially conserved with this family and related families. Mutations within the basic columns particularly debilitate spreading from the parS sites and impair SMC recruitment. The C-terminal domain contains a HTH DNA-binding motif and is the primary homo-dimerization domain, and binds to parS DNA sites. Additional homo-dimerization contacts are found along the N-terminal domain, but dimerization of the N-terminus may only occur after concentration at ParB-parS foci.','cd16411','ParB_N_like'),(978443,'gnl|CDD|172387',_binary 'plasmid partitioning protein RepB; Provisional.','PRK13866','PRK13866'),(978450,'gnl|CDD|315376',_binary 'Protein of unknown function (DUF3800).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea, eukaryotes and viruses. Proteins in this family are typically between 215 and 302 amino acids in length. There is a DE motif at the N-terminus and a QXXD motif at the C-terminus that may be functionally important.','pfam12686','DUF3800'),(978451,'gnl|CDD|213355',_binary 'Escherichia coli CorA-Salmonella typhimurium ZntB_like family.  A family of the MIT superfamily of essential membrane proteins involved in transporting divalent cations (uptake or efflux) across membranes. Members of this family are found in all three kingdoms of life. It is a functionally diverse family, including the Mg2+ transporters Escherichia coli and Salmonella typhimurium CorAs (which can also transport Co2+, and Ni2+ ), and the Zn2+ transporter Salmonella typhimurium ZntB which mediates the efflux of Zn2+ (and Cd2+). It also includes two Saccharomyces cerevisiae members: the inner membrane Mg2+ transporters Mfm1p/Lpe10p, and Mrs2p, and a family of Arabidopsis thaliana members (AtMGTs) some of which are localized to distinct tissues, and not all of which can transport Mg2+. Structures of the intracellular domain of Vibrio parahaemolyticus and Salmonella typhimurium ZntB form funnel-shaped homopentamers, the tip of the funnel is formed from two C-terminal transmembrane (TM) helices from each monomer, and the large opening of the funnel from the N-terminal cytoplasmic domains. The GMN signature motif of the MIT superfamily occurs just after TM1, mutation within this motif is known to abolish Mg2+ transport through Salmonella typhimurium CorA, and Mrs2p. Natural variants such as GVN and GIN, such as occur in some ZntB family proteins, may be associated with the transport of different divalent cations, such as zinc and cadmium. The functional diversity of MIT transporters may also be due to minor structural differences regulating gating, substrate selection, and transport.','cd12821','EcCorA_ZntB-like'),(978783,'gnl|CDD|132132',_binary 'sugar transferase, PEP-CTERM/EpsH1 system associated.  Members of this family include a match to the pfam00534 Glycosyl transferases group 1 domain. Nearly all are found in species that encode the PEP-CTERM/exosortase system predicted to act in protein sorting in a number of Gram-negative bacteria. In particular, these transferases are found proximal to a particular variant of exosortase, EpsH1, which appears to travel with a conserved group of genes summarized by Genome Property GenProp0652. The nature of the sugar transferase reaction catalyzed by members of this clade is unknown and may conceivably be variable with respect to substrate by species, but we hypothesize a conserved substrate.','TIGR03088','stp2'),(978785,'gnl|CDD|99979',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. WbnK in Shigella dysenteriae has been shown to be involved in the type 7 O-antigen biosynthesis.','cd03807','GT1_WbnK_like'),(978786,'gnl|CDD|100000',_binary 'This family is most closely related to the GT1 family of glycosyltransferases and is named after WbdM in Escherichia coli. In general glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found in bacteria.','cd04951','GT1_WbdM_like'),(978788,'gnl|CDD|99985',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found mainly in bacteria and eukaryotes.','cd03814','GT1_like_2'),(978794,'gnl|CDD|181176',_binary 'DNA polymerase III subunit epsilon; Provisional.','PRK07942','PRK07942'),(978799,'gnl|CDD|180524',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK06309','PRK06309'),(978803,'gnl|CDD|315926',_binary 'UvrD-like helicase C-terminal domain.  This domain is found at the C-terminus of a wide variety of helicase enzymes. This domain has a AAA-like structural fold.','pfam13361','UvrD_C'),(978804,'gnl|CDD|130473',_binary 'DNA polymerase III, epsilon subunit, Proteobacterial.  This model represents DnaQ, the DNA polymerase III epsilon subunit, as found in most Proteobacteria. It consists largely of an exonuclease domain as described in pfam00929. In Gram-positive bacteria, closely related regions are found both in the Gram-positive type DNA polymerase III alpha subunit and as an additional N-terminal domain of a DinG-family helicase. Both are excluded from this model, as are smaller proteins, also outside the Proteobacteria, that are similar in size to the epsilon subunit but as different in sequence as are the epsilon-like regions found in Gram-positive bacteria. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01406','DNA_polymerase_III_subuni'),(978808,'gnl|CDD|180525',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK06310','PRK06310'),(978809,'gnl|CDD|129663',_binary 'exonuclease, DNA polymerase III, epsilon subunit family.  All proteins in this family for which functions are known are components of the DNA polymerase III complex (epsilon subunit). There is, however, an outgroup that includes paralogs in some gamma-proteobacteria and the n-terminal region of DinG from some low GC gram positive bacteria. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, Degradation of DNA].','TIGR00573','DNA_polymerase_III_subuni'),(978814,'gnl|CDD|236392',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK09146','PRK09146'),(978815,'gnl|CDD|180377',_binary 'DNA polymerase III subunit epsilon; Provisional.','PRK06063','PRK06063'),(978817,'gnl|CDD|236085',_binary 'hypothetical protein; Provisional.','PRK07740','PRK07740'),(978828,'gnl|CDD|314819',_binary 'MgsA AAA+ ATPase C terminal.  The MgsA protein possesses DNA-dependent ATPase and ssDNA annealing activities. MgsA contributes to the recovery of stalled replication forks and therefore prevents genomic instability caused by aberrant DNA replication. Additionally, MgsA may play a role in chromosomal segregation. This is consistent with a report that MgsA co-localizes with the replisome and affects chromosome segregation. This domain represents the C terminal region of MgsA.','pfam12002','MgsA_C'),(978829,'gnl|CDD|237355',_binary 'recombination factor protein RarA; Reviewed.','PRK13342','PRK13342'),(978830,'gnl|CDD|225165',_binary 'ATPase related to the helicase subunit of the Holliday junction resolvase [DNA replication, recombination, and repair].','COG2256','MGS1'),(978831,'gnl|CDD|237354',_binary 'recombination factor protein RarA/unknown domain fusion protein; Reviewed.','PRK13341','PRK13341'),(978839,'gnl|CDD|99987',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. UDP-glucose-diacylglycerol glucosyltransferase (UGDG; also known as 1,2-diacylglycerol 3-glucosyltransferase) catalyzes the transfer of glucose from UDP-glucose to 1,2-diacylglycerol forming 3-D-glucosyl-1,2-diacylglycerol.','cd03817','GT1_UGDG_like'),(978843,'gnl|CDD|99998',_binary 'This family is most closely related to the GT1 family of glycosyltransferases and is named after gtfA in Streptococcus gordonii, where it plays a role in the O-linked glycosylation of GspB, a cell surface glycoprotein involved in platelet binding.  In general glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found in bacteria.','cd04949','GT1_gtfA_like'),(979511,'gnl|CDD|236984',_binary 'N-acetyl-anhydromuranmyl-L-alanine amidase; Provisional.','PRK11789','PRK11789'),(979573,'gnl|CDD|222995',_binary 'dUTPase; Provisional.','PHA03130','PHA03130'),(979575,'gnl|CDD|165395',_binary 'dUTPase; Provisional.','PHA03123','PHA03123'),(979583,'gnl|CDD|214947',_binary 'This entry represents the putative helicase A859L.','smart00974','T5orf172'),(979584,'gnl|CDD|313714',_binary 'T5orf172 domain.  This domain was identified by Iyer and colleagues.','pfam10544','T5orf172'),(979664,'gnl|CDD|236967',_binary 'dihydropyrimidine dehydrogenase subunit A; Provisional.','PRK11749','PRK11749'),(979726,'gnl|CDD|226075',_binary 'Predicted esterase of the alpha/beta hydrolase fold [General function prediction only].','COG3545','COG3545'),(979734,'gnl|CDD|223812',_binary 'Soluble lytic murein transglycosylase and related regulatory proteins (some contain LysM/invasin domains) [Cell envelope biogenesis, outer membrane].','COG0741','MltE'),(979756,'gnl|CDD|224125',_binary 'Superfamily II helicase [General function prediction only].','COG1204','COG1204'),(979767,'gnl|CDD|315782',_binary 'AAA ATPase domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily.','pfam13191','AAA_16'),(979768,'gnl|CDD|316156',_binary 'AAA domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily. Many of the proteins in this family are conjugative transfer proteins. There is a Walker A and Walker B.','pfam13604','AAA_30'),(979806,'gnl|CDD|274023',_binary 'Glutaredoxin-like family.  This family of glutaredoxin-like proteins is aparrently limited to plants. Multiple isoforms are found in A. thaliana and O.sativa.','TIGR02189','glutaredoxin-like_protein'),(979808,'gnl|CDD|274016',_binary 'Glutaredoxin.  Glutaredoxins are thioltransferases (disulfide reductases) which utilize glutathione and NADPH as cofactors. Oxidized glutathione is regenerated by glutathione reductase. Together these components compose the glutathione system. Glutaredoxins utilize the CXXC motif common to thioredoxins and are involved in multiple cellular processes including protection from redox stress, reduction of critical enzymes such as ribonucleotide reductase and the generation of reduced sulfur for iron sulfur cluster formation. Glutaredoxins are capable of reduction of mixed disulfides of glutathione as well as the formation of glutathione mixed disulfides. This model represents eukaryotic glutaredoxins and includes sequences from fungi, plants and metazoans as well as viruses.','TIGR02180','Uncharacterized_monothiol'),(979900,'gnl|CDD|273916',_binary 'PTS system, trehalose-specific IIBC component.  This model represents the fused enzyme II B and C components of the trehalose-specific PTS sugar transporter system. Trehalose is converted to trehalose-6-phosphate in the process of translocation into the cell. These transporters lack their own IIA domains and instead use the glucose IIA protein (IIAglc or Crr). The exceptions to this rule are Staphylococci and Streptococci which contain their own A domain as a C-terminal fusion. This family is closely related to the sucrose transporting PTS IIBC enzymes and the B and C domains of each are described by subfamily-domain level TIGRFAMs models (TIGR00826 and TIGR00852, respectively). In E. coli, B. subtilis and P. fluorescens the presence of this gene is associated with the presence of trehalase which degrades T6P to glucose and glucose-6-P. Trehalose may also be transported (in Salmonella) via the mannose PTS or galactose permease systems, or (in Sinorhizobium, Thermococcus and Sulfolobus, for instance) by ABC transporters.','TIGR01992','phosphotransferase_system'),(979990,'gnl|CDD|308611',_binary 'Domain of unknown function (DUF239).  This is a family of plant and bacterial proteins, a small number of which are putative carboxy-terminal peptidases.','pfam03080','DUF239'),(980097,'gnl|CDD|311742',_binary 'TPR/MLP1/MLP2-like protein.  The sequences featured in this family are similar to a region of human TPR protein and to yeast myosin-like proteins 1 (MLP1) and 2 (MLP2). These proteins share a number of features; for example, they all have coiled-coil regions and all three are associated with nuclear pores. TPR is thought to be a component of nuclear pore complex- attached intra-nuclear filaments, and is implicated in nuclear protein import. Moreover, its N-terminal region is involved in the activation of oncogenic kinases, possibly by mediating the dimerisation of kinase domains or by targeting these kinases to the nuclear pore complex. MLP1 and MLP2 are involved in the process of telomere length regulation, where they are thought to interact with proteins such as Tel1p and modulate their activity.','pfam07926','TPR_MLP1_2'),(980185,'gnl|CDD|319836',_binary 'molecular class C acid phosphatases, similar to Haemophilus influenzae e (P4) acid phosphatase; belongs to the haloacid dehalogenase-like hydrolase superfamily.  Molecular class C acid phosphatases (CAPs) are nonspecific acid phosphatases with generally broad substrate specificity and optimum activity at neutral to acidic pH. Members include Haemophilus influenzae lipoprotein e (P4), Elizabethkingia meningosepticum OlpA, Helicobacter pylori HppA, Enterobacter sp. 4 acid phosphatase PhoI, and Streptococcus pyogenes M1 GAS LppC. Lipoprotein e (P4) exhibits phosphomonoesterase activity with aryl phosphate substrates including nicotinamide mononucleotide (NMN), tyrosine phosphate, phenyl phosphate, p-nitrophenyl phosphate, and 4-methylumbelliferyl phosphate. The role of P4 in NAD+ uptake appears to be the dephosphorylation of NMN to nicotinamide riboside, which is then taken up by the organism. Elizabethkingia meningosepticum OlpA is a broad-spectrum nucleotidase with preference for 5\'-nucleotides, it efficiently hydrolyzes nucleotide monophosphates, with a strong preference for 5\'-nucleotides and for 3\'-AMP; OlpA can also hydrolyze sugar phosphates and beta-glycerol phosphate, although with a lower efficiency. Helicobacter pylori HppA is also a 5\' nucleotidase. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd07534','HAD_CAP'),(980905,'gnl|CDD|310368',_binary 'Permuted papain-like amidase enzyme, YaeF/YiiX, C92 family.  Amidase_YiiX is a family of permuted papain-like amidases. It has amidase specificity for the amide bond between a lipid and an amino acid (or peptide). From the structure, a tetramer, each monomer is made up of a layered alpha-beta fold with a central, 6-stranded, antiparallel beta-sheet that is protected by helices on either side. The catalytic Cys154 in UniProtKB:Q74NK7, Structure 3kw0, is located on the N-terminus of helix alphaF. The two additional helices located above Cys154 contribute to the formation of the active site, where the lysine ligand is bound.','pfam05708','Peptidase_C92'),(980906,'gnl|CDD|236642',_binary 'hypothetical protein; Provisional.','PRK10030','PRK10030'),(980915,'gnl|CDD|273910',_binary 'ubiquinone biosynthesis O-methyltransferase.  This model represents an O-methyltransferase believed to act at two points in the ubiquinone biosynthetic pathway in bacteria (UbiG) and fungi (COQ3). A separate methylase (MenG/UbiE) catalyzes the single C-methylation step. The most commonly used names for genes in this family do not indicate whether this gene is an O-methyl, or C-methyl transferase. [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR01983','Ubiquinone_biosynthesis_O'),(980917,'gnl|CDD|282961',_binary 'Methyltransferase small domain.  This domain is found in ribosomal RNA small subunit methyltransferase C as well as other methyltransferases.','pfam05175','MTS'),(980927,'gnl|CDD|225021',_binary 'Predicted phosphatase homologous to the C-terminal domain of histone macroH2A1 [General function prediction only].','COG2110','COG2110'),(980929,'gnl|CDD|277317',_binary 'metallophosphatase superfamily, metallophosphatase domain.  Metallophosphatases (MPPs), also known as metallophosphoesterases, phosphodiesterases (PDEs), binuclear metallophosphoesterases, and dimetal-containing phosphoesterases (DMPs), represent a diverse superfamily of enzymes with a conserved domain containing an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. This superfamily includes: the phosphoprotein phosphatases (PPPs), Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd00838','MPP_superfamily'),(980930,'gnl|CDD|277331',_binary 'Bacillus subtilis YkuE and related proteins, C-terminal metallophosphatase domain.  YkuE is an uncharacterized Bacillus subtilis protein with a C-terminal metallophosphatase domain and an N-terminal twin-arginine (RR) motif. An RR-signal peptide derived from the Bacillus subtilis YkuE protein can direct Tat-dependent secretion of agarase in Streptomyces lividans. This is an indication that YkuE is transported by the Bacillus subtilis Tat (Twin-arginine translocation) pathway machinery.  YkuE belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07385','MPP_YkuE_C'),(980932,'gnl|CDD|273417',_binary 'DNA topoisomerase VI, B subunit.  This model describes DNA topoisomerase VI, an archaeal type II DNA topoisomerase (DNA gyrase). [DNA metabolism, DNA replication, recombination, and repair].','TIGR01052','Type_2_DNA_topoisomerase_'),(980933,'gnl|CDD|316143',_binary 'Histidine kinase-, DNA gyrase B-, and HSP90-like ATPase.  This family represents, additionally, the structurally related ATPase domains of histidine kinase, DNA gyrase B and HSP90.','pfam13589','HATPase_c_3'),(980937,'gnl|CDD|215625',_binary 'Voltage-dependent potassium channel; Provisional.','PLN03192','PLN03192'),(980938,'gnl|CDD|316169',_binary 'KTSC domain.  This short domain is named after Lysine tRNA synthetase C-terminal domain. It is found at the C-terminus of some Lysyl tRNA synthetases as well as a single domain in bacterial proteins. The domain is about 60 amino acids in length and contains a reasonably conserved YXY motif in the centre of the sequence. The function of this domain is unknown but it could be an RNA binding domain.','pfam13619','KTSC'),(980946,'gnl|CDD|239293',_binary 'PDIa family, C-terminal TRX domain (a\') subfamily; composed of the C-terminal redox active a\' domains of PDI, ERp72, ERp57 (or ERp60) and EFP1. PDI, ERp72 and ERp57 are endoplasmic reticulum (ER)-resident eukaryotic proteins involved in oxidative protein folding. They are oxidases, catalyzing the formation of disulfide bonds of newly synthesized polypeptides in the ER. They also exhibit reductase activity in acting as isomerases to correct any non-native disulfide bonds, as well as chaperone activity to prevent protein aggregation and facilitate the folding of newly synthesized proteins. PDI and ERp57 have the abb\'a\' domain structure (where a and a\' are redox active TRX domains while b and b\' are redox inactive TRX-like domains). PDI also contains an acidic region (c domain) after the a\' domain that is absent in ERp57. ERp72 has an additional a domain at the N-terminus (a\'abb\'a\' domain structure). ERp57 interacts with the lectin chaperones, calnexin and calreticulin, and specifically promotes the oxidative folding of glycoproteins, while PDI shows a wider substrate specificity. ERp72 associates with several ER chaperones and folding factors to form complexes in the ER that bind nascent proteins. EFP1 is a binding partner protein of thyroid oxidase, which is responsible for the generation of hydrogen peroxide, a crucial substrate of thyroperoxidase, which functions to iodinate thyroglobulin and synthesize thyroid hormones.','cd02995','PDI_a_PDI_a\'_C'),(980962,'gnl|CDD|226685',_binary 'Thiol:disulfide interchange protein [Posttranslational modification, protein turnover, chaperones / Energy production and conversion].','COG4232','COG4232'),(980963,'gnl|CDD|259815',_binary 'Alloslipin, a subgroup of the stomatin-like proteins (slipins) familyfamily; belonging to the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes a subgroup of the stomatin-like protein family (SLPs or slipins) that is found in some eukaryotes and viruses. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. This diverse subgroup of the SLPs remains largely uncharacterized.','cd13437','SPFH_alloslipin'),(980965,'gnl|CDD|214581',_binary 'prohibitin homologues.  prohibitin homologues.','smart00244','PHB'),(980966,'gnl|CDD|239294',_binary 'PDIa family, endoplasmic reticulum protein 44 (ERp44) subfamily; ERp44 is an ER-resident protein, induced during stress, involved in thiol-mediated ER retention. It contains an N-terminal TRX domain, similar to that of PDIa, with a CXFS motif followed by two redox inactive TRX-like domains, homologous to the b and b\' domains of PDI. The CXFS motif in the N-terminal domain allows ERp44 to form stable reversible mixed disulfides with its substrates. Through this activity, ERp44 mediates the ER localization of Ero1alpha, a protein that oxidizes protein disulfide isomerases into their active form. ERp44 also prevents the secretion of unassembled cargo protein with unpaired cysteines. It also modulates the activity of inositol 1,4,5-triphosphate type I receptor (IP3R1), an intracellular channel protein that mediates calcium release from the ER to the cytosol.','cd02996','PDI_a_ERp44'),(980967,'gnl|CDD|239302',_binary 'PDIa family, C-terminal ERdj5 subfamily; ERdj5, also known as  JPDI and macrothioredoxin, is a protein containing an N-terminal DnaJ domain and four redox active TRX domains. This subfamily is composed of the three TRX domains located at the C-terminal half of the protein. ERdj5 is a ubiquitous protein localized in the endoplasmic reticulum (ER) and is abundant in secretory cells. It\'s transcription is induced during ER stress. It interacts with BiP through its DnaJ domain in an ATP-dependent manner. BiP, an ER-resident member of the Hsp70 chaperone family, functions in ER-associated degradation and protein translocation. Also included in the alignment is the single complete TRX domain of an uncharacterized protein from Tetraodon nigroviridis, which also contains a DnaJ domain at its N-terminus.','cd03004','PDI_a_ERdj5_C'),(980969,'gnl|CDD|239264',_binary 'TlpA-like family; composed of  TlpA, ResA, DsbE and similar proteins. TlpA, ResA and DsbE are bacterial protein disulfide reductases with important roles in cytochrome maturation. They are membrane-anchored proteins with a soluble TRX domain containing a CXXC motif located in the periplasm. The TRX domains of this family contain an insert, approximately 25 residues in length, which correspond to an extra alpha helix and a beta strand when compared with TRX. TlpA catalyzes an essential reaction in the biogenesis of cytochrome aa3, while ResA and DsbE are essential proteins in cytochrome c maturation. Also included in this family are proteins containing a TlpA-like TRX domain with domain architectures similar to E. coli DipZ protein, and the N-terminal TRX domain of PilB protein from Neisseria which acts as a disulfide reductase that can recylce methionine sulfoxide reductases.','cd02966','TlpA_like_family'),(980970,'gnl|CDD|239251',_binary 'DsbD gamma family; DsbD gamma is the C-terminal periplasmic domain of the bacterial protein DsbD. It contains a CXXC motif in a TRX fold and shuttles the reducing potential from the membrane domain (DsbD beta) to the N-terminal periplasmic domain (DsbD alpha).  DsbD beta, a transmembrane domain comprising of eight helices, acquires its reducing potential from the cytoplasmic thioredoxin. DsbD alpha transfers the acquired reducing potential from DsbD gamma to target proteins such as the periplasmic protein disulphide isomerases, DsbC and DsbG. This flow of reducing potential from the cytoplasm through DsbD allows DsbC and DsbG to act as isomerases in the oxidizing environment of the bacterial periplasm. DsbD also transfers reducing potential from the cytoplasm to specific reductases in the periplasm which are involved in the maturation of cytochromes.','cd02953','DsbDgamma'),(980971,'gnl|CDD|185622',_binary 'Thioredoxin domain-containing protein; Provisional.','PTZ00443','PTZ00443'),(980972,'gnl|CDD|239255',_binary 'Phosducin (Phd)-like family; composed of Phd and Phd-like proteins (PhLP), characterized as cytosolic regulators of G protein functions. Phd and PhLPs specifically bind G protein betagamma (Gbg)-subunits with high affinity, resulting in the solubilization of Gbg from the plasma membrane and impeding G protein-mediated signal transduction by inhibiting the formation of a functional G protein trimer (G protein alphabetagamma). Phd also inhibits the GTPase activity of G protein alpha. Phd can be phosphorylated by protein kinase A and G protein-coupled receptor kinase 2, leading to its inactivation. Phd was originally isolated from the retina, where it is highly expressed and has been implicated to play an important role in light adaptation. It is also found in the pineal gland, liver, spleen, striated muscle and the brain. The C-terminal domain of Phd adopts a thioredoxin fold, but it does not contain a CXXC motif. Phd interacts with G protein beta mostly through the N-terminal helical domain. Also included in this family is a PhLP characterized as a viral inhibitor of apoptosis (IAP)-associated factor, named VIAF, that functions in caspase activation during apoptosis.','cd02957','Phd_like'),(980973,'gnl|CDD|177357',_binary 'thioredoxin-like protein.','PHA02278','PHA02278'),(980975,'gnl|CDD|239297',_binary 'PDIa family, endoplasmic reticulum protein 44 (ERp44)-like subfamily; composed of uncharacterized PDI-like eukaryotic proteins containing only one redox active TRX (a) domain with a CXXS motif, similar to ERp44. CXXS is still a redox active motif; however, the mixed disulfide formed with the substrate is more stable than those formed by CXXC motif proteins. PDI-related proteins are usually involved in the oxidative protein folding in the ER by acting as catalysts and folding assistants. ERp44 is involved in thiol-mediated retention in the ER.','cd02999','PDI_a_ERp44_like'),(980976,'gnl|CDD|239309',_binary 'TlpA-like family, suppressor for copper sensitivity D protein (ScsD) and actinobacterial DsbE homolog subfamily; composed of ScsD, the DsbE homolog of Mycobacterium tuberculosis (MtbDsbE) and similar proteins, all containing a redox-active CXXC motif. The Salmonella typhimurium ScsD is a thioredoxin-like protein which confers copper tolerance to copper-sensitive mutants of E. coli. MtbDsbE has been characterized as an oxidase in vitro, catalyzing the disulfide bond formation of substrates like hirudin. The reduced form of MtbDsbE is more stable than its oxidized form, consistent with an oxidase function. This is in contrast to the function of DsbE from gram-negative bacteria which is a specific reductase of apocytochrome c.','cd03011','TlpA_like_ScsD_MtbDsbE'),(980977,'gnl|CDD|239290',_binary 'PDIa family, Quiescin-sulfhydryl oxidase (QSOX) subfamily; QSOX is a eukaryotic protein containing an N-terminal redox active TRX domain, similar to that of PDI, and a small C-terminal flavin adenine dinucleotide (FAD)-binding domain homologous to the yeast ERV1p protein. QSOX oxidizes thiol groups to disulfides like PDI, however, unlike PDI, this oxidation is accompanied by the reduction of oxygen to hydrogen peroxide. QSOX is localized in high concentrations in cells with heavy secretory load and prefers peptides and proteins as substrates, not monothiols like glutathione. Inside the cell, QSOX is found in the endoplasmic reticulum and Golgi. The flow of reducing equivalents in a QSOX-catalyzed reaction goes from the dithiol substrate -> dithiol of the QSOX TRX domain -> dithiols of the QSOX ERV1p domain -> FAD -> oxygen.','cd02992','PDI_a_QSOX'),(980978,'gnl|CDD|316422',_binary 'Thioredoxin-like.  Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond.','pfam13899','Thioredoxin_7'),(981019,'gnl|CDD|238503',_binary 'Goose Egg White Lysozyme domain. Eukaryotic \'goose-type\' lysozymes (GEWL). These enzymes catalyze the cleavage of the beta-1,4-glycosidic bond between N-acetylmuramic acid (MurNAc) and N-acetylglucosamine (GlcNAc).  Members include tunicate, Japanese flounder, ostrich, and mouse.','cd01021','GEWL'),(981027,'gnl|CDD|316589',_binary 'Domain of unknown function (DUF4258).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea and eukaryotes. Proteins in this family are typically between 95 and 124 amino acids in length.','pfam14076','DUF4258'),(981034,'gnl|CDD|225023',_binary 'Predicted Ser/Thr protein kinase [Signal transduction mechanisms].','COG2112','COG2112'),(981064,'gnl|CDD|236314',_binary 'hypothetical protein; Provisional.','PRK08624','PRK08624'),(981162,'gnl|CDD|239110',_binary 'C1 Peptidase family (MEROPS database nomenclature), also referred to as the papain family; composed of two subfamilies of cysteine peptidases (CPs), C1A (papain) and C1B (bleomycin hydrolase). Papain-like enzymes are mostly endopeptidases with some exceptions like cathepsins B, C, H and X, which are exopeptidases. Papain-like CPs have different functions in various organisms. Plant CPs are used to mobilize storage proteins in seeds while mammalian CPs are primarily lysosomal enzymes responsible for protein degradation in the lysosome. Papain-like CPs are synthesized as inactive proenzymes with N-terminal propeptide regions, which are removed upon activation. Bleomycin hydrolase (BH) is a CP that detoxifies bleomycin by hydrolysis of an amide group. It acts as a carboxypeptidase on its C-terminus to convert itself into an aminopeptidase and peptide ligase. BH is found in all tissues in mammals as well as in many other eukaryotes. It forms a hexameric ring barrel structure with the active sites imbedded in the central channel. Some members of the C1 family are proteins classified as non-peptidase homologs which lack peptidase activity or have missing active site residues.','cd02619','Peptidase_C1'),(981166,'gnl|CDD|222890',_binary 'long tail fiber, proximal subunit; Provisional.','PHA02584','34'),(981176,'gnl|CDD|132992',_binary 'Uncharacterized proteins involved in stress response, similar to tellurium resistance terD.  Tellurium resistance terD like proteins. This family is composed of uncharacterized proteins involved in stress response, such as the tellurium resistance proteins, chemical-damaging agent resistance proteins, and general stress proteins from a variety of organisms. The tellurium resistance proteins are homologous terA,-D,-E,-F,-Z,-X gene products, which confer tellurium resistance mediated by plasmids. Currently, the biochemical mechanism of tellurium resistance remains unknown. The family also contains several ter gene homologues, YceC, YceD, YceE, for which there is no clear evidence for any involvement in the tellurium resistance. A putative cAMP-binding protin CABP1 shows a significant similarity to the terD protein and is also included in this family.','cd06974','TerD_like'),(981177,'gnl|CDD|308129',_binary 'TerD domain.  The TerD domain is found in TerD family proteins that include the paralogous TerD, TerA, TerE, TerF and TerZ proteins It is found in a stress response operon with TerB and TerC. TerD has a maximum of two calcium-binding sites depending on the conservation of aspartates. It has various fusions to nuclease domains, RNA binding domains, ubiquitin related domains, and metal binding domains. The ter gene products lie at the centre of membrane-linked metal recognition complexes with regulatory ramifications encompassing phosphorylation-dependent signal transduction, RNA-dependent regulation, biosynthesis of nucleoside-like metabolites and DNA processing linked to novel pathways.','pfam02342','TerD'),(981487,'gnl|CDD|238738',_binary 'vWA_interalpha trypsin inhibitor (ITI): ITI is a glycoprotein composed of three polypeptides- two heavy chains and one light chain (bikunin). Bikunin confers the protease-inhibitor function while the heavy chains are involved in rendering stability to the extracellular matrix by binding to hyaluronic acid. The heavy chains carry the VWA domain with a conserved MIDAS motif. Although the exact role of the VWA domains remains unknown, it has been speculated to be involved in mediating protein-protein interactions with the components of the extracellular matrix.','cd01461','vWA_interalpha_trypsin_in'),(981498,'gnl|CDD|259805',_binary 'Uncharacterized family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes an uncharacterized family of proteins similar to stomatin, prohibitin, flotillin, HflK/C (SPFH) and podocin. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Many superfamily members are associated with lipid rafts. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Microdomains formed from flotillin proteins may in addition be dynamic units with their own regulatory functions. Flotillins have been implicated in signal transduction, vesicle trafficking, cytoskeleton rearrangement and are known to interact with a variety of proteins. Stomatin interacts with and regulates members of the degenerin/epithelia Na+ channel family in mechanosensory cells of Caenorhabditis elegans and vertebrate neurons and participates in trafficking of Glut1 glucose transporters. Prohibitin may act as a chaperone for the stabilization of mitochondrial proteins. Prokaryotic HflK/C plays a role in the decision between lysogenic and lytic cycle growth during lambda phage infection. Flotillins have been implicated in the progression of prion disease, in the pathogenesis of neurodegenerative diseases such as Parkinson\'s and Alzheimer\'s disease and, in cancer invasion and metastasis. Mutations in the podocin gene give rise to autosomal recessive steroid resistant nephritic syndrome.','cd03407','SPFH_like_u4'),(981501,'gnl|CDD|238736',_binary 'VWA Copine: Copines are phospholipid-binding proteins originally identified in paramecium. They are found in human and orthologues have been found in C. elegans and Arabidopsis Thaliana. None have been found in D. Melanogaster or S. Cereviciae. Phylogenetic distribution suggests that copines have been lost in some eukaryotes. No functional properties have been assigned to the VWA domains present in copines. The members of this subgroup contain a functional MIDAS motif based on their preferential binding to magnesium and manganese. However, the MIDAS motif is not totally conserved, in most cases the MIDAS consists of the sequence DxTxS instead of the motif DxSxS that is found in most cases. The C2 domains present in copines mediate phospholipid binding.','cd01459','vWA_copine_like'),(981504,'gnl|CDD|284432',_binary 'Copine.  This family represents a conserved region approximately 220 residues long within eukaryotic copines. Copines are Ca(2+)-dependent phospholipid-binding proteins that are thought to be involved in membrane-trafficking, and may also be involved in cell division and growth.','pfam07002','Copine'),(981506,'gnl|CDD|117022',_binary 'FR47-like protein.  The members of this family are similar to the C-terminal region of the D. melanogaster hypothetical protein FR47. This protein has been found to consist of two N-acyltransferase-like domains swapped with the C-terminal strands.','pfam08445','FR47'),(981516,'gnl|CDD|226648',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4185','COG4185'),(981551,'gnl|CDD|133998',_binary 'thioredoxin-like protein.','PHA02125','PHA02125'),(981573,'gnl|CDD|153102',_binary 'DPS protein, ferritin-like diiron-binding domain.  DPS (DNA Protecting protein under Starved conditions) domain is a member of a broad superfamily of ferritin-like diiron-carboxylate proteins. Some DPS proteins nonspecifically bind DNA, protecting it from cleavage caused by reactive oxygen species such as the hydroxyl radicals produced during oxidation of Fe(II) by hydrogen peroxide. These proteins assemble into dodecameric structures, some form DPS-DNA co-crystalline complexes, and possess iron and H2O2 detoxification capabilities. Expression of DPS is induced by oxidative or nutritional stress, including metal ion starvation. Members of the DPS family are homopolymers formed by 12 four-helix bundle subunits that assemble with 23 symmetry into a hollow shell. The DPS ferroxidase site is unusual in that it is not located in a four-helix bundle as in ferritin, but is shared by 2-fold symmetry-related subunits providing the iron ligands. Many DPS sequences (e.g., E. coli) display an N-terminal extension of variable length that contains two or three positively charged lysine residues that extends into the solvent and is thought to play an important role in the stabilization of the complex with DNA. DPS Listeria Flp, Bacillus anthracis Dlp-1 and Dlp-2, and Helicobacter pylori HP-NAP which lack the N-terminal extension, do not bind DNA. DPS proteins from Helicobacter pylori, Treponema pallidum, and Borrelia burgdorferi are highly immunogenic.','cd01043','DPS'),(981574,'gnl|CDD|223854',_binary 'DNA-binding ferritin-like protein (oxidative damage protectant) [Inorganic ion transport and metabolism].','COG0783','Dps'),(981576,'gnl|CDD|236521',_binary 'DNA starvation/stationary phase protection protein Dps; Provisional.','PRK09448','PRK09448'),(981577,'gnl|CDD|306676',_binary 'Ferritin-like domain.  This family contains ferritins and other ferritin-like proteins such as members of the DPS family and bacterioferritins.','pfam00210','Ferritin'),(981712,'gnl|CDD|273628',_binary 'methyltransferase, FkbM family.  Members of this family are characterized by two well-conserved short regions separated by a variable in both sequence and length. The first of the two regions is found in a large number of proteins outside this subfamily, a number of which have been characterized as methyltransferases. One member of the present family, FkbM, was shown to be required for a specific methylation in the biosynthesis of the immunosuppressant FK506 in Streptomyces strain MA6548.','TIGR01444','2-O-methyltransferase_Noe'),(981713,'gnl|CDD|309963',_binary 'Methyltransferase FkbM domain.  This family has members from bacteria to human, and appears to be a methyltransferase.','pfam05050','Methyltransf_21'),(981714,'gnl|CDD|316079',_binary 'AAA domain.','pfam13521','AAA_28'),(981737,'gnl|CDD|235316',_binary 'cell division protein MukB; Provisional.','PRK04863','mukB'),(981834,'gnl|CDD|313366',_binary 'Mu-like prophage I protein.  Members of this family of proteins comprise various viral Mu-like prophage I proteins.','pfam10123','Mu-like_Pro'),(981858,'gnl|CDD|238212',_binary 'Glutamine amidotransferases class-II (GATase). The glutaminase domain catalyzes an amide nitrogen transfer from glutamine to the appropriate substrate. In this process, glutamine is hydrolyzed to glutamic acid and ammonia. This domain is related to members of the Ntn (N-terminal nucleophile) hydrolase superfamily and is found at the N-terminus of enzymes such as glucosamine-fructose 6-phosphate synthase (GLMS or GFAT), glutamine phosphoribosylpyrophosphate (Prpp) amidotransferase (GPATase), asparagine synthetase B (AsnB), beta lactam synthetase (beta-LS) and glutamate synthase (GltS). GLMS catalyzes the formation of glucosamine 6-phosphate from fructose 6-phosphate and glutamine in amino sugar synthesis. GPATase catalyzes the first step in purine biosynthesis, an amide transfer from glutamine to PRPP, resulting in phosphoribosylamine, pyrophosphate and glutamate.  Asparagine synthetase B  synthesizes asparagine from aspartate and glutamine. Beta-LS catalyzes the formation of the beta-lactam ring in the beta-lactamase inhibitor clavulanic acid. GltS synthesizes L-glutamate from 2-oxoglutarate and L-glutamine. These enzymes are generally dimers, but GPATase also exists as a homotetramer.','cd00352','Gn_AT_II'),(981859,'gnl|CDD|238366',_binary 'Glutamine amidotransferases class-II (Gn-AT)_GFAT-type. This domain is found at the N-terminus of glucosamine-6P synthase (GlmS, or GFAT in humans).  The glutaminase domain catalyzes amide nitrogen transfer from glutamine to the appropriate substrate. In this process, glutamine is hydrolyzed to glutamic acid and ammonia. In humans, GFAT catalyzes the first and rate-limiting step of hexosamine metabolism, the conversion of D-fructose-6P (Fru6P) into D-glucosamine-6P using L-glutamine as a nitrogen source.  The end product of this pathway, UDP-N-acetyl glucosamine, is a major building block of the bacterial peptidoglycan and fungal chitin.','cd00714','GFAT'),(981860,'gnl|CDD|234729',_binary 'glucosamine--fructose-6-phosphate aminotransferase; Reviewed.','PRK00331','PRK00331'),(981861,'gnl|CDD|223526',_binary 'Glucosamine 6-phosphate synthetase, contains amidotransferase and phosphosugar isomerase domains [Cell envelope biogenesis, outer membrane].','COG0449','GlmS'),(981862,'gnl|CDD|273462',_binary 'glucosamine--fructose-6-phosphate aminotransferase (isomerizing).  The member from Methanococcus jannaschii contains an intein. [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan, Central intermediary metabolism, Amino sugars].','TIGR01135','glmS'),(981863,'gnl|CDD|173585',_binary 'glucosamine-fructose-6-phosphate aminotransferase; Provisional.','PTZ00394','PTZ00394'),(981864,'gnl|CDD|223199',_binary 'Predicted glutamine amidotransferase [General function prediction only].','COG0121','COG0121'),(981865,'gnl|CDD|238889',_binary 'Glutamine amidotransferases class-II (Gn-AT)_YafJ-type.  YafJ is a glutamine amidotransferase-like protein of unknown function found in prokaryotes, eukaryotes and archaea.  YafJ has a conserved structural fold similar to those of other class II glutamine amidotransferases including lucosamine-fructose 6-phosphate synthase (GLMS or GFAT), glutamine phosphoribosylpyrophosphate (Prpp) amidotransferase (GPATase),  asparagine synthetase B (AsnB), beta lactam synthetase (beta-LS) and glutamate synthase (GltS).  The YafJ fold is also somwhat similar to the Ntn (N-terminal nucleophile) hydrolase fold of the proteasomal alpha and beta subunits.','cd01908','YafJ'),(981888,'gnl|CDD|197761',_binary 'Intron encoded nuclease repeat motif.  Repeat of unknown function, but possibly DNA-binding via helix-turn-helix motif (Ponting, unpublished).','smart00497','IENR1'),(982049,'gnl|CDD|113356',_binary 'Reovirus sigma C capsid protein.','pfam04582','Reo_sigmaC'),(982073,'gnl|CDD|223403',_binary 'Molecular chaperone, HSP90 family [Posttranslational modification, protein turnover, chaperones].','COG0326','HtpG'),(982075,'gnl|CDD|223400',_binary 'DNA mismatch repair enzyme (predicted ATPase) [DNA replication, recombination, and repair].','COG0323','MutL'),(982076,'gnl|CDD|214643',_binary 'Histidine kinase-like ATPases.  Histidine kinase-, DNA gyrase B-, phytochrome-like ATPases.','smart00387','HATPase_c'),(982077,'gnl|CDD|234630',_binary 'DNA mismatch repair protein; Reviewed.','PRK00095','mutL'),(982078,'gnl|CDD|273155',_binary 'DNA mismatch repair protein MutL.  All proteins in this family for which the functions are known are involved in the process of generalized mismatch repair. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00585','DNA_mismatch_repair_prote'),(982107,'gnl|CDD|307611',_binary 'Transposase.  Transposase proteins are necessary for efficient DNA transposition. This family includes an amino-terminal region of the pilin gene inverting protein (PIVML) and of members of the IS111A/IS1328/IS1533 family of transposases. The C-terminus is represented by family pfam02371.','pfam01548','DEDD_Tnp_IS110'),(982108,'gnl|CDD|308145',_binary 'Transposase IS116/IS110/IS902 family.  Transposases are needed for efficient transposition of the insertion sequence or transposon DNA. This family includes transposases for IS116, IS110 and IS902. This region is often found with pfam01548. The exact function of this region is uncertain. This family contains a HHH motif suggesting a DNA-binding function.','pfam02371','Transposase_20'),(982109,'gnl|CDD|226077',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG3547','COG3547'),(982128,'gnl|CDD|225718',_binary 'Fic family protein [Function unknown].','COG3177','COG3177'),(982130,'gnl|CDD|308340',_binary 'Fic/DOC family.  This family consists of the Fic (filamentation induced by cAMP) protein and doc (death on curing). The Fic protein is involved in cell division and is suggested to be involved in the synthesis of PAB or folate, indicating that the Fic protein and cAMP are involved in a regulatory mechanism of cell division via folate metabolism. This family contains a central conserved motif HPFXXGNG in most members. The exact molecular function of these proteins is uncertain. P1 lysogens of Escherichia coli carry the prophage as a stable low copy number plasmid. The frequency with which viable cells cured of prophage are produced is about 10(-5) per cell per generation. A significant part of this remarkable stability can be attributed to a plasmid-encoded mechanism that causes death of cells that have lost P1. In other words, the lysogenic cells appear to be addicted to the presence of the prophage. The plasmid withdrawal response depends on a gene named doc (death on curing) that is represented by this family. Doc induces a reversible growth arrest of E. coli cells by targetting the protein synthesis machinery. Doc hosts the C-terminal domain of its antitoxin partner Phd (prevents host death) through fold complementation, a domain that is intrinsically disordered in solution but that folds into an alpha-helix on binding to Doc.This domain forms complexes with Phd antitoxins containing pfam02604.','pfam02661','Fic'),(982150,'gnl|CDD|227091',_binary 'Uncharacterized conserved protein [Function unknown].','COG4748','COG4748'),(982152,'gnl|CDD|225367',_binary 'Predicted type IV restriction endonuclease [Defense mechanisms].','COG2810','COG2810'),(982155,'gnl|CDD|316142',_binary 'Type I restriction enzyme R protein N terminus (HSDR_N).  This family consists of a number of N terminal regions found in type I restriction enzyme R (HSDR) proteins. Restriction and modification (R/M) systems are found in a wide variety of prokaryotes and are thought to protect the host bacterium from the uptake of foreign DNA. Type I restriction and modification systems are encoded by three genes: hsdR, hsdM, and hsdS. The three polypeptides, HsdR, HsdM, and HsdS, often assemble to give an enzyme (R2M2S1) that modifies hemimethylated DNA and restricts unmethylated DNA.','pfam13588','HSDR_N_2'),(982158,'gnl|CDD|309446',_binary 'Type I restriction enzyme R protein N terminus (HSDR_N).  This family consists of a number of N terminal regions found in type I restriction enzyme R (HSDR) proteins. Restriction and modification (R/M) systems are found in a wide variety of prokaryotes and are thought to protect the host bacterium from the uptake of foreign DNA. Type I restriction and modification systems are encoded by three genes: hsdR, hsdM, and hsdS. The three polypeptides, HsdR, HsdM, and HsdS, often assemble to give an enzyme (R2M2S1) that modifies hemimethylated DNA and restricts unmethylated DNA.','pfam04313','HSDR_N'),(982226,'gnl|CDD|314670',_binary 'Alanine-zipper, major outer membrane lipoprotein.  This is a family of a major outer membrane lipoprotein, OprL that is an alanine-zipper. Zipper motifs are a seven-repeat motif where the first and fourth positions are occupied by an aliphatic residue, usually a leucine. These residues are positioned on the outside of the coil such as to bind firmly to one or more monomers of the protein to create a triple or five-helical coiled-coil that probably forms a seam in a membrane.','pfam11839','Alanine_zipper'),(982608,'gnl|CDD|224242',_binary 'Predicted nucleotidyltransferase [General function prediction only].','COG1323','COG1323'),(982609,'gnl|CDD|234912',_binary 'phosphopantetheine adenylyltransferase/unknown domain fusion protein; Provisional.','PRK01170','PRK01170'),(982653,'gnl|CDD|106974',_binary 'major capsid protein.','PHA00665','PHA00665'),(982706,'gnl|CDD|313343',_binary 'Predicted membrane protein (DUF2335).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10097','DUF2335'),(982707,'gnl|CDD|227650',_binary 'Predicted membrane protein [Function unknown].','COG5346','COG5346'),(982733,'gnl|CDD|317102',_binary 'AP2-like DNA-binding integrase domain.  This family includes AP2-like domains found in a variety of phage integrase proteins. Presumably these domains are DNA-binding.','pfam14657','Integrase_AP2'),(982825,'gnl|CDD|165252',_binary 'putative transposase; Provisional.','PHA02942','PHA02942'),(982936,'gnl|CDD|180071',_binary '50S ribosomal protein L1; Validated.','PRK05424','rplA'),(982967,'gnl|CDD|308664',_binary 'LAGLIDADG DNA endonuclease family.  This is a family of site-specific DNA endonucleases encoded by DNA mobile elements. Similar to pfam00961, the members of this family are also LAGLIDADG endonucleases.','pfam03161','LAGLIDADG_2'),(983029,'gnl|CDD|224136',_binary 'Glycosyltransferases, probably involved in cell wall biogenesis [Cell envelope biogenesis, outer membrane].','COG1215','COG1215'),(983043,'gnl|CDD|237985',_binary 'DNA-binding domain found in transcription regulators in plants such as APETALA2 and EREBP (ethylene responsive element binding protein). In EREBPs the domain specifically binds to the 11bp GCC box of the ethylene response element (ERE), a promotor element essential for ethylene responsiveness. EREBPs and the C-repeat binding factor CBF1, which is involved in stress response, contain a single copy of the AP2 domain. APETALA2-like proteins, which play a role in plant  development contain two copies.','cd00018','AP2'),(983044,'gnl|CDD|312331',_binary 'CotH protein.  Members of this family include the spore coat protein H (cotH).','pfam08757','CotH'),(983048,'gnl|CDD|226487',_binary 'Uncharacterized protein contain chitin-binding domain type 3 [General function prediction only].','COG3979','COG3979'),(983224,'gnl|CDD|311012',_binary 'Prophage tail length tape measure protein.  This family represents a conserved region located towards the N-terminal end of prophage tail length tape measure protein (TMP). TMP is important for assembly of phage tails and involved in tail length determination. Mutated forms TMP cause tail fibres to be shortened.','pfam06791','TMP_2'),(983230,'gnl|CDD|274228',_binary 'addiction module antidote protein, HigA family.  Members of this family form a distinct clade within the larger family HTH_3 of helix-turn-helix proteins, described by pfam01381. Members of this clade are strictly bacterial and nearly always shorter than 110 amino acids. This family includes the characterized member HigA, without which the killer protein HigB cannot be cloned. The hig (host inhibition of growth) system is noted to be unusual in that killer protein is uncoded by the upstream member of the gene pair. [Regulatory functions, DNA interactions, Regulatory functions, Protein interactions, Mobile and extrachromosomal element functions, Other].','TIGR02607','Virulence-associated_prot'),(983317,'gnl|CDD|308260',_binary 'D-alanyl-D-alanine carboxypeptidase.','pfam02557','VanY'),(983325,'gnl|CDD|224313',_binary 'Predicted transcriptional regulator [Transcription].','COG1395','COG1395'),(983442,'gnl|CDD|179690',_binary 'putative transcriptional regulator; Provisional.','PRK03975','tfx'),(983594,'gnl|CDD|318671',_binary 'Leucine zipper of domain CC2 of NEMO, NF-kappa-B essential modulator.  CC2-LZ is a leucine-zipper domain associated with the CC2 coiled-coil region of NF-kappa-B essential modulator, NEMO. It plays a regulatory role, along with the very C-terminal zinc-finger; it contains a ubiquitin-binding domain (UBD) and represents one region that contributes to NEMO oligomerization. NEMO itself is an integral part of the IkappaB kinase complex and serves as a molecular switch via which the NF-kappaB signalling pathway is regulated.','pfam16516','CC2-LZ'),(983660,'gnl|CDD|238073',_binary 'Dual specificity phosphatases (DSP); Ser/Thr and Tyr protein phosphatases. Structurally similar to tyrosine-specific phosphatases but with a shallower active site cleft and a distinctive active site signature motif, HCxxGxxR. Characterized as VHR- or Cdc25-like.','cd00127','DSPc'),(983661,'gnl|CDD|214551',_binary 'Dual specificity phosphatase, catalytic domain.','smart00195','DSPc'),(983662,'gnl|CDD|225297',_binary 'Predicted protein-tyrosine phosphatase [Signal transduction mechanisms].','COG2453','CDC14'),(983663,'gnl|CDD|307089',_binary 'Dual specificity phosphatase, catalytic domain.  Ser/Thr and Tyr protein phosphatases. The enzyme\'s tertiary fold is highly similar to that of tyrosine-specific phosphatases, except for a \'recognition\' region.','pfam00782','DSPc'),(983664,'gnl|CDD|214649',_binary 'Protein tyrosine phosphatase, catalytic domain motif.','smart00404','PTPc_motif'),(983665,'gnl|CDD|214469',_binary 'Protein tyrosine phosphatase, catalytic domain, undefined specificity.  Protein tyrosine phosphatases. Homologues detected by this profile and not by those of \'PTPc\' or \'DSPc\' are predicted to be protein phosphatases with a similar fold to DSPs and PTPs, yet with unpredicted specificities.','smart00012','PTPc_DSPc'),(983666,'gnl|CDD|225240',_binary 'Protein tyrosine/serine phosphatase [Signal transduction mechanisms].','COG2365','COG2365'),(983843,'gnl|CDD|188267',_binary 'RNA polymerase sigma-70 factor, Myxococcales family 1.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are found in multiple copies in the order Myxococcales. This model supercedes TIGR02233, which has now been retired.','TIGR03001','Sig-70_gmx1'),(983867,'gnl|CDD|223498',_binary 'Spermidine synthase [Amino acid transport and metabolism].','COG0421','SpeE'),(983877,'gnl|CDD|234786',_binary 'ribosomal protein L11 methyltransferase; Reviewed.','PRK00517','prmA'),(983885,'gnl|CDD|225317',_binary 'tRNA(1-methyladenosine) methyltransferase and related methyltransferases [Translation, ribosomal structure and biogenesis].','COG2519','GCD14'),(983925,'gnl|CDD|238869',_binary 'Endoglucanase E-like members of the SGNH hydrolase family; Endoglucanase E catalyzes the endohydrolysis of 1,4-beta-glucosidic linkages in cellulose, lichenin and cereal beta-D-glucans.','cd01831','Endoglucanase_E_like'),(984003,'gnl|CDD|308511',_binary 'Phage integrase, N-terminal SAM-like domain.','pfam02899','Phage_int_SAM_1'),(984156,'gnl|CDD|275271',_binary 'radical SAM/CxCxxxxC motif protein YfkAB.  Members of this highly conserved family in some Firmicutes have an N-terminal radical SAM domain (pfam04055) and a C-terminal pfam08756 domain with a CxCxxxxC motif that suggests binding to an additional metallocluster. It appears all correct sequences in this family are about 370 amino acids in length, containing the YfkA and YfkB regions originally reported as separate ORFs in Bacillus subtilis. Partial Phylogenetic Profiling shows occurrences almost exclusively in the Bacilli, with very few examples of either lateral transfer or gene loss. The essentially monophyletic distribution suggests a housekeeping function. Members have no well-conserved gene neighborhood. The function is unknown. [Unknown function, Enzymes of unknown specificity].','TIGR04478','radical_SAM_superfamily_d'),(984948,'gnl|CDD|236941',_binary 'ATP-dependent RNA helicase DeaD; Provisional.','PRK11634','PRK11634'),(984950,'gnl|CDD|236877',_binary 'ATP-dependent RNA helicase SrmB; Provisional.','PRK11192','PRK11192'),(984954,'gnl|CDD|236722',_binary 'ATP-dependent RNA helicase RhlE; Provisional.','PRK10590','PRK10590'),(984960,'gnl|CDD|273162',_binary 'primosomal protein N\'.  All proteins in this family for which functions are known are components of the primosome which is involved in replication, repair, and recombination.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00595','Primosomal_protein_N\''),(984984,'gnl|CDD|319837',_binary 'vegetative storage proteins similar to soybean VSPalpha and VSPbeta proteins; belongs to the haloacid dehalogenase-like superfamily.  Soybean [Glycine max (L.) Merr.] vegetative storage protein VSPalpha and VSPbeta levels were identified as storage proteins due to their abundance and pattern of expression in plant tissues, they accumulate to almost one-half the amount of soluble leaf protein when soybean plants are continually depodded. They possess acid phosphatase activity which appears to be low compared to several other plant acid phosphatases, it increases in the leaves of depodded soybean plants, but to no more than 0.1% of the total acid phosphatase activity in these leaves. This acid phosphatase activity has maximal activity at pH 5.0 - 5.5, and can liberate Pi from different substrates such as napthyl acid phosphate, carboxyphenyl phosphate, sugar-phosphates, glyceraldehyde 3-phosphate, dihydroxyacetone phosphate, phosphoenolpyruvate, ATP, ADP, PPi, and short chain polyphosphates; they cleave phosphoenolpyruvate, ATP, ADP, PPI, and polyphosphates most efficiently. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Soybean VSPalpha and VSPbeta lack this active site aspartate, other members of this family have this aspartate and may be more active.  Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd07535','HAD_VSP'),(984989,'gnl|CDD|307699',_binary 'IstB-like ATP binding protein.  This protein contains an ATP/GTP binding P-loop motif. It is found associated with IS21 family insertion sequences. The function of this protein is unknown, but it may perform a transposase function.','pfam01695','IstB_IS21'),(985000,'gnl|CDD|226031',_binary 'Phage protein D [General function prediction only].','COG3500','COG3500'),(985001,'gnl|CDD|311038',_binary 'T4-like virus tail tube protein gp19.  This family consists of several tail tube protein gp19 sequences from the T4-like viruses. This family also contains bacterial members which suggest lateral transfer of genes.','pfam06841','Phage_T4_gp19'),(985002,'gnl|CDD|274049',_binary 'conserved hypothetical phage tail region protein.  This family consists of uncharacterized proteins. All members so far represent bacterial genes found in apparent phage or otherwisely laterally transferred regions of the chromosome. Tentatively identified neighboring proteins tend to be phage tail region proteins. In some species, including Photorhabdus luminescens TTO1, several members of this family may be encoded near each other.','TIGR02241','hypothetical_protein_SCD8'),(985008,'gnl|CDD|226028',_binary 'Phage tail sheath protein FI [General function prediction only].','COG3497','COG3497'),(985009,'gnl|CDD|222862',_binary 'tail sheath protein; Provisional.','PHA02539','18'),(985048,'gnl|CDD|239308',_binary 'TlpA-like family, DsbE (also known as CcmG and CycY) subfamily; DsbE is a membrane-anchored, periplasmic TRX-like reductase containing a CXXC motif that specifically donates reducing equivalents to apocytochrome c via CcmH, another cytochrome c maturation (Ccm) factor with a redox active CXXC motif. Assembly of cytochrome c requires the ligation of heme to reduced thiols of the apocytochrome. In bacteria, this assembly occurs in the periplasm. The reductase activity of DsbE in the oxidizing environment of the periplasm is crucial in the maturation of cytochrome c.','cd03010','TlpA_like_DsbE'),(985053,'gnl|CDD|129481',_binary 'periplasmic protein thiol:disulfide oxidoreductases, DsbE subfamily.  Involved in the biogenesis of c-type cytochromes as well as in disulfide bond formation in some periplasmic proteins. [Protein fate, Protein folding and stabilization].','TIGR00385','Thiol:disulfide_interchan'),(985054,'gnl|CDD|239298',_binary 'PDIa family, TMX3 subfamily; composed of eukaryotic proteins similar to human TMX3, a TRX related transmembrane protein containing one redox active TRX domain at the N-terminus and a classical ER retrieval sequence for type I transmembrane proteins at the C-terminus. The TMX3 transcript is found in a variety of tissues with the highest levels detected in skeletal muscle and the heart. In vitro, TMX3 showed oxidase activity albeit slightly lower than that of protein disulfide isomerase.','cd03000','PDI_a_TMX3'),(985056,'gnl|CDD|239248',_binary 'TRX-like protein A (TxlA) family; TxlA was originally isolated from the cyanobacterium Synechococcus. It is found only in oxygenic photosynthetic organisms. TRX is a small enzyme that participate in redox reactions, via the reversible oxidation of an active site dithiol present in a CXXC motif. Disruption of the txlA gene suggests that the protein is involved in the redox regulation  of the structure and function of photosynthetic apparatus. The plant homolog (designated as HCF164) is localized in the chloroplast and is involved in the assembly of the cytochrome b6f complex, which takes a central position in photosynthetic electron transport.','cd02950','TxlA'),(985060,'gnl|CDD|239260',_binary 'TMX2 family; composed of proteins similar to human TMX2, a 372-amino acid TRX-related transmembrane protein, identified and characterized through the cloning of its cDNA from a human fetal library. It contains a TRX domain but the redox active CXXC motif is replaced with SXXC. Sequence analysis predicts that TMX2 may be a Type I membrane protein, with its C-terminal half protruding on the luminal side of the endoplasmic reticulum (ER). In addition to the TRX domain, transmembrane region and ER-retention signal, TMX2 also contains a Myb DNA-binding domain repeat signature and a dileucine motif in the tail.','cd02962','TMX2'),(985062,'gnl|CDD|316427',_binary 'Thioredoxin-like.  Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond.','pfam13905','Thioredoxin_8'),(985064,'gnl|CDD|239307',_binary 'Tryparedoxin (TryX)-like family, TryX and nucleoredoxin (NRX) subfamily; TryX and NRX are thioredoxin (TRX)-like protein disulfide oxidoreductases that alter the redox state of target proteins via the reversible oxidation of an active center CXXC motif. TryX is involved in the regulation of oxidative stress in parasitic trypanosomatids by reducing TryX peroxidase, which in turn catalyzes the reduction of hydrogen peroxide and organic hydroperoxides. TryX derives reducing equivalents from reduced trypanothione, a polyamine peptide conjugate unique to trypanosomatids, which is regenerated by the NADPH-dependent flavoprotein trypanothione reductase. Vertebrate NRX is a 400-amino acid nuclear protein with one redox active TRX domain containing a CPPC active site motif followed by one redox inactive TRX-like domain. Mouse NRX transcripts are expressed in all adult tissues but is restricted to the nervous system and limb buds in embryos. Plant NRX, longer than the vertebrate NRX by about 100-200 amino acids, is a nuclear protein containing a redox inactive TRX-like domain between two redox active TRX domains. Both vertebrate and plant NRXs show thiol oxidoreductase activity in vitro. Their localization in the nucleus suggests a role in the redox regulation of nuclear proteins such as transcription factors.','cd03009','TryX_like_TryX_NRX'),(985065,'gnl|CDD|179545',_binary 'thiol-disulfide oxidoreductase; Provisional.','PRK03147','PRK03147'),(985066,'gnl|CDD|317049',_binary 'Thioredoxin.','pfam14595','Thioredoxin_9'),(985068,'gnl|CDD|239271',_binary 'Thioredoxin (TRX)-Glutaredoxin (GRX)-like family; composed of archaeal and bacterial proteins that show similarity to both TRX and GRX, including the C-terminal TRX-fold subdomain of Pyrococcus furiosus protein disulfide oxidoreductase (PfPDO). All members contain a redox-active CXXC motif and may function as PDOs. The archaeal proteins Mj0307 and Mt807 show structures more similar to GRX, but activities more similar to TRX. Some members of the family are similar to PfPDO in that they contain a second CXXC motif located in a second TRX-fold subdomain at the N-terminus; the superimposable N- and C-terminal TRX subdomains form a compact structure. PfPDO is postulated to be the archaeal counterpart of bacterial DsbA and eukaryotic protein disulfide isomerase (PDI). The C-terminal CXXC motif of PfPDO is required for its oxidase, reductase and isomerase activities. Also included in the family is the C-terminal TRX-fold subdomain of the N-terminal domain (NTD) of bacterial AhpF, which has a similar fold as PfPDO with two TRX-fold subdomains but without the second CXXC motif.','cd02973','TRX_GRX_like'),(985070,'gnl|CDD|185310',_binary 'thiol:disulfide interchange protein DsbE; Provisional.','PRK15412','PRK15412'),(985074,'gnl|CDD|184456',_binary 'trifunctional thioredoxin/methionine sulfoxide reductase A/B protein; Provisional.','PRK14018','PRK14018'),(985075,'gnl|CDD|306949',_binary 'AhpC/TSA family.  This family contains proteins related to alkyl hydroperoxide reductase (AhpC) and thiol specific antioxidant (TSA).','pfam00578','AhpC-TSA'),(985078,'gnl|CDD|225192',_binary 'Uncharacterized proteins involved in stress response, homologs of TerZ and putative cAMP-binding protein CABP1 [Signal transduction mechanisms].','COG2310','TerZ'),(985082,'gnl|CDD|279710',_binary 'Ribbon-helix-helix protein, copG family.  The structure of this protein repressor, which is the shortest reported to date and the first isolated from a plasmid, has a homodimeric ribbon-helix-helix arrangement. The helix-turn-helix-like structure is involved in dimerisation and not DNA binding as might have been expected.','pfam01402','RHH_1'),(985104,'gnl|CDD|282135',_binary 'Stage V sporulation protein S (SpoVS).  In Bacillus subtilis this protein interferes with sporulation at an early stage and this inhibitory effect is overcome by SpoIIB and SpoVG. SpoVS seems to play a positive role in allowing progression beyond stage V of sporulation. Null mutations in the spoVS gene block sporulation at stage V, impairing the development of heat resistance and coat assembly.','pfam04232','SpoVS'),(985105,'gnl|CDD|225234',_binary 'Stage V sporulation protein SpoVS [Function unknown].','COG2359','SpoVS'),(985130,'gnl|CDD|222869',_binary 'single-stranded DNA binding protein; Provisional.','PHA02550','32'),(985133,'gnl|CDD|211905',_binary 'ribonucleoside-diphosphate reductase, class 1b, alpha subunit.  Members of this family are NrdE, the alpha subunit of class 1b ribonucleotide reductase. This form uses a dimanganese moiety associated with a tyrosine radical to reduce the cellular requirement for iron.','TIGR04170','RNR_1b_NrdE'),(985140,'gnl|CDD|235952',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07187','PRK07187'),(985141,'gnl|CDD|236069',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07632','PRK07632'),(985147,'gnl|CDD|178056',_binary 'ribonucleoside--diphosphate reductase large subunit.','PLN02437','PLN02437'),(985149,'gnl|CDD|181700',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK09209','PRK09209'),(985151,'gnl|CDD|237077',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Provisional.','PRK12364','PRK12364'),(985157,'gnl|CDD|236266',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK08447','PRK08447'),(985161,'gnl|CDD|222883',_binary 'ribonucleoside-diphosphate reductase subunit alpha; Provisional.','PHA02572','nrdA'),(985165,'gnl|CDD|236179',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK08188','PRK08188'),(985167,'gnl|CDD|180921',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07306','PRK07306'),(985169,'gnl|CDD|181649',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK09103','PRK09103'),(986426,'gnl|CDD|129780',_binary 'conserved hypothetical integral membrane protein.  All known members of this family are proteins or 210-250 amino acids in length. Conserved regions of hydrophobicity suggest that all members of the family are integral membrane proteins. [Hypothetical proteins, Conserved].','TIGR00697','Uncharacterized_protein_H'),(986427,'gnl|CDD|308287',_binary 'Putative vitamin uptake transporter.','pfam02592','Vut_1'),(986428,'gnl|CDD|224652',_binary 'Uncharacterized member of the PurR regulon [General function prediction only].','COG1738','yhhQ'),(986429,'gnl|CDD|183041',_binary 'hypothetical protein; Provisional.','PRK11212','PRK11212'),(986442,'gnl|CDD|213285',_binary 'The adenylation (A) domain of siderophore-synthesizing nonribosomal peptide synthetases (NRPS).  The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. This family of siderophore-synthesizing NRPS includes the third adenylation domain of SidN from the endophytic fungus Neotyphodium lolii, ferrichrome siderophore synthetase, HC-toxin synthetase, and enniatin synthase. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.','cd05918','A_NRPS_SidN3_like'),(986443,'gnl|CDD|227020',_binary 'Microcystin-dependent protein [Function unknown].','COG4675','MdpB'),(986444,'gnl|CDD|311431',_binary 'Phage Tail Collar Domain.  This region is occasionally found in conjunction with pfam03335. Most of the family appear to be phage tail proteins; however some appear to be involved in other processes. For instance a member from Rhizobium leguminosarum may be involved in plant-microbe interactions. A related protein MrpB is involved in the pathogenicity of Microcystis aeruginosa. The finding of this family in a structural component of the phage tail fibre baseplate suggests that its function is structural rather than enzymatic. Structural studies show this region consists of a helix and a loop and three beta-strands. This alignment does not catch the third strand as it is separated from the rest of the structure by around 100 residues. This strand is conserved in homologs but the intervening sequence is not. Much of the function of phage T4 short tail fiber protein gp12 appears to reside in this intervening region. In the tertiary structure of the phage baseplate this domain forms part of the \'collar\'. The domain may bind SO4, however the residues accredited with this vary between the PDB file and the Swiss-Prot entry. The long unconserved region maybe due to domain swapping in and out of a loop or reflective of rapid evolution.','pfam07484','Collar'),(986469,'gnl|CDD|235224',_binary 'hypothetical protein; Provisional.','PRK04140','PRK04140'),(986477,'gnl|CDD|133385',_binary 'Helix-Turn-Helix DNA binding domain of the SoxR transcription regulator.  Helix-turn-helix (HTH) transcriptional regulator SoxR. The global regulator, SoxR, up-regulates gene expression of another transcription activator, SoxS, which directly stimulates the oxidative stress regulon genes in E. coli. The soxRS response renders the bacterial cell resistant to superoxide-generating agents, macrophage-generated nitric oxide, organic solvents, and antibiotics. The SoxR proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the unusually long spacer between the -35 and -10 promoter elements. They also harbor a regulatory C-terminal domain containing an iron-sulfur center.','cd01110','HTH_SoxR'),(986479,'gnl|CDD|133398',_binary 'Helix-Turn-Helix DNA binding domain of Heavy Metal Resistance transcription regulators.  Helix-turn-helix (HTH) heavy metal resistance transcription regulators (HMRTR): MerR1 (mercury), CueR (copper),  CadR (cadmium),  PbrR (lead), ZntR (zinc), and other related proteins. These transcription regulators mediate responses to heavy metal stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04770','HTH_HMRTR'),(986480,'gnl|CDD|133379',_binary 'Helix-Turn-Helix DNA binding domain of the transcription regulators MlrA and CarA.  Helix-turn-helix (HTH) transcription regulator MlrA (merR-like regulator A), N-terminal domain. The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium.  Its close homolog, CarA from Myxococcus xanthus, is involved in activation of the carotenoid biosynthesis genes by light. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules. Many MlrA- and CarA-like proteins in this group appear to lack the long dimerization helix seen in the N-terminal domains of typical MerR-like proteins.','cd01104','HTH_MlrA-CarA'),(986629,'gnl|CDD|237798',_binary 'DNA polymerase II large subunit; Provisional.','PRK14714','PRK14714'),(986633,'gnl|CDD|223788',_binary 'Flavodoxins [Energy production and conversion].','COG0716','FldA'),(986634,'gnl|CDD|179443',_binary 'flavoprotein NrdI; Provisional.','PRK02551','PRK02551'),(986635,'gnl|CDD|311777',_binary 'NrdI Flavodoxin like.','pfam07972','Flavodoxin_NdrI'),(986636,'gnl|CDD|188042',_binary 'ribonucleoside-diphosphate reductase 2, operon protein nrdI.  Ribonucleotide reductases (RNRs) are enzymes that provide the precursors of DNA synthesis. The three characterized classes of RNRs differ by their metal cofactor and their stable organic radical. The exact function of nrdI within the ribonucleotide reductases has not yet been fully characterised. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR00333','Protein_NrdI'),(986694,'gnl|CDD|223686',_binary 'Predicted metal-dependent phosphoesterases (PHP family) [General function prediction only].','COG0613','COG0613'),(986713,'gnl|CDD|224119',_binary 'Primosomal protein N\' (replication factor Y) - superfamily II helicase [DNA replication, recombination, and repair].','COG1198','PriA'),(986779,'gnl|CDD|311999',_binary 'Ribonucleotide reductase N-terminal.  This domain is found at the N-terminus of bacterial ribonucleoside-diphosphate reductases (ribonucleotide reductases, RNRs) which catalyze the formation of deoxyribonucleotides. It occurs together with the RNR all-alpha domain (pfam00317) and the RNR barrel domain (pfam02867).','pfam08343','RNR_N'),(986816,'gnl|CDD|212147',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from bacteria and archaea.  This family corresponds to a group of uncharacterized hypothetical proteins from bacteria and archaea, showing a high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain that contains a well conserved divalent ion-binding motif EXX[E/D].','cd11540','NTP-PPase_u3'),(987226,'gnl|CDD|306655',_binary 'Dihydrofolate reductase.','pfam00186','DHFR_1'),(987227,'gnl|CDD|238127',_binary 'Dihydrofolate reductase (DHFR). Reduces 7,8-dihydrofolate to 5,6,7,8-tetrahydrofolate with NADPH as a cofactor. This is an essential step in the biosynthesis of deoxythymidine phosphate since 5,6,7,8-tetrahydrofolate is required to regenerate 5,10-methylenetetrahydrofolate which is then utilized by thymidylate synthase. Inhibition of DHFR interrupts thymidilate synthesis and DNA replication, inhibitors of DHFR (such as Methotrexate) are used in cancer chemotherapy.  5,6,7,8-tetrahydrofolate also is involved in glycine, serine, and threonine metabolism and aminoacyl-tRNA biosynthesis.','cd00209','DHFR'),(987228,'gnl|CDD|223340',_binary 'Dihydrofolate reductase [Coenzyme metabolism].','COG0262','FolA'),(987229,'gnl|CDD|240299',_binary 'bifunctional dihydrofolate reductase-thymidylate synthase; Provisional.','PTZ00164','PTZ00164'),(987230,'gnl|CDD|182714',_binary 'dihydrofolate reductase; Provisional.','PRK10769','folA'),(987231,'gnl|CDD|234776',_binary 'segregation and condensation protein A/unknown domain fusion protein; Provisional.','PRK00478','scpA'),(987241,'gnl|CDD|129841',_binary 'uracil-DNA glycosylase, family 4.  This well-conserved family of proteins is about 200 residues in length and homologous to the N-terminus of the DNA polymerase of phage SPO1 of Bacillus subtilis. The member from Thermus thermophilus HB8 is known to act as uracil-DNA glycosylase, an enzyme of DNA base excision repair. Its appearance as a domain of phage DNA polymerases could be consistent with uracil-DNA glycosylase activity. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00758','UDG_fam4'),(987242,'gnl|CDD|308668',_binary 'Uracil DNA glycosylase superfamily.','pfam03167','UDG'),(987243,'gnl|CDD|224489',_binary 'Uracil-DNA glycosylase [DNA replication, recombination, and repair].','COG1573','COG1573'),(987244,'gnl|CDD|198432',_binary 'Uncharacterized subfamily of Uracil-DNA glycosylases.  This is a subfamily of Uracil-DNA glycosylase superfamily. Uracil-DNA glycosylases (UDG) catalyze the removal of uracil from DNA to initiate DNA base excision repair pathway. Uracil in DNA can arise as a result of mis-incorporation of dUMP residues by DNA polymerase or deamination of cytosine. Uracil mispaired with guanine in DNA is one of the major pro-mutagenic events, causing G:C->A:T mutations. UDG is an essential enzyme for maintaining the integrity of genetic information. This ubiquitously found enzyme hydrolyzes the N-glycosidic bond of deoxyuridine in DNA.','cd10034','UDG_like_2'),(987245,'gnl|CDD|214956',_binary 'Uracil DNA glycosylase superfamily.','smart00986','UDG'),(987246,'gnl|CDD|214957',_binary 'UreE urease accessory protein, C-terminal domain.  UreE is a urease accessory protein. Urease hydrolyses urea into ammonia and carbamic acid. The C-terminal region of members of this family contains a His rich Nickel binding site.','smart00987','UreE_C'),(987247,'gnl|CDD|198424',_binary 'Uracil-DNA glycosylases (UDG) and related enzymes.  Uracil-DNA glycosylases (UDG) catalyzes the removal of uracil from DNA, which initiates the DNA base excision repair pathway. Uracil in DNA can arise as a result of mis-incorporation of dUMP residues by DNA polymerase or via deamination of cytosine. Uracil in DNA mispaired with guanine is one of the major pro-mutagenic events, causing G:C->A:T mutations. Thus, UDG is an essential enzyme for maintaining the integrity of genetic information. At least five UDG families have been characterized so far; these families share similar overall folds and common active site motifs. They demonstrate different substrate specificities, but often the function of one enzyme can be complemented by the other. Family 1 enzymes are active against uracil in both ssDNA and dsDNA, and recognize uracil explicitly in an extrahelical conformation via a combination of protein and bound-water interactions. Family 2 enzymes are mismatch specific and explicitly recognize the widowed guanine on the complementary strand, rather than the extrahelical scissile pyrimidine. This allows a broader specificity so that some Family 2 enzymes can excise uracil as well as 3, N(4)-ethenocytosine from mismatches with guanine. A Family 3 UDG from human was first characterized to remove Uracil from ssDNA, hence the name hSMUG (single-strand-selective monofunctional uracil-DNA glycosylase). However, subsequent research has shown that hSMUG1 and its rat ortholog can remove uracil and its oxidized pyrimidine derivatives from both, ssDNA and dsDNA. Enzymes in Families 4 and 5 are both thermostable. Family 4 enzymes specifically recognize uracil in a manner similar to human UDG (Family 1), rather than guanine in the complementary strand DNA, as does E. coli MUG (Family 2). These results suggest that the mechanism by which Family 4 UDGs remove uracils from DNA is similar to that of Family 1 enzyme. Although Family 5 enzymes are close relatives of Family 4, they show different substrate specificities.','cd09593','UDG_like'),(987250,'gnl|CDD|130696',_binary 'phage virion morphogenesis (putative tail completion) protein.  This model describes protein S of phage P2, suggested experimentally to act in tail completion and stable head joining, and related proteins from a number of phage. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01635','Mu-like_prophage_FluMu_G_'),(987265,'gnl|CDD|227608',_binary 'Phage-related tail protein [Function unknown].','COG5283','COG5283'),(987289,'gnl|CDD|129732',_binary 'DNA primase-related protein.  The DNA primase DnaG of E. coli and its apparent orthologs in other eubacterial species are approximately 600 residues in length. Within this set, a conspicuous outlier in percent identity, as seen in a UPGMA difference tree, is the branch containing the Mycoplasmas. This lineage is also unique in containing the small, DNA primase-related protein modelled here, which is homologous to the central third of DNA primase. Several small regions of sequence similarity specifically to Mycoplasma sequences rather than to all DnaG homologs suggests that the divergence of this protein from DnaG post-dated the separation of bacterial lineages. The function of this DNA primase-related protein is unknown. [Unknown function, General].','TIGR00646','Uncharacterized_protein_M'),(987290,'gnl|CDD|222863',_binary 'DNA primase; Provisional.','PHA02540','61'),(987293,'gnl|CDD|173777',_binary 'TOPRIM_ RNase M5_like: The topoisomerase-primase (TOPRIM) nucleotidyl transferase/hydrolase domain found in Ribonuclease M5: (RNase M5) and other small primase-like proteins from bacteria and archaea.  RNase M5 catalyzes the maturation of 5S rRNA in low G+C Gram-positive bacteria. The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). The conserved glutamate may act as a general base in nucleotide polymerization by primases. The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.','cd01027','TOPRIM_RNase_M5_like'),(987301,'gnl|CDD|223315',_binary 'Dephospho-CoA kinase [Coenzyme metabolism].','COG0237','CoaE'),(987303,'gnl|CDD|315796',_binary 'AAA domain.','pfam13207','AAA_17'),(987316,'gnl|CDD|306750',_binary 'Thymidylate synthase.  This is a family of proteins that are flavin-dependent thymidylate synthases.','pfam00303','Thymidylat_synt'),(987317,'gnl|CDD|234984',_binary 'thymidylate synthase; Reviewed.','PRK01827','thyA'),(987319,'gnl|CDD|213790',_binary 'thymidylate synthase.  Members of this protein family are thymidylate synthase, an enzyme that produces dTMP from dUMP. In prokaryotes, its gene usually is found close to that for dihydrofolate reductase, and in some systems the two enzymes are found as a fusion protein. This model excludes a set of related proteins (TIGR03283) that appears to replace this family in archaeal methanogens, where tetrahydrofolate is replaced by tetrahydromethanopterin. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR03284','thym_sym'),(987323,'gnl|CDD|184347',_binary 'thymidylate synthase; Provisional.','PRK13821','thyA'),(987324,'gnl|CDD|179181',_binary 'thymidylate synthase; Provisional.','PRK00956','thyA'),(987325,'gnl|CDD|132326',_binary 'thymidylate synthase, methanogen type.  Thymidylate synthase makes dTMP for DNA synthesis, and is among the most widely distributed of all enzymes. Members of this protein family are encoded within a completed genome sequence if and only if that species is one of the methanogenenic archaea. In these species, tetrahydromethanopterin replaces tetrahydrofolate, The member from Methanobacterium thermoautotrophicum was shown to behave as a thymidylate synthase based on similar side reactions (the exchange of a characteristic proton with water), although the full reaction was not reconstituted. Partial sequence data showed no similarity to known thymidylate synthases simply because the region sequenced was from a distinctive N-terminal region not found in other thymidylate synthases. Members of this protein family appear, therefore, to a novel, tetrahydromethanopterin-dependent thymidylate synthase. [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR03283','thy_syn_methano'),(987334,'gnl|CDD|306632',_binary 'Sigma-54 interaction domain.','pfam00158','Sigma54_activat'),(987336,'gnl|CDD|274079',_binary 'propionate catabolism operon regulatory protein PrpR.  At least five distinct pathways exists for the catabolism of propionate by way of propionyl-CoA. Members of this family represent the transcriptional regulatory protein PrpR, whose gene is found in most cases divergently transcribed from an operon for the methylcitric acid cycle of propionate catabolism. 2-methylcitric acid, a catabolite by this pathway, is a coactivator of PrpR. [Regulatory functions, DNA interactions].','TIGR02329','Propionate_catabolism_ope'),(987337,'gnl|CDD|224142',_binary 'Transcriptional regulators containing an AAA-type ATPase domain and a DNA-binding domain [Transcription / Signal transduction mechanisms].','COG1221','PspF'),(987338,'gnl|CDD|185070',_binary 'response regulator GlrR; Provisional.','PRK15115','PRK15115'),(987340,'gnl|CDD|224160',_binary 'Mg-chelatase subunit ChlI [Coenzyme metabolism].','COG1239','ChlI'),(987341,'gnl|CDD|285027',_binary 'ATPase family associated with various cellular activities (AAA).  This Pfam entry includes some of the AAA proteins not detected by the pfam00004 model.','pfam07726','AAA_3'),(987413,'gnl|CDD|133060',_binary 'EpsO protein participates in the methanolan synthesis.  The Methylobacillus sp EpsO protein is predicted to participate in the methanolan synthesis. Methanolan is an exopolysaccharide (EPS), composed of glucose, mannose and galactose.  A 21 genes cluster was predicted to participate in the methanolan synthesis. Gene disruption analysis revealed that EpsO is one of the glycosyltransferase enzymes involved in the synthesis of repeating sugar units onto the lipid carrier.','cd06438','EpsO_like'),(987415,'gnl|CDD|274866',_binary 'poly-beta-1,6 N-acetyl-D-glucosamine synthase.  Members of this protein family are biofilm-forming enzymes that polymerize N-acetyl-D-glucosamine residues in beta(1,6) linkage. One named members is IcaA (intercellular adhesin protein A), an enzyme that acts (with aid of subunit IcaD) in Polysaccharide Intercellular Adhesin (PIA) biosynthesis in Staphylococcus epidermis). The homologous member in E. coli is designated PgaC. Members are often encoded next to a polysaccharide deacetylase and involved in biofilm formation. Note that chitin, although also made from N-acetylglucosamine, is formed with beta-1,4 linkages.','TIGR03937','Poly-beta-16-N-acetyl-D-g'),(987416,'gnl|CDD|316189',_binary 'Glycosyltransferase like family 2.  Members of this family of prokaryotic proteins include putative glucosyltransferase, which are involved in bacterial capsule biosynthesis.','pfam13641','Glyco_tranf_2_3'),(987418,'gnl|CDD|236882',_binary 'N-glycosyltransferase; Provisional.','PRK11204','PRK11204'),(987419,'gnl|CDD|133061',_binary 'CESA_like_1 is a member of the cellulose synthase (CESA) superfamily.  This is a subfamily of cellulose synthase (CESA) superfamily.  CESA superfamily includes a wide variety of glycosyltransferase family 2 enzymes that share the common characteristic of catalyzing the elongation of polysaccharide chains.  The members of the superfamily include cellulose synthase catalytic subunit, chitin synthase, glucan biosynthesis protein and other families of CESA-like proteins.','cd06439','CESA_like_1'),(987422,'gnl|CDD|227721',_binary 'Endopygalactorunase [Cell envelope biogenesis, outer membrane].','COG5434','PGU1'),(987423,'gnl|CDD|133022',_binary 'DPM_DPG-synthase_like is a member of the Glycosyltransferase 2 superfamily.  DPM1 is the catalytic subunit of eukaryotic dolichol-phosphate mannose (DPM) synthase. DPM synthase is required for synthesis of the glycosylphosphatidylinositol (GPI) anchor, N-glycan precursor, protein O-mannose, and C-mannose. In higher eukaryotes,the enzyme has three subunits, DPM1, DPM2 and DPM3. DPM is synthesized from dolichol phosphate and GDP-Man on the cytosolic surface of the ER membrane by DPM synthase and then is flipped onto the luminal side and used as a donor substrate. In lower eukaryotes, such as Saccharomyces cerevisiae and Trypanosoma brucei, DPM synthase consists of a single component (Dpm1p and TbDpm1, respectively) that possesses one predicted transmembrane region near the C terminus for anchoring to the ER membrane. In contrast, the Dpm1 homologues of higher eukaryotes, namely fission yeast, fungi, and animals, have no transmembrane region, suggesting the existence of adapter molecules for membrane anchoring. This family also includes bacteria and archaea DPM1_like enzymes. However, the enzyme structure and mechanism of function are not well understood. The UDP-glucose:dolichyl-phosphate glucosyltransferase (DPG_synthase) is a transmembrane-bound enzyme of the endoplasmic reticulum involved in protein N-linked glycosylation. This enzyme catalyzes the transfer of glucose from UDP-glucose to dolichyl phosphate. This protein family belongs to Glycosyltransferase 2 superfamily.','cd04179','DPM_DPG-synthase_like'),(987424,'gnl|CDD|315392',_binary 'Pectate lyase superfamily protein.  This family of proteins possesses a beta helical structure like Pectate lyase. This family is most closely related to glycosyl hydrolase family 28.','pfam12708','Pectate_lyase_3'),(987425,'gnl|CDD|133013',_binary 'GT_2_like_a represents a glycosyltransferase family-2 subfamily with unknown function.  Glycosyltransferase family 2 (GT-2) subfamily of unknown function. GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd02522','GT_2_like_a'),(987426,'gnl|CDD|165802',_binary 'polygalacturonase.','PLN02155','PLN02155'),(987427,'gnl|CDD|316181',_binary 'Glycosyl transferase family group 2.  Members of this family of prokaryotic proteins include putative glucosyltransferases, which are involved in bacterial capsule biosynthesis.','pfam13632','Glyco_trans_2_3'),(987428,'gnl|CDD|184755',_binary 'N-glycosyltransferase; Provisional.','PRK14583','hmsR'),(987429,'gnl|CDD|133057',_binary 'NdvC_like  proteins in this family are putative bacterial beta-(1,6)-glucosyltransferase.  NdvC_like  proteins in this family are putative bacterial beta-(1,6)-glucosyltransferase. Bradyrhizobium japonicum synthesizes periplasmic cyclic beta-(1,3),beta-(1,6)-D-glucans during growth under hypoosmotic conditions. Two genes (ndvB, ndvC) are involved in the beta-(1, 3), beta-(1,6)-glucan synthesis. The ndvC mutant strain resulted in synthesis of altered cyclic beta-glucans composed almost entirely of beta-(1, 3)-glycosyl linkages. The periplasmic cyclic beta-(1,3),beta-(1,6)-D-glucans function for osmoregulation. The ndvC mutation also affects the ability of the bacteria to establish a successful symbiotic interaction with host plant. Thus, the beta-glucans may function as suppressors of a host defense response.','cd06435','CESA_NdvC_like'),(987430,'gnl|CDD|133058',_binary 'N-acetyl-glucosamine transferase is involved in the synthesis of Poly-beta-1,6-N-acetyl-D-glucosamine.  N-acetyl-glucosamine transferase is responsible for the synthesis of bacteria Poly-beta-1,6-N-acetyl-D-glucosamine (PGA). Poly-beta-1,6-N-acetyl-D-glucosamine is a homopolymer that serves as an adhesion for the maintenance of biofilm structural stability in diverse eubacteria. N-acetyl-glucosamine transferase is the product of gene pgaC. Genetic analysis indicated that all four genes of the pgaABCD locus were required for the PGA production, pgaC being a glycosyltransferase.','cd06436','GlcNAc-1-P_transferase'),(987431,'gnl|CDD|132155',_binary 'putative glycosyltransferase, exosortase G-associated.  Members of this protein family are probable glycosyltransferases of family 2, whose genes are near those for the exosortase homolog XrtG (TIGR03110), which is restricted to Gram-positive bacteria. Other genes in the conserved gene neighborhood include a 6-pyruvoyl tetrahydropterin synthase homolog (TIGR03112) and an uncharacterized intergral membrane protein (TIGR03766).','TIGR03111','glyc2_xrt_Gpos1'),(987433,'gnl|CDD|133031',_binary 'DPG_synthase is involved in protein N-linked glycosylation.  UDP-glucose:dolichyl-phosphate glucosyltransferase (DPG_synthase) is a transmembrane-bound enzyme of the endoplasmic reticulum involved in protein N-linked glycosylation. This enzyme catalyzes the transfer of glucose from UDP-glucose to dolichyl phosphate.','cd04188','DPG_synthase'),(987434,'gnl|CDD|275103',_binary 'transferase 2, rSAM/selenodomain-associated.  This enzyme may transfer a nucleotide, or it sugar moiety, as part of a biosynthetic pathway. Other proposed members of the pathway include another transferase (TIGR04282), a phosphoesterase, and a radical SAM enzyme (TIGR04167) whose C-terminal domain (pfam12345) frequently contains a selenocysteine. [Unknown function, Enzymes of unknown specificity].','TIGR04283','glycosyl_transferase_fami'),(987435,'gnl|CDD|237556',_binary 'putative glucosyl-3-phosphoglycerate synthase; Provisional.','PRK13915','PRK13915'),(987437,'gnl|CDD|275035',_binary 'glycosyltransferase, TIGR04182 family.  Members of this family are glycosyltransferases restricted to the archaea. All but two members are from species with the PGF-CTERM/archaeosortase A system, a proposed maturation system for exported, glycosylated proteins as are found often in S-layers.','TIGR04182','glyco_TIGR04182'),(987438,'gnl|CDD|133062',_binary 'DPM1_like represents putative enzymes similar to eukaryotic DPM1.  Proteins similar to eukaryotic DPM1, including enzymes from bacteria and archaea; DPM1 is the catalytic subunit of eukaryotic dolichol-phosphate mannose (DPM) synthase. DPM synthase is required for synthesis of the glycosylphosphatidylinositol (GPI) anchor, N-glycan precursor, protein O-mannose, and C-mannose. In higher eukaryotes,the enzyme has three subunits, DPM1, DPM2 and DPM3. DPM is synthesized from dolichol phosphate and GDP-Man on the cytosolic surface of the ER membrane by DPM synthase and then is flipped onto the luminal side and used as a donor substrate. In lower eukaryotes, such as Saccharomyces cerevisiae and Trypanosoma brucei, DPM synthase consists of a single component (Dpm1p and TbDpm1, respectively) that possesses one predicted transmembrane region near the C terminus for anchoring to the ER membrane. In contrast, the Dpm1 homologues of higher eukaryotes, namely fission yeast, fungi, and animals, have no transmembrane region, suggesting the existence of adapter molecules for membrane anchoring. This family also includes bacteria and archaea DPM1_like enzymes. However, the enzyme structure and mechanism of function are not well understood. This protein family belongs to Glycosyltransferase 2 superfamily.','cd06442','DPM1_like'),(987439,'gnl|CDD|133043',_binary 'CESA_CelA_like are involved in the elongation of the glucan chain of cellulose.  Family of proteins related to  Agrobacterium tumefaciens CelA and  Gluconacetobacter xylinus BscA. These proteins are involved in the elongation of the glucan chain of cellulose, an aggregate of unbranched polymers of beta-1,4-linked glucose residues. They are putative catalytic subunit of cellulose synthase, which is a glycosyltransferase using UDP-glucose as the substrate. The catalytic subunit is an integral membrane protein with 6 transmembrane segments and it is postulated that the protein is anchored in the membrane at the N-terminal end.','cd06421','CESA_CelA_like'),(987440,'gnl|CDD|133030',_binary 'Bacterial DPM1_like enzymes are related to eukaryotic DPM1.  A family of  bacterial enzymes related to eukaryotic DPM1; Although the mechanism of eukaryotic enzyme is well studied, the mechanism of the  bacterial enzymes is not well understood. The eukaryotic DPM1 is the catalytic subunit of eukaryotic Dolichol-phosphate mannose (DPM) synthase. DPM synthase is required for synthesis of the glycosylphosphatidylinositol (GPI) anchor, N-glycan precursor, protein O-mannose, and C-mannose. The enzyme has three subunits, DPM1, DPM2 and DPM3. DPM is synthesized from dolichol phosphate and GDP-Man on the cytosolic surface of the ER membrane by DPM synthase and then is flipped onto the luminal side and used as a donor substrate. This protein family belongs to Glycosyltransferase 2 superfamily.','cd04187','DPM1_like_bac'),(987441,'gnl|CDD|133035',_binary 'Subfamily of Glycosyltransferase Family GT2 of unknown function.  GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd04192','GT_2_like_e'),(987443,'gnl|CDD|133056',_binary 'Hyaluronan synthases catalyze polymerization of hyaluronan.  Hyaluronan synthases (HASs) are bi-functional glycosyltransferases that catalyze polymerization of hyaluronan. HASs transfer both GlcUA and GlcNAc in beta-(1,3) and beta-(1,4) linkages, respectively to the hyaluronan chain using UDP-GlcNAc and UDP-GlcUA as substrates. HA is made as a free glycan, not attached to a protein or lipid. HASs do not need a primer for HA synthesis; they initiate HA biosynthesis de novo with only UDP-GlcNAc, UDP-GlcUA, and Mg2+. Hyaluronan (HA) is a linear heteropolysaccharide composed of (1-3)-linked beta-D-GlcUA-beta-D-GlcNAc disaccharide repeats. It can be found in vertebrates and a few microbes and is typically on the cell surface or in the extracellular space, but is also found inside mammalian cells. Hyaluronan has several physiochemical and biological functions such as space filling, lubrication, and providing a hydrated matrix through which cells can migrate.','cd06434','GT2_HAS'),(987444,'gnl|CDD|213815',_binary 'hopene-associated glycosyltransferase HpnB.  This family of genes include a glycosyl transferase, group 2 domain (pfam00535) which are responsible, generally for the transfer of nucleotide-diphosphate sugars to substrates such as polysaccharides and lipids. The genes of this family are often found in the same genetic locus with squalene-hopene cyclase genes, and are never associated with genes for the metabolism of phytoene. Indeed, the members of this family appear to never be found in a genome lacking squalene-hopene cyclase (SHC), although not all genomes encoding SHC have this glycosyl transferase. In the organism Zymomonas mobilis the linkage of this gene to hopanoid biosynthesis has been noted and the gene named HpnB. Hopanoids are known to feature polar glycosyl head groups in many organisms.','TIGR03469','HpnB'),(987445,'gnl|CDD|133042',_binary 'N-terminal domain of Chondroitin polymerase functions as a GalNAc transferase.  Chondroitin polymerase is a two domain, bi-functional protein. The N-terminal domain functions as a GalNAc transferase. The bacterial chondroitin polymerase catalyzes elongation of the chondroitin chain by alternatively transferring the GlcUA and GalNAc moiety from UDP-GlcUA and UDP-GalNAc to the non-reducing ends of the chondroitin chain. The enzyme consists of N-terminal and C-terminal domains in which the two active sites catalyze the addition of GalNAc and GlcUA, respectively. Chondroitin chains range from 40 to over 100 repeating units of the disaccharide. Sulfated chondroitins are involved in the regulation of various biological functions such as central nervous system development, wound repair, infection, growth factor signaling, and morphogenesis, in addition to its conventional structural roles. In Caenorhabditis elegans, chondroitin is an essential factor for the worm to undergo cytokinesis and cell division. Chondroitin is synthesized as proteoglycans, sulfated and secreted to the cell surface or extracellular matrix.','cd06420','GT2_Chondriotin_Pol_N'),(987476,'gnl|CDD|315718',_binary 'AsmA-like C-terminal region.  This family is similar to the C-terminal of the AsmA protein of E. coli.','pfam13109','AsmA_1'),(987844,'gnl|CDD|235859',_binary 'homoserine O-acetyltransferase; Provisional.','PRK06765','PRK06765'),(988651,'gnl|CDD|318193',_binary 'Coiled-coil domain-containing protein 158.  CCDC158 is a family of proteins found in eukaryotes. The function is not known.','pfam15921','CCDC158'),(988994,'gnl|CDD|315491',_binary 'Mitochondrial ribosomal protein subunit L20.  This family is the essential mitochondrial ribosomal protein subunit L20 of fungi.','pfam12824','MRP-L20'),(989031,'gnl|CDD|307927',_binary 'RyR domain.  This domain is called RyR for Ryanodine receptor. The domain is found in four copies in the ryanodine receptor. The function of this domain is unknown.','pfam02026','RyR'),(989086,'gnl|CDD|198027',_binary 'Rho termination factor, N-terminal domain.  The Rho termination factor disengages newly transcribed RNA from its DNA template at certain, specific transcripts. It it thought that two copies of Rho bind to RNA and that Rho functions as a hexamer of protomers. This domain is found to the N-terminus of the RNA binding domain.','smart00959','Rho_N'),(989087,'gnl|CDD|311441',_binary 'Rho termination factor, N-terminal domain.  The Rho termination factor disengages newly transcribed RNA from its DNA template at certain, specific transcripts. It it thought that two copies of Rho bind to RNA and that Rho functions as a hexamer of protomers. This domain is found to the N-terminus of the RNA binding domain (pfam07497).','pfam07498','Rho_N'),(989090,'gnl|CDD|306576',_binary 'von Willebrand factor type A domain.','pfam00092','VWA'),(989093,'gnl|CDD|274135',_binary 'cobaltochelatase subunit.  Cobaltochelatase is responsible for the insertion of cobalt into the corrin ring of coenzyme B12 during its biosynthesis. Two versions have been well described. CbiK/CbiX is a monomeric, anaerobic version which acts early in the biosynthesis (pfam06180). CobNST is a trimeric, ATP-dependent, aerobic version which acts late in the biosynthesis (TIGR02257/TIGR01650/TIGR01651). A number of genomes (actinobacteria, cyanobacteria, betaproteobacteria and pseudomonads) which apparently biosynthesize B12, encode a cobN gene but are demonstrably lacking cobS and cobT. These genomes do, however contain a homolog (modelled here) of the magnesium chelatase subunits BchI/BchD family. Aside from the cyanobacteria (which have a separate magnesium chelatase trimer), these species do not make chlorins, so do not have any use for a magnesium chelatase. Furthermore, in nearly all cases the members of this family are proximal to either CobN itself or other genes involved in cobalt transport or B12 biosynthesis.','TIGR02442','Uncharacterized_protein_R'),(989094,'gnl|CDD|224161',_binary 'Mg-chelatase subunit ChlD [Coenzyme metabolism].','COG1240','ChlD'),(989367,'gnl|CDD|279101',_binary 'DNA polymerase III beta subunit, N-terminal domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam00712','DNA_pol3_beta'),(989371,'gnl|CDD|184910',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14946','PRK14946'),(989422,'gnl|CDD|316037',_binary 'AAA domain.','pfam13476','AAA_23'),(989424,'gnl|CDD|224116',_binary 'Recombinational DNA repair ATPase (RecF pathway) [DNA replication, recombination, and repair].','COG1195','RecF'),(989425,'gnl|CDD|213194',_binary 'ATP-binding cassette domain of non-transporter proteins.  ABC-type Class 2 contains systems involved in cellular processes other than transport. These families are characterized by the fact that the ABC subunit is made up of duplicated, fused ABC modules (ABC2). No known transmembrane proteins or domains are associated with these proteins.','cd03227','ABC_Class2'),(989426,'gnl|CDD|213207',_binary 'ATP-binding cassette domain of Rad50.  The catalytic domains of Rad50 are similar to the ATP-binding cassette of ABC transporters, but are not associated with membrane-spanning domains. The conserved ATP-binding motifs common to Rad50 and the ABC transporter family include the Walker A and Walker B motifs, the Q loop, a histidine residue in the switch region, a D-loop, and a conserved LSGG sequence. This conserved sequence, LSGG, is the most specific and characteristic motif of this family and is thus known as the ABC signature sequence.','cd03240','ABC_Rad50'),(989427,'gnl|CDD|213243',_binary 'ATP-binding cassette domain of eukaryotic SM6 proteins.  The structural maintenance of chromosomes (SMC) proteins are large (approximately 110 to 170 kDa), and each is arranged into five recognizable domains. Amino-acid sequence homology of SMC proteins between species is largely confined to the amino- and carboxy-terminal globular domains. The amino-terminal domain contains a \'Walker A\' nucleotide-binding domain (GxxGxGKS/T, in the single-letter amino-acid code), which by mutational studies has been shown to be essential in several proteins. The carboxy-terminal domain contains a sequence (the DA-box) that resembles a \'Walker B\' motif, and a motif with homology to the signature sequence of the ATP-binding cassette (ABC) family of ATPases. The sequence homology within the carboxy-terminal domain is relatively high within the SMC1-SMC4 group, whereas SMC5 and SMC6 show some divergence in both of these sequences. In eukaryotic cells, the proteins are found as heterodimers of SMC1 paired with SMC3, SMC2 with SMC4, and SMC5 with SMC6 (formerly known as Rad18).','cd03276','ABC_SMC6_euk'),(989428,'gnl|CDD|315766',_binary 'AAA ATPase domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily.','pfam13175','AAA_15'),(989455,'gnl|CDD|316773',_binary 'PD-(D/E)XK nuclease superfamily.  Members of this family belong to the PD-(D/E)XK nuclease superfamily.','pfam14281','PDDEXK_4'),(990064,'gnl|CDD|182697',_binary 'lysophospholipase L2; Provisional.','PRK10749','PRK10749'),(990098,'gnl|CDD|238865',_binary 'sialate O-acetylesterase_like family of the SGNH hydrolases, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01827','sialate_O-acetylesterase_'),(990105,'gnl|CDD|223532',_binary 'Acetyltransferases [General function prediction only].','COG0456','RimI'),(990106,'gnl|CDD|306954',_binary 'Acetyltransferase (GNAT) family.  This family contains proteins with N-acetyltransferase functions such as Elp3-related proteins.','pfam00583','Acetyltransf_1'),(990107,'gnl|CDD|316066',_binary 'Acetyltransferase (GNAT) domain.  This domain catalyzes N-acetyltransferase reactions.','pfam13508','Acetyltransf_7'),(990112,'gnl|CDD|227235',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4898','COG4898'),(990243,'gnl|CDD|172486',_binary 'putative oxidoreductase; Provisional.','PRK13984','PRK13984'),(990263,'gnl|CDD|100796',_binary 'chromosome segregation protein; Provisional.','PRK01156','PRK01156'),(990316,'gnl|CDD|224236',_binary 'Flagellar biosynthesis/type III secretory pathway protein [Cell motility and secretion / Intracellular trafficking and secretion].','COG1317','FliH'),(990326,'gnl|CDD|225028',_binary 'Predicted subunit of tRNA(5-methylaminomethyl-2-thiouridylate) methyltransferase, contains the PP-loop ATPase domain [Translation, ribosomal structure and biogenesis].','COG2117','COG2117'),(990327,'gnl|CDD|237510',_binary 'hypothetical protein; Provisional.','PRK13795','PRK13795'),(990334,'gnl|CDD|239945',_binary 'Members of the SGNH-hydrolase superfamily, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad from other serine hydrolases, but may lack the carboxlic acid.','cd04501','SGNH_hydrolase_like_4'),(990394,'gnl|CDD|311706',_binary 'CopG-like RHH_1 or ribbon-helix-helix domain, RHH_5.  This family contains bacterial proteins that form a ribbon-helix-helix fold. This fold occurs in many examples of bacterial antitoxins.','pfam07878','RHH_5'),(990416,'gnl|CDD|236137',_binary 'NAD-dependent DNA ligase LigA; Validated.','PRK07956','ligA'),(990417,'gnl|CDD|238062',_binary 'NAD+ dependent DNA ligase adenylation domain. DNA ligases catalyze the crucial step of joining the breaks in duplex DNA during DNA replication, repair and recombination, utilizing either ATP or NAD(+) as a cofactor, but using the same basic reaction mechanism. The enzyme reacts with the cofactor to form a phosphoamide-linked AMP with the amino group of a conserved Lysine in the KXDG motif, and subsequently transfers it to the DNA substrate to yield adenylated DNA. This alignment contains members of the NAD+ dependent subfamily only.','cd00114','LIGANc'),(990418,'gnl|CDD|223350',_binary 'NAD-dependent DNA ligase (contains BRCT domain type II) [DNA replication, recombination, and repair].','COG0272','Lig'),(990419,'gnl|CDD|214709',_binary 'Ligase N family.','smart00532','LIGANc'),(990420,'gnl|CDD|307673',_binary 'NAD-dependent DNA ligase adenylation domain.  DNA ligases catalyze the crucial step of joining the breaks in duplex DNA during DNA replication, repair and recombination, utilising either ATP or NAD(+) as a cofactor. This domain is the catalytic adenylation domain. The NAD+ group is covalently attached to this domain at the lysine in the KXDG motif of this domain. This enzyme- adenylate intermediate is an important feature of the proposed catalytic mechanism.','pfam01653','DNA_ligase_aden'),(990463,'gnl|CDD|273148',_binary 'DNA ligase, NAD-dependent.  All proteins in this family with known functions are NAD-dependent DNA ligases. Functions of these proteins include DNA repair, DNA replication, and DNA recombination. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). The member of this family from Treponema pallidum differs in having three rather than just one copy of the BRCT (BRCA1 C Terminus) domain (pfam00533) at the C-terminus. It is included in the seed. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00575','DNA_ligase'),(990488,'gnl|CDD|99921',_binary 'DNA polymerase type-II B subfamily catalytic domain. Bacteria contain five DNA polymerases (I, II, III, IV and V). DNA polymerase II (Pol II) is a prototype for the B-family of polymerases. The role of Pol II in a variety of cellular activities, such as repair of DNA damaged by UV irradiation or oxidation has been proved by genetic studies. DNA polymerase III is the main enzyme responsible for replication of the bacterial chromosome; however, In vivo studies have also shown that Pol II is able to participate in chromosomal DNA replication with larger role in lagging-strand replication.','cd05538','POLBc_Pol_II_B'),(990669,'gnl|CDD|176499',_binary 'Glycerophosphodiester phosphodiesterase domain as found in prokaryota and eukaryota, and similar proteins.  The typical glycerophosphodiester phosphodiesterase domain (GDPD) consists of a TIM barrel and a small insertion domain named the GDPD-insertion (GDPD-I) domain, which is specific for GDPD proteins. This family corresponds to both typical GDPD domain and GDPD-like domain which lacks the GDPD-I region. Members in this family mainly consist of a large family of prokaryotic and eukaryotic glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46), and a number of uncharacterized homologs. Sphingomyelinases D (SMases D) (sphingomyelin phosphodiesterase D, EC 3.1.4.41) from spider venom, SMases D-like proteins, and phospholipase D (PLD) from several pathogenic bacteria are also included in this family. GDPD plays an essential role in glycerol metabolism and catalyzes the hydrolysis of glycerophosphodiesters to sn-glycerol-3-phosphate (G3P) and the corresponding alcohols are major sources of carbon and phosphate. Its catalytic mechanism is based on the metal ion-dependent acid-base reaction, which is similar to that of phosphoinositide-specific phospholipases C (PI-PLCs, EC 3.1.4.11). Both, GDPD related proteins and PI-PLCs, belong to the superfamily of PI-PLC-like phosphodiesterases.','cd08556','GDPD'),(990670,'gnl|CDD|176543',_binary 'Glycerophosphodiester phosphodiesterase domain of Staphylococcus aureus and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in uncharacterized glycerophosphodiester phosphodiesterase (GP-GDE, EC 3.1.4.46) from Staphylococcus aureus, Bacillus subtilis and similar proteins. Members in this family show very high sequence similarity to Escherichia coli periplasmic phosphodiesterase GlpQ, which catalyzes the Ca2+-dependent degradation of periplasmic glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08601','GDPD_SaGlpQ_like'),(990671,'gnl|CDD|308577',_binary 'Glycerophosphoryl diester phosphodiesterase family.  E. coli has two sequence related isozymes of glycerophosphoryl diester phosphodiesterase (GDPD) - periplasmic and cytosolic. This family also includes agrocinopine synthase, the similarity to GDPD has been noted. This family appears to have weak but not significant matches to mammalian phospholipase C pfam00388, which suggests that this family may adopt a TIM barrel fold.','pfam03009','GDPD'),(990672,'gnl|CDD|176502',_binary 'Periplasmic glycerophosphodiester phosphodiesterase domain (GlpQ) and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in bacterial and eukaryotic glycerophosphodiester phosphodiesterase (GP-GDE, EC 3.1.4.46) similar to Escherichia coli periplasmic phosphodiesterase GlpQ. GP-GDEs are involved in glycerol metabolism and catalyze the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols, which are major sources of carbon and phosphate. In E. coli, there are two major G3P uptake systems: Glp and Ugp, which contain genes coding for two different GP-GDEs. GlpQ gene from the glp operon codes for a periplasmic phosphodiesterase GlpQ. GlpQ is a dimeric enzyme that hydrolyzes periplasmic glycerophosphodiesters, such as glycerophosphocholine (GPC), glycerophosphoethanolanmine (GPE), glycerophosphoglycerol (GPG), glycerophosphoinositol (GPI), and glycerophosphoserine (GPS), to the corresponding alcohols and G3P, which is subsequently transported into the cell through the GlpT transport system. Ca2+ is required for GlpQ enzymatic activity. This subfamily also includes some GP-GDEs in higher plants and their eukaryotic homologs, which show very high sequence similarities with bacterial periplasmic GP-GDEs.','cd08559','GDPD_periplasmic_GlpQ_lik'),(990673,'gnl|CDD|176509',_binary 'Glycerophosphodiester phosphodiesterase domain of Agrobacterium tumefaciens and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Agrobacterium tumefaciens glycerophosphodiester phosphodiesterase (AtGDE, EC 3.1.4.46) and its uncharacterized eukaryotic homolgoues. Members in this family shows high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. AtGDE exists as a hexamer that is a trimer of dimers, which is unique among current known GDPD family members. However, it remains unclear if the hexamer plays a physiological role in AtGDE enzymatic function.','cd08566','GDPD_AtGDE_like'),(990674,'gnl|CDD|176517',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE4-like proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE4 (also known as glycerophosphodiester phosphodiesterase domain-containing protein 1 (GDPD1)) and similar proteins. Mammalian GDE4 is a transmembrane protein whose cellular function is not elucidated. It is expressed widely, including in placenta, liver, kidney, pancreas, spleen, thymus, ovary, small intestine and peripheral blood leukocytes. It is also expressed in the growth cones in neuroblastoma Neuro2a cells, which suggests mammalian GDE4 may play some distinct role from other members of mammalian GDEs family. Also included in this subfamily are uncharacterized mammalian glycerophosphodiester phosphodiesterase domain-containing protein 3 (GDPD3) and similar proteins which display very high sequence homology to mammalian GDE4.','cd08575','GDPD_GDE4_like'),(990675,'gnl|CDD|176523',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial glycerophosphodiester phosphodiesterases.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized bacterial glycerophosphodiester phosphodiesterase and similar proteins. They show high sequence similarity to Escherichia coli glycerophosphodiester phosphodiesterase, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08581','GDPD_like_1'),(990676,'gnl|CDD|176544',_binary 'Glycerophosphodiester phosphodiesterase domain of Streptomycin coelicolor (GlpQ1) and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present  in a group of putative bacterial and eukaryotic glycerophosphodiester phosphodiesterases (GP-GDE, EC 3.1.4.46) similar to Escherichia coli periplasmic phosphodiesterase GlpQ, as well as plant glycerophosphodiester phosphodiesterases (GP-PDEs), all of which catalyzes the Ca2+-dependent degradation of periplasmic glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. The prototypes of this family include putative secreted phosphodiesterase encoded by gene glpQ1 (SCO1565) from the pho regulon in Streptomyces coelicolor genome, and in plants, two distinct Arabidopsis thaliana genes, AT5G08030 and AT1G74210, coding putative GP-PDEs from the cell walls and vacuoles, respectively.','cd08602','GDPD_ScGlpQ1_like'),(990677,'gnl|CDD|223657',_binary 'Glycerophosphoryl diester phosphodiesterase [Energy production and conversion].','COG0584','UgpQ'),(990678,'gnl|CDD|176504',_binary 'Glycerophosphodiester phosphodiesterase domain of Streptomyces coelicolor cytoplasmic phosphodiesterases UgpQ2 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized cytoplasmic phosphodiesterases which predominantly exist in bacteria. The prototype of this family is a putative cytoplasmic phosphodiesterase encoded by gene ulpQ2 (SCO1419) in the Streptomyces coelicolor genome. It is distantly related to the Escherichia coli cytoplasmic phosphodiesterases UgpQ that catalyzes the hydrolysis of glycerophosphodiesters at the inner side of the cytoplasmic membrane to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08561','GDPD_cytoplasmic_ScUgpQ2_'),(990679,'gnl|CDD|176524',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial glycerophosphodiester phosphodiesterases.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized bacterial glycerophosphodiester phosphodiesterase and similar proteins. They show high sequence similarity to Escherichia coli glycerophosphodiester phosphodiesterase, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08582','GDPD_like_2'),(990680,'gnl|CDD|176512',_binary 'Glycerophosphodiester phosphodiesterase domain of Saccharomyces cerevisiae YPL206cp and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Saccharomyces cerevisiae YPL206cp and uncharacterized hypothetical homologs existing in fungi. The product of S. cerevisiae ORF YPL206c (PGC1), YPL206cp (Pgc1p), displays homology to bacterial and mammalian glycerophosphodiester phosphodiesterases (GP-GDE, EC 3.1.4.46), which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. S. cerevisiae YPL206cp is an integral membrane protein with a single GDPD domain following by a short hydrophobic C-terminal tail that may function as a membrane anchor. This protein plays an essential role in the regulation of the cardiolipin (CL) biosynthetic pathway in yeast by removing the excess phosphatidylglycerol (PG) content of membranes via a phospholipase C-type degradation mechanism. YPL206cp has been characterized as a PG-specific phospholipase C that selectively catalyzes the cleavage of PG, not glycerophosphoinositol (GPI) or glycerophosphocholine (GPC), to diacylglycerol (DAG) and glycerophosphate.  Members in this family are distantly related to S. cerevisiae YPL110cp, which selectively hydrolyzes glycerophosphocholine (GPC), not glycerophosphoinositol (GPI), to generate choline and glycerolphosphate, and has been characterized as a cytoplasmic GPC-specific phosphodiesterase.','cd08570','GDPD_YPL206cp_fungi'),(990681,'gnl|CDD|176505',_binary 'Glycerophosphodiester phosphodiesterase domain in Escherichia coli cytosolic glycerophosphodiester phosphodiesterase UgpQ and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Escherichia coli cytosolic glycerophosphodiester phosphodiesterase (GP-GDE, EC 3.1.4.46), UgpQ, and similar proteins. GP-GDE plays an essential role in the metabolic pathway of E. coli. It catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols, which are major sources of carbon and phosphate. E. coli possesses two major G3P uptake systems: Glp and Ugp, which contain genes coding for two distinct GP-GDEs. UgpQ gene from the E. coli ugp operon codes for a cytosolic phosphodiesterase GlpQ, which is the prototype of this family. Various glycerophosphodiesters, such as glycerophosphocholine (GPC), glycerophosphoethanolanmine (GPE), glycerophosphoglycerol (GPG), glycerophosphoinositol (GPI), and glycerophosphoserine (GPS), can only be hydrolyzed by UgpQ during transport at the inner side of the cytoplasmic membrane to alcohols and G3P, which is a source of phosphate. In contrast to Ca2+-dependent periplasmic phosphodiesterase GlpQ, cytosolic phosphodiesterase UgpQ requires divalent cations, such as Mg2+, Co2+, or Mn2+, for its enzyme activity.','cd08562','GDPD_EcUgpQ_like'),(990682,'gnl|CDD|176553',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE4 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE4 (also known as glycerophosphodiester phosphodiesterase domain-containing protein 1 (GDPD1)) and similar proteins. Mammalian GDE4 is a transmembrane protein whose cellular function has not yet been elucidated. It is expressed widely, including in placenta, liver, kidney, pancreas, spleen, thymus, ovary, small intestine and peripheral blood leukocytes. It is also expressed in the growth cones in neuroblastoma Neuro2a cells, which suggests GDE4 may play some distinct role from other members of the GDE family.','cd08612','GDPD_GDE4'),(990683,'gnl|CDD|176503',_binary 'Glycerophosphodiester phosphodiesterase domain similar to Escherichia coli periplasmic phosphodiesterase (GlpQ) include uncharacterized proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized glycerophosphodiester phosphodiesterases (GP-GDE, EC 3.1.4.46) and their hypothetical homologs. Members in this subfamily show high sequence similarity to Escherichia coli periplasmic phosphodiesterase GlpQ, which catalyzes the Ca2+-dependent degradation of periplasmic glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08560','GDPD_EcGlpQ_like_1'),(990684,'gnl|CDD|176498',_binary 'Catalytic domain of phosphoinositide-specific phospholipase C-like phosphodiesterases superfamily.  The PI-PLC-like phosphodiesterases superfamily represents the catalytic domains of bacterial phosphatidylinositol-specific phospholipase C (PI-PLC, EC 4.6.1.13), eukaryotic phosphoinositide-specific phospholipase C (PI-PLC, EC 3.1.4.11), glycerophosphodiester phosphodiesterases (GP-GDE, EC 3.1.4.46), sphingomyelinases D (SMases D) (sphingomyelin phosphodiesterase D, EC 3.1.4.41) from spider venom, SMases D-like proteins, and phospholipase D (PLD) from several pathogenic bacteria, as well as their uncharacterized homologs found in organisms ranging from bacteria and archaea to metazoans, plants, and fungi. PI-PLCs are ubiquitous enzymes hydrolyzing the membrane lipid phosphoinositides to yield two important second messengers, inositol phosphates and diacylglycerol (DAG). GP-GDEs play essential roles in glycerol metabolism and catalyze the hydrolysis of glycerophosphodiesters to sn-glycerol-3-phosphate (G3P) and the corresponding alcohols that are major sources of carbon and phosphate. Both, PI-PLCs and GP-GDEs, can hydrolyze the 3\'-5\' phosphodiester bonds in different substrates, and utilize a similar mechanism of general base and acid catalysis with conserved histidine residues, which consists of two steps, a phosphotransfer and a phosphodiesterase reaction. This superfamily also includes Neurospora crassa ankyrin repeat protein NUC-2 and its Saccharomyces cerevisiae counterpart, Phosphate system positive regulatory protein PHO81, glycerophosphodiester phosphodiesterase (GP-GDE)-like protein SHV3 and SHV3-like proteins (SVLs). The residues essential for enzyme activities and metal binding are not conserved in these sequence homologs, which might suggest that the function of catalytic domains in these proteins might be distinct from those in typical PLC-like phosphodiesterases.','cd08555','PI-PLCc_GDPD_SF'),(990685,'gnl|CDD|176521',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial glycerophosphodiester phosphodiesterases.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in uncharacterized bacterial glycerophosphodiester phosphodiesterases. In addition to a C-terminal GDPD domain, most members in this family have an N-terminus that functions as a membrane anchor.','cd08579','GDPD_memb_like'),(990686,'gnl|CDD|176542',_binary 'Glycerophosphodiester phosphodiesterase domain of Escherichia coli (GlpQ) and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Escherichia coli periplasmic glycerophosphodiester phosphodiesterase (GP-GDE, EC 3.1.4.46), GlpQ, and similar proteins. GP-GDE plays an essential role in the metabolic pathway of E. coli. It catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols, which are major sources of carbon and phosphate. E. coli possesses two major G3P uptake systems: Glp and Ugp, which contain genes coding for two different GP-GDEs. GlpQ gene from the E. coli glp operon codes for a periplasmic phosphodiesterase GlpQ, which is the prototype of this family. GlpQ is a dimeric enzyme that hydrolyzes periplasmic glycerophosphodiesters, such as glycerophosphocholine (GPC), glycerophosphoethanolanmine (GPE), glycerophosphoglycerol (GPG), glycerophosphoinositol (GPI), and glycerophosphoserine (GPS), to the corresponding alcohols and G3P, which is subsequently transported into the cell through the GlpT transport system. Ca2+ is required for the enzymatic activity of GlpQ.  This family also includes a surface-exposed lipoprotein, protein D (HPD), from Haemophilus influenza Type b and nontypeable strains, which shows very high sequence similarity with E. coli GlpQ. HPD has been characterized as a human immunoglobulin D-binding protein with glycerophosphodiester phosphodiesterase activity. It can hydrolyze phosphatidylcholine from host membranes to produce free choline on the lipopolysaccharides on the surface of pathogenic bacteria.','cd08600','GDPD_EcGlpQ_like'),(990748,'gnl|CDD|235735',_binary 'DNA polymerase III subunit epsilon; Validated.','PRK06195','PRK06195'),(990749,'gnl|CDD|181186',_binary 'exodeoxyribonuclease X; Provisional.','PRK07983','PRK07983'),(990759,'gnl|CDD|198390',_binary 'GIY-YIG domain of uncharacterized hypothetical protein found in phycodnavirus PBCV-1 DNA virus, T. thermophila Tlr element eoncoding protein Tlr8p, and similar proteins found in bacteria.  The family includes a group of diverse uncharacterized hypothetical proteins with a GIY-YIG domain that shows statistically significant similarity to the N-terminal catalytic domains of GIY-YIG family of intron-encoded homing endonuclease  I-TevI.  Similar to I-TevI, family members from phycodnavirus PBCV-1 DNA virus have nuclease-associated modular DNA-binding domains (NUMODs) and a helix-turn-helix (HTH) domain C-terminally fused to the GIY-YIG domain, which suggests that these PBCV-1 acquired the I-TevI-like homing endonucleases from phages by horizontal gene transfer. This family also includes proteins that appear to connect homing endonucleases with Penelope elements, such as Tetrahymena thermophila Tlr element encoding protein Tlr8p that possess additional N-terminal and central structural regions, followed by a putative superfamily 1 helicase domain and I-TevI-like GIY-YIG domain, but lacks the NUMOD domains and HTH domain. It is suggested that the Tlr8p element could have acquired its GIY-YIG domain w ithin the nucleus of the ciliate cell infected by the Phycodnavirus. Some family members only contain a standalone GIY-YIG domain and their biological functions are unclear.','cd10443','GIY-YIG_HE_Tlr8p_PBC-V_li'),(990772,'gnl|CDD|176508',_binary 'Glycerophosphodiester phosphodiesterase domain of putative Agrobacterium tumefaciens glycerophosphodiester phosphodiesterase and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in putative Agrobacterium tumefaciens glycerophosphodiester phosphodiesterase (pAtGDE, EC 3.1.4.46) and its uncharacterized homologs. Members in this family show high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08565','GDPD_pAtGDE_like'),(990778,'gnl|CDD|236524',_binary 'cytoplasmic glycerophosphodiester phosphodiesterase; Provisional.','PRK09454','ugpQ'),(990780,'gnl|CDD|176552',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE6 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE6 (also known as glycerophosphodiester phosphodiesterase domain-containing protein 4 (GDPD4)) and their metazoan homologs. Mammalian GDE6 is a transmembrane protein predominantly expressed in the spermatocytes of testis. Although the specific physiological function of mammalian GDE6 has not been elucidated, its different pattern of tissue distribution suggests it might play a critical role in the completion of meiosis during male germ cell differentiation.','cd08610','GDPD_GDE6'),(990783,'gnl|CDD|176515',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE1 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE1 (also known as MIR16, membrane interacting protein of RGS16) and their metazoan homologs. GDE1 is widely expressed in mammalian tissues, including the heart, brain, liver, and kidney. It shows sequence homology to bacterial glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46), which catalyzes the hydrolysis of various glycerophosphodiesters, and produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. GDE1 has been characterized as GPI-GDE (EC 3.1.4.44) that selectively hydrolyzes extracellular glycerophosphoinositol (GPI) to generate glycerol phosphate and inositol. It functions as an integral membrane-bound glycoprotein interacting with regulator of G protein signaling protein RGS16, and is modulated by G protein-coupled receptor (GPCR) signaling. In addition, GDE1 may interact with PRA1 domain family, member 2 (PRAF2, also known as JM4), which is an interacting protein of the G protein-coupled chemokine receptor CCR5. The catalytic activity, which is dependent on the integrity of the GDPD domain, is required for GDE1 cellular function.','cd08573','GDPD_GDE1'),(990785,'gnl|CDD|176522',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial protein Rv2277c and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in uncharacterized bacterial protein Rv2277c and similar proteins. Members in this subfamily are bacterial homologous of mammalian GDE4, a transmembrane protein whose cellular function has not yet been elucidated.','cd08580','GDPD_Rv2277c_like'),(990787,'gnl|CDD|176506',_binary 'Glycerophosphodiester phosphodiesterase domain of Thermoanaerobacter tengcongensis and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Thermoanaerobacter tengcongensis glycerophosphodiester phosphodiesterase (TtGDE, EC 3.1.4.46) and its uncharacterized homologs. Members in this family show high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. Despite the fact that most of GDPD family members exist as the monomer, TtGDE can function as a dimeric unit. Its catalytic mechanism is based on the general base-acid catalysis, which is similar to that of phosphoinositide-specific phospholipases C (PI-PLCs, EC 3.1.4.11). A divalent metal cation is required for the enzyme activity of TtGDE.','cd08563','GDPD_TtGDE_like'),(990788,'gnl|CDD|176527',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial glycerophosphodiester phosphodiesterases.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized bacterial glycerophosphodiester phosphodiesterase and similar proteins. They show high sequence similarity with Escherichia coli glycerophosphodiester phosphodiesterase, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08585','GDPD_like_3'),(990789,'gnl|CDD|176516',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE2, GDE3, GDE6-like proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian glycerophosphodiester phosphodiesterase domain-containing protein subtype 5 (GDE2), subtype 2 (GDE3), subtype 1 (GDE6), and their eukaryotic homologs. Mammalian GDE2, GDE3, and GDE6 show very high sequence similarity to each other and have been classified into the same family. Although they are all transmembrane proteins, based on different pattern of tissue distribution, these enzymes might display diverse cellular functions. Mammalian GDE2 is primarily expressed in mature neurons. It selectively hydrolyzes glycerophosphocholine (GPC) and mainly functions in a complex with an antioxidant scavenger peroxiredoxin1 (Prdx1) to control motor neuron differentiation in the spinal cord.  Mammalian GDE3 is specifically expressed in bone tissues and spleen. It selectively hydrolyzes extracellular glycerophosphoinositol (GPI) to generate inositol 1-phosphate (Ins1P) and glycerol and functions as an inducer of osteoblast differentiation. Mammalian GDE6 is predominantly expressed in the spermatocytes of testis, and its specific physiological function has not been elucidated yet.','cd08574','GDPD_GDE_2_3_6'),(990860,'gnl|CDD|274805',_binary 'putative zinc finger/helix-turn-helix protein, YgiT family.  This model describes a family of predicted regulatory proteins with a conserved zinc finger/HTH architecture. The amino-terminal region contains a novel domain, featuring two CXXC motifs and occuring in a number of small bacterial proteins as well as in the present family. The carboxyl-terminal region consists of a helix-turn-helix domain, modeled by pfam01381. The predicted function is DNA binding and transcriptional regulation.','TIGR03830','transcriptional_regulator'),(990911,'gnl|CDD|275121',_binary 'tungsten cofactor oxidoreducase radical SAM maturase.  Members of this family are radical SAM enzymes involved in the maturation of tungsten (W)-containing cofactors in the enzymes aldehyde ferredoxin oxidoreductase, formaldehyde ferredoxin oxidoreductase, and others, and tend to be encoded by an adjacent gene.','TIGR04317','W_rSAM_matur'),(991218,'gnl|CDD|316403',_binary 'ESCO1/2 acetyl-transferase.','pfam13880','Acetyltransf_13'),(991221,'gnl|CDD|224167',_binary 'N-acetylglutamate synthase and related acetyltransferases [Amino acid transport and metabolism].','COG1246','ArgA'),(991222,'gnl|CDD|173926',_binary 'N-Acyltransferase superfamily: Various enzymes that characteristically catalyze the transfer of an acyl group to a substrate.  NAT (N-Acyltransferase) is a large superfamily of enzymes that mostly catalyze the transfer of an acyl group to a substrate and are implicated in a variety of functions, ranging from bacterial antibiotic resistance to circadian rhythms in mammals. Members include GCN5-related N-Acetyltransferases (GNAT) such as Aminoglycoside N-acetyltransferases, Histone N-acetyltransferase (HAT) enzymes, and Serotonin N-acetyltransferase, which catalyze the transfer of an acetyl group to a substrate. The kinetic mechanism of most GNATs involves the ordered formation of a ternary complex: the reaction begins with Acetyl Coenzyme A (AcCoA) binding, followed by binding of substrate, then direct transfer of the acetyl group from AcCoA to the substrate, followed by product and subsequent CoA release. Other family members include Arginine/ornithine N-succinyltransferase, Myristoyl-CoA: protein N-myristoyltransferase, and Acyl-homoserinelactone synthase which have a similar catalytic mechanism but differ in types of acyl groups transferred. Leucyl/phenylalanyl-tRNA-protein transferase and FemXAB nonribosomal peptidyltransferases which catalyze similar peptidyltransferase reactions are also included.','cd04301','NAT_SF'),(991261,'gnl|CDD|238866',_binary 'sialate_O-acetylesterase_like subfamily of the SGNH-hydrolases, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01828','sialate_O-acetylesterase_'),(991790,'gnl|CDD|224450',_binary 'DNA repair photolyase [DNA replication, recombination, and repair].','COG1533','SplB'),(992911,'gnl|CDD|273106',_binary 'SOS regulatory protein LexA.  LexA acts as a homodimer to repress a number of genes involved in the response to DNA damage (SOS response), including itself and RecA. RecA, in the presence of single-stranded DNA, acts as a co-protease to activate a latent autolytic protease activity (EC 3.4.21.88) of LexA, where the active site Ser is part of LexA. The autolytic cleavage site is an Ala-Gly bond in LexA (at position 84-85 in E. coli LexA; this sequence is replaced by Gly-Gly in Synechocystis). The cleavage leads to derepression of the SOS regulon and eventually to DNA repair. LexA in Bacillus subtilis is called DinR. LexA is much less broadly distributed than RecA. [DNA metabolism, DNA replication, recombination, and repair, Regulatory functions, DNA interactions].','TIGR00498','LexA_repressor.'),(992912,'gnl|CDD|201938',_binary 'LexA DNA binding domain.  This is the DNA binding domain of the LexA SOS regulon repressor which prevents expression of DNA repair proteins. The aligned region contains a variant form of the helix-turn-helix DNA binding motif. This domain is found associated with pfam00717 the auto-proteolytic domain of LexA EC:3.4.21.88.','pfam01726','LexA_DNA_bind'),(992913,'gnl|CDD|234688',_binary 'LexA repressor; Validated.','PRK00215','PRK00215'),(992914,'gnl|CDD|224885',_binary 'SOS-response transcriptional repressors (RecA-mediated autopeptidases) [Transcription / Signal transduction mechanisms].','COG1974','LexA'),(992915,'gnl|CDD|171489',_binary 'LexA repressor; Provisional.','PRK12423','PRK12423'),(993043,'gnl|CDD|185524',_binary 'protein tyrosine phosphatase; Provisional.','PTZ00242','PTZ00242'),(993096,'gnl|CDD|313302',_binary 'Protein of unknown function (DUF2283).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10049','DUF2283'),(993216,'gnl|CDD|316375',_binary 'Growth-arrest specific micro-tubule binding.  This family is the highly conserved central region of a number of metazoan proteins referred to as growth-arrest proteins. In mouse, Gas8 is predominantly a testicular protein, whose expression is developmentally regulated during puberty and spermatogenesis. In humans, it is absent in infertile males who lack the ability to generate gametes. The localization of Gas8 in the motility apparatus of post-meiotic gametocytes and mature spermatozoa, together with the detection of Gas8 also in cilia at the apical surfaces of epithelial cells lining the pulmonary bronchi and Fallopian tubes suggests that the Gas8 protein may have a role in the functioning of motile cellular appendages. Gas8 is a microtubule-binding protein localized to regions of dynein regulation in mammalian cells.','pfam13851','GAS'),(993220,'gnl|CDD|310310',_binary 'HOOK protein.  This family consists of several HOOK1, 2 and 3 proteins from different eukaryotic organisms. The different members of the human gene family are HOOK1, HOOK2 and HOOK3. Different domains have been identified in the three human HOOK proteins, and it was demonstrated that the highly conserved NH2-domain mediates attachment to microtubules, whereas the central coiled-coil motif mediates homodimerisation and the more divergent C-terminal domains are involved in binding to specific organelles (organelle-binding domains). It has been demonstrated that endogenous HOOK3 binds to Golgi membranes, whereas both HOOK1 and HOOK2 are localized to discrete but unidentified cellular structures. In mice the Hook1 gene is predominantly expressed in the testis. Hook1 function is necessary for the correct positioning of microtubular structures within the haploid germ cell. Disruption of Hook1 function in mice causes abnormal sperm head shape and fragile attachment of the flagellum to the sperm head.','pfam05622','HOOK'),(993224,'gnl|CDD|224755',_binary 'Phage shock protein A (IM30), suppresses sigma54-dependent transcription [Transcription / Signal transduction mechanisms].','COG1842','PspA'),(993226,'gnl|CDD|317892',_binary 'Domain of unknown function (DUF4659).  This family of proteins is found in eukaryotes. Proteins in this family are typically between 427 and 674 amino acids in length. There are two completely conserved residues (D and I) that may be functionally important.','pfam15558','DUF4659'),(993243,'gnl|CDD|253442',_binary 'O-phosphoseryl-tRNA(Sec) selenium transferase, SepSecS.  Early annotation suggested this family, SepSecS, of several eukaryotic and archaeal proteins, was involved in antigen-antibodies responses in the liver and pancreas. Structural studies show that the family is O-phosphoseryl-tRNA(Sec) selenium transferase, an enzyme involved in the synthesis of the amino acid selenocysteine (Sec). Sec is the only amino acid whose biosynthesis occurs on its cognate transfer RNA (tRNA). SepSecS catalyzes the final step in the formation of the amino acid. The early observation that autoantibodies isolated from patients with type I autoimmune hepatitis targeted a ribonucleoprotein complex containing tRNASec led to the identification and characterization of the archaeal and the human SepSecS. SepSecS forms its own branch in the family of fold-type I pyridoxal phosphate (PLP) enzymes that goes back to the last universal common ancestor which explains why the archaeal sequences spcS and MK0229 are annotated as being pyridoxal phosphate-dependent enzymes.','pfam05889','SepSecS'),(993542,'gnl|CDD|223415',_binary 'Site-specific DNA methylase [DNA replication, recombination, and repair].','COG0338','Dam'),(993557,'gnl|CDD|306682',_binary 'Bacterial DNA-binding protein.','pfam00216','Bac_DNA_binding'),(993558,'gnl|CDD|259853',_binary 'histone-like DNA-binding protein HU.  This subfamily includes HU and HU-like domains. HU is a conserved nucleoid-associated protein (NAP) which binds non-specifically to duplex DNA with a particular preference for targeting nicked and bent DNA. It is highly basic and contributes to chromosomal compaction and maintenance of negative supercoiling, thus often referred to as histone-like protein. HU can induce DNA bends, condense DNA in a fiber and also interact with single stranded DNA. It contains two homologous subunits, alpha and beta, typically forming homodimers (alpha-alpha and beta-beta), except in E. coli and other enterobacteria, which form heterodimers (alpha-beta). In E. coli, HU binds uniformly to the chromosome, with a preference for damaged or distorted DNA structures and can introduce negative supercoils into closed circular DNA in the presence of topoisomerase I. Anabaena HU (AHU) shows preference for A/T-rich region in the center of its DNA binding site.','cd13831','HU'),(993559,'gnl|CDD|223847',_binary 'Bacterial nucleoid DNA-binding protein [DNA replication, recombination, and repair].','COG0776','HimA'),(993560,'gnl|CDD|259852',_binary 'DNA sequence specific (IHF) and non-specific (HU) domains.  This family includes integration host factor (IHF) and HU, also called type II DNA-binding proteins (DNABII), which are small dimeric proteins that specifically bind the DNA minor groove, inducing large bends in the DNA and serving as architectural factors in a variety of cellular processes such as recombination, initiation of replication/transcription and gene regulation. IHF binds DNA in a sequence specific manner while HU displays little or no sequence preference. IHF homologs are usually heterodimers, while HU homologs are typically homodimers (except HU heterodimers from E. coli and other enterobacteria). HU is highly basic and contributes to chromosomal compaction and maintenance of negative supercoiling, thus often referred to as histone-like protein. IHF is an essential cofactor in phage lambda site-specific recombination, having an architectural role during assembly of specialized nucleoprotein structures (snups). Bacillus phage SPO1-encoded transcription factor 1 (TF1) is another related type II DNA-binding protein. Like IHF, TF1 binds DNA specifically and bends DNA sharply.','cd00591','HU_IHF'),(993561,'gnl|CDD|197709',_binary 'bacterial (prokaryotic) histone like domain.','smart00411','BHL'),(993562,'gnl|CDD|259859',_binary 'Bacteriophage SPO1-encoded TF1 binds and bends DNA.  This group contains proteins related to bacillus phage SPO1-encoded transcription factor 1 (TF1),  a type II DNA-binding protein related to the DNA sequence specific (IHF) and non-specific (HU) domains. Type II DNA-binding proteins bind and bend DNA as dimers. Like IHF, TF1 binds DNA specifically and bends DNA sharply.  Bacteriophage SPO1-encoded TF1 recognizes SPO1 phage DNA containing 5-(hydroxymethyl)-2\'-deoxyuridine as opposed to thymine,   Related  family members includes integration host factor (IHF) and HU, also called type II DNA-binding proteins (DNABII), which are small dimeric proteins that specifically bind the DNA minor groove, inducing large bends in the DNA and serving as architectural factors in a variety of cellular processes such as recombination, initiation of replication/transcription and gene regulation. IHF binds DNA in a sequence specific manner while HU displays little or no sequence preference. IHF homologs are usually heterodimers, while HU homologs are typically homodimers (except HU heterodimers from E. coli and other enterobacteria). HU is highly basic and contributes to chromosomal compaction and maintenance of negative supercoiling, thus often referred to as histone-like protein. IHF is an essential cofactor in phage lambda site-specific recombination, having an architectural role during assembly of specialized nucleoprotein structures (snups).','cd14435','SPO1_TF1_like'),(993563,'gnl|CDD|259854',_binary 'Integration host factor (IHF) and similar proteins.  This subfamily includes integration host factor (IHF) and IHF-like domains. IHF is a nucleoid-associated protein (NAP) that binds and sharply bends many DNA targets in a sequence specific manner. It is a heterodimeric protein composed of two highly homologous subunits IHFA (IHF-alpha) and IHFB (IHF-beta). It is known to act as a transcription factor at many gene regulatory regions in E. coli. IHF is an essential cofactor in phage lambda site-specific recombination, having an architectural role during assembly of specialized nucleoprotein structures (snups). IHF is also involved in formation as well as maintenance of bacterial biofilms since it is found in complex with extracellular DNA (eDNA) within the extracellular polymeric substances (EPS) matrix of many biofilms. This subfamily also includes the protein Hbb from tick-borne spirochete Borrelia burgdorferi, responsible for causing Lyme disease in humans. Hbb, a homodimer, shows DNA sequence preferences that are related, yet distinct from those of IHF.','cd13832','IHF'),(993564,'gnl|CDD|259858',_binary 'Beta subunit of integration host factor (IHFB).  This subfamily consists of the beta subunit of integration host factor (IHF) and IHF-like domains. IHF is a nucleoid-associated protein (NAP) that binds and sharply bends many DNA targets in a sequence specific manner. It is a heterodimeric protein composed of two highly homologous subunits IHFA (IHF-alpha) and IHFB (IHF-beta). It is known to act as a transcription factor at many gene regulatory regions in E. coli. IHF is an essential cofactor in phage lambda site-specific recombination, having an architectural role during assembly of specialized nucleoprotein structures (snups). IHF is also involved in formation as well as maintenance of bacterial biofilms since it is found in complex with extracellular DNA (eDNA) within the extracellular polymeric substances (EPS) matrix of many biofilms.','cd13836','IHF_B'),(993565,'gnl|CDD|259857',_binary 'Alpha subunit of integration host factor (IHFA).  This subfamily consists of the alpha subunit of integration host factor (IHF) and IHF-like domains. IHF is a nucleoid-associated protein (NAP) that binds and sharply bends many DNA targets in a sequence specific manner. It is a heterodimeric protein composed of two highly homologous subunits IHFA (IHF-alpha) and IHFB (IHF-beta). It is known to act as a transcription factor at many gene regulatory regions in E. coli. IHF is an essential cofactor in phage lambda site-specific recombination, having an architectural role during assembly of specialized nucleoprotein structures (snups). IHF is also involved in formation as well as maintenance of bacterial biofilms since it is found in complex with extracellular DNA (eDNA) within the extracellular polymeric substances (EPS) matrix of many biofilms.','cd13835','IHF_A'),(993566,'gnl|CDD|130061',_binary 'integration host factor, beta subunit.  This protein forms a site-specific DNA-binding heterodimer with the homologous integration host factor alpha subunit. It is closely related to the DNA-binding protein HU. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00988','Integration_host_factor_s'),(993567,'gnl|CDD|178925',_binary 'integration host factor subunit beta; Reviewed.','PRK00199','ihfB'),(993568,'gnl|CDD|178961',_binary 'integration host factor subunit alpha; Reviewed.','PRK00285','ihfA'),(993571,'gnl|CDD|311157',_binary 'Protein of unknown function (DUF1320).  This family consists of both hypothetical bacterial and phage proteins of around 145 residues in length. The function of this family is unknown.','pfam07030','DUF1320'),(993588,'gnl|CDD|306664',_binary 'Bacterial regulatory proteins, luxR family.','pfam00196','GerE'),(993595,'gnl|CDD|312292',_binary 'Primase C terminal 1 (PriCT-1).  This alpha helical domain is found at the C terminal of primases.','pfam08708','PriCT_1'),(993613,'gnl|CDD|314272',_binary 'Protein of unknown function (DUF3090).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11290','DUF3090'),(993617,'gnl|CDD|309029',_binary 'ADP-ribosylglycohydrolase.  This family includes enzymes that ADP-ribosylations, for example ADP-ribosylarginine hydrolase EC:3.2.2.19 cleaves ADP-ribose-L-arginine. The family also includes dinitrogenase reductase activating glycohydrolase. Most surprisingly the family also includes jellyfish crystallins, these proteins appear to have lost the presumed active site residues.','pfam03747','ADP_ribosyl_GH'),(993618,'gnl|CDD|224315',_binary 'ADP-ribosylglycohydrolase [Posttranslational modification, protein turnover, chaperones].','COG1397','DraG'),(993619,'gnl|CDD|131710',_binary 'ADP-ribosyl-[dinitrogen reductase] hydrolase.  Members of this family are the enzyme ADP-ribosyl-[dinitrogen reductase] hydrolase (EC 3.2.2.24), better known as Dinitrogenase Reductase Activating Glycohydrolase, DRAG. This enzyme reverses a regulatory inactivation of dinitrogen reductase caused by the action of NAD(+)--dinitrogen-reductase ADP-D-ribosyltransferase (EC 2.4.2.37) (DRAT). This enzyme is restricted to nitrogen-fixing bacteria and belongs to the larger family of ADP-ribosylglycohydrolases described by pfam03747. [Central intermediary metabolism, Nitrogen fixation].','TIGR02662','ADP-ribosyl-_glycohydrola'),(993847,'gnl|CDD|275056',_binary 'SH3 domain protein.  Members of this protein family have a signal peptide, a strongly conserved SH3 domain, a variable region, and then a C-terminal hydrophobic transmembrane alpha helix region.','TIGR04211','hypothetical_protein'),(994308,'gnl|CDD|226971',_binary 'Predicted soluble lytic transglycosylase fused to an ABC-type amino acid-binding protein [Cell envelope biogenesis, outer membrane].','COG4623','COG4623'),(994671,'gnl|CDD|133381',_binary 'Helix-Turn-Helix DNA binding domain of the transcription regulators TipAL, Mta, and SkgA.  Helix-turn-helix (HTH) TipAL, Mta, and SkgA transcription regulators, and related proteins, N-terminal domain. TipAL regulates resistance to and activation by numerous cyclic thiopeptide antibiotics, such as thiostrepton. Mta is a global transcriptional regulator; the N-terminal DNA-binding domain of Mta interacts directly with the promoters of mta, bmr, blt, and ydfK, and induces transcription of these multidrug-efflux transport genes. SkgA has been shown to control stationary-phase expression of catalase-peroxidase in Caulobacter crescentus. These proteins are comprised of distinct domains that harbor an  N-terminal active (DNA-binding) site and a regulatory (effector-binding) site. The conserved N-terminal domain of these transcription regulators contains winged HTH motifs that mediate DNA binding. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Unique to this family, is a TipAL-like, lineage specific Bacilli subgroup, which has five conserved cysteines in the C-terminus of the protein.','cd01106','HTH_TipAL-Mta'),(994673,'gnl|CDD|133388',_binary 'Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily.  Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 3). Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd01282','HTH_MerR-like_sg3'),(994674,'gnl|CDD|133392',_binary 'Helix-Turn-Helix DNA binding domain of putative MlrA-like transcription regulators.  Putative helix-turn-helix (HTH) MlrA-like transcription regulators (subgroup 1). The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules. Many MlrA-like proteins in this group appear to lack the long dimerization helix seen in the N-terminal domains of typical MerR-like proteins.','cd04764','HTH_MlrA-like_sg1'),(994675,'gnl|CDD|133415',_binary 'Helix-Turn-Helix DNA binding domain of the transcription regulators NolA and AlbR.  Helix-turn-helix (HTH) transcription regulators NolA and AlbR, N-terminal domain. In Bradyrhizobium (Arachis) sp. NC92, NolA is required for efficient nodulation of host plants. In Xanthomonas albilineans, AlbR regulates the expression of the pathotoxin, albicidin. These proteins are putatively comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains are often unrelated and bind specific coactivator molecules. They share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.','cd04788','HTH_NolA-AlbR'),(994677,'gnl|CDD|184964',_binary 'redox-sensitivie transcriptional activator SoxR; Provisional.','PRK15002','PRK15002'),(995421,'gnl|CDD|226079',_binary 'Plasmid maintenance system killer protein [General function prediction only].','COG3549','HigB'),(995422,'gnl|CDD|282822',_binary 'RelE-like toxin of type II toxin-antitoxin system HigB.  This family carries several different examples of type II bacterial toxins of toxin-antitoxin systems including many HigB-like ones. The fold is referred to as the RelE/YoeB/Txe/Yeeu fold suggesting all examples of these are present in this family. Several plasmids with proteic killer gene-systems have been reported. All of them encode a stable toxin and an unstable antidote. Upon loss of the plasmid, the less stable inhibitor is inactivated more rapidly than the toxin, allowing the toxin to be activated. The activation of these systems results in cell filamentation and cessation of viable cell production. It has been verified that both the stable killer and the unstable inhibitor of the systems are short polypeptides. This family corresponds to the toxin.','pfam05015','HigB-like_toxin'),(995435,'gnl|CDD|308809',_binary 'Peptidase C39 family.  Lantibiotic and non-lantibiotic bacteriocins are synthesized as precursor peptides containing N-terminal extensions (leader peptides) which are cleaved off during maturation. Most non-lantibiotics and also some lantibiotics have leader peptides of the so-called double-glycine type. These leader peptides share consensus sequences and also a common processing site with two conserved glycine residues in positions -1 and -2. The double- glycine-type leader peptides are unrelated to the N-terminal signal sequences which direct proteins across the cytoplasmic membrane via the sec pathway. Their processing sites are also different from typical signal peptidase cleavage sites, suggesting that a different processing enzyme is involved. Peptide bacteriocins are exported across the cytoplasmic membrane by a dedicated ATP-binding cassette (ABC) transporter. The ABC transporter is the maturation protease and its proteolytic domain resides in the N-terminal part of the protein. This peptidase domain is found in a wide range of ABC transporters, however the presumed catalytic cysteine and histidine are not conserved in all members of this family.','pfam03412','Peptidase_C39'),(995437,'gnl|CDD|239099',_binary 'A sub-family of peptidase family C39. Peptidase family C39 mostly contains bacteriocin-processing endopeptidases from bacteria. The cysteine peptidases in family C39 cleave the \'double-glycine\' leader peptides from the precursors of various bacteriocins (mostly non-lantibiotic). The cleavage is mediated by the transporter as part of the secretion process. Bacteriocins are antibiotic proteins secreted by some species of bacteria that inhibit the growth of other bacterial species. The bacteriocin is synthesized as a precursor with an N-terminal leader peptide, and processing involves removal of the leader peptide by cleavage at a Gly-Gly bond, followed by translocation of the mature bacteriocin across the cytoplasmic membrane. Most endopeptidases of family C39 are N-terminal domains in larger proteins (ABC transporters) that serve both functions. The proposed protease active site is conserved in this sub-family.','cd02418','Peptidase_C39B'),(995466,'gnl|CDD|225177',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG2268','COG2268'),(995468,'gnl|CDD|235334',_binary 'electron transport complex protein RnfC; Provisional.','PRK05035','PRK05035'),(995560,'gnl|CDD|193534',_binary 'M18 Peptidase Aspartyl aminopeptidase.  Peptidase M18 family, Aspartyl aminopeptidase (DAP; EC  3.4.11.21) subfamily. DAP is widely distributed in bacteria and eukaryotes, but only the mammalian aspartyl aminopeptidase has been characterized to date. DAP cleaves only unblocked N-terminal acidic amino-acid residues. It is a cytosolic enzyme and is highly conserved; for example, the human enzyme has 51% identity to an aspartyl aminopeptidase-like protein in Arabidopsis thaliana. The mammalian DAP is highly selective for hydrolysis of N-terminal aspartate or glutamate residues from peptides. Unlike glutamyl aminopeptidase (M42), DAP does not cleave simple aminoaryl-arylamide substrates. Although there is lack of understanding of the function of this enzyme, it is thought to act in concert with other aminopeptidases to facilitate protein turnover because of their restricted specificities for the N-terminal aspartic and glutamic acid, which cannot be cleaved by any other aminopeptidases. The mammalian aspartyl aminopeptidase is possibly contributing to the catabolism of peptides, including those produced by the proteasome. It may also trim the N-terminus of peptides that are intended for the MHC class I system.  In humans, DAP has been implicated in the the specific function of converting angiotensin II to the vasoactive angiotensin III within the brain.','cd05658','M18_DAP'),(995561,'gnl|CDD|224281',_binary 'Aspartyl aminopeptidase [Amino acid transport and metabolism].','COG1362','LAP4'),(995562,'gnl|CDD|235073',_binary 'putative aminopeptidase 2; Provisional.','PRK02813','PRK02813'),(995590,'gnl|CDD|227406',_binary 't-SNARE complex subunit, syntaxin [Intracellular trafficking and secretion].','COG5074','COG5074'),(995738,'gnl|CDD|235648',_binary 'RNA polymerase factor sigma-54; Reviewed.','PRK05932','PRK05932'),(995763,'gnl|CDD|226180',_binary 'Prophage maintenance system killer protein [General function prediction only].','COG3654','Doc'),(995765,'gnl|CDD|273687',_binary 'death-on-curing family protein.  The characterized member of this family is the death-on-curing (DOC) protein of phage P1. It is part of a two protein operon with prevents-host-death (phd) that forms an addiction module. DOC lacks homology to analogous addiction module post-segregational killing proteins involved in plasmid maintenance. These modules work as a combination of a long lived poison (e.g. this protein) and a more abundant but shorter lived antidote. Members of this family have a well-conserved central motif HxFx[ND][AG]NKR. A similar region, with K replaced by G, is found in the huntingtin interacting protein (HYPE) family. [Unknown function, General].','TIGR01550','Protein_kinase_doc'),(995774,'gnl|CDD|236491',_binary 'elongation factor Ts; Provisional.','PRK09377','tsf'),(995778,'gnl|CDD|274788',_binary 'NHLM bacteriocin system ABC transporter, peptidase/ATP-binding protein.  This protein describes a multidomain ABC transporter subunit that is one of three protein families associated with some regularity with a distinctive family of putative bacteriocins. It includes a bacteriocin-processing peptidase domain at the N-terminus. Model TIGR03793 describes a conserved propeptide region for this bacteriocin family, unusual because it shows obvious homology a region of the enzyme nitrile hydratase up to the classic Gly-Gly cleavage motif. This family is therefore predicted to be a subunit of a bacteriocin processing and export system characteristic to this system that we designate NHLM, Nitrile Hydratase Leader Microcin. [Transport and binding proteins, Amino acids, peptides and amines, Cellular processes, Biosynthesis of natural products].','TIGR03796','ABC_transporter_related'),(995807,'gnl|CDD|224425',_binary 'DNA-directed RNA polymerase specialized sigma subunit, sigma54 homolog [Transcription].','COG1508','RpoN'),(995894,'gnl|CDD|227066',_binary 'Phage-related protein [Function unknown].','COG4722','COG4722'),(996054,'gnl|CDD|313022',_binary 'Transmembrane protein.  This entry is a highly conserved protein present in eukaryotes.','pfam09726','Macoilin'),(996404,'gnl|CDD|314281',_binary 'Protein of unknown function (DUF3102).  This family of proteins has no known function.','pfam11300','DUF3102'),(996426,'gnl|CDD|226603',_binary 'Antitoxin of toxin-antitoxin stability system [Cell division and chromosome partitioning].','COG4118','Phd'),(996427,'gnl|CDD|308945',_binary 'Carbohydrate esterase, sialic acid-specific acetylesterase.  The catalytic triad of this esterase enzyme comprises residues Ser127, His403 and Asp391 in UniProtKB:P70665.','pfam03629','SASA'),(996431,'gnl|CDD|318076',_binary 'ParE-like toxin of type II bacterial toxin-antitoxin system.','pfam15781','ParE-like_toxin'),(996432,'gnl|CDD|211740',_binary 'addiction module toxin, RelE/StbE family.  Plasmids may be maintained stably in bacterial populations through the action of addiction modules, in which a toxin and antidote are encoded in a cassette on the plasmid. In any daughter cell that lacks the plasmid, the toxin persists and is lethal after the antidote protein is depleted. Toxin/antitoxin pairs are also found on main chromosomes, and likely represent selfish DNA. Sequences in the seed for this alignment all are found adjacent to RelB/DinJ family antitoxin genes (TIGR02384), as are most genes found by the resulting model. StbE from Morganella morganii plasmid R485 shows typical behaviour for an addiction module toxin. It cannot be cloned without its partner (the antitoxin), whereas its partner cannot confer plasmid stability without StbE. [Cellular processes, Toxin production and resistance, Mobile and extrachromosomal element functions, Other].','TIGR02385','Hypothetical_relE_protein'),(997140,'gnl|CDD|182158',_binary 'DNA-binding transcriptional repressor PuuR; Provisional.','PRK09943','PRK09943'),(997198,'gnl|CDD|311960',_binary 'Peptidase M15.','pfam08291','Peptidase_M15_3'),(997200,'gnl|CDD|225650',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3108','COG3108'),(997201,'gnl|CDD|283577',_binary 'Bacterial protein of unknown function (DUF882).  This family consists of a series of hypothetical bacterial proteins of unknown function.','pfam05951','Peptidase_M15_2'),(997221,'gnl|CDD|179316',_binary 'transcriptional regulator Spx; Reviewed.','PRK01655','spxA'),(997225,'gnl|CDD|239333',_binary 'Arsenate Reductase (ArsC) family, Yffb subfamily; Yffb is an uncharacterized bacterial protein encoded by the yffb gene, related to the thioredoxin-fold arsenic reductases, ArsC. The structure of Yffb and the conservation of the catalytic cysteine suggest that it is likely to function as a glutathione (GSH)-dependent thiol reductase. ArsC catalyzes the reduction of arsenate [As(V)] to arsenite [As(III)], using reducing equivalents derived from GSH via glutaredoxin, through a single catalytic cysteine.','cd03035','ArsC_Yffb'),(997226,'gnl|CDD|79035',_binary 'transcriptional regulator Spx; Provisional.','PRK12559','PRK12559'),(997234,'gnl|CDD|237125',_binary 'RNA polymerase sigma factor; Provisional.','PRK12519','PRK12519'),(997236,'gnl|CDD|131893',_binary 'RNA polymerase sigma-K factor.  The sporulation-specific transcription factor sigma-K (also called sigma-27) is expressed in the mother cell compartment of endospore-forming bacteria such as Bacillus subtilis. Like its close homolog sigma-E (sigma-29) (see TIGR02835), also specific to the mother cell compartment, it must be activated by a proteolytic cleavage. Note that in Bacillus subtilis (and apparently also Clostridium tetani), but not in other endospore forming species such as Bacillus anthracis, the sigK gene is generated by a non-germline (mother cell only) chromosomal rearrangement that recombines coding regions for the N-terminal and C-terminal regions of sigma-K. [Transcription, Transcription factors, Cellular processes, Sporulation and germination].','TIGR02846','RNA_polymerase_sigma_subu'),(997238,'gnl|CDD|236228',_binary 'sporulation sigma factor SigE; Reviewed.','PRK08301','PRK08301'),(997240,'gnl|CDD|180266',_binary 'sporulation sigma factor SigK; Reviewed.','PRK05803','PRK05803'),(997297,'gnl|CDD|179584',_binary 'inorganic polyphosphate/ATP-NAD kinase; Provisional.','PRK03501','ppnK'),(997897,'gnl|CDD|274819',_binary 'FMN-dependent oxidoreductase, nitrilotriacetate monooxygenase family.  This model represents a distinctive clade, in which all characterized members are FMN-binding, within the larger family of luciferase-like monooxygenases (LLM), among which there are both FMN- and F420-binding enzymes. A well-characterized member is nitrilotriacetate monooxygenase from Aminobacter aminovorans (Chelatobacter heintzii), where nitrilotriacetate is a chelating agent used in detergents. [Unknown function, Enzymes of unknown specificity].','TIGR03860','FMN_nitrolo'),(997958,'gnl|CDD|178419',_binary 'hydrolase, alpha/beta fold family protein.','PLN02824','PLN02824'),(998009,'gnl|CDD|224788',_binary 'D-alanyl-D-alanine carboxypeptidase [Cell envelope biogenesis, outer membrane].','COG1876','VanY'),(998012,'gnl|CDD|285916',_binary 'dUTPase.  2-Deoxyuridine 5-triphosphate nucleotidohydrolase (dUTPase) catalyzes the hydrolysis of dUTP to dUMP and pyrophosphate (EC:3.6.1.23). Members of this family have a novel all-alpha fold and are unrelated to the all-beta fold found in dUTPases of the majority of organisms. This family contains both dUTPase homologs of dUTPase including dCTPase of phage T4.','pfam08761','dUTPase_2'),(998059,'gnl|CDD|311949',_binary 'DNA primase catalytic core, N-terminal domain.','pfam08275','Toprim_N'),(998496,'gnl|CDD|309549',_binary 'tRNAHis guanylyltransferase.  The Thg1 protein from Saccharomyces cerevisiae is responsible for adding a GMP residue to the 5\' end of tRNA His. The catalytic domain Thg1 contains a RRM (ferredoxin) fold palm domain, just like the viral RNA-dependent RNA polymerases, reverse transcriptases, family A and B DNA polymerases, adenylyl cyclases, diguanylate cyclases (GGDEF domain) and the predicted polymerase of the CRISPR system. Thg1 possesses an active site with three acidic residues that chelate Mg++ cations. Thg1 catalyzes polymerization similar to the 5\'-3\' polymerases.','pfam04446','Thg1'),(998497,'gnl|CDD|226508',_binary 'Uncharacterized conserved protein [Function unknown].','COG4021','COG4021'),(998609,'gnl|CDD|177849',_binary 'glutathione gamma-glutamylcysteinyltransferase.','PLN02198','PLN02198'),(998934,'gnl|CDD|223056',_binary 'single-stranded binding protein UL29; Provisional.','PHA03362','PHA03362'),(999546,'gnl|CDD|224001',_binary 'Predicted acetyltransferases and hydrolases with the alpha/beta hydrolase fold [General function prediction only].','COG1075','LipA'),(1000618,'gnl|CDD|313100',_binary 'Essential protein Yae1, N terminal.  Members of this family are found in the N terminal region of the essential protein Yae1. Their exact function has not, as yet, been determined. The family DUF1715, pfam08215 has now been merged into this family.','pfam09811','Yae1_N'),(1000707,'gnl|CDD|316316',_binary 'Domain of unknown function (DUF4177).','pfam13783','DUF4177'),(1000771,'gnl|CDD|184905',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14941','PRK14941'),(1000915,'gnl|CDD|314336',_binary 'Protein of unknown function (DUF3168).  This family of proteins has no known function but is likely to be a component of bacteriophage.','pfam11367','DUF3168'),(1000917,'gnl|CDD|227724',_binary 'Predicted secreted protein [Function unknown].','COG5437','COG5437'),(1000918,'gnl|CDD|310649',_binary 'Phage tail tube protein.  characterized members are major tail tube proteins from various phages, including lactococcal temperate bacteriophage TP901-1.','pfam06199','Phage_tail_2'),(1000921,'gnl|CDD|309961',_binary 'Periplasmic copper-binding protein (NosD).  NosD is a periplasmic protein which is thought to insert copper into the exported reductase apoenzyme (NosZ). This region forms a parallel beta helix domain.','pfam05048','NosD'),(1000922,'gnl|CDD|315812',_binary 'Right handed beta helix region.  This region contains a parallel beta helix region that shares some similarity with Pectate lyases.','pfam13229','Beta_helix'),(1000973,'gnl|CDD|310254',_binary 'Phage head-tail joining protein.','pfam05521','Phage_H_T_join'),(1000974,'gnl|CDD|227901',_binary 'Bacteriophage head-tail adaptor [General function prediction only].','COG5614','COG5614'),(1000975,'gnl|CDD|273692',_binary 'phage head-tail adaptor, putative, SPP1 family.  This family describes a small protein of about 100 amino acids found in bacteriophage and in bacterial prophage regions. Examples include gp9 of phage HK022 and gp16 of phage SPP1. This minor structural protein is suggested to be a head-tail adaptor protein (although the source of this annotation was not traced during construction of this model). [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01563','hypothetical_protein_AGR_'),(1001599,'gnl|CDD|225145',_binary 'Predicted phosphoribosyltransferases [General function prediction only].','COG2236','COG2236'),(1001613,'gnl|CDD|238985',_binary 'Adenosine 5\'-phosphosulfate kinase (APSK) catalyzes the phosphorylation of adenosine 5\'-phosphosulfate to form 3\'-phosphoadenosine 5\'-phosphosulfate (PAPS). The end-product PAPS is a biologically \'activated\' sulfate form important for the assimilation of inorganic sulfate.','cd02027','APSK'),(1001730,'gnl|CDD|240612',_binary 'YadA-like, left-handed beta-roll.  This group contains the collagen-binding domain virulence factor YadA an adhesion proteins of several Yersinia species, and related cell surface proteins, including Moraxella catarrhalis UspA-like proteins. The collagen-binding portion is found in the hydrophobic N-terminal region. YadA forms a matrix on the bacterial outer membrane, which mediates binding to collagen and epithelial cells. YadA inhibits the complement-activating pathway with the coating of the cell surface with factor H, which impedes C3b molecules. These domains form a left handed beta roll made up of a series of short repeated elements. UspA1 and UspA2 are part of a class of pathogenicity factors that act as cell surface adhesion molecules, in which N-terminal head and neck domains extend from the bacterial outer membrane. The UspA1 head domain of Moraxella catarrhalis, is formed from trimeric left-handed parallel beta-helices of 14-16 amino acid repeats. The UspA1 head domain connects to a neck region of large extended, charged loops that maybe be ligand binding, which is in turn connected to an extended coiled coil domain that tethers the head and neck region to the cell surface via a transmembrane region.','cd12820','LbR_YadA-like'),(1001738,'gnl|CDD|226155',_binary 'Phage baseplate assembly protein W [General function prediction only].','COG3628','COG3628'),(1001746,'gnl|CDD|274373',_binary 'RNA polymerase sigma-70 factor, sigma-B/F/G subfamily.  This group of similar sigma-70 factors includes clades found in Bacilli (including the sporulation factors SigF:TIGR02885 and SigG:TIGR02850 as well as SigB:TIGR02941), and the high GC gram positive bacteria (Actinobacteria) where a variable number of them are found depending on the lineage.','TIGR02980','SigBFG'),(1001753,'gnl|CDD|182933',_binary 'ATP-dependent DNA helicase RecQ; Provisional.','PRK11057','PRK11057'),(1001754,'gnl|CDD|129701',_binary 'ATP-dependent DNA helicase, RecQ family.  All proteins in this family for which functions are known are 3\'-5\' DNA-DNA helicases. These proteins are used for recombination, recombinational repair, and possibly maintenance of chromosome stability. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00614','Werner_syndrome_ATP-depen'),(1001755,'gnl|CDD|273594',_binary 'ATP-dependent DNA helicase RecQ.  The ATP-dependent DNA helicase RecQ of E. coli is about 600 residues long. This model represents bacterial proteins with a high degree of similarity in domain architecture and in primary sequence to E. coli RecQ. The model excludes eukaryotic and archaeal proteins with RecQ-like regions, as well as more distantly related bacterial helicases related to RecQ. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01389','recQ'),(1001756,'gnl|CDD|214682',_binary '5\'-3\' exonuclease.','smart00475','53EXOc'),(1001759,'gnl|CDD|223336',_binary '5\'-3\' exonuclease (including N-terminal domain of PolI) [DNA replication, recombination, and repair].','COG0258','Exo'),(1001760,'gnl|CDD|237877',_binary '5\'-3\' exonuclease; Provisional.','PRK14976','PRK14976'),(1001761,'gnl|CDD|189029',_binary 'PIN domain of the 5\'-3\' exonuclease of Taq DNA polymerase I and homologs.  The 5\'-3\' exonuclease (53EXO) PIN (PilT N terminus) domain of multi-domain DNA polymerase I and single domain protein homologs are included in this family.  Taq contains a polymerase domain for synthesizing a new DNA strand and a 53EXO PIN domain for cleaving RNA primers or damaged DNA strands. Taq\'s 53EXO PIN domain recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp. The 53EXO PIN domain cleaves the unpaired 5\'-arm of the overlap flap DNA substrate. 5\'-3\' exonucleases are members of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. These nucleases contain a PIN domain with a helical arch/clamp region (I domain) of variable length (approximately 16 residues in 53EXO PIN domains) and a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region.  Both the H3TH domain (not included here) and the helical arch/clamp region are involved in DNA/RNA binding. The active site includes a set of conserved acidic residues that are essential for binding divalent metal ions required for nuclease activity.','cd09859','PIN_53EXO'),(1001762,'gnl|CDD|308394',_binary '5\'-3\' exonuclease, N-terminal resolvase-like domain.','pfam02739','5_3_exonuc_N'),(1001763,'gnl|CDD|181896',_binary 'flap endonuclease-like protein; Provisional.','PRK09482','PRK09482'),(1001764,'gnl|CDD|307498',_binary '5\'-3\' exonuclease, C-terminal SAM fold.','pfam01367','5_3_exonuc'),(1001765,'gnl|CDD|188618',_binary 'H3TH domain of the 5\'-3\' exonuclease of Taq DNA polymerase I and homologs.  H3TH (helix-3-turn-helix) domains of the 5\'-3\' exonuclease (53EXO) of mutli-domain DNA polymerase I and single domain protein homologs are included in this family. Taq DNA polymerase I contains a polymerase domain for synthesizing a new DNA strand and a 53EXO domain for cleaving RNA primers or damaged DNA strands. Taq\'s 53EXO recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp. The 53EXO cleaves the unpaired 5\'-arm of the overlap flap DNA substrate. 5\'-3\' exonucleases are members of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+ or Mn2+ or Zn2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and two Asp residues from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5\' nucleases.','cd09898','H3TH_53EXO'),(1001766,'gnl|CDD|189030',_binary 'PIN domain of bacteriophage T3, T4 RNase H, T5-5\'nuclease, and homologs.  PIN (PilT N terminus) domain of bacteriophage T5-5\'nuclease (5\'-3\' exonuclease or T5FEN), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar 5\' nucleases are included in this family. T5-5\'nuclease is a 5\'-3\'exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5\'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5\'- 3\'exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. Bacteriophage T3 is believed to function in the removal of DNA-linked RNA primers and is essential for phage DNA replication and also necessary for host DNA degradation and phage genetic recombination. These nucleases are members of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. They have a PIN domain with a helical arch/clamp region (I domain) of variable length (approximately 20 to 30 residues in PIN T5-like domains) and a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region.  Both the H3TH domain (not included here) and the helical arch/clamp region are involved in DNA/RNA binding. The active site includes a set of conserved acidic residues that are essential for binding divalent metal ions required for nuclease activity. In the T5-5\'nuclease, structure-specific endonuclease activity requires binding of a single metal ion in the high-affinity, metal binding site 1, whereas exonuclease activity requires both, the high-affinity, metal binding site 1 and the low-affinity, metal binding site 2 to be occupied by a divalent cofactor.  The T5-5\'nuclease is reported to be able to bind several metal ions including, Mg2+, Mn2+, Zn2+ and Co2+, as co-factors.','cd09860','PIN_T4-like'),(1001767,'gnl|CDD|189019',_binary 'PIN domains of the 5\'-3\' exonucleases of DNA polymerase I, bacteriophage T4 RNase H and T5-5\' nucleases, and homologs.  PIN (PilT N terminus) domains of the 5\'-3\' exonucleases (53EXO) of mutli-domain DNA polymerase I and single domain protein homologs, as well as, the PIN domains of bacteriophage T5-5\'nuclease (T5FEN or 5\'-3\'exonuclease), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar  nucleases are included in this family.  The 53EXO of DNA polymerase I recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp.  The T5-5\'nuclease is a 5\'-3\'exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5\'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5\'- 3\'exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. These nucleases are members of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. They contain a PIN domain with a helical arch/clamp region (I domain) of variable length (approximately 16 to 30 residues in 53EXO-like PIN domains) and a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region.  Both the H3TH domain (not included here) and the helical arch/clamp region are involved in DNA binding. The active site of the 53EXO of Taq DNA polymerase I includes a set of conserved acidic residues that are essential for binding three divalent metal ions (two Mn2+ ions and one Zn2+ ion) required for nuclease activity. T5-5\'nuclease requires at least two bound divalent metal ions for nuclease activity and is reported to be able to use Mg2+, Mn2+ or Co2+ as co-factors.','cd00008','PIN_53EXO-like'),(1001768,'gnl|CDD|188619',_binary 'H3TH domain of bacteriophage T3, T4 RNase H, T5-5\' nucleases, and homologs.  H3TH (helix-3-turn-helix) domains of bacteriophage T5-5\'nuclease (5\'-3\' exonuclease or T5FEN), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar 5\' nucleases are included in this family. The T5-5\'nuclease is a 5\'-3\' exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5\'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5\'- 3\' exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. Bacteriophage T3 is believed to function in the removal of DNA-linked RNA primers and is essential for phage DNA replication and also necessary for host DNA degradation and phage genetic recombination. These nucleases are members of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. They contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors required for nuclease activity. The first metal binding site (MBS-1) is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site (MBS-2) is composed generally of two Asp residues from the PIN domain and two Asp residues from the H3TH domain. In the T5-5\'nuclease, structure-specific endonuclease activity requires binding of a single metal ion in the high-affinity, MBS-1, whereas exonuclease activity requires both, the high-affinity, MBS-1 and the low-affinity, MBS-2 to be occupied by a divalent cofactor. The T5-5\'nuclease is reported to be able to bind several metal ions including, Mg2+, Mn2+, Zn2+ and Co2+, as co-factors. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5\' nucleases.','cd09899','H3TH_T4-like'),(1001769,'gnl|CDD|197623',_binary 'Helix-hairpin-helix class 2 (Pol1 family) motifs.','smart00279','HhH2'),(1001770,'gnl|CDD|188616',_binary 'H3TH domains of structure-specific 5\' nucleases (or flap endonuclease-1-like) involved in DNA replication, repair, and recombination.  The 5\' nucleases of this superfamily are capable of both 5\'-3\' exonucleolytic activity and cleaving bifurcated or branched DNA, in an endonucleolytic, structure-specific manner, and are involved in DNA replication, repair, and recombination. The superfamily includes the H3TH (helix-3-turn-helix) domains of Flap Endonuclease-1 (FEN1), Exonuclease-1 (EXO1), Mkt1, Gap Endonuclease 1 (GEN1) and Xeroderma pigmentosum complementation group G (XPG) nuclease. Also included are the H3TH domains of the 5\'-3\' exonucleases of DNA polymerase I and single domain protein homologs, as well as, the bacteriophage T4 RNase H, T5-5\'nuclease, and other homologs. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the C-terminal region of the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. Typically, the nucleases within this superfamily have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+, Mn2+, Zn2+, or Co2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one or two Asp residues from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5\' nucleases.','cd00080','H3TH_StructSpec-5\'-nuclea'),(1001771,'gnl|CDD|189023',_binary 'PIN domains of structure-specific 5\' nucleases (or flap endonuclease-1-like) involved in DNA replication, repair, and recombination.  Structure-specific 5\' nucleases are capable of both 5\'-3\' exonucleolytic activity and cleaving bifurcated or branched DNA, in an endonucleolytic, structure-specific manner. The family includes the PIN (PilT N terminus) domains of Flap Endonuclease-1 (FEN1), Exonuclease-1 (EXO1), Mkt1, Gap Endonuclease 1 (GEN1), and Xeroderma pigmentosum complementation group G (XPG) nuclease. Also included are the PIN domains of the 5\'-3\' exonucleases of DNA polymerase I and single domain protein homologs, as well as, the bacteriophage T4- and T5-5\' nucleases, and other homologs. These nucleases contain a PIN domain with a helical arch/clamp region (I domain) of variable length (approximately 16 to 800 residues) and a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region. Both the H3TH domain (not included here) and the helical arch/clamp region are involved in DNA binding. With the exception of Mkt1, the nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (Mg2+, Mn2+, Zn2+, or Co2+).','cd09853','PIN_StructSpec-5\'-nucleas'),(1001772,'gnl|CDD|188623',_binary 'H3TH domain of Flap Endonuclease-1, a structure-specific, divalent-metal-ion dependent, 5\' nuclease: Archaeal homologs.  Members of this subgroup include the H3TH (helix-3-turn-helix) domains of archaeal Flap endonuclease-1 (FEN1), 5\' nucleases. FEN1 is involved in multiple DNA metabolic pathways, including DNA replication processes (5\' flap DNA endonuclease activity and double stranded DNA 5\'-exonuclease activity) and DNA repair processes (long-patch base excision repair) in eukaryotes and archaea. Interaction between FEN1 and PCNA (Proliferating cell nuclear antigen) is an essential prerequisite to FEN1\'s DNA replication functionality and stimulates FEN1 nuclease activity by 10-50 fold. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this subfamily have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (Mg2+ or Mn2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one Asp residue from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5\' nucleases. Also, FEN1 has a C-terminal extension containing residues forming the consensus PIP-box - Qxx(M/L/I)xxF(Y/F) which serves to anchor FEN1 to PCNA.','cd09903','H3TH_FEN1-Arc'),(1001773,'gnl|CDD|222882',_binary 'RnaseH; Provisional.','PHA02567','rnh'),(1001785,'gnl|CDD|223080',_binary 'Glutamate-1-semialdehyde aminotransferase [Coenzyme metabolism].','COG0001','HemL'),(1001788,'gnl|CDD|183493',_binary 'glutamate-1-semialdehyde aminotransferase; Provisional.','PRK12389','PRK12389'),(1001799,'gnl|CDD|234607',_binary 'glutamate-1-semialdehyde aminotransferase; Provisional.','PRK00062','PRK00062'),(1001832,'gnl|CDD|178100',_binary 'glutamate-1-semialdehyde 2,1-aminomutase.','PLN02482','PLN02482'),(1001926,'gnl|CDD|99839',_binary 'DEDDh 3\'-5\' exonuclease domain of three prime repair exonuclease (TREX)1, TREX2, and similar proteins.  Three prime repair exonuclease (TREX)1 and TREX2 are closely related DEDDh-type DnaQ-like 3\'-5\' exonucleases. They contain three conserved sequence motifs known as ExoI, II, and III, with a specific Hx(4)D conserved pattern at ExoIII. These motifs contain four conserved acidic residues that participate in coordination of divalent metal ions required for catalysis. Both proteins play a role in the metabolism and clearance of DNA. TREX1 is the major 3\'-5\' exonuclease activity detected in mammalian cells. Mutations in the human TREX1 gene can cause Aicardi-Goutieres syndrome (AGS), which is characterized by perturbed innate immunity and presents itself as a severe neurological disease. TREX1 degrades ssDNA generated by aberrant replication intermediates to prevent checkpoint activation and autoimmune disease. There are distinct structural differences between TREX1 and TREX2 that point to different biological roles for these proteins. The main difference is the presence of about 70 amino acids at the C-terminus of TREX1. In addition, TREX1 has a nonrepetitive proline-rich region that is not present in the TREX2 protein. Furthermore, TREX2 contains a conserved DNA binding loop positioned adjacent to the active site that has a sequence distinct from the corresponding loop in TREX1. Truncations in the C-terminus of human TREX1 cause autosomal dominant retinal vasculopathy with cerebral leukodystrophy (RVCL), a neurovascular syndrome featuring a progressive loss of visual acuity combined with a variable neurological picture.','cd06136','TREX1_2'),(1001933,'gnl|CDD|212519',_binary 'Histone deacetylase class IV also known as histone deacetylase 11.  Class IV histone deacetylases (HDAC11; EC 3.5.1.98) are predicted Zn-dependent enzymes. This class includes animal HDAC11, plant HDA2 and related bacterial deacetylases. Enzymes in this subfamily participate in regulation of a number of different processes through protein modification (deacetylation). They catalyze hydrolysis of N(6)-acetyl-lysine of histones (or other proteins) to yield a deacetylated proteins. Histone deacetylases often act as members of large multi-protein complexes such as mSin3A or SMRT/N-CoR. Human HDAC11 does not associate with them but can interact with HDAC6 in vivo. It has been suggested that HDAC11 and HDAC6 may use non-histone proteins as their substrates and play a role other than to directly modulate chromatin structure. In normal tissues, expression of HDAC11 is limited to kidney, heart, brain, skeletal muscle and testis, suggesting that its function might be tissue-specific. In mammals, HDAC11 proteins are known to be involved in progression of various tumors. HDAC11 plays an essential role in regulating OX40 ligand (OX40L) expression in Hodgkin lymphoma (HL); selective inhibition of HDAC11 expression significantly up-regulates OX40L and induces apoptosis in HL cell lines. Thus, inhibition of HDAC11 could be a therapeutic drug option for antitumor immune response in HL patients.','cd09993','HDAC_classIV'),(1001934,'gnl|CDD|223201',_binary 'Deacetylases, including yeast histone deacetylase and acetoin utilization protein [Chromatin structure and dynamics / Secondary metabolites biosynthesis, transport, and catabolism].','COG0123','AcuC'),(1001935,'gnl|CDD|307135',_binary 'Histone deacetylase domain.  Histones can be reversibly acetylated on several lysine residues. Regulation of transcription is caused in part by this mechanism. Histone deacetylases catalyze the removal of the acetyl group. Histone deacetylases are related to other proteins.','pfam00850','Hist_deacetyl'),(1001936,'gnl|CDD|212518',_binary 'Histone deacetylases and histone-like deacetylases, classII.  Class II histone deacetylases are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. This group includes animal HDAC4,5,6,7,8,9,10, fungal HOS3 and HDA1, plant HDA5 and HDA15 as well as other eukaryotes, archaeal and bacterial histone-like deacetylases. Eukaryotic deacetylases mostly use histones (H2, H3, H4) as substrates for deacetylation; however, non-histone substrates are known (for example, tubulin). Substrates for prokaryotic histone-like deacetylases are not known. Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Interaction partners of class II deacetylases include 14-3-3 proteins, MEF2 family of transcriptional factors, CtBP, calmodulin (CaM), SMRT, N-CoR, BCL6, HP1alpha and SUMO. Histone deacetylases play a role in the regulation of cell cycle, cell differentiation and survival. Class II mammalian HDACs are differentially inhibited by structurally diverse compounds with known antitumor activities, thus presenting them as potential drug targets for human diseases resulting from aberrant acetylation.','cd09992','HDAC_classII'),(1001937,'gnl|CDD|212541',_binary 'Histone deacetylases and histone-like deacetylases, classII.  This subfamily includes eukaryotic as well as bacterial Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. In D. discoideum, where four homologs (HdaA, HdaB, HdaC, HdaD) have been identified, HDAC activity is important for regulating the timing of gene expression during development. Also, inhibition of HDAC activity by trichostatin A is shown to cause hyperacetylation of the histone and a delay in cell aggregation and differentiation.','cd11599','HDAC_classII_2'),(1001938,'gnl|CDD|212512',_binary 'Histone deacetylase (HDAC) classes I, II, IV and related proteins.  The HDAC/HDAC-like family includes Zn-dependent histone deacetylase classes I, II and IV (class III HDACs, also called sirtuins, are NAD-dependent and structurally unrelated, and therefore not part of this family). Histone deacetylases catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98), as opposed to the acetylation reaction by some histone acetyltransferases (EC 2.3.1.48). Deacetylases of this family are involved in signal transduction through histone and other protein modification, and can repress/activate transcription of a number of different genes. They usually act via the formation of large multiprotein complexes. They are involved in various cellular processes, including cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and post-translational control of the acetyl coenzyme A synthetase. In mammals, they are known to be involved in progression of different tumors. Specific inhibitors of mammalian histone deacetylases are an emerging class of promising novel anticancer drugs.','cd09301','HDAC'),(1001939,'gnl|CDD|212520',_binary 'Class I histone deacetylase AcuC (Acetoin utilization protein)-like enzymes.  AcuC (Acetoin utilization protein) is a class I deacetylase found only in bacteria and is involved in post-translational control of the acetyl-coenzyme A synthetase (AcsA). Deacetylase AcuC works in coordination with deacetylase SrtN (class III), possibly to maintain AcsA in active (deacetylated) form and let the cell grow under low concentration of acetate. B. subtilis AcuC is a member of operon acuABC; this operon is repressed by the presence of glucose and does not show induction by acetoin; acetoin is a bacterial fermentation product that can be converted to acetate via the butanediol cycle in absence of other carbon sources. Inactivation of AcuC leads to slower growth and lower cell yield under low-acetate conditions in Bacillus subtilis. In general, Class I histone deacetylases (HDACs) are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase.','cd09994','HDAC_AcuC_like'),(1001940,'gnl|CDD|212521',_binary 'Histone deacetylases and histone-like deacetylases, classII.  This subfamily includes bacterial as well as eukaryotic Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. Included in this family is a bacterial HDAC-like amidohydrolase (Bordetella/Alcaligenes species FB18817, denoted as FB188 HDAH) shown to be most similar in sequence and function to class II HDAC6 domain 3 or b (HDAC6b). FB188 HDAH is able to remove the acetyl moiety from acetylated histones, and can be inhibited by common HDAC inhibitors such as SAHA (suberoylanilide hydroxamic acid) as well as class II-specific but not class I specific inhibitors.','cd09996','HDAC_classII_1'),(1001941,'gnl|CDD|212527',_binary 'Histone deacetylase 6, domain 2.  Histone deacetylase 6 is a class IIb Zn-dependent enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. Known interaction partners of HDAC6 are alpha tubulin and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD).','cd10003','HDAC6-dom2'),(1001942,'gnl|CDD|212526',_binary 'Histone deacetylase 6, domain 1 and histone deacetylase 10.  Histone deacetylases 6 and 10 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. HDAC10 has an N-terminal deacetylase domain and a C-terminal pseudo-repeat that shares significant similarity with its catalytic domain. It is located in the nucleus and cytoplasm, and is involved in regulation of melanogenesis. It transcriptionally down-regulates thioredoxin-interacting protein (TXNIP), leading to altered reactive oxygen species (ROS) signaling in human gastric cancer cells. Known interaction partners of HDAC6 are alpha tubulin (substrate) and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD) while interaction partners of HDAC10 are Pax3, KAP1, hsc70 and HDAC3 proteins.','cd10002','HDAC10_HDAC6-dom1'),(1001943,'gnl|CDD|212525',_binary 'Histone deacetylase class IIa.  This subfamily includes bacterial acetylpolyamine amidohydrolase (APAH) as well as other Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. Mycoplana ramosa APAH exhibits broad substrate specificity and catalyzes the deacetylation of polyamines such as putrescine, spermidine, and spermine by cleavage of a non-peptide amide bond.','cd10001','HDAC_classII_APAH'),(1001944,'gnl|CDD|212545',_binary 'Histone deacetylase 6, domain 1.  Histone deacetylases 6 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. Known interaction partners of HDAC6 are alpha tubulin (substrate) and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD).','cd11682','HDAC6-dom1'),(1001945,'gnl|CDD|212546',_binary 'Histone deacetylase 10.  Histone deacetylases 10 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC10 has an N-terminal deacetylase domain and a C-terminal pseudo-repeat that shares significant similarity with its catalytic domain. It is located in the nucleus and cytoplasm, and is involved in regulation of melanogenesis. It transcriptionally down-regulates thioredoxin-interacting protein (TXNIP), leading to altered reactive oxygen species (ROS) signaling in human gastric cancer cells. Known interaction partners of HDAC10 are Pax3, KAP1, hsc70 and HDAC3 proteins.','cd11683','HDAC10'),(1001946,'gnl|CDD|212544',_binary 'Histone deacetylases, class IIa.  Class IIa histone deacetylases are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) to yield deacetylated histones. This subclass includes animal HDAC4, HDAC5, HDAC7, and HDCA9. Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, they have N-terminal regulatory domain with two or three conserved serine residues, phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC9 is involved in regulation of gene expression and dendritic growth in developing cortical neurons. It also plays a role in hematopoiesis. HDAC7 is involved in regulation of myocyte migration and differentiation. HDAC5 is involved in integration of chronic drug (cocaine) addiction and depression with changes in chromatin structure and gene expression. HDAC4 participates in regulation of chondrocyte hypertrophy and skeletogenesis.','cd11681','HDAC_classIIa'),(1001947,'gnl|CDD|212517',_binary 'Class I histone deacetylases.  Class I histone deacetylases (HDACs) are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. This group includes animal HDAC1, HDAC2, HDAC3, HDAC8, fungal RPD3, HOS1 and HOS2, plant HDA9, protist, archaeal and bacterial (AcuC) deacetylases. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase. In mammals, they are known to be involved in progression of various tumors. Specific inhibitors of mammalian histone deacetylases are an emerging class of promising novel anticancer drugs.','cd09991','HDAC_classI'),(1001948,'gnl|CDD|212534',_binary 'Histone deacetylase 1 (HDAC1).  Histone deacetylase 1 (HDAC1) is a Zn-dependent class I enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDAC1 is involved in regulation through association with DNA binding proteins to target specific chromatin regions. In particular, HDAC1 appears to play a major role in pre-implantation embryogenesis in establishing a repressive chromatin state. Its interaction with retinoblastoma tumor-suppressor protein is essential in the control of cell proliferation and differentiation. Together with metastasis-associated protein-2 (MTA2), it deacetylates p53, thereby modulating its effect on cell growth and apoptosis. It participates in DNA-damage response, along with HDAC2; together, they promote DNA non-homologous end-joining. HDAC1 is also involved in tumorogenesis; its overexpression modulates cancer progression. Specific inhibitors of HDAC1 are currently used in cancer therapy.','cd10010','HDAC1'),(1001949,'gnl|CDD|212524',_binary 'Histone deacetylase 8 (HDAC8).  HDAC8 is a Zn-dependent class I histone deacetylase that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. HDAC8 is found in human cytoskeleton-bound protein fraction and insoluble cell pellets. It plays a crucial role in intramembraneous bone formation; germline deletion of HDAC8 is detrimental to skull bone formation. HDAC8 is possibly associated with the smooth muscle actin cytockeleton and may regulate the contractive capacity of smooth muscle cells. HDAC8 is also involved in the metabolic control of the estrogen receptor related receptor (ERR)-alpha/peroxisome proliferator activated receptor (PPAR) gamma coactivator 1 alpha (PGC1-alpha) transcriptional complex as well as in the development of neuroblastoma and T-cell lymphoma. HDAC8-selective small-molecule inhibitors could be a therapeutic drug option for these diseases.','cd10000','HDAC8'),(1001950,'gnl|CDD|212535',_binary 'Histone deacetylase 2 (HDAC2).  Histone deacetylase 2 (HDAC2) is a Zn-dependent class I enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDAC2 is involved in regulation through association with DNA binding proteins to target specific chromatin regions. It forms transcriptional repressor complexes by associating with several proteins, including the mammalian zinc-finger transcription factor YY1, thus playing an important role in transcriptional regulation, cell cycle progression and developmental events. Additionally, a few non-histone HDAC2 substrates have been found. HDAC2 plays a role in embryonic development and cytokine signaling important for immune response, and is over-expressed in several solid tumors including oral, prostate, ovarian, endometrial and gastric cancer. It participates in DNA-damage response, along with HDAC1; together, they can promote DNA non-homologous end-joining. HDAC2 is considered an important cancer prognostic marker. Inhibitors specifically targeting HDAC2 could be a therapeutic drug option.','cd10011','HDAC2'),(1001951,'gnl|CDD|212531',_binary 'Histone deacetylase 5.  Histone deacetylase 5 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC5 is involved in integration of chronic drug (cocaine) addiction and depression with changes in chromatin structure and gene expression; cocaine regulates HDAC5 function to antagonize the rewarding impact of cocaine, possibly by blocking drug-stimulated gene expression that supports drug-induced behavioral change. It is also involved in regulation of angiogenesis and cell cycle as well as immune system development. HDAC5 and HDAC9 have been found to be significantly up-regulated in high-risk medulloblastoma compared with low-risk and may potentially be novel drug targets.','cd10007','HDAC5'),(1001952,'gnl|CDD|212530',_binary 'Histone deacetylase 4.  Histone deacetylase 4 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC4 participates in regulation of chondrocyte hypertrophy and skeletogenesis. However, biological substrates for HDAC4 have not been identified; only low lysine deacetylation activity has been demonstrated and active site mutant has enhanced activity toward acetylated lysines. HDAC4 does not bind DNA directly, but through transcription factors MEF2C (myocyte enhancer factor-2C) and MEF2D. Other known interaction partners of the protein are 14-3-3 proteins, SMRT and N-CoR co-repressors, BCL6, HP1, SUMO-1 ubiquitin-like protein, and ANKRA2. It appears to interact in a multiprotein complex with RbAp48 and HDAC3. Furthermore, HDAC4 is required for TGFbeta1-induced myofibroblastic differentiation.','cd10006','HDAC4'),(1001953,'gnl|CDD|212542',_binary 'Class II Histone deacetylase  Clr3 and similar proteins.  Clr3 is a class II Histone deacetylase Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Clr3 is the homolog of the class-II HDAC HdaI in S. cerevisiae, and is essential for silencing in heterochromatin regions, such as centromeric regions, ribosomal DNA, the mating-type region and telomeric loci. Clr3 has also been implicated in the regulation of stress-related genes; the histone acetyltransferase, Gcn5, in S. cerevisiae, preferentially acetylates global histone H3K14 while Clr3 preferentially deacetylates H3K14ac, and therefore, interplay between Gcn5 and Clr3 is crucial for the regulation of many stress-response genes.','cd11600','HDAC_Clr3'),(1001954,'gnl|CDD|212533',_binary 'Histone deacetylase 9.  Histone deacetylase 9 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, they have N-terminal regulatory domain with two or three conserved serine residues, phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC9 is involved in regulation of gene expression and dendritic growth in developing cortical neurons. It also plays a role in hematopoiesis. Its deregulated expression may be associated with some human cancers. HDAC5 and HDAC9 have been found to be significantly up-regulated in high-risk medulloblastoma compared with low-risk and may potentially be novel drug targets.','cd10009','HDAC9'),(1001955,'gnl|CDD|212528',_binary 'reduced potassium dependency-3 (RPD3)-like.  Proteins of the Rpd3-like family are class I Zn-dependent Histone deacetylases that catalyze hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). RPD3 is the yeast homolog of class I HDACs. The main function of RPD3-like group members is regulation of a number of different processes through protein (mostly different histones) modification (deacetylation). This group includes fungal RPD3 and acts via the formation of large multiprotein complexes. Members of this group are involved in cell cycle regulation, DNA damage response, embryonic development and cytokine signaling important for immune response. Histone deacetylation by yeast RPD3 represses genes regulated by the Ash1 and Ume6 DNA-binding proteins. In mammals, they are known to be involved in progression of various tumors. Specific inhibitors of mammalian histone deacetylases could be a therapeutic drug option.','cd10004','RPD3-like'),(1001956,'gnl|CDD|212532',_binary 'Histone deacetylase 7.  Histone deacetylase 7 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC7 is involved in regulation of myocyte migration and differentiation. Known interaction partners of class IIa HDAC7 are myocyte enhancer factors - MEF2A, -2C, and -2D, 14-3-3 proteins, SMRT and N-CoR co-repressors, HDAC3, ETA (endothelin receptor). This enzyme is also involved in the development of the immune system as well as brain and heart development. Multiple alternatively spliced transcript variants encoding several isoforms have been found for this gene.','cd10008','HDAC7'),(1001957,'gnl|CDD|212540',_binary 'Class I histone deacetylases including ScHos2 and SpPhd1.  This subfamily includes Class I histone deacetylase (HDAC) Hos2 from Saccharomyces cerevisiae as well as a histone deacetylase Phd1 from Schizosaccharomyces pombe. Hos2 binds to the coding regions of genes during gene activation, specifically it deacetylates the lysines in H3 and H4 histone tails. It is preferentially associated with genes of high activity genome-wide and is shown to be necessary for efficient transcription. Thus, Hos2 is directly required for gene activation in contrast to other class I histone deacetylases. Protein encoded by phd1 is inhibited by trichostatin A (TSA), a specific inhibitor of histone deacetylase, and is involved in the meiotic cell cycle in S. pombe. Class 1 HDACs are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98).','cd11598','HDAC_Hos2'),(1001958,'gnl|CDD|212543',_binary 'Class I histone deacetylases Hos1 and related proteins.  Saccharomyces cerevisiae Hos1 is responsible for Smc3 deacetylation. Smc3 is an important player during the establishment of sister chromatid cohesion. Hos1 belongs to the class I histone deacetylases (HDACs). HDACs are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. Other class I HDACs are animal HDAC1, HDAC2, HDAC3, HDAC8, fungal RPD3 and HOS2, plant HDA9, protist, archaeal and bacterial (AcuC) deacetylases. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase.','cd11680','HDAC_Hos1'),(1001959,'gnl|CDD|212522',_binary 'Class II histone deacetylases Hos3 and related proteins.  Fungal histone deacetylase Hos3 from Saccharomyces cerevisiae is a Zn-dependent enzyme belonging to HDAC class II. It catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Hos3 deacetylase is homodimer, in vitro it shows specificity to H4, H3 and H2A.','cd09998','HDAC_Hos3'),(1001960,'gnl|CDD|240251',_binary 'histone deacetylase; Provisional.','PTZ00063','PTZ00063'),(1001961,'gnl|CDD|212529',_binary 'Histone deacetylase 3 (HDAC3).  HDAC3 is a Zn-dependent class I histone deacetylase that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. In order to target specific chromatin regions, HDAC3 can interact with DNA-binding proteins (transcriptional factors) either directly or after forming complexes with a number of other proteins, as observed for the SMPT/N-CoR complex which recruits human HDAC3 to specific chromatin loci and activates deacetylation. Human HDAC3 is also involved in deacetylation of non-histone substrates such as RelA, SPY and p53 factors. This protein can also down-regulate p53 function and subsequently modulate cell growth and apoptosis. This gene is therefore regarded as a potential tumor suppressor gene. HDAC3 plays a role in various physiological processes, including subcellular protein localization, cell cycle progression, cell differentiation, apoptosis and survival. HDAC3 has been found to be overexpressed in some tumors including leukemia, lung carcinoma, colon cancer and maxillary carcinoma. Thus, inhibitors precisely targeting HDAC3 (in some cases together with retinoic acid or hyperthermia) could be a therapeutic drug option.','cd10005','HDAC3'),(1001987,'gnl|CDD|238879',_binary 'NnaC (CMP-NeuNAc synthetase) _like subfamily of SGNH_hydrolases, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles two of the three components of typical Ser-His-Asp(Glu) triad from other serine hydrolases. E. coli NnaC appears to be involved in polysaccharide synthesis.','cd01841','NnaC_like'),(1002036,'gnl|CDD|214906',_binary 'The HIRAN protein (HIP116, Rad5p N-terminal) is found in the N-terminal regions of the SWI2/SNF2 proteins typified by HIP116 and Rad5p.  HIRAN is found as a standalone protein in several bacteria and prophages, or fused to other catalytic domains, such as a nuclease of the restriction endonuclease fold and TDP1-like DNA phosphoesterases, in the eukaryotes. It has been predicted that this protein functions as a DNA-binding domain that probably recognises features associated with damaged DNA or stalled replication forks.','smart00910','HIRAN'),(1002037,'gnl|CDD|312360',_binary 'HIRAN domain.  The HIRAN domain (HIP116, Rad5p N-terminal) is found in the N-terminal regions of the SWI2/SNF2 proteins typified by HIP116 and Rad5p. The HIRAN domain is found as a standalone protein in several bacteria and prophages, or fused to other catalytic domains, such as a nuclease of the restriction endonuclease fold and TDP1-like DNA phosphoesterases, in the eukaryotes. It has been predicted that this domain functions as a DNA-binding domain that probably recognizes features associated with damaged DNA or stalled replication forks.','pfam08797','HIRAN'),(1002793,'gnl|CDD|239384',_binary 'This protein family\'s function is unkown. It contains nucleotide binding site. It uses NTP as energy source to transfer electron or ion.','cd03110','Fer4_NifH_child'),(1002932,'gnl|CDD|185079',_binary '2\'-5\' RNA ligase; Provisional.','PRK15124','PRK15124'),(1002933,'gnl|CDD|260008',_binary 'non-LTR RNase HI domain of reverse transcriptases.  Ribonuclease H (RNase H) is classified into two families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). Ribonuclease HI (RNase HI) is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes. RNase HI has also been observed as an adjunct domain to the reverse transcriptase gene in retroviruses, long-term repeat (LTR)-bearing retrotransposons and non-LTR retrotransposons. RNase HI in LTR retrotransposons perform degradation of the original RNA template, generation of a polypurine tract (the primer for plus-strand DNA synthesis), and final removal of RNA primers from newly synthesized minus and plus strands. The catalytic residues for RNase H enzymatic activity, three aspartatic acids and one glutamic acid residue (DEDD), are unvaried across all RNase H domains. The position of the RNase domain of non-LTR and LTR transposons is at the carboxyl terminal of the reverse transcriptase (RT) domain and their RNase domains group together, indicating a common evolutionary origin. Many non-LTR transposons have lost the RNase domain because their activity is at the nucleus and cellular RNase may suffice; however LTR retrotransposons always encode their own RNase domain because it requires RNase activity in RNA-protein particles in the cytoplasm. RNase H inhibitors have been explored as an anti-HIV drug target because RNase H inactivation inhibits reverse transcription.','cd09276','Rnase_HI_RT_non_LTR'),(1002954,'gnl|CDD|129526',_binary 'phosophoadenylyl-sulfate reductase (thioredoxin).  This enzyme, involved in the assimilation of inorganic sulfate, is designated cysH in Bacteria and MET16 in Saccharomyces cerevisiae. Synonyms include phosphoadenosine phosphosulfate reductase, PAPS reductase, and PAPS reductase, thioredoxin-dependent. In a reaction requiring reduced thioredoxin and NADPH, it converts 3(prime)-phosphoadenylylsulfate (PAPS) to sulfite and adenosine 3(prime),5(prime) diphosphate (PAP). A related family of plant enzymes, scoring below the trusted cutoff, differs in having a thioredoxin-like C-terminal domain, not requiring thioredoxin, and in having a preference for 5(prime)-adenylylsulfate (APS) over PAPS. [Central intermediary metabolism, Sulfur metabolism].','TIGR00434','Phosphoadenosine_phosphos'),(1002955,'gnl|CDD|223115',_binary 'tRNA(Ile)-lysidine synthase MesJ [Cell cycle control, cell division, chromosome partitioning].','COG0037','MesJ'),(1002956,'gnl|CDD|234997',_binary 'phosphoadenosine phosphosulfate reductase; Provisional.','PRK02090','PRK02090'),(1002961,'gnl|CDD|310665',_binary 'Phage tail assembly chaperone.','pfam06222','Phage_TAC_1'),(1002966,'gnl|CDD|311353',_binary 'Bacteriophage Mu Gam like protein.  This family consists of bacterial and phage Gam proteins. The gam gene of bacteriophage Mu encodes a protein which protects linear double stranded DNA from exonuclease degradation in vitro and in vivo.','pfam07352','Phage_Mu_Gam'),(1002970,'gnl|CDD|224427',_binary 'Predicted transcriptional regulators [Transcription].','COG1510','COG1510'),(1002972,'gnl|CDD|225016',_binary 'Uncharacterized conserved protein [Function unknown].','COG2105','COG2105'),(1002973,'gnl|CDD|310586',_binary 'Gamma-glutamyl cyclotransferase, AIG2-like.  GGACT, gamma-glutamylamine cyclotransferase, is a ubiquitous enzyme found in bacteria, plants, and metazoans from Dictyostelium through to humans. It converts gamma-glutamylamines to free amines and 5-oxoproline.','pfam06094','GGACT'),(1002974,'gnl|CDD|119400',_binary 'GGCT-like domains, also called AIG2-like family. Gamma-glutamyl cyclotransferase (GGCT) catalyzes the formation of pyroglutamic acid (5-oxoproline) from dipeptides containing gamma-glutamyl, and is a dimeric protein. In Homo sapiens, the protein is encoded by the gene C7orf24, and the enzyme participates in the gamma-glutamyl cycle. Hereditary defects in the gamma-glutamyl cycle have been described for some of the genes involved, but not for C7orf24. The synthesis and metabolism of glutathione (L-gamma-glutamyl-L-cysteinylglycine) ties the gamma-glutamyl cycle to numerous cellular processes; glutathione acts as a ubiquitous reducing agent in reductive mechanisms involved in protein and DNA synthesis, transport processes, enzyme activity, and metabolism. AIG2 (avrRpt2-induced gene) is an Arabidopsis protein that exhibits RPS2- and avrRpt2-dependent induction early after infection with Pseudomonas syringae pv maculicola strain ES4326 carrying avrRpt2. avrRpt2 is an avirulence gene that can convert virulent strains of P. syringae to avirulence on Arabidopsis thaliana, soybean, and bean. The family also includes bacterial tellurite-resistance proteins (trgB); tellurium (Te) compounds are used in industrial processes and had been used as antimicrobial agents in the past. Some members have been described proteins involved in cation transport (chaC).','cd06661','GGCT_like'),(1002980,'gnl|CDD|314998',_binary 'Putative amidoligase enzyme.  This family of proteins are likely to act as amidoligase enzymes Protein in this family are found in conserved gene neighborhoods encoding a glutamine amidotransferase-like thiol peptidase (in proteobacteria) or an Aig2 family cyclotransferase protein (in firmicutes).','pfam12224','Amidoligase_2'),(1002984,'gnl|CDD|316080',_binary 'Glutamine amidotransferase domain.  This domain is a class-II glutamine amidotransferase domain found in a variety of enzymes, such as asparagine synthetase and glutamine--fructose-6-phosphate transaminase.','pfam13522','GATase_6'),(1002992,'gnl|CDD|240349',_binary 'glucosamine-fructose-6-phosphate aminotransferase; Provisional.','PTZ00295','PTZ00295'),(1002994,'gnl|CDD|316095',_binary 'Glutamine amidotransferase domain.  This domain is a class-II glutamine amidotransferase domain found in a variety of enzymes such as asparagine synthetase and glutamine-fructose-6-phosphate transaminase.','pfam13537','GATase_7'),(1002995,'gnl|CDD|223112',_binary 'Glutamine phosphoribosylpyrophosphate amidotransferase [Nucleotide transport and metabolism].','COG0034','PurF'),(1003147,'gnl|CDD|128941',_binary 'Animal peptidoglycan recognition proteins homologous to Bacteriophage T3 lysozyme.  The bacteriophage molecule, but not its moth homologue, has been shown to have N-acetylmuramoyl-L-alanine amidase activity. One member of this family, Tag7, is a cytokine.','smart00701','PGRP'),(1003150,'gnl|CDD|177282',_binary 'lysozyme.','PHA00447','PHA00447'),(1003233,'gnl|CDD|316217',_binary 'Acetyltransferase (GNAT) domain.  This family contains proteins with N-acetyltransferase functions such as Elp3-related proteins.','pfam13673','Acetyltransf_10'),(1003235,'gnl|CDD|273701',_binary 'ribosomal-protein-alanine acetyltransferase.  Members of this model belong to the GCN5-related N-acetyltransferase (GNAT) superfamily. This model covers prokarotes and the archaea. The seed contains a characterized accession for Gram negative E. coli. An untraceable characterized accession (PIR|S66013) for Gram positive B. subtilis scores well (205.0) in the full alignment. Characterized members are lacking in the archaea. Noise cutoff (72.4) was set to exclude M. loti paralog of rimI. Trusted cutoff (80.0) was set at next highest scoring member in the mini-database. [Protein synthesis, Ribosomal proteins: synthesis and modification].','TIGR01575','rimI'),(1003238,'gnl|CDD|224584',_binary 'Acetyltransferases, including N-acetylases of ribosomal proteins [Translation, ribosomal structure and biogenesis].','COG1670','RimL'),(1003239,'gnl|CDD|237916',_binary 'spermidine N1-acetyltransferase; Provisional.','PRK15130','PRK15130'),(1003240,'gnl|CDD|225261',_binary 'Predicted acetyltransferase [General function prediction only].','COG2388','COG2388'),(1003241,'gnl|CDD|235142',_binary 'putative acetyltransferase; Provisional.','PRK03624','PRK03624'),(1003269,'gnl|CDD|237171',_binary 'transcription termination factor Rho; Provisional.','PRK12678','PRK12678'),(1003272,'gnl|CDD|309952',_binary 'Protein of unknown function (DUF669).  Members of this family are found in various phage proteins.','pfam05037','DUF669'),(1003831,'gnl|CDD|222857',_binary 'portal vertex protein; Provisional.','PHA02531','20'),(1003928,'gnl|CDD|224122',_binary 'Lhr-like helicases [General function prediction only].','COG1201','Lhr'),(1004312,'gnl|CDD|307032',_binary 'haloacid dehalogenase-like hydrolase.  This family is structurally different from the alpha/beta hydrolase family (pfam00561). This family includes L-2-haloacid dehalogenase, epoxide hydrolases and phosphatases. The structure of the family consists of two domains. One is an inserted four helix bundle, which is the least well conserved region of the alignment, between residues 16 and 96 of Pseudomonas sp. (S)-2-haloacid dehalogenase 1. The rest of the fold is composed of the core alpha/beta domain. Those members with the characteristic DxD triad at the N-terminus are probably phosphatidylglycerolphosphate (PGP) phosphatases involved in cardiolipin biosynthesis in the mitochondria.','pfam00702','Hydrolase'),(1004725,'gnl|CDD|226041',_binary 'Cephalosporin hydroxylase [Defense mechanisms].','COG3510','CmcI'),(1004746,'gnl|CDD|240399',_binary 'protein tyrosine phosphatase; Provisional.','PTZ00393','PTZ00393'),(1004754,'gnl|CDD|307645',_binary 'Transposase DDE domain.  Transposase proteins are necessary for efficient DNA transposition. This domain is a member of the DDE superfamily, which contain three carboxylate residues that are believed to be responsible for coordinating metal ions needed for catalysis. The catalytic activity of this enzyme involves DNA cleavage at a specific site followed by a strand transfer reaction. This family contains transposases for IS4, IS421, IS5377, IS427, IS402, IS1355, IS5, which was original isolated in bacteriophage lambda.','pfam01609','DDE_Tnp_1'),(1004755,'gnl|CDD|315909',_binary 'Putative transposase of IS4/5 family (DUF4096).','pfam13340','DUF4096'),(1004756,'gnl|CDD|290319',_binary 'Transposase DDE domain.  Transposase proteins are necessary for efficient DNA transposition. This domain is a member of the DDE superfamily, which contain three carboxylate residues that are believed to be responsible for coordinating metal ions needed for catalysis.','pfam13586','DDE_Tnp_1_2'),(1004757,'gnl|CDD|225830',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG3293','COG3293'),(1004758,'gnl|CDD|184957',_binary 'methyl viologen resistance protein SmvA; Provisional.','PRK14995','PRK14995'),(1004759,'gnl|CDD|129794',_binary 'drug resistance transporter, EmrB/QacA subfamily.  This subfamily of drug efflux proteins, a part of the major faciliator family, is predicted to have 14 potential membrane-spanning regions. Members with known activities include EmrB (multiple drug resistance efflux pump) in E. coli, FarB (antibacterial fatty acid resistance) in Neisseria gonorrhoeae, TcmA (tetracenomycin C resistance) in Streptomyces glaucescens, etc. In most cases, the efflux pump is described as having a second component encoded in the same operon, such as EmrA of E. coli. [Cellular processes, Toxin production and resistance, Transport and binding proteins, Other].','TIGR00711','Uncharacterized_MFS-type_'),(1004760,'gnl|CDD|311564',_binary 'Major Facilitator Superfamily.','pfam07690','MFS_1'),(1004761,'gnl|CDD|273318',_binary 'Multidrug resistance protein.','TIGR00880','Chromaffin_granule_amine_'),(1004762,'gnl|CDD|119392',_binary 'The Major Facilitator Superfamily (MFS) is a large and diverse group of secondary transporters that includes uniporters, symporters, and antiporters. MFS proteins facilitate the transport across cytoplasmic or internal membranes of a variety of substrates including ions, sugar phosphates, drugs, neurotransmitters, nucleosides, amino acids, and peptides. They do so using the electrochemical potential of the transported substrates. Uniporters transport a single substrate, while symporters and antiporters transport two substrates in the same or in opposite directions, respectively, across membranes. MFS proteins are typically 400 to 600 amino acids in length, and the majority contain 12 transmembrane alpha helices (TMs) connected by hydrophilic loops. The N- and C-terminal halves of these proteins display weak similarity and may be the result of a gene duplication/fusion event. Based on kinetic studies and the structures of a few bacterial superfamily members, GlpT (glycerol-3-phosphate transporter), LacY (lactose permease), and EmrD (multidrug transporter), MFS proteins are thought to function through a single substrate binding site, alternating-access mechanism involving a rocker-switch type of movement. Bacterial members function primarily for nutrient uptake, and as drug-efflux pumps to confer antibiotic resistance. Some MFS proteins have medical significance in humans such as the glucose transporter Glut4, which is impaired in type II diabetes, and glucose-6-phosphate transporter (G6PT), which causes glycogen storage disease when mutated.','cd06174','MFS'),(1004763,'gnl|CDD|182502',_binary 'putative transporter; Provisional.','PRK10504','PRK10504'),(1004764,'gnl|CDD|273326',_binary 'D-galactonate transporter.  [Transport and binding proteins, Carbohydrates, organic alcohols, and acids].','TIGR00893','Probable_glucarate_transp'),(1005525,'gnl|CDD|187639',_binary 'clavulanic acid dehydrogenase (CAD), classical (c) SDR.  CAD catalyzes the NADP-dependent reduction of clavulanate-9-aldehyde to clavulanic acid, a beta-lactamase inhibitor. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08934','CAD_SDR_c'),(1005555,'gnl|CDD|211705',_binary '3-hydroxybutyrate dehydrogenase.  This model represents a subfamily of the short chain dehydrogenases. Characterized members so far as 3-hydroxybutyrate dehydrogenases and are found in species that accumulate ester polmers called polyhydroxyalkanoic acids (PHAs) under certain conditions. Several members of the family are from species not known to accumulate PHAs, including Oceanobacillus iheyensis and Bacillus subtilis. However, polymer formation is not required for there be a role for 3-hydroxybutyrate dehydrogenase; it may be members of this family have the same function in those species.','TIGR01963','D-beta-hydroxybutyrate_de'),(1005558,'gnl|CDD|235693',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK06077','fabG'),(1006193,'gnl|CDD|225420',_binary 'Predicted transcriptional regulator containing an HTH domain and an uncharacterized domain shared with the mammalian protein Schlafen [Transcription].','COG2865','COG2865'),(1006197,'gnl|CDD|316285',_binary 'Putative ATP-dependent DNA helicase recG C-terminal.  This domain may well interact selectively and non-covalently with ATP, adenosine 5\'-triphosphate, a universally important coenzyme and enzyme regulator.','pfam13749','HATPase_c_4'),(1006200,'gnl|CDD|309457',_binary 'Putative DNA-binding domain.  This family belongs to the AlbA clan of DNA-binding domains.','pfam04326','AlbA_2'),(1006211,'gnl|CDD|309469',_binary 'Protein of unknown function, DUF488.  This family includes several proteins of uncharacterized function.','pfam04343','DUF488'),(1006212,'gnl|CDD|227770',_binary 'Uncharacterized conserved protein [Function unknown].','COG5483','COG5483'),(1006609,'gnl|CDD|238858',_binary 'PAF_acetylhydrolase (PAF-AH)_like subfamily of SGNH-hydrolases. Platelet-activating factor (PAF) and PAF-AH are key players in inflammation and in atherosclerosis. PAF-AH is a calcium independent phospholipase A2 which exhibits strong substrate specificity towards PAF, hydrolyzing an acetyl ester at the sn-2 position. PAF-AH also degrades a family of oxidized PAF-like phospholipids with short sn-2 residues.  In addition,  PAF and PAF-AH are associated with neural migration and mammalian reproduction.','cd01820','PAF_acetylesterase_like'),(1006621,'gnl|CDD|177330',_binary 'helix-turn-helix protein.','PHA01976','PHA01976'),(1007288,'gnl|CDD|226809',_binary 'Uncharacterized protein conserved in bacteria with the myosin-like domain [Function unknown].','COG4372','COG4372'),(1007579,'gnl|CDD|225263',_binary 'Transcriptional regulator, contains sigma factor-related N-terminal domain [Transcription].','COG2390','DeoR'),(1007581,'gnl|CDD|166660',_binary 'carbonic anhydrase.','PLN03019','PLN03019'),(1007589,'gnl|CDD|238264',_binary '6-pyruvoyl tetrahydropterin synthase (PTPS). Folate derivatives are essential cofactors in the biosynthesis of purines, pyrimidines, and amino acids, as well as formyl-tRNA. Mammalian cells are able to utilize pre-formed folates after uptake by a carrier-mediated active transport system. Most microbes and plants lack this system and must synthesize folates de novo from guanosine triphosphate. One enzyme from this pathway is PTPS which catalyzes the conversion of dihydroneopterin triphosphate to 6-pyruvoyl tetrahydropterin. The functional enzyme is a hexamer of identical subunits.','cd00470','PTPS');
INSERT INTO `domain` VALUES (1007595,'gnl|CDD|179167',_binary 'GMP synthase subunit B; Validated.','PRK00919','PRK00919'),(1007610,'gnl|CDD|274251',_binary 'probable molybdenum cofactor biosynthesis protein A, archaeal.  This model describes an archaeal family related, and predicted to be functionally equivalent, to molybdenum cofactor biosynthesis protein A (MoaA) of bacteria (see TIGR02666). [Biosynthesis of cofactors, prosthetic groups, and carriers, Molybdopterin].','TIGR02668','moaA_archaeal'),(1007614,'gnl|CDD|225449',_binary 'Molybdenum cofactor biosynthesis enzyme [Coenzyme metabolism].','COG2896','MoaA'),(1007616,'gnl|CDD|211974',_binary 'SynChlorMet cassette radical SAM/SPASM protein ScmF.  A biosynthesis cassette found in Syntrophobacter fumaroxidans MPOB, Chlorobium limicola DSM 245, Methanocella paludicola SANAE, and delta proteobacterium NaphS2 contains two PqqE-like radical SAM/SPASM domain proteins, a PqqD homolog, and a conserved hypothetical protein. These components suggest modification of a ribosomally produced peptide precursor, but the precursor has not been identified. Of the two PqqE homologs of the cassette, this family is the more distant in sequence.','TIGR04251','SCM_rSAM_ScmF'),(1007619,'gnl|CDD|100105',_binary 'Radical SAM superfamily. Enzymes of this family generate radicals by combining a 4Fe-4S cluster and S-adenosylmethionine (SAM) in close proximity. They are characterized by a conserved CxxxCxxC motif, which coordinates the conserved iron-sulfur cluster. Mechanistically, they share the transfer of a single electron from the iron-sulfur cluster to SAM, which leads to its reductive cleavage to methionine and a 5\'-deoxyadenosyl radical, which, in turn, abstracts a hydrogen from the appropriately positioned carbon atom. Depending on the enzyme, SAM is consumed during this process or it is restored and reused. Radical SAM enzymes catalyze steps in metabolism, DNA repair, the biosynthesis of vitamins and coenzymes, and the biosynthesis of many antibiotics. Examples are biotin synthase (BioB), lipoyl synthase (LipA), pyruvate formate-lyase (PFL), coproporphyrinogen oxidase (HemN), lysine 2,3-aminomutase (LAM), anaerobic ribonucleotide reductase (ARR), and  MoaA, an enzyme of the biosynthesis of molybdopterin.','cd01335','Radical_SAM'),(1007620,'gnl|CDD|275259',_binary 'cytosylglucuronate decarboxylase.  BlsE, part of the blasticidin S biosynthetic pathway, is a radical SAM enzyme that performs a decarboxylation at C5 of the glucoside residue. MilG in mildiomycin biosynthesis is equivalent. This enzyme follows CGA synthase and makes the pyranoside core moiety of a class of peptidyl nucleoside antibiotics. [Cellular processes, Toxin production and resistance].','TIGR04466','hypothetical_protein_Radi'),(1007622,'gnl|CDD|188566',_binary 'heme d1 biosynthesis radical SAM protein NirJ.  Heme d1 occurs in the cytochrome cd1 subunit of nitrite reductase in species such as Pseudomonas stutzeri. NirJ is a radical SAM protein involved in its bioynthesis. In a number of species, distinct genes NirJ1 and NirJ2 are found in similar genomic regions; this model describe authentic NirJ from genomes with NirJ only. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR04051','rSAM_NirJ'),(1007624,'gnl|CDD|275338',_binary 'heme b synthase.  Members of this family are AhbD (alternative heme biosynthetic protein D), a radical SAM enzyme in sulfate-reducing bacteria and methanogens that performs the last decarboxylations to synthesize heme b from Fe-coproporphyrin III. Members include DVU_0855, previously included in error in TIGR04055, the NirJ2 family thought to be involved in heme d1 biosynthesis. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR04545','rSAM_ahbD_hemeb'),(1007627,'gnl|CDD|275143',_binary 'pseudo-rSAM protein/SPASM domain protein.  Members of this family all have a C-terminal SPASM domain (see model TIGR04085), a region usually found as a C-terminal second 4Fe-4S domain of radical SAM domain (see pfam04055) proteins. A majority of rSAM/SPASM proteins modify ribosomally produced peptides. In a few members of this family, the key Cys residues of the radical SAM domain have been lost, making this a pseudo-rSAM family. Members of this family are restricted so far to Haloarchaea, always occur next a member of family TIGR04031, and are often accompanied by another rSAM/SPASM domain protein. The function of this two or three gene cassette is unknown.','TIGR04347','pseudo_SAM_Halo'),(1007628,'gnl|CDD|234672',_binary 'molybdenum cofactor biosynthesis protein A; Reviewed.','PRK00164','moaA'),(1007639,'gnl|CDD|313984',_binary 'Protein of unknown function (DUF2634).  Some members in this family of proteins are annotated as phage related, xkdS however currently there is no known function.','pfam10934','DUF2634'),(1007724,'gnl|CDD|180803',_binary 'extracytoplasmic-function sigma-70 factor; Validated.','PRK07037','PRK07037'),(1007760,'gnl|CDD|318852',_binary 'Domain of unknown function (DUF5067).','pfam16729','DUF5067'),(1007774,'gnl|CDD|314348',_binary 'Stealth protein CR2, conserved region 2.  Stealth_CR2 is the second of several highly conserved regions on stealth proteins in metazoa and bacteria. There are up to four CR regions on all member proteins. CR2 carries a well-conserved NDD sequence-motif. The domain is found in tandem with CR1, CR3 and CR4 on both potential metazoan hosts and pathogenic eubacterial species that are capsular polysaccharide phosphotransferases. The CR domains appear on eukaryotic proteins such as GNPTAB, N-acetylglucosamine-1-phosphotransferase subunits alpha/beta. Horizontal gene-transfer seems to have occurred between host and bacteria of these sequence-regions in order for the bacteria to evade detection by the host innate immune system.','pfam11380','Stealth_CR2'),(1007812,'gnl|CDD|107267',_binary 'Ligand-binding domain of DNA transcription repressor for the hexuronate utilization operon from Bacillus species and its close homologs from other bacteria, all of which are a member of the LacI-GalR family of bacterial transcription regulators.  Ligand-binding domain of DNA transcription repressor for the hexuronate utilization operon from Bacillus species and its close homologs from other bacteria, all of which are a member of the LacI-GalR family of bacterial transcription regulators. The LacI-GalR family repressors are composed of two functional domains: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal ligand-binding domain, which is homologous to the sugar-binding domain of ABC-type transport systems that contain the type I periplasmic binding protein-like fold. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcription repressor undergoes a conformational change upon ligand binding which in turn changes the DNA binding affinity of the repressor.','cd06272','PBP1_hexuronate_repressor'),(1009681,'gnl|CDD|239068',_binary 'Peptidase C1A subfamily (MEROPS database nomenclature); composed of cysteine peptidases (CPs) similar to papain, including the mammalian CPs (cathepsins B, C, F, H, L, K, O, S, V, X and W). Papain is an endopeptidase with specific substrate preferences, primarily for bulky hydrophobic or aromatic residues at the S2 subsite, a hydrophobic pocket in papain that accommodates the P2 sidechain of the substrate (the second residue away from the scissile bond). Most members of the papain subfamily are endopeptidases. Some exceptions to this rule can be explained by specific details of the catalytic domains like the occluding loop in cathepsin B which confers an additional carboxydipeptidyl activity and the mini-chain of cathepsin H resulting in an N-terminal exopeptidase activity. Papain-like CPs have different functions in various organisms. Plant CPs are used to mobilize storage proteins in seeds. Parasitic CPs act extracellularly to help invade tissues and cells, to hatch or to evade the host immune system. Mammalian CPs are primarily lysosomal enzymes with the exception of cathepsin W, which is retained in the endoplasmic reticulum. They are responsible for protein degradation in the lysosome. Papain-like CPs are synthesized as inactive proenzymes with N-terminal propeptide regions, which are removed upon activation. In addition to its inhibitory role, the propeptide is required for proper folding of the newly synthesized enzyme and its stabilization in denaturing pH conditions. Residues within the propeptide region also play a role in the transport of the proenzyme to lysosomes or acidified vesicles. Also included in this subfamily are proteins classified as non-peptidase homologs, which lack peptidase activity or have missing active site residues.','cd02248','Peptidase_C1A'),(1009682,'gnl|CDD|239111',_binary 'Cathepsin B group; composed of cathepsin B and similar proteins, including tubulointerstitial nephritis antigen (TIN-Ag). Cathepsin B is a lysosomal papain-like cysteine peptidase which is expressed in all tissues and functions primarily as an exopeptidase through its carboxydipeptidyl activity. Together with other cathepsins, it is involved in the degradation of proteins, proenzyme activation, Ag processing, metabolism and apoptosis. Cathepsin B has been implicated in a number of human diseases such as cancer, rheumatoid arthritis, osteoporosis and Alzheimer\'s disease. The unique carboxydipeptidyl activity of cathepsin B is attributed to the presence of an occluding loop in its active site which favors the binding of the C-termini of substrate proteins. Some members of this group do not possess the occluding loop. TIN-Ag is an extracellular matrix basement protein which was originally identified as a target Ag involved in anti-tubular basement membrane antibody-mediated interstitial nephritis. It plays a role in renal tubulogenesis and is defective in hereditary tubulointerstitial disorders. TIN-Ag is exclusively expressed in kidney tissues. .','cd02620','Peptidase_C1A_CathepsinB'),(1009683,'gnl|CDD|306594',_binary 'Papain family cysteine protease.','pfam00112','Peptidase_C1'),(1009690,'gnl|CDD|237495',_binary 'ATP-dependent protease Lon; Provisional.','PRK13765','PRK13765'),(1009691,'gnl|CDD|273259',_binary 'lon-related putative ATP-dependent protease.  This model represents a set of proteins with extensive C-terminal homology to the ATP-dependent protease La, product of the lon gene of E. coli. The model is based on a seed alignment containing only archaeal members, but several bacterial proteins match the model well. Because several species, including Thermotoga maritima and Treponema pallidum, contain both a close homolog of the lon protease and nearly full-length homolog of the members of this family, we suggest there may also be a functional division between the two families. Members of this family from Pyrococcus horikoshii and Pyrococcus abyssi each contain a predicted intein. [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00764','Archaeal_Lon_protease'),(1009702,'gnl|CDD|273883',_binary 'HflC protein.  HflK and HflC are paralogs encoded by tandem genes in Proteobacteria, spirochetes, and some other bacterial lineages. The HflKC complex is anchored in the membrane and exposed to the periplasm. The complex is not active as a protease, but rather binds to and appears to modulate the ATP-dependent protease FtsH. The overall function of HflKC is not fully described. [Protein fate, Degradation of proteins, peptides, and glycopeptides, Regulatory functions, Protein interactions].','TIGR01932','Protein_HflC'),(1009713,'gnl|CDD|274682',_binary 'probable DNA repair protein.  Members of this protein family are bacterial proteins of about 900 amino acids in length. Members show extended homology to proteins in TIGR02786, the AddB protein of double-strand break repair via homologous recombination. Members of this family, therefore, may be DNA repair proteins.','TIGR03623','TIGR03623'),(1009748,'gnl|CDD|184430',_binary 'thymidylate kinase; Provisional.','PRK13975','PRK13975'),(1009789,'gnl|CDD|213028',_binary 'Putative catalytic NodB homology domain of uncharacterized bacterial polysaccharide deacetylases which consist of a 5-stranded beta/alpha barrel.  This family contains many uncharacterized bacterial polysaccharide deacetylases. Although their biological functions remain unknown, all members of the family are predicted to contain a conserved domain with a 5-stranded beta/alpha barrel, which is similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, belonging to the larger carbohydrate esterase 4 (CE4) superfamily.','cd10973','CE4_DAC_u4_5s'),(1009790,'gnl|CDD|307596',_binary 'Polysaccharide deacetylase.  This domain is found in polysaccharide deacetylase. This family of polysaccharide deacetylases includes NodB (nodulation protein B from Rhizobium) which is a chitooligosaccharide deacetylase. It also includes chitin deacetylase from yeast, and endoxylanases which hydrolyses glucosidic bonds in xylan.','pfam01522','Polysacc_deac_1'),(1009791,'gnl|CDD|223798',_binary 'Predicted xylanase/chitin deacetylase [Carbohydrate transport and metabolism].','COG0726','CDA1'),(1009831,'gnl|CDD|227069',_binary 'Transcriptional activator, adenine-specific DNA methyltransferase [Signal transduction mechanisms / Transcription].','COG4725','IME4'),(1009832,'gnl|CDD|309971',_binary 'MT-A70.  MT-A70 is the S-adenosylmethionine-binding subunit of human mRNA:m6A methyl-transferase (MTase), an enzyme that sequence-specifically methylates adenines in pre-mRNAs.','pfam05063','MT-A70'),(1010428,'gnl|CDD|224275',_binary 'Transcriptional regulator [DNA replication, recombination and repair].','COG1356','tfx'),(1010468,'gnl|CDD|306641',_binary 'Aldehyde dehydrogenase family.  This family of dehydrogenases act on aldehyde substrates. Members use NADP as a cofactor. The family includes the following members: The prototypical members are the aldehyde dehydrogenases EC:1.2.1.3. Succinate-semialdehyde dehydrogenase EC:1.2.1.16. Lactaldehyde dehydrogenase EC:1.2.1.22. Benzaldehyde dehydrogenase EC:1.2.1.28. Methylmalonate-semialdehyde dehydrogenase EC:1.2.1.27. Glyceraldehyde-3-phosphate dehydrogenase EC:1.2.1.9. Delta-1-pyrroline-5-carboxylate dehydrogenase EC: 1.5.1.12. Acetaldehyde dehydrogenase EC:1.2.1.10. Glutamate-5-semialdehyde dehydrogenase EC:1.2.1.41. This family also includes omega crystallin, an eye lens protein from squid and octopus that has little aldehyde dehydrogenase activity.','pfam00171','Aldedh'),(1010737,'gnl|CDD|235514',_binary 'primosome assembly protein PriA; Validated.','PRK05580','PRK05580'),(1010972,'gnl|CDD|318812',_binary 'DNA-packaging protein gp3.  DNA-packaging protein gp3 (terminase small subunit) is involved in DNA packing in bacteriophage. it contains a channel where DNA is bound and passed to DNA-packaging protein gp2 (terminase large subunit).','pfam16677','GP3_package'),(1010982,'gnl|CDD|99982',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. WabH in Klebsiella pneumoniae has been shown to transfer a GlcNAc residue from UDP-GlcNAc onto the acceptor GalUA residue in the cellular outer core.','cd03811','GT1_WabH_like'),(1010988,'gnl|CDD|99968',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. wbuB in E. coli is involved in the biosynthesis of the O26 O-antigen.  It has been proposed to function as an N-acetyl-L-fucosamine (L-FucNAc) transferase.','cd03794','GT1_wbuB_like'),(1010990,'gnl|CDD|223528',_binary 'Nucleoside-diphosphate-sugar epimerases [Cell envelope biogenesis, outer membrane / Carbohydrate transport and metabolism].','COG0451','WcaG'),(1010992,'gnl|CDD|212494',_binary 'extended (e) SDRs.  Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd08946','SDR_e'),(1010994,'gnl|CDD|187566',_binary 'UDP-N-acetylglucosamine 4-epimerase, extended (e) SDRs.  This subgroup contains UDP-N-acetylglucosamine 4-epimerase of Pseudomonas aeruginosa, WbpP,  an extended SDR, that catalyzes the NAD+ dependent conversion of UDP-GlcNAc and UDPGalNA to UDP-Glc and UDP-Gal.  This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05256','UDP_AE_SDR_e'),(1010996,'gnl|CDD|307501',_binary 'NAD dependent epimerase/dehydratase family.  This family of proteins utilize NAD as a cofactor. The proteins in this family use nucleotide-sugar substrates for a variety of chemical reactions.','pfam01370','Epimerase'),(1010997,'gnl|CDD|99994',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. wcfI in Bacteroides fragilis has been shown to be involved in the capsular polysaccharide biosynthesis.','cd03825','GT1_wcfI_like'),(1010998,'gnl|CDD|223645',_binary 'Uridine kinase [Nucleotide transport and metabolism].','COG0572','Udk'),(1010999,'gnl|CDD|273487',_binary 'UDP-glucose-4-epimerase GalE.  Alternate name: UDPgalactose 4-epimerase This enzyme interconverts UDP-glucose and UDP-galactose. A set of related proteins, some of which are tentatively identified as UDP-glucose-4-epimerase in Thermotoga maritima, Bacillus halodurans, and several archaea, but deeply branched from this set and lacking experimental evidence, are excluded from this model and described by a separate model. [Energy metabolism, Sugars].','TIGR01179','UDP-glucose_4-epimerase'),(1011000,'gnl|CDD|187559',_binary 'ADP-L-glycero-D-mannoheptose 6-epimerase (GME), extended (e) SDRs.  This subgroup contains ADP-L-glycero-D-mannoheptose 6-epimerase, an extended SDR, which catalyzes the NAD-dependent interconversion of ADP-D-glycero-D-mannoheptose and ADP-L-glycero-D-mannoheptose.  This subgroup has the canonical active site tetrad and NAD(P)-binding motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05248','ADP_GME_SDR_e'),(1011001,'gnl|CDD|187557',_binary 'dTDP-D-glucose 4,6-dehydratase, extended (e) SDRs.  This subgroup contains dTDP-D-glucose 4,6-dehydratase and related proteins, members of the extended-SDR family, with the characteristic Rossmann fold core region, active site tetrad and NAD(P)-binding motif. dTDP-D-glucose 4,6-dehydratase is closely related to other sugar epimerases of the SDR family. dTDP-D-dlucose 4,6,-dehydratase catalyzes the second of four steps in the dTDP-L-rhamnose pathway (the dehydration of dTDP-D-glucose to dTDP-4-keto-6-deoxy-D-glucose) in the synthesis of L-rhamnose, a cell wall component of some pathogenic bacteria. In many gram negative bacteria, L-rhamnose is an important constituent of lipopoylsaccharide O-antigen. The larger N-terminal portion of dTDP-D-Glucose 4,6-dehydratase forms a Rossmann fold NAD-binding domain, while the C-terminus binds the sugar substrate. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05246','dTDP_GD_SDR_e'),(1011002,'gnl|CDD|224012',_binary 'UDP-glucose 4-epimerase [Cell envelope biogenesis, outer membrane].','COG1087','GalE'),(1011003,'gnl|CDD|187558',_binary 'UDP-glucose 4 epimerase, subgroup 1, extended (e) SDRs.  UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05247','UDP_G4E_1_SDR_e'),(1011004,'gnl|CDD|133024',_binary 'NTP_transferases catalyze the transfer of nucleotides onto phosphosugars.  Nucleotidyltransferases transfer nucleotides onto phosphosugars.  The enzyme family includes Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase. The products are activated sugars that are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides.','cd04181','NTP_transferase'),(1011005,'gnl|CDD|273489',_binary 'dTDP-glucose 4,6-dehydratase.  This protein is related to UDP-glucose 4-epimerase (GalE) and likewise has an NAD cofactor. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01181','dTDP-glucose_46-dehydrata'),(1011006,'gnl|CDD|224129',_binary 'Nucleoside-diphosphate-sugar pyrophosphorylase involved in lipopolysaccharide biosynthesis/translation initiation factor 2B, gamma/epsilon subunits (eIF-2Bgamma/eIF-2Bepsilon) [Cell envelope biogenesis, outer membrane / Translation, ribosomal structure and biogenesis].','COG1208','GCD1'),(1011007,'gnl|CDD|274028',_binary 'ADP-L-glycero-D-manno-heptose-6-epimerase.  This family consists of examples of ADP-L-glycero-D-mannoheptose-6-epimerase, an enzyme involved in biosynthesis of the inner core of lipopolysaccharide (LPS) for Gram-negative bacteria. This enzyme is homologous to UDP-glucose 4-epimerase (TIGR01179) and belongs to the NAD dependent epimerase/dehydratase family (pfam01370). [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR02197','heptose_epim'),(1011008,'gnl|CDD|133044',_binary 'NTP_transferase_like_1 is a member of the nucleotidyl transferase family.  This is a subfamily of nucleotidyl transferases. Nucleotidyl transferases transfer nucleotides onto phosphosugars. The activated sugars are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides. Other subfamilies of nucleotidyl transferases include Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase.','cd06422','NTP_transferase_like_1'),(1011009,'gnl|CDD|224013',_binary 'dTDP-D-glucose 4,6-dehydratase [Cell envelope biogenesis, outer membrane].','COG1088','RfbB'),(1011010,'gnl|CDD|224134',_binary 'Predicted sugar nucleotidyltransferases [Cell envelope biogenesis, outer membrane].','COG1213','COG1213'),(1011011,'gnl|CDD|187550',_binary 'GDP-fucose synthetase, extended (e) SDRs.  GDP-fucose synthetase (aka 3, 5-epimerase-4-reductase) acts in the NADP-dependent synthesis of GDP-fucose from GDP-mannose. Two activities have been proposed for the same active site: epimerization and reduction. Proteins in this subgroup are extended SDRs, which have a characteristic active site tetrad and an NADP-binding motif, [AT]GXXGXXG, that is a close match to the archetypical form. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05239','GDP_FS_SDR_e'),(1011012,'gnl|CDD|133001',_binary 'The N-terminal of eIF-2B_gamma_like is predicted to have glycosyltransferase activity.  N-terminal domain of eEIF-2B epsilon and gamma, subunits of eukaryotic translation initiators, is a subfamily of glycosyltranferase 2 and is predicted to have glycosyltranferase activity. eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit epsilon shares sequence similarity with gamma subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.','cd02507','eIF-2B_gamma_N_like'),(1011013,'gnl|CDD|318561',_binary 'GDP-mannose 4,6 dehydratase.','pfam16363','GDP_Man_Dehyd'),(1011014,'gnl|CDD|133041',_binary 'The N-terminal domain of gamma subunit of the eIF-2B is a subfamily of glycosyltransferase 2.  N-terminal domain of gamma subunit of the eukaryotic translation initiation factor 2B (eIF-2B): eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit gamma shares sequence similarity with epsilon subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.','cd04198','eIF-2B_gamma_N'),(1011015,'gnl|CDD|187574',_binary 'UDP-glucose 4-epimerase (G4E), subgroup 5, extended (e) SDRs.  This subgroup partially conserves the characteristic active site tetrad and NAD-binding motif of the extended SDRs, and has been identified as possible UDP-glucose 4-epimerase (aka UDP-galactose 4-epimerase), a homodimeric member of the extended SDR family. UDP-glucose 4-epimerase catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05264','UDP_G4E_5_SDR_e'),(1011016,'gnl|CDD|133065',_binary 'WcbM_like is a subfamily of nucleotidyl transferases.  WcbM protein of Burkholderia mallei is involved in the biosynthesis, export or translocation of capsule. It is a subfamily of nucleotidyl transferases that transfer nucleotides onto phosphosugars.','cd06915','NTP_transferase_WcbM_like'),(1011017,'gnl|CDD|215146',_binary 'probable rhamnose biosynthetic enzyme.','PLN02260','PLN02260'),(1011018,'gnl|CDD|187570',_binary 'GDP-mannose 4,6 dehydratase, extended (e) SDRs.  GDP-mannose 4,6 dehydratase, a homodimeric SDR, catalyzes the NADP(H)-dependent conversion of GDP-(D)-mannose to GDP-4-keto, 6-deoxy-(D)-mannose in the fucose biosynthesis pathway. These proteins have the canonical active site triad and NAD-binding pattern, however the active site Asn is often missing and may be substituted with Asp. A Glu residue has been identified as an important active site base. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05260','GDP_MD_SDR_e'),(1011019,'gnl|CDD|187541',_binary 'UDP-glucuronate decarboxylase (UGD) and related proteins, extended (e) SDRs.  UGD catalyzes the formation of UDP-xylose from UDP-glucuronate; it is an extended-SDR, and has the characteristic glycine-rich NAD-binding pattern, TGXXGXXG, and active site tetrad.  Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05230','UGD_SDR_e'),(1011020,'gnl|CDD|187564',_binary 'dTDP-6-deoxy-L-lyxo-4-hexulose reductase and related proteins, extended (e) SDRs.  dTDP-6-deoxy-L-lyxo-4-hexulose reductase, an extended SDR, synthesizes dTDP-L-rhamnose from alpha-D-glucose-1-phosphate,  providing the precursor of L-rhamnose, an essential cell wall component of many pathogenic bacteria. This subgroup has the characteristic active site tetrad and NADP-binding motif. This subgroup also contains human MAT2B, the regulatory subunit of methionine adenosyltransferase (MAT); MAT catalyzes S-adenosylmethionine synthesis. The human gene encoding MAT2B encodes two major splicing variants which are induced in human cell liver cancer and regulate HuR, an mRNA-binding protein which stabilizes the mRNA of several cyclins, to affect cell proliferation. Both MAT2B variants include this extended SDR domain. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05254','dTDP_HR_like_SDR_e'),(1011021,'gnl|CDD|187567',_binary 'Arna decarboxylase_like, extended (e) SDRs.  Decarboxylase domain of ArnA. ArnA, is an enzyme involved in the modification of outer membrane protein lipid A of gram-negative bacteria. It is a bifunctional enzyme that catalyzes the NAD-dependent decarboxylation of UDP-glucuronic acid and N-10-formyltetrahydrofolate-dependent formylation of UDP-4-amino-4-deoxy-l-arabinose; its NAD-dependent decaboxylating activity is in the C-terminal 360 residues. This subgroup belongs to the extended SDR family, however the NAD binding motif is not a perfect match and the upstream Asn of the canonical active site tetrad is not conserved. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05257','Arna_like_SDR_e'),(1011022,'gnl|CDD|187563',_binary 'UDP glucuronic acid epimerase, extended (e) SDRs.  This subgroup contains UDP-D-glucuronic acid 4-epimerase, an extended SDR, which catalyzes the conversion of UDP-alpha-D-glucuronic acid to UDP-alpha-D-galacturonic acid. This group has the SDR\'s canonical catalytic tetrad and the TGxxGxxG NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05253','UDP_GE_SDE_e'),(1011023,'gnl|CDD|187543',_binary 'UDP-glucose 4 epimerase, subgroup 4, extended (e) SDRs.  UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup is comprised of bacterial proteins, and includes the Staphylococcus aureus capsular polysaccharide Cap5N, which may have a role in the synthesis of UDP-N-acetyl-d-fucosamine. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05232','UDP_G4E_4_SDR_e'),(1011024,'gnl|CDD|177883',_binary 'UDP-glucose 4-epimerase.','PLN02240','PLN02240'),(1011025,'gnl|CDD|187548',_binary 'UDP-Glcnac (UDP-linked N-acetylglucosamine) inverting 4,6-dehydratase, extended (e) SDRs.  UDP-Glcnac inverting 4,6-dehydratase was identified in Helicobacter pylori as the hexameric flaA1 gene product (FlaA1). FlaA1 is hexameric, possesses UDP-GlcNAc-inverting 4,6-dehydratase activity,  and catalyzes the first step in the creation of a pseudaminic acid derivative in protein glycosylation. Although this subgroup has the NADP-binding motif characteristic of extended SDRs, its members tend to have a Met substituted for the active site Tyr found in most SDR families. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05237','UDP_invert_4-6DH_SDR_e'),(1011026,'gnl|CDD|187581',_binary 'Arabidopsis thaliana GDP-mannose-3\',5\'-epimerase (GME)-like, extended (e) SDRs.  This subgroup of NDP-sugar epimerase/dehydratases are extended SDRs; they have the characteristic active site tetrad, and an NAD-binding motif: TGXXGXX[AG], which is a close match to the canonical NAD-binding motif. Members include Arabidopsis thaliana GDP-mannose-3\',5\'-epimerase (GME) which catalyzes the epimerization of two positions of GDP-alpha-D-mannose to form GDP-beta-L-galactose. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05273','GME-like_SDR_e'),(1011027,'gnl|CDD|187575',_binary 'atypical (a) SDRs, subgroup 1.  Atypical SDRs in this subgroup are poorly defined and have been identified putatively as isoflavones reductase, sugar dehydratase, mRNA binding protein etc. Atypical SDRs are distinct from classical SDRs. Members of this subgroup retain the canonical active site triad (though not the upstream Asn found in most SDRs) but have an unusual putative glycine-rich NAD(P)-binding motif, GGXXXXG, in the usual location. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05265','SDR_a1'),(1011028,'gnl|CDD|224016',_binary 'dTDP-4-dehydrorhamnose reductase [Cell envelope biogenesis, outer membrane].','COG1091','RfbD'),(1011029,'gnl|CDD|187673',_binary 'human NSDHL (NAD(P)H steroid dehydrogenase-like protein)-like, extended (e) SDRs.  This subgroup includes human NSDHL and related proteins. These proteins have the characteristic active site tetrad of extended SDRs, and also have a close match to their  NAD(P)-binding motif.  Human NSDHL is a 3beta-hydroxysteroid dehydrogenase (3 beta-HSD) which functions in the cholesterol biosynthetic pathway.  3 beta-HSD catalyzes the oxidative conversion of delta 5-3 beta-hydroxysteroids to the delta 4-3-keto configuration; this activity is essential for the biosynthesis of all classes of hormonal steroids. Mutations in the gene encoding NSDHL cause CHILD syndrome (congenital hemidysplasia with ichthyosiform nevus and limb defects), an X-linked dominant, male-lethal trait.  This subgroup also includes an unusual bifunctional [3beta-hydroxysteroid dehydrogenase (3b-HSD)/C-4 decarboxylase from Arabidopsis thaliana, and Saccharomyces cerevisiae ERG26, a 3b-HSD/C-4 decarboxylase, involved in the synthesis of ergosterol, the major sterol of yeast.  Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid sythase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd09813','3b-HSD-NSDHL-like_SDR_e'),(1011030,'gnl|CDD|187660',_binary 'Bordetella bronchiseptica enzymes WbmH and WbmG-like, extended (e) SDRs.  Bordetella bronchiseptica enzymes WbmH and WbmG, and related proteins. This subgroup exhibits the active site tetrad and NAD-binding motif of the extended SDR family. It has been proposed that the active site in Bordetella WbmG and WbmH cannot function as an epimerase, and that it plays a role in O-antigen synthesis pathway from UDP-2,3-diacetamido-2,3-dideoxy-l-galacturonic acid. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd08957','WbmH_like_SDR_e'),(1011031,'gnl|CDD|187549',_binary 'Escherichia coli Gne (a nucleoside-diphosphate-sugar 4-epimerase)-like, extended (e) SDRs.  Nucleoside-diphosphate-sugar 4-epimerase has the characteristic active site tetrad and NAD-binding motif of the extended SDR, and is related to more specifically defined epimerases such as UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), which catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup includes Escherichia coli 055:H7 Gne, a UDP-GlcNAc 4-epimerase, essential for O55 antigen synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05238','Gne_like_SDR_e'),(1011032,'gnl|CDD|182639',_binary 'UDP-galactose-4-epimerase; Provisional.','PRK10675','PRK10675'),(1011033,'gnl|CDD|182313',_binary 'dTDP-glucose 4,6-dehydratase; Provisional.','PRK10217','PRK10217'),(1011034,'gnl|CDD|187545',_binary 'UDP-glucose 4 epimerase, subgroup 2, extended (e) SDRs.  UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup is comprised of archaeal and bacterial proteins, and has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05234','UDP_G4E_2_SDR_e'),(1011035,'gnl|CDD|309453',_binary 'RmlD substrate binding domain.  L-rhamnose is a saccharide required for the virulence of some bacteria. Its precursor, dTDP-L-rhamnose, is synthesized by four different enzymes the final one of which is RmlD. The RmlD substrate binding domain is responsible for binding a sugar nucleotide.','pfam04321','RmlD_sub_bind'),(1011036,'gnl|CDD|187539',_binary 'uncharacterized subgroup of aldehyde reductase and flavonoid reductase related proteins, extended (e) SDRs.  This subgroup contains proteins of unknown function related to aldehyde reductase and flavonoid reductase of the extended SDR-type. Aldehyde reductase I (aka carbonyl reductase) is an NADP-binding SDR; it has an NADP-binding motif consensus that is slightly different from the canonical SDR form and lacks the Asn of the extended SDR active site tetrad. Aldehyde reductase I catalyzes the NADP-dependent  reduction of ethyl 4-chloro-3-oxobutanoate to ethyl (R)-4-chloro-3-hydroxybutanoate. The related flavonoid reductases act in the NADP-dependent reduction of  flavonoids, ketone-containing plant secondary metabolites. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05228','AR_FR_like_1_SDR_e'),(1011037,'gnl|CDD|187568',_binary 'CDP-tyvelose 2-epimerase, extended (e) SDRs.  CDP-tyvelose 2-epimerase is a tetrameric SDR that catalyzes the conversion of CDP-D-paratose to CDP-D-tyvelose, the last step in tyvelose biosynthesis. This subgroup is a member of the extended SDR subfamily, with a characteristic active site tetrad and NAD-binding motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05258','CDP_TE_SDR_e'),(1011038,'gnl|CDD|187552',_binary '3beta-hydroxysteroid dehydrogenases (3b-HSD)-like, extended (e) SDRs.  Extended SDR family domains belonging to this subgroup have the characteristic active site tetrad and a fairly well-conserved NAD(P)-binding motif. 3b-HSD catalyzes the NAD-dependent conversion of various steroids, such as pregnenolone to progesterone, or androstenediol to testosterone. This subgroup includes an unusual bifunctional 3b-HSD/C-4 decarboxylase from Arabidopsis thaliana, and Saccharomyces cerevisiae ERG26, a 3b-HSD/C-4 decarboxylase, involved in the synthesis of ergosterol, the major sterol of yeast. It also includes human 3 beta-HSD/HSD3B1 and C(27) 3beta-HSD/ [3beta-hydroxy-delta(5)-C(27)-steroid oxidoreductase; HSD3B7].  C(27) 3beta-HSD/HSD3B7 is a membrane-bound enzyme of the endoplasmic reticulum, that catalyzes the isomerization and oxidation of 7alpha-hydroxylated sterol intermediates, an early step in bile acid biosynthesis. Mutations in the human NSDHL (NAD(P)H steroid dehydrogenase-like protein) cause CHILD syndrome (congenital hemidysplasia with ichthyosiform nevus and limb defects), an X-linked dominant, male-lethal trait. Mutations in the human gene encoding C(27) 3beta-HSD underlie a rare autosomal recessive form of neonatal cholestasis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid sythase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05241','3b-HSD-like_SDR_e'),(1011039,'gnl|CDD|273505',_binary 'dTDP-4-dehydrorhamnose reductase.  This enzyme catalyzes the last of 4 steps in making dTDP-rhamnose, a precursor of LPS core antigen, O-antigen, etc. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01214','dTDP-4-keto-L-rhamnose_re'),(1011040,'gnl|CDD|185103',_binary 'Vi polysaccharide biosynthesis protein TviC; Provisional.','PRK15181','PRK15181'),(1011041,'gnl|CDD|187537',_binary 'Extended (e) and atypical (a) SDRs.  Extended or atypical short-chain dehydrogenases/reductases (SDRs, aka tyrosine-dependent oxidoreductases) are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05226','SDR_e_a'),(1011042,'gnl|CDD|165812',_binary 'dTDP-glucose 4,6-dehydratase.','PLN02166','PLN02166'),(1011043,'gnl|CDD|182998',_binary 'ADP-L-glycero-D-mannoheptose-6-epimerase; Provisional.','PRK11150','rfaD'),(1011052,'gnl|CDD|309551',_binary 'Protein of unknown function (DUF551).  This family represents the carboxy terminus of a protein of unknown function, found in dsDNA viruses with no RNA stage, including bacteriophages lambda and P22, and also in some Escherichia coli prophages.','pfam04448','DUF551'),(1011062,'gnl|CDD|310757',_binary 'Protein of unknown function (DUF1073).  This family consists of several hypothetical bacterial proteins. The function of this family is unknown.','pfam06381','DUF1073'),(1011068,'gnl|CDD|313242',_binary 'Uncharacterized protein conserved in bacteria (DUF2213).  Members of this family of bacterial proteins comprise various hypothetical and phage-related proteins. The exact function of these proteins has not, as yet, been determined.','pfam09979','DUF2213'),(1011069,'gnl|CDD|226096',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3566','COG3566'),(1011072,'gnl|CDD|198384',_binary 'N-terminal catalytic domain of GIY-YIG intron endonuclease I-TevI, I-BmoI, I-BanI, I-BthII and similar proteins.  I-TevI is a site-specific GIY-YIG homing endonuclease encoded within the group I intron of the thymidylate synthase gene (td) from Escherichia coli phage T4. It functions as an endonuclease that catalyzes the first step in intron homing by generating a double-strand break in the intronless td allele within a sequence designated the homing site. I-TevI recognizes its extensive 37 base pair DNA target in a site-specific, but sequence-tolerant manner. The cleavage site is located at 23 (upper strand) and 25 (lower strand) nucleotides upstream of the intron insertion site. A divalent cation, such as Mg2+, is required for the catalysis. I-TevI also acts as a repressor of its own transcription. It binds an operator that is located upstream of the I-TevI coding sequence and overlaps the T4 late promoter, which drives I-TevI expression from within the td intron. I-TevI binds the homing sites and the operator with the same affinity, but cleaves the homing site more efficiently than the operator. I-TevI consists of an N-terminal catalytic domain, containing the GIY-YIG motif, and a C-terminal DNA-binding domain that binds DNA as a monomer, joined by a flexible linker. The C-terminal domain includes three subdomains: a zinc finger, a minor-groove binding alpha-helix (NUMOD3, nuclease-associated modular domain 3), and a helix-turn-helix domain (HTH). The last two are responsible for DNA-binding. The zinc finger is part of the linker and not required for DNA-binding. It is implicated as a distance sensor to constrain the catalytic domain to cleave the homing site at a fixed position. None of other GIY-YIG endonucleases have been found to have the zinc finger motif. This family also includes a reduced activity isoschizomer of I-TevI, I-BmoI, which is encoded within the group I intron of the thymidylate synthase (TS) gene (thyA) from Bacillus mojavensis. I-BmoI catalyzes the first step in intron homing by generating a double-strand break in the intronless td allele within a sequence designated the homing site in the presence of a divalent cation cofactor, such as Mg2+. In the absence of Mg2+, I-Bmol only nicks one of the strands. Both I-BmoI and I-TevI bind a homologous stretch of TS-encoding DNA as monomers, but use different strategies to distinguish intronless from intron-containing substrates. I-TevI recognizes substrates at the level of DNA-binding. However, I-BmoI binds both intron-containing and intronless TS-encoding substrates, but efficiently cleaves only intronless substrate. Afterwards they cleave their respective intronless substrates in the same positions, and both require a critical G-C base pair adjacent to the top strand site for efficient cleavage. The C-terminal domain of I-BmoI has nuclease-associated modular DNA-binding domains (NUMODs), but lacks the zinc finger, which is different from that of I-TevI. Although the zinc finger implicated as a distance determination in I-TevI is absent, I-BmoI still possesses some cleavage distance discrimination. Besides I-TevI and I-BmoI, this family contains a putative GIY-YIG homing endonuclease, I-BanI, encoded within the self-splicing group I intron of nrdE gene from Bacillus anthracis. It contains two major domains, the N-terminal GIY-YIG domain and the C-terminal DNA-binding domain that consists of a minor-groove DNA binding alpha-helix motif and a helix-turn-helix (HTH) motif. I-BanI generates a double-strand break (DSB) in the intronless nrdE gene. The cleavage site is located at 5 and 7 nucleotides upstream of the intron insertion site, with 2-nucleotide 3\' extensions. The recognition site is 35 to 40 base pairs and covers the cleavage site with a bias toward the downstream region including the (intervening sequence) IVS insertion site. Moreover, this family contains another putative GIY-YIG homing endonuclease, I-BthII, encoded within the self-splicing group I intron of nrdF gene from Bacillus thuringiensis ssp. pakistani. It contains a GIY-YIG motif that generates a double-strand break (DSB) in the intronless nrdF gene. The cleavage site is located at 7 and 9 nucleotides upstream of the intron insertion site, leaving 2-nucleotide 3\' extensions. The recognition site is 27 to 29 base pairs with the DSB cleavage site at the 5\'-end of the top strand, and with the intervening sequence (IVS) insertion site approximately in the middle of the recognition site.','cd10437','GIY-YIG_HE_I-TevI_like'),(1011074,'gnl|CDD|198392',_binary 'Catalytic GIY-YIG domain of putative intron-encoded endonuclease bI1 and similar proteins.  The prototype of this family is a putative intron-encoded mitochondrial DNA endonuclease bI1 found in mitochondrion Ustilago maydis. This protein may arise from proteolytic cleavage of an in-frame translation of COB exon 1 plus intron 1, containing the bI1 open reading frame. It contains an N-terminal truncated non-functional cytochrome b region and a C-terminal intron-encoded endonuclease bI1 region. The bI1 region shows high sequence similarity to endonucleases of group I introns of fungi and phage and might be involved in intron homing. Many uncharacterized bI1 homologs existing in fungi and chlorophyta in this family do not contain the cytochrome b region, but have a standalone bI1-like region, which contains a GIY-YIG domain and a minor-groove binding alpha-helix nuclease-associated modular domain (NUMOD). This family also includes a Yarrowia lipolytica mobile group-II intron COX1-i1, also called intron alpha, encoding protein with reverse transcriptase activity. The group-II intron COX1-i1 may be involv ed both in the generation of the circular multimeric DNA molecules (senDNA alpha) which amplify during the senescence syndrome and in the generation of the site-specific deletion which accumulates in the premature-death syndrome.','cd10445','GIY-YIG_bI1_like'),(1011076,'gnl|CDD|198398',_binary 'GIY-YIG domain of LuxR and ArsR family transcriptional regulators, and uncharacterized hypothetical proteins found in bacteria.  The family includes some bacterial LuxR and ArsR family transcriptional regulators.  The a C-terminal conserved domain shows sequence similarity to the N-terminal catalytic GIY-YIG domains of intron-encoded homing endonucleases. Besides, they have an N-terminally fused transcriptional regulators module, comprising the winged helix-turn-helix (wHTH) domain and uncharacterized domain DUF2087. At this point, they are distinct from GIY-YIG homing endonucleases, which typically contain a variety of C-terminally fused nuclease-associated modular DNA-binding domains (NUMODs). Moreover, some key residues relevant to catalysis in GIY-YIG endonucleases are mutanted or absent in this family, which suggests that members in this family might lose the catalytic function that GIY-YIG endonucleases possess. This family also includes many uncharacterized hypothetical proteins that consist of a standalone GIY-YIG like domain.','cd10451','GIY-YIG_LuxR_like'),(1011081,'gnl|CDD|312882',_binary 'Conserved phage C-terminus (Phg_2220_C).  This entry represents the conserved C-terminal domain of a family of proteins found exclusively in bacteriophage and in bacterial prophage regions. The functions of this domain and the proteins containing it are unknown.','pfam09524','Phg_2220_C'),(1011082,'gnl|CDD|274039',_binary 'phage conserved hypothetical protein, C-terminal domain.  This model represents the conserved C-terminal domain of a family of proteins found exclusively in bacteriophage and in bacterial prophage regions. The functions of this domain and the proteins containing it are unknown. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR02220','hypothetical_protein_phag'),(1011542,'gnl|CDD|288299',_binary 'Loader and inhibitor of phage G40P.  G39P inhibits the initiation of DNA replication by blocking G40P replicative helicase. G39P has a bipartite stricture consisting of a folded N-terminal domain and an unfolded C-terminal domain. The C terminal is essential for helicase interaction.','pfam11417','Inhibitor_G39P'),(1011636,'gnl|CDD|307788',_binary 'IPT/TIG domain.  This family consists of a domain that has an immunoglobulin like fold. These domains are found in cell surface receptors such as Met and Ron as well as in intracellular transcription factors where it is involved in DNA binding. CAUTION: This family does not currently recognize a significant number of members.','pfam01833','TIG'),(1011637,'gnl|CDD|238338',_binary 'IPT domain (domain D) of cyclodextrin glycosyltransferase (CGTase) and similar enzymes. These enzymes are involved in the enzymatic hydrolysis of alpha-1,4 linkages of starch polymers and belong to the glycosyl hydrolase family 13. Most consist of three domains (A,B,C) but CGTase is more complex and has two additional domains (D,E). The function of the IPT/D domain is unknown.','cd00604','IPT_CGTD'),(1011687,'gnl|CDD|309932',_binary 'Nucleoside 2-deoxyribosyltransferase.  Nucleoside 2-deoxyribosyltransferase EC:2.4.2.6 catalyzes the cleavage of the glycosidic bonds of 2`-deoxyribonucleosides.','pfam05014','Nuc_deoxyrib_tr'),(1011721,'gnl|CDD|133380',_binary 'Helix-Turn-Helix DNA binding domain of GlnR-like transcription regulators.  Helix-turn-helix (HTH) transcription regulator GlnR and related proteins, N-terminal domain. The GlnR and TnrA (also known as ScgR) proteins have been shown to regulate expression of glutamine synthetase as well as several genes involved in nitrogen metabolism. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.  A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.','cd01105','HTH_GlnR-like'),(1011998,'gnl|CDD|226040',_binary 'Poly(3-hydroxybutyrate) depolymerase [Secondary metabolites biosynthesis, transport, and catabolism].','COG3509','LpqC'),(1012000,'gnl|CDD|178549',_binary 'Probable pheophorbidase.','PLN02965','PLN02965'),(1012029,'gnl|CDD|313221',_binary 'Uncharacterized protein conserved in bacteria (DUF2188).  This domain, found in various hypothetical bacterial proteins, has no known function.','pfam09954','DUF2188'),(1012031,'gnl|CDD|226420',_binary 'Predicted transcriptional regulator [Transcription].','COG3905','COG3905'),(1012094,'gnl|CDD|309402',_binary 'Protein of unknown function (DUF433).','pfam04255','DUF433'),(1012095,'gnl|CDD|225291',_binary 'Uncharacterized conserved protein [Function unknown].','COG2442','COG2442'),(1012695,'gnl|CDD|318456',_binary 'GxGYxY sequence motif in domain of unknown function N-terminal.  This domain is found in bacteria, archaea and eukaryotes, and is typically between 213 and 231 amino acids in length. This domain is found in association with pfam14323.','pfam16216','GxGYxYP_N'),(1013377,'gnl|CDD|236300',_binary 'hypothetical protein; Provisional.','PRK08576','PRK08576'),(1013378,'gnl|CDD|273945',_binary 'thioredoxin-dependent adenylylsulfate APS reductase.  This model describes recently identified adenosine 5\'-phosphosulfate (APS) reductase activity found in sulfate-assimilatory prokaryotes, thus separating it from the traditionally described phosphoadenosine 5\'-phosphosulfate (PAPS) reductases found in bacteria and fungi. Homologous to PAPS reductase in enterobacteria, cyanobacteria, and yeast, APS reductase here clusters with, and demonstrates greater homology to plant APS reductase. Additionally, the presence of two conserved C-terminal motifs (CCXXRKXXPL & SXGCXXCT) distinguishes APS substrate specificity and serves as a FeS cluster. [Central intermediary metabolism, Sulfur metabolism].','TIGR02055','Phosphoadenosine_phosphos'),(1013379,'gnl|CDD|184745',_binary 'hypothetical protein; Provisional.','PRK14561','PRK14561'),(1013380,'gnl|CDD|274658',_binary 'N-acetyl sugar amidotransferase.  This enzyme has been implicated in the formation of the acetamido moiety (sugar-NC(=NH)CH3) which is found on some exopolysaccharides and is positively charged at neutral pH. The reaction involves ligation of ammonia with a sugar N-acetyl group, displacing water. In E. coli (O145 strain) and Pseudomonas aeruginosa (O12 strain) this gene is known as wbuX and ifnA respectively and likely acts on sialic acid. In Campylobacter jejuni, the gene is known as pseA and acts on pseudaminic acid in the process of flagellin glycosylation. In other Pseudomonas strains and various organisms it is unclear what the identity of the sugar substrate is, and in fact, the phylogenetic tree of this family sports a considerably deep branching suggestive of possible major differences in substrate structure. Nevertheless, the family is characterized by a conserved tetracysteine motif (CxxC.....[GN]xCxxC) possibly indicative of a metal binding site, as well as an invariable contextual association with homologs of the HisH and HisF proteins known as WbuY and WbuZ, respectively. These two proteins are believed to supply the enzyme with ammonium by hydrolysis of glutamine and delivery through an ammonium conduit.','TIGR03573','WbuX'),(1013477,'gnl|CDD|309218',_binary 'Leo1-like protein.  Members of this family are part of the Paf1/RNA polymerase II complex. The Paf1 complex probably functions during the elongation phase of transcription. The Leo1 subunit of the yeast Paf1-complex binds RNA and contributes to complex recruitment. The subunit acts by co-ordinating co-transcriptional chromain modifications and helping recruitment of mRNA 3prime-end processing factors.','pfam04004','Leo1'),(1013478,'gnl|CDD|239651',_binary 'Coenzyme B12-dependent-methylmalonyl coenzyme A (CoA) mutase (MCM) family, Alpha subunit-like subfamily; contains proteins similar to the alpha subunit of Propionbacterium shermanni MCM, as well as human and E. coli MCM. Members of this subfamily contain an N-terminal MCM domain and a C-terminal coenzyme B12 binding domain. MCM catalyzes the isomerization of methylmalonyl-CoA to succinyl-CoA. The reaction proceeds via radical intermediates beginning with a substrate-induced homolytic cleavage of the Co-C bond of coenzyme B12 to produce cob(II)alamin and the deoxyadenosyl radical. MCM plays an important role in the conversion of propionyl-CoA to succinyl-CoA during the degradation of propionate for the Krebs cycle. In higher animals, MCM is involved in the breakdown of odd-chain fatty acids, several amino acids, and cholesterol. Methylobacterium extorquens MCM participates in the glyoxylate regeneration pathway. In M. extorquens, MCM forms a complex with MeaB; MeaB may protect MCM from irreversible inactivation. In some bacteria, MCM is involved in the reverse metabolic reaction, the rearrangement of succinyl-CoA to methylmalonyl-CoA. Examples include P. shermanni MCM during propionic acid fermentation, E.coli MCM in a pathway for the conversion of succinate to propionate and Streptomyces MCM in polyketide biosynthesis. Sinorhizobium meliloti strain SU47 MCM plays a role in the polyhydroxyalkanoate degradation pathway. P. shermanni and Streptomyces cinnamonensis MCMs are alpha/beta heterodimers. It has been shown for P. shermanni MCM that only the alpha subunit binds coenzyme B12 and substrates. Human MCM is a homodimer with two active sites. Mouse and E.coli MCMs are also homodimers. In humans, impaired activity of MCM results in methylmalonic aciduria, a disorder of propionic acid metabolism.','cd03679','MM_CoA_mutase_alpha_like'),(1013859,'gnl|CDD|274884',_binary 'mycofactocin system glycosyltransferase.  Members of this protein family are putative glycosyltransferases, members of pfam00535 (glycosyl transferase family 2). Members appear mostly in the Actinobacteria, where they appear to be part of a system for converting a precursor peptide (TIGR03969) into a novel redox carrier designated mycofactocin. A radical SAM enzyme, TIGR03962, is a proposed to be a key maturase for mycofactocin.','TIGR03965','glycosyltransferase_Rv069'),(1013871,'gnl|CDD|284686',_binary 'Type III secretion protein YscO.  This family contains the bacterial type III secretion protein YscO, which is approximately 150 residues long. YscO has been shown to be required for high-level expression and secretion of the anti-host proteins V antigen and Yops in Yersinia pestis.','pfam07321','YscO'),(1013900,'gnl|CDD|178945',_binary 'putative inner membrane protein translocase component YidC; Validated.','PRK00247','PRK00247'),(1013951,'gnl|CDD|274996',_binary 'metallophosphoesterase, DNA ligase-associated.  Members of this protein family are an uncharacterized putative metallophosphoesterase associated with a DNA ligase, a helicase, and a putative exonuclease. It may play a role in DNA repair. Its system is present in about 12 % of prokaryotic reference genomes.','TIGR04123','hypothetical_protein'),(1013978,'gnl|CDD|315760',_binary 'AAA domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily. Many of the proteins in this family are conjugative transfer proteins. This family includes the PrrC protein that is thought to be the active component of the anticodon nuclease.','pfam13166','AAA_13'),(1013979,'gnl|CDD|310624',_binary 'Septation ring formation regulator, EzrA.  During the bacterial cell cycle, the tubulin-like cell-division protein FtsZ polymerizes into a ring structure that establishes the location of the nascent division site. EzrA modulates the frequency and position of FtsZ ring formation.','pfam06160','EzrA'),(1014071,'gnl|CDD|310363',_binary 'Weak chloroplast movement under blue light.  WEMBL consists of several plant proteins required for the chloroplast avoidance response under high intensity blue light. This avoidance response consists in the relocation of chloroplasts on the anticlinal side of exposed cells. Acts in association with PMI2 to maintain the velocity of chloroplast photo-relocation movement via the regulation of cp-actin filaments. Thus several member-sequences are described as \'myosin heavy chain-like\'.','pfam05701','WEMBL'),(1014093,'gnl|CDD|237790',_binary 'putative mercuric reductase; Provisional.','PRK14694','PRK14694'),(1014161,'gnl|CDD|227618',_binary 'Phage-related tail fibre protein [General function prediction only].','COG5301','COG5301'),(1014167,'gnl|CDD|308471',_binary 'LigT like Phosphoesterase.  Members of this family are bacterial and archaeal RNA ligases that are able to ligate tRNA half molecules containing 2\',3\'-cyclic phosphate and 5\' hydroxyl termini to products containing the 2\',5\' phosphodiester linkage. Each member of this family contains an internal duplication, each of which contains an HXTX motif that defines the family. The structure of a related protein is known. They belong to the 2H phosphoesterase superfamily. They share a common active site, characterized by two conserved histidines, with vertebrate myelin-associated 2\',3\' phosphodiesterases, plant Arabidopsis thaliana CPDases and several several bacteria and virus proteins.','pfam02834','LigT_PEase'),(1014247,'gnl|CDD|238739',_binary 'VWA YIEM type: Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses  In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains. Members of this subgroup have a conserved MIDAS motif, however, their biochemical function is not well characterised.','cd01462','VWA_YIEM_type'),(1014472,'gnl|CDD|274912',_binary 'mycofactocin system FadH/OYE family oxidoreductase 2.  The yeast protein called old yellow enzyme and FadH from Escherichia coli (2,4-dienoyl CoA reductase) are enzymes with 4Fe-4S, FMN, and FAD prosthetic groups, and interact with NADPH as well as substrate. Members of this related protein family occur in the vicinity of the putative mycofactocin biosynthesis operon in a number of Actinobacteria such as Frankia sp. and Rhodococcus sp., in Pelotomaculum thermopropionicum SI (Firmicutes), and in Geobacter uraniireducens Rf4 (Deltaproteobacteria). The function of this oxidoreductase is unknown.','TIGR03997','NADH:flavin_oxidoreductas'),(1014669,'gnl|CDD|198385',_binary 'Catalytic GIY-YIG domain of eukaryotic DNA mismatch repair protein MutS homologs.  This family represents a putative GIY-YIG nuclease domain C-terminally fused to the DNA-repair ATPase on a small group of eukaryotic DNA mismatch repair protein mutS homologs (MSH). The MSH proteins in this family do not have the zinc finger domain, but have a predicted mitochondrial localization. They might play roles in the recognition and repair of errors made during the replication of DNA. The prototype of this family is the protein encoded by the chloroplast mutator (CHM) locus from Arabidopsis thaliana. It is suggested that this protein could be involved in the maintenance of mitochondrial genome stability.','cd10438','GIY-YIG_MSH'),(1014688,'gnl|CDD|129721',_binary 'Holliday junction DNA helicase, RuvB subunit.  All proteins in this family for which functions are known are 5\'-3\' DNA helicases that, as part of a complex with RuvA homologs serve as a 5\'-3\' Holliday junction helicase. RuvA specifically binds Holliday junctions as a sandwich of two tetramers and maintains the configuration of the junction. It forms a complex with two hexameric rings of RuvB, the subunit that contains helicase activity. The complex drives ATP-dependent branch migration of the Holliday junction recombination intermediate. The endonuclease RuvC resolves junctions. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00635','Holliday_junction_ATP-dep'),(1014691,'gnl|CDD|315699',_binary 'AAA domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily. Many of the proteins in this family are conjugative transfer proteins.','pfam13086','AAA_11'),(1014727,'gnl|CDD|153120',_binary 'Ribonucleotide Reductase R2-like protein, Mn/Fe-binding domain.  Rv0233 is a Mycobacterium tuberculosis ribonucleotide reductase R2 protein with a  heterodinuclear manganese/iron-carboxylate cofactor located in its metal center. The Rv0233-like family may represent a structural/functional counterpart of the evolutionary ancestor of the RNRR2\'s (Ribonucleotide Reductase, R2/beta subunit) and the bacterial multicomponent monooxygenases.  RNRR2s belong to a broad superfamily of ferritin-like diiron-carboxylate proteins. The RNR protein catalyzes the conversion of ribonucleotides to deoxyribonucleotides and is found in prokaryotes and archaea. The catalytically active form of RNR is a proposed alpha2-beta2 tetramer. The homodimeric alpha subunit (R1) contains the active site and redox active cysteines as well as the allosteric binding sites.','cd07911','RNRR2_Rv0233_like'),(1014772,'gnl|CDD|161676',_binary 'dTMP kinase.  Function: phosphorylation of DTMP to form DTDP in both de novo and salvage pathways of DTTP synthesis. Catalytic activity: ATP + thymidine 5\'-phosphate = ADP + thymidine 5\'-diphosphate. [Purines, pyrimidines, nucleosides, and nucleotides, Nucleotide and nucleoside interconversions].','TIGR00041','Thymidylate_kinase'),(1014784,'gnl|CDD|307680',_binary 'Macro domain.  This domain is an ADP-ribose binding module. It is found in a number of otherwise unrelated proteins. It is found at the C-terminus of the macro-H2A histone protein. This domain is found in the non-structural proteins of several types of ssRNA viruses such as NSP3 from alphaviruses. This domain is also found on its own in a family of proteins from bacteria, archaebacteria and eukaryotes.','pfam01661','Macro'),(1014821,'gnl|CDD|313658',_binary 'Leucine-rich repeats of kinetochore protein Cenp-F/LEK1.  Cenp-F, a centromeric kinetochore, microtubule-binding protein consisting of two 1,600-amino acid-long coils, is essential for the full functioning of the mitotic checkpoint pathway. There are several leucine-rich repeats along the sequence of LEK1 that are considered to be zippers, though they do not appear to be binding DNA directly in this instance.','pfam10473','CENP-F_leu_zip'),(1015704,'gnl|CDD|318152',_binary 'Junction-mediating and -regulatory protein.  JMY, Junction-mediating and -regulatory protein is also a WASP homolog-associated protein with actin, membranes and microtubules. This middle region is the coiled-coil region that putatively binds microtubules to the scaffold. This ability to interact with microtubules plays a role in membrane tubulation.','pfam15871','JMY'),(1015757,'gnl|CDD|234814',_binary 'thymidylate kinase; Validated.','PRK00698','tmk'),(1015758,'gnl|CDD|238835',_binary 'Thymidine monophosphate kinase (TMPK), also known as thymidylate kinase, catalyzes the phosphorylation of thymidine monophosphate (TMP) to thymidine diphosphate (TDP) utilizing ATP as its preferred phophoryl donor. TMPK represents the rate-limiting step in either de novo or salvage biosynthesis of thymidine triphosphate (TTP).','cd01672','TMPK'),(1015759,'gnl|CDD|238026',_binary 'Guanosine monophosphate kinase (GMPK, EC 2.7.4.8), also known as guanylate kinase (GKase), catalyzes the reversible phosphoryl transfer from adenosine triphosphate (ATP) to guanosine monophosphate (GMP) to yield adenosine diphosphate (ADP) and guanosine diphosphate (GDP). It plays an essential role in the biosynthesis of guanosine triphosphate (GTP). This enzyme is also important for the activation of some antiviral and anticancer agents, such as acyclovir, ganciclovir, carbovir, and thiopurines.','cd00071','GMPK'),(1015760,'gnl|CDD|223203',_binary 'Thymidylate kinase [Nucleotide transport and metabolism].','COG0125','Tmk'),(1016357,'gnl|CDD|99916',_binary 'DNA polymerase type-B delta subfamily catalytic domain. Three DNA-dependent DNA polymerases type B (alpha, delta, and epsilon) have been identified as essential for nuclear DNA replication in eukaryotes. Presently, no direct data is available regarding the strand specificity of DNA polymerase during DNA replication in vivo. However, mutation analysis supports the hypothesis that DNA polymerase delta is the enzyme responsible for both elongation and maturation of Okazaki fragments on the lagging strand.','cd05533','POLBc_delta'),(1016415,'gnl|CDD|236135',_binary 'hypothetical protein; Provisional.','PRK07945','PRK07945'),(1016416,'gnl|CDD|213994',_binary 'Polymerase and Histidinol Phosphatase domain of Histidinol phosphate phosphatase of Hisj like.  Bacillus subtilis YtvP HisJ has strong histidinol phosphate phosphatase (HisPPase) activity. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. HisPPase catalyzes the eighth step of histidine biosynthesis, in which L-histidinol phosphate undergoes dephosphorylation to produce histidinol. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. The PHP domain of HisPPase is structurally homologous to other members of the PHP family that have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd12110','PHP_HisPPase_Hisj_like'),(1016460,'gnl|CDD|213552',_binary '(p)ppGpp synthetase, RelA/SpoT family.  The functions of E. coli RelA and SpoT differ somewhat. RelA (EC 2.7.6.5) produces pppGpp (or ppGpp) from ATP and GTP (or GDP). SpoT (EC 3.1.7.2) degrades ppGpp, but may also act as a secondary ppGpp synthetase. The two proteins are strongly similar. In many species, a single homolog to SpoT and RelA appears reponsible for both ppGpp synthesis and ppGpp degradation. (p)ppGpp is a regulatory metabolite of the stringent response, but appears also to be involved in antibiotic biosynthesis in some species. [Cellular processes, Adaptations to atypical conditions].','TIGR00691','PppGpp_synthetase.'),(1016461,'gnl|CDD|223394',_binary 'Guanosine polyphosphate pyrophosphohydrolases/synthetases [Signal transduction mechanisms / Transcription].','COG0317','SpoT'),(1016462,'gnl|CDD|236843',_binary 'bifunctional (p)ppGpp synthetase II/ guanosine-3\',5\'-bis pyrophosphate 3\'-pyrophosphohydrolase; Provisional.','PRK11092','PRK11092'),(1016463,'gnl|CDD|315898',_binary 'HD domain.  HD domains are metal dependent phosphohydrolases.','pfam13328','HD_4'),(1016464,'gnl|CDD|307883',_binary 'HD domain.  HD domains are metal dependent phosphohydrolases.','pfam01966','HD'),(1016465,'gnl|CDD|226789',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4339','COG4339'),(1016479,'gnl|CDD|179535',_binary 'ribose-phosphate pyrophosphokinase; Provisional.','PRK03092','PRK03092'),(1016494,'gnl|CDD|275231',_binary 'small Trp-rich protein.  Members of this bacterial protein family average 80 residues in length, and average nearly 6 Trp residues (two of which are invariant) in the first 45, which are strongly hydrophobic. Past this region, the protein is highly charged, with large numbers of Lys, Arg, Asp, and Glu residues. Members usually are divergently transcribed from a gene encoding a c-type cytochrome.','TIGR04438','hypothetical_protein'),(1016529,'gnl|CDD|239446',_binary 'Macro domain, Unknown family 2. The macro domain is a high-affinity ADP-ribose binding module found in a variety of proteins as a stand-alone domain or in combination with other domains like in histone macroH2A and some PARPs (poly ADP-ribose polymerases). Some macro domains recognize poly ADP-ribose as a ligand. Previously identified as displaying an Appr-1\'-p (ADP-ribose-1\'-monophosphate) processing activity, the macro domain may play roles in distinct ADP-ribose pathways, such as the ADP-ribosylation of proteins, an important post-translational modification which occurs in DNA repair, transcription, chromatin biology, and long-term memory formation, among other processes. This family is composed of uncharacterized proteins containing a stand-alone macro domain.','cd03330','Macro_2'),(1016530,'gnl|CDD|239231',_binary 'Macro domain, BAL_like family. The macro domain is a high-affinity ADP-ribose binding module found in a variety of proteins as a stand-alone domain or in combination with other domains like in histone macroH2A and some PARPs (poly ADP-ribose polymerases).  Some macro domains recognize poly ADP-ribose as a ligand. Previously identified as displaying an Appr-1\'-p (ADP-ribose-1\'-monophosphate) processing activity, the macro domain may play roles in distinct ADP-ribose pathways, such as the ADP-ribosylation of proteins, an important post-translational modification which occurs in DNA repair, transcription, chromatin biology, and long-term memory formation, among other processes. Members of this family show similarity to BAL (B-aggressive lymphoma) proteins, which contain one to three macro domains. Most BAL family macro domains belong to this family except for the most N-terminal domain in multiple-domain containing proteins. Most BAL proteins also contain a C-terminal PARP active site and are also named as PARPs. Human BAL1  (or PARP-9) was originally identified as a risk-related gene in diffuse large B-cell lymphoma that promotes malignant B-cell migration. Some BAL family proteins exhibit PARP activity. Poly (ADP-ribosyl)ation is an immediate DNA-damage-dependent post-translational modification of histones and other nuclear proteins. BAL proteins may also function as transcription repressors.','cd02903','Macro_BAL_like'),(1016536,'gnl|CDD|234717',_binary 'thiol:disulfide interchange protein precursor; Provisional.','PRK00293','dipZ'),(1016537,'gnl|CDD|223868',_binary 'Lipoproteins [Cell envelope biogenesis, outer membrane].','COG0797','RlpA'),(1016548,'gnl|CDD|181354',_binary 'cobalt-precorrin-6Y C(15)-methyltransferase; Validated.','PRK08287','PRK08287'),(1016568,'gnl|CDD|225203',_binary 'Predicted metalloprotease [General function prediction only].','COG2321','COG2321'),(1016569,'gnl|CDD|282131',_binary 'Putative neutral zinc metallopeptidase.  Members of this family have a predicted zinc binding motif characteristic of neutral zinc metallopeptidases (Prosite:PDOC00129).','pfam04228','Zn_peptidase'),(1016991,'gnl|CDD|223783',_binary 'F0F1-type ATP synthase, subunit b [Energy production and conversion].','COG0711','AtpF'),(1017047,'gnl|CDD|237118',_binary 'hypothetical protein; Provisional.','PRK12496','PRK12496'),(1017454,'gnl|CDD|237756',_binary 'putative protein serine/threonine phosphatase; Provisional.','PRK14559','PRK14559'),(1018116,'gnl|CDD|226883',_binary 'Negative regulator of septation ring formation [Cell division and chromosome partitioning].','COG4477','EzrA'),(1018664,'gnl|CDD|225318',_binary 'Predicted methyltransferase [General function prediction only].','COG2520','COG2520'),(1019134,'gnl|CDD|234938',_binary 'ATP-dependent RNA helicase RhlB; Provisional.','PRK01297','PRK01297'),(1019214,'gnl|CDD|181044',_binary 'short chain dehydrogenase; Provisional.','PRK07577','PRK07577'),(1019216,'gnl|CDD|181113',_binary 'short chain dehydrogenase; Provisional.','PRK07775','PRK07775'),(1019223,'gnl|CDD|235925',_binary 'sorbitol dehydrogenase; Provisional.','PRK07067','PRK07067'),(1019224,'gnl|CDD|235914',_binary 'short chain dehydrogenase; Provisional.','PRK07041','PRK07041'),(1019231,'gnl|CDD|236209',_binary 'short chain dehydrogenase; Provisional.','PRK08265','PRK08265'),(1019233,'gnl|CDD|237220',_binary 'short chain dehydrogenase; Provisional.','PRK12828','PRK12828'),(1019235,'gnl|CDD|187623',_binary '7 alpha-hydroxysteroid dehydrogenase (7 alpha-HSDH), classical (c) SDRs.  This bacterial subgroup contains 7 alpha-HSDHs,  including Escherichia coli 7 alpha-HSDH. 7 alpha-HSDH, a member of the SDR family, catalyzes the NAD+ -dependent dehydrogenation of a hydroxyl group at position 7 of  the steroid skeleton of bile acids. In humans the two primary bile acids are cholic and chenodeoxycholic acids, these are formed from cholesterol in the liver. Escherichia coli 7 alpha-HSDH dehydroxylates these bile acids in the human intestine. Mammalian 7 alpha-HSDH activity has been found in livers. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05365','7_alpha_HSDH_SDR_c'),(1019249,'gnl|CDD|180399',_binary 'short chain dehydrogenase; Provisional.','PRK06101','PRK06101'),(1019259,'gnl|CDD|187666',_binary 'human estrogenic 17beta-hydroxysteroid dehydrogenase type 1 (type 1 17beta-HSD)-like, classical (c) SDRs.  17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. This classical SDR subgroup includes human type 1 17beta-HSD, human retinol dehydrogenase 8, zebrafish photoreceptor associated retinol dehydrogenase type 2, and a chicken ovary-specific 17beta-hydroxysteroid dehydrogenase. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd09806','type1_17beta-HSD-like_SDR'),(1019271,'gnl|CDD|132250',_binary '2-hydroxycyclohexanecarboxyl-CoA dehydrogenase.  Members of this protein family are the enzyme 2-hydroxycyclohexanecarboxyl-CoA dehydrogenase. The enzymatic properties were confirmed experimentally in Rhodopseudomonas palustris; the enzyme is homotetrameric, and not sensitive to oxygen. This enzyme is part of proposed pathway for degradation of benzoyl-CoA to 3-hydroxypimeloyl-CoA that differs from the analogous in Thauera aromatica. It also may occur in degradation of the non-aromatic compound cyclohexane-1-carboxylate.','TIGR03206','benzo_BadH'),(1019273,'gnl|CDD|187628',_binary 'classical (c) SDR, subgroup 2.  Short-chain dehydrogenases/reductases (SDRs, aka Tyrosine-dependent oxidoreductases) are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05370','SDR_c2'),(1019280,'gnl|CDD|236308',_binary 'enoyl-(acyl carrier protein) reductase; Provisional.','PRK08594','PRK08594'),(1020259,'gnl|CDD|313024',_binary 'Myosin-like coiled-coil protein.  Taxilin contains an extraordinarily long coiled-coil domain in its C-terminal half and is ubiquitously expressed. It is a novel binding partner of several syntaxin family members and is possibly involved in Ca2+-dependent exocytosis in neuroendocrine cells. Gamma-taxilin, described as leucine zipper protein Factor Inhibiting ATF4-mediated Transcription (FIAT), localizes to the nucleus in osteoblasts and dimerizes with ATF4 to form inactive dimers, thus inhibiting ATF4-mediated transcription.','pfam09728','Taxilin'),(1020336,'gnl|CDD|235128',_binary '23S rRNA methyluridine methyltransferase; Reviewed.','PRK03522','rumB'),(1020356,'gnl|CDD|235628',_binary 'short chain dehydrogenase; Validated.','PRK05855','PRK05855'),(1020359,'gnl|CDD|223999',_binary 'Hydrolases of the alpha/beta superfamily [General function prediction only].','COG1073','COG1073'),(1020360,'gnl|CDD|223489',_binary 'Dienelactone hydrolase and related enzymes [Secondary metabolites biosynthesis, transport, and catabolism].','COG0412','COG0412'),(1020362,'gnl|CDD|224561',_binary 'Esterase/lipase [General function prediction only].','COG1647','COG1647'),(1020363,'gnl|CDD|225749',_binary 'Predicted thioesterase involved in non-ribosomal peptide biosynthesis [Secondary metabolites biosynthesis, transport, and catabolism].','COG3208','GrsT'),(1020365,'gnl|CDD|273437',_binary 'adenine phosphoribosyltransferase.  A phylogenetic analysis suggested omitting the bi-directional best hit homologs from the spirochetes from the seed for this model and making only tentative predictions of adenine phosphoribosyltransferase function for this lineage. The trusted cutoff score is made high for this reason. Most proteins scoring between the trusted and noise cutoffs are likely to act as adenine phosphotransferase. [Purines, pyrimidines, nucleosides, and nucleotides, Salvage of nucleosides and nucleotides].','TIGR01090','Adenine_phosphoribosyltra'),(1020380,'gnl|CDD|307599',_binary 'Transposase.  Transposase proteins are necessary for efficient DNA transposition. This family consists of various E. coli insertion elements and other bacterial transposases some of which are members of the IS3 family.','pfam01527','HTH_Tnp_1'),(1020381,'gnl|CDD|225511',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG2963','COG2963'),(1020561,'gnl|CDD|316108',_binary 'Winged helix-turn helix.  This helix-turn-helix domain is often found in transferases and is likely to be DNA-binding.','pfam13551','HTH_29'),(1020650,'gnl|CDD|319794',_binary 'alpha-subunit of Na(+)/K(+)-ATPases and of gastric H(+)/K(+)-ATPase, similar to the human Na(+)/K(+)-ATPase alpha subunits 1-4.  This subfamily includes the alpha subunit of Na(+)/K(+)-ATPase a heteromeric transmembrane protein composed of an alpha- and beta-subunit and an optional third subunit belonging to the FXYD proteins which are more tissue specific regulatory subunits of the enzyme. The alpha-subunit is the catalytic subunit responsible for transport activities of the enzyme. This subfamily includes all four isotopes of the human alpha subunit: (alpha1-alpha4, encoded by the ATP1A1- ATP1A4 genes).  Na(+)/K(+)-ATPase functions chiefly as an ion pump, hydrolyzing one molecule of ATP to pump three Na(+) out of the cell in exchange for two K(+)entering the cell per pump cycle. In addition Na(+)/K(+)-ATPase acts as a signal transducer. This subfamily also includes Oreochromis mossambicus (tilapia) Na(+)/K(+)-ATPase alpha 1 and alpha 3 subunits, and gastric H(+)/K(+)-ATPase which exchanges hydronium ion with potassium and is responsible for gastric acid secretion. Gastric H(+)/K(+)-ATPase is an alpha,beta-heterodimeric enzyme. This subfamily belongs to the P-type ATPases, a large family of integral membrane transporters that are of critical importance in all kingdoms of life. They generate and maintain (electro-) chemical gradients across cellular membranes, by translocating cations, heavy metals and lipids, and are distinguished from other main classes of transport ATPases (F- , V- , and ABC- type) by the formation of a phosphorylated (P-) intermediate state in the catalytic cycle.','cd02608','P-type_ATPase_Na-K_like'),(1020875,'gnl|CDD|309591',_binary 'Baculovirus polyhedron envelope protein, PEP, C terminus.  Polyhedra are large crystalline occlusion bodies containing nucleopolyhedrovirus virions, and surrounded by an electron-dense structure called the polyhedron envelope or polyhedron calyx. The polyhedron envelope (associated) protein PEP is thought to be an integral part of the polyhedron envelope. PEP is concentrated at the surface of polyhedra, and is thought to be important for the proper formation of the periphery of polyhedra. It is thought that PEP may stabilize polyhedra and protect them from fusion or aggregation.','pfam04513','Baculo_PEP_C'),(1021517,'gnl|CDD|239332',_binary 'Arsenate Reductase (ArsC) family, ArsC subfamily; arsenic reductases similar to that encoded by arsC on the R733 plasmid of Escherichia coli. E. coli ArsC catalyzes the reduction of arsenate [As(V)] to arsenite [As(III)], the first step in the detoxification of arsenic, using reducing equivalents derived from glutathione (GSH) via glutaredoxin (GRX). ArsC contains a single catalytic cysteine, within a thioredoxin fold, that forms a covalent thiolate-As(V) intermediate, which is reduced by GRX through a mixed GSH-arsenate intermediate. This family of predominantly bacterial enzymes is unrelated to two other families of arsenate reductases which show similarity to low-molecular-weight acid phosphatases and phosphotyrosyl phosphatases.','cd03034','ArsC_ArsC'),(1022212,'gnl|CDD|227642',_binary 'Spore coat assembly protein [Cell envelope biogenesis, outer membrane].','COG5337','CotH'),(1022653,'gnl|CDD|184989',_binary 'arginine decarboxylase; Provisional.','PRK15029','PRK15029'),(1022782,'gnl|CDD|165281',_binary 'hypothetical protein; Provisional.','PHA02977','PHA02977'),(1022797,'gnl|CDD|309289',_binary 'Rad52/22 family double-strand break repair protein.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to Rad52. These proteins contain two helix-hairpin-helix motifs.','pfam04098','Rad52_Rad22'),(1022838,'gnl|CDD|182648',_binary 'purine nucleoside phosphoramidase; Provisional.','PRK10687','PRK10687'),(1022839,'gnl|CDD|238609',_binary 'aprataxin related: Aprataxin, a HINT family hydrolase is mutated in ataxia oculomotor apraxia syndrome. All the members of this subgroup have the conserved HxHxHxx (where x is a hydrophobic residue) signature motif. Members of this subgroup are predominantly eukaryotic in origin.','cd01278','aprataxin_related'),(1023100,'gnl|CDD|99825',_binary 'Uncharacterized bacterial subgroup of the DEDDy 3\'-5\' exonuclease domain of family-B DNA polymerases.  A subfamily of the 3\'-5\' exonuclease domain of family-B DNA polymerases. This subfamily is composed of uncharacterized bacterial family-B DNA polymerases. Family-B DNA polymerases contain an N-terminal DEDDy DnaQ-like exonuclease domain in the same polypeptide chain as the polymerase domain, similar to family-A DNA polymerases. This exonuclease domain contains three sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are involved in metal binding and catalysis. The exonuclease domain of family-B DNA polymerases has a fundamental role in proofreading activity. It contains a beta hairpin structure that plays an important role in active site switching in the event of a nucleotide misincorporation. Family-B DNA polymerases are predominantly involved in DNA replication and DNA repair.','cd05782','DNA_polB_like1_exo'),(1023101,'gnl|CDD|225895',_binary 'Predicted exonuclease [DNA replication, recombination, and repair].','COG3359','COG3359'),(1023128,'gnl|CDD|183157',_binary 'hypothetical protein; Provisional.','PRK11479','PRK11479'),(1023129,'gnl|CDD|226381',_binary 'Uncharacterized distant relative of cell wall-associated hydrolases [Function unknown].','COG3863','COG3863'),(1023428,'gnl|CDD|307504',_binary 'Glycosyl hydrolase family 46.  This family are chitosanase enzymes.','pfam01374','Glyco_hydro_46'),(1023429,'gnl|CDD|238479',_binary 'Glycosyl hydrolase family 46 chitosanase domain. This family are composed of the chitosanase enzymes which hydrolyzes chitosan, a biopolymer of beta (1,4)-linked -D-glucosamine (GlcN) residues produced by partial or full deacetylation of chitin. Chitosanases play a role in defense against pathogens such as fungi and are found in microorganisms, fungi, viruses, and plants. Microbial chitosanases who members are the most prevalent can be divided into 3 subclasses based on the specificity of the cleavage positions for partial acetylated chitosan.  Subclass I chitosanases such as N174 can split GlcN-GlcN and GlcNAc-GlcN linkages, whereas subclass II chitosanases such as Bacillus sp. no. 7-M can cleave only GlcN-GlcN linkages. Subclass III chitosanases such as MH-K1 chitosanase are the most versatile and can split both GlcN-GlcN and GlcN-GlcNAc linkages.','cd00978','chitosanase_glyco_hydro_4'),(1024102,'gnl|CDD|131274',_binary 'putative phage cell wall peptidase, NlpC/P60 family.  Members of this family show sequence similarity to members of the NlpC/P60 family described by pfam00877 and by Anantharaman and Aravind (). The NlpC/P60 family includes a number of characterized bacterial cell wall hydrolases. Members of this related family are all found in prophage regions of bacterial genomes. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR02219','hypothetical_protein_CC_2'),(1024111,'gnl|CDD|315834',_binary 'Protein of unknown function (DUF4043).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 369 and 424 amino acids in length. There is a single completely conserved residue G that may be functionally important.','pfam13252','DUF4043'),(1025080,'gnl|CDD|214005',_binary 'Novel bacterial esterase that cleaves esters on halogenated cyclic compounds.  This family contains proteins similar to a novel bacterial esterase (Alcaligenes esterase 713) with the alpha/beta hydrolase fold but does not contain the GXSXXG pentapeptide around the active site serine residue as commonly seen in other enzymes of this class. Esterase 713 shows negligible sequence homology to other esterase and lipase enzymes. It is active as a dimer and cleaves esters on halogenated cyclic compounds though its natural substrate is unknown. This enzyme is possibly exported from the cytosol to the periplasmic space. A large majority of sequences in this family have yet to be characterized.','cd12806','Esterase_713_like'),(1025574,'gnl|CDD|180090',_binary 'aspartate alpha-decarboxylase; Provisional.','PRK05449','PRK05449'),(1025575,'gnl|CDD|132994',_binary 'Aspartate alpha-decarboxylase or L-aspartate 1-decarboxylase, a pyruvoyl group-dependent  decarboxylase in beta-alanine production.  Decarboxylation of aspartate is  the major route of beta-alanine production in bacteria, and is catalyzed  by the enzyme L-aspartate decarboxylase (ADC), EC:4.1.1.11 which  requires a pyruvoyl group for its activity. The pyruvoyl cofactor is  covalently bound to the enzyme. The protein is synthesized as a  proenzyme and cleaved via self-processing at Gly23-Ser24 to yield an  alpha chain (C-terminal fragment) and beta chain (N-terminal fragment),  and the pyruvoyl group. Beta-alanine is required for the biosynthesis of  pantothenate, in which the enzyme plays a critical regulatory role. The  active site of the tetrameric enzyme is located at the interface of two  subunits, with a Lysine and a Histidine from the beta chain of one  subunit forming the active site with residues from the alpha chain of  the adjacent subunit. This alignment model spans the precursor (or both  beta and alpha chains) of aspartate decarboxylase.','cd06919','Asp_decarbox'),(1025576,'gnl|CDD|308076',_binary 'Aspartate decarboxylase.  Decarboxylation of aspartate is the major route of beta-alanine production in bacteria, and is catalyzed by the enzyme aspartate decarboxylase EC:4.1.1.11 which requires a pyruvoyl group for its activity. It is synthesized initially as a proenzyme which is then proteolytically cleaved to an alpha (C-terminal) and beta (N-terminal) subunit and a pyruvoyl group. This family contains both chains of aspartate decarboxylase.','pfam02261','Asp_decarbox'),(1025577,'gnl|CDD|225265',_binary 'ABC-type ATPase fused to a predicted acetyltransferase domain [General function prediction only].','COG2401','COG2401'),(1025578,'gnl|CDD|223922',_binary 'Aspartate 1-decarboxylase [Coenzyme metabolism].','COG0853','PanD'),(1025579,'gnl|CDD|213179',_binary 'ATP-binding cassette transporter nucleotide-binding domain.  ABC transporters are a large family of proteins involved in the transport of a wide variety of different compounds, like sugars, ions, peptides, and more complex organic molecules. The nucleotide-binding domain shows the highest similarity between all members of the family. ABC transporters are a subset of nucleotide hydrolases that contain a signature motif, Q-loop, and H-loop/switch region, in addition to, the Walker A motif/P-loop and Walker B motif commonly found in a number of ATP- and GTP-binding and hydrolyzing proteins.','cd00267','ABC_ATPase'),(1025580,'gnl|CDD|129327',_binary 'L-aspartate-alpha-decarboxylase.  Members of this family are aspartate 1-decarboxylase, the enzyme that makes beta-alanine and C02 from aspartate. Beta-alanine is then used to make the vitamin pantothenate, from which coenzyme A is made. Aspartate 1-decarboxylase is synthesized as a proenzyme, then cleaved to an alpha (C-terminal) and beta (N-terminal) subunit with a pyruvoyl group. [Biosynthesis of cofactors, prosthetic groups, and carriers, Pantothenate and coenzyme A].','TIGR00223','Aspartate_1-decarboxylase'),(1025581,'gnl|CDD|213184',_binary 'ABC-type transport system involved in Fe-S cluster assembly, ATPase component.  Biosynthesis of iron-sulfur clusters (Fe-S) depends on multi-protein systems. The SUF system of E. coli and Erwinia chrysanthemi is important for Fe-S biogenesis under stressful conditions. The SUF system is made of six proteins: SufC is an atypical cytoplasmic ABC-ATPase, which forms a complex with SufB and SufD; SufA plays the role of a scaffold protein for assembly of iron-sulfur clusters and delivery to target proteins; SufS is a cysteine desulfurase which mobilizes the sulfur atom from cysteine and provides it to the cluster; SufE has no associated function yet.','cd03217','ABC_FeS_Assembly'),(1025582,'gnl|CDD|181965',_binary 'cysteine desulfurase ATPase component; Reviewed.','PRK09580','sufC'),(1025583,'gnl|CDD|224046',_binary 'ABC-type Mn/Zn transport systems, ATPase component [Inorganic ion transport and metabolism].','COG1121','ZnuC'),(1025584,'gnl|CDD|213181',_binary 'ATP-binding component of iron-siderophores, vitamin B12 and hemin transporters and related proteins.  ABC transporters, involved in the uptake of siderophores, heme, and vitamin B12, are widely conserved in bacteria and archaea. Only very few species lack representatives of the siderophore family transporters. The E. coli BtuCD protein is an ABC transporter mediating vitamin B12 uptake. The two ATP-binding cassettes (BtuD) are in close contact with each other, as are the two membrane-spanning subunits (BtuC); this arrangement is distinct from that observed for the E. coli lipid flippase MsbA. The BtuC subunits provide 20 transmembrane helices grouped around a translocation pathway that is closed to the cytoplasm by a gate region, whereas the dimer arrangement of the BtuD subunits resembles the ATP-bound form of the Rad50 DNA repair enzyme. A prominent cytoplasmic loop of BtuC forms the contact region with the ATP-binding cassette and represent a conserved motif among the ABC transporters.','cd03214','ABC_Iron-Siderophores_B12'),(1025586,'gnl|CDD|223562',_binary 'ATPase components of ABC transporters with duplicated ATPase domains [General function prediction only].','COG0488','Uup'),(1025587,'gnl|CDD|224044',_binary 'ABC-type molybdenum transport system, ATPase component/photorepair protein PhrA [Inorganic ion transport and metabolism].','COG1119','ModF'),(1025590,'gnl|CDD|213202',_binary 'ATP-binding cassette domain of the metal-type transporters.  This family includes transporters involved in the uptake of various metallic cations such as iron, manganese, and zinc. The ATPases of this group of transporters are very similar to members of iron-siderophore uptake family suggesting that they share a common ancestor. The best characterized metal-type ABC transporters are the YfeABCD system of Y. pestis, the SitABCD system of Salmonella enterica serovar Typhimurium, and the SitABCD transporter of Shigella flexneri. Moreover other uncharacterized homologs of these metal-type transporters are mainly found in pathogens like Haemophilus or enteroinvasive E. coli isolates.','cd03235','ABC_Metallic_Cations'),(1025591,'gnl|CDD|213214',_binary 'ATP-binding cassette domain of CydCD, subfamily C.  The CYD subfamily implicated in cytochrome bd biogenesis. The CydC and CydD proteins are important for the formation of cytochrome bd terminal oxidase of E. coli and it has been proposed that they were necessary for biosynthesis of the cytochrome bd quinol oxidase and for periplasmic c-type cytochromes. CydCD were proposed to determine a heterooligomeric complex important for heme export into the periplasm or to be involved in the maintenance of the proper redox state of the periplasmic space. In Bacillus subtilis, the absence of CydCD does not affect the presence of halo-cytochrome c in the membrane and this observation suggests that CydCD proteins are not involved in the export of heme in this organism.','cd03247','ABCC_cytochrome_bd'),(1025592,'gnl|CDD|226646',_binary 'ABC-type uncharacterized transport system, permease and ATPase components [General function prediction only].','COG4178','COG4178'),(1025594,'gnl|CDD|223473',_binary 'Cysteine desulfurase activator ATPase [Posttranslational modification, protein turnover, chaperones].','COG0396','sufC'),(1025595,'gnl|CDD|213213',_binary 'ATP-binding cassette domain of PrtD, subfamily C.  This family represents the ABC component of the protease secretion system PrtD, a 60-kDa integral membrane protein sharing 37% identity with HlyB, the ABC component of the alpha-hemolysin secretion pathway, in the C-terminal domain. They export degradative enzymes by using a type I protein secretion system and lack an N-terminal signal peptide, but contain a C-terminal secretion signal. The Type I secretion apparatus is made up of three components, an ABC transporter, a membrane fusion protein (MFP), and an outer membrane protein (OMP). For the HlyA transporter complex, HlyB (ABC transporter) and HlyD (MFP) reside in the inner membrane of E. coli. The OMP component is TolC, which is thought to interact with the MFP to form a continuous channel across the periplasm from the cytoplasm to the exterior. HlyB belongs to the family of ABC transporters, which are ubiquitous, ATP-dependent transmembrane pumps or channels. The spectrum of transport substrates ranges from inorganic ions, nutrients such as amino acids, sugars, or peptides, hydrophobic drugs, to large polypeptides, such as HlyA.','cd03246','ABCC_Protease_Secretion'),(1025596,'gnl|CDD|315964',_binary 'AAA domain.','pfam13401','AAA_22'),(1025599,'gnl|CDD|185016',_binary 'manganese/iron transporter ATP-binding protein; Provisional.','PRK15056','PRK15056'),(1025600,'gnl|CDD|273360',_binary 'Peroxysomal Fatty Acyl CoA Transporter (FAT) Family protei.  [Transport and binding proteins, Carbohydrates, organic alcohols, and acids].','TIGR00954','Peroxisomal_long-chain_fa'),(1025601,'gnl|CDD|308234',_binary 'CoA-transferase family III.  CoA-transferases are found in organisms from all lines of descent. Most of these enzymes belong to two well-known enzyme families, but recent work on unusual biochemical pathways of anaerobic bacteria has revealed the existence of a third family of CoA-transferases. The members of this enzyme family differ in sequence and reaction mechanism from CoA-transferases of the other families. Currently known enzymes of the new family are a formyl-CoA: oxalate CoA-transferase, a succinyl-CoA: (R)-benzylsuccinate CoA-transferase, an (E)-cinnamoyl-CoA: (R)-phenyllactate CoA-transferase, and a butyrobetainyl-CoA: (R)-carnitine CoA-transferase. In addition, a large number of proteins of unknown or differently annotated function from Bacteria, Archaea and Eukarya apparently belong to this enzyme family. Properties and reaction mechanisms of the CoA-transferases of family III are described and compared to those of the previously known CoA-transferases.','pfam02515','CoA_transf_3'),(1025602,'gnl|CDD|224717',_binary 'Predicted acyl-CoA transferases/carnitine dehydratase [Energy production and conversion].','COG1804','CaiB'),(1025603,'gnl|CDD|213269',_binary 'ATP-binding cassette domain 2 of NFT1, subfamily C.  Domain 2 of NFT1 (New full-length MRP-type transporter 1). NFT1 belongs to the MRP (multidrug resistance-associated protein) family of ABC transporters. Some of the MRP members have five additional transmembrane segments in their N-terminus, but the function of these additional membrane-spanning domains is not clear. The MRP was found in the multidrug-resisting lung cancer cell in which p-glycoprotein was not overexpressed. MRP exports glutathione by drug stimulation, as well as, certain substrates in conjugated forms with anions such as glutathione, glucuronate, and sulfate.','cd03369','ABCC_NFT1'),(1026014,'gnl|CDD|197669',_binary 'helix_turn_helix gluconate operon transcriptional repressor.','smart00345','HTH_GNTR'),(1026015,'gnl|CDD|224639',_binary 'Predicted transcriptional regulators [Transcription].','COG1725','COG1725'),(1026016,'gnl|CDD|306822',_binary 'Bacterial regulatory proteins, gntR family.  This family of regulatory proteins consists of the N-terminal HTH region of GntR-like bacterial transcription factors. At the C-terminus there is usually an effector-binding/oligomerization domain. The GntR-like proteins include the following sub-families: MocR, YtrR, FadR, AraR, HutC and PlmA, DevA, DasR. Many of these proteins have been shown experimentally to be autoregulatory, enabling the prediction of operator sites and the discovery of cis/trans relationships. The DasR regulator has been shown to be a global regulator of primary metabolism and development in Streptomyces coelicolor.','pfam00392','GntR'),(1026017,'gnl|CDD|225097',_binary 'Transcriptional regulators [Transcription].','COG2186','FadR'),(1026018,'gnl|CDD|182065',_binary 'DNA-binding transcriptional repressor MngR; Provisional.','PRK09764','PRK09764'),(1026019,'gnl|CDD|224089',_binary 'Transcriptional regulators containing a DNA-binding HTH domain and an aminotransferase domain (MocR family) and their eukaryotic orthologs [Transcription / Amino acid transport and metabolism].','COG1167','ARO8'),(1026020,'gnl|CDD|183118',_binary 'DNA-binding transcriptional regulator FrlR; Provisional.','PRK11402','PRK11402'),(1026021,'gnl|CDD|274116',_binary 'trehalose operon repressor, B. subtilis-type.  This family consists of repressors of the GntR family typically associated with trehalose utilization operons. Trehalose is imported as trehalose-6-phosphate and then hydrolyzed by alpha,alpha-phosphotrehalase to glucose and glucose-6-P. This family includes repressors mostly from Gram-positive lineages and does not include the TreR from E. coli. [Regulatory functions, DNA interactions].','TIGR02404','Trehalose_operon_transcri'),(1026022,'gnl|CDD|188194',_binary 'histidine utilization repressor, proteobacterial.  This model represents a proteobacterial histidine utilization repressor. It is usually found clustered with the enzymes HutUHIG so that it can regulate its own expression as well. A number of species have several paralogs and may fine-tune the regulation according to levels of degradation intermediates such as urocanate. This family belongs to the larger GntR family of transcriptional regulators. [Energy metabolism, Amino acids and amines, Regulatory functions, DNA interactions].','TIGR02018','Histidine_utilization_rep'),(1026023,'gnl|CDD|184961',_binary 'histidine utilization repressor; Provisional.','PRK14999','PRK14999'),(1026024,'gnl|CDD|131378',_binary 'phosphonates metabolism transcriptional regulator PhnF.  All members of the seed alignment for this family are predicted helix-turn-helix transcriptional regulatory proteins of the broader gntR and are found associated with genes for the import and degradation of phosphonates and/or related compounds (e.g. phosphonites) with a direct C-P bond. [Transport and binding proteins, Anions, Regulatory functions, DNA interactions].','TIGR02325','Probable_transcriptional_'),(1026025,'gnl|CDD|236690',_binary 'DNA-binding transcriptional repressor LldR; Provisional.','PRK10421','PRK10421'),(1026026,'gnl|CDD|182227',_binary 'phosphonate metabolism transcriptional regulator PhnF; Provisional.','PRK10079','PRK10079'),(1026027,'gnl|CDD|224715',_binary 'Transcriptional regulators [Transcription].','COG1802','GntR'),(1026028,'gnl|CDD|182318',_binary 'DNA-binding transcriptional repressor UxuR; Provisional.','PRK10225','PRK10225'),(1026029,'gnl|CDD|181879',_binary 'transcriptional regulator PdhR; Reviewed.','PRK09464','pdhR'),(1026030,'gnl|CDD|132380',_binary 'phosphonate utilization transcriptional regulator PhnR.  This family of proteins are members of the GntR family (pfam00392) containing an N-terminal helix-turn-helix (HTH) motif. This clade is found adjacent to or inside of operons for the degradation of 2-aminoethylphosphonate (AEP) in Salmonella, Vibrio Aeromonas hydrophila, Hahella chejuensis and Psychromonas ingrahamii. [Regulatory functions, DNA interactions].','TIGR03337','phnR'),(1026031,'gnl|CDD|163028',_binary 'fatty acid metabolism transcriptional regulator FadR.  Members of this family are FadR, a transcriptional regulator of fatty acid metabolism, including both biosynthesis and beta-oxidation. It is found exclusively in a subset of Gammaproteobacteria, with strictly one copy per genome. It has an N-terminal DNA-binding domain and a less well conserved C-terminal long chain acyl-CoA-binding domain. FadR from this family heterologously expressed in Escherichia coli show differences in regulatory response and fatty acid binding profiles. The family is nevertheless designated equivalog, as all member proteins have at least nominally the same function. [Fatty acid and phospholipid metabolism, Biosynthesis, Fatty acid and phospholipid metabolism, Degradation, Regulatory functions, DNA interactions].','TIGR02812','Fatty_acid_metabolism_reg'),(1026032,'gnl|CDD|183176',_binary 'DNA-binding transcriptional repressor ExuR; Provisional.','PRK11523','PRK11523'),(1026033,'gnl|CDD|182186',_binary 'DNA-binding transcriptional regulator GlcC; Provisional.','PRK09990','PRK09990'),(1026034,'gnl|CDD|235166',_binary 'transcriptional regulator NanR; Provisional.','PRK03837','PRK03837'),(1026035,'gnl|CDD|185378',_binary 'transcriptional regulatory protein PtsJ; Provisional.','PRK15481','PRK15481'),(1026036,'gnl|CDD|179909',_binary 'fatty acid metabolism regulator; Provisional.','PRK04984','PRK04984'),(1026112,'gnl|CDD|235461',_binary 'seryl-tRNA synthetase; Provisional.','PRK05431','PRK05431'),(1026115,'gnl|CDD|223250',_binary 'Seryl-tRNA synthetase [Translation, ribosomal structure and biogenesis].','COG0172','SerS'),(1026116,'gnl|CDD|311978',_binary 'Spc7 kinetochore protein.  This domain is found in cell division proteins which are required for kinetochore-spindle association.','pfam08317','Spc7'),(1026117,'gnl|CDD|314215',_binary 'Subunit 21 of Mediator complex.  Med21 has been known as Srb7 in yeasts, hSrb7 in humans and Trap 19 in Drosophila. The heterodimer of the two subunits Med7 and Med21 appears to act as a hinge between the middle and the tail regions of Mediator.','pfam11221','Med21'),(1026118,'gnl|CDD|308162',_binary 'Seryl-tRNA synthetase N-terminal domain.  This domain is found associated with the Pfam tRNA synthetase class II domain (pfam00587) and represents the N-terminal domain of seryl-tRNA synthetase.','pfam02403','Seryl_tRNA_N'),(1028165,'gnl|CDD|237048',_binary 'GTPase CgtA; Reviewed.','PRK12299','obgE'),(1028166,'gnl|CDD|131827',_binary 'P-type conjugative transfer protein TrbJ.  The TrbJ protein is found in the trb locus of Agrobacterium Ti plasmids where it is involved in the type IV secretion system for plasmid conjugative transfer. TrbJ is a homolog of the F-type TraE protein (which is believed to be an inner membrane pore-forming protein, TIGR02761) as well as the vir system VirB5 protein.','TIGR02780','TrbJ_protein'),(1029073,'gnl|CDD|288491',_binary 'PD-(D/E)XK endonuclease.  This family of endonucleases includes a group I intron-encoded endonuclease. This family belongs to the PD-(D/E)XK superfamily.','pfam11645','PDDEXK_5'),(1030234,'gnl|CDD|239283',_binary 'TRX family, chloroplastic drought-induced stress protein of 32 kD (CDSP32); CDSP32 is composed of two TRX domains, a C-terminal TRX domain which contains a redox active CXXC motif and an N-terminal TRX-like domain which contains an SXXS sequence instead of the redox active motif. CDSP32 is a stress-inducible TRX, i.e., it acts as a TRX by reducing protein disulfides and is induced by environmental and oxidative stress conditions. It plays a critical role in plastid defense against oxidative damage, a role related to its function as a physiological electron donor to BAS1, a plastidic 2-cys peroxiredoxin. Plants lacking CDSP32 exhibit decreased photosystem II photochemical efficiencies and chlorophyll retention compared to WT controls, as well as an increased proportion of BAS1 in its overoxidized monomeric form.','cd02985','TRX_CDSP32'),(1030395,'gnl|CDD|177290',_binary 'CRISPR-associated Cas4-like protein.','PHA00619','PHA00619'),(1030889,'gnl|CDD|184491',_binary 'recombination protein F; Provisional.','PRK14079','recF'),(1030890,'gnl|CDD|129705',_binary 'exonuclease SbcC.  All proteins in this family for which functions are known are part of an exonuclease complex with sbcD homologs. This complex is involved in the initiation of recombination to regulate the levels of palindromic sequences in DNA. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00618','Nuclease_SbcCD_subunit_C'),(1030891,'gnl|CDD|273173',_binary 'recF protein.  All proteins in this family for which functions are known are DNA binding proteins that assist the filamentation of RecA onto DNA for the initiation of recombination or recombinational repair. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00611','DNA_replication_and_repai'),(1030971,'gnl|CDD|213217',_binary 'ATP-binding cassette domain 1 of multidrug resistance-associated protein, subfamily C.  This subfamily is also known as MRP (multidrug resistance-associated protein). Some of the MRP members have five additional transmembrane segments in their N-terminus, but the function of these additional membrane-spanning domains is not clear. The MRP was found in the multidrug-resisting lung cancer cell in which p-glycoprotein was not overexpressed. MRP exports glutathione by drug stimulation, as well as, certain substrates in conjugated forms with anions, such as glutathione, glucuronate, and sulfate.','cd03250','ABCC_MRP_domain1'),(1030972,'gnl|CDD|213195',_binary 'ATP-binding cassette domain of multidrug resistance protein-like transporters.  The MRP (Multidrug Resistance Protein)-like transporters are involved in drug, peptide, and lipid export. They belong to the subfamily C of the ATP-binding cassette (ABC) superfamily of transport proteins. The ABCC subfamily contains transporters with a diverse functional spectrum that includes ion transport, cell surface receptor, and toxin secretion activities. The MRP-like family, similar to all ABC proteins, have a common four-domain core structure constituted by two membrane-spanning domains, each composed of six transmembrane (TM) helices, and two nucleotide-binding domains (NBD). ABC transporters are a subset of nucleotide hydrolases that contain a signature motif, Q-loop, and H-loop/switch region, in addition to, the Walker A motif/P-loop and Walker B motif commonly found in a number of ATP- and GTP-binding and hydrolyzing proteins.','cd03228','ABCC_MRP_Like'),(1031207,'gnl|CDD|224391',_binary 'Cdc6-related protein, AAA superfamily ATPase [DNA replication, recombination, and repair / Posttranslational modification, protein turnover, chaperones].','COG1474','CDC6'),(1031209,'gnl|CDD|274354',_binary 'orc1/cdc6 family replication initiation protein.  Members of this protein family are found exclusively in the archaea. This set of DNA binding proteins shows homology to the origin recognition complex subunit 1/cell division control protein 6 family in eukaryotes. Several members may be found in genome and interact with each other. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02928','ORC1-type_DNA_replication'),(1031210,'gnl|CDD|234751',_binary 'cell division control protein 6; Reviewed.','PRK00411','cdc6'),(1031285,'gnl|CDD|198396',_binary 'Catalytic GIY-YIG domain of yeast structure-specific endonuclease subunit SLX1 and its homologs.  Structure-specific endonuclease subunit SLX1 is a highly conserved protein from yeast to human, with an N-terminal GIY-YIG endonuclease domain and a C-terminal PHD-type zinc finger postulated to mediate protein-protein or protein-DNA interaction. SLX1 forms active heterodimeric complexes with its SLX4 partner, which has additional roles in the DNA damage response that are distinct from the function of the heterodimeric SLX1-SLX4 nuclease. In yeast, the SLX1-SLX4 complex functions as a 5\' flap endonuclease that maintains ribosomal DNA copy number, where SLX1 and SLX4 are shown to be catalytic and regulatory subunits, respectively. This endonuclease introduces single-strand cuts in duplex DNA on the 3\' side of junctions with single-strand DNA. In addition to 5\' flap endonuclease activity, human SLX1-SLX4 complex has been identified as a Holliday junction resolvase that promotes symmetrical cleavage of static and migrating Holliday junctions. SLX1 also associates with MUS81, EME1, C20orf94, PLK1, and ERCC1. Some eukaryotic SLX1 homologs lack the zinc finger domain, but possess intrinsically unstructured extensions of unknown function. These unstructured segments might be involved in interactions with other proteins.','cd10449','GIY-YIG_SLX1_like'),(1031647,'gnl|CDD|307738',_binary 'Acyltransferase family.  This family includes a range of acyltransferase enzymes. This domain is found in many as yet uncharacterized C. elegans proteins and it is approximately 300 amino acids long.','pfam01757','Acyl_transf_3'),(1031653,'gnl|CDD|131662',_binary 'mobile mystery protein B.  Members of this protein family, which we designate mobile mystery protein B, are found in mobization-related contexts more often than not, including within a CRISPR-associated gene region in Geobacter sulfurreducens PCA, and on plasmids in Agrobacterium tumefaciens and Coxiella burnetii, always together with mobile mystery protein A (TIGR02612), a member of the family of helix-turn-helix DNA binding proteins (pfam01381). This protein is encoded by the downstream member of the gene pair and belongs to the Fic protein family (pfam02661), where Fic (filamentation induced by cAMP) is a regulator of cell division. The characteristics of having a two-gene operon in a varied context and often on plasmids, with one member affecting cell division and the other able to bind DNA, suggests similarity to addiction modules.','TIGR02613','cell_filamentation_protei'),(1031715,'gnl|CDD|274476',_binary 'DGQHR domain.  This highly divergent, uncharacterized domain has several absolutely conserved residues, including a QR pair and FxxxN motif. Its most striking feature, however, is a near invariant pentapeptide motif DGQHR. Several different subfamilies occur specifically as a part of DNA phosphorothioation systems, previously called DND (DNA instability during electrophoresis), while others (e.g. CPS_2936) occur in other contexts suggestive of lateral gene transfer (sporadic distribution of helicase-containing cassettes). The region described by this model is about 280 amino acids in length; additional sequences show local sequence similarity.','TIGR03187','hypothetical_protein'),(1032122,'gnl|CDD|223497',_binary 'DNA repair exonuclease [DNA replication, recombination, and repair].','COG0420','SbcD'),(1032167,'gnl|CDD|235815',_binary 'hypothetical protein; Provisional.','PRK06489','PRK06489'),(1033691,'gnl|CDD|130902',_binary 'type I secretion membrane fusion protein, HlyD family.  Type I secretion is an ABC transport process that exports proteins, without cleavage of any signal sequence, from the cytosol to extracellular medium across both inner and outer membranes. The secretion signal is found in the C-terminus of the transported protein. This model represents the adaptor protein between the ATP-binding cassette (ABC) protein of the inner membrane and the outer membrane protein, and is called the membrane fusion protein. This model selects a subfamily closely related to HlyD; it is defined narrowly and excludes, for example, colicin V secretion protein CvaA and multidrug efflux proteins. [Protein fate, Protein and peptide secretion and trafficking].','TIGR01843','Hemolysin_secretion_prote'),(1033763,'gnl|CDD|311666',_binary 'PGAP1-like protein.  The sequences found in this family are similar to PGAP1. This is an endoplasmic reticulum membrane protein with a catalytic serine containing motif that is conserved in a number of lipases. PGAP1 functions as a GPI inositol-deacylase; this deacylation is important for the efficient transport of GPI-anchored proteins from the endoplasmic reticulum to the Golgi body.','pfam07819','PGAP1'),(1033796,'gnl|CDD|235244',_binary 'cytidylate kinase; Provisional.','PRK04182','PRK04182'),(1033880,'gnl|CDD|183452',_binary '2-phosphoglycerate kinase; Provisional.','PRK12337','PRK12337'),(1033891,'gnl|CDD|223092',_binary 'Alanyl-tRNA synthetase [Translation, ribosomal structure and biogenesis].','COG0013','AlaS'),(1034902,'gnl|CDD|260012',_binary 'Eukaryotic RNase H is essential and is longer and more complex than their prokaryotic counterparts.  Ribonuclease H (RNase H) is classified into two families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is involved in DNA replication, repair and transcription. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. RNase H is widely present in various organisms, including bacteria, archaea and eukaryote and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD) residues and have the same catalytic mechanism and functions in cells. Eukaryotic RNase H is longer and more complex than in prokaryotes. Almost all eukaryotic RNase HI have highly conserved regions at their N-termini called hybrid binding domain (HBD). It is speculated that the HBD contributes to binding the RNA/DNA hybrid. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability, but RNase H is essential in higher eukaryotes. RNase H knockout mice lack mitochondrial DNA replication and die as embryos.','cd09280','RNase_HI_eukaryote_like'),(1034904,'gnl|CDD|260010',_binary 'RNase HI family found mainly in prokaryotes.  Ribonuclease H (RNase H) is classified into two evolutionarily unrelated families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is involved in DNA replication, repair and transcription. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD), residues and have the same catalytic mechanism and functions in cells. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. Prokaryotic RNase H varies greatly in domain structures and substrate specificities. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability.','cd09278','RNase_HI_prokaryote_like'),(1034906,'gnl|CDD|223405',_binary 'Ribonuclease HI [DNA replication, recombination, and repair].','COG0328','RnhA'),(1034908,'gnl|CDD|259998',_binary 'Ribonuclease H-like superfamily, including RNase H, HI, HII, HIII, and RNase-like domain IV of spliceosomal protein Prp8.  Ribonuclease H (RNase H) enzymes are divided into two major families, Type 1 and Type 2, based on amino acid sequence similarities and biochemical properties. RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner in the presence of divalent cations. It is widely present in various organisms, including bacteria, archaea, and eukaryotes. Most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site residues and have the same catalytic mechanism and functions in cells. RNase H is involved in DNA replication, repair and transcription. An important RNase H function is to remove Okazaki fragments during DNA replication. RNase H inhibitors have been explored as anti-HIV drug targets since RNase H inactivation inhibits reverse transcription. This model also includes the Prp8 domain IV, which adopts the RNase fold but shows low sequence homology; domain IV is implicated in key spliceosomal interactions.','cd06222','RNase_H_like'),(1034910,'gnl|CDD|306562',_binary 'RNase H.  RNase H digests the RNA strand of an RNA/DNA hybrid. Important enzyme in retroviral replication cycle, and often found as a domain associated with reverse transcriptases. Structure is a mixed alpha+beta fold with three a/b/a layers.','pfam00075','RNase_H'),(1034912,'gnl|CDD|260009',_binary 'Bacterial RNase HI containing a hybrid binding domain (HBD) at the N-terminus.  Ribonuclease H (RNase H) enzymes are divided into two major families, Type 1 and Type 2, based on amino acid sequence similarities and biochemical properties. RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner in the presence of divalent cations. RNase H is involved in DNA replication, repair and transcription. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, Type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD) residues and have the same catalytic mechanism and functions in cells. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. Prokaryotic RNase H varies greatly in domain structures and substrate specificities. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability. Some bacteria distinguished from other bacterial RNase HI in the presence of a hybrid binding domain (HBD) at the N-terminus which is commonly present at the N-termini of eukaryotic RNase HI. It has been reported that this domain is required for dimerization and processivity of RNase HI upon binding to RNA-DNA hybrids.','cd09277','RNase_HI_bacteria_like'),(1034978,'gnl|CDD|235611',_binary 'amidophosphoribosyltransferase; Provisional.','PRK05793','PRK05793'),(1034979,'gnl|CDD|215531',_binary 'glucosamine:fructose-6-phosphate aminotransferase.','PLN02981','PLN02981'),(1034980,'gnl|CDD|273676',_binary 'asparagine synthase (glutamine-hydrolyzing).  This model describes the glutamine-hydrolysing asparagine synthase. A poorly conserved C-terminal extension was removed from the model. Bacterial members of the family tend to have a long, poorly conserved insert lacking from archaeal and eukaryotic sequences. Multiple isozymes have been demonstrated, such as in Bacillus subtilis. Long-branch members of the phylogenetic tree (which typically were also second or third candidate members from their genomes) were removed from the seed alignment and score below trusted cutoff. [Amino acid biosynthesis, Aspartate family].','TIGR01536','Asparagine_synthetase_1'),(1035002,'gnl|CDD|238322',_binary 'Proliferating Cell Nuclear Antigen (PCNA) domain found in eukaryotes and archaea.  These polymerase processivity factors play a role in DNA replication and repair.  PCNA encircles duplex DNA in its central cavity, providing a DNA-bound platform for the attachment of the polymerase. The trimeric PCNA ring is structurally similar to the dimeric ring formed by the DNA polymerase processivity factors in bacteria (beta subunit DNA polymerase III holoenzyme) and in bacteriophages (catalytic subunits in T4 and RB69). This structural correspondence further substantiates the mechanistic connection between eukaryotic and prokaryotic DNA replication that has been suggested on biochemical grounds.   PCNA is also involved with proteins involved in cell cycle processes such as DNA repair and apoptosis. Many of these proteins contain a highly conserved motif known as the PIP-box (PCNA interacting protein box) which contains the sequence Qxx[LIM]xxF[FY]. .','cd00577','PCNA'),(1035004,'gnl|CDD|273158',_binary 'proliferating cell nuclear antigen (pcna).  All proteins in this family for which functions are known form sliding DNA clamps that are used in DNA replication processes. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00590','Proliferating_cell_nuclea'),(1035035,'gnl|CDD|273185',_binary 'excinuclease ABC, B subunit.  All proteins in this family for wich functions are known are DNA helicases that function in the nucleotide excision repair and are endonucleases that make the 3\' incision next to DNA damage. They are part of a pathway requiring UvrA, UvrB, UvrC, and UvrD homologs. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00631','UvrABC_system_protein_B'),(1036678,'gnl|CDD|280223',_binary 'Protein of unknown function DUF134.  This family of archaeal proteins has no known function.','pfam02001','DUF134'),(1036679,'gnl|CDD|180707',_binary 'RNA polymerase factor sigma-70; Validated.','PRK06811','PRK06811'),(1036748,'gnl|CDD|213992',_binary 'Polymerase and Histidinol Phosphatase domain of Ycdx like.  PHP Ycdx-like is a stand alone PHP domain similar to Ycdx E. coli protein with an unknown physiological role. The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. It has also been shown that the PHP domain functions in DNA repair. The PHP structures have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel. YcdX may be involved in swarming.','cd07437','PHP_HisPPase_Ycdx_like'),(1036751,'gnl|CDD|236429',_binary 'putative hydrolase; Validated.','PRK09248','PRK09248'),(1038031,'gnl|CDD|178452',_binary 'long-chain-fatty-acid-CoA ligase.','PLN02861','PLN02861'),(1038635,'gnl|CDD|307644',_binary 'I/LWEQ domain.  I/LWEQ domains bind to actin. It has been shown that the I/LWEQ domains from mouse talin and yeast Sla2p interact with F-actin. I/LWEQ domains can be placed into four major groups based on sequence similarity: (1) Metazoan talin; (2) Dictyostelium TalA/TalB and SLA110; (3) metazoan Hip1p and (4) yeast Sla2p. The domain has four conserved blocks, the name of the domain is derived from the initial conserved amino acid of each of the four blocks.','pfam01608','I_LWEQ'),(1038677,'gnl|CDD|129018',_binary 'PhnA Zinc-Ribbon.  This protein family includes an uncharacterised member designated phnA in Escherichia coli, part of a large operon associated with alkylphosphonate uptake and carbon-phosphorus bond cleavage. This protein is not related to the characterised phosphonoacetate hydrolase designated PhnA.','smart00782','PhnA_Zn_Ribbon'),(1039120,'gnl|CDD|307646',_binary 'Transposase.  Transposase proteins are necessary for efficient DNA transposition. Contains transposases for IS204, IS1001, IS1096 and IS1165.','pfam01610','DDE_Tnp_ISL3'),(1039122,'gnl|CDD|225995',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG3464','COG3464'),(1039124,'gnl|CDD|317134',_binary 'zinc-finger of transposase IS204/IS1001/IS1096/IS1165.','pfam14690','zf-ISL3'),(1039126,'gnl|CDD|311745',_binary 'Plasmid pRiA4b ORF-3-like protein.  Members of this family are similar to the protein product of ORF-3 found on plasmid pRiA4 in the bacterium Agrobacterium rhizogenes. This plasmid is responsible for tumorigenesis at wound sites of plants infected by this bacterium, but the ORF-3 product does not seem to be involved in the pathogenetic process. Other proteins found in this family are annotated as being putative TnpR resolvases, but no further evidence was found to back this. Moreover, another member of this family is described as a probable lexA repressor and in fact carries a LexA DNA binding domain (pfam01726), but no references were found to expand on this.','pfam07929','PRiA4_ORF3'),(1039211,'gnl|CDD|316965',_binary 'Helix-turn-helix domain.  This presumed domain is found at the C-terminus of a large number of helicase proteins.','pfam14493','HTH_40'),(1039212,'gnl|CDD|307963',_binary 'Transcriptional regulator.  This family is related to pfam001022 and other transcription regulation families (personal obs: Yeats C).','pfam02082','Rrf2'),(1039417,'gnl|CDD|314033',_binary 'Protein of unknown function (DUF2815).  This is a phage related family of proteins with unknown function.','pfam10991','DUF2815'),(1039442,'gnl|CDD|307079',_binary 'Ezrin/radixin/moesin family.  This family of proteins contain a band 4.1 domain (pfam00373), at their amino terminus. This family represents the rest of these proteins.','pfam00769','ERM'),(1040743,'gnl|CDD|239389',_binary 'The signal recognition particle (SRP) mediates the transport to or across the plasma membrane in bacteria and the endoplasmic reticulum in eukaryotes. SRP recognizes N-terminal sighnal sequences of newly synthesized polypeptides at the ribosome. The SRP-polypeptide complex is then targeted to the membrane by an interaction between SRP and its cognated receptor (SR). In mammals, SRP consists of six protein subunits and a 7SL RNA. One of these subunits is a 54 kd protein (SRP54), which is a GTP-binding protein that interacts with the signal sequence when it emerges from the ribosome. SRP54 is a multidomain protein that consists of an N-terminal domain, followed by a central G (GTPase) domain and a C-terminal M domain.','cd03115','SRP'),(1040751,'gnl|CDD|315506',_binary 'Helix-turn-helix domain.  This domain represents a DNA-binding Helix-turn-helix domain found in transcriptional regulatory proteins.','pfam12840','HTH_20'),(1040753,'gnl|CDD|282780',_binary 'HTH DNA binding domain.','pfam04967','HTH_10'),(1040754,'gnl|CDD|225947',_binary 'Predicted DNA binding protein [General function prediction only].','COG3413','COG3413'),(1041310,'gnl|CDD|316531',_binary 'Domain of unknown function (DUF4234).  This presumed integral membrane protein domain is functionally uncharacterized. This domain family is found in bacteria and archaea, and is approximately 70 amino acids in length.','pfam14018','DUF4234'),(1041537,'gnl|CDD|311546',_binary 'Domain of Unknown Function (DUF1599).','pfam07659','DUF1599'),(1042159,'gnl|CDD|182222',_binary 'putative transcriptional regulator; Provisional.','PRK10072','PRK10072'),(1042293,'gnl|CDD|185628',_binary '104 kDa microneme/rhoptry antigen; Provisional.','PTZ00449','PTZ00449'),(1042296,'gnl|CDD|223021',_binary 'large tegument protein UL36; Provisional.','PHA03247','PHA03247'),(1042896,'gnl|CDD|316691',_binary 'Domain of unknown function (DUF4314).  This presumed domain is functionally uncharacterized. This domain family is found in bacteria, and is typically between 56 and 93 amino acids in length.','pfam14192','DUF4314'),(1042919,'gnl|CDD|224706',_binary 'ATP-dependent DNA ligase [DNA replication, recombination, and repair].','COG1793','CDC9'),(1042920,'gnl|CDD|185711',_binary 'Adenylation domain of archaeal and bacterial LigB-like DNA ligases.  ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. ATP-dependent ligases are present in many organisms such as viruses, bacteriophages, eukarya, archaea and bacteria. Bacterial DNA ligases are divided into two broad classes: NAD-dependent and ATP-dependent. All bacterial species have a NAD-dependent DNA ligase (LigA). Some bacterial genomes contain multiple genes for DNA ligases that are predicted to use ATP as their cofactor, including Mycobacterium tuberculosis LigB, LigC, and LigD. This group is composed of archaeal DNA ligases and bacterial proteins similar to Mycobacterium tuberculosis LigB. Members of this group contain adenylation and C-terminal oligonucleotide/oligosaccharide binding (OB)-fold domains, comprising a catalytic core unit that is common to most members of the ATP-dependent DNA ligase family. The adenylation domain binds ATP and contains many of the active-site residues. The common catalytic core unit comprises six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases.','cd07901','Adenylation_DNA_ligase_Ar'),(1042921,'gnl|CDD|185704',_binary 'Adenylation domain of proteins similar to ATP-dependent polynucleotide ligases.  ATP-dependent polynucleotide ligases catalyze the phosphodiester bond formation of nicked nucleic acid substrates using ATP as a cofactor in a three step reaction mechanism. This family includes ATP-dependent DNA and RNA ligases. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. ATP-dependent DNA ligases have a highly modular architecture, consisting of a unique arrangement of two or more discrete domains, including a DNA-binding domain, an adenylation or nucleotidyltransferase (NTase) domain, and an oligonucleotide/oligosaccharide binding (OB)-fold domain. The adenylation domain binds ATP and contains many active site residues. Together with the C-terminal OB-fold domain, it comprises a catalytic core unit that is common to most members of the ATP-dependent DNA ligase family. The catalytic core contains six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases including eukaryotic GRP-dependent mRNA-capping enzymes. The catalytic core contains both the active site as well as many DNA-binding residues. The RNA circularization protein from archaea and bacteria contains the minimal catalytic unit, the adenylation domain, but does not contain an OB-fold domain. This family also includes the m3G-cap binding domain of snurportin, a nuclear import adaptor that binds m3G-capped spliceosomal U small nucleoproteins (snRNPs), but doesn\'t have enzymatic activity.','cd06846','Adenylation_DNA_ligase_li'),(1042922,'gnl|CDD|185708',_binary 'Adenylation domain of putative bacterial ATP-dependent DNA ligases.  Bacterial DNA ligases are divided into two broad classes: NAD-dependent and ATP-dependent. All bacterial species have a NAD-dependent DNA ligase (LigA). Some bacterial genomes contain multiple genes for DNA ligases that are predicted to use ATP as their cofactor, including Mycobacterium tuberculosis LigB, LigC, and LigD. This group is composed of predicted bacterial ATP-dependent DNA ligases. ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three-step reaction mechanism. The adenylation and C-terminal oligonucleotide/oligosaccharide binding (OB)-fold domains comprise a catalytic core unit that is common to most members of the ATP-dependent DNA ligase family, including this group. The adenylation domain binds ATP and contains many of the active site residues.','cd07897','Adenylation_DNA_ligase_Ba'),(1042923,'gnl|CDD|185715',_binary 'Adenylation domain of Mycobacterium tuberculosis LigD and LigC-like ATP-dependent DNA ligases.  Bacterial DNA ligases are divided into two broad classes: NAD-dependent and ATP-dependent. All bacterial species have a NAD-dependent DNA ligase (LigA). Some bacterial genomes contain multiple genes for DNA ligases that are predicted to use ATP as their cofactor, including Mycobacterium tuberculosis LigB, LigC, and LigD. This group is composed of ATP-dependent DNA ligases similar to Mycobacterium tuberculosis LigC. ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. Members of this group contain adenylation and C-terminal oligonucleotide/oligosaccharide binding (OB)-fold domains, comprising a catalytic core unit that is common to all members of the ATP-dependent DNA ligase family. The adenylation domain binds ATP and contains many of the active-site residues. The common catalytic core unit comprises six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases. LigD consists of a central ATP-dependent DNA ligase catalytic core unit fused to a C-terminal polymerase domain and an N-terminal 3\'-phosphoesterase (PE) module. LigD catalyzes the end-healing and end-sealing steps during non-homologous end joining.','cd07906','Adenylation_DNA_ligase_Li'),(1042924,'gnl|CDD|185713',_binary 'Adenylation domain of DNA Ligase IV.  ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. ATP-dependent ligases are present in many organisms such as viruses, bacteriophages, eukarya, archaea and bacteria. There are three classes of ATP-dependent DNA ligase in eukaryotic cells (I, III and IV). DNA ligase IV is required for DNA non-homologous end joining pathways, including recombination of the V(D)J immunoglobulin gene segments in cells of the mammalian immune system. DNA ligase IV is stabilized by forming a complex with XRCC4, a nuclear phosphoprotein, which is phosphorylated by DNA-dependent protein kinase. DNA ligases have a highly modular architecture consisting of a unique arrangement of two or more discrete domains. The adenylation and C-terminal oligonucleotide/oligosaccharide binding (OB)-fold domains comprise a catalytic core unit that is common to all members of the ATP-dependent DNA ligase family. The adenylation domain binds ATP and contains many of the active-site residues. The common catalytic unit comprises six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases.','cd07903','Adenylation_DNA_ligase_IV'),(1042925,'gnl|CDD|185714',_binary 'Adenylation domain of Mycobacterium tuberculosis LigC-like ATP-dependent DNA ligases.  Bacterial DNA ligases are divided into two broad classes: NAD-dependent and ATP-dependent. All bacterial species have a NAD-dependent DNA ligase (LigA). Some bacterial genomes contain multiple genes for DNA ligases that are predicted to use ATP as their cofactor, including Mycobacterium tuberculosis LigB, LigC, and LigD. This group is composed of ATP-dependent DNA ligases similar to Mycobacterium tuberculosis LigC. ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. Members of this group contain adenylation and C-terminal oligonucleotide/oligosaccharide binding (OB)-fold domains, comprising a catalytic core unit that is common to all members of the ATP-dependent DNA ligase family. The adenylation domain binds ATP and contains many of the active-site residues. The common catalytic core unit comprises six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases.','cd07905','Adenylation_DNA_ligase_Li'),(1042926,'gnl|CDD|236599',_binary 'ATP-dependent DNA ligase; Reviewed.','PRK09632','PRK09632'),(1042927,'gnl|CDD|279417',_binary 'ATP dependent DNA ligase domain.  This domain belongs to a more diverse superfamily, including pfam01331 and pfam01653.','pfam01068','DNA_ligase_A_M'),(1042928,'gnl|CDD|234900',_binary 'ATP-dependent DNA ligase; Provisional.','PRK01109','PRK01109'),(1042929,'gnl|CDD|185709',_binary 'Adenylation domain of ATP-dependent DNA Ligases.  ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation using nicked nucleic acid substrates with the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. DNA ligases play a vital role in the diverse processes of DNA replication, recombination and repair. ATP-dependent ligases are present in many organisms such as viruses, bacteriophages, eukarya, archaea and bacteria. Some organisms express a variety of different ligases which appear to be targeted to specific functions. ATP-dependent DNA ligases have a highly modular architecture consisting of a unique arrangement of two or more discrete domains including a DNA-binding domain, an adenylation (nucleotidyltransferase (NTase)) domain, and an oligonucleotide/oligosaccharide binding (OB)-fold domain. The adenylation domain binds ATP and contains many of the active-site residues. The adenylation and C-terminal OB-fold domains comprise a catalytic core unit that is common to most members of the ATP-dependent DNA ligase family. The catalytic core unit contains six conserved sequence motifs (I, III, IIIa, IV, V and VI) that define this family of related nucleotidyltransferases.','cd07898','Adenylation_DNA_ligase'),(1042930,'gnl|CDD|236428',_binary 'ATP-dependent DNA ligase; Validated.','PRK09247','PRK09247'),(1042931,'gnl|CDD|274993',_binary 'DNA ligase, ATP-dependent, PP_1105 family.  This model describes a family of ATP-dependent DNA ligases present in about 12 % of prokaryotic genomes. It occurs as part of a four-gene system with an exonuclease, a helicase and a phosphoesterase, with all four genes clustered or at least the first two and last two paired. This family resembles DNA ligase I (see TIGR00574 and pfam01068), and its presumed function may be in DNA repair, replication, or recombination.','TIGR04120','DNA_ligase'),(1042938,'gnl|CDD|181904',_binary 'ribosomal-protein-alanine N-acetyltransferase; Provisional.','PRK09491','rimI'),(1042947,'gnl|CDD|131112',_binary 'phosphoadenosine phosphosulfate reductase, thioredoxin dependent.  Requiring thioredoxin as an electron donor, phosphoadenosine phosphosulfate reductase catalyzes the reduction of 3\'-phosphoadenylylsulfate (PAPS) to sulfite and phospho-adenosine-phosphate (PAP). Found in enterobacteria, cyanobacteria, and yeast, PAPS reductase is related to a group of plant (TIGR00424) and bacterial (TIGR02055) enzymes preferring 5\'-adenylylsulfate (APS) over PAPS as a substrate for reduction to sulfite. [Central intermediary metabolism, Sulfur metabolism].','TIGR02057','Phosphoadenosine_phosphos'),(1042982,'gnl|CDD|213858',_binary 'integrating conjugative element, PFGI_1 class, ParB family protein.  Members of this protein family carry the ParB-type nuclease domain and are found in integrating conjugative elements (ICE) in the same class as PFGI-1 of Pseudomonas fluorescens Pf-5.','TIGR03764','ICE_PFGI_1_parB'),(1043294,'gnl|CDD|236798',_binary 'putative mechanosensitive channel protein; Provisional.','PRK10929','PRK10929'),(1043456,'gnl|CDD|226696',_binary 'Uncharacterized protein encoded in toxicity protection region of plasmid R478, contains von Willebrand factor (vWF) domain [General function prediction only].','COG4245','TerY'),(1043500,'gnl|CDD|274399',_binary 'nucleotide sugar dehydrogenase.  Enzymes in this family catalyze the NAD-dependent alcohol-to-acid oxidation of nucleotide-linked sugars. Examples include UDP-glucose 6-dehydrogenase (1.1.1.22), GDP-mannose 6-dehydrogenase (1.1.1.132), UDP-N-acetylglucosamine 6-dehydrogenase (1.1.1.136), UDP-N-acetyl-D-galactosaminuronic acid dehydrogenase, and UDP-N-acetyl-D-mannosaminuronic acid dehydrogenase. These enzymes are most often involved in the biosynthesis of polysaccharides and are often found in operons devoted to that purpose. All of these enzymes contain three Pfam domains, pfam03721, pfam00984, and pfam03720 for the N-terminal, central, and C-terminal regions respectively.','TIGR03026','NDP-sugDHase'),(1043501,'gnl|CDD|223936',_binary 'Predicted UDP-glucose 6-dehydrogenase [Cell envelope biogenesis, outer membrane].','COG1004','Ugd'),(1043502,'gnl|CDD|307223',_binary 'UDP-glucose/GDP-mannose dehydrogenase family, central domain.  The UDP-glucose/GDP-mannose dehydrogenaseses are a small group of enzymes which possesses the ability to catalyze the NAD-dependent 2-fold oxidation of an alcohol to an acid without the release of an aldehyde intermediate.','pfam00984','UDPG_MGDP_dh'),(1043503,'gnl|CDD|185017',_binary 'UDP-glucose 6-dehydrogenase; Provisional.','PRK15057','PRK15057'),(1043504,'gnl|CDD|177986',_binary 'probable UDP-glucose 6-dehydrogenase.','PLN02353','PLN02353'),(1045141,'gnl|CDD|238139',_binary 'Chloramphenicol (Cm) phosphotransferase (CPT). Cm-inactivating enzyme; modifies the primary (C-3) hydroxyl of the antibiotic. Related structurally to shikimate kinase II.','cd00227','CPT'),(1045179,'gnl|CDD|238861',_binary 'SEST_like. A family of secreted SGNH-hydrolases similar to Streptomyces scabies esterase (SEST), a causal agent of the potato scab disease, which hydrolyzes a specific ester bond in suberin, a plant lipid. The tertiary fold of this enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles two of the three components of typical Ser-His-Asp(Glu) triad from other serine hydrolases, but may lack the carboxylic acid.','cd01823','SEST_like'),(1045261,'gnl|CDD|179711',_binary 'hypothetical protein; Provisional.','PRK04017','PRK04017'),(1045481,'gnl|CDD|270838',_binary 'Catalytic domain of the Serine/Threonine Kinase, Cyclin-Dependent protein Kinase Like 5.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Mutations in the gene encoding CDKL5, previously called STK9, are associated with early onset epilepsy and severe mental retardation [X-linked infantile spasm syndrome (ISSX) or West syndrome]. In addition, CDKL5 mutations also sometimes cause a phenotype similar to Rett syndrome (RTT), a progressive neurodevelopmental disorder. These pathogenic mutations are located in the N-terminal portion of the protein within the kinase domain. CDKs belong to a large family of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. The CDKL5 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd07848','STKc_CDKL5'),(1046645,'gnl|CDD|239583',_binary 'The Delta6 Fatty Acid Desaturase (Delta6-FADS)-like CD includes the integral-membrane enzymes: delta-4, delta-5, delta-6, delta-8, delta-8-sphingolipid, and delta-11 desaturases found in vertebrates, higher plants, fungi, and bacteria. These desaturases are required for the synthesis of highly unsaturated fatty acids (HUFAs), which are mainly esterified into phospholipids and contribute to maintaining membrane fluidity. While HUFAs may be required for cold tolerance in bacteria, plants and fish, the primary role of HUFAs in mammals is cell signaling. These enzymes are described as front-end desaturases because they introduce a double bond between the pre-exiting double bond and the carboxyl (front) end of the fatty acid. Various substrates are involved, with both acyl-coenzyme A (CoA) and acyl-lipid desaturases present in this CD. Acyl-lipid desaturases are localized in the membranes of cyanobacterial thylakoid, plant endoplasmic reticulum (ER), and plastid; and acyl-CoA desaturases are present in ER membrane. ER-bound plant acyl-lipid desaturases and acyl-CoA desaturases require cytochrome b5 as an electron donor. Most of the eukaryotic desaturase domains have an adjacent N-terminal cytochrome b5-like domain. This domain family has extensive hydrophobic regions that would be capable of spanning the membrane bilayer at least twice. Comparison of sequences also reveals the existence of three regions of conserved histidine cluster motifs that contain the residues: HXXXH, HXX(X)HH, and Q/HXXHH. These histidine residues are reported to be catalytically essential and proposed to be the ligands for the iron atoms contained within the homolog, stearoyl CoA desaturase.','cd03506','Delta6-FADS-like'),(1046646,'gnl|CDD|178740',_binary 'delta6-acyl-lipid desaturase-like protein; Provisional.','PLN03199','PLN03199'),(1046647,'gnl|CDD|306890',_binary 'Fatty acid desaturase.','pfam00487','FA_desaturase'),(1046648,'gnl|CDD|178739',_binary 'delta6-acyl-lipid desaturase; Provisional.','PLN03198','PLN03198'),(1046649,'gnl|CDD|225779',_binary 'Fatty acid desaturase [Lipid metabolism].','COG3239','DesA'),(1046650,'gnl|CDD|238511',_binary 'The membrane fatty acid desaturase (Membrane_FADS)-like CD includes membrane FADSs, alkane hydroxylases, beta carotene ketolases (CrtW-like), hydroxylases (CrtR-like), and other related proteins. They are present in all groups of organisms with the exception of archaea. Membrane FADSs are non-heme, iron-containing, oxygen-dependent enzymes involved in regioselective introduction of double bonds in fatty acyl aliphatic chains. They play an important role in the maintenance of the proper structure and functioning of biological membranes. Alkane hydroxylases are bacterial, integral-membrane di-iron enzymes that share a requirement for iron and oxygen for activity similar to that of membrane FADSs, and are involved in the initial oxidation of inactivated alkanes. Beta-carotene ketolase and beta-carotene hydroxylase are carotenoid biosynthetic enzymes for astaxanthin and zeaxanthin, respectively. This superfamily domain has extensive hydrophobic regions that would be capable of spanning the membrane bilayer at least twice. Comparison of these sequences also reveals three regions of conserved histidine cluster motifs that contain eight histidine residues: HXXX(X)H, HXX(X)HH, and HXXHH (an additional conserved histidine residue is seen between clusters 2 and 3). Spectroscopic and genetic evidence point to a nitrogen-rich coordination environment located in the cytoplasm with as many as eight histidines coordinating the two iron ions and a carboxylate residue bridging the two metals in the Pseudomonas oleovorans alkane hydroxylase (AlkB). In addition, the eight histidine residues are reported to be catalytically essential and proposed to be the ligands for the iron atoms contained within the rat stearoyl CoA delta-9 desaturase.','cd01060','Membrane-FADS-like'),(1046652,'gnl|CDD|239588',_binary 'This CD includes the putative hydrocarbon oxygenase, MocD, a bacterial rhizopine (3-O-methyl-scyllo-inosamine, 3-O-MSI) oxygenase, and other related proteins. It has been proposed that MocD, MocE (Rieske-like ferredoxin), and MocF (ferredoxin reductase) under the regulation of MocR, act in concert to form a ferredoxin oxygenase system that demethylates 3-O-MSI to form scyllo-inosamine.  This domain family appears to be structurally related to the membrane fatty acid desaturases and the alkane hydroxylases. They all share in common extensive hydrophobic regions that would be capable of spanning the membrane bilayer at least twice. Comparison of sequences also reveals the existence of three regions of conserved histidine cluster motifs that contain eight histidine residues: HXXXH, HXXHH, and HXXHH. These histidine residues are reported to be catalytically essential and proposed to be the ligands for the iron atoms contained within homologs, stearoyl CoA desaturase and alkane hydroxylase.','cd03511','Rhizopine-oxygenase-like'),(1046653,'gnl|CDD|239587',_binary 'This CD includes the dihydrorhizobitoxine fatty acid desaturase (RtxC) characterized in Bradyrhizobium japonicum USDA110, and other related proteins. Dihydrorhizobitoxine desaturase is reported to be involved in the final step of rhizobitoxine biosynthesis. This domain family appears to be structurally related to the membrane fatty acid desaturases and the alkane hydroxylases. They all share in common extensive hydrophobic regions that would be capable of spanning the membrane bilayer at least twice. Comparison of sequences also reveals the existence of three regions of conserved histidine cluster motifs that contain eight histidine residues: HXXXH, HXX(X)HH, and HXXHH. These histidine residues are reported to be catalytically essential and proposed to be the ligands for the iron atoms contained within homologs, stearoyl CoA desaturase and alkane hydroxylase.','cd03510','Rhizobitoxine-FADS-like'),(1046654,'gnl|CDD|311798',_binary 'Bacterial PH domain.  This family contains many bacterial hypothetical proteins. The structures of Structure 3hsa and Structure 3dcx show similarities to the PH or pleckstrin homology domain. First evidence of PH-like domains in bacteria suggests role in cell envelope stress response.','pfam08000','bPH_1'),(1046655,'gnl|CDD|270045',_binary 'Pleckstrin homology (PH)-like domains in bacteria (PHb).  Pleckstrin homology (PH) domains were first identified in eukaryotic proteins. Recently PH-like domains have been identified in bacteria as well. These PHb form dome-shaped oligomeric rings with a conserved hydrophilic surface at the intersection of the beta-strands of adjacent protomers that likely mediates protein-protein interactions. It is now thought that the PH domain superfamily is more widespread than previous thought and appears to have existed before prokaryotes and eukaryotes diverged. PH domains have diverse functions, but in general are involved in targeting proteins to the appropriate cellular location or in the interaction with a binding partner. They share little sequence conservation, but all have a common fold, which is electrostatically polarized. Less than 10% of PH domains bind phosphoinositide phosphates (PIPs) with high affinity and specificity. PH domains are distinguished from other PIP-binding domains by their specific high-affinity binding to PIPs with two vicinal phosphate groups: PtdIns(3,4)P2, PtdIns(4,5)P2 or PtdIns(3,4,5)P3 which results in targeting some PH domain proteins to the plasma membrane. A few display strong specificity in lipid binding. Any specificity is usually determined by loop regions or insertions in the N-terminus of the domain, which are not conserved across all PH domains. PH domains are found in cellular signaling proteins such as serine/threonine kinase, tyrosine kinases, regulators of G-proteins, endocytotic GTPases, adaptors, as well as cytoskeletal associated molecules and in lipid associated enzymes.','cd13225','PH-like_bacteria'),(1046656,'gnl|CDD|234921',_binary 'phosphoribosylformylglycinamidine synthase II; Provisional.','PRK01213','PRK01213'),(1046657,'gnl|CDD|223124',_binary 'Phosphoribosylformylglycinamidine (FGAM) synthase, synthetase domain [Nucleotide transport and metabolism].','COG0046','PurL'),(1046658,'gnl|CDD|273781',_binary 'phosphoribosylformylglycinamidine synthase II.  Phosphoribosylformylglycinamidine synthase is a single, long polypeptide in most Proteobacteria and eukarotes. Three proteins are required in Bacillus subtilis and many other species. This is the longest of the three and is designated PurL, phosphoribosylformylglycinamidine synthase II, or FGAM synthase II. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01736','FGAM_synth_II'),(1046659,'gnl|CDD|100034',_binary 'PurL subunit of the formylglycinamide ribonucleotide amidotransferase (FGAR-AT), first repeat. FGAR-AT catalyzes the ATP-dependent conversion of formylglycinamide ribonucleotide (FGAR) and glutamine to formylglycinamidine ribonucleotide (FGAM), ADP, phosphate, and glutamate in the fourth step of the purine biosynthetic pathway. In eukaryotes and Gram-negative bacteria, FGAR-AT is encoded by the purL gene as a multidomain protein with a molecular mass of about 140 kDa. In Gram-positive bacteria and archaea FGAR-AT is a complex of three proteins: PurS, PurL, and PurQ. PurL itself contains two tandem N- and C-terminal domains (four domains altogether). The N-terminal domains bind ATP and are related to the ATP-binding domains of HypE, ThiL, SelD and PurM.','cd02203','PurL_repeat1'),(1046661,'gnl|CDD|184499',_binary 'phosphoribosylformylglycinamidine synthase II; Provisional.','PRK14090','PRK14090'),(1046663,'gnl|CDD|100035',_binary 'PurL subunit of the formylglycinamide ribonucleotide amidotransferase (FGAR-AT), second repeat. FGAR-AT catalyzes the ATP-dependent conversion of formylglycinamide ribonucleotide (FGAR) and glutamine to formylglycinamidine ribonucleotide (FGAM), ADP, phosphate, and glutamate in the fourth step of the purine biosynthetic pathway. In eukaryotes and Gram-negative bacteria, FGAR-AT is encoded by the purL gene as a multidomain protein with a molecular mass of about 140 kDa. In Gram-positive bacteria and archaea FGAR-AT is a complex of three proteins: PurS, PurL, and PurQ. PurL itself contains two tandem N- and C-terminal domains (four domains altogether). The N-terminal domains bind ATP and are related to the ATP-binding domains of HypE, ThiL, SelD and PurM.','cd02204','PurL_repeat2'),(1046665,'gnl|CDD|100029',_binary 'Formylglycinamide ribonucleotide amidotransferase (FGAR-AT) catalyzes the ATP-dependent conversion of formylglycinamide ribonucleotide (FGAR) and glutamine to formylglycinamidine ribonucleotide (FGAM), ADP, phosphate, and glutamate in the fourth step of the purine biosynthetic pathway. In eukaryotes and Gram-negative bacteria, FGAR-AT is encoded by the purL gene as a multidomain protein with a molecular mass of about 140 kDa. In Gram-positive bacteria and archaea FGAR-AT is a complex of three proteins: PurS, PurL, and PurQ. PurL itself contains two tandem N- and C-terminal domains (four domains altogether).  The N-terminal domains bind ATP and are related to the ATP-binding domains of HypE, ThiL, SelD and PurM.','cd02193','PurL'),(1046667,'gnl|CDD|130916',_binary 'phosphoribosylformylglycinamidine synthase, clade II.  This model represents a single-molecule form of phosphoribosylformylglycinamidine synthase, also called FGAM synthase, an enzyme of purine de novo biosynthesis. This model represents a second clade of these enzymes found in Clostridia, Bifidobacteria and Streptococcus species. This enzyme performs the fourth step in IMP biosynthesis (the precursor of all purines) from PRPP. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01857','FGAM-synthase'),(1046668,'gnl|CDD|188163',_binary 'phosphoribosylformylglycinamidine synthase, single chain form.  This model represents a single-molecule form of phosphoribosylformylglycinamidine synthase, also called FGAM synthase, an enzyme of purine de novo biosynthesis. This form is found mostly in eukaryotes and Proteobacteria. In Bacillus subtilis PurL (FGAM synthase II) and PurQ (FGAM synthase I), homologous to different parts of this model, perform the equivalent function; the unrelated small protein PurS is also required and may be a third subunit. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01735','Phosphoribosylformylglyci'),(1046669,'gnl|CDD|308418',_binary 'AIR synthase related protein, C-terminal domain.  This family includes Hydrogen expression/formation protein HypE, AIR synthases EC:6.3.3.1, FGAM synthase EC:6.3.5.3 and selenide, water dikinase EC:2.7.9.3. The function of the C-terminal domain of AIR synthase is unclear, but the cleft formed between N and C domains is postulated as a sulphate binding site.','pfam02769','AIRS_C'),(1046671,'gnl|CDD|100027',_binary 'AIR (aminoimidazole ribonucleotide) synthase related protein. This family includes Hydrogen expression/formation protein HypE, AIR synthases, FGAM (formylglycinamidine ribonucleotide) synthase and Selenophosphate synthetase (SelD). The N-terminal domain of AIR synthase forms the dimer interface of the protein, and is suggested as a putative ATP binding domain.','cd00396','PurM-like'),(1046673,'gnl|CDD|306956',_binary 'AIR synthase related protein, N-terminal domain.  This family includes Hydrogen expression/formation protein HypE, AIR synthases EC:6.3.3.1, FGAM synthase EC:6.3.5.3 and selenide, water dikinase EC:2.7.9.3. The N-terminal domain of AIR synthase forms the dimer interface of the protein, and is suggested as a putative ATP binding domain.','pfam00586','AIRS'),(1046675,'gnl|CDD|235394',_binary 'phosphoribosylformylglycinamidine synthase; Provisional.','PRK05297','PRK05297'),(1046678,'gnl|CDD|178745',_binary 'phosphoribosylformylglycinamidine synthase; Provisional.','PLN03206','PLN03206'),(1046680,'gnl|CDD|223058',_binary 'FGAM-synthase; Provisional.','PHA03366','PHA03366'),(1046681,'gnl|CDD|100037',_binary 'AIR synthase (PurM) related protein, subgroup 1 of unknown function. The family of PurM related proteins includes Hydrogen expression/formation protein HypE, AIR synthases, FGAM synthase and Selenophosphate synthetase (SelD). They all contain two conserved domains and seem to dimerize. The N-terminal domain forms the dimer interface and is a putative ATP binding domain.','cd06061','PurM-like1'),(1046683,'gnl|CDD|223386',_binary 'Hydrogenase maturation factor [Posttranslational modification, protein turnover, chaperones].','COG0309','HypE'),(1046684,'gnl|CDD|273784',_binary 'herpesvirus tegument protein/v-FGAM-synthase.  This model describes a family of large proteins of herpesvirues. The protein is described variably as tegument protein or phosphoribosylformylglycinamidine synthase (FGAM-synthase). Most of the length of the protein shows homology to eukaryotic FGAM-synthase. Functional characterizations were not verified during construction of this model.','TIGR01739','Probable_membrane_antigen'),(1046685,'gnl|CDD|132311',_binary 'putative methanogenesis marker protein 2.  A single member of this protein family is found in each of the first ten complete genome sequences of archaeal methanogens, and nowhere else. Sequence similarity to various bacterial proteins is reflected in Pfam models pfam00586 and pfam02769, AIR synthase related protein N-terminal and C-terminal domains, respectively. The functions of proteins in this family are unknown, but their role is likely one essential to methanogenesis. [Energy metabolism, Methanogenesis].','TIGR03267','methan_mark_2'),(1046686,'gnl|CDD|225055',_binary 'Selenophosphate synthetase-related proteins [General function prediction only].','COG2144','COG2144'),(1046687,'gnl|CDD|100036',_binary 'AIR synthase (PurM) related protein, archaeal subgroup 2 of unknown function. The family of PurM related proteins includes Hydrogen expression/formation protein HypE, AIR synthases, FGAM synthase and Selenophosphate synthetase (SelD). They all contain two conserved domains and seem to dimerize. The N-terminal domain forms the dimer interface and is a putative ATP binding domain.','cd02691','PurM-like2'),(1046688,'gnl|CDD|188564',_binary 'AIR synthase-related protein, sll0787 family.  Members of this family include sll0787 from Synechocystis sp. PCC 6803 and resemble the C-terminal region of MSMEG_0567 from Mycobacterium smegmatis, where the N-terminal is a GNAT family N-acetyltransferase. The conserved cluster is found broadly (Cyanobacteria, Proteobacteria, Actinobacteria) in about 8 percent of genomes and appears to be biosynthetic. The product is unkown. [Unknown function, Enzymes of unknown specificity].','TIGR04049','AIR_rel_sll0787'),(1047012,'gnl|CDD|272869',_binary 'phosphoesterase, MJ0936 family.  Members of this largely uncharacterized family share a motif approximating DXH(X25)GDXXD(X25)GNHD as found in several phosphoesterases, including the nucleases SbcD and Mre11, and a family of uncharacterized archaeal putative phosphoesterases described by TIGR00024. In this family, the His residue in GNHD portion of the motif is not conserved. The member MJ0936, one of two from Methanococcus jannaschii, was shown () to act on model phosphodiesterase substrates; a divalent cation was required. [Unknown function, Enzymes of unknown specificity].','TIGR00040','Putative_metallophosphoes'),(1047119,'gnl|CDD|143628',_binary 'Adenylyl cyclase (AC) class IV-like, a subgroup of the CYTH-like superfamily.  This subgroup contains class IV ACs and similar proteins. AC catalyzes the conversion of ATP to 3\',5\'-cyclic AMP (cAMP) and PPi. cAMP is a key signaling molecule which conveys a variety of signals in different cell types. In prokaryotes, cAMP is a catabolite derepression signal which triggers the expression of metabolic pathways including the lactose operon. Six non-homologous classes of ACs have been identified (I-VI). Class IV ACs are found in this group. In bacteria, the gene encoding Class IV AC has been designated cyaB and the protein as AC2. AC-IV occurs in addition to AC-I in bacterial pathogens such as Yersinia pestis (plague disease). The role of AC-IV is unknown but it has been speculated that it may be a factor in pathogenesis, perhaps providing cAMP for a secondary internal signaling function, or for secretion and uptake into host cells, where it may disrupt normal cellular processes. This subgroup belongs to the CYTH/triphosphate tunnel metalloenzyme (TTM)-like superfamily, whose enzymes have a unique active site located within an eight-stranded beta barrel.','cd07890','CYTH-like_AC_IV-like'),(1047120,'gnl|CDD|224354',_binary 'Adenylate cyclase, class 2 (thermophilic) [Nucleotide transport and metabolism].','COG1437','CyaB'),(1047121,'gnl|CDD|307858',_binary 'CYTH domain.  These sequences are functionally identified as members of the adenylate cyclase family, which catalyzes the conversion of ATP to 3\',5\'-cyclic AMP and pyrophosphate. Six distinct non-homologous classes of AC have been identified. The structure of three classes of adenylyl cyclases have been solved.','pfam01928','CYTH'),(1047122,'gnl|CDD|273010',_binary 'adenylyl cyclase CyaB, putative.  The protein CyaB from Aeromonas hydrophila is a second adenylyl cyclase from that species, as demonstrated by complementation in E. coli and by assay of the enzymatic properties of purified recombinant protein. It has no detectable homology to any other protein of known function, and has several unusual properties, including an optimal temperature of 65 degrees and an optimal pH of 9.5. A cluster of uncharaterized archaeal homologs may be orthologous and serve (under certain circumstances) to produce the regulatory metabolite cyclic AMP (cAMP). [Regulatory functions, Small molecule interactions].','TIGR00318','Putative_adenylate_cyclas'),(1047124,'gnl|CDD|284810',_binary 'Phage DNA packaging protein Nu1.  Terminase, the DNA packaging enzyme of bacteriophage lambda, is a heteromultimer composed of subunits Nu1 and A. The smaller Nu1 terminase subunit has a low-affinity ATPase stimulated by non-specific DNA.','pfam07471','Phage_Nu1'),(1047397,'gnl|CDD|320683',_binary 'Oma1 peptidase-like, integral membrane metallopeptidase.  This family contains peptidase M48 subfamily C (also known as Oma1 peptidase or mitochondrial metalloendopeptidase OMA1), including similar peptidases containing tetratricopeptide (TPR) repeats, as well as uncharacterized proteins such as E. coli bepA (formerly yfgC), ycaL and loiP (formerly yggG), considered to be putative metallopeptidases. Oma1 peptidase is part of the quality control system in the inner membrane of mitochondria, with its catalytic site facing the matrix space. It cleaves and thereby promotes the turnover of mistranslated or misfolded membrane proteins. Oma1 can cleave the misfolded multi-pass membrane protein Oxa1, thus exerting a function similar to the ATP-dependent m-AAA protease for quality control of inner membrane proteins. It has been proposed that in the absence of m-AAA protease, proteolysis of Oxa1 is mediated by Oma1 in an ATP-independent manner. Homologs of Oma1 are present in higher eukaryotes, eubacteria and archaebacteria, suggesting that Oma1 is the founding member of a conserved family of membrane-embedded metallopeptidases, all containing the zinc metalloprotease motif (HEXXH). M48 peptidases proteolytically remove the C-terminal three residues of farnesylated proteins.','cd07324','M48C_Oma1-like'),(1047398,'gnl|CDD|320692',_binary 'Peptidase M48C Ste24p bepA-like, integral membrane protein.  This family contains peptidase M48C Ste24p protease bepA (formerly yfgC)-like proteins considered to be putative metallopeptidases, containing a zinc-binding motif, HEXXH, and a COOH-terminal ER retrieval signal (KKXX). They proteolytically remove the C-terminal three residues of farnesylated proteins. They are integral membrane proteins associated with the endoplasmic reticulum and golgi, binding one zinc ion per subunit.  In eukaryotes, Ste24p is required for the first NH2-terminal proteolytic processing event within the a-factor precursor, which takes place after COOH-terminal CAAX modification (C is cysteine; A is usually aliphatic; X is one of several amino acids) is complete. Mutation studies have shown that the HEXXH protease motif, which is extracellular but adjacent to a transmembrane domain and therefore close to the membrane surface, is critical for Ste24p activity.  Several members of this family also contain tetratricopeptide (TPR) repeat motifs, which are involved in a variety of functions including protein-protein interactions. BepA has been shown to possess protease activity and is responsible for the degradation of incorrectly folded LptD, an essential outer-membrane protein (OMP) involved in OM transport and assembly of lipopolysaccharide. Overexpression of the bepA protease causes abnormal biofilm architecture.','cd07333','M48C_bepA_like'),(1047400,'gnl|CDD|320704',_binary 'Peptidase M48 subfamily A-like, putative CaaX prenyl protease.  This family contains peptidase family M48 subfamily A-like CaaX prenyl protease 1, most of which are uncharacterized. Some of these contain tetratricopeptide (TPR) repeats at the C-terminus. Proteins in this family contain the zinc metalloprotease motif (HEXXH), likely exposed on the cytoplasmic side. They are thought to be possibly associated with the endoplasmic reticulum (ER), regardless of whether their genes possess the conventional signal motif (KKXX) in the C-terminal. These proteins putatively remove the C-terminal three residues of farnesylated proteins proteolytically.','cd07345','M48A_Ste24p-like'),(1047402,'gnl|CDD|239217',_binary 'Nudix hydrolase is a superfamily of enzymes found in all three kingdoms of life, and it catalyzes the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+ for their activity. Members of this family are recognized by a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which forms a structural motif that functions as a metal binding and catalytic site. Substrates of nudix hydrolase include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance and \'house-cleaning\' enzymes. Substrate specificity is used to define child families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required. This superfamily consists of at least nine families: IPP (isopentenyl diphosphate) isomerase, ADP ribose pyrophosphatase, mutT pyrophosphohydrolase, coenzyme-A pyrophosphatase, MTH1-7,8-dihydro-8-oxoguanine-triphosphatase, diadenosine tetraphosphate hydrolase, NADH pyrophosphatase, GDP-mannose hydrolase and the c-terminal portion of the mutY adenine glycosylase.','cd02883','Nudix_Hydrolase'),(1047403,'gnl|CDD|306742',_binary 'NUDIX domain.','pfam00293','NUDIX'),(1047404,'gnl|CDD|239517',_binary 'The MutT pyrophosphohydrolase is a prototypical Nudix hydrolase that catalyzes the hydrolysis of nucleoside and deoxynucleoside triphosphates (NTPs and dNTPs) by substitution at a beta-phosphorus to yield a nucleotide monophosphate (NMP) and inorganic pyrophosphate (PPi). This enzyme requires two divalent cations for activity; one coordinates the phosphoryl groups of the NTP/dNTP substrate, and the other coordinates to the enzyme. It also contains the Nudix motif, a highly conserved 23-residue block (GX5EX7REUXEEXGU, where U = I, L or V), that functions as metal binding and catalytic site. MutT pyrophosphohydrolase is important in preventing errors in DNA replication by hydrolyzing mutagenic nucleotides such as 8-oxo-dGTP (a product of oxidative damage), which can mispair with template adenine during DNA replication, to guanine nucleotides.','cd03425','MutT_pyrophosphohydrolase'),(1047405,'gnl|CDD|240030',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04673','Nudix_Hydrolase_15'),(1047407,'gnl|CDD|239519',_binary 'MutT homolog-1 (MTH1) is a member of the Nudix hydrolase superfamily. MTH1, the mammalian counterpart of MutT, hydrolyzes oxidized purine nucleoside triphosphates, such as 8-oxo-dGTP and 2-hydroxy-ATP, to monophosphates, thereby preventing the incorporation of such oxygen radicals during replication. This is an important step in the repair mechanism in genomic and mitochondrial DNA.  Like other members of the Nudix family, it requires a divalent cation, such as Mg2+ or Mn2+, for activity, and contain the Nudix motif, a highly conserved 23-residue block (GX5EX7REUXEEXGU, where U = I, L or V), that functions as a metal binding and catalytic site. MTH1 is predominantly localized in the cytoplasm and mitochondria. Structurally, this enzyme adopts a similar fold to MutT despite low sequence similarity outside the conserved nudix motif. The most distinctive structural difference between MutT and MTH1 is the presence of a beta-hairpin, which is absent in MutT. This results in a much deeper and narrower substrate binding pocket. Mechanistically, MTH1 contains dual specificity for nucleotides that contain 2-OH-adenine bases and those that contain 8-oxo-guanine bases.','cd03427','MTH1'),(1047409,'gnl|CDD|223979',_binary 'ADP-ribose pyrophosphatase [Nucleotide transport and metabolism].','COG1051','COG1051'),(1047411,'gnl|CDD|223568',_binary 'NTP pyrophosphohydrolases including oxidative damage repair enzymes [DNA replication, recombination, and repair / General function prediction only].','COG0494','MutT'),(1047413,'gnl|CDD|240054',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04699','Nudix_Hydrolase_39'),(1047415,'gnl|CDD|240038',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04682','Nudix_Hydrolase_23'),(1047416,'gnl|CDD|234608',_binary 'recombination protein F; Reviewed.','PRK00064','recF'),(1047417,'gnl|CDD|240036',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04680','Nudix_Hydrolase_21'),(1047419,'gnl|CDD|240040',_binary 'Contains a crystal structure of the Nudix hydrolase from Enterococcus faecalis, which has an unknown function. In general, members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity. They also contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which forms a structural motif that functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04684','Nudix_Hydrolase_25'),(1047420,'gnl|CDD|239520',_binary 'Diadenosine tetraphosphate (Ap4A) hydrolase is a member of the Nudix hydrolase superfamily. Ap4A hydrolases are well represented in a variety of prokaryotic and eukaryotic organisms. Phylogenetic analysis reveals two distinct subgroups where plant enzymes fall into one subfamily and fungi/animals/archaea enzymes, represented by this subfamily, fall into another. Bacterial enzymes are found in both subfamilies. Ap4A is a potential by-product of aminoacyl tRNA synthesis, and accumulation of Ap4A has been implicated in a range of biological events, such as DNA replication, cellular differentiation, heat shock, metabolic stress, and apoptosis. Ap4A hydrolase cleaves Ap4A asymmetrically into ATP and AMP. It is important in the invasive properties of bacteria and thus presents a potential target for inhibition of such invasive bacteria. Besides the signature nudix motif (G[X5]E[X7]REUXEEXGU, where U is Ile, Leu, or Val) that functions as a metal binding and catalytic site, and a required divalent cation, Ap4A hydrolase is structurally similar to the other members of the nudix superfamily with some degree of variation. Several regions in the sequences are poorly defined and substrate and metal binding sites are only predicted based on kinetic studies.','cd03428','Ap4A_hydrolase_human_like'),(1047422,'gnl|CDD|239646',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity. They also contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, U=I, L or V), which forms a structural motif that functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd03674','Nudix_Hydrolase_1'),(1047423,'gnl|CDD|236361',_binary 'hypothetical protein; Provisional.','PRK08999','PRK08999'),(1047424,'gnl|CDD|182721',_binary 'nucleoside triphosphate pyrophosphohydrolase; Provisional.','PRK10776','PRK10776'),(1047425,'gnl|CDD|240032',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04676','Nudix_Hydrolase_17'),(1047427,'gnl|CDD|240033',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04677','Nudix_Hydrolase_18'),(1047428,'gnl|CDD|239643',_binary 'Diadenosine tetraphosphate (Ap4A) hydrolase is a member of the Nudix hydrolase superfamily. Members of this family are well represented in a variety of prokaryotic and eukaryotic organisms. Phylogenetic analysis reveals two distinct subgroups where plant enzymes fall into one group (represented by this subfamily) and fungi/animals/archaea enzymes fall into another. Bacterial enzymes are found in both subfamilies. Ap4A is a potential by-product of aminoacyl tRNA synthesis, and accumulation of Ap4A has been implicated in a range of biological events, such as DNA replication, cellular differentiation, heat shock, metabolic stress, and apoptosis. Ap4A hydrolase cleaves Ap4A asymmetrically into ATP and AMP. It is important in the invasive properties of bacteria and thus presents a potential target for the inhibition of such invasive bacteria. Besides the signature nudix motif (G[X5]E[X7]REUXEEXGU where U is Ile, Leu, or Val), Ap4A hydrolase is structurally similar to the other members of the nudix superfamily with some degree of variations. Several regions in the sequences are poorly defined and substrate and metal binding sites are only predicted based on kinetic studies.','cd03671','Ap4A_hydrolase_plant_like'),(1047429,'gnl|CDD|240025',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04667','Nudix_Hydrolase_10'),(1047430,'gnl|CDD|239647',_binary 'Contains a crystal structure of the Nudix hydrolase from Nitrosomonas europaea, which has an unknown function. In general, members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity. They also contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which forms a structural motif that functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd03675','Nudix_Hydrolase_2'),(1047431,'gnl|CDD|240023',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04665','Nudix_Hydrolase_8'),(1047432,'gnl|CDD|240026',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04669','Nudix_Hydrolase_11'),(1047433,'gnl|CDD|131752',_binary 'nucleoside triphosphatase YtkD.  The functional assignment to the proteins of this family is contentious, with papers disagreeing in both interpretation and enzyme assay results. This protein belongs to the nudix family and shares some sequence identity with E. coli MutT but appears not to be functionally interchangeable with it. [DNA metabolism, DNA replication, recombination, and repair].','TIGR02705','conserved_hypothetical_pr'),(1047434,'gnl|CDD|182536',_binary 'pyrimidine (deoxy)nucleoside triphosphate pyrophosphohydrolase; Provisional.','PRK10546','PRK10546'),(1047435,'gnl|CDD|240039',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04683','Nudix_Hydrolase_24'),(1047436,'gnl|CDD|240041',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily requires a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04685','Nudix_Hydrolase_26'),(1047437,'gnl|CDD|239516',_binary 'ADP-ribose pyrophosphatase (ADPRase) catalyzes the hydrolysis of ADP-ribose and a variety of additional ADP-sugar conjugates to AMP and ribose-5-phosphate. Like other members of the Nudix hydrolase superfamily, it requires a divalent cation, such as Mg2+, for its activity. It also contains a highly conserved 23-residue Nudix motif (GX5EX7REUXEEXGU, where U = I, L or V) which functions as a metal binding site/catalytic site. In addition to the Nudix motif, there are additional conserved amino acid residues, distal from the signature sequence, that correlate with substrate specificity. In humans, there are four distinct ADPRase activities, three putative cytosolic enzymes (ADPRase-I, -II, and -Mn) and a single mitochondrial enzyme (ADPRase-m). Human ADPRase-II is also referred to as NUDT5. It lacks the N-terminal target sequence unique to mitochondrial ADPRase. The different cytosolic types are distinguished by their specificities for substrate and specific requirement for metal ions. NUDT5 forms a homodimer.','cd03424','ADPRase_NUDT5'),(1047438,'gnl|CDD|240055',_binary 'DR1025 from Deinococcus radiodurans, a member of the Nudix hydrolase superfamily, show nucleoside triphosphatase and dinucleoside polyphosphate pyrophosphatase activities. Like other enzymes belonging to this superfamily, it requires a divalent cation, in this case Mg2+, for its activity. It also contains a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. In general, substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04700','DR1025_like'),(1047439,'gnl|CDD|240027',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04670','Nudix_Hydrolase_12'),(1047441,'gnl|CDD|240037',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04681','Nudix_Hydrolase_22'),(1047442,'gnl|CDD|240034',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04678','Nudix_Hydrolase_19'),(1047445,'gnl|CDD|240035',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04679','Nudix_Hydrolase_20'),(1047447,'gnl|CDD|240028',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04671','Nudix_Hydrolase_13'),(1047449,'gnl|CDD|239645',_binary 'Diadenosine hexaphosphate (Ap6A) hydrolase is a member of the Nudix hydrolase superfamily. Ap6A hydrolase specifically hydrolyzes diadenosine polyphosphates, but not ATP or diadenosine triphosphate, and it generates ATP as the product. Ap6A, the most preferred substrate, hydrolyzes to produce two ATP molecules, which is a novel hydrolysis mode for Ap6A. These results indicate that Ap6A  hydrolase is a diadenosine polyphosphate hydrolase. It requires the presence of a divalent cation, such as Mn2+, Mg2+, Zn2+, and Co2+, for activity. Members of the Nudix superfamily are recognized by a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which forms a structural motif that functions as a metal binding and catalytic site.','cd03673','Ap6A_hydrolase'),(1047451,'gnl|CDD|240022',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04664','Nudix_Hydrolase_7'),(1047453,'gnl|CDD|240031',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04674','Nudix_Hydrolase_16'),(1047455,'gnl|CDD|240021',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belong to this superfamily requires a divalent cation, such as Mg2+ or Mn2+ for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, U=I, L or V) which functions as metal binding and catalytic site. Substrates of nudix hydrolase include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04663','Nudix_Hydrolase_6'),(1047457,'gnl|CDD|200031',_binary 'mutator mutT protein.  All proteins in this family for which functions are known are involved in repairing oxidative damage to dGTP (they are 8-oxo-dGTPases). This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). Lowering the threshold picks up members of MutT superfamily well. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00586','8-oxo-dGTP_diphosphatase'),(1047459,'gnl|CDD|235436',_binary 'bifunctional nicotinamide mononucleotide adenylyltransferase/ADP-ribose pyrophosphatase; Provisional.','PRK05379','PRK05379'),(1047461,'gnl|CDD|240052',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04696','Nudix_Hydrolase_37'),(1047465,'gnl|CDD|240024',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04666','Nudix_Hydrolase_9'),(1047467,'gnl|CDD|234820',_binary 'RNA pyrophosphohydrolase; Reviewed.','PRK00714','PRK00714'),(1047469,'gnl|CDD|317251',_binary 'NUDIX domain.','pfam14815','NUDIX_4'),(1047471,'gnl|CDD|236516',_binary 'dihydroneopterin triphosphate pyrophosphatase; Provisional.','PRK09438','nudB'),(1047473,'gnl|CDD|240046',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04690','Nudix_Hydrolase_31'),(1047475,'gnl|CDD|240029',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04672','Nudix_Hydrolase_14'),(1047477,'gnl|CDD|239518',_binary 'Coenzyme A pyrophosphatase (CoAse), a member of the Nudix hydrolase superfamily, functions to catalyze the elimination of oxidized inactive CoA, which can inhibit CoA-utilizing enzymes. The need of CoAses mainly arises under conditions of oxidative stress. CoAse has a conserved Nudix fold and requires a single divalent cation for catalysis. In addition to a signature Nudix motif G[X5]E[X7]REUXEEXGU, where U is  Ile, Leu, or Val, CoAse contains an additional motif upstream called the NuCoA motif (LLTXT(SA)X3RX3GX3FPGG) which is postulated to be involved in CoA recognition. CoA plays a central role in lipid metabolism. It is involved in the initial steps of fatty acid sythesis in the cytosol, in the oxidation of fatty acids and the citric acid cycle in the mitochondria, and in the oxidation of long-chain fatty acids in peroxisomes. CoA has the important role of activating fatty acids for further modification into key biological signalling molecules.','cd03426','CoAse'),(1047479,'gnl|CDD|239948',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, U=I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04511','Nudix_Hydrolase_4'),(1047481,'gnl|CDD|240044',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04688','Nudix_Hydrolase_29'),(1047483,'gnl|CDD|240047',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04691','Nudix_Hydrolase_32'),(1047486,'gnl|CDD|240042',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04686','Nudix_Hydrolase_27'),(1047490,'gnl|CDD|198164',_binary 'Carbamoyl-phosphate synthetase large chain, oligomerisation domain.  Carbamoyl-phosphate synthase catalyses the ATP-dependent synthesis of carbamyl-phosphate from glutamine or ammonia and bicarbonate. The carbamoyl-phosphate synthase (CPS) enzyme in prokaryotes is a heterodimer of a small and large chain.','smart01096','CPSase_L_D3'),(1047491,'gnl|CDD|235393',_binary 'carbamoyl phosphate synthase large subunit; Reviewed.','PRK05294','carB'),(1047492,'gnl|CDD|292600',_binary 'Tn7-like transposition protein D.  TnsD is a family of putative Tn7-like transposition proteins type D.','pfam15978','TnsD'),(1047494,'gnl|CDD|237320',_binary '7-cyano-7-deazaguanine reductase; Provisional.','PRK13258','PRK13258'),(1047496,'gnl|CDD|213775',_binary '7-cyano-7-deazaguanine reductase.  This enzyme catalyzes the 4-electron reduction of the cyano group of 7-cyano-7-deazaguanine (proQ1) to an amine. Although related to a large family of GTP cyclohydrolases (pfam01227), the relationship is structural and not germane to the catalytic mechanism. This mode represents the shorter, gram-positive version of the enzyme as found in B. subtilis. The enzymatic step represents the first point at which the biosynthesis of queuosine in bacteria and eukaryotes is distinguished from the biosynthesis of archaeosine in archaea.','TIGR03139','QueF-II'),(1047498,'gnl|CDD|223851',_binary 'Enzyme related to GTP cyclohydrolase I [General function prediction only].','COG0780','COG0780'),(1047500,'gnl|CDD|316961',_binary 'QueF-like protein.  This protein is involved in the biosynthesis of queuosine. In some proteins this domain appears to be fused to pfam06508.','pfam14489','QueF'),(1047503,'gnl|CDD|274443',_binary '7-cyano-7-deazaguanine reductase.  This enzyme catalyzes the 4-electron reduction of the cyano group of 7-cyano-7-deazaguanine (preQ0) to an amine. Although related to a large family of GTP cyclohydrolases (pfam01227), the relationship is structural and not germane to the catalytic mechanism. This mode represents the longer, gram-negative version of the enzyme as found in E. coli. The enzymatic step represents the first point at which the biosynthesis of queuosine in bacteria and eukaryotes is distinguished from the biosynthesis of archaeosine in archaea. [Transcription, RNA processing].','TIGR03138','QueF'),(1047509,'gnl|CDD|317255',_binary 'Nitrile reductase, 7-cyano-7-deazaguanine-reductase N-term.  The QueF monomer is made up of two ferredoxin-like domains aligned together with their beta-sheets that have additional embellishments. This subunit is composed of a three-stranded beta-sheet and two alpha-helices. QueF reduces a nitrile bond to a primary amine. The two monomer units together create suitable substrate-binding pockets.','pfam14819','QueF_N'),(1047512,'gnl|CDD|236986',_binary '7-cyano-7-deazaguanine reductase; Provisional.','PRK11792','queF'),(1047521,'gnl|CDD|307055',_binary 'Asparagine synthase.  This family is always found associated with pfam00310. Members of this family catalyze the conversion of aspartate to asparagine.','pfam00733','Asn_synthase'),(1047555,'gnl|CDD|200384',_binary 'radical SAM enzyme, rSAM/lipoprotein system.  Members of this protein family are radical SAM enzymes with an additional 4Fe4S cluster-binding C-terminal domain (TIGR04085) shared with PqqE and many other peptide and protein-modifying radical SAM enzymes. All members occur in the context of a predicted lipoprotein that usually is encoded by an adjacent gene.','TIGR04133','coenzyme_PQQ_synthesis_pr'),(1047557,'gnl|CDD|131546',_binary 'pyruvate formate-lyase 1-activating enzyme.  An iron-sulfur protein with a radical-SAM domain (pfam04055). A single glycine residue in EC 2.3.1.54, formate C-acetyltransferase (formate-pyruvate lyase), is oxidized to the corresponding radical by transfer of H from its CH2 to AdoMet with concomitant cleavage of the latter. The reaction requires Fe2+. The first stage is reduction of the AdoMet to give methionine and the 5\'-deoxyadenosin-5-yl radical, which then abstracts a hydrogen radical from the glycine residue. [Energy metabolism, Anaerobic, Protein fate, Protein modification and repair].','TIGR02493','PFLA'),(1048695,'gnl|CDD|222889',_binary 'baseplate wedge subunit and tail pin; Provisional.','PHA02583','11'),(1050227,'gnl|CDD|306637',_binary 'Ribosomal protein S4/S9 N-terminal domain.  This family includes small ribosomal subunit S9 from prokaryotes and S16 from metazoans. This domain is predicted to bind to ribosomal RNA. This domain is composed of four helices in the known structure. However the domain is discontinuous in sequence and the alignment for this family contains only the first three helices.','pfam00163','Ribosomal_S4'),(1050247,'gnl|CDD|183953',_binary 'L-ectoine synthase; Reviewed.','PRK13290','ectC'),(1050248,'gnl|CDD|310736',_binary 'Ectoine synthase.  This family consists of several bacterial ectoine synthase proteins. The ectABC genes encode the diaminobutyric acid acetyltransferase (EctA), the diaminobutyric acid aminotransferase (EctB), and the ectoine synthase (EctC). Together these proteins constitute the ectoine biosynthetic pathway.','pfam06339','Ectoine_synth'),(1050255,'gnl|CDD|314747',_binary 'Domain of unknown function (DUF3440).  This presumed domain is functionally uncharacterized. This domain is found in bacteria. This domain is typically between 53 to 190 amino acids in length. This domain is found associated with pfam01507. This domain has a conserved KND sequence motif.','pfam11922','DUF3440'),(1050378,'gnl|CDD|224422',_binary 'Serine proteases of the peptidase family S9A [Amino acid transport and metabolism].','COG1505','COG1505'),(1051629,'gnl|CDD|226607',_binary 'Predicted O-methyltransferase [General function prediction only].','COG4122','COG4122'),(1051630,'gnl|CDD|307638',_binary 'O-methyltransferase.  Members of this family are O-methyltransferases. The family includes catechol o-methyltransferase, caffeoyl-CoA O-methyltransferase and a family of bacterial O-methyltransferases that may be involved in antibiotic production.','pfam01596','Methyltransf_3'),(1051631,'gnl|CDD|178094',_binary 'O-methyltransferase.','PLN02476','PLN02476'),(1051632,'gnl|CDD|166230',_binary 'caffeoyl-CoA O-methyltransferase.','PLN02589','PLN02589'),(1051634,'gnl|CDD|215417',_binary 'Probable caffeoyl-CoA O-methyltransferase.','PLN02781','PLN02781'),(1051649,'gnl|CDD|234740',_binary 'cobalt-precorrin-6Y C(15)-methyltransferase; Provisional.','PRK00377','cbiT'),(1051650,'gnl|CDD|223813',_binary 'N6-adenine-specific methylase [DNA replication, recombination, and repair].','COG0742','COG0742'),(1051696,'gnl|CDD|318169',_binary 'Nucleoside 2-deoxyribosyltransferase like.','pfam15891','Nuc_deoxyri_tr2'),(1052000,'gnl|CDD|319975',_binary 'formate dehydrogenase accessory protein FdhE and similar proteins.  This family contains formate dehydrogenase accessory protein FdhE and FdhE-like protein, found largely in gamma- and some beta-Proteobacteria, where the fdhE genes are almost always genetically-linked to the structural genes for formate dehydrogenases.  FdhE is  required for the assembly of formate dehydrogenase although not present in the final complex. In E. coli, FdhE interacts with the catalytic subunits of the respiratory formate dehydrogenases. Purification of recombinant FdhE demonstrates the protein is an iron-binding rubredoxin that can adopt monomeric and homodimeric forms. E. coli FdhE interacts with the catalytic subunits, FdnG and FdoG, of the Tat- dependent respiratory formate dehydrogenases. Site-directed mutagenesis has shown that conserved cysteine motifs are essential for the physiological activity of the FdhE protein and are also involved in Fe(III) ligation. The iron likely is redox active, suggesting that the switch from aerobic to anaerobic conditions may be important in modulating FdhE function. Alternatively, FdhE may be involved in an electron transfer reaction, similar to other rubredoxins.','cd16341','FdhE'),(1052514,'gnl|CDD|318170',_binary 'BNR repeat-containing family member.  BNR_4 is a family which carries the unique sequence motif SxDxGxTW which is so characteristic of the repeats of the BNR family, pfam02012. It is unclear whether or not this unit is repeated throughout the sequences of this family, but if it is then the family is likely to be bacterial neuraminidase.','pfam15892','BNR_4'),(1052881,'gnl|CDD|224411',_binary 'Fructose-1,6-bisphosphatase/sedoheptulose 1,7-bisphosphatase and related proteins [Carbohydrate transport and metabolism].','COG1494','GlpX'),(1055478,'gnl|CDD|237025',_binary 'putative bifunctional UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase/UDP-N-acetylmuramoyl-tripeptide:D-alanyl-D-alanine ligase; Provisional.','PRK11929','PRK11929'),(1056001,'gnl|CDD|224694',_binary 'Protein involved in ribonucleotide reduction [Nucleotide transport and metabolism].','COG1780','NrdI'),(1056002,'gnl|CDD|179603',_binary 'ribonucleotide reductase stimulatory protein; Reviewed.','PRK03600','nrdI'),(1056188,'gnl|CDD|237791',_binary 'reverse gyrase; Provisional.','PRK14701','PRK14701'),(1056189,'gnl|CDD|236049',_binary 'ribonucleotide-diphosphate reductase subunit alpha; Validated.','PRK07562','PRK07562'),(1056557,'gnl|CDD|223003',_binary 'hypothetical protein; Provisional.','PHA03169','PHA03169'),(1056794,'gnl|CDD|119347',_binary 'ASC-1 homology domain, subfamily similar to Pyrococcus furiosus Pf0470. The ASCH domain, a small beta-barrel domain found in all three kingdoms of life, resembles the RNA-binding PUA domain and may also interact with RNA. ASCH has been proposed to function as an RNA-binding domain during coactivation, RNA-processing and the regulation of prokaryotic translation.','cd06555','ASCH_PF0470_like'),(1058590,'gnl|CDD|130619',_binary 'L-rhamnosyltransferase.  This model subfamily is comprised of gamma proteobacteria whose proteins function as L-rhamnosyltransferases in the synthesis of their respective surface polysaccharides. Rhamnolipids are glycolipids containing mono- or di- L-rhamnose molecules. Rhamnolipid synthesis occurs by sequential glycosyltransferase reactions involving two distinct rhamnosyltransferase enzymes. In P.aeruginosa, the synthesis of mono-rhamnolipids is catalyzed by rhamnosyltransferase 1, and proceeds by a glycosyltransfer reaction catalyzed by rhamnosyltransferase 2 to yield di-rhamnolipids. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01556','dTDP-rhamnosyl_transferas'),(1059327,'gnl|CDD|222853',_binary 'putative transposase OrfB; Reviewed.','PHA02517','PHA02517'),(1059328,'gnl|CDD|307008',_binary 'Integrase core domain.  Integrase mediates integration of a DNA copy of the viral genome into the host chromosome. Integrase is composed of three domains. The amino-terminal domain is a zinc binding domain pfam02022. This domain is the central catalytic domain. The carboxyl terminal domain that is a non-specific DNA binding domain pfam00552. The catalytic domain acts as an endonuclease when two nucleotides are removed from the 3\' ends of the blunt-ended viral DNA made by reverse transcription. This domain also catalyzes the DNA strand transfer reaction of the 3\' ends of the viral DNA to the 5\' ends of the integration site.','pfam00665','rve'),(1059329,'gnl|CDD|225361',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG2801','Tra5'),(1059330,'gnl|CDD|315857',_binary 'HTH-like domain.  This domain contains a predicted helix-turn-helix suggesting a DNA-binding function.','pfam13276','HTH_21'),(1059331,'gnl|CDD|316225',_binary 'Integrase core domain.','pfam13683','rve_3'),(1059332,'gnl|CDD|290072',_binary 'Integrase core domain.','pfam13333','rve_2'),(1059510,'gnl|CDD|224748',_binary 'Predicted acyltransferases [Lipid metabolism].','COG1835','COG1835'),(1059797,'gnl|CDD|223829',_binary 'Predicted Rossmann fold nucleotide-binding protein involved in DNA uptake [DNA replication, recombination, and repair / Intracellular trafficking and secretion].','COG0758','Smf'),(1060455,'gnl|CDD|238727',_binary 'Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses  In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains  .','cd01450','vWFA_subfamily_ECM'),(1060623,'gnl|CDD|224240',_binary 'Mn-dependent transcriptional regulator [Transcription].','COG1321','TroR'),(1060680,'gnl|CDD|279374',_binary 'Bacterial regulatory protein, arsR family.  Members of this family contains a DNA binding \'helix-turn-helix\' motif. This family includes other proteins which are not included in the Prosite definition.','pfam01022','HTH_5'),(1060843,'gnl|CDD|131106',_binary 'Hg(II)-responsive transcriptional regulator.  This model represents the mercury (II) responsive transcriptional activator of the mer organomercurial resistance operon. This protein is a member of the MerR family of transcriptional activators (pfam00376) and contains a distinctive pattern of cysteine residues in its metal binding loop, Cys-X(8)-Cys-Pro, as well as a conserved and critical cysteine at the N-terminal end of the dimerization helix. [Cellular processes, Detoxification, Regulatory functions, DNA interactions].','TIGR02051','Mercuric_resistance_opero'),(1060845,'gnl|CDD|184302',_binary 'putative transcriptional regulator MerR; Provisional.','PRK13752','PRK13752'),(1060920,'gnl|CDD|131813',_binary 'cryptochrome, plant family.  At least five major families of cryptochomes and photolyases share FAD cofactor binding, sequence homology, and the ability to react to short wavelengths of visible light. Photolysases are responsible for light-dependent DNA repair by removal of two types of uv-induced DNA dimerizations. Cryptochromes have other functions, often regulatory and often largely unknown, which may include circadian clock entrainment and control of development. Members of this subfamily are known so far only in plants; they may show some photolyase activity in vitro but appear mostly to be regulatory proteins that respond to blue light.','TIGR02766','crypt_chrom_pln'),(1061177,'gnl|CDD|273392',_binary 'capsular exopolysaccharide family.  This model describes the capsular exopolysaccharide proteins in bacteria. The exopolysaccharide gene cluster consists of several genes which encode a number of proteins which regulate the exoploysaccharide biosynthesis(EPS). Atleast 13 genes espA to espM in streptococcus species seem to direct the EPS proteins and all of which share high homology. Functional roles were characterized by gene disruption experiments which resulted in exopolysaccharide-deficient phenotypes. [Transport and binding proteins, Carbohydrates, organic alcohols, and acids].','TIGR01007','Tyrosine-protein_kinase_C'),(1061178,'gnl|CDD|274798',_binary 'helicase/secretion neighborhood CpaE-like protein.  Members of this protein family belong to the MinD/ParA family of P-loop NTPases, and in particular show homology to the CpaE family of pilus assembly proteins (see ). Nearly all members are found, not only in a gene context consistent with pilus biogenesis or a pilus-like secretion apparatus, but also near a DEAD/DEAH-box helicase, suggesting an involvement in DNA transfer activity. The model describes a clade restricted to the Actinobacteria.','TIGR03815','CpaE_hom_Actino'),(1061179,'gnl|CDD|177011',_binary 'photochlorophyllide reductase subunit L.','CHL00072','chlL'),(1061181,'gnl|CDD|280525',_binary 'Anion-transporting ATPase.  This Pfam family represents a conserved domain, which is sometimes repeated, in an anion-transporting ATPase. The ATPase is involved in the removal of arsenate, antimonite, and arsenate from the cell.','pfam02374','ArsA_ATPase'),(1062339,'gnl|CDD|212155',_binary 'VbhA antitoxin and related proteins.  VbhA is the antitoxin to VbhT. The VbhT toxin of the mammalian pathogen Bartonella schoenbuchensis is responsible for the disruptive adenylation of host proteins. VbhT also induces FIC-domain-mediated growth arrest in bacteria; it is inhibited by this antitoxin which binds to block the ATP binding site of the VbhT FIC domain.','cd11586','VbhA_like'),(1062419,'gnl|CDD|235054',_binary 'thiamine-phosphate pyrophosphorylase; Provisional.','PRK02615','PRK02615'),(1062420,'gnl|CDD|238317',_binary 'Thiamine monophosphate synthase (TMP synthase)/TenI. TMP synthase catalyzes an important step in the thiamine biosynthesis pathway, the substitution of the pyrophosphate of 2-methyl-4-amino-5- hydroxymethylpyrimidine pyrophosphate by 4-methyl-5- (beta-hydroxyethyl) thiazole phosphate to yield thiamine phosphate. TenI is a enzymatically inactive regulatory protein involved in the regulation of several extracellular enzymes. This superfamily also contains other enzymatically inactive proteins with unknown functions.','cd00564','TMP_TenI'),(1063091,'gnl|CDD|314130',_binary 'Minor capsid protein.  Most of the members of this family are annotated as being minor capsid proteins. The genomes carrying the genes usually have three similar proteins adjacent to each other, hence this one being named as No.2.','pfam11114','Minor_capsid_2'),(1063515,'gnl|CDD|277335',_binary 'Bacillus subtilis PhoD and related proteins, metallophosphatase domain.  PhoD (also known as alkaline phosphatase D/APaseD in Bacillus subtilis) is a secreted phosphodiesterase encoded by phoD of the Pho regulon in Bacillus subtilis. PhoD homologs are found in prokaryotes, eukaryotes, and archaea. PhoD contains a twin arginine (RR) motif and is transported by the Tat (Twin-arginine translocation) translocation pathway machinery (TatAyCy). This family also includes the Fusarium oxysporum Fso1 protein. PhoD belongs to the metallophosphatase (MPP) superfamily. MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases). The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07389','MPP_PhoD'),(1063516,'gnl|CDD|312802',_binary 'PhoD-like phosphatase.','pfam09423','PhoD'),(1063517,'gnl|CDD|226070',_binary 'Phosphodiesterase/alkaline phosphatase D [Inorganic ion transport and metabolism].','COG3540','PhoD'),(1063559,'gnl|CDD|277368',_binary 'Shewanella-like phosphatases, metallophosphatase domain.  This family includes bacterial, eukaryotic, and archeal proteins orthologous to the Shewanella cold-active protein-tyrosine phosphatase, CAPTPase.  CAPTPase is an uncharacterized protein that belongs to the Shelph (Shewanella-like phosphatase) family of PPP (phosphoprotein phosphatases).  The PPP family is one of two known protein phosphatase families specific for serine and threonine.  In addition to Shelps, the PPP family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07425','MPP_Shelphs'),(1063564,'gnl|CDD|224307',_binary 'DNA topoisomerase VI, subunit B [DNA replication, recombination, and repair].','COG1389','COG1389'),(1063567,'gnl|CDD|235246',_binary 'DNA topoisomerase VI subunit B; Validated.','PRK04184','PRK04184'),(1063595,'gnl|CDD|100102',_binary 'Ribosomal protein L7/L12. Ribosomal protein L7/L12 refers to the large ribosomal subunit proteins L7 and L12, which are identical except that L7 is acetylated at the N terminus. It is a component of the L7/L12 stalk, which is located at the surface of the ribosome. The stalk base consists of a portion of the 23S rRNA and ribosomal proteins L11 and L10. An extended C-terminal helix of L10 provides the binding site for L7/L12. L7/L12 consists of two domains joined by a flexible hinge, with the helical N-terminal domain (NTD) forming pairs of homodimers that bind to the extended helix of L10. It is the only multimeric ribosomal component, with either four or six copies per ribosome that occur as two or three dimers bound to the L10 helix. L7/L12 is the only ribosomal protein that does not interact directly with rRNA, but instead has indirect interactions through L10. The globular C-terminal domains of L7/L12 are highly mobile. They are exposed to the cytoplasm and contain binding sites for other molecules. Initiation factors, elongation factors, and release factors are known to interact with the L7/L12 stalk during their GTP-dependent cycles. The binding site for the factors EF-Tu and EF-G comprises L7/L12, L10, L11, the L11-binding region of 23S rRNA, and the sarcin-ricin loop of 23S rRNA. Removal of L7/L12 has minimal effect on factor binding and it has been proposed that L7/L12 induces the catalytically active conformation of EF-Tu and EF-G, thereby stimulating the GTPase activity of both factors. In eukaryotes, the proteins that perform the equivalent function to L7/L12 are called P1 and P2, which do not share sequence similarity with L7/L12. However, a bacterial L7/L12 homolog is found in some eukaryotes, in mitochondria and chloroplasts. In archaea, the protein equivalent to L7/L12 is called aL12 or L12p, but it is closer in sequence to P1 and P2 than to L7/L12.','cd00387','Ribosomal_L7_L12'),(1063596,'gnl|CDD|306923',_binary 'Ribosomal protein L7/L12 C-terminal domain.','pfam00542','Ribosomal_L12'),(1063597,'gnl|CDD|234671',_binary '50S ribosomal protein L7/L12; Reviewed.','PRK00157','rplL'),(1063598,'gnl|CDD|273301',_binary 'ribosomal protein L7/L12.  Ribosomal proteins L7 and L12 are synonymous except for post-translational modification of the N-terminal amino acid. THis model resembles pfam00542 but matches the full length of prokaryotic and organellar proteins rather than just the C-terminus. [Protein synthesis, Ribosomal proteins: synthesis and modification].','TIGR00855','50S_ribosomal_protein_L7/'),(1063599,'gnl|CDD|223300',_binary 'Ribosomal protein L7/L12 [Translation, ribosomal structure and biogenesis].','COG0222','RplL'),(1063692,'gnl|CDD|234750',_binary 'recombination and DNA strand exchange inhibitor protein; Reviewed.','PRK00409','PRK00409'),(1063702,'gnl|CDD|236545',_binary 'cell envelope integrity inner membrane protein TolA; Provisional.','PRK09510','tolA'),(1063979,'gnl|CDD|224380',_binary 'ABC-type transport system involved in resistance to organic solvents, periplasmic component [Secondary metabolites biosynthesis, transport, and catabolism].','COG1463','Ttg2C'),(1064094,'gnl|CDD|184753',_binary 'outer membrane N-deacetylase; Provisional.','PRK14581','hmsF'),(1064095,'gnl|CDD|184754',_binary 'outer membrane N-deacetylase; Provisional.','PRK14582','pgaB'),(1064096,'gnl|CDD|274867',_binary 'poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase PgaB.  Two well-characterized systems produce polysaccharide based on N-acetyl-D-glucosamine in straight chains with beta-1,6 linkages. These are encoded by the icaADBC operon in Staphylococcus species, where the system is designated polysaccharide intercellular adhesin (PIA), and the pgaABCD operon in Gram-negative bacteria such as E. coli. Both systems include a putative polysaccharide deacetylase. The PgaB protein, described here, contains an additional domain lacking from its Gram-positive counterpart IcaB (TIGR03933). Deacetylation by this protein appears necessary to allow export through the porin PgaA [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR03938','hypothetical_protein_Pcar'),(1064097,'gnl|CDD|200586',_binary 'N-terminal putative catalytic polysaccharide deacetylase domain of bacterial poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase PgaB, and similar proteins.  This family is represented by an outer membrane lipoprotein, poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase (PgaB, EC 3.5.1.-), encoded by Escherichia coli pgaB gene from the pgaABCD (formerly ycdSRQP) operon, which affects biofilm development by promoting abiotic surface binding and intercellular adhesion. PgaB catalyzes the N-deacetylation of poly-beta-1,6-N-acetyl-D-glucosamine (PGA), a biofilm adhesin polysaccharide that stabilizes biofilms of E. coli and other bacteria. PgaB contains an N-terminal NodB homology domain with a 5-stranded beta/alpha barrel, and a C-terminal carbohydrate binding domain required for PGA N-deacetylation, which may be involved in binding to unmodified poly-beta-1,6-GlcNAc and assisting catalysis by the deacetylase domain. This family also includes several orthologs of PgaB, such as the hemin storage system HmsF protein, encoded by Yersinia pestis hmsF gene from the hmsHFRS operon, which is essential for Y. pestis biofilm formation. Like PgaB, HmsF is an outer membrane protein with an N-terminal NodB homology domain, which is likely involved in the modification of the exopolysaccharide (EPS) component of the biofilm. HmsF also has a conserved but uncharacterized C-terminal domain that is present in other HmsF-like proteins in Gram-negative bacteria. This alignment model corresponds to the N-terminal NodB homology domain.','cd10964','CE4_PgaB_5s'),(1064098,'gnl|CDD|200587',_binary 'Putative catalytic polysaccharide deacetylase domain of bacterial intercellular adhesion protein IcaB and similar proteins.  The family is represented by the surface-attached protein intercellular adhesion protein IcaB (Poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase, EC 3.5.1.-), encoded by Staphylococcus epidermidis icaB gene from the icaABC gene cluster that is involved in the synthesis of polysaccharide intercellular adhesin (PIA), which is located mainly on the cell surface. IcaB is a secreted, cell wall-associated protein that plays a crucial role in exopolysaccharide modification in bacterial biofilm formation. It catalyzes the N-deacetylation of poly-beta-1,6-N-acetyl-D-glucosamine (PNAG, also referred to as PIA), a biofilm adhesin polysaccharide. IcaB shows high homology to the N-terminal NodB homology domain of Escherichia coli PgaB. At this point, they are classified in the same family.','cd10965','CE4_IcaB_5s'),(1064100,'gnl|CDD|213026',_binary 'Putative catalytic NodB homology domain of a hypothetical protein Ecf1 from Escherichia coli and similar proteins.  This family contains a hypothetical protein Ecf1 from Escherichia coli and its prokaryotic homologs. Although their biochemical properties remain to be determined, members in this family contain a conserved domain with a 5-stranded beta/alpha barrel, which is similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, belonging to the larger carbohydrate esterase 4 (CE4) superfamily.','cd10969','CE4_Ecf1_like_5s'),(1064340,'gnl|CDD|139929',_binary 'plasmid-partitioning protein RepA; Provisional.','PRK13869','PRK13869'),(1064342,'gnl|CDD|235662',_binary 'cob(I)alamin adenolsyltransferase/cobinamide ATP-dependent adenolsyltransferase; Validated.','PRK05986','PRK05986'),(1064344,'gnl|CDD|225020',_binary 'ATP:corrinoid adenosyltransferase [Coenzyme metabolism].','COG2109','BtuR'),(1064347,'gnl|CDD|282122',_binary 'CENP-B N-terminal DNA-binding domain.  Centromere Protein B (CENP-B) is a DNA-binding protein localized to the centromere. Within the N-terminal 125 residues, there is a DNA-binding region, which binds to a corresponding 17bp CENP-B box sequence. CENP-B dimers either bind two separate DNA molecules or alternatively, they may bind two CENP-B boxes on one DNA molecule, with the intervening stretch of DNA forming a loop structure. The CENP-B DNA-binding domain consists of two repeating domains, RP1 and RP2. This family corresponds to RP1 has been shown to consist of four helices in a helix-turn-helix structure.','pfam04218','CENP-B_N'),(1066840,'gnl|CDD|313271',_binary 'Uncharacterized protein conserved in bacteria (DUF2256).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10013','DUF2256'),(1067782,'gnl|CDD|310145',_binary 'Protein of unknown function (DUF745).  This family consists of several uncharacterized Drosophila melanogaster proteins of unknown function.','pfam05335','DUF745'),(1068588,'gnl|CDD|280843',_binary 'Proliferating cell nuclear antigen, C-terminal domain.  N-terminal and C-terminal domains of PCNA are topologically identical. Three PCNA molecules are tightly associated to form a closed ring encircling duplex DNA.','pfam02747','PCNA_C'),(1068751,'gnl|CDD|153097',_binary 'Ferritin-like superfamily of diiron-containing four-helix-bundle proteins.  Ferritin-like, diiron-carboxylate proteins participate in a range of functions including iron regulation, mono-oxygenation, and reactive radical production. These proteins are characterized by the fact that they catalyze dioxygen-dependent oxidation-hydroxylation reactions within diiron centers; one exception is manganese catalase, which catalyzes peroxide-dependent oxidation-reduction within a dimanganese center. Diiron-carboxylate proteins are further characterized by the presence of duplicate metal ligands, glutamates and histidines (ExxH) and two additional glutamates within a four-helix bundle. Outside of these conserved residues there is little obvious homology. Members include bacterioferritin, ferritin, rubrerythrin, aromatic and alkene monooxygenase hydroxylases (AAMH), ribonucleotide reductase R2 (RNRR2), acyl-ACP-desaturases (Acyl_ACP_Desat), manganese (Mn) catalases, demethoxyubiquinone hydroxylases (DMQH), DNA protecting proteins (DPS), and ubiquinol oxidases (AOX), and the aerobic cyclase system, Fe-containing subunit (ACSF).','cd00657','Ferritin_like'),(1068808,'gnl|CDD|129694',_binary 'rad50.  All proteins in this family for which functions are known are involvedin recombination, recombinational repair, and/or non-homologous end joining.They are components of an exonuclease complex with MRE11 homologs. This family is distantly related to the SbcC family of bacterial proteins.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University).','TIGR00606','RAD50_homologue_ceRAD50'),(1069006,'gnl|CDD|225117',_binary 'AraC-type DNA-binding domain-containing proteins [Transcription].','COG2207','AraC'),(1070011,'gnl|CDD|307151',_binary 'Transposase, Mutator family.','pfam00872','Transposase_mut'),(1070012,'gnl|CDD|315109',_binary 'Protein of unknown function (DUF3644).  This domain family is found in bacteria, and is typically between 65 and 80 amino acids in length.','pfam12358','DUF3644'),(1070013,'gnl|CDD|225865',_binary 'Transposase and inactivated derivatives [DNA replication, recombination, and repair].','COG3328','COG3328'),(1070014,'gnl|CDD|313718',_binary 'MULE transposase domain.  This domain was identified by Babu and colleagues.','pfam10551','MULE'),(1070021,'gnl|CDD|320687',_binary 'M48 Ste24 endopeptidase-like, integral membrane metallopeptidase.  This family contains peptidase M48-like proteins that are as yet uncharacterized, but probably function as intracellular, membrane-associated zinc metalloproteases; they all contain the HEXXH Zn-binding motif, which is critical for Ste24p activity. They likely remove the C-terminal three residues of farnesylated proteins proteolytically and are possibly associated with the endoplasmic reticulum and golgi.','cd07328','M48_Ste24p_like'),(1070040,'gnl|CDD|315832',_binary 'Domain of unknown function (DUF4041).  This presumed domain is functionally uncharacterized. This domain family is found in bacteria, archaea and viruses, and is approximately 60 amino acids in length. The family is found in association with pfam10544.','pfam13250','DUF4041'),(1070411,'gnl|CDD|308366',_binary 'Preprotein translocase subunit.  See.','pfam02699','YajC'),(1070661,'gnl|CDD|214922',_binary 'Outer membrane protein (OmpH-like).  This family includes outer membrane proteins such as OmpH among others. Skp (OmpH) has been characterized as a molecular chaperone that interacts with unfolded proteins as they emerge in the periplasm from the Sec translocation machinery.','smart00935','OmpH'),(1071234,'gnl|CDD|222840',_binary 'TM2 domain-containing protein.','PHA01886','PHA01886'),(1071862,'gnl|CDD|99841',_binary 'N-terminal DEDDh 3\'-5\' exonuclease domain of Escherichia coli exonuclease I and similar proteins.  This subfamily is composed of the N-terminal domain of Escherichia coli exonuclease I (ExoI) and similar proteins. ExoI is a monomeric enzyme that hydrolyzes single stranded DNA in the 3\' to 5\' direction. It plays a role in DNA recombination and repair. It primarily functions in repairing frameshift mutations. The N-terminal domain of ExoI is a DEDDh-type DnaQ-like 3\'-5 exonuclease containing three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific Hx(4)D conserved pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The ExoI structure is unique among DnaQ family enzymes in that there is a large distance between the two metal ions required for catalysis and the catalytic histidine is oriented away from the active site.','cd06138','ExoI_N'),(1071867,'gnl|CDD|225072',_binary 'Antitoxin of toxin-antitoxin stability system [Cell division and chromosome partitioning].','COG2161','StbD'),(1072101,'gnl|CDD|214934',_binary 'Region found in RelA / SpoT proteins.  The functions of Escherichia coli RelA and SpoT differ somewhat. RelA produces pppGpp (or ppGpp) from ATP and GTP (or GDP). SpoT degrades ppGpp, but may also act as a secondary ppGpp synthetase. The two proteins are strongly similar. In many species, a single homolog to SpoT and RelA appears reponsible for both ppGpp synthesis and ppGpp degradation. (p)ppGpp is a regulatory metabolite of the stringent response, but appears also to be involved in antibiotic biosynthesis in some species.','smart00954','RelA_SpoT'),(1072102,'gnl|CDD|143389',_binary 'Nucleotidyltransferase (NT) domain of RelA- and SpoT-like ppGpp synthetases and hydrolases.  This family includes the catalytic domains of Escherichia coli ppGpp synthetase (RelA), ppGpp synthetase/hydrolase (SpoT), and related proteins. RelA synthesizes (p)ppGpp in response to amino-acid starvation and in association with ribosomes. (p)ppGpp triggers the bacterial stringent response. SpoT catalyzes (p)ppGpp synthesis under carbon limitation in a ribosome-independent manner. It also catalyzes (p)ppGpp degradation. Gram-negative bacteria have two enzymes involved in (p)ppGpp metabolism while most Gram-positive organisms have a single Rel-Spo enzyme (Rel), which both synthesizes and degrades (p)ppGpp. The Arabidopsis thaliana Rel-Spo proteins, At-RSH1,-2, and-3 appear to regulate a rapid (p)ppGpp-mediated response to pathogens and other stresses. This catalytic domain is found in association with an N-terminal HD domain and a C-terminal metal dependent phosphohydrolase domain (TGS). Some Rel-Spo proteins also have a C-terminal regulatory ACT domain. This subgroup belongs to the Pol beta-like NT superfamily. In the majority of enzymes in this superfamily, two carboxylates, Dx[D/E], together with a third more distal carboxylate, coordinate two divalent metal cations involved in a two-metal ion mechanism of nucleotide addition.Two of the three catalytic carboxylates are found in Rel-Spo enzymes, with the second carboxylate of the DXD motif missing. Evidence supports a single-cation synthetase mechanism.','cd05399','NT_Rel-Spo_like'),(1072103,'gnl|CDD|225232',_binary 'PpGpp synthetase catalytic domain [General function prediction only].','COG2357','COG2357'),(1072104,'gnl|CDD|309652',_binary 'Region found in RelA / SpoT proteins.  This region of unknown function is found in RelA and SpoT of Escherichia coli, and their homologs in plants and in other eubacteria. RelA is a guanosine 3\',5\'-bis-pyrophosphate (ppGpp) synthetase (EC:2.7.6.5) while SpoT is thought to be a bifunctional enzyme catalyzing both ppGpp synthesis and degradation (ppGpp 3\'-pyrophosphohydrolase, (EC:3.1.7.2)). This region is often found in association with HD (pfam01966), a metal-dependent phosphohydrolase, TGS (pfam02824) which is a possible nucleotide-binding region, and the ACT regulatory domain (pfam01842).','pfam04607','RelA_SpoT'),(1072105,'gnl|CDD|227504',_binary 'Uncharacterized conserved protein [Function unknown].','COG5177','COG5177'),(1072135,'gnl|CDD|286429',_binary 'IclR helix-turn-helix domain.','pfam09339','HTH_IclR'),(1072144,'gnl|CDD|224332',_binary 'Transcriptional regulator [Transcription].','COG1414','IclR'),(1072147,'gnl|CDD|316102',_binary 'Crp-like helix-turn-helix domain.  This family represents a crp-like helix-turn-helix domain that is likely to bind DNA.','pfam13545','HTH_Crp_2'),(1072359,'gnl|CDD|224589',_binary 'Transcription initiation factor IIE, alpha subunit [Transcription].','COG1675','TFA1'),(1073462,'gnl|CDD|273294',_binary 'sodium/hydrogen exchanger 3.  The Monovalent Cation:Proton Antiporter-1 (CPA1) Family (TC 2.A.36)The CPA1 family is a large family of proteins derived from Gram-positive and Gram-negative bacteria, blue green bacteria, yeast, plants and animals.Transporters from eukaryotes have been functionally characterized, and all of these catalyze Na+:H+ exchange. Their primary physiological functions may be in(1) cytoplasmic pH regulation, extruding the H+ generated during metabolism, and (2) salt tolerance (in plants), due to Na+ uptake into vacuoles.This model is specific for the eukaryotic members members of this family. [Transport and binding proteins, Cations and iron carrying compounds].','TIGR00840','Endosomal/prevacuolar_sod'),(1074694,'gnl|CDD|114219',_binary 'Synaptonemal complex protein 1 (SCP-1).  Synaptonemal complex protein 1 (SCP-1) is the major component of the transverse filaments of the synaptonemal complex. Synaptonemal complexes are structures that are formed between homologous chromosomes during meiotic prophase.','pfam05483','SCP-1'),(1075310,'gnl|CDD|235058',_binary 'putative inner membrane protein translocase component YidC; Provisional.','PRK02654','PRK02654'),(1075507,'gnl|CDD|310504',_binary 'Bacterial protein of unknown function (DUF885).  This family consists of several hypothetical bacterial proteins several of which are putative membrane proteins.','pfam05960','DUF885'),(1075508,'gnl|CDD|227142',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4805','COG4805'),(1075676,'gnl|CDD|274512',_binary '2-aminoethylphosphonate aminotransferase.  This family includes a number of 2-aminoethylphosphonate aminotransferases, some of which are indicated to operate in the catabolism of 2-aminoethylphosphonate (AEP) and others which are involved in the biosynthesis of the same compound. The catabolic enzyme (PhnW) is known to use pyruvate:alanine as the transfer partner and is modeled by the equivalog-level model (TIGR02326). The PhnW family is apparently a branch of a larger tree including genes (AepZ) adjacent to others responsible for the biosynthesis of phosphonoacetaldehyde. The identity of the transfer partner is unknown for these enzymes and considering the reversed flux compared to PhnW, it may very well be different.','TIGR03301','PhnW-AepZ'),(1075984,'gnl|CDD|275079',_binary 'nitrous oxide reductase family maturation protein NosD.  Members of this family include NosD, a repetitive periplasmic protein required for the maturation of the copper-containing enzyme nitrous-oxide reductase. NosD appears to be part of a complex with NosF (an ABC transporter family ATP-binding protein) and NosY (a six-helix transmembrane protein in the ABC-2 permease family). However, NosDFY-like complexes appear to occur also in species whose copper requiring enzymes are something other than nitrous-oxide reductase.','TIGR04247','nitrous_oxide_maturation_'),(1076006,'gnl|CDD|227645',_binary 'Predicted transcriptional regulator [Transcription].','COG5340','COG5340'),(1076421,'gnl|CDD|133384',_binary 'Helix-Turn-Helix DNA binding domain of the MerR-like transcription regulators YyaN and YraB.  Putative helix-turn-helix (HTH) MerR-like transcription regulators of Bacillus subtilis, YyaN and YraB, and related proteins; N-terminal domain. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd01109','HTH_YyaN'),(1076452,'gnl|CDD|133387',_binary 'Helix-Turn-Helix DNA binding domain of HspR-like transcription regulators.  Helix-turn-helix (HTH) transcription regulator HspR and related proteins, N-terminal domain. Heat shock protein regulators (HspR) have been shown to regulate expression of specific regulons in response to high temperature or high osmolarity in Streptomyces and Helicobacter, respectively. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.','cd01279','HTH_HspR-like'),(1076454,'gnl|CDD|133416',_binary 'Helix-Turn-Helix DNA binding domain of the Cfa transcription regulator.  Putative helix-turn-helix (HTH) MerR-like transcription regulator; the N-terminal domain of Cfa, a cyclopropane fatty acid synthase and other related methyltransferases. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04789','HTH_Cfa'),(1076456,'gnl|CDD|133391',_binary 'Helix-Turn-Helix DNA binding domain of MlrA-like transcription regulators.  Helix-turn-helix (HTH) transcription regulator MlrA (merR-like regulator A) and related proteins, N-terminal domain. The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium. Its close homolog, CarA from Myxococcus xanthus, is involved in activation of the carotenoid biosynthesis genes by light. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules. Many MlrA-like proteins in this group appear to lack the long dimerization helix seen in the N-terminal domains of typical MerR-like proteins.','cd04763','HTH_MlrA-like'),(1076457,'gnl|CDD|133383',_binary 'Helix-Turn-Helix DNA binding domain of CueR-like transcription regulators.  Helix-turn-helix (HTH) transcription regulators CueR and ActP, copper efflux regulators. In Bacillus subtilis, copper induced CueR regulates the copZA operon, preventing copper toxicity. In Rhizobium leguminosarum, ActP controls copper homeostasis; it detects cytoplasmic copper stress and activates transcription in response to increasing copper concentrations. These proteins are comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain winged HTH motifs that mediate DNA binding, while the C-terminal domains have two conserved cysteines that define a monovalent copper ion binding site. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.','cd01108','HTH_CueR'),(1076458,'gnl|CDD|133402',_binary 'Helix-Turn-Helix DNA binding domain of Cfa-like transcription regulators.  Putative helix-turn-helix (HTH) MerR-like transcription regulators; the HTH domain of Cfa, a cyclopropane fatty acid synthase, and other related methyltransferases, as well as, the N-terminal domain of a conserved, uncharacterized ~172 a.a. protein. Based on sequence similarity of the N-terminal domain, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04775','HTH_Cfa-like'),(1076461,'gnl|CDD|133393',_binary 'Helix-Turn-Helix DNA binding domain of putative MlrA-like transcription regulators.  Putative helix-turn-helix (HTH) MlrA-like transcription regulators (subgroup 2), N-terminal domain. The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.','cd04765','HTH_MlrA-like_sg2'),(1076830,'gnl|CDD|215405',_binary '4-aminobutyrate:pyruvate transaminase.','PLN02760','PLN02760'),(1076832,'gnl|CDD|180471',_binary 'glutamate-1-semialdehyde 2,1-aminomutase; Provisional.','PRK06209','PRK06209'),(1076953,'gnl|CDD|316274',_binary 'tRNA nucleotidyltransferase domain 2 putative.','pfam13735','tRNA_NucTran2_2'),(1076962,'gnl|CDD|222859',_binary 'terminase ATPase subunit; Provisional.','PHA02535','P'),(1076982,'gnl|CDD|236088',_binary 'acetyltransferase; Provisional.','PRK07757','PRK07757'),(1078369,'gnl|CDD|310273',_binary 'Mitotic checkpoint protein.  This family consists of several eukaryotic mitotic checkpoint (Mitotic arrest deficient or MAD) proteins. The mitotic spindle checkpoint monitors proper attachment of the bipolar spindle to the kinetochores of aligned sister chromatids and causes a cell cycle arrest in prometaphase when failures occur. Multiple components of the mitotic spindle checkpoint have been identified in yeast and higher eukaryotes. In S.cerevisiae, the existence of a Mad1-dependent complex containing Mad2, Mad3, Bub3 and Cdc20 has been demonstrated.','pfam05557','MAD'),(1078370,'gnl|CDD|310531',_binary 'Cell division protein ZapB.  ZapB is a non-essential, abundant cell division factor that is required for proper Z-ring formation.','pfam06005','ZapB'),(1078372,'gnl|CDD|309416',_binary 'Phosphomevalonate kinase.  Phosphomevalonate kinase (EC:2.7.4.2) catalyzes the phosphorylation of 5-phosphomevalonate into 5-diphosphomevalonate, an essential step in isoprenoid biosynthesis via the mevalonate pathway. This family represents the animal type of the enzyme. The other is the ERG8 type, found in plants and fungi, and some bacteria (see pfam00288).','pfam04275','P-mevalo_kinase'),(1079687,'gnl|CDD|319247',_binary 'N-terminal domain of transcriptional regulators similar to SbnI.  Siderophore staphylobactin biosynthesis protein SbnI of Staphylococcus aureus is a ParB/Spo0J like protein required for the expression of genes in the sbn operon, which is responsible for staphyloferrin B (SB) biosynthesis. SnbI forms dimers and binds DNA upstream of sdnD. SbnI binds heme, which inhibits DNA binding of SbnI, leading to a suppression of sbn operon expression.','cd16388','SbnI_like_N'),(1080305,'gnl|CDD|226297',_binary 'Mannosyltransferase OCH1 and related enzymes [Cell envelope biogenesis, outer membrane].','COG3774','OCH1'),(1080306,'gnl|CDD|309577',_binary 'Glycosyltransferase sugar-binding region containing DXD motif.  The DXD motif is a short conserved motif found in many families of glycosyltransferases, which add a range of different sugars to other sugars, phosphates and proteins. DXD-containing glycosyltransferases all use nucleoside diphosphate sugars as donors and require divalent cations, usually manganese. The DXD motif is expected to play a carbohydrate binding role in sugar-nucleoside diphosphate and manganese dependent glycosyltransferases.','pfam04488','Gly_transf_sug'),(1080307,'gnl|CDD|308374',_binary 'N-terminal domain of galactosyltransferase.  This is the N-terminal domain of a family of galactosyltransferases from a wide range of Metazoa with three related galactosyltransferases activities, all three of which are possessed by one sequence in some cases. EC:2.4.1.90, N-acetyllactosamine synthase; EC:2.4.1.38, Beta-N-acetylglucosaminyl-glycopeptide beta-1,4- galactosyltransferase; and EC:2.4.1.22 Lactose synthase. Note that N-acetyllactosamine synthase is a component of Lactose synthase along with alpha-lactalbumin, in the absence of alpha-lactalbumin EC:2.4.1.90 is the catalyzed reaction.','pfam02709','Glyco_transf_7C'),(1080309,'gnl|CDD|161703',_binary 'tRNA (guanine-N(7)-)-methyltransferase.  This predicted S-adenosylmethionine-dependent methyltransferase is found in a single copy in most Bacteria. It is also found, with a short amino-terminal extension in eukaryotes. Its function is unknown. In E. coli, this protein flanks the DNA repair protein MutY, also called micA. [Protein synthesis, tRNA and rRNA base modification].','TIGR00091','tRNA_guanine-N7--methyltr'),(1080310,'gnl|CDD|280538',_binary 'Putative methyltransferase.  This is a family of putative methyltransferases. The aligned region contains the GXGXG S-AdoMet binding site suggesting a putative methyltransferase activity.','pfam02390','Methyltransf_4'),(1080312,'gnl|CDD|223298',_binary 'Predicted S-adenosylmethionine-dependent methyltransferase [General function prediction only].','COG0220','COG0220'),(1080318,'gnl|CDD|274376',_binary 'RNA polymerase sigma-70 factor, Bacteroides expansion family 1.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are found primarily in the genus Bacteroides. This family appears to have resulted from a lineage-specific expansion as B. thetaiotaomicron VPI-5482, Bacteroides forsythus ATCC 43037, Bacteroides fragilis YCH46 and Bacteroides fragilis NCTC 9343 contain 25, 12, 24 and 23 members, respectively. There are currentlyonly two known members of this family outside of the Bacteroides, in Rhodopseudomonas and Bradyrhizobium.','TIGR02985','Sig70_bacteroi1'),(1081809,'gnl|CDD|129804',_binary 'DNA-binding protein, Tfx family.  PSI-BLAST starting with one member of this family converges with significant hits only to other members of the family, which is restricted to the Archaea. Homology is strongest in the helix-turn-helix-containing N-terminal region. Tfx from Methanobacterium thermoautotrophicum is associated with the operon for molybdenum formyl-methanofuran dehydrogenase and binds a DNA sequence near its promoter. [Regulatory functions, DNA interactions].','TIGR00721','Tfx_protein'),(1082857,'gnl|CDD|235825',_binary 'hypothetical protein; Provisional.','PRK06547','PRK06547'),(1083691,'gnl|CDD|235367',_binary '5-methyltetrahydropteroyltriglutamate--homocysteine S-methyltransferase; Provisional.','PRK05222','PRK05222'),(1084361,'gnl|CDD|236346',_binary 'coproporphyrinogen III oxidase; Provisional.','PRK08898','PRK08898'),(1084829,'gnl|CDD|311648',_binary 'Protein of unknown function (DUF1643).  The members of this family are all sequences found within hypothetical proteins expressed by various bacterial species. The region concerned is approximately 150 residues long.','pfam07799','DUF1643'),(1084830,'gnl|CDD|226783',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4333','COG4333'),(1084843,'gnl|CDD|238427',_binary 'Chalcone and stilbene synthases; plant-specific polyketide synthases (PKS) and related enzymes, also called type III PKSs. PKS generate an array of different products, dependent on the nature of the starter molecule. They share a common chemical strategy, after the starter molecule is loaded onto the active site cysteine, a carboxylative condensation reation extends the polyketide chain. Plant-specific PKS are dimeric iterative PKSs, using coenzyme A esters to deliver substrate to the active site, but they differ in the choice of starter molecule and the number of condensation reactions.','cd00831','CHS_like'),(1084844,'gnl|CDD|225958',_binary 'Predicted naringenin-chalcone synthase [Secondary metabolites biosynthesis, transport, and catabolism].','COG3424','BcsA'),(1084845,'gnl|CDD|178715',_binary 'chalcone synthase-like protein; Provisional.','PLN03171','PLN03171'),(1084846,'gnl|CDD|178714',_binary 'chalcone synthase; Provisional.','PLN03170','PLN03170'),(1084847,'gnl|CDD|178717',_binary 'chalcone synthase; Provisional.','PLN03173','PLN03173'),(1084848,'gnl|CDD|178716',_binary 'chalcone synthase family protein; Provisional.','PLN03172','PLN03172'),(1084849,'gnl|CDD|238423',_binary '\'initiating\' condensing enzymes are a subclass of decarboxylating condensing enzymes, including beta-ketoacyl [ACP] synthase, type III and polyketide synthases, type III, which include chalcone synthase and related enzymes. They are characterized by the utlization of CoA substrate primers, as well as the nature of their active site residues.','cd00827','init_cond_enzymes'),(1084850,'gnl|CDD|178712',_binary 'chalcone synthase; Provisional.','PLN03168','PLN03168'),(1084851,'gnl|CDD|238426',_binary 'Ketoacyl-acyl carrier protein synthase III (KASIII) initiates the elongation in type II fatty acid synthase systems. It is found in bacteria and plants. Elongation of fatty acids in the type II systems occurs by Claisen condensation of malonyl-acyl carrier protein (ACP) with acyl-ACP. KASIII initiates this process by specifically using acetyl-CoA over acyl-CoA.','cd00830','KAS_III'),(1084852,'gnl|CDD|215612',_binary 'chalcone synthase family protein; Provisional.','PLN03169','PLN03169'),(1084853,'gnl|CDD|223409',_binary '3-oxoacyl-[acyl-carrier-protein].','COG0332','FabH'),(1084854,'gnl|CDD|306663',_binary 'Chalcone and stilbene synthases, N-terminal domain.  The C-terminal domain of Chalcone synthase is reported to be structurally similar to domains in thiolase and beta-ketoacyl synthase. The differences in activity are accounted for by differences in this N-terminal domain.','pfam00195','Chal_sti_synt_N'),(1084855,'gnl|CDD|273249',_binary '3-oxoacyl-(acyl-carrier-protein) synthase III.  FabH in general initiate elongation in type II fatty acid synthase systems found in bacteria and plants. The two members of this subfamily from Bacillus subtilis differ from each other, and from FabH from E. coli, in acyl group specificity. Active site residues include Cys112, His244 and Asn274 of E. coli FabH. Cys-112 is the site of acyl group attachment. [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR00747','3-oxoacyl-_synthase_3'),(1084856,'gnl|CDD|225959',_binary '3-hydroxy-3-methylglutaryl CoA synthase [Lipid metabolism].','COG3425','PksG'),(1084857,'gnl|CDD|312149',_binary '3-Oxoacyl-[acyl-carrier-protein (ACP)] synthase III.  This domain is found on 3-Oxoacyl-[acyl-carrier-protein (ACP)] synthase III EC:2.3.1.180, the enzyme responsible for initiating the chain of reactions of the fatty acid synthase in plants and bacteria.','pfam08545','ACP_syn_III'),(1084858,'gnl|CDD|312038',_binary 'FAE1/Type III polyketide synthase-like protein.  The members of this family are described as 3-ketoacyl-CoA synthases, type III polyketide synthases, fatty acid elongases and fatty acid condensing enzymes, and are found in both prokaryotic and eukaryotic (mainly plant) species. The region featured in this family contains the active site residues, as well as motifs involved in substrate binding.','pfam08392','FAE1_CUT1_RppA'),(1084859,'gnl|CDD|236037',_binary '3-oxoacyl-(acyl carrier protein) synthase III; Reviewed.','PRK07515','PRK07515'),(1084860,'gnl|CDD|238201',_binary 'Condensing enzymes; Family of enzymes that catalyze a (decarboxylating or non-decarboxylating) Claisen-like condensation reaction. Members are share strong structural similarity, and are involved in the synthesis and degradation of fatty acids, and the production of polyketides, a diverse group of natural products.','cd00327','cond_enzymes'),(1084861,'gnl|CDD|238421',_binary 'decarboxylating condensing enzymes; Family of enzymes that catalyze the formation of a new carbon-carbon bond by a decarboxylating Claisen-like condensation reaction. Members are involved in the synthesis of fatty acids and polyketides, a diverse group of natural products. Both pathways are an iterative series of additions of small carbon units, usually acetate, to a nascent acyl group. There are 2 classes of decarboxylating condensing enzymes, which can be distinguished by sequence similarity, type of active site residues and type of primer units (acetyl CoA or acyl carrier protein (ACP) linked units).','cd00825','decarbox_cond_enzymes'),(1084862,'gnl|CDD|237245',_binary '3-oxoacyl-(acyl carrier protein) synthase III; Reviewed.','PRK12879','PRK12879'),(1084863,'gnl|CDD|312145',_binary '3-Oxoacyl-[acyl-carrier-protein (ACP)] synthase III C terminal.  This domain is found on 3-Oxoacyl-[acyl-carrier-protein (ACP)] synthase III EC:2.3.1.41, the enzyme responsible for initiating the chain of reactions of the fatty acid synthase in plants and bacteria.','pfam08541','ACP_syn_III_C'),(1084864,'gnl|CDD|236475',_binary '3-oxoacyl-(acyl carrier protein) synthase III; Reviewed.','PRK09352','PRK09352'),(1084865,'gnl|CDD|180328',_binary '3-oxoacyl-(acyl carrier protein) synthase II; Reviewed.','PRK05963','PRK05963'),(1084866,'gnl|CDD|308442',_binary 'Chalcone and stilbene synthases, C-terminal domain.  This domain of chalcone synthase is reported to be structurally similar to domains in thiolase and beta-ketoacyl synthase. The differences in activity are accounted for by differences in the N-terminal domain.','pfam02797','Chal_sti_synt_C'),(1084867,'gnl|CDD|312026',_binary 'Nuclease-related domain.  The nuclease-related domain (NERD) is found in a range of bacterial as well as archaeal and plant proteins. It has distant similarity to endonucleases (hence its name) and its predicted secondary structure is helix - sheet - sheet - sheet - sheet - weak sheet/long loop - helix - sheet - sheet. The majority of NERD-containing proteins are single-domain, but in several cases proteins containing NERD have additional domains which in 75% of cases are involved in DNA processing.','pfam08378','NERD'),(1084869,'gnl|CDD|274393',_binary 'transcriptional regulator EpsA.  Proteins in this family include a C-terminal LuxR transcriptional regulator domain (pfam00196). These proteins are positioned proximal to either EpsH-containing exopolysaccharide biosynthesis operons of the Methylobacillus type, or the associated PEP-CTERM-containing genes.','TIGR03020','Transcriptional_regulator'),(1084872,'gnl|CDD|225360',_binary 'DNA-binding HTH domain-containing proteins [Transcription].','COG2771','CsgD'),(1084874,'gnl|CDD|225461',_binary 'ATP-dependent transcriptional regulator [Transcription].','COG2909','MalT'),(1084875,'gnl|CDD|182241',_binary 'DNA-binding transcriptional regulator CsgD; Provisional.','PRK10100','PRK10100'),(1084876,'gnl|CDD|182431',_binary 'transcriptional regulator NarP; Provisional.','PRK10403','PRK10403'),(1084877,'gnl|CDD|132580',_binary 'LuxR family transcriptional regulatory, chaperone HchA-associated.  Members of this protein family belong to the LuxR transcriptional regulator family, and contain both autoinducer binding (pfam03472) and transcriptional regulator (pfam00196) domains. Members, however, occur only in a few members of the Gammaproteobacteria that have the chaperone/aminopeptidase HchA, and are always encoded by the adjacent gene.','TIGR03541','reg_near_HchA'),(1084878,'gnl|CDD|185267',_binary 'two component system sensor kinase SsrB; Provisional.','PRK15369','PRK15369'),(1084888,'gnl|CDD|310110',_binary 'Protein of unknown function (DUF721).  This family contains several actinomycete proteins of unknown function.','pfam05258','DUF721'),(1084914,'gnl|CDD|307965',_binary 'D12 class N6 adenine-specific DNA methyltransferase.','pfam02086','MethyltransfD12'),(1084915,'gnl|CDD|273146',_binary 'DNA adenine methylase (dam).  All proteins in this family for which functions are known are DNA-adenine methyltransferases. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). The DNA adenine methylase (dam) of E. coli and related species is instrumental in distinguishing the newly synthesized strand during DNA replication for methylation-directed mismatch repair. This family includes several phage methylases and a number of different restriction enzyme chromosomal site-specific modification systems. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00571','DNA_adenine_methylase'),(1084925,'gnl|CDD|308714',_binary 'Protein of unknown function DUF262.','pfam03235','DUF262'),(1084949,'gnl|CDD|176551',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE3 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE3 (also known as glycerophosphodiester phosphodiesterase domain-containing protein 2 (GDPD2), Osteoblast differentiation promoting factor) and their metazoan homologs. Mammalian GDE3 is a transmembrane protein specifically expressed in bone tissues and spleen. It is a mammalian homolog of bacterial glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46), which catalyzes the hydrolysis of various glycerophosphodiesters, and produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. Mammalian GDE3 has been characterized as glycerophosphoinositol inositolphosphodiesterase (EC 3.1.4.43) that selectively hydrolyzes extracellular glycerophosphoinositol (GPI) to generate inositol 1-phosphate (Ins1P) and glycerol. Mammalian GDE3 functions as an inducer of osteoblast differentiation. It also plays a critical role for actin cytoskeletal modulation. The catalytic activity of GDPD domain is essential for mammalian GDE3 cellular function.','cd08609','GDPD_GDE3'),(1084950,'gnl|CDD|176550',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE2 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian GDE2 (also known as glycerophosphodiester phosphodiesterase domain-containing protein 5 (GDPD5)) and their metazoan homologs. Mammalian GDE2 is transmembrane protein primarily expressed in mature neurons. It is a mammalian homolog of bacterial glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46), which catalyze the hydrolysis of various glycerophosphodiesters, and produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. Mammalian GDE2 selectively hydrolyzes glycerophosphocholine (GPC) and has been characterized as GPC-GDE (EC 3.1.4.2) that contributes to osmotic regulation of cellular GPC. Mammalian GDE2 functions in a complex with an antioxidant scavenger peroxiredoxin1 (Prdx1) to control motor neuron differentiation in the spinal cord. Mammalian GDE2 also plays a critical role for retinoid-induced neuronal outgrowth. The catalytic activity of GDPD domain is essential for mammalian GDE2 cellular function.','cd08608','GDPD_GDE2'),(1084952,'gnl|CDD|176510',_binary 'Glycerophosphodiester phosphodiesterase domain of putative Silicibacter pomeroyi glycerophosphodiester phosphodiesterase and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized bacterial glycerophosphodiester phosphodiesterases (GP-GDE, EC 3.1.4.46) and similar proteins. The prototype of this CD is a putative GP-GDE from Silicibacter pomeroyi (SpGDE). It shows high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08567','GDPD_SpGDE_like'),(1084989,'gnl|CDD|240045',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U=I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04689','Nudix_Hydrolase_30'),(1084995,'gnl|CDD|239522',_binary 'GDP-mannose glycosyl hydrolase (AKA GDP-mannose mannosyl hydrolase (GDPMH)) is a member of the Nudix hydrolase superfamily. This class of enzymes is unique from other members of the superfamily in two aspects. First, it contains a modified Nudix signature sequence. The slight changes to the conserved sequence motif, GX5EX7REUXEEXGU, where U = I, L or V), are believed to contribute to the removal of all magnesium binding sites but one, retaining only the metal site that coordinates the pyrophosphate of the substrate. Secondly, it is not a pyrophosphatase that substitutes at a phosphorus; instead, it hydrolyzes nucleotide sugars such as GDP-mannose to GDP and mannose, cleaving the phosphoglycosyl bond by substituting at a carbon position. GDP-mannose provides mannosyl components for cell wall synthesis and is required for the synthesis of other glycosyl donors (such as GDP-fucose and colitose) for the cell wall. The importance of GDP-sugar hydrolase activities is thus closely related to the regulation of cell wall biosynthesis. Enzymes in this family are believed to regulate the concentration of GDP-mannose and GDP-glucose in the bacterial cell wall.','cd03430','GDPMH'),(1084999,'gnl|CDD|240050',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04694','Nudix_Hydrolase_35'),(1085003,'gnl|CDD|215184',_binary 'nudix hydrolase.','PLN02325','PLN02325'),(1085006,'gnl|CDD|240043',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04687','Nudix_Hydrolase_28'),(1085016,'gnl|CDD|185369',_binary 'nucleoside triphosphatase NudI; Provisional.','PRK15472','PRK15472'),(1085018,'gnl|CDD|240048',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04692','Nudix_Hydrolase_33'),(1085020,'gnl|CDD|183303',_binary 'adenosine nucleotide hydrolase NudE; Provisional.','PRK11762','nudE'),(1085026,'gnl|CDD|239644',_binary 'mRNA decapping enzyme 2 (Dcp2p), the catalytic subunit, and Dcp1p are the two components of the decapping enzyme complex. Decapping is a key step in both general and nonsense-mediated 5\'->3\' mRNA-decay pathways. Dcp2p contains an all-alpha helical N-terminal domain and a C-terminal domain which has the Nudix fold. While decapping is not dependent on the N-terminus of Dcp2p, it does affect its efficiency. Dcp1p binds the N-terminal domain of Dcp2p stimulating the decapping activity of Dcp2p. Decapping permits the degradation of the transcript and is a site of numerous control inputs. It is responsible for nonsense-mediated decay as well as AU-rich element (ARE)-mediated decay. In addition, it may also play a role in the levels of mRNA. Enzymes belonging to the Nudix superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and are recognized by a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V).','cd03672','Dcp2p'),(1085029,'gnl|CDD|311574',_binary 'UTRA domain.  The UbiC transcription regulator-associated (UTRA) domain is a conserved ligand-binding domain that has a similar fold to pfam04345. It is believed to modulate activity of bacterial transcription factors in response to binding small molecules.','pfam07702','UTRA'),(1085030,'gnl|CDD|240051',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04695','Nudix_Hydrolase_36'),(1085031,'gnl|CDD|214869',_binary 'The UbiC transcription regulator-associated (UTRA) domain is a conserved ligand-binding domain.  It has a similar fold to HutC/FarR-like bacterial transcription factors of the GntR family. It is believed to modulate activity of bacterial transcription factors in response to binding small molecules.','smart00866','UTRA'),(1085036,'gnl|CDD|239521',_binary 'NADH pyrophosphatase, a member of the Nudix hydrolase superfamily, catalyzes the cleavage of NADH into reduced nicotinamide mononucleotide (NMNH) and AMP. Like other members of the Nudix family, it requires a divalent cation, such as Mg2+ or Mn2+, for activity. Members of this family are also recognized by the Nudix motif, a highly conserved 23-residue block (GX5EX7REUXEEXGU, where U = I, L or V), that functions as a metal binding and catalytic site. A block of 8 conserved amino acids downstream of the nudix motif is thought to give NADH pyrophosphatase its specificity for NADH. NADH pyrophosphatase forms a dimer.','cd03429','NADH_pyrophosphatase'),(1085038,'gnl|CDD|225373',_binary 'NTP pyrophosphohydrolases containing a Zn-finger, probably nucleic-acid-binding [DNA replication, recombination, and repair].','COG2816','NPY1'),(1085040,'gnl|CDD|239642',_binary 'ADP-ribose pyrophosphatase (ADPRase) catalyzes the hydrolysis of ADP-ribose to AMP and ribose-5-P.  Like other members of the Nudix hydrolase superfamily of enzymes, it is thought to require a divalent cation, such as Mg2+, for its activity. It also contains a 23-residue Nudix motif (GX5EX7REUXEEXGU, where U = I, L or V) which functions as a metal binding site/catalytic site. In addition to the Nudix motif, there are additional conserved amino acid residues, distal from the signature sequence, that correlate with substrate specificity. In humans, there are four distinct ADPRase activities, three putative cytosolic (ADPRase-I, -II, and -Mn) and a single mitochondrial enzyme (ADPRase-m). ADPRase-m is also known as NUDT9. It can be distinugished from the cytosolic ADPRase by a N-terminal target sequence unique to mitochondrial ADPRase. NUDT9 functions as a monomer.','cd03670','ADPRase_NUDT9'),(1085044,'gnl|CDD|240049',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04693','Nudix_Hydrolase_34'),(1085046,'gnl|CDD|235156',_binary 'isopentenyl-diphosphate delta-isomerase; Provisional.','PRK03759','PRK03759'),(1085050,'gnl|CDD|239218',_binary 'Isopentenyl diphosphate (IPP) isomerase, a member of the Nudix hydrolase superfamily, is a key enzyme in the isoprenoid biosynthetic pathway. Isoprenoids comprise a large family of natural products including sterols, carotenoids, dolichols and prenylated proteins. These compounds are synthesized from two precursors: isopentenyl diphosphate (IPP) and dimethylallyl diphosphate (DMAPP). IPP isomerase catalyzes the interconversion of IPP and DMAPP by a stereoselective antarafacial transposition of hydrogen. The enzyme requires one Mn2+ or Mg2+ ion in its active site to fold into an active conformation and also contains the Nudix motif, a highly conserved 23-residue block (GX5EX7REUXEEXGU, where U = I, L or V), that functions as a metal binding and catalytic site. The metal binding site is present within the active site and plays structural and catalytical roles. IPP isomerase is well represented in several bacteria, archaebacteria and eukaryotes, including fungi, mammals and plants. Despite sequence variations (mainly at the N-terminus), the core structure is highly conserved.','cd02885','IPP_Isomerase'),(1085051,'gnl|CDD|240020',_binary 'Members of the Nudix hydrolase superfamily catalyze the hydrolysis of NUcleoside DIphosphates linked to other moieties, X. Enzymes belonging to this superfamily require a divalent cation, such as Mg2+ or Mn2+, for their activity and contain a highly conserved 23-residue nudix motif (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. Substrates of nudix hydrolases include intact and oxidatively damaged nucleoside triphosphates, dinucleoside polyphosphates, nucleotide-sugars and dinucleotide enzymes. These substrates are metabolites or cell signaling molecules that require regulation during different stages of the cell cycle or during periods of stress. In general, the role of the nudix hydrolase is to sanitize the nucleotide pools and to maintain cell viability, thereby serving as surveillance & \'house-cleaning\' enzymes. Substrate specificity is used to define families within the superfamily. Differences in substrate specificity are determined by the N-terminal extension or by residues in variable loop regions. Mechanistically, substrate hydrolysis occurs by a nucleophilic substitution reaction, with variation in the numbers and roles of divalent cations required.','cd04662','Nudix_Hydrolase_5'),(1085831,'gnl|CDD|227944',_binary 'CAS/CSE protein involved in chromosome segregation [Cell division and chromosome partitioning].','COG5657','CSE1'),(1085894,'gnl|CDD|277347',_binary 'Enterobacter aerogenes GpdQ and related proteins, metallophosphatase domain.  GpdQ (glycerophosphodiesterase Q, also known as Rv0805 in Mycobacterium tuberculosis) is a binuclear metallophosphoesterase from Enterobacter aerogenes that catalyzes the hydrolysis of mono-, di-, and triester substrates, including some organophosphate pesticides and products of the degradation of nerve agents.  The GpdQ homolog, Rv0805, has 2\',3\'-cyclic nucleotide phosphodiesterase activity. GpdQ and Rv0805 belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07402','MPP_GpdQ'),(1087953,'gnl|CDD|115174',_binary 'Equine infectious anaemia virus S2 protein.  This family consists of several equine infectious anaemia virus S2 proteins. The function of this family is unknown.','pfam06502','Equine_IAV_S2'),(1088030,'gnl|CDD|132129',_binary 'TIGR03085 family protein.  This family, like pfam07398 and TIGRFAMs family TIGR03084, belongs to the larger set of probable enzymes defined in family TIGR03083. Members are found primarily in the Actinobacteria (Mycobacterium, Streptomyces, etc.). The family is uncharacterized. [Hypothetical proteins, Conserved].','TIGR03085','TIGR03085'),(1088459,'gnl|CDD|132044',_binary 'RNA polymerase sigma factor, TIGR02999 family.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are found in a variety of species including Rhodopirellula baltica which encodes a paralogous group of five.','TIGR02999','Sig-70_X6'),(1088988,'gnl|CDD|197842',_binary 'SANT SWI3, ADA2, N-CoR and TFIIIB\'\' DNA-binding domains.','smart00717','SANT'),(1089445,'gnl|CDD|193541',_binary 'M20 Peptidases Aminoacyclase-1 indole-3-acetic-L-aspartic acid hydrolase from bacteria and archaea.  Peptidase M20 family, Bacterial and archaeal Aminoacyclase-1 indole-3-acetic-L-aspartic acid hydrolase (IAA-Asp hydrolase; IAAspH; IAAH; IAA amidohydrolase; EC 3.5.1.-) subfamily. IAAspH hydrolyzes indole-3-acetyl-N-aspartic acid (IAA or auxin) to indole-3-acetic acid. Genes encoding IAA-amidohydrolases were first cloned from Arabidopsis; ILR1, IAR3, ILL1 and ILL2 encode active IAA- amino acid hydrolases, and three additional amidohydrolase-like genes (ILL3, ILL5, ILL6) have been isolated. In higher plants, the growth regulator indole-3-acetic acid (IAA or auxin) is found both free and conjugated via amide bonding to a variety of amino acids and peptides, and via an ester linkage to carbohydrates. IAA-Asp conjugates are involved in homeostatic control, protection, storing and subsequent use of free IAA. IAA-Asp is also found in some plants as a unique intermediate for entering into IAA non-decarboxylative oxidative pathway. IAA amidohydrolase cleaves the amide bond between the auxin and the conjugated amino acid. Enterobacter agglomerans IAAspH has very strong enzyme activity and substrate specificity towards IAA-Asp, although its substrate affinity is weaker compared to Arabidopsis enzymes of the ILR1 gene family. Enhanced IAA-hydrolase activity has been observed during clubroot disease in Chinese cabbage.','cd05665','M20_Acy1_IAAspH_bact'),(1089891,'gnl|CDD|114011',_binary 'Borrelia P83/100 protein.  This family consists of several Borrelia P83/P100 antigen proteins.','pfam05262','Borrelia_P83'),(1089893,'gnl|CDD|282014',_binary 'Protein of unknown function (DUF390).  This is a family of long proteins currently only found in the rice genome. They have no known function. However they may be some kind of transposable element.','pfam04094','DUF390'),(1090116,'gnl|CDD|225973',_binary 'Predicted restriction endonuclease [Defense mechanisms].','COG3440','COG3440'),(1090143,'gnl|CDD|178337',_binary 'long-chain acyl-CoA synthetase.','PLN02736','PLN02736'),(1090158,'gnl|CDD|238030',_binary 'Histidine kinase-like ATPases; This family includes several ATP-binding proteins for example: histidine kinase, DNA gyrase B, topoisomerases, heat shock protein HSP90, phytochrome-like ATPases and DNA mismatch repair proteins.','cd00075','HATPase_c'),(1090925,'gnl|CDD|314931',_binary 'Spherulation-specific family 4.  This protein is found in bacteria, archaea and eukaryotes. Proteins in this family are typically between 250 and 398 amino acids in length. There is a conserved NPG sequence motif and there are two completely conserved G residues that may be functionally important. Starvation will often induce spherulation - the production of spores - and this process may involve DNA-methylation. Changes in the methylation of spherulin4 are associated with the formation of spherules, but these changes are probably transient. Methylation of the gene accompanies its transcriptional activation, and spherulin4 mRNA is only detectable in late spherulating cultures and mature spherules. It is a spherulation-specific protein.','pfam12138','Spherulin4'),(1091417,'gnl|CDD|116881',_binary 'PrkA AAA domain.  This is a family of PrkA bacterial and archaeal serine kinases approximately 630 residues long. This is the N-terminal AAA domain.','pfam08298','AAA_PrkA'),(1092196,'gnl|CDD|236630',_binary 'putative selenate reductase subunit YgfK; Provisional.','PRK09853','PRK09853'),(1092227,'gnl|CDD|315969',_binary 'Transglycosylase SLT domain.  This family is related to the SLT domain pfam01464.','pfam13406','SLT_2'),(1092230,'gnl|CDD|236754',_binary 'murein hydrolase B; Provisional.','PRK10760','PRK10760'),(1092282,'gnl|CDD|130060',_binary 'integration host factor, alpha subunit.  This protein forms a site-specific DNA-binding heterodimer with the integration host factor beta subunit. It is closely related to the DNA-binding protein HU. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00987','Integration_host_factor_a'),(1092287,'gnl|CDD|138142',_binary 'transcriptional regulator HU subunit alpha; Provisional.','PRK10753','PRK10753'),(1092288,'gnl|CDD|259856',_binary 'DNA-binding proteins similar to HU domains.  This subfamily consists of DNA-binding proteins similar to HU domains. HU is a conserved nucleoid-associated protein (NAP) which binds non-specifically to duplex DNA with a particular preference for targeting nicked and bent DNA. It is highly basic and contributes to chromosomal compaction and maintenance of negative supercoiling, thus often referred to as histone-like protein. HU can induce DNA bends, condense DNA in a fiber and also interact with single stranded DNA. It contains two homologous subunits, alpha and beta, typically forming homodimers (alpha-alpha and beta-beta), except in E. coli and other enterobacteria, which form heterodimers (alpha-beta).','cd13834','HU_like'),(1092289,'gnl|CDD|170612',_binary 'transcriptional regulator HU subunit beta; Provisional.','PRK10664','PRK10664'),(1092310,'gnl|CDD|187572',_binary 'atypical (a) SDRs, subgroup 7.  This subgroup contains atypical SDRs of unknown function. Members of this subgroup have a glycine-rich NAD(P)-binding motif consensus that matches the extended SDRs, TGXXGXXG, but lacks the characteristic active site residues of the SDRs. This subgroup has basic residues (HXXXR) in place of the active site motif YXXXK, these may have a catalytic role. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05262','SDR_a7'),(1092311,'gnl|CDD|275033',_binary 'NAD dependent epimerase/dehydratase, LLPSF_EDH_00030 family.  This clade within the NAD dependent epimerase/dehydratase superfamily (pfam01370) is characterized by inclusion of its members within a cassette of seven distinctive enzymes. These include four genes homologous to the elements of the neuraminic (sialic) acid biosynthesis cluster (NeuABCD), an aminotransferase and a nucleotidyltransferase in addition to the epimerase/dehydratase. Together it is very likely that these enzymes direct the biosynthesis of a nine-carbon sugar analagous to CMP-neuraminic acid. These seven genes form the core of the cassette, although they are often accompanied by additional genes that may further modify the product sugar. Although this cassette is widely distributed in bacteria, the family nomenclature arises from the instance in Leptospira interrogans serovar Lai, str. 56601, where it appears as the 30th gene in the 91-gene lipopolysaccharide biosynthesis cluster.','TIGR04180','NAD-dependent_epimerase/d'),(1092313,'gnl|CDD|178298',_binary 'GDP-D-mannose-3\',5\'-epimerase.','PLN02695','PLN02695'),(1092314,'gnl|CDD|187551',_binary 'UDP-glucose 4 epimerase (G4E), subgroup 3, extended (e) SDRs.  Members of this bacterial subgroup are identified as possible sugar epimerases, such as UDP-glucose 4 epimerase. However, while the NAD(P)-binding motif is fairly well conserved, not all members retain the canonical active site tetrad of the extended SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05240','UDP_G4E_3_SDR_e'),(1092315,'gnl|CDD|177856',_binary 'UDP-glucuronate decarboxylase.','PLN02206','PLN02206'),(1092317,'gnl|CDD|187579',_binary 'NADH dehydrogenase (ubiquinone) 1 alpha subcomplex, subunit 9, 39 kDa, (NDUFA9) -like, atypical (a) SDRs.  This subgroup of extended SDR-like proteins are atypical SDRs. They have a glycine-rich NAD(P)-binding motif similar to the typical SDRs, GXXGXXG, and have the YXXXK active site motif (though not the other residues of the SDR tetrad). Members identified include NDUFA9 (mitochondrial) and putative nucleoside-diphosphate-sugar epimerase. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05271','NDUFA9_like_SDR_a'),(1092324,'gnl|CDD|187672',_binary '3beta-hydroxysteroid dehydrogenase (3b-HSD)-like, subgroup1, extended (e) SDRs.  An uncharacterized subgroup of the 3b-HSD-like extended-SDR family. Proteins in this subgroup have the characteristic active site tetrad and NAD(P)-binding motif of extended-SDRs. 3 beta-HSD catalyzes the oxidative conversion of delta 5-3 beta-hydroxysteroids to the delta 4-3-keto configuration; this activity is essential for the biosynthesis of all classes of hormonal steroids. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid sythase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd09812','3b-HSD_like_1_SDR_e'),(1092328,'gnl|CDD|224011',_binary 'Predicted nucleoside-diphosphate sugar epimerases [Cell envelope biogenesis, outer membrane / Carbohydrate transport and metabolism].','COG1086','COG1086'),(1092330,'gnl|CDD|187573',_binary 'Pseudomonas fluorescens MupV-like, extended (e) SDRs.  This subgroup of extended SDR family domains have the characteristic active site tetrad and a well-conserved NAD(P)-binding motif. This subgroup is not well characterized, its members are annotated as having a variety of putative functions. One characterized member is Pseudomonas fluorescens MupV a protein  involved in the biosynthesis of Mupirocin, a polyketide-derived antibiotic. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05263','MupV_like_SDR_e'),(1092332,'gnl|CDD|223774',_binary 'Predicted nucleoside-diphosphate-sugar epimerases [Cell envelope biogenesis, outer membrane / Carbohydrate transport and metabolism].','COG0702','COG0702'),(1092333,'gnl|CDD|187562',_binary 'CDP-D-glucose 4,6-dehydratase, extended (e) SDRs.  This subgroup contains CDP-D-glucose 4,6-dehydratase, an extended SDR, which catalyzes the conversion of CDP-D-glucose to CDP-4-keto-6-deoxy-D-glucose. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05252','CDP_GD_SDR_e'),(1092335,'gnl|CDD|187554',_binary 'atypical (a) SDRs, subgroup 5.  This subgroup contains atypical SDRs, some of which are identified as putative NAD(P)-dependent epimerases, one as a putative NAD-dependent epimerase/dehydratase. Atypical SDRs are distinct from classical SDRs. Members of this subgroup have a glycine-rich NAD(P)-binding motif that is very similar to the extended SDRs, GXXGXXG, and binds NADP. Generally, this subgroup has poor conservation of the active site tetrad; however, individual sequences do contain matches to the YXXXK active site motif, the upstream Ser, and there is a highly conserved Asp in place of the usual active site Asn throughout the subgroup. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05243','SDR_a5'),(1092336,'gnl|CDD|187553',_binary 'atypical (a) SDRs, subgroup 8.  This subgroup contains atypical SDRs of unknown function. Proteins in this subgroup have a glycine-rich NAD(P)-binding motif consensus that resembles that of the extended SDRs, (GXXGXXG or GGXGXXG), but lacks the characteristic active site residues of the SDRs. A Cys often replaces the usual Lys of the YXXXK active site motif, while the upstream Ser is generally present and Arg replaces the usual Asn. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05242','SDR_a8'),(1092337,'gnl|CDD|163279',_binary 'hopanoid-associated sugar epimerase.  The sequences in this family are members of the pfam01370 superfamily of NAD-dependent epimerases and dehydratases typically acting on nucleotide-sugar substrates. The genes of the family modeled here are generally in the same locus with genes involved in the biosynthesis and elaboration of hopene, the cyclization product of the polyisoprenoid squalene. This gene and its association with hopene biosynthesis in Zymomonas mobilis has been noted in the literature where the gene symbol hpnA was assigned. Hopanoids are known to be components of the plasma membrane and to have polar sugar head groups in Z. mobilis and other species.','TIGR03466','HpnA'),(1092339,'gnl|CDD|178326',_binary 'GDP-4-keto-6-deoxymannose-3,5-epimerase-4-reductase.','PLN02725','PLN02725'),(1092340,'gnl|CDD|308379',_binary 'Polysaccharide biosynthesis protein.  This is a family of diverse bacterial polysaccharide biosynthesis proteins including the CapD protein, WalL protein, mannosyl-transferase, and several putative epimerases (e.g. WbiI).','pfam02719','Polysacc_synt_2'),(1092343,'gnl|CDD|273800',_binary 'TIGR01777 family protein.  This model represents a clade of proteins of unknown function including the E. coli yfcH protein. [Hypothetical proteins, Conserved].','TIGR01777','Hypothetical_UPF0105_prot'),(1092344,'gnl|CDD|187540',_binary 'atypical (a) SDRs, subgroup 3.  These atypical SDR family members of unknown function have a glycine-rich NAD(P)-binding motif consensus that is very similar to the extended SDRs, GXXGXXG.  Generally, this group has poor conservation of the active site tetrad, However, individual sequences do contain matches to the YXXXK active site motif, and generally Tyr or Asn in place of the upstream Ser found in most SDRs. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05229','SDR_a3'),(1092345,'gnl|CDD|316022',_binary 'NAD(P)H-binding.','pfam13460','NAD_binding_10'),(1092347,'gnl|CDD|225462',_binary 'Putative NADH-flavin reductase [General function prediction only].','COG2910','COG2910'),(1092349,'gnl|CDD|279420',_binary '3-beta hydroxysteroid dehydrogenase/isomerase family.  The enzyme 3 beta-hydroxysteroid dehydrogenase/5-ene-4-ene isomerase (3 beta-HSD) catalyzes the oxidation and isomerisation of 5-ene-3 beta-hydroxypregnene and 5-ene-hydroxyandrostene steroid precursors into the corresponding 4-ene-ketosteroids necessary for the formation of all classes of steroid hormones.','pfam01073','3Beta_HSD'),(1092350,'gnl|CDD|235630',_binary 'hypothetical protein; Provisional.','PRK05865','PRK05865'),(1092351,'gnl|CDD|187556',_binary 'atypical (a) SDRs, subgroup 2.  This subgroup contains atypical SDRs, one member is identified as Escherichia coli protein ybjT, function unknown. Atypical SDRs are distinct from classical SDRs. Members of this subgroup have a glycine-rich NAD(P)-binding motif consensus that generally matches the extended SDRs, TGXXGXXG, but lacks the characteristic active site residues of the SDRs. This subgroup has basic residues (HXXXR) in place of the active site motif YXXXK, these may have a catalytic role. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05245','SDR_a2'),(1092352,'gnl|CDD|236649',_binary 'dTDP-glucose 4,6 dehydratase; Provisional.','PRK10084','PRK10084'),(1092353,'gnl|CDD|132628',_binary 'UDP-N-acetylglucosamine 4,6-dehydratase (inverting).  This enzyme catalyzes the first step in the biosynthesis of pseudaminic acid, the conversion of UDP-N-acetylglucosamine to UDP-4-keto-6-deoxy-N-acetylglucosamine. These sequences are members of the broader pfam01073 (3-beta hydroxysteroid dehydrogenase/isomerase family) family.','TIGR03589','PseB'),(1092354,'gnl|CDD|187536',_binary 'aldehyde reductase, flavonoid reductase, and related proteins, extended (e) SDRs.  This subgroup contains aldehyde reductase and flavonoid reductase of the extended SDR-type and related proteins. Proteins in this subgroup have a complete SDR-type active site tetrad and a close match to the canonical extended SDR NADP-binding motif. Aldehyde reductase I (aka carbonyl reductase) is an NADP-binding SDR; it catalyzes  the NADP-dependent  reduction of ethyl 4-chloro-3-oxobutanoate to ethyl (R)-4-chloro-3-hydroxybutanoate. The related flavonoid reductases act in the NADP-dependent reduction of  flavonoids, ketone-containing plant secondary metabolites. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05193','AR_like_SDR_e'),(1092355,'gnl|CDD|274236',_binary 'CDP-glucose 4,6-dehydratase.  Members of this protein family are CDP-glucose 4,6-dehydratase from a variety of Gram-negative and Gram-positive bacteria. Members typically are encoded next to a gene that encodes a glucose-1-phosphate cytidylyltransferase, which produces the substrate, CDP-D-glucose, used by this enzyme to produce CDP-4-keto-6-deoxyglucose. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR02622','CDP-glucose_46-dehydratas'),(1092356,'gnl|CDD|187555',_binary 'biliverdin IX beta reductase (BVR-B, aka flavin reductase)-like proteins; atypical (a) SDRs.  Human BVR-B catalyzes pyridine nucleotide-dependent production of bilirubin-IX beta during fetal development; in the adult BVR-B has flavin and ferric reductase activities. Human BVR-B catalyzes the reduction of FMN, FAD, and riboflavin. Recognition of flavin occurs mostly by hydrophobic interactions, accounting for the broad substrate specificity. Atypical SDRs are distinct from classical SDRs. BVR-B does not share the key catalytic triad, or conserved tyrosine typical of SDRs. The glycine-rich NADP-binding motif of BVR-B is GXXGXXG, which is similar but not identical to the pattern seen in extended SDRs. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05244','BVR-B_like_SDR_a'),(1092357,'gnl|CDD|187538',_binary 'aldehyde reductase, extended (e) SDRs.  This subgroup contains aldehyde reductase of the extended SDR-type and related proteins. Aldehyde reductase I (aka carbonyl reductase) is an NADP-binding SDR; it has an NADP-binding motif consensus that is slightly different from the canonical SDR form and lacks the Asn of the extended SDR active site tetrad. Aldehyde reductase I catalyzes the NADP-dependent  reduction of ethyl 4-chloro-3-oxobutanoate to ethyl (R)-4-chloro-3-hydroxybutanoate. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05227','AR_SDR_e'),(1092358,'gnl|CDD|187671',_binary 'human 3beta-HSD (hydroxysteroid dehydrogenase) and HSD3B1(delta 5-delta 4-isomerase)-like, extended (e) SDRs.  This extended-SDR subgroup includes human 3 beta-HSD/HSD3B1 and C(27) 3beta-HSD/ [3beta-hydroxy-delta(5)-C(27)-steroid oxidoreductase; HSD3B7], and related proteins. These proteins have the characteristic active site tetrad and NAD(P)-binding motif of extended SDRs. 3 beta-HSD catalyzes the oxidative conversion of delta 5-3 beta-hydroxysteroids to the delta 4-3-keto configuration; this activity is essential for the biosynthesis of all classes of hormonal steroids. C(27) 3beta-HSD is a membrane-bound enzyme of the endoplasmic reticulum, it catalyzes the isomerization and oxidation of 7alpha-hydroxylated sterol intermediates, an early step in bile acid biosynthesis. Mutations in the human gene encoding C(27) 3beta-HSD underlie a rare autosomal recessive form of neonatal cholestasis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid sythase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd09811','3b-HSD_HSDB1_like_SDR_e'),(1092359,'gnl|CDD|215029',_binary 'RNA-binding protein; Provisional.','PLN00016','PLN00016'),(1092361,'gnl|CDD|224015',_binary 'Predicted nucleoside-diphosphate sugar epimerase [General function prediction only].','COG1090','COG1090'),(1092362,'gnl|CDD|178484',_binary 'cinnamyl-alcohol dehydrogenase.','PLN02896','PLN02896'),(1092363,'gnl|CDD|178263',_binary '3,8-divinyl protochlorophyllide a 8-vinyl reductase.','PLN02657','PLN02657'),(1092364,'gnl|CDD|187565',_binary 'UDP_sulfoquinovose_synthase (Arabidopsis thaliana SQD1 and related proteins), extended (e) SDRs.  Arabidopsis thaliana UDP-sulfoquinovose-synthase ( SQD1), an extended SDR,  catalyzes the transfer of SO(3)(-) to UDP-glucose in the biosynthesis of plant sulfolipids. Members of this subgroup share the conserved SDR catalytic residues, and a partial match to the characteristic extended-SDR NAD-binding motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05255','SQD1_like_SDR_e'),(1092365,'gnl|CDD|275003',_binary 'UDP-N-acetylglucosamine 4,6-dehydratase/5-epimerase.  The FnlA enzyme is the first step in the biosynthesis of UDP-FucNAc from UDP-GlcNAc in E. coli (along with FnlB and FnlC). The proteins identified by this model include FnlA homologs in the O-antigen clusters of O4, O25, O26, O29 (Shigella D11), O118, O145 and O172 serotype strains, all of which produce O-antigens containing FucNAc (or the further modified FucNAm). A homolog from Pseudomonas aerugiosa serotype O11, WbjB, also involved in the biosynthesis of UDP-FucNAc has been characterized and is now believed to carry out both the initial 4,6-dehydratase reaction and the subsequent epimerization of the resulting methyl group at C-5. A phylogenetic tree of related sequences shows a distinct clade of enzymes involved in the biosynthesis of UDP-QuiNAc (Qui=qinovosamine). This clade appears to be descendant from the common ancestor of the Pseudomonas and E. coli fucose-biosynthesis enzymes. It has been hypothesized that the first step in the biosynthesis of these two compounds may be the same, and thus that these enzymes all have the same function. At present, lacking sufficient confirmation of this, the current model trusted cutoff only covers the tree segment surrounding the E. coli genes. The clades containing the Pseudomonas and QuiNAc biosynthesis enzymes score above the noise cutoff. Immediately below the noise cutoff are enzymes involved in the biosynthesis of UDP-RhaNAc (Rha=rhamnose), which again may or may not produce the same product.','TIGR04130','L-fucosamine_synthetase'),(1092416,'gnl|CDD|278729',_binary '\'Cold-shock\' DNA-binding domain.','pfam00313','CSD'),(1092417,'gnl|CDD|224197',_binary 'Cold shock proteins [Transcription].','COG1278','CspC'),(1092418,'gnl|CDD|239905',_binary 'Cold-Shock Protein (CSP) contains an S1-like cold-shock domain (CSD) that is found in eukaryotes, prokaryotes, and archaea.  CSP\'s include the major cold-shock proteins CspA and CspB in bacteria and the eukaryotic gene regulatory factor Y-box protein. CSP expression is up-regulated by an abrupt drop in growth temperature. CSP\'s are also expressed under normal condition at lower level. The function of cold-shock proteins is not fully understood. They preferentially bind poly-pyrimidine region of single-stranded RNA and DNA.  CSP\'s are thought to bind mRNA and regulate ribosomal translation, mRNA degradation, and  the rate of transcription termination. The human Y-box protein, which contains a CSD, regulates transcription and translation of genes that contain the Y-box sequence in their promoters. This specific ssDNA-binding properties of CSD are required for the binding of Y-box protein to the promoter\'s Y-box sequence, thereby regulating transcription.','cd04458','CSP_CDS'),(1092419,'gnl|CDD|131434',_binary 'cold shock domain protein CspD.  This model represents what appears to be a phylogenetically distinct clade, containing E. coli CspD (SP|P24245) and related proteobacterial proteins within the larger family of cold shock domain proteins described by pfam00313. The gene symbol cspD may have been used idependently for other subfamilies of cold shock domain proteins, such as for B. subtilis CspD. These proteins typically are shorter than 70 amino acids. In E. coli, CspD is a stress response protein induced in stationary phase. This homodimer binds single-stranded DNA and appears to inhibit DNA replication. [DNA metabolism, DNA replication, recombination, and repair, Cellular processes, Adaptations to atypical conditions].','TIGR02381','Cold_shock-like_protein_C'),(1092420,'gnl|CDD|169931',_binary 'cold shock protein CspE; Reviewed.','PRK09507','cspE'),(1092421,'gnl|CDD|182402',_binary 'RNA chaperone/anti-terminator; Provisional.','PRK10354','PRK10354'),(1092422,'gnl|CDD|214633',_binary 'Cold shock protein domain.  RNA-binding domain that functions as a RNA-chaperone in bacteria and is involved in regulating translation in eukaryotes. Contains sub-family of RNA-binding domains in the Rho transcription termination factor.','smart00357','CSP'),(1092423,'gnl|CDD|170841',_binary 'cold shock-like protein CspC; Provisional.','PRK10943','PRK10943'),(1092424,'gnl|CDD|77467',_binary 'cold shock protein CspG; Provisional.','PRK09890','PRK09890'),(1092425,'gnl|CDD|184960',_binary 'cold shock-like protein CspD; Provisional.','PRK14998','PRK14998'),(1092426,'gnl|CDD|77494',_binary 'stationary phase/starvation inducible regulatory protein CspD; Provisional.','PRK09937','PRK09937'),(1092427,'gnl|CDD|185360',_binary 'cold shock-like protein CspF; Provisional.','PRK15463','PRK15463'),(1092428,'gnl|CDD|185361',_binary 'cold shock-like protein CspH; Provisional.','PRK15464','PRK15464'),(1092465,'gnl|CDD|99966',_binary 'Trehalose phosphorylase (TP) reversibly catalyzes trehalose synthesis and degradation from alpha-glucose-1-phosphate (alpha-Glc-1-P) and glucose. The catalyzing activity includes the phosphorolysis of trehalose, which produce alpha-Glc-1-P and glucose, and the subsequent synthesis of trehalose. This family is most closely related to the GT1 family of glycosyltransferases.','cd03792','GT1_Trehalose_phosphoryla'),(1092487,'gnl|CDD|189004',_binary 'Peptidase M4 family includes thermolysin, protealysin, aureolysin and neutral protease.  This peptidase M4 family includes several endopeptidases such as thermolysin (EC 3.4.24.27), aureolysin (the extracellular metalloproteinase from Staphylococcus aureus), neutral protease from Bacillus cereus and protealysin. These enzymes have a two-domain structure with the active site between the domains. The N-terminal domain contains the HEXXH zinc-binding motif while the helical C-terminal domain, which is unique for the family, carries the third zinc ligand. Most of these secreted proteases degrade extracellular proteins and peptides for bacterial nutrition, especially prior to sporulation. They have N-terminal propeptides that assist in folding and are removed autocatalytically. Thermolysin is widely used as a nonspecific protease to obtain fragments for peptide sequencing. It has also been used in production of the artificial sweetener aspartame.','cd09597','M4_neutral_protease'),(1092488,'gnl|CDD|225768',_binary 'Zinc metalloprotease (elastase) [Amino acid transport and metabolism].','COG3227','LasB'),(1092489,'gnl|CDD|308486',_binary 'Thermolysin metallopeptidase, alpha-helical domain.','pfam02868','Peptidase_M4_C'),(1092490,'gnl|CDD|223288',_binary 'Superfamily I DNA and RNA helicases [DNA replication, recombination, and repair].','COG0210','UvrD'),(1092491,'gnl|CDD|188992',_binary 'Peptidase M4 family includes thermolysin, pseudolysin, aureolysin, neutral protease and bacillolysin.  M4 Peptidase family belong to the Glu-zincins family and includes numerous zinc-dependent metallopeptidases that hydrolyze peptide bonds such as thermolysin (EC 3.4.24.27), pseudolysin (the extracellullar elastase of Pseudomonas aeruginosa), aureolysin (the extracellular metalloproteinase from Staphylococcus aureus), neutral protease from Bacillus cereus as well as fungalysin (from Aspergillus fumigates) and bacillolysin (EC 3.4.24.28). Typically, the M4 peptidases consist of a presequence (signal sequence), a propeptide sequence and a peptidase unit. The presequence is cleaved off during export while the propeptide has inhibitory and chaperone functions and facilitates folding. The propeptide remains attached until the peptidase is secreted and can be safely activated. All peptidases in this family bind a single catalytic zinc ion which is tetrahedrally co-ordinated by three amino acid ligands and a water molecule that forms the nucleophile on activation during catalysis. These peptidases are secreted eubacterial endopeptidases from Gram-positive or Gram-negative sources that degrade extracellular proteins and peptides for bacterial nutrition. They contain HEXXH and EXXXD motifs as part of their active site and are selectively inhibited by Steptomyces metalloproteinase inhibitor (SMPI) as well as by phosphoramidon from Streptomyces tanashiensis. A large number of these enzymes are implicated as key factors in the pathogenesis of various diseases, including  gastritis, peptic ulcer, gastric carcinoma, cholera and several types of bacterial infections, and are therefore important drug targets. Some enzymes of the family can function at extremes of temperatures, while some function in organic solvents, thus rendering them novel targets for biotechnological applications.','cd02699','M4'),(1092494,'gnl|CDD|306951',_binary 'UvrD/REP helicase N-terminal domain.  The Rep family helicases are composed of four structural domains. The Rep family function as dimers. REP helicases catalyze ATP dependent unwinding of double stranded DNA to single stranded DNA. Some members have large insertions near to the carboxy-terminus relative to other members of the family.','pfam00580','UvrD-helicase'),(1092495,'gnl|CDD|307552',_binary 'Thermolysin metallopeptidase, catalytic domain.','pfam01447','Peptidase_M4'),(1092496,'gnl|CDD|273172',_binary 'exodeoxyribonuclease V, beta subunit.  The RecBCD holoenzyme is a multifunctional nuclease with potent ATP-dependent exodeoxyribonuclease activity. Ejection of RecD, as occurs at chi recombinational hotspots, cripples exonuclease activity in favor of recombinagenic helicase activity. All proteins in this family for which functions are known are DNA-DNA helicases that are used as part of an exonuclease-helicase complex (made up of RecBCD homologs) that function to generate substrates for the initiation of recombination and recombinational repair. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00609','RecBCD_enzyme_subunit_Rec'),(1092497,'gnl|CDD|189001',_binary 'Peptidase Gluzincin family (thermolysin-like proteinases, TLPs) includes peptidases M1, M2, M3, M4, M13, M32 and M36 (fungalysins).  Gluzincin family (thermolysin-like peptidases or TLPs) includes several zinc-dependent metallopeptidases such as the M1, M2, M3, M4, M13, M32, M36 peptidases (MEROPS classification), and contain HEXXH and EXXXD motifs as part of their active site. All peptidases in this family bind a single catalytic zinc ion which is tetrahedrally co-ordinated by three amino acid ligands and a water molecule that forms the nucleophile on activation during catalysis. M1 family includes aminopeptidase N (APN) and leukotriene A4 hydrolase (LTA4H).  APN preferentially cleaves neutral amino acids from the N-terminus of oligopeptides and is present in a variety of human tissues and cell types. LTA4H is a bifunctional enzyme, possessing an aminopeptidase as well as an epoxide hydrolase activity such that the two activities occupy different, but overlapping sites. The peptidase M3 or neurolysin-like family, includes M3, M2 and M32 metallopeptidases.  The M3 peptidases have two subfamilies: M3A, includes thimet oligopeptidase (TOP; endopeptidase 3.4.24.15), neurolysin (3.4.24.16), and the mitochondrial intermediate peptidase; M3B contains oligopeptidase F. M2 peptidase angiotensin converting enzyme (ACE, EC 3.4.15.1) catalyzes the conversion of decapeptide angiotensin I to the potent vasopressor octapeptide angiotensin II. ACE is a key part of the renin-angiotensin system that regulates blood pressure, thus ACE inhibitors are important for the treatment of hypertension. M32 family includes two eukaryotic enzymes from protozoa Trypanosoma cruzi, a causative agent of Chagas\' disease, and Leishmania major, a parasite that causes leishmaniasis, making them attractive targets for drug development. The M4 family includes secreted protease  thermolysin (EC 3.4.24.27), pseudolysin, aureolysin, neutral protease as well as fungalysin and bacillolysin (EC 3.4.24.28) that degrade extracellular proteins and peptides for bacterial nutrition, especially prior to sporulation. Thermolysin is widely used as a nonspecific protease to obtain fragments for peptide sequencing as well as in production of the artificial sweetener aspartame. M13 family includes neprilysin (EC 3.4.24.11) and endothelin-converting enzyme I (ECE-1, EC 3.4.24.71), which fulfill a broad range of physiological roles due to the greater variation in the S2\' subsite allowing substrate specificity and are prime therapeutic targets for selective inhibition. Peptidase M36 (fungamysin) family includes endopeptidases from pathogenic fungi. Fungalysin hydrolyzes extracellular matrix proteins such as elastin and keratin. Aspergillus fumigatus causes the pulmonary disease aspergillosis by invading the lungs of immuno-compromised animals and secreting fungalysin that possibly breaks down proteinaceous structural barriers.','cd09594','GluZincin'),(1092499,'gnl|CDD|189003',_binary 'Peptidase M36 family also known as fungalysin family.  Peptidase M36 protease family, also known as fungalysin (elastinolytic metalloproteinase) family, includes endopeptidases from pathogenic fungi. Fungalysin can hydrolyze extracellular matrix proteins such as elastin and keratin, with a preference for cleavage on the amino side of hydrophobic residues with bulky side-chains. This family is similar to the M4 (thermolysin) family due to the presence of the active site residues in HEXXH and EXXXD motifs, as well as its fold prediction. Some of these enzymes also contain a protease-associated (PA) domain insert. The eukaryotic M36 and bacterial M4 families of metalloproteases also share a conserved domain in their propeptides called FTP (fungalysin/thermolysin propeptide). Aspergillus fumigatus causes the pulmonary disease aspergillosis by invading the lungs of immuno-compromised animals. It secretes fungalysin that possibly breaks down proteinaceous structural barriers. A solid lesion known as an aspergilloma can grow in a lung cavity, particularly following recovery from tuberculosis.','cd09596','M36'),(1092500,'gnl|CDD|130146',_binary 'ATP-dependent DNA helicase Rep.  Designed to identify rep members of the uvrD/rep subfamily. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01074','ATP-dependent_DNA_helicas'),(1092501,'gnl|CDD|316096',_binary 'UvrD-like helicase C-terminal domain.  This domain is found at the C-terminus of a wide variety of helicase enzymes. This domain has a AAA-like structural fold.','pfam13538','UvrD_C_2'),(1092502,'gnl|CDD|273429',_binary 'ATP-dependent DNA helicase PcrA.  Designed to identify pcrA members of the uvrD/rep subfamily. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01073','ATP-dependent_DNA_helicas'),(1092503,'gnl|CDD|236784',_binary 'exonuclease V subunit beta; Provisional.','PRK10876','recB'),(1092504,'gnl|CDD|309442',_binary 'LexA-binding, inner membrane-associated putative hydrolase.  YdjM is a family of putative LexA-binding proteins. Members are predicted to be membrane-bound metal-dependent hydrolases that may be acting as phospholipases. It is a member of the SOS network, that rescues cells from UV and other DNA-damage. Expression of YdjM is regulated by LexA.','pfam04307','YdjM'),(1092505,'gnl|CDD|183256',_binary 'inner membrane protein; Provisional.','PRK11648','PRK11648'),(1092699,'gnl|CDD|316844',_binary 'Abortive infection C-terminus.  This domain is found at the C-terminus of the Lactococcus lactis abortive infection protein Abi-859. This protein confers bacteriophage resistance.','pfam14355','Abi_C'),(1093487,'gnl|CDD|143437',_binary 'Bacillus subtilis NAD+-dependent betaine aldehyde dehydrogenase-like.  Included in this CD is the NAD+-dependent, betaine aldehyde dehydrogenase (BADH, GbsA, EC=1.2.1.8) of Bacillus subtilis involved in the synthesis of the osmoprotectant glycine betaine from choline or glycine betaine aldehyde.','cd07119','ALDH_BADH-GbsA'),(1093689,'gnl|CDD|273921',_binary 'Fe-S cluster assembly protein NifU.  Three different but partially homologous Fe-S cluster assembly systems have been described: Isc, Suf, and Nif. The latter is associated with donation of an Fe-S cluster to nitrogenase in a number of nitrogen-fixing species. NifU, described here, consists of an N-terminal domain (pfam01592) and a C-terminal domain (pfam01106). Homologs with an equivalent domain archictecture from Helicobacter and Campylobacter, however, are excluded from this model by a high trusted cutoff. The model, therefore, is specific for NifU involved in nitrogenase maturation. The related model TIGR01999 homologous to the N-terminus of this model describes IscU from the Isc system as in E. coli, Saccharomyces cerevisiae, and Homo sapiens. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other, Central intermediary metabolism, Nitrogen fixation].','TIGR02000','Nitrogen_fixation_protein'),(1093819,'gnl|CDD|211591',_binary 'DNA mismatch endonuclease Vsr.  All proteins in this family for which functions are known are G:T mismatch endonucleases that function in a specialized mismatch repair process used usually to repair G:T mismatches in specific sections of the genome. This family was based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). Members of this family typically are found near to a DNA cytosine methyltransferase. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00632','HpaII_very_short_patch_re'),(1093820,'gnl|CDD|238136',_binary 'Very Short Patch Repair (Vsr) Endonuclease. Endonucleases in DNA repair that recognize damaged DNA and cleave the phosphodiester backbone. Vsr endonucleases have a common endonuclease topology that has been tailored for recognition of TG mismatches.','cd00221','Vsr'),(1093821,'gnl|CDD|225408',_binary 'Very-short-patch-repair endonuclease [Replication, recombination,    and repair].','COG2852','COG2852'),(1093822,'gnl|CDD|238509',_binary 'Domain of unknown function, appears to be related to a diverse group of endonucleases.','cd01038','Endonuclease_DUF559'),(1093823,'gnl|CDD|282351',_binary 'Protein of unknown function (DUF559).','pfam04480','DUF559'),(1093824,'gnl|CDD|226250',_binary 'DNA G:T-mismatch repair endonuclease [DNA replication, recombination, and repair].','COG3727','Vsr'),(1093825,'gnl|CDD|238508',_binary 'Superfamily of nucleases including Short Patch Repair (Vsr) Endonucleases, archaeal Holliday junction resolvases, MutH methy-directed DNA mismatch-repair endonucleases, and catalytic domains of many restriction endonucleases, such as EcoRI, BamHI, and FokI.','cd01037','Restriction_endonuclease_'),(1093871,'gnl|CDD|152957',_binary 'Protein of unknown function (DUF3725).  This domain family is found in viruses, and is approximately 70 amino acids in length. The family is found in association with pfam01577. There is a conserved FLE sequence motif.','pfam12523','DUF3725'),(1093987,'gnl|CDD|283558',_binary 'Prophage CP4-57 regulatory protein (AlpA).  This family consists of several short bacterial and phage proteins which are related to the E. coli protein AlpA. AlpA suppress two phenotypes of a delta lon protease mutant, overproduction of capsular polysaccharide and sensitivity to UV light. Several of the sequences in this family are thought to be DNA-binding proteins.','pfam05930','Phage_AlpA'),(1094180,'gnl|CDD|306740',_binary 'Pyridoxal-phosphate dependent enzyme.  Members of this family are all pyridoxal-phosphate dependent enzymes. This family includes: serine dehydratase EC:4.2.1.13 P20132, threonine dehydratase EC:4.2.1.16, tryptophan synthase beta chain EC:4.2.1.20, threonine synthase EC:4.2.99.2, cysteine synthase EC:4.2.99.8 P11096, cystathionine beta-synthase EC:4.2.1.22, 1-aminocyclopropane-1-carboxylate deaminase EC:4.1.99.4.','pfam00291','PALP'),(1094181,'gnl|CDD|172537',_binary '1-aminocyclopropane-1-carboxylate deaminase; Provisional.','PRK14045','PRK14045'),(1094182,'gnl|CDD|107202',_binary 'Tryptophan synthase beta superfamily (fold type II); this family of pyridoxal phosphate (PLP)-dependent enzymes catalyzes beta-replacement and beta-elimination reactions. This CD corresponds to aminocyclopropane-1-carboxylate deaminase (ACCD), tryptophan synthase beta chain (Trp-synth_B), cystathionine beta-synthase (CBS), O-acetylserine sulfhydrylase (CS), serine dehydratase (Ser-dehyd), threonine dehydratase (Thr-dehyd), diaminopropionate ammonia lyase (DAL), and threonine synthase (Thr-synth). ACCD catalyzes the conversion of 1-aminocyclopropane-1-carboxylate  to alpha-ketobutyrate and ammonia. Tryptophan synthase folds into a tetramer, where the beta chain is the catalytic PLP-binding subunit and catalyzes the formation of L-tryptophan from indole and L-serine. CBS is a tetrameric hemeprotein that catalyzes condensation of serine and homocysteine to cystathionine. CS is a homodimer that catalyzes the formation of L-cysteine from O-acetyl-L-serine. Ser-dehyd catalyzes the conversion of L- or D-serine  to pyruvate and ammonia. Thr-dehyd is active as a homodimer and catalyzes the conversion of L-threonine to 2-oxobutanoate and ammonia. DAL is also a homodimer and catalyzes the alpha, beta-elimination reaction of both L- and D-alpha, beta-diaminopropionate to form pyruvate and ammonia. Thr-synth catalyzes the formation of threonine and inorganic phosphate from O-phosphohomoserine.','cd00640','Trp-synth-beta_II'),(1094589,'gnl|CDD|214615',_binary 'Staphylococcal nuclease homologues.','smart00318','SNc'),(1094594,'gnl|CDD|184841',_binary 'undecaprenyl pyrophosphate synthase; Provisional.','PRK14831','PRK14831'),(1095230,'gnl|CDD|315465',_binary 'Mechanosensitive ion channel porin domain.  The small mechanosensitive channel, MscS, is a part of the turgor-driven solute efflux system that protects bacteria from lysis in the event of osmotic shock. The MscS protein alone is sufficient to form a functional mechanosensitive channel gated directly by tension in the lipid bilayer. The MscS proteins are heptamers of three transmembrane subunits with seven converging M3 domains, and this MscS_porin is towards the N-terminal of the molecules. The high concentration of negative charges at the extracellular entrance of the pore helps select the cations for efflux.','pfam12795','MscS_porin'),(1095333,'gnl|CDD|237112',_binary 'nicotinate phosphoribosyltransferase; Provisional.','PRK12484','PRK12484'),(1096567,'gnl|CDD|223603',_binary 'Adenylylsulfate kinase and related kinases [Inorganic ion transport and metabolism].','COG0529','CysC'),(1096570,'gnl|CDD|238836',_binary 'Deoxyribonucleoside kinase (dNK) catalyzes the phosphorylation of deoxyribonucleosides to yield corresponding monophosphates (dNMPs). This family consists of various deoxynucleoside kinases including deoxyribo- cytidine (EC 2.7.1.74), guanosine (EC 2.7.1.113), adenosine (EC 2.7.1.76), and thymidine (EC 2.7.1.21) kinases. They are key enzymes in the salvage of deoxyribonucleosides originating from extra- or intracellular breakdown of DNA.','cd01673','dNK'),(1096607,'gnl|CDD|239105',_binary 'A sub-family of peptidase family C39. Peptidase family C39 mostly contains bacteriocin-processing endopeptidases from bacteria. The cysteine peptidases in family C39 cleave the \'double-glycine\' leader peptides from the precursors of various bacteriocins (mostly non-lantibiotic). The cleavage is mediated by the transporter as part of the secretion process. Bacteriocins are antibiotic proteins secreted by some species of bacteria that inhibit the growth of other bacterial species. The bacteriocin is synthesized as a precursor with an N-terminal leader peptide, and processing involves removal of the leader peptide by cleavage at a Gly-Gly bond, followed by translocation of the mature bacteriocin across the cytoplasmic membrane. Most endopeptidases of family C39 are N-terminal domains in larger proteins (ABC transporters) that serve both functions. The proposed protease active site is conserved in this sub-family.','cd02425','Peptidase_C39F'),(1096631,'gnl|CDD|182315',_binary 'hypothetical protein; Provisional.','PRK10220','PRK10220'),(1096673,'gnl|CDD|226919',_binary 'Glutaredoxin-related protein [Posttranslational modification, protein turnover, chaperones].','COG4545','COG4545'),(1097249,'gnl|CDD|224095',_binary 'ABC-type proline/glycine betaine transport systems, permease component [Amino acid transport and metabolism].','COG1174','OpuBB'),(1097427,'gnl|CDD|274148',_binary 'precorrin-6Y C5,15-methyltransferase (decarboxylating), CbiT subunit.  This model recognizes the CbiT methylase which is responsible, in part (along with CbiE), for methylating precorrin-6y (or cobalt-precorrin-6y) at both the 5 and 15 positions as well as the concomitant decarbozylation at C-12. In many organisms, this protein is fused to the CbiE subunit. The fused protein, when found in organisms catalyzing the oxidative version of the cobalamin biosynthesis pathway, is called CobL. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR02469','CbiT'),(1097428,'gnl|CDD|225151',_binary 'Precorrin-6B methylase 2 [Coenzyme metabolism].','COG2242','CobL'),(1098521,'gnl|CDD|179525',_binary 'GTP-binding protein Der; Reviewed.','PRK03003','PRK03003'),(1101197,'gnl|CDD|223363',_binary 'Type I restriction-modification system methyltransferase subunit [Defense mechanisms].','COG0286','HsdM'),(1101676,'gnl|CDD|311452',_binary 'Protein of unknown function (DUF1523).','pfam07509','DUF1523'),(1101774,'gnl|CDD|213240',_binary 'ATP-binding cassette domain of eukaryotic SMC2 proteins.  The structural maintenance of chromosomes (SMC) proteins are large (approximately 110 to 170 kDa), and each is arranged into five recognizable domains. Amino-acid sequence homology of SMC proteins between species is largely confined to the amino- and carboxy-terminal globular domains. The amino-terminal domain contains a \'Walker A\' nucleotide-binding domain (GxxGxGKS/T, in the single-letter amino-acid code), which by mutational studies has been shown to be essential in several proteins. The carboxy-terminal domain contains a sequence (the DA-box) that resembles a \'Walker B\' motif, and a motif with homology to the signature sequence of the ATP-binding cassette (ABC) family of ATPases. The sequence homology within the carboxy-terminal domain is relatively high within the SMC1-SMC4 group, whereas SMC5 and SMC6 show some divergence in both of these sequences. In eukaryotic cells, the proteins are found as heterodimers of SMC1 paired with SMC3, SMC2 with SMC4, and SMC5 with SMC6 (formerly known as Rad18).','cd03273','ABC_SMC2_euk'),(1101992,'gnl|CDD|213020',_binary 'Catalytic NodB homology domain of the carbohydrate esterase 4 superfamily.  The carbohydrate esterase 4 (CE4) superfamily mainly includes chitin deacetylases (EC 3.5.1.41), bacterial peptidoglycan N-acetylglucosamine deacetylases (EC 3.5.1.-), and acetylxylan esterases (EC 3.1.1.72), which catalyze the N- or O-deacetylation of substrates such as acetylated chitin, peptidoglycan, and acetylated xylan, respectively. Members in this superfamily contain a NodB homology domain that adopts a deformed (beta/alpha)8 barrel fold, which encompasses a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad, closely associated with the conserved catalytic base (aspartic acid) and acid (histidine) to carry out acid/base catalysis. The NodB homology domain of CE4 superfamily is remotely related to the 7-stranded beta/alpha barrel catalytic domain of the superfamily consisting of family 38 glycoside hydrolases (GH38), family 57 heat stable retaining glycoside hydrolases (GH57), lactam utilization protein LamB/YcsF family proteins, and YdjC-family proteins.','cd10585','CE4_SF'),(1103026,'gnl|CDD|213996',_binary 'Polymerase and Histidinol Phosphatase domain of Chlorobi like.  The PHP (also called histidinol phosphatase-2/HIS2) domain is associated with several types of DNA polymerases, such as PolIIIA and family X DNA polymerases, stand alone histidinol phosphate phosphatases (HisPPases), and a number of uncharacterized protein families. Chlorobi PHP is uncharacterized protein. HisPPase catalyzes the eighth step of histidine biosynthesis, in which L-histidinol phosphate undergoes dephosphorylation to produce histidinol. The HisPPase can be classified into two types: the bifunctional Hisppase found in proteobacteria that belongs to the DDDD superfamily and the monofunctional Bacillus subtilis type that is a member of the PHP family. The PHP domain has four conserved sequence motifs and contains an invariant histidine that is involved in metal ion coordination. The PHP domain of HisPPase is structurally homologous to other members of the PHP family that have a distorted (beta/alpha)7 barrel fold with a trinuclear metal site on the C-terminal side of the barrel.','cd12112','PHP_HisPPase_Chlorobi_lik'),(1106818,'gnl|CDD|214926',_binary 'Primase C terminal 1 (PriCT-1).  This alpha helical domain is found at the C terminal of primases.','smart00942','PriCT_1'),(1106819,'gnl|CDD|223094',_binary 'Adenylosuccinate lyase [Nucleotide transport and metabolism].','COG0015','PurB'),(1106821,'gnl|CDD|273345',_binary 'adenylosuccinate lyase.  This family consists of adenylosuccinate lyase, the enzyme that catalyzes step 8 in the purine biosynthesis pathway for de novo synthesis of IMP and also the final reaction in the two-step sequence from IMP to AMP. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR00928','Adenylosuccinate_lyase'),(1106823,'gnl|CDD|176467',_binary 'Adenylsuccinate lyase (ASL)_like.  This group contains ASL, prokaryotic-type 3-carboxy-cis,cis-muconate cycloisomerase (pCMLE), and related proteins. These proteins are members of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and; the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP). pCMLE catalyzes the cyclization of 3-carboxy-cis,cis-muconate (3CM) to 4-carboxy-muconolactone, in the beta-ketoadipate pathway. ASL deficiency has been linked to several pathologies including psychomotor retardation with autistic features, epilepsy and muscle wasting.','cd01595','Adenylsuccinate_lyase_lik'),(1106825,'gnl|CDD|176464',_binary 'Adenylsuccinate lyase (ASL)_subgroup 1.  This subgroup contains bacterial and archeal proteins similar to ASL, a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and, the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP).','cd01360','Adenylsuccinate_lyase_1'),(1106827,'gnl|CDD|236283',_binary 'adenylosuccinate lyase; Reviewed.','PRK08540','PRK08540'),(1106829,'gnl|CDD|181437',_binary 'adenylosuccinate lyase; Provisional.','PRK08470','PRK08470'),(1106831,'gnl|CDD|176471',_binary 'Adenylsuccinate lyase (ASL)_subgroup 2.  This subgroup contains mainly eukaryotic proteins similar to ASL, a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and, the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP). ASL deficiency has been linked to several pathologies including psychomotor retardation with autistic features, epilepsy and muscle wasting.','cd03302','Adenylsuccinate_lyase_2'),(1106833,'gnl|CDD|176461',_binary 'Lyase class I family; a group of proteins which catalyze similar beta-elimination reactions.  The Lyase class I family contains class II fumarase, aspartase, adenylosuccinate lyase (ASL), argininosuccinate lyase (ASAL), prokaryotic-type 3-carboxy-cis,cis-muconate cycloisomerase (pCMLE), and related proteins. It belongs to the Lyase_I superfamily. Proteins of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits.','cd01334','Lyase_I'),(1106835,'gnl|CDD|180954',_binary 'adenylosuccinate lyase; Provisional.','PRK07380','PRK07380'),(1106836,'gnl|CDD|181000',_binary 'adenylosuccinate lyase; Provisional.','PRK07492','PRK07492'),(1106837,'gnl|CDD|235792',_binary 'adenylosuccinate lyase; Provisional.','PRK06390','PRK06390'),(1106838,'gnl|CDD|236352',_binary 'adenylosuccinate lyase; Provisional.','PRK08937','PRK08937'),(1106839,'gnl|CDD|176469',_binary 'prokaryotic 3-carboxy-cis,cis-muconate cycloisomerase (CMLE)_like.  This subgroup contains pCLME and related proteins, and belongs to the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. CMLE catalyzes the cyclization of 3-carboxy-cis,cis-muconate (3CM) to 4-carboxy-muconolactone in the beta-ketoadipate pathway. This pathway is responsible for the catabolism of a variety of aromatic compounds into intermediates of the citric cycle in prokaryotic and eukaryotic micro-organisms.','cd01597','pCLME'),(1106840,'gnl|CDD|176466',_binary 'Lyase class I_like superfamily: contains the lyase class I family, histidine ammonia-lyase and phenylalanine ammonia-lyase, which catalyze similar beta-elimination reactions.  Lyase class I_like superfamily of enzymes that catalyze beta-elimination reactions and are active as homotetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. This superfamily contains the lyase class I family, histidine ammonia-lyase and phenylalanine ammonia-lyase. The lyase class I family comprises proteins similar to class II fumarase, aspartase, adenylosuccinate lyase, argininosuccinate lyase, and 3-carboxy-cis, cis-muconate lactonizing enzyme which, for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. Histidine or phenylalanine ammonia-lyase catalyze a beta-elimination of ammonia from histidine and phenylalanine, respectively.','cd01594','Lyase_I_like'),(1106841,'gnl|CDD|178440',_binary 'adenylosuccinate lyase.','PLN02848','PLN02848'),(1106842,'gnl|CDD|278629',_binary 'Lyase.','pfam00206','Lyase_1'),(1106843,'gnl|CDD|274128',_binary '3-carboxy-cis,cis-muconate cycloisomerase.  Members of this family are 3-carboxy-cis,cis-muconate cycloisomerase, the enzyme the catalyzes the second step in the protocatechuate degradation to beta-ketoadipate and then to succinyl-CoA and acetyl-CoA. 4-hydroxybenzoate, 3-hydroxybenzoate, and vanillate all can be converted in one step to protocatechuate. All members of the seed alignment for this model were chosen from within protocatechuate degradation operons of at least three genes of the pathway, from genomes with the complete pathway through beta-ketoadipate. [Energy metabolism, Other].','TIGR02426','3-carboxy-ciscis-muconate'),(1107357,'gnl|CDD|227104',_binary 'Predicted membrane protein [Function unknown].','COG4763','COG4763'),(1107359,'gnl|CDD|226122',_binary 'Fucose 4-O-acetylase and related acetyltransferases [Carbohydrate transport and metabolism].','COG3594','NolL'),(1107360,'gnl|CDD|225813',_binary 'Predicted O-acyltransferase [General function prediction only].','COG3274','COG3274'),(1107361,'gnl|CDD|287372',_binary 'Protein of unknown function (DUF2441).  This is a family of highly conserved, predicted, proteins from Bacillus species. The structure forms a homo-dimer. The function is unknown.','pfam10386','DUF2441'),(1107475,'gnl|CDD|316242',_binary 'Transposase DDE domain group 1.  Transposase proteins are necessary for efficient DNA transposition. This domain is a member of the DDE superfamily, which contain three carboxylate residues that are believed to be responsible for coordinating metal ions needed for catalysis.','pfam13701','DDE_Tnp_1_4'),(1107755,'gnl|CDD|312649',_binary 'Alpha-L-arabinofuranosidase B, catalytic.  Members of this family, which are present in fungal alpha-L-arabinofuranosidase B, adopt a beta-sandwich fold similar to that of Concanavalin A-like lectins/glucanase. The beta-sandwich fold consists of two anti-parallel beta-sheets with seven and and six strands, respectively. In addition, there are four helices outside of the beta-strands. The beta-sandwich strands are closely packed and curved with a jelly roll topology, creating a small catalytic pocket. The domain catalyzes the hydrolysis of alpha-1,2-, alpha-1,3- and alpha-1,5-L-arabinofuranosidic bonds in L-arabinose-containing hemicelluloses such as arabinoxylan and L-arabinan.','pfam09206','ArabFuran-catal'),(1110167,'gnl|CDD|212150',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from bacteria and archaea.  This family corresponds to a group of uncharacterized hypothetical proteins from bacteria, showing a high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain.','cd11543','NTP-PPase_u6'),(1110441,'gnl|CDD|224126',_binary 'Distinct helicase family with a unique C-terminal domain including a metal-binding cysteine cluster [General function prediction only].','COG1205','COG1205'),(1111191,'gnl|CDD|197666',_binary 'helix_turn_helix, arabinose operon control protein.','smart00342','HTH_ARAC'),(1112311,'gnl|CDD|224261',_binary 'Predicted DNA-binding proteins [General function prediction only].','COG1342','COG1342'),(1113158,'gnl|CDD|185705',_binary 'Adenylation domain of RNA circularization proteins.  RNA circularization proteins are capable of circularizing RNA molecules in an ATP-dependent reaction. RNA circularization may protect RNA from exonuclease activity. This model comprises the adenylation domain, the minimal catalytic unit that is common to all members of the ATP-dependent DNA ligase family, and the carboxy-terminal extension of RNA circularization protein that serves as a dimerization module. ATP-dependent polynucleotide ligases catalyze phosphodiester bond formation of nicked nucleic acid substrates using the high energy nucleotide of ATP as a cofactor in a three step reaction mechanism. The adenylation domain binds ATP and contains many active site residues.','cd07894','Adenylation_RNA_ligase'),(1113733,'gnl|CDD|224414',_binary 'Predicted transcriptional regulator [Transcription].','COG1497','COG1497'),(1115324,'gnl|CDD|274346',_binary 'stage VI sporulation protein D.  SpoVID, the stage VI sporulation protein D, is restricted to endospore-forming members of the bacteria, all of which are found among the Firmicutes. It is widely distributed but not quite universal in this group. Between well-conserved N-terminal and C-terminal domains is a poorly conserved, low-complexity region of variable length, rich enough in glutamic acid to cause spurious BLAST search results unless a filter is used. The seed alignment for this model was trimmed, in effect, by choosing member sequences in which these regions are relatively short. SpoVID is involved in spore coat assembly by the mother cell compartment late in the process of sporulation. [Cellular processes, Sporulation and germination].','TIGR02907','Stage_VI_sporulation_prot'),(1115440,'gnl|CDD|114305',_binary 'Vibrio cholerae RfbT protein.  This family consists of several RfbT proteins from Vibrio cholerae. It has been found that genetic alteration of the rfbT gene is responsible for serotype conversion of Vibrio cholerae O1 and determines the difference between the Ogawa and Inaba serotypes, in that the presence of rfbT is sufficient for Inaba-to-Ogawa serotype conversion.','pfam05575','V_cholerae_RfbT'),(1116079,'gnl|CDD|226530',_binary 'Preprotein translocase subunit Sec61beta [Intracellular    trafficking, secretion, and vesicular transport].','COG4043','COG4043'),(1116080,'gnl|CDD|309411',_binary 'ASCH domain.  The ASCH domain adopts a beta-barrel fold similar to the pfam01472 domain. It is thought to function as an RNA-binding domain during coactivation, RNA-processing and possibly during prokaryotic translation regulation.','pfam04266','ASCH'),(1116081,'gnl|CDD|119343',_binary 'ASC-1 homology or ASCH domain, a small beta-barrel domain found in all three kingdoms of life. ASCH resembles the RNA-binding PUA domain and may also interact with RNA. ASCH has been proposed to function as an RNA-binding domain during coactivation, RNA-processing and the regulation of prokaryotic translation. The domain has been named after the ASC-1 protein, the activating signal cointegrator 1 or thyroid hormone receptor interactor protein 4 (TRIP4). ASC-1 is conserved in many eukaryotes and has been suggested to participate in a protein complex that interacts with RNA. It has been shown that ASC-1 mediates the interaction between various transciption factors and the basal transcriptional machinery.','cd06541','ASCH'),(1116082,'gnl|CDD|214979',_binary 'The ASCH domain adopts a beta-barrel fold similar to that of the PUA domain.  It is thought to function as an RNA-binding domain during coactivation, RNA-processing and possibly during prokaryotic translation regulation.','smart01022','ASCH'),(1116092,'gnl|CDD|227388',_binary 'Recombination DNA repair protein (RAD52 pathway) [DNA replication, recombination, and repair].','COG5055','RAD52'),(1116093,'gnl|CDD|129695',_binary 'recombination protein rad52.  All proteins in this family for which functions are known are involved in recombination and recombination repair. Their exact biochemical activity is not yet known.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00607','DNA_repair_protein_RAD52_'),(1116481,'gnl|CDD|215128',_binary 'methyl indole-3-acetate methyltransferase.','PLN02211','PLN02211'),(1116727,'gnl|CDD|223577',_binary 'Adenine/guanine phosphoribosyltransferases and related PRPP-binding proteins [Nucleotide transport and metabolism].','COG0503','Apt'),(1117740,'gnl|CDD|288642',_binary 'Domain of unknown function (DUF3330).  This family of proteins are functionally uncharacterized. This family is only found in bacteria.','pfam11809','DUF3330'),(1117824,'gnl|CDD|214358',_binary 'ribosomal protein L12.','CHL00083','rpl12'),(1118553,'gnl|CDD|131005',_binary 'redox-sensitive transcriptional activator SoxR.  SoxR is a MerR-family homodimeric transcription factor with a 2Fe-2S cluster in each monomer. The motif CIGCGCxxxxxC is conserved. Oxidation of the iron-sulfur cluster activates SoxR. The physiological role in E. coli is response to oxidative stress. It is activated by superoxide, singlet oxygen, nitric oxide (NO), and hydrogen peroxide. In E. coli, SoxR increases expression of transcription factor SoxS; different downstream targets may exist in other species. [Cellular processes, Detoxification, Regulatory functions, DNA interactions].','TIGR01950','HTH-type_transcriptional_'),(1119101,'gnl|CDD|280533',_binary 'N-6 DNA Methylase.  Restriction-modification (R-M) systems protect a bacterial cell against invasion of foreign DNA by endonucleolytic cleavage of DNA that lacks a site specific modification. The R-M system is a complex containing three polypeptides: M (this family), S (pfam01420), and R. This family consists of N-6 adenine-specific DNA methylase EC:2.1.1.72 from Type I and Type IC restriction systems. These methylases have the same sequence specificity as their corresponding restriction enzymes.','pfam02384','N6_Mtase'),(1119103,'gnl|CDD|211578',_binary 'type I restriction system adenine methylase (hsdM).  Function: methylation of specific adenine residues; required for both restriction and modification activities. The ECOR124/3 I enzyme recognizes 5\'GAA(N7)RTCG. for E.coli see (J. Mol. Biol. 257: 960-969 (1996)). [DNA metabolism, Restriction/modification].','TIGR00497','hsdM'),(1119196,'gnl|CDD|128898',_binary 'Ribosomal RNA adenine dimethylases.','smart00650','rADc'),(1119197,'gnl|CDD|223109',_binary 'Dimethyladenosine transferase (rRNA methylation) [Translation, ribosomal structure and biogenesis].','COG0030','KsgA'),(1119902,'gnl|CDD|100003',_binary 'Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility.','cd05844','GT1_like_7'),(1120965,'gnl|CDD|182797',_binary '(p)ppGpp synthetase I/GTP pyrophosphokinase; Provisional.','PRK10872','relA'),(1122124,'gnl|CDD|240090',_binary 'Dihydroorotate dehydrogenase (DHOD) like proteins.  DHOD catalyzes the oxidation of (S)-dihydroorotate to orotate. This is the fourth step and the only redox reaction in the de novo biosynthesis of UMP, the precursor of all pyrimidine nucleotides. DHOD requires FMN as co-factor. DHOD divides into class 1 and class 2 based on their amino acid sequences and cellular location. Members of class 1 are cytosolic enzymes and multimers while class 2 enzymes are membrane associated and monomeric. The class 1 enzymes can be further divided into subtypes 1A and 1B which are homodimers and heterotetrameric proteins, respectively.  This subgroup has the conserved FMN binding site, but lacks some catalytic residues and may therefore be inactive.','cd04739','DHOD_like'),(1123476,'gnl|CDD|309856',_binary 'Dynamitin.  Dynamitin is a subunit of the microtubule-dependent motor complex and in implicated in cell adhesion by binding to macrophage-enriched myristoylated alanine-rice C kinase substrate (MacMARCKS).','pfam04912','Dynamitin'),(1127319,'gnl|CDD|153445',_binary 'Putative Head-Tail Connector Protein Yqbg from Bacillus subtilis and similar proteins.  The uncharacterized Bacillus subtilis Yqbg protein, whose gene is part of the unusual genetic element called skin, shows a similar structure to the connector proteins gp6 and gp15 from bacteriophage HK97 and SPP1, respectively. gp6 and gp15 are critical in the assembly of the connector, a specialized structure that serves as an interface for head and tail attachment, as well as a point at which DNA exits the head during infection by the bacteriophage. They form dodecameric ring structures that comprise the middle ring of the connector, located between the portal protein (attached to the head) and the gp7/gp16 ring (attached to the tail). The Yqbg gene is surrounded with genes similar to genes in the Bacillus subtilis prophage-like element PBSX, which encode for proteins comprising contractile-tailed phage-like particles that are produced upon mitomycin C treatment. Yqbg likely acts as a head-tail connector protein, similar to gp6 and gp15, of the PBSX-like prophage encoded in the skin element.','cd08053','Yqbg'),(1127464,'gnl|CDD|239390',_binary 'Molybdenum is an essential trace element in the form of molybdenum cofactor (Moco) which is associated with the metabolism of nitrogen, carbon and sulfur by redox active enzymes. In E. coli, the synthesis of Moco involves genes from several loci: moa, mob, mod, moe and mog. The mob locus contains mobA and mobB genes. MobB catalyzes the attachment of the guanine dinucleotide to molybdopterin.','cd03116','MobB'),(1127618,'gnl|CDD|270831',_binary 'Catalytic domain of Cyclin-Dependent protein Kinase 4 and 6-like Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. CDK4 and CDK6 partner with D-type cyclins to regulate the early G1 phase of the cell cycle. They are the first kinases activated by mitogenic signals to release cells from the G0 arrested state. CDK4 and CDK6 are both expressed ubiquitously, associate with all three D cyclins (D1, D2 and D3), and phosphorylate the retinoblastoma (pRb) protein. They are also regulated by the INK4 family of inhibitors which associate with either the CDK alone or the CDK/cyclin complex. CDK4 and CDK6 show differences in subcellular localization, sensitivity to some inhibitors, timing in activation, tumor selectivity, and possibly substrate profiles. Although CDK4 and CDK6 seem to show some redundancy, they also have discrete, nonoverlapping functions. CDK6 plays an important role in cell differentiation. CDKs belong to a large family of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. The CDK4/6-like subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd07838','STKc_CDK4_6_like'),(1127622,'gnl|CDD|270852',_binary 'Catalytic domain of the Serine/Threonine Kinase, PFTAIRE-2 kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. PFTAIRE-2 is also referred to as ALS2CR7 (amyotrophic lateral sclerosis 2 (juvenile) chromosome region candidate 7). It may be associated with amyotrophic lateral sclerosis 2 (ALS2), an autosomal recessive form of juvenile ALS. The function of PFTAIRE-2 is not yet known. It shares sequence similarity with Cyclin-Dependent Kinases (CDKs), which belong to a large family of STKs that are regulated by their cognate cyclins. Together, CDKs and cyclins are involved in the control of cell-cycle progression, transcription, and neuronal function. The PFTAIRE-2 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd07870','STKc_PFTAIRE2'),(1127630,'gnl|CDD|270889',_binary 'Catalytic domain of the Serine/Threonine kinase, SH3 Binding Kinase 1.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. SBK1, also called BSK146, is predominantly expressed in the brain. Its expression is increased in the developing brain during the late embryonic stage, coinciding with dramatic neuronal proliferation, migration, and maturation. SBK1 may play an important role in regulating brain development. The SBK1 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd13987','STKc_SBK1'),(1127632,'gnl|CDD|143374',_binary 'Catalytic domain of the Serine/Threonine Kinase, PFTAIRE-1 kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. PFTAIRE-1 is widely expressed except in the spleen and thymus. It is highly expressed in the brain, heart, pancreas, testis, and ovary, and is localized in the cytoplasm. It is regulated by cyclin D3 and is inhibited by the p21 cell cycle inhibitor. It has also been shown to interact with the membrane-associated cyclin Y, which recruits the protein to the plasma membrane. PFTAIRE-1 shares sequence similarity with Cyclin-Dependent Kinases (CDKs), which belong to a large family of STKs that are regulated by their cognate cyclins. Together, CDKs and cyclins are involved in the control of cell-cycle progression, transcription, and neuronal function. The PFTAIRE-1 subfamily is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd07869','STKc_PFTAIRE1'),(1128274,'gnl|CDD|306957',_binary 'SpoU rRNA Methylase family.  This family of proteins probably use S-AdoMet.','pfam00588','SpoU_methylase'),(1128275,'gnl|CDD|223640',_binary 'rRNA methylases [Translation, ribosomal structure and biogenesis].','COG0566','SpoU'),(1128276,'gnl|CDD|129290',_binary 'rRNA methylase, putative, group 3.  this is part of the trmH (spoU) family of rRNA methylases [Protein synthesis, tRNA and rRNA base modification].','TIGR00186','Uncharacterized_tRNA/rRNA'),(1128277,'gnl|CDD|223297',_binary 'Predicted rRNA methylase (SpoU class) [Translation, ribosomal structure and biogenesis].','COG0219','CspR'),(1128278,'gnl|CDD|211559',_binary 'tRNA (cytidine(34)-2\'-O)-methyltransferase.  TrmL (previously YibK) is responsible for 2\'-O-methylation at tRNA(Leu) position 34. [Protein synthesis, tRNA and rRNA base modification].','TIGR00185','Putative_tRNA_cytidine34-'),(1128279,'gnl|CDD|223639',_binary 'rRNA methylase [Translation, ribosomal structure and biogenesis].','COG0565','LasT'),(1128280,'gnl|CDD|236779',_binary 'putative methyltransferase; Provisional.','PRK10864','PRK10864'),(1128281,'gnl|CDD|236837',_binary 'tRNA guanosine-2\'-O-methyltransferase; Provisional.','PRK11081','PRK11081'),(1128295,'gnl|CDD|236749',_binary 'putative methyltransferase; Provisional.','PRK10742','PRK10742'),(1128297,'gnl|CDD|273632',_binary 'helicase, putative, RecD/TraA family.  This model describes a family similar to RecD, the exodeoxyribonuclease V alpha chain of TIGR01447. Members of this family, however, are not found in a context of RecB and RecC and are longer by about 200 amino acids at the amino end. Chlamydia muridarum has both a member of this family and a RecD. [Unknown function, Enzymes of unknown specificity].','TIGR01448','recD_rel'),(1128298,'gnl|CDD|223581',_binary 'ATP-dependent exoDNAse (exonuclease V), alpha subunit - helicase superfamily I member [DNA replication, recombination, and repair].','COG0507','RecD'),(1128299,'gnl|CDD|315827',_binary 'AAA domain.','pfam13245','AAA_19'),(1128301,'gnl|CDD|273631',_binary 'exodeoxyribonuclease V, alpha subunit.  This family describes the exodeoxyribonuclease V alpha subunit, RecD. RecD is part of a RecBCD complex. A related family in the Gram-positive bacteria separates in a phylogenetic tree, has an additional N-terminal extension of about 200 residues, and is not supported as a member of a RecBCD complex by neighboring genes. The related family is consequently described by a different model. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01447','recD'),(1128303,'gnl|CDD|236783',_binary 'exonuclease V subunit alpha; Provisional.','PRK10875','recD'),(1128305,'gnl|CDD|224361',_binary 'Predicted P-loop ATPase fused to an acetyltransferase [General function prediction only].','COG1444','COG1444'),(1128307,'gnl|CDD|131079',_binary 'glutamate formiminotransferase.  This model represents the tetrahydrofolate (THF) dependent glutamate formiminotransferase involved in the histidine utilization pathway. This enzyme interconverts L-glutamate and N-formimino-L-glutamate. The enzyme is bifunctional as it also catalyzes the cyclodeaminase reaction on N-formimino-THF, converting it to 5,10-methenyl-THF and releasing ammonia - part of the process of regenerating THF. This model covers enzymes from metazoa as well as gram-positive bacteria and archaea. In humans, deficiency of this enzyme results in a disease phenotype. The crystal structure of the enzyme has been studied in the context of the catalytic mechanism. [Energy metabolism, Amino acids and amines].','TIGR02024','FtcD'),(1128376,'gnl|CDD|273807',_binary 'TonB-dependent heme/hemoglobin receptor family protein.  This model represents the TonB-dependent outer membrane heme/hemoglobin receptor/transporter found in bacteria which live in contact with animals (which contain hemoglobin or other heme-bearing globins) or legumes (which contain leghemoglobin). Some species having hits to this model such as Nostoc, Caulobacter and Chlorobium do not have an obvious source of hemoglobin-like proteins in their biological niche and so the possibility exists that they act on some other substance. [Transport and binding proteins, Cations and iron carrying compounds, Transport and binding proteins, Porins].','TIGR01785','Heme/hemopexin_utilizatio'),(1129110,'gnl|CDD|222858',_binary 'large terminase protein; Provisional.','PHA02533','17'),(1130039,'gnl|CDD|227893',_binary 'Uncharacterized conserved small protein [Function unknown].','COG5606','COG5606'),(1130107,'gnl|CDD|173947',_binary 'Iron-containing alcohol dehydrogenases-like.  Iron-containing alcohol dehydrogenase-like. Alcohol dehydrogenase catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is belonged to the alcohol dehydrogenase-like superfamily. They are distinct from other alcohol dehydrogenases which contain different protein domains.  Proteins of this family have not been characterized. Their specific function is unknown.','cd08188','Fe-ADH4'),(1130191,'gnl|CDD|238874',_binary 'SGNH_hydrolase subfamily, FeeA, FeeB and similar esterases/lipases. FeeA and FeeB are part of a biosynthetic gene cluster and may participate in the biosynthesis of long-chain N-acyltyrosines by providing saturated and unsaturated fatty acids, which it turn are loaded onto the acyl carrier protein FeeL. SGNH hydrolases are a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01836','FeeA_FeeB_like'),(1131167,'gnl|CDD|238167',_binary 'DEAD-box helicases. A diverse family of proteins involved in ATP-dependent RNA unwinding, needed in a variety of cellular processes including splicing, ribosome biogenesis and RNA degradation. The name derives from the sequence of the Walker  B motif (motif II). This domain contains the ATP- binding region.','cd00268','DEADc'),(1131173,'gnl|CDD|307864',_binary 'Domain of unknown function DUF87.  The function of this prokaryotic domain is unknown. It contains several conserved aspartates and histidines that could be metal ligands.','pfam01935','DUF87'),(1131896,'gnl|CDD|106982',_binary 'hypothetical protein.','PHA00675','PHA00675'),(1132671,'gnl|CDD|227579',_binary 'Predicted membrane protein [Function unknown].','COG5254','ARV1'),(1133293,'gnl|CDD|197714',_binary 'helix_turn_helix, Deoxyribose operon repressor.','smart00420','HTH_DEOR'),(1133295,'gnl|CDD|285436',_binary 'DeoR-like helix-turn-helix domain.','pfam08220','HTH_DeoR'),(1133297,'gnl|CDD|214629',_binary 'helix_turn_helix isocitrate lyase regulation.','smart00346','HTH_ICLR'),(1133299,'gnl|CDD|275137',_binary 'putative metallohydrolase, TIGR04338 family.  This protein family is restricted to the Actinomycetales, including Mycobacterium, Rhodococcus, Nocardia, Gordonii, and others. The invariant motif HEXXH, at the core of the best conserved region in the protein, suggests metallohydrolase activity, as does local sequence similarity in this region to other metallohydrolases.','TIGR04338','hypothetical_protein'),(1133302,'gnl|CDD|203311',_binary 'Cyclin-dependent kinase inhibitor 3 (CDKN3).  This family consists of cyclin-dependent kinase inhibitor 3 or kinase associated phosphatase proteins from several mammalian species. The cyclin-dependent kinase (Cdk)-associated protein phosphatase (KAP) is a human dual specificity protein phosphatase that dephosphorylates Cdk2 on threonine 160 in a cyclin-dependent manner.','pfam05706','CDKN3'),(1134240,'gnl|CDD|226434',_binary 'Signal transduction histidine kinase [Signal transduction mechanisms].','COG3920','COG3920'),(1134479,'gnl|CDD|180928',_binary 'adenine phosphoribosyltransferase; Provisional.','PRK07322','PRK07322'),(1134483,'gnl|CDD|129436',_binary 'orotate phosphoribosyltransferase.  Orotate phosphoribosyltransferase (OPRTase) is involved in the biosynthesis of pyrimidine nucleotides. Alpha-D-ribosyldiphosphate 5-phosphate (PRPP) and orotate are utilized to form pyrophosphate and orotidine 5\'-monophosphate (OMP) in the presence of divalent cations, preferably Mg2+. In a number of eukaryotes, this protein is fused to a domain that catalyses the reaction (EC 4.1.1.23). The combined activity of EC 2.4.2.10 and EC 4.1.1.23 is termed uridine 5\'-monophosphate synthase. The conserved Lys (K) residue at position 101 of the seed alignment has been proposed as the active site for the enzyme. [Purines, pyrimidines, nucleosides, and nucleotides, Pyrimidine ribonucleotide biosynthesis].','TIGR00336','Orotate_phosphoribosyltra'),(1138497,'gnl|CDD|225083',_binary 'Anti-sigma regulatory factor (Ser/Thr protein kinase) [Signal transduction mechanisms].','COG2172','RsbW'),(1138498,'gnl|CDD|179626',_binary 'anti-sigma F factor; Provisional.','PRK03660','PRK03660'),(1138500,'gnl|CDD|223200',_binary '3-methyladenine DNA glycosylase/8-oxoguanine DNA glycosylase [DNA replication, recombination, and repair].','COG0122','AlkA'),(1138512,'gnl|CDD|280400',_binary 'Thymidylate kinase.','pfam02223','Thymidylate_kin'),(1138516,'gnl|CDD|184429',_binary 'thymidylate kinase; Provisional.','PRK13973','PRK13973'),(1138520,'gnl|CDD|172477',_binary 'thymidylate kinase; Provisional.','PRK13974','PRK13974'),(1138522,'gnl|CDD|237574',_binary 'thymidylate kinase; Provisional.','PRK13976','PRK13976'),(1138524,'gnl|CDD|236133',_binary 'thymidylate kinase; Validated.','PRK07933','PRK07933'),(1138526,'gnl|CDD|223963',_binary 'Fe-S oxidoreductase [Energy production and conversion].','COG1032','COG1032'),(1138528,'gnl|CDD|214792',_binary 'Elongator protein 3, MiaB family, Radical SAM.  This superfamily contains MoaA, NifB, PqqE, coproporphyrinogen III oxidase, biotin synthase and MiaB families, and includes a representative in the eukaryotic elongator subunit, Elp-3. Some members of the family are methyltransferases.','smart00729','Elp3'),(1138529,'gnl|CDD|275272',_binary 'radical SAM P-methyltransferase, PhpK family.  Characterized members of this family are B12-binding domain/radical SAM domain enzymes that use methylcobalamin as a methyl donor to methylate a phosphorous atom during the biosynthesis of natural products such as bialaphos and phosalacine. These syntheses create an extremely rare C-P-C bond. All members of the seed alignment derive from genomic regions that include a non-ribosomal peptide synthase. Note that a single organism, Pelosinus fermentans JBW45 from Cr(VI)-contaminated groundwater, has eight additional homologs of unknown function that score between trusted and noise cutoffs of this model. [Cellular processes, Toxin production and resistance].','TIGR04479','bcpD_PhpK_rSAM'),(1138532,'gnl|CDD|132511',_binary 'hopanoid biosynthesis associated radical SAM protein HpnJ.  The sequences represented by this model are members of the radical SAM superfamily of enzymes (pfam04055). These enzymes utilize an iron-sulfur redox cluster and S-adenosylmethionine to carry out diverse radical mediated reactions. The member of this clade from Acidithiobacillus ferrooxidans ATCC 23270 (AFE_0975) is found in the same locus as squalene-hopene cyclase (SHC, TIGR01507) and other genes associated with the biosynthesis of hopanoid natural products. Similarly, in Ralstonia eutropha JMP134 (Reut_B4901) this gene is adjacent to HpnAB, IspH and HpnH (TIGR03470), although SHC itself is elsewhere in the genome. Notably, this gene (here named HpnJ) and three others form a conserved set (HpnIJKL) which occur in a subset of all genomes containing the SHC enzyme. This relationship was discerned using the method of partial phylogenetic profiling. This group includes Zymomonas mobilis, the organism where the initial hopanoid biosynthesis locus was described consisting of the genes HpnA-E and SHC (HpnF). Continuing past SHC are found a phosphorylase enzyme (ZMO0873, i.e. HpnG, TIGR03468) and another radical SAM enzyme (ZMO0874), HpnH. Although discontinuous in Z. mobilis, we continue the gene symbol sequence with HpnIJKL. One of the well-described hopanoid intermediates is bacteriohopanetetrol. In the conversion from hopene several reactions must occur in the side chain for which a radical mechanism might be reasonable. These include the four (presumably anaerobic) hydroxylations and a methyl shift.','TIGR03471','HpnJ'),(1138534,'gnl|CDD|274918',_binary 'B12-binding domain/radical SAM domain protein, MJ_0865 family.  Members of this family have both a B12 binding homology domain (pfam02310) and a radical SAM domain (pfam04055), and occur only once per genome. This protein occurs so far only in methanogenic archaea. Some species with members of this family have a related protein with similar domain architecture (see TIGR04013). [Unknown function, Enzymes of unknown specificity].','TIGR04014','B12_SAM_MJ_0865'),(1138539,'gnl|CDD|274893',_binary 'ribosomal peptide maturation radical SAM protein 1.  Models TIGR03793 and TIGR03798 describe bacteriocin precursor families to occur often in large paralogous families and are subject to various modifications, including by LanM family lantibiotic synthases and by cyclodehydratases. This model represents a radical SAM protein family that regularly occurs in the context of these bacteriocins, and may occur where other familiar peptide modification enzymes are absent. [Cellular processes, Toxin production and resistance].','TIGR03975','hypothetical_protein_PPSI'),(1138542,'gnl|CDD|275178',_binary 'putative variant cofactor biosynthesis B12-binding domain/radical SAM domain protein 1.  Members of this protein family are one of two tandem B12-binding domain/radical SAM domain proteins that occur in a genome context with a pair of homologs to ThiC (phosphomethylpyrimidine synthase, EC 4.1.99.17), an enzyme that performs a complex rearrangement involved in thiamin biosynthesis, and a putative CobT (nicotinate-nucleotide--dimethylbenzimidazole phosphoribosyltransferase), an enzyme of cobalamin biosynthesis.','TIGR04385','radical_SAM_superfamily_p'),(1138545,'gnl|CDD|274917',_binary 'B12-binding domain/radical SAM domain protein, MJ_1487 family.  Members of this family have both a B12 binding homology domain (pfam02310) and a radical SAM domain (pfam04055), and occur only once per genome. Some species with members of this family have a related protein with similar domain architecture. This protein is occurs largely in archaeal methanogens but also in a few bacteria, including Thermotoga maritima and Myxococcus xanthus. [Unknown function, Enzymes of unknown specificity].','TIGR04013','Radical_SAM_domain_protei'),(1138548,'gnl|CDD|131081',_binary 'magnesium-protoporphyrin IX monomethyl ester anaerobic oxidative cyclase.  This model represents the cobalamin-dependent oxidative cyclase, a radical SAM enzyme responsible for forming the distinctive E-ring of the chlorin ring system under anaerobic conditions. This step is essential in the biosynthesis of both bacteriochlorophyll and chlorophyll under anaerobic conditions (a separate enzyme, AcsF, acts under aerobic conditions). This model identifies two clades of sequences, one from photosynthetic, non-cyanobacterial bacteria and another including Synechocystis and several non-photosynthetic bacteria. The function of the Synechocystis gene is supported by gene clustering with other photosynthetic genes, so the purpose of the gene in the non-photosynthetic bacteria is uncertain. Note that homologs of this gene are not found in plants which rely solely on the aerobic cyclase.','TIGR02026','BchE'),(1138551,'gnl|CDD|275160',_binary 'hopanoid C-3 methylase HpnR.  Members of this are family are a B12-binding domain/radical SAM domain protein required for 3-methylhopanoid production. Activity was confirmed by mutant phenotype by disrupting this gene in Methylococcus capsulatus strain Bath. This protein family should only occur in genomes that encode a squalene-hopene cyclase (see TIGR01507). [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR04367','protein_of_the_SAM_superf'),(1138554,'gnl|CDD|275111',_binary 'B12-binding domain/radical SAM domain protein, rhizo-twelve system.  A variety of bacteria, including multiple species of Bradyrhizobium, Mesorhizobium, and Methylobacterium, have a typically twelve-gene cassette (hence the designation rhizo-twelve) for the biosynthesis of some unknown oligosaccaride. This family is a B12-binding domain/radical SAM domain protein found in roughly have of these cassettes, but nowhere else.','TIGR04295','B12_rSAM_oligo'),(1138557,'gnl|CDD|272900',_binary 'radical SAM methylthiotransferase, MiaB/RimO family.  This subfamily contains the tRNA-i(6)A37 modification enzyme, MiaB (TIGR01574). The phylogenetic tree indicates 4 distinct clades, one of which corresponds to MiaB. The other three clades are modelled by hypothetical equivalogs (TIGR01125, TIGR01579 and TIGR01578). Together, the four models hit every sequence hit by the subfamily model without any overlap between them. This subfamily is aparrently a part of a larger superfamily of enzymes utilizing both a 4Fe4S cluster and S-adenosyl methionine (SAM) to initiate radical reactions. MiaB acts on a particular isoprenylated Adenine base of certain tRNAs causing thiolation at an aromatic carbon, and probably also transferring a methyl grouyp from SAM to the thiol. The particular substrate of the three other clades is unknown but may be very closely related.','TIGR00089','Ribosomal_protein_S12_met'),(1138560,'gnl|CDD|184630',_binary '(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional.','PRK14334','PRK14334'),(1138563,'gnl|CDD|211913',_binary 'B12-binding domain/radical SAM domain protein, Ta0216 family.  Members of this family are enzymes with an N-terminal B12-binding domain and central radical SAM domain. Families TIGR03975, TIGR04013 and TIGR04014 exhibit a similar architecture, which may be associated with lipid metabolism.','TIGR04190','Fe-S_oxidoreductase'),(1138569,'gnl|CDD|223694',_binary '2-methylthioadenine synthetase [Translation, ribosomal structure and biogenesis].','COG0621','MiaB'),(1138572,'gnl|CDD|239019',_binary 'B12 binding domain_like associated with radical SAM domain. This domain shows similarity with B12 (adenosylcobamide) binding domains found in several enzymes, such as glutamate mutase, methionine synthase and methylmalonyl-CoA mutase, but it lacks the signature motif Asp-X-His-X-X-Gly, which contains the histidine that acts as a cobalt ligand. The function of this domain remains unclear.','cd02068','radical_SAM_B12_BD'),(1138575,'gnl|CDD|275227',_binary 'B12-binding domain/radical SAM domain protein, Pput_1520 family.  Members of this family are radical SAM domain (pfam04055) enzymes with an N-terminal B12-binding domain (pfam02310), as is fairly common for radical SAM enzymes with lipid substrates. However, both domains as found in this family seem to be long-branch. The function is unknown, but all cases a PLP-dependent enzyme (a cysteine desulfurase homolog) is found nearby.','TIGR04434','hypothetical_protein_D187'),(1138578,'gnl|CDD|308110',_binary 'B12 binding domain.  This domain binds to B12 (adenosylcobamide), it is found in several enzymes, such as glutamate mutase, methionine synthase, and methylmalonyl-CoA mutase. It contains a conserved DxHxxGx(41)SxVx(26)GG motif, which is important for B12 binding.','pfam02310','B12-binding'),(1138582,'gnl|CDD|273700',_binary 'tRNA-N(6)-(isopentenyl)adenosine-37 thiotransferase enzyme MiaB.  This model represents homologs of the MiaB enzyme responsible for the modification of the isopentenylated adenine-37 base of most bacterial and eukaryotic tRNAs that read codons beginning with uracil (all except tRNA(I,V) Ser). Adenine-37 is next to the anticodon on the 3\' side in these tRNA\'s, and lack of modification at this site leads to an increased spontaneous mutation frequency. Isopentenylated A-37 is modified by methylthiolation at position 2, either by MiaB alone or in concert with a separate methylase yet to be discovered (MiaC?). MiaB contains a 4Fe-4S cluster which is labile under oxidizing conditions. Additionally, the sequence is homologous (via PSI-BLAST searches) to the biotin synthetase, BioB, which utilizes both an iron-sulfur cluster and S-adenosym methionine (SAM) to generate a radical which is responsible for initiating the insertion of sulfur into the substrate. It is reasonable to surmise that the methyl group of SAM becomes the methyl group of the product, but this has not been shown, and the possibility of a separate methylase exists. This equivalog is a member of a subfamily (TIGR00089) which contains several other hypothetical equivalogs which are all probably enzymes with similar function acting on different substrates. These enzymes contain a TRAM domain (pfam01938) which is believed to be responsible for binding to tRNAs. Hits to this model span all major groups of bacteria and eukaryotes, but not archaea, which are known to lack this particular tRNA modification. The enzyme from Thermotoga maritima has been cloned, expressed, spectroscopically characterized and shown to complement the E. coli MiaB enzyme. [Protein synthesis, tRNA and rRNA base modification].','TIGR01574','Hypothetical_UPF0004_prot'),(1138583,'gnl|CDD|226604',_binary 'Predicted NTP pyrophosphohydrolase [DNA replication, recombination, and repair / General function prediction only].','COG4119','COG4119'),(1138584,'gnl|CDD|273704',_binary 'MiaB-like tRNA modifying enzyme.  This clade of sequences is closely related to MiaB, a modifier of isopentenylated adenosine-37 of certain eukaryotic and bacterial tRNAs (see TIGR01574). Sequence alignments suggest that this equivalog performs the same chemical transformation as MiaB, perhaps on a different (or differently modified) tRNA base substrate. This clade is a member of a subfamily (TIGR00089) and spans low GC Gram positive bacteria, alpha and epsilon proteobacteria, Campylobacter, Porphyromonas, Aquifex, Thermotoga, Chlamydia, Treponema and Fusobacterium. [Protein synthesis, tRNA and rRNA base modification].','TIGR01579','MiaB-like-C'),(1138587,'gnl|CDD|184628',_binary '(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional.','PRK14331','PRK14331'),(1138590,'gnl|CDD|184627',_binary '(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional.','PRK14330','PRK14330'),(1138592,'gnl|CDD|237673',_binary '(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional.','PRK14325','PRK14325'),(1138594,'gnl|CDD|237676',_binary '(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional.','PRK14329','PRK14329'),(1138596,'gnl|CDD|224163',_binary 'Predicted Fe-S oxidoreductase [General function prediction only].','COG1242','COG1242'),(1138598,'gnl|CDD|273502',_binary 'radical SAM enzyme, TIGR01210 family.  This family of exclusively archaeal radical SAM enzymes has no characterized close homologs. [Hypothetical proteins, Conserved].','TIGR01210','Uncharacterized_protein_M'),(1138630,'gnl|CDD|225477',_binary 'Exonuclease I [DNA replication, recombination, and repair].','COG2925','SbcB'),(1139652,'gnl|CDD|317968',_binary 'Family with sequence similarity 184, A and B.  The function of FAM184 is not known.','pfam15665','FAM184'),(1142803,'gnl|CDD|225606',_binary 'Membrane protein involved in colicin uptake [Cell envelope biogenesis, outer membrane].','COG3064','TolA'),(1142804,'gnl|CDD|310345',_binary 'MAP7 (E-MAP-115) family.  The organisation of microtubules varies with the cell type and is presumably controlled by tissue-specific microtubule-associated proteins (MAPs). The 115-kDa epithelial MAP (E-MAP-115/MAP7) has been identified as a microtubule-stabilizing protein predominantly expressed in cell lines of epithelial origin. The binding of this microtubule associated protein is nucleotide independent.','pfam05672','MAP7'),(1142805,'gnl|CDD|311259',_binary 'Modifier of rudimentary (Mod(r)) protein.  This family represents a conserved region approximately 150 residues long within a number of eukaryotic proteins that show homology with Drosophila melanogaster Modifier of rudimentary (Mod(r)) proteins. The N-terminal half of Mod(r) proteins is acidic, whereas the C-terminal half is basic, and both of these regions are represented in this family. Members of this family include the Vps37 subunit of the endosomal sorting complex ESCRT-I, a complex involved in recruiting transport machinery for protein sorting at the multivesicular body (MVB). The yeast ESCRT-I complex consists of three proteins (Vps23, Vps28 and Vps37). The mammalian homolog of Vps37 interacts with Tsg101 (Pfam: PF05743) through its mod(r) domain and its function is essential for lysosomal sorting of EGF receptors.','pfam07200','Mod_r'),(1142806,'gnl|CDD|317089',_binary 'Domain of unknown function (DUF4455).  This domain family is found in bacteria and eukaryotes, and is approximately 480 amino acids in length. There are two completely conserved residues (W and P) that may be functionally important.','pfam14643','DUF4455'),(1142810,'gnl|CDD|310784',_binary 'Mitochondrial genome maintenance MGM101.  The mgm101 gene was identified as essential for maintenance of the mitochondrial genome in Saccharomyces cerevisiae. Based on its DNA-binding activity, and experimental work with a temperature-sensitive mgm101 mutant, it has been proposed that the mgm101 gene product performs an essential function in the repair of oxidatively damaged mitochondrial DNA.','pfam06420','Mgm101p'),(1143378,'gnl|CDD|227786',_binary 'Predicted transcription regulator containing HTH domain [Transcription].','COG5499','COG5499'),(1144347,'gnl|CDD|309094',_binary 'DNA polymerase II large subunit DP2.','pfam03833','PolC_DP2'),(1144636,'gnl|CDD|225174',_binary 'SAM-dependent methyltransferases related to tRNA (uracil-5-)-methyltransferase [Translation, ribosomal structure and biogenesis].','COG2265','TrmA'),(1144816,'gnl|CDD|227512',_binary 'Protein involved in chromosome segregation, interacts with SMC proteins [Cell division and chromosome partitioning].','COG5185','HEC1'),(1144817,'gnl|CDD|313412',_binary 'RIM-binding protein of the cytomatrix active zone.  This is a family of proteins that form part of the CAZ (cytomatrix at the active zone) complex which is involved in determining the site of synaptic vesicle fusion. The C-terminus is a PDZ-binding motif that binds directly to RIM (a small G protein Rab-3A effector). The family also contains four coiled-coil domains.','pfam10174','Cast'),(1147402,'gnl|CDD|259835',_binary 'Mg-dependent tRNAse of the contact-dependent growth inhibition (CDI) system of Burkholderia pseudomallei 1026b, and related proteins.  CDI toxins are expressed by gram-negative bacteria as part of a mechanism to inhibit the growth of neighboring cells. This model represents the C-terminal toxin domain of CdiA effector proteins. CdiA secretion is dependent on the outer membrane protein CdiB. Upon binding to a receptor on the surface of target bacteria, the CDI toxin is delivered. A wide variety of C-terminal toxin domains appear to exist; this particular example from Burkholderia pseudomallei 1026b and other bacteria appears to function as a Mg2+-dependent RNAse cleaving tRNA, most likely in the aminoacyl acceptor stem.','cd13442','CDI_toxin_Bp1026b_like'),(1147410,'gnl|CDD|308334',_binary 'WhiA C-terminal HTH domain.  This domain is found at the C-terminus of the sporulation regulator WhiA. It is predicted to form a DNA-binding helix-turn-helix structure. The WhiA protein also contains two N-terminal domains that are distant homologs of LAGLIDADG homing endonucleases.','pfam02650','HTH_WhiA'),(1147411,'gnl|CDD|224398',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG1481','COG1481'),(1147413,'gnl|CDD|273194',_binary 'DNA-binding protein WhiA.  This family describes a DNA-binding protein widely conserved in Gram-positive bacteria, and occasionally occurring elsewhere, such as in Thermotoga. It is associated with cell division, and in sporulating organisms with sporulation. [Cellular processes, Cell division].','TIGR00647','DNA_bind_WhiA'),(1148074,'gnl|CDD|214940',_binary 'SRP54-type protein, GTPase domain.  This entry represents the GTPase domain of the 54 kDa SRP54 component, a GTP-binding protein that interacts with the signal sequence when it emerges from the ribosome. SRP54 of the signal recognition particle has a three-domain structure: an N-terminal helical bundle domain, a GTPase domain, and the M-domain that binds the 7s RNA and also binds the signal sequence. The extreme C-terminal region is glycine-rich and lower in complexity and poorly conserved between species. The GTPase domain is evolutionary related to P-loop NTPase domains found in a variety of other proteins.','smart00962','SRP54'),(1148106,'gnl|CDD|225253',_binary 'Predicted transcriptional regulator [Transcription].','COG2378','COG2378'),(1148194,'gnl|CDD|310464',_binary 'Siphovirus protein of unknown function (DUF859).  This family consists of several uncharacterized proteins from the Siphoviruses as well as one bacterial sequence. Some of the members of this family are described as putative minor structural proteins.','pfam05895','DUF859'),(1148262,'gnl|CDD|225891',_binary 'Predicted transcriptional regulator [Transcription].','COG3355','COG3355'),(1149431,'gnl|CDD|182049',_binary 'antitoxin HipB; Provisional.','PRK09726','PRK09726'),(1151485,'gnl|CDD|223943',_binary 'Predicted hydrolase (HAD superfamily) [General function prediction only].','COG1011','COG1011'),(1151486,'gnl|CDD|274056',_binary 'REG-2-like, HAD superfamily (subfamily IA) hydrolase.  This family of proteins includes uncharacterized sequences from eukaryotes, cyanobacteria and Leptospira as well as the DREG-2 protein from Drosophila melanogaster which has been identified as a rhythmically (diurnally) regulated gene. This family is a member of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called \'capping domain\', or the absence of such a domain. This family is a member of subfamily 1A in which the cap domain consists of a predicted alpha helical bundle found in between the first and second catalytic motifs. A distinctive feature of this family is a conserved tandem pair of tryptophan residues in the cap domain. The most divergent sequences included within the scope of this model are from plants and have \'FW\' at this position instead. Most likely, these sequences, like the vast majority of HAD sequences, represent phosphatase enzymes.','TIGR02252','Rhythmically_expressed_ge'),(1151487,'gnl|CDD|319852',_binary 'uncharacterized family of the haloacid dehalogenase-like superfamily, similar to uncharacterized Drosophila melanogaster rhythmically expressed gene 2 protein and human haloacid dehalogenase-like hydrolase domain-containing protein 3.  The haloacid dehalogenase-like (HAD) hydrolases are a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. Members are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd16415','HAD_dREG-2_like'),(1151489,'gnl|CDD|274057',_binary 'HAD superfamily (subfamily IA) hydrolase, TIGR02253.  This family of sequences from archaea and metazoans includes the human uncharacterized protein CTE7. Pyrococcus species appear to have three different forms of this enzyme, so it is unclear whether all members of this family have the same function. This family is a member of the haloacid dehalogenase (HAD) superfamily of hydrolases which are characterized by three conserved sequence motifs. By virtue of an alpha helical domain in-between the first and second conserved motif, this family is a member of subfamily IA (TIGR01549).','TIGR02253','Uncharacterized_HAD-hydro'),(1151490,'gnl|CDD|319787',_binary 'L-2-haloacid dehalogenase.  L-2-Haloacid dehalogenase catalyzes the hydrolytic dehalogenation of L-2-haloacids to produce the corresponding D-2-hydroxyacids with an inversion of the C2-configuration. 2-haloacid dehalogenases are of interest for their potential to degrade recalcitrant halogenated environmental pollutants and their use in the synthesis of industrial chemicals. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd02588','HAD_L2-DEX'),(1151491,'gnl|CDD|319800',_binary 'human N-acetylneuraminate-9-phosphate phosphatase, Escherichia coli house-cleaning phosphatase YjjG, and related phosphatases.  N-acetylneuraminate-9- phosphatase (Neu5Ac-9-Pase; E.C. 3.1.3.29) catalyzes the dephosphorylation of N-acylneuraminate 9-phosphate during the synthesis of N-acetylneuraminate; Escherichia coli nucleotide phosphatase YjjG has a broad pyrimidine nucleotide activity spectrum and functions as an in vivo house-cleaning phosphatase for noncanonical pyrimidine nucleotides. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd04305','HAD_Neu5Ac-Pase_like'),(1151493,'gnl|CDD|319797',_binary 'pyrophosphatase similar to Bacillus subtilis PpaX.  This family includes Bacillus subtilis PpaX which hydrolyzes pyrophosphate formed during serine-46-phosphorylated HPr (P-Ser-HPr) dephosphorylation by the bifunctional enzyme HPr kinase/phosphorylase. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd02616','HAD_PPase'),(1151494,'gnl|CDD|319790',_binary 'N-terminal lipase phosphatase domain of human soluble epoxide hydrolase, Escherichia coli YihX/HAD4 alpha-D-glucose 1-phosphate phosphatase, and related domains, may be inactive.  This family includes the N-terminal phosphatase domain of human soluble epoxide hydrolase (sEH). sEH is a bifunctional enzyme with two distinct enzyme activities, the C-terminal domain has epoxide hydrolysis activity and the N-terminal domain (Ntermphos), which belongs to this family, has lipid phosphatase activity. The latter prefers mono-phosphate esters, and lysophosphatidic acids (LPAs) are the best natural substrates found to date.  In addition this family includes Gallus gallus sEH and Xenopus sEH which appears to lack phosphatase activity, and Escherichia coli YihX/HAD4 which selectively hydrolyzes alpha-Glucose-1-P, phosphatase, has significant phosphatase activity against pyridoxal phosphate, and has low beta phosphoglucomutase activity. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd02603','HAD_sEH-N_like'),(1151496,'gnl|CDD|162788',_binary 'noncanonical pyrimidine nucleotidase, YjjG family.  This HAD superfamily includes including YjjG from E. coli and YfnB from B. subtilis. YjjG has been shown to act as a house-cleaning enzyme, cleaving nucleotides with non-canonical nucleotide bases. This family is a member of the haloacid dehalogenase (HAD) superfamily of hydrolases which are characterized by three conserved sequence motifs. By virtue of an alpha helical domain in-between the first and second conserved motif, this family is a member of subfamily IA (TIGR01549).','TIGR02254','hydrolase_haloacid_dehalo'),(1151497,'gnl|CDD|273742',_binary 'HAD-superfamily hydrolase, subfamily IIIA.  This subfamily falls within the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The Class III subfamilies are characterized by the lack of any domains located between either between the first and second conserved catalytic motifs (as in the Class I subfamilies, TIGR01493, TIGR01509, TIGR01488 and TIGR01494) or between the second and third conserved catalytic motifs (as in the Class II subfamilies, TIGR01460 and TIGR01484) of the superfamily domain. The IIIA subfamily contains five major clades: histidinol-phosphatase (TIGR01261) and histidinol-phosphatase-related protein (TIGR00213) which together form a subfamily (TIGR01656), DNA 3\'-phosphatase (TIGR01663, TIGR01664), YqeG (TIGR01668) and YrbI (TIGR01670). In the case of histidinol phosphatase and PNK-3\'-phosphatase, this model represents a domain of a bifunctional system. In the histidinol phosphatase HisB, a C-terminal domain is an imidazoleglycerol-phosphate dehydratase which catalyzes a related step in histidine biosynthesis. In PNK-3\'-phosphatase, N- and C-terminal domains constitute the polynucleotide kinase and DNA-binding components of the enzyme. [Unknown function, Enzymes of unknown specificity].','TIGR01662','Uncharacterized_protein_Y'),(1151498,'gnl|CDD|130495',_binary '2-haloalkanoic acid dehalogenase, type II.  Catalyzes the hydrolytic dehalogenation of small L-2-haloalkanoic acids to yield the corresponding D-2-hydroxyalkanoic acids. Belongs to the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases (pfam00702), class (subfamily) I. Note that the Type I HAD enzymes have not yet been fully characterized, but clearly utilize a substantially different catalytic mechanism and are thus unlikely to be related.','TIGR01428','Haloacetate_dehalogenase'),(1151499,'gnl|CDD|273686',_binary 'haloacid dehalogenase superfamily, subfamily IA, variant 1 with third motif having Dx(3-4)D or Dx(3-4)E.  This model represents part of one structural subfamily of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called \'capping domain\', or the absence of such a domain. Subfamily I consists of sequences in which the capping domain is found in between the first and second catalytic motifs. Subfamily II consists of sequences in which the capping domain is found between the second and third motifs. Subfamily III sequences have no capping domain in either of these positions.The Subfamily IA and IB capping domains are predicted by PSI-PRED to consist of an alpha helical bundle. Subfamily I encompasses such a wide region of sequence space (the sequences are highly divergent) that representing it with a single model is impossible, resulting in an overly broad description which allows in many unrelated sequences. Subfamily IA and IB are separated based on an aparrent phylogenetic bifurcation. Subfamily IA is still too broad to model, but cannot be further subdivided into large chunks based on phylogenetic trees. Of the three motifs defining the HAD superfamily, the third has three variant forms: (1) hhhhsDxxx(x)(D/E), (2) hhhhssxxx(x)D and (3) hhhhDDxxx(x)s where _s_ refers to a small amino acid and _h_ to a hydrophobic one. All three of these variants are found in subfamily IA. Individual models were made based on seeds exhibiting only one of the variants each. Variant 1 (this model) is found in the enzymes phosphoglycolate phosphatase (TIGR01449) and enolase-phosphatase. These three variant models (see also TIGR01493 and TIGR01509) were created withthe knowledge that there will be overlap among them - this is by design and serves the purpose of eliminating the overlap with models of more distantly relatedHAD subfamilies caused by an overly broad single model. [Unknown function, Enzymes of unknown specificity].','TIGR01549','HAD-SF-IA-v1'),(1151500,'gnl|CDD|315824',_binary 'HAD-hyrolase-like.','pfam13242','Hydrolase_like'),(1151501,'gnl|CDD|319854',_binary 'Escherichia coli Gph phosphoglycolate phosphatase and related proteins; belongs to the haloacid dehalogenase-like superfamily.  Phosphoglycolate phosphatase (PGP; EC 3.1.3.18) catalyzes the conversion of 2-phosphoglycolate into glycolate and phosphate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd16417','HAD_PGPase'),(1151502,'gnl|CDD|273662',_binary 'haloacid dehalogenase superfamily, subfamily IA, variant 3 with third motif having DD or ED.  This model represents part of one structural subfamily of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called \'capping domain\', or the absence of such a domain. Subfamily I consists of sequences in which the capping domain is found in between the first and second catalytic motifs. Subfamily II consists of sequences in which the capping domain is found between the second and third motifs. Subfamily III sequences have no capping domain in either of these positions. The Subfamily IA and IB capping domains are predicted by PSI-PRED to consist of an alpha helical bundle. Subfamily I encompasses such a wide region of sequence space (the sequences are highly divergent) that representing it with a single model is impossible, resulting in an overly broad description which allows in many unrelated sequences. Subfamily IA and IB are separated based on an aparrent phylogenetic bifurcation. Subfamily IA is still too broad to model, but cannot be further subdivided into large chunks based on phylogenetic trees. Of the three motifs defining the HAD superfamily, the third has three variant forms: (1) hhhhsDxxx(x)D, (2) hhhhssxxx(x)D and (3) hhhhDDxxx(x)s where _s_ refers to a small amino acid and _h_ to a hydrophobic one. All three of these variants are found in subfamily IA. Individual models were made based on seeds exhibiting only one of the variants each. Variant 3 (this model) is found in the enzymes beta-phosphoglucomutase (TIGR01990) and deoxyglucose-6-phosphatase, while many other enzymes of subfamily IA exhibit this variant as well as variant 1 (TIGR01549). These three variant models were created with the knowledge that there will be overlap among them - this is by design and serves the purpose of eliminating the overlap with models of more distantly related HAD subfamilies caused by an overly broad single model. [Unknown function, Enzymes of unknown specificity].','TIGR01509','HAD-SF-IA-v3'),(1151503,'gnl|CDD|181865',_binary 'dUMP phosphatase; Provisional.','PRK09449','PRK09449'),(1151505,'gnl|CDD|315982',_binary 'Haloacid dehalogenase-like hydrolase.','pfam13419','HAD_2'),(1152231,'gnl|CDD|283157',_binary 'Mycobacterium membrane protein.  This family contains several membrane proteins from Mycobacterium species.','pfam05423','Mycobact_memb'),(1152475,'gnl|CDD|278873',_binary 'KOW motif.  This family has been extended to coincide with ref. The KOW (Kyprides, Ouzounis, Woese) motif is found in a variety of ribosomal proteins and NusG.','pfam00467','KOW'),(1153093,'gnl|CDD|319269',_binary 'DNA sulfur modification protein DndB.  dndB acts in the regulation of DNA modifications, including DNA phosphorothioation. DndB may act by binding near the phosphorothioate modification site and regulating access of the Dnd modification machinery to DNA. These proteins show similarity to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, and other members of the ParB/Srx superfamily.','cd16412','dndB'),(1153481,'gnl|CDD|313027',_binary 'Mitochondrial inner membrane protein.  Mitofilin controls mitochondrial cristae morphology. Mitofilin is enriched in the narrow space between the inner boundary and the outer membranes, where it forms a homotypic interaction and assembles into a large multimeric protein complex. The first 78 amino acids contain a typical amino-terminal-cleavable mitochondrial presequence rich in positive-charged and hydroxylated residues and a membrane anchor domain. In addition, it has three centrally located coiled coil domains.','pfam09731','Mitofilin'),(1155280,'gnl|CDD|307132',_binary 'AP2 domain.  This 60 amino acid residue domain can bind to DNA and is found in transcription factor proteins.','pfam00847','AP2'),(1155283,'gnl|CDD|213198',_binary 'Cytochrome c biogenesis ATP-binding export protein.  CcmA, the ATP-binding component of the bacterial CcmAB transporter. The CCM family is involved in bacterial cytochrome c biogenesis. Cytochrome c maturation in E. coli requires the ccm operon, which encodes eight membrane proteins (CcmABCDEFGH). CcmE is a periplasmic heme chaperon that binds heme covalently and transfers it onto apocytochrome c in the presence of CcmF, CcmG, and CcmH. The CcmAB proteins represent an ABC transporter and the CcmCD proteins participate in heme transfer to CcmE.','cd03231','ABC_CcmA_heme_exporter'),(1155285,'gnl|CDD|213196',_binary 'ATP-binding cassette domain of the binding protein-dependent transport systems.  This class is comprised of all BPD (Binding Protein Dependent) systems that are largely represented in archaea and eubacteria and are primarily involved in scavenging solutes from the environment. ABC transporters are a large family of proteins involved in the transport of a wide variety of different compounds, like sugars, ions, peptides, and more complex organic molecules. The nucleotide binding domain shows the highest similarity between all members of the family. ABC transporters are a subset of nucleotide hydrolases that contain a signature motif, Q-loop, and H-loop/switch region, in addition to, the Walker A motif/P-loop and Walker B motif commonly found in a number of ATP- and GTP-binding and hydrolyzing proteins.','cd03229','ABC_Class3'),(1155287,'gnl|CDD|184125',_binary 'cytochrome c biogenesis protein CcmA; Provisional.','PRK13538','PRK13538'),(1155288,'gnl|CDD|274323',_binary 'thiol reductant ABC exporter, CydD subunit.  The gene pair cydCD encodes an ABC-family transporter in which each gene contains an N-terminal membrane-spanning domain (pfam00664) and a C-terminal ATP-binding domain (pfam00005). In E. coli these genes were discovered as mutants which caused the terminal heme-copper oxidase complex cytochrome bd to fail to assemble. Recent work has shown that the transporter is involved in export of redox-active thiol compounds such as cysteine and glutathione. The linkage to assembly of the cytochrome bd complex is further supported by the conserved operon structure found outside the gammaproteobacteria (cydABCD) containing both the transporter and oxidase genes components. The genes used as the seed members for this model are all either found in the gammproteobacterial context or the CydABCD context. All members of this family scoring above trusted at the time of its creation were from genomes which encode a cytochrome bd complex. Unfortunately, the gene symbol nomenclature adopted based on this operon in B. subtilis assigns cydC to the third gene in the operon where this gene is actually homologous to the E. coli cydD gene. We have chosen to name all homologs in this family in accordance with the precedence of publication of the E. coli name, CydD.','TIGR02857','transport_ATP-binding_pro'),(1155290,'gnl|CDD|274331',_binary 'thiol reductant ABC exporter, CydC subunit.  The gene pair cydCD encodes an ABC-family transporter in which each gene contains an N-terminal membrane-spanning domain (pfam00664) and a C-terminal ATP-binding domain (pfam00005). In E. coli these genes were discovered as mutants which caused the terminal heme-copper oxidase complex cytochrome bd to fail to assemble. Recent work has shown that the transporter is involved in export of redox-active thiol compounds such as cysteine and glutathione. The linkage to assembly of the cytochrome bd complex is further supported by the conserved operon structure found outside the gammaproteobacteria (cydABCD) containing both the transporter and oxidase genes components. The genes used as the seed members for this model are all either found in the gammproteobacterial context or the CydABCD context. All members of this family scoring above trusted at the time of its creation were from genomes which encode a cytochrome bd complex.','TIGR02868','ABC_transporter_ATP-bindi'),(1155291,'gnl|CDD|226617',_binary 'ABC-type transport system involved in cytochrome c biogenesis, ATPase component [Posttranslational modification, protein turnover, chaperones].','COG4133','CcmA'),(1155293,'gnl|CDD|184588',_binary 'phosphate transporter ATP-binding protein; Provisional.','PRK14243','PRK14243'),(1155294,'gnl|CDD|224047',_binary 'ABC-type cobalt transport system, ATPase component [Inorganic ion transport and metabolism].','COG1122','CbiO'),(1155298,'gnl|CDD|226952',_binary 'ABC-type uncharacterized transport system, ATPase component [General function prediction only].','COG4586','COG4586'),(1155301,'gnl|CDD|213234',_binary 'ATP-binding cassette domain of an uncharacterized transporter similar in sequence to NatA.  NatA is the ATPase component of a bacterial ABC-type Na+ transport system called NatAB, which catalyzes ATP-dependent electrogenic Na+ extrusion without mechanically coupled to proton or K+ uptake. NatB possess six putative membrane spanning regions at its C-terminus. In B. subtilis, NatAB is inducible by agents such as ethanol and protonophores, which lower the proton-motive force across the membrane. The closest sequence similarity to NatA is exhibited by DrrA of the two-component daunorubicin- and doxorubicin-efflux system. Hence, the functional NatAB is presumably assembled with two copies of the single ATP-binding protein and the single integral membrane protein.','cd03267','ABC_NatA_like'),(1155302,'gnl|CDD|187724',_binary 'Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalyzes the transfer of one-carbon groups, specifically the formyl- or hydroxymethyl- group.  This domain contains a Rossmann fold and it is the catalytic domain of the enzyme.','cd08822','FMT_core_like_2'),(1155303,'gnl|CDD|187712',_binary 'Formyltransferase, catalytic core domain.  Formyltransferase, catalytic core domain. The proteins of this superfamily contain a formyltransferase domain that hydrolyzes the removal of a formyl group from its substrate as part of a multistep transfer mechanism, and this alignment model represents the catalytic core of the formyltransferase domain.  This family includes the following known members; Glycinamide Ribonucleotide Transformylase (GART), Formyl-FH4 Hydrolase, Methionyl-tRNA Formyltransferase, ArnA, and 10-Formyltetrahydrofolate Dehydrogenase (FDH).  Glycinamide Ribonucleotide Transformylase  (GART) catalyzes the third step in de novo purine biosynthesis, the transfer of a formyl group to 5\'-phosphoribosylglycinamide. Formyl-FH4 Hydrolase catalyzes the hydrolysis of 10-formyltetrahydrofolate (formyl-FH4) to FH4 and formate. Methionyl-tRNA Formyltransferase transfers a formyl group onto the amino terminus of the acyl moiety of the methionyl aminoacyl-tRNA, which plays important role in translation initiation. ArnA is required for the modification of lipid A with 4-amino-4-deoxy-l-arabinose (Ara4N) that leads to resistance to cationic antimicrobial peptides (CAMPs) and clinical antimicrobials such as polymyxin. 10-formyltetrahydrofolate dehydrogenase (FDH) catalyzes the conversion of 10-formyltetrahydrofolate, a precursor for nucleotide biosynthesis, to tetrahydrofolate. Members of this family are multidomain proteins. The formyltransferase domain is located at the N-terminus of FDH, Methionyl-tRNA Formyltransferase and ArnA, and at the C-terminus of Formyl-FH4 Hydrolase.  Prokaryotic Glycinamide Ribonucleotide Transformylase (GART) is a single domain protein while eukaryotic GART is a trifunctional protein that catalyzes the second, third and fifth steps in de novo purine biosynthesis.','cd08369','FMT_core'),(1155304,'gnl|CDD|187716',_binary '10-formyltetrahydrofolate dehydrogenase (FDH), N-terminal hydrolase domain.  This family represents the N-terminal hydrolase domain of the bifunctional protein 10-formyltetrahydrofolate dehydrogenase (FDH). This domain contains a 10-formyl-tetrahydrofolate (10-formyl-THF) binding site and shares sequence homology and structural topology with other enzymes utilizing this substrate. This domain functions as a hydrolase, catalyzing the conversion of 10-formyl-THF, a precursor for nucleotide biosynthesis, to tetrahydrofolate (THF). The overall FDH reaction mechanism is a coupling of two sequential reactions, a hydrolase and a formyl dehydrogenase, bridged by a substrate transfer step.  The N-terminal hydrolase domain removes the formyl group from 10-formyl-THF and the C-terminal NADP-dependent dehydrogenase domain then reduces the formyl group to carbon dioxide.  The two catalytic domains are connected by a third intermediate linker domain that transfers the formyl group, covalently attached to the sulfhydryl group of the phosphopantetheine arm, from the N-terminal domain to the C-terminal domain.','cd08647','FMT_core_FDH_N'),(1155305,'gnl|CDD|223301',_binary 'Methionyl-tRNA formyltransferase [Translation, ribosomal structure and biogenesis].','COG0223','Fmt'),(1155306,'gnl|CDD|187715',_binary 'Methionyl-tRNA formyltransferase, N-terminal hydrolase domain.  Methionyl-tRNA formyltransferase (Met-tRNA-FMT), N-terminal formyltransferase domain.  Met-tRNA-FMT transfers a formyl group from N-10 formyltetrahydrofolate to the amino terminal end of a methionyl-aminoacyl-tRNA acyl moiety, yielding formyl-Met-tRNA. Formyl-Met-tRNA plays essential role in protein translation initiation by forming complex with IF2. The formyl group plays a dual role in the initiator identity of N-formylmethionyl-tRNA by promoting its recognition by IF2 and by impairing its binding to EFTU-GTP.  The N-terminal domain contains a Rossmann fold and it is the catalytic domain of the enzyme.','cd08646','FMT_core_Met-tRNA-FMT_N'),(1155307,'gnl|CDD|187713',_binary 'ArnA, N-terminal formyltransferase domain.  ArnA_N:  ArnA is a bifunctional enzyme required for the modification of lipid A with 4-amino-4-deoxy-L-arabinose (Ara4N) that leads to resistance to cationic antimicrobial peptides (CAMPs) and clinical antimicrobials such as polymyxin.  The C-terminal dehydrogenase domain of ArnA catalyzes the oxidative decarboxylation of UDP-glucuronic acid (UDP-GlcUA) to UDP-4-keto-arabinose (UDP-Ara4O), while the N-terminal formyltransferase domain of ArnA catalyzes the addition of a formyl group to UDP-4-amino-4-deoxy-L-arabinose (UDP-L-Ara4N) to form UDP-L-4-formamido-arabinose (UDP-L-Ara4FN). This domain family represents the catalytic core of the N-terminal formyltransferase domain. The formyltransferase also contains a smaller C-terminal domain the may be  involved in substrate binding. ArnA forms a hexameric structure, in which the dehydrogenase domains are arranged at the center of the particle with the transformylase domains on the outside of the particle.','cd08644','FMT_core_ArnA_N'),(1155308,'gnl|CDD|187720',_binary 'Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalyzes the transfer of one-carbon groups, specifically the formyl- or hydroxymethyl- group.  This domain contains a Rossmann fold and it is the catalytic domain of the enzyme.','cd08651','FMT_core_like_4'),(1155309,'gnl|CDD|187725',_binary 'Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalytic core domain found in a group of proteins with unknown functions.  Formyl transferase catalyzes the transfer of one-carbon groups, specifically the formyl- or hydroxymethyl- group.  This domain contains a Rossmann fold and it is the catalytic domain of the enzyme.','cd08823','FMT_core_like_5'),(1155310,'gnl|CDD|273088',_binary 'methionyl-tRNA formyltransferase.  The top-scoring characterized proteins other than methionyl-tRNA formyltransferase (fmt) itself are formyltetrahydrofolate dehydrogenases. The mitochondrial methionyl-tRNA formyltransferases are so divergent that, in a multiple alignment of bacterial fmt, mitochondrial fmt, and formyltetrahydrofolate dehydrogenases, the mitochondrial fmt appears the most different. However, because both bacterial and mitochondrial fmt are included in the seed alignment, all credible fmt sequences score higher than any non-fmt sequence. This enzyme modifies Met on initiator tRNA to f-Met. [Protein synthesis, tRNA aminoacylation].','TIGR00460','Methionyl-tRNA_formyltran'),(1155311,'gnl|CDD|215159',_binary 'methionyl-tRNA formyltransferase.','PLN02285','PLN02285'),(1155312,'gnl|CDD|278950',_binary 'Formyl transferase.  This family includes the following members. Glycinamide ribonucleotide transformylase catalyzes the third step in de novo purine biosynthesis, the transfer of a formyl group to 5\'-phosphoribosylglycinamide. Formyltetrahydrofolate deformylase produces formate from formyl- tetrahydrofolate. Methionyl-tRNA formyltransferase transfers a formyl group onto the amino terminus of the acyl moiety of the methionyl aminoacyl-tRNA. Inclusion of the following members is supported by PSI-blast. HOXX_BRAJA (P31907) contains a related domain of unknown function. PRTH_PORGI (P46071) contains a related domain of unknown function. Y09P_MYCTU (Q50721) contains a related domain of unknown function.','pfam00551','Formyl_trans_N'),(1155313,'gnl|CDD|236156',_binary 'bifunctional UDP-glucuronic acid decarboxylase/UDP-4-amino-4-deoxy-L-arabinose formyltransferase; Validated.','PRK08125','PRK08125'),(1155314,'gnl|CDD|223376',_binary 'Folate-dependent phosphoribosylglycinamide formyltransferase PurN [Nucleotide transport and metabolism].','COG0299','PurN'),(1155343,'gnl|CDD|129833',_binary 'LAO/AO transport system ATPase.  In E. coli, mutation of this kinase blocks phosphorylation of two transporter system periplasmic binding proteins and consequently inhibits those transporters. This kinase is also found in Gram-positive bacteria, archaea, and the roundworm C. elegans. It may have a more general, but still unknown function. Mutations have also been found that do not phosphorylate the periplasmic binding proteins, yet still allow transport. The ATPase activity of this protein seems to be necessary, however. [Transport and binding proteins, Amino acids, peptides and amines, Regulatory functions, Protein interactions].','TIGR00750','Hypothetical_protein_Rv14'),(1155489,'gnl|CDD|316082',_binary 'Glycosyl transferases group 1.','pfam13524','Glyco_trans_1_2'),(1155507,'gnl|CDD|176511',_binary 'Glycerophosphodiester phosphodiesterase domain of Thermotoga maritime and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in Thermotoga maritime glycerophosphodiester phosphodiesterase (TmGDE, EC 3.1.4.46) and its uncharacterized  homologs. Members in this family show high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. TmGDE exists as a monomer that might be the biologically relevant form.','cd08568','GDPD_TmGDE_like'),(1155516,'gnl|CDD|236859',_binary 'glycerophosphodiester phosphodiesterase; Provisional.','PRK11143','glpQ'),(1155518,'gnl|CDD|176507',_binary 'Glycerophosphodiester phosphodiesterase domain of putative Galdieria sulphuraria glycerophosphodiester phosphodiesterase and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in putative Galdieria sulphuraria glycerophosphodiester phosphodiesterase (GsGDE, EC 3.1.4.46) and its uncharacterized eukaryotic homologs. Members in this family show high sequence similarity to Escherichia coli GP-GDE, which catalyzes the degradation of glycerophosphodiesters to produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08564','GDPD_GsGDE_like'),(1155519,'gnl|CDD|176514',_binary 'Glycerophosphodiester phosphodiesterase domain of mammalian glycerophosphodiester phosphodiesterase GDE5-like proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in mammalian glycerophosphodiester phosphodiesterase GDE5-like proteins. GDE5 is widely expressed in mammalian tissues, with highest expression in spinal chord. Although its biological function remains unclear, mammalian GDE5 shows higher sequence homology to fungal and plant  glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46) than to other bacterial and mammalian GP-GDEs. It may also hydrolyze glycerophosphodiesters to sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.','cd08572','GDPD_GDE5_like'),(1155522,'gnl|CDD|176549',_binary 'Glycerophosphodiester phosphodiesterase domain of putative mammalian glycerophosphodiester phosphodiesterase GDE5 and similar proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in putative mammalian GDE5 and similar proteins. Mammalian GDE5 is widely expressed in mammalian tissues, with highest expression in the spinal chord. Although its biological function remains unclear, mammalian GDE5 shows higher sequence homology to fungal and plant  glycerophosphodiester phosphodiesterases (GP-GDEs, EC 3.1.4.46) than to other bacterial and mammalian GP-GDEs. It may also hydrolyze glycerophosphodiesters to sn-glycerol-3-phosphate (G3P) and the corresponding alcohols. In addition to C-terminal GDPD domain, all members in this subfamily have a starch binding domain (CBM20) in the N-terminus, which suggests these proteins may play a distinct role in glycerol metabolism.','cd08607','GDPD_GDE5'),(1155523,'gnl|CDD|176554',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized bacterial  homologs of mammalian glycerophosphodiester phosphodiesterase GDE4.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in uncharacterized bacterial homologs of mammalian GDE4, a transmembrane protein whose cellular function has not been elucidated yet.','cd08613','GDPD_GDE4_like_1'),(1155524,'gnl|CDD|176513',_binary 'Glycerophosphodiester phosphodiesterase domain of glycerophosphodiester phosphodiesterase-like protein SHV3 and SHV3-like proteins.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase (GDPD) domain present in glycerophosphodiester phosphodiesterase (GP-GDE)-like protein SHV3 and SHV3-like proteins (SVLs), which may play an important role in cell wall organization. The prototype of this family is a glycosylphosphatidylinositol (GPI) anchored protein SHV3 encoded by shaven3 (shv3) gene from Arabidopsis thaliana. Members in this family show sequence homology to bacterial GP-GDEs (EC 3.1.4.46) that catalyze the hydrolysis of various glycerophosphodiesters, and produce sn-glycerol-3-phosphate (G3P) and the corresponding alcohols.  Both, SHV3 and SVLs, have two tandemly repeated GDPD domains whose biochemical functions remain unclear. The residues essential for interactions with the substrates and calcium ions in bacterial GP-GDEs are not conserved in SHV3 and SVLs, which suggests that the function of GDPD domains in these proteins might be distinct from those in typical bacterial GP-GDEs. In addition, the two tandem repeats show low sequence similarity to each other, suggesting they have different biochemical function. Most members of this family are Arabidopsis-specific gene products. To date, SHV3 orthologues are only found in Physcomitrella patens.','cd08571','GDPD_SHV3_plant'),(1155527,'gnl|CDD|176547',_binary 'Glycerophosphodiester phosphodiesterase domain of uncharacterized plant glycerophosphodiester phosphodiesterase-like proteins similar to mammalian GDE5.  This subfamily corresponds to the glycerophosphodiester phosphodiesterase domain (GDPD) present in a group of uncharacterized plant glycerophosphodiester phosphodiesterase (GP-PDE)-like proteins. Members in this family show very high sequence homology to mammalian glycerophosphodiester phosphodiesterase GDE5 and are distantly related to plant GP-PDEs.','cd08605','GDPD_GDE5_like_1_plant'),(1155528,'gnl|CDD|224120',_binary 'Rad3-related DNA helicases [Transcription / DNA replication, recombination, and repair].','COG1199','DinG'),(1155529,'gnl|CDD|236966',_binary 'ATP-dependent DNA helicase DinG; Provisional.','PRK11747','dinG'),(1155530,'gnl|CDD|315883',_binary 'Helicase C-terminal domain.  This domain is found at the C-terminus of DEAD-box helicases.','pfam13307','Helicase_C_2'),(1155534,'gnl|CDD|214694',_binary 'helicase superfamily c-terminal domain.','smart00491','HELICc2'),(1155536,'gnl|CDD|273169',_binary 'DNA repair helicase (rad3).  All proteins in this family for which funcitons are known are DNA-DNA helicases that funciton in the initiation of transcription and nucleotide excision repair as part of the TFIIH complex. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00604','ATP-dependent_RNA_helicas'),(1155537,'gnl|CDD|187839',_binary 'CRISPR/Cas system-associated DinG family helicase Csf4.  CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) and associated Cas proteins comprise a system for heritable host defense by prokaryotic cells against phage and other foreign DNA; DinG family DNA helicase.','cd09708','Csf4_U'),(1155538,'gnl|CDD|274436',_binary 'CRISPR type AFERR-associated DEAD/DEAH-box helicase Csf4.  Members of this family show up near CRISPR repeats in Acidithiobacillus ferrooxidans ATCC 23270, Azoarcus sp. EbN1, and Rhodoferax ferrireducens DSM 15236. In the latter two species, the CRISPR/cas locus is found on a plasmid. This family is one of several characteristic of a type of CRISPR-associated (cas) gene cluster we designate Aferr after A. ferrooxidans, where it is both chromosomal and the only type of cas gene cluster found. The gene is designated csf4 (CRISPR/cas Subtype as in A. ferrooxidans protein 1), as it lies farthest (fourth closest) from the repeats in the A. ferrooxidans genome.','TIGR03117','cas_csf4'),(1155600,'gnl|CDD|129520',_binary 'competence protein ComEA helix-hairpin-helix repeat region.  Members of the subfamily recognized by this model include competence protein ComEA and closely related proteins from a number of species that exhibit competence for transformation by exongenous DNA, including Streptococcus pneumoniae, Bacillus subtilis, Neisseria meningitidis, and Haemophilus influenzae. This model represents a region of two tandem copies of a helix-hairpin-helix domain (pfam00633), each about 30 residues in length. Limited sequence similarity can be found among some members of this family N-terminal to the region covered by this model. [Cellular processes, DNA transformation].','TIGR00426','Hypothetical_protein_ybaV'),(1156910,'gnl|CDD|273258',_binary 'endopeptidase La.  This protein, the ATP-dependent serine endopeptidase La, is induced by heat shock and other stresses in E. coli, B. subtilis, and other species. The yeast member, designated PIM1, is located in the mitochondrial matrix, required for mitochondrial function, and also induced by heat shock. [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00763','Lon_protease'),(1157434,'gnl|CDD|169423',_binary 'hypothetical protein; Provisional.','PRK08392','PRK08392'),(1157543,'gnl|CDD|236233',_binary 'putative RNA polymerase sigma factor SigI; Reviewed.','PRK08311','PRK08311'),(1157562,'gnl|CDD|316946',_binary 'Domain of unknown function (DUF4429).  This presumed domain is functionally uncharacterized. This domain family is found in bacteria, archaea and viruses, and is approximately 90 amino acids in length. This domain is often found in two tandem copies.','pfam14472','DUF4429'),(1157720,'gnl|CDD|238084',_binary 'polycystic kidney disease I (PKD) domain; similar to other cell-surface modules, with an IG-like fold; domain probably functions as a ligand binding site in protein-protein or protein-carbohydrate interactions; a single instance of the repeat is presented here. The domain is also found in microbial collagenases and chitinases.','cd00146','PKD'),(1157727,'gnl|CDD|274475',_binary 'DNA sulfur modification protein DndD.  This model describes the DndB protein encoded by an operon associated with a sulfur-containing modification to DNA. The operon is sporadically distributed in bacteria, much like some restriction enzyme operons. DndD is described as a putative ATPase. The small number of examples known so far include species from among the Firmicutes, Actinomycetes, Proteobacteria, and Cyanobacteria. [DNA metabolism, Restriction/modification].','TIGR03185','DNA_S_dndD'),(1157746,'gnl|CDD|222994',_binary 'dUTPase; Provisional.','PHA03129','PHA03129'),(1157816,'gnl|CDD|119349',_binary 'The GH18 (glycosyl hydrolase, family 18) type II chitinases hydrolyze chitin, an abundant polymer of beta-1,4-linked N-acetylglucosamine (GlcNAc) which is a major component of the cell wall of fungi and the exoskeleton of arthropods.  Chitinases have been identified in viruses, bacteria, fungi, protozoan parasites, insects, and plants. The structure of the GH18 domain is an eight-stranded beta/alpha barrel with a pronounced active-site cleft at the C-terminal end of the beta-barrel.  The GH18 family includes chitotriosidase, chitobiase, hevamine, zymocin-alpha, narbonin, SI-CLP (stabilin-1 interacting chitinase-like protein), IDGF (imaginal disc growth factor), CFLE (cortical fragment-lytic enzyme) spore hydrolase, the type III and type V plant chitinases, the endo-beta-N-acetylglucosaminidases, and the chitolectins.  The GH85 (glycosyl hydrolase, family 85) ENGases (endo-beta-N-acetylglucosaminidases) are closely related to the GH18 chitinases and are included in this alignment model.','cd00598','GH18_chitinase-like'),(1157872,'gnl|CDD|284793',_binary 'NUMOD1 domain.  This domain probably represents a DNA-binding helix-turn-helix based on its similarity to other families (Bateman A pers obs).','pfam07453','NUMOD1'),(1158056,'gnl|CDD|176653',_binary 'DEDDy 3\'-5\' exonuclease domain of WRN and similar proteins.  WRN is a unique RecQ DNA helicase exhibiting an exonuclease activity. It contains a DEDDy-type DnaQ-like 3\'-5\' exonuclease domain possessing three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. Mutations in the WRN gene cause Werner syndrome, an autosomal recessive disorder associated with premature aging and increased susceptibility to cancer and type II diabetes. WRN interacts with key proteins involved in DNA replication, recombination, and repair. It is believed to maintain genomic stability and life span by participating in DNA processes. WRN is stimulated by Ku70/80, an important regulator of genomic stability.','cd06141','WRN_exo'),(1158057,'gnl|CDD|274382',_binary 'RNA polymerase sigma factor, cyanobacterial RpoD-like family.  This family includes a number of closely related sigma-70 (TIGR02937) factors in the cyanobacteria. All appear most closely related to the essential sigma-70 factor RpoD, and some score above trusted to the RpoD C-terminal domain model (TIGR02393).','TIGR02997','RNA_polymerase_sigma_subu'),(1158058,'gnl|CDD|236306',_binary 'RNA polymerase sigma factor SigB; Validated.','PRK08583','PRK08583'),(1158106,'gnl|CDD|99917',_binary 'DNA polymerase type-B zeta subfamily catalytic domain. DNA polymerase (Pol) zeta is a member of the eukaryotic B-family of DNA polymerases and distantly related to DNA Pol delta. Pol zeta plays a major role in translesion replication and the production of either spontaneous or induced mutations. Apart from its role in translesion replication, Pol zeta also appears to be involved in somatic hypermutability in B lymphocytes, an important element for the production of high affinity antibodies in response to an antigen.','cd05534','POLBc_zeta'),(1158179,'gnl|CDD|238859',_binary 'Rhamnogalacturan_acetylesterase_like subgroup of SGNH-hydrolases. Rhamnogalacturan acetylesterase removes acetyl esters from rhamnogalacturonan substrates, and renders them susceptible to degradation by rhamnogalacturonases. Rhamnogalacturonans are highly branched regions in pectic polysaccharides, consisting of repeating -(1,2)-L-Rha-(1,4)-D-GalUA disaccharide units, with many rhamnose residues substituted by neutral oligosaccharides such as arabinans, galactans and arabinogalactans. Extracellular enzymes participating in the degradation of plant cell wall polymers, such as Rhamnogalacturonan acetylesterase, would typically be found in saprophytic and plant pathogenic fungi and bacteria.','cd01821','Rhamnogalacturan_acetyles'),(1158219,'gnl|CDD|315502',_binary 'Integrase.  This is a family of DNA-binding prophage integrases found in Proteobacteria.','pfam12835','Integrase_1'),(1158294,'gnl|CDD|259816',_binary 'Uncharacterized prokaryotic subgroup of the stomatin-like proteins (slipins) family; belonging to the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes a subgroup of the stomatin-like protein family (SLPs or slipins) that is found in bacteria. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Bacterial SLPs remain uncharacterized.','cd13438','SPFH_eoslipins_u2'),(1158295,'gnl|CDD|259808',_binary 'Uncharacterized prokaryotic subgroup of the stomatin-like proteins (slipins) family; belonging to the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes a subgroup of the stomatin-like protein family (SLPs or slipins) that is found in bacteria and archaebacteria. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Bacterial and archaebacterial SLPs remain uncharacterized. This subgroup contains PH1511 from the hyperthermophilic archaeon Pyrococcus horikoshi.','cd08826','SPFH_eoslipins_u1'),(1158296,'gnl|CDD|259812',_binary 'Stomatin-like proteins (slipins) family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes proteins similar to stomatin, podocin, and other members of the stomatin-like protein family (SLPs or slipins). The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Stomatin interacts with and regulates members of the degenerin/epithelia Na+ channel family in mechanosensory cells of Caenorhabditis elegans and vertebrate neurons and participates in trafficking of Glut1 glucose transporters. Mutations in the podocin gene give rise to autosomal recessive steroid resistant nephritic syndrome. Bacterial and archaebacterial SLPs and many of the eukaryotic family members remain uncharacterized.','cd13434','SPFH_SLPs'),(1158297,'gnl|CDD|259813',_binary 'Slipin-4 (SLP-4), an uncharacterized subgroup of the stomatin-like proteins (slipins) family; belonging to the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes a subgroup of the stomatin-like protein family (SLPs or slipins) that is found in arthropods. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Members of this divergent slipin subgroup remain largely uncharacterized. It contains Drosophila Mec2, the gene for which was identified in a screen for genes required for nephrocyte function; it may function together with Sns in maintaining nephrocyte diaphragm.','cd13435','SPFH_SLP-4'),(1158316,'gnl|CDD|237861',_binary 'DNA polymerase III subunit beta; Provisional.','PRK14947','PRK14947'),(1158582,'gnl|CDD|185720',_binary 'Glycosyl hydrolase families 32 and 68, which for the clan GH-J.  This glycosyl hydrolase family clan J (according to carbohydrate-active enzymes database (CAZY)) includes family 32 (GH32) and 68 (GH68). The overall sequence homology between the two families is low (<15% identity), but common sequence motifs have been identified. GH32 enzymes are invertases that also include other fructofuranosidases such as inulinase (EC 3.2.1.7), exo-inulinase (EC 3.2.1.80), levanase (EC 3.2.1.65), and transfructosidases such sucrose:sucrose 1-fructosyltransferase (EC 2.4.1.99), fructan:fructan 1-fructosyltransferase (EC 2.4.1.100), sucrose:fructan 6-fructosyltransferase (EC 2.4.1.10), fructan:fructan 6G-fructosyltransferase (EC 2.4.1.243) and levan fructosyltransferases (EC 2.4.1.-). GH32 enzymes cleave sucrose into fructose and glucose via beta-fructofuranosidase activity, producing invert sugar that is a mixture of dextrorotatory D-glucose and levorotatory D-fructose, thus named invertase (EC 3.2.1.26). GH68 consists of frucosyltransferases (FTFs) that include levansucrase (EC 2.4.1.10); beta-fructofuranosidase (EC 3.2.1.26); inulosucrase (EC 2.4.1.9), all of which use sucrose as their preferential donor substrate. A common structural feature of all these enzymes is a 5-bladed beta-propeller domain, similar to GH43, that contains the catalytic acid and catalytic base. A long V-shaped groove, partially enclosed at one end, forms a single extended substrate-binding surface across the face of the propeller.','cd08979','GH_J'),(1158862,'gnl|CDD|227606',_binary 'Phage-related minor tail protein [Function unknown].','COG5281','COG5281'),(1159025,'gnl|CDD|99920',_binary 'DNA polymerase type-II subfamily catalytic domain. Bacteria contain five DNA polymerases (I, II, III, IV and V). DNA polymerase II (Pol II) is a prototype for the B-family of polymerases. The role of Pol II in a variety of cellular activities, such as repair of DNA damaged by UV irradiation or oxidation has been proven by genetic studies. DNA polymerase III is the main enzyme responsible for replication of the bacterial chromosome; however, In vivo studies have also shown that Pol II is able to participate in chromosomal DNA replication with larger role in lagging-strand replication.','cd05537','POLBc_Pol_II'),(1159052,'gnl|CDD|185679',_binary 'FAD synthetase, N-terminal domain of the bifunctional enzyme.  FAD synthetase_N.  N-terminal domain of the bifunctional riboflavin biosynthesis protein riboflavin kinase/FAD synthetase. These enzymes have both ATP:riboflavin 5\'-phosphotransferase and ATP:FMN-adenylyltransferase activities.  The N-terminal domain is believed to play a role in the adenylylation reaction of FAD synthetases. The C-terminal domain is thought to have kinase activity.  FAD synthetase is present among all kingdoms of life.  However, the bifunctional enzyme is not found in mammals, which use separate enzymes for FMN and FAD formation.','cd02064','FAD_synthetase_N'),(1159053,'gnl|CDD|223274',_binary 'FAD synthase [Coenzyme metabolism].','COG0196','RibF'),(1159054,'gnl|CDD|185680',_binary 'Nicotinamide/nicotinate mononucleotide adenylyltransferase.  Nicotinamide/nicotinate mononucleotide (NMN/ NaMN)adenylyltransferase (NMNAT).  NMNAT represents the primary bacterial and eukaryotic adenylyltransferases for nicotinamide-nucleotide and for the deamido form, nicotinate nucleotide.  It is an indispensable enzyme in the biosynthesis of NAD(+) and NADP(+). Nicotinamide-nucleotide adenylyltransferase synthesizes NAD via the salvage pathway, while nicotinate-nucleotide adenylyltransferase synthesizes the immediate precursor of NAD via the de novo pathway. Human NMNAT displays unique dual substrate specificity toward both NMN and NaMN, and can participate in both de novo and salvage pathways of NAD synthesis.','cd02165','NMNAT'),(1159055,'gnl|CDD|272898',_binary 'riboflavin kinase/FMN adenylyltransferase.  multifunctional enzyme: riboflavin kinase (EC 2.7.1.26) (flavokinase) / FMN adenylyltransferase (EC 2.7.7.2) (FAD pyrophosphorylase) (FAD synthetase). [Biosynthesis of cofactors, prosthetic groups, and carriers, Riboflavin, FMN, and FAD].','TIGR00083','Putative_riboflavin_biosy'),(1159059,'gnl|CDD|275028',_binary 'DNA phosphorothioation-associated DGQHR protein 1.  The DND system produces an phosphorothioation modification to DNA, replacing a non-bridging oxygen of a phosphate group with sulfur. The modification causes DNA degradation during electrophoresis in Tris buffer. This protein, like DndB (TIGR03233), contains a DGQHR domain (TIGR03187), which also occurs in several contexts that suggest lateral transfer rather than DNA phosphorothioation-dependent restriction.','TIGR04172','putative_dgqhr_domain_con'),(1159060,'gnl|CDD|316585',_binary 'DNA-sulfur modification-associated.  This is family of bacterial proteins likely to be necessary for binding to DNA and recognising the modification sites. Members are found in bacteria, archaea and on viral plasmids, and are typically between 354 and 474 amino acids in length. There is a conserved DGQHR sequence motif.','pfam14072','DndB'),(1159062,'gnl|CDD|319271',_binary 'DNA-sulfur modification-associated domain.  Family of proteins related to dndB. dndB acts in the regulation of DNA modifications, including DNA phosphorothioation. Both have a conserved DGQHR sequence motif. These proteins show similarity to the N-terminal domain of ParB, a DNA-binding component of the prokaryotic parABS partitioning system, and other members of the ParB/Srx superfamily.','cd16414','dndB_like'),(1159083,'gnl|CDD|224787',_binary 'NYN ribonuclease and ATPase of PhoH family domains [General    function prediction only].','COG1875','COG1875'),(1159084,'gnl|CDD|308265',_binary 'PhoH-like protein.  PhoH is a cytoplasmic protein and predicted ATPase that is induced by phosphate starvation.','pfam02562','PhoH'),(1159085,'gnl|CDD|189053',_binary 'PIN domain of bacterial Smg6-like homologs with PhoH-like ATPase domains.  PIN (PilT N terminus) domain of Smg6-like bacterial proteins with C-terminal PhoH-like ATPase domains and other similar homologs are included in this family. Eukaryotic Smg5 and Smg6 nucleases are essential factors in nonsense-mediated mRNA decay (NMD), a post-transcriptional regulatory pathway that recognizes and rapidly degrades mRNAs containing premature translation termination codons. In vivo, the Smg6 PIN domain elicits degradation of bound mRNAs, as well as, metal ion dependent, degradation of single-stranded RNA, in vitro. These PIN domains are homologs of flap endonuclease-1 (FEN1)-like PIN domains, but lack the extensive arch/clamp region  and the H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region, seen in FEN1-like PIN domains.  PIN domains within this subgroup contain four highly conserved acidic residues (putative metal-binding, active site residues). Many of the bacterial homologs in this group have an N-terminal PIN domain and a C-terminal PhoH-like ATPase domain and are predicted to be ATPases which are induced by phosphate starvation.','cd09883','PIN_PhoHL-ATPase'),(1159086,'gnl|CDD|224616',_binary 'Phosphate starvation-inducible protein PhoH, predicted ATPase [Signal transduction mechanisms].','COG1702','PhoH'),(1159087,'gnl|CDD|316186',_binary 'PIN domain.  Members of this family of bacterial domains are predicted to be RNases (from similarities to 5\'-exonucleases).','pfam13638','PIN_4'),(1159088,'gnl|CDD|182529',_binary 'hypothetical protein; Provisional.','PRK10536','PRK10536'),(1159089,'gnl|CDD|189050',_binary 'PIN domain.  PIN (PilT N terminus) domain of nonsense-mediated decay (NMD) factors, Smg5 and Smg6, and homologs are included in this family. Smg5 and Smg6 are essential factors in NMD, a post-transcriptional regulatory pathway that recognizes and rapidly degrades mRNAs containing premature translation termination codons. In vivo, the Smg6 PIN domain elicits degradation of bound mRNAs, as well as, metal-ion dependent, degradation of single-stranded RNA, in vitro. These PIN domains are structural homologs of flap endonuclease-1 (FEN1)-like PIN domains, but lack the extensive arch/clamp region and the H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region, seen in FEN1-like PIN domains.  Point mutation studies of the conserved aspartate residues in the catalytic center of the Smg6 PIN domain revealed that Smg6 is the endonuclease involved in human NMD. However, Smg5 lacks several of these key catalytic residues and does not degrade single-stranded RNA, in vivo. Eukaryotic Smg5 and Smg6-like PIN domains are present at the C-terminal end of telomerase activating proteins, Est1. Many of the bacterial homologs in this group have an N-terminal PIN domain and a C-terminal PhoH-like ATPase domain.','cd09880','PIN_Smg5-Smg6-like'),(1159123,'gnl|CDD|307069',_binary 'F5/8 type C domain.  This domain is also known as the discoidin (DS) domain family.','pfam00754','F5_F8_type_C'),(1159224,'gnl|CDD|316776',_binary 'PcfJ-like protein.  The PcfJ-like protein family includes the E. faecalis PcfJ protein, which is functionally uncharacterized. It is found in bacteria and viruses, and is typically between 159 and 170 amino acids in length. There is a conserved HCV sequence motif.','pfam14284','PcfJ'),(1159226,'gnl|CDD|306938',_binary 'Staphylococcal nuclease homolog.  Present in all three domains of cellular life. Four copies in the transcriptional coactivator p100: these, however, appear to lack the active site residues of Staphylococcal nuclease. Positions 14 (Asp-21), 34 (Arg-35), 39 (Asp-40), 42 (Glu-43) and 110 (Arg-87) [SNase numbering in parentheses] are thought to be involved in substrate-binding and catalysis.','pfam00565','SNase'),(1159227,'gnl|CDD|238102',_binary 'Staphylococcal nuclease homologues. SNase homologues are found in bacteria, archaea, and eukaryotes. They contain no disufide bonds.','cd00175','SNc'),(1159228,'gnl|CDD|224442',_binary 'Micrococcal nuclease (thermonuclease) homologs [DNA replication, recombination, and repair].','COG1525','COG1525'),(1159284,'gnl|CDD|311793',_binary 'Male sterility protein.  This family represents the C-terminal region of the male sterility protein in a number of arabidopsis and drosophila. A sequence-related jojoba acyl CoA reductase is also included.','pfam07993','NAD_binding_4'),(1159293,'gnl|CDD|224014',_binary 'GDP-D-mannose dehydratase [Cell envelope biogenesis, outer membrane].','COG1089','Gmd'),(1159299,'gnl|CDD|187546',_binary 'extended (e) SDRs, subgroup 1.  This family consists of an SDR module of multidomain proteins identified as putative polyketide sythases fatty acid synthases (FAS), and nonribosomal peptide synthases, among others. However, unlike the usual ketoreductase modules of FAS and polyketide synthase, these domains are related to the extended SDRs, and have canonical NAD(P)-binding motifs and an active site tetrad. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05235','SDR_e1'),(1159319,'gnl|CDD|273787',_binary 'thioester reductase domain.  This model includes the terminal domain from the fungal alpha aminoadipate reductase enzyme (also known as aminoadipate semialdehyde dehydrogenase) which is involved in the biosynthesis of lysine, as well as the reductase-containing component of the myxochelin biosynthetic gene cluster, MxcG. The mechanism of reduction involves activation of the substrate by adenylation and transfer to a covalently-linked pantetheine cofactor as a thioester. This thioester is then reduced to give an aldehyde (thus releasing the product) and a regenerated pantetheine thiol. (In myxochelin biosynthesis this aldehyde is further reduced to an alcohol or converted to an amine by an aminotransferase.) This is a fundamentally different reaction than beta-ketoreductase domains of polyketide synthases which act at a carbonyl two carbons removed from the thioester and forms an alcohol as a product. This domain is invariably found at the C-terminus of the proteins which contain it (presumably because it results in the release of the product). The majority of hits to this model are non-ribosomal peptide synthetases in which this domain is similarly located proximal to a thiolation domain (pfam00550). In some cases this domain is found at the end of a polyketide synthetase enzyme, but is unlike ketoreductase domains which are found before the thiolase domains. Exceptions to this observed relationship with the thiolase domain include three proteins which consist of stand-alone reductase domains (GP|466833 from M. leprae, GP|435954 from Anabaena and OMNI|NTL02SC1199 from Strep. coelicolor) and one protein (OMNI|NTL01NS2636 from Nostoc) which contains N-terminal homology with a small group of hypothetical proteins but no evidence of a thiolation domain next to the putative reductase domain. Below the noise cutoff to this model are proteins containing more distantly related ketoreductase and dehydratase/epimerase domains. It has been suggested that a NADP-binding motif can be found in the N-terminal portion of this domain that may form a Rossman-type fold.','TIGR01746','L-aminoadipate-semialdehy'),(1159321,'gnl|CDD|183375',_binary 'NAD-dependent epimerase/dehydratase family protein; Provisional.','PRK11908','PRK11908'),(1159324,'gnl|CDD|225857',_binary 'Putative dehydrogenase domain of multifunctional non-ribosomal peptide synthetases and related enzymes [Secondary metabolites biosynthesis, transport, and catabolism].','COG3320','COG3320'),(1159328,'gnl|CDD|273644',_binary 'GDP-mannose 4,6-dehydratase.  Alternate name: GDP-D-mannose dehydratase. This enzyme converts GDP-mannose to GDP-4-dehydro-6-deoxy-D-mannose, the first of three steps for the conversion of GDP-mannose to GDP-fucose in animals, plants, and bacteria. In bacteria, GDP-L-fucose acts as a precursor of surface antigens such as the extracellular polysaccharide colanic acid of E. coli. Excluded from this model are members of the clade that score poorly because of highly dervied (phylogenetically long-branch) sequences, e.g. Aneurinibacillus thermoaerophilus Gmd, described as a bifunctional GDP-mannose 4,6-dehydratase/GDP-6-deoxy-D-lyxo-4-hexulose reductase (PUBMED:11096116). [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01472','GDP-mannose_46_dehydratas'),(1159336,'gnl|CDD|213180',_binary 'Eye pigment and drug resistance transporter subfamily G of the ATP-binding cassette superfamily.  ABCG transporters are involved in eye pigment (EP) precursor transport, regulation of lipid-trafficking mechanisms, and pleiotropic drug resistance (DR). DR is a well-described phenomenon occurring in fungi and shares several similarities with processes in bacteria and higher eukaryotes. Compared to other members of the ABC transporter subfamilies, the ABCG transporter family is composed of proteins that have an ATP-binding cassette domain at the N-terminus and a TM (transmembrane) domain at the C-terminus.','cd03213','ABCG_EPDR'),(1159348,'gnl|CDD|240507',_binary 'KOW domain of Spt5, repeat 3.  Spt5, an eukaryotic ortholog of NusG, contains multiple KOW motifs at its C-terminus. Spt5 is involved in transcription elongation and termination. KOW domain is known as an RNA-binding motif that is shared so far among some families of ribosomal proteins, the essential bacterial transcriptional elongation factor NusG, the eukaryotic chromatin elongation factor Spt5, the higher eukaryotic KIN17 proteins and Mtr4. KOW_Spt5 domains play critical roles in recruitment of multiple other eukaryotic transcription elongation and RNA biogenesis factors and additionally are involved in the binding of the eukaryotic Spt5 proteins to RNA polymerases.','cd06083','KOW_Spt5_3'),(1159349,'gnl|CDD|200471',_binary 'Alpha amylase catalytic domain found in oligo-1,6-glucosidase (also called isomaltase; sucrase-isomaltase; alpha-limit dextrinase), trehalose synthase (also called maltose alpha-D-glucosyltransferase), and related proteins.  Oligo-1,6-glucosidase (EC 3.2.1.10) hydrolyzes the alpha-1,6-glucosidic linkage of isomaltooligosaccharides, pannose, and dextran. Unlike alpha-1,4-glucosidases (EC 3.2.1.20), it fails to hydrolyze the alpha-1,4-glucosidic bonds of maltosaccharides. Trehalose synthase (EC 5.4.99.16) catalyzes the isomerization of maltose to produce trehalulose. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11332','AmyAc_OligoGlu_TS'),(1159350,'gnl|CDD|200472',_binary 'Alpha amylase catalytic domain found in Sucrose isomerases, oligo-1,6-glucosidase (also called isomaltase; sucrase-isomaltase; alpha-limit dextrinase), dextran glucosidase (also called glucan 1,6-alpha-glucosidase), and related proteins.  The sucrose isomerases (SIs) Isomaltulose synthase (EC 5.4.99.11) and Trehalose synthase (EC 5.4.99.16) catalyze the isomerization of sucrose and maltose to produce isomaltulose and trehalulose, respectively. Oligo-1,6-glucosidase (EC 3.2.1.10) hydrolyzes the alpha-1,6-glucosidic linkage of isomaltooligosaccharides, pannose, and dextran. Unlike alpha-1,4-glucosidases (EC 3.2.1.20), it fails to hydrolyze the alpha-1,4-glucosidic bonds of maltosaccharides. Dextran glucosidase (DGase, EC 3.2.1.70) hydrolyzes alpha-1,6-glucosidic linkages at the non-reducing end of panose, isomaltooligosaccharides and dextran to produce alpha-glucose.The common reaction chemistry of the alpha-amylase family enzymes is based on a two-step acid catalytic mechanism that requires two critical carboxylates: one acting as a general acid/base (Glu) and the other as a nucleophile (Asp). Both hydrolysis and transglycosylation proceed via the nucleophilic substitution reaction between the anomeric carbon, C1 and a nucleophile. Both enzymes contain the three catalytic residues (Asp, Glu and Asp) common to the alpha-amylase family as well as two histidine residues which are predicted to be critical to binding the glucose residue adjacent to the scissile bond in the substrates. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11333','AmyAc_SI_OligoGlu_DGase'),(1159351,'gnl|CDD|200470',_binary 'Alpha amylase catalytic domain found in oligo-1,6-glucosidase (also called isomaltase; sucrase-isomaltase; alpha-limit dextrinase) and related proteins.  Oligo-1,6-glucosidase (EC 3.2.1.10) hydrolyzes the alpha-1,6-glucosidic linkage of isomalto-oligosaccharides, pannose, and dextran. Unlike alpha-1,4-glucosidases (EC 3.2.1.20), it fails to hydrolyze the alpha-1,4-glucosidic bonds of maltosaccharides. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11331','AmyAc_OligoGlu_like'),(1159352,'gnl|CDD|200469',_binary 'Alpha amylase catalytic domain found in oligo-1,6-glucosidase (also called isomaltase; sucrase-isomaltase; alpha-limit dextrinase) and related proteins.  Oligo-1,6-glucosidase (EC 3.2.1.10) hydrolyzes the alpha-1,6-glucosidic linkage of isomalto-oligosaccharides, pannose, and dextran. Unlike alpha-1,4-glucosidases (EC 3.2.1.20), it fails to hydrolyze the alpha-1,4-glucosidic bonds of maltosaccharides. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11330','AmyAc_OligoGlu'),(1159353,'gnl|CDD|200467',_binary 'Alpha amylase catalytic domain found in maltase (also known as alpha glucosidase) and related proteins.  Maltase (EC 3.2.1.20) hydrolyzes the terminal, non-reducing (1->4)-linked alpha-D-glucose residues in maltose, releasing alpha-D-glucose. In most cases, maltase is equivalent to alpha-glucosidase, but the term \'maltase\' emphasizes the disaccharide nature of the substrate from which glucose is cleaved, and the term \'alpha-glucosidase\' emphasizes the bond, whether the substrate is a disaccharide or polysaccharide. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11328','AmyAc_maltase'),(1159354,'gnl|CDD|200494',_binary 'Alpha amylase catalytic domain found in Solute Carrier family 3 member 1 proteins.  SLC3A1, also called Neutral and basic amino acid transport protein rBAT or NBAT, plays a role in amino acid and cystine absorption. Mutations in the gene encoding SLC3A1 causes cystinuria, an autosomal recessive disorder characterized by the failure of proximal tubules to reabsorb filtered cystine and dibasic amino acids. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11359','AmyAc_SLC3A1'),(1159355,'gnl|CDD|274115',_binary 'alpha,alpha-phosphotrehalase.  Trehalose is a glucose disaccharide that serves in many biological systems as a compatible solute for protection against hyperosmotic and thermal stress. This family describes trehalose-6-phosphate hydrolase, product of the treC (or treA) gene, which is often found together with a trehalose uptake transporter and a trehalose operon repressor.','TIGR02403','Trehalose-6-phosphate_hyd'),(1159356,'gnl|CDD|200473',_binary 'Alpha amylase catalytic domain found in Trehalose synthetase.  Trehalose synthetase (TreS) catalyzes the reversible interconversion of trehalose and maltose. The enzyme catalyzes the reaction in both directions, but the preferred substrate is maltose. Glucose is formed as a by-product of this reaction. It is believed that the catalytic mechanism may involve the cutting of the incoming disaccharide and transfer of a glucose to an enzyme-bound glucose. This enzyme also catalyzes production of a glucosamine disaccharide from maltose and glucosamine. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11334','AmyAc_TreS'),(1159357,'gnl|CDD|223443',_binary 'Glycosidases [Carbohydrate transport and metabolism].','COG0366','AmyA'),(1159358,'gnl|CDD|200455',_binary 'Alpha amylase catalytic domain found in bacterial Alpha-amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes Chloroflexi, Dictyoglomi, and Fusobacteria. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11316','AmyAc_bac2_AmyA'),(1159359,'gnl|CDD|182849',_binary 'trehalose-6-phosphate hydrolase; Provisional.','PRK10933','PRK10933'),(1159360,'gnl|CDD|274140',_binary 'trehalose synthase.  Trehalose synthase interconverts maltose and alpha, alpha-trehalose by transglucosylation. This is one of at least three mechanisms for biosynthesis of trehalose, an important and widespread compatible solute. However, it is not driven by phosphate activation of sugars and its physiological role may tend toward trehalose degradation. This view is accentuated by numerous examples of fusion to a probable maltokinase domain. The sequence region described by this model is found both as the whole of a trehalose synthase and as the N-terminal region of a larger fusion protein that includes trehalose synthase activity. Several of these fused trehalose synthases have a domain homologous to proteins with maltokinase activity from Actinoplanes missouriensis and Streptomyces coelicolor (). [Energy metabolism, Biosynthesis and degradation of polysaccharides].','TIGR02456','Trehalose_synthase'),(1159361,'gnl|CDD|306608',_binary 'Alpha amylase, catalytic domain.  Alpha amylase is classified as family 13 of the glycosyl hydrolases. The structure is an 8 stranded alpha/beta barrel containing the active site, interrupted by a ~70 a.a. calcium-binding domain protruding between beta strand 3 and alpha helix 3, and a carboxyl-terminal Greek key beta-barrel domain.','pfam00128','Alpha-amylase'),(1159362,'gnl|CDD|200486',_binary 'Alpha amylase catalytic domain found in an uncharacterized protein family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The catalytic triad (DED) is not present here. The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11348','AmyAc_2'),(1159363,'gnl|CDD|200477',_binary 'Alpha amylase catalytic domain found in cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11338','AmyAc_CMD'),(1159364,'gnl|CDD|214758',_binary 'Alpha-amylase domain.','smart00642','Aamy'),(1159365,'gnl|CDD|200452',_binary 'Alpha amylase catalytic domain found in archaeal and bacterial Alpha-amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes firmicutes, bacteroidetes, and proteobacteria. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11313','AmyAc_arch_bac_AmyA'),(1159366,'gnl|CDD|200468',_binary 'Alpha amylase catalytic domain family found in maltase.  Maltase (EC 3.2.1.20) hydrolyzes the terminal, non-reducing (1->4)-linked alpha-D-glucose residues in maltose, releasing alpha-D-glucose. The catalytic triad (DED) which is highly conserved in the other maltase group is not present in this subfamily. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11329','AmyAc_maltase-like'),(1159367,'gnl|CDD|200463',_binary 'Alpha amylase catalytic domain found in Amylosucrase.  Amylosucrase is a glucosyltransferase that catalyzes the transfer of a D-glucopyranosyl moiety from sucrose onto an acceptor molecule. When the acceptor is another saccharide, only alpha-1,4 linkages are produced. Unlike most amylopolysaccharide synthases, it does not require any alpha-D-glucosyl nucleoside diphosphate substrate. In the presence of glycogen it catalyzes the transfer of a D-glucose moiety onto a glycogen branch, but in its absence, it hydrolyzes sucrose and synthesizes polymers, smaller maltosaccharides, and sucrose isoforms. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11324','AmyAc_Amylosucrase'),(1159368,'gnl|CDD|200451',_binary 'Alpha amylase catalytic domain family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; and C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost this catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd00551','AmyAc_family'),(1159370,'gnl|CDD|200459',_binary 'Alpha amylase catalytic domain found in maltogenic amylases, cyclodextrin glycosyltransferase, and related proteins.  Enzymes such as amylases, cyclomaltodextrinase (CDase), and cyclodextrin glycosyltransferase (CGTase) degrade starch to smaller oligosaccharides by hydrolyzing the alpha-D-(1,4) linkages between glucose residues. In the case of CGTases, an additional cyclization reaction is catalyzed yielding mixtures of cyclic oligosaccharides which are referred to as alpha-, beta-, or gamma-cyclodextrins (CDs), consisting of six, seven, or eight glucose residues, respectively. CGTases are characterized depending on the major product of the cyclization reaction. Besides having similar catalytic site residues, amylases and CGTases contain carbohydrate binding domains that are distant from the active site and are implicated in attaching the enzyme to raw starch granules and in guiding the amylose chain into the active site. The maltogenic alpha-amylase from Bacillus is a five-domain structure, unlike most alpha-amylases, but similar to that of cyclodextrin glycosyltransferase. In addition to the A, B, and C domains, they have a domain D and a starch-binding domain E. Maltogenic amylase is an endo-acting amylase that has activity on cyclodextrins, terminally modified linear maltodextrins, and amylose. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11320','AmyAc_AmyMalt_CGTase_like'),(1159371,'gnl|CDD|200481',_binary 'Alpha amylase catalytic domain found in sucrose phosphorylase (also called sucrose glucosyltransferase, disaccharide glucosyltransferase, and sucrose-phosphate alpha-D glucosyltransferase).  Sucrose phosphorylase is a bacterial enzyme that catalyzes the phosphorolysis of sucrose to yield glucose-1-phosphate and fructose. These enzymes do not have the conserved calcium ion present in other alpha amylase family enzymes. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11343','AmyAc_Sucrose_phosphoryla'),(1159372,'gnl|CDD|200479',_binary 'Alpha amylase catalytic domain found in bacterial cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). This group of CMDs is bacterial. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11340','AmyAc_bac_CMD_like_3'),(1159373,'gnl|CDD|236759',_binary 'maltodextrin glucosidase; Provisional.','PRK10785','PRK10785'),(1159374,'gnl|CDD|200491',_binary 'Alpha amylase catalytic domain found in bacterial cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). This group of CMDs is bacterial. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11354','AmyAc_bac_CMD_like'),(1159375,'gnl|CDD|200493',_binary 'Alpha amylase catalytic domain found in sucrose phosphorylase-like proteins (also called sucrose glucosyltransferase, disaccharide glucosyltransferase, and sucrose-phosphate alpha-D glucosyltransferase).  Sucrose phosphorylase is a bacterial enzyme that catalyzes the phosphorolysis of sucrose to yield glucose-1-phosphate and fructose. These enzymes do not have the conserved calcium ion present in other alpha amylase family enzymes. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11356','AmyAc_Sucrose_phosphoryla'),(1159376,'gnl|CDD|200490',_binary 'Alpha amylase catalytic domain found in eukaryotic and bacterial cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). This group of CMDs is mainly bacterial. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11353','AmyAc_euk_bac_CMD_like'),(1159377,'gnl|CDD|200489',_binary 'Alpha amylase catalytic domain found in an uncharacterized protein family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11352','AmyAc_5'),(1159378,'gnl|CDD|200488',_binary 'Alpha amylase catalytic domain found in an uncharacterized protein family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11350','AmyAc_4'),(1159379,'gnl|CDD|200476',_binary 'Alpha amylase catalytic domain found in cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). This group of CMDs is mainly bacterial. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11337','AmyAc_CMD_like'),(1159380,'gnl|CDD|200492',_binary 'Alpha amylase catalytic domain found in sucrose phosphorylase (also called sucrose glucosyltransferase, disaccharide glucosyltransferase, and sucrose-phosphate alpha-D glucosyltransferase).  Sucrose phosphorylase is a bacterial enzyme that catalyzes the phosphorolysis of sucrose to yield glucose-1-phosphate and fructose. These enzymes do not have the conserved calcium ion present in other alpha amylase family enzymes. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11355','AmyAc_Sucrose_phosphoryla'),(1159381,'gnl|CDD|200478',_binary 'Alpha amylase catalytic domain found in bacterial cyclomaltodextrinases and related proteins.  Cyclomaltodextrinase (CDase; EC3.2.1.54), neopullulanase (NPase; EC 3.2.1.135), and maltogenic amylase (MA; EC 3.2.1.133) catalyze the hydrolysis of alpha-(1,4) glycosidic linkages on a number of substrates including cyclomaltodextrins (CDs), pullulan, and starch. These enzymes hydrolyze CDs and starch to maltose and pullulan to panose by cleavage of alpha-1,4 glycosidic bonds whereas alpha-amylases essentially lack activity on CDs and pullulan. They also catalyze transglycosylation of oligosaccharides to the C3-, C4- or C6-hydroxyl groups of various acceptor sugar molecules. Since these proteins are nearly indistinguishable from each other, they are referred to as cyclomaltodextrinases (CMDs). This group of CMDs is bacterial. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11339','AmyAc_bac_CMD_like_2'),(1159382,'gnl|CDD|200454',_binary 'Alpha amylase catalytic domain found in bacterial Alpha-amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes Firmicutes, Proteobacteria, Actinobacteria, and Cyanobacteria. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11315','AmyAc_bac1_AmyA'),(1159383,'gnl|CDD|225819',_binary 'Maltooligosyl trehalose synthase [Carbohydrate transport and metabolism].','COG3280','TreY'),(1159384,'gnl|CDD|274113',_binary 'malto-oligosyltrehalose synthase.  This enzyme, formally named (1->4)-alpha-D-glucan 1-alpha-D-glucosylmutase, is the TreY enzyme of the TreYZ pathway of trehalose biosynthesis, an alternative to the OtsAB pathway. Trehalose may be incorporated into more complex compounds but is best known as compatible solute. It is one of the most effective osmoprotectants, and unlike the various betaines does not require nitrogen for its synthesis. [Energy metabolism, Biosynthesis and degradation of polysaccharides].','TIGR02401','Putative_maltooligosyl_tr'),(1159385,'gnl|CDD|200475',_binary 'Alpha amylase catalytic domain found in maltooligosyl trehalose synthase (MTSase).  Maltooligosyl trehalose synthase (MTSase) domain. MTSase and maltooligosyl trehalose trehalohydrolase (MTHase) work together to produce trehalose. MTSase is responsible for converting the alpha-1,4-glucosidic linkage to an alpha,alpha-1,1-glucosidic linkage at the reducing end of the maltooligosaccharide through an intramolecular transglucosylation reaction, while MTHase hydrolyzes the penultimate alpha-1,4 linkage of the reducing end, resulting in the release of trehalose. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11336','AmyAc_MTSase'),(1159386,'gnl|CDD|200483',_binary 'Alpha amylase catalytic domain found in solute carrier family 3 member 2 proteins.  4F2 cell-surface antigen heavy chain (hc) is a protein that in humans is encoded by the SLC3A2 gene. 4F2hc is a multifunctional type II membrane glycoprotein involved in amino acid transport and cell fusion, adhesion, and transformation. It is related to bacterial alpha-glycosidases, but lacks alpha-glycosidase activity. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11345','AmyAc_SLC3A2'),(1159387,'gnl|CDD|200487',_binary 'Alpha amylase catalytic domain found in an uncharacterized protein family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11349','AmyAc_3'),(1159388,'gnl|CDD|237740',_binary 'maltooligosyl trehalose synthase; Provisional.','PRK14511','PRK14511'),(1159389,'gnl|CDD|223373',_binary '1,4-alpha-glucan branching enzyme [Carbohydrate transport and metabolism].','COG0296','GlgB'),(1159390,'gnl|CDD|236543',_binary 'alpha-amylase; Reviewed.','PRK09505','malS'),(1159391,'gnl|CDD|200458',_binary 'Alpha amylase catalytic domain found in eukaryotic Alpha-amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes eukaryotic alpha-amylases including proteins from fungi, sponges, and protozoans. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11319','AmyAc_euk_AmyA'),(1159392,'gnl|CDD|237739',_binary 'putative bifunctional 4-alpha-glucanotransferase/glycogen debranching enzyme; Provisional.','PRK14510','PRK14510'),(1159393,'gnl|CDD|200482',_binary 'Alpha amylase catalytic domain found in GlgE-like proteins.  GlgE is a (1,4)-a-D-glucan:phosphate a-D-maltosyltransferase, involved in a-glucan biosynthesis in bacteria. It is also an anti-tuberculosis drug target. GlgE isoform I from Streptomyces coelicolor has the same catalytic and very similar kinetic properties to GlgE from Mycobacterium tuberculosis. GlgE from Streptomyces coelicolor forms a homodimer with each subunit comprising five domains (A, B, C, N, and S) and 2 inserts. Domain A is a catalytic alpha-amylase-type domain that along with domain N, which has a beta-sandwich fold and forms the core of the dimer interface, binds cyclodextrins. Domain A, B, and the 2 inserts define a well conserved donor pocket that binds maltose. Cyclodextrins competitively inhibit the binding of maltooligosaccharides to the S. coelicolor enzyme, indicating that the hydrophobic patch overlaps with the acceptor binding site. This is not the case in M. tuberculosis GlgE because cyclodextrins do not inhibit this enzyme, despite acceptor length specificity being conserved. Domain C is hypothesized to help stabilize domain A and could be involved in substrate binding. Domain S is a helix bundle that is inserted within the N domain and it plays a role in the dimer interface and interacts directly with domain B. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11344','AmyAc_GlgE_like'),(1159394,'gnl|CDD|236518',_binary 'cytoplasmic alpha-amylase; Reviewed.','PRK09441','PRK09441'),(1159395,'gnl|CDD|200457',_binary 'Alpha amylase catalytic domain found in bacterial and fungal Alpha amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes bacterial and fungal proteins. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11318','AmyAc_bac_fung_AmyA'),(1159396,'gnl|CDD|200485',_binary 'Alpha amylase catalytic domain found in an uncharacterized protein family.  The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11347','AmyAc_1'),(1159397,'gnl|CDD|200453',_binary 'Alpha amylase catalytic domain found in archaeal, bacterial, and plant Alpha-amylases (also called 1,4-alpha-D-glucan-4-glucanohydrolase).  AmyA (EC 3.2.1.1) catalyzes the hydrolysis of alpha-(1,4) glycosidic linkages of glycogen, starch, related polysaccharides, and some oligosaccharides. This group includes AmyA from bacteria, archaea, water fleas, and plants. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11314','AmyAc_arch_bac_plant_AmyA'),(1159398,'gnl|CDD|237737',_binary 'putative bifunctional 4-alpha-glucanotransferase/malto-oligosyltrehalose synthase; Provisional.','PRK14507','PRK14507'),(1159399,'gnl|CDD|237528',_binary 'sucrose phosphorylase; Provisional.','PRK13840','PRK13840'),(1159400,'gnl|CDD|273667',_binary 'alpha-1,4-glucan:alpha-1,4-glucan 6-glycosyltransferase.  This model describes the glycogen branching enzymes which are responsible for the transfer of chains of approx. 7 alpha(1--4)-linked glucosyl residues to other similar chains (in new alpha(1--6) linkages) in the biosynthesis of glycogen. This enzyme is a member of the broader amylase family of starch hydrolases which fold as (beta/alpha)8 barrels, the so-called TIM-barrel structure. All of the sequences comprising the seed of this model have been experimentally characterized. This model encompasses both bacterial and eukaryotic species. No archaea have this enzyme, although Aquifex aolicus does. Two species, Bacillus thuringiensis and Clostridium perfringens have two sequences each which are annotated as amylases. These annotations are aparrently in error. GP|18143720 from C. perfringens, for instance, contains the note \'674 aa, similar to gp:A14658_1 amylase (1,4-alpha-glucan branching enzyme (EC 2.4.1.18) ) from Bacillus thuringiensis (648 aa); 51.1% identity in 632 aa overlap.\' A branching enzyme from Porphyromonas gingivales, OMNI|PG1793, appears to be more closely related to the eukaryotic species (across a deep phylogenetic split) and may represent an instance of lateral transfer from this species\' host. A sequence from Arabidopsis thaliana, GP|9294564, scores just above trusted, but appears either to contain corrupt sequence or, more likely, to be a pseudogene as some of the conserved catalytic residues common to the alpha amylase family are not conserved here. [Energy metabolism, Biosynthesis and degradation of polysaccharides].','TIGR01515','branching_enzym'),(1159909,'gnl|CDD|315878',_binary 'Acetyltransferase (GNAT) domain.  This domain catalyzes N-acetyltransferase reactions.','pfam13302','Acetyltransf_3'),(1159910,'gnl|CDD|317008',_binary 'GCN5-related N-acetyl-transferase.  This family of GCN5-related N-acetyl-transferases bind both CoA and acetyl-CoA. They are characterized by highly conserved glycine, a cysteine residue in the acetyl-CoA binding site near the acetyl group, their small size compared with other GNATs and a lack of of an obvious substrate-binding site. It is proposed that they transfer an acetyl group from acetyl-CoA to one or more unidentified aliphatic amines via an acetyl (cysteine) enzyme intermediate. The substrate might be another macromolecule.','pfam14542','Acetyltransf_CG'),(1159911,'gnl|CDD|225928',_binary 'Predicted acetyltransferase [General function prediction only].','COG3393','COG3393'),(1159912,'gnl|CDD|316085',_binary 'Acetyltransferase (GNAT) domain.  This domain catalyzes N-acetyltransferase reactions.','pfam13527','Acetyltransf_9'),(1159913,'gnl|CDD|315324',_binary 'Arabinose-binding domain of AraC transcription regulator, N-term.  AraC is a bacterial transcriptional regulatory protein with a DNA-binding domain at the C-terminus, HTH_AraC, pfam00165, and this dimerisation domain which harbours the arabinose-binding pocket at the N-terminus. AraC positively and negatively regulates expression of the proteins required for the uptake and catabolism of the sugar L-arabinose 1,2,3].','pfam12625','Arabinose_bd'),(1159919,'gnl|CDD|239731',_binary 'proteasome beta type-6 subunit. The 20S proteasome, multisubunit proteolytic complex, is the central enzyme of nonlysosomal protein degradation in both the cytosol and nucleus. It is composed of 28 subunits arranged as four homoheptameric rings that stack on top of one another forming an elongated alpha-beta-beta-alpha cylinder with a central cavity. The proteasome alpha and beta subunits are members of the N-terminal nucleophile (Ntn)-hydrolase superfamily. Their N-terminal threonine residues are exposed as a nucleophile in peptide bond hydrolysis. Mammals have 7 alpha and 7 beta proteasome subunits while archaea have one of each.','cd03762','proteasome_beta_type_6'),(1159991,'gnl|CDD|312795',_binary 'RNA ligase.  This is a family of RNA ligases. The enzyme repairs RNA strand breaks in nicked DNA:RNA and RNA:RNA but not in DNA:DNA duplexes.','pfam09414','RNA_ligase'),(1160038,'gnl|CDD|214722',_binary 'LamG-like jellyroll fold domain.','smart00560','LamGL'),(1160048,'gnl|CDD|173886',_binary 'legume lectins.  The L-type (legume-type) lectins are a highly diverse family of carbohydrate binding proteins that generally display no enzymatic activity toward the sugars they bind.  This family includes arcelin, concanavalinA, the lectin-like receptor kinases, the ERGIC-53/VIP36/EMP46 type1 transmembrane proteins, and an alpha-amylase inhibitor.  L-type lectins have a dome-shaped beta-barrel carbohydrate recognition domain with a curved seven-stranded beta-sheet referred to as the \'front face\' and a flat six-stranded beta-sheet referred to as the \'back face\'.  This domain homodimerizes so that adjacent back sheets form a contiguous 12-stranded sheet and homotetramers occur by a back-to-back association of these homodimers.  Though L-type lectins exhibit both sequence and structural similarity to one another, their carbohydrate binding specificities differ widely.','cd01951','lectin_L-type'),(1160056,'gnl|CDD|309185',_binary 'Mnd1 family.  This family of proteins includes MND1 from S. cerevisiae. The mnd1 protein forms a complex with hop2 to promote homologous chromosome pairing and meiotic double-strand break repair.','pfam03962','Mnd1'),(1160204,'gnl|CDD|313019',_binary 'Zinc ribbon domain.  This entry represents a region of about 41 amino acids found in a number of small proteins in a wide range of bacteria. The region usually begins with the initiator Met and contains two CxxC motifs separated by 17 amino acids. One protein in this entry has been noted as a putative regulatory protein, designated FmdB. Most proteins in this entry have a C-terminal region containing highly degenerate sequence.','pfam09723','Zn-ribbon_8'),(1160251,'gnl|CDD|132489',_binary 'mycothiol synthase.  Members of this family are MshD, the acetyltransferase that catalyzes the final step of mycothiol biosynthesis in various members of the Actinomyctes, Mycothiol replaces glutathione in these species. [Biosynthesis of cofactors, prosthetic groups, and carriers, Glutathione and analogs].','TIGR03448','mycothiol_MshD'),(1160253,'gnl|CDD|226489',_binary 'Predicted acetyltransferase [General function prediction only].','COG3981','COG3981'),(1160646,'gnl|CDD|215433',_binary 'diaminohydroxyphosphoribosylaminopyrimidine deaminase.','PLN02807','PLN02807'),(1161279,'gnl|CDD|129632',_binary 'histidine decarboxylase, pyruvoyl type.  This enzyme converts histadine to histamine in a single step by catalyzing the release of CO2. This type is synthesized as an inactive single chain precursor, then cleaved into two chains. The Ser at the new N-terminus at the cleavage site is converted to a pyruvoyl group essential for activity. This type of histidine decarboxylase appears is known so far only in some Gram-positive bacteria, where it may play a role in amino acid catabolism. There is also a pyridoxal phosphate type histidine decarboxylase, as found in human, where histamine is a biologically active amine. [Energy metabolism, Amino acids and amines].','TIGR00541','Contains:_Histidine_decar'),(1161386,'gnl|CDD|187661',_binary 'flavonoid reductase (FR), extended (e) SDRs.  This subgroup contains FRs of the extended SDR-type and related proteins. These FRs act in the NADP-dependent reduction of  flavonoids, ketone-containing plant secondary metabolites; they have the characteristic active site triad of the SDRs (though not the upstream active site Asn) and a NADP-binding motif that is very similar to the typical extended SDR motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd08958','FR_SDR_e'),(1161389,'gnl|CDD|187654',_binary 'daunorubicin C-13 ketoreductase (KR), classical (c)-like SDRs.  Daunorubicin is a clinically important therapeutic compound used in some cancer treatments. Daunorubicin C-13 ketoreductase is member of the classical SDR family with a canonical glycine-rich NAD(P)-binding motif, but lacking a complete match to the active site tetrad characteristic of this group. The critical Tyr, plus the Lys and upstream Asn are present, but the catalytic Ser is replaced, generally by Gln. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type KRs have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd08951','DR_C-13_KR_SDR_c_like'),(1161398,'gnl|CDD|309701',_binary 'Protein similar to CwfJ C-terminus 1.  This region is found in the N terminus of Schizosaccharomyces pombe protein CwfJ. CwfJ is part of the Cdc5p complex involved in mRNA splicing.','pfam04677','CwfJ_C_1'),(1161490,'gnl|CDD|99972',_binary 'This is a family of GT1 glycosyltransferases found specifically in certain bacteria. amsK in Erwinia amylovora, has been reported to be involved in the biosynthesis of amylovoran, a exopolysaccharide acting as a virulence factor.','cd03799','GT1_amsK_like'),(1161492,'gnl|CDD|99969',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP-linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found mainly in bacteria and eukaryotes.','cd03795','GT1_like_4'),(1161513,'gnl|CDD|193550',_binary 'M20 Peptidase, carboxypeptidase yscS-like.  Peptidase M20 family, yscS (GlyX-carboxypeptidase, CPS1, carboxypeptidase S, carboxypeptidase a, carboxypeptidase yscS, glycine carboxypeptidase)-like subfamily. This group contains proteins that have been uncharacterized to date with similarity to vacuolar proteins involved in nitrogen metabolism which are essential for use of certain peptides that are sole nitrogen sources. YscS releases a C-terminal amino acid from a peptide that has glycine as the penultimate residue. It is synthesized as one polypeptide chain precursor which yields two active precursor molecules after carbohydrate modification in the secretory pathway. The proteolytically unprocessed forms are associated with the membrane, whereas the mature forms of the enzyme are soluble. Enzymes in this subfamily may also cleave intracellularly generated peptides in order to recycle amino acids for protein synthesis.','cd05675','M20_yscS_like'),(1161526,'gnl|CDD|130147',_binary 'DNA helicase II.  Designed to identify uvrD members of the uvrD/rep subfamily. [DNA metabolism, DNA replication, recombination, and repair].','TIGR01075','DNA_helicase_II'),(1161538,'gnl|CDD|225501',_binary 'Membrane-bound lytic murein transglycosylase B [Cell envelope biogenesis, outer membrane].','COG2951','MltB'),(1165212,'gnl|CDD|315278',_binary 'NMDA receptor-regulated protein 1.  This domain family is found in eukaryotes, and is approximately 40 amino acids in length. The family is found in association with pfam07719, pfam00515. There is a single completely conserved residue L that may be functionally important. NARP1 is the mammalian homolog of a yeast N-terminal acetyltransferase that regulates entry into the G(0) phase of the cell cycle.','pfam12569','NARP1'),(1165572,'gnl|CDD|226895',_binary '5\'(3\')-deoxyribonucleotidase [Nucleotide transport and metabolism].','COG4502','COG4502'),(1165574,'gnl|CDD|227950',_binary 'Uncharacterized conserved protein [Function unknown].','COG5663','COG5663'),(1165636,'gnl|CDD|222778',_binary 'lower collar protein.','PHA00148','PHA00148'),(1165637,'gnl|CDD|133855',_binary 'upper collar protein.','PHA00147','PHA00147'),(1165638,'gnl|CDD|147504',_binary 'Phage Connector (GP10).  The head-tail connector of bacteriophage 29 is composed of 12 36 kDa subunits with 12 fold symmetry. It is the central component of a rotary motor that packages the genomic dsDNA into pre-formed proheads. This motor consists of the head-tail connector, surrounded by a 29-encoded, 174-base, RNA and a viral ATPase protein.','pfam05352','Phage_connector'),(1165639,'gnl|CDD|222777',_binary 'major head protein.','PHA00144','PHA00144'),(1165643,'gnl|CDD|222817',_binary 'tail protein.','PHA00965','PHA00965'),(1165644,'gnl|CDD|133872',_binary 'tail protein.','PHA00380','PHA00380'),(1165646,'gnl|CDD|222779',_binary 'DNA encapsidation protein.','PHA00149','PHA00149'),(1165647,'gnl|CDD|310463',_binary 'Podovirus DNA encapsidation protein (Gp16).  This family consists of several DNA encapsidation protein (Gp16) sequences from the phi-29-like viruses. Gene product 16 catalyzes the in vivo and in vitro genome-encapsidation reaction.','pfam05894','Podovirus_Gp16'),(1165664,'gnl|CDD|222823',_binary 'hypothetical protein.','PHA01548','PHA01548'),(1165665,'gnl|CDD|222879',_binary 'DNA polymerase; Provisional.','PHA02563','PHA02563'),(1165666,'gnl|CDD|281206',_binary 'DNA polymerase type B, organellar and viral.  Like pfam00136, members of this family are also DNA polymerase type B proteins. Those included here are found in plant and fungal mitochondria, and in viruses.','pfam03175','DNA_pol_B_2'),(1165804,'gnl|CDD|225064',_binary 'Predicted acyltransferase [General function prediction only].','COG2153','ElaA'),(1165907,'gnl|CDD|236183',_binary 'purine nucleoside phosphorylase; Provisional.','PRK08202','PRK08202'),(1165934,'gnl|CDD|130988',_binary 'HflK protein.  HflK and HflC are paralogs encoded by tandem genes in Proteobacteria, spirochetes, and some other bacterial lineages. The HflKC complex is anchored in the membrane and exposed to the periplasm. The complex is not active as a protease, but rather binds to and appears to modulate the ATP-dependent protease FtsH. The overall function of HflKC is not fully described.//Regulation of FtsH by HflKC appears to be negative (SS 8/27/03].','TIGR01933','Protein_HflK'),(1165937,'gnl|CDD|259801',_binary 'Stomatin, a subgroup of the stomatin-like proteins (slipins) family; belonging to the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  Stomatin (or band 7) is widely expressed and, highly expressed in red blood cells. It localizes predominantly to the plasma membrane and to intracellular vesicles of the endocytic pathway, where it is present in higher order homo-oligomeric complexes (of between 9 and 12 monomers). Stomatin interacts with and regulates members of the degenerin/epithelia Na+ channel family in mechanosensory cells of Caenorhabditis elegans and vertebrate neurons and, is implicated in trafficking of Glut1 glucose transporters. This subgroup found in animals, also contains proteins similar to Caenorhabditis elegans MEC-2. MEC-2 interacts with MEC-4, which is part of the degenerin channel complex required for response to gentle body touch.','cd03403','SPFH_stomatin'),(1165939,'gnl|CDD|259798',_binary 'Flotillin or reggie family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  The flotillin (reggie) like proteins are lipid raft-associated. Individual proteins of this SPFH family may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. In addition, microdomains formed from flotillin proteins may be dynamic units with their own regulatory functions. Flotillins have been implicated in signal transduction, vesicle trafficking, cytoskeleton rearrangement and interact with a variety of proteins. They may play a role in the progression of prion disease, in the pathogenesis of neurodegenerative diseases such as Parkinson\'s and Alzheimer\'s disease and in cancer invasion, and metastasis.','cd03399','SPFH_flotillin'),(1166411,'gnl|CDD|227819',_binary 'Uncharacterized conserved protein [Function unknown].','COG5532','COG5532'),(1167179,'gnl|CDD|197870',_binary 'This domain is found in Saccharomyces cerevisiae protein SMP2, proteins with an N-terminal lipin domain and phosphatidylinositol transfer proteins.  SMP2 is involved in plasmid maintenance and respiration. Lipin proteins are involved in adipose tissue development and insulin resistance.','smart00775','LNS2'),(1168385,'gnl|CDD|223558',_binary 'Predicted tRNA(5-methylaminomethyl-2-thiouridylate) methyltransferase, contains the PP-loop ATPase domain [Translation, ribosomal structure and biogenesis].','COG0482','TrmU'),(1168386,'gnl|CDD|234664',_binary 'tRNA-specific 2-thiouridylase MnmA; Reviewed.','PRK00143','mnmA'),(1168387,'gnl|CDD|281101',_binary 'tRNA methyl transferase.  This family represents tRNA(5-methylaminomethyl-2-thiouridine)-methyltransferase which is involved in the biosynthesis of the modified nucleoside 5-methylaminomethyl-2-thiouridine present in the wobble position of some tRNAs.','pfam03054','tRNA_Me_trans'),(1168958,'gnl|CDD|313461',_binary 'Clusterin-associated protein-1.  This protein is conserved from worms to humans. The protein of 413 amino acids contains a central coiled-coil domain, possibly the region that binds to clusterin. Cluap1 expression is highest in the nucleus and gradually increases during late S to G2/M phases of the cell cycle and returns to the basal level in the G0/G1 phases. In addition, it is upregulated in colon cancer tissues compared to corresponding non-cancerous mucosa. It thus plays a crucial role in the life of the cell.','pfam10234','Cluap1'),(1169108,'gnl|CDD|213246',_binary 'ATP-binding cassette domain of sbcCD.  SbcCD and other Mre11/Rad50 (MR) complexes are implicated in the metabolism of DNA ends. They cleave ends sealed by hairpin structures and are thought to play a role in removing protein bound to DNA termini.','cd03279','ABC_sbcCD'),(1169235,'gnl|CDD|237110',_binary 'hypothetical protein; Provisional.','PRK12472','PRK12472'),(1169240,'gnl|CDD|306717',_binary 'Tropomyosin.','pfam00261','Tropomyosin'),(1169351,'gnl|CDD|100053',_binary 'Maltose O-acyltransferase (MAT)-like: This family is composed of maltose O-acetyltransferase, galactoside O-acetyltransferase (GAT), xenobiotic acyltransferase (XAT) and similar proteins. MAT and GAT catalyze the CoA-dependent acetylation of the 6-hydroxyl group of their respective sugar substrates. MAT acetylates maltose and glucose exclusively while GAT specifically acetylates galactopyranosides. XAT catalyzes the CoA-dependent acetylation of a variety of hydroxyl-bearing acceptors such as chloramphenicol and streptogramin, among others. XATs are implicated in inactivating xenobiotics leading to xenobiotic resistance in patients. Members of this family contain a a left-handed parallel beta-helix (LbH) domain with at least 5 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). They are trimeric in their active form.','cd04647','LbH_MAT_like'),(1169352,'gnl|CDD|100047',_binary 'Maltose O-acetyltransferase (MAT) and Galactoside O-acetyltransferase (GAT): MAT and GAT catalyze the CoA-dependent acetylation of the 6-hydroxyl group of their respective sugar substrates. MAT acetylates maltose and glucose exclusively at the C6 position of the nonreducing end glucosyl moiety. GAT specifically acetylates galactopyranosides. Furthermore, MAT shows higher affinity toward artificial substrates containing an alkyl or hydrophobic chain as well as a glucosyl unit. Active MAT and GAT are homotrimers, with each subunit consisting of an N-terminal alpha-helical region and a C-terminal left-handed parallel alpha-helix (LbH) subdomain with 6 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).','cd03357','LbH_MAT_GAT'),(1169353,'gnl|CDD|223188',_binary 'Acetyltransferase (isoleucine patch superfamily) [General function prediction only].','COG0110','WbbJ'),(1169354,'gnl|CDD|236703',_binary 'putative acyl transferase; Provisional.','PRK10502','PRK10502'),(1169355,'gnl|CDD|100063',_binary 'wcaF-like: This group is composed of the protein product of the E. coli wcaF gene and similar proteins. WcaF is part of the gene cluster responsible for the biosynthesis of the extracellular polysaccharide colanic acid. The wcaF protein is predicted to contain a left-handed parallel beta-helix (LbH) domain encoded by imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. Many are trimeric in their active forms.','cd05825','LbH_wcaF_like'),(1169356,'gnl|CDD|100040',_binary 'Xenobiotic acyltransferase (XAT): The XAT class of hexapeptide acyltransferases is composed of a large number of microbial enzymes that catalyze the CoA-dependent acetylation of a variety of hydroxyl-bearing acceptors such as chloramphenicol and streptogramin, among others. Members of this class of enzymes include Enterococcus faecium streptogramin A acetyltransferase and Pseudomonas aeruginosa chloramphenicol acetyltransferase. They contain repeated copies of a six-residue hexapeptide repeat sequence motif (X-[STAV]-X-[LIV]-[GAED]-X) and adopt a left-handed parallel beta helix (LbH) structure. The active enzyme is a trimer with CoA and substrate binding sites at the interface of two separate LbH subunits. XATs are implicated in inactivating xenobiotics leading to xenobiotic resistance in patients.','cd03349','LbH_XAT'),(1169357,'gnl|CDD|100048',_binary 'WcxM-like, Left-handed parallel beta-Helix (LbH) N-terminal domain: This group is composed of Xanthomonas campestris WcxM and proteins with similarity to the WcxM N-terminal domain. WcxM is thought to be bifunctional, catalyzing both the isomerization and transacetylation reactions of keto-hexoses. It contains an N-terminal LbH domain responsible for the transacetylation function and a C-terminal isomerase domain. The LbH domain contains imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), typical of enzymes with acyltransferase activity.','cd03358','LbH_WxcM_N_like'),(1169359,'gnl|CDD|182235',_binary 'maltose O-acetyltransferase; Provisional.','PRK10092','PRK10092'),(1169361,'gnl|CDD|181930',_binary 'galactoside O-acetyltransferase; Reviewed.','PRK09527','lacA'),(1169624,'gnl|CDD|309098',_binary 'Recombination protein U.','pfam03838','RecU'),(1169626,'gnl|CDD|225868',_binary 'Penicillin-binding protein-related factor A, putative recombinase [General function prediction only].','COG3331','PrfA'),(1170170,'gnl|CDD|176652',_binary 'inactive DEDDy 3\'-5\' exonuclease domain of Bacillus stearothermophilus DNA polymerase I and similar family-A DNA polymerases.  Bacillus stearothermophilus-like Polymerase I (Pol I), a subgroup of the family-A DNA polymerases, contains an inactive DnaQ-like 3\'-5\' exonuclease domain in the same polypeptide chain as the polymerase region. The exonuclease-like domain of these proteins possess the same fold as the Klenow fragment (KF) of Escherichia coli Pol I, but does not contain the four critical metal-binding residues necessary for activity. The function of this domain is unknown. It might act as a spacer between the polymerase and the 5\'-3\' exonuclease domains. Some members of this subgroup, such as those from Bacillus sphaericus and Thermus aquaticus, are thermostable DNA polymerases.','cd06140','DNA_polA_I_Bacillus_like_'),(1170212,'gnl|CDD|309182',_binary 'Serine hydrolase (FSH1).  This is a family of serine hydrolases.','pfam03959','FSH1'),(1170698,'gnl|CDD|316944',_binary 'Bacterial PH domain.  Proteins in this family are distantly related to PH domains.','pfam14470','bPH_3'),(1174279,'gnl|CDD|107038',_binary 'hypothetical protein.','PHA01516','PHA01516'),(1174294,'gnl|CDD|238977',_binary 'Nucleoside/nucleotide kinase (NK) is a protein superfamily consisting of multiple families of enzymes that share structural similarity and are functionally related to the catalysis of the reversible phosphate group transfer from nucleoside triphosphates to nucleosides/nucleotides, nucleoside monophosphates, or sugars. Members of this family play a wide variety of essential roles in nucleotide metabolism, the biosynthesis of coenzymes and aromatic compounds, as well as the metabolism of sugar and sulfate.','cd02019','NK'),(1174426,'gnl|CDD|273166',_binary 'DNA excision repair protein (rad2).  All proteins in this family for which functions are known are flap endonucleases that generate the 3\' incision next to DNA damage as part of nucleotide excision repair. This family is related to many other flap endonuclease families including the fen1 family. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00600','DNA_repair_protein_comple'),(1174427,'gnl|CDD|189038',_binary 'PIN domain of Xeroderma pigmentosum complementation group G (XPG) nuclease, a structure-specific, divalent-metal-ion dependent, 5\' nuclease and homologs.  The Xeroderma pigmentosum complementation group G (XPG) nuclease plays a central role in nucleotide excision repair (NER) in cleaving DNA bubble structures or loops. XPG is a member of the structure-specific, 5\' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region (I domain). In XPG PIN domains, this arch region can be quite variable and extensive (400 - 800 residues) in length and is required for NER activity and for efficient processing of bubble substrates. Inserted within the PIN domain of these 5\' nucleases is a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region. Both the H3TH domain (not included here) and the helical arch/clamp region are involved in DNA binding. Nucleases within this group also have a carboxylate-rich active site that is involved in binding essential divalent metal ion cofactors (Mg2+/Mn2+).','cd09868','PIN_XPG'),(1174866,'gnl|CDD|223444',_binary 'Asparagine synthase (glutamine-hydrolyzing) [Amino acid transport and metabolism].','COG0367','AsnB'),(1174869,'gnl|CDD|238950',_binary 'N-terminal domain of predicted ATPase of the PP-loop faimly implicated in cell cycle control [Cell division and chromosome partitioning]. This is a subfamily of Adenine nucleotide alpha hydrolases superfamily.Adeninosine nucleotide alpha hydrolases superfamily  includes N type ATP PPases and ATP sulphurylases. It forms a apha/beta/apha fold which  binds to Adenosine group.  This domain has  a strongly conserved motif SGGXD at the N terminus.','cd01992','PP-ATPase'),(1174870,'gnl|CDD|274129',_binary 'tRNA(Ile)-lysidine synthetase, N-terminal domain.  The only examples in which the wobble position of a tRNA must discriminate between G and A of mRNA are AUA (Ile) vs. AUG (Met) and UGA (stop) vs. UGG (Trp). In all bacteria, the wobble position of the tRNA(Ile) recognizing AUA is lysidine, a lysine derivative of cytidine. This family describes a protein domain found, apparently, in all bacteria in a single copy. Eukaryotic sequences appear to be organellar. The domain archictecture of this protein family is variable; some, including characterized proteins of E. coli and B. subtilis known to be tRNA(Ile)-lysidine synthetase, include a conserved 50-residue domain that many other members lack. This protein belongs to the ATP-binding PP-loop family ( pfam01171). It appears in the literature and protein databases as TilS, YacA, and putative cell cycle protein MesJ (a misnomer). [Protein synthesis, tRNA and rRNA base modification].','TIGR02432','tRNAIle-lysidine_synthase'),(1174871,'gnl|CDD|279510',_binary 'PP-loop family.  This family of proteins belongs to the PP-loop superfamily.','pfam01171','ATP_bind_3'),(1174873,'gnl|CDD|238949',_binary 'The C-terminal domain of Asparagine Synthase B. This domain is always found associated n-terminal amidotransferase domain. Family members that contain this domain catalyse the conversion of aspartate to asparagine. Asparagine synthetase B  catalyzes the assembly of asparagine from aspartate, Mg(2+)ATP, and glutamine. The three-dimensional architecture of the N-terminal domain of asparagine synthetase B is similar to that observed for glutamine phosphoribosylpyrophosphate amidotransferase while the molecular motif of the C-domain is reminiscent to that observed for GMP synthetase .','cd01991','Asn_Synthase_B_C'),(1174874,'gnl|CDD|238944',_binary 'Adenine nucleotide alpha hydrolases superfamily  including N type ATP PPases and ATP sulphurylases. The domain forms a apha/beta/apha fold which  binds to Adenosine group..','cd01986','Alpha_ANH_like'),(1175931,'gnl|CDD|315810',_binary 'Protein of unknown function (DUF4035).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 67 and 93 amino acids in length.','pfam13227','DUF4035'),(1175935,'gnl|CDD|315473',_binary 'MobA-like NTP transferase domain.  This family includes the MobA protein (Molybdopterin-guanine dinucleotide biosynthesis protein A). The family also includes a wide range of other NTP transferase domain.','pfam12804','NTP_transf_3'),(1176110,'gnl|CDD|223256',_binary 'Excinuclease ATPase subunit [DNA replication, recombination, and repair].','COG0178','UvrA'),(1176505,'gnl|CDD|307549',_binary 'Apolipoprotein A1/A4/E domain.  These proteins contain several 22 residue repeats which form a pair of alpha helices. This family includes: Apolipoprotein A-I. Apolipoprotein A-IV. Apolipoprotein E.','pfam01442','Apolipoprotein'),(1179429,'gnl|CDD|306631',_binary 'Pou domain - N-terminal to homeobox domain.','pfam00157','Pou'),(1179451,'gnl|CDD|320686',_binary 'HtpX-like membrane-bound metallopeptidase.  This family contains peptidase M48 subfamily B, also known as HtpX, which consists of proteins smaller than Ste24p, with homology restricted to the C-terminal half of Ste24p. HtpX, an integral membrane (IM) metallopeptidase, is widespread in bacteria and archaea, and plays a central role in protein quality control by preventing the accumulation of misfolded proteins in the membrane. Its expression is controlled by the Cpx stress response system, which senses abnormal membrane proteins. HtpX participates in the proteolytic quality control of these misfolded proteins by undergoing self-degradation and eliminating them by collaborating with FtsH, a membrane-bound and ATP-dependent protease. HtpX contains the zinc binding motif (HEXXH), has an FtsH-like topology, and is capable of introducing endoproteolytic cleavages into SecY (also an FtsH substrate). However, HtpX does not have an ATPase activity and will only act against cytoplasmic regions of a target membrane protein. Thus, HtpX and FtsH have overlapping and/or complementary functions, which are especially important at high temperature; in E. coli and Xylella fastidiosa, HtpX is heat-inducible, while in Streptococcus gordonii it is not. Mutation studies of HtpX-like M48 metalloprotease from Leptospira interrogans (LA4131) has been shown to result in altered expression of a subset of metal toxicity and stress response genes.','cd07327','M48B_HtpX_like'),(1180879,'gnl|CDD|183271',_binary 'murein transglycosylase C; Provisional.','PRK11671','mltC'),(1181655,'gnl|CDD|280928',_binary 'Carbohydrate binding domain.  This short domain is found in many different glycosyl hydrolase enzymes and is presumed to have a carbohydrate binding function. The domain has six aromatic groups that may be important for binding.','pfam02839','CBM_5_12'),(1181711,'gnl|CDD|283041',_binary 'Virulence-associated protein E.  This family contains several bacterial virulence-associated protein E like proteins. These proteins contain a P-loop motif.','pfam05272','VirE'),(1181713,'gnl|CDD|227832',_binary 'Predicted P-loop ATPase and inactivated derivatives [General function prediction only].','COG5545','COG5545'),(1182842,'gnl|CDD|100050',_binary 'Putative Acyltransferase (AT), Left-handed parallel beta-Helix (LbH) domain; This group is composed of mostly uncharacterized proteins containing an N-terminal helical subdomain followed by a LbH domain. The alignment contains 6 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. A few members are identified as NeuD, a sialic acid (Sia) O-acetyltransferase that is required for Sia synthesis and surface polysaccharide sialylation.','cd03360','LbH_AT_putative'),(1182844,'gnl|CDD|137467',_binary 'putative lipopolysaccharide biosynthesis O-acetyl transferase WbbJ; Provisional.','PRK09677','PRK09677'),(1182846,'gnl|CDD|100051',_binary 'Gamma carbonic anhydrase-like: This family is composed of gamma carbonic anhydrase (CA), Ferripyochelin Binding Protein (FBP), E. coli paaY protein, and similar proteins. CAs are zinc-containing enzymes that catalyze the reversible hydration of carbon dioxide in a two-step mechanism, involving the nucleophilic attack of a zinc-bound hydroxide ion on carbon dioxide, followed by the regeneration of the active site by ionization of the zinc-bound water molecule and removal of a proton from the active site. They are ubiquitous enzymes involved in fundamental processes like photosynthesis, respiration, pH homeostasis and ion transport. There are three evolutionary distinct groups - alpha, beta and gamma carbonic anhydrases - which show no significant sequence identity or structural similarity. Gamma CAs are trimeric enzymes with left-handed parallel beta helix (LbH) structural domain.','cd04645','LbH_gamma_CA_like'),(1182849,'gnl|CDD|132351',_binary 'phosphonate metabolim protein, transferase hexapeptide repeat family.  This family of proteins contains copies of the Bacterial transferase hexapeptide repeat family (pfam00132) and is only found in operons encoding the phosphonate C-P lyase system (GenProp0232). Many C-P lyase operons, however, lack a homolog of this protein.','TIGR03308','phn_thr-fam'),(1182853,'gnl|CDD|274656',_binary 'sugar O-acyltransferase, sialic acid O-acetyltransferase NeuD family.  This family of proteins includes the characterized NeuD sialic acid O-acetyltransferase enzymes from E. coli and Streptococcus agalactiae (group B strep). These two are quite closely related to one another, so extension of this annotation to other members of the family in unsupported without additional independent evidence. The neuD gene is often observed in close proximity to the neuABC genes for the biosynthesis of CMP-N-acetylneuraminic acid (CMP-sialic acid), and NeuD sequences from these organisms were used to construct the seed for this model. Nevertheless, there are numerous instances of sequences identified by this model which are observed in a different genomic context (although almost universally in exopolysaccharide biosynthesis-related loci), as well as in genomes for which the biosynthesis of sialic acid (SA) is undemonstrated. Even in the cases where the association with SA biosynthesis is strong, it is unclear in the literature whether the biological substrate is SA iteself, CMP-SA, or a polymer containing SA. Similarly, it is unclear to what extent the enzyme has a preference for acetylation at the 7, 8 or 9 positions. In the absence of evidence of association with SA, members of this family may be involved with the acetylation of differring sugar substrates, or possibly the delivery of alternative acyl groups. The closest related sequences to this family (and those used to root the phylogenetic tree constructed to create this model) are believed to be succinyltransferases involved in lysine biosynthesis. These proteins contain repeats of the bacterial transferase hexapeptide (pfam00132), although often these do not register above the trusted cutoff.','TIGR03570','NeuD_NnaD'),(1182858,'gnl|CDD|132571',_binary '2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-acetyltransferase.  This enzyme is part of the diaminopimelate pathway of lysine biosynthesis. Alternate name: tetrahydrodipicolinate N-acetyltransferase. Note that IUBMB lists this alternate name as the accepted name. Unfortunately, the related succinyl transferase acting on the same substrate (EC:2.3.1.117, TIGR00695) uses the opposite standard. We have decided to give these two enzymes names which more clearly indicated that they act on the same substrate.','TIGR03532','DapD_Ac'),(1182859,'gnl|CDD|225082',_binary 'Tetrahydrodipicolinate N-succinyltransferase [Amino acid transport and metabolism].','COG2171','DapD'),(1182860,'gnl|CDD|223975',_binary 'Serine acetyltransferase [Amino acid transport and metabolism].','COG1045','CysE'),(1182861,'gnl|CDD|223735',_binary 'Carbonic anhydrases/acetyltransferases, isoleucine patch superfamily [General function prediction only].','COG0663','PaaY'),(1182862,'gnl|CDD|100045',_binary 'Serine acetyltransferase (SAT): SAT catalyzes the CoA-dependent acetylation of the side chain hydroxyl group of L-serine to form O-acetylserine, as the first step of a two-step biosynthetic pathway in bacteria and plants leading to the formation of L-cysteine. This reaction represents a key metabolic point of regulation for the cysteine biosynthetic pathway due to its feedback inhibition by cysteine. The enzyme is a 175 kDa homohexamer, composed of a dimer of homotrimers. Each subunit contains an N-terminal alpha helical region and a C-terminal left-handed beta-helix (LbH) subdomain with 5 turns, each containing a hexapeptide repeat motif characteristic of the acyltransferase superfamily of enzymes. The trimer interface mainly involves the C-terminal LbH subdomain while the dimer (of trimers) interface is mediated by the N-terminal alpha helical subdomain.','cd03354','LbH_SAT'),(1182863,'gnl|CDD|234858',_binary 'UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase; Provisional.','PRK00892','lpxD'),(1182865,'gnl|CDD|188523',_binary 'colanic acid biosynthesis acetyltransferase WcaF.  This gene is one of the glycosyl transferases involved in the biosynthesis of colanic acid, an exopolysaccharide expressed in Enterobacteraceae species. This acetyltransferase is believed to catalyze the addition of the acetyl group that is attached through an O linkage to the first fucosyl residue of the colanic acid repetitive unit (E unit).','TIGR04008','WcaF'),(1182866,'gnl|CDD|100043',_binary 'UDP-3-O-acyl-glucosamine N-acyltransferase (LpxD): The enzyme catalyzes the transfer of 3-hydroxymyristic acid or 3-hydroxy-arachidic acid, depending on the organism, from the acyl carrier protein (ACP) to UDP-3-O-acyl-glucosamine to produce UDP-2,3-diacyl-GlcNAc. This constitutes the third step in the lipid A biosynthetic pathway in Gram-negative bacteria. LpxD is a homotrimer, with each subunit consisting of a novel combination of an N-terminal uridine-binding domain, a core lipid-binding left-handed parallel beta helix (LbH) domain, and a C-terminal alpha-helical extension. The LbH domain contains 9 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).','cd03352','LbH_LpxD'),(1182870,'gnl|CDD|100038',_binary 'Left-handed parallel beta-Helix (LbetaH or LbH) domain: The alignment contains 5 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity, however, some subfamilies in this hierarchy also show activities related to ion transport or translation initiation. Many are trimeric in their active forms.','cd00208','LbetaH'),(1182871,'gnl|CDD|178297',_binary 'serine O-acetyltransferase.','PLN02694','PLN02694'),(1182872,'gnl|CDD|273834',_binary 'UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase LpxD.  This model describes LpxD, an enzyme for the biosynthesis of lipid A, a component oflipopolysaccharide (LPS) in the outer membrane outer leaflet of most Gram-negative bacteria. Some differences are found between lipid A of different species. This protein represents the third step from UDP-N-acetyl-D-glucosamine. The group added at this step generally is 14:0(3-OH) (myristate) but may vary; in Aquifex it appears to be 16:0(3-OH) (palmitate). [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01853','UDP-3-O-acylglucosamine_N'),(1182877,'gnl|CDD|200082',_binary 'serine O-acetyltransferase.  Cysteine biosynthesis [Amino acid biosynthesis, Serine family].','TIGR01172','Serine_acetyltransferase'),(1182878,'gnl|CDD|215205',_binary 'serine acetyltransferase.','PLN02357','PLN02357'),(1182879,'gnl|CDD|100041',_binary '2,3,4,5-tetrahydropyridine-2,6-dicarboxylate (THDP) N-succinyltransferase (also called THP succinyltransferase): THDP N-succinyltransferase catalyzes the conversion of tetrahydrodipicolinate and succinyl-CoA to N-succinyltetrahydrodipicolinate and CoA. It is the committed step in the succinylase pathway by which bacteria synthesize L-lysine and meso-diaminopimelate, a component of peptidoglycan. The enzyme is homotrimeric and each subunit contains an N-terminal region with alpha helices and hairpin loops, as well as a C-terminal region with a left-handed parallel alpha-helix (LbH) structural motif encoded by hexapeptide repeat motifs.','cd03350','LbH_THP_succinylT'),(1182881,'gnl|CDD|306610',_binary 'Bacterial transferase hexapeptide (six repeats).','pfam00132','Hexapep'),(1182882,'gnl|CDD|317055',_binary 'Hexapeptide repeat of succinyl-transferase.','pfam14602','Hexapep_2'),(1182883,'gnl|CDD|100049',_binary 'Dynactin 5 (or subunit p25); Dynactin is a major component of the activator complex that stimulates dynein-mediated vesicle transport. Dynactin is a heterocomplex of at least eight subunits, including a 150,000-MW protein called Glued, the actin-capping protein Arp1, and dynamatin. In vitro binding experiments show that dynactin enhances dynein-dependent motility, possibly through interaction with microtubules and vesicles. Subunit p25 is part of the pointed-end subcomplex in dynactin that also includes p26, p27, and Arp11. This subcomplex interacts with membranous cargoes. p25 and p27 contain imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), indicating a left-handed parallel beta helix (LbH) structural domain. Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.','cd03359','LbH_Dynactin_5'),(1182884,'gnl|CDD|215394',_binary 'serine acetyltransferase.','PLN02739','PLN02739'),(1182886,'gnl|CDD|182987',_binary 'serine acetyltransferase; Provisional.','PRK11132','cysE'),(1182887,'gnl|CDD|235390',_binary 'UDP-N-acetylglucosamine acyltransferase; Provisional.','PRK05289','PRK05289'),(1182889,'gnl|CDD|223974',_binary 'UDP-3-O-[3-hydroxymyristoyl].','COG1044','LpxD'),(1182894,'gnl|CDD|236996',_binary '2,3,4,5-tetrahydropyridine-2,6-carboxylate N-succinyltransferase; Provisional.','PRK11830','dapD'),(1182895,'gnl|CDD|100042',_binary 'UDP-N-acetylglucosamine O-acyltransferase (UDP-GlcNAc acyltransferase): Proteins in this family catalyze the transfer of (R)-3-hydroxymyristic acid from its acyl carrier protein thioester to UDP-GlcNAc. It is the first enzyme in the lipid A biosynthetic pathway and is also referred to as LpxA. Lipid A is essential for the growth of Escherichia coli and related bacteria. It is also essential for maintaining the integrity of the outer membrane. UDP-GlcNAc acyltransferase is a homotrimer of left-handed parallel beta helix (LbH) subunits. Each subunit contains an N-terminal LbH region with 9 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), and a C-terminal alpha-helical region.','cd03351','LbH_UDP-GlcNAc_AT'),(1182897,'gnl|CDD|274093',_binary 'non-ribosomal peptide synthetase terminal domain of unknown function.  This domain is found exclusively in non-ribosomal peptide synthetases and always as the final domain in the polypeptide. This domain is roughly 700 amino acids in size and is found in polypeptides roughly twice that size.','TIGR02353','NON-RIBOSOMAL_PEPTIDE_SYN'),(1182898,'gnl|CDD|100058',_binary 'paaY-like: This group is composed by uncharacterized proteins with similarity to the protein product of the E. coli paaY gene, which is part of the paa gene cluster responsible for phenylacetic acid degradation. Proteins in this group are expected to adopt the left-handed parallel beta-helix (LbH) structure. They contain imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Similarity to gamma carbonic anhydrase and Ferripyochelin Binding Protein (FBP) may suggest metal binding capacity.','cd04745','LbH_paaY_like'),(1182899,'gnl|CDD|100060',_binary 'Putative glucose-1-phosphate thymidylyltransferase, C-terminal Left-handed parallel beta-Helix (LbH) domain: Proteins in this family show simlarity to glucose-1-phosphate adenylyltransferases in that they contain N-terminal catalytic domains that resemble a dinucleotide-binding Rossmann fold and C-terminal LbH fold domains. Members in this family are predicted to be glucose-1-phosphate thymidylyltransferases, which are involved in the dTDP-L-rhamnose biosynthetic pathway. Glucose-1-phosphate thymidylyltransferase catalyzes the synthesis of deoxy-thymidine di-phosphate (dTDP)-L-rhamnose, an important component of the cell wall of many microorganisms. The C-terminal LbH domain contains multiple turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.','cd05636','LbH_G1P_TT_C_like'),(1182900,'gnl|CDD|182295',_binary 'putative acyl transferase; Provisional.','PRK10191','PRK10191'),(1182901,'gnl|CDD|183539',_binary 'UDP-N-acetylglucosamine acyltransferase; Provisional.','PRK12461','PRK12461'),(1182902,'gnl|CDD|237685',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK14355','glmU'),(1182903,'gnl|CDD|188531',_binary 'colanic acid biosynthesis acetyltransferase WcaB.  This gene is one of the acetyl transferases involved in the biosynthesis of colanic acid, an exopolysaccharide expressed in Enterobacteraceae species.','TIGR04016','WcaB'),(1182905,'gnl|CDD|100044',_binary 'N-acetyl-glucosamine-1-phosphate uridyltransferase (GlmU), C-terminal left-handed beta-helix (LbH) acetyltransferase domain: GlmU is also known as UDP-N-acetylglucosamine pyrophosphorylase. It is a bifunctional bacterial enzyme that catalyzes two consecutive steps in the formation of UDP-N-acetylglucosamine (UDP-GlcNAc), an important precursor in bacterial cell wall formation. The two enzymatic activities, uridyltransferase and acetyltransferase, are carried out by two independent domains. The C-terminal LbH domain possesses the acetyltransferase activity. It catalyzes the CoA-dependent acetylation of GlcN-1-phosphate to GlcNAc-1-phosphate. The LbH domain contains 10 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X. The acetyltransferase active site is located at the interface between two subunits of the active LbH trimer.','cd03353','LbH_GlmU_C'),(1182907,'gnl|CDD|237686',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK14356','glmU'),(1182908,'gnl|CDD|224128',_binary 'N-acetylglucosamine-1-phosphate uridyltransferase (contains nucleotidyltransferase and I-patch acetyltransferase domains) [Cell envelope biogenesis, outer membrane].','COG1207','GlmU'),(1182909,'gnl|CDD|100055',_binary 'Ferripyochelin Binding Protein (FBP): FBP is an outer membrane protein which plays a role in iron acquisition. It binds iron when it is complexed with pyochelin. It adopts the left-handed parallel beta-helix (LbH) structure, and contains imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. Acyltransferase activity has not been observed in this group.','cd04650','LbH_FBP'),(1182910,'gnl|CDD|223973',_binary 'Acyl-[acyl carrier protein].','COG1043','LpxA'),(1182911,'gnl|CDD|131340',_binary 'phenylacetic acid degradation protein PaaY.  Members of this family are located next to other genes organized into apparent operons for phenylacetic acid degradation. PaaY is located near the end of these gene clusters and often next to PaaX, a transcriptional regulator. [Energy metabolism, Other].','TIGR02287','Phenylacetic_acid_degrada'),(1182912,'gnl|CDD|188173',_binary 'acyl-[acyl-carrier-protein]--UDP-N-acetylglucosamine O-acyltransferase.  This model describes LpxA, an enzyme for the biosynthesis of lipid A, a component oflipopolysaccharide (LPS) in the outer membrane outer leaflet of most Gram-negative bacteria. Some differences are found between lipid A of different species, but this protein represents the first step (from UDP-N-acetyl-D-glucosamine) and appears to be conserved in function. Proteins from this family contain many copies of the bacterial transferase hexapeptide repeat (pfam00132). [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR01852','Acyl---UDP-N-acetylglucos'),(1182913,'gnl|CDD|184642',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK14353','glmU'),(1182914,'gnl|CDD|274908',_binary 'UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase.  The MJ_1101 protein from Methanococcus jannaschii has been characterized as the GlmU enzyme catalyzing the final two steps of UDP-GlcNAc biosynthesis. Many of the genes identified by this model are in proximity to the GlmS and GlmM genes and are also presumed to be GlmU. However, some archaeal genomes contain multiple closely-related homologs from this family and it is not clear what the substrate specificity is for each of them.','TIGR03992','Arch_glmU'),(1182915,'gnl|CDD|237687',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK14357','glmU'),(1182916,'gnl|CDD|130038',_binary '2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase.  This enzyme is part of the diaminopimelate pathway of lysine biosynthesis. Alternate name: tetrahydrodipicolinate N-succinyltransferase. The closely related TabB protein of Pseudomonas syringae (pv. tabaci), SP|P31852|TABB_PSESZ, appears to act in the biosynthesis of tabtoxin rather than lysine. The trusted cutoff is set high enough to exclude this gene. Sequences below trusted also include a version of this enzyme which apparently utilize acetate rather than succinate (EC: 2.3.1.89). [Amino acid biosynthesis, Aspartate family].','TIGR00965','dapD'),(1183349,'gnl|CDD|225564',_binary 'Predicted metal-binding protein [General function prediction only].','COG3019','COG3019'),(1183427,'gnl|CDD|237918',_binary 'putative aldose-1-epimerase; Provisional.','PRK15172','PRK15172'),(1184141,'gnl|CDD|314902',_binary 'Protein of unknown function (DUF3574).  This family of proteins is functionally uncharacterized. This protein is found in bacteria and viruses. Proteins in this family are typically between 144 to 163 amino acids in length. This protein has a conserved TPRF sequence motif.','pfam12098','DUF3574'),(1185316,'gnl|CDD|306629',_binary 'Aminotransferase class I and II.','pfam00155','Aminotran_1_2'),(1185869,'gnl|CDD|318794',_binary 'Purple acid Phosphatase, N-terminal domain.  This domain is found at the N-terminus of Purple acid phosphatase proteins.','pfam16656','Pur_ac_phosph_N'),(1187633,'gnl|CDD|213244',_binary 'ATP-binding cassette domain of eukaryotic SMC5 proteins.  The structural maintenance of chromosomes (SMC) proteins are large (approximately 110 to 170 kDa), and each is arranged into five recognizable domains. Amino-acid sequence homology of SMC proteins between species is largely confined to the amino- and carboxy-terminal globular domains. The amino-terminal domain contains a \'Walker A\' nucleotide-binding domain (GxxGxGKS/T, in the single-letter amino-acid code), which by mutational studies has been shown to be essential in several proteins. The carboxy-terminal domain contains a sequence (the DA-box) that resembles a \'Walker B\' motif, and a motif with homology to the signature sequence of the ATP-binding cassette (ABC) family of ATPases. The sequence homology within the carboxy-terminal domain is relatively high within the SMC1-SMC4 group, whereas SMC5 and SMC6 show some divergence in both of these sequences. In eukaryotic cells, the proteins are found as heterodimers of SMC1 paired with SMC3, SMC2 with SMC4, and SMC5 with SMC6 (formerly known as Rad18).','cd03277','ABC_SMC5_euk'),(1187689,'gnl|CDD|238948',_binary 'This is a subfamily of Adenine nucleotide alpha hydrolases superfamily. Adenine nucleotide alpha hydrolases superfamily  includes N type ATP PPases and ATP sulphurylases. It forms a apha/beta/apha fold which  binds to Adenosine group.  This subfamily   of proteins probably binds ATP. This domain is about 200 amino acids long with a strongly conserved motif SGGKD at the N terminus.','cd01990','Alpha_ANH_like_I'),(1187691,'gnl|CDD|224522',_binary 'ATP-utilizing enzymes of the PP-loop superfamily [General function prediction only].','COG1606','COG1606'),(1187710,'gnl|CDD|274967',_binary 'putative peptide-modifying radical SAM enzyme, AF0577 family.  This radical SAM family contains a C-terminal region motif CXXCX5CX3C that is found in PqqE and other radical SAM enzymes that act on peptide substrates. Members of this family are found primarily in the Archaea, but also several eukaryotes (Trichomonas vaginalis G3, Entamoeba dispar SAW760, Giardia intestinalis ATCC 50581, etc.). The function is unknown.','TIGR04084','rSAM_AF0577'),(1187717,'gnl|CDD|274595',_binary 'hopanoid biosynthesis associated radical SAM protein HpnH.  The sequences represented by this model are members of the radical SAM superfamily of enzymes (pfam04055). These enzymes utilize an iron-sulfur redox cluster and S-adenosylmethionine to carry out diverse radical mediated reactions. The members of this clade are frequently found in the same locus as squalene-hopene cyclase (SHC, TIGR01507) and other genes associated with the biosynthesis of hopanoid natural products. The linkage between SHC and this radical SAM enzyme is strong; one is nearly always observed in the same genome where the other is found. A hopanoid biosynthesis locus was described in Zymomonas mobilis consisting of the genes HpnA-E and SHC (HpnF). Continuing past SHC are found a phosphorylase enzyme (ZMO0873, i.e. HpnG, TIGR03468) and this radical SAM enzyme (ZMO0874) which we name here HpnH. Granted, in Z. mobilis, HpnH is in a convergent orientation with respect to HpnA-G, but one gene beyond HpnH and running in the same convergent direction is IspH (ZM0875, 4-hydroxy-3-methylbut-2-enyl diphosphate reductase), an essential enzyme of IPP biosynthesis and therefore essential for the biosynthesis of hopanoids. One of the well-described hopanoid intermediates is bacteriohopanetetrol. In the conversion from hopene several reactions must occur in the side chain for which a radical mechanism might be reasonable. These include the four (presumably anaerobic) hydroxylations and a methyl shift.','TIGR03470','HpnH'),(1187725,'gnl|CDD|238351',_binary 'Tunnelling fold (T-fold). The five known T-folds are found in five different enzymes with different functions: dihydroneopterin-triphosphate epimerase (DHNTPE), dihydroneopterin aldolase (DHNA) , GTP cyclohydrolase I (GTPCH-1),  6-pyrovoyl tetrahydropterin synthetase (PTPS), and uricase (UO,uroate/urate oxidase). They bind to substrates belonging to the purine or pterin families, and share a fold-related binding site with a glutamate or glutamine residue anchoring the substrate and a lot of conserved interactions. They also share a similar oligomerization mode: several T-folds join together to form a beta(2n)alpha(n) barrel, then two barrels join together in a head-to-head fashion to made up the native enzymes. The functional enzyme is a tetramer for UO, a hexamer for PTPS, an octamer for DHNA/DHNTPE and a decamer for GTPCH-1. The substrate is located in a deep and narrow pocket at the interface between monomers. In PTPS, the active site is located at the interface of three monomers, two from one trimer and one from the other trimer. In GTPCH-1, it is also located at the interface of three subunits, two from one pentamer and one from the other pentamer. There are four equivalent active sites in UO, six in PTPS, eight in DHNA/DHNTPE and ten in GTPCH-1.   Each globular multimeric enzyme encloses a tunnel which is lined with charged residues for DHNA and UO, and with basic residues in PTPS. The N and C-terminal ends are located on one side of the T-fold while the residues involved in the catalytic activity are located at the opposite side. In PTPS, UO and DHNA/DHNTPE, the N and C-terminal extremities of the enzyme are located on the exterior side of the functional multimeric enzyme. In GTPCH-1, the extra C-terminal helix places the extremity inside the tunnel.','cd00651','TFold'),(1187744,'gnl|CDD|130214',_binary 'ATP synthase, F0 subunit b.  This model describes the F1/F0 ATP synthase b subunit in bacteria only. Scoring just below the trusted cutoff are the N-terminal domains of Mycobacterial b/delta fusion proteins and a subunit from an archaeon, Methanosarcina barkeri, in which the ATP synthase homolog differs in architecture and is not experimentally confirmed. This model helps resolve b from the related b\' subunit. Within the family is an example from a sodium-translocating rather than proton-translocating ATP synthase. [Energy metabolism, ATP-proton motive force interconversion].','TIGR01144','ATP_synthase_subunit_b'),(1187746,'gnl|CDD|180240',_binary 'F0F1 ATP synthase subunit B; Validated.','PRK05759','PRK05759'),(1187857,'gnl|CDD|313750',_binary 'AIPR protein.  This family of proteins was identified in as an abortive infection phage resistance protein often found in restriction modification system operons.','pfam10592','AIPR'),(1188205,'gnl|CDD|308214',_binary 'DNA recombination-mediator protein A.  The SMF family, of DNA processing chain A, dprA, are a group of bacterial proteins. In H. pylori, dprA is required for natural chromosomal and plasmid transformation. It has now been shown that DprA is found to bind cooperatively to single-stranded DNA (ssDNA) and to interact with RecA. In the process, DprA-RecA-ssDNA filaments are produced and these filaments catalyze the homology-dependent formation of joint molecules. While the E.coli SSB protein limits access of RecA to ssDNA, DprA alleviates this barrier. It is proposed that DprA is a new member of the recombination-mediator protein family, dedicated to natural bacterial transformation.','pfam02481','DNA_processg_A'),(1188207,'gnl|CDD|273238',_binary 'DNA protecting protein DprA.  Disruption of this gene in both Haemophilus influenzae and Helicobacter pylori drastically reduces the efficiency of transformation with exogenous DNA, but with different levels of effect on chromosomal (linear) and plasmid (circular) DNA. This difference suggests the DprA is not active in recombination, and it has been shown not to affect DNA binding, leaving the intermediate step in natural transformation, DNA processing. In Strep. pneumoniae, inactivation of dprA had no effect on the uptake of DNA. All of these data indicated that DprA is required at a later stage in transformation. Subsequently DprA and RecA were both shown in S. pneumoniae to be required to protect incoming ssDNA from immediate degradation. Role of DprA in non-transformable species is not known. The gene symbol smf was assigned in E. coli, but without assignment of function. [Cellular processes, DNA transformation].','TIGR00732','Protein_smf'),(1188208,'gnl|CDD|236747',_binary 'hypothetical protein; Provisional.','PRK10736','PRK10736'),(1192111,'gnl|CDD|225237',_binary 'D-aminopeptidase [Amino acid transport and metabolism].','COG2362','DppA'),(1193029,'gnl|CDD|181619',_binary 'hypothetical protein; Validated.','PRK09039','PRK09039'),(1194808,'gnl|CDD|235485',_binary 'CDP-diacylglycerol pyrophosphatase; Provisional.','PRK05471','PRK05471'),(1194884,'gnl|CDD|308633',_binary 'Bacterial RNA polymerase, alpha chain C terminal domain.  The alpha subunit of RNA polymerase consists of two independently folded domains, referred to as amino-terminal and carboxyl terminal domains. The amino terminal domain is involved in the interaction with the other subunits of the RNA polymerase. The carboxyl-terminal domain interacts with the DNA and activators. The amino acid sequence of the alpha subunit is conserved in prokaryotic and chloroplast RNA polymerases. There are three regions of particularly strong conservation, two in the amino-terminal and one in the carboxyl- terminal.','pfam03118','RNA_pol_A_CTD'),(1194885,'gnl|CDD|273934',_binary 'DNA-directed RNA polymerase, alpha subunit, bacterial and chloroplast-type.  This family consists of the bacterial (and chloroplast) DNA-directed RNA polymerase alpha subunit, encoded by the rpoA gene. The RNA polymerase catalyzes the transcription of DNA into RNA using the four ribonucleoside triphosphates as substrates. The amino terminal domain is involved in dimerizing and assembling the other RNA polymerase subunits into a transcriptionally active enzyme. The carboxy-terminal domain contains determinants for interaction with DNA and with transcriptional activator proteins. [Transcription, DNA-dependent RNA polymerase].','TIGR02027','DNA-directed_RNA_polymera'),(1197126,'gnl|CDD|315954',_binary 'HNH endonuclease.','pfam13391','HNH_2'),(1197160,'gnl|CDD|234794',_binary 'DNA-directed RNA polymerase subunit beta\'; Provisional.','PRK00566','PRK00566'),(1197751,'gnl|CDD|314030',_binary 'Putative auto-transporter adhesin, head GIN domain.  This bacterial family of proteins shows structural similarity to other pectin lyase families. Although structures from this family align with acetyl-transferases, there is no conservation of catalytic residues found. It is likely that the function is one of cell-adhesion. In Structure 3jx8, it is interesting to note that the sequence of contains several well defined sequence repeats, centred around GSG motifs defining the tight beta turn between the two sheets of the super-helix; there are 8 such repeats in the C-terminal half of the protein, which could be grouped into 4 repeats of two. It seems likely that this family belongs to the superfamily of trimeric auto-transporter adhesins (TAAs), which are important virulence factors in Gram-negative pathogens. In the case of Parabacteroides distasonis, which is a component of the normal distal human gut microbiota, TAA-like complexes probably modulate adherence to the host (information derived from TOPSAN).','pfam10988','DUF2807'),(1203489,'gnl|CDD|239246',_binary 'TRX domain, TRX and NDP-kinase (NDPK) fusion protein family; most members of this group are fusion proteins which contain one redox active TRX domain containing a CXXC motif and three NDPK domains, and are characterized as intermediate chains (ICs) of axonemal outer arm dynein. Dyneins are molecular motors that generate force against microtubules to produce cellular movement, and are divided into two classes: axonemal and cytoplasmic. They are supramolecular complexes consisting of three protein groups classified according to size: dynein heavy, intermediate and light chains. Axonemal dyneins form two structures, the inner and outer arms, which are attached to doublet microtubules throughout the cilia and flagella. The human homolog is the sperm-specific Sptrx-2, presumed to be a  component of the human sperm axoneme architecture. Included in this group is another human protein, TRX-like protein 2, a smaller fusion protein containing one TRX and one NDPK domain, which is also associated with microtubular structures. The other members of this group are hypothetical insect proteins containing a TRX domain and outer arm dynein light chains (14 and 16kDa) of Chlamydomonas reinhardtii. Using standard assays, the fusion proteins have shown no TRX enzymatic activity.','cd02948','TRX_NDPK'),(1203519,'gnl|CDD|273997',_binary 'glycogen synthase, Corynebacterium family.  This model describes Corynebacterium glutamicum GlgA and closely related proteins in several other species. This enzyme is required for glycogen biosynthesis and appears to replace the distantly related TIGR02095 family of ADP-glucose type glycogen synthase in Corynebacterium glutamicum, Mycobacterium tuberculosis, Bifidobacterium longum, and Streptomyces coelicolor. [Energy metabolism, Biosynthesis and degradation of polysaccharides].','TIGR02149','glgA_Coryne'),(1204538,'gnl|CDD|312439',_binary 'WbqC-like protein family.  This family of proteins are functionally uncharacterized. However it is found in an O-antigen gene cluster in E. coli and other bacteria suggesting a role in O-antigen production. Feng et al. suggest that wbnG may code for a glycine transferase.','pfam08889','WbqC'),(1205324,'gnl|CDD|239257',_binary 'Endoplasmic reticulum protein 19 (ERp19) family; ERp19 is also known as ERp18, a protein located in the ER containing one redox active TRX domain. Denaturation studies indicate that the reduced form is more stable than the oxidized form, suggesting that the protein is involved in disulfide bond formation. In vitro, ERp19 has been shown to possess thiol-disulfide oxidase activity which is dependent on the presence of both active site cysteines. Although described as protein disulfide isomerase (PDI)-like, the protein does not complement for PDI activity. ERp19 shows a wide tissue distribution but is most abundant in liver, testis, heart and kidney.','cd02959','ERp19'),(1205371,'gnl|CDD|100002',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility. The members of this family are found mainly in bacteria, while some of them are also found in Archaea and eukaryotes.','cd04962','GT1_like_5'),(1205400,'gnl|CDD|273909',_binary '2-polyprenylphenol 6-hydroxylase.  This model represents the enzyme (UbiB) which catalyzes the first hydroxylation step in the ubiquinone biosynthetic pathway in bacteria. It is believed that the reaction is 2-polyprenylphenol -> 6-hydroxy-2-polyprenylphenol. This model finds hits primarily in the proteobacteria. The gene is also known as AarF in certain species. [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR01982','Probable_protein_kinase_U'),(1205425,'gnl|CDD|225724',_binary 'Predicted restriction endonuclease [Defense mechanisms].','COG3183','COG3183'),(1206076,'gnl|CDD|275320',_binary 'two transmembrane protein.  Members of this family are uncharacterized proteins from the genus Mycoplasma, typically about 260 amino acids long, with a hydrophobic predicted transmembrane alpha helix toward each end. Often two family members are encoded in tandem, e.g. MG_279 and MG_280 from Mycoplasma genitalium.','TIGR04527','mycoplas_twoTM'),(1206167,'gnl|CDD|181627',_binary '3-carboxy-cis,cis-muconate cycloisomerase; Provisional.','PRK09053','PRK09053'),(1206168,'gnl|CDD|176470',_binary 'PurB_like adenylosuccinases (adenylsuccinate lyase, ASL).  This subgroup contains EcASL, the product of the purB gene in Escherichia coli, and related proteins. It is a member of the Lyase class I family of the Lyase_I superfamily. Members of the Lyase class I family function as homotetramers to catalyze similar beta-elimination reactions in which a Calpha-N or Calpha-O bond is cleaved with the subsequent release of fumarate as one of the products. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two non-sequential steps in the de novo purine biosynthesis pathway: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and; the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP).','cd01598','PurB'),(1206169,'gnl|CDD|168324',_binary '3-carboxy-cis,cis-muconate cycloisomerase; Provisional.','PRK05975','PRK05975'),(1206170,'gnl|CDD|236452',_binary 'adenylosuccinate lyase; Provisional.','PRK09285','PRK09285'),(1206172,'gnl|CDD|176463',_binary 'Argininosuccinate lyase (argininosuccinase, ASAL).  This group contains ASAL and related proteins. It is a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASAL is a cytosolic enzyme which catalyzes the reversible breakdown of argininosuccinate to arginine and fumarate during arginine biosynthesis. In ureotelic species ASAL also catalyzes a reaction involved in the production of urea. Included in this group are the major soluble avian eye lens proteins from duck, delta 1 and delta 2 crystallin. Of these two isoforms only delta 2 has retained ASAL activity. These crystallins may have evolved by, gene recruitment of ASAL followed by gene duplication. In humans, mutations in ASAL result in the autosomal recessive disorder argininosuccinic aciduria.','cd01359','Argininosuccinate_lyase'),(1206173,'gnl|CDD|223243',_binary 'Argininosuccinate lyase [Amino acid transport and metabolism].','COG0165','ArgH'),(1206174,'gnl|CDD|235010',_binary 'argininosuccinate lyase; Provisional.','PRK02186','PRK02186'),(1206175,'gnl|CDD|223192',_binary 'Fumarase [Energy production and conversion].','COG0114','FumC'),(1206176,'gnl|CDD|171490',_binary 'fumarate hydratase; Provisional.','PRK12425','PRK12425'),(1206177,'gnl|CDD|176468',_binary 'aspartase (L-aspartate ammonia-lyase) and fumarase class II enzymes.  This group contains aspartase (L-aspartate ammonia-lyase), fumarase class II enzymes, and related proteins. It is a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. Aspartase catalyzes the reversible deamination of aspartic acid. Fumarase catalyzes the reversible hydration/dehydration of fumarate to L-malate during the Krebs cycle.','cd01596','Aspartase_like'),(1206178,'gnl|CDD|223958',_binary 'Aspartate ammonia-lyase [Amino acid transport and metabolism].','COG1027','AspA'),(1206179,'gnl|CDD|213564',_binary 'aspartate ammonia-lyase.  This enzyme, aspartate ammonia-lyase, shows local homology to a number of other lyases, as modeled by pfam00206. Fumarate hydratase scores as high as 570 bits against this model. [Energy metabolism, Amino acids and amines].','TIGR00839','Aspartate_ammonia-lyase'),(1206180,'gnl|CDD|237743',_binary 'aspartate ammonia-lyase; Provisional.','PRK14515','PRK14515'),(1206181,'gnl|CDD|315860',_binary 'WYL domain.  This presumed domain is around 170 amino acids in length. It is found to the C-terminus of a DNA-binding helix-turn-helix domain. This domain may be involved in binding to an as yet unknown ligand that allows a transcriptional regulation response to that molecule. There are a number of proteins that contain two tandem copies of this domain. This suggests that this domain may form a dimeric arrangement.','pfam13280','WYL'),(1206715,'gnl|CDD|226728',_binary 'Uncharacterized conserved protein [Function unknown].','COG4278','COG4278'),(1206728,'gnl|CDD|226181',_binary 'Predicted transcriptional regulator [Transcription].','COG3655','COG3655'),(1206756,'gnl|CDD|226595',_binary 'Uncharacterized protein involved in stress response [General function prediction only].','COG4110','COG4110'),(1206760,'gnl|CDD|309458',_binary 'Cysteine protease Prp.  This is a family of cysteine protease that are found to cleave the N-terminus extension of ribosomal subunit L27 in eubacteria. Proteins in this family are distinguished by a pair of invariant histidine and cysteine residues with conserved spacing that form the classic catalytic dyad of a cysteine protease.','pfam04327','Peptidase_Prp'),(1206761,'gnl|CDD|319990',_binary 'conserved uncharacterized protein similar to Bacillus subtilis YsxB.  family of conserved uncharacterized proteins similar to Bacillus subtilis YsxB.','cd16332','YsxB-like'),(1206762,'gnl|CDD|184740',_binary 'hypothetical protein; Provisional.','PRK14553','PRK14553'),(1206763,'gnl|CDD|225423',_binary 'Predicted ribosomal protein [Translation, ribosomal structure and biogenesis].','COG2868','COG2868'),(1206764,'gnl|CDD|224759',_binary 'Transcriptional regulators [Transcription].','COG1846','MarR'),(1206765,'gnl|CDD|224609',_binary 'Predicted transcriptional regulators [Transcription].','COG1695','COG1695'),(1206766,'gnl|CDD|275076',_binary 'chitooligosaccharide synthase NodC.  Members of this family are NodC, an N-acetylglucosaminyltransferase involved in the production of nodulation factors through which rhizobia establish symbioses with leguminous plants.','TIGR04242','glycosyl_transferase_fami'),(1213670,'gnl|CDD|236512',_binary 'Dna-J like membrane chaperone protein; Provisional.','PRK09430','djlA'),(1215662,'gnl|CDD|238742',_binary 'VWA subgroup: Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses  In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains. Not much is known about the function of the VWA domain in these proteins. The members do have a conserved MIDAS motif. The biochemical function however is not known.','cd01465','vWA_subgroup'),(1215664,'gnl|CDD|318994',_binary 'Ribosomal proteins L26 eukaryotic, L24P archaeal.  Ribosomal_L26 is a family of the 50S and the 60S ribosomal proteins from eukaryotes - L26 - and archaea - L25.','pfam16906','Ribosomal_L26'),(1215752,'gnl|CDD|239291',_binary 'PDIa family, 5\'-Adenylylsulfate (APS) reductase subfamily; composed of plant-type APS reductases containing a C-terminal redox active TRX domain and an N-terminal reductase domain which is part of a superfamily that includes N type ATP PPases. APS reductase catalyzes the reduction of activated sulfate to sulfite, a key step in the biosynthesis of sulfur-containing metabolites. Sulfate is first activated by ATP sulfurylase, forming APS, which can be phosphorylated to 3\'-phosphoadenosine-5\'-phosphosulfate (PAPS). Depending on the organism, either APS or PAPS can be used for sulfate reduction. Prokaryotes and fungi use PAPS, whereas plants use both APS and PAPS. Since plant-type APS reductase uses glutathione (GSH) as its electron donor, the C-terminal domain may function like glutaredoxin, a GSH-dependent member of the TRX superfamily. The flow of reducing equivalents goes from GSH -> C-terminal TRX domain -> N-terminal reductase domain -> APS. Plant-type APS reductase shows no homology to that of dissimilatory sulfate-reducing bacteria, which is an iron-sulfur flavoenzyme. Also included in the alignment is EYE2 from Chlamydomonas reinhardtii, a protein required for eyespot assembly.','cd02993','PDI_a_APS_reductase'),(1215758,'gnl|CDD|215175',_binary '5\'-adenylylsulfate reductase.','PLN02309','PLN02309'),(1215767,'gnl|CDD|273072',_binary '5\'-adenylylsulfate reductase, thioredoxin-independent.  This enzyme, involved in the assimilation of inorganic sulfate, is closely related to the thioredoxin-dependent PAPS reductase of Bacteria (CysH) and Saccharomyces cerevisiae. However, it has its own C-terminal thioredoxin-like domain and is not thioredoxin-dependent. Also, it has a substrate preference for 5\'-adenylylsulfate (APS) over 3\'-phosphoadenylylsulfate (PAPS) so the pathway does not require an APS kinase (CysC) to convert APS to PAPS. Arabidopsis thaliana appears to have three isozymes, all able to complement E. coli CysH mutants (even in backgrounds lacking thioredoxin or APS kinase) but likely localized to different compartments in Arabidopsis. [Central intermediary metabolism, Sulfur metabolism].','TIGR00424','APS_reductase'),(1216117,'gnl|CDD|315806',_binary 'Protein of unknown function (DUF4031).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 91 and 130 amino acids in length. There is a conserved HYD sequence motif.','pfam13223','DUF4031'),(1217172,'gnl|CDD|184956',_binary 'SAM-dependent 16S ribosomal RNA C1402 ribose 2\'-O-methyltransferase; Provisional.','PRK14994','PRK14994'),(1217239,'gnl|CDD|314107',_binary 'Endoribonuclease GhoS.  GhoS is part of the GhoT-GhoS type V toxin-antitoxin (TA) system. GhoT is inhibited by antitoxin GhoS, which specifically cleaves its mRNA.','pfam11080','GhoS'),(1217492,'gnl|CDD|223506',_binary 'Predicted hydrolase of the alpha/beta-hydrolase fold [General function prediction only].','COG0429','COG0429'),(1217493,'gnl|CDD|307987',_binary 'X-Pro dipeptidyl-peptidase (S15 family).','pfam02129','Peptidase_S15'),(1217515,'gnl|CDD|133407',_binary 'Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily.  Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 5), N-terminal domain. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04780','HTH_MerR-like_sg5'),(1217518,'gnl|CDD|131102',_binary 'Cd(II)/Pb(II)-responsive transcriptional regulator.  This model represents the cadmium(II) and/or lead(II) responsive transcriptional activator of the proteobacterial metal efflux system. This protein is a member of the MerR family of transcriptional activators (pfam00376) and contains a distinctive pattern of cysteine residues in its metal binding loop, Cys-X(6-9)-Cys, as well as a conserved and critical cysteine at the N-terminal end of the dimerization helix. [Regulatory functions, DNA interactions].','TIGR02047','putative_regulatory_prote'),(1218117,'gnl|CDD|212140',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in uncharacterized protein from Archaeoglobus fulgidus (Af0060) and its bacterial homologs.  This family includes an uncharacterized protein from Archaeoglobus fulgidus (Af0060) and its homologs from bacteria. Although its biological role remains unclear, Af0060 shows high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain that contains a well conserved divalent ion-binding motif EXX[E/D].','cd11533','NTP-PPase_Af0060_like'),(1218118,'gnl|CDD|212145',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from bacteria.  This family corresponds to a group of uncharacterized hypothetical proteins from bacteria, showing a high sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, members in this family consist of a single MazG-like domain that contains a well conserved divalent ion-binding motif EXX[E/D].','cd11538','NTP-PPase_u1'),(1218384,'gnl|CDD|236605',_binary 'RNA polymerase sigma factor SigL; Provisional.','PRK09645','PRK09645'),(1218424,'gnl|CDD|274646',_binary 'luciferase family oxidoreductase, group 1.  The Pfam domain family pfam00296 is named for luciferase-like monooxygenases, but the family also contains several coenzyme F420-dependent enzymes. This protein family represents a well-resolved clade within family pfam00296 and shows no restriction to coenzyme F420-positive species, unlike some other clades within pfam00296. [Unknown function, Enzymes of unknown specificity].','TIGR03558','oxido_grp_1'),(1218886,'gnl|CDD|239946',_binary 'Members of the SGNH-hydrolase superfamily, a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad from other serine hydrolases, but may lack the carboxlic acid.','cd04502','SGNH_hydrolase_like_7'),(1218889,'gnl|CDD|238873',_binary 'SGNH_hydrolase subfamily. SGNH hydrolases are a diverse family of lipases and esterases. The tertiary fold of the enzyme is substantially different from that of the alpha/beta hydrolase family and unique among all known hydrolases; its active site closely resembles the Ser-His-Asp(Glu) triad found in other serine hydrolases.','cd01835','SGNH_hydrolase_like_3'),(1218891,'gnl|CDD|182521',_binary 'multifunctional acyl-CoA thioesterase I and protease I and lysophospholipase L1; Provisional.','PRK10528','PRK10528'),(1219042,'gnl|CDD|235850',_binary 'flagellar assembly protein H; Validated.','PRK06669','fliH'),(1219079,'gnl|CDD|315920',_binary 'Beta-lactamase enzyme family.  This family is closely related to Beta-lactamase, pfam00144, the serine beta-lactamase-like superfamily, which contains the distantly related pfam00905 and PF00768 D-alanyl-D-alanine carboxypeptidase.','pfam13354','Beta-lactamase2'),(1219080,'gnl|CDD|225242',_binary 'Beta-lactamase class A [Defense mechanisms].','COG2367','PenP'),(1219084,'gnl|CDD|308120',_binary 'Outer membrane efflux protein.  The OEP family (Outer membrane efflux protein) form trimeric channels that allow export of a variety of substrates in Gram negative bacteria. Each member of this family is composed of two repeats. The trimeric channel is composed of a 12 stranded all beta sheet barrel that spans the outer membrane, and a long all helical barrel that spans the periplasm.','pfam02321','OEP'),(1220139,'gnl|CDD|181867',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK09451','glmU'),(1220140,'gnl|CDD|273482',_binary 'UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase.  This protein is a bifunctional enzyme, GlmU, which catalyzes last two reactions in the four-step pathway of UDP-N-acetylglucosamine biosynthesis from fructose-6-phosphate. Its reaction product is required from peptidoglycan biosynthesis, LPS biosynthesis in species with LPS, and certain other processes. [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan, Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides, Central intermediary metabolism, Amino sugars].','TIGR01173','Bifunctional_glmU_protein'),(1220142,'gnl|CDD|215167',_binary 'carbonate dehydratase.','PLN02296','PLN02296'),(1220149,'gnl|CDD|184643',_binary 'bifunctional N-acetylglucosamine-1-phosphate uridyltransferase/glucosamine-1-phosphate acetyltransferase; Provisional.','PRK14354','glmU'),(1220157,'gnl|CDD|100039',_binary 'Gamma carbonic anhydrases (CA): Carbonic anhydrases are zinc-containing enzymes that catalyze the reversible hydration of carbon dioxide in a two-step mechanism, involving the nucleophilic attack of a zinc-bound hydroxide ion on carbon dioxide, followed by the regeneration of the active site by ionization of the zinc-bound water molecule and removal of a proton from the active site. They are ubiquitous enzymes involved in fundamental processes like photosynthesis, respiration, pH homeostasis and ion transport. There are three distinct groups of  carbonic anhydrases - alpha, beta and gamma - which show no significant sequence identity or structural similarity. Gamma CAs are homotrimeric enzymes, with each subunit containing a left-handed parallel beta helix (LbH) structural domain.','cd00710','LbH_gamma_CA'),(1220158,'gnl|CDD|100052',_binary 'Dynactin 6 (or subunit p27): Dynactin is a major component of the activator complex that stimulates dynein-mediated vesicle transport. Dynactin is a heterocomplex of at least eight subunits, including a 150,000-MW protein called Glued, the actin-capping protein Arp1, and dynamatin. In vitro binding experiments show that dynactin enhances dynein-dependent motility, possibly through interaction with microtubules and vesicles. Subunit p27 is part of the pointed-end subcomplex in dynactin that also includes p25, p26, and Arp11. This subcomplex interacts with membranous cargoes. p25 and p27 contain the imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), indicating a left-handed parallel beta helix (LbH) structural domain. Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.','cd04646','LbH_Dynactin_6'),(1221072,'gnl|CDD|307559',_binary 'Uncharacterized protein family (UPF0051).','pfam01458','UPF0051'),(1221073,'gnl|CDD|273908',_binary 'FeS assembly protein SufD.  This protein, SufD, forms a cytosolic complex SufBCD. This complex enhances the cysteine desulfurase of SufSE. The system, together with SufA, is believed to act in iron-sulfur cluster formation during oxidative stress. SufB and SufD are homologous. Note that SufC belongs to the family of ABC transporter ATP binding proteins, so this protein, encoded by an adjacent gene, has often been annotated as a transporter component. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other].','TIGR01981','UPF0051_protein_Rv1462/MT'),(1221074,'gnl|CDD|223791',_binary 'Cysteine desulfurase activator SufB [Posttranslational modification, protein turnover, chaperones].','COG0719','SufB'),(1221085,'gnl|CDD|308834',_binary 'MgtE intracellular N domain.  This domain is found at the N-terminus of eubacterial magnesium transporters of the MgtE family pfam01769. This domain is an intracellular domain that has an alpha-helical structure. The crystal structure of the MgtE transporter shows two of 5 magnesium ions are in the interface between the N domain and the CBS domains. In the absence of magnesium there is a large shift between the N and CBS domains.','pfam03448','MgtE_N'),(1221171,'gnl|CDD|251894',_binary 'Phage Terminase.  The majority of the members of this family are bacteriophage proteins, several of which are thought to be terminase large subunit proteins. There are also a number of bacterial proteins of unknown function.','pfam03354','Terminase_1'),(1221172,'gnl|CDD|253031',_binary 'Phage portal protein, SPP1 Gp6-like.  This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. During SPP1 morphogenesis, Gp6 participates in the procapsid assembly reaction. This family also includes the old Pfam family Phage_min_cap (PF05126).','pfam05133','Phage_prot_Gp6'),(1221174,'gnl|CDD|253082',_binary 'Hom_end-associated Hint.  Homing endonucleases are encoded by mobile DNA elements that are found inserted within host genes in all domains of life. The crystal structure of the homing nuclease PI-Sce revealed two domains: an endonucleolytic centre resembling the C-terminal domain of Drosophila melanogaster Hedgehog protein, and a a second domain containing the protein-splicing active site. This Domain corresponds to the latter protein-splicing domain.','pfam05203','Hom_end_hint'),(1221179,'gnl|CDD|258668',_binary 'LAGLIDADG-like domain.  This domain is part of the LAGLIDADG superfamily.','pfam14528','LAGLIDADG_3'),(1221180,'gnl|CDD|252984',_binary 'Phage capsid family.  Family of bacteriophage hypothetical proteins and capsid proteins.','pfam05065','Phage_capsid'),(1221182,'gnl|CDD|233467',_binary 'phage major capsid protein, HK97 family.  This model family represents the major capsid protein component of the heads (capsids) of bacteriophage HK97, phi-105, P27, and related phage. This model represents one of several analogous families lacking detectable sequence similarity. The gene encoding this component is typically located in an operon encoding the small and large terminase subunits, the portal protein and the prohead or maturation protease [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01554','major_cap_HK97'),(1221184,'gnl|CDD|150131',_binary 'Phage protein Gp19/Gp15/Gp42.  This family of proteins are functionally uncharacterized. They are found in a variety of bacteriophage.','pfam09355','Phage_Gp19'),(1221187,'gnl|CDD|151357',_binary 'Protein of unknown function (DUF2744).  This is a viral family of proteins with unknown function.','pfam10910','DUF2744'),(1221189,'gnl|CDD|249626',_binary 'Beta-lactamase.  This family appears to be distantly related to pfam00905 and PF00768 D-alanyl-D-alanine carboxypeptidase.','pfam00144','Beta-lactamase'),(1221203,'gnl|CDD|249707',_binary 'Resolvase, N terminal domain.  The N-terminal domain of the resolvase family (this family) contains the active site and the dimer interface. The extended arm at the C-terminus of this domain connects to the C-terminal helix-turn-helix domain of resolvase - see pfam02796.','pfam00239','Resolvase'),(1221207,'gnl|CDD|254248',_binary 'Recombinase.  This domain is usually found associated with pfam00239 in putative integrases/recombinases of mobile genetic elements of diverse bacteria and phages.','pfam07508','Recombinase'),(1221208,'gnl|CDD|257733',_binary 'Recombinase zinc beta ribbon domain.  This short bacterial protein contains a zinc ribbon domain that is likely to be DNA-binding. This domain is found in site specific recombinase proteins. This family appears most closely related to pfam04606.','pfam13408','Zn_ribbon_recom'),(1221209,'gnl|CDD|249888',_binary 'DNA polymerase family A.','pfam00476','DNA_pol_A'),(1221215,'gnl|CDD|233038',_binary 'DNA polymerase I.  All proteins in this family for which functions are known are DNA polymerases Many also have an exonuclease motif. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00593','pola'),(1221237,'gnl|CDD|250744',_binary '3\'-5\' exonuclease.  This domain is responsible for the 3\'-5\' exonuclease proofreading activity of Escherichia coli DNA polymerase I (polI) and other enzymes, it catalyses the hydrolysis of unpaired or mismatched nucleotides. This domain consists of the amino-terminal half of the Klenow fragment in Escherichia coli polI it is also found in the Werner syndrome helicase (WRN), focus forming activity 1 protein (FFA-1) and ribonuclease D (RNase D). Werner syndrome is a human genetic disorder causing premature aging; the WRN protein has helicase activity in the 3\'-5\' direction. The FFA-1 protein is required for formation of a replication foci and also has helicase activity; it is a homologue of the WRN protein. RNase D is a 3\'-5\' exonuclease involved in tRNA processing. Also found in this family is the autoantigen PM/Scl thought to be involved in polymyositis-scleroderma overlap syndrome.','pfam01612','DNA_pol_A_exo1'),(1221246,'gnl|CDD|233392',_binary 'DNA primase, catalytic core.  Members of this family are DNA primase, a ubiquitous bacteria protein. Most members of this family contain nearly two hundred additional residues C-terminal to the region represented here, but conservation between species is poor and the C-terminal region was not included in the seed alignment. This protein contains a CHC2 zinc finger (pfam01807) and a Toprim domain (pfam01751) [DNA metabolism, DNA replication, recombination, and repair].','TIGR01391','dnaG'),(1221247,'gnl|CDD|257526',_binary 'Toprim-like.  This is a family or Toprim-like proteins.','pfam13155','Toprim_2'),(1221249,'gnl|CDD|257969',_binary 'Toprim domain.  The toprim domain is found in a wide variety of enzymes involved in nucleic acid manipulation.','pfam13662','Toprim_4'),(1221253,'gnl|CDD|110780',_binary 'CHC2 zinc finger.  This domain is principally involved in DNA binding in DNA primases.','pfam01807','zf-CHC2'),(1221260,'gnl|CDD|251634',_binary 'Recombination endonuclease VII.','pfam02945','Endonuclease_7'),(1221268,'gnl|CDD|249878',_binary 'Glutaredoxin.','pfam00462','Glutaredoxin'),(1221269,'gnl|CDD|131251',_binary 'Glutaredoxin-like protein, YruB-family.  This glutaredoxin-like protein family contains the conserved CxxC motif and includes the Clostridium pasteurianum protein YruB which has been cloned from a rubredoxin operon. Somewhat related to NrdH, it is unknown whether this protein actually interacts with glutathione/glutathione reducatase, or, like NrdH, some other reductant system.','TIGR02196','GlrX_YruB'),(1221270,'gnl|CDD|233766',_binary 'Glutaredoxin, GrxC family.  Glutaredoxins are thioltransferases (disulfide reductases) which utilize glutathione and NADPH as cofactors. Oxidized glutathione is regenerated by glutathione reductase. Together these components compose the glutathione system. Glutaredoxins utilize the CXXC motif common to thioredoxins and are involved in multiple cellular processes including protection from redox stress, reduction of critical enzymes such as ribonucleotide reductase and the generation of reduced sulfur for iron sulfur cluster formation. Glutaredoxins are capable of reduction of mixed disulfides of glutathione as well as the formation of glutathione mixed disulfides. This family of glutaredoxins includes the E. coli protein GrxC (Grx3) which appears to have a secondary role in reducing ribonucleotide reductase (in the absence of GrxA) possibly indicating a role in the reduction of other protein disulfides [Energy metabolism, Electron transport].','TIGR02181','GRX_bact'),(1221284,'gnl|CDD|257805',_binary 'AAA domain.  This AAA domain is found in a wide variety of presumed DNA repair proteins.','pfam13481','AAA_25'),(1221287,'gnl|CDD|258543',_binary 'Superinfection immunity protein.  This family includes the Escherichia coli bacteriophage T4 superinfection immunity (imm) protein. When Escherichia coli is sequentially infected with two T-even type bacteriophage the DNA of the superinfecting phage is excluded from the host, into the periplasmic space. The immunity protein plays a role in this process.','pfam14373','Imm_superinfect'),(1221288,'gnl|CDD|257237',_binary 'PD-(D/E)XK nuclease superfamily.  Members of this family belong to the PD-(D/E)XK nuclease superfamily.','pfam12705','PDDEXK_1'),(1221293,'gnl|CDD|257966',_binary 'Methyltransferase domain.  This family contains methyltransferase domains.','pfam13659','Methyltransf_26'),(1221294,'gnl|CDD|233085',_binary 'DNA-methyltransferase (dcm).  All proteins in this family for which functions are known are DNA-cytosine methyltransferases. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00675','dcm'),(1221295,'gnl|CDD|258120',_binary 'Methyltransferase domain.  This family appears to be have methyltransferase activity.','pfam13847','Methyltransf_31'),(1221296,'gnl|CDD|249627',_binary 'C-5 cytosine-specific DNA methylase.','pfam00145','DNA_methylase'),(1221297,'gnl|CDD|255371',_binary 'RNA cap guanine-N2 methyltransferase.  RNA cap guanine-N2 methyltransferases such as Schizosaccharomyces pombe Tgs1 and Giardia lamblia Tgs2 catalyse methylation of the exocyclic N2 amine of 7-methylguanosine.','pfam09445','Methyltransf_15'),(1221301,'gnl|CDD|257858',_binary 'D-alanyl-D-alanine carboxypeptidase.  This family resembles VanY, pfam02557, which is part of the peptidase M15 family.','pfam13539','Peptidase_M15_4'),(1221305,'gnl|CDD|257717',_binary 'HNH endonuclease.  This is a zinc-binding loop of Fold group 7 as found in endo-deoxy-ribonucleases and HNH nucleases.','pfam13392','HNH_3'),(1221307,'gnl|CDD|251820',_binary 'Terminase-like family.  This family represents a group of terminase proteins.','pfam03237','Terminase_6'),(1221309,'gnl|CDD|257877',_binary 'Helix-turn-helix domain.  This domain is a helix-turn-helix domain that probably binds to DNA.','pfam13560','HTH_31'),(1221310,'gnl|CDD|257343',_binary 'Helix-turn-helix domain.  Members of this family contains a DNA-binding helix-turn-helix domain.','pfam12844','HTH_19'),(1221314,'gnl|CDD|250573',_binary 'Helix-turn-helix.  This large family of DNA binding helix-turn helix proteins includes Cro and CI. Within the Neisseria gonorrhoeae phage associated protein NGO0477, the full protein fold incorporates a helix-turn-helix motif, but the function of this member is unlikely to be that of a DNA-binding regulator, the function of most other members, so is not necessarily characteristic of the whole family.','pfam01381','HTH_3'),(1221315,'gnl|CDD|250701',_binary 'Peptidase family M23.  Members of this family are zinc metallopeptidases with a range of specificities. The peptidase family M23 is included in this family, these are Gly-Gly endopeptidases. Peptidase family M23 are also endopeptidases. This family also includes some bacterial lipoproteins such as Escherichia coli murein hydrolase activator NlpD, for which no proteolytic activity has been demonstrated. This family also includes leukocyte cell-derived chemotaxin 2 (LECT2) proteins. LECT2 is a liver-specific protein which is thought to be linked to hepatocyte growth although the exact function of this protein is unknown.','pfam01551','Peptidase_M23'),(1221319,'gnl|CDD|250643',_binary 'Putative peptidoglycan binding domain.  This domain is composed of three alpha helices. This domain is found at the N or C terminus of a variety of enzymes involved in bacterial cell wall degradation. This domain may have a general peptidoglycan binding function. This family is found N-terminal to the catalytic domain of matrixins. The domain is found to bind peptidoglycan experimentally.','pfam01471','PG_binding_1'),(1221325,'gnl|CDD|250673',_binary 'N-acetylmuramoyl-L-alanine amidase.  This family includes zinc amidases that have N-acetylmuramoyl-L-alanine amidase activity EC:3.5.1.28. This enzyme domain cleaves the amide bond between N-acetylmuramoyl and L-amino acids in bacterial cell walls (preferentially: D-lactyl-L-Ala). The structure is known for the bacteriophage T7 structure and shows that two of the conserved histidines are zinc binding.','pfam01510','Amidase_2'),(1221330,'gnl|CDD|249733',_binary 'Helicase conserved C-terminal domain.  The Prosite family is restricted to DEAD/H helicases, whereas this domain family is found in a wide variety of helicases and helicase related proteins. It may be that this is not an autonomously folding unit, but an integral part of the helicase.','pfam00271','Helicase_C'),(1221337,'gnl|CDD|249732',_binary 'DEAD/DEAH box helicase.  Members of this family include the DEAD and DEAH box helicases. Helicases are involved in unwinding nucleic acids. The DEAD box helicases are involved in various aspects of RNA metabolizm, including nuclear transcription, pre mRNA splicing, ribosome biogenesis, nucleocytoplasmic transport, translation, RNA decay and organellar gene expression.','pfam00270','DEAD'),(1221339,'gnl|CDD|252834',_binary 'Type III restriction enzyme, res subunit.','pfam04851','ResIII'),(1221341,'gnl|CDD|251073',_binary 'Crossover junction endodeoxyribonuclease RuvC.','pfam02075','RuvC'),(1221344,'gnl|CDD|255250',_binary 'Bifunctional DNA primase/polymerase, N-terminal.  Members of this family adopt a structure consisting of a core of antiparallel beta sheets. They are found in various bacterial hypothetical proteins, and have been shown to harbour both primase and polymerase activities.','pfam09250','Prim-Pol'),(1221353,'gnl|CDD|233793',_binary 'DNA repair and recombination protein RadB.  This family consists exclusively of archaeal RadB protein, a homolog of bacterial RecA (TIGR02012), eukaryotic RAD51 (TIGR02239) and DMC1 (TIGR02238), and archaeal RadA (TIGR02236).','TIGR02237','recomb_radB'),(1221364,'gnl|CDD|257349',_binary 'Oxygenase domain of the 2OGFeDO superfamily.  A double-stranded beta helix (DSBH) fold domain of the 2-oxoglutarate (2OG)-Fe(II)-dependent dioxygenase (2OGFeDO) superfamily found in various eukaryotes, bacteria and bacteriophages. Members of this family catalyze nucleic acid modifications, such as thymidine hydroxylation during base J synthesis in kinetoplastids, and the conversion of 5 methyl-cytosine (5-mC) to 5-hydroxymethyl-cytosine (hmC), or further oxidation to 5-formylcytosine (5fC) and 5-carboxylcytosine (5caC). Metazoan TET proteins contain a cysteine-rich region inserted into the core of the DSBH fold. Vertebrate TET proteins are oncogenes that are mutated in various myeloid cancers. Fungal and algal versions of this family are linked to a predicted transposase and show lineage-specific expansions.','pfam12851','Tet_JBP'),(1221366,'gnl|CDD|251394',_binary 'GlcNAc-PI de-N-acetylase.  Members of this family are related to PIG-L an N-acetylglucosaminylphosphatidylinositol de-N-acetylase (EC:3.5.1.89) that catalyses the second step in GPI biosynthesis.','pfam02585','PIG-L'),(1221367,'gnl|CDD|234439',_binary 'bacillithiol biosynthesis deacetylase BshB1.  Members of this protein family are BshB1 (YpjG), an enzyme of bacillithiol biosynthesis; either BshB1 or BshB2 (YojG) must be present, and often both are present. Bacillithiol is a low-molecular-weight thiol, an analog of glutathione and mycothiol, and is found largely in the Firmicutes [Biosynthesis of cofactors, prosthetic groups, and carriers, Glutathione and analogs].','TIGR04001','thiol_BshB1'),(1221370,'gnl|CDD|257848',_binary 'Peptidase_C39 like family.','pfam13529','Peptidase_C39_2'),(1221371,'gnl|CDD|253713',_binary 'Zeta toxin.  This family consists of several bacterial zeta toxin proteins. Zeta toxin is thought to be part of a postregulational killing system in bacteria. It relies on antitoxin/toxin systems that secure stable inheritance of low and medium copy number plasmids during cell division and kill cells that have lost the plasmid.','pfam06414','Zeta_toxin'),(1221389,'gnl|CDD|234468',_binary 'DNA phosphorothioation system restriction enzyme.  The DNA phosphorothioate modification system dnd (DNA instability during electrophoresis) recently has been shown to provide a modification essential to a restriction system. This protein family was detected by Partial Phylogenetic Profiling as linked to dnd, and its members usually are clustered with the dndABCDE genes.','TIGR04095','dnd_restrict_1'),(1221420,'gnl|CDD|257710',_binary 'Homeodomain-like domain.','pfam13384','HTH_23'),(1221424,'gnl|CDD|253543',_binary 'Domain of unknown function (DUF932).  Family of prokaryotic proteins with unknown function. Contains a number of highly conserved polar residues that could suggest an enzymatic activity.','pfam06067','DUF932'),(1221425,'gnl|CDD|234162',_binary 'phage/plasmid-like protein TIGR03299.  Members of this uncharacterized protein family are found in various Mycobacterium phage genomes, in Streptomyces coelicolor plasmid SCP1, and in bacterial genomes near various markers that suggest lateral gene transfer. The function is unknown [Mobile and extrachromosomal element functions, Other].','TIGR03299','LGT_TIGR03299'),(1221429,'gnl|CDD|256862',_binary 'Protein of unknown function (DUF3584).  This protein is found in bacteria and eukaryotes. Proteins in this family are typically between 943 to 1234 amino acids in length. This family contains a P-loop motif suggesting it is a nucleotide binding protein. It may be involved in replication.','pfam12128','DUF3584'),(1221433,'gnl|CDD|253144',_binary 'Protein of unknown function (DUF732).  This family consists of several uncharacterized Mycobacterium tuberculosis and leprae proteins of unknown function.','pfam05305','DUF732'),(1221434,'gnl|CDD|238594',_binary 'phiLC3 phage and phage-related integrases, site-specific recombinases, DNA breaking-rejoining enzymes, C-terminal catalytic domain. This CD includes various bacterial (mainly gram positive) and phage integrases, including those similar to Lactococcus phage phiLC3, TPW22, Tuc2009, BK5-T, A2,  bIL285, bIL286, bIL311, ul36 and phi g1e; Staphylococcus aureus phage phi13 and phi42; Oenococcus oeni phage fOg44; Streptococcus thermophilus phage O1205 and Sfi21; and Streptococcus pyogenes phage T12 and T270.','cd01189','INT_phiLC3_C'),(1221435,'gnl|CDD|238587',_binary 'DNA breaking-rejoining enzymes, intergrase/recombinases, C-terminal catalytic domain. The tyrosine recombinase/integrase family share the same catalytic domain containing six conserved active site residues. The best-studied members of this diverse family include the bacteriophage lambda integrase, the bacteriophage P1 Cre recombinase, the yeast Flp recombinase and the bacterial XerD/C recombinases. Their overall reaction mechanism is essentially identical and involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. Many intergrase/recombinases also have N-terminal domains, which show little sequence or structure similarity.','cd01182','INT_REC_C'),(1221436,'gnl|CDD|249984',_binary 'Phage integrase family.  Members of this family cleave DNA substrates by a series of staggered cuts, during which the protein becomes covalently linked to the DNA through a catalytic tyrosine residue at the carboxy end of the alignment. The catalytic site residues in CRE recombinase are Arg-173, His-289, Arg-292 and Tyr-324.','pfam00589','Phage_integrase'),(1221437,'gnl|CDD|238597',_binary 'P22-like integrases, site-specific recombinases, DNA breaking-rejoining enzymes, C-terminal catalytic domain. This CD includes various bacterial and phage integrases, including those similar to phage P22-like integrases, DLP12 and APSE-1.','cd01192','INT_P22_C'),(1221438,'gnl|CDD|238604',_binary 'Tn1545-related conjugative transposon integrases, site-specific recombinases, DNA breaking-rejoining enzymes, C-terminal catalytic domain. This CD includes bacterial (gram positive) and phage integrases, including those similar to Tn1545, Tn5252, and Tn5276 conjugative transposon integrases and Lactobacillus phage phi adh integrase.','cd01199','INT_Tn1545_C'),(1221439,'gnl|CDD|238413',_binary 'XerD and XerC integrases, DNA breaking-rejoining enzymes, N- and C-terminal domains. XerD-like integrases are involved in the site-specific integration and excision of lysogenic bacteriophage genomes, transposition of conjugative transposons, termination of chromosomal replication, and stable plasmid inheritance. They share the same fold in their catalytic domain containing six conserved active site residues and the overall reaction mechanism with the DNA breaking-rejoining enzyme superfamily.  In Escherichia coli, the Xer site-specific recombination system acts to convert dimeric chromosomes, which are formed by homologous recombination to monomers. Two related recombinases, XerC and XerD, bind cooperatively to a recombination site present in the E. coli chromosome. Each recombinase catalyzes the exchange of one pair of DNA strand in a reaction that proceeds through a Holliday junction intermediate. These enzymes can bridge two different and well-separated DNA sequences called arm- and core-sites. The C-terminal domain binds, cleaves and re-ligates DNA strands at the core-sites, while the N-terminal domain is largely responsible for high-affinity binding to the arm-type sites.','cd00798','INT_XerDC'),(1221441,'gnl|CDD|233788',_binary 'tyrosine recombinase XerC.  The phage integrase family describes a number of recombinases with tyrosine active sites that transiently bind covalently to DNA. Many are associated with mobile DNA elements, including phage, transposons, and phase variation loci. This model represents XerC, one of two closely related chromosomal proteins along with XerD (TIGR02225). XerC and XerD are site-specific recombinases which help resolve chromosome dimers to monomers for cell division after DNA replication. In species with a large chromosome and homologs of XerC on other replicons, the chomosomal copy was preferred for building this model. This model does not detect all XerC, as some apparent XerC examples score in the gray zone between trusted (450) and noise (410) cutoffs, along with some XerD examples. XerC and XerD interact with cell division protein FtsK [DNA metabolism, DNA replication, recombination, and repair].','TIGR02224','recomb_XerC'),(1221442,'gnl|CDD|238596',_binary 'phiCTX phage and phage-related integrases, site-specific recombinases, DNA breaking-rejoining enzymes, C-terminal catalytic domain. This CD includes various phage and bacterial integrases, including those similar to phage integrases: Bordetella and Pseudomonas phiCTX;  E. coli  Rac, Qin, and Shiga toxin 2 933W; and Salmonella typhimurium LT2 Gifsy-2 and Fels-1; and a putative pore-forming cytotoxin integrase from Vibrio parahaemolyticus O3:K6.','cd01191','INT_phiCTX_C'),(1221443,'gnl|CDD|238231',_binary 'DNA breaking-rejoining enzymes, C-terminal catalytic domain. The DNA breaking-rejoining enzyme superfamily includes type IB topoisomerases and tyrosine recombinases that share the same fold in their catalytic domain containing six conserved active site residues. The best-studied members of this diverse superfamily include human topoisomerase I, the bacteriophage lambda integrase, the bacteriophage P1 Cre recombinase, the yeast Flp recombinase and the bacterial XerD/C recombinases. Their overall reaction mechanism is essentially identical and involves cleavage of a single strand of a DNA duplex by nucleophilic attack of a conserved tyrosine to give a 3\' phosphotyrosyl protein-DNA adduct. In the second rejoining step, a terminal 5\' hydroxyl attacks the covalent adduct to release the enzyme and generate duplex DNA. The enzymes differ in that topoisomerases cleave and then rejoin the same 5\' and 3\' termini, whereas a site-specific recombinase transfers a 5\' hydroxyl generated by recombinase cleavage to a new 3\' phosphate partner located in a different duplex region.  Many DNA breaking-rejoining enzymes also have N-terminal domains, which show little sequence or structure similarity.','cd00397','DNA_BRE_C'),(1221446,'gnl|CDD|233789',_binary 'tyrosine recombinase XerD.  The phage integrase family describes a number of recombinases with tyrosine active sites that transiently bind covalently to DNA. Many are associated with mobile DNA elements, including phage, transposons, and phase variation loci. This model represents XerD, one of two closely related chromosomal proteins along with XerC (TIGR02224). XerC and XerD are site-specific recombinases which help resolve chromosome dimers to monomers for cell division after DNA replication. In species with a large chromosome and with homologs of XerD on other replicons, the chomosomal copy was preferred for building this model. This model does not detect all XerD, as some apparent XerD examples score below the trusted and noise cutoff scores. XerC and XerD interact with cell division protein FtsK [DNA metabolism, DNA replication, recombination, and repair].','TIGR02225','recomb_XerD'),(1221447,'gnl|CDD|238416',_binary 'Bacteriophage P4 integrase. P4-like integrases are found in temperate bacteriophages, integrative plasmids, pathogenicity and symbiosis islands, and other mobile genetic elements.  They share the same fold in their catalytic domain and the overall reaction mechanism with the superfamily of DNA breaking-rejoining enzymes. The P4 integrase mediates integrative and excisive site-specific recombination between two sites, called attachment sites, located on the phage genome and the bacterial chromosome. The phage attachment site is often found adjacent to the integrase gene, while the host attachment sites are typically situated near tRNA genes.','cd00801','INT_P4'),(1221449,'gnl|CDD|238599',_binary 'Tn544A and related transposases, DNA breaking-rejoining enzymes, integrase/recombinases, C-terminal catalytic domain. This CD includes various bacterial transposases similar to TnpA from transposon Tn554.','cd01194','INT_Tn554A_C'),(1221451,'gnl|CDD|238412',_binary 'Phage HP1 integrase, C-terminal catalytic domain.  Bacteriophage HP1 and related integrases are found in eubacteria, plasmids and temperate bacteriophages of the P2 family.  They belong to the DNA breaking-rejoining enzyme superfamily, which includes tyrosine recombinases and type IB topoisomerases. These enzymes share the same fold in their C-terminal catalytic domain containing six conserved active site residues and the overall reaction mechanism. The HP1 recombinase controls phage replication by site-specific recombination between the HP1 genome and the chromosomal DNA. It is a heterobifunctional DNA-binding protein, which recognizes two different DNA sequence motifs (type I and type II binding sites). The C-terminal catalytic domain of the HP1 integrase binds to the type I site, while the less conserved N-terminal domain is largely responsible for binding to the type II site.','cd00797','HP1_INT_C'),(1221454,'gnl|CDD|111373',_binary 'Transcription factor WhiB.  WhiB is a putative transcription factor in Actinobacteria, required for differentiation and sporulation.','pfam02467','Whib'),(1221456,'gnl|CDD|238598',_binary 'IntI (E2) integrases, site-specific tyrosine recombinases, DNA breaking-rejoining enzymes, N- and C-terminal domains. This CD includes integrases which are components of multiresistant integrons and mediate recombination between a proximal attI site and a secondary target called the attC (or 59-base element) present on various mobile gene cassettes. Integron-integrases are present in many natural occurring mobile elements, including transposons and conjugative plasmids. Vibrio, Shewanella, Xanthomonas and Pseudomonas species harbor chromosomal super-integrons. All integron-integrases carry large inserts unlike the TnpF ermF-like proteins also seen in this group.','cd01193','INT_IntI'),(1221457,'gnl|CDD|238589',_binary 'INT_SG2, DNA breaking-rejoining enzymes, integrase/recombinases subgroup 2, C-terminal catalytic domain. The CD contains mainly predicted integrase/recombinases and phage-related integrases. Some have N-terminal domains, which show little sequence similarity to each other. Members of this subgroup are predominantly found in proteobacteria.','cd01184','INT_SG2_C'),(1221458,'gnl|CDD|238602',_binary 'FimB and FimE and related proteins, DNA breaking-rejoining enzymes, integrase/recombinases, catalytic domain. This CD includes those proteins similar to E.coli FimE and FimB regulatory proteins and Proteus mirabilis MrpI.','cd01197','INT_FimBE_C'),(1221459,'gnl|CDD|238595',_binary 'INT_SG5, DNA breaking-rejoining enzymes, integrase/recombinases subgroup 5, N- and C-terminal domains. The CD contains mainly predicted bacterial integrase/recombinases.','cd01190','INT_SG5'),(1221460,'gnl|CDD|238588',_binary 'INT_SG1, DNA breaking-rejoining enzymes, integrase/recombinases subgroup 1, C-terminal catalytic domain. The CD contains mainly predicted integrase/recombinase and site-specific XerD recombinases. The members of this CD are found predominantly in proteobacteria. These proteins have not been biochemically characerised as yet.','cd01183','INT_SG1_C'),(1221461,'gnl|CDD|258797',_binary 'Phage integrase, N-terminal SAM-like domain.  This domain is found in a variety of phage integrase proteins.','pfam14659','Phage_int_SAM_3'),(1221462,'gnl|CDD|238593',_binary 'pAE1 and related integrases, DNA breaking-rejoining enzymes, integrase/recombinases, C-terminal domain. This CD includes various bacterial integrases, including the predicted integrase of the deletion-prone region of plasmid pAE1 of Alcaligenes eutrophus H1.','cd01188','INT_pAE1'),(1221463,'gnl|CDD|238592',_binary 'INT_SG4, DNA breaking-rejoining enzymes, integrase/recombinases subgroup 4, N- and C-terminal domains. The CD contains mainly predicted bacterial integrase/recombinases for which not much biochemical characterization is available.','cd01187','INT_SG4'),(1221464,'gnl|CDD|238600',_binary 'Tn544B and related transposases, DNA breaking-rejoining enzymes, integrase/recombinases, catalytic domain. This CD includes various bacterial transposases similar to TnpB from transposon Tn554.','cd01195','INT_Tn544B_C'),(1221465,'gnl|CDD|204733',_binary 'Protein of unknwon function (DUF3310).  This is a family of conserved bacteriophage proteins of unknown function.','pfam11753','DUF3310'),(1221468,'gnl|CDD|250234',_binary 'Exonuclease.  This family includes a variety of exonuclease proteins, such as ribonuclease T and the epsilon subunit of DNA polymerase III.;.','pfam00929','RNase_T'),(1221471,'gnl|CDD|233398',_binary 'DnaQ family exonuclease/DinG family helicase, putative.  This model represents a family of proteins in Gram-positive bacteria. The N-terminal region of about 200 amino acids resembles the epsilon subunit of E. coli DNA polymerase III and the homologous region of the Gram-positive type DNA polymerase III alpha subunit. The epsilon subunit contains an exonuclease domain. The remainder of this protein family resembles a predicted ATP-dependent helicase, the DNA damage-inducible protein DinG of E. coli [DNA metabolism, DNA replication, recombination, and repair].','TIGR01407','dinG_rel'),(1221475,'gnl|CDD|253429',_binary 'Endodeoxyribonuclease RusA.  This family consists of several bacterial and phage Holliday junction resolvase (RusA) like proteins. The RusA protein of Escherichia coli is an endonuclease that can resolve Holliday intermediates and correct the defects in genetic recombination and DNA repair associated with inactivation of RuvAB or RuvC.','pfam05866','RusA'),(1221479,'gnl|CDD|255861',_binary 'SprT-like family.  This family represents a domain found in eukaryotes and prokaryotes. The domain contains a characteristic motif of the zinc metallopeptidases. This family includes the bacterial SprT protein.','pfam10263','SprT-like'),(1221480,'gnl|CDD|130674',_binary 'phage/plasmid primase, P4 family, C-terminal domain.  This model represents a clade within a larger family of proteins from viruses of bacteria and animals. Members of this family are found in phage and plasmids of bacteria and archaea only. The model describes a domain of about 300 residues, found generally toward the protein C-terminus [DNA metabolism, DNA replication, recombination, and repair, Mobile and extrachromosomal element functions, Prophage functions].','TIGR01613','primase_Cterm'),(1221484,'gnl|CDD|254987',_binary 'D5 N terminal like.  This domain is found in D5 proteins of DNA viruses and bacteriophage P4 DNA primases phages.','pfam08706','D5_N'),(1221487,'gnl|CDD|250393',_binary 'tRNA-splicing ligase RtcB.  This family of RNA ligases (EC:6.5.1.3) join 2\',3\'-cyclic phosphate and 5\'-OH ends. They catalyse the splicing of tRNA and may also participate in tRNA repair and recovery from stress-induced RNA damage.','pfam01139','RtcB'),(1221490,'gnl|CDD|213769',_binary 'release factor H-coupled RctB family protein.  Members of this family are related to RctB. RctB a protein of known structure but unknown function that often is encoded near RNA cyclase and therefore is suggested to be a tRNA or mRNA processing enzyme. This family of RctB-like proteins in encoded upstream of, and apparently is translationally coupled to, the putative peptide chain release factor RF-H (TIGR03072), product of the prfH gene. Note that a large deletion at the junction between this gene and the prfH gene in Escherichia coli K-12 marks both as probable pseudogenes [Protein synthesis, Other].','TIGR03073','release_rtcB'),(1221492,'gnl|CDD|257720',_binary 'HNH endonuclease.  This HNH nuclease domain is found in CRISPR-related proteins.','pfam13395','HNH_4'),(1221493,'gnl|CDD|250911',_binary 'HNH endonuclease.','pfam01844','HNH'),(1221497,'gnl|CDD|233605',_binary 'CRISPR-associated protein Cas9/Csn1, subtype II/NMEMI.  CRISPR loci appear to be mobile elements with a wide host range. This model represents a protein found only in CRISPR-containing species, near other CRISPR-associated proteins (cas), as part of the NMENI subtype of CRISPR/Cas locus. The species range so far for this protein is animal pathogens and commensals only [Mobile and extrachromosomal element functions, Other].','TIGR01865','cas_Csn1'),(1221503,'gnl|CDD|258607',_binary 'Prokaryotic homologs of the JAB domain.  These are metalloenzymes that function as the ubiquitin isopeptidase/ deubiquitinase in the ubiquitin-based signaling and protein turnover pathways in eukaryotes. Prokaryotic JAB domains are predicted to have a similar role in their cognates of the ubiquitin modification pathway. The domain is widely found in bacteria, archaea and phages where they are present in several gene contexts in addition to those that correspond to the prokaryotic cognates of the eukaryotic Ub pathway. Other contexts in which JAB domains are present include gene neighbor associations with ubiquitin fold domains in cysteine and siderophore biosynthesis, and phage tail morphogenesis, where they are shown or predicted to process the associated ubiquitin. A distinct family, the RadC-like JAB domains are widespread in bacteria and are predicted to function as nucleases. In halophilic archaea the JAB domain shows strong gene-neighborhood associations with a nucleotidyltransferase suggesting a role in nucleotide metabolizm.','pfam14464','Prok-JAB'),(1221506,'gnl|CDD|250588',_binary 'JAB1/Mov34/MPN/PAD-1 ubiquitin protease.  Members of this family are found in proteasome regulatory subunits, eukaryotic initiation factor 3 (eIF3) subunits and regulators of transcription factors. This family is also known as the MPN domain and PAD-1-like domain, JABP1 domain or JAMM domain. These are metalloenzymes that function as the ubiquitin isopeptidase/ deubiquitinase in the ubiquitin-based signaling and protein turnover pathways in eukaryotes. Versions of the domain in prokaryotic cognates of the ubiquitin-modification pathway are predicted to have a similar role.','pfam01398','JAB'),(1221508,'gnl|CDD|253885',_binary 'Transglycosylase-like domain.  This family of proteins are very likely to act as transglycosylase enzymes related to pfam00062 and pfam01464. These other families are weakly matched by this family, and include the known active site residues.','pfam06737','Transglycosylas'),(1221518,'gnl|CDD|254192',_binary 'Domain of unknown function (DUF1508).  This family represents a series of bacterial domains of unknown function of around 50 residues in length. Members of this family are often found as tandem repeats and in some cases represent the whole protein. All member proteins are described as being hypothetical.','pfam07411','DUF1508'),(1221520,'gnl|CDD|256224',_binary 'Protein of unknown function (DUF2786).  This family of proteins has no known function.','pfam10979','DUF2786'),(1221523,'gnl|CDD|249654',_binary 'SNF2 family N-terminal domain.  This domain is found in proteins involved in a variety of processes including transcription regulation (e.g., SNF2, STH1, brahma, MOT1), DNA repair (e.g., ERCC6, RAD16, RAD5), DNA recombination (e.g., RAD54), and chromatin unwinding (e.g., ISWI) as well as a variety of other proteins with little functional information (e.g., lodestar, ETL1).','pfam00176','SNF2_N'),(1221527,'gnl|CDD|257803',_binary 'AAA domain.  This AAA domain is found in a wide variety of presumed phage proteins.','pfam13479','AAA_24'),(1221531,'gnl|CDD|254685',_binary 'Methyltransferase domain.  Members of this family are SAM dependent methyltransferases.','pfam08241','Methyltransf_11'),(1221533,'gnl|CDD|253984',_binary '5\' nucleotidase, deoxy (Pyrimidine), cytosolic type C protein (NT5C).  This family consists of several 5\' nucleotidase, deoxy (Pyrimidine), cytosolic type C (NT5C) proteins. 5\'(3\')-Deoxyribonucleotidase is a ubiquitous enzyme in mammalian cells whose physiological function is not known.','pfam06941','NT5C'),(1221546,'gnl|CDD|213929',_binary '2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase.  Members of this family are 2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase (EC 1.3.1.28), the third enzyme in the biosynthesis of 2,3-dihydroxybenzoic acid (DHB) from chorismate. The first two enzymes are isochorismate synthase (EC 5.4.4.2) and isochorismatase (EC 3.3.2.1). Synthesis is often followed by adenylation by the enzyme DHBA-AMP ligase (EC 2.7.7.58) to activate (DHB) for a non-ribosomal peptide synthetase.','TIGR04316','dhbA_paeA'),(1221553,'gnl|CDD|251342',_binary 'Thymidylate synthase complementing protein.  Thymidylate synthase complementing protein (Thy1) complements the thymidine growth requirement of the organisms in which it is found, but shows no homology to thymidylate synthase.','pfam02511','Thy1'),(1221559,'gnl|CDD|233759',_binary 'thymidylate synthase, flavin-dependent.  Two forms of microbial thymidylate synthase are known: ThyA (2.1.1.45) and ThyX (2.1.1.148). This model describes ThyX, a homotetrameric flavoprotein. Both enzymes convert dUMP to dTMP. Under oxygen-limiting conditions, thyX can complement a thyA mutation [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02170','thyX'),(1221562,'gnl|CDD|257878',_binary 'Enoyl-(Acyl carrier protein) reductase.','pfam13561','adh_short_C2'),(1221564,'gnl|CDD|233590',_binary '3-oxoacyl-(acyl-carrier-protein) reductase.  This model represents 3-oxoacyl-[ACP] reductase, also called 3-ketoacyl-acyl carrier protein reductase, an enzyme of fatty acid biosynthesis [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR01830','3oxo_ACP_reduc'),(1221575,'gnl|CDD|234422',_binary 'oxidoreductase, SDR family.  Members of this protein subfamily are putative oxidoreductases belonging to the larger SDR family. Members of the present subfamily may occur several to a genome and are largely restricted to genomes that contain members of families TIGR03962, TIGR03967, and TIGR03969. Many members have been annotated by homology as carveol dehydrogenases.','TIGR03971','SDR_subfam_1'),(1221580,'gnl|CDD|252000',_binary 'ADP-ribosyltransferase exoenzyme.  This is a family of bacterial and viral bi-glutamic acid ADP-ribosyltransferases; in aexT from Aeromonas salmonicida E403 is the catalytic residue and E401 contributes to the transfer of ADP-ribose to the target protein. In clostridial species it is actin that is being ADP-ribosylated; this result is lethal and dermonecrotic in infected mammals.','pfam03496','ADPrib_exo_Tox'),(1221589,'gnl|CDD|233039',_binary 'DNA-directed DNA polymerase III (polc).  All proteins in this family for which functions are known are DNA polymerases. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00594','polc'),(1221599,'gnl|CDD|254393',_binary 'Bacterial DNA polymerase III alpha subunit.','pfam07733','DNA_pol3_alpha'),(1221624,'gnl|CDD|233397',_binary 'DNA polymerase III, alpha chain, Gram-positive type.  This model describes a polypeptide chain of DNA polymerase III. Full-length homologs of this protein are restricted to the Gram-positive lineages, including the Mycoplasmas. This protein is designated alpha chain and given the gene symbol polC, but is not a full-length homolog of other polC genes. The N-terminal region of about 200 amino acids is rich in low-complexity sequence, poorly alignable, and not included n this model [DNA metabolism, DNA replication, recombination, and repair].','TIGR01405','polC_Gram_pos'),(1221628,'gnl|CDD|251545',_binary 'PHP domain.  The PHP (Polymerase and Histidinol Phosphatase) domain is a putative phosphoesterase domain.','pfam02811','PHP'),(1221648,'gnl|CDD|258717',_binary 'Helix-hairpin-helix motif.  The HHH domain is a short DNA-binding domain.','pfam14579','HHH_6'),(1221650,'gnl|CDD|257337',_binary 'Helix-hairpin-helix motif.  The HhH domain is a short DNA-binding domain.','pfam12836','HHH_3'),(1221658,'gnl|CDD|258660',_binary 'Helix-hairpin-helix domain.','pfam14520','HHH_5'),(1221679,'gnl|CDD|130890',_binary '3-oxoacyl-(acyl-carrier-protein) reductase, putative.  This model represents a small, very well conserved family of proteins closely related to the FabG family, TIGR01830, and possibly equal in function. In all completed genomes with a member of this family, a FabG in TIGR01830 is also found [Fatty acid and phospholipid metabolism, Biosynthesis].','TIGR01831','fabG_rel'),(1221686,'gnl|CDD|249592',_binary 'short chain dehydrogenase.  This family contains a wide variety of dehydrogenases.','pfam00106','adh_short'),(1221687,'gnl|CDD|188169',_binary 'acetoacetyl-CoA reductase.  This model represent acetoacetyl-CoA reductase, a member of the family short-chain-alcohol dehydrogenases. Note that, despite the precision implied by the enzyme name, the reaction of EC 1.1.1.36 is defined more generally as (R)-3-hydroxyacyl-CoA + NADP+ = 3-oxoacyl-CoA + NADPH. Members of this family may act in the biosynthesis of poly-beta-hydroxybutyrate (e.g. Rhizobium meliloti) and related poly-beta-hydroxyalkanoates. Note that the member of this family from Azospirillum brasilense, designated NodG, appears to lack acetoacetyl-CoA reductase activity and to act instead in the production of nodulation factor. This family is downgraded to subfamily for this NodG. Other proteins designated NodG, as from Rhizobium, belong to related but distinct protein families.','TIGR01829','AcAcCoA_reduct'),(1221689,'gnl|CDD|258554',_binary 'Domain of unknown function (DUF4417).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 220 and 340 amino acids in length. There is a single completely conserved residue G that may be functionally important.','pfam14386','DUF4417'),(1221708,'gnl|CDD|151323',_binary 'Protein of unknown function (DUF2746).  This family of proteins has no known function.','pfam10874','DUF2746'),(1221709,'gnl|CDD|256204',_binary 'Protein of unknown function (DUF2637).  This family of proteins has no known function.','pfam10935','DUF2637'),(1221710,'gnl|CDD|257254',_binary 'Helix-turn-helix domain.  This domain is a DNA-binding helix-turn-helix domain.','pfam12728','HTH_17'),(1221718,'gnl|CDD|238590',_binary 'Tn4399 and related integrases, DNA breaking-rejoining enzymes, integrase/recombinases, N- and C-terminal domains. This CD includes various bacterial integrases, including cLV25, a Bacteroides fragilis chromosomal transfer factor integrase similar to the Bacteroides mobilizable transposon, Tn4399, integrase.','cd01185','INT_Tn4399'),(1221720,'gnl|CDD|238411',_binary 'Rci recombinase, C-terminal catalytic domain.  Rci enzymes are found in IncI1 incompatibility group plasmids such as R64.  These recombinases belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The R64 Rci recombinase mediates site specific recombination at the highly mobile DNA segments called shufflon located in the C-terminal region of the pilV gene, which determines the recipient specificity in liquid mating. This gene encodes a thin pilus component that recognizes recipient\'s receptors required for liquid mating. The recombination occurs between any of the seven inverted repeats that separate four DNA segments of the shufflon. The segments can be inverted independently or in groups, resulting in a complex DNA rearrangement. The catalytic domain of Rci is linked to a variable N-terminal domain, whose function is unknown.','cd00796','INT_Rci'),(1221749,'gnl|CDD|233901',_binary 'ribonucleoside-triphosphate reductase, adenosylcobalamin-dependent.  This model represents a group of adenosylcobalamin(B12)-dependent ribonucleotide reductases (RNR) related to the characterized species from Lactococcus leichmannii. RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. Thus model identifies NrdJ enzymes only in cyanobacteria, lactococcus and certain bacteriophage. A separate model (TIGR02504) identifies a larger group of divergent B12-dependent RNR\'s [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02505','RTPR'),(1221751,'gnl|CDD|233900',_binary 'ribonucleoside-diphosphate reductase, adenosylcobalamin-dependent.  This model represents a group of adenosylcobalamin(B12)-dependent ribonucleotide reductases (Class II RNRs) related to the characterized species from Pyrococcus , Thermoplasma , Corynebacterium and Deinococcus. RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. This model identifies genes in a wide range of deeply branching bacteria. All are structurally related to the class I (non-heme iron dependent) RNRs. In most species this gene is known as NrdJ, while in mycobacteria it is called NrdZ [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02504','NrdJ_Z'),(1221754,'gnl|CDD|249630',_binary 'Calcineurin-like phosphoesterase.  This family includes a diverse range of phosphoesterases, including protein phosphoserine phosphatases, nucleotidases, sphingomyelin phosphodiesterases and 2\'-3\' cAMP phosphodiesterases as well as nucleases such as bacterial SbcD or yeast MRE11. The most conserved regions in this superfamily centre around the metal chelating residues.','pfam00149','Metallophos'),(1221758,'gnl|CDD|251578',_binary 'Ribonucleotide reductase, barrel domain.','pfam02867','Ribonuc_red_lgC'),(1221769,'gnl|CDD|233902',_binary 'ribonucleoside-diphosphate reductase, alpha subunit.  This model represents the alpha (large) chain of the class I ribonucleotide reductase (RNR). RNR\'s are responsible for the conversion of the ribose sugar of RNA into the deoxyribose sugar of DNA. This is the rate-limiting step of DNA biosynthesis. Class I RNR\'s generate the required radical (on tyrosine) via a \'non-heme\' iron cofactor which resides in the beta (small) subunit. The alpha subunit contains the catalytic and allosteric regulatory sites. The mechanism of this enzyme requires molecular oxygen. E. Coli contains two versions of this enzyme which are regulated independently (NrdAB and NrdEF, where NrdA and NrdE are the large chains ). Most organisms contain only one, but the application of the gene symbols NrdA and NrdE are somewhat arbitrary. This model identifies RNR\'s in diverse clades of bacteria, eukaryotes as well as numerous DNA viruses and phage [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02506','NrdE_NrdA'),(1221770,'gnl|CDD|253377',_binary 'Glutaredoxin-like domain (DUF836).  These proteins are related to the pfam00462 family.','pfam05768','DUF836'),(1221788,'gnl|CDD|250836',_binary 'Toprim domain.  This is a conserved region from DNA primase. This corresponds to the Toprim domain common to DnaG primases, topoisomerases, OLD family nucleases and RecR proteins. Both DnaG motifs IV and V are present in the alignment, the DxD (V) motif may be involved in Mg2+ binding and mutations to the conserved glutamate (IV) completely abolish DnaG type primase activity. DNA primase EC:2.7.7.6 is a nucleotidyltransferase it synthesizes the oligoribonucleotide primers required for DNA replication on the lagging strand of the replication fork; it can also prime the leading stand and has been implicated in cell division. This family also includes the atypical archaeal A subunit from type II DNA topoisomerases. Type II DNA topoisomerases catalyse the relaxation of DNA supercoiling by causing transient double strand breaks.','pfam01751','Toprim'),(1221797,'gnl|CDD|257690',_binary 'Toprim domain.  The toprim domain is found in a wide variety of enzymes involved in nucleic acid manipulation.','pfam13362','Toprim_3'),(1221801,'gnl|CDD|151183',_binary 'Protein of unknown function (DUF2493).  Members of this family are all Proteobacteria. The function is not known.','pfam10686','DUF2493'),(1221806,'gnl|CDD|253890',_binary 'KaiC.  This family represents a conserved region within bacterial and archaeal proteins, most of which are hypothetical. More than one copy is sometimes found in each protein. This family includes KaiC, which is one of the Kai proteins among which direct protein-protein association may be a critical process in the generation of circadian rhythms in cyanobacteria.','pfam06745','KaiC'),(1221810,'gnl|CDD|251333',_binary 'BRO family, N-terminal domain.  This family includes the N-terminus of baculovirus BRO and ALI motif proteins. The function of BRO proteins is unknown. It has been suggested that BRO-A and BRO-C are DNA binding proteins that influence host DNA replication and/or transcription. This Pfam domain does not include the characteristic invariant alanine, leucine, isoleucine motif of the ALI proteins.','pfam02498','Bro-N'),(1221811,'gnl|CDD|251908',_binary 'Phage antirepressor protein KilAC domain.  This domain was called the KilAC domain by Iyer and colleagues.','pfam03374','ANT'),(1221816,'gnl|CDD|232944',_binary 'CRISPR-associated protein Cas4.  This model represents a family of proteins associated with CRISPR repeats in a wide set of prokaryotic genomes. This scope of this model has been broadened since it was first built to describe an archaeal subset only. The function of the protein is undefined. Distantly related proteins, excluded from this model, include ORFs from Mycobacteriophage D29 and Sulfolobus islandicus filamentous virus and a region of the Schizosaccharomyces pombe DNA replication helicase Dna2p [Mobile and extrachromosomal element functions, Other].','TIGR00372','cas4'),(1221881,'gnl|CDD|250703',_binary 'DNA methylase.  Members of this family are DNA methylases. The family contains both N-4 cytosine-specific DNA methylases and N-6 Adenine-specific DNA methylases.','pfam01555','N6_N4_Mtase'),(1221894,'gnl|CDD|254056',_binary 'Protein of unknown function (DUF1360).  This family consists of several bacterial proteins of around 115 residues in length. Members of this family are found in Bacillus species and Streptomyces coelicolor, the function of the family is unknown.','pfam07098','DUF1360'),(1221895,'gnl|CDD|252468',_binary 'Phage Mu protein F like protein.  Members of this family are found in double-stranded DNA bacteriophages, and in some bacteria. A member of this family is required for viral head morphogenesis in bacteriophage SPP1. This family is possibly a minor head protein. This family may be related to the family TT_ORF1 (pfam02956).','pfam04233','Phage_Mu_F'),(1221899,'gnl|CDD|201832',_binary 'Phosphoadenosine phosphosulfate reductase family.  This domain is found in phosphoadenosine phosphosulfate (PAPS) reductase enzymes or PAPS sulfotransferase. PAPS reductase is part of the adenine nucleotide alpha hydrolases superfamily also including N type ATP PPases and ATP sulphurylases. The enzyme uses thioredoxin as an electron donor for the reduction of PAPS to phospho-adenosine-phosphate (PAP). It is also found in NodP nodulation protein P from Rhizobium which has ATP sulfurylase activity (sulfate adenylate transferase).','pfam01507','PAPS_reduct'),(1221904,'gnl|CDD|251149',_binary 'ParB-like nuclease domain.','pfam02195','ParBc'),(1221923,'gnl|CDD|257644',_binary 'Helicase C-terminal domain.  This domain is the second of two tandem AAA domains found in a wide variety of helicase enzymes.','pfam13307','Helicase_C_2'),(1221950,'gnl|CDD|252390',_binary 'Low affinity iron permease.','pfam04120','Iron_permease'),(1221952,'gnl|CDD|257828',_binary 'Acetyltransferase (GNAT) domain.  This domain catalyses N-acetyltransferase reactions.','pfam13508','Acetyltransf_7'),(1221953,'gnl|CDD|189752',_binary 'NlpC/P60 family.  The function of this domain is unknown. It is found in several lipoproteins.','pfam00877','NLPC_P60'),(1221979,'gnl|CDD|233853',_binary 'diaminobutyrate--2-oxoglutarate aminotransferase.  Members of this family of class III pyridoxal-phosphate-dependent aminotransferases are diaminobutyrate--2-oxoglutarate aminotransferase (EC 2.6.1.76) that catalyze the first step in ectoine biosynthesis from L-aspartate beta-semialdehyde. This family is readily separated phylogenetically from enzymes with the same substrate and product but involved in other process such as siderophore (SP|Q9Z3R2) or 1,3-diaminopropane (SP|P44951) biosynthesis. The family TIGR00709 previously included both groups but has now been revised to exclude the ectoine biosynthesis proteins of this family. Ectoine is a compatible solute particularly effective in conferring salt tolerance [Cellular processes, Adaptations to atypical conditions].','TIGR02407','ectoine_ectB'),(1221980,'gnl|CDD|233098',_binary 'transaminase, acetylornithine/succinylornithine family.  This family of proteins, for which ornithine aminotransferases form an outgroup, consists mostly of proteins designated acetylornithine aminotransferase. However, the two very closely related members from E. coli are assigned different enzymatic activities. One is acetylornithine aminotransferase (EC 2.6.1.11), ArgD, an enzyme of arginine biosynthesis, while another is succinylornithine aminotransferase, an enzyme of the arginine succinyltransferase pathway, an ammonia-generating pathway of arginine catabolism (See MEDLINE:98361920). Members of this family may also act on ornithine, like ornithine aminotransferase (EC 2.6.1.13) (see MEDLINE:90337349) and on succinyldiaminopimelate, like N-succinyldiaminopmelate-aminotransferase (EC 2.6.1.17, DapC, an enzyme of lysine biosynthesis) (see MEDLINE:99175097).','TIGR00707','argD'),(1221981,'gnl|CDD|249674',_binary 'Aminotransferase class-III.','pfam00202','Aminotran_3'),(1221990,'gnl|CDD|213836',_binary 'phage replicative helicase, DnaB family, HK022 subfamily.  Members of this family are phage (or prophage-region) homologs of the bacterial homohexameric replicative helicase DnaB. Some phage may rely on host DnaB, while others encode their own verions. This model describes the largest phage-specific clade among the close homologs of DnaB, but there are, or course, other DnaB homologs from phage that fall outside the scope of this model [Mobile and extrachromosomal element functions, Prophage functions].','TIGR03600','phage_DnaB'),(1221992,'gnl|CDD|252168',_binary 'DnaB-like helicase C terminal domain.  The hexameric helicase DnaB unwinds the DNA duplex at the Escherichia coli chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerization of the N-terminal domain has been observed and may occur during the enzymatic cycle. This C-terminal domain contains an ATP-binding site and is therefore probably the site of ATP hydrolysis.','pfam03796','DnaB_C'),(1221996,'gnl|CDD|233080',_binary 'replicative DNA helicase.  This model describes the helicase DnaB, a homohexameric protein required for DNA replication. The homohexamer can form a ring around a single strand of DNA near a replication fork. An intein of > 400 residues is found at a conserved location in DnaB of Synechocystis PCC6803, Rhodothermus marinus (both experimentally confirmed), and Mycobacterium tuberculosis. The intein removes itself by a self-splicing reaction. The seed alignment contains inteins so that the model built from the seed alignment will model a low cost at common intein insertion sites [DNA metabolism, DNA replication, recombination, and repair].','TIGR00665','DnaB'),(1222000,'gnl|CDD|254701',_binary 'DNA primase catalytic core, N-terminal domain.','pfam08275','Toprim_N'),(1222012,'gnl|CDD|249696',_binary 'DnaJ domain.  DnaJ domains (J-domains) are associated with hsp70 heat-shock system and it is thought that this domain mediates the interaction. DnaJ-domain is therefore part of a chaperone (protein folding) system. The T-antigens, although not in Prosite are confirmed as DnaJ containing domains from literature.','pfam00226','DnaJ'),(1222021,'gnl|CDD|249886',_binary 'RF-1 domain.  This domain is found in peptide chain release factors such as RF-1 and RF-2, and a number of smaller proteins of unknown function. This domain contains the peptidyl-tRNA hydrolase activity. The domain contains a highly conserved motif GGQ, where the glutamine is thought to coordinate the water that mediates the hydrolysis.','pfam00472','RF-1'),(1222029,'gnl|CDD|232785',_binary 'peptide chain release factor 2.  In many but not all taxa, there is a conserved real translational frameshift at a TGA codon. RF-2 helps terminate translation at TGA codons and can therefore regulate its own production by readthrough when RF-2 is insufficient. There is a PFAM model called \'RF-1\' for the superfamily of RF-1, RF-2, mitochondrial, RF-H, etc [Protein synthesis, Translation factors].','TIGR00020','prfB'),(1222037,'gnl|CDD|249635',_binary 'recA bacterial DNA recombination protein.  RecA is a DNA-dependent ATPase and functions in DNA repair systems. RecA protein catalyses an ATP-dependent DNA strand-exchange reaction that is the central step in the repair of dsDNA breaks by homologous recombination.','pfam00154','RecA'),(1222077,'gnl|CDD|253355',_binary 'TROVE domain.  This presumed domain is found in TEP1 and Ro60 proteins, that are RNA-binding components of Telomerase, Ro and Vault RNPs. This domain has been named TROVE, (after Telomerase, Ro and Vault). This domain is probably RNA-binding.','pfam05731','TROVE'),(1222080,'gnl|CDD|255763',_binary 'Glycosyltransferase like family 2.  Members of this family of prokaryotic proteins include putative glucosyltransferase, which are involved in bacterial capsule biosynthesis.','pfam10111','Glyco_tranf_2_2'),(1222084,'gnl|CDD|250396',_binary 'SPFH domain / Band 7 family.  This family has been called SPFH, Band 7 or PHB domain. Recent phylogenetic analysis has shown this domain to be a slipin or Stomatin-like integral membrane domain conserved from protozoa to mammals.','pfam01145','Band_7'),(1222092,'gnl|CDD|173623',_binary 'Catalytic domain of Protein Kinases.  Protein Kinases (PKs), catalytic (c) domain. PKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine or tyrosine residues on protein substrates. The PK family is part of a larger superfamily that includes the catalytic domains of RIO kinases, aminoglycoside phosphotransferase, choline kinase, phosphoinositide 3-kinase (PI3K), and actin-fragmin kinase. PKs make up a large family of serine/threonine kinases, protein tyrosine kinases (PTKs), and dual-specificity PKs that phosphorylate both serine/threonine and tyrosine residues of target proteins. Majority of protein phosphorylation, about 95%, occurs on serine residues while only 1% occurs on tyrosine residues. Protein phosphorylation is a mechanism by which a wide variety of cellular proteins, such as enzymes and membrane channels, are reversibly regulated in response to certain stimuli. PKs often function as components of signal transduction pathways in which one kinase activates a second kinase, which in turn, may act on other kinases; this sequential action transmits a signal from the cell surface to target proteins, which results in cellular responses. The PK family is one of the largest known protein families with more than 100 homologous yeast enzymes and 550 human proteins. A fraction of PK family members are pseudokinases that lack crucial residues for catalytic activity. The mutiplicity of kinases allows for specific regulation according to substrate, tissue distribution, and cellular localization. PKs regulate many cellular processes including proliferation, division, differentiation, motility, survival, metabolism, cell-cycle progression, cytoskeletal rearrangement, immunity, and neuronal functions. Many kinases are implicated in the development of various human diseases including different types of cancer.','cd00180','PKc'),(1222095,'gnl|CDD|173756',_binary 'Pseudokinase domain of STE20-related kinase adapter protein.  Protein Kinase family, STE20-related kinase adapter protein (STRAD) subfamily, pseudokinase domain. The STRAD subfamily is part of a larger superfamily that includes the catalytic domains of serine/threonine kinases (STKs), protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase. The pseudokinase domain shows similarity to protein kinases but lacks crucial residues for catalytic activity. STRAD forms a complex with the scaffolding protein MO25, and the STK, LKB1, resulting in the activation of the kinase. In the complex, LKB1 phosphorylates and activates adenosine monophosphate-activated protein kinases (AMPKs), which regulate cell energy metabolism and cell polarity. LKB1 is a tumor suppressor linked to the rare inherited disease, Peutz-Jeghers syndrome, which is characterized by a predisposition to benign polyps and hyperpigmentation of the buccal mucosa. There are two forms of STRAD, alpha and beta, that complex with LKB1 and MO25. The structure of STRAD-alpha is available and shows that this protein binds ATP, has an ordered activation loop, and adopts a closed conformation typical of fully active protein kinases. It does not possess activity due to nonconservative substitutions of essential catalytic residues. ATP binding enhances the affinity of STRAD for MO25.  The conformation of STRAD-alpha stabilized through ATP and MO25 may be needed to activate LKB1.','cd08216','PK_STRAD'),(1222096,'gnl|CDD|173672',_binary 'Catalytic domain of the Protein Serine/Threonine Kinase, Phosphoinositide-dependent kinase 1.  Serine/Threonine Kinases (STKs), Phosphoinositide-dependent kinase 1 (PDK1) subfamily, catalytic (c) domain. STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The PDK1 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase (PI3K). PDK1 carries an N-terminal catalytic domain and a C-terminal pleckstrin homology (PH) domain that binds phosphoinositides. It phosphorylates the activation loop of AGC kinases that are regulated by PI3K such as PKB, SGK, and PKC, among others, and is crucial for their activation. Thus, it contributes in regulating many processes including metabolism, growth, proliferation, and survival. PDK1 also has the ability to autophosphorylate and is constitutively active in mammalian cells. PDK1 is essential for normal embryo development and is important in regulating cell volume.','cd05581','STKc_PDK1'),(1222097,'gnl|CDD|173670',_binary 'Catalytic domain of Microtubule-associated serine/threonine kinase-like proteins.  Serine/Threonine Kinases (STKs), Microtubule-associated serine/threonine (MAST) kinase subfamily, catalytic (c) domain. STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The MAST kinase subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase. The MAST kinase subfamily includes MAST kinases, MAST-like (MASTL) kinases, and fungal kinases with similarity to Saccharomyces cerevisiae Rim15 and Schizosaccharomyces pombe cek1. MAST kinases contain an N-terminal domain of unknown function, a central catalytic domain, and a C-terminal PDZ domain that mediates protein-protein interactions. MASTL kinases carry only a catalytic domain which contains a long insert relative to other kinases. The fungal kinases in this subfamily harbor other domains in addition to a central catalytic domain, which also contains an insert relative to MAST kinases like MASTL. Rim15 contains a C-terminal signal receiver (REC) domain while cek1 contains an N-terminal PAS domain. MAST kinases are cytoskeletal associated kinases of unknown function that are also expressed at neuromuscular junctions and postsynaptic densities. The fungal proteins Rim15 and cek1 are involved in the regulation of meiosis and mitosis, respectively.','cd05579','STKc_MAST_like'),(1222098,'gnl|CDD|143338',_binary 'Catalytic domain of Cyclin-Dependent protein Kinase Like Serine/Threonine Kinases.  Serine/Threonine Kinases (STKs), Cyclin-dependent protein kinase like (CDKL) subfamily, catalytic (c) domain. STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The CDKL subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase. CDKs belong to a large family of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. This subfamily is composed of CDKL1-5 and similar proteins. Some CDKLs, like CDKL1 and CDKL3, may be implicated in transformation and others, like CDKL3 and CDKL5, are associated with mental retardation when impaired. CDKL2 plays a role in learning and memory.','cd07833','STKc_CDKL'),(1222099,'gnl|CDD|249558',_binary 'Protein kinase domain.','pfam00069','Pkinase'),(1222102,'gnl|CDD|233416',_binary 'intein N-terminal splicing region.  This model is based on interated search results, starting with a curated collection of intein N-terminal splicing regions from InBase, the New England Biolabs Intein Database, as presented on its web site. It is designed to recognize inteins but not the related region of the sonic hedgehog protein.','TIGR01445','intein_Nterm'),(1222103,'gnl|CDD|250073',_binary 'Adenylosuccinate synthetase.','pfam00709','Adenylsucc_synt'),(1222106,'gnl|CDD|232863',_binary 'adenylosuccinate synthase.  Alternate name IMP--aspartate ligase [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR00184','purA'),(1222120,'gnl|CDD|255774',_binary 'Predicted nucleotidyltransferase.  Members of this family of bacterial proteins catalyze the transfer of nucleotide residues from nucleoside diphosphates or triphosphates into dimer or polymer forms.','pfam10127','Nuc-transf'),(1222123,'gnl|CDD|205918',_binary 'Helix-turn-helix domain.  Members of this family contains a DNA-binding helix-turn-helix domain.','pfam13744','HTH_37'),(1222128,'gnl|CDD|257768',_binary 'Cro/C1-type HTH DNA-binding domain.  This is a helix-turn-helix domain that probably binds to DNA.','pfam13443','HTH_26'),(1222138,'gnl|CDD|257880',_binary '2\'-5\' RNA ligase superfamily.  This family contains proteins related to pfam02834. These proteins are likely to be enzymes, but they may not share the RNA ligase activity.','pfam13563','2_5_RNA_ligase2'),(1222141,'gnl|CDD|250647',_binary 'LysM domain.  The LysM (lysin motif) domain is about 40 residues long. It is found in a variety of enzymes involved in bacterial cell wall degradation. This domain may have a general peptidoglycan binding function. The structure of this domain is known.','pfam01476','LysM'),(1222160,'gnl|CDD|250775',_binary 'CobQ/CobB/MinD/ParA nucleotide binding domain.  This family consists of various cobyrinic acid a,c-diamide synthases. These include CbiA and CbiP from S.typhimurium, and CobQ from R. capsulatus. These amidases catalyse amidations to various side chains of hydrogenobyrinic acid or cobyrinic acid a,c-diamide in the biosynthesis of cobalamin (vitamin B12) from uroporphyrinogen III. Vitamin B12 is an important cofactor and an essential nutrient for many plants and animals and is primarily produced by bacteria. The family also contains dethiobiotin synthetases as well as the plasmid partitioning proteins of the MinD/ParA family.','pfam01656','CbiA'),(1222168,'gnl|CDD|234188',_binary 'cellulose synthase operon protein YhjQ.  Members of this family are the YhjQ protein, found immediately upsteam of bacterial cellulose synthase (bcs) genes in a broad range of bacteria, including both copies of the bcs locus in Klebsiella pneumoniae. In several species it is seen clearly as part of the bcs operon. It is identified as a probable component of the bacterial cellulose metabolic process not only by gene location, but also by partial phylogenetic profiling, or Haft-Selengut algorithm (PMID:16930487), based on a bacterial cellulose biosynthesis genome property profile. Cellulose plays an important role in biofilm formation and structural integrity in some bacteria. Mutants in yhjQ in Escherichia coli, show altered morphology an growth, but the function of YhjQ has not yet been determined [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR03371','cellulose_yhjQ'),(1222170,'gnl|CDD|234215',_binary 'plasmid partitioning protein RepA.  Members of this family are the RepA (or ParA) protein involved in replicon partitioning. All known examples occur in bacterial species with two or more replicons, on a plasmid or the smaller chromosome. Note that an apparent exception may be seen as a pseudomolecule from assembly of an incompletely sequenced genome. Members of this family belong to a larger family that also includes the enzyme cobyrinic acid a,c-diamide synthase, but assignment of that name to members of this family would be in error [Mobile and extrachromosomal element functions, Plasmid functions].','TIGR03453','partition_RepA'),(1222180,'gnl|CDD|255204',_binary 'ATPase MipZ.  MipZ is an ATPase that forms a complex with the chromosome partitioning protein ParB near the chromosomal origin of replication. It is responsible for the temporal and spatial regulation of FtsZ ring formation.','pfam09140','MipZ'),(1222210,'gnl|CDD|163617',_binary 'Escherichia coli YfcE and related proteins, metallophosphatase domain.  YfcE is a manganase-dependent metallophosphatase, found in bacteria and archaea, that cleaves bis-p-nitrophenyl phosphate, thymidine 5\'-monophosphate-p-nitrophenyl ester, and p-nitrophenyl phosphorylcholine, but is unable to hydrolyze 2\',3 \' or 3\',5\' cyclic nucleic phosphodiesters, and various phosphomonoesters, including p-nitrophenyl phosphate. This family also includes the Bacilus subtilis YsnB and Methanococcus jannaschii MJ0936 proteins.  This domain family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd00841','MPP_YfcE'),(1222212,'gnl|CDD|163616',_binary 'Mre11 nuclease, N-terminal metallophosphatase domain.  Mre11 (also known as SbcD in Escherichia coli) is a subunit of the MRX protein complex. This complex includes: Mre11, Rad50, and Xrs2/Nbs1, and plays a vital role in several nuclear processes including DNA double-strand break repair, telomere length maintenance, cell cycle checkpoint control, and meiotic recombination, in eukaryotes.  During double-strand break repair, the MRX complex is required to hold the two ends of a broken chromosome together.  In vitro studies show that Mre11 has 3\'-5\' exonuclease activity on dsDNA templates and endonuclease activity on dsDNA and ssDNA templates. In addition to the N-terminal phosphatase domain, the eukaryotic MRE11 members of this family have a C-terminal DNA binding domain (not included in this alignment model).  MRE11-like proteins are found in prokaryotes and archaea was well as in eukaryotes.  Mre11 belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd00840','MPP_Mre11_N'),(1222272,'gnl|CDD|256050',_binary 'T5orf172 domain.  This domain was identified by Iyer and colleagues.','pfam10544','T5orf172'),(1222334,'gnl|CDD|257796',_binary 'GDSL-like Lipase/Acylhydrolase family.  This family of presumed lipases and related enzymes are similar to pfam00657.','pfam13472','Lipase_GDSL_2'),(1222337,'gnl|CDD|250036',_binary 'GDSL-like Lipase/Acylhydrolase.','pfam00657','Lipase_GDSL'),(1222340,'gnl|CDD|233560',_binary 'phage tail tape measure protein, TP901 family, core region.  This model represents a reasonably well conserved core region of a family of phage tail proteins. The member from phage TP901-1 was characterized as a tail length tape measure protein in that a shortened form of the protein leads to phage with proportionately shorter tails [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01760','tape_meas_TP901'),(1222341,'gnl|CDD|255789',_binary 'Phage-related minor tail protein.  Members of this family are found in putative phage tail tape measure proteins.','pfam10145','PhageMin_Tail'),(1222342,'gnl|CDD|257806',_binary 'RNase_H superfamily.','pfam13482','RNase_H_2'),(1222348,'gnl|CDD|253118',_binary 'CHAP domain.  This domain corresponds to an amidase function. Many of these proteins are involved in cell wall metabolizm of bacteria. This domain is found at the N-terminus of E. coli bifunctional glutathionylspermidine synthetase/amidase, where is functions as a glutathionylspermidine amidase EC:3.5.1.78. This domain is found to be the catalytic domain of PlyCA.','pfam05257','CHAP'),(1222351,'gnl|CDD|252962',_binary 'Protein of unknown function (DUF669).  Members of this family are found in various phage proteins.','pfam05037','DUF669'),(1222356,'gnl|CDD|257520',_binary 'Protein of unknown function (DUF3987).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences.','pfam13148','DUF3987'),(1222372,'gnl|CDD|257977',_binary 'AAA domain.  This family of domains contain only a P-loop motif, that is characteristic of the AAA superfamily. Many of the proteins in this family are just short fragments so there is no Walker B motif.','pfam13671','AAA_33'),(1222375,'gnl|CDD|255405',_binary 'RNA ligase.  Members of this family include T4 phage proteins with ATP-dependent RNA ligase activity. Host defence to phage may include cleavage and inactivation of specific tRNA molecules; members of this family act to reverse this RNA damage. The enzyme is adenylated, transiently, on a Lys residue in a motif KXDGSL. This family also includes fungal tRNA ligases that have adenylyltransferase activity. tRNA ligases are enzymes required for the splicing of precursor tRNA molecules containing introns.','pfam09511','RNA_lig_T4_1'),(1222471,'gnl|CDD|256053',_binary 'P22_AR N-terminal domain.  This domain was identified by Iyer and colleagues.','pfam10547','P22_AR_N'),(1222478,'gnl|CDD|252196',_binary 'RecT family.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to RecT.','pfam03837','RecT'),(1222480,'gnl|CDD|252839',_binary 'Phage portal protein.  Bacteriophage portal proteins form a dodecamer and is located at a five-fold vertex of the viral capsid. The portal complex forms a channel through which the viral DNA is packaged into the capsid, and exits during infection. The portal protein is though to rotate during DNA packaging. Portal proteins from different phage show little sequence homology, so this family does not represent all portal proteins.','pfam04860','Phage_portal'),(1222481,'gnl|CDD|233455',_binary 'phage portal protein, HK97 family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01537','portal_HK97'),(1222485,'gnl|CDD|249856',_binary 'Single-strand binding protein family.  This family includes single stranded binding proteins and also the primosomal replication protein N (PriB). PriB forms a complex with PriA, PriC and ssDNA.','pfam00436','SSB'),(1222487,'gnl|CDD|233057',_binary 'single stranded DNA-binding protein (ssb).  All proteins in this family for which functions are known are single-stranded DNA binding proteins that function in many processes including transcription, repair, replication and recombination. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00621','ssb'),(1222501,'gnl|CDD|233460',_binary 'phage prohead protease, HK97 family.  This model describes the prohead protease of HK97 and related phage. It is generally encoded next to the gene for the capsid protein that it processes, and in some cases may be fused to it. This family does not show similarity to the prohead protease of phage T4 (see pfam03420) [Mobile and extrachromosomal element functions, Prophage functions, Protein fate, Other].','TIGR01543','proheadase_HK97'),(1222503,'gnl|CDD|252685',_binary 'Caudovirus prohead protease.  Family of Caudovirus prohead proteases also found in a number of bacteria possibly as the result of horizontal transfer.','pfam04586','Peptidase_U35'),(1222509,'gnl|CDD|115907',_binary 'Putative transposase DNA-binding domain.  This putative domain is found at the C-terminus of a large number of transposase proteins. This domain contains four conserved cysteines suggestive of a zinc binding domain. Given the need for transposases to bind DNA as well as the large number of DNA-binding zinc fingers we hypothesise this domain is DNA-binding.','pfam07282','OrfB_Zn_ribbon'),(1222511,'gnl|CDD|250577',_binary 'Probable transposase.  This family includes IS891, IS1136 and IS1341. DUF1225, pfam06774, has now been merged into this family.','pfam01385','OrfB_IS605'),(1222555,'gnl|CDD|238601',_binary 'VanD integrase, IntD, and related integrases, DNA breaking-rejoining enzymes, integrase/recombinases, N- and C-terminal domains. This CD includes various bacterial integrases including those similar to IntD, a putative integrase-like protein, a component of the vanD glycopeptide resistance cluster in Enterococcus faecium BM4339. Members of this CD are predominantly bacterial in origin.','cd01196','INT_VanD'),(1222586,'gnl|CDD|252098',_binary 'Uncharacterized protein family (UPF0150).  This family of small proteins is uncharacterized. In CC_3185 this domain is found next to a DNA binding helix-turn-helix domain pfam01402, which suggests that this is some kind of ligand binding domain. The structure of this domain suggests that these domains oligomerise and due to structural similarities may bind to RNA. The monomer adopts an alpha-beta-beta-beta-alpha fold and forms a homotetramer. Based on the properties and functions of structural homologues of the HB8 monomer, the protein is speculated to be involved in RNA metabolizm, including RNA binding and cleavage.','pfam03681','UPF0150'),(1222630,'gnl|CDD|238603',_binary 'Archaeal site-specific recombinase A (ASSRA), DNA breaking-rejoining enzymes, integrase/recombinases, C-terminal catalytic domain. Members of this CD are archael in origin. No biochemical characterization is available for the proteins of this subgroup at this point.','cd01198','INT_ASSRA_C'),(1222669,'gnl|CDD|257726',_binary 'AAA domain.','pfam13401','AAA_22'),(1222713,'gnl|CDD|205166',_binary 'Domain of Unknown Function with PDB structure (DUF3850).  The search results from NCBI sequence alignment indicates a conserved domain belonging to ASCH superfamily. Dali searching results show that the protein is a structurally similar to the PUA domain, suggesting it may be involved in RNA recognition. It has been reported that the deletion of PUA genes results in impaired growth (RluD) and competitive disadvantage (TruB) in Escherichia coli. Suggestions have been put forward that, apart from their usual catalytic role, certain PUS enzymes (e.g. TruB) may also act as chaperones for RNA folding. The interface interaction indicates that the biomolecule of protein NP_809782.1 should be a dimer.','pfam12961','DUF3850'),(1222722,'gnl|CDD|257220',_binary 'PDDEXK-like domain of unknown function (DUF3799).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 265 and 420 amino acids in length. It appears that these proteins are distantly related to the PDDEXK superfamily and so these domains are likely to be nucleases. This family has a C-terminal cysteine cluster similar to that found in pfam01930.','pfam12684','DUF3799'),(1222806,'gnl|CDD|257231',_binary 'Alpha/beta hydrolase family.  This family contains alpha/beta hydrolase enzymes of diverse specificity.','pfam12697','Abhydrolase_6'),(1222808,'gnl|CDD|249959',_binary 'alpha/beta hydrolase fold.  This catalytic domain is found in a very wide range of enzymes.','pfam00561','Abhydrolase_1'),(1222809,'gnl|CDD|257229',_binary 'Alpha/beta hydrolase family.  This family contains a diverse range of alpha/beta hydrolase enzymes.','pfam12695','Abhydrolase_5'),(1222811,'gnl|CDD|234315',_binary '2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase.  This protein catalyzes the formation of SHCHC, or (1 R,6 R)-2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate, by elmination of pyruvate from 2-succinyl-5-enolpyruvyl-6-hydroxy-3-cyclohexene-1-carboxylate (SEPHCHC). Note that SHCHC synthase activity previously was attributed to MenD, which in fact is SEPHCHC synthase [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR03695','menH_SHCHC'),(1222848,'gnl|CDD|256129',_binary 'Protein of unknown function (DUF2510).  This is family of proteins conserved in Actinobacteria. Many members are annotated as putative membrane proteins but this could not be confirmed.','pfam10708','DUF2510'),(1222865,'gnl|CDD|249817',_binary 'Cytidine and deoxycytidylate deaminase zinc-binding region.','pfam00383','dCMP_cyt_deam_1'),(1222872,'gnl|CDD|258198',_binary 'Helix-turn-helix domain.  This helix-turn-helix domain is often found in transferases and is likely to be DNA-binding.','pfam13936','HTH_38'),(1222941,'gnl|CDD|233393',_binary 'homoserine O-acetyltransferase.  This family describes homoserine-O-acetyltransferase, an enzyme of methionine biosynthesis. This model has been rebuilt to identify sequences more broadly, including a number of sequences suggested to be homoserine O-acetyltransferase based on proximity to other Met biosynthesis genes [Amino acid biosynthesis, Aspartate family].','TIGR01392','homoserO_Ac_trn'),(1222952,'gnl|CDD|234014',_binary 'double-strand break repair protein AddB, alphaproteobacterial type.  AddAB is a system well described in the Firmicutes as a replacement for RecBCD in many prokaryotes for the repair of double stranded break DNA damage. More recently, a distantly related gene pair conserved in many alphaproteobacteria was shown also to function in double-stranded break repair in Rhizobium etli. This family consists of AddB proteins of the alphaproteobacteial type [DNA metabolism, DNA replication, recombination, and repair].','TIGR02786','addB_alphas'),(1222958,'gnl|CDD|253774',_binary 'Queuosine biosynthesis protein QueC.  This family of proteins participate in the biosynthesis of 7-carboxy-7-deazaguanine. They catalyse the conversion of 7-deaza-7-carboxyguanine to preQ0.','pfam06508','QueC'),(1222961,'gnl|CDD|250094',_binary 'Asparagine synthase.  This family is always found associated with pfam00310. Members of this family catalyse the conversion of aspartate to asparagine.','pfam00733','Asn_synthase'),(1222962,'gnl|CDD|250470',_binary '6-pyruvoyl tetrahydropterin synthase.  6-Pyruvoyl tetrahydrobiopterin synthase catalyses the conversion of dihydroneopterin triphosphate to 6-pyruvoyl tetrahydropterin, the second of three enzymatic steps in the synthesis of tetrahydrobiopterin from GTP. The functional enzyme is a hexamer of identical subunits.','pfam01242','PTPS'),(1222964,'gnl|CDD|234186',_binary 'queuosine biosynthesis protein QueD.  Members of this protein family, closely related to eukaryotic 6-pyruvoyl tetrahydrobiopterin synthase enzymes, are the QueD protein of queuosine biosynthesis. Queuosine is a hypermodified base in the wobble position of tRNAs for Tyr, His, Asp, and Asn in many species. This modification, although widespread, appears not to be important for viability. The queuosine precursor made by this enzyme may be converted instead to archeaosine as in some Archaea [Protein synthesis, tRNA and rRNA base modification].','TIGR03367','queuosine_QueD'),(1222965,'gnl|CDD|232793',_binary '6-pyruvoyl tetrahydropterin synthase/QueD family protein.  This model has been downgraded from hypothetical_equivalog to subfamily. The animal enzymes are known to be 6-pyruvoyl tetrahydropterin synthase. The function of the bacterial branch of the sequence lineage had been thought to be the same, but many are now taken to be QueD, and enzyme of queuosine biosynthesis. Queuosine is a hypermodified base in the wobble position of some tRNAs in most species. A new model is built to be the QueD equivalog model [Protein synthesis, tRNA and rRNA base modification].','TIGR00039','6PTHBS'),(1222970,'gnl|CDD|249660',_binary 'Chitinase class I.','pfam00182','Glyco_hydro_19'),(1222972,'gnl|CDD|257719',_binary '4Fe-4S single cluster domain.','pfam13394','Fer4_14'),(1222973,'gnl|CDD|132408',_binary '7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE.  This uncharacterized enzyme, designated QueE, participates in the biosynthesis, from GTP, of 7-cyano-7-deazaguanosine, also called preQ0 because in many species it is a precursor of queuosine. In most Archaea, it is instead the precursor of a different tRNA modified base, archaeosine [Protein synthesis, tRNA and rRNA base modification].','TIGR03365','Bsubt_queE'),(1222975,'gnl|CDD|213935',_binary 'putative 7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE.  Members of this radical SAM domain protein family appear to be the E. coli form of the queuosine biosynthesis protein QueE. QueE is involved in making preQ0 (7-cyano-7-deazaquanine), a precursor of both the bacterial/eukaryotic modified tRNA base queuosine and the archaeal modified base archaeosine. Members occur in species that lack known forms of QueE but usually are not found in queuosine biosynthesis operons. Members of this family tend to form bi-directional best hit matches to members of known (TIGR03365) and putative (TIGR03963) QueE families from other lineages.','TIGR04322','rSAM_QueE_Ecoli'),(1222976,'gnl|CDD|252342',_binary 'Radical SAM superfamily.  Radical SAM proteins catalyse diverse reactions, including unusual methylations, isomerization, sulphur insertion, ring formation, anaerobic oxidation and protein radical formation.','pfam04055','Radical_SAM'),(1222977,'gnl|CDD|213962',_binary 'putative 7-cyano-7-deazaguanosine (preQ0) biosynthesis protein QueE, gammaproteobacterial type.  Members of this radical SAM domain protein family appear to be a form of the queuosine biosynthesis protein QueE. QueE is involved in making preQ0 (7-cyano-7-deazaquanine), a precursor of both the bacterial/eukaryotic modified tRNA base queuosine and the archaeal modified base archaeosine. Members occur in preQ0 operons species that lack members of related protein family TIGR03365.','TIGR04349','rSAM_QueE_gams'),(1222980,'gnl|CDD|233896',_binary 'anaerobic ribonucleoside-triphosphate reductase activating protein.  This enzyme is a member of the radical-SAM family (pfam04055). It is often gene clustered with the class III (anaerobic) ribonucleotide triphosphate reductase (NrdD, TIGR02487) and presumably fulfills the identical function as NrdG which utilizes S-adenosyl methionine, an iron-sulfur cluster and a reductant (dihydroflavodoxin) to produce a glycine-centered radical in NrdD [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism, Protein fate, Protein modification and repair].','TIGR02495','NrdG2'),(1222981,'gnl|CDD|234542',_binary 'spiro-SPASM protein.  This three-domain protein is restricted to the spirochetes and widely distributed (excepting Borrelia). It has a conserved C-terminal SPASM domain, a 4Fe-4S binding domain shared by a number of peptide-modifying and heme-modifying radical SAM proteins. It has a central radical SAM domain, although half the members have lost the signature 4Fe-4S-binding Cys residues, fail to register with the radical SAM domain definition of pfam04055, and must be considered pseudo-SAM proteins. PSI-BLAST shows a relationship between the N-terminal domain and various predicted glycosyltransferases (e.g. Bacillus subtilis SpsF) and cytidyltransferases. In some Treponema species, this protein appears to split into two tandem genes.','TIGR04321','spiroSPASM'),(1222982,'gnl|CDD|257682',_binary '4Fe-4S single cluster domain.  This family includes proteins containing domains which bind to iron-sulfur clusters. Members include bacterial ferredoxins, various dehydrogenases, and various reductases. The structure of the domain is an alpha-antiparallel beta sandwich.','pfam13353','Fer4_12'),(1222983,'gnl|CDD|188569',_binary 'putative heme d1 biosynthesis radical SAM protein NirJ1.  Members of this radical SAM protein subfamily, designated NirJ1, occur in genomic contexts with a paralog NirJ2 and with other nitrite reductase operon genes associated with heme d1 biosynthesis, as in Heliobacillus mobilis and Heliophilum fasciatum. NirJ1 is presumed by bioinformatics analysis (Xiong, et al.) to be a heme d1 biosynthesis protein by context, perhaps involved in conversions of acetate groups to methyl groups in conversion from uroporphyrinogen III.','TIGR04054','rSAM_NirJ1'),(1223001,'gnl|CDD|250460',_binary 'GTP cyclohydrolase I.  This family includes GTP cyclohydrolase enzymes and a family of related bacterial proteins.','pfam01227','GTP_cyclohydroI'),(1223148,'gnl|CDD|232920',_binary 'riboflavin biosynthesis protein RibD.  This model describes the ribD protein as found in Escherichia coli. The N-terminal domain includes the conserved zinc-binding site region captured in the model dCMP_cyt_deam and shared by proteins such as cytosine deaminase, mammalian apolipoprotein B mRNA editing protein, blasticidin-S deaminase, and Bacillus subtilis competence protein comEB. The C-terminal domain is homologous to the full length of yeast HTP reductase, a protein required for riboflavin biosynthesis. A number of archaeal proteins believed related to riboflavin biosynthesis contain only this C-terminal domain and are not found as full-length matches to this model [Biosynthesis of cofactors, prosthetic groups, and carriers, Riboflavin, FMN, and FAD].','TIGR00326','eubact_ribD'),(1223149,'gnl|CDD|255349',_binary 'RNA ligase.  This is a family of RNA ligases. The enzyme repairs RNA strand breaks in nicked DNA:RNA and RNA:RNA but not in DNA:DNA duplexes.','pfam09414','RNA_ligase'),(1223205,'gnl|CDD|249765',_binary 'Ribonucleotide reductase, all-alpha domain.','pfam00317','Ribonuc_red_lgN'),(1223375,'gnl|CDD|249637',_binary 'Phosphoribosyl transferase domain.  This family includes a range of diverse phosphoribosyl transferase enzymes. This family includes: Adenine phosphoribosyl-transferase EC:2.4.2.7. Hypoxanthine-guanine-xanthine phosphoribosyl-transferase. Hypoxanthine phosphoribosyl-transferase EC:2.4.2.8. Ribose-phosphate pyrophosphokinase i EC:2.7.6.1. Amidophosphoribosyltransferase EC:2.4.2.14. Orotate phosphoribosyl-transferase EC:2.4.2.10. Uracil phosphoribosyl-transferase EC:2.4.2.9. Xanthine-guanine phosphoribosyl-transferase EC:2.4.2.22. In Arabidopsis, At the very N-terminus of this domain is the P-Loop NTPase domain.','pfam00156','Pribosyltran'),(1223385,'gnl|CDD|254136',_binary 'Antirestriction protein (ArdA).  This family consists of several bacterial antirestriction (ArdA) proteins.  ArdA functions in bacterial conjugation to allow an unmodified plasmid to evade restriction in the recipient bacterium and yet acquire cognate modification.','pfam07275','ArdA'),(1223450,'gnl|CDD|259027',_binary 'Intein splicing domain.  Inteins are segments of protein which excise themselves from a precursor protein and mediate the rejoining of the remainder of the precursor (the extein). Most inteins consist of a splicing domain which is split into two segments by a homing endonuclease domain. This domain represents the splicing domain.','pfam14890','Intein_splicing'),(1223525,'gnl|CDD|250462',_binary 'HIT domain.','pfam01230','HIT'),(1223638,'gnl|CDD|253083',_binary 'Homing endonuclease.  Homing endonucleases are encoded by mobile DNA elements that are found inserted within host genes in all domains of life.','pfam05204','Hom_end'),(1223655,'gnl|CDD|249939',_binary 'Glycosyl transferase family 2.  Diverse family, transferring sugar from UDP-glucose, UDP-N-acetyl- galactosamine, GDP-mannose or CDP-abequose, to a range of substrates including cellulose, dolichol phosphate and teichoic acids.','pfam00535','Glycos_transf_2'),(1223714,'gnl|CDD|257779',_binary 'Meiotically up-regulated gene 113.  This is a family of fungal proteins found to be up-regulated in meiosis.','pfam13455','MUG113'),(1223718,'gnl|CDD|250840',_binary 'Glycosyltransferase family 25 (LPS biosynthesis protein).  Members of this family belong to Glycosyltransferase family 25 This is a family of glycosyltransferases involved in lipopolysaccharide (LPS) biosynthesis. These enzymes catalyse the transfer of various sugars onto the growing LPS chain during its biosynthesis.','pfam01755','Glyco_transf_25'),(1223719,'gnl|CDD|257894',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13578','Methyltransf_24'),(1223720,'gnl|CDD|191350',_binary 'Macrocin-O-methyltransferase (TylF).  This family consists of bacterial macrocin O-methyltransferase (TylF) proteins. TylF is responsible for the methylation of macrocin to produce tylosin. Tylosin is a macrolide antibiotic used in veterinary medicine to treat infections caused by Gram-positive bacteria and as an animal growth promoter in the swine industry. It is produced by several Streptomyces species. As with other macrolides, the antibiotic activity of tylosin is due to the inhibition of protein biosynthesis by a mechanism that involves the binding of tylosin to the ribosome, preventing the formation of the mRNA-aminoacyl-tRNA-ribosome complex. The structure of one representative sequence from this family, NovP, shows it to be an S-adenosyl-l-methionine-dependent O-methyltransferase that catalyses the penultimate step in the biosynthesis of the aminocoumarin antibiotic novobiocin. Specifically, it methylates at 4-OH of the noviose moiety, and the resultant methoxy group is important for the potency of the mature antibiotic. It is likely that the key structural features of NovP are common to the rest of the family and include: a helical \'lid\' region that gates access to the co-substrate binding pocket and an active centre that contains a 3-Asp putative metal binding site. A further conserved Asp probably acts as the general base that initiates the reaction by de-protonating the 4-OH group of the noviose unit.','pfam05711','TylF'),(1223722,'gnl|CDD|257942',_binary 'Glycosyl transferase family group 2.  Members of this family of prokaryotic proteins include putative glucosyltransferases, which are involved in bacterial capsule biosynthesis.','pfam13632','Glyco_trans_2_3'),(1223739,'gnl|CDD|250694',_binary 'GIY-YIG catalytic domain.  This domain called GIY-YIG is found in the amino terminal region of excinuclease abc subunit c (uvrC), bacteriophage T4 endonucleases segA, segB, segC, segD and segE; it is also found in putative endonucleases encoded by group I introns of fungi and phage. The structure of I-TevI a GIY-YIG endonuclease, reveals a novel alpha/beta-fold with a central three-stranded antiparallel beta-sheet flanked by three helices. The most conserved and putative catalytic residues are located on a shallow, concave surface and include a metal coordination site.','pfam01541','GIY-YIG'),(1223769,'gnl|CDD|258450',_binary 'Domain of unknown function (DUF4355).  This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 180 and 214 amino acids in length.','pfam14265','DUF4355'),(1223859,'gnl|CDD|251254',_binary 'Bacterial Ig-like domain (group 2).  This family consists of bacterial domains with an Ig-like fold. Members of this family are found in bacterial and phage surface proteins such as intimins.','pfam02368','Big_2'),(1223864,'gnl|CDD|250424',_binary 'Glycosyl hydrolases family 25.','pfam01183','Glyco_hydro_25'),(1223882,'gnl|CDD|251640',_binary 'Bacterial regulatory protein, Fis family.','pfam02954','HTH_8'),(1223888,'gnl|CDD|205869',_binary 'Winged helix-turn-helix DNA-binding.','pfam13693','HTH_35'),(1223890,'gnl|CDD|233461',_binary 'phage terminase, large subunit, PBSX family.  This model detects members of a highly divergent family of the large subunit of phage terminase. All members are encoded by phage genomes or within prophage regions of bacterial genomes. This is a distinct family from pfam03354 [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01547','phage_term_2'),(1223939,'gnl|CDD|234065',_binary 'RNA polymerase sigma factor, sigma-70 family.  This model encompasses all varieties of the sigma-70 type sigma factors including the ECF subfamily. A number of sigma factors have names with a different number than 70 (i.e. sigma-38), but in fact, all except for the Sigma-54 family (TIGR02395) are included within this family. Several PFAM models hit segments of these sequences including Sigma-70 region 2 (pfam04542) and Sigma-70, region 4 (pfam04545), but not always above their respective trusted cutoffs.','TIGR02937','sigma70-ECF'),(1223946,'gnl|CDD|257728',_binary 'Hint domain.  This domain is found in inteins.','pfam13403','Hint_2'),(1224009,'gnl|CDD|253431',_binary 'DNA N-6-adenine-methyltransferase (Dam).  This family consists of several bacterial and phage DNA N-6-adenine-methyltransferase (Dam) like sequences.','pfam05869','Dam'),(1224044,'gnl|CDD|252854',_binary 'Bacteriophage HK97-gp10, putative tail-component.  This family of proteins is found in the caudovirales. It may be a tail component.','pfam04883','HK97-gp10_like'),(1224055,'gnl|CDD|253047',_binary 'TM2 domain.  This family is composed of a pair of transmembrane alpha helices connected by a short linker. The function of this domain is unknown, however it occurs in a wide range or protein contexts.','pfam05154','TM2'),(1224060,'gnl|CDD|249811',_binary 'MerR family regulatory protein.','pfam00376','MerR'),(1224061,'gnl|CDD|257736',_binary 'MerR HTH family regulatory protein.','pfam13411','MerR_1'),(1224105,'gnl|CDD|238414',_binary 'Cre recombinase, C-terminal catalytic domain.  Cre-like recombinases belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism.  The bacteriophage P1 Cre recombinase maintains the circular phage replicon in a monomeric state by catalyzing a site-specific recombination between two loxP sites.  The catalytic core domain of Cre recombinase is linked to a more divergent helical N-terminal domain, which interacts primarily with the DNA major groove proximal to the crossover region.','cd00799','INT_Cre'),(1224117,'gnl|CDD|255339',_binary 'Ftsk gamma domain.  This domain directs oriented DNA translocation and forms a winged helix structure. Mutated proteins with substitutions in the FtsK gamma DNA-recognition helix are impaired in DNA binding.','pfam09397','Ftsk_gamma'),(1224157,'gnl|CDD|257926',_binary 'Helix-turn-helix of DDE superfamily endonuclease.  This domain is the probable DNA-binding region of transposase enzymes, necessary for efficient DNA transposition. Most of the members derive from the IS superfamily IS5 and rather fewer from IS4.','pfam13613','HTH_Tnp_4'),(1224245,'gnl|CDD|257265',_binary 'Chlorophyllase enzyme.  This family consists of several chlorophyllase and chlorophyllase-2 (EC:3.1.1.14) enzymes. Chlorophyllase (Chlase) is the first enzyme involved in chlorophyll (Chl) degradation and catalyses the hydrolysis of an ester bond to yield chlorophyllide and phytol. The family includes both plant and Amphioxus members.','pfam12740','Chlorophyllase2'),(1224253,'gnl|CDD|256201',_binary 'Protein of unknown function (DUF2800).  This is a family of uncharacterized proteins found in bacteria and viruses. Some members of this family are annotated as being Phi APSE P51-like proteins.','pfam10926','DUF2800'),(1224394,'gnl|CDD|252988',_binary 'Phage virion morphogenesis family.  Protein S of phage P2 is thought to be involved in tail completion and stable head joining.','pfam05069','Phage_tail_S'),(1224396,'gnl|CDD|252922',_binary 'Phage tail sheath protein.  This family includes a variety of phage tail sheath proteins.','pfam04984','Phage_sheath_1'),(1224404,'gnl|CDD|252614',_binary 'Phage terminase large subunit.  Initiation of packaging of double-stranded viral DNA involves the specific interaction of the prohead with viral DNA in a process mediated by a phage-encoded terminase protein. The terminase enzymes are usually hetero-oligomers composed of a small and a large subunit. This region is found on the large subunit and possess an endonuclease and ATPase activity that require Mg2+ and a neutral or slightly basic reaction. This region is also found in bacterial sequences.','pfam04466','Terminase_3'),(1224408,'gnl|CDD|252910',_binary 'Gene 25-like lysozyme.  This family includes the phage protein Gene 25 from T4 which is a structural component of the outer wedge of the baseplate that has acidic lysozyme activity. The family also includes relatives from bacteria that are also presumably lysozymes.','pfam04965','GPW_gp25'),(1224410,'gnl|CDD|253545',_binary 'Protein of unknown function (DUF935).  This family consists of several bacterial proteins of unknown function as well as the Bacteriophage Mu gp29 protein.','pfam06074','DUF935'),(1224411,'gnl|CDD|233797',_binary 'conserved hypothetical protein, phage tail-like region.  This family consists of a large, conserved hypothetical protein in phage tail-like regions of at least six bacterial genomes: Gloeobacter violaceus PCC 7421, Geobacter sulfurreducens PCA, Streptomyces coelicolor A3(2), Streptomyces avermitilis MA-4680, Mesorhizobium loti, and Myxococcus xanthus. The function is unknown [Mobile and extrachromosomal element functions, Prophage functions].','TIGR02243','TIGR02243'),(1224413,'gnl|CDD|252842',_binary 'Baseplate J-like protein.  The P2 bacteriophage J protein lies at the edge of the baseplate. This family also includes a number of bacterial homologues, which are thought to have been horizontally transferred.','pfam04865','Baseplate_J'),(1224419,'gnl|CDD|255437',_binary 'YqaJ-like viral recombinase domain.  This protein family is found in many different bacterial species but is of viral origin. The protein forms an oligomer and functions as a processive alkaline exonuclease that digests linear double-stranded DNA in a Mg(2+)-dependent reaction, It has a preference for 5\'-phosphorylated DNA ends. It thus forms part of the two-component SynExo viral recombinase functional unit.','pfam09588','YqaJ'),(1224445,'gnl|CDD|259774',_binary 'Papain fold toxin 1.  A papain fold toxin domain found in bacterial polymorphic toxin systems. In these systems they might function either as a releasing peptidase or toxin.','pfam15644','Tox-PL'),(1224449,'gnl|CDD|234533',_binary 'nucleoid occlusion protein.  This model describes nucleoid occlusion protein, a close homolog to ParB chromosome partitioning proteins including Spo0J in Bacillus subtilis. Its gene often is located near the gene for the Spo0J ortholog. This protein bind a specific DNA sequence and blocks cytokinesis from happening until chromosome segregation is complete.','TIGR04285','nucleoid_noc'),(1224450,'gnl|CDD|232860',_binary 'ParB/RepB/Spo0J family partition protein.  This model represents the most well-conserved core of a set of chromosomal and plasmid partition proteins related to ParB, including Spo0J, RepB, and SopB. Spo0J has been shown to bind a specific DNA sequence that, when introduced into a plasmid, can serve as partition site. Study of RepB, which has nicking-closing activity, suggests that it forms a transient protein-DNA covalent intermediate during the strand transfer reaction.','TIGR00180','parB_part'),(1224451,'gnl|CDD|234336',_binary 'PRTRC system ParB family protein.  A novel genetic system characterized by six major proteins, included a ParB homolog and a ThiF homolog, is designated PRTRC, or ParB-Related,ThiF-Related Cassette. It is often found on plasmids. This protein family the member related to ParB, and is designated PRTRC system ParB family protein.','TIGR03734','PRTRC_parB'),(1224456,'gnl|CDD|233078',_binary 'DNA polymerase III, beta subunit.  All proteins in this family for which functions are known are components of the DNA polymerase III complex (beta subunit). This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00663','dnan'),(1224461,'gnl|CDD|111641',_binary 'DNA polymerase III beta subunit, C-terminal domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam02768','DNA_pol3_beta_3'),(1224462,'gnl|CDD|202381',_binary 'DNA polymerase III beta subunit, central domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam02767','DNA_pol3_beta_2'),(1224537,'gnl|CDD|233676',_binary 'SUF system FeS assembly protein, NifU family.  Three iron-sulfur cluster assembly systems are known so far. ISC is broadly distributed while NIF tends to be associated with nitrogenase in nitrogen-fixing bacteria. The most recently described is SUF, believed to be important to maintain the function during aerobic stress of enzymes with labile Fe-S clusters. It is fairly widely distributed. This family represents one of two different proteins proposed to act as a scaffold on which the Fe-S cluster is built and from which it is transferred [Biosynthesis of cofactors, prosthetic groups, and carriers, Other].','TIGR01994','SUF_scaf_2'),(1224539,'gnl|CDD|250728',_binary 'NifU-like N terminal domain.  This domain is found in NifU in combination with pfam01106. This domain is found on isolated in several bacterial species. The nif genes are responsible for nitrogen fixation. However this domain is found in bacteria that do not fix nitrogen, so it may have a broader significance in the cell than nitrogen fixation. These proteins appear to be scaffold proteins for iron-sulfur clusters.','pfam01592','NifU_N'),(1224599,'gnl|CDD|252278',_binary 'ArsC family.  This family is related to glutaredoxins pfam00462.','pfam03960','ArsC'),(1224600,'gnl|CDD|233495',_binary 'transcriptional regulator, Spx/MgsR family.  This model represents a portion of the proteins within the larger set covered by pfam03960. That larger family includes a glutaredoxin-dependent arsenate reductase (TIGR00014). Characterized members of this family include Spx and MgsR from Bacillus subtili. Spx is a global regulator for response to thiol-specific oxidative stress. It interacts with RNA polymerase. MgsR (modulator of the general stress response, also called YqgZ) provides a second level of regulation for more than a third of the proteins in the B. subtilis general stress regulon controlled by Sigma-B [Regulatory functions, DNA interactions].','TIGR01617','arsC_related'),(1224614,'gnl|CDD|254214',_binary 'NUMOD4 motif.  NUMOD4 is a putative DNA-binding motif found in homing endonucleases and related proteins.','pfam07463','NUMOD4'),(1224736,'gnl|CDD|240159',_binary 'Aminoglycoside 3\'-phosphotransferase (APH) and Choline Kinase (ChoK) family. The APH/ChoK family is part of a larger superfamily that includes the catalytic domains of other kinases, such as the typical serine/threonine/tyrosine protein kinases (PKs), RIO kinases, actin-fragmin kinase (AFK), and phosphoinositide 3-kinase (PI3K). The family is composed of APH, ChoK, ethanolamine kinase (ETNK), macrolide 2\'-phosphotransferase (MPH2\'), an unusual homoserine kinase, and uncharacterized proteins with similarity to the N-terminal domain of acyl-CoA dehydrogenase 10 (ACAD10). The members of this family catalyze the transfer of the gamma-phosphoryl group from ATP (or CTP) to small molecule substrates such as aminoglycosides, macrolides, choline, ethanolamine, and homoserine. Phosphorylation of the antibiotics, aminoglycosides and macrolides, leads to their inactivation and to bacterial antibiotic resistance. Phosphorylation of choline, ethanolamine, and homoserine serves as precursors to the synthesis of important biological compounds, such as the major phospholipids, phosphatidylcholine and phosphatidylethanolamine and the amino acids, threonine, methionine, and isoleucine.','cd05120','APH_ChoK_like'),(1224765,'gnl|CDD|257738',_binary 'Helix-turn-helix domain.  This domain is a helix-turn-helix domain that probably binds to DNA.','pfam13413','HTH_25'),(1224817,'gnl|CDD|116893',_binary 'LGFP repeat.  This 54 amino acid repeat is found in many hypothetical proteins. Several hypothetical proteins from C.glutamicum and C.efficiens along with PS1 protein contain this repeat region. The N-terminus region of PS1 contains an esterase domain which transfers corynomycolic acid. The C-terminus region consists of 4 tandem LGFP repeats. It is hypothesised that the PS1 proteins in Corynebacterium, when associated with the cell wall, may be anchored via the LGFP tandem repeats that may be important for maintaining cell wall integrity [Adindla et al. Comparative and Functional Genomics 2004; 5:2-16]. Deletion of Corynebacterium glutamicum protein PS1 results in a 10-fold increase in the cell volume of the organism and infers the corresponding proteins involvement in the cell shape formation. The secondary structure of each repeat is predicted to comprise two beta-strands and one alpha-helix [Adindla et al. 2004].','pfam08310','LGFP'),(1224819,'gnl|CDD|254681',_binary 'PE-PPE domain.  This domain is found C terminal to the PE (pfam00934) and PPE (pfam00823) domains. The secondary structure of this domain is predicted to be a mixture of alpha helices and beta strands.','pfam08237','PE-PPE'),(1224821,'gnl|CDD|250350',_binary 'Cutinase.','pfam01083','Cutinase'),(1224843,'gnl|CDD|249970',_binary 'Clp protease.  The Clp protease has an active site catalytic triad. In Escherichia coli Clp protease, ser-111, his-136 and asp-185 form the catalytic triad. Cyanophora parodoxa clpP-B has lost all of these active site residues and is therefore inactive. Some members contain one or two large insertions.','pfam00574','CLP_protease'),(1224900,'gnl|CDD|145844',_binary 'Phage integrase, N-terminal SAM-like domain.','pfam02899','Phage_int_SAM_1'),(1224902,'gnl|CDD|257818',_binary 'Phage integrase, N-terminal SAM-like domain.','pfam13495','Phage_int_SAM_4'),(1224903,'gnl|CDD|238591',_binary 'INT_SG3, DNA breaking-rejoining enzymes, integrase/recombinases subgroup 3, catalytic domain. The CD contains various predicted bacterial and phage integrase/recombinase sequences for which not much experimental characterization is available.','cd01186','INT_SG3_C'),(1224917,'gnl|CDD|258409',_binary 'Domain of unknown function (DUF4326).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea, eukaryotes and viruses. Proteins in this family are typically between 100 and 162 amino acids in length. There are two completely conserved residues (P and C) that may be functionally important.','pfam14216','DUF4326'),(1224991,'gnl|CDD|206527',_binary 'Domain of unknown function (DUF4406).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 98 and 145 amino acids in length.','pfam14359','DUF4406'),(1225009,'gnl|CDD|254791',_binary 'Rad51.  Rad51 is a DNA repair and recombination protein and is a homologue of the bacterial ATPase RecA protein.','pfam08423','Rad51'),(1225019,'gnl|CDD|233794',_binary 'DNA repair protein RAD51.  This eukaryotic sequence family consists of RAD51, a protein involved in DNA homologous recombination and repair. It is similar in sequence the exclusively meiotic recombinase DMC1 (TIGR02238), to archaeal families RadA (TIGR02236) and RadB (TIGR02237), and to bacterial RecA (TIGR02012).','TIGR02239','recomb_RAD51'),(1225058,'gnl|CDD|256611',_binary 'Lsr2.  Lsr2 is a small, basic DNA-bridging protein present in Mycobacterium and related actinomycetes. It is a functional homologue of the H-NS-like proteins. H-NS proteins play a role in nucleoid organisation and also function as a pleiotropic regulator of gene expression.','pfam11774','Lsr2'),(1225063,'gnl|CDD|253021',_binary 'Phage terminase, small subunit.','pfam05119','Terminase_4'),(1225067,'gnl|CDD|233468',_binary 'phage terminase, small subunit, putative, P27 family.  This model describes a distinct family of phage (and integrated prophage) putative terminase small subunit. Members tend to be adjacent to the phage terminase large subunit gene [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01558','sm_term_P27'),(1225095,'gnl|CDD|109845',_binary 'Pentapeptide repeats (8 copies).  These repeats are found in many cyanobacterial proteins. The repeats were first identified in hglK. The function of these repeats is unknown. The structure of this repeat has been predicted to be a beta-helix. The repeat can be approximately described as A(D/N)LXX, where X can be any amino acid.','pfam00805','Pentapeptide'),(1225101,'gnl|CDD|257914',_binary 'Pentapeptide repeats (9 copies).','pfam13599','Pentapeptide_4'),(1225119,'gnl|CDD|250122',_binary 'DnaB-like helicase N terminal domain.  The hexameric helicase DnaB unwinds the DNA duplex at the Escherichia coli chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerization of the N-terminal domain has been observed and may occur during the enzymatic cycle. This N-terminal domain is required both for interaction with other proteins in the primosome and for DnaB helicase activity.','pfam00772','DnaB'),(1225125,'gnl|CDD|232966',_binary 'DNA repair protein RadA.  The gene protuct codes for a probable ATP-dependent protease involved in both DNA repair and degradation of proteins, peptides, glycopeptides. Also known as sms. Residues 11-28 of the SEED alignment contain a putative Zn binding domain. Residues 110-117 of the seed contain a putative ATP binding site both documented in Haemophilus and in Listeria monocytogenes. for E.coli see ( J. BACTERIOL. 178:5045-5048(1996)) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00416','sms'),(1225142,'gnl|CDD|233818',_binary 'RNA ligase, DRB0094 family.  The member of this family from Deinococcus radiodurans, a species that withstands and recovers from extensive radiation or dessication damage, is an apparent RNA ligase. It repairs RNA stand breaks in nicked DNA:RNA and RNA:RNA but not DNA:DNA duplexes. It has adenylyltransferase activity associated with the C-terminal domain. Related proteins also in this family are found in Streptomyces avermitilis MA-4680 and in bacteriophage 44RR2.8t. The phage example is unsurprising since one mechanism of host cell defense against phage is cleavage and inactivation of certain tRNA molecules. A fungal sequence from Neurospora crassa scores between trusted and noise cutofffs and may be similar in function.','TIGR02306','RNA_lig_DRB0094'),(1225863,'gnl|CDD|233546',_binary 'phage protein, HK97 gp10 family.  This model represents an uncharacterized, highly divergent bacteriophage family. The family includes gp10 from HK022 and HK97. It appears related to TIGR01635, a phage morphogenesis family believed to be involved in tail completion [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01725','phge_HK97_gp10'),(1225874,'gnl|CDD|254025',_binary 'Bacteriophage CI repressor helix-turn-helix domain.  This family consists of several phage CI repressor proteins and related bacterial sequences. The CI repressor is known to function as a transcriptional switch, determining whether transcription is lytic or lysogenic.','pfam07022','Phage_CI_repr'),(1225876,'gnl|CDD|147495',_binary 'Protein of unknown function (DUF739).  This family contains several bacteriophage proteins. Some of the proteins in this family have been labeled putative cro repressor proteins.','pfam05339','DUF739'),(1225988,'gnl|CDD|233459',_binary 'phage tail tape measure protein, lambda family.  This model represents a relatively well-conserved region near the C-terminus of the tape measure protein of a lambda and related phage. This protein, which controls phage tail length, is typically about 1000 residues in length. Both low-complexity sequence and insertion/deletion events appear common in this family. Mutational studies suggest a ruler or template role in the determination of phage tail length. Similar behavior is attributed to proteins from distantly related or unrelated families in other phage [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01541','tape_meas_lam_C'),(1225991,'gnl|CDD|257711',_binary 'Concanavalin A-like lectin/glucanases superfamily.  This domain belongs to the Concanavalin A-like lectin/glucanases superfamily.','pfam13385','Laminin_G_3'),(1225997,'gnl|CDD|253342',_binary 'Phage tail protein.  This family consists of several Siphovirus and other phage tail component proteins as well as some bacterial proteins of unknown function.','pfam05709','Sipho_tail'),(1225998,'gnl|CDD|258732',_binary 'Siphovirus ReqiPepy6 Gp37-like protein.  This family includes numerous phage proteins from Siphoviruses. The function of this protein is uncertain, but it is related to pfam06605. In Rhodococcus phage ReqiPepy6 this protein is called Gp37.','pfam14594','Sipho_Gp37'),(1226000,'gnl|CDD|249979',_binary 'Acetyltransferase (GNAT) family.  This family contains proteins with N-acetyltransferase functions such as Elp3-related proteins.','pfam00583','Acetyltransf_1'),(1226020,'gnl|CDD|249756',_binary 'Thymidylate synthase.','pfam00303','Thymidylat_synt'),(1226307,'gnl|CDD|252926',_binary 'Cephalosporin hydroxylase.  Members of this family are about 220 amino acids long. The CmcI protein is presumed to represent the cephalosporin-7--hydroxylase. However this has not been experimentally verified.','pfam04989','CmcI'),(1226308,'gnl|CDD|232997',_binary 'SOS regulatory protein LexA.  LexA acts as a homodimer to repress a number of genes involved in the response to DNA damage (SOS response), including itself and RecA. RecA, in the presence of single-stranded DNA, acts as a co-protease to activate a latent autolytic protease activity (EC 3.4.21.88) of LexA, where the active site Ser is part of LexA. The autolytic cleavage site is an Ala-Gly bond in LexA (at position 84-85 in E. coli LexA; this sequence is replaced by Gly-Gly in Synechocystis). The cleavage leads to derepression of the SOS regulon and eventually to DNA repair. LexA in Bacillus subtilis is called DinR. LexA is much less broadly distributed than RecA [DNA metabolism, DNA replication, recombination, and repair, Regulatory functions, DNA interactions].','TIGR00498','lexA'),(1226318,'gnl|CDD|163633',_binary 'Aquifex aeolicus AQ1575 and related proteins, metallophosphatase domain.  This family includes bacterial and archeal proteins homologous to AQ1575, an uncharacterized Aquifex aeolicus protein.  AQ1575 may play an accessory role in DNA repair, based on the close proximity of its gene to Holliday junction resolvasome genes.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07390','MPP_AQ1575'),(1226325,'gnl|CDD|256539',_binary 'P22 coat protein - gene protein 5.  This family of proteins represents gene product 5 from bacteriophage P22. This protein is involved in the formation of the pro-capsid shells in the bacteriophage. In total, there are 415 molecules of the coat protein which are arranged in an icosahedral shell.','pfam11651','P22_CoatProtein'),(1226339,'gnl|CDD|233757',_binary 'chromosome segregation protein SMC, common bacterial type.  SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins].','TIGR02168','SMC_prok_B'),(1226361,'gnl|CDD|252574',_binary 'ERF superfamily.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to ERF.','pfam04404','ERF'),(1226400,'gnl|CDD|250832',_binary 'Poly A polymerase head domain.  This family includes nucleic acid independent RNA polymerases, such as Poly(A) polymerase, which adds the poly (A) tail to mRNA EC:2.7.7.19. This family also includes the tRNA nucleotidyltransferase that adds the CCA to the 3\' of the tRNA EC:2.7.7.25. This family is part of the nucleotidyltransferase superfamily.','pfam01743','PolyA_pol'),(1226414,'gnl|CDD|257174',_binary 'Probable RNA and SrmB- binding site of polymerase A.  This region encompasses much of the RNA and SrmB binding motifs on polymerase A.','pfam12627','PolyA_pol_RNAbd'),(1226424,'gnl|CDD|251754',_binary 'Plant protein of unknown function.  The function of the plant proteins constituting this family is unknown.','pfam03140','DUF247'),(1226440,'gnl|CDD|250129',_binary 'Dual specificity phosphatase, catalytic domain.  Ser/Thr and Tyr protein phosphatases. The enzyme\'s tertiary fold is highly similar to that of tyrosine-specific phosphatases, except for a \'recognition\' region.','pfam00782','DSPc'),(1226448,'gnl|CDD|234012',_binary 'double-strand break repair helicase AddA, alphaproteobacterial type.  AddAB, also called RexAB, substitutes for RecBCD in several bacterial lineages. These DNA recombination proteins act before synapse and are particularly important for DNA repair of double-stranded breaks by homologous recombination. The term AddAB is used broadly, with AddA homologous between the alphaproteobacteria (as modeled here) and the Firmicutes, while the partner AddB proteins show no strong homology across the two groups of species [DNA metabolism, DNA replication, recombination, and repair].','TIGR02784','addA_alphas'),(1226471,'gnl|CDD|233458',_binary 'phage portal protein, PBSX family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa. This family shows clear homology to TIGR01537. The alignment for this group was trimmed of poorly alignable N-terminal sequence of about 50 residues and of C-terminal regions present in some but not all members of up 180 residues [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01540','portal_PBSX'),(1226475,'gnl|CDD|204386',_binary 'Protein of unknown function (DUF2283).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10049','DUF2283'),(1226492,'gnl|CDD|163614',_binary 'metallophosphatase superfamily, metallophosphatase domain.  Metallophosphatases (MPPs), also known as metallophosphoesterases, phosphodiesterases (PDEs), binuclear metallophosphoesterases, and dimetal-containing phosphoesterases (DMPs), represent a diverse superfamily of enzymes with a conserved domain containing an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. This superfamily includes: the phosphoprotein phosphatases (PPPs), Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd00838','MPP_superfamily'),(1226524,'gnl|CDD|250719',_binary 'FtsK/SpoIIIE family.  FtsK has extensive sequence similarity to wide variety of proteins from prokaryotes and plasmids, termed the FtsK/SpoIIIE family. This domain contains a putative ATP binding P-loop motif. It is found in the FtsK cell division protein from Escherichia coli and the stage III sporulation protein E SpoIIIE, which has roles in regulation of prespore specific gene expression in B. subtilis. A mutation in FtsK causes a temperature sensitive block in cell division and it is involved in peptidoglycan synthesis or modification. The SpoIIIE protein is implicated in intercellular chromosomal DNA transfer.','pfam01580','FtsK_SpoIIIE'),(1226525,'gnl|CDD|234402',_binary 'type VII secretion protein EssC, C-terminal domain.  This model describes the C-terminal domain, or longer subunit, of the Firmicutes type VII secretion protein EssC. This protein (homologous to EccC in Actinobacteria) and the WXG100 target proteins are the only homologous parts of type VII secretion between Firmicutes and Actinobacteria [Protein fate, Protein and peptide secretion and trafficking].','TIGR03928','T7_EssCb_Firm'),(1226528,'gnl|CDD|234400',_binary 'type VII secretion protein EccCa.  This model represents the N-terminal domain or EccCa subunit of the type VII secretion protein EccC as found in the Actinobacteria. Type VII secretion is defined more broadly as including secretion systems for ESAT-6-like proteins in the Firmicutes as well as in the Actinobacteria, but this family does not show close homologs in the Firmicutes [Protein fate, Protein and peptide secretion and trafficking].','TIGR03924','T7SS_EccC_a'),(1226529,'gnl|CDD|234401',_binary 'type VII secretion protein EccCb.  This model represents the C-terminal domain or EccCb subunit of the type VII secretion protein EccC as found in the Actinobacteria. Type VII secretion is defined more broadly as including secretion systems for ESAT-6-like proteins in the Firmicutes as well as in the Actinobacteria, but this family does not show close homologs in the Firmicutes [Protein fate, Protein and peptide secretion and trafficking].','TIGR03925','T7SS_EccC_b'),(1226531,'gnl|CDD|257344',_binary 'AAA-like domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily. Many of the proteins in this family are conjugative transfer proteins.','pfam12846','AAA_10'),(1226533,'gnl|CDD|234004',_binary 'Ku protein, prokaryotic.  Members of this protein family are Ku proteins of non-homologous end joining (NHEJ) DNA repair in bacteria and in at least one member of the archaea (Archaeoglobus fulgidus). Most members are encoded by a gene adjacent to the gene for the DNA ligase that completes the repair. The NHEJ system is broadly but rather sparsely distributed, being present in about one fifth of the first 250 completed prokarytotic genomes. A few species (e.g. Archaeoglobus fulgidus and Bradyrhizobium japonicum) have multiple copies that appear to represent recent paralogous family expansion [DNA metabolism, DNA replication, recombination, and repair].','TIGR02772','Ku_bact'),(1226536,'gnl|CDD|251503',_binary 'Ku70/Ku80 beta-barrel domain.  The Ku heterodimer (composed of Ku70 and Ku80) contributes to genomic integrity through its ability to bind DNA double-strand breaks and facilitate repair by the non-homologous end-joining pathway. This is the central DNA-binding beta-barrel domain. This domain is found in both the Ku70 and Ku80 proteins that form a DNA binding heterodimer.','pfam02735','Ku'),(1226578,'gnl|CDD|233758',_binary 'chromosome segregation protein SMC, primarily archaeal type.  SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins].','TIGR02169','SMC_prok_A'),(1226594,'gnl|CDD|251311',_binary 'RecF/RecN/SMC N terminal domain.  This domain is found at the N terminus of SMC proteins. The SMC (structural maintenance of chromosomes) superfamily proteins have ATP-binding domains at the N- and C-termini, and two extended coiled-coil domains separated by a hinge in the middle. The eukaryotic SMC proteins form two kind of heterodimers: the SMC1/SMC3 and the SMC2/SMC4 types. These heterodimers constitute an essential part of higher order complexes, which are involved in chromatin and DNA dynamics.   This family also includes the RecF and RecN proteins that are involved in DNA metabolizm and recombination.','pfam02463','SMC_N'),(1226599,'gnl|CDD|253248',_binary 'Zinc-binding loop region of homing endonuclease.  This domain is the short zinc-binding loops region of a number of much longer chain homing endonucleases. Such loops are probably stabilized by the zinc and may be viewed as small but separate domains. The common structural feature of these domains is that at least three zinc ligands lie very close to each other in the sequence and are not incorporated into regular secondary structural elements. The biological roles played by these small zinc-binding domains are presently unknown.','pfam05551','zf-His_Me_endon'),(1226601,'gnl|CDD|258135',_binary 'tumor suppressor, Mitostatin.  Trichoplein or mitostatin, was first defined as a meiosis-specific nuclear structural protein. It has since been linked with mitochondrial movement. It is associated with the mitochondrial outer membrane, and over-expression leads to reduction in mitochondrial motility whereas lack of it enhances mitochondrial movement. The activity appears to be mediated through binding the mitochondria to the actin intermediate filaments (IFs).','pfam13868','Trichoplein'),(1226605,'gnl|CDD|255941',_binary 'Putative cell-wall binding lipoprotein.  YkyA is a family of proteins containing a lipoprotein signal and a hydrolase domain. It is similar to cell wall binding proteins and might also be recognisable by a host immune defence system. It is thus likely to belong to pathways important for pathogenicity.','pfam10368','YkyA'),(1226606,'gnl|CDD|144972',_binary 'Myosin tail.  The myosin molecule is a multi-subunit complex made up of two heavy chains and four light chains it is a fundamental contractile protein found in all eukaryote cell types. This family consists of the coiled-coil myosin heavy chain tail region. The coiled-coil is composed of the tail from two molecules of myosin. These can then assemble into the macromolecular thick filament. The coiled-coil region provides the structural backbone the thick filament.','pfam01576','Myosin_tail_1'),(1226607,'gnl|CDD|259749',_binary 'Ciliary protein causing Leber congenital amaurosis disease.  Lebercilin is a family of eukaryotic ciliary proteins. Mutations in the gene, LCA5, are implicated in the disease Leber congenital amaurosis. In photoreceptors, lebercilin is uniquely localized at the cilium that bridges the inner and outer segments. Lebercilin functions as an integral element of selective protein transport through photoreceptor cilia. Lebercilin specifically interacts with the intraflagellar transport (IFT), and disruption of IFT can lead to Leber congenital amaurosis.','pfam15619','Lebercilin'),(1226612,'gnl|CDD|253253',_binary 'Mitotic checkpoint protein.  This family consists of several eukaryotic mitotic checkpoint (Mitotic arrest deficient or MAD) proteins. The mitotic spindle checkpoint monitors proper attachment of the bipolar spindle to the kinetochores of aligned sister chromatids and causes a cell cycle arrest in prometaphase when failures occur. Multiple components of the mitotic spindle checkpoint have been identified in yeast and higher eukaryotes. In S.cerevisiae, the existence of a Mad1-dependent complex containing Mad2, Mad3, Bub3 and Cdc20 has been demonstrated.','pfam05557','MAD'),(1226621,'gnl|CDD|234342',_binary 'integrating conjugative element protein, PFL_4705 family.  Members of this protein family are found occasionally on plasmids such as the Pseudomonas putida toluene catabolic TOL plasmid pWWO_p085. Usually, however, they are found on the bacterial main chromosome in regions flanked by markers of conjugative transfer and/or transposition [Mobile and extrachromosomal element functions, Plasmid functions].','TIGR03752','conj_TIGR03752'),(1226648,'gnl|CDD|254474',_binary 'alpha/beta hydrolase fold.  This catalytic domain is found in a very wide range of enzymes.','pfam07859','Abhydrolase_3'),(1226651,'gnl|CDD|249771',_binary 'Prolyl oligopeptidase family.','pfam00326','Peptidase_S9'),(1226654,'gnl|CDD|249618',_binary 'Carboxylesterase family.','pfam00135','COesterase'),(1226730,'gnl|CDD|254388',_binary 'AAA domain (dynein-related subfamily).  This Pfam entry includes some of the AAA proteins not detected by the pfam00004 model.','pfam07728','AAA_5'),(1226733,'gnl|CDD|249500',_binary 'ATPase family associated with various cellular activities (AAA).  AAA family proteins often perform chaperone-like functions that assist in the assembly, operation, or disassembly of protein complexes.','pfam00004','AAA'),(1226783,'gnl|CDD|256460',_binary 'Domain of unknown function (DUF2828).  This is a uncharacterized domain found in eukaryotes and viruses.','pfam11443','DUF2828'),(1226784,'gnl|CDD|257839',_binary 'von Willebrand factor type A domain.','pfam13519','VWA_2'),(1226799,'gnl|CDD|130318',_binary 'ribose-phosphate pyrophosphokinase.  Alternate name: phosphoribosylpyrophosphate synthetase In some systems, close homologs lacking enzymatic activity exist and perform regulatory functions. The model is designated subfamily rather than equivalog for this reason [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01251','ribP_PPkin'),(1226823,'gnl|CDD|257574',_binary 'AAA domain.','pfam13207','AAA_17'),(1226825,'gnl|CDD|254798',_binary 'Chromatin associated protein KTI12.  This is a family of chromatin associated proteins which interact with the Elongator complex, a component of the elongating form of RNA polymerase II. The Elongator complex has histone acetyltransferase activity.','pfam08433','KTI12'),(1226828,'gnl|CDD|252368',_binary 'Nicotinate phosphoribosyltransferase (NAPRTase) family.  Nicotinate phosphoribosyltransferase (EC:2.4.2.11) is the rate limiting enzyme that catalyses the first reaction in the NAD salvage synthesis. This family also includes Pre-B cell enhancing factor that is a cytokine. This family is related to Quinolinate phosphoribosyltransferase pfam01729.','pfam04095','NAPRTase'),(1226834,'gnl|CDD|257595',_binary 'AAA domain.','pfam13238','AAA_18'),(1226908,'gnl|CDD|145774',_binary 'Helix-turn-helix domain of resolvase.','pfam02796','HTH_7'),(1226909,'gnl|CDD|189960',_binary 'Trp repressor protein.  This protein binds to tryptophan and represses transcription of the Trp operon.','pfam01371','Trp_repressor'),(1227118,'gnl|CDD|240172',_binary 'Choline Kinase (ChoK). The ChoK subfamily is part of a larger superfamily that includes the catalytic domains of other kinases, such as the typical serine/threonine/tyrosine protein kinases (PKs), RIO kinases, actin-fragmin kinase (AFK), and phosphoinositide 3-kinase (PI3K). It is composed of bacterial and eukaryotic choline kinases, as well as eukaryotic ethanolamine kinase. ChoK catalyzes the transfer of the gamma-phosphoryl group from ATP (or CTP) to its substrate, choline, producing phosphorylcholine (PCho), a precursor to the biosynthesis of two major membrane phospholipids, phosphatidylcholine (PC), and sphingomyelin (SM). Although choline is the preferred substrate, ChoK also shows substantial activity towards ethanolamine and its N-methylated derivatives. Bacterial ChoK is also referred to as licA protein. ETNK catalyzes the transfer of the gamma-phosphoryl group from CTP to ethanolamine (Etn), the first step in the CDP-Etn pathway for the formation of the major phospholipid, phosphatidylethanolamine (PtdEtn). Unlike ChoK, ETNK shows specific activity for its substrate and displays negligible activity towards N-methylated derivatives of Etn. ChoK plays an important role in cell signaling pathways and the regulation of cell growth.','cd05151','ChoK'),(1227120,'gnl|CDD|233983',_binary 'thiamine kinase.  Members of this family are the ycfN gene product of Escherichia coli, now identified as the salvage enzyme thiamine kinase (thiK), and additional proteobacterial homologs taken to be orthologs with equivalent function [Biosynthesis of cofactors, prosthetic groups, and carriers, Thiamine].','TIGR02721','ycfN_thiK'),(1227172,'gnl|CDD|256401',_binary 'Protein of unknown function (DUF3102).  This family of proteins has no known function.','pfam11300','DUF3102'),(1227188,'gnl|CDD|257869',_binary 'Winged helix-turn helix.  This helix-turn-helix domain is often found in transferases and is likely to be DNA-binding.','pfam13551','HTH_29'),(1227209,'gnl|CDD|153048',_binary 'Ribosome recycling factor.  This family of proteins is found in bacteria and viruses. Proteins in this family are approximately 130 amino acids in length. There are two conserved sequence motifs: LPS and LKR. Overproduction of ribosome recycling factor (RRF) reduces tna operon expression and increases the rate of cleavage of TnaC-tRNA(2)(Pro), relieving the growth inhibition associated with plasmid-mediated tnaC overexpression.','pfam12614','RRF_GI'),(1227269,'gnl|CDD|213553',_binary 'signal peptide peptidase SppA, 36K type.  The related but duplicated, double-length protein SppA (protease IV) of E. coli was shown experimentally to degrade signal peptides as are released by protein processing and secretion. This protein shows stronger homology to the C-terminal region of SppA than to the N-terminal domain or to the related putative protease SuhB. The member of this family from Bacillus subtilis was shown to have properties consistent with a role in degrading signal peptides after cleavage from precursor proteins, although it was not demonstrated conclusively [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00706','SppA_dom'),(1227366,'gnl|CDD|250321',_binary 'MarR family.  The Mar proteins are involved in the multiple antibiotic resistance, a non-specific resistance system. The expression of the mar operon is controlled by a repressor, MarR. A large number of compounds induce transcription of the mar operon. This is thought to be due to the compound binding to MarR, and the resulting complex stops MarR binding to the DNA. With the MarR repression lost, transcription of the operon proceeds. The structure of MarR is known and shows MarR as a dimer with each subunit containing a winged-helix DNA binding motif.','pfam01047','MarR'),(1227372,'gnl|CDD|251041',_binary 'Carbohydrate binding domain.  This family includes diverse carbohydrate binding domains.','pfam02018','CBM_4_9'),(1227391,'gnl|CDD|252617',_binary 'Restriction endonuclease.  Prokaryotic family found in type II restriction enzymes containing the hallmark (D/E)-(D/E)XK active site. Presence of catalytic residues implicates this region in the enzymatic cleavage of DNA.','pfam04471','Mrr_cat'),(1227392,'gnl|CDD|257222',_binary 'Protein of unknown function (DUF3800).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria, archaea, eukaryotes and viruses. Proteins in this family are typically between 215 and 302 amino acids in length. There is a DE motif at the N-terminus and a QXXD motif at the C-terminus that may be functionally important.','pfam12686','DUF3800'),(1227413,'gnl|CDD|258795',_binary 'AP2-like DNA-binding integrase domain.  This family includes AP2-like domains found in a variety of phage integrase proteins. Presumably these domains are DNA-binding.','pfam14657','Integrase_AP2'),(1227429,'gnl|CDD|255729',_binary 'Uncharacterized conserved protein (DUF2303).  Members of this family of hypothetical bacterial proteins have no known function.','pfam10065','DUF2303'),(1227530,'gnl|CDD|233303',_binary 'TIGR01177 family protein.  This family is found exclusively in the Archaea [Hypothetical proteins, Conserved].','TIGR01177','TIGR01177'),(1227614,'gnl|CDD|250124',_binary 'Dioxygenase.','pfam00775','Dioxygenase_C'),(1227712,'gnl|CDD|249533',_binary 'Fibronectin type III domain.','pfam00041','fn3'),(1227806,'gnl|CDD|190992',_binary 'Protein of unknown function (DUF550).  This family is found in a range of Proteobacteria and a few P-22 dsDNA virus particles. The function is currently not known.','pfam04447','DUF550'),(1228091,'gnl|CDD|254989',_binary 'Primase C terminal 1 (PriCT-1).  This alpha helical domain is found at the C terminal of primases.','pfam08708','PriCT_1'),(1228098,'gnl|CDD|201398',_binary 'dUTPase.  dUTPase hydrolyses dUTP to dUMP and pyrophosphate.','pfam00692','dUTPase'),(1228100,'gnl|CDD|129666',_binary 'deoxyuridine 5\'-triphosphate nucleotidohydrolase (dut).  The main function of these proteins is in maintaining the levels of dUTP in the cell to prevent dUTP incorporation into DNA during DNA replication. Pol proteins in viruses are very similar to this protein family. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). Changed role from 132 to 123. RTD [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR00576','dut'),(1228110,'gnl|CDD|233806',_binary 'deoxycytidine triphosphate deaminase.  Members of this family include the Escherichia coli monofunctional deoxycytidine triphosphate deaminase (dCTP deaminase) and a Methanocaldococcus jannaschii bifunctional dCTP deaminase (3.5.4.13)/dUTP diphosphatase (EC 3.6.1.23), which has the EC number 3.5.4.30 for the overall operation [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR02274','dCTP_deam'),(1228139,'gnl|CDD|250076',_binary 'DNA polymerase III beta subunit, N-terminal domain.  A dimer of the beta subunit of DNA polymerase beta forms a ring which encircles duplex DNA. Each monomer contains three domains of identical topology and DNA clamp fold.','pfam00712','DNA_pol3_beta'),(1228160,'gnl|CDD|112623',_binary 'MazG nucleotide pyrophosphohydrolase domain.  This domain is about 100 amino acid residues in length. It is found in the MazG protein from Escherichia coli. It contains four conserved negatively charged residues that probably form an active site or metal binding site. This domain is found in isolation in some proteins as well as associated with pfam00590. This domain is clearly related to pfam01503 another pyrophosphohydrolase involved in histidine biosynthesis. This family may be structurally related to the NUDIX domain pfam00293 (Bateman A pers. obs.).','pfam03819','MazG'),(1228324,'gnl|CDD|255042',_binary 'HIRAN domain.  The HIRAN domain (HIP116, Rad5p N-terminal) is found in the N-terminal regions of the SWI2/SNF2 proteins typified by HIP116 and Rad5p. The HIRAN domain is found as a standalone protein in several bacteria and prophages, or fused to other catalytic domains, such as a nuclease of the restriction endonuclease fold and TDP1-like DNA phosphoesterases, in the eukaryotes. It has been predicted that this domain functions as a DNA-binding domain that probably recognizes features associated with damaged DNA or stalled replication forks.','pfam08797','HIRAN'),(1228495,'gnl|CDD|234013',_binary 'helicase-exonuclease AddAB, AddA subunit, Firmicutes type.  AddAB, also called RexAB, substitutes for RecBCD in several bacterial lineages. These DNA recombination proteins act before synapse and are particularly important for DNA repair of double-stranded breaks by homologous recombination. The term AddAB is used broadly, with AddA homologous between the Firmicutes (as modeled here) and the alphaproteobacteria, while the partner AddB proteins show no strong homology across the two groups of species [DNA metabolism, DNA replication, recombination, and repair].','TIGR02785','addA_Gpos'),(1228694,'gnl|CDD|233819',_binary 'RNA ligase, Rnl2 family.  Members of this family ligate (seal breaks in) RNA. Members so far include phage proteins that can counteract a host defense of cleavage of specific tRNA molecules, trypanosome ligases involved in RNA editing, but no prokaryotic host proteins [Mobile and extrachromosomal element functions, Prophage functions, Transcription, RNA processing].','TIGR02307','RNA_lig_RNL2'),(1228971,'gnl|CDD|256653',_binary 'Protein of unknown function (DUF3349).  This family of proteins are functionally uncharacterized. This protein is found in bacteria. Proteins in this family are typically between 99 to 124 amino acids in length.','pfam11829','DUF3349'),(1228972,'gnl|CDD|257902',_binary 'Transposase DDE domain.  Transposase proteins are necessary for efficient DNA transposition. This domain is a member of the DDE superfamily, which contain three carboxylate residues that are believed to be responsible for coordinating metal ions needed for catalysis.','pfam13586','DDE_Tnp_1_2'),(1228973,'gnl|CDD|250742',_binary 'Transposase DDE domain.  Transposase proteins are necessary for efficient DNA transposition. This domain is a member of the DDE superfamily, which contain three carboxylate residues that are believed to be responsible for coordinating metal ions needed for catalysis. The catalytic activity of this enzyme involves DNA cleavage at a specific site followed by a strand transfer reaction. This family contains transposases for IS4, IS421, IS5377, IS427, IS402, IS1355, IS5, which was original isolated in bacteriophage lambda.','pfam01609','DDE_Tnp_1'),(1228974,'gnl|CDD|257670',_binary 'Putative transposase of IS4/5 family (DUF4096).','pfam13340','DUF4096'),(1228978,'gnl|CDD|254360',_binary 'Major Facilitator Superfamily.','pfam07690','MFS_1'),(1228979,'gnl|CDD|233166',_binary 'Multidrug resistance protein.','TIGR00880','2_A_01_02'),(1228982,'gnl|CDD|233174',_binary 'D-galactonate transporter.  [Transport and binding proteins, Carbohydrates, organic alcohols, and acids].','TIGR00893','2A0114'),(1229015,'gnl|CDD|238415',_binary 'Lambda integrase, C-terminal catalytic domain. Lambda-type integrases catalyze site-specific integration and excision of temperate bacteriophages and other mobile genetic elements to and from the bacterial host chromosome.  They belong to the superfamily of DNA breaking-rejoining enzymes, which share the same fold in their catalytic domain and the overall reaction mechanism. The phage lambda integrase can bridge two different and well-separated DNA sequences called arm- and core-sites.  The C-terminal domain binds, cleaves and re-ligates DNA strands at the core-sites, while the N-terminal domain is largely responsible for high-affinity binding to the arm-type sites.','cd00800','INT_Lambda_C'),(1229067,'gnl|CDD|257841',_binary 'AAA domain.','pfam13521','AAA_28'),(1229238,'gnl|CDD|148935',_binary 'Pretoxin HINT domain.  A member of the HINT superfamily of proteases that is usually found N-terminal to the toxin module in polymorphic toxin systems. The domain is predicted to function in releasing the toxin domain by autoproteolysis.','pfam07591','PT-HINT'),(1229590,'gnl|CDD|233328',_binary 'AAA family ATPase, CDC48 subfamily.  This subfamily of the AAA family ATPases includes two members each from three archaeal species. It also includes yeast CDC48 (cell division control protein 48) and the human ortholog, transitional endoplasmic reticulum ATPase (valosin-containing protein). These proteins in eukaryotes are involved in the budding and transfer of membrane from the transitional endoplasmic reticulum to the Golgi apparatus.','TIGR01243','CDC48'),(1229594,'gnl|CDD|233327',_binary 'ATP-dependent metalloprotease FtsH.  HflB(FtsH) is a pleiotropic protein required for correct cell division in bacteria. It has ATP-dependent zinc metalloprotease activity. It was formerly designated cell division protein FtsH [Cellular processes, Cell division, Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR01241','FtsH_fam'),(1229618,'gnl|CDD|250173',_binary 'AP2 domain.  This 60 amino acid residue domain can bind to DNA and is found in transcription factor proteins.','pfam00847','AP2'),(1229939,'gnl|CDD|251410',_binary 'Antitoxin Phd_YefM, type II toxin-antitoxin system.  Members of this family act as antitoxins in type II toxin-antitoxin systems. When bound to their toxin partners, they can bind DNA via the N-terminus and repress the expression of operons containing genes encoding the toxin and the antitoxin. This domain complexes with Txe toxins containing pfam06769, Fic/DOC toxins containing pfam02661 and YafO toxins containing pfam13957.','pfam02604','PhdYeFM_antitox'),(1229940,'gnl|CDD|252949',_binary 'Plasmid stabilisation system protein.  Members of this family are involved in plasmid stabilisation. The exact molecular function of this protein is not known. This family also encompasses RelE/ParE described in.','pfam05016','Plasmid_stabil'),(1229941,'gnl|CDD|233465',_binary 'prevent-host-death family protein.  This model recognizes a region of about 55 amino acids toward the N-terminal end of bacterial proteins of about 85 amino acids in length. The best-characterized member is prevent-host-death (phd) of bacteriophage P1, the antidote partner of death-on-curing (doc) (TIGR01550) in an addiction module. Addiction modules prevent plasmid curing by killing the host cell as the longer-lived killing protein persists while the gene for the shorter-lived antidote is lost. Note, however, that relatively few members of this family appear to be plasmid or phage-encoded. Also, there is little overlap, except for phage P1 itself, of species with this family and with the doc family [Cellular processes, Toxin production and resistance, Mobile and extrachromosomal element functions, Other].','TIGR01552','phd_fam'),(1230012,'gnl|CDD|189996',_binary 'Transglycosylase SLT domain.  This family is distantly related to pfam00062. Members are found in phages, type II, type III and type IV secretion systems.','pfam01464','SLT'),(1230147,'gnl|CDD|250902',_binary 'IPT/TIG domain.  This family consists of a domain that has an immunoglobulin like fold. These domains are found in cell surface receptors such as Met and Ron as well as in intracellular transcription factors where it is involved in DNA binding. CAUTION: This family does not currently recognize a significant number of members.','pfam01833','TIG'),(1230192,'gnl|CDD|258026',_binary 'Helix-turn-helix domain.','pfam13730','HTH_36'),(1230204,'gnl|CDD|252947',_binary 'Nucleoside 2-deoxyribosyltransferase.  Nucleoside 2-deoxyribosyltransferase EC:2.4.2.6 catalyses the cleavage of the glycosidic bonds of 2`-deoxyribonucleosides.','pfam05014','Nuc_deoxyrib_tr'),(1230229,'gnl|CDD|250542',_binary 'OB-fold nucleic acid binding domain.  This family contains OB-fold domains that bind to nucleic acids. The family includes the anti-codon binding domain of lysyl, aspartyl, and asparaginyl -tRNA synthetases (See pfam00152). Aminoacyl-tRNA synthetases catalyse the addition of an amino acid to the appropriate tRNA molecule EC:6.1.1.-. This family also includes part of RecG helicase involved in DNA repair. Replication factor A is a heterotrimeric complex, that contains a subunit in this family. This domain is also found at the C-terminus of bacterial DNA polymerase III alpha chain.','pfam01336','tRNA_anti-codon'),(1230599,'gnl|CDD|234142',_binary 'DGQHR domain.  This highly divergent, uncharacterized domain has several absolutely conserved residues, including a QR pair and FxxxN motif. Its most striking feature, however, is a near invariant pentapeptide motif DGQHR. Several different subfamilies occur specifically as a part of DNA phosphorothioation systems, previously called DND (DNA instability during electrophoresis), while others (e.g. CPS_2936) occur in other contexts suggestive of lateral gene transfer (sporadic distribution of helicase-containing cassettes). The region described by this model is about 280 amino acids in length; additional sequences show local sequence similarity.','TIGR03187','DGQHR'),(1231276,'gnl|CDD|256874',_binary 'Putative lysophospholipase.  This domain is found in bacteria and eukaryotes and is approximately 110 amino acids in length. It is found in association with pfam00561. Many members are annotated as being lysophospholipases, and others as alpha-beta hydrolase fold-containing proteins.','pfam12146','Hydrolase_4'),(1231309,'gnl|CDD|254225',_binary 'Phage Tail Collar Domain.  This region is occasionally found in conjunction with pfam03335. Most of the family appear to be phage tail proteins; however some appear to be involved in other processes. For instance rhiB from Rhizobium leguminosarum may be involved in plant-microbe interactions. A related protein mrpB is involved in the pathogenicity of Microcystis aeruginosa. The finding of this family in a structural component of the phage tail fibre baseplate suggests that its function is structural rather than enzymatic. Structural studies show this region consists of a helix and a loop and three beta-strands. This alignment does not catch the third strand as it is separated from the rest of the structure by around 100 residues. This strand is conserved in homologues but the intervening sequence is not. Much of the function of the Bacteriophage T4 short tail fiber protein appears to reside in this intervening region. In the tertiary structure of the phage baseplate this domain forms part of the \'collar\'. The domain may bind SO4, however the residues accredited with this vary between the PDB file and the Swiss-Prot entry. The long unconserved region maybe due to domain swapping in and out of a loop or reflective of rapid evolution.','pfam07484','Collar'),(1231316,'gnl|CDD|257239',_binary 'Pectate lyase superfamily protein.  This family of proteins possesses a beta helical structure like Pectate lyase. This family is most closely related to glycosyl hydrolase family 28.','pfam12708','Pectate_lyase_3'),(1231322,'gnl|CDD|254676',_binary 'Cpl-7 lysozyme C-terminal domain.  This domain was originally found in the C-terminal moiety of the Cpl-7 lysozyme encoded by the Streptococcus pneumoniae bacteriophage Cp-7. It is assumed that these repeats represent cell wall binding motifs although no direct evidence has been obtained so far.','pfam08230','Cpl-7'),(1233300,'gnl|CDD|203898',_binary 'Sigma-70, region 4.  Region 4 of sigma-70 like sigma-factors are involved in binding to the -35 promoter element via a helix-turn-helix motif.','pfam08281','Sigma70_r4_2'),(1233490,'gnl|CDD|258151',_binary 'Chaperone of endosialidase.  This is the very C-terminal, chaperone, domain of the bacteriophage protein endosialidase. It releases itself, via the serine-lysine dyad at the N-terminus, from the remainder of the end-tail-spike. Cleavage occurs after the threonine which is the final residue of the End-tail-spike family, pfam12219. The endosialidase protein forms homotrimeric molecules in bacteriophages. The catalytic dyad allows this portion of the molecule to be cleaved from the more N-terminal region such that the latter can fold and presumably bind to DNA.','pfam13884','Peptidase_S74'),(1233937,'gnl|CDD|257431',_binary 'Domain of unknown function (DUF3846).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences. This domain is found associated with an pfam07275 like domain. This suggests that this family may also be involved in evading host restriction.','pfam12957','DUF3846'),(1233951,'gnl|CDD|233119',_binary 'ATP-dependent protease La.  This protein is induced by heat shock and other stresses in E. coli, B. subtilis, and other species. The yeast member, designated PIM1, is located in the mitochondrial matrix, required for mitochondrial function, and also induced by heat shock [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR00763','lon'),(1233979,'gnl|CDD|254705',_binary 'HTH domain.  This family includes helix-turn-helix domains in a wide variety of proteins.','pfam08279','HTH_11'),(1234063,'gnl|CDD|233446',_binary 'putative nicotinate phosphoribosyltransferase.  A deep split separates two related families of proteins, one of which includes experimentally characterized examples of nicotinate phosphoribosyltransferase, an the first enzyme of NAD salvage biosynthesis. This model represents the other family. Members have a different (longer) spacing of several key motifs and have an additional C-terminal domain of up to 100 residues. One argument suggesting that this family represents the same enzyme is that no species has a member of both families. Another is that the gene encoding this protein is located near other NAD salvage biosynthesis genes in Nostoc and in at least four different Gram-positive bacteria. NAD and NADP are ubiquitous in life. Most members of this family are Gram-positive bacteria. An additional set of mutually closely related archaeal sequences score between the trusted and noise cutoffs [Biosynthesis of cofactors, prosthetic groups, and carriers, Pyridine nucleotides].','TIGR01513','NAPRTase_put'),(1234075,'gnl|CDD|234457',_binary 'polynucleotide kinase-phosphatase.  Members of this protein family are the bacterial polynucleotide kinase-phosphatase (Pnkp) whose genes occur paired with genes for the 3\' terminal RNA ribose 2\'-O-methyltransferase Hen1. All members of the seed alignment belong to a cassette with the Hen1. The pair acts in bacterial RNA repair. This enzyme performs end-healing reactions on broken RNA, preparing from the RNA ligase to close the break. The working hypothesis is that the combination of Pnkp (RNA repair) and Hen1 (RNA modification) serves to first repair RNA damage from ribotoxins and then perform a modification that prevents the damage from recurring [Transcription, RNA processing].','TIGR04075','bacter_Pnkp'),(1234404,'gnl|CDD|252212',_binary 'Phage major capsid protein E.  Major capsid protein E is involved with the stabilisation of the condensed form of the DNA molecule in phage heads.','pfam03864','Phage_cap_E'),(1234455,'gnl|CDD|256790',_binary 'MgsA AAA+ ATPase C terminal.  The MgsA protein possesses DNA-dependent ATPase and ssDNA annealing activities. MgsA contributes to the recovery of stalled replication forks and therefore prevents genomic instability caused by aberrant DNA replication. Additionally, MgsA may play a role in chromosomal segregation. This is consistent with a report that MgsA co-localizes with the replisome and affects chromosome segregation. This domain represents the C terminal region of MgsA.','pfam12002','MgsA_C'),(1234462,'gnl|CDD|249938',_binary 'Glycosyl transferases group 1.  Mutations in this domain of human PIGA lead to disease (Paroxysmal Nocturnal haemoglobinuria). Members of this family transfer activated sugars to a variety of substrates, including glycogen, Fructose-6-phosphate and lipopolysaccharides. Members of this family transfer UDP, ADP, GDP or CMP linked sugars. The eukaryotic glycogen synthases may be distant members of this family.','pfam00534','Glycos_transf_1'),(1234467,'gnl|CDD|257813',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13489','Methyltransf_23'),(1234468,'gnl|CDD|257958',_binary 'Methyltransferase domain.  This family appears to be a methyltransferase domain.','pfam13649','Methyltransf_25'),(1235024,'gnl|CDD|255031',_binary 'VRR-NUC domain.','pfam08774','VRR_NUC'),(1235026,'gnl|CDD|258688',_binary 'Putative phage protease XkdF.  This domain is largely found on phage proteins. In a number of cases the domain is associated with a SAM-dependent methyltransferase.','pfam14550','Peptidase_U35_2'),(1235031,'gnl|CDD|233802',_binary '2\'-5\' RNA ligase.  This protein family consists of bacterial and archaeal proteins with two tandem copies of Pfam domain pfam02834. Members for which activity has been measured perform a reversible, ATP-independent 2\'-5\'-ligation of what is presumably a non-phyiological substrate: half-tRNA splice intermediates from an intron-containing yeast tRNA. The physiological substrate(s) in prokaryotes may include small 2\'-5\'-link-containing oligonucleotides, perhaps with regulatory or biosynthetic roles [Transcription, RNA processing].','TIGR02258','2_5_ligase'),(1235040,'gnl|CDD|249564',_binary 'RNase H.  RNase H digests the RNA strand of an RNA/DNA hybrid. Important enzyme in retroviral replication cycle, and often found as a domain associated with reverse transcriptases. Structure is a mixed alpha+beta fold with three a/b/a layers.','pfam00075','RNase_H'),(1235060,'gnl|CDD|148055',_binary 'Phage tail assembly chaperone.','pfam06222','Phage_TAC'),(1235068,'gnl|CDD|254169',_binary 'Bacteriophage Mu Gam like protein.  This family consists of bacterial and phage Gam proteins. The gam gene of bacteriophage Mu encodes a protein which protects linear double stranded DNA from exonuclease degradation in vitro and in vivo.','pfam07352','Phage_Mu_Gam'),(1235077,'gnl|CDD|253558',_binary 'AIG2-like family.  AIG2 is an Arabidopsis proteins that exhibit RPS2- and avrRpt2-dependent induction early after infection with Pseudomonas syringae pv maculicola strain ES4326 carrying avrRpt2.','pfam06094','AIG2'),(1235086,'gnl|CDD|257842',_binary 'Glutamine amidotransferase domain.  This domain is a class-II glutamine amidotransferase domain found in a variety of enzymes, such as asparagine synthetase and glutamine--fructose-6-phosphate transaminase.','pfam13522','GATase_6'),(1235087,'gnl|CDD|256909',_binary 'Putative amidoligase enzyme.  This family of proteins are likely to act as amidoligase enzymes Protein in this family are found in conserved gene neighborhoods encoding a glutamine amidotransferase-like thiol peptidase (in proteobacteria) or an Aig2 family cyclotransferase protein (in firmicutes).','pfam12224','Amidoligase_2'),(1235091,'gnl|CDD|130205',_binary 'glucosamine--fructose-6-phosphate aminotransferase (isomerizing).  The member from Methanococcus jannaschii contains an intein [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan, Central intermediary metabolism, Amino sugars].','TIGR01135','glmS'),(1235101,'gnl|CDD|233285',_binary 'amidophosphoribosyltransferase.  Alternate name: glutamine phosphoribosylpyrophosphate (PRPP) amidotransferase [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR01134','purF'),(1235102,'gnl|CDD|249761',_binary 'Glutamine amidotransferases class-II.','pfam00310','GATase_2'),(1235104,'gnl|CDD|233454',_binary 'asparagine synthase (glutamine-hydrolyzing).  This model describes the glutamine-hydrolysing asparagine synthase. A poorly conserved C-terminal extension was removed from the model. Bacterial members of the family tend to have a long, poorly conserved insert lacking from archaeal and eukaryotic sequences. Multiple isozymes have been demonstrated, such as in Bacillus subtilis. Long-branch members of the phylogenetic tree (which typically were also second or third candidate members from their genomes) were removed from the seed alignment and score below trusted cutoff [Amino acid biosynthesis, Aspartate family].','TIGR01536','asn_synth_AEB'),(1235110,'gnl|CDD|250119',_binary 'Ezrin/radixin/moesin family.  This family of proteins contain a band 4.1 domain (pfam00373), at their amino terminus. This family represents the rest of these proteins.','pfam00769','ERM'),(1235129,'gnl|CDD|188067',_binary 'proliferating cell nuclear antigen (pcna).  All proteins in this family for which functions are known form sliding DNA clamps that are used in DNA replication processes. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00590','pcna'),(1235162,'gnl|CDD|233063',_binary 'excinuclease ABC, B subunit.  All proteins in this family for wich functions are known are DNA helicases that function in the nucleotide excision repair and are endonucleases that make the 3\' incision next to DNA damage. They are part of a pathway requiring UvrA, UvrB, UvrC, and UvrD homologs. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00631','uvrb'),(1235265,'gnl|CDD|254663',_binary 'APOBEC-like N-terminal domain.  A mechanism of generating protein diversity is mRNA editing. Members of this family are C-to-U editing enzymes. The N-terminal domain of APOBEC-1 like proteins is the catalytic domain, while the C-terminal domain is a pseudocatalyitc domain. More specifically, the catalytic domain is a zinc dependent deaminases domain and is essential for cytidine deamination.APOBEC-3 like members contain two copies of this domain. RNA editing by APOBEC-1 requires homodimerization and this complex interacts with RNA binding proteins to from the editosome (and references therein). This family also includes the functionally homologous activation induced deaminase (AID), which is essential for the development of antibody diversity in B lymphocytes, and the sea lamprey PmCDA1 and PmCDA2, which are predicted to play an AID-like role in the adaptive immune response of jawless vertebrates. Divergent members of this family are present in various eukaryotes such as Nematostella, C. elegans, Micromonas and Emiliania, and prokaryotes such as Wolbachia and Pseudomonas brassicacearum.','pfam08210','APOBEC_N'),(1235277,'gnl|CDD|249574',_binary 'Thioredoxin.  Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond. Some members with only the active site are not separated from the noise.','pfam00085','Thioredoxin'),(1235279,'gnl|CDD|257499',_binary 'Thioredoxin-like domain.','pfam13098','Thioredoxin_2'),(1235282,'gnl|CDD|218139',_binary 'Sigma-70, region 4.  Region 4 of sigma-70 like sigma-factors are involved in binding to the -35 promoter element via a helix-turn-helix motif. Due to the way Pfam works, the threshold has been set artificially high to prevent overlaps with other helix-turn-helix families. Therefore there are many false negatives.','pfam04545','Sigma70_r4'),(1236585,'gnl|CDD|255772',_binary 'Mu-like prophage I protein.  Members of this family of proteins comprise various viral Mu-like prophage I proteins.','pfam10123','Mu-like_Pro'),(1236718,'gnl|CDD|163542',_binary 'putative zinc finger/helix-turn-helix protein, YgiT family.  This model describes a family of predicted regulatory proteins with a conserved zinc finger/HTH architecture. The amino-terminal region contains a novel domain, featuring two CXXC motifs and occuring in a number of small bacterial proteins as well as in the present family. The carboxyl-terminal region consists of a helix-turn-helix domain, modeled by pfam01381. The predicted function is DNA binding and transcriptional regulation.','TIGR03830','CxxCG_CxxCG_HTH'),(1237029,'gnl|CDD|256529',_binary 'Domain of unknown function (DUF4352).  Members of these family are poutative lipoproteins that fall into the Antigen MPT63/MPB63 (immunoprotective extracellular protein) superfamily.','pfam11611','DUF4352'),(1237082,'gnl|CDD|130799',_binary 'pimelyl-[acyl-carrier protein] methyl ester esterase.  This CoA-binding enzyme is required for the production of pimeloyl-coenzyme A, the substrate of the BioF protein early in the biosynthesis of biotin. Its exact function is unknown, but is proposed in ref 2. This enzyme belongs to the alpha/beta hydrolase fold family (Pfam model pfam00561). Members of this family are restricted to the Proteobacteria [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin].','TIGR01738','bioH'),(1237617,'gnl|CDD|256766',_binary 'Scavenger mRNA decapping enzyme C-term binding.  This family consists of several scavenger mRNA decapping enzymes (DcpS) and is the C-terminal region. DcpS is a scavenger pyrophosphatase that hydrolyses the residual cap structure following 3\' to 5\' decay of an mRNA. The association of DcpS with 3\' to 5\' exonuclease exosome components suggests that these two activities are linked and there is a coupled exonucleolytic decay-dependent decapping pathway. The C-terminal domain contains a histidine triad (HIT) sequence with three histidines separated by hydrophobic residues. The central histidine within the DcpS HIT motif is critical for decapping activity and defines the HIT motif as a new mRNA decapping domain, making DcpS the first member of the HIT family of proteins with a defined biological function.','pfam11969','DcpS_C'),(1237624,'gnl|CDD|252279',_binary 'Protein of unknown function (DUF342).  This family of bacterial proteins has no known function. The proteins are in the region of 500-600 amino acid residues in length.','pfam03961','DUF342'),(1237634,'gnl|CDD|249581',_binary 'von Willebrand factor type A domain.','pfam00092','VWA'),(1237682,'gnl|CDD|132290',_binary 'succinylornithine transaminase family.  Members of the seed alignment for this protein family are the enzyme succinylornithine transaminase (EC 2.6.1.81), which catalyzes the third of five steps in arginine succinyltransferase (AST) pathway, an ammonia-releasing pathway of arginine degradation. All seed alignment sequences are found within arginine succinyltransferase operons, and all proteins that score above 820.0 bits should function as succinylornithine transaminase. However, a number of sequences extremely closely related in sequence, found in different genomic contexts, are likely to act in different biological processes and may act on different substrates. This model is desigated subfamily rather than equivalog, pending further consideration, for this reason [Energy metabolism, Amino acids and amines].','TIGR03246','arg_catab_astC'),(1237686,'gnl|CDD|132295',_binary 'L-lysine 6-transaminase.  Characterized members of this protein family are L-lysine 6-transaminase, also called lysine epsilon-aminotransferase (LAT). The immediate product of the reaction of this enzyme on lysine, 2-aminoadipate 6-semialdehyde, becomes 1-piperideine 6-carboxylate, or P6C. This product may be converted subsequently to pipecolate or alpha-aminoadipate, lysine catabolites that may be precursors of certain seconary metabolites.','TIGR03251','LAT_fam'),(1237693,'gnl|CDD|233829',_binary 'chaperone protein DnaJ.  This model represents bacterial forms of DnaJ, part of the DnaK-DnaJ-GrpE chaperone system. The three components typically are encoded by consecutive genes. DnaJ homologs occur in many genomes, typically not near DnaK and GrpE-like genes; most such genes are not included by this family. Eukaryotic (mitochondrial and chloroplast) forms are not included in the scope of this family.','TIGR02349','DnaJ_bact'),(1237707,'gnl|CDD|233000',_binary 'adenosylmethionine-8-amino-7-oxononanoate transaminase.  All members of the seed alignment have been demonstrated experimentally to act as EC 2.6.1.62, an enzyme in the biotin biosynthetic pathway. Alternate names include 7,8-diaminopelargonic acid aminotransferase, DAPA aminotransferase, and adenosylmethionine-8-amino-7-oxononanoate aminotransferase. The gene symbol is bioA in E. coli and BIO3 in S. cerevisiae [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin].','TIGR00508','bioA'),(1237721,'gnl|CDD|130940',_binary 'ornithine aminotransferase.  This model describes the final step in the biosynthesis of ornithine from glutamate via the non-acetylated pathway. Ornithine amino transferase takes L-glutamate 5-semialdehyde and makes it into ornithine, which is used in the urea cycle, as well as in the biosynthesis of arginine. This model includes low-GC bacteria and eukaryotic species. The genes from two species are annotated as putative acetylornithine aminotransferases - one from Porphyromonas gingivalis (OMNI|PG1271), and the other from Staphylococcus aureus (OMNI|SA0170). After homology searching using BLAST it was determined that these two sequences were most closely related to ornithine aminotransferases. This model\'s seed includes one characterized hit, from Bacillus subtilis (SP|P38021).','TIGR01885','Orn_aminotrans'),(1237752,'gnl|CDD|234368',_binary 'terminal organelle assembly protein TopJ.  This model describes TopJ (MG_200, CbpA), a DnaJ homolog and probable assembly protein of the Mycoplasma terminal organelle. The terminal organelle is involved in both cytadherence and gliding motility [Cellular processes, Chemotaxis and motility].','TIGR03835','termin_org_DnaJ'),(1237773,'gnl|CDD|233100',_binary 'glutamate-1-semialdehyde-2,1-aminomutase.  This enzyme, glutamate-1-semialdehyde-2,1-aminomutase (glutamate-1-semialdehyde aminotransferase, GSA aminotransferase), contains a pyridoxal phosphate attached at a Lys residue at position 283 of the seed alignment. It is in the family of class III aminotransferases [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR00713','hemL'),(1237821,'gnl|CDD|256595',_binary 'Protein of unknown function (DUF3307).  This family of bacterial proteins has no known function.','pfam11750','DUF3307'),(1237828,'gnl|CDD|258090',_binary 'Domain of unknown function (DUF4185).','pfam13810','DUF4185'),(1237834,'gnl|CDD|251573',_binary 'Clp amino terminal domain.  This short domain is found in one or two copies at the amino terminus of ClpA and ClpB proteins from bacteria and eukaryotes. The function of these domains is uncertain but they may form a protein binding site.','pfam02861','Clp_N'),(1237836,'gnl|CDD|234173',_binary 'ATP-dependent chaperone ClpB.  Members of this protein family are the bacterial ATP-dependent chaperone ClpB. This protein belongs to the AAA family, ATPases associated with various cellular activities (pfam00004). This molecular chaperone does not act as a protease, but rather serves to disaggregate misfolded and aggregated proteins [Protein fate, Protein folding and stabilization].','TIGR03346','chaperone_ClpB'),(1237838,'gnl|CDD|258433',_binary 'Domain of unknown function (DUF4343).  This domain family is found in bacteria, eukaryotes and viruses, and is typically between 127 and 142 amino acids in length.','pfam14243','DUF4343'),(1237841,'gnl|CDD|213892',_binary 'ribonucleoside-diphosphate reductase, class 1b, beta subunit.  Members of this family are NrdF, the beta subunit of class 1b ribonucleotide reductase. This form uses a dimanganese moiety associated with a tyrosine radical to reduce the cellular requirement for iron [Purines, pyrimidines, nucleosides, and nucleotides, 2\'-Deoxyribonucleotide metabolism].','TIGR04171','RNR_1b_NrdF'),(1237848,'gnl|CDD|249730',_binary 'Ribonucleotide reductase, small chain.','pfam00268','Ribonuc_red_sm'),(1237855,'gnl|CDD|254488',_binary 'Ion channel.  This family includes the two membrane helix type ion channels found in bacteria.','pfam07885','Ion_trans_2'),(1237867,'gnl|CDD|253353',_binary 'NACHT domain.  This NTPase domain is found in apoptosis proteins as well as those involved in MHC transcription activation. This family is closely related to pfam00931.','pfam05729','NACHT'),(1237884,'gnl|CDD|250279',_binary 'Sec1 family.','pfam00995','Sec1'),(1237887,'gnl|CDD|258255',_binary 'Protein of unknown function (DUF4236).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 69 and 402 amino acids in length.','pfam14020','DUF4236'),(1237902,'gnl|CDD|251013',_binary 'Peptidyl-tRNA hydrolase PTH2.  Peptidyl-tRNA hydrolases are enzymes that release tRNAs from peptidyl-tRNA during translation.','pfam01981','PTH2'),(1237926,'gnl|CDD|258343',_binary 'Domain of unknown function (DUF4294).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria. Proteins in this family are typically between 192 and 226 amino acids in length.','pfam14127','DUF4294'),(1238685,'gnl|CDD|146915',_binary 'Baculovirus polyhedron envelope protein, PEP, C terminus.  Polyhedra are large crystalline occlusion bodies containing nucleopolyhedrovirus virions, and surrounded by an electron-dense structure called the polyhedron envelope or polyhedron calyx. The polyhedron envelope (associated) protein PEP is thought to be an integral part of the polyhedron envelope. PEP is concentrated at the surface of polyhedra, and is thought to be important for the proper formation of the periphery of polyhedra. It is thought that PEP may stabilize polyhedra and protect them from fusion or aggregation.','pfam04513','Baculo_PEP_C'),(1238926,'gnl|CDD|257951',_binary 'Glycosyltransferase like family 2.  Members of this family of prokaryotic proteins include putative glucosyltransferase, which are involved in bacterial capsule biosynthesis.','pfam13641','Glyco_tranf_2_3'),(1238948,'gnl|CDD|234331',_binary 'Kae1-associated kinase Bud32.  Members of this protein family are the Bud32 protein associated with Kae1 (kinase-associated endopeptidase 1) in the Archaea. In many Archaeal genomes, Kae1 and Bud32 are fused. The complex is homologous to the Kae1 and Bud32 subunits of the eukaryotic KEOPS complex, an apparently ancient protein kinase-containing molecular machine [Unknown function, General].','TIGR03724','arch_bud32'),(1238954,'gnl|CDD|250760',_binary 'Phosphotransferase enzyme family.  This family consists of bacterial antibiotic resistance proteins, which confer resistance to various aminoglycosides they include: aminoglycoside 3\'-phosphotransferase or kanamycin kinase / neomycin-kanamycin phosphotransferase and streptomycin 3\'\'-kinase or streptomycin 3\'\'-phosphotransferase. The aminoglycoside phosphotransferases inactivate aminoglycoside antibiotics via phosphorylation. This family also includes homoserine kinase. This family is related to fructosamine kinase pfam03881.','pfam01636','APH'),(1238962,'gnl|CDD|254686',_binary 'Methyltransferase domain.  Members of this family are SAM dependent methyltransferases.','pfam08242','Methyltransf_12'),(1239022,'gnl|CDD|250668',_binary 'Phosphoribosyl-ATP pyrophosphohydrolase.  This enzyme catalyses the second step in the histidine biosynthetic pathway.','pfam01503','PRA-PH'),(1239134,'gnl|CDD|255375',_binary 'H-type lectin domain.  The H-type lectin domain is a unit of six beta chains, combined into a homo-hexamer. It is involved in self/non-self recognition of cells, through binding with carbohydrates. It is sometimes found in association with the F5_F8_type_C domain pfam00754.','pfam09458','H_lectin'),(1239571,'gnl|CDD|251054',_binary 'SAP domain.  The SAP (after SAF-A/B, Acinus and PIAS) motif is a putative DNA/RNA binding domain found in diverse nuclear and cytoplasmic proteins.','pfam02037','SAP'),(1239599,'gnl|CDD|250732',_binary 'O-methyltransferase.  Members of this family are O-methyltransferases. The family includes catechol o-methyltransferase, caffeoyl-CoA O-methyltransferase, and a family of bacterial O-methyltransferases that may be involved in antibiotic production.','pfam01596','Methyltransf_3'),(1239607,'gnl|CDD|234492',_binary 'methyltransferase, ATP-grasp peptide maturase system.  Members of this protein family are predicted SAM-dependent methyltransferases that regularly occur in the context of a putative peptide modification ATP-grasp enzyme (TIGR04187, related to enzymes of microviridin maturation) and a putative ribosomal peptide modification target (TIGR04186).','TIGR04188','methyltr_grsp'),(1239615,'gnl|CDD|232816',_binary 'protein-L-isoaspartate(D-aspartate) O-methyltransferase.  This is an all-kingdom (but not all species) full-length ortholog enzyme for repairing aging proteins. Among the prokaryotes, the gene name is pcm. Among eukaryotes, pimt [Protein fate, Protein modification and repair].','TIGR00080','pimt'),(1239616,'gnl|CDD|234563',_binary 'methyltransferase, FxLD system.  Members of this family resemble occur regularly in the vicinity of lantibiotic biosynthesis enzymes and their probable target, the FxLD family of putative ribosomal natural product precursor (TIGR04363). Members resemble protein-L-isoaspartate O-methyltransferase (TIGR00080) and a predicted methyltranserase, TIGR04188, of another putative peptide modification system.','TIGR04364','methyltran_FxLD'),(1240779,'gnl|CDD|256010',_binary 'Leucine-rich repeats of kinetochore protein Cenp-F/LEK1.  Cenp-F, a centromeric kinetochore, microtubule-binding protein consisting of two 1,600-amino acid-long coils, is essential for the full functioning of the mitotic checkpoint pathway. There are several leucine-rich repeats along the sequence of LEK1 that are considered to be zippers, though they do not appear to be binding DNA directly in this instance.','pfam10473','CENP-F_leu_zip'),(1241034,'gnl|CDD|255783',_binary 'vWA found in TerF C terminus.  vWA domain fused to TerD domain typified by the TerF protein. Some times found as solos.','pfam10138','vWA-TerF-like'),(1241110,'gnl|CDD|257737',_binary 'Winged helix-turn-helix DNA-binding.','pfam13412','HTH_24'),(1241113,'gnl|CDD|251077',_binary 'Transcriptional regulator.  This family is related to pfam001022 and other transcription regulation families (personal obs: Yeats C).','pfam02082','Rrf2'),(1241118,'gnl|CDD|251622',_binary 'Bacteriophage lambda head decoration protein D.','pfam02924','HDPD'),(1241134,'gnl|CDD|233341',_binary 'exodeoxyribonuclease VII, small subunit.  This protein is the small subunit for exodeoxyribonuclease VII. Exodeoxyribonuclease VII is made of a complex of four small subunits to one large subunit. The complex degrades single-stranded DNA into large acid-insoluble oligonucleotides. These nucleotides are then degraded further into acid-soluble oligonucleotides [DNA metabolism, Degradation of DNA].','TIGR01280','xseB'),(1241136,'gnl|CDD|251414',_binary 'Exonuclease VII small subunit.  This family consist of exonuclease VII, small subunit EC:3.1.11.6 This enzyme catalyses exonucleolytic cleavage in either 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. This exonuclease VII enzyme is composed of one large subunit and 4 small ones.','pfam02609','Exonuc_VII_S'),(1241142,'gnl|CDD|251408',_binary 'Exonuclease VII, large subunit.  This family consist of exonuclease VII, large subunit EC:3.1.11.6 This enzyme catalyses exonucleolytic cleavage in either 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. This exonuclease VII enzyme is composed of one large subunit and 4 small ones.','pfam02601','Exonuc_VII_L'),(1241143,'gnl|CDD|161783',_binary 'exodeoxyribonuclease VII, large subunit.  This family consist of exodeoxyribonuclease VII, large subunit XseA which catalyses exonucleolytic cleavage in either the 5\'->3\' or 3\'->5\' direction to yield 5\'-phosphomononucleotides. Exonuclease VII consists of one large subunit and four small subunits [DNA metabolism, Degradation of DNA].','TIGR00237','xseA'),(1241145,'gnl|CDD|258036',_binary 'OB-fold nucleic acid binding domain.  This family contains OB-fold domains that bind to nucleic acids.','pfam13742','tRNA_anti_2'),(1241152,'gnl|CDD|163634',_binary 'Pyrococcus furiosus PF1019 and related proteins, metallophosphatase domain.  This family includes bacterial and archeal proteins homologous to PF1019, an uncharacterized Pyrococcus furiosus protein.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07391','MPP_PF1019'),(1241157,'gnl|CDD|151680',_binary 'Protein of unknown function (DUF3039).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11238','DUF3039'),(1241449,'gnl|CDD|258299',_binary 'DNA-sulfur modification-associated.  This is family of bacterial proteins likely to be necessary for binding to DNA and recognizing the modification sites. Members are found in bacteria, archaea and on viral plasmids, and are typically between 354 and 474 amino acids in length. There is a conserved DGQHR sequence motif.','pfam14072','DndB'),(1241510,'gnl|CDD|204704',_binary 'Protein of unknown function (DUF3263).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11662','DUF3263'),(1241554,'gnl|CDD|233633',_binary 'phage recombination protein Bet.  This model represents the phage recombination protein Bet from a number of phage, including phage lambda. All members of this family are found in phage genomes or in putative prophage regions of bacterial genomes [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01913','bet_lambda'),(1241563,'gnl|CDD|250686',_binary 'Transposase.  Transposase proteins are necessary for efficient DNA transposition. This family consists of various Escherichia coli insertion elements and other bacterial transposases some of which are members of the IS3 family.','pfam01527','HTH_Tnp_1'),(1241566,'gnl|CDD|250040',_binary 'Integrase core domain.  Integrase mediates integration of a DNA copy of the viral genome into the host chromosome. Integrase is composed of three domains. The amino-terminal domain is a zinc binding domain pfam02022. This domain is the central catalytic domain. The carboxyl terminal domain that is a non-specific DNA binding domain pfam00552. The catalytic domain acts as an endonuclease when two nucleotides are removed from the 3\' ends of the blunt-ended viral DNA made by reverse transcription. This domain also catalyses the DNA strand transfer reaction of the 3\' ends of the viral DNA to the 5\' ends of the integration site.','pfam00665','rve'),(1241567,'gnl|CDD|257619',_binary 'HTH-like domain.  This domain contains a predicted helix-turn-helix suggesting a DNA-binding function.','pfam13276','HTH_21'),(1241568,'gnl|CDD|257986',_binary 'Integrase core domain.','pfam13683','rve_3'),(1241569,'gnl|CDD|257663',_binary 'Integrase core domain.','pfam13333','rve_2'),(1242103,'gnl|CDD|233456',_binary 'phage portal protein, SPP1 family.  This model represents one of several distantly related families of phage portal protein. This protein forms a hole, or portal, that enables DNA passage during packaging and ejection. It also forms the junction between the phage head (capsid) and the tail proteins. It functions as a dodecamer of a single polypeptide of average mol. wt. of 40-90 KDa [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01538','portal_SPP1'),(1242105,'gnl|CDD|250841',_binary 'Acyltransferase family.  This family includes a range of acyltransferase enzymes. This domain is found in many as yet uncharacterized C. elegans proteins and it is approximately 300 amino acids long.','pfam01757','Acyl_transf_3'),(1242137,'gnl|CDD|233541',_binary 'phage N-6-adenine-methyltransferase.  This model is a fragment-mode model for a phage-borne DNA N-6-adenine-methyltransferase [Mobile and extrachromosomal element functions, Prophage functions, DNA metabolism, Restriction/modification].','TIGR01712','phage_N6A_met'),(1243194,'gnl|CDD|163672',_binary 'human MPPE1 and related proteins, metallophosphatase domain.  MPPE1 is a functionally uncharacterized metallophosphatase domain-containing protein. The MPPE1 gene is located on chromosome 18 and is a candidate susceptibility gene for Bipolar disorder.  MPPE1 belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd08165','MPP_MPPE1'),(1243196,'gnl|CDD|234479',_binary 'metallophosphoesterase, DNA ligase-associated.  Members of this protein family are an uncharacterized putative metallophosphoesterase associated with a DNA ligase, a helicase, and a putative exonuclease. It may play a role in DNA repair. Its system is present in about 12 % of prokaryotic reference genomes.','TIGR04123','P_estr_lig_assc'),(1243443,'gnl|CDD|253160',_binary 'Putative Ig domain.  This alignment represents the conserved core region of ~90 residue repeat found in several haemagglutinins and other cell surface proteins. Sequence similarities to (pfam02494) and (pfam00801) suggest an Ig-like fold (personal obs:C. Yeats). So this family may be similar in function to the (pfam02639) and (pfam02638) domains. This domain is also found in the WisP family of proteins of Tropheryma whipplei.','pfam05345','He_PIG'),(1243449,'gnl|CDD|253507',_binary 'Laminin Domain I.  coiled-coil structure. It has been suggested that the domains I and II from laminin A, B1 and B2 may come together to form a triple helical coiled-coil structure.','pfam06008','Laminin_I'),(1243450,'gnl|CDD|233973',_binary 'TIGR02680 family protein.  Members of this protein family belong to a conserved gene four-gene neighborhood found sporadically in a phylogenetically broad range of bacteria: Nocardia farcinica, Symbiobacterium thermophilum, and Streptomyces avermitilis (Actinobacteria), Geobacillus kaustophilus (Firmicutes), Azoarcus sp. EbN1 and Ralstonia solanacearum (Betaproteobacteria). Proteins in this family average over 1400 amino acids in length [Hypothetical proteins, Conserved].','TIGR02680','TIGR02680'),(1243457,'gnl|CDD|257309',_binary 'MarR family.  The Mar proteins are involved in the multiple antibiotic resistance, a non-specific resistance system. The expression of the mar operon is controlled by a repressor, MarR. A large number of compounds induce transcription of the mar operon. This is thought to be due to the compound binding to MarR, and the resulting complex stops MarR binding to the DNA. With the MarR repression lost, transcription of the operon proceeds. The structure of MarR is known and shows MarR as a dimer with each subunit containing a winged-helix DNA binding motif.','pfam12802','MarR_2'),(1243458,'gnl|CDD|233344',_binary 'nitrogenase cofactor biosynthesis protein NifB.  This model describes NifB, a protein required for the biosynthesis of the iron-molybdenum (or iron-vanadium) cofactor used by the nitrogen-fixing enzyme nitrogenase. NifB belongs to the radical SAM family, and the FeMo cluster biosynthesis process requires S-adenosylmethionine. Archaeal homologs lack the most C-terminal region and score between the trusted and noise cutoffs of this model [Biosynthesis of cofactors, prosthetic groups, and carriers, Other, Central intermediary metabolism, Nitrogen fixation].','TIGR01290','nifB'),(1243511,'gnl|CDD|250300',_binary 'Bacterial regulatory protein, arsR family.  Members of this family contains a DNA binding \'helix-turn-helix\' motif. This family includes other proteins which are not included in the Prosite definition.','pfam01022','HTH_5'),(1243738,'gnl|CDD|152211',_binary 'Cobalamin biosynthesis protein CobT VWA domain.  This family consists of several bacterial cobalamin biosynthesis (CobT) proteins. CobT is involved in the transformation of precorrin-3 into cobyrinic acid.','pfam11775','CobT_C'),(1244443,'gnl|CDD|130080',_binary 'capsular exopolysaccharide family.  This model describes the capsular exopolysaccharide proteins in bacteria. The exopolysaccharide gene cluster consists of several genes which encode a number of proteins which regulate the exoploysaccharide biosynthesis(EPS). Atleast 13 genes espA to espM in streptococcus species seem to direct the EPS proteins and all of which share high homology. Functional roles were characterized by gene disruption experiments which resulted in exopolysaccharide-deficient phenotypes [Transport and binding proteins, Carbohydrates, organic alcohols, and acids].','TIGR01007','eps_fam'),(1244444,'gnl|CDD|234363',_binary 'helicase/secretion neighborhood CpaE-like protein.  Members of this protein family belong to the MinD/ParA family of P-loop NTPases, and in particular show homology to the CpaE family of pilus assembly proteins (see PMID:12370432). Nearly all members are found, not only in a gene context consistent with pilus biogenesis or a pilus-like secretion apparatus, but also near a DEAD/DEAH-box helicase, suggesting an involvement in DNA transfer activity. The model describes a clade restricted to the Actinobacteria.','TIGR03815','CpaE_hom_Actino'),(1244447,'gnl|CDD|251258',_binary 'Anion-transporting ATPase.  This Pfam family represents a conserved domain, which is sometimes repeated, in an anion-transporting ATPase. The ATPase is involved in the removal of arsenate, antimonite, and arsenate from the cell.','pfam02374','ArsA_ATPase'),(1244448,'gnl|CDD|130348',_binary 'light-independent protochlorophyllide reductase, iron-sulfur ATP-binding protein.  The BchL peptide (ChlL in chloroplast and cyanobacteria) is an ATP-binding iron-sulfur protein of the dark form protochlorophyllide reductase, an enzyme similar to nitrogenase. This subunit resembles the nitrogenase NifH subunit [Biosynthesis of cofactors, prosthetic groups, and carriers, Chlorophyll and bacteriochlorphyll].','TIGR01281','DPOR_bchL'),(1244512,'gnl|CDD|233771',_binary 'Glutaredoxin-like family.  This family of glutaredoxin-like proteins is aparrently limited to plants. Multiple isoforms are found in A. thaliana and O.sativa.','TIGR02189','GlrX-like_plant'),(1244665,'gnl|CDD|254725',_binary 'Spc7 kinetochore protein.  This domain is found in cell division proteins which are required for kinetochore-spindle association.','pfam08317','Spc7'),(1244936,'gnl|CDD|256981',_binary 'Helix-turn-helix domain.  This is the N terminal helix-turn-helix domain of Transposase_2 pfam01385.','pfam12323','HTH_OrfB_IS605'),(1244937,'gnl|CDD|233563',_binary 'transposase, IS605 OrfB family, central region.  This model represents a region of a sequence similarity between a family of putative transposases of Thermoanaerobacter tengcongensis, smaller related proteins from Bacillus anthracis, putative transposes described by Pfam model pfam01385, and other proteins [Mobile and extrachromosomal element functions, Transposon functions].','TIGR01766','tspaseT_teng_C'),(1245440,'gnl|CDD|254245',_binary 'Phage protein Gp37/Gp68.  Homologues of phage proteins Gp37 and Gp68 are found in several bacteria.','pfam07505','Gp37_Gp68'),(1245812,'gnl|CDD|148137',_binary 'Ectoine synthase.  This family consists of several bacterial ectoine synthase proteins. The ectABC genes encode the diaminobutyric acid acetyltransferase (EctA), the diaminobutyric acid aminotransferase (EctB), and the ectoine synthase (EctC). Together these proteins constitute the ectoine biosynthetic pathway.','pfam06339','Ectoine_synth'),(1245818,'gnl|CDD|256728',_binary 'Domain of unknown function (DUF3440).  This presumed domain is functionally uncharacterized. This domain is found in bacteria. This domain is typically between 53 to 190 amino acids in length. This domain is found associated with pfam01507. This domain has a conserved KND sequence motif.','pfam11922','DUF3440'),(1245859,'gnl|CDD|204152',_binary 'Domain of unknown function (DUF1937).  This domain is found in a set of hypothetical bacterial proteins. Their exact function has not, as yet, been described.','pfam09152','DUF1937'),(1245880,'gnl|CDD|232794',_binary 'phosphoesterase, MJ0936 family.  Members of this largely uncharacterized family share a motif approximating DXH(X25)GDXXD(X25)GNHD as found in several phosphoesterases, including the nucleases SbcD and Mre11, and a family of uncharacterized archaeal putative phosphoesterases described by TIGR00024. In this family, the His residue in GNHD portion of the motif is not conserved. The member MJ0936, one of two from Methanococcus jannaschii, was shown (PMID:15128743) to act on model phosphodiesterase substrates; a divalent cation was required [Unknown function, Enzymes of unknown specificity].','TIGR00040','yfcE'),(1246298,'gnl|CDD|255508',_binary 'Protein of unknown function (DUF2384).  Proteins in this family are found almost exclusively in the Proteobacteria, but also in Gloeobacter violaceus PCC 7421, a cyanobacterium. The function is unknown.','pfam09722','DUF2384'),(1246620,'gnl|CDD|257744',_binary 'Haloacid dehalogenase-like hydrolase.','pfam13419','HAD_2'),(1246957,'gnl|CDD|233267',_binary 'adenine phosphoribosyltransferase.  A phylogenetic analysis suggested omitting the bi-directional best hit homologs from the spirochetes from the seed for This model and making only tentative predictions of adenine phosphoribosyltransferase function for this lineage. The trusted cutoff score is made high for this reason. Most proteins scoring between the trusted and noise cutoffs are likely to act as adenine phosphotransferase [Purines, pyrimidines, nucleosides, and nucleotides, Salvage of nucleosides and nucleotides].','TIGR01090','apt'),(1246984,'gnl|CDD|256299',_binary 'Minor capsid protein.  Most of the members of this family are annotated as being minor capsid proteins. The genomes carrying the genes usually have three similar proteins adjacent to each other, hence this one being named as No.2.','pfam11114','Minor_capsid_2'),(1246997,'gnl|CDD|253569',_binary 'Domain of unknown function (DUF955).  Family of bacterial and viral proteins with undetermined function. A conserved H-E-X-X-H motif is suggestive of a catalytic active site and shows similarity to pfam01435.','pfam06114','DUF955'),(1247070,'gnl|CDD|252484',_binary 'Protein of unknown function (DUF433).','pfam04255','DUF433'),(1247420,'gnl|CDD|255356',_binary 'PhoD-like phosphatase.','pfam09423','PhoD'),(1247421,'gnl|CDD|163632',_binary 'Bacillus subtilis PhoD and related proteins, metallophosphatase domain.  PhoD (also known as alkaline phosphatase D/APaseD  in Bacillus subtilis) is a secreted phosphodiesterase encoded by phoD of the Pho regulon in Bacillus subtilis.  PhoD homologs are found in prokaryotes, eukaryotes, and archaea.  PhoD contains a twin arginine (RR) motif and is transported by the Tat (Twin-arginine translocation) translocation pathway machinery (TatAyCy).  This family also includes the Fusarium oxysporum Fso1 protein.  PhoD belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07389','MPP_PhoD'),(1247424,'gnl|CDD|253341',_binary 'Orthopoxvirus protein of unknown function (DUF830).  This family consists of several Orthopoxvirus proteins of unknown function.','pfam05708','DUF830'),(1247439,'gnl|CDD|257979',_binary 'Acetyltransferase (GNAT) domain.  This family contains proteins with N-acetyltransferase functions such as Elp3-related proteins.','pfam13673','Acetyltransf_10'),(1247465,'gnl|CDD|163668',_binary 'Shewanella-like phosphatases, metallophosphatase domain.  This family includes bacterial, eukaryotic, and archeal proteins orthologous to the Shewanella cold-active protein-tyrosine phosphatase, CAPTPase.  CAPTPase is an uncharacterized protein that belongs to the Shelph (Shewanella-like phosphatase) family of PPP (phosphoprotein phosphatases).  The PPP family is one of two known protein phosphatase families specific for serine and threonine.  In addition to Shelps, the PPP family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07425','MPP_Shelphs'),(1247468,'gnl|CDD|233250',_binary 'DNA topoisomerase VI, B subunit.  This model describes DNA topoisomerase VI, an archaeal type II DNA topoisomerase (DNA gyrase) [DNA metabolism, DNA replication, recombination, and repair].','TIGR01052','top6b'),(1247473,'gnl|CDD|205797',_binary 'KTSC domain.  This short domain is named after Lysine tRNA synthetase C-terminal domain. It is found at the C-terminus of some Lysyl tRNA synthetases as well as a single domain in bacterial proteins. The domain is about 60 amino acids in length and contains a reasonably conserved YXY motif in the centre of the sequence. The function of this domain is unknown but it could be an RNA binding domain.','pfam13619','KTSC'),(1247483,'gnl|CDD|254998',_binary 'Domain of unknown function (DUF1768).  This is a domain of unknown function. It is alpha helical in structure. The GO annotation for this protein suggests it is involved in nematode larval development and has a positive regulation on growth rate.','pfam08719','DUF1768'),(1247487,'gnl|CDD|233876',_binary 'conserved hypothetical protein, ribA/ribD-fused.  This model describes a sequence region that occurs in at least three different polypeptide contexts. It is found fused to GTP cyclohydrolase II, the RibA of riboflavin biosynthesis (TIGR00505), as in Vibrio vulnificus. It is found fused to riboflavin biosynthesis protein RibD (TIGR00326) in rice and Arabidopsis. It occurs as a standalone protein in a number of bacterial species in varied contexts, including single gene operons and bacteriophage genomes. The member from E. coli currently is named YbiA. The function(s) of members of this family is unknown.','TIGR02464','ribofla_fusion'),(1247491,'gnl|CDD|200074',_binary 'protein disulfide-isomerase domain.  This model describes a domain of eukaryotic protein disulfide isomerases, generally found in two copies. The high cutoff for total score reflects the expectation of finding both copies. The domain is similar to thioredoxin but the redox-active disulfide region motif is APWCGHCK [Protein fate, Protein folding and stabilization].','TIGR01126','pdi_dom'),(1247498,'gnl|CDD|258172',_binary 'Thioredoxin-like.  Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond.','pfam13905','Thioredoxin_8'),(1247501,'gnl|CDD|249945',_binary 'Ribosomal protein L7/L12 C-terminal domain.','pfam00542','Ribosomal_L12'),(1247503,'gnl|CDD|233152',_binary 'ribosomal protein L7/L12.  Ribosomal proteins L7 and L12 are synonymous except for post-translational modification of the N-terminal amino acid. THis model resembles pfam00542 but matches the full length of prokaryotic and organellar proteins rather than just the C-terminus [Protein synthesis, Ribosomal proteins: synthesis and modification].','TIGR00855','L12'),(1247515,'gnl|CDD|249681',_binary 'Ferritin-like domain.  This family contains ferritins and other ferritin-like proteins such as members of the DPS family and bacterioferritins.','pfam00210','Ferritin'),(1247525,'gnl|CDD|232837',_binary 'cytidyltransferase-like domain.  Protein families that contain at least one copy of this domain include citrate lyase ligase, pantoate-beta-alanine ligase, glycerol-3-phosphate cytidyltransferase, ADP-heptose synthase, phosphocholine cytidylyltransferase, lipopolysaccharide core biosynthesis protein KdtB, the bifunctional protein NadR, and a number whose function is unknown. Many of these proteins are known to use CTP or ATP and release pyrophosphate.','TIGR00125','cyt_tran_rel'),(1247532,'gnl|CDD|250640',_binary 'Cytidylyltransferase.  This family includes: Cholinephosphate cytidylyltransferase and Glycerol-3-phosphate cytidylyltransferase.','pfam01467','CTP_transf_2'),(1247550,'gnl|CDD|234448',_binary 'glycosyltransferase, MSMEG_0565 family.  A conserved gene cluster found sporadically from Actinobacteria to Proteobacteria to Cyanobacteria features a radical SAM protein, an N-acetyltransferase, an oxidoreductase, and two additional proteins whose functional classes are unclear. The metabolic role of the cluster is probably biosynthetic. This glycosyltransferase, named from member MSMEG_0565 from Mycobacterium smegmatis, occurs in most but not all instances of the cluster [Unknown function, Enzymes of unknown specificity].','TIGR04047','MSMEG_0565_glyc'),(1247600,'gnl|CDD|256812',_binary 'Domain of unknown function (DUF3523).  This presumed domain is functionally uncharacterized. This domain is found in eukaryotes. This domain is typically between 257 to 277 amino acids in length. This domain is found associated with pfam00004. This domain has a conserved LER sequence motif.','pfam12037','DUF3523'),(1247688,'gnl|CDD|250977',_binary 'Domain of unknown function DUF83.  This domain has no known function. The domain contains three conserved cysteines at its C terminus.','pfam01930','Cas_Cas4'),(1247985,'gnl|CDD|233421',_binary 'group I intron endonuclease.  This model represents one subfamily of endonucleases containing the endo/excinuclease amino terminal domain, pfam01541 at its amino end. A distinct subfamily includes excinuclease abc subunit c (uvrC). Members of pfam01541 are often termed GIY-YIG endonucleases after conserved motifs near the amino end. This subfamily in This model is found in open reading frames of group I introns in both phage and mitochondria. The closely related endonucleases of phage T4: segA, segB, segC, segD and segE, score below the trusted cutoff for the family.','TIGR01453','grpIintron_endo'),(1248091,'gnl|CDD|255111',_binary 'Domain of unknown function (DUF1906).  This domain is found in a set of uncharacterized hypothetical bacterial proteins.','pfam08924','DUF1906'),(1248110,'gnl|CDD|250683',_binary 'Polysaccharide deacetylase.  This domain is found in polysaccharide deacetylase. This family of polysaccharide deacetylases includes NodB (nodulation protein B from Rhizobium) which is a chitooligosaccharide deacetylase. It also includes chitin deacetylase from yeast, and endoxylanases which hydrolyses glucosidic bonds in xylan.','pfam01522','Polysacc_deac_1'),(1248112,'gnl|CDD|188453',_binary 'poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase PgaB.  Two well-characterized systems produce polysaccharide based on N-acetyl-D-glucosamine in straight chains with beta-1,6 linkages. These are encoded by the icaADBC operon in Staphylococcus species, where the system is designated polysaccharide intercellular adhesin (PIA), and the pgaABCD operon in Gram-negative bacteria such as E. coli. Both systems include a putative polysaccharide deacetylase. The PgaB protein, described here, contains an additional domain lacking from its Gram-positive counterpart IcaB (TIGR03933). Deacetylation by this protein appears necessary to allow export through the porin PgaA [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR03938','deacetyl_PgaB'),(1248361,'gnl|CDD|113008',_binary 'CENP-B N-terminal DNA-binding domain.  Centromere Protein B (CENP-B) is a DNA-binding protein localized to the centromere. Within the N-terminal 125 residues, there is a DNA-binding region, which binds to a corresponding 17bp CENP-B box sequence. CENP-B dimers either bind two separate DNA molecules or alternatively, they may bind two CENP-B boxes on one DNA molecule, with the intervening stretch of DNA forming a loop structure. The CENP-B DNA-binding domain consists of two repeating domains, RP1 and RP2. This family corresponds to RP1 has been shown to consist of four helices in a helix-turn-helix structure.','pfam04218','CENP-B_N'),(1248454,'gnl|CDD|234283',_binary 'probable DNA repair protein.  Members of this protein family are bacterial proteins of about 900 amino acids in length. Members show extended homology to proteins in TIGR02786, the AddB protein of double-strand break repair via homologous recombination. Members of this family, therefore, may be DNA repair proteins.','TIGR03623','TIGR03623'),(1248910,'gnl|CDD|163636',_binary 'Deinococcus radiodurans DR1119 and related proteins, metallophosphatase domain.  DR1119 is an uncharacterized Deinococcus radiodurans protein with a metallophosphatase domain.  The domain present in members of this family belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07393','MPP_DR1119'),(1248914,'gnl|CDD|163646',_binary 'Thermus thermophilus TTHA0053 and related proteins, metallophosphatase domain.  TTHA0053 is an uncharacterized Thermus thermophilus protein with a domain that belongs to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd07403','MPP_TTHA0053'),(1248938,'gnl|CDD|258800',_binary 'Coiled-coil region of CCDC155.  This is a small family of eukaryotic proteins of unknown function.ThiS is the central coiled-coil region.','pfam14662','CCDC155'),(1249755,'gnl|CDD|252601',_binary 'tRNAHis guanylyltransferase.  The Thg1 protein from Saccharomyces cerevisiae is responsible for adding a GMP residue to the 5\' end of tRNA His. The catalytic domain Thg1 contains a RRM (ferredoxin) fold palm domain, just like the viral RNA-dependent RNA polymerases, reverse transcriptases, family A and B DNA polymerases, adenylyl cyclases, diguanylate cyclases (GGDEF domain) and the predicted polymerase of the CRISPR system. Thg1 possesses an active site with three acidic residues that chelate Mg++ cations. Thg1 catalyzes polymerization similar to the 5\'-3\' polymerases.','pfam04446','Thg1'),(1249757,'gnl|CDD|258575',_binary 'Thg1 C terminal domain.  Thg1 polymerases contain an additional region of conservation C-terminal to the core palm domain that comprise of 5 helices and two strands. This region has several well-conserved charged residues including a basic residue found towards the end of the first helix of this unit might contribute to the Thg1-specific active site. This C-terminal module of Thg1 is predicted to form a helical bundle that functions equivalently to the fingers of the other nucleic acid polymerases, probably in interacting with the template HtRNA.','pfam14413','Thg1C'),(1249789,'gnl|CDD|234419',_binary 'mycofactocin system glycosyltransferase.  Members of this protein family are putative glycosyltransferases, members of pfam00535 (glycosyl transferase family 2). Members appear mostly in the Actinobacteria, where they appear to be part of a system for converting a precursor peptide (TIGR03969) into a novel redox carrier designated mycofactocin. A radical SAM enzyme, TIGR03962, is a proposed to be a key maturase for mycofactocin.','TIGR03965','mycofact_glyco'),(1249830,'gnl|CDD|233687',_binary 'magnesium protoporphyrin O-methyltransferase.  This model represents the S-adenosylmethionine-dependent O-methyltransferase responsible for methylation of magnesium protoporphyrin IX. This step is essentiasl for the biosynthesis of both chlorophyll and bacteriochlorophyll. This model encompasses two closely related clades, from cyanobacteria (and plants) where it is called ChlM and other photosynthetic bacteria where it is known as BchM [Biosynthesis of cofactors, prosthetic groups, and carriers, Chlorophyll and bacteriochlorphyll].','TIGR02021','BchM-ChlM'),(1249837,'gnl|CDD|233668',_binary 'ubiquinone biosynthesis O-methyltransferase.  This model represents an O-methyltransferase believed to act at two points in the ubiquinone biosynthetic pathway in bacteria (UbiG) and fungi (COQ3). A separate methylase (MenG/UbiE) catalyzes the single C-methylation step. The most commonly used names for genes in this family do not indicate whether this gene is an O-methyl, or C-methyl transferase [Biosynthesis of cofactors, prosthetic groups, and carriers, Menaquinone and ubiquinone].','TIGR01983','UbiG'),(1252024,'gnl|CDD|253155',_binary 'Protein of unknown function (DUF745).  This family consists of several uncharacterized Drosophila melanogaster proteins of unknown function.','pfam05335','DUF745'),(1253112,'gnl|CDD|202372',_binary 'Proliferating cell nuclear antigen, C-terminal domain.  N-terminal and C-terminal domains of PCNA are topologically identical. Three PCNA molecules are tightly associated to form a closed ring encircling duplex DNA.','pfam02747','PCNA_C'),(1253199,'gnl|CDD|257742',_binary 'Glutathione S-transferase, N-terminal domain.','pfam13417','GST_N_3'),(1253336,'gnl|CDD|250257',_binary 'LAGLIDADG endonuclease.','pfam00961','LAGLIDADG_1'),(1255545,'gnl|CDD|233050',_binary 'exodeoxyribonuclease V, beta subunit.  The RecBCD holoenzyme is a multifunctional nuclease with potent ATP-dependent exodeoxyribonuclease activity. Ejection of RecD, as occurs at chi recombinational hotspots, cripples exonuclease activity in favor of recombinagenic helicase activity. All proteins in this family for which functions are known are DNA-DNA helicases that are used as part of an exonuclease-helicase complex (made up of RecBCD homologs) that function to generate substrates for the initiation of recombination and recombinational repair. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00609','recB'),(1256065,'gnl|CDD|255478',_binary 'Phage regulatory protein Rha (Phage_pRha).  Members of this protein family are found in temperate phage and bacterial prophage regions. Members include the product of the rha gene of the lambdoid phage phi-80, a late operon gene. The presence of this gene interferes with infection of bacterial strains that lack integration host factor (IHF), which regulates the rha gene. It is suggested that Rha is a phage regulatory protein.','pfam09669','Phage_pRha'),(1256105,'gnl|CDD|178979',_binary 'GIY-YIG nuclease superfamily protein; Validated.','PRK00329','PRK00329'),(1256270,'gnl|CDD|250158',_binary 'PPE family.  This family named after a PPE motif near to the amino terminus of the domain. The PPE family of proteins all contain an amino-terminal region of about 180 amino acids. The carboxyl terminus of this family are variable, and on the basis of this region fall into at least three groups. The MPTR subgroup has tandem copies of a motif NXGXGNXG. The second subgroup contains a conserved motif at about position 350. The third group are only related in the amino terminal region. The function of these proteins is uncertain but it has been suggested that they may be related to antigenic variation of Mycobacterium tuberculosis.','pfam00823','PPE'),(1256271,'gnl|CDD|259778',_binary 'Restriction endonuclease fold toxin 5.  A predicted toxin of the restriction endonuclease fold present in bacterial polymorphic toxin systems. In bacterial polymorphic toxin systems, the toxin is exported by the type 2, type 5, type 6, or PrsW-peptidase dependent secretion system. Versions of this domain are also found in caudoviruses.','pfam15648','Tox-REase-5'),(1256324,'gnl|CDD|258460',_binary 'HNH endonuclease.  This domain is related to other HNH domain families such as pfam01844. Suggesting that these proteins have a nucleic acid cleaving function.','pfam14279','HNH_5'),(1257536,'gnl|CDD|252698',_binary 'Region found in RelA / SpoT proteins.  This region of unknown function is found in RelA and SpoT of Escherichia coli, and their homologues in plants and in other eubacteria. RelA is a guanosine 3\',5\'-bis-pyrophosphate (ppGpp) synthetase (EC:2.7.6.5) while SpoT is thought to be a bifunctional enzyme catalyzing both ppGpp synthesis and degradation (ppGpp 3\'-pyrophosphohydrolase, (EC:3.1.7.2)). This region is often found in association with HD (pfam01966), a metal-dependent phosphohydrolase, TGS (pfam02824) which is a possible nucleotide-binding region, and the ACT regulatory domain (pfam01842).','pfam04607','RelA_SpoT'),(1257547,'gnl|CDD|258042',_binary 'ATP-dependent DNA helicase recG C-terminal.  This domain may well interact selectively and non-covalently with ATP, adenosine 5\'-triphosphate, a universally important coenzyme and enzyme regulator.','pfam13749','HATPase_c_4'),(1257549,'gnl|CDD|252528',_binary 'Divergent AAA domain.  This family is related to the pfam00004 family, and presumably has the same function (ATP-binding).','pfam04326','AAA_4'),(1257561,'gnl|CDD|150115',_binary 'IclR helix-turn-helix domain.','pfam09339','HTH_IclR'),(1257641,'gnl|CDD|152729',_binary 'Protein of unknown function (DUF3626).  This family of proteins is found in bacteria. Proteins in this family are typically between 294 and 374 amino acids in length.','pfam12294','DUF3626'),(1257882,'gnl|CDD|233377',_binary 'orotate phosphoribosyltransferase, Thermus family.  This model represents a distinct clade of orotate phosphoribosyltransferases. Members include the experimentally determined example from Thermus aquaticus and additional examples from Caulobacter crescentus, Helicobacter pylori, Mesorhizobium loti, and related species [Purines, pyrimidines, nucleosides, and nucleotides, Pyrimidine ribonucleotide biosynthesis].','TIGR01367','pyrE_Therm'),(1257903,'gnl|CDD|250268',_binary 'Thioesterase domain.  Peptide synthetases are involved in the non-ribosomal synthesis of peptide antibiotics. Next to the operons encoding these enzymes, in almost all cases, are genes that encode proteins that have similarity to the type II fatty acid thioesterases of vertebrates. There are also modules within the peptide synthetases that also share this similarity. With respect to antibiotic production, thioesterases are required for the addition of the last amino acid to the peptide antibiotic, thereby forming a cyclic antibiotic. Thioesterases (non-integrated) have molecular masses of 25-29 kDa.','pfam00975','Thioesterase'),(1258009,'gnl|CDD|234501',_binary 'SH3 domain protein.  Members of this protein family have a signal peptide, a strongly conserved SH3 domain, a variable region, and then a C-terminal hydrophobic transmembrane alpha helix region.','TIGR04211','SH3_and_anchor'),(1258274,'gnl|CDD|148750',_binary 'Type III secretion protein YscO.  This family contains the bacterial type III secretion protein YscO, which is approximately 150 residues long. YscO has been shown to be required for high-level expression and secretion of the anti-host proteins V antigen and Yops in Yersinia pestis.','pfam07321','YscO'),(1258749,'gnl|CDD|233147',_binary 'sodium/hydrogen exchanger 3.  The Monovalent Cation:Proton Antiporter-1 (CPA1) Family (TC 2.A.36)The CPA1 family is a large family of proteins derived from Gram-positive and Gram-negative bacteria, blue green bacteria, yeast, plants and animals.Transporters from eukaryotes have been functionally characterized, and all of these catalyze Na+:H+ exchange. Their primary physiological functions may be in(1) cytoplasmic pH regulation, extruding the H+ generated during metabolism, and (2) salt tolerance (in plants), due to Na+ uptake into vacuoles.This model is specific for the eukaryotic members members of this family [Transport and binding proteins, Cations and iron carrying compounds].','TIGR00840','b_cpa1'),(1259083,'gnl|CDD|202581',_binary 'Alkylmercury lyase.  Alkylmercury lyase (EC:4.99.1.2) cleaves the carbon-mercury bond of organomercurials such as phenylmercuric acetate.','pfam03243','MerB'),(1259692,'gnl|CDD|205391',_binary 'Domain of unknown function (DUF4018).  This presumed domain is functionally uncharacterized. This domain family is found in bacteria, and is approximately 190 amino acids in length.','pfam13210','DUF4018'),(1260237,'gnl|CDD|257537',_binary 'AAA domain.  This family of domains contain a P-loop motif that is characteristic of the AAA superfamily. Many of the proteins in this family are conjugative transfer proteins. This family includes the PrrC protein that is thought to be the active component of the anticodon nuclease.','pfam13166','AAA_13'),(1260238,'gnl|CDD|255838',_binary 'Clusterin-associated protein-1.  This protein is conserved from worms to humans. The protein of 413 amino acids contains a central coiled-coil domain, possibly the region that binds to clusterin. Cluap1 expression is highest in the nucleus and gradually increases during late S to G2/M phases of the cell cycle and returns to the basal level in the G0/G1 phases. In addition, it is upregulated in colon cancer tissues compared to corresponding non-cancerous mucosa. It thus plays a crucial role in the life of the cell.','pfam10234','Cluap1'),(1261466,'gnl|CDD|239004',_binary 'SERine Proteinase INhibitors (serpins), prokaryotic subgroup. Little information about specific functions is available for this subgroup, most likely they are inhibitory members of the serpin superfamily. In general, serpins exhibit conformational polymorphism shifting from native to cleaved, latent, delta, or polymorphic forms. Many serpins, such as antitrypsin and antichymotrypsin, function as serine protease inhibitors.','cd02049','bacterial_SERPIN'),(1261486,'gnl|CDD|232933',_binary 'type I site-specific deoxyribonuclease, HsdR family.  This gene is part of the type I restriction and modification system which is composed of three polypeptides R (restriction endonuclease), M (modification) and S (specificity). This group of enzymes recognize specific short DNA sequences and have an absolute requirement for ATP (or dATP) and S-adenosyl-L-methionine. They also catalyse the reactions of EC 2.1.1.72 and EC 2.1.1.73, with similar site specificity.(J. Mol. Biol. 271 (3), 342-348 (1997)). Members of this family are assumed to differ from each other in DNA site specificity [DNA metabolism, Restriction/modification].','TIGR00348','hsdR'),(1262521,'gnl|CDD|258067',_binary 'Domain of unknown function (DUF4177).','pfam13783','DUF4177'),(1262608,'gnl|CDD|257800',_binary 'AAA domain.','pfam13476','AAA_23'),(1262643,'gnl|CDD|116129',_binary 'Protein of unknown function (DUF1523).','pfam07509','DUF1523'),(1263533,'gnl|CDD|203773',_binary 'PGAP1-like protein.  The sequences found in this family are similar to PGAP1. This is an endoplasmic reticulum membrane protein with a catalytic serine containing motif that is conserved in a number of lipases. PGAP1 functions as a GPI inositol-deacylase; this deacylation is important for the efficient transport of GPI-anchored proteins from the endoplasmic reticulum to the Golgi body.','pfam07819','PGAP1'),(1263553,'gnl|CDD|254521',_binary 'YcfA-like protein.  The viral, archaeal and bacterial proteins making up this family are similar to the YcfA protein expressed by Escherichia coli. Most of these proteins are hypothetical proteins of unknown function.','pfam07927','YcfA'),(1265021,'gnl|CDD|113256',_binary 'Protein of unknown function (DUF559).','pfam04480','DUF559'),(1265089,'gnl|CDD|233282',_binary 'protein disulfide isomerase, eukaryotic.  This model represents eukaryotic protein disulfide isomerases retained in the endoplasmic reticulum (ER) and closely related forms. Some members have been assigned alternative or additional functions such as prolyl 4-hydroxylase and dolichyl-diphosphooligosaccharide-protein glycotransferase. Members of this family have at least two protein-disulfide domains, each similar to thioredoxin but with the redox-active disulfide in the motif PWCGHCK, and an ER retention signal at the extreme C-terminus (KDEL, HDEL, and similar motifs).','TIGR01130','ER_PDI_fam'),(1265292,'gnl|CDD|249746',_binary 'Pyridoxal-phosphate dependent enzyme.  Members of this family are all pyridoxal-phosphate dependent enzymes. This family includes: serine dehydratase EC:4.2.1.13 P20132, threonine dehydratase EC:4.2.1.16, tryptophan synthase beta chain EC:4.2.1.20, threonine synthase EC:4.2.99.2, cysteine synthase EC:4.2.99.8 P11096, cystathionine beta-synthase EC:4.2.1.22, 1-aminocyclopropane-1-carboxylate deaminase EC:4.1.99.4.','pfam00291','PALP'),(1265322,'gnl|CDD|251797',_binary 'Molybdopterin guanine dinucleotide synthesis protein B.  This protein contains a P-loop.','pfam03205','MobB'),(1265508,'gnl|CDD|206603',_binary 'MafB19-like deaminase.  A member of the nucleic acid/nucleotide deaminase superfamily prototyped by Neisseria MafB19. Members of this family are present in a wide phyletic range of bacteria and are predicted to function as toxins in bacterial polymorphic toxin systems.','pfam14437','MafB19-deam'),(1266205,'gnl|CDD|257434',_binary 'Domain of unknown function (DUF3854).  A family of uncharacterized proteins found by clustering human gut metagenomic sequences. This domain is likely to be related to the Toprim domain.','pfam12965','DUF3854'),(1266223,'gnl|CDD|233046',_binary 'DNA repair helicase rad25.  All proteins in this family for which functions are known are DNA-DNA helicases used for the initiation of nucleotide excision repair and transacription as part of the TFIIH complex.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair].','TIGR00603','rad25'),(1266645,'gnl|CDD|255602',_binary 'Uncharacterized conserved protein (DUF2075).  This domain, found in various prokaryotic proteins (including putative ATP/GTP binding proteins), has no known function.','pfam09848','DUF2075'),(1266827,'gnl|CDD|251453',_binary 'Fic/DOC family.  This family consists of the Fic (filamentation induced by cAMP) protein and doc (death on curing). The Fic protein is involved in cell division and is suggested to be involved in the synthesis of PAB or folate, indicating that the Fic protein and cAMP are involved in a regulatory mechanism of cell division via folate metabolizm. This family contains a central conserved motif HPFXXGNG in most members. The exact molecular function of these proteins is uncertain. P1 lysogens of Escherichia coli carry the prophage as a stable low copy number plasmid. The frequency with which viable cells cured of prophage are produced is about 10(-5) per cell per generation. A significant part of this remarkable stability can be attributed to a plasmid-encoded mechanism that causes death of cells that have lost P1. In other words, the lysogenic cells appear to be addicted to the presence of the prophage. The plasmid withdrawal response depends on a gene named doc (death on curing) that is represented by this family. Doc induces a reversible growth arrest of Escherichia coli cells by targetting the protein synthesis machinery. Doc hosts the C-terminal domain of its antitoxin partner Phd (prevents host death) through fold complementation, a domain that is intrinsically disordered in solution but that folds into an alpha-helix on binding to Doc.This domain forms complexes with Phd antitoxins containing pfam02604.','pfam02661','Fic'),(1266828,'gnl|CDD|188156',_binary 'death-on-curing family protein.  The characterized member of this family is the death-on-curing (DOC) protein of phage P1. It is part of a two protein operon with prevents-host-death (phd) that forms an addiction module. DOC lacks homology to analogous addiction module post-segregational killing proteins involved in plasmid maintenance. These modules work as a combination of a long lived poison (e.g. this protein) and a more abundant but shorter lived antidote. Members of this family have a well-conserved central motif HxFx[ND][AG]NKR. A similar region, with K replaced by G, is found in the huntingtin interacting protein (HYPE) family [Unknown function, General].','TIGR01550','DOC_P1'),(1266857,'gnl|CDD|251933',_binary 'Peptidase C39 family.  Lantibiotic and non-lantibiotic bacteriocins are synthesized as precursor peptides containing N-terminal extensions (leader peptides) which are cleaved off during maturation. Most non-lantibiotics and also some lantibiotics have leader peptides of the so-called double-glycine type. These leader peptides share consensus sequences and also a common processing site with two conserved glycine residues in positions -1 and -2. The double- glycine-type leader peptides are unrelated to the N-terminal signal sequences which direct proteins across the cytoplasmic membrane via the sec pathway. Their processing sites are also different from typical signal peptidase cleavage sites, suggesting that a different processing enzyme is involved. Peptide bacteriocins are exported across the cytoplasmic membrane by a dedicated ATP-binding cassette (ABC) transporter. The ABC transporter is the maturation protease and its proteolytic domain resides in the N-terminal part of the protein. This peptidase domain is found in a wide range of ABC transporters, however the presumed catalytic cysteine and histidine are not conserved in all members of this family.','pfam03412','Peptidase_C39'),(1266861,'gnl|CDD|163508',_binary 'NHLM bacteriocin system ABC transporter, peptidase/ATP-binding protein.  This protein describes a multidomain ABC transporter subunit that is one of three protein families associated with some regularity with a distinctive family of putative bacteriocins. It includes a bacteriocin-processing peptidase domain at the N-terminus. Model TIGR03793 describes a conserved propeptide region for this bacteriocin family, unusual because it shows obvious homology a region of the enzyme nitrile hydratase up to the classic Gly-Gly cleavage motif. This family is therefore predicted to be a subunit of a bacteriocin processing and export system characteristic to this system that we designate NHLM, Nitrile Hydratase Leader Microcin [Transport and binding proteins, Amino acids, peptides and amines, Cellular processes, Biosynthesis of natural products].','TIGR03796','NHLM_micro_ABC1'),(1266959,'gnl|CDD|233949',_binary 'addiction module antidote protein, HigA family.  Members of this family form a distinct clade within the larger family HTH_3 of helix-turn-helix proteins, described by pfam01381. Members of this clade are strictly bacterial and nearly always shorter than 110 amino acids. This family includes the characterized member HigA, without which the killer protein HigB cannot be cloned. The hig (host inhibition of growth) system is noted to be unusual in that killer protein is uncoded by the upstream member of the gene pair [Regulatory functions, DNA interactions, Regulatory functions, Protein interactions, Mobile and extrachromosomal element functions, Other].','TIGR02607','antidote_HigA'),(1266992,'gnl|CDD|252536',_binary 'Protein of unknown function, DUF488.  This family includes several proteins of uncharacterized function.','pfam04343','DUF488'),(1267203,'gnl|CDD|254437',_binary 'Protein of unknown function (DUF1643).  The members of this family are all sequences found within hypothetical proteins expressed by various bacterial species. The region concerned is approximately 150 residues long.','pfam07799','DUF1643'),(1267285,'gnl|CDD|252412',_binary 'Domain of unknown function (DUF397).  The function of this family is unknown. It has been suggested that some members of this family are regulators of transcription. In particular, it is thought that this may regulator of antibiotic production in Streptomyces coelicolor.','pfam04149','DUF397'),(1267301,'gnl|CDD|257345',_binary 'Methyltransferase domain.  Protein in this family function as methyltransferases.','pfam12847','Methyltransf_18'),(1267306,'gnl|CDD|233880',_binary 'precorrin-6Y C5,15-methyltransferase (decarboxylating), CbiT subunit.  This model recognizes the CbiT methylase which is responsible, in part (along with CbiE), for methylating precorrin-6y (or cobalt-precorrin-6y) at both the 5 and 15 positions as well as the concomitant decarbozylation at C-12. In many organisms, this protein is fused to the CbiE subunit. The fused protein, when found in organisms catalyzing the oxidative version of the cobalamin biosynthesis pathway, is called CobL [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin].','TIGR02469','CbiT'),(1267325,'gnl|CDD|148013',_binary 'Phage minor capsid protein 2.  Family of related phage minor capsid proteins.','pfam06152','Phage_min_cap2'),(1267415,'gnl|CDD|257336',_binary 'Integrase.  This is a family of DNA-binding prophage integrases found in Proteobacteria.','pfam12835','Integrase_1'),(1269316,'gnl|CDD|257394',_binary 'Antitoxin of toxin-antitoxin stability system N-terminal.  This domain appears to be the N-terminus of the RelB antitoxin of toxin-antitoxin stability system or prevent-host death system. Together RelE toxin and the RelB antitoxin form a non-toxic complex. Although toxin-antitoxin gene cassettes were first found in plasmids, it is clear that these loci are abundant in free-living prokaryotes, including many pathogenic bacteria, and these toxin-antitoxin loci provide a control mechanism that helps free-living prokaryotes cope with nutritional stress.','pfam12910','RelB_N'),(1269317,'gnl|CDD|252068',_binary 'Domain of unknown function (DUF303).  Distribution of this domain seems limited to prokaryotes and viruses.','pfam03629','DUF303'),(1269322,'gnl|CDD|253485',_binary 'Phage derived protein Gp49-like (DUF891).  This family consists of hypothetical bacterial proteins of unknown function as well as phage Gp49 proteins.','pfam05973','Gp49'),(1270320,'gnl|CDD|253235',_binary 'Phage head-tail joining protein.','pfam05521','Phage_H_T_join'),(1270322,'gnl|CDD|256434',_binary 'Protein of unknown function (DUF3168).  This family of proteins has no known function but is likely to be a component of bacteriophage.','pfam11367','DUF3168'),(1270323,'gnl|CDD|253612',_binary 'Phage major tail protein 2.  characterized members are major tail proteins from various phage, including lactococcal temperate bacteriophage TP901-1.','pfam06199','Phage_tail_2'),(1272580,'gnl|CDD|255022',_binary 'dUTPase.  2-Deoxyuridine 5-triphosphate nucleotidohydrolase (dUTPase) catalyses the hydrolysis of dUTP to dUMP and pyrophosphate (EC:3.6.1.23). Members of this family have a novel all-alpha fold and are unrelated to the all-beta fold found in dUTPases of the majority of organisms. This family contains both dUTPase homologues of dUTPase including dCTPase of phage T4.','pfam08761','dUTPase_2'),(1274239,'gnl|CDD|253534',_binary 'Domain of unknown function (DUF927).  Family of bacterial proteins of unknown function. The C-terminal half of this family contains a P-loop motif.','pfam06048','DUF927'),(1274241,'gnl|CDD|233192',_binary 'adenylosuccinate lyase.  This family consists of adenylosuccinate lyase, the enzyme that catalyzes step 8 in the purine biosynthesis pathway for de novo synthesis of IMP and also the final reaction in the two-step sequence from IMP to AMP [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis].','TIGR00928','purB'),(1274266,'gnl|CDD|249677',_binary 'Lyase.','pfam00206','Lyase_1'),(1274267,'gnl|CDD|233863',_binary '3-carboxy-cis,cis-muconate cycloisomerase.  Members of this family are 3-carboxy-cis,cis-muconate cycloisomerase, the enzyme the catalyzes the second step in the protocatechuate degradation to beta-ketoadipate and then to succinyl-CoA and acetyl-CoA. 4-hydroxybenzoate, 3-hydroxybenzoate, and vanillate all can be converted in one step to protocatechuate. All members of the seed alignment for this model were chosen from within protocatechuate degradation operons of at least three genes of the pathway, from genomes with the complete pathway through beta-ketoadipate [Energy metabolism, Other].','TIGR02426','protocat_pcaB'),(1275059,'gnl|CDD|255233',_binary 'Alpha-L-arabinofuranosidase B, catalytic.  Members of this family, which are present in fungal alpha-L-arabinofuranosidase B, adopt a beta-sandwich fold similar to that of Concanavalin A-like lectins/glucanase. The beta-sandwich fold consists of two anti-parallel beta-sheets with seven and and six strands, respectively. In addition, there are four helices outside of the beta-strands. The beta-sandwich strands are closely packed and curved with a jelly roll topology, creating a small catalytic pocket. The domain catalyses the hydrolysis of alpha-1,2-, alpha-1,3- and alpha-1,5-L-arabinofuranosidic bonds in L-arabinose-containing hemicelluloses such as arabinoxylan and L-arabinan.','pfam09206','ArabFuran-catal'),(1277977,'gnl|CDD|234078',_binary 'RNA polymerase sigma-70 factor, sigma-B/F/G subfamily.  This group of similar sigma-70 factors includes clades found in Bacilli (including the sporulation factors SigF:TIGR02885 and SigG:TIGR02850 as well as SigB:TIGR02941), and the high GC gram positive bacteria (Actinobacteria) where a variable number of them are found depending on the lineage.','TIGR02980','SigBFG'),(1280733,'gnl|CDD|257897',_binary 'Histidine kinase-like ATPase domain.','pfam13581','HATPase_c_2'),(1280736,'gnl|CDD|252494',_binary 'ASCH domain.  The ASCH domain adopts a beta-barrel fold similar to the pfam01472 domain. It is thought to function as an RNA-binding domain during coactivation, RNA-processing and possibly during prokaryotic translation regulation.','pfam04266','ASCH'),(1280740,'gnl|CDD|258015',_binary 'GNAT acetyltransferase 2.  This domain has N-acetyltransferase activity. It has a GCN5-related N-acetyltransferase (GNAT) fold.','pfam13718','GNAT_acetyltr_2'),(1280747,'gnl|CDD|252371',_binary 'Rad52/22 family double-strand break repair protein.  The DNA single-strand annealing proteins (SSAPs), such as RecT, Red-beta, ERF and Rad52, function in RecA-dependent and RecA-independent DNA recombination pathways. This family includes proteins related to Rad52. These proteins contain two helix-hairpin-helix motifs.','pfam04098','Rad52_Rad22'),(1282255,'gnl|CDD|250568',_binary 'Glycosyl hydrolase family 46.  This family are chitosanase enzymes.','pfam01374','Glyco_hydro_46'),(1282510,'gnl|CDD|251262',_binary 'N-6 DNA Methylase.  Restriction-modification (R-M) systems protect a bacterial cell against invasion of foreign DNA by endonucleolytic cleavage of DNA that lacks a site specific modification. The R-M system is a complex containing three polypeptides: M (this family), S (pfam01420), and R. This family consists of N-6 adenine-specific DNA methylase EC:2.1.1.72 from Type I and Type IC restriction systems. These methylases have the same sequence specificity as their corresponding restriction enzymes.','pfam02384','N6_Mtase'),(1282515,'gnl|CDD|205766',_binary 'Type I restriction enzyme R protein N terminus (HSDR_N).  This family consists of a number of N terminal regions found in type I restriction enzyme R (HSDR) proteins. Restriction and modification (R/M) systems are found in a wide variety of prokaryotes and are thought to protect the host bacterium from the uptake of foreign DNA. Type I restriction and modification systems are encoded by three genes: hsdR, hsdM, and hsdS. The three polypeptides, HsdR, HsdM, and HsdS, often assemble to give an enzyme (R2M2S1) that modifies hemimethylated DNA and restricts unmethylated DNA.','pfam13588','HSDR_N_2'),(1282631,'gnl|CDD|233115',_binary 'dimethyladenosine transferase.  In both E. coli and Saccharomyces cerevisiae, this protein is responsible for the dimethylation of two adjacent adenosine residues in a conserved hairpin of 16S rRNA in bacteria, 18S rRNA in eukaryotes. This adjacent dimethylation is the only rRNA modification shared by bacteria and eukaryotes. A single member of this family is present in each of the first 20 completed microbial genomes. This protein is essential in yeast, but not in E. coli, where its deletion leads to resistance to the antibiotic kasugamycin. Alternate name: S-adenosylmethionine--6-N\',N\'-adenosyl (rRNA) dimethyltransferase [Protein synthesis, tRNA and rRNA base modification].','TIGR00755','ksgA'),(1282917,'gnl|CDD|233679',_binary 'Fe-S cluster assembly protein NifU.  Three different but partially homologous Fe-S cluster assembly systems have been described: Isc, Suf, and Nif. The latter is associated with donation of an Fe-S cluster to nitrogenase in a number of nitrogen-fixing species. NifU, described here, consists of an N-terminal domain (pfam01592) and a C-terminal domain (pfam01106). Homologs with an equivalent domain archictecture from Helicobacter and Campylobacter, however, are excluded from this model by a high trusted cutoff. The model, therefore, is specific for NifU involved in nitrogenase maturation. The related model TIGR01999 homologous to the N-terminus of this model describes IscU from the Isc system as in E. coli, Saccharomyces cerevisiae, and Homo sapiens [Biosynthesis of cofactors, prosthetic groups, and carriers, Other, Central intermediary metabolism, Nitrogen fixation].','TIGR02000','NifU_proper'),(1284925,'gnl|CDD|255658',_binary 'Uncharacterized protein conserved in bacteria (DUF2188).  This domain, found in various hypothetical bacterial proteins, has no known function.','pfam09954','DUF2188'),(1286090,'gnl|CDD|256396',_binary 'Protein of unknown function (DUF3090).  This family of proteins with unknown function appears to be restricted to Actinobacteria.','pfam11290','DUF3090'),(1287162,'gnl|CDD|143375',_binary 'Catalytic domain of the Serine/Threonine Kinase, PFTAIRE-2 kinase.  Serine/Threonine Kinases (STKs), PFTAIRE-2 subfamily, catalytic (c) domain. STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The PFTAIRE-2 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase. PFTAIRE-2 shares sequence similarity with Cyclin-Dependent Kinases (CDKs), which belong to a large family of STKs that are regulated by their cognate cyclins. Together, CDKs and cyclins are involved in the control of cell-cycle progression, transcription, and neuronal function. PFTAIRE-2 is also referred to as ALS2CR7 (amyotrophic lateral sclerosis 2 (juvenile) chromosome region candidate 7). It may be associated with amyotrophic lateral sclerosis 2 (ALS2), an autosomal recessive form of juvenile ALS. The function of PFTAIRE-2 is not yet known.','cd07870','STKc_PFTAIRE2'),(1287164,'gnl|CDD|173739',_binary 'Catalytic domain of Cyclin-Dependent protein Kinase 4 and 6-like Serine/Threonine Kinases.  Serine/Threonine Kinases (STKs), Cyclin-dependent protein kinase 4 (CDK4) and CDK6-like subfamily, catalytic (c) domain. STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. The CDK4/6-like subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase. CDKs belong to a large family of STKs that are regulated by their cognate cyclins. Together, they are involved in the control of cell-cycle progression, transcription, and neuronal function. CDK4 and CDK6 partner with D-type cyclins to regulate the early G1 phase of the cell cycle. They are the first kinase activated by mitogenic signals to release cells from the G0 arrested state. CDK4 and CDK6 are both expressed ubiquitously, associate with all three D cyclins (D1, D2 and D3), and phosphorylate the retinoblastoma (pRb) protein. They are also regulated by the INK4 family of inhibitors which associate with either the CDK alone or the CDK/cyclin complex. CDK4 and CDK6 show differences in subcellular localization, sensitivity to some inhibitors, timing in activation, tumor selectivity, and possibly substrate profiles. Although CDK4 and CDK6 seem to show some redundancy, they also have discrete, nonoverlapping functions. CDK6 plays an important role in cell differentiation.','cd07838','STKc_CDK4_6_like'),(1287857,'gnl|CDD|250592',_binary 'Ribbon-helix-helix protein, copG family.  The structure of this protein repressor, which is the shortest reported to date and the first isolated from a plasmid, has a homodimeric ribbon-helix-helix arrangement. The helix-turn-helix-like structure is involved in dimerization and not DNA binding as might have been expected.','pfam01402','RHH_1'),(1288599,'gnl|CDD|250347',_binary 'Hint module.  This is an alignment of the Hint module in the Hedgehog proteins. It does not include any Inteins which also possess the Hint module.','pfam01079','Hint'),(1288880,'gnl|CDD|257658',_binary 'HD domain.  HD domains are metal dependent phosphohydrolases.','pfam13328','HD_4'),(1288893,'gnl|CDD|251234',_binary 'TerD domain.  The TerD domain is found in TerD family proteins that include the paralogous TerD, TerA, TerE, TerF and TerZ proteins. It is found in a stress response operon with TerB and TerC. TerD has a maximum of two calcium binding sites {2] depending on the conservation of aspartates {2]. It has various fusions to nuclease domains, RNA binding domains, ubiquitin related domains, and metal binding domains. The ter gene products lie at the center of membrane-linked metal recognition complexes with regulatory ramifications encompassing phosphorylation- dependent signal transduction, RNA-dependent regulation, biosynthesis of nucleoside-like metabolites and DNA processing linked to novel pathways.','pfam02342','TerD'),(1288906,'gnl|CDD|250772',_binary 'NAD-dependent DNA ligase adenylation domain.  DNA ligases catalyse the crucial step of joining the breaks in duplex DNA during DNA replication, repair and recombination, utilizing either ATP or NAD(+) as a cofactor. This domain is the catalytic adenylation domain. The NAD+ group is covalently attached to this domain at the lysine in the KXDG motif of this domain. This enzyme- adenylate intermediate is an important feature of the proposed catalytic mechanism.','pfam01653','DNA_ligase_aden'),(1288908,'gnl|CDD|233030',_binary 'DNA ligase, NAD-dependent.  All proteins in this family with known functions are NAD-dependent DNA ligases. Functions of these proteins include DNA repair, DNA replication, and DNA recombination. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). The member of this family from Treponema pallidum differs in having three rather than just one copy of the BRCT (BRCA1 C Terminus) domain (pfam00533) at the C-terminus. It is included in the seed [DNA metabolism, DNA replication, recombination, and repair].','TIGR00575','dnlj'),(1288950,'gnl|CDD|250779',_binary 'Macro domain.  This domain is an ADP-ribose binding module. It is found in a number of otherwise unrelated proteins. It is found at the C-terminus of the macro-H2A histone protein. This domain is found in the non-structural proteins of several types of ssRNA viruses such as NSP3 from alphaviruses. This domain is also found on its own in a family of proteins from bacteria, archaebacteria, and eukaryotes.','pfam01661','Macro'),(1288959,'gnl|CDD|252144',_binary 'ADP-ribosylglycohydrolase.  This family includes enzymes that ADP-ribosylations, for example ADP-ribosylarginine hydrolase EC:3.2.2.19 cleaves ADP-ribose-L-arginine. The family also includes dinitrogenase reductase activating glycohydrolase. Most surprisingly the family also includes jellyfish crystallins, these proteins appear to have lost the presumed active site residues.','pfam03747','ADP_ribosyl_GH'),(1288967,'gnl|CDD|233415',_binary 'methyltransferase, FkbM family.  Members of this family are characterized by two well-conserved short regions separated by a variable in both sequence and length. The first of the two regions is found in a large number of proteins outside this subfamily, a number of which have been characterized as methyltransferases. One member of the present family, FkbM, was shown to be required for a specific methylation in the biosynthesis of the immunosuppressant FK506 in Streptomyces strain MA6548.','TIGR01444','fkbM_fam'),(1288968,'gnl|CDD|252973',_binary 'Methyltransferase FkbM domain.  This family has members from bacteria to human, and appears to be a methyltransferase.','pfam05050','Methyltransf_21'),(1288980,'gnl|CDD|252464',_binary 'Putative neutral zinc metallopeptidase.  Members of this family have a predicted zinc binding motif characteristic of neutral zinc metallopeptidases.','pfam04228','Zn_peptidase'),(1289345,'gnl|CDD|152081',_binary 'PD-(D/E)XK endonuclease.  This family of endonucleases includes a group I intron-encoded endonuclease. This family belongs to the PD-(D/E)XK superfamily.','pfam11645','PDDEXK_5'),(1289942,'gnl|CDD|250980',_binary 'Domain of unknown function DUF87.  The function of this prokaryotic domain is unknown. It contains several conserved aspartates and histidines that could be metal ligands.','pfam01935','DUF87'),(1290010,'gnl|CDD|249747',_binary 'NUDIX domain.','pfam00293','NUDIX'),(1290499,'gnl|CDD|254238',_binary 'Rho termination factor, N-terminal domain.  The Rho termination factor disengages newly transcribed RNA from its DNA template at certain, specific transcripts. It it thought that two copies of Rho bind to RNA and that Rho functions as a hexamer of protomers. This domain is found to the N-terminus of the RNA binding domain (pfam07497).','pfam07498','Rho_N'),(1292938,'gnl|CDD|251347',_binary 'Histidine kinase-, DNA gyrase B-, and HSP90-like ATPase.  This family represents the structurally related ATPase domains of histidine kinase, DNA gyrase B and HSP90.','pfam02518','HATPase_c'),(1293217,'gnl|CDD|258781',_binary 'Domain of unknown function (DUF4455).  This domain family is found in bacteria and eukaryotes, and is approximately 480 amino acids in length. There are two completely conserved residues (W and P) that may be functionally important.','pfam14643','DUF4455'),(1293345,'gnl|CDD|251319',_binary 'Met-10+ like-protein.  The methionine-10 mutant allele of N. crassa codes for a protein of unknown function. However, homologous proteins have been found in yeast suggesting this protein may be involved in methionine biosynthesis, transport and/or utilisation.','pfam02475','Met_10'),(1293520,'gnl|CDD|200948',_binary 'Intermediate filament protein.','pfam00038','Filament'),(1293522,'gnl|CDD|255286',_binary 'Domain of unknown function (DUF1978).  Members of this family are found in various hypothetical proteins produced by the bacterium Chlamydia pneumoniae. Their exact function has not, as yet, been identified.','pfam09321','DUF1978'),(1293533,'gnl|CDD|255019',_binary 'CotH protein.  Members of this family include the spore coat protein H (cotH).','pfam08757','CotH'),(1297146,'gnl|CDD|233518',_binary 'phage minor structural protein, N-terminal region.  This model represents the conserved N-terminal region, typically from about residue 25 to about residue 350, of a family of uncharacterized phage proteins 500 to 1700 residues in length [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01665','put_anti_recept'),(1297147,'gnl|CDD|253819',_binary 'Prophage endopeptidase tail.  This family is of prophage tail proteins that are probably acting as endopeptidases.','pfam06605','Prophage_tail'),(1297154,'gnl|CDD|250070',_binary 'Glycosyl hydrolases family 18.','pfam00704','Glyco_hydro_18'),(1297199,'gnl|CDD|257604',_binary 'zinc-ribbon domain.  This family consists of a single zinc ribbon domain, ie half of a pair as in family DZR. pfam12773.','pfam13248','zf-ribbon_3'),(1297200,'gnl|CDD|257597',_binary 'zinc-ribbon domain.  This family consists of a single zinc ribbon domain, ie half of a pair as in family DZR. pfam12773.','pfam13240','zinc_ribbon_2'),(1297392,'gnl|CDD|116875',_binary 'Peptidase M15.','pfam08291','Peptidase_M15_3'),(1297395,'gnl|CDD|253473',_binary 'Bacterial protein of unknown function (DUF882).  This family consists of a series of hypothetical bacterial proteins of unknown function.','pfam05951','Peptidase_M15_2'),(1297396,'gnl|CDD|251373',_binary 'D-alanyl-D-alanine carboxypeptidase.','pfam02557','VanY'),(1297478,'gnl|CDD|249664',_binary 'Dihydrofolate reductase.','pfam00186','DHFR_1'),(1297493,'gnl|CDD|253375',_binary 'Bacteriophage Lambda NinG protein.  NinG or Rap is involved in recombination. Rap (recombination adept with plasmid) increases lambda-by-plasmid recombination catalysed by Escherichia coli\'s RecBCD pathway.','pfam05766','NinG'),(1298331,'gnl|CDD|253446',_binary 'Siphovirus protein of unknown function (DUF859).  This family consists of several uncharacterized proteins from the Siphoviruses as well as one bacterial sequence. Some of the members of this family are described as putative minor structural proteins.','pfam05895','DUF859'),(1298332,'gnl|CDD|258160',_binary 'RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain).  The RRM motif is probably diagnostic of an RNA binding protein. RRMs are found in a variety of RNA binding proteins, including various hnRNP proteins, proteins implicated in regulation of alternative splicing, and protein components of snRNPs. The motif also appears in a few single stranded DNA binding proteins.','pfam13893','RRM_5'),(1298387,'gnl|CDD|251771',_binary 'Uracil DNA glycosylase superfamily.','pfam03167','UDG'),(1298429,'gnl|CDD|313664',_binary 'Cenp-F N-terminal domain.  Mitosin or centromere-associated protein-F (Cenp-F) is found bound across the centromere as one of the proteins of the outer layer of the kinetochore. Most of the kinetochore/centromere functions appear to depend upon binding of the C-terminal par to f the molecule, whereas the N-terminal part, here, may be a cytoplasmic player in controlling the function of microtubules and dynein.','pfam10481','CENP-F_N'),(1301309,'gnl|CDD|249687',_binary 'Bacterial DNA-binding protein.','pfam00216','Bac_DNA_binding'),(1301322,'gnl|CDD|254029',_binary 'Protein of unknown function (DUF1320).  This family consists of both hypothetical bacterial and phage proteins of around 145 residues in length. The function of this family is unknown.','pfam07030','DUF1320'),(1301323,'gnl|CDD|254054',_binary 'Intracellular growth attenuator protein IgaA.  This family consists of several bacterial intracellular growth attenuator (IgaA) proteins. IgaA is involved in negative control of bacterial proliferation within fibroblasts. IgaA is homologous to the Escherichia coli YrfF and P. mirabilis UmoB proteins. Whereas the biological function of YrfF is currently unknown, UmoB has been shown elsewhere to act as a positive regulator of FlhDC, the master regulator of flagella and swarming. FlhDC has been shown to repress cell division during P. mirabilis swarming, suggesting that UmoB could repress cell division via FlhDC. This biological function, if maintained in S. enterica, could sustain a putative negative control of cell division and growth exerted by IgaA in intracellular bacteria.','pfam07095','IgaA'),(1301391,'gnl|CDD|109261',_binary 'Bacterial regulatory proteins, luxR family.','pfam00196','GerE'),(1301667,'gnl|CDD|201202',_binary 'Bacterial regulatory proteins, gntR family.  This family of regulatory proteins consists of the N-terminal HTH region of GntR-like bacterial transcription factors. At the C-terminus there is usually an effector-binding/oligomerization domain. The GntR-like proteins include the following sub-families: MocR, YtrR, FadR, AraR, HutC and PlmA, DevA, DasR. Many of these proteins have been shown experimentally to be autoregulatory, enabling the prediction of operator sites and the discovery of cis/trans relationships. The DasR regulator has been shown to be a global regulator of primary metabolizm and development in Streptomyces coelicolor.','pfam00392','GntR'),(1301682,'gnl|CDD|233852',_binary 'trehalose operon repressor, B. subtilis-type.  This family consists of repressors of the GntR family typically associated with trehalose utilization operons. Trehalose is imported as trehalose-6-phosphate and then hydrolyzed by alpha,alpha-phosphotrehalase to glucose and glucose-6-P. This family includes repressors mostly from Gram-positive lineages and does not include the TreR from E. coli [Regulatory functions, DNA interactions].','TIGR02404','trehalos_R_Bsub'),(1302485,'gnl|CDD|197317',_binary 'Exonuclease-Endonuclease-Phosphatase domain; uncharacterized family 1.  This family of uncharacterized proteins belongs to a superfamily that includes the catalytic domain (exonuclease/endonuclease/phosphatase, EEP, domain) of a diverse set of proteins including the ExoIII family of apurinic/apyrimidinic (AP) endonucleases, inositol polyphosphate 5-phosphatases (INPP5), neutral sphingomyelinases (nSMases), deadenylases (such as the vertebrate circadian-clock regulated nocturnin), bacterial cytolethal distending toxin B (CdtB), deoxyribonuclease 1 (DNase1), the endonuclease domain of the non-LTR retrotransposon LINE-1, and related domains. These diverse enzymes share a common catalytic mechanism of cleaving phosphodiester bonds. Their substrates range from nucleic acids to phospholipids and perhaps, proteins.','cd09083','EEP-1'),(1302522,'gnl|CDD|255575',_binary 'Essential protein Yae1, N terminal.  Members of this family are found in the N terminal region of the essential protein Yae1. Their exact function has not, as yet, been determined. The family DUF1715, pfam08215 has now been merged into this family.','pfam09811','Yae1_N'),(1302582,'gnl|CDD|213756',_binary 'RNA polymerase sigma-70 factor, Bacteroides expansion family 1.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are found primarily in the genus Bacteroides. This family appears to have resulted from a lineage-specific expansion as B. thetaiotaomicron VPI-5482, Bacteroides forsythus ATCC 43037, Bacteroides fragilis YCH46 and Bacteroides fragilis NCTC 9343 contain 25, 12, 24 and 23 members, respectively. There are currentlyonly two known members of this family outside of the Bacteroides, in Rhodopseudomonas and Bradyrhizobium.','TIGR02985','Sig70_bacteroi1'),(1302585,'gnl|CDD|227207',_binary 'Cysteine protease [Posttranslational modification, protein turnover, chaperones].','COG4870','COG4870'),(1302593,'gnl|CDD|251566',_binary 'Carbohydrate binding domain.  This short domain is found in many different glycosyl hydrolase enzymes and is presumed to have a carbohydrate binding function. The domain has six aromatic groups that may be important for binding.','pfam02839','CBM_5_12'),(1302598,'gnl|CDD|250625',_binary 'Apolipoprotein A1/A4/E domain.  These proteins contain several 22 residue repeats which form a pair of alpha helices. This family includes: Apolipoprotein A-I. Apolipoprotein A-IV, and Apolipoprotein E.','pfam01442','Apolipoprotein'),(1302635,'gnl|CDD|256231',_binary 'Protein of unknown function (DUF2815).  This is a phage related family of proteins with unknown function.','pfam10991','DUF2815'),(1302641,'gnl|CDD|253123',_binary 'Virulence-associated protein E.  This family contains several bacterial virulence-associated protein E like proteins. These proteins contain a P-loop motif.','pfam05272','VirE'),(1459116,'gnl|CDD|223727',_binary '2-polyprenyl-6-methoxyphenol hydroxylase and related FAD-dependent oxidoreductases [Coenzyme metabolism / Energy production and conversion].','COG0654','UbiH'),(1459117,'gnl|CDD|272992',_binary 'flavoprotein, HI0933 family.  The model when searched with a partial length search brings in proteins with a dinucleotide-binding motif (Rossman fold) over the initial 40 residues of the model, including oxidoreductases and dehydrogenases. Partially characterized members include an FAD-binding protein from Bacillus cereus and flavoprotein HI0933 from Haemophilus influenzae. [Unknown function, Enzymes of unknown specificity].','TIGR00275','TIGR00275'),(1459118,'gnl|CDD|225915',_binary 'Predicted NAD/FAD-dependent oxidoreductase [General function prediction only].','COG3380','COG3380'),(1459119,'gnl|CDD|168629',_binary '2-octaprenyl-6-methoxyphenyl hydroxylase; Validated.','PRK06617','PRK06617'),(1459120,'gnl|CDD|281484',_binary 'HI0933-like protein.','pfam03486','HI0933_like'),(1459121,'gnl|CDD|224992',_binary 'Predicted flavoproteins [General function prediction only].','COG2081','COG2081'),(1537919,'gnl|CDD|316076',_binary 'Helix-turn-helix domain.  This helix-turn-helix domain is often found in transposases and is likely to be DNA-binding.','pfam13518','HTH_28'),(1537921,'gnl|CDD|320698',_binary 'Peptidase M48 subfamily B HtpX-like membrane-bound metallopeptidase.  This HtpX family of peptidase M48 subfamily B includes uncharacterized HtpX homologs and consists of proteins smaller than Ste24p, with homology restricted to the C-terminal half of Ste24p. HtpX expression is controlled by the Cpx stress response system, which senses abnormal membrane proteins. HtpX participates in the proteolytic quality control of these misfolded proteins by undergoing self-degradation and collaborating with FtsH, a membrane-bound and ATP-dependent protease, to eliminate them. HtpX, a zinc metalloprotease with an active site motif HEXXH, has an FtsH-like topology, and is capable of introducing endoproteolytic cleavages into SecY (also an FtsH substrate). However, HtpX does not have an ATPase activity and will only act against cytoplasmic regions of a target membrane protein. Thus, HtpX and FtsH have overlapping and/or complementary functions, which are especially important at high temperature; in E. coli and Xylella fastidiosa, HtpX is heat-inducible, while in Streptococcus gordonii it is not.','cd07339','M48B_HtpX_like'),(1538715,'gnl|CDD|180381',_binary 'flagellar accessory protein FlaH; Validated.','PRK06067','PRK06067'),(1539393,'gnl|CDD|223584',_binary 'Thiamine kinase and related kinases [Coenzyme transport and metabolism].','COG0510','ycfN'),(1539394,'gnl|CDD|270870',_binary 'Catalytic domain of the Protein Kinase superfamily.  The PK superfamily contains the large family of typical PKs that includes serine/threonine kinases (STKs), protein tyrosine kinases (PTKs), and dual-specificity PKs that phosphorylate both serine/threonine and tyrosine residues of target proteins, as well as pseudokinases that lack crucial residues for catalytic activity and/or ATP binding. It also includes phosphoinositide 3-kinases (PI3Ks), aminoglycoside 3\'-phosphotransferases (APHs), choline kinase (ChoK), Actin-Fragmin Kinase (AFK), and the atypical RIO and Abc1p-like protein kinases. These proteins catalyze the transfer of the gamma-phosphoryl group from ATP to their target substrates; these include serine/threonine/tyrosine residues in proteins for typical or atypical PKs, the 3-hydroxyl of the inositol ring of D-myo-phosphatidylinositol (PtdIns) or its derivatives for PI3Ks, the 4-hydroxyl of PtdIns for PI4Ks, and other small molecule substrates for APH/ChoK and similar proteins such as aminoglycosides, macrolides, choline, ethanolamine, and homoserine.','cd13968','PKc_like'),(1539559,'gnl|CDD|273261',_binary 'alpha-L-glutamate ligase, RimK family.  This family, related to bacterial glutathione synthetases, contains at least three different alpha-L-glutamate ligases. One is RimK, as in E. coli, which adds additional Glu residues to the native Glu-Glu C-terminus of ribosomal protein S6, but not to Lys-Glu mutants. Most species with a member of this subfamily lack an S6 homolog ending in Glu-Glu, however. Members in Methanococcus jannaschii act instead as a tetrahydromethanopterin:alpha-l-glutamate ligase (MJ0620) and a gamma-F420-2:alpha-l-glutamate ligase (MJ1001).','TIGR00768','Tetrahydromethanopterin:a'),(1540010,'gnl|CDD|319200',_binary 'OSK domain.  This entry represents the OSK domain defined by Jeske and colleagues. The domain is related to SGNH hydrolases but lacks the active site residues. The domain binds to RNA.','pfam17182','OSK'),(1540326,'gnl|CDD|183772',_binary '1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase; Provisional.','PRK12823','benD'),(1541880,'gnl|CDD|275083',_binary 'GxxExxY protein.  Members of this protein family average about 130 residues in length and include an almost perfectly conserved motif GxxExxY. Members occur in a wide range of prokaryotes, including Proteobacteria, Perrucomicrobia, Cyanobacteria, Bacteriodetes, Archaea, etc.','TIGR04256','conserved_hypothetical_pr'),(1541882,'gnl|CDD|315931',_binary 'PD-(D/E)XK nuclease superfamily.  Members of this family belong to the PD-(D/E)XK nuclease superfamily.','pfam13366','PDDEXK_3'),(1541900,'gnl|CDD|315480',_binary 'Bax inhibitor 1 like.  The Bax-inhibitor-1 region of the receptor molecules is conserved from bacteria to humans.','pfam12811','BaxI_1'),(1542032,'gnl|CDD|273147',_binary 'DNA ligase I, ATP-dependent (dnl1).  All proteins in this family with known functions are ATP-dependent DNA ligases. Functions include DNA repair, DNA replication, and DNA recombination (or any process requiring ligation of two single-stranded DNA sections). This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair].','TIGR00574','DNA_ligase'),(1542271,'gnl|CDD|224055',_binary 'ABC-type multidrug transport system, ATPase and permease components [Defense mechanisms].','COG1132','MdlB'),(1542272,'gnl|CDD|224048',_binary 'ATPase components of various ABC-type transport systems, contain duplicated ATPase [General function prediction only].','COG1123','COG1123'),(1542633,'gnl|CDD|133399',_binary 'First Helix-Turn-Helix DNA binding domain of the regulatory protein TioE.  Putative helix-turn-helix (HTH) regulatory protein, TioE, and related proteins. TioE is part of the thiocoraline gene cluster, which is involved in the biosynthesis of the antitumor thiocoraline from the marine actinomycete, Micromonospora. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Proteins in this family are unique within the MerR superfamily in that they are composed of just two adjacent MerR-like N-terminal domains; this CD contains the N-terminal or first repeat (rpt1) of these tandem MerR-like domain proteins.','cd04772','HTH_TioE_rpt1'),(1543291,'gnl|CDD|214720',_binary 'Filamin-type immunoglobulin domains.  These form a rod-like structure in the actin-binding cytoskeleton protein, filamin. The C-terminal repeats of filamin bind beta1-integrin (CD29).','smart00557','IG_FLMN'),(1543803,'gnl|CDD|223822',_binary 'Glycyl-tRNA synthetase, beta subunit [Translation, ribosomal structure and biogenesis].','COG0751','GlyS'),(1543841,'gnl|CDD|214648',_binary 'zinc finger binding to DNA consensus sequence [AT]GATA[AG].','smart00401','ZnF_GATA'),(1544124,'gnl|CDD|212481',_binary 'Bacillus subtilis YxlS-like, C-terminal domain.  YxlS is a Bacillus subtilis gene of unknown function with two domains that each have an alpha/beta fold. The N-terminal domain is composed of two alpha-helices and a three-stranded beta-sheet, while the C-terminal domain is composed of one alpha-helix and a five-stranded beta-sheet. This CD represents the C-terminal domain which has a fold similar to the Sm fold of proteins like Sm-D3.','cd01734','YlxS_C'),(1544244,'gnl|CDD|308992',_binary 'Bacterial PH domain.  Domain found in uncharacterized family of membrane proteins. 1-3 copies found in each protein, with each copy flanked by transmembrane helices. Members of this family have a PH domain like structure.','pfam03703','bPH_2'),(1544717,'gnl|CDD|270896',_binary 'Catalytic domain of Fungal Halotolerance protein 4-like Serine/Threonine kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. This subfamily is composed of HAL4, Saccharomyces cerevisiae Ptk2/Stk2, and similar fungal proteins. Proteins in this subfamily are involved in regulating ion transporters. In budding and fission yeast, HAL4 promotes potassium ion uptake, which increases cellular resistance to other cations such as sodium, lithium, and calcium ions. HAL4 stabilizes the major high-affinity K+ transporter Trk1 at the plasma membrane under low K+ conditions, which prevents endocytosis and vacuolar degradation. Budding yeast Ptk2 phosphorylates and regulates the plasma membrane H+ ATPase, Pma1. The HAL4-like subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd13994','STKc_HAL4_like'),(1544969,'gnl|CDD|224027',_binary 'Cytidylate kinase [Nucleotide transport and metabolism].','COG1102','Cmk'),(1544970,'gnl|CDD|223637',_binary 'Adenylate kinase and related kinases [Nucleotide transport and metabolism].','COG0563','Adk'),(1545347,'gnl|CDD|270898',_binary 'Catalytic domain of the Serine/Threonine kinase, eukaryotic translation Initiation Factor 2-Alpha Kinase.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. EIF2AKs phosphorylate the alpha subunit of eIF-2, resulting in the downregulation of protein synthesis. eIF-2 phosphorylation is induced in response to cellular stresses including virus infection, heat shock, nutrient deficiency, and the accummulation of unfolded proteins, among others. There are four distinct kinases that phosphorylate eIF-2 and control protein synthesis under different stress conditions: General Control Non-derepressible-2 (GCN2) which is activated during amino acid or serum starvation; protein kinase regulated by RNA (PKR) which is activated by double stranded RNA; heme-regulated inhibitor kinase (HRI) which is activated under heme-deficient conditions; and PKR-like endoplasmic reticulum kinase (PERK) which is activated when misfolded proteins accumulate in the ER. The EIF2AK subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and phosphoinositide 3-kinase.','cd13996','STKc_EIF2AK'),(1547105,'gnl|CDD|188159',_binary 'putative phage tail component, N-terminal domain.  This model represents the best-conserved region of about 125 amino acids, toward the N-terminus, of a family of proteins from temperate phage of a number of Gram-positive bacteria. These phage proteins range in length from 230 to 525 amino acids. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01633','unnamed_protein_product'),(1547977,'gnl|CDD|235028',_binary 'adenine phosphoribosyltransferase; Provisional.','PRK02304','PRK02304'),(1549437,'gnl|CDD|187630',_binary 'Enoyl acyl carrier protein (ACP) reductase (ENR), divergent SDR.  This bacterial subgroup of ENRs includes Escherichia coli ENR. ENR catalyzes the NAD(P)H-dependent reduction of enoyl-ACP in the last step of fatty acid biosynthesis. De novo fatty acid biosynthesis is catalyzed by the fatty acid synthetase complex, through the serial addition of 2-carbon subunits. In bacteria and plants,ENR catalyzes one of six synthetic steps in this process. Oilseed rape ENR, and also apparently the NADH-specific form of Escherichia coli ENR, is tetrameric.  Although similar to the classical SDRs, this group does not have the canonical catalytic tetrad, nor does it have the typical Gly-rich NAD-binding pattern. Such so-called divergent SDRs have a GXXXXXSXA NAD-binding motif and a YXXMXXXK (or YXXXMXXXK) active site motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05372','ENR_SDR'),(1550248,'gnl|CDD|306855',_binary 'Type II/IV secretion system protein.  This family contains both type II and type IV pathway secretion proteins from bacteria. VirB11 ATPase is a subunit of the Agrobacterium tumefaciens transfer DNA (T-DNA) transfer system, a type IV secretion pathway required for delivery of T-DNA and effector proteins to plant cells during infection.','pfam00437','T2SSE'),(1551337,'gnl|CDD|309295',_binary 'Autophagy protein Apg6.  In yeast, 15 Apg proteins coordinate the formation of autophagosomes. Autophagy is a bulk degradation process induced by starvation in eukaryotic cells. Apg6/Vps30p has two distinct functions in the autophagic process, either associated with the membrane or in a retrieval step of the carboxypeptidase Y sorting pathway.','pfam04111','APG6'),(1551490,'gnl|CDD|180965',_binary 'RNA polymerase sigma factor SigF; Reviewed.','PRK07408','PRK07408'),(1551529,'gnl|CDD|238050',_binary 'Immunoglobulin-like fold, Plexins, Transcription factors (IPT). IPTs are also known as Transcription factor ImmunoGlobin (TIG) domains. They are present in intracellular transcription factors, cell surface receptors (such as plexins and scatter factor receptors), as well as, cyclodextrin glycosyltransferase and similar enzymes. Although they are involved in DNA binding in transcription factors, their function in other proteins is unknown. In these transcription factors, IPTs form homo- or heterodimers with the exception of the nuclear factor of activated Tcells (NFAT) transcription factors which are mainly monomers.','cd00102','IPT'),(1551613,'gnl|CDD|315220',_binary 'Phosphate ATP-binding cassette transporter.  This domain family is found in bacteria, and is typically between 143 and 173 amino acids in length. The family is found in association with pfam00528. There is a single completely conserved residue P that may be functionally important.','pfam12501','DUF3708'),(1551782,'gnl|CDD|239326',_binary 'Glutaredoxin (GRX) family, PKC-interacting cousin of TRX (PICOT)-like subfamily; composed of PICOT and GRX-PICOT-like proteins. The non-PICOT members of this family contain only the GRX-like domain, whereas PICOT contains an N-terminal TRX-like domain followed by one to three GRX-like domains. It is interesting to note that PICOT from plants contain three repeats of the GRX-like domain, metazoan proteins (except for insect) have two repeats, while fungal sequences contain only one copy of the domain. PICOT is a protein that interacts with protein kinase C (PKC) theta, a calcium independent PKC isoform selectively expressed in skeletal muscle and T lymphocytes. PICOT inhibits the activation of c-Jun N-terminal kinase and the transcription factors, AP-1 and NF-kB, induced by PKC theta or T-cell activating stimuli. Both GRX and TRX domains of PICOT are required for its activity. Characterized non-PICOT members of this family include CXIP1, a CAX-interacting protein in Arabidopsis thaliana, and PfGLP-1, a GRX-like protein from Plasmodium falciparum.','cd03028','GRX_PICOT_like'),(1552884,'gnl|CDD|238978',_binary 'Cytidine monophosphate kinase (CMPK) catalyzes the reversible phosphorylation of cytidine monophosphate (CMP) to produce cytidine diphosphate (CDP), using ATP as the preferred phosphoryl donor.','cd02020','CMPK'),(1553274,'gnl|CDD|316016',_binary 'FAD-NAD(P)-binding.','pfam13454','NAD_binding_9'),(1553990,'gnl|CDD|259848',_binary 'Largest subunit (Rpb1) of eukaryotic RNA polymerase II (RNAP II), N-terminal domain.  The two largest subunits of RNA polymerase II (RNAP II), Rpb1 and Rpb2, form the active site, DNA entry channel and RNA exit channel. RNAP II is a large multi-subunit complex responsible for the synthesis of mRNA in eukaryotes. RNAP II consists of a 10-subunit core enzyme and a peripheral heterodimer of two subunits. Structure studies suggest that RNAP complexes from different organisms share a crab-claw-shape structure. In yeast, Rpb1 and Rpb2, each makes up one clamp, one jaw, and part of the cleft. Rpb1_N contains part of the active site, forms the head and core of the one clamp, and makes up the pore and funnel regions of RNAP II.','cd02733','RNAP_II_RPB1_N'),(1554486,'gnl|CDD|224071',_binary 'MinD superfamily P-loop ATPase containing an inserted ferredoxin domain [Energy production and conversion].','COG1149','COG1149'),(1555740,'gnl|CDD|312208',_binary 'Autophagy protein 16 (ATG16).  Autophagy is a ubiquitous intracellular degradation system for eukaryotic cells. During autophagy, cytoplasmic components are enclosed in autophagosomes and delivered to lysosomes/vacuoles. ATG16 (also known as Apg16) has been shown to be bind to Apg5 and is required for the function of the Apg12p-Apg5p conjugate in the yeast autophagy pathway.','pfam08614','ATG16'),(1557130,'gnl|CDD|224678',_binary 'Organic hydroperoxide reductase [Secondary metabolites biosynthesis, transport and catabolism].','COG1764','osmC'),(1559909,'gnl|CDD|224228',_binary 'Transcriptional regulator [Transcription].','COG1309','AcrR'),(1562554,'gnl|CDD|224327',_binary 'Predicted phosphohydrolases [General function prediction only].','COG1409','Icc'),(1562826,'gnl|CDD|308367',_binary 'Phosphoribosylformylglycinamidine (FGAM) synthase.  This family forms a component of the de novo purine biosynthesis pathway.','pfam02700','PurS'),(1563317,'gnl|CDD|238367',_binary 'Glutamine amidotransferases class-II (GN-AT)_GPAT- type. This domain is found at the N-terminus of  glutamine phosphoribosylpyrophosphate (Prpp) amidotransferase (GPATase) . The glutaminase domain catalyzes amide nitrogen transfer from glutamine to the appropriate substrate. In this process, glutamine is hydrolyzed to glutamic acid and ammonia. GPATase catalyzes the first step in purine biosynthesis, an amide transfer from glutamine to PRPP,  resulting in phosphoribosylamine, pyrophosphate and glutamate. GPATase crystalizes as a homotetramer, but can also exist as a homdimer.','cd00715','GPATase_N'),(1563327,'gnl|CDD|222881',_binary 'ADP-ribosyltransferase; Provisional.','PHA02566','alt'),(1563608,'gnl|CDD|224870',_binary 'Predicted transcriptional regulator [Transcription].','COG1959','COG1959'),(1564469,'gnl|CDD|240019',_binary 'Mitochondrial ribosomal protein L46 (MRP L46) is a component of the large subunit (39S) of the mammalian mitochondrial ribosome and a member of the Nudix hydrolase superfamily. MRPs are thought to be involved in the maintenance of the mitochondrial DNA. In general, members of the Nudix superfamily require a divalent cation, such as Mg2+ or Mn2+, for activity and contain the Nudix motif, a highly conserved 23-residue block (GX5EX7REUXEEXGU, where U = I, L or V), which functions as a metal binding and catalytic site. MRP L46 appears to contain a modified nudix motif.','cd04661','MRP_L46'),(1564513,'gnl|CDD|274421',_binary 'membrane protein AbrB duplication.  The model describes a hydrophobic sequence region that is duplicated to form the AbrB protein of Escherichia coli (not to be confused with a Bacillus subtilis protein with the same gene symbol). In some species, notably the Cyanobacteria and Thermus thermophilus, proteins consist of a single copy rather than two copies. The member from Pseudomonas putida, PP_1415, was suggested to be an ammonia monooxygenase characteristic of heterotrophic nitrifiers, based on an experimental indication of such activity in the organism and a glimmer of local sequence similarity between parts of P. putida protein and an instance of the AmoA protein from Nitrosomonas europaea (; we do not believe the sequence similarity to be meaningful. The member from E. coli (b0715, ybgN) appears to be the largely uncharacterized AbrB (aidB regulator) protein of E. coli cited in Volkert, et al. (PMID 8002588), although we did not manage to trace the origin of association of the article to the sequence.','TIGR03082','Gneg_AbrB_dup'),(1564934,'gnl|CDD|225164',_binary 'Holliday junction resolvasome, helicase subunit [DNA replication, recombination, and repair].','COG2255','RuvB'),(1565205,'gnl|CDD|193498',_binary 'M28 Zn-peptidase Streptomyces griseus Aminopeptidase and similar proteins.  Peptidase family M28; Streptomyces griseus Aminopeptidase (SGAP, Leucine aminopeptidase (LAP), aminopeptidase S, Mername-AA022 peptidase) subfamily. SGAP is a di-zinc exopeptidase with high preference towards large hydrophobic amino-terminal residues, with Leu being the most efficiently cleaved. It can accommodate all except Pro and Glu residues in the P1\' position. It is a monomeric (30 kDa), calcium-activated and calcium-stabilized enzyme; its activation by calcium correlates with substrate specificity and it has thermal stability only in the presence of calcium. Although SGAP contains a calcium binding site, it is not conserved in many members of this subfamily. SGAP is present in the extracellular fluid of S. griseus cultures.','cd03876','M28_SGAP_like'),(1565345,'gnl|CDD|234647',_binary 'putative DNA-binding protein; Validated.','PRK00118','PRK00118'),(1567016,'gnl|CDD|317179',_binary 'Solute carrier (proton/amino acid symporter), TRAMD3 or PAT1.  PAT1 (proton amino acid transporter 1), also known as TRAMD3 of AAT-1, is the molecular correlate of the intestinal imino acid carrier. It is a proton-amino acid co-transporter having a stoichiometry of 1:1. Due to its mechanism, PAT1 activity increases at acidic pH, which correlates well with the acidic micro-climate close to the brush-border in the intestine. Glycine, proline, and alanine are the preferred substrates of the transporter. The maximum velocity is similar for the three substrates. All substrates are transported with low affinity, showing Km values in the range of 2-10 mM. The transporter does not discriminate between L- and D-isoforms of these amino acids; in addition, beta-alanine is transported with similar affinity as alpha-alanine. Similar to the IMINO transporter, the amino acid analog MeAIB is recognized by PAT1. The transporter is strongly expressed in the small intestine, colon, kidney, and brain.','pfam14738','PaaSYMP'),(1567377,'gnl|CDD|179933',_binary 'bifunctional N-succinyldiaminopimelate-aminotransferase/acetylornithine transaminase protein; Reviewed.','PRK05093','argD'),(1567454,'gnl|CDD|234619',_binary 'Holliday junction DNA helicase RuvB; Reviewed.','PRK00080','ruvB'),(1567479,'gnl|CDD|238895',_binary 'Hybrid cluster protein (HCP), formerly known as prismane, is thought to play a role in nitrogen metabolism but its specific function is unknown. HCP has three structural domains, an N-terminal alpha-helical domain, and two similar domains comprising a central beta-sheet flanked by alpha-helices. HCP contains two iron-sulfur clusters, one of which is a [Fe4-S4] cubane cluster similar to that of carbon monoxide dehydrogenase (CODH).  The second cluster, referred to as the hybrid cluster, is a hybrid [Fe4-S2-O2] center located at the interface of the three domains. Although the hybrid cluster is buried within the protein, it is accessible through a large hydrophobic cavity.','cd01914','HCP'),(1567538,'gnl|CDD|236347',_binary 'DnaA regulatory inactivator Hda; Validated.','PRK08903','PRK08903'),(1568568,'gnl|CDD|177633',_binary 'very late expression factor 1; Provisional.','PHA03397','vlf-1'),(1571280,'gnl|CDD|179877',_binary 'septation ring formation regulator EzrA; Provisional.','PRK04778','PRK04778'),(1571544,'gnl|CDD|306864',_binary 'SRP54-type protein, GTPase domain.  This family includes relatives of the G-domain of the SRP54 family of proteins.','pfam00448','SRP54'),(1572981,'gnl|CDD|274615',_binary 'FimV N-terminal domain.  This region is found at, or about 200 amino acids from, the N-terminus of FimV from Pseudomonas aeruginosa, TspA of Neisseria meningitidis, and related proteins. Disruption of FimV blocks twitching motility from type IV pili; Semmler, et al. suggest a role for this family in peptidoglycan layer remodelling required by type IV fimbrial systems. Most but not all members of this protein family have a C-terminal region recognized by TIGR03504. In between is a highly variable, often repeat-filled region rich in the negatively charged amino acids Asp and Glu.','TIGR03505','FimV_core'),(1573376,'gnl|CDD|153271',_binary 'The Bin/Amphiphysin/Rvs (BAR) domain, a dimerization module that binds membranes and detects membrane curvature.  BAR domains are dimerization, lipid binding and curvature sensing modules found in many different proteins with diverse functions including organelle biogenesis, membrane trafficking or remodeling, and cell division and migration. Mutations in BAR containing proteins have been linked to diseases and their inactivation in cells leads to altered membrane dynamics. A BAR domain with an additional N-terminal amphipathic helix (an N-BAR) can drive membrane curvature. These N-BAR domains are found in amphiphysins and endophilins, among others. BAR domains are also frequently found alongside domains that determine lipid specificity, such as the Pleckstrin Homology (PH) and Phox Homology (PX) domains which are present in beta centaurins (ACAPs and ASAPs) and sorting nexins, respectively. A FES-CIP4 Homology (FCH) domain together with a coiled coil region is called the F-BAR domain and is present in Pombe/Cdc15 homology (PCH) family proteins, which include Fes/Fes tyrosine kinases, PACSIN or syndapin, CIP4-like proteins, and srGAPs, among others. The Inverse (I)-BAR or IRSp53/MIM homology Domain (IMD) is found in multi-domain proteins, such as IRSp53 and MIM, that act as scaffolding proteins and transducers of a variety of signaling pathways that link membrane dynamics and the underlying actin cytoskeleton. BAR domains form dimers that bind to membranes, induce membrane bending and curvature, and may also be involved in protein-protein interactions. The I-BAR domain induces membrane protrusions in the opposite direction compared to classical BAR and F-BAR domains, which produce membrane invaginations. BAR domains that also serve as protein interaction domains include those of arfaptin and OPHN1-like proteins, among others, which bind to Rac and Rho GAP domains, respectively.','cd07307','BAR'),(1573810,'gnl|CDD|316325',_binary 'N-terminal domain of ribose phosphate pyrophosphokinase.  This family is frequently found N-terminal to the Pribosyltran, pfam00156.','pfam13793','Pribosyltran_N'),(1573818,'gnl|CDD|236693',_binary 'putative RNA methyltransferase; Provisional.','PRK10433','PRK10433'),(1573831,'gnl|CDD|309548',_binary 'Putative SAM-dependent methyltransferase.  This is a family of putative SAM-dependent methyltransferases.','pfam04445','SAM_MT'),(1573902,'gnl|CDD|310476',_binary 'RIB43A.  This family consists of several RIB43A-like eukaryotic proteins. Ciliary and flagellar microtubules contain a specialized set of protofilaments, termed ribbons, that are composed of tubulin and several associated proteins. RIB43A was first characterized in the unicellular biflagellate, Chlamydomonas reinhardtii although highly related sequences are present in several higher eukaryotes including humans. The function of this protein is unknown although the structure of RIB43A and its association with the specialized protofilament ribbons and with basal bodies is relevant to the proposed role of ribbons in forming and stabilizing doublet and triplet microtubules and in organising their three-dimensional structure. Human RIB43A homologs could represent a structural requirement in centriole replication in dividing cells.','pfam05914','RIB43A'),(1574230,'gnl|CDD|182670',_binary 'flagella biosynthesis regulator; Provisional.','PRK10715','flk'),(1578738,'gnl|CDD|274634',_binary 'protein-(glutamine-N5) methyltransferase, release factor-specific.  Members of this protein family are HemK (PrmC), a protein once thought to be involved in heme biosynthesis but now recognized to be a protein-glutamine methyltransferase that modifies the peptide chain release factors. All members of the seed alignment are encoded next to the release factor 1 gene (prfA) and confirmed by phylogenetic analysis. SIMBAL analysis (manuscript in prep.) shows the motif [LIV]PRx[DE]TE (in Escherichia coli, IPRPDTE) confers specificity for the release factors rather than for ribosomal protein L3. [Protein fate, Protein modification and repair].','TIGR03534','RF_mod_PrmC'),(1579188,'gnl|CDD|307944',_binary 'Flagellar FliJ protein.','pfam02050','FliJ'),(1579662,'gnl|CDD|197331',_binary 'C-terminal deadenylase domain of CCR4 and related domains.  This subfamily contains the C-terminal catalytic domain of the deadenylases, Saccharomyces cerevisiae Ccr4p and two vertebrate homologs (CCR4a and CCR4b), and related domains. CCR4 belongs to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. CCR4 is the major deadenylase subunit of the CCR4-NOT transcription complex, which contains two deadenylase subunits and several noncatalytic subunits. The other deadenylase subunit, Caf1 (called Pop2 in yeast), is a DEDD-type protein and does not belong in this superfamily. Saccharomyces cerevisiae CCR4 (or Ccr4p) is a 3\'-5\' poly(A) RNA and ssDNA exonuclease. It is the catalytic subunit of the yeast mRNA deadenylase (Ccr4p/Pop2p/Not complex). This complex participates in various ways in mRNA metabolism, including transcription initiation and elongation, and mRNA degradation. Ccr4p degrades both poly(A) and single-stranded DNA. There are two vertebrate homologs of Ccr4p, CCR4a (also called CCR4-NOT transcription complex subunit 6 or CNOT6) and CCR4b (also called CNOT6-like or CNOT6L), which independently associate with other components to form distinct CCR4-NOT multisubunit complexes. The nuclease domain of CNOT6 and CNOT6L exhibits Mg2+-dependent deadenylase activity, with specificity for poly (A) RNA as substrate. CCR4a is a component of P-bodies and is necessary for foci formation. CCR4b regulates p27/Kip1 mRNA levels, thereby influencing cell cycle progression. They both contribute to the prevention of cell death by regulating insulin-like growth factor-binding protein 5.','cd09097','Deadenylase_CCR4'),(1580312,'gnl|CDD|198397',_binary 'GIY-YIG domain found in CAXIP1-like proteins, iron-sulfur cluster assembly proteins, and similar proteins.  The family includes CAX-interacting protein-1 (CXIP1)-like proteins and iron-sulfur cluster assembly proteins, both of which contain a GIY-YIG domain that shows statistically significant similarity to the N-terminal catalytic domains of GIY-YIG family of intron-encoded homing endonuclease I-TevI and catalytic GIY-YIG domain of nucleotide excision repair endonuclease UvrC. CAXIP1 is a novel PICOT (protein kinase C-interacting cousin of thioredoxin) domain-containing Arabidopsis protein that activates H+/Ca2+ exchanger CAX1, and its homolog CAX4, but not CAX2 or CAX3. Iron-sulfur cluster assembly proteins in this family also contain a C-terminal NifU-like domain that corresponds to a common region between the NifU protein from nitrogen-fixing bacteria and rhodobacterial species. The biochemical function of NifU is unknown.','cd10450','GIY-YIG_AtGrxS16_like'),(1581382,'gnl|CDD|235239',_binary 'phenylalanyl-tRNA synthetase subunit alpha; Provisional.','PRK04172','pheS'),(1582002,'gnl|CDD|128645',_binary 'Paired Box domain.','smart00351','PAX'),(1582875,'gnl|CDD|212139',_binary 'Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) MazG-like domain found in a group of uncharacterized proteins from archaea and bacteria.  The family includes some uncharacterized hypothetical proteins from archaea and bacteria. Although their biological roles remain unclear, the family members show significant sequence similarity to the dimeric 2-deoxyuridine 5\'-triphosphate nucleotidohydrolase (dUTP pyrophosphatase or dUTPase) and NTP-PPase MazG proteins. However, unlike typical tandem-domain MazG proteins, the family contains a single MazG-like domain.','cd11532','NTP-PPase_COG4997'),(1583826,'gnl|CDD|133411',_binary 'Helix-Turn-Helix DNA binding domain of the CadR and PbrR transcription regulators.  Helix-turn-helix (HTH) CadR and PbrR transcription regulators including Pseudomonas aeruginosa CadR and Ralstonia metallidurans PbrR that regulate expression of the cadmium and lead resistance operons, respectively. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines which form a putative metal binding site. Some members in this group have a histidine-rich C-terminal extension. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.','cd04784','HTH_CadR-PbrR'),(1584470,'gnl|CDD|236969',_binary 'DNA-binding transcriptional dual regulator Crp; Provisional.','PRK11753','PRK11753'),(1584656,'gnl|CDD|213190',_binary 'ATP-binding cassette domain of peroxisomal transporter, subfamily D.  Peroxisomal ATP-binding cassette transporter (Pat) is involved in the import of very long-chain fatty acids (VLCFA) into the peroxisome. The peroxisomal membrane forms a permeability barrier for a wide variety of metabolites required for and formed during fatty acid beta-oxidation. To communicate with the cytoplasm and mitochondria, peroxisomes need dedicated proteins to transport such hydrophilic molecules across their membranes. X-linked adrenoleukodystrophy (X-ALD) is caused by mutations in the ALD gene, which encodes ALDP (adrenoleukodystrophy protein ), a peroxisomal integral membrane protein that is a member of the ATP-binding cassette (ABC) transporter protein family. The disease is characterized by a striking and unpredictable variation in phenotypic expression. Phenotypes include the rapidly progressive childhood cerebral form (CCALD), the milder adult form, adrenomyeloneuropathy (AMN), and variants without neurologic involvement (i.e. asymptomatic).','cd03223','ABCD_peroxisomal_ALDP'),(1584753,'gnl|CDD|308674',_binary 'Putative carbohydrate binding domain.  This domain represents the N terminal domain in chitobiases and beta-hexosaminidases EC:3.2.1.52. It is composed of a beta sandwich structure that is similar in structure to the cellulose binding domain of cellulase from Cellulomonas fimi. This suggests that this may be a carbohydrate binding domain.','pfam03173','CHB_HEX'),(1588150,'gnl|CDD|275151',_binary 'CofD-related protein, GAK system.  Members of this family are distantly related to CofD, the enzyme LPPG:FO 2-phospho-L-lactate transferase, involved in coenzyme F420 biosynthesis. This family appears to belong to a biosynthesis cassette of unknown function.','TIGR04357','conserved_hypothetical_pr'),(1592124,'gnl|CDD|238713',_binary 'Adenylate kinase (ADK) catalyzes the reversible phosphoryl transfer from adenosine triphosphates (ATP) to adenosine monophosphates (AMP) and to yield adenosine diphosphates (ADP). This enzyme is required for the biosynthesis of ADP and is essential for homeostasis of adenosine phosphates.','cd01428','ADK'),(1593416,'gnl|CDD|273387',_binary 'bacteriocin secretion accessory protein.  This family represents an accessory protein that works with the bacteriocin maturation and ABC transport secretion protein described by TIGR01193. [Transport and binding proteins, Other].','TIGR01000','Mesentericin_Y105_secreti'),(1593798,'gnl|CDD|310338',_binary 'Coiled stalk of trimeric autotransporter adhesin.  This short motif is found in invasins and haemagglutinins, normally associated with (pfam05658).','pfam05662','YadA_stalk'),(1594696,'gnl|CDD|224241',_binary 'Predicted nuclease of restriction endonuclease-like fold, RmuC family [General function prediction only].','COG1322','COG1322'),(1595340,'gnl|CDD|319966',_binary 'C-terminal of Bacillus subtilis CatE like protein, a vicinal oxygen chelate subfamily.  Uncharacterized subfamily of vicinal oxygen chelate (VOC) superfamily contains Bacillus subtilis CatE and similar proteins. CatE is proposed to function as Catechol-2,3-dioxygenase. VOC  is composed of structurally related proteins with paired beta.alpha.beta.beta.beta motifs that provide a metal coordination environment with two or three open or readily accessible coordination sites to promote direct electrophilic participation of the metal ion in catalysis. VOC domain is found in a variety of structurally related metalloproteins, including the bleomycin resistance protein, glyoxalase I, and type I ring-cleaving dioxygenases. A bound metal ion is required for protein activities for the members of this superfamily. A variety of metal ions have been found in the catalytic centers of these proteins including Fe(II), Mn(II), Zn(II), Ni(II) and Mg(II). The protein superfamily contains members with or without domain swapping. The proteins of this family share three conserved metal binding amino acids with the type I extradiol dioxygenases, which shows no domain swapping.','cd16359','VOC_BsCatE_like_C'),(1598362,'gnl|CDD|314569',_binary 'Invariant surface glycoprotein.  This family is found in Trypanosome species, and appears to be one of two invariant surface glycoproteins, ISG65 and ISG75. that are found in the mammalian stage of the parasitic protozoan. the sequence suggests the two families are polypeptides with N-terminal signal sequences, hydrophilic extracellular domains, single trans-membrane alpha-helices and short cytoplasmic domains. they are both expressed in the bloodstream form but not in the midgut stage. Both polypeptides are distributed over the entire surface of the parasite.','pfam11727','ISG65-75'),(1599805,'gnl|CDD|317945',_binary 'GHH signature containing HNH/Endo VII superfamily nuclease toxin.  A predicted toxin of the HNH/Endonuclease VII fold present in bacterial polymorphic toxin systems with a characteristic sG[HQ]H signature motif. In bacterial polymorphic toxin systems, the toxin is exported by the type 2, type 6, type 7 or TcdB/TcaC-type secretion system. The metazoan teneurin proteins possess an inactive of this domain at their C-terminus.','pfam15636','Tox-GHH'),(1599821,'gnl|CDD|318210',_binary 'Putative antitoxin of bacterial toxin-antitoxin system, YdaS/YdaT.  YdaS_antitoxin is a family of putative bacterial antitoxins, neutralising the toxin YdaT, family pfam06254.','pfam15943','YdaS_antitoxin'),(1600982,'gnl|CDD|319818',_binary 'phosphatase, similar to Escherichia coli Cof and Thermotoga maritima TM0651; belongs to the haloacid dehalogenase-like superfamily.  Escherichia coli Cof is involved in the hydrolysis of HMP-PP (4-amino-2-methyl-5-hydroxymethylpyrimidine pyrophosphate, an intermediate in thiamin biosynthesis), Cof also has phosphatase activity against the coenzymes pyridoxal phosphate (PLP) and FMN. Thermotoga maritima TM0651 acts as a phosphatase with a phosphorylated carbohydrate molecule as a possible substrate. Escherichia coli YbhA is also a member of this family and catalyzes the dephosphorylation of PLP, YbhA can also hydrolyze erythrose-4-phosphate and fructose-1,6-bis-phosphate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd07516','HAD_Pase'),(1604255,'gnl|CDD|188620',_binary 'H3TH domains of Flap endonuclease-1 (FEN1)-like structure specific 5\' nucleases: FEN1 (archaeal), GEN1, YEN1, and XPG.  The 5\' nucleases within this family are capable of both 5\'-3\' exonucleolytic activity and cleaving bifurcated or branched DNA, in an endonucleolytic, structure-specific manner, and are involved in DNA replication, repair, and recombination. This family includes the H3TH (helix-3-turn-helix) domains of archaeal Flap Endonuclease-1 (FEN1), Gap Endonuclease 1 (GEN1), Yeast Endonuclease 1 (YEN1), Xeroderma pigmentosum complementation group G (XPG) nuclease, and other eukaryotic and archaeal homologs. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. With the except of the Mkt1-like proteins, the nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+, Mn2+, Zn2+, or Co2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one Asp residue from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5\' nucleases.','cd09900','H3TH_XPG-like'),(1604360,'gnl|CDD|214707',_binary 'AP endonuclease family 2.  These endonucleases play a role in DNA repair. Cleave phosphodiester bonds at apurinic or apyrimidinic sites.','smart00518','AP2Ec'),(1609677,'gnl|CDD|236988',_binary 'N5-glutamine S-adenosyl-L-methionine-dependent methyltransferase; Provisional.','PRK11805','PRK11805'),(1609747,'gnl|CDD|272905',_binary 'Cof subfamily of IIB subfamily of haloacid dehalogenase superfamily.  This subfamily of sequences falls within the Class-IIB subfamily (TIGR01484) of the Haloacid Dehalogenase superfamily of aspartate-nucleophile hydrolases. The use of the name \'Cof\' as an identifier here is arbitrary and refers to the E. coli Cof protein. This subfamily is notable for the large number of recent paralogs in many species. Listeria, for instance, has 12, Clostridium, Lactococcus and Streptococcus pneumoniae have 8 each, Enterococcus and Salmonella have 7 each, and Bacillus subtilus, Mycoplasma, Staphylococcus and E. coli have 6 each. This high degree of gene duplication is limited to the gamma proteobacteria and low-GC gram positive lineages. The profusion of genes in this subfamily is not coupled with a high degree of divergence, so it is impossible to determine an accurate phylogeny at the equivalog level. Considering the relationship of this subfamily to the other known members of the HAD-IIB subfamily (TIGR01484), sucrose and trehalose phosphatases and phosphomannomutase, it seems a reasonable hypothesis that these enzymes act on phosphorylated sugars. Possibly the diversification of genes in this subfamily represents the diverse sugars and polysaccharides that various bacteria find in their biological niches. The members of this subfamily are restricted almost exclusively to bacteria (one sequences from S. pombe scores above trusted, while another is between trusted and noise). It is notable that no archaea are found in this group, the closest relations to the archaea found here being two Deinococcus sequences. [Unknown function, Enzymes of unknown specificity].','TIGR00099','Uncharacterized_protein_M'),(1610997,'gnl|CDD|240310',_binary 'cysteine proteinase; Provisional.','PTZ00200','PTZ00200'),(1610998,'gnl|CDD|214761',_binary 'Papain family cysteine protease.','smart00645','Pept_C1'),(1612192,'gnl|CDD|316887',_binary 'Butirosin biosynthesis protein H, N-terminal.  BtrH_N is the N-terminus of the acyl carrier protein:aminoglycoside acyltransferase BtrH. Alternatively it can be referred to as butirosin biosynthesis protein H. BtrH transfers the unique (S)-4-amino-2-hydroxybutyrate (AHBA) side chain, which protects the antibiotic butirosin from several common resistance mechanisms. Butirosin, an aminoglycoside antibiotic produced by Bacillus circulans, exhibits improved antibiotic properties over its parent molecule and retains bactericidal activity toward many aminoglycoside-resistant strains. Butirosin is unique in carrying the AHBA side-chain. BtrH transfers the AHBA from the acyl carrier protein BtrI to the parent aminoglycoside ribostamycin as a gamma-glutamylated dipeptide.','pfam14399','BtrH_N'),(1612321,'gnl|CDD|99983',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. capH in Staphylococcus aureus has been shown to be required for the biosynthesis of the type 1 capsular polysaccharide (CP1).','cd03812','GT1_CapH_like'),(1612324,'gnl|CDD|99990',_binary 'This family is most closely related to the GT1 family of glycosyltransferases. AmSD in Erwinia amylovora has been shown to be involved in the biosynthesis of amylovoran, the acidic exopolysaccharide acting as a virulence factor. This enzyme may be responsible for the formation of  galactose alpha-1,6 linkages in amylovoran.','cd03820','GT1_amsD_like'),(1612334,'gnl|CDD|133014',_binary 'Phosphocholine cytidylyltransferases catalyze the synthesis of CDP-choline.  This family contains proteins similar to prokaryotic phosphocholine (P-cho) cytidylyltransferases. Phosphocholine (PC) cytidylyltransferases catalyze the transfer of a cytidine monophosphate from CTP to phosphocholine to form CDP-choline. PC is the most abundant phospholipid in eukaryotic membranes and it is also important in prokaryotic membranes. For pathogenic prokaryotes, the cell surface PC facilitates the interaction with host surface and induces attachment and invasion. In addition cell wall PC serves as scaffold for a group of choline-binding proteins that are secreted from the cells. Phosphocholine (PC) cytidylyltransferase is a key enzyme in the prokaryotic choline metabolism pathway. It has been hypothesized to consist of a choline transport system, a choline kinase, CTP:phosphocholine cytidylyltransferase, and a choline phosphotransferase that transfers P-Cho from CDP-Cho to either lipoteichoic acid or lipopolysaccharide.','cd02523','PC_cytidylyltransferase'),(1612379,'gnl|CDD|313218',_binary 'Uncharacterized protein conserved in bacteria (DUF2184).  This domain, found in various hypothetical bacterial proteins, has no known function.','pfam09950','DUF2184'),(1613184,'gnl|CDD|179289',_binary 'Trp operon repressor; Provisional.','PRK01381','PRK01381'),(1613799,'gnl|CDD|133025',_binary 'GT_2_like_f is a subfamily of the glycosyltransferase family 2 (GT-2) with unknown function.  GT-2 includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. Glycosyltransferases have been classified into more than 90 distinct sequence based families.','cd04182','GT_2_like_f'),(1613980,'gnl|CDD|240515',_binary 'NusG contains an NGN domain at its N-terminus and KOW motif at its C-terminus.  KOW_NusG motif is one of the two domains of N-Utilization Substance G (NusG) a transcription elongation and Rho-termination factor in bacteria and archaea. KOW domain is known as an RNA-binding motif that is shared so far among some families of ribosomal proteins, the essential bacterial transcriptional elongation factor NusG, the eukaryotic chromatin elongation factor Spt5, the higher eukaryotic KIN17 proteins and Mtr4. The eukaryotic ortholog of NusG is Spt5 with multiple KOW motifs at its C-terminus.','cd06091','KOW_NusG'),(1615059,'gnl|CDD|315907',_binary 'Transcriptional regulator, AbiEi antitoxin.  AbiEi_4 is the cognate antitoxin of the type IV toxin-antitoxin \'innate immunity\' bacterial abortive infection (Abi) system that protects bacteria from the spread of a phage infection. The Abi system is activated upon infection with phage to abort the cell thus preventing the spread of phage through viral replication. There are some 20 or more Abis, and they are predominantly plasmid-encoded lactococcal systems. TA, toxin-antitoxin, systems on plasmids function by killing cells that lose the plasmid upon division. AbiE phage resistance systems function as novel Type IV TAs and are widespread in bacteria and archaea. The cognate antitoxin is pfam13338.','pfam13338','AbiEi_4'),(1617559,'gnl|CDD|309556',_binary 'Encapsulating protein for peroxidase.  The Linocin_M18 is found in eubacteria and archaea. These proteins, referred to as encapsulins, form nanocompartments within the bacterium which contain ferritin-like proteins or peroxidases, enzymes involved in oxidative-stress response. These enzymes are targeted to the interior of encapsulins via unique C-terminal extensions.','pfam04454','Linocin_M18');
INSERT INTO `domain` VALUES (1618525,'gnl|CDD|224623',_binary 'Predicted transcriptional regulator [Transcription].','COG1709','COG1709'),(1620027,'gnl|CDD|310973',_binary 'DEAD_2.  This represents a conserved region within a number of RAD3-like DNA-binding helicases that are seemingly ubiquitous - members include proteins of eukaryotic, bacterial and archaeal origin. RAD3 is involved in nucleotide excision repair, and forms part of the transcription factor TFIIH in yeast.','pfam06733','DEAD_2'),(1620081,'gnl|CDD|176462',_binary 'Aspartase.  This subgroup contains Escherichia coli aspartase (L-aspartate ammonia-lyase), Bacillus aspartase and related proteins. It is a member of the Lyase class I family, which includes both aspartase (L-aspartate ammonia-lyase) and fumarase class II enzymes. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. Aspartase catalyzes the reversible deamination of aspartic acid.','cd01357','Aspartase'),(1621444,'gnl|CDD|259850',_binary 'Cis (Z)-Isoprenyl Diphosphate Synthases.  Cis (Z)-Isoprenyl Diphosphate Synthases (cis-IPPS) catalyze the successive 1\'-4 condensation of the isopentenyl diphosphate (IPP) molecule to trans,trans-farnesyl diphosphate (FPP) or to cis,trans-FPP to form long-chain polyprenyl diphosphates. A few can also catalyze the condensation of IPP to trans-geranyl diphosphate to form the short-chain cis,trans- FPP. In prokaryotes, the cis-IPPS, undecaprenyl diphosphate synthase (UPP synthase), catalyzes the formation of the carrier lipid UPP in bacterial cell wall peptidoglycan biosynthesis. Similarly, in eukaryotes, the cis-IPPS, dehydrodolichyl diphosphate (dedol-PP) synthase catalyzes the formation of the polyisoprenoid glycosyl carrier lipid dolichyl monophosphate. cis-IPPS form homodimers and are mechanistically and structurally distinct from trans-IPPS, which lack the DDXXD motifs, yet require Mg2+ for activity.','cd00475','Cis_IPPS'),(1623256,'gnl|CDD|213206',_binary 'The SMC head domain belongs to the ATP-binding cassette superfamily.  The structural maintenance of chromosomes (SMC) proteins are essential for successful chromosome transmission during replication and segregation of the genome in all organisms. SMCs are generally present as single proteins in bacteria, and as at least six distinct proteins in eukaryotes. The proteins range in size from approximately 110 to 170 kDa, and each has five distinct domains: amino- and carboxy-terminal globular domains, which contain sequences characteristic of ATPases, two coiled-coil regions separating the terminal domains , and a central flexible hinge. SMC proteins function together with other proteins in a range of chromosomal transactions, including chromosome condensation, sister-chromatid cohesion, recombination, DNA repair, and epigenetic silencing of gene expression.','cd03239','ABC_SMC_head'),(1624261,'gnl|CDD|227085',_binary 'Predicted transcriptional regulator [Transcription].','COG4742','COG4742'),(1625224,'gnl|CDD|206685',_binary 'Obg GTPase.  The Obg nucleotide binding protein subfamily has been implicated in stress response, chromosome partitioning, replication initiation, mycelium development, and sporulation. Obg proteins are among a large group of GTP binding proteins conserved from bacteria to humans. The E. coli homolog, ObgE is believed to function in ribosomal biogenesis. Members of the subfamily contain two equally and highly conserved domains, a C-terminal GTP binding domain and an N-terminal glycine-rich domain.','cd01898','Obg'),(1625288,'gnl|CDD|237865',_binary 'DNA polymerase III subunits gamma and tau; Provisional.','PRK14951','PRK14951'),(1626038,'gnl|CDD|187580',_binary 'L-threonine dehydrogenase, extended (e) SDRs.  This subgroup contains members identified as L-threonine dehydrogenase (TDH). TDH catalyzes the zinc-dependent formation of 2-amino-3-ketobutyrate from L-threonine via NAD(H)-dependent oxidation. This group is distinct from TDHs that are members of the medium chain dehydrogenase/reductase family. This group has the NAD-binding motif and active site tetrad of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05272','TDH_SDR_e'),(1626092,'gnl|CDD|315916',_binary 'Putative adhesin.  This has a putative all-beta structure with a twenty-residue repeat with a highly conserved repeating GD, gly-asp, motif. It may form part of a bacterial adhesin.','pfam13349','DUF4097'),(1626146,'gnl|CDD|224035',_binary 'Reverse gyrase [DNA replication, recombination, and repair].','COG1110','COG1110'),(1627629,'gnl|CDD|316548',_binary 'YlzJ-like protein.  The YlzJ-like protein family includes the B. subtilis YlzJ protein, which is functionally uncharacterized. This family of proteins is found in bacteria. Proteins in this family are typically between 61 and 72 amino acids in length. There are two completely conserved residues (L and G) that may be functionally important.','pfam14035','YlzJ'),(1627990,'gnl|CDD|317800',_binary 'PHD/FYVE-zinc-finger like domain.  This family appears to be a combination domain of several consecutive zinc-binding regions.','pfam15446','zf-PHD-like'),(1628678,'gnl|CDD|131526',_binary 'flagellar export protein FliJ.  Members of this family are the FliJ protein found, in nearly every case, in the midst of other flagellar biosynthesis genes in bacgterial genomes. Typically the fliJ gene is found adjacent to the gene for the flagellum-specific ATPase FliI. Sequence scoring in the gray zone between trusted and noise cutoffs include both probable FliJ proteins and components of bacterial type III secretion systems.','TIGR02473','conserved_hypothetical_pr'),(1630933,'gnl|CDD|236707',_binary 'multidrug transporter membrane component/ATP-binding component; Provisional.','PRK10522','PRK10522'),(1632671,'gnl|CDD|236776',_binary 'cytoskeletal protein RodZ; Provisional.','PRK10856','PRK10856'),(1635292,'gnl|CDD|316625',_binary 'Domain of unknown function (DUF4286).  This family of proteins is functionally uncharacterized. This family of proteins is found in bacteria and eukaryotes. Proteins in this family are typically between 100 and 112 amino acids in length.','pfam14114','DUF4286'),(1635897,'gnl|CDD|316113',_binary 'PucR C-terminal helix-turn-helix domain.  This helix-turn-helix domain is often found at the C-terminus of PucR-like transcriptional regulators such as Bacillus subtilis pucR and is likely to be DNA-binding.','pfam13556','HTH_30'),(1636306,'gnl|CDD|240439',_binary 'major vault protein; Provisional.','PTZ00491','PTZ00491'),(1636744,'gnl|CDD|223280',_binary 'DNA-directed RNA polymerase, alpha subunit/40 kD subunit [Transcription].','COG0202','RpoA'),(1638926,'gnl|CDD|99999',_binary 'Glycosyltransferases catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds. The acceptor molecule can be a lipid, a protein, a heterocyclic compound, or another carbohydrate residue. This group of glycosyltransferases is most closely related to the previously defined glycosyltransferase family 1 (GT1). The members of this family may transfer UDP, ADP, GDP, or CMP linked sugars. The diverse enzymatic activities among members of this family reflect a wide range of biological functions. The protein structure available for this family has the GTB topology, one of the two protein topologies observed for nucleotide-sugar-dependent glycosyltransferases. GTB proteins have distinct N- and C- terminal domains each containing a typical Rossmann fold. The two domains have high structural homology despite minimal sequence homology. The large cleft that separates the two domains includes the catalytic center and permits a high degree of flexibility.','cd04950','GT1_like_1'),(1638975,'gnl|CDD|275156',_binary 'choice-of-anchor C domain.  This family describes an extracellular bacterial domain that occurs on a number of proteins with PEP-CTERM (exosortase recognition site) sequences at the C-terminus, as well some with an apparent alternate anchor sequence. Note that related pfam04862 (DUF642), as of release 26, is double the length of this model because it has two tandem regions homologous to this domain. pfam04862, in turn, belongs to a Pfam clan called the galactose-binding domain-like superfamily.','TIGR04362','hypothetical_protein'),(1639028,'gnl|CDD|200474',_binary 'Alpha amylase catalytic domain found in maltosyltransferase.  Maltosyltransferase (MTase), a maltodextrin glycosyltransferase, acts on starch and maltooligosaccharides. It catalyzes the transfer of maltosyl units from alpha-1,4-linked glucans or maltooligosaccharides to other alpha-1,4-linked glucans, maltooligosaccharides or glucose. MTase is a homodimer. The catalytic core domain has the (beta/alpha) 8 barrel fold with the active-site cleft formed at the C-terminal end of the barrel. Substrate binding experiments have led to the location of two distinct maltose-binding sites: one lies in the active-site cleft and the other is located in a pocket adjacent to the active-site cleft. It is a member of the alpha-amylase family, but unlike typical alpha-amylases, MTase does not require calcium for activity and lacks two histidine residues which are predicted to be critical for binding the glucose residue adjacent to the scissile bond in the substrates. The common reaction chemistry of the alpha-amylase family of enzymes is based on a two-step acid catalytic mechanism that requires two critical carboxylates: one acting as a general acid/base (Glu) and the other as a nucleophile (Asp). Both hydrolysis and transglycosylation proceed via the nucleophilic substitution reaction between the anomeric carbon, C1 and a nucleophile. The Alpha-amylase family comprises the largest family of glycoside hydrolases (GH), with the majority of enzymes acting on starch, glycogen, and related oligo- and polysaccharides. These proteins catalyze the transformation of alpha-1,4 and alpha-1,6 glucosidic linkages with retention of the anomeric center. The protein is described as having 3 domains: A, B, C. A is a (beta/alpha) 8-barrel; B is a loop between the beta 3 strand and alpha 3 helix of A; C is the C-terminal extension characterized by a Greek key. The majority of the enzymes have an active site cleft found between domains A and B where a triad of catalytic residues (Asp, Glu and Asp) performs catalysis. Other members of this family have lost the catalytic activity as in the case of the human 4F2hc, or only have 2 residues that serve as the catalytic nucleophile and the acid/base, such as Thermus A4 beta-galactosidase with 2 Glu residues (GH42) and human alpha-galactosidase with 2 Asp residues (GH31). The family members are quite extensive and include: alpha amylase, maltosyltransferase, cyclodextrin glycotransferase, maltogenic amylase, neopullulanase, isoamylase, 1,4-alpha-D-glucan maltotetrahydrolase, 4-alpha-glucotransferase, oligo-1,6-glucosidase, amylosucrase, sucrose phosphorylase, and amylomaltase.','cd11335','AmyAc_MTase_N'),(1639656,'gnl|CDD|308346',_binary 'Taurine catabolism dioxygenase TauD, TfdA family.  This family consists of taurine catabolism dioxygenases of the TauD, TfdA family. TauD from E. coli is a alpha-ketoglutarate-dependent taurine dioxygenase. This enzyme catalyzes the oxygenolytic release of sulfite from taurine. TfdA from Burkholderia sp. is a 2,4-dichlorophenoxyacetic acid/alpha-ketoglutarate dioxygenase. TfdA from Alcaligenes eutrophus JMP134 is a 2,4-dichlorophenoxyacetate monooxygenase. Also included are gamma-Butyrobetaine hydroxylase enzymes EC:1.14.11.1.','pfam02668','TauD'),(1640476,'gnl|CDD|213192',_binary 'First domain of the ATP-binding cassette component of cobalt transport system.  Domain I of the ABC component of a cobalt transport family found in bacteria, archaea, and eukaryota. The transition metal cobalt is an essential component of many enzymes and must be transported into cells in appropriate amounts when needed. This ABC transport system of the CbiMNQO family is involved in cobalt transport in association with the cobalamin (vitamin B12) biosynthetic pathways. Most of cobalt (Cbi) transport systems possess a separate CbiN component, the cobalt-binding periplasmic protein, and they are encoded by the conserved gene cluster cbiMNQO. Both the CbiM and CbiQ proteins are integral cytoplasmic membrane proteins, and the CbiO protein has the linker peptide and the Walker A and B motifs commonly found in the ATPase components of the ABC-type transport systems.','cd03225','ABC_cobalt_CbiO_domain1'),(1640553,'gnl|CDD|239584',_binary 'The Delta12 Fatty Acid Desaturase (Delta12-FADS)-like CD includes the integral-membrane enzymes, delta-12 acyl-lipid desaturases, oleate 12-hydroxylases, omega3 and omega6 fatty acid desaturases, and other related proteins, found in a wide range of organisms including higher plants, green algae, diatoms, nematodes, fungi, and bacteria. The expression of these proteins appears to be temperature dependent: decreases in temperature result in increased levels of fatty acid desaturation within membrane lipids subsequently altering cell membrane fluidity. An important enzyme for the production of polyunsaturates in plants is the oleate delta-12 desaturase (Arabidopsis FAD2) of the endoplasmic reticulum. This enzyme accepts l-acyl-2-oleoyl-sn-glycero-3-phosphocholine as substrate and requires NADH:cytochrome b oxidoreductase, cytochrome b, and oxygen for activity. FAD2 converts oleate(18:1) to linoleate (18:2) and is closely related to oleate 12-hydroxylase which catalyzes the hydroxylation of oleate to ricinoleate. Plastid-bound desaturases (Arabidopsis delta-12 desaturase (FAD6), omega-3 desaturase (FAD8), omega-6 desaturase (FAD6)), as well as, the cyanobacterial thylakoid-bound FADSs require oxygen, ferredoxin, and ferredoxin oxidoreductase for activity. As in higher plants, the cyanobacteria delta-12 (DesA) and omega-3 (DesB) FADSs desaturate oleate (18:1) to linoleate (18:2) and linoleate (18:2) to linolenate (18:3), respectively. Omega-3 (DesB/FAD8) and omega-6 (DesD/FAD6) desaturases catalyze reactions that introduce a double bond between carbons three and four, and carbons six and seven, respectively, from the methyl end of fatty acids. As with other members of this superfamily, this domain family has extensive hydrophobic regions that would be capable of spanning the membrane bilayer at least twice. Comparison of sequences also reveals the existence of three regions of conserved histidine cluster motifs that contain eight histidine residues: HXXXH, HXX(X)HH, and HXXHH. These histidine residues are reported to be catalytically essential and proposed to be the ligands for the iron atoms contained within the homologue, stearoyl CoA desaturase. Mutation of any one of four of these histidines in the Synechocystis delta-12 acyl-lipid desaturase resulted in complete inactivity.','cd03507','Delta12-FADS-like'),(1640839,'gnl|CDD|274241',_binary 'ATP-dependent Clp protease ATP-binding subunit clpA.  [Protein fate, Degradation of proteins, peptides, and glycopeptides].','TIGR02639','ATP-dependent_Clp_proteas'),(1641189,'gnl|CDD|213209',_binary 'ATP-binding cassette domain of RecF.  RecF is a recombinational DNA repair ATPase that maintains replication in the presence of DNA damage. When replication is prematurely disrupted by DNA damage, several recF pathway gene products play critical roles processing the arrested replication fork, allowing it to resume and complete its task. This CD represents the nucleotide binding domain of RecF. RecF belongs to a large superfamily of ABC transporters involved in the transport of a wide variety of different compounds including sugars, ions, peptides, and more complex organic molecules. The nucleotide binding domain shows the highest similarity between all members of the family. ABC transporters are a subset of nucleotide hydrolases with a signature motif, Q-loop, and H-loop/switch region, in addition to, the Walker A motif/P-loop and Walker B motif commonly found in a number of ATP- and GTP-binding and hydrolyzing proteins.','cd03242','ABC_RecF'),(1642066,'gnl|CDD|181707',_binary 'beta alanine--pyruvate transaminase; Provisional.','PRK09221','PRK09221'),(1642804,'gnl|CDD|274303',_binary 'TolA protein.  TolA couples the inner membrane complex of itself with TolQ and TolR to the outer membrane complex of TolB and OprL (also called Pal). Most of the length of the protein consists of low-complexity sequence that may differ in both length and composition from one species to another, complicating efforts to discriminate TolA (the most divergent gene in the tol-pal system) from paralogs such as TonB. Selection of members of the seed alignment and criteria for setting scoring cutoffs are based largely conserved operon struction. //The Tol-Pal complex is required for maintaining outer membrane integrity. Also involved in transport (uptake) of colicins and filamentous DNA, and implicated in pathogenesis. Transport is energized by the proton motive force. TolA is an inner membrane protein that interacts with periplasmic TolB and with outer membrane porins ompC, phoE and lamB. [Transport and binding proteins, Other, Cellular processes, Pathogenesis].','TIGR02794','Protein_TolA'),(1642806,'gnl|CDD|281941',_binary 'PspA/IM30 family.  This family includes PspA a protein that suppresses sigma54-dependent transcription. The PspA protein, a negative regulator of the Escherichia coli phage shock psp operon, is produced when virulence factors are exported through secretins in many Gram-negative pathogenic bacteria and its homolog in plants, VIPP1, plays a critical role in thylakoid biogenesis, essential for photosynthesis. Activation of transcription by the enhancer-dependent bacterial sigma(54) containing RNA polymerase occurs through ATP hydrolysis-driven protein conformational changes enabled by activator proteins that belong to the large AAA(+) mechanochemical protein family. It has been shown that PspA directly and specifically acts upon and binds to the AAA(+) domain of the PspF transcription activator.','pfam04012','PspA_IM30'),(1642898,'gnl|CDD|240268',_binary 'S-adenosylmethionine synthase; Provisional.','PTZ00104','PTZ00104'),(1643999,'gnl|CDD|236242',_binary 'ribonucleotide-diphosphate reductase subunit beta; Validated.','PRK08326','PRK08326'),(1648955,'gnl|CDD|225542',_binary 'Uncharacterized paraquat-inducible protein A [Function unknown].','COG2995','PqiA'),(1655762,'gnl|CDD|182927',_binary 'manganese transport regulator MntR; Provisional.','PRK11050','PRK11050'),(1656407,'gnl|CDD|311893',_binary 'CASP C terminal.  This domain is the C-terminal region of the CASP family of proteins. It is a Golgi membrane protein which is thought to have a role in vesicle transport.','pfam08172','CASP_C'),(1657172,'gnl|CDD|238314',_binary 'ATP:corrinoid adenosyltransferase BtuR/CobO/CobP. This family consists of the BtuR, CobO, CobP proteins all of which are Cob(I)alamin (vitamin B12) adenosyltransferase, which is involved in cobalamin (vitamin B12) biosynthesis. This enzyme is a homodimer,  which catalyzes the adenosylation reaction: ATP + cob(I)alamin + H2O <=> phosphate + diphosphate + adenosylcobalamin.','cd00561','CobA_CobO_BtuR'),(1658746,'gnl|CDD|238247',_binary 'Cytosol aminopeptidase family, N-terminal and catalytic domains.  Family M17 contains zinc- and manganese-dependent exopeptidases ( EC  3.4.11.1), including leucine aminopeptidase. They catalyze removal of amino acids from the N-terminus of a protein and play a key role in protein degradation and in the metabolism of biologically active peptides. They do not contain HEXXH motif (which is used as one of the signature patterns to group the peptidase families) in the metal-binding site. The two associated zinc ions and the active site are entirely enclosed within the C-terminal catalytic domain in leucine aminopeptidase. The enzyme is a hexamer, with the catalytic domains clustered around the three-fold axis, and the two trimers related to one another by a two-fold rotation. The N-terminal domain is structurally similar to the ADP-ribose binding Macro domain. This family includes proteins from bacteria, archaea, animals and plants.','cd00433','Peptidase_M17'),(1659146,'gnl|CDD|235015',_binary 'Holliday junction-specific endonuclease; Reviewed.','PRK02234','recU'),(1660080,'gnl|CDD|236124',_binary 'enoyl-(acyl carrier protein) reductase; Provisional.','PRK07889','PRK07889'),(1662339,'gnl|CDD|313495',_binary 'Transmembrane Fragile-X-F protein.  This is a family of conserved transmembrane proteins that appear in humans to be expressed from a region upstream of the FragileXF site and to be intimately linked with the Fragile-X syndrome. Absence of TMEM185A does not necessarily lead to developmental delay, but might in combination with other, yet unknown, factors. Otherwise, the lack of the TMEM185A protein is either disposable (redundant) or its function can be complemented by the highly similar chromosome 2 retro-pseudogene product, TMEM185B.','pfam10269','Tmemb_185A'),(1662868,'gnl|CDD|293877',_binary 'uncharacterized Thermus thermophilus TTHA1623 and related proteins; MBL-fold metallo hydrolase domain.  Includes the MBL-fold metallo hydrolase domain of uncharacterized Thermus thermophilus TTHA1623 and related proteins. Members of this subgroup belong to the MBL-fold metallo-hydrolase superfamily which is comprised mainly of hydrolytic enzymes which carry out a variety of biological functions. This family includes homologs present in a wide range of bacteria and archaea and some eukaryota. Members of the MBL-fold metallo-hydrolase superfamily exhibit a variety of active site metallo-chemistry, TTHA1623 exhibiting a uniquely shaped putative substrate-binding pocket with a glyoxalase II-type metal-coordination mode.','cd16322','TTHA1623-like_MBL-fold'),(1667920,'gnl|CDD|310525',_binary 'Cytoplasmic Fragile-X interacting family.  CYFIP1/2 (Cytoplasmic fragile X mental retardation interacting protein) like proteins for a highly conserved protein family. The function of CYFIPs is unclear, but CYFIP interaction with fragile X mental retardation interacting protein (FMRP) involves the domain of FMRP which also mediating homo- and heteromerization.','pfam05994','FragX_IP'),(1674269,'gnl|CDD|316924',_binary 'OTT_1508-like deaminase.  A member of the nucleic acid/nucleotide deaminase superfamily prototyped by Orientia OTT_1508. Members of this family are present in a wide phyletic range of bacteria,including several intracellular parasites and eukaryotes such as fungi, Leishmania, Selaginella, and some apicomplexa. In bacteria, these deaminases are predicted to function as toxins in bacterial polymorphic toxin systems. Versions in intracellular bacteria lack immunity proteins and are likely to be deployed against their eukaryotic hosts. Eukaryotic versions are predicted to function as nucleic acid (either DNA or RNA) deaminases. Among eukaryotes, some fungi show lineage-specific expansions of this family. Many fungal versions are fused to a distinct N-terminal globular domain. Various fungal versions are fused to domains involved in chromatin function. Apicomplexan versions are fused to tRNA guanine transglycosylase domain.','pfam14441','OTT_1508_deam'),(1674581,'gnl|CDD|274271',_binary 'Obg family GTPase CgtA.  This model describes a univeral, mostly one-gene-per-genome GTP-binding protein that associates with ribosomal subunits and appears to play a role in ribosomal RNA maturation. This GTPase, related to the nucleolar protein Obg, is designated CgtA in bacteria. Mutations in this gene are pleiotropic, but it appears that effects on cellular functions such as chromosome partition may be secondary to the effect on ribosome structure. Recent work done in Vibrio cholerae shows an essential role in the stringent response, in which RelA-dependent ability to synthesize the alarmone ppGpp is required for deletion of this GTPase to be lethal. [Protein synthesis, Other].','TIGR02729','Obg_CgtA'),(1674975,'gnl|CDD|278838',_binary 'ATP synthase B/B\' CF(0).  Part of the CF(0) (base unit) of the ATP synthase. The base unit is thought to translocate protons through membrane (inner membrane in mitochondria, thylakoid membrane in plants, cytoplasmic membrane in bacteria). The B subunits are thought to interact with the stalk of the CF(1) subunits. This domain should not be confused with the ab CF(1) proteins (in the head of the ATP synthase) which are found in pfam00006.','pfam00430','ATP-synt_B'),(1676082,'gnl|CDD|307893',_binary 'Sugar-specific transcriptional regulator TrmB.  One member of this family, TrmB, has been shown to be a sugar-specific transcriptional regulator of the trehalose/maltose ABC transporter in Thermococcus litoralis.','pfam01978','TrmB'),(1676759,'gnl|CDD|238220',_binary 'Ribonuclease T2 (RNase T2) is a widespread family of secreted RNases found in every organism examined thus far.  This family includes RNase Rh, RNase MC1, RNase LE, and self-incompatibility RNases (S-RNases).  Plant T2 RNases are expressed during leaf senescence in order to scavenge phosphate from ribonucleotides. They are also expressed in response to wounding or pathogen invasion. S-RNases are thought to prevent self-fertilization by acting as selective cytotoxins of \'self\' pollen.','cd00374','RNase_T2'),(1679826,'gnl|CDD|213325',_binary 'The adenylation domain of nonribosomal peptide synthetases (NRPS), including Bacillus subtilis termination module Surfactin (SrfA-C).  The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and, in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions. This family includes the adenylation domain of the Bacillus subtilis termination module (Surfactin domain, SrfA-C) which recognizes a specific amino acid building block, which is then activated and transferred to the terminal thiol of the 4\'-phosphopantetheine (Ppan) arm of the downstream peptidyl carrier protein (PCP) domain.','cd12117','A_NRPS_Srf_like'),(1680425,'gnl|CDD|239783',_binary 'AAK_NAGK-C: N-Acetyl-L-glutamate kinase - cyclic (NAGK-C) catalyzes the phosphorylation of the gamma-COOH group of N-acetyl-L-glutamate (NAG) by ATP in the second step of arginine biosynthesis found in some bacteria and photosynthetic organisms using the non-acetylated, cyclic route of ornithine biosynthesis. In this pathway, glutamate is first N-acetylated and then phosphorylated by NAGK to give phosphoryl NAG, which is converted to NAG-ornithine. There are two variants of this pathway. In one, typified by the pathway in Thermotoga maritima and Pseudomonas aeruginosa, the acetyl group is recycled by reversible transacetylation from acetylornithine to glutamate. The phosphorylation of NAG by NAGK is feedback inhibited by arginine. In photosynthetic organisms, NAGK is the target of the nitrogen-signaling protein PII. Hexameric formation of NAGK domains appears to be essential to both arginine inhibition and NAGK-PII complex formation. NAGK-C are members of the Amino Acid Kinase Superfamily (AAK).','cd04250','AAK_NAGK-C'),(1680911,'gnl|CDD|234649',_binary 'tRNA (guanine-N(7)-)-methyltransferase; Reviewed.','PRK00121','trmB'),(1681823,'gnl|CDD|235375',_binary 'sulfate adenylyltransferase subunit 2; Provisional.','PRK05253','PRK05253'),(1686690,'gnl|CDD|312405',_binary 'Protein of unknown function (DUF1822).  This family of proteins are functionally uncharacterized.','pfam08852','DUF1822'),(1687581,'gnl|CDD|213872',_binary 'conserved hypothetical protein.  The conserved hypothetical protein described here occurs as part of the trio of uncharacterized proteins common in the Actinobacteria.','TIGR03847','TIGR03847'),(1687899,'gnl|CDD|237190',_binary 'phospho-2-dehydro-3-deoxyheptonate aldolase; Provisional.','PRK12755','PRK12755'),(1690834,'gnl|CDD|240538',_binary 'Uncharacterized subfamily of the multidrug and toxic compound extrusion (MATE) proteins.  The integral membrane proteins from the MATE family are involved in exporting metabolites across the cell membrane and are responsible for multidrug resistance (MDR) in many bacteria and animals. A number of family members are involved in the synthesis of peptidoglycan components in bacteria.','cd13133','MATE_like_7'),(1690899,'gnl|CDD|185352',_binary 'PrkA family serine protein kinase; Provisional.','PRK15455','PRK15455'),(1690900,'gnl|CDD|235150',_binary 'vitamin B12-transporter ATPase; Provisional.','PRK03695','PRK03695'),(1691418,'gnl|CDD|223039',_binary 'transcriptional regulator ICP4; Provisional.','PHA03307','PHA03307'),(1691737,'gnl|CDD|236766',_binary 'ribonuclease E; Reviewed.','PRK10811','rne'),(1691829,'gnl|CDD|310165',_binary 'NmrA-like family.  NmrA is a negative transcriptional regulator involved in the post-translational modification of the transcription factor AreA. NmrA is part of a system controlling nitrogen metabolite repression in fungi. This family only contains a few sequences as iteration results in significant matches to other Rossmann fold families.','pfam05368','NmrA'),(1691830,'gnl|CDD|187542',_binary 'NmrA (a transcriptional regulator) and triphenylmethane reductase (TMR) like proteins, subgroup 1, atypical (a) SDRs.  Atypical SDRs related to NMRa, TMR, and HSCARG (an NADPH sensor). This subgroup resembles the SDRs and has a partially conserved characteristic [ST]GXXGXXG NAD-binding motif, but lacks the conserved active site residues. NmrA is a negative transcriptional regulator of various fungi, involved in the post-translational modulation of the GATA-type transcription factor AreA. NmrA lacks the canonical GXXGXXG NAD-binding motif and has altered residues at the catalytic triad, including a Met instead of the critical Tyr residue. NmrA may bind nucleotides but appears to lack any dehydrogenase activity. HSCARG has been identified as a putative NADP-sensing molecule, and redistributes and restructures in response to NADPH/NADP ratios. Like NmrA, it lacks most of the active site residues of the SDR family, but has an NAD(P)-binding motif similar to the extended SDR family, GXXGXXG. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Atypical SDRs are distinct from classical SDRs. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.','cd05231','NmrA_TMR_like_1_SDR_a'),(1699665,'gnl|CDD|309951',_binary 'Sporulation related domain.  This 70 residue domain is composed of two 35 residue repeats found in proteins involved in sporulation and cell division such as FtsN, DedD, and CwlM. This domain is involved in binding peptidoglycan. Two tandem repeats fold into a pseudo-2-fold symmetric single-domain structure containing numerous contacts between the repeats. FtsN is an essential cell division protein with a simple bitopic topology, a short N-terminal cytoplasmic segment fused to a large carboxy periplasmic domain through a single transmembrane domain. These repeats lay at the periplasmic C-terminus. FtsN localizes to the septum ring complex.','pfam05036','SPOR'),(1699788,'gnl|CDD|225172',_binary 'Predicted RNA methylase [Translation, ribosomal structure and biogenesis].','COG2263','COG2263'),(1699789,'gnl|CDD|225173',_binary 'Ribosomal protein L11 methylase [Translation, ribosomal structure and biogenesis].','COG2264','PrmA'),(1703379,'gnl|CDD|235185',_binary 'flap endonuclease-1; Provisional.','PRK03980','PRK03980'),(1708749,'gnl|CDD|176874',_binary 'Ligand-binding SRPBCC domain of an uncharacterized subfamily of proteins.  Uncharacterized group of the SRPBCC (START/RHO_alpha_C/PITP/Bet_v1/CoxG/CalC) domain superfamily. SRPBCC domains have a deep hydrophobic ligand-binding pocket and they bind diverse ligands. SRPBCC domains include the steroidogenic acute regulatory protein (StAR)-related lipid transfer (START) domains of mammalian STARD1-STARD15, the C-terminal catalytic domains of the alpha oxygenase subunit of Rieske-type non-heme iron aromatic ring-hydroxylating oxygenases (RHOs_alpha_C), Class I and II phosphatidylinositol transfer proteins (PITPs), Bet v 1 (the major pollen allergen of white birch, Betula verrucosa), CoxG, CalC, and related proteins. Other members of the superfamily include PYR/PYL/RCAR plant proteins, the aromatase/cyclase (ARO/CYC) domains of proteins such as Streptomyces glaucescens tetracenomycin, and the SRPBCC domains of Streptococcus mutans Smu.440 and related proteins.','cd08865','SRPBCC_10'),(1720367,'gnl|CDD|224323',_binary 'Transcription initiation factor TFIIIB, Brf1 subunit/Transcription initiation factor TFIIB [Transcription].','COG1405','SUA7'),(1723109,'gnl|CDD|239934',_binary 'RecG_wedge_OBF: A subfamily of OB folds corresponding to the OB fold found in the N-terminal (wedge) domain of Escherichia coli RecG. RecG is a branched-DNA-specific helicase, which catalyzes the interconversion of a DNA replication fork to a four-stranded (Holliday) junction in vivo and in vitro. This interconversion provides a route to repair stalled forks. The RecG monomer contains three domains. The N-terminal domain is named for its wedge structure, and may provide the specificity of RecG for binding branched-DNA structures. During the reversal of fork to Holliday junction, the wedge domain is fixed at the junction of the fork where the leading and lagging strand duplex arms meet, and is thought to promote the unwinding of the nascent leading and lagging strands. In order to form the Holliday junction, these nascent strands would be annealed, and the parental strands reannealed. The wedge domain may also be a processivity factor of RecG on these branched chain substrates.','cd04488','RecG_wedge_OBF'),(1723416,'gnl|CDD|226932',_binary 'Response regulator [Signal transduction mechanisms].','COG4566','TtrR'),(1725509,'gnl|CDD|234792',_binary 'excinuclease ABC subunit C; Validated.','PRK00558','uvrC'),(1727641,'gnl|CDD|311694',_binary 'Nitrogen fixation protein of unknown function.  This domain is found in the Cyanobacteria, and may be involved in nitrogen fixation, but no role has been assigned.','pfam07862','Nif11'),(1734429,'gnl|CDD|273252',_binary 'ribosomal RNA small subunit methyltransferase A.  In both E. coli and Saccharomyces cerevisiae, this protein is responsible for the dimethylation of two adjacent adenosine residues in a conserved hairpin of 16S rRNA in bacteria, 18S rRNA in eukaryotes. This adjacent dimethylation is the only rRNA modification shared by bacteria and eukaryotes. A single member of this family is present in each of the first 20 completed microbial genomes. This protein is essential in yeast, but not in E. coli, where its deletion leads to resistance to the antibiotic kasugamycin. Alternate name: S-adenosylmethionine--6-N\',N\'-adenosyl (rRNA) dimethyltransferase [Protein synthesis, tRNA and rRNA base modification].','TIGR00755','Ribosomal_RNA_small_subun'),(1737035,'gnl|CDD|274609',_binary 'flagellar biosynthetic protein FlhF.  [Cellular processes, Chemotaxis and motility].','TIGR03499','FlhF'),(1749212,'gnl|CDD|270882',_binary 'Catalytic domain of the Serine/Threonine kinase, Vacuolar protein sorting-associated protein 15.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. Vps15 is a large protein consisting of an N-terminal kinase domain, a C-terminal WD-repeat containing domain, and an intermediate bridge domain that contain HEAT repeats. The kinase domain is necessary for the signaling functions of Vps15. Human Vps15 was previously called p150. It associates and regulates Vps34, also called Class III phosphoinositide 3-kinase (PI3K), which catalyzes the phosphorylation of D-myo-phosphatidylinositol (PtdIns). Vps34 is the only PI3K present in yeast. It plays an important role in the regulation of protein and vesicular trafficking and sorting, autophagy, trimeric G-protein signaling, and phagocytosis. The Vps15 subfamily is part of a larger superfamily that includes the catalytic domains of other protein STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and PI3K.','cd13980','STKc_Vps15'),(1754007,'gnl|CDD|223914',_binary 'Membrane-fusion protein [Cell envelope biogenesis, outer membrane].','COG0845','AcrA'),(1754073,'gnl|CDD|311792',_binary 'Pyridine nucleotide-disulphide oxidoreductase.  This family includes both class I and class II oxidoreductases and also NADH oxidases and peroxidases. This domain is actually a small NADH binding domain within a larger FAD binding domain.','pfam07992','Pyr_redox_2'),(1759599,'gnl|CDD|238956',_binary 'tRNA methyl transferase. This family represents tRNA(5-methylaminomethyl-2-thiouridine)-methyltransferase which is involved in the biosynthesis of the modified nucleoside 5-methylaminomethyl-2-thiouridine present in the wobble position of some tRNAs. This family of enzyme only presents in bacteria and eukaryote. The  archaeal counterpart of this enzyme performs same function, but is completely unrelated in sequence.','cd01998','tRNA_Me_trans'),(1759601,'gnl|CDD|238951',_binary 'This is a subfamily of Adenine nucleotide alpha hydrolases superfamily.Adeninosine nucleotide alpha hydrolases superfamily  includes N type ATP PPases and ATP sulphurylases. It forms a apha/beta/apha fold which  binds to Adenosine group.  This subfamily   of proteins is predicted to  bind ATP. This domainhas  a strongly conserved motif SGGKD at the N terminus.','cd01993','Alpha_ANH_like_II'),(1760682,'gnl|CDD|318854',_binary 'Glutamic acid/alanine-rich protein of Trypanosoma.  GARP, or glutamic acid/alanine-rich protein, is one of a subset of major surface molecules on Trypanosoma species. They are all surface-orientated, immunodominant, and highly charged. GARP is interesting as ts expression coincides with the loss and gain of variant surface glycoprotein (VSG) molecules in the tsetse vector. It has an extended helical bundle structure that is homologous to the core surface structure of VSG, suggesting that it might replace the bloodstream VSG as the trypanosomes differentiate inside the tsetse vector after a blood-meal.','pfam16731','GARP'),(1764718,'gnl|CDD|280612',_binary 'Met-10+ like-protein.  The methionine-10 mutant allele of N. crassa codes for a protein of unknown function. However, homologous proteins have been found in yeast, suggesting this protein may be involved in methionine biosynthesis, transport and/or utilisation.','pfam02475','Met_10'),(1768348,'gnl|CDD|311593',_binary 'AAA domain (Cdc48 subfamily).  This Pfam entry includes some of the AAA proteins not detected by the pfam00004 model.','pfam07724','AAA_2'),(1770197,'gnl|CDD|223319',_binary 'Histidinol phosphatase and related phosphatases [Amino acid transport and metabolism].','COG0241','HisB'),(1770212,'gnl|CDD|319853',_binary 'Uncharacterized family of the the haloacid dehalogenase-like superfamily, similar to the uncharacterized protein Bacillus subtilis YqeG.  The haloacid dehalogenase-like (HAD) hydrolases are a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. Members are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.','cd16416','HAD_BsYqeG-like'),(1771889,'gnl|CDD|310592',_binary 'Domain of unknown function (DUF947).  Family of eukaryotic proteins with unknown function.','pfam06102','DUF947'),(1772618,'gnl|CDD|184523',_binary 'cell division suppressor protein YneA; Provisional.','PRK14125','PRK14125'),(1773392,'gnl|CDD|306511',_binary 'ABC transporter.  ABC transporters for a large family of proteins responsible for translocation of a variety of compounds across biological membranes. ABC transporters are the largest family of proteins in many completely sequenced bacteria. ABC transporters are composed of two copies of this domain and two copies of a transmembrane domain pfam00664. These four domains may belong to a single polypeptide as in CFTR, or belong in different polypeptide chains.','pfam00005','ABC_tran'),(1773398,'gnl|CDD|237421',_binary 'cytochrome c biogenesis protein CcmA; Provisional.','PRK13539','PRK13539'),(1773447,'gnl|CDD|224617',_binary 'Putative periplasmic protein kinase ArgK and related GTPases of G3E family [Amino acid transport and metabolism].','COG1703','ArgK'),(1777389,'gnl|CDD|315885',_binary 'HTH domain.  This domain is a helix-turn-helix domain that is likely to act as a DNA-binding domain.','pfam13309','HTH_22'),(1782577,'gnl|CDD|277318',_binary 'purple acid phosphatases of the metallophosphatase superfamily, metallophosphatase domain.  Purple acid phosphatases (PAPs) belong to a diverse family of binuclear metallohydrolases that have been identified and characterized in plants, animals, and fungi. PAPs contain a binuclear metal center and their characteristic pink or purple color derives from a charge-transfer transition between a tyrosine residue and a chromophoric ferric ion within the binuclear center. PAPs catalyze the hydrolysis of a wide range of activated phosphoric acid mono- and di-esters and anhydrides. PAPs are distinguished from the other phosphatases by their insensitivity to L-(+) tartrate inhibition and are therefore also known as tartrate resistant acid phosphatases (TRAPs). While only a few copies of PAP-like genes are present in mammalian and fungal genomes, multiple copies are present in plant genomes. PAPs belong to the metallophosphatase (MPP) superfamily. MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases). The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets. This domain is thought to allow for productive metal coordination.','cd00839','MPP_PAPs'),(1782675,'gnl|CDD|314193',_binary 'Protein of unknown function (DUF2829).  This is a uncharacterized family of proteins found in bacteria and bacteriphages.','pfam11195','DUF2829'),(1782775,'gnl|CDD|132933',_binary 'Signal peptide peptidase A (SppA) 36K type, a serine protease, has catalytic Ser-Lys dyad.  Signal peptide peptidase A (SppA; Peptidase S49; Protease IV) 36K type: SppA is found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Members in this subfamily are all bacterial and include sohB peptidase and protein C. These are sometimes referred to as 36K type since they contain only one domain, unlike E. coli SppA that also contains an amino-terminal domain. Site-directed mutagenesis and sequence analysis have shown these SppAs to be serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases.','cd07022','S49_Sppa_36K_type'),(1783121,'gnl|CDD|235139',_binary 'spermidine synthase; Provisional.','PRK03612','PRK03612'),(1783122,'gnl|CDD|234708',_binary '16S ribosomal RNA methyltransferase KsgA/Dim1 family protein; Reviewed.','PRK00274','ksgA'),(1783123,'gnl|CDD|279850',_binary 'Spermine/spermidine synthase.  Spermine and spermidine are polyamines. This family includes spermidine synthase that catalyzes the fifth (last) step in the biosynthesis of spermidine from arginine, and spermine synthase.','pfam01564','Spermine_synth'),(1784856,'gnl|CDD|224308',_binary 'Archaeal/vacuolar-type H+-ATPase subunit E [Energy production and conversion].','COG1390','NtpE'),(1784890,'gnl|CDD|310726',_binary 'Major fimbrial subunit protein (FimA).  This family consists of several Porphyromonas gingivalis major fimbrial subunit protein (FimA) sequences. Fimbriae of Porphyromonas gingivalis, a periodontopathogen, play an important role in its adhesion to and invasion of host cells. The fimA genes encoding fimbrillin (FimA), a subunit protein of fimbriae, have been classified into five types, types I to V, based on nucleotide sequences. It has been found that type II FimA can bind to epithelial cells most efficiently through specific host receptors. Human dental plaque is a multispecies microbial biofilm that is associated with two common oral diseases, dental caries and periodontal disease. There is an inter-species contact-dependent communication system between P. gingivalis and S. cristatus that involces the Arc-A enzyme.','pfam06321','P_gingi_FimA'),(1785001,'gnl|CDD|319095',_binary 'zinc-ribbon family.  This zinc-ribbon region is found on a set of largely microsporidia-specific proteins.','pfam17032','zinc_ribbon_15'),(1785154,'gnl|CDD|163686',_binary 'Mpr1p, Pad1p N-terminal (MPN) domains.  MPN (also known as Mov34, PAD-1, JAMM, JAB, MPN+) domains are found in the N-terminal termini of proteins with a variety of functions; they are components of the proteasome regulatory subunits, the signalosome (CSN), eukaryotic translation initiation factor 3 (eIF3) complexes, and regulators of transcription factors.  These domains are isopeptidases that release ubiquitin from ubiquitinated proteins (thus having deubiquitinating (DUB) activity) that are tagged for degradation. Catalytically active MPN domains contain a metalloprotease signature known as the JAB1/MPN/Mov34 metalloenzyme (JAMM) motif. For example, Rpn11 (also known as POH1 or PSMD14), a subunit of the 19S proteasome lid is involved in the ATP-dependent degradation of ubiquitinated proteins, contains the conserved JAMM motif involved in zinc ion coordination. Poh1 is a regulator of c-Jun, an important regulator of cell proliferation, differentiation, survival and death. JAB1 is a component of the COP9 signalosome (CSN), a regulatory particle of the ubiquitin (Ub)/26S proteasome system occurring in all eukaryotic cells; it cleaves the ubiquitin-like protein NEDD8 from the cullin subunit of the SCF (Skp1, Cullins, F-box proteins) family of E3 ubiquitin ligases. AMSH (associated molecule with the SH3 domain of STAM, also known as STAMBP), a member of JAMM/MPN+ deubiquitinases (DUBs), specifically cleaves Lys 63-linked polyubiquitin (poly-Ub) chains, thus facilitating the recycling and subsequent trafficking of receptors to the cell surface.  Similarly, BRCC36, part of the nuclear complex that includes BRCA1 protein and is targeted to DNA damage foci after irradiation, specifically disassembles K63-linked polyUb. BRCC36 is aberrantly expressed in sporadic breast tumors, indicative of a potential role in the pathogenesis of the disease. Some variants of the JAB1/MPN domains lack key residues in their JAMM motif and are unable to coordinate a metal ion. Comparisons of key catalytic and metal binding residues explain why the MPN-containing proteins Mov34/PSMD7, Rpn8, CSN6, Prp8p, and the translation initiation factor 3 subunits f (p47) and h (p40) do not show catalytic isopeptidase activity. It has been proposed that the MPN domain in these proteins has a primarily structural function.','cd07767','MPN'),(1785166,'gnl|CDD|310820',_binary 'Protein of unknown function (DUF1090).  This family consists of several bacterial proteins of unknown function and is known as YqjC in E. coli.','pfam06476','DUF1090'),(1785167,'gnl|CDD|313392',_binary 'rRNA-processing protein Efg1.  Efg1 is involved in rRNA processing.','pfam10153','Efg1'),(1785169,'gnl|CDD|315513',_binary 'Methyltransferase domain.  Protein in this family function as methyltransferases.','pfam12847','Methyltransf_18'),(1785233,'gnl|CDD|187604',_binary 'classical (c) SDR, subgroup 5.  These proteins are members of the classical SDR family, with a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05346','SDR_c5'),(1785249,'gnl|CDD|180446',_binary 'short chain dehydrogenase; Provisional.','PRK06180','PRK06180'),(1785251,'gnl|CDD|236372',_binary 'short chain dehydrogenase; Provisional.','PRK09072','PRK09072'),(1785257,'gnl|CDD|235739',_binary '2,3-dihydroxy-2,3-dihydrophenylpropionate dehydrogenase; Provisional.','PRK06200','PRK06200'),(1785267,'gnl|CDD|180462',_binary 'short chain dehydrogenase; Provisional.','PRK06198','PRK06198'),(1785269,'gnl|CDD|187598',_binary 'human 17-beta-hydroxysteroid dehydrogenase XI-like, classical (c) SDRs.  17-beta-hydroxysteroid dehydrogenases (17betaHSD) are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. 17betaHSD type XI, a classical SDR, preferentially converts 3alpha-adiol to androsterone but not numerous other tested steroids. This subgroup of classical SDRs also includes members identified as retinol dehydrogenases, which convert retinol to retinal, a property that overlaps with 17betaHSD activity. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05339','17beta-HSDXI-like_SDR_c'),(1785279,'gnl|CDD|180617',_binary '3-ketoacyl-(acyl-carrier-protein) reductase; Provisional.','PRK06550','fabG'),(1785281,'gnl|CDD|187595',_binary 'dihydropteridine reductase (DHPR), classical (c) SDRs.  Dihydropteridine reductase is an NAD-binding protein related to the SDRs. It converts dihydrobiopterin into tetrahydrobiopterin, a cofactor necessary in catecholamines synthesis. Dihydropteridine reductase has the YXXXK of these tyrosine-dependent oxidoreductases, but lacks the typical upstream Asn and Ser catalytic residues. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.','cd05334','DHPR_SDR_c_like'),(1785285,'gnl|CDD|181335',_binary 'short chain dehydrogenase; Validated.','PRK08264','PRK08264'),(1785288,'gnl|CDD|235910',_binary 'short chain dehydrogenase; Provisional.','PRK07024','PRK07024'),(1785290,'gnl|CDD|212492',_binary 'retinol dehydrogenase (retinol-DH), Light dependent Protochlorophyllide (Pchlide) OxidoReductase (LPOR) and related proteins, classical (c) SDRs.  Classical SDR subgroup containing retinol-DHs, LPORs, and related proteins. Retinol is processed by a medium chain alcohol dehydrogenase followed by retinol-DHs. Pchlide reductases act in chlorophyll biosynthesis. There are distinct enzymes that catalyze Pchlide reduction in light or dark conditions. Light-dependent reduction is via an NADP-dependent SDR, LPOR. Proteins in this subfamily share the glycine-rich NAD-binding motif of the classical SDRs, have a partial match to the canonical active site tetrad, but lack the typical active site Ser. This subgroup includes the human proteins: retinol dehydrogenase -12, -13 ,and -14, dehydrogenase/reductase SDR family member (DHRS)-12 , -13 and -X (a DHRS on chromosome X), and WWOX (WW domain-containing oxidoreductase), as well as a Neurospora crassa SDR encoded by the blue light inducible bli-4 gene. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.','cd05327','retinol-DH_like_SDR_c_lik'),(1785291,'gnl|CDD|181295',_binary 'gluconate 5-dehydrogenase; Provisional.','PRK08213','PRK08213'),(1785294,'gnl|CDD|236099',_binary 'short chain dehydrogenase; Provisional.','PRK07791','PRK07791'),(1785445,'gnl|CDD|237100',_binary '3-hydroxybutyrate dehydrogenase; Provisional.','PRK12429','PRK12429'),(1785447,'gnl|CDD|226476',_binary 'Short-chain dehydrogenase involved in D-alanine esterification of lipoteichoic acid and wall teichoic acid (D-alanine transfer protein) [Cell envelope biogenesis, outer membrane].','COG3967','DltE'),(1787483,'gnl|CDD|307424',_binary 'Putative undecaprenyl diphosphate synthase.  Previously known as uncharacterized protein family UPF0015, a single member of this family has been identified as an undecaprenyl diphosphate synthase.','pfam01255','Prenyltransf'),(1787484,'gnl|CDD|223099',_binary 'Undecaprenyl pyrophosphate synthase [Lipid metabolism].','COG0020','UppS'),(1787485,'gnl|CDD|184840',_binary 'undecaprenyl pyrophosphate synthase; Provisional.','PRK14830','PRK14830'),(1787505,'gnl|CDD|239034',_binary 'This family contains the NifH (iron protein) of nitrogenase, L subunit (BchL/ChlL) of the  protochlorophyllide reductase and the BchX subunit of the Chlorophyllide reductase. Members of this family use energey from ATP hydrolysis and transfer electrons through a Fe4-S4 cluster to other subunit for reduction of substrate.','cd02117','NifH_like'),(1787506,'gnl|CDD|238996',_binary 'NifH gene encodes component II (iron protein) of nitrogenase. Nitrogenase is responsible for the biological nitrogen fixation, i.e. reduction of molecular nitrogen to ammonia. NifH consists of two oxygen-sensitive metallosulfur proteins: the mollybdenum-iron (alternatively, vanadium-iron or iron-iron) protein (commonly referred to as component 1), and the iron protein (commonly referred to as component 2). The iron protein is a homodimer, with an Fe4S4 cluster bound between the subunits and two ATP-binding domains. It supplies energy by ATP hydrolysis, and transfers electrons from reduced ferredoxin or flavodoxin to component 1 for the reduction of molecular nitrogen to ammonia.','cd02040','NifH'),(1787507,'gnl|CDD|224267',_binary 'Nitrogenase subunit NifH (ATPase) [Inorganic ion transport and metabolism].','COG1348','NifH'),(1787509,'gnl|CDD|274392',_binary 'exopolysaccharide/PEP-CTERM locus tyrosine autokinase.  Members of this protein family are related to a known protein-tyrosine autokinase and to numerous homologs from exopolysaccharide biosynthesis region proteins, many of which are designated as chain length determinants. Most members of this family contain a short region, immediately C-terminal to the region modeled here, with an abundance of Tyr residues. These C-terminal tyrosine residues are likely to be autophosphorylation sites. Some members of this family are fusion proteins.','TIGR03018','pepcterm_TyrKin'),(1787510,'gnl|CDD|310869',_binary 'Cellulose biosynthesis protein BcsQ.  This is a family of bacterial proteins involved in cellulose biosynthesis. (Roemling U. and Galperin M.Y. \'Bacterial cellulose biosynthesis. Diversity of operons and subunits\' (manuscript in preparation)). A second component of the extracellular matrix of the multicellular morphotype (rdar) of Salmonella typhimurium and Escherichia coli is cellulose. The family does contain a P-loop sequence motif suggesting a nucleotide binding function, but this has not been confirmed.','pfam06564','CBP_BcsQ'),(1787511,'gnl|CDD|237293',_binary 'protochlorophyllide reductase iron-sulfur ATP-binding protein; Provisional.','PRK13185','chlL'),(1787518,'gnl|CDD|237509',_binary 'hypothetical protein; Provisional.','PRK13794','PRK13794'),(1787527,'gnl|CDD|307718',_binary 'FtsJ-like methyltransferase.  This family consists of FtsJ from various bacterial and archaeal sources FtsJ is a methyltransferase, but actually has no effect on cell division. FtsJ\'s substrate is the 23S rRNA. The 1.5 A crystal structure of FtsJ in complex with its cofactor S-adenosylmethionine revealed that FtsJ has a methyltransferase fold. This family also includes the N terminus of flaviviral NS5 protein. It has been hypothesized that the N-terminal domain of NS5 is a methyltransferase involved in viral RNA capping.','pfam01728','FtsJ'),(1787540,'gnl|CDD|223715',_binary 'Signal transduction histidine kinase [Signal transduction mechanisms].','COG0642','BaeS'),(1787541,'gnl|CDD|226951',_binary 'Signal transduction histidine kinase [Signal transduction mechanisms].','COG4585','COG4585'),(1787585,'gnl|CDD|307486',_binary 'Glycosyl hydrolases family 6.','pfam01341','Glyco_hydro_6'),(1787608,'gnl|CDD|236608',_binary 'RNA polymerase sigma factor RpoE; Provisional.','PRK09652','PRK09652'),(1787609,'gnl|CDD|183566',_binary 'RNA polymerase sigma factor; Provisional.','PRK12513','PRK12513'),(1787610,'gnl|CDD|183384',_binary 'RNA polymerase sigma factor; Provisional.','PRK11924','PRK11924'),(1787611,'gnl|CDD|236602',_binary 'RNA polymerase sigma factor SigX; Provisional.','PRK09639','PRK09639'),(1787613,'gnl|CDD|182015',_binary 'RNA polymerase sigma factor SigK; Reviewed.','PRK09646','PRK09646'),(1787614,'gnl|CDD|236601',_binary 'RNA polymerase sigma factor SigZ; Provisional.','PRK09637','PRK09637'),(1787617,'gnl|CDD|182010',_binary 'RNA polymerase sigma factor SigY; Reviewed.','PRK09638','PRK09638'),(1787619,'gnl|CDD|274379',_binary 'RNA polymerase sigma-70 factor, Rhodopirellula/Verrucomicrobium family.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are abundantly found in the species Rhodopirellula baltica (11), and Verrucomicrobium spinosum (16) and to a lesser extent in Gemmata obscuriglobus (2).','TIGR02989','Sig-70_gvs1'),(1787620,'gnl|CDD|238044',_binary 'helix_turn_helix, cAMP Regulatory protein C-terminus; DNA binding domain of prokaryotic regulatory proteins belonging to the catabolite activator protein family.','cd00092','HTH_CRP'),(1787623,'gnl|CDD|235530',_binary 'RNA polymerase sigma factor; Reviewed.','PRK05602','PRK05602'),(1787627,'gnl|CDD|236607',_binary 'RNA polymerase sigma factor SigD; Reviewed.','PRK09648','PRK09648'),(1787629,'gnl|CDD|274375',_binary 'RNA polymerase sigma-70 factor, Planctomycetaceae-specific subfamily 1.  This group of sigma factors are members of the sigma-70 family (TIGR02937) and are apparently found only in the Planctomycetaceae family including the genuses Gemmata and Pirellula (in which seven sequences are found).','TIGR02984','Sig-70_plancto1'),(1787631,'gnl|CDD|236367',_binary 'RNA polymerase factor sigma-70; Validated.','PRK09047','PRK09047'),(1787632,'gnl|CDD|236604',_binary 'RNA polymerase sigma factor SigM; Reviewed.','PRK09643','PRK09643'),(1787634,'gnl|CDD|237024',_binary 'RNA polymerase sigma factor; Provisional.','PRK11922','PRK11922'),(1787643,'gnl|CDD|225310',_binary 'Predicted membrane-associated trancriptional regulator    [Transcription].','COG2512','COG2512'),(1798760,'gnl|CDD|235319',_binary 'ATP-dependent helicase HepA; Validated.','PRK04914','PRK04914'),(1798788,'gnl|CDD|283628',_binary 'Laminin Domain II.  It has been suggested that the domains I and II from laminin A, B1 and B2 may come together to form a triple helical coiled-coil structure.','pfam06009','Laminin_II'),(1801796,'gnl|CDD|236846',_binary 'sensory histidine kinase CreC; Provisional.','PRK11100','PRK11100'),(1801803,'gnl|CDD|237551',_binary 'alanyl-tRNA synthetase; Provisional.','PRK13902','alaS'),(1801806,'gnl|CDD|310622',_binary 'Protein of unknown function (DUF972).  This family consists of several hypothetical bacterial sequences. The function of this family is unknown.','pfam06156','DUF972'),(1801807,'gnl|CDD|274722',_binary 'alanyl-tRNA synthetase.  This family of alanyl-tRNA synthetases is limited to the archaea, and is a subset of those sequences identified by the model pfam07973 covering the second additional domain (SAD) of alanyl and threonyl tRNA synthetases .','TIGR03683','A-tRNA_syn_arch'),(1801871,'gnl|CDD|183876',_binary 'DNA replication intiation control protein YabA; Reviewed.','PRK13169','PRK13169'),(1809034,'gnl|CDD|214795',_binary 'Dystroglycan-type cadherin-like domains.  Cadherin-homologous domains present in metazoan dystroglycans and alpha/epsilon sarcoglycans, yeast Axl2p and in a very large protein from magnetotactic bacteria. Likely to bind calcium ions.','smart00736','CADG'),(1809036,'gnl|CDD|235401',_binary 'translation initiation factor IF-2; Validated.','PRK05306','infB'),(1809099,'gnl|CDD|316839',_binary 'Beta protein.  This family includes the beta protein from Bacteriophage T4. Beta protein prevents the gop protein from killing the bacterial host cell.','pfam14350','Beta_protein'),(1810052,'gnl|CDD|133412',_binary 'Helix-Turn-Helix DNA binding domain of the CadR- and PbrR-like transcription regulators.  Helix-turn-helix (HTH) CadR- and PbrR-like transcription regulators. CadR and PbrR regulate expression of the cadmium and lead resistance operons, respectively. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines which comprise a putative metal binding site. Some members in this group have a histidine-rich C-terminal extension. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.','cd04785','HTH_CadR-PbrR-like'),(1810227,'gnl|CDD|308788',_binary 'Endonuclease/Exonuclease/phosphatase family.  This large family of proteins includes magnesium dependent endonucleases and a large number of phosphatases involved in intracellular signalling. This family includes: AP endonuclease proteins EC:4.2.99.18, DNase I proteins EC:3.1.21.1, Synaptojanin an inositol-1,4,5-trisphosphate phosphatase EC:3.1.3.56, Sphingomyelinase EC:3.1.4.12, and Nocturnin.','pfam03372','Exo_endo_phos'),(1810228,'gnl|CDD|197306',_binary 'Exonuclease-Endonuclease-Phosphatase (EEP) domain superfamily.  This large superfamily includes the catalytic domain (exonuclease/endonuclease/phosphatase or EEP domain) of a diverse set of proteins including the ExoIII family of apurinic/apyrimidinic (AP) endonucleases, inositol polyphosphate 5-phosphatases (INPP5), neutral sphingomyelinases (nSMases), deadenylases (such as the vertebrate circadian-clock regulated nocturnin), bacterial cytolethal distending toxin B (CdtB), deoxyribonuclease 1 (DNase1), the endonuclease domain of the non-LTR retrotransposon LINE-1, and related domains. These diverse enzymes share a common catalytic mechanism of cleaving phosphodiester bonds; their substrates range from nucleic acids to phospholipids and perhaps proteins.','cd08372','EEP'),(1810230,'gnl|CDD|197313',_binary 'Streptococcus agalactiae RgfB, part of a putative two component signal transduction system, and related proteins.  This family includes Streptococcus agalactiae RgfB (for regulator of fibrinogen binding) and related proteins. The function of RgfB is unknown. It is part of a putative two component signal transduction system designated rgfBDAC (the rgf locus was identified in a screen for mutants of Streptococcus agalactiae with altered binding to fibrinogen). RgfA,-C,and -D do not belong to this superfamily: rgfA encodes a putative response regulator, and rgfC, a putative histidine kinase. All four genes are co-transcribed, and may be involved in regulating expression of bacterial cell surface components. This family belongs to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds.','cd09079','RgfB-like'),(1810232,'gnl|CDD|197318',_binary 'Exonuclease-Endonuclease-Phosphatase (EEP) domain superfamily; uncharacterized family 2.  This family of uncharacterized proteins belongs to a superfamily that includes the catalytic domain (exonuclease/endonuclease/phosphatase, EEP, domain) of a diverse set of proteins including the ExoIII family of apurinic/apyrimidinic (AP) endonucleases, inositol polyphosphate 5-phosphatases (INPP5), neutral sphingomyelinases (nSMases), deadenylases (such as the vertebrate circadian-clock regulated nocturnin), bacterial cytolethal distending toxin B (CdtB), deoxyribonuclease 1 (DNase1), the endonuclease domain of the non-LTR retrotransposon LINE-1, and related domains. These diverse enzymes share a common catalytic mechanism of cleaving phosphodiester bonds; their substrates range from nucleic acids to phospholipids and perhaps, proteins.','cd09084','EEP-2'),(1810234,'gnl|CDD|197312',_binary 'Neutral sphingomyelinases (nSMase) catalyze the hydrolysis of sphingomyelin in biological membranes to ceramide and phosphorylcholine.  Sphingomyelinases (SMase) are phosphodiesterases that catalyze the hydrolysis of sphingomyelin to ceramide and phosphorylcholine. Eukaryotic SMases have been classified according to their pH optima and are known as acid SMase, alkaline SMase, and neutral SMase (nSMase). Eukaryotic proteins in this family are nSMases, and are activated by a variety of stress-inducing agents such as cytokines or UV radiation. Ceramides and other metabolic derivatives, including sphingosine, are lipid \'second messenger\' molecules that participate in the regulation of stress-induced cellular responses, including cell death, adhesion, differentiation, and proliferation. Bacterial neutral SMases, which also belong to this domain family, are secreted proteins that act as membrane-damaging virulence factors. They promote colonization of the host tissue. This family belongs to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds.','cd09078','nSMase'),(1810236,'gnl|CDD|225565',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3021','COG3021'),(1810308,'gnl|CDD|235079',_binary 'cyanate hydratase; Validated.','PRK02866','PRK02866'),(1810844,'gnl|CDD|198372',_binary 'uncharacterized xylulose kinase-like proteins; a subgroup of the FGGY family of carbohydrate kinases.  This subgroup is composed of uncharacterized bacterial and archaeal xylulose kinases-like proteins with similarity to bacterial D-xylulose kinases (XK, also known as xylulokinase; EC 2.7.1.17), which catalyze the rate-limiting step in the ATP-dependent phosphorylation of D-xylulose to produce D-xylulose 5-phosphate (X5P) and ADP. The presence of Mg2+ or Mn2+ is required for catalytic activity. D-XK exists as a dimer with an active site that lies at the interface between the N- and C-terminal domains. This model includes both the N-terminal domain, which adopts a ribonuclease H-like fold, and the structurally related C-terminal domain. Members of this subgroup belong to the FGGY family of carbohydrate kinases.','cd07804','FGGY_XK_like_1'),(1811870,'gnl|CDD|173847',_binary 'Short chain acyl-CoA dehydrogenases and eukaryotic short/branched chain acyl-CoA dehydrogenases.  Short chain acyl-CoA dehydrogenase (SCAD). SCAD is a mitochondrial beta-oxidation enzyme. It catalyzes the alpha,beta dehydrogenation of the corresponding trans-enoyl-CoA by FAD, which becomes reduced. The reduced form of SCAD is reoxidized in the oxidative half-reaction by electron-transferring flavoprotein (ETF), from which the electrons are transferred to the mitochondrial respiratory chain coupled with ATP synthesis.  This subgroup also contains the eukaryotic short/branched chain acyl-CoA dehydrogenase(SBCAD), the bacterial butyryl-CoA dehydorgenase(BCAD) and 2-methylbutyryl-CoA dehydrogenase, which is involved in isoleucine catabolism.  These enzymes are homotetramers.','cd01158','SCAD_SBCAD'),(1811952,'gnl|CDD|275211',_binary 'primosomal replication protein PriB.  Members of this protein family are primosomal replication protein N (PriB), virtually always encoded between rpsF (ribosomal protein S6) and rpsR (ribosomal protein S18). Note that only this short form, as found primarily in the gamma-Proteobacteria, of the single-stranded DNA binding protein family (see model TIGR00621) may partner with PriA and be involved in priming for re-initiation of replication. [DNA metabolism, DNA replication, recombination, and repair].','TIGR04418','PriB_gamma'),(1818839,'gnl|CDD|279152',_binary 'D-alanyl-D-alanine carboxypeptidase.','pfam00768','Peptidase_S11'),(1818847,'gnl|CDD|226149',_binary 'Hydroxypyruvate isomerase [Carbohydrate transport and metabolism].','COG3622','Hfi'),(1818971,'gnl|CDD|237951',_binary 'two component system sensor kinase SsrA; Provisional.','PRK15347','PRK15347'),(1818991,'gnl|CDD|227938',_binary 'PPE-repeat proteins [Cell motility and secretion].','COG5651','COG5651'),(1818992,'gnl|CDD|312152',_binary 'Fungal domain of unknown function (DUF1750).  This is a fungal domain of unknown function.','pfam08549','SWI-SNF_Ssr4'),(1819342,'gnl|CDD|226136',_binary 'Predicted transcriptional regulators containing the CopG/Arc/MetJ DNA-binding domain [Transcription].','COG3609','COG3609'),(1819923,'gnl|CDD|274782',_binary 'marine proteobacterial sortase target protein.  Members of this protein family are restricted to the Proteobacteria. Each contains a C-terminal sortase-recognition motif, transmembrane domain, and basic residues cluster at the the C-terminus, and is encoded adjacent to a sortase gene. This protein is frequently the only sortase target in its genome, which is as unusual its occurrence in Gram-negative rather than Gram-positive genomes. Many bacteria with this system are marine. In addition to the LPXTG signal, members carry a vault protein inter-alpha-trypsin inhibitor domain (pfam08487) and a von Willebrand factor type A domain (pfam00092).','TIGR03788','marine_srt_targ'),(1819929,'gnl|CDD|238744',_binary 'VWA BatA type: Von Willebrand factor type A (vWA) domain was originally found in the blood coagulation protein von Willebrand factor (vWF). Typically, the vWA domain is made up of approximately 200 amino acid residues folded into a classic a/b para-rossmann type of fold. The vWA domain, since its discovery, has drawn great interest because of its widespread occurrence and its involvement in a wide variety of important cellular functions. These include basal membrane formation, cell migration, cell differentiation, adhesion, haemostasis, signaling, chromosomal stability, malignant transformation and in immune defenses. In integrins these domains form heterodimers while in vWF it forms multimers. There are different interaction surfaces of this domain as seen by the various molecules it complexes with. Ligand binding in most cases is mediated by the presence of a metal ion dependent adhesion site termed as the MIDAS motif that is a characteristic feature of most, if not all A domains. Members of this subgroup are bacterial in origin. They are typified by the presence of a MIDAS motif.','cd01467','vWA_BatA_type'),(1819960,'gnl|CDD|259797',_binary 'core domain of the SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes proteins similar to stomatin, prohibitin, flotillin, HflK/C (SPFH) and podocin. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Many superfamily members are associated with lipid rafts. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Microdomains formed from flotillin proteins may in addition be dynamic units with their own regulatory functions. Flotillins have been implicated in signal transduction, vesicle trafficking, cytoskeleton rearrangement and are known to interact with a variety of proteins. Stomatin interacts with and regulates members of the degenerin/epithelia Na+ channel family in mechanosensory cells of Caenorhabditis elegans and vertebrate neurons, and participates in trafficking of Glut1 glucose transporters. Prohibitin may act as a chaperone for the stabilization of mitochondrial proteins. Prokaryotic HflK/C plays a role in the decision between lysogenic and lytic cycle growth during lambda phage infection. Flotillins have been implicated in the progression of prion disease, in the pathogenesis of neurodegenerative diseases such as Parkinson\'s and Alzheimer\'s disease, and in cancer invasion and metastasis. Mutations in the podocin gene give rise to autosomal recessive steroid resistant nephritic syndrome.','cd02106','SPFH_like'),(1819968,'gnl|CDD|316750',_binary 'Domain of unknown function (DUF4349).  This family of proteins is found in bacteria and archaea. Proteins in this family are typically between 282 and 353 amino acids in length. There is a single completely conserved residue D that may be functionally important.','pfam14257','DUF4349'),(1820122,'gnl|CDD|317619',_binary 'Coiled-coil domain-containing protein 66.  This protein family, named Coiled-coil domain-containing protein 66 (CCDC) refers to a protein domain found in eukaryotes, and is approximately 160 amino acids in length. CCDC66 protein is detected mainly in the inner segments of photoreceptors in many vertebrates including mice and humans. It has been found in dogs, that a mutation in the CCDC66 gene causes generalized progressive retinal atrophy (gPRA). This shows that the protein encoded for by this gene is vital for healthy vision and guards against photoreceptor cell degeneration. The structure of CCDC66 proteins includes a heptad repeat pattern which contains at least one coiled-coil domain. There are at least two or more alpha-helices which form a cable-like structure.','pfam15236','CCDC66'),(1820128,'gnl|CDD|153279',_binary 'The Bin/Amphiphysin/Rvs (BAR) domain of Rich-like Rho GTPase Activating Proteins.  BAR domains are dimerization, lipid binding and curvature sensing modules found in many different proteins with diverse functions. This subfamily is composed of Rho and Rac GTPase activating proteins (GAPs) with similarity to GAP interacting with CIP4 homologs proteins (Rich). Members contain an N-terminal BAR domain, followed by a Rho GAP domain, and a C-terminal prolin-rich region. Vertebrates harbor at least three Rho GAPs in this subfamily including Rich1, Rich2, and SH3-domain binding protein 1 (SH3BP1). Rich1 and Rich2 play complementary roles in the establishment and maintenance of cell polarity. Rich1 is a Cdc42- and Rac-specific GAP that binds to polarity proteins through the scaffold protein angiomotin and plays a role in maintaining the integrity of tight junctions. Rich2 is a Rac GAP that interacts with CD317 and plays a role in actin cytoskeleton organization and the maintenance of microvilli in polarized epithelial cells. SH3BP1 is a Rac GAP that inhibits Rac-mediated platelet-derived growth factor (PDGF)-induced membrane ruffling. BAR domains form dimers that bind to membranes, induce membrane bending and curvature, and may also be involved in protein-protein interactions. The BAR domain of Rich1 has been shown to form oligomers, bind membranes and induce membrane tubulation.','cd07595','BAR_RhoGAP_Rich-like'),(1820139,'gnl|CDD|309164',_binary 'Outer membrane protein (OmpH-like).  This family includes outer membrane proteins such as OmpH among others. Skp (OmpH) has been characterized as a molecular chaperone that interacts with unfolded proteins as they emerge in the periplasm from the Sec translocation machinery.','pfam03938','OmpH'),(1820144,'gnl|CDD|316426',_binary 'Domain of unknown function (DUF4207).  This family is found in eukaryotes; it has several conserved tryptophan residues. The function is not known.','pfam13904','DUF4207'),(1822298,'gnl|CDD|310761',_binary 'CDK-activating kinase assembly factor MAT1.  MAT1 is an assembly/targeting factor for cyclin-dependent kinase-activating kinase (CAK), which interacts with the transcription factor TFIIH. The domain found to the N-terminal side of this domain is a C3HC4 RING finger.','pfam06391','MAT1'),(1822412,'gnl|CDD|236090',_binary 'DNA polymerase III subunits gamma and tau; Validated.','PRK07764','PRK07764'),(1823042,'gnl|CDD|270693',_binary 'Catalytic domain of AGC family Serine/Threonine Kinases.  STKs catalyze the transfer of the gamma-phosphoryl group from ATP to serine/threonine residues on protein substrates. AGC kinases regulate many cellular processes including division, growth, survival, metabolism, motility, and differentiation. Many are implicated in the development of various human diseases. Members of this family include cAMP-dependent Protein Kinase (PKA), cGMP-dependent Protein Kinase (PKG), Protein Kinase C (PKC), Protein Kinase B (PKB), G protein-coupled Receptor Kinase (GRK), Serum- and Glucocorticoid-induced Kinase (SGK), and 70 kDa ribosomal Protein S6 Kinase (p70S6K or S6K), among others. AGC kinases share an activation mechanism based on the phosphorylation of up to three sites: the activation loop (A-loop), the hydrophobic motif (HM) and the turn motif. Phosphorylation at the A-loop is required of most AGC kinases, which results in a disorder-to-order transition of the A-loop. The ordered conformation results in the access of substrates and ATP to the active site. A subset of AGC kinases with C-terminal extensions containing the HM also requires phosphorylation at this site. Phosphorylation at the HM allows the C-terminal extension to form an ordered structure that packs into the hydrophobic pocket of the catalytic domain, which then reconfigures the kinase into an active bi-lobed state. In addition, growth factor-activated AGC kinases such as PKB, p70S6K, RSK, MSK, PKC, and SGK, require phosphorylation at the turn motif (also called tail or zipper site), located N-terminal to the HM at the C-terminal extension. The AGC family is part of a larger superfamily that includes the catalytic domains of other STKs, protein tyrosine kinases, RIO kinases, aminoglycoside phosphotransferase, choline kinase, and Phosphoinositide 3-Kinase.','cd05123','STKc_AGC'),(1823815,'gnl|CDD|316912',_binary 'SCP1.201-like deaminase.  A member of the nucleic acid/nucleotide deaminase superfamily prototyped by Streptomyces SCP1.201. Members of this family are predicted to function as toxins in bacterial polymorphic toxin systems.','pfam14428','SCP1201-deam'),(1824545,'gnl|CDD|314181',_binary 'Protein of unknown function (DUF2968).  This family of proteins has no known function.','pfam11180','DUF2968'),(1825381,'gnl|CDD|313831',_binary 'Protein of unknown function (DUF2507).  This family is conserved in Firmicutes. The function is not known.','pfam10702','DUF2507'),(1826129,'gnl|CDD|316774',_binary 'FlxA-like protein.  This family includes FlxA from E. coli. The expression of FlxA is regulated by the FliA sigma factor, a transcription factor specific for class 3 flagellar operons. However FlxA is not required for flagellar function or formation.','pfam14282','FlxA'),(1826369,'gnl|CDD|318708',_binary 'GIT coiled-coil Rho guanine nucleotide exchange factor.  GIT-CC is the coiled-coil region of GIT (G protein-coupled receptor kinase-interacting) proteins. This coiled-coil region is the surface that associates with the equivalent binding-region on beta-PIX, or p21-activated kinase-interacting exchange factor proteins. Both GIT and PIX complex together to form a scaffold for the formation of multi-protein assemblies. On its own the GIT-CC region assembles into a parallel two-stranded CC in the asymmetric unit. Similarly the PIX coiled-coil region assembles into a trimer. At least in vitro the two regions associate together into a stable heteropentameric complex that consists of one PIX trimer and one GIT dimer.','pfam16559','GIT_CC'),(1827279,'gnl|CDD|224657',_binary 'Adenine-specific DNA methylase containing a Zn-ribbon [DNA replication, recombination, and repair].','COG1743','COG1743'),(1827565,'gnl|CDD|286561',_binary 'Bacterial type III secretion protein (HrpB7).  This entry represents proteins encoded by genes which are found in type III secretion operons in a narrow range of species including Xanthomonas, Burkholderia and Ralstonia.','pfam09486','HrpB7'),(1828389,'gnl|CDD|319040',_binary 'RNA-binding signal recognition particle 68.  SRP68 is a family that is part of the SRP or signal recognition particle complex. This complex, consisting of six proteins and a 7SL-RNA is necessary for guiding the emerging proteins designed for the membrane towards the translocation pore. SRP68 forms a stable heterodimer with SRP72, a protein with a TPR repeat. Specific RNA-binding of SRP68 is mediated by the N-terminal domain of approximately 200 residues of this family.','pfam16969','SRP68'),(1828430,'gnl|CDD|235600',_binary 'V-type ATP synthase subunit I; Validated.','PRK05771','PRK05771'),(1829076,'gnl|CDD|223970',_binary 'Predicted amidophosphoribosyltransferases [General function prediction only].','COG1040','ComFC'),(1829082,'gnl|CDD|224527',_binary 'Predicted Rossmann fold nucleotide-binding protein [General function prediction only].','COG1611','COG1611'),(1829142,'gnl|CDD|273713',_binary 'phage major tail protein, phi13 family.  This model describes a set of proteins that share low levels of sequence similarity but similar lengths and similar patterns of charged, hydrophobic, and Gly/Pro residues. All members (except one attributed to mouse embryo cDNA) belong to phage of Gram-positive bacteria. Several are identified as phage major tail proteins. Some members of this family have additional C-terminal regions of about 100 residues not included in this model. [Mobile and extrachromosomal element functions, Prophage functions].','TIGR01603','Uncharacterized_phage_rel'),(1829227,'gnl|CDD|275116',_binary 'putative metalloenzyme radical SAM/SPASM domain maturase.  This model describes a family of radical SAM/SPASM enzymes largely from the deltaproteobacteria. The family is most closely related to radical SAM enzyme family regularly in archaea in the vicinity of tungsten-containing oxidoreductases. A single member of the family in archaea may correspond to multiple tungsten enzymes, e.g. five in Pyrococcus furiosus. Therefore, the lack of a conserved gene neighborhood for members of this family in deltaprotebacteria suggests members may be involved in the maturation of multiple metalloenzymes.','TIGR04311','Radical_SAM_domain_protei'),(1829541,'gnl|CDD|274577',_binary 'VWFA-related Acidobacterial domain.  Members of this family are bacterial domains that include a region related to the von Willebrand factor type A (VWFA) domain (pfam00092). These domains are restricted to, and have undergone a large paralogous family expansion in, the Acidobacteria, including Solibacter usitatus and Acidobacterium capsulatum ATCC 51196.','TIGR03436','acidobact_VWFA'),(1829637,'gnl|CDD|274334',_binary 'sporulation integral membrane protein YtvI.  Three lines of evidence show this protein to be involved in sporulation. First, it is under control of a sporulation-specific sigma factor, sigma-E. Second, mutation leads to a sporulation defect. Third, it if found in exactly those genomes whose bacteria are capable of sporulation, except for being absent in Clostridium acetobutylicum ATCC824. This protein has extensive hydrophobic regions and is likely an integral membrane protein. [Cellular processes, Sporulation and germination].','TIGR02872','UPF0118_membrane_protein_'),(1829918,'gnl|CDD|314771',_binary 'Protein of unknown function (DUF3464).  This family of proteins are functionally uncharacterized. This protein is found in bacteria and eukaryotes. Proteins in this family are typically between 137 to 196 amino acids in length.','pfam11947','DUF3464'),(1829975,'gnl|CDD|274386',_binary 'polysaccharide chain length determinant protein, PEP-CTERM locus subfamily.  Members of this protein family belong to the family of polysaccharide chain length determinant proteins (pfam02706). All are found in species that encode the PEP-CTERM/exosortase system predicted to act in protein sorting in a number of Gram-negative bacteria, and are found near the epsH homolog that is the putative exosortase gene. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides].','TIGR03007','pepcterm_ChnLen'),(1830170,'gnl|CDD|313570',_binary 'Putative cell-wall binding lipoprotein.  YkyA is a family of proteins containing a lipoprotein signal and a hydrolase domain. It is similar to cell wall binding proteins and might also be recognisable by a host immune defense system. It is thus likely to belong to pathways important for pathogenicity.','pfam10368','YkyA'),(1830180,'gnl|CDD|313469',_binary 'Microtubule-binding protein MIP-T3.  This protein, which interacts with both microtubules and TRAF3 (tumor necrosis factor receptor-associated factor 3), is conserved from worms to humans. The N-terminal region is the microtubule binding domain and is well-conserved; the C-terminal 100 residues, also well-conserved, constitute the coiled-coil region which binds to TRAF3. The central region of the protein is rich in lysine and glutamic acid and carries KKE motifs which may also be necessary for tubulin-binding, but this region is the least well-conserved.','pfam10243','MIP-T3'),(1830182,'gnl|CDD|225288',_binary 'Uncharacterized conserved protein [Function unknown].','COG2433','COG2433'),(1830187,'gnl|CDD|226400',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3883','COG3883'),(1830189,'gnl|CDD|226889',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG4487','COG4487'),(1830191,'gnl|CDD|313034',_binary 'LRRFIP family.  LRRFIP1 is a transcriptional repressor which preferentially binds to the GC-rich consensus sequence (5\'- AGCCCCCGGCG-3\') and may regulate expression of TNF, EGFR and PDGFA. LRRFIP2 may function as activator of the canonical Wnt signalling pathway, in association with DVL3, upstream of CTNNB1/beta-catenin.','pfam09738','LRRFIP'),(1830243,'gnl|CDD|318198',_binary 'Cancer susceptibility candidate 1 N-terminus.  This presumed domain is functionally uncharacterized. This domain family is found in eukaryotes, and is approximately 200 amino acids in length. The family is found in association with pfam12366. There are two completely conserved residues (N and W) that may be functionally important.','pfam15927','Casc1_N'),(1830245,'gnl|CDD|310129',_binary 'Protein of unknown function (DUF737).  This family consists of several uncharacterized mammalian proteins of unknown function.','pfam05300','DUF737'),(1830254,'gnl|CDD|311717',_binary 'Rrp15p.  Rrp15p is required for the formation of 60S ribosomal subunits.','pfam07890','Rrp15p'),(1830561,'gnl|CDD|308381',_binary 'CDC45-like protein.  CDC45 is an essential gene required for initiation of DNA replication in S. cerevisiae, forming a complex with MCM5/CDC46. homologs of CDC45 have been identified in human, mouse and smut fungus, among others.','pfam02724','CDC45'),(1830572,'gnl|CDD|259842',_binary 'Apolipophorin-III and similar insect proteins.  Exchangeable apolipoproteins play vital roles in the transport of lipids and lipoprotein metabolism. Apolipophorin III (apoLp-III) assists in the loading of diacylglycerol, generated from triacylglycerol stores in the fat body through the action of adipokinetic hormone, into lipophorin, the hemolymph lipoprotein. ApoLp-III increases the lipid carrying capacity of lipophorin by covering the expanding hydrophobic surface resulting from diacylglycerol uptake. It plays a critical role in the transport of lipids during insect flight, and may also play a role in defense mechanisms and innate immunity.','cd13769','ApoLp-III_like'),(1830578,'gnl|CDD|314881',_binary 'Domain of unknown function (DUF3552).  This presumed domain is functionally uncharacterized. This domain is found in bacteria, archaea and eukaryotes. This domain is about 200 amino acids in length. This domain is found associated with pfam00013, pfam01966. This domain has a single completely conserved residue A that may be functionally important.','pfam12072','DUF3552'),(1830583,'gnl|CDD|309330',_binary 'IncA protein.  Chlamydia trachomatis is an obligate intracellular bacterium that develops within a parasitophorous vacuole termed an inclusion. The inclusion is non-fusogenic with lysosomes but intercepts lipids from a host cell exocytic pathway. Initiation of chlamydial development is concurrent with modification of the inclusion membrane by a set of C. trachomatis-encoded proteins collectively designated Incs. One of these Incs, IncA, is functionally associated with the homotypic fusion of inclusions. This family probably includes members of the wider Inc family rather than just IncA. Members are usually either 2 or 4TM proteins.','pfam04156','IncA'),(1831153,'gnl|CDD|235250',_binary 'replication factor C large subunit; Provisional.','PRK04195','PRK04195'),(1831245,'gnl|CDD|214510',_binary 'Repeats in polycystic kidney disease 1 (PKD1) and other proteins.  Polycystic kidney disease 1 protein contains 14 repeats, present elsewhere such as in microbial collagenases.','smart00089','PKD'),(1831261,'gnl|CDD|225153',_binary 'Membrane protein involved in the export of O-antigen and teichoic acid [General function prediction only].','COG2244','RfbX'),(1831265,'gnl|CDD|223769',_binary 'Permeases of the drug/metabolite transporter (DMT) superfamily [Carbohydrate transport and metabolism / Amino acid transport and metabolism / General function prediction only].','COG0697','RhaT'),(1831330,'gnl|CDD|313424',_binary 'Vacuolar sorting 38 and autophagy-related subunit 14.  The Atg14 or Apg14 proteins are hydrophilic proteins with a predicted molecular mass of 40.5 kDa, and have a coiled-coil motif at the N terminus region. Yeast cells with mutant Atg14 are defective not only in autophagy but also in sorting of carboxypeptidase Y (CPY), a vacuolar-soluble hydrolase, to the vacuole. Subcellular fractionation indicate that Apg14p and Apg6p are peripherally associated with a membrane structure(s). Apg14p was co-immunoprecipitated with Apg6p, suggesting that they form a stable protein complex. These results imply that Apg6/Vps30p has two distinct functions: in the autophagic process and in the vacuolar protein sorting pathway. Apg14p may be a component specifically required for the function of Apg6/Vps30p through the autophagic pathway. There are 17 auto-phagosomal component proteins which are categorized into six functional units, one of which is the AS-PI3K complex (Vps30/Atg6 and Atg14). The AS-PI3K complex and the Atg2-Atg18 complex are essential for nucleation, and the specific function of the AS-PI3K apparently is to produce phosphatidylinositol 3-phosphate (PtdIns(3)P) at the pre-autophagosomal structure (PAS). The localization of this complex at the PAS is controlled by Atg14. Autophagy mediates the cellular response to nutrient deprivation, protein aggregation, and pathogen invasion in humans, and malfunction of autophagy has been implicated in multiple human diseases including cancer. This effect seems to be mediated through direct interaction of the human Atg14 with Beclin 1 in the human phosphatidylinositol 3-kinase class III complex.','pfam10186','Atg14'),(1831563,'gnl|CDD|273491',_binary 'heme ABC exporter, ATP-binding protein CcmA.  This model describes the cyt c biogenesis protein encoded by ccmA in bacteria. An exception is, an arabidopsis protein. Quite likely this is encoded by an organelle. Bacterial c-type cytocromes are located on the periplasmic side of the cytoplasmic membrane. Several gene products encoded in a locus designated as \'ccm\' are implicated in the transport and assembly of the functional cytochrome C. This cluster includes genes: ccmA;B;C;D;E;F;G and H. The posttranslational pathway includes the transport of heme moiety, the secretion of the apoprotein and the covalent attachment of the heme with the apoprotein. The proteins ccmA and B represent an ABC transporter; ccmC and D participate in heme transfer to ccmE, which function as a periplasmic heme chaperone. The presence of ccmF, G and H is suggested to be obligatory for the final functional assembly of cytochrome c. [Protein fate, Protein and peptide secretion and trafficking, Transport and binding proteins, Other].','TIGR01189','ccmA'),(1831566,'gnl|CDD|224058',_binary 'ABC-type metal ion transport system, ATPase component [Inorganic ion transport and metabolism].','COG1135','AbcC'),(1831567,'gnl|CDD|237452',_binary 'cobalt transporter ATP-binding subunit; Provisional.','PRK13632','cbiO'),(1831585,'gnl|CDD|238182',_binary 'Usp: Universal stress protein family. The universal stress protein Usp is a small cytoplasmic bacterial protein whose expression is enhanced when the cell is exposed to stress agents. Usp enhances the rate of cell survival during prolonged exposure to such conditions, and may provide a general \'stress endurance\' activity. The crystal structure of Haemophilus influenzae Usp reveals an alpha/beta fold similar to that of the Methanococcus jannaschii MJ0577 protein, which binds ATP, athough Usp lacks ATP-binding activity.','cd00293','USP_Like'),(1831603,'gnl|CDD|239388',_binary 'The function of this protein family is unkown. The protein sequences are similar to the ArgK protein in E. coli. ArgK protein is a membrane ATPase which is required for transporting arginine, ornithine and lysine into the cells by the arginine and ornithine (AO system) and lysine, arginine and ornithine (LAO) transport systems.','cd03114','ArgK-like'),(1831719,'gnl|CDD|234645',_binary 'Holliday junction DNA helicase RuvA; Reviewed.','PRK00116','ruvA'),(1831720,'gnl|CDD|223705',_binary 'Holliday junction resolvasome, DNA-binding subunit [DNA replication, recombination, and repair].','COG0632','RuvA'),(1831722,'gnl|CDD|238013',_binary 'endonuclease III; includes endonuclease III (DNA-(apurinic or apyrimidinic site) lyase), alkylbase DNA glycosidases (Alka-family) and other DNA glycosidases.','cd00056','ENDO3c'),(1831792,'gnl|CDD|215628',_binary 'MOC1-like protein; Provisional.','PLN03196','PLN03196'),(1831794,'gnl|CDD|223701',_binary 'Predicted permease, member of the PurR regulon [General function prediction only].','COG0628','yhhT'),(1832142,'gnl|CDD|225281',_binary 'Uncharacterized protein containing a von Willebrand factor type A (vWA) domain [General function prediction only].','COG2425','COG2425'),(1832290,'gnl|CDD|237560',_binary 'rod shape-determining protein MreC; Provisional.','PRK13922','PRK13922'),(1832637,'gnl|CDD|224202',_binary 'Na+/phosphate symporter [Inorganic ion transport and metabolism].','COG1283','NptA'),(1832775,'gnl|CDD|310000',_binary 'AF-4 proto-oncoprotein.  This family consists of AF4 (Proto-oncogene AF4) and FMR2 (Fragile X E mental retardation syndrome) nuclear proteins. These proteins have been linked to human diseases such as acute lymphoblastic leukaemia and mental retardation. The family also contains a Drosophila AF4 protein homolog Lilliputian which contains an AT-hook domain. Lilliputian represents a novel pair-rule gene that acts in cytoskeleton regulation, segmentation and morphogenesis in Drosophila.','pfam05110','AF-4'),(1833030,'gnl|CDD|316301',_binary 'von Willebrand factor type A domain.','pfam13768','VWA_3'),(1833093,'gnl|CDD|176223',_binary 'Alcohol dehydrogenases of the MDR family.  The medium chain dehydrogenases/reductase (MDR)/zinc-dependent alcohol dehydrogenase-like family, which contains the zinc-dependent alcohol dehydrogenase (ADH-Zn) and related proteins, is a diverse group of proteins related to the first identified member, class I mammalian ADH.  MDRs display a broad range of activities and are distinguished from the smaller short chain dehydrogenases (~ 250 amino acids vs. the ~ 350 amino acids of the MDR).  The MDR proteins have 2 domains: a C-terminal NAD(P)-binding Rossmann fold domain of a beta-alpha form and an N-terminal catalytic domain with distant homology to GroES.  The MDR group contains a host of activities, including the founding alcohol dehydrogenase (ADH), quinone reductase, sorbitol dehydrogenase, formaldehyde dehydrogenase, butanediol DH, ketose reductase, cinnamyl reductase, and numerous others. The zinc-dependent alcohol dehydrogenases (ADHs) catalyze the NAD(P)(H)-dependent interconversion of alcohols to aldehydes or ketones.  Active site zinc has a catalytic role, while structural zinc aids in stability.  ADH-like proteins typically form dimers (typically higher plants, mammals) or tetramers (yeast, bacteria), and generally have 2 tightly bound zinc atoms per subunit. The active site zinc is coordinated by a histidine, two cysteines, and a water molecule. The second zinc seems to play a structural role, affects subunit interactions, and is typically coordinated by 4 cysteines.','cd08262','Zn_ADH8'),(1833122,'gnl|CDD|311417',_binary 'NUMOD3 motif (2 copies).  NUMOD3 is a DNA-binding motif found in homing endonucleases and related proteins. It occurs on its own or in tandem repeats in GIY-YIG (pfam01541) and HTH proteins. It constitutes a beta-turn-loop-helix subregion of the the DNA-binding domain of I-TevI homing endonuclease.','pfam07460','NUMOD3'),(1833351,'gnl|CDD|235093',_binary 'nickel responsive regulator; Provisional.','PRK02967','PRK02967'),(1833602,'gnl|CDD|239060',_binary 'This family contains NADPH-dependent flavin reductase and oxygen-insensitive nitroreductase. These enzymes are homodimeric flavoproteins that contain one FMN per monomer as a cofactor. Flavin reductase catalyzes the reduction of flavin by using NADPH as an electron donor. Oxygen-insensitive nitroreductase, such as NfsA protein in Escherichia coli, catalyzes reduction of nitrocompounds using NADPH as electron donor.','cd02146','NfsA_FRP'),(1833722,'gnl|CDD|272993',_binary 'epoxyqueuosine reductase.  This model was rebuilt to exclude archaeal homologs, now that there is new information that bacterial members are epoxyqueuosine reductase, QueG, involved in queuosine biosynthesis for tRNA maturation. [Protein synthesis, tRNA and rRNA base modification].','TIGR00276','iron-sulfur_cluster-bindi'),(1833741,'gnl|CDD|235026',_binary 'V-type ATP synthase subunit E; Provisional.','PRK02292','PRK02292'),(1833938,'gnl|CDD|313130',_binary 'Zn-ribbon containing protein.  This archaeal protein has no known function.','pfam09845','DUF2072'),(1833955,'gnl|CDD|234567',_binary 'methionyl-tRNA formyltransferase; Reviewed.','PRK00005','fmt'),(1833973,'gnl|CDD|259806',_binary 'Uncharacterized family; SPFH (stomatin, prohibitin, flotillin, and HflK/C) superfamily.  This model summarizes an uncharacterized family of proteins similar to stomatin, prohibitin, flotillin, HflK/C (SPFH) and podocin. The conserved domain common to the SPFH superfamily has also been referred to as the Band 7 domain. Many superfamily members are associated with lipid rafts. Individual proteins of the SPFH superfamily may cluster to form membrane microdomains which may in turn recruit multiprotein complexes. Microdomains formed from flotillin proteins may in addition be dynamic units with their own regulatory functions. Flotillins have been implicated in signal transduction, vesicle trafficking, cytoskeleton rearrangement and are known to interact with a variety of proteins. Stomatin interacts with and regulates members of the degenerin/epithelia Na+ channel family in mechanosensory cells of Caenorhabditis elegans and vertebrate neurons and participates in trafficking of Glut1 glucose transporters. Prohibitin may act as a chaperone for the stabilization of mitochondrial proteins. Prokaryotic HflK/C plays a role in the decision between lysogenic and lytic cycle growth during lambda phage infection. Flotillins have been implicated in the progression of prion disease, in the pathogenesis of neurodegenerative diseases such as Parkinson\'s and Alzheimer\'s disease and, in cancer invasion and metastasis. Mutations in the podocin gene give rise to autosomal recessive steroid resistant nephritic syndrome.','cd03408','SPFH_like_u1'),(1834100,'gnl|CDD|224086',_binary 'Oligoendopeptidase F [Amino acid transport and metabolism].','COG1164','COG1164'),(1834103,'gnl|CDD|316386',_binary 'Domain of unknown function (DUF4200).  This family is found in eukaryotes. It is a coiled-coil domain of unknwon function.','pfam13863','DUF4200'),(1834120,'gnl|CDD|225248',_binary 'Large extracellular alpha-helical protein [General function prediction only].','COG2373','COG2373'),(1834123,'gnl|CDD|291649',_binary 'Domain of unknown function (DUF4515).  This family of proteins is found in bacteria and eukaryotes. Proteins in this family are typically between 198 and 469 amino acids in length. There are two completely conserved L residues that may be functionally important.','pfam14988','DUF4515'),(1834133,'gnl|CDD|274091',_binary 'chaperone protein DnaK.  Members of this family are the chaperone DnaK, of the DnaK-DnaJ-GrpE chaperone system. All members of the seed alignment were taken from completely sequenced bacterial or archaeal genomes and (except for Mycoplasma sequence) found clustered with other genes of this systems. This model excludes DnaK homologs that are not DnaK itself, such as the heat shock cognate protein HscA (TIGR01991). However, it is not designed to distinguish among DnaK paralogs in eukaryotes. Note that a number of dnaK genes have shadow ORFs in the same reverse (relative to dnaK) reading frame, a few of which have been assigned glutamate dehydrogenase activity. The significance of this observation is unclear; lengths of such shadow ORFs are highly variable as if the presumptive protein product is not conserved. [Protein fate, Protein folding and stabilization].','TIGR02350','Chaperone_protein_DnaK'),(1834365,'gnl|CDD|318530',_binary 'Domain of unknown function (DUF4959).  This family consists of uncharacterized proteins around 400 residues in length and is mainly found in various Bacteroides, Pedobacter and Parabacteroides species. Several proteins are annotated as Galactose-binding like proteins, but the specific function of this protein is unknown.','pfam16323','DUF4959'),(1834413,'gnl|CDD|224212',_binary 'Predicted RNA-binding protein homologous to eukaryotic snRNP [Transcription].','COG1293','COG1293'),(1834696,'gnl|CDD|313417',_binary 'Uncharacterized conserved protein (DUF2369).  This is a proline-rich region of a group of proteins found from plants to fungi. The function is not known.','pfam10179','DUF2369'),(1834698,'gnl|CDD|274855',_binary 'type VII secretion integral membrane protein EccD.  Members of this family are EccD, a component of actinobacterial type VII secretion systems (T7SS) with ten to eleven predicted transmembrane helix regions. [Protein fate, Protein and peptide secretion and trafficking].','TIGR03920','T7SS_EccD'),(1834713,'gnl|CDD|306614',_binary 'DNA polymerase family B.  This region of DNA polymerase B appears to consist of more than one structural domain, possibly including elongation, DNA-binding and dNTP binding activities.','pfam00136','DNA_pol_B'),(1834870,'gnl|CDD|235549',_binary 'RNA polymerase sigma factor RpoD; Validated.','PRK05658','PRK05658'),(1834926,'gnl|CDD|315199',_binary 'Polo kinase kinase.  This domain family is found in eukaryotes, and is approximately 140 amino acids in length. The family is found in association with pfam00069. Polo-like kinase 1 (Plx1) is essential during mitosis for the activation of Cdc25C, for spindle assembly, and for cyclin B degradation. This family is Polo kinase kinase (PKK) which phosphorylates Polo kinase and Polo-like kinase to activate them. PKK is a serine/threonine kinase.','pfam12474','PKK'),(1834935,'gnl|CDD|153342',_binary 'The F-BAR (FES-CIP4 Homology and Bin/Amphiphysin/Rvs) domain of Nitric Oxide Synthase TRaffic INducer (NOSTRIN).  F-BAR domains are dimerization modules that bind and bend membranes and are found in proteins involved in membrane dynamics and actin reorganization. Nitric Oxide Synthase TRaffic INducer (NOSTRIN) is expressed in endothelial and epithelial cells and is involved in the regulation, trafficking and targeting of endothelial NOS (eNOS). NOSTRIN facilitates the endocytosis of eNOS by coordinating the functions of dynamin and the Wiskott-Aldrich syndrome protein (WASP). Increased expression of NOSTRIN may be correlated to preeclampsia. NOSTRIN contains an N-terminal F-BAR domain and a C-terminal SH3 domain. F-BAR domains form banana-shaped dimers with a positively-charged concave surface that binds to negatively-charged lipid membranes. They can induce membrane deformation in the form of long tubules. The F-BAR domain of NOSTRIN is necessary and sufficient for its membrane association and is responsible for its subcellular localization.','cd07658','F-BAR_NOSTRIN'),(1834942,'gnl|CDD|310305',_binary 'Tho complex subunit 7.  The Tho complex is involved in transcription elongation and mRNA export from the nucleus.','pfam05615','THOC7'),(1835096,'gnl|CDD|316757',_binary 'Domain of unknown function (DUF4355).  This family of proteins is found in bacteria and viruses. Proteins in this family are typically between 180 and 214 amino acids in length.','pfam14265','DUF4355'),(1835116,'gnl|CDD|316355',_binary 'Domain of unknown function (DUF4191).','pfam13829','DUF4191'),(1835117,'gnl|CDD|308989',_binary 'Uncharacterized protein family (UPF0182).  This family contains uncharacterized integral membrane proteins.','pfam03699','UPF0182'),(1835144,'gnl|CDD|310509',_binary 'PIF1-like helicase.  This family includes homologs of the PIF1 helicase, which inhibits telomerase activity and is cell cycle regulated. This family includes a large number of largely uncharacterized plant proteins. This family includes a P-loop motif that is involved in nucleotide binding.','pfam05970','PIF1'),(1835243,'gnl|CDD|99748',_binary 'Low-specificity threonine aldolase (TA). This family belongs to pyridoxal phosphate (PLP)-dependent aspartate aminotransferase superfamily (fold I).  TA catalyzes the conversion of L-threonine or L-allo-threonine to glycine and acetaldehyde in a secondary glycine biosynthetic pathway.','cd06502','TA_like'),(1835251,'gnl|CDD|314944',_binary 'Putative metal-binding domain of cation transport ATPase.  This domain is found in bacteria, and is approximately 90 amino acids in length. It is found associated with pfam00403, pfam00122, pfam00702. The cysteine-rich nature and composition suggest this might be a cation-binding domain; most members are annotated as being cation transport ATPases.','pfam12156','ATPase-cat_bd'),(1835268,'gnl|CDD|308731',_binary 'NapC/NirT cytochrome c family, N-terminal region.  Within the NapC/NirT family of cytochrome c proteins, some members, such as NapC and NirT, bind four haem groups, while others, such as TorC, bind five haems. This family aligns the common N-terminal region that contains four haem-binding C-X(2)-CH motifs.','pfam03264','Cytochrom_NNT'),(1835450,'gnl|CDD|307833',_binary 'DNA primase small subunit.  DNA primase synthesizes the RNA primers for the Okazaki fragments in lagging strand DNA synthesis. DNA primase is a heterodimer of large and small subunits. This family also includes baculovirus late expression factor 1 or LEF-1 proteins. Baculovirus LEF-1 is a DNA primase enzyme. The family also contains many bacterial DNA primases.','pfam01896','DNA_primase_S'),(1835451,'gnl|CDD|240130',_binary 'AE_Prim_S: primase domain similar to that found in the small subunit of archaeal and eukaryotic (A/E) DNA primases. Primases are DNA-dependent RNA polymerases which synthesis the short RNA primers required for DNA replication. In addition to its catalytic role in replication, DNA primase may play a role in coupling replication to DNA damage repair and in checkpoint control during S phase. In eukaryotes, this small catalytically active primase subunit (p50) and a larger primase subunit (p60), referred to jointly as the core primase, associate with the B subunit and the DNA polymerase alpha subunit in a complex, called Pol alpha-pri. The function of the larger primase subunit is unclear. Included in this group are Pfu41 and Pfu46, these two proteins comprise the primase complex of the archaea Pyrococcus furiosus; Pfu41 and Pfu46 have sequence identity to the eukaryotic p50 and p60 primase proteins respectively. Pfu41 preferentially uses dNTPs as substrate. Pfu46 regulates the primase activity of Pfu41.','cd04860','AE_Prim_S'),(1835463,'gnl|CDD|224768',_binary 'ATPase (PilT family) [General function prediction only].','COG1855','COG1855'),(1835464,'gnl|CDD|214771',_binary 'Large family of predicted nucleotide-binding domains.  From similarities to 5\'-exonucleases, these domains are predicted to be RNases. PINc domains in nematode SMG-5 and yeast NMD4p are predicted to be involved in RNAi.','smart00670','PINc'),(1835465,'gnl|CDD|189047',_binary 'PIN domain of Thermus Thermophilus Hb8, uncharacterized Bacillus subtilis YacL, and other bacterial homologs.  PIN (PilT N terminus) domain of the conserved membrane protein of unknown function of Thermus Thermophilus Hb8, Bacillus subtilis YacL and other similar homologs are included in this family. These PIN domains are structural homologs of flap endonuclease-1 (FEN1)-like PIN domains, but lack the extensive arch/clamp region  and the H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region, seen in FEN1-like PIN domains. PIN domain homologs within this group contain four highly conserved acidic residues (putative metal-binding, active site residues) which cluster at the C-terminal end of the beta-sheet and form a negatively charged pocket near the center of the molecule.  Proteins in this group have a C-terminal TRAM domain whose function is unknown but predicted to be a RNA-binding domain common to tRNA uracil methylation and adenine thiolation enzymes.','cd09877','PIN_YacL'),(1835466,'gnl|CDD|189048',_binary 'PIN domain of the Methanocaldococcus Jannaschii Dsm 2661 protein, Thermococcus sibiricus MM 739 predicted ATPase, and other similar archaeal homologs.  PIN (PilT N terminus) domain present N-terminal of AAA+, VirB11-like ATPases, as well as, the PIN domains of proteins from Methanopyrus kandleri AV19 and Thermococcus sibiricus MM 739, and other similar archaeal homologs are included in this family. Several members of this subfamily possess an AAA+, VirB11-like ATPase domain, flanked by PIN and KH nucleic acid-binding domains.  VirB11-ATPase is a type IV secretory pathway component required for T-pilus biogenesis and virulence. These PIN domains are structural homologs of flap endonuclease-1 (FEN1)-like PIN domains, but lack the extensive arch/clamp region  and the H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region, seen in FEN1-like PIN domains. PIN domains within this subgroup contain four highly conserved acidic residues (putative metal-binding, active site residues) which cluster at the C-terminal end of the beta-sheet and form a negatively charged pocket near the center of the molecule.','cd09878','PIN_VirB11L-ATPase'),(1835467,'gnl|CDD|224041',_binary 'ABC-type nitrate/sulfonate/bicarbonate transport system, ATPase component [Inorganic ion transport and metabolism].','COG1116','TauB'),(1835468,'gnl|CDD|213260',_binary 'ATP-binding cassette domain of the nitrate and sulfonate transporters.  NrtD and SsuB are the ATP-binding subunits of the bacterial ABC-type nitrate and sulfonate transport systems, respectively. ABC transporters are a large family of proteins involved in the transport of a wide variety of different compounds, like sugars, ions, peptides, and more complex organic molecules. The nucleotide binding domain shows the highest similarity between all members of the family. ABC transporters are a subset of nucleotide hydrolases that contain a signature motif, Q-loop, and H-loop/switch region, in addition to, the Walker A motif/P-loop and Walker B motif commonly found in a number of ATP- and GTP-binding and hydrolyzing proteins.','cd03293','ABC_NrtD_SsuB_transporter'),(1835469,'gnl|CDD|189033',_binary 'PIN domain of human telomerase-binding protein EST1, Smg6, and ribosome assembly factor, Nob1, Archaeoglobus fulgidus AF0591 protein and other eukaryotic, bacterial, and archaeal homologs.  PIN (PilT N terminus) domains of eukaryotic ribonucleases such as Smg5 and Smg6, essential factors in nonsense-mediated mRNA decay (NMD), and Nob1, a ribosome assembly factor critical in pre-rRNA processing, as well as, the uncharacterized archaeal Archaeoglobus fulgidus AF0591 protein and other eukaryotic, bacterial, and archaeal homologs are included in this family. Smg5 and Smg6 are essential factors in NMD, a post-transcriptional regulatory pathway that recognizes and rapidly degrades mRNAs containing premature translation termination codons. In vivo, the Smg6 PIN domain elicits degradation of bound mRNAs, as well as, metal-ion dependent, degradation of single-stranded RNA, in vitro. The Nob1 PIN domain binds the single-stranded cleavage site D at the 3\'end of 18S rRNA. Recombinant Nob1 binds as a tetramer to pre-18S rRNA fragments containing cleavage site D and believed to cleave at this site.  These PIN domains are structural homologs of flap endonuclease-1 (FEN1)-like PIN domains, but lack the extensive arch/clamp region and the H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region, that is seen in FEN1-like PIN domains. PIN domains within this group typically contain three or four highly conserved acidic residues (putative metal-binding, active site residues) which cluster at the C-terminal end of the beta-sheet and form a negatively charged pocket near the center of the molecule.','cd09863','PIN_Smg6-like'),(1835470,'gnl|CDD|280094',_binary 'PIN domain.','pfam01850','PIN'),(1835472,'gnl|CDD|311773',_binary 'Leukocidin/Hemolysin toxin family.','pfam07968','Leukocidin'),(1835493,'gnl|CDD|310471',_binary 'Poly-gamma-glutamate hydrolase.  This family consists of a number of bacterial and phage proteins that function as gamma-PGA hydrolase enzymes. Structurally the protein in this family adopted an open alpha/beta mixed core structure with a seven-stranded parallel/anti-parallel beta-sheet. This structure shows similarity to mammalian carboxypeptidase A and related enzymes.','pfam05908','Gamma_PGA_hydro'),(1835495,'gnl|CDD|259794',_binary 'C-terminal zinc ribbon domain of RPC11 subunit of RNA polymerase III.  The C-terminal zinc ribbon domain (C_ribbon) of subunit C11 (C-ribbon_RPC11) in RNA polymerase (Pol) III is required for intrinsic transcript cleavage. RPC11 is also involved in Pol III termination. Eukaryote genomes are transcribed by three nuclear RNA polymerases (Pol I, II and III) that share some subunits. RPC11 have strong homology to RPB9 of Pol II and RPA12 of Pol I. C-ribbon_RPC11 is homologous to Pol II elongation factor TFIIS domain III. C11 has two zinc-binding domains separated by a flexible linker.','cd10509','Zn-ribbon_RPC11'),(1835509,'gnl|CDD|178512',_binary 'thymidylate kinase.','PLN02924','PLN02924'),(1835527,'gnl|CDD|226112',_binary 'Uncharacterized protein conserved in bacteria [Function unknown].','COG3584','COG3584'),(1835528,'gnl|CDD|274850',_binary 'uracil-DNA glycosylase family domain.  This model represents a clade within the uracil-DNA glycosylase superfamily. Among characterized proteins, it most closely resembles the Thermus thermophilus uracil-DNA glycosylase TTUDGA, which acts uracil (deamidated cytosine) in both single-stranded DNA and U/G pairs of double-stranded DNA. This domain may occur either as a stand-alone protein or as the C-terminal domain of a fusion with another domain that always pairs with a particular radical-SAM family protein.','TIGR03914','SPO1_DNA_polymerase-relat'),(1835529,'gnl|CDD|270620',_binary 'Non-mltA associated 3D domain containing proteins, named for 3 conserved aspartate residues.  This family contains the 3D domain, named for its 3 conserved aspartates, including similar uncharacterized proteins. These proteins contain the critical active site aspartate of mltA-like lytic transglycosylases where the 3D domain forms a larger domain with the N-terminal region. This domain is also found in conjunction with numerous other domains such as the Escherichia coli MltA, a membrane-bound lytic transglycosylase comprised of 2 domains separated by a large groove, where the peptidoglycan strand binds. Domain A is made up of an N-terminal and a C-terminal portion, corresponding to the 3D domain and Domain B is inserted within the linear sequence of domain A.  MltA is distinct from other bacterial LTs, which are similar to each other. Escherichia coli peptidoglycan lytic transglycosylase (LT) initiates cell wall recycling in response to damage, during bacterial fission, and cleaves peptidoglycan (PG) to create functional spaces in its wall. PG chains (also known as murein), the major components of the bacterial cell wall, are comprised of alternating beta-1-4-linked N-acetylmuramic acid (MurNAc) and N-acetyl-D-glucosamine (GlcNAc), and lytic transglycosylases cleave this beta-1-4 bond.','cd14667','3D_containing_proteins'),(1835532,'gnl|CDD|310967',_binary '3D domain.  This short presumed domain contains three conserved aspartate residues, hence the name 3D. It has been shown to be part of the catalytic double psi beta barrel domain of MltA.','pfam06725','3D'),(1835535,'gnl|CDD|270619',_binary '3D domain, named for 3 conserved aspartate residues, is found in mltA-like lytic transglycosylases and numerous other contexts.  This family contains the 3D domain, named for its 3 conserved aspartates. It is found in conjunction with numerous other domains such as MltA (membrane-bound lytic murein transglycosylase A).  These aspartates are critical active site residues of mltA-like lytic transglycosylases. Escherichia coli peptidoglycan lytic transglycosylase (LT) initiates cell wall recycling in response to damage, during bacterial fission, and cleaves peptidoglycan (PG) to create functional spaces in its wall. MltA has 2 domains, separated by a large groove, where the peptidoglycan strand binds. The C-terminus has a double-psi beta barrel fold within the 3D domain, which forms the larger A domain along with the N-terminal region of Mlts, but is also found in various other domain architectures. Peptigoglycan (also known as murein) chains, the primary structural component of bacterial cells walls, are comprised of alternating beta-1-4-linked N-acetylmuramic acid (MurNAc) and N-acetyl-D-glucosamine (GlcNAc); lytic transglycosylases (LTs) cleave this beta-1-4 bond. Typically, LTs are exolytic, releasing Metabolite 1 (GlcNAc-anhMurNAc-L-Ala-D-Glu-m-Dap-D-Ala-D-Ala) from the ends of the PG strands. In contrast, membrane-bound lytic murein transglycosylase E (MltE) is endolytic , cleaving in the middle of PG strands, with further processing to Metabolite 1 accomplished by other LTs. In E. coli, there are six membrane- bound LTs: MltA-MltF and soluble Slt70. Slt35 is a soluble fragment cleaved from MltB. Bacterial LTs are classified in 4 families: Family 1 includes slt70 MltC-MltF, Family 2 includes MltA, Family 3 includes MltB, and family 4 of bacteriophage origin. While most LTs are related members of the lysozyme-like lytic transglycosylase family, MltA represents a distinct fold and sequence conservation.','cd14486','3D_domain'),(1835537,'gnl|CDD|198429',_binary 'Family-5 Uracil-DNA glycosylases (UDG), found in thermophilic organisms.  Family-5 Uracil-DNA glycosylases (UDG), found in thermophilic organisms, are DNA base excision repair enzymes.  This family is represented by the enzyme TTUDGB from Thermus thermophilus HB8. Members of this family exhibit high structural and sequence similarity to Family 4 UDGs, which are also found in thermophilic organisms.  However, Family 4 and Family 5 enzymes demonstrate differences in substrate specificity and catalytic mechanisms. Both TTUDGA (Family 4) and TTUDGB (Family 5) are capable of removing uracil from double stranded DNA. However, TTUDGA can also remove uracil from single-stranded DNA, while TTUDGB does not. TTUDGB also excises thymine from G:T mismatched DNA. In contrast, TTUDGA cannot remove thymine from a G:T mismatch. Furthermore, TTUDGB removes analogs of uracil from DNA, including 5-hydroxymethyluracil (hmU) and 5-fluorouracil (fU).','cd10031','UDG_F5_TTUDGB_like'),(1835538,'gnl|CDD|270618',_binary 'Domain A of MltA and related lytic transglycosylase; domain A is interrupted by domain B.  Escherichia coli MltA is a membrane-bound lytic transglycosylase comprised of two domains separated by a large groove, where the peptidoglycan strand binds. Domain A is made up of an N-terminal and a C-terminal portion, which correspond to the 3D domain, named for 3 conserved aspartate residues. Domain B is inserted within the linear sequence of domain A. MltA is distinct from other bacterial lytic transglycosylases (LTs), which are similar to each other. Escherichia coli peptidoglycan lytic transglycosylase (LT) initiates cell wall recycling in response to damage, during bacterial fission, and cleaves peptidoglycan (PG) to create functional spaces in its wall. PG chains (also known as murein), the major components of the bacterial cell wall, are comprised of alternating beta-1-4-linked N-acetylmuramic acid (MurNAc) and N-acetyl-D-glucosamine (GlcNAc), and lytic transglycosylases cleave this beta-1-4 bond. Typically, peptidoglycan lytic transglycosylases (LT) are exolytic, releasing Metabolite 1 (GlcNAc-anhMurNAc-L-Ala-D-Glu-m-Dap-D-Ala-D-Ala) from the ends of the PG strands. In contrast, MltE is endolytic , cleaving in the middle of PG strands, with further processing to Metabolite 1 accomplished by other LTs. In E. coli, there are six membrane-bound LTs: MltA-MltF and soluble Slt70. Slt35 is a soluble fragment cleaved from MltB. Bacterial LTs are classified in 4 families: Family 1 includes slt70 MltC-MltF, Family 2 includes MltA, Family 3 includes MltB, and Family 4 of bacteriophage origin. While most of the LT family members are similar in structure and sequence with a lysozyme-like fold, Family 2 (including mltA) is distinct.','cd14485','mltA_like_LT_A'),(1835540,'gnl|CDD|225377',_binary 'Membrane-bound lytic murein transglycosylase [Cell envelope biogenesis, outer membrane].','COG2821','MltA'),(1835543,'gnl|CDD|320684',_binary 'M48 Ste24 endopeptidase-like, integral membrane metallopeptidase.  This family contains peptidase M48 family Ste24p-like proteins that are as yet uncharacterized, but probably function as intracellular, membrane-associated zinc metalloproteases; they all contain the HEXXH Zn-binding motif, which is critical for Ste24p activity. They likely remove the C-terminal three residues of farnesylated proteins proteolytically and are possibly associated with the endoplasmic reticulum and golgi. Some members also contain ankyrin domains which occur in very diverse families of proteins and mediate protein-protein interactions.','cd07325','M48_Ste24p_like'),(1835554,'gnl|CDD|225268',_binary 'Predicted phosphohydrolase (DHH superfamily) [General function prediction only].','COG2404','COG2404'),(1835555,'gnl|CDD|223691',_binary 'Exopolyphosphatase-related proteins [General function prediction only].','COG0618','COG0618'),(1835557,'gnl|CDD|236976',_binary 'DNA-dependent helicase II; Provisional.','PRK11773','uvrD'),(1835563,'gnl|CDD|182930',_binary 'DNA helicase IV; Provisional.','PRK11054','helD'),(1835565,'gnl|CDD|237554',_binary 'putative recombination protein RecB; Provisional.','PRK13909','PRK13909'),(1835571,'gnl|CDD|307550',_binary 'Viral (Superfamily 1) RNA helicase.  Helicase activity for this family has been demonstrated and NTPase activity. This helicase has multiple roles at different stages of viral RNA replication, as dissected by mutational analysis.','pfam01443','Viral_helicase1'),(1835598,'gnl|CDD|173803',_binary 'Peptidase S8 family domain in Subtilisin proteins.  This group is composed of many different subtilisins: Pro-TK-subtilisin, subtilisin Carlsberg, serine protease Pb92 subtilisin, and BPN subtilisins just to name a few. Pro-TK-subtilisin is a serine protease from the hyperthermophilic archaeon Thermococcus kodakaraensis and consists of a signal peptide, a propeptide, and a mature domain.  TK-subtilisin is matured from pro-TK-subtilisin upon autoprocessing and degradation of the propeptide. Unlike other subtilisins though, the folding of the unprocessed form of pro-TK-subtilisin is induced by Ca2+ binding which is almost completed prior to autoprocessing. Ca2+ is required for activity unlike the bacterial subtilisins. The propeptide is not required for folding of the mature domain unlike the bacterial subtilases because of the stability produced from Ca2+ binding.  Subtilisin Carlsberg is extremely similar in structure to subtilisin BPN\'/Novo thought it has a 30% difference in amino acid sequence.  The substrate binding regions are also similar and 2 possible Ca2+ binding sites have been identified recently. Subtilisin Carlsberg possesses the highest commercial importance as a proteolytic additive for detergents. Serine protease Pb92, the serine protease from the alkalophilic Bacillus strain PB92, also contains two calcium ions and the overall  folding of the polypeptide chain closely resembles that of the subtilisins.   Members of the peptidases S8 and S35 clan include endopeptidases, exopeptidases and also a tripeptidyl-peptidase. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The S53 family contains a catalytic triad Glu/Asp/Ser. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07477','Peptidases_S8_Subtilisin_'),(1835600,'gnl|CDD|173787',_binary 'Peptidase domain in the S8 and S53 families.  Members of the peptidases S8 (subtilisin and kexin) and S53 (sedolisin) family include endopeptidases and  exopeptidases. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. Serine acts as a nucleophile, aspartate as an electrophile, and histidine as a base. The S53 family contains a catalytic triad Glu/Asp/Ser with an additional acidic residue Asp in the oxyanion hole, similar to that of subtilisin.  The serine residue here is the nucleophilic equivalent of the serine residue in the S8 family, while glutamic acid has the same role here as the histidine base.   However, the aspartic acid residue that acts as an electrophile is quite different.  In S53, it follows glutamic acid, while in S8 it precedes histidine. The stability of these enzymes may be enhanced by calcium; some members have been shown to bind up to 4 ions via binding sites with different affinity.  There is a great diversity in the characteristics of their members: some contain disulfide bonds, some are intracellular while others are extracellular, some function at extreme temperatures, and others at high or low pH values.','cd00306','Peptidases_S8_S53'),(1835601,'gnl|CDD|173812',_binary 'Peptidase S8 family domain, uncharacterized subfamily 1.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07487','Peptidases_S8_1'),(1835602,'gnl|CDD|173810',_binary 'Peptidase S8 family domain in Thermitase-like proteins.  Thermitase is a non-specific, trypsin-related serine protease with a very high specific activity.  It contains a subtilisin like domain. The tertiary structure of thermitase is similar to that of subtilisin BPN\'.  It contains a Asp/His/Ser catalytic triad. Members of the peptidases S8 (subtilisin and kexin) and S53 (sedolisin) clan include endopeptidases and  exopeptidases. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. Serine acts as a nucleophile, aspartate as an electrophile, and histidine as a base. The S53 family contains a catalytic triad Glu/Asp/Ser with an additional acidic residue Asp in the oxyanion hole, similar to that of subtilisin.  The serine residue here is the nucleophilic equivalent of the serine residue in the S8 family, while glutamic acid has the same role here as the histidine  base.   However, the aspartic acid residue that acts as an electrophile  is quite different.  In S53 the it follows glutamic acid, while in S8 it precedes histidine. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity.  There is a great diversity in the characteristics of their members: some contain disulfide bonds, some are intracellular while others are extracellular, some function at extreme temperatures, and others at high or low pH values.','cd07484','Peptidases_S8_Thermitase_'),(1835603,'gnl|CDD|173799',_binary 'Peptidase S8 family domain in Subtilisin-like proteins.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07473','Peptidases_S8_Subtilisin_'),(1835604,'gnl|CDD|173806',_binary 'Peptidase S8 family domain, uncharacterized subfamily 12.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07480','Peptidases_S8_12'),(1835605,'gnl|CDD|173800',_binary 'Peptidase S8 family domain in Vpr-like proteins.  The maturation of the peptide antibiotic (lantibiotic) subtilin in Bacillus subtilis ATCC 6633 includes posttranslational modifications of the propeptide and proteolytic cleavage of the leader peptide.  Vpr was identified as one of the proteases,  along with WprA, that are capable of processing subtilin.    Asp, Ser, His triadPeptidases S8 or Subtilases are a serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07474','Peptidases_S8_subtilisin_'),(1835606,'gnl|CDD|173814',_binary 'Peptidase S8 family domain, uncharacterized subfamily 5.  gap in seq This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07489','Peptidases_S8_5'),(1835607,'gnl|CDD|173801',_binary 'Peptidase S8 family domain in Streptococcal C5a peptidases.  Streptococcal C5a peptidase (SCP), is a highly specific protease and adhesin/invasin.  The subtilisin-like protease domain is located at the N-terminus and contains a protease-associated domain inserted into a loop.  There are three fibronectin type III (Fn) domains at the C-terminus. SCP binds to integrins with the help of Arg-Gly-Asp motifs which are thought to stabilize conformational changes required for substrate binding.  Peptidases S8 or Subtilases are a serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07475','Peptidases_S8_C5a_Peptida'),(1835608,'gnl|CDD|173790',_binary 'Peptidase S8 family domain in ProteinaseK-like proteins.  The peptidase S8 or Subtilase clan of proteases have a Asp/His/Ser catalytic triad that is not homologous to trypsin. This CD contains several members of this clan including: PCSK9 (Proprotein convertase subtilisin/kexin type 9), Proteinase_K, Proteinase_T, and other subtilisin-like serine proteases.  PCSK9 posttranslationally regulates hepatic low-density lipoprotein receptors (LDLRs) by binding to LDLRs on the cell surface, leading to their degradation. The binding site of PCSK9 has been localized to the epidermal growth factor-like repeat A (EGF-A) domain of the LDLR. Characterized Proteinases K are secreted endopeptidases with a high degree of sequence conservation.  Proteinases K are not substrate-specific and function in a wide variety of species in different pathways. It can hydrolyze keratin and other proteins with subtilisin-like specificity. The number of calcium-binding motifs found in these differ. Proteinase T is a novel proteinase from the fungus Tritirachium album Limber. The amino acid sequence of proteinase T as deduced from the nucleotide sequence is about 56% identical to that of proteinase K.','cd04077','Peptidases_S8_PCSK9_Prote'),(1835609,'gnl|CDD|274856',_binary 'type VII secretion-associated serine protease mycosin.  Members of this family are subtilisin-related serine proteases, found strictly in the Actinobacteria and associated with type VII secretion operons. The designation mycosin is used for members from Mycobacterium. [Protein fate, Protein and peptide secretion and trafficking, Protein fate, Protein modification and repair].','TIGR03921','T7SS_mycosin'),(1835610,'gnl|CDD|224322',_binary 'Subtilisin-like serine proteases [Posttranslational modification, protein turnover, chaperones].','COG1404','AprE'),(1835611,'gnl|CDD|306567',_binary 'Subtilase family.  Subtilases are a family of serine proteases. They appear to have independently and convergently evolved an Asp/Ser/His catalytic triad, like that found in the trypsin serine proteases (see pfam00089). Structure is an alpha/beta fold containing a 7-stranded parallel beta sheet, order 2314567.','pfam00082','Peptidase_S8'),(1835612,'gnl|CDD|173797',_binary 'Peptidase S8 family domain, uncharacterized subfamily 4.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd05561','Peptidases_S8_4'),(1835613,'gnl|CDD|173794',_binary 'Peptidase S8 family domain in Autotransporter serine proteases.  Autotransporter serine proteases belong to Peptidase S8 or Subtilase family. Subtilases, or subtilisin-like serine proteases, have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure (an example of convergent evolution).  Autotransporters are a superfamily of outer membrane/secreted proteins of gram-negative bacteria.  The presence of these subtilisin-like domains in these autotransporters are may enable them to be auto-catalytic and may also serve to allow them to act as a maturation protease cleaving other outer membrane proteins at the cell surface.','cd04848','Peptidases_S8_Autotranspo'),(1835614,'gnl|CDD|173817',_binary 'Peptidase S8 family domain, uncharacterized subfamily 8.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07492','Peptidases_S8_8'),(1835615,'gnl|CDD|173822',_binary 'Peptidase S8 family domain, uncharacterized subfamily 15.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07498','Peptidases_S8_15'),(1835616,'gnl|CDD|173816',_binary 'Peptidase S8 family domain, uncharacterized subfamily 7.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07491','Peptidases_S8_7'),(1835617,'gnl|CDD|173802',_binary 'Peptidase S8 family domain in Thiazoline oxidase/subtilisin-like proteases.  Thiazoline oxidase/subtilisin-like protease is produced by the symbiotic bacteria Prochloron spp. that inhabit didemnid family ascidians.  The cyclic peptides of the patellamide class found in didemnid extracts are now known to be synthesized by the Prochloron spp.  The prepatellamide is heterocyclized to form thiazole and oxazoline rings and the peptide is cleaved to form the two cyclic patellamides A and C.  Subtilases, or subtilisin-like serine proteases, have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure (an example of convergent evolution).','cd07476','Peptidases_S8_thiazoline_'),(1835618,'gnl|CDD|173808',_binary 'Peptidase S8 family domain in Lantiobiotic (lanthionine-containing antibiotics) specific proteases.  Lantiobiotic (lanthionine-containing antibiotics) specific proteases are very similar in structure to serine proteases.  Lantibiotics are ribosomally synthesised antimicrobial agents derived from ribosomally synthesised peptides with antimicrobial activities against Gram-positive bacteria. The proteases that cleave the N-terminal leader peptides from lantiobiotics include:  epiP, nsuP, mutP, and nisP.  EpiP, from Staphylococcus, is thought to cleave matured epidermin. NsuP, a dehydratase from Streptococcus and NisP, a membrane-anchored subtilisin-like serine protease from Lactococcus cleave nisin.  MutP is highly similar to epiP and nisP and is thought to process the prepeptide mutacin III of S. mutans. Members of the peptidases S8 (subtilisin and kexin) and S53 (sedolisin) clan include endopeptidases and  exopeptidases. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. Serine acts as a nucleophile, aspartate as an electrophile, and histidine as a base. The S53 family contains a catalytic triad Glu/Asp/Ser with an additional acidic residue Asp in the oxyanion hole, similar to that of subtilisin.  The serine residue here is the nucleophilic equivalent of the serine residue in the S8 family, while glutamic acid has the same role here as the histidine  base.   However, the aspartic acid residue that acts as an electrophile  is quite different.  In S53 the it follows glutamic acid, while in S8 it precedes histidine. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity.  There is a great diversity in the characteristics of their members: some contain disulfide bonds, some are intracellular while others are extracellular, some function at extreme temperatures, and others at high or low pH values.','cd07482','Peptidases_S8_Lantibiotic'),(1835619,'gnl|CDD|173795',_binary 'Peptidase S8 family domain, uncharacterized subfamily 3.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd04852','Peptidases_S8_3'),(1835620,'gnl|CDD|173791',_binary 'Peptidase S8 family domain in Kp43 proteases.  Kp43 proteases are members of the peptidase S8 or Subtilase clan of proteases. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure (an example of convergent evolution). Kp43 is topologically similar to kexin and furin both of which are proprotein convertases, but differ in amino acids sequence and the position of its C-terminal barrel.  Kp43 has 3 Ca2+ binding sites that differ from the corresponding sites in the other known subtilisin-like proteases.  KP-43 protease is known to be an oxidation-resistant protease when compared with the other subtilisin-like proteases.','cd04842','Peptidases_S8_Kp43_protea'),(1835622,'gnl|CDD|173807',_binary 'Peptidase S8 family domain in BacillopeptidaseF-like proteins.  Bacillus subtilis produces and secretes proteases and other types of exoenzymes at the end of the exponential phase of growth. The ones that make up this group is known as bacillopeptidase F, encoded by bpr,  a serine protease with high esterolytic activity which is inhibited by PMSF.  Like other members of the peptidases S8 family these have a Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity.','cd07481','Peptidases_S8_Bacillopept'),(1835624,'gnl|CDD|173818',_binary 'Peptidase S8 family domain, uncharacterized subfamily 9.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07493','Peptidases_S8_9'),(1835626,'gnl|CDD|173789',_binary 'Peptidase S8 family domain in Protein convertases.  Protein convertases, whose members include furins and kexins, are members of the peptidase S8 or Subtilase clan of proteases. They have an Asp/His/Ser catalytic triad that is not homologous to trypsin. Kexins are involved in the activation of peptide hormones, growth factors, and viral proteins.  Furin cleaves cell surface vasoactive peptides and proteins involved in cardiovascular tissue remodeling in the TGN, at cell surface, or in endosomes but rarely in the ER.  Furin also plays a key role in blood pressure regulation though the activation of transforming growth factor (TGF)-beta. High specificity is seen for cleavage after dibasic (Lys-Arg or Arg-Arg) or multiple basic residues in protein convertases.  There is also strong sequence conservation.','cd04059','Peptidases_S8_Protein_con'),(1835628,'gnl|CDD|173809',_binary 'Peptidase S8 family domain in Subtilisin_Novo-like proteins.  Subtilisins are a group of alkaline proteinases originating from different strains of Bacillus subtilis.  Novo is one of the strains that produced enzymes belonging to this group.  The enzymes obtained from the Novo and BPN\' strains are identical.  The Carlsburg and Novo subtilisins are thought to have arisen from a common ancestral protein.  They have similar peptidase and esterase activities, pH profiles, catalyze transesterification reactions, and are both inhibited by diispropyl fluorophosphate, though they differ in 85 positions in the amino acid sequence.  Members of the peptidases S8 and S35 clan include endopeptidases, exopeptidases and also a tripeptidyl-peptidase. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The S53 family contains a catalytic triad Glu/Asp/Ser with an additional acidic residue Asp in the oxyanion hole, similar to that of subtilisin.. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07483','Peptidases_S8_Subtilisin_'),(1835630,'gnl|CDD|173793',_binary 'Peptidase S8 family domain in Subtilisin-like proteins.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd04847','Peptidases_S8_Subtilisin_'),(1835632,'gnl|CDD|173804',_binary 'Peptidase S8 family domain in CspA-like proteins.  GSP (germination-specific protease) converts the spore peptidoglycan hydrolase (SleC) precursor to an active enzyme during germination of Clostridium perfringens S40 spores.  Analysis of an enzyme fraction of GSP showed that it was composed of a gene cluster containing the processed forms of products of cspA, cspB, and cspC which are positioned in a tandem array just upstream of the 5\' end of sleC. The amino acid sequences deduced from the nucleotide sequences of the csp genes showed significant similarity and showed a high degree of homology with those of the catalytic domain and the oxyanion binding region of subtilisin-like serine proteases.   Members of the peptidases S8 and S35 clan include endopeptidases, exopeptidases and also a tripeptidyl-peptidase. The S8 family has an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The S53 family contains a catalytic triad Glu/Asp/Ser. The stability of these enzymes may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07478','Peptidases_S8_CspA-like'),(1835636,'gnl|CDD|173820',_binary 'Peptidase S8 family domain, uncharacterized subfamily 13.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07496','Peptidases_S8_13'),(1835638,'gnl|CDD|173796',_binary 'Peptidase S8 family domain in Tripeptidyl aminopeptidases_II.  Tripeptidyl aminopeptidases II are member of the peptidase S8 or Subtilase family. Subtilases, or subtilisin-like serine proteases, have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure (an example of convergent evolution).  Tripeptidyl aminopeptidase II removes tripeptides from the free N terminus of oligopeptides as well as having endoproteolytic activity.  Some tripeptidyl aminopeptidases have been shown to cleave tripeptides and small peptides, e.g. angiotensin II and glucagon, while others are believed to be involved in MHC I processing.','cd04857','Peptidases_S8_Tripeptidyl'),(1835641,'gnl|CDD|173813',_binary 'Peptidase S8 family domain, uncharacterized subfamily 2.  This family is a member of the Peptidases S8 or Subtilases serine endo- and exo-peptidase clan. They have an Asp/His/Ser catalytic triad similar to that found in trypsin-like proteases, but do not share their three-dimensional structure and are not homologous to trypsin. The stability of subtilases may be enhanced by calcium, some members have been shown to bind up to 4 ions via binding sites with different affinity. Some members of this clan contain disulfide bonds. These enzymes can be intra- and extracellular, some function at extreme temperatures and pH values.','cd07488','Peptidases_S8_2'),(1835644,'gnl|CDD|293797',_binary 'uncharacterized human metallo-beta-lactamase domain-containing protein 1 and related proteins; MBL-fold metallo hydrolase domain.  Includes the MBL-fold metallo hydrolase domain of uncharacterized human MBLAC1 and related proteins. Members of this subgroup belong to the MBL-fold metallo-hydrolase superfamily which is comprised mainly of hydrolytic enzymes which carry out a variety of biological functions.','cd07711','MBLAC1-like_MBL-fold'),(1835718,'gnl|CDD|163441',_binary 'putative phosphoesterase.  Members of this protein family belong to the larger family pfam00149 (calcineurin-like phosphoesterase), a family largely defined by small motifs of metal-chelating residues. The subfamily in this model shows a good but imperfect co-occurrence in species with domain TIGR03715 that defines a novel class of signal peptide typical of the accessory secretory system.','TIGR03729','acc_ester'),(1835722,'gnl|CDD|311698',_binary 'Protein of unknown function (DUF1653).  This is a family of hypothetical bacterial proteins of unknown function.','pfam07866','DUF1653'),(1835728,'gnl|CDD|238032',_binary 'Metal dependent phosphohydrolases with conserved \'HD\' motif.','cd00077','HDc'),(1835730,'gnl|CDD|214679',_binary 'Metal dependent phosphohydrolases with conserved \'HD\' motif.  Includes eukaryotic cyclic nucleotide phosphodiesterases (PDEc). This profile/HMM does not detect HD homologues in bacterial glycine aminoacyl-tRNA synthetases (beta subunit).','smart00471','HDc'),(1835731,'gnl|CDD|273761',_binary '[Protein-PII] uridylyltransferase.  This model describes GlnD, the uridylyltransferase/uridylyl-removing enzyme for the nitrogen regulatory protein PII. Not all homologs of PII share the property of uridylyltransferase modification on the characteristic Tyr residue (see Prosite pattern PS00496 and document PDOC00439), but the modification site is preserved in the PII homolog of all species with a member of this family. [Central intermediary metabolism, Nitrogen metabolism, Regulatory functions, Protein interactions].','TIGR01693','UTase_glnD'),(1835732,'gnl|CDD|312965',_binary 'YopX protein.  YopX is a protein that is largely helical, with three identical chains probably complexing into a twelve-chain structure.','pfam09643','YopX'),(1835738,'gnl|CDD|163448',_binary 'PRTRC system ThiF family protein.  A novel genetic system characterized by six major proteins, included a ParB homolog and a ThiF homolog, is designated PRTRC, or ParB-Related,ThiF-Related Cassette. This family is the PRTRC system ThiF family protein.','TIGR03736','PRTRC_ThiF'),(1835739,'gnl|CDD|238760',_binary 'Superfamily of activating enzymes (E1) of the ubiquitin-like proteins. This family includes classical ubiquitin-activating enzymes E1, ubiquitin-like (ubl) activating enzymes and other mechanistic homologes, like MoeB, Thif1 and others. The common reaction mechanism catalyzed by MoeB, ThiF and the E1 enzymes begins with a nucleophilic attack of the C-terminal carboxylate of MoaD, ThiS and ubiquitin, respectively, on the alpha-phosphate of an ATP molecule bound at the active site of the activating enzymes, leading to the formation of a high-energy acyladenylate intermediate and subsequently to the formation of a thiocarboxylate at the C termini of MoaD and ThiS.','cd01483','E1_enzyme_family'),(1835740,'gnl|CDD|163447',_binary 'PRTRC system protein A.  A novel genetic system characterized by six major proteins, included a ParB homolog and a ThiF homolog, is designated PRTRC, or ParB-Related,ThiF-Related Cassette. It is often found on plasmids. This protein family is designated protein A.','TIGR03735','PRTRC_A'),(1835741,'gnl|CDD|274094',_binary 'thiazole biosynthesis adenylyltransferase ThiF, E. coli subfamily.  Members of the HesA/MoeB/ThiF family of proteins (pfam00899) include a number of members encoded in the midst of thiamine biosynthetic operons. This mix of known and putative ThiF proteins shows a deep split in phylogenetic trees, with the Escherichia. coli ThiF and the E. coli MoeB proteins seemingly more closely related than E. coli ThiF and Campylobacter (for example) ThiF. This model represents the more widely distributed clade of ThiF proteins such found in E. coli. [Biosynthesis of cofactors, prosthetic groups, and carriers, Thiamine].','TIGR02356','molybdopterin_biosynthesi'),(1835742,'gnl|CDD|279270',_binary 'ThiF family.  This domain is found in ubiquitin activating E1 family and members of the bacterial ThiF/MoeB/HesA family. It is repeated in Ubiquitin-activating enzyme E1.','pfam00899','ThiF'),(1835743,'gnl|CDD|238386',_binary 'ThiF_MoeB_HesA. Family of E1-like enzymes involved in molybdopterin and thiamine biosynthesis family. The common reaction mechanism catalyzed by MoeB and ThiF, like other E1 enzymes, begins with a nucleophilic attack of the C-terminal carboxylate of MoaD and ThiS, respectively, on the alpha-phosphate of an ATP molecule bound at the active site of the activating enzymes, leading to the formation of a high-energy acyladenylate intermediate and subsequently to the formation of  a thiocarboxylate at the C termini of MoaD and ThiS. MoeB, as the MPT synthase (MoaE/MoaD complex) sulfurase, is involved in the biosynthesis of the molybdenum cofactor, a derivative of the tricyclic pterin, molybdopterin (MPT). ThiF catalyzes the adenylation of ThiS, as part of the biosynthesis pathway of thiamin pyrophosphate (vitamin B1). .','cd00757','ThiF_MoeB_HesA_family'),(1835744,'gnl|CDD|224100',_binary 'Dinucleotide-utilizing enzymes involved in molybdopterin and thiamine biosynthesis family 1 [Coenzyme metabolism].','COG1179','COG1179'),(1835745,'gnl|CDD|238384',_binary 'Family of activating enzymes (E1) of ubiquitin-like proteins related to the E.coli hypothetical protein ygdL. The common reaction mechanism catalyzed by E1-like enzymes begins with a nucleophilic attack of the C-terminal carboxylate of the ubiquitin-like substrate, on the alpha-phosphate of an ATP molecule bound at the active site of the activating enzymes, leading to the formation of a high-energy acyladenylate intermediate and subsequently to the formation of a thiocarboxylate at the C termini of the substrate. The exact function of this family is unknown.','cd00755','YgdL_like'),(1835746,'gnl|CDD|236320',_binary 'thiamine biosynthesis protein ThiF; Provisional.','PRK08644','PRK08644'),(1835749,'gnl|CDD|115518',_binary 'Protein of unknown function (DUF1256).  This family consists of several uncharacterized bacterial proteins which seem to be specific to the orders Clostridia and Bacillales. Family members are typically around 180 residues in length. The function of this family is unknown. These proteins are related to peptidase family M63 and so may be peptidases.','pfam06866','DUF1256'),(1835772,'gnl|CDD|277316',_binary 'phosphoprotein phosphatases of the metallophosphatase superfamily, metallophosphatase domain.  The PPP (phosphoprotein phosphatase) family is one of two known protein phosphatase families specific for serine and threonine.  This family includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd00144','MPP_PPP_family'),(1835773,'gnl|CDD|277366',_binary 'Bacillus subtilis PrpE and related proteins, metallophosphatase domain.  PrpE (protein phosphatase E) is a bacterial member of the PPP (phosphoprotein phosphatase) family of serine/threonine phosphatases and a key signal transduction pathway component controlling the expression of spore germination receptors GerA and GerK in Bacillus subtilis. PrpE is closely related to ApaH (also known symmetrical Ap(4)A hydrolase and bis(5\'nucleosyl)-tetraphosphatase).  PrpE has specificity for phosphotyrosine only, unlike the serine/threonine phosphatases to which it is related. The Bacilli members of this family are single domain proteins while the other members have N- and C-terminal domains in addition to this phosphatase domain. Pnkp is the end-healing and end-sealing component of an RNA repair system present in bacteria. It is composed of three catalytic modules: an N-terminal polynucleotide 5\' kinase, a central 2\',3\' phosphatase, and a C-terminal ligase. Pnkp is a Mn(2+)-dependent phosphodiesterase-monoesterase that dephosphorylates 2\',3\'-cyclic phosphate RNA ends. An RNA binding site is suggested by a continuous tract of positive surface potential flanking the active site. The PPP (phosphoprotein phosphatase) family, to which PrpE belongs, is one of two known protein phosphatase families specific for serine and threonine.  The PPP family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07423','MPP_Prp_like'),(1835774,'gnl|CDD|277361',_binary 'PP2B, metallophosphatase domain.  PP2B (calcineurin) is a unique serine/threonine protein phosphatase in its regulation by a second messenger (calcium and calmodulin).  PP2B is involved in many biological processes including immune responses, the second messenger cAMP pathway, sodium/potassium ion transport in the nephron, cell cycle progression in lower eukaryotes, cardiac hypertrophy, and memory formation.  PP2B is highly conserved from yeast to humans, but is absent from plants.  PP2B is a heterodimer consisting of a catalytic subunit (CnA) and a regulatory subunit (CnB); CnB  contains four Ca2+ binding motifs referred to as EF hands.  The PPP (phosphoprotein phosphatase) family, to which PP2B belongs, is one of two known protein phosphatase families specific for serine and threonine.  The PPP family also includes: PP1, PP2A, PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07416','MPP_PP2B'),(1835775,'gnl|CDD|277367',_binary 'PrpA and PrpB, metallophosphatase domain.  PrpA and PrpB are bacterial type I serine/threonine and tyrosine phosphatases thought to modulate the expression of proteins that protect the cell upon accumulation of misfolded proteins in the periplasm.  The PPP (phosphoprotein phosphatase) family, to which PrpA and PrpB belong, is one of two known protein phosphatase families specific for serine and threonine.  This family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07424','MPP_PrpA_PrpB'),(1835776,'gnl|CDD|197547',_binary 'Protein phosphatase 2A homologues, catalytic domain.  Large family of serine/threonine phosphatases, that includes PP1, PP2A and PP2B (calcineurin) family members.','smart00156','PP2Ac'),(1835777,'gnl|CDD|234673',_binary 'diadenosine tetraphosphatase; Reviewed.','PRK00166','apaH'),(1835778,'gnl|CDD|277365',_binary 'Escherichia coli ApaH and related proteins, metallophosphatase domain.  ApaH (also known as symmetrically cleaving Ap4A hydrolase and bis(5\'nucleosyl)-tetraphosphatase) is a bacterial member of the PPP (phosphoprotein phosphatase) family of serine/threonine phosphatases that hydrolyzes the nucleotide-signaling molecule diadenosine tetraphosphate (Ap(4)A) into two ADP and also hydrolyzes Ap(5)A, Gp(4)G, and other extending compounds.  Null mutations in apaH result in high intracellular levels of Ap(4)A which correlate with multiple phenotypes, including a decreased expression of catabolite-repressible genes, a reduction in the expression of flagellar operons, and an increased sensitivity to UV  and heat.  Ap4A hydrolase is important in responding to heat shock and oxidative stress via regulating the concentration of Ap4A in bacteria.  Ap4A hydrolase is also thought to play a role in siderophore production, but the mechanism by which ApaH interacts with siderophore pathways in unknown.  The PPP (phosphoprotein phosphatase) family, to which ApaH belongs, is one of two known protein phosphatase families specific for serine and threonine.  The PPP family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, and PrpA/PrpB. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07422','MPP_ApaH'),(1835779,'gnl|CDD|277360',_binary 'PP2A, PP4, and PP6 phosphoprotein phosphatases, metallophosphatase domain.  PP2A-like family of phosphoprotein phosphatases (PPP\'s) including PP4 and PP6.  PP2A (Protein phosphatase 2A) is a critical regulator of many cellular activities.  PP2A comprises about 1% of total cellular proteins.  PP2A, together with protein phosphatase 1 (PP1), accounts for more than 90% of all serine/threonine phosphatase activities in most cells and tissues. The PP2A subunit  in addition to having a catalytic domain homologous to PP1, has a unique C-terminal tail, containing a motif that is conserved in the catalytic subunits of all PP2A-like phosphatases including PP4 and PP6, and has an important role in PP2A regulation.  The PP2A-like family of phosphatases all share a similar heterotrimeric architecture, that includes: a 65kDa scaffolding subunit (A), a 36kDa catalytic subunit (C), and one of 18 regulatory subunits (B).  The PPP (phosphoprotein phosphatase) family, to which PP2A belongs, is one of two known protein phosphatase families specific for serine and threonine.  The PPP family also includes: PP1, PP2B (calcineurin), PP4, PP5, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07415','MPP_PP2A_PP4_PP6'),(1835787,'gnl|CDD|277362',_binary 'PP5, C-terminal metallophosphatase domain.  Serine/threonine protein phosphatase-5 (PP5) is a member of the PPP gene family of protein phosphatases that is highly conserved among eukaryotes and widely expressed in mammalian tissues. PP5 has a C-terminal phosphatase domain and an extended N-terminal TPR (tetratricopeptide repeat) domain containing three TPR motifs.  The PPP (phosphoprotein phosphatase) family, to which PP5 belongs, is one of two known protein phosphatase families specific for serine and threonine.  The PPP family also includes: PP1, PP2A, PP2B (calcineurin), PP4, PP6, PP7, Bsu1, RdgC, PrpE, PrpA/PrpB, and ApA4 hydrolase. The PPP catalytic domain is defined by three conserved motifs (-GDXHG-, -GDXVDRG- and -GNHE-).  The PPP enzyme family is ancient with members found in all eukaryotes, and in most bacterial and archeal genomes.  Dephosphorylation of phosphoserines and phosphothreonines on target proteins plays a central role in the regulation of many cellular processes.  PPPs belong to the metallophosphatase (MPP) superfamily.  MPPs are functionally diverse, but all share a conserved domain with an active site consisting of two metal ions (usually manganese, iron, or zinc) coordinated with octahedral geometry by a cage of histidine, aspartate, and asparagine residues. The MPP superfamily includes: Mre11/SbcD-like exonucleases, Dbr1-like RNA lariat debranching enzymes, YfcE-like phosphodiesterases, purple acid phosphatases (PAPs), YbbF-like UDP-2,3-diacylglucosamine hydrolases, and acid sphingomyelinases (ASMases).  The conserved domain is a double beta-sheet sandwich with a di-metal active site made up of residues located at the C-terminal side of the sheets.  This domain is thought to allow for productive metal coordination.','cd07417','MPP_PP5_C'),(1835790,'gnl|CDD|237461',_binary 'hypothetical protein; Provisional.','PRK13660','PRK13660'),(1835801,'gnl|CDD|289422',_binary 'Ribbon-helix-helix domain.  This short bacterial protein contains a ribbon-helix-helix domain that is likely to be DNA-binding.','pfam12651','RHH_3'),(1835802,'gnl|CDD|271323',_binary 'Uncharacterized NHL-repeat domain in bacterial proteins.  This bacterial family of NHL-repeat domains is found in a variety of domain architectures. The NHL (NCL-1, HT2A and LIN-41) repeat is found in multiple tandem copies, typically as 6 instances. It is about 40 residues long and resembles the WD repeat and other beta-propeller structures.','cd14953','NHL_like_1'),(1835822,'gnl|CDD|315794',_binary 'Bacterial Ig-like domain.','pfam13205','Big_5'),(1835827,'gnl|CDD|183236',_binary 'lytic murein transglycosylase; Provisional.','PRK11619','PRK11619'),(1835828,'gnl|CDD|236778',_binary 'membrane-bound lytic transglycosylase F; Provisional.','PRK10859','PRK10859'),(1835840,'gnl|CDD|318325',_binary 'Domain of unknown function (DUF4815).','pfam16075','DUF4815'),(1835846,'gnl|CDD|153282',_binary 'The Bin/Amphiphysin/Rvs (BAR) domain of Family with sequence similarity 92 (FAM92).  BAR domains are dimerization, lipid binding and curvature sensing modules found in many different proteins with diverse functions including organelle biogenesis, membrane trafficking or remodeling, and cell division and migration. This group is composed of proteins from the family with sequence similarity 92 (FAM92), which were originally identified by the presence of the unknown domain DUF1208. This domain shows similarity to the BAR domains of sorting nexins. Mammals contain at least two member types, FAM92A and FAM92B, which may exist in many variants. The Xenopus homolog of FAM92A1, xVAP019, is essential for embryo survival and cell differentiation. FAM92A1 may be involved in regulating cell proliferation and apoptosis. BAR domains form dimers that bind to membranes, induce membrane bending and curvature, and may also be involved in protein-protein interactions.','cd07598','BAR_FAM92'),(1835848,'gnl|CDD|309996',_binary 'Bacteriophage holin family.  Phage holins and lytic enzymes are both necessary for bacterial lysis and virus dissemination. This family also includes TcdE/UtxA involved in toxin secretion in Clostridium difficile. The 1.E.10 family is represented by Bacillus subtilis phi29 holin; 1.E.16 represents the Cph1 holin; and the 1.E.19 family is represented by the Clostridium difficile TcdE holin. Toxigenic strains of C. difficile produce two large toxins (TcdA and TcdB) encoded within a pathogenicity locus. tcdE, encoded between tcdA and tcdB, encodes a 166 aa protein which causes death to E. coli when expressed, and the structure of TcdE resembles holins. TcdE acts on the bacterial membrane. Since TcdA and TcdB lack signal peptides, they may be released via TcdE either prior to or subsequent to cell lysis.','pfam05105','Phage_holin_4_1'),(1836226,'gnl|CDD|184118',_binary 'regulatory ATPase RavA; Provisional.','PRK13531','PRK13531'),(1836229,'gnl|CDD|307292',_binary 'Magnesium chelatase, subunit ChlI.  Magnesium-chelatase is a three-component enzyme that catalyzes the insertion of Mg2+ into protoporphyrin IX. This is the first unique step in the synthesis of (bacterio)chlorophyll. Due to this, it is thought that Mg-chelatase has an important role in channelling inter- mediates into the (bacterio)chlorophyll branch in response to conditions suitable for photosynthetic growth. ChlI and BchD have molecular weight between 38-42 kDa.','pfam01078','Mg_chelatase'),(1836302,'gnl|CDD|224482',_binary 'Multidrug resistance efflux pump [Defense mechanisms].','COG1566','EmrA'),(1836310,'gnl|CDD|235999',_binary 'F0F1 ATP synthase subunit B\'; Validated.','PRK07353','PRK07353'),(1836992,'gnl|CDD|273152',_binary 'transcription-repair coupling factor (mfd).  All proteins in this family for which functions are known are DNA-dependent ATPases that function in the process of transcription-coupled DNA repair in which the repair of the transcribed strand of actively transcribed genes is repaired at a higher rate than the repair of non-transcribed regions of the genome and than the non-transcribed strand of the same gene. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). This family is closely related to the RecG and UvrB families. [DNA metabolism, DNA replication, recombination, and repair].','TIGR00580','Transcription-repair-coup'),(1837020,'gnl|CDD|273829',_binary 'type I secretion outer membrane protein, TolC family.  Members of this model are outer membrane proteins from the TolC subfamily within the RND (Resistance-Nodulation-cell Division) efflux systems. These proteins, unlike the NodT subfamily, appear not to be lipoproteins. All are believed to participate in type I protein secretion, an ABC transporter system for protein secretion without cleavage of a signal sequence, although they may, like TolC, participate also in the efflux of smaller molecules as well. This family includes the well-documented examples TolC (E. coli), PrtF (Erwinia), and AprF (Pseudomonas aeruginosa). [Protein fate, Protein and peptide secretion and trafficking, Transport and binding proteins, Porins].','TIGR01844','Proteases_secretion_prote'),(1837021,'gnl|CDD|197675',_binary 'helix_turn _helix lactose operon repressor.','smart00354','HTH_LACI'),(1837651,'gnl|CDD|274528',_binary 'type VI secretion ATPase, ClpV1 family.  Members of this protein family are homologs of ClpB, an ATPase associated with chaperone-related functions. These ClpB homologs, designated ClpV1, are a key component of the bacterial pathogenicity-associated type VI secretion system. [Protein fate, Protein and peptide secretion and trafficking, Cellular processes, Pathogenesis].','TIGR03345','VI_ClpV1'),(1837747,'gnl|CDD|237429',_binary 'anthranilate synthase; Provisional.','PRK13566','PRK13566'),(1837775,'gnl|CDD|239652',_binary 'Coenzyme B12-dependent-methylmalonyl coenzyme A (CoA) mutase (MCM) family, isobutyryl-CoA mutase (ICM)-like subfamily; contains archaeal and bacterial proteins similar to the large subunit of Streptomyces cinnamonensis coenzyme B12-dependent ICM. ICM from S. cinnamonensis is comprised of a large and a small subunit. The holoenzyme appears to be an alpha2beta2 heterotetramer with up to 2 molecules of coenzyme B12 bound. The small subunit binds coenzyme B12. ICM catalyzes the reversible rearrangement of n-butyryl-CoA to isobutyryl-CoA, intermediates in fatty acid and valine catabolism, which in S. cinnamonensis can be converted to methylmalonyl-CoA and used in polyketide synthesis.','cd03680','MM_CoA_mutase_ICM_like'),(1837780,'gnl|CDD|310118',_binary 'SH3 domain-binding protein 5 (SH3BP5).  This family consists of several eukaryotic SH3 domain-binding protein 5 or c-Jun N-terminal kinase (JNK)-interacting proteins (SH3BP5 or Sab). Sab binds to and serves as a substrate for JNK in vitro, and has been found to interact with the Src homology 3 (SH3) domain of Bruton\'s tyrosine kinase (Btk). Inspection of the sequence of Sab reveals the presence of two putative mitogen-activated protein kinase interaction motifs (KIMs) similar to that found in the JNK docking domain of the c-Jun transcription factor, and four potential serine-proline JNK phosphorylation sites in the C-terminal half of the molecule.','pfam05276','SH3BP5'),(1837829,'gnl|CDD|238295',_binary 'Phosphotriesterase (PTE) catalyzes the hydrolysis of organophosphate nerve agents, including the chemical warfare agents VX, soman, and sarin as well as the insecticide paraoxon. PTE exists as a homodimer with one active site per monomer. The active site is located next to a binuclear metal center, at the C-terminal end of a TIM alpha- beta barrel motif.  The native enzyme contains two zinc ions at the active site however these can be replaced with other metals such as cobalt, cadmium, nickel or manganese and the enzyme remains active.','cd00530','PTE'),(1837879,'gnl|CDD|310988',_binary 'Enhancer of Polycomb C-terminus.  This family represents the C-terminus of eukaryotic enhancer of polycomb proteins, which have roles in heterochromatin formation. This family contains several conserved motifs.','pfam06752','E_Pc_C'),(1838071,'gnl|CDD|223961',_binary 'Membrane-bound serine protease (ClpP class) [Posttranslational modification, protein turnover, chaperones].','COG1030','NfeD'),(1838330,'gnl|CDD|225249',_binary 'Predicted extracellular nuclease [General function prediction only].','COG2374','COG2374'),(1838408,'gnl|CDD|238728',_binary 'Magnesium chelatase: Mg-chelatase catalyses the insertion of Mg into protoporphyrin IX (Proto). In chlorophyll biosynthesis, insertion of Mg2+ into protoporphyrin IX is catalysed by magnesium chelatase in an ATP-dependent reaction. Magnesium chelatase is a three sub-unit (BchI, BchD and BchH) enzyme with a novel arrangement of domains: the C-terminal helical domain is located behind the nucleotide binding site. The BchD domain contains a AAA domain at its N-terminus and a VWA domain at its C-terminus. The VWA domain has been speculated to be involved in mediating protein-protein interactions.','cd01451','vWA_Magnesium_chelatase'),(1838558,'gnl|CDD|316100',_binary 'Helix-turn-helix domain of transposase family ISL3.','pfam13542','HTH_Tnp_ISL3'),(1838573,'gnl|CDD|214015',_binary 'Uncharacterized protein conserved in Bacteroidetes.  A well-conserved family of 16-stranded beta barrels resembling outer membrane porins. The interior of the barrels is mostly occupied by an insert with partially helical structure.','cd12871','Bacuni_01323_like'),(1838583,'gnl|CDD|309871',_binary 'O-Antigen ligase.  This group of bacterial proteins is involved in the synthesis of O-antigen, a lipopolysaccharide found in the outer membrane in gram-negative bacteria. This family includes O-antigen ligases such as E. coli RfaL.','pfam04932','Wzy_C'),(1838765,'gnl|CDD|188306',_binary 'ribonuclease Y.  Members of this family are RNase Y, an endoribonuclease. The member from Bacillus subtilis, YmdA, has been shown to be involved in turnover of yitJ riboswitch. [Transcription, Degradation of RNA].','TIGR03319','RNase_Y'),(1838840,'gnl|CDD|316312',_binary 'Domain of unknown function (DUF4175).','pfam13779','DUF4175'),(1840058,'gnl|CDD|225035',_binary 'Cytochrome P450 [Secondary metabolites biosynthesis, transport, and catabolism].','COG2124','CypX'),(1840061,'gnl|CDD|237030',_binary 'alpha-ketoglutarate decarboxylase; Reviewed.','PRK12270','kgd'),(1840063,'gnl|CDD|237015',_binary 'hypothetical protein; Reviewed.','PRK11901','PRK11901'),(1840173,'gnl|CDD|226127',_binary 'Cell division initiation protein [Cell division and chromosome partitioning].','COG3599','DivIVA'),(1840175,'gnl|CDD|306914',_binary 'HlyD membrane-fusion protein of T1SS.  HlyD is a component of the prototypical alpha-haemolysin (HlyA) bacterial type I secretion system, along with the other components HlyB and TolC. HlyD and HlyB are inner-membrane proteins and specific components of the transport apparatus of alpha-haemolysin. HlyD is anchored in the cytoplasmic membrane by a single transmembrane domain and has a large periplasmic domain within the carboxy-terminal 100 amino acids, HlyB and HlyD form a stable complex that binds the recombinant protein bearing a C-terminal HlyA signal sequence and ATP in the cytoplasm. HlyD, HlyB and TolC combine to form the three-component ABC transporter complex that forms a trans-membrane channel or pore through which HlyA can be transferred directly to the extracellular medium. Cutinase has been shown to be transported effectively through this pore.','pfam00529','HlyD'),(1840552,'gnl|CDD|310735',_binary 'ComK protein.  This family consists of several bacterial ComK proteins. The ComK protein of Bacillus subtilis positively regulates the transcription of several late competence genes as well as comK itself. It has been found that ClpX plays an important role in the regulation of ComK at the post-transcriptional level.','pfam06338','ComK'),(1841219,'gnl|CDD|318244',_binary 'Bacterial collagen, middle region.  Collagen_mid is the conserved central region of bacterial collagen triple helix repeat proteins.','pfam15984','Collagen_mid'),(1841493,'gnl|CDD|132999',_binary 'Beta-4-Galactosyltransferase is involved in the formation of the poly-N-acetyllactosamine core structures present in glycoproteins and glycosphingolipids.  Beta-4-Galactosyltransferase transfers galactose from uridine diphosphogalactose to the terminal beta-N-acetylglucosamine residues, hereby forming the poly-N-acetyllactosamine core structures present in glycoproteins and glycosphingolipids. At least seven homologous beta-4-galactosyltransferase isoforms have been identified that use different types of glycoproteins and glycolipids as substrates. Of the seven identified members of the beta-1,4-galactosyltransferase subfamily (beta1,4-Gal-T1 to -T7), b1,4-Gal-T1 is most characterized (biochemically). It is a Golgi-resident type II membrane enzyme with a cytoplasmic domain, membrane spanning region, and a stem region and catalytic domain facing the lumen.','cd00899','b4GalT'),(1842071,'gnl|CDD|225437',_binary 'Flagellar biosynthesis chaperone [Cell motility and secretion / Intracellular trafficking and secretion / Posttranslational modification, protein turnover, chaperones].','COG2882','FliJ'),(1842236,'gnl|CDD|311626',_binary 'Nop53 (60S ribosomal biogenesis).  This nucleolar family of proteins are involved in 60S ribosomal biogenesis. They are specifically involved in the processing beyond the 27S stage of 25S rRNA maturation. This family contains sequences that bear similarity to the glioma tumor suppressor candidate region gene 2 protein (p60). This protein has been found to interact with herpes simplex type 1 regulatory proteins.','pfam07767','Nop53'),(1842238,'gnl|CDD|309584',_binary 'Family of unknown function (DUF572).  Family of eukaryotic proteins with undetermined function.','pfam04502','DUF572'),(1843538,'gnl|CDD|133408',_binary 'Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily.  Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 6) with at least two conserved cysteines present in the C-terminal portion of the protein. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.','cd04781','HTH_MerR-like_sg6');
/*!40000 ALTER TABLE `domain` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `gene`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `gene` (
  `GeneID` varchar(35) NOT NULL DEFAULT '',
  `PhageID` varchar(25) NOT NULL,
  `Start` mediumint(9) NOT NULL,
  `Stop` mediumint(9) NOT NULL,
  `Length` mediumint(9) NOT NULL,
  `Name` varchar(50) NOT NULL,
  `TypeID` varchar(10) DEFAULT NULL,
  `translation` varchar(5000) DEFAULT NULL,
  `StartCodon` enum('ATG','GTG','TTG') DEFAULT NULL,
  `StopCodon` enum('TAA','TAG','TGA') DEFAULT NULL,
  `Orientation` enum('F','R') DEFAULT NULL,
  `GC1` float DEFAULT NULL,
  `GC2` float DEFAULT NULL,
  `GC3` float DEFAULT NULL,
  `GC` float DEFAULT NULL,
  `LeftNeighbor` varchar(25) DEFAULT NULL,
  `RightNeighbor` varchar(25) DEFAULT NULL,
  `Notes` blob,
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `clustalw_status` enum('avail','pending','stale','done') NOT NULL DEFAULT 'avail',
  `blast_status` enum('avail','pending','stale','done') NOT NULL DEFAULT 'avail',
  `cdd_status` tinyint(1) NOT NULL DEFAULT '0',
  `LocusTag` varchar(50) DEFAULT NULL,
  PRIMARY KEY (`GeneID`),
  KEY `PhageID` (`PhageID`),
  KEY `id` (`id`),
  CONSTRAINT `gene_ibfk_2` FOREIGN KEY (`PhageID`) REFERENCES `phage` (`PhageID`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=746996 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `gene` WRITE;
/*!40000 ALTER TABLE `gene` DISABLE KEYS */;
/*!40000 ALTER TABLE `gene` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `gene_domain`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `gene_domain` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `GeneID` varchar(35) DEFAULT NULL,
  `hit_id` varchar(25) NOT NULL,
  `query_start` int(10) unsigned NOT NULL,
  `query_end` int(10) unsigned NOT NULL,
  `expect` double unsigned NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `GeneID__hit_id` (`GeneID`,`hit_id`),
  KEY `hit_id` (`hit_id`),
  CONSTRAINT `gene_domain_ibfk_1` FOREIGN KEY (`GeneID`) REFERENCES `gene` (`GeneID`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `gene_domain_ibfk_2` FOREIGN KEY (`hit_id`) REFERENCES `domain` (`hit_id`)
) ENGINE=InnoDB AUTO_INCREMENT=1026657 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `gene_domain` WRITE;
/*!40000 ALTER TABLE `gene_domain` DISABLE KEYS */;
/*!40000 ALTER TABLE `gene_domain` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `host`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `host` (
  `ID` int(11) NOT NULL AUTO_INCREMENT,
  `Name` varchar(50) NOT NULL,
  `Accession` varchar(15) DEFAULT NULL,
  PRIMARY KEY (`ID`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `host` WRITE;
/*!40000 ALTER TABLE `host` DISABLE KEYS */;
INSERT INTO `host` VALUES (1,'Mtb',NULL),(2,'MKD8',NULL),(3,'Jucho',NULL);
/*!40000 ALTER TABLE `host` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `host_range`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `host_range` (
  `PhageID` varchar(25) NOT NULL,
  `host_id` int(11) NOT NULL,
  `plating_efficiency` double unsigned NOT NULL,
  `below_detection` tinyint(1) NOT NULL,
  KEY `PhageID` (`PhageID`),
  KEY `host_id` (`host_id`),
  CONSTRAINT `host_range_ibfk_1` FOREIGN KEY (`PhageID`) REFERENCES `phage` (`PhageID`),
  CONSTRAINT `host_range_ibfk_2` FOREIGN KEY (`host_id`) REFERENCES `host` (`ID`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `host_range` WRITE;
/*!40000 ALTER TABLE `host_range` DISABLE KEYS */;
/*!40000 ALTER TABLE `host_range` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `node`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `node` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `platform` varchar(15) DEFAULT NULL,
  `hostname` varchar(50) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `hostname_index` (`hostname`)
) ENGINE=InnoDB AUTO_INCREMENT=2536 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `node` WRITE;
/*!40000 ALTER TABLE `node` DISABLE KEYS */;
INSERT INTO `node` VALUES (2388,'linux2','phage'),(2389,'linux2','phamerator'),(2390,'linux2','ip-10-31-145-29'),(2391,'linux2','loki'),(2392,'linux2','ip-10-91-131-70'),(2393,'linux2','ip-10-91-166-111'),(2394,'linux2','ip-10-91-128-77'),(2395,'linux2','ip-10-91-167-56'),(2396,'linux2','ip-10-91-131-228'),(2397,'linux2','ip-10-91-166-31'),(2398,'linux2','ip-10-91-132-38'),(2399,'linux2','ip-10-91-167-21'),(2400,'linux2','ip-10-91-131-54'),(2401,'linux2','ip-10-91-167-55'),(2402,'linux2','ip-10-119-23-168'),(2403,'linux2','ip-10-98-214-197'),(2404,'linux2','ip-10-78-143-14'),(2405,'linux2','ip-10-84-178-112'),(2406,'linux2','ip-10-39-61-32'),(2407,'linux2','ip-10-38-221-118'),(2408,'linux2','ip-10-98-193-161'),(2409,'linux2','ip-10-79-29-206'),(2410,'linux2','ip-10-37-23-251'),(2411,'linux2','ip-10-212-218-64'),(2412,'linux2','ip-10-196-206-146'),(2413,'linux2','ip-10-79-77-160'),(2414,'linux2','ip-10-78-241-18'),(2415,'linux2','ip-10-99-74-64'),(2416,'linux2','ip-10-76-114-225'),(2417,'linux2','ip-10-242-138-214'),(2418,'linux2','ip-10-242-99-95'),(2419,'linux2','ip-10-244-242-194'),(2420,'linux2','ip-10-194-155-2'),(2421,'linux2','ip-10-32-167-156'),(2422,'linux2','ip-10-91-138-254'),(2423,'linux2','ip-10-234-37-53'),(2424,'linux2','ip-10-91-166-73'),(2425,'linux2','ip-10-91-168-8'),(2426,'linux2','ip-10-91-128-45'),(2427,'linux2','ip-10-91-135-36'),(2428,'linux2','ip-10-91-134-172'),(2429,'linux2','ip-10-91-130-156'),(2430,'linux2','ip-10-91-128-111'),(2431,'linux2','ip-10-91-142-89'),(2432,'linux2','ip-10-66-46-131'),(2433,'linux2','ip-10-3-105-49'),(2434,'linux2','ip-10-70-215-119'),(2435,'linux2','ip-10-47-31-144'),(2436,'linux2','domU-12-31-39-0F-6A-34'),(2437,'linux2','domU-12-31-39-06-A4-4A'),(2438,'linux2','domU-12-31-39-12-A8-64'),(2439,'linux2','domU-12-31-39-0F-52-93'),(2440,'linux2','domU-12-31-39-13-6D-B2'),(2441,'linux2','domU-12-31-39-0F-32-04'),(2442,'linux2','ip-10-91-165-91'),(2443,'linux2','domU-12-31-39-12-98-4D'),(2444,'linux2','ip-10-108-218-115'),(2445,'linux2','ip-10-159-24-183'),(2446,'linux2','ip-10-91-163-164'),(2447,'linux2','ip-10-64-161-101'),(2448,'linux2','ip-10-108-83-217'),(2449,'linux2','ip-10-159-10-24'),(2450,'linux2','ip-10-91-140-204'),(2451,'linux2','ip-10-91-165-112'),(2452,'linux2','ip-10-122-242-120'),(2453,'linux2','domU-12-31-38-04-1D-F9'),(2454,'linux2','ip-10-126-42-35'),(2455,'linux2','domU-12-31-38-07-05-19'),(2456,'linux2','domU-12-31-38-04-20-94'),(2457,'linux2','domU-12-31-38-04-88-DD'),(2458,'linux2','ip-10-122-223-9'),(2459,'linux2','ip-10-124-149-26'),(2460,'linux2','ip-10-122-238-33'),(2461,'linux2','domU-12-31-38-04-89-B3'),(2462,'linux2','odin'),(2463,'linux2','ip-10-91-167-60'),(2464,'linux2','ip-10-91-166-37'),(2465,'linux2','ip-10-91-167-17'),(2466,'linux2','ip-10-91-128-49'),(2467,'linux2','ip-10-91-129-188'),(2468,'linux2','ip-10-91-131-222'),(2469,'linux2','ip-10-91-134-76'),(2470,'linux2','ip-10-91-140-198'),(2471,'linux2','ip-10-91-130-8'),(2472,'linux2','ip-10-91-163-233'),(2473,'linux2','ip-10-205-25-78'),(2474,'linux2','ip-10-98-22-119'),(2475,'linux2','ip-10-98-115-242'),(2476,'linux2','ip-10-204-234-47'),(2477,'linux2','ip-10-195-209-187'),(2478,'linux2','ip-10-112-210-78'),(2479,'linux2','ip-10-76-17-93'),(2480,'linux2','ip-10-78-185-130'),(2481,'linux2','ip-10-36-251-168'),(2482,'linux2','ip-10-204-183-47'),(2483,'linux2','ip-10-35-13-34'),(2484,'linux2','ip-10-99-78-73'),(2485,'linux2','ip-10-39-17-50'),(2486,'linux2','ip-10-99-59-73'),(2487,'linux2','ip-10-212-217-155'),(2488,'linux2','ip-10-202-229-93'),(2489,'linux2','ip-10-242-255-130'),(2490,'linux2','ip-10-115-90-231'),(2491,'linux2','ip-10-195-130-143'),(2492,'linux2','ip-10-84-129-145'),(2493,'linux2','hatfull-file'),(2494,'linux2','hatfull-web'),(2495,'linux2','ip-10-202-231-166'),(2496,'linux2','ip-10-242-15-2'),(2497,'linux2','ip-10-119-11-180'),(2498,'linux2','ip-10-195-147-81'),(2499,'linux2','ip-10-32-191-115'),(2500,'linux2','ip-10-38-249-216'),(2501,'linux2','ip-10-84-175-232'),(2502,'linux2','ip-10-76-78-100'),(2503,'linux2','ip-10-78-49-23'),(2504,'linux2','ip-10-115-46-179'),(2505,'linux2','ip-10-98-238-227'),(2506,'linux2','ip-10-98-213-238'),(2507,'linux2','ip-10-115-98-74'),(2508,'linux2','ip-10-243-138-223'),(2509,'linux2','ip-10-195-153-229'),(2510,'linux2','n275'),(2511,'linux2','n277'),(2512,'linux2','n278'),(2513,'linux2','n276'),(2514,'linux2','n50'),(2515,'linux2','n6'),(2516,'linux2','n4'),(2517,'linux2','n7'),(2518,'linux2','n258'),(2519,'linux2','n257'),(2520,'linux2','n259'),(2521,'linux2','n256'),(2522,'linux2','n178'),(2523,'linux2','n243'),(2524,'linux2','n245'),(2525,'linux2','n247'),(2526,'linux2','n246'),(2527,'linux2','n265'),(2528,'linux2','n266'),(2529,'linux2','n269'),(2530,'linux2','n267'),(2531,'linux2','n48'),(2532,'linux2','n9'),(2533,'linux2','n49'),(2534,'linux2','n5'),(2535,'linux2','n8');
/*!40000 ALTER TABLE `node` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `phage`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `phage` (
  `PhageID` varchar(25) NOT NULL,
  `Accession` varchar(15) NOT NULL,
  `Name` varchar(50) NOT NULL,
  `Isolated` varchar(100) DEFAULT NULL,
  `HostStrain` varchar(50) DEFAULT NULL,
  `Sequence` mediumblob NOT NULL,
  `SequenceLength` mediumint(9) NOT NULL,
  `Prophage` enum('yes','no') DEFAULT NULL,
  `ProphageOffset` int(11) DEFAULT NULL,
  `DateLastModified` datetime DEFAULT NULL,
  `DateLastSearched` datetime DEFAULT NULL,
  `Notes` blob,
  `GC` float DEFAULT NULL,
  `Cluster` varchar(5) DEFAULT NULL,
  `status` varchar(5) DEFAULT NULL,
  `RetrieveRecord` tinyint(1) NOT NULL DEFAULT '0',
  `AnnotationQC` tinyint(1) NOT NULL DEFAULT '0',
  `AnnotationAuthor` tinyint(1) NOT NULL DEFAULT '0',
  `Cluster2` varchar(5) DEFAULT NULL,
  `Subcluster2` varchar(5) DEFAULT NULL,
  PRIMARY KEY (`PhageID`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `phage` WRITE;
/*!40000 ALTER TABLE `phage` DISABLE KEYS */;
/*!40000 ALTER TABLE `phage` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `pham`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `pham` (
  `GeneID` varchar(35) NOT NULL DEFAULT '',
  `name` int(10) unsigned DEFAULT NULL,
  `orderAdded` int(5) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`GeneID`),
  KEY `orderAdded_index` (`orderAdded`),
  KEY `name_index` (`name`),
  CONSTRAINT `pham_ibfk_1` FOREIGN KEY (`GeneID`) REFERENCES `gene` (`GeneID`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=281951 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `pham` WRITE;
/*!40000 ALTER TABLE `pham` DISABLE KEYS */;
/*!40000 ALTER TABLE `pham` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `pham_color`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `pham_color` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `name` int(10) unsigned NOT NULL,
  `color` char(7) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=23407 DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `pham_color` WRITE;
/*!40000 ALTER TABLE `pham_color` DISABLE KEYS */;
INSERT INTO `pham_color` VALUES (1,2,'#41f8f7'),(2,3,'#cf3ac9'),(3,4,'#d00f3c'),(4,7,'#dc4bcd'),(5,9,'#050fcd'),(6,10,'#da7e00'),(7,11,'#e467b5'),(8,12,'#fef044'),(9,15,'#d88a1a'),(10,16,'#fe9e30'),(11,17,'#FFFFFF'),(12,19,'#FFFFFF'),(13,23,'#97f87a'),(14,25,'#ed2cae'),(15,26,'#d57318'),(16,27,'#7bfe5a'),(17,29,'#42ea21'),(18,30,'#f48a36'),(19,31,'#fcc86c'),(20,33,'#703fd1'),(21,34,'#FFFFFF'),(22,37,'#31dbc2'),(23,38,'#e7179f'),(24,39,'#f5b96e'),(25,42,'#a80ace'),(26,43,'#82eb23'),(27,44,'#5e80e1'),(28,51,'#fd5653'),(29,56,'#6fcefe'),(30,58,'#ad3bf8'),(31,59,'#da246f'),(32,60,'#FFFFFF'),(33,61,'#FFFFFF'),(34,62,'#3fefdd'),(35,63,'#b306e7'),(36,64,'#FFFFFF'),(37,65,'#FFFFFF'),(38,66,'#e41792'),(39,69,'#16fc5e'),(40,70,'#FFFFFF'),(41,71,'#cbef48'),(42,73,'#19dc5b'),(43,74,'#6197fd'),(44,76,'#f56674'),(45,79,'#668ced'),(46,80,'#d59807'),(47,81,'#f4f86d'),(48,84,'#5318d9'),(49,88,'#3fd657'),(50,89,'#dfe564'),(51,91,'#f77b78'),(52,93,'#f6f16f'),(53,94,'#6df6ad'),(54,96,'#44ef49'),(55,98,'#06df73'),(56,99,'#37a2db'),(57,100,'#FFFFFF'),(58,101,'#8056cf'),(59,102,'#79f927'),(60,103,'#6578cd'),(61,104,'#e733e6'),(62,105,'#e36df0'),(63,106,'#ebc936'),(64,107,'#FFFFFF'),(65,109,'#f2eb11'),(66,114,'#16efc8'),(67,115,'#36ddfd'),(68,116,'#49bad8'),(69,118,'#6b8be5'),(70,120,'#d04959'),(71,121,'#FFFFFF'),(72,123,'#3be5af'),(73,124,'#e0922b'),(74,125,'#2ce7d7'),(75,126,'#8849e3'),(76,128,'#FFFFFF'),(77,129,'#FFFFFF'),(78,130,'#df436f'),(79,131,'#30cdc3'),(80,134,'#FFFFFF'),(81,135,'#6931da'),(82,137,'#f86e7f'),(83,140,'#6ef080'),(84,142,'#4982df'),(85,144,'#FFFFFF'),(86,145,'#5985d6'),(87,146,'#68d467'),(88,147,'#f73213'),(89,151,'#c7f65d'),(90,153,'#3fb7ee'),(91,155,'#d3935b'),(92,156,'#e6e710'),(93,158,'#5bf018'),(94,159,'#fdc358'),(95,162,'#69e0cb'),(96,163,'#26e037'),(97,164,'#fdc401'),(98,165,'#3857d8'),(99,166,'#d25743'),(100,167,'#e328ad'),(101,169,'#67fb05'),(102,170,'#4df1bf'),(103,175,'#dafe4f'),(104,176,'#f577c7'),(105,177,'#75fa6f'),(106,178,'#FFFFFF'),(107,182,'#ea5c4b'),(108,187,'#cff02c'),(109,188,'#ce1d05'),(110,191,'#33fe41'),(111,192,'#676cf6'),(112,193,'#6656ee'),(113,194,'#6053cd'),(114,196,'#ae46d8'),(115,200,'#de8e3c'),(116,202,'#e61ccd'),(117,203,'#FFFFFF'),(118,204,'#d8b129'),(119,208,'#4bd062'),(120,209,'#e7625f'),(121,210,'#945ad1'),(122,212,'#fad522'),(123,213,'#f66900'),(124,214,'#d69225'),(125,215,'#db02ba'),(126,216,'#6fe9c6'),(127,218,'#9f4bd2'),(128,219,'#efb174'),(129,220,'#e65f64'),(130,222,'#fb7d28'),(131,223,'#aae014'),(132,224,'#37d4af'),(133,226,'#cc9800'),(134,228,'#d5722e'),(135,230,'#20e2ba'),(136,231,'#29dbd5'),(137,233,'#e2c146'),(138,234,'#17d4d6'),(139,235,'#eeea48'),(140,237,'#d504aa'),(141,238,'#10dc7f'),(142,239,'#af6fec'),(143,240,'#4ee005'),(144,241,'#40da76'),(145,242,'#FFFFFF'),(146,243,'#b76af9'),(147,246,'#52f152'),(148,247,'#adef4f'),(149,248,'#60d08a'),(150,252,'#FFFFFF'),(151,253,'#4fa3ce'),(152,254,'#e3de6c'),(153,255,'#06d940'),(154,256,'#cee53e'),(155,260,'#d3598f'),(156,262,'#dd6023'),(157,264,'#89dd5c'),(158,265,'#8d11ce'),(159,266,'#e58e5d'),(160,268,'#4af599'),(161,269,'#26a4d5'),(162,272,'#55dbfa'),(163,275,'#893bdb'),(164,276,'#d8fa67'),(165,278,'#FFFFFF'),(166,279,'#9838e9'),(167,280,'#8e56eb'),(168,282,'#f68516'),(169,284,'#62eaa6'),(170,286,'#b1f15b'),(171,287,'#bfcc0e'),(172,288,'#50ade9'),(173,289,'#59cd25'),(174,291,'#b3e13e'),(175,293,'#cf3bc3'),(176,294,'#2364fd'),(177,295,'#cd13b3'),(178,297,'#FFFFFF'),(179,300,'#ea4c37'),(180,302,'#70e275'),(181,304,'#42d376'),(182,305,'#34ee9b'),(183,306,'#7489ed'),(184,308,'#d92b4a'),(185,312,'#f9159e'),(186,314,'#e44b9b'),(187,317,'#d5f46d'),(188,319,'#3ef83b'),(189,321,'#f8b309'),(190,322,'#2cda44'),(191,323,'#f1ca0c'),(192,326,'#45ee11'),(193,328,'#44e2fb'),(194,330,'#ce9d0f'),(195,332,'#5cf711'),(196,335,'#d1537c'),(197,336,'#FFFFFF'),(198,337,'#e831be'),(199,344,'#11ecc7'),(200,345,'#c2f858'),(201,346,'#c466fd'),(202,349,'#f19832'),(203,351,'#afe258'),(204,353,'#1a3fed'),(205,356,'#36ecf4'),(206,360,'#FFFFFF'),(207,362,'#FFFFFF'),(208,363,'#FFFFFF'),(209,364,'#de26ee'),(210,365,'#af53e5'),(211,368,'#db3a17'),(212,369,'#894cd9'),(213,374,'#06f6d1'),(214,375,'#ddf403'),(215,379,'#dd845a'),(216,380,'#FFFFFF'),(217,382,'#4c99fa'),(218,383,'#ed5cd8'),(219,385,'#3dde1a'),(220,386,'#9c65d6'),(221,387,'#4198df'),(222,388,'#ce3378'),(223,391,'#2f19fa'),(224,393,'#cc4b1b'),(225,395,'#10d38f'),(226,397,'#eb074f'),(227,398,'#cd855b'),(228,399,'#d5060b'),(229,400,'#d46149'),(230,402,'#FFFFFF'),(231,405,'#FFFFFF'),(232,406,'#1ce671'),(233,407,'#FFFFFF'),(234,409,'#12f3b8'),(235,410,'#9114f7'),(236,411,'#FFFFFF'),(237,413,'#755ed0'),(238,417,'#8b3bd3'),(239,420,'#FFFFFF'),(240,421,'#a7ec6b'),(241,423,'#FFFFFF'),(242,425,'#db6902'),(243,428,'#6bdb8b'),(244,429,'#0cce60'),(245,431,'#656ef4'),(246,432,'#FFFFFF'),(247,434,'#FFFFFF'),(248,436,'#4cf5a2'),(249,437,'#f94f6c'),(250,442,'#7a5ad4'),(251,444,'#8105e8'),(252,445,'#de6eb2'),(253,446,'#bddb5e'),(254,448,'#d02e9d'),(255,449,'#5ecdc9'),(256,451,'#ef03c9'),(257,452,'#4bcea0'),(258,454,'#5ae5a3'),(259,455,'#6486f5'),(260,456,'#83e824'),(261,459,'#d63b38'),(262,461,'#d031fa'),(263,462,'#2cf21d'),(264,463,'#771dce'),(265,464,'#f10301'),(266,465,'#f370cc'),(267,467,'#2ffe1f'),(268,468,'#b814e8'),(269,472,'#ea3679'),(270,473,'#f63b8e'),(271,479,'#113ad5'),(272,480,'#b018de'),(273,481,'#e60202'),(274,482,'#3a68db'),(275,483,'#00ea2f'),(276,484,'#1bf2e9'),(277,486,'#e3f235'),(278,487,'#FFFFFF'),(279,491,'#6719f6'),(280,492,'#47d3f2'),(281,493,'#7c2afa'),(282,494,'#cc3692'),(283,497,'#FFFFFF'),(284,499,'#58deba'),(285,507,'#5ac1ef'),(286,508,'#fd1281'),(287,509,'#6bdb31'),(288,510,'#e8553f'),(289,512,'#d2661a'),(290,513,'#6b71fb'),(291,515,'#ce6160'),(292,517,'#cbf52a'),(293,518,'#e16e91'),(294,520,'#fe4d56'),(295,523,'#e00bb9'),(296,524,'#FFFFFF'),(297,525,'#2bc1e2'),(298,526,'#de7a31'),(299,531,'#acfa18'),(300,532,'#d662d1'),(301,533,'#5ac4f5'),(302,534,'#cc10a8'),(303,535,'#FFFFFF'),(304,536,'#eb6ba1'),(305,538,'#FFFFFF'),(306,540,'#66d110'),(307,541,'#a645fd'),(308,542,'#b3e65f'),(309,543,'#2dd84a'),(310,544,'#FFFFFF'),(311,548,'#2bb2d2'),(312,549,'#a458e5'),(313,552,'#FFFFFF'),(314,555,'#FFFFFF'),(315,557,'#9972f5'),(316,558,'#f36fb1'),(317,560,'#7c48ef'),(318,561,'#dc6186'),(319,562,'#c0ee58'),(320,565,'#14ebdb'),(321,566,'#5be240'),(322,567,'#2f98cc'),(323,568,'#FFFFFF'),(324,569,'#891ece'),(325,570,'#67f2fa'),(326,572,'#FFFFFF'),(327,576,'#9c36d3'),(328,577,'#efb544'),(329,578,'#d87a00'),(330,579,'#20f37b'),(331,580,'#f46d62'),(332,581,'#63f730'),(333,582,'#d66346'),(334,585,'#bf51cd'),(335,587,'#36e0a1'),(336,588,'#58d1ce'),(337,594,'#95e91a'),(338,595,'#4fe447'),(339,597,'#d4197a'),(340,599,'#d2905f'),(341,600,'#2edebf'),(342,601,'#2bdfc9'),(343,602,'#FFFFFF'),(344,604,'#8729e9'),(345,605,'#FFFFFF'),(346,606,'#672be0'),(347,607,'#4af1ec'),(348,609,'#52f8dd'),(349,610,'#31f064'),(350,611,'#b35cda'),(351,614,'#FFFFFF'),(352,615,'#fa43a7'),(353,616,'#8dfd07'),(354,618,'#bacd27'),(355,621,'#6ee5a3'),(356,624,'#1770f9'),(357,629,'#FFFFFF'),(358,630,'#b8da66'),(359,631,'#ecb372'),(360,635,'#ec8f46'),(361,636,'#16e0bf'),(362,637,'#FFFFFF'),(363,638,'#2cf049'),(364,640,'#07cce9'),(365,642,'#FFFFFF'),(366,643,'#ec5ffe'),(367,644,'#FFFFFF'),(368,645,'#bd68e2'),(369,648,'#3f69f8'),(370,653,'#2e1aed'),(371,654,'#88f32b'),(372,655,'#ee4692'),(373,656,'#841bf2'),(374,657,'#f16ead'),(375,658,'#51ecf1'),(376,659,'#5debf6'),(377,661,'#e53ae2'),(378,662,'#FFFFFF'),(379,665,'#fdd35f'),(380,666,'#03ef56'),(381,668,'#0ee5f1'),(382,672,'#e85c50'),(383,673,'#0f9df8'),(384,674,'#e4b01e'),(385,675,'#1d91fd'),(386,679,'#FFFFFF'),(387,682,'#d83b9e'),(388,684,'#e23497'),(389,685,'#0c5eeb'),(390,686,'#d8cf67'),(391,687,'#d86248'),(392,689,'#62ddab'),(393,692,'#d417b5'),(394,695,'#f94d79'),(395,698,'#0df996'),(396,700,'#bc76f0'),(397,702,'#8323f3'),(398,704,'#FFFFFF'),(399,706,'#cc4f65'),(400,708,'#FFFFFF'),(401,710,'#f3e72f'),(402,711,'#fb499a'),(403,712,'#11f58d'),(404,713,'#77c7fc'),(405,715,'#777ef7'),(406,716,'#f873d3'),(407,718,'#55a5e2'),(408,719,'#1f3cda'),(409,722,'#f0896c'),(410,724,'#6757da'),(411,729,'#74a0f3'),(412,730,'#ed156e'),(413,732,'#2af257'),(414,733,'#d9240f'),(415,736,'#d555a7'),(416,738,'#e9628f'),(417,739,'#0f8ad6'),(418,741,'#da7e66'),(419,742,'#d35d65'),(420,743,'#FFFFFF'),(421,744,'#1866d1'),(422,746,'#70d034'),(423,748,'#75f6bd'),(424,750,'#ce3876'),(425,751,'#f8012f'),(426,752,'#b8fe22'),(427,753,'#e0796a'),(428,756,'#5bd370'),(429,757,'#FFFFFF'),(430,759,'#91fb24'),(431,760,'#eb6c6d'),(432,762,'#e0ef26'),(433,763,'#b942d1'),(434,765,'#3c45e3'),(435,769,'#3772d3'),(436,772,'#c2d520'),(437,773,'#c7eb50'),(438,774,'#d7c039'),(439,775,'#83d730'),(440,776,'#0c32cc'),(441,777,'#2b1cd1'),(442,778,'#FFFFFF'),(443,779,'#FFFFFF'),(444,780,'#3b5cda'),(445,781,'#cdf53a'),(446,783,'#29ed6c'),(447,785,'#FFFFFF'),(448,787,'#30d4cf'),(449,788,'#d01e0e'),(450,789,'#ce8b3d'),(451,790,'#ea9f39'),(452,793,'#f92e9d'),(453,794,'#648fe8'),(454,798,'#15a9fd'),(455,800,'#8ee759'),(456,803,'#46e0f7'),(457,808,'#aff018'),(458,812,'#55feb6'),(459,815,'#d20fd5'),(460,818,'#dd25dd'),(461,820,'#1fe796'),(462,822,'#d1b767'),(463,823,'#f46598'),(464,826,'#85d352'),(465,828,'#67eeb5'),(466,832,'#FFFFFF'),(467,833,'#4b74d5'),(468,834,'#83d40d'),(469,836,'#5f18f3'),(470,837,'#ef74a2'),(471,838,'#33e93a'),(472,839,'#5013d0'),(473,840,'#1ae876'),(474,842,'#2ed5f2'),(475,843,'#dc4386'),(476,845,'#d7348e'),(477,848,'#960be7'),(478,849,'#944ad3'),(479,857,'#5c86e3'),(480,860,'#daea5f'),(481,861,'#ebee62'),(482,862,'#54c6d9'),(483,863,'#7ccaf9'),(484,864,'#92ef2f'),(485,866,'#24d735'),(486,867,'#FFFFFF'),(487,868,'#5e00ea'),(488,869,'#de816b'),(489,870,'#45e9d7'),(490,872,'#673af5'),(491,876,'#48e925'),(492,878,'#cad25a'),(493,879,'#e14ada'),(494,880,'#FFFFFF'),(495,881,'#3ad212'),(496,884,'#d8af53'),(497,886,'#ec3a22'),(498,887,'#e39c06'),(499,890,'#fa0297'),(500,893,'#2670cc'),(501,895,'#15d839'),(502,896,'#aa49e8'),(503,898,'#dc6237'),(504,899,'#4b05e5'),(505,900,'#c63def'),(506,901,'#e54b3d'),(507,902,'#b15cf3'),(508,903,'#FFFFFF'),(509,904,'#5378d2'),(510,905,'#36cc54'),(511,906,'#cc4f81'),(512,909,'#1013d8'),(513,910,'#fd5ab9'),(514,912,'#62e730'),(515,913,'#6de28d'),(516,915,'#e79a06'),(517,916,'#FFFFFF'),(518,917,'#57ccb3'),(519,921,'#d99448'),(520,922,'#3bfec4'),(521,923,'#556ef0'),(522,924,'#62cf41'),(523,927,'#5a75d4'),(524,928,'#a2f34d'),(525,929,'#0ed40c'),(526,930,'#d3ba20'),(527,931,'#6043da'),(528,932,'#2822f6'),(529,933,'#a9f954'),(530,935,'#8247eb'),(531,939,'#FFFFFF'),(532,940,'#a875fe'),(533,941,'#FFFFFF'),(534,944,'#d8771e'),(535,946,'#01fca8'),(536,947,'#e6c972'),(537,948,'#41e13c'),(538,949,'#28cd67'),(539,950,'#FFFFFF'),(540,953,'#672de8'),(541,955,'#d070e4'),(542,956,'#d4bf5f'),(543,958,'#2bd248'),(544,959,'#a835fb'),(545,960,'#ef74ed'),(546,963,'#30cc7c'),(547,964,'#645fde'),(548,965,'#FFFFFF'),(549,966,'#e743c4'),(550,967,'#d13fe8'),(551,968,'#FFFFFF'),(552,969,'#FFFFFF'),(553,971,'#7f0cfd'),(554,974,'#46f2e3'),(555,976,'#FFFFFF'),(556,977,'#d074f7'),(557,978,'#03e0e4'),(558,982,'#c7ed39'),(559,984,'#dc3b61'),(560,985,'#68fb40'),(561,986,'#f3625c'),(562,987,'#7ded2c'),(563,988,'#FFFFFF'),(564,989,'#264ff9'),(565,990,'#66cfbf'),(566,991,'#6471e3'),(567,992,'#d93b59'),(568,993,'#77d55d'),(569,994,'#61e488'),(570,995,'#d7564f'),(571,996,'#ddab09'),(572,997,'#0aa3f8'),(573,1000,'#dc392b'),(574,1001,'#21f525'),(575,1002,'#82d512'),(576,1005,'#FFFFFF'),(577,1006,'#69f284'),(578,1008,'#FFFFFF'),(579,1009,'#cd0a29'),(580,1010,'#ddbd3d'),(581,1011,'#FFFFFF'),(582,1012,'#07d01d'),(583,1013,'#1cdb4e'),(584,1014,'#87cc29'),(585,1018,'#10e7f0'),(586,1019,'#59e5d4'),(587,1022,'#4087cd'),(588,1024,'#cacc39'),(589,1026,'#2d81cd'),(590,1029,'#faab29'),(591,1030,'#dc5be0'),(592,1031,'#d49b53'),(593,1033,'#bfe157'),(594,1041,'#1bd270'),(595,1042,'#e91b5a'),(596,1044,'#bbcc50'),(597,1049,'#cdca41'),(598,1051,'#e19425'),(599,1053,'#ee3235'),(600,1054,'#FFFFFF'),(601,1057,'#0dd443'),(602,1058,'#d29e41'),(603,1061,'#31d219'),(604,1062,'#FFFFFF'),(605,1063,'#FFFFFF'),(606,1064,'#ed0b45'),(607,1066,'#4987f5'),(608,1068,'#7af9e2'),(609,1069,'#6d2dd3'),(610,1071,'#FFFFFF'),(611,1073,'#cd3351'),(612,1074,'#FFFFFF'),(613,1075,'#ee7074'),(614,1079,'#f7f134'),(615,1080,'#4c5fee'),(616,1082,'#d12fb6'),(617,1083,'#ded935'),(618,1086,'#4428e0'),(619,1088,'#25e84b'),(620,1089,'#66db90'),(621,1090,'#b7e53f'),(622,1093,'#4e3bdb'),(623,1094,'#d7b95f'),(624,1095,'#5e09ef'),(625,1096,'#b675f0'),(626,1097,'#e22c18'),(627,1098,'#df4b37'),(628,1099,'#a9d82b'),(629,1103,'#4ed4ca'),(630,1104,'#22d34f'),(631,1107,'#04e393'),(632,1111,'#66ded1'),(633,1112,'#e51fe6'),(634,1113,'#8ee72d'),(635,1115,'#d33966'),(636,1118,'#FFFFFF'),(637,1120,'#8f65e5'),(638,1121,'#FFFFFF'),(639,1122,'#FFFFFF'),(640,1125,'#f25250'),(641,1127,'#c6d03c'),(642,1129,'#7ad436'),(643,1130,'#8b71e2'),(644,1133,'#FFFFFF'),(645,1135,'#FFFFFF'),(646,1136,'#cea74a'),(647,1139,'#eaca07'),(648,1141,'#c1d357'),(649,1143,'#f35cb5'),(650,1146,'#fb31c2'),(651,1147,'#b4d25e'),(652,1149,'#fb9618'),(653,1150,'#67d8ac'),(654,1151,'#678bd6'),(655,1152,'#1ae4cd'),(656,1155,'#9167e7'),(657,1156,'#FFFFFF'),(658,1158,'#ec1974'),(659,1160,'#d814cd'),(660,1161,'#b9d41c'),(661,1162,'#d26f17'),(662,1163,'#f1830b'),(663,1165,'#fbfc56'),(664,1166,'#36fa34'),(665,1168,'#0becb1'),(666,1169,'#5b0cda'),(667,1170,'#a955f6'),(668,1172,'#dbe06e'),(669,1174,'#58d982'),(670,1176,'#e19914'),(671,1177,'#44b3e2'),(672,1178,'#00fe28'),(673,1180,'#FFFFFF'),(674,1182,'#FFFFFF'),(675,1183,'#f3770c'),(676,1184,'#5d17fc'),(677,1185,'#0bf993'),(678,1189,'#3e54d1'),(679,1190,'#6f6beb'),(680,1191,'#68ec2b'),(681,1195,'#2fd8e9'),(682,1197,'#5bf99b'),(683,1199,'#637dfa'),(684,1202,'#ed44c4'),(685,1206,'#d54349'),(686,1208,'#FFFFFF'),(687,1209,'#FFFFFF'),(688,1211,'#efed70'),(689,1213,'#a7d050'),(690,1214,'#c5cf16'),(691,1216,'#FFFFFF'),(692,1221,'#6684f8'),(693,1228,'#53f8a9'),(694,1231,'#ce985e'),(695,1232,'#4f54e6'),(696,1233,'#4bcf4c'),(697,1234,'#fef14b'),(698,1235,'#bff923'),(699,1236,'#f72a0f'),(700,1242,'#511be5'),(701,1247,'#ba3ef2'),(702,1248,'#2cdf28'),(703,1251,'#f24093'),(704,1260,'#fec61e'),(705,1262,'#498ae2'),(706,1264,'#73df6a'),(707,1265,'#FFFFFF'),(708,1267,'#728fe7'),(709,1268,'#2bd6d3'),(710,1269,'#c22ee8'),(711,1270,'#1175cd'),(712,1271,'#1362d0'),(713,1272,'#16e7fc'),(714,1273,'#FFFFFF'),(715,1274,'#4638ce'),(716,1275,'#df1326'),(717,1278,'#FFFFFF'),(718,1279,'#3756ea'),(719,1280,'#FFFFFF'),(720,1281,'#fa3a8c'),(721,1292,'#06ce13'),(722,1294,'#FFFFFF'),(723,1295,'#6ce5aa'),(724,1296,'#09d779'),(725,1297,'#d40451'),(726,1299,'#de2a2a'),(727,1300,'#6821f5'),(728,1301,'#d50ff3'),(729,1302,'#FFFFFF'),(730,1307,'#f31f4c'),(731,1309,'#67edee'),(732,1311,'#f33e7b'),(733,1315,'#FFFFFF'),(734,1316,'#d78b01'),(735,1318,'#f20679'),(736,1319,'#37cc95'),(737,1320,'#5a4be8'),(738,1323,'#7b28de'),(739,1325,'#4ef9ea'),(740,1327,'#fdea26'),(741,1328,'#ef56eb'),(742,1330,'#d1cb56'),(743,1331,'#FFFFFF'),(744,1333,'#cc0ea9'),(745,1334,'#05d597'),(746,1336,'#b37cfa'),(747,1338,'#53bed5'),(748,1339,'#1f83df'),(749,1340,'#d948b0'),(750,1341,'#79ea37'),(751,1342,'#8ef534'),(752,1343,'#9fef11'),(753,1344,'#41dc99'),(754,1345,'#e03079'),(755,1346,'#7ee109'),(756,1347,'#b05fcf'),(757,1349,'#d19c44'),(758,1351,'#ed734c'),(759,1354,'#FFFFFF'),(760,1356,'#c9d15f'),(761,1357,'#670de1'),(762,1362,'#cd34c6'),(763,1363,'#e2834f'),(764,1364,'#8a4fdf'),(765,1365,'#149df9'),(766,1366,'#290edf'),(767,1369,'#cfd74e'),(768,1371,'#f278f5'),(769,1373,'#7574ec'),(770,1375,'#FFFFFF'),(771,1378,'#f81375'),(772,1382,'#ebce31'),(773,1386,'#3f9bd9'),(774,1388,'#ce4e1e'),(775,1393,'#FFFFFF'),(776,1394,'#FFFFFF'),(777,1395,'#77ec55'),(778,1398,'#d0b845'),(779,1399,'#72f3fd'),(780,1400,'#fd7db3'),(781,1401,'#e76551'),(782,1402,'#2acf8a'),(783,1404,'#3b43d4'),(784,1405,'#9a60fb'),(785,1408,'#52dbf8'),(786,1409,'#6328eb'),(787,1411,'#fd7666'),(788,1412,'#FFFFFF'),(789,1415,'#FFFFFF'),(790,1418,'#79dd5f'),(791,1419,'#735ad1'),(792,1422,'#FFFFFF'),(793,1428,'#41dd47'),(794,1429,'#0bf254'),(795,1431,'#a562e1'),(796,1432,'#d59568'),(797,1433,'#b86ce0'),(798,1434,'#f42d2d'),(799,1435,'#bff076'),(800,1436,'#4474e9'),(801,1437,'#cbf754'),(802,1439,'#5111ee'),(803,1441,'#6fce46'),(804,1442,'#7c4dd8'),(805,1443,'#e98466'),(806,1444,'#fc5589'),(807,1445,'#69e37c'),(808,1446,'#FFFFFF'),(809,1447,'#3168cc'),(810,1450,'#df128e'),(811,1452,'#e58665'),(812,1453,'#f43d87'),(813,1454,'#fb8d63'),(814,1456,'#04c6ef'),(815,1457,'#db0882'),(816,1459,'#dd4770'),(817,1464,'#e34486'),(818,1468,'#0c9ad1'),(819,1470,'#67e1d6'),(820,1474,'#f8f83c'),(821,1475,'#3bd250'),(822,1476,'#ee0b06'),(823,1479,'#fa499c'),(824,1480,'#FFFFFF'),(825,1481,'#67b5dd'),(826,1484,'#10e741'),(827,1485,'#4e58e6'),(828,1486,'#FFFFFF'),(829,1487,'#6356ed'),(830,1488,'#70fa31'),(831,1489,'#b6f306'),(832,1491,'#d7124f'),(833,1492,'#39afcc'),(834,1496,'#5a5de6'),(835,1498,'#d2a064'),(836,1499,'#f2801d'),(837,1500,'#64ecd2'),(838,1503,'#ae40db'),(839,1504,'#9c6feb'),(840,1506,'#38b5f8'),(841,1507,'#36cf8d'),(842,1508,'#a8ee43'),(843,1510,'#f248f5'),(844,1512,'#2cf293'),(845,1513,'#db190a'),(846,1517,'#FFFFFF'),(847,1519,'#9b23ef'),(848,1521,'#5493e3'),(849,1522,'#3191ed'),(850,1523,'#06dd90'),(851,1524,'#d7c002'),(852,1525,'#a0db12'),(853,1526,'#fe28e6'),(854,1529,'#60f8f6'),(855,1531,'#d9d746'),(856,1535,'#ee9c09'),(857,1545,'#270ee4'),(858,1547,'#ce34d0'),(859,1548,'#8950fd'),(860,1549,'#56b1f2'),(861,1550,'#c5e63a'),(862,1553,'#FFFFFF'),(863,1554,'#f33f9a'),(864,1556,'#cd63c4'),(865,1557,'#10ce5e'),(866,1564,'#edc74e'),(867,1565,'#f21133'),(868,1566,'#8320d6'),(869,1568,'#83f16a'),(870,1571,'#FFFFFF'),(871,1572,'#26c3db'),(872,1573,'#bee11c'),(873,1577,'#d12200'),(874,1578,'#41bafa'),(875,1579,'#10fb6f'),(876,1580,'#e2c847'),(877,1581,'#e255d4'),(878,1584,'#FFFFFF'),(879,1586,'#42cbe9'),(880,1587,'#11f064'),(881,1588,'#6bdec7'),(882,1591,'#FFFFFF'),(883,1592,'#5664e4'),(884,1593,'#4699dd'),(885,1594,'#a0e368'),(886,1595,'#7ccc45'),(887,1599,'#fbd768'),(888,1600,'#f870dc'),(889,1601,'#b379f9'),(890,1602,'#78fcca'),(891,1603,'#583ad0'),(892,1604,'#dded25'),(893,1605,'#5ae2eb'),(894,1606,'#e46d21'),(895,1608,'#fc8045'),(896,1609,'#0fdd6e'),(897,1610,'#1055e7'),(898,1611,'#FFFFFF'),(899,1612,'#06ea33'),(900,1613,'#d45219'),(901,1615,'#5bec2d'),(902,1617,'#FFFFFF'),(903,1618,'#fb7f3d'),(904,1621,'#61f296'),(905,1633,'#5bebe1'),(906,1635,'#e263fa'),(907,1637,'#FFFFFF'),(908,1638,'#d61d18'),(909,1639,'#4af2b7'),(910,1641,'#2ff68e'),(911,1642,'#bfde1a'),(912,1643,'#fe3c43'),(913,1644,'#2ddb89'),(914,1646,'#a03cd5'),(915,1647,'#FFFFFF'),(916,1648,'#2cfda2'),(917,1657,'#74e9a0'),(918,1658,'#5852d7'),(919,1660,'#f35088'),(920,1661,'#5b0ee4'),(921,1662,'#26d4f2'),(922,1663,'#FFFFFF'),(923,1664,'#56d3d8'),(924,1669,'#fda902'),(925,1671,'#54e36a'),(926,1672,'#d7148b'),(927,1673,'#666ad5'),(928,1675,'#47ded7'),(929,1677,'#503bcd'),(930,1678,'#88ed21'),(931,1684,'#07d33c'),(932,1686,'#4cd4ae'),(933,1694,'#FFFFFF'),(934,1697,'#da63bc'),(935,1699,'#61d7de'),(936,1700,'#df721b'),(937,1701,'#FFFFFF'),(938,1702,'#FFFFFF'),(939,1703,'#eb5193'),(940,1704,'#21fe89'),(941,1707,'#62c7f9'),(942,1708,'#4e21dc'),(943,1709,'#18f93e'),(944,1711,'#5df8f9'),(945,1712,'#fac938'),(946,1714,'#d1ce66'),(947,1715,'#3dcc7b'),(948,1716,'#d3cb67'),(949,1718,'#bb25e5'),(950,1723,'#e66941'),(951,1726,'#58df06'),(952,1729,'#9464ce'),(953,1730,'#54baf6'),(954,1732,'#0626de'),(955,1734,'#da7523'),(956,1736,'#FFFFFF'),(957,1737,'#82f90f'),(958,1739,'#d5295b'),(959,1740,'#f26e70'),(960,1741,'#32fa8f'),(961,1743,'#d73be9'),(962,1744,'#FFFFFF'),(963,1745,'#9460e4'),(964,1746,'#FFFFFF'),(965,1747,'#a448ed'),(966,1748,'#0442e2'),(967,1749,'#66e84f'),(968,1750,'#6cdbf6'),(969,1753,'#FFFFFF'),(970,1754,'#FFFFFF'),(971,1756,'#9f6aeb'),(972,1758,'#70cefd'),(973,1759,'#FFFFFF'),(974,1760,'#6897d1'),(975,1761,'#df47ac'),(976,1765,'#f48733'),(977,1769,'#1bd129'),(978,1770,'#f5c459'),(979,1771,'#73ea86'),(980,1772,'#d6dd3f'),(981,1773,'#cc4e42'),(982,1778,'#d54834'),(983,1779,'#FFFFFF'),(984,1780,'#d07110'),(985,1781,'#fbef17'),(986,1783,'#FFFFFF'),(987,1784,'#aedf5b'),(988,1785,'#b7da45'),(989,1787,'#FFFFFF'),(990,1790,'#2ef456'),(991,1791,'#ea6982'),(992,1792,'#16bbfb'),(993,1795,'#13fd20'),(994,1796,'#c6ce06'),(995,1797,'#cc662b'),(996,1798,'#6a5ff1'),(997,1800,'#a364e5'),(998,1802,'#f7ee2b'),(999,1803,'#f07808'),(1000,1805,'#2b23f4'),(1001,1806,'#e43f9d'),(1002,1808,'#FFFFFF'),(1003,1812,'#d25973'),(1004,1814,'#1fe5aa'),(1005,1816,'#eb43ab'),(1006,1821,'#FFFFFF'),(1007,1822,'#ec46ee'),(1008,1824,'#f56772'),(1009,1825,'#7fce23'),(1010,1828,'#FFFFFF'),(1011,1829,'#e8ec5b'),(1012,1830,'#74e711'),(1013,1831,'#c7e064'),(1014,1833,'#70e17b'),(1015,1836,'#cd7b06'),(1016,1838,'#46d5ac'),(1017,1839,'#96dc5f'),(1018,1840,'#fa3675'),(1019,1842,'#6bd6e1'),(1020,1844,'#d33379'),(1021,1846,'#d829be'),(1022,1847,'#dc4123'),(1023,1852,'#5acf3c'),(1024,1853,'#677ae2'),(1025,1854,'#e1c20a'),(1026,1855,'#5d5eda'),(1027,1856,'#840ed3'),(1028,1857,'#c154d5'),(1029,1859,'#94da3f'),(1030,1860,'#fa02f8'),(1031,1862,'#4c26db'),(1032,1863,'#24fc4b'),(1033,1868,'#fa5bcc'),(1034,1869,'#71ccf7'),(1035,1870,'#4ef54f'),(1036,1873,'#e84260'),(1037,1875,'#f57219'),(1038,1877,'#f9cb7b'),(1039,1878,'#d75fe8'),(1040,1879,'#a94adc'),(1041,1880,'#e39d58'),(1042,1883,'#10a6f4'),(1043,1885,'#d65606'),(1044,1886,'#fd0b5b'),(1045,1889,'#23fdc9'),(1046,1890,'#7bee08'),(1047,1891,'#be03e7'),(1048,1895,'#e48116'),(1049,1898,'#b15afe'),(1050,1900,'#dd4635'),(1051,1902,'#FFFFFF'),(1052,1904,'#82fe4e'),(1053,1906,'#1bd76a'),(1054,1907,'#36dc35'),(1055,1909,'#bed267'),(1056,1912,'#57d495'),(1057,1913,'#da4038'),(1058,1914,'#2dfe39'),(1059,1915,'#fdae32'),(1060,1917,'#FFFFFF'),(1061,1918,'#FFFFFF'),(1062,1919,'#f96ceb'),(1063,1920,'#24a2cf'),(1064,1921,'#6434d5'),(1065,1922,'#dc1360'),(1066,1924,'#506cce'),(1067,1926,'#f172f6'),(1068,1927,'#f8f659'),(1069,1928,'#616cdb'),(1070,1929,'#a422e8'),(1071,1930,'#2fe987'),(1072,1931,'#06c0e8'),(1073,1932,'#2de6e4'),(1074,1937,'#b723d3'),(1075,1938,'#41d2e3'),(1076,1939,'#29e389'),(1077,1940,'#5c97f7'),(1078,1941,'#902eda'),(1079,1943,'#21cdbe'),(1080,1944,'#e02c38'),(1081,1945,'#fc951d'),(1082,1948,'#FFFFFF'),(1083,1950,'#17b5e4'),(1084,1951,'#df3a18'),(1085,1952,'#e82380'),(1086,1953,'#FFFFFF'),(1087,1956,'#FFFFFF'),(1088,1958,'#d38829'),(1089,1963,'#ce13ae'),(1090,1968,'#eaf43a'),(1091,1970,'#fc4fa7'),(1092,1971,'#d952e1'),(1093,1973,'#0a20cf'),(1094,1974,'#3421d8'),(1095,1975,'#FFFFFF'),(1096,1976,'#FFFFFF'),(1097,1978,'#42d7a7'),(1098,1979,'#e47073'),(1099,1980,'#6999f6'),(1100,1981,'#f155da'),(1101,1982,'#6887fd'),(1102,1983,'#8aea39'),(1103,1984,'#f0ed5c'),(1104,1985,'#3df435'),(1105,1986,'#0dd9ee'),(1106,1989,'#06fe3e'),(1107,1991,'#3e1bf0'),(1108,1992,'#9ae93b'),(1109,1993,'#d815cb'),(1110,1995,'#FFFFFF'),(1111,1997,'#61bbf1'),(1112,1998,'#eb5a01'),(1113,1999,'#579ccf'),(1114,2002,'#3813e7'),(1115,2005,'#FFFFFF'),(1116,2006,'#fa4700'),(1117,2007,'#f3b730'),(1118,2008,'#d61cb6'),(1119,2009,'#cfef4b'),(1120,2010,'#d63c05'),(1121,2011,'#FFFFFF'),(1122,2015,'#e5e660'),(1123,2016,'#1feec5'),(1124,2017,'#9fd132'),(1125,2018,'#cce561'),(1126,2020,'#76f4b9'),(1127,2022,'#e30dfc'),(1128,2025,'#f7355e'),(1129,2027,'#57dc59'),(1130,2028,'#FFFFFF'),(1131,2029,'#e810d3'),(1132,2030,'#06fd87'),(1133,2031,'#f77ab1'),(1134,2032,'#332ef0'),(1135,2033,'#23a9d4'),(1136,2035,'#b51acc'),(1137,2036,'#617ad5'),(1138,2040,'#48dcd0'),(1139,2041,'#01eb1b'),(1140,2045,'#27ec32'),(1141,2049,'#c1d133'),(1142,2057,'#50cf61'),(1143,2060,'#d31a9d'),(1144,2061,'#d3a313'),(1145,2063,'#b7fd44'),(1146,2064,'#79e358'),(1147,2065,'#4011ed'),(1148,2066,'#68e04f'),(1149,2067,'#b419e3'),(1150,2068,'#FFFFFF'),(1151,2069,'#51d87f'),(1152,2070,'#d750b4'),(1153,2072,'#63dc88'),(1154,2075,'#7839e0'),(1155,2076,'#170ce7'),(1156,2078,'#9f6af6'),(1157,2079,'#57b0f4'),(1158,2080,'#5412d3'),(1159,2081,'#8ad450'),(1160,2084,'#69cbd4'),(1161,2087,'#4db5cd'),(1162,2092,'#d71310'),(1163,2093,'#e9cb49'),(1164,2094,'#1eccac'),(1165,2097,'#1ccef1'),(1166,2098,'#FFFFFF'),(1167,2101,'#f314bd'),(1168,2103,'#0f93ef'),(1169,2104,'#FFFFFF'),(1170,2105,'#FFFFFF'),(1171,2106,'#e743c4'),(1172,2107,'#FFFFFF'),(1173,2108,'#ccf574'),(1174,2109,'#e53f87'),(1175,2110,'#cbe344'),(1176,2112,'#6cd90c'),(1177,2114,'#e9478d'),(1178,2115,'#f65110'),(1179,2116,'#e7b941'),(1180,2119,'#29c3e0'),(1181,2120,'#e7d236'),(1182,2121,'#FFFFFF'),(1183,2122,'#FFFFFF'),(1184,2123,'#35e0e1'),(1185,2124,'#64f49b'),(1186,2126,'#6cf15e'),(1187,2129,'#5de58d'),(1188,2130,'#2dded8'),(1189,2131,'#f212a1'),(1190,2132,'#e85c5d'),(1191,2134,'#33edc5'),(1192,2137,'#f3e45e'),(1193,2139,'#faf32e'),(1194,2141,'#f57aa8'),(1195,2142,'#0edac9'),(1196,2144,'#92ce4a'),(1197,2147,'#883ce9'),(1198,2148,'#76db0c'),(1199,2149,'#00e263'),(1200,2151,'#4070da'),(1201,2152,'#7637d8'),(1202,2153,'#e309f8'),(1203,2154,'#0fd68f'),(1204,2155,'#0522cf'),(1205,2157,'#cfb346'),(1206,2158,'#63f133'),(1207,2160,'#1d42cc'),(1208,2161,'#913af4'),(1209,2164,'#d5eb37'),(1210,2165,'#f0e272'),(1211,2169,'#d733df'),(1212,2172,'#cfe36e'),(1213,2173,'#44e932'),(1214,2174,'#e06a96'),(1215,2175,'#cf3412'),(1216,2176,'#31ec44'),(1217,2177,'#FFFFFF'),(1218,2178,'#15ed29'),(1219,2179,'#f9892d'),(1220,2181,'#88d20b'),(1221,2182,'#69f0bc'),(1222,2183,'#1886f6'),(1223,2184,'#98d40d'),(1224,2186,'#5d24fd'),(1225,2191,'#6334d0'),(1226,2194,'#FFFFFF'),(1227,2195,'#FFFFFF'),(1228,2196,'#d95761'),(1229,2199,'#e0ae62'),(1230,2200,'#03dd66'),(1231,2201,'#FFFFFF'),(1232,2202,'#e2be35'),(1233,2203,'#f15dd0'),(1234,2204,'#cc4036'),(1235,2205,'#10dc54'),(1236,2206,'#23f722'),(1237,2207,'#d02703'),(1238,2208,'#e04634'),(1239,2210,'#d3942d'),(1240,2212,'#4bd513'),(1241,2217,'#9673ef'),(1242,2218,'#f2c16d'),(1243,2219,'#FFFFFF'),(1244,2223,'#df9468'),(1245,2224,'#6fd340'),(1246,2225,'#46d3b0'),(1247,2226,'#312cf5'),(1248,2227,'#3fce7a'),(1249,2233,'#a9f64b'),(1250,2234,'#f0fe06'),(1251,2235,'#e21dd3'),(1252,2236,'#44e082'),(1253,2237,'#cd5734'),(1254,2238,'#bae367'),(1255,2240,'#804ffb'),(1256,2241,'#09d32d'),(1257,2243,'#0bda98'),(1258,2244,'#a255d1'),(1259,2246,'#42d514'),(1260,2247,'#f25ced'),(1261,2248,'#63a1e2'),(1262,2249,'#6595f3'),(1263,2250,'#0eec85'),(1264,2251,'#d055cb'),(1265,2252,'#d111b3'),(1266,2255,'#0925db'),(1267,2257,'#FFFFFF'),(1268,2258,'#0ef787'),(1269,2260,'#a61fda'),(1270,2263,'#cc643d'),(1271,2266,'#41e665'),(1272,2269,'#df1f4d'),(1273,2270,'#e63f6c'),(1274,2271,'#8aef07'),(1275,2273,'#db2393'),(1276,2275,'#cec64b'),(1277,2278,'#d4be64'),(1278,2280,'#39aae4'),(1279,2282,'#d21add'),(1280,2283,'#ec361e'),(1281,2286,'#f0276c'),(1282,2288,'#a5ec38'),(1283,2290,'#f38f52'),(1284,2291,'#e610d4'),(1285,2292,'#34e47a'),(1286,2294,'#30ce37'),(1287,2295,'#601ecd'),(1288,2296,'#e309cb'),(1289,2298,'#24f483'),(1290,2299,'#815ce5'),(1291,2300,'#5f88f6'),(1292,2302,'#2ee6ef'),(1293,2304,'#4cd90f'),(1294,2305,'#f96a1e'),(1295,2306,'#e26209'),(1296,2307,'#FFFFFF'),(1297,2308,'#ee76cd'),(1298,2309,'#e4136f'),(1299,2310,'#FFFFFF'),(1300,2311,'#FFFFFF'),(1301,2312,'#d1ba63'),(1302,2317,'#06ecb1'),(1303,2319,'#17fc42'),(1304,2321,'#de9f3d'),(1305,2322,'#8f56ce'),(1306,2324,'#3df12a'),(1307,2328,'#2fd5b5'),(1308,2330,'#cff720'),(1309,2331,'#c61fe4'),(1310,2332,'#ad3bf2'),(1311,2334,'#FFFFFF'),(1312,2335,'#FFFFFF'),(1313,2337,'#43e8d9'),(1314,2339,'#33a3e8'),(1315,2340,'#3d2cf1'),(1316,2341,'#d27157'),(1317,2342,'#dd9f36'),(1318,2343,'#209bdd'),(1319,2346,'#7120f1'),(1320,2348,'#ec168c'),(1321,2350,'#FFFFFF'),(1322,2353,'#e1b353'),(1323,2357,'#ec39ee'),(1324,2358,'#ade614'),(1325,2361,'#FFFFFF'),(1326,2363,'#0ddca2'),(1327,2366,'#32db99'),(1328,2368,'#72ecc1'),(1329,2371,'#d23eac'),(1330,2372,'#18b3d6'),(1331,2373,'#274dde'),(1332,2380,'#FFFFFF'),(1333,2381,'#FFFFFF'),(1334,2382,'#FFFFFF'),(1335,2384,'#db2a1a'),(1336,2385,'#5b94fe'),(1337,2386,'#0adfce'),(1338,2388,'#6e69d3'),(1339,2389,'#e7680f'),(1340,2390,'#5636ef'),(1341,2391,'#3de69e'),(1342,2392,'#19b1de'),(1343,2395,'#d0a41a'),(1344,2396,'#22f71e'),(1345,2398,'#e74a4b'),(1346,2400,'#ad11d5'),(1347,2402,'#33aecf'),(1348,2409,'#cbd758'),(1349,2410,'#FFFFFF'),(1350,2412,'#FFFFFF'),(1351,2414,'#30e60f'),(1352,2415,'#31e22b'),(1353,2419,'#30cc4a'),(1354,2422,'#efc02d'),(1355,2424,'#33b5ce'),(1356,2425,'#4c76d2'),(1357,2426,'#ec4d3f'),(1358,2427,'#0ef862'),(1359,2431,'#bedb66'),(1360,2432,'#c80ef8'),(1361,2435,'#c5d42b'),(1362,2437,'#41eaa7'),(1363,2440,'#46f91e'),(1364,2441,'#cada45'),(1365,2442,'#38ee33'),(1366,2443,'#ea0e01'),(1367,2446,'#cc4f4c'),(1368,2448,'#fa127d'),(1369,2451,'#e2e416'),(1370,2452,'#74f6ef'),(1371,2455,'#3364e4'),(1372,2456,'#fe231b'),(1373,2460,'#2045e4'),(1374,2462,'#f57d4e'),(1375,2463,'#3332f7'),(1376,2464,'#a062fc'),(1377,2469,'#544fdd'),(1378,2470,'#d739ab'),(1379,2475,'#FFFFFF'),(1380,2477,'#6cfab1'),(1381,2478,'#66e4d9'),(1382,2480,'#594df7'),(1383,2483,'#addf40'),(1384,2485,'#10d476'),(1385,2487,'#db0be7'),(1386,2488,'#7c29de'),(1387,2490,'#d43ee6'),(1388,2491,'#49c4f4'),(1389,2493,'#8e12eb'),(1390,2494,'#96ec4b'),(1391,2497,'#b611f0'),(1392,2498,'#e25bc5'),(1393,2500,'#b03fd2'),(1394,2502,'#2ee7c5'),(1395,2504,'#ee7e59'),(1396,2505,'#FFFFFF'),(1397,2507,'#d3324b'),(1398,2508,'#FFFFFF'),(1399,2509,'#29d420'),(1400,2510,'#3beb74'),(1401,2512,'#FFFFFF'),(1402,2513,'#FFFFFF'),(1403,2514,'#FFFFFF'),(1404,2515,'#6ec4f3'),(1405,2516,'#e043ab'),(1406,2517,'#1e37f1'),(1407,2518,'#62e450'),(1408,2520,'#2b99d9'),(1409,2522,'#78e936'),(1410,2524,'#FFFFFF'),(1411,2526,'#4fe595'),(1412,2527,'#e254ba'),(1413,2533,'#bedc27'),(1414,2534,'#FFFFFF'),(1415,2537,'#b51dd8'),(1416,2539,'#b82aea'),(1417,2540,'#af2fe3'),(1418,2541,'#f4ac43'),(1419,2542,'#35b4fc'),(1420,2543,'#FFFFFF'),(1421,2545,'#4d7cdb'),(1422,2551,'#2157f4'),(1423,2553,'#b252d0'),(1424,2555,'#6e20d8'),(1425,2557,'#724ee5'),(1426,2560,'#FFFFFF'),(1427,2561,'#0dcec8'),(1428,2564,'#f755ee'),(1429,2565,'#d03c93'),(1430,2566,'#e6688a'),(1431,2567,'#99f71b'),(1432,2568,'#185cd4'),(1433,2571,'#f0688f'),(1434,2575,'#55fe10'),(1435,2578,'#FFFFFF'),(1436,2580,'#35e2c0'),(1437,2586,'#d7a74e'),(1438,2590,'#9f5df7'),(1439,2592,'#52bfed'),(1440,2593,'#63d726'),(1441,2596,'#5be353'),(1442,2598,'#dc7355'),(1443,2600,'#FFFFFF'),(1444,2601,'#9ad86a'),(1445,2602,'#b65ae2'),(1446,2603,'#1983f3'),(1447,2604,'#FFFFFF'),(1448,2606,'#FFFFFF'),(1449,2609,'#b002ce'),(1450,2612,'#55e652'),(1451,2613,'#e6be54'),(1452,2614,'#e67a42'),(1453,2615,'#c5ed0e'),(1454,2620,'#FFFFFF'),(1455,2623,'#FFFFFF'),(1456,2624,'#FFFFFF'),(1457,2625,'#59e0c1'),(1458,2626,'#f2d03d'),(1459,2627,'#96d965'),(1460,2628,'#1de462'),(1461,2629,'#32e313'),(1462,2633,'#3defa6'),(1463,2636,'#ef6881'),(1464,2638,'#655cdc'),(1465,2639,'#db41ba'),(1466,2640,'#a2d502'),(1467,2641,'#11d3be'),(1468,2644,'#e03c45'),(1469,2646,'#85e06d'),(1470,2648,'#FFFFFF'),(1471,2649,'#dc03df'),(1472,2651,'#f20cc9'),(1473,2656,'#FFFFFF'),(1474,2659,'#33fdcc'),(1475,2661,'#52de7a'),(1476,2665,'#86e95f'),(1477,2668,'#fa0c2a'),(1478,2669,'#eb292b'),(1479,2670,'#c65bfc'),(1480,2671,'#15da5a'),(1481,2673,'#FFFFFF'),(1482,2674,'#3290d3'),(1483,2675,'#FFFFFF'),(1484,2676,'#f76fd9'),(1485,2677,'#FFFFFF'),(1486,2678,'#da2ab6'),(1487,2679,'#18faab'),(1488,2680,'#c6cd4f'),(1489,2684,'#e17252'),(1490,2685,'#e68c68'),(1491,2687,'#f8e77a'),(1492,2689,'#784fd6'),(1493,2690,'#03a8fe'),(1494,2691,'#07e212'),(1495,2692,'#0ce59f'),(1496,2693,'#7d35de'),(1497,2694,'#1955f7'),(1498,2697,'#6ade48'),(1499,2698,'#e78520'),(1500,2703,'#3d5fe2'),(1501,2704,'#6c26ce'),(1502,2705,'#a368f6'),(1503,2706,'#eee06f'),(1504,2707,'#badb42'),(1505,2709,'#dd02f9'),(1506,2714,'#74d15e'),(1507,2715,'#8b4af5'),(1508,2716,'#0764e5'),(1509,2718,'#FFFFFF'),(1510,2721,'#f6348e'),(1511,2722,'#fbc647'),(1512,2726,'#d86e4b'),(1513,2733,'#FFFFFF'),(1514,2737,'#c84af7'),(1515,2738,'#efc75e'),(1516,2739,'#34f400'),(1517,2740,'#ce5dbf'),(1518,2742,'#1c13d0'),(1519,2745,'#257cf5'),(1520,2746,'#8462e4'),(1521,2747,'#ee2c19'),(1522,2748,'#54da25'),(1523,2749,'#fc5691'),(1524,2750,'#5016d5'),(1525,2751,'#d75ad1'),(1526,2753,'#08fcb0'),(1527,2757,'#5a5dd1'),(1528,2760,'#FFFFFF'),(1529,2761,'#d1a73b'),(1530,2763,'#59e9a9'),(1531,2767,'#cc1ffe'),(1532,2768,'#ede108'),(1533,2770,'#7f20dd'),(1534,2771,'#FFFFFF'),(1535,2777,'#d93682'),(1536,2778,'#7e63ce'),(1537,2779,'#089fdd'),(1538,2780,'#da3abd'),(1539,2781,'#ed4367'),(1540,2784,'#f4450d'),(1541,2785,'#FFFFFF'),(1542,2786,'#4bd5e1'),(1543,2787,'#5b19dd'),(1544,2791,'#f3248e'),(1545,2792,'#eb494d'),(1546,2793,'#df375a'),(1547,2794,'#52e0e2'),(1548,2795,'#cdea5c'),(1549,2796,'#25f758'),(1550,2798,'#61d7bb'),(1551,2799,'#5bf091'),(1552,2800,'#41e746'),(1553,2802,'#FFFFFF'),(1554,2803,'#266ee8'),(1555,2805,'#6f22d4'),(1556,2806,'#ef4521'),(1557,2807,'#FFFFFF'),(1558,2810,'#f47194'),(1559,2815,'#FFFFFF'),(1560,2816,'#ee3bbf'),(1561,2817,'#c42cd2'),(1562,2821,'#0fe0bb'),(1563,2828,'#4366f1'),(1564,2829,'#c3e530'),(1565,2830,'#4b62f9'),(1566,2831,'#10e2c7'),(1567,2833,'#c2d64b'),(1568,2834,'#FFFFFF'),(1569,2835,'#68f490'),(1570,2836,'#c130da'),(1571,2837,'#bdcc4f'),(1572,2840,'#5279d0'),(1573,2841,'#d4d74f'),(1574,2844,'#a3fe0c'),(1575,2846,'#e09836'),(1576,2847,'#bd09ea'),(1577,2849,'#5accb8'),(1578,2850,'#e8cd04'),(1579,2851,'#4a17f2'),(1580,2852,'#20d35a'),(1581,2855,'#a8fc78'),(1582,2857,'#57fa4d'),(1583,2859,'#9ff267'),(1584,2860,'#8afb77'),(1585,2861,'#735fe9'),(1586,2862,'#57dea7'),(1587,2863,'#f2ad56'),(1588,2865,'#ec6145'),(1589,2866,'#d70dce'),(1590,2867,'#5ee39a'),(1591,2868,'#5bdce3'),(1592,2871,'#4659f2'),(1593,2875,'#e1a262'),(1594,2876,'#cd9802'),(1595,2877,'#ebe72a'),(1596,2878,'#d326d6'),(1597,2881,'#e96499'),(1598,2882,'#d2974a'),(1599,2884,'#b865e1'),(1600,2886,'#e5c053'),(1601,2888,'#2f8aeb'),(1602,2889,'#fdba49'),(1603,2890,'#404ef2'),(1604,2891,'#18d3ae'),(1605,2895,'#FFFFFF'),(1606,2899,'#FFFFFF'),(1607,2900,'#8ee45f'),(1608,2901,'#9c29e5'),(1609,2902,'#e03400'),(1610,2904,'#eead5b'),(1611,2908,'#6561d1'),(1612,2912,'#b365fd'),(1613,2913,'#FFFFFF'),(1614,2914,'#70ee0f'),(1615,2916,'#28e92c'),(1616,2919,'#9bd264'),(1617,2920,'#f48e3f'),(1618,2923,'#4b4efa'),(1619,2924,'#d95448'),(1620,2928,'#4277f8'),(1621,2929,'#2fd083'),(1622,2930,'#9768d6'),(1623,2931,'#dc355f'),(1624,2932,'#FFFFFF'),(1625,2933,'#FFFFFF'),(1626,2934,'#d218db'),(1627,2937,'#FFFFFF'),(1628,2938,'#b9cf0c'),(1629,2939,'#6be2ba'),(1630,2940,'#bb4ee4'),(1631,2941,'#e4607c'),(1632,2942,'#d12de7'),(1633,2943,'#ced529'),(1634,2944,'#6201f2'),(1635,2945,'#FFFFFF'),(1636,2949,'#FFFFFF'),(1637,2950,'#e97a3d'),(1638,2958,'#d934e0'),(1639,2959,'#4f25f2'),(1640,2960,'#4661e0'),(1641,2965,'#4320ce'),(1642,2967,'#50f416'),(1643,2970,'#9ecf26'),(1644,2972,'#1bf830'),(1645,2974,'#dd449d'),(1646,2975,'#4fc5d9'),(1647,2976,'#9fd66a'),(1648,2978,'#7267d2'),(1649,2985,'#FFFFFF'),(1650,2991,'#44f9c7'),(1651,2992,'#f8d605'),(1652,2993,'#6de815'),(1653,2999,'#ec6e1e'),(1654,3004,'#32a7d8'),(1655,3005,'#9933e9'),(1656,3007,'#2010fc'),(1657,3008,'#5ad960'),(1658,3010,'#5e8ddb'),(1659,3016,'#060ffe'),(1660,3018,'#a46cf8'),(1661,3019,'#6632df'),(1662,3022,'#4a35cd'),(1663,3023,'#12ee5c'),(1664,3025,'#FFFFFF'),(1665,3026,'#FFFFFF'),(1666,3029,'#9c29d4'),(1667,3032,'#bedc3a'),(1668,3035,'#5adf96'),(1669,3038,'#e2823e'),(1670,3043,'#FFFFFF'),(1671,3046,'#2e2bec'),(1672,3048,'#f833d0'),(1673,3049,'#65a4e4'),(1674,3053,'#3dd7b2'),(1675,3054,'#96e214'),(1676,3057,'#eb5fe9'),(1677,3059,'#d69901'),(1678,3060,'#06cafb'),(1679,3062,'#FFFFFF'),(1680,3065,'#e05339'),(1681,3067,'#b8cd01'),(1682,3075,'#5c83d3'),(1683,3076,'#56dfec'),(1684,3078,'#a8e568'),(1685,3080,'#b6f707'),(1686,3083,'#6729d4'),(1687,3084,'#FFFFFF'),(1688,3088,'#d0dd23'),(1689,3089,'#FFFFFF'),(1690,3093,'#0bbde2'),(1691,3094,'#8a51de'),(1692,3095,'#e10f4a'),(1693,3096,'#512bce'),(1694,3097,'#FFFFFF'),(1695,3099,'#9e3eea'),(1696,3100,'#9bde34'),(1697,3102,'#9645fd'),(1698,3104,'#e7fb23'),(1699,3106,'#534ffc'),(1700,3107,'#298bcc'),(1701,3108,'#65d6a4'),(1702,3110,'#2234f4'),(1703,3111,'#57da1b'),(1704,3112,'#d0ee45'),(1705,3116,'#f9b013'),(1706,3122,'#d4a352'),(1707,3123,'#caf520'),(1708,3124,'#df6685'),(1709,3125,'#e80613'),(1710,3126,'#73d05d'),(1711,3127,'#3ece56'),(1712,3128,'#1753de'),(1713,3129,'#1579db'),(1714,3130,'#c7d908'),(1715,3131,'#38eaaf'),(1716,3132,'#0279fe'),(1717,3133,'#f7ae38'),(1718,3138,'#09ee48'),(1719,3139,'#a0e912'),(1720,3142,'#3fd34c'),(1721,3143,'#f69977'),(1722,3146,'#26da80'),(1723,3147,'#ed13ea'),(1724,3149,'#a333df'),(1725,3151,'#1f23cd'),(1726,3152,'#6ee7af'),(1727,3153,'#96d92c'),(1728,3154,'#FFFFFF'),(1729,3155,'#2015e1'),(1730,3158,'#bbe36e'),(1731,3159,'#bfd166'),(1732,3160,'#d0bc5d'),(1733,3162,'#ce1766'),(1734,3164,'#fa8458'),(1735,3167,'#f036e1'),(1736,3168,'#fac55a'),(1737,3169,'#3ad315'),(1738,3174,'#FFFFFF'),(1739,3175,'#FFFFFF'),(1740,3176,'#4cd58b'),(1741,3177,'#ca3bd8'),(1742,3178,'#FFFFFF'),(1743,3179,'#52cd9f'),(1744,3180,'#966bf4'),(1745,3183,'#49ddc6'),(1746,3185,'#33ce21'),(1747,3186,'#5806d2'),(1748,3187,'#FFFFFF'),(1749,3189,'#08a8ce'),(1750,3193,'#16e165'),(1751,3197,'#6bccdc'),(1752,3198,'#e73d81'),(1753,3201,'#5b91f9'),(1754,3203,'#e230bd'),(1755,3206,'#e7dc03'),(1756,3208,'#5bdc15'),(1757,3210,'#de2ffa'),(1758,3211,'#d054e1'),(1759,3212,'#6ace54'),(1760,3213,'#FFFFFF'),(1761,3215,'#7f5ae6'),(1762,3217,'#5c15da'),(1763,3220,'#e3205d'),(1764,3221,'#4ccaf0'),(1765,3223,'#6af3dc'),(1766,3225,'#6eec08'),(1767,3226,'#41e7c7'),(1768,3227,'#FFFFFF'),(1769,3228,'#98cf2b'),(1770,3231,'#4db1d0'),(1771,3232,'#46a3d3'),(1772,3233,'#2eb0cc'),(1773,3235,'#df5ac9'),(1774,3238,'#d9458e'),(1775,3242,'#8067e2'),(1776,3243,'#fad761'),(1777,3245,'#fbc46c'),(1778,3246,'#cebc21'),(1779,3248,'#24b0e1'),(1780,3249,'#2cb4ce'),(1781,3250,'#57f9df'),(1782,3251,'#FFFFFF'),(1783,3254,'#da1db9'),(1784,3255,'#556fd9'),(1785,3257,'#7082ef'),(1786,3259,'#c636e2'),(1787,3260,'#f91a89'),(1788,3261,'#1c09f7'),(1789,3262,'#cc8964'),(1790,3263,'#d2b15b'),(1791,3265,'#7127dd'),(1792,3267,'#5bfadc'),(1793,3269,'#f0d137'),(1794,3271,'#4ae8ea'),(1795,3272,'#fbe412'),(1796,3273,'#e15ffc'),(1797,3274,'#0ecf74'),(1798,3275,'#34d9d6'),(1799,3276,'#fabc45'),(1800,3277,'#FFFFFF'),(1801,3279,'#d265cb'),(1802,3284,'#3ed634'),(1803,3285,'#fde215'),(1804,3286,'#FFFFFF'),(1805,3289,'#60d829'),(1806,3290,'#e26646'),(1807,3291,'#00e03c'),(1808,3293,'#ee47fa'),(1809,3295,'#FFFFFF'),(1810,3296,'#FFFFFF'),(1811,3297,'#dcbf31'),(1812,3298,'#FFFFFF'),(1813,3299,'#fdcb68'),(1814,3300,'#4655ea'),(1815,3303,'#FFFFFF'),(1816,3307,'#468be7'),(1817,3308,'#09cc34'),(1818,3313,'#091cec'),(1819,3315,'#c1dd0c'),(1820,3317,'#4e95d1'),(1821,3318,'#f7bc5c'),(1822,3320,'#FFFFFF'),(1823,3321,'#bee86c'),(1824,3322,'#c54bf5'),(1825,3323,'#6878d5'),(1826,3324,'#a331ed'),(1827,3326,'#e52731'),(1828,3327,'#a2fc54'),(1829,3329,'#35fa89'),(1830,3330,'#0740ef'),(1831,3332,'#01f8e1'),(1832,3333,'#13faf8'),(1833,3334,'#f8732e'),(1834,3338,'#8122ec'),(1835,3340,'#4432cc'),(1836,3341,'#FFFFFF'),(1837,3344,'#FFFFFF'),(1838,3350,'#39c4ef'),(1839,3352,'#dc3503'),(1840,3353,'#76fd49'),(1841,3354,'#5066d4'),(1842,3357,'#61e4ee'),(1843,3358,'#e37338'),(1844,3359,'#eb354a'),(1845,3362,'#6ee16f'),(1846,3367,'#55e3de'),(1847,3368,'#cd5b0e'),(1848,3369,'#5bd26f'),(1849,3370,'#d567e4'),(1850,3371,'#d4ce49'),(1851,3374,'#ab78f5'),(1852,3375,'#fe10a6'),(1853,3378,'#4a3ad2'),(1854,3381,'#aa17ec'),(1855,3383,'#cd401e'),(1856,3384,'#beda0b'),(1857,3388,'#b0d238'),(1858,3390,'#1689cf'),(1859,3393,'#d07131'),(1860,3395,'#d81a0e'),(1861,3397,'#03f792'),(1862,3398,'#2343e7'),(1863,3400,'#eea449'),(1864,3401,'#cace18'),(1865,3403,'#31d297'),(1866,3406,'#6987d9'),(1867,3407,'#de09b3'),(1868,3408,'#f1641b'),(1869,3409,'#66cc83'),(1870,3410,'#dc04f7'),(1871,3411,'#38d4ef'),(1872,3412,'#b2d569'),(1873,3415,'#386fdb'),(1874,3416,'#dffe7e'),(1875,3417,'#FFFFFF'),(1876,3419,'#116ff3'),(1877,3422,'#57d123'),(1878,3426,'#8cfe00'),(1879,3427,'#5fd94d'),(1880,3428,'#d437f8'),(1881,3429,'#ed6991'),(1882,3431,'#e763ac'),(1883,3432,'#d59569'),(1884,3433,'#4fa2dd'),(1885,3434,'#FFFFFF'),(1886,3435,'#221dde'),(1887,3436,'#4de118'),(1888,3437,'#27dd54'),(1889,3438,'#dc529d'),(1890,3440,'#fd50d6'),(1891,3444,'#00fcfb'),(1892,3445,'#FFFFFF'),(1893,3449,'#9adc44'),(1894,3451,'#8ed468'),(1895,3454,'#d72971'),(1896,3455,'#eff135'),(1897,3456,'#e764c6'),(1898,3457,'#4c4ef0'),(1899,3462,'#303ae7'),(1900,3463,'#f2d776'),(1901,3464,'#5049ea'),(1902,3465,'#49e06a'),(1903,3466,'#9920fb'),(1904,3467,'#9ad462'),(1905,3469,'#FFFFFF'),(1906,3472,'#765bd0'),(1907,3473,'#7c15f5'),(1908,3474,'#5cdfda'),(1909,3479,'#68ecb0'),(1910,3481,'#eb145f'),(1911,3483,'#1079d6'),(1912,3488,'#b1fc50'),(1913,3490,'#d4db1a'),(1914,3492,'#6b64d0'),(1915,3495,'#ce9e55'),(1916,3496,'#70b9e4'),(1917,3498,'#63d5cd'),(1918,3501,'#d3377f'),(1919,3505,'#72f747'),(1920,3511,'#ae1ed0'),(1921,3512,'#47db27'),(1922,3514,'#ded72f'),(1923,3515,'#dcb20e'),(1924,3516,'#d6d360'),(1925,3517,'#89fb56'),(1926,3520,'#10eed7'),(1927,3521,'#30cafe'),(1928,3522,'#62ace1'),(1929,3526,'#FFFFFF'),(1930,3527,'#e83821'),(1931,3528,'#d86060'),(1932,3529,'#fb7496'),(1933,3530,'#FFFFFF'),(1934,3531,'#FFFFFF'),(1935,3535,'#61c7cf'),(1936,3536,'#14d11f'),(1937,3538,'#51d1c6'),(1938,3539,'#678ce7'),(1939,3540,'#a1d752'),(1940,3541,'#d953f5'),(1941,3542,'#32d8f8'),(1942,3544,'#dc7624'),(1943,3545,'#f2446d'),(1944,3546,'#4561dd'),(1945,3553,'#6ef0d2'),(1946,3554,'#FFFFFF'),(1947,3555,'#FFFFFF'),(1948,3556,'#e32d55'),(1949,3557,'#e2e718'),(1950,3559,'#FFFFFF'),(1951,3560,'#e7d864'),(1952,3562,'#89ee00'),(1953,3564,'#de34e2'),(1954,3567,'#8839cf'),(1955,3569,'#64d3ce'),(1956,3570,'#d135bf'),(1957,3571,'#0adbf0'),(1958,3574,'#87e31d'),(1959,3575,'#09f0b5'),(1960,3577,'#6756e1'),(1961,3578,'#ef111e'),(1962,3581,'#4c6af9'),(1963,3585,'#199bea'),(1964,3586,'#ec1eda'),(1965,3587,'#4de8c9'),(1966,3589,'#3d1deb'),(1967,3590,'#fb575c'),(1968,3591,'#a6e947'),(1969,3594,'#6f67fa'),(1970,3595,'#4ad5a6'),(1971,3596,'#FFFFFF'),(1972,3598,'#19fb85'),(1973,3599,'#9d0fe5'),(1974,3600,'#b4dd6d'),(1975,3601,'#9ace5d'),(1976,3603,'#0be094'),(1977,3604,'#ce58d5'),(1978,3607,'#2517db'),(1979,3609,'#fed640'),(1980,3610,'#6bdbb4'),(1981,3611,'#42faa1'),(1982,3614,'#c71fce'),(1983,3615,'#eb5b59'),(1984,3616,'#a8d60c'),(1985,3617,'#FFFFFF'),(1986,3620,'#7bf889'),(1987,3623,'#b6e423'),(1988,3624,'#de7407'),(1989,3625,'#baeb43'),(1990,3627,'#e75721'),(1991,3631,'#FFFFFF'),(1992,3633,'#d670e3'),(1993,3635,'#FFFFFF'),(1994,3637,'#ecc75e'),(1995,3639,'#FFFFFF'),(1996,3640,'#FFFFFF'),(1997,3641,'#df2f96'),(1998,3643,'#9619f8'),(1999,3647,'#34b2fe'),(2000,3648,'#4ddcc7'),(2001,3651,'#7f42e6'),(2002,3655,'#c74ecd'),(2003,3656,'#04f5dd'),(2004,3657,'#ec5ee0'),(2005,3658,'#941efc'),(2006,3666,'#e92929'),(2007,3667,'#d6b331'),(2008,3669,'#d96baa'),(2009,3671,'#1c08d3'),(2010,3674,'#51fbcd'),(2011,3675,'#3af460'),(2012,3676,'#92e145'),(2013,3677,'#2937e2'),(2014,3680,'#7cacfd'),(2015,3681,'#f05f23'),(2016,3683,'#d69e26'),(2017,3684,'#37add1'),(2018,3687,'#f62222'),(2019,3689,'#4afe83'),(2020,3691,'#40c5d5'),(2021,3693,'#21a3f3'),(2022,3697,'#FFFFFF'),(2023,3698,'#e05d94'),(2024,3699,'#fedc7a'),(2025,3700,'#befc6c'),(2026,3701,'#743dec'),(2027,3703,'#d85fe0'),(2028,3705,'#cf8262'),(2029,3706,'#377fee'),(2030,3707,'#527cf5'),(2031,3709,'#115afd'),(2032,3710,'#e76d59'),(2033,3711,'#30d9d4'),(2034,3713,'#e670a6'),(2035,3715,'#51d22d'),(2036,3716,'#e2c60a'),(2037,3717,'#feea12'),(2038,3718,'#FFFFFF'),(2039,3719,'#bc14d6'),(2040,3720,'#cdbc5d'),(2041,3723,'#43f7a3'),(2042,3724,'#6ad009'),(2043,3732,'#3ef40a'),(2044,3733,'#FFFFFF'),(2045,3735,'#a838e1'),(2046,3737,'#fa10d5'),(2047,3739,'#975df1'),(2048,3741,'#FFFFFF'),(2049,3745,'#FFFFFF'),(2050,3746,'#6bf162'),(2051,3747,'#243bf5'),(2052,3749,'#3bd0c7'),(2053,3752,'#31dffb'),(2054,3755,'#d2744c'),(2055,3760,'#fc521d'),(2056,3761,'#c7f473'),(2057,3762,'#f88950'),(2058,3763,'#FFFFFF'),(2059,3765,'#1438cc'),(2060,3766,'#b04eee'),(2061,3768,'#6cd902'),(2062,3769,'#d06644'),(2063,3770,'#FFFFFF'),(2064,3776,'#FFFFFF'),(2065,3777,'#ebac15'),(2066,3784,'#FFFFFF'),(2067,3785,'#de1e0c'),(2068,3789,'#FFFFFF'),(2069,3790,'#96e468'),(2070,3792,'#fe8f48'),(2071,3798,'#7d3fce'),(2072,3801,'#78b7f9'),(2073,3802,'#e52849'),(2074,3803,'#2be955'),(2075,3805,'#87e716'),(2076,3806,'#82e32c'),(2077,3808,'#ec910f'),(2078,3816,'#FFFFFF'),(2079,3817,'#d8320d'),(2080,3818,'#e98f06'),(2081,3821,'#FFFFFF'),(2082,3822,'#d658e7'),(2083,3823,'#eb5115'),(2084,3827,'#4a8de7'),(2085,3829,'#6f1cde'),(2086,3831,'#e96bce'),(2087,3832,'#d244a2'),(2088,3835,'#17db2d'),(2089,3836,'#d864b9'),(2090,3837,'#cc3a65'),(2091,3838,'#51c7f1'),(2092,3840,'#e3dc61'),(2093,3843,'#dd6f48'),(2094,3844,'#1fe938'),(2095,3846,'#c2f64f'),(2096,3849,'#4118cd'),(2097,3852,'#FFFFFF'),(2098,3853,'#f62dfb'),(2099,3854,'#fb5944'),(2100,3855,'#e2fa4a'),(2101,3857,'#b1f75c'),(2102,3858,'#fe28a3'),(2103,3859,'#4746e8'),(2104,3860,'#e0624d'),(2105,3861,'#27f821'),(2106,3862,'#4efef6'),(2107,3864,'#8973ea'),(2108,3866,'#5672f5'),(2109,3867,'#addb04'),(2110,3869,'#46fd6a'),(2111,3871,'#3af25a'),(2112,3872,'#2ffc70'),(2113,3874,'#d91395'),(2114,3876,'#d226ae'),(2115,3878,'#62e55e'),(2116,3879,'#cd6081'),(2117,3880,'#57f349'),(2118,3882,'#60daa9'),(2119,3883,'#e3fa36'),(2120,3884,'#afda24'),(2121,3886,'#4077f8'),(2122,3887,'#dcaa66'),(2123,3888,'#d32a74'),(2124,3889,'#FFFFFF'),(2125,3890,'#e99b27'),(2126,3891,'#e4321a'),(2127,3893,'#ad50ef'),(2128,3895,'#40efec'),(2129,3896,'#f6715e'),(2130,3897,'#981fe4'),(2131,3899,'#1c4df4'),(2132,3900,'#419fe6'),(2133,3901,'#fedb42'),(2134,3905,'#FFFFFF'),(2135,3906,'#dc6b67'),(2136,3908,'#20bbf9'),(2137,3910,'#08d9a4'),(2138,3911,'#d42c80'),(2139,3913,'#43eded'),(2140,3914,'#bed13e'),(2141,3915,'#a90ce0'),(2142,3916,'#d5ab3d'),(2143,3919,'#FFFFFF'),(2144,3920,'#ed6590'),(2145,3921,'#6d9bea'),(2146,3923,'#51a4f4'),(2147,3924,'#FFFFFF'),(2148,3925,'#099add'),(2149,3930,'#4cf3b0'),(2150,3932,'#6aef3d'),(2151,3933,'#65ee3e'),(2152,3935,'#d8c239'),(2153,3936,'#432ad4'),(2154,3937,'#f001db'),(2155,3939,'#d55e53'),(2156,3940,'#aace0d'),(2157,3941,'#42e648'),(2158,3943,'#d62d06'),(2159,3944,'#931ee7'),(2160,3945,'#fd188f'),(2161,3946,'#4e89d4'),(2162,3949,'#292df9'),(2163,3950,'#0cfafb'),(2164,3951,'#39f31d'),(2165,3953,'#cef82d'),(2166,3957,'#FFFFFF'),(2167,3959,'#9532f7'),(2168,3961,'#aa35e2'),(2169,3963,'#9461f0'),(2170,3965,'#e0f478'),(2171,3966,'#e4dd51'),(2172,3967,'#ed36e2'),(2173,3970,'#28e7c3'),(2174,3972,'#d1356a'),(2175,3974,'#f6be16'),(2176,3975,'#8f1bd9'),(2177,3976,'#c62cd9'),(2178,3978,'#4eedf6'),(2179,3980,'#bffc5c'),(2180,3981,'#FFFFFF'),(2181,3982,'#d58163'),(2182,3984,'#df053e'),(2183,3985,'#77e2f2'),(2184,3987,'#43d230'),(2185,3988,'#FFFFFF'),(2186,3989,'#fa41a9'),(2187,3990,'#d054ac'),(2188,3992,'#a31ed2'),(2189,3993,'#cc456c'),(2190,3996,'#e4c937'),(2191,3997,'#62cfa8'),(2192,3998,'#18d9fd'),(2193,3999,'#dc4f97'),(2194,4003,'#e19b66'),(2195,4005,'#f5f424'),(2196,4007,'#cad439'),(2197,4009,'#8344e5'),(2198,4010,'#1dbefe'),(2199,4011,'#3ec8d4'),(2200,4014,'#3992d4'),(2201,4020,'#FFFFFF'),(2202,4025,'#e46cb7'),(2203,4026,'#68fc3a'),(2204,4027,'#f73ecd'),(2205,4029,'#dea93f'),(2206,4030,'#3fd0ad'),(2207,4031,'#FFFFFF'),(2208,4033,'#5f58ea'),(2209,4034,'#669cea'),(2210,4036,'#d738d6'),(2211,4037,'#FFFFFF'),(2212,4039,'#e95547'),(2213,4045,'#23f626'),(2214,4048,'#e218bc'),(2215,4049,'#e160ea'),(2216,4051,'#00ccad'),(2217,4053,'#d9fc23'),(2218,4054,'#3cf84f'),(2219,4055,'#e86155'),(2220,4056,'#45f035'),(2221,4057,'#92db65'),(2222,4058,'#85cc20'),(2223,4061,'#FFFFFF'),(2224,4064,'#f92b3d'),(2225,4065,'#2adcac'),(2226,4066,'#646def'),(2227,4067,'#c2de21'),(2228,4068,'#f8384c'),(2229,4069,'#7242e8'),(2230,4070,'#52e7d8'),(2231,4072,'#4b2ee3'),(2232,4073,'#4f59f4'),(2233,4074,'#42e0c3'),(2234,4075,'#54a7e8'),(2235,4076,'#37ef24'),(2236,4078,'#dd1aae'),(2237,4079,'#ee07f5'),(2238,4084,'#b2e714'),(2239,4087,'#cf41e2'),(2240,4088,'#f55a5b'),(2241,4089,'#5ef49e'),(2242,4092,'#59cd33'),(2243,4093,'#cc1df2'),(2244,4094,'#FFFFFF'),(2245,4095,'#66da48'),(2246,4096,'#adfb6b'),(2247,4097,'#f64361'),(2248,4100,'#e645dc'),(2249,4102,'#c840fc'),(2250,4104,'#0eafdf'),(2251,4106,'#7bf8d0'),(2252,4107,'#ea635b'),(2253,4110,'#FFFFFF'),(2254,4111,'#12d6d4'),(2255,4112,'#7f52db'),(2256,4114,'#FFFFFF'),(2257,4115,'#eefe11'),(2258,4123,'#FFFFFF'),(2259,4124,'#82d250'),(2260,4127,'#64ea58'),(2261,4128,'#FFFFFF'),(2262,4129,'#cd9d09'),(2263,4131,'#a255e4'),(2264,4132,'#9ee43a'),(2265,4134,'#8967d7'),(2266,4137,'#b047fb'),(2267,4140,'#bd34f6'),(2268,4145,'#b330e5'),(2269,4146,'#54d21d'),(2270,4147,'#c5f11e'),(2271,4153,'#49a0ee'),(2272,4154,'#deae46'),(2273,4155,'#FFFFFF'),(2274,4158,'#f275c5'),(2275,4160,'#cf7259'),(2276,4162,'#FFFFFF'),(2277,4163,'#d8803f'),(2278,4164,'#2cdaa1'),(2279,4165,'#8438e9'),(2280,4167,'#ec348b'),(2281,4169,'#d14f42'),(2282,4170,'#fa0741'),(2283,4171,'#e42561'),(2284,4173,'#ba74f0'),(2285,4175,'#e67a5b'),(2286,4178,'#FFFFFF'),(2287,4180,'#9de900'),(2288,4181,'#d00148'),(2289,4183,'#b3f34d'),(2290,4185,'#e1886c'),(2291,4187,'#0822e1'),(2292,4188,'#35e465'),(2293,4192,'#d765ca'),(2294,4193,'#78ec26'),(2295,4194,'#4bd42d'),(2296,4195,'#6ad325'),(2297,4197,'#0ab9cd'),(2298,4199,'#e3741d'),(2299,4202,'#ec58fe'),(2300,4203,'#7e32d0'),(2301,4204,'#44edf4'),(2302,4206,'#d9a81f'),(2303,4214,'#e351dc'),(2304,4216,'#FFFFFF'),(2305,4217,'#FFFFFF'),(2306,4218,'#ae48e2'),(2307,4220,'#995fdc'),(2308,4221,'#e5bf1c'),(2309,4223,'#fc0647'),(2310,4224,'#fe7d90'),(2311,4230,'#f41ecc'),(2312,4232,'#FFFFFF'),(2313,4233,'#dbee34'),(2314,4234,'#d28c41'),(2315,4235,'#20d44d'),(2316,4237,'#de4b47'),(2317,4238,'#33cc1b'),(2318,4239,'#27ee86'),(2319,4240,'#71da2e'),(2320,4242,'#01d712'),(2321,4243,'#43d287'),(2322,4244,'#b0ea28'),(2323,4245,'#5dd313'),(2324,4246,'#1aa9e8'),(2325,4247,'#59ddd6'),(2326,4250,'#dc2578'),(2327,4251,'#a2e80b'),(2328,4253,'#41dfe1'),(2329,4254,'#6ff23d'),(2330,4255,'#f9dd4e'),(2331,4259,'#FFFFFF'),(2332,4260,'#FFFFFF'),(2333,4263,'#c566e6'),(2334,4264,'#4204d1'),(2335,4266,'#feda5b'),(2336,4267,'#e4984a'),(2337,4269,'#033ed5'),(2338,4271,'#60e488'),(2339,4274,'#b2f177'),(2340,4275,'#b936d9'),(2341,4276,'#4a32f9'),(2342,4279,'#db8c15'),(2343,4283,'#f62fa0'),(2344,4285,'#43d867'),(2345,4287,'#a40df9'),(2346,4288,'#57fbd4'),(2347,4289,'#ae21db'),(2348,4290,'#bf44e0'),(2349,4291,'#25cd8b'),(2350,4292,'#fe9c22'),(2351,4294,'#5c34ec'),(2352,4297,'#5127d0'),(2353,4298,'#6a16d5'),(2354,4299,'#FFFFFF'),(2355,4300,'#5fe699'),(2356,4301,'#4ddd2c'),(2357,4302,'#ef6632'),(2358,4308,'#a6fc4d'),(2359,4309,'#4598e6'),(2360,4310,'#050eeb'),(2361,4312,'#5cefe6'),(2362,4313,'#9f6ff6'),(2363,4317,'#3de6fb'),(2364,4318,'#d2f766'),(2365,4325,'#43db0f'),(2366,4326,'#a9e561'),(2367,4329,'#2a29e2'),(2368,4330,'#9b61dd'),(2369,4331,'#df826b'),(2370,4332,'#1969ed'),(2371,4333,'#bf48da'),(2372,4334,'#dd8747'),(2373,4336,'#4fdbd6'),(2374,4338,'#d3871f'),(2375,4339,'#6233fb'),(2376,4341,'#6869de'),(2377,4343,'#84fb16'),(2378,4345,'#170ce4'),(2379,4346,'#3fd182'),(2380,4347,'#bbd727'),(2381,4349,'#88d041'),(2382,4350,'#ef7f3f'),(2383,4352,'#d64513'),(2384,4353,'#ee9309'),(2385,4355,'#6fd021'),(2386,4357,'#2cc3fb'),(2387,4358,'#e27e68'),(2388,4359,'#782ccd'),(2389,4360,'#29d383'),(2390,4361,'#09de8c'),(2391,4364,'#4cfc55'),(2392,4365,'#277bd5'),(2393,4367,'#d3e114'),(2394,4368,'#d519bb'),(2395,4371,'#865ad5'),(2396,4372,'#58cda5'),(2397,4374,'#bc65d4'),(2398,4379,'#2bec97'),(2399,4380,'#3ae2ce'),(2400,4381,'#5c07d6'),(2401,4382,'#15b1f7'),(2402,4383,'#d811c5'),(2403,4384,'#c715e4'),(2404,4391,'#36d6d1'),(2405,4392,'#d9621e'),(2406,4394,'#d09819'),(2407,4398,'#f61ad0'),(2408,4400,'#d08816'),(2409,4402,'#69a2f6'),(2410,4403,'#e638ae'),(2411,4405,'#e6ae1b'),(2412,4407,'#d29b5b'),(2413,4408,'#f7da24'),(2414,4409,'#FFFFFF'),(2415,4410,'#f77d36'),(2416,4411,'#9efd58'),(2417,4413,'#d039ef'),(2418,4414,'#f9b64e'),(2419,4416,'#FFFFFF'),(2420,4419,'#d332c5'),(2421,4421,'#c7f219'),(2422,4425,'#FFFFFF'),(2423,4426,'#FFFFFF'),(2424,4427,'#FFFFFF'),(2425,4430,'#60d2a9'),(2426,4433,'#d564dc'),(2427,4434,'#fe4511'),(2428,4436,'#944fe2'),(2429,4437,'#FFFFFF'),(2430,4438,'#cd6080'),(2431,4440,'#f6e55f'),(2432,4441,'#f18b66'),(2433,4443,'#FFFFFF'),(2434,4445,'#6e34dd'),(2435,4446,'#2e05d5'),(2436,4448,'#00dbef'),(2437,4452,'#f27092'),(2438,4453,'#7c5aea'),(2439,4456,'#FFFFFF'),(2440,4457,'#d9fe59'),(2441,4464,'#f5ce18'),(2442,4468,'#69d5b0'),(2443,4469,'#d75a21'),(2444,4470,'#47a0d1'),(2445,4471,'#2182cc'),(2446,4472,'#9f60cf'),(2447,4474,'#6bdb4a'),(2448,4477,'#04ce54'),(2449,4479,'#cb36d3'),(2450,4480,'#FFFFFF'),(2451,4484,'#d99148'),(2452,4485,'#4bce25'),(2453,4487,'#f7a322'),(2454,4488,'#FFFFFF'),(2455,4489,'#da01b9'),(2456,4490,'#0358df'),(2457,4491,'#e7d628'),(2458,4492,'#1be9ce'),(2459,4494,'#e68e54'),(2460,4497,'#d03a2d'),(2461,4499,'#20f96e'),(2462,4500,'#4d62e0'),(2463,4501,'#d717e5'),(2464,4502,'#FFFFFF'),(2465,4505,'#d8846a'),(2466,4507,'#71e0ef'),(2467,4508,'#FFFFFF'),(2468,4509,'#ace64d'),(2469,4510,'#FFFFFF'),(2470,4511,'#3c91cd'),(2471,4512,'#dd065f'),(2472,4513,'#c546d7'),(2473,4516,'#de680f'),(2474,4519,'#FFFFFF'),(2475,4520,'#25c1d3'),(2476,4521,'#5afbe2'),(2477,4525,'#4590e5'),(2478,4529,'#FFFFFF'),(2479,4530,'#fcaa3f'),(2480,4531,'#FFFFFF'),(2481,4533,'#e9863e'),(2482,4535,'#8560ed'),(2483,4536,'#d88f25'),(2484,4537,'#4da6f1'),(2485,4538,'#1650eb'),(2486,4539,'#FFFFFF'),(2487,4540,'#FFFFFF'),(2488,4541,'#eb10e8'),(2489,4542,'#44d0b3'),(2490,4545,'#FFFFFF'),(2491,4546,'#e8263b'),(2492,4549,'#FFFFFF'),(2493,4553,'#dd9442'),(2494,4558,'#d448ec'),(2495,4561,'#f57360'),(2496,4563,'#8508d2'),(2497,4567,'#d7147f'),(2498,4569,'#67d2e7'),(2499,4570,'#6c78e3'),(2500,4572,'#5f8df4'),(2501,4573,'#852aee'),(2502,4574,'#fa9374'),(2503,4575,'#a0ee0e'),(2504,4576,'#51f23f'),(2505,4578,'#bd31d5'),(2506,4581,'#6d21cd'),(2507,4586,'#cc3a97'),(2508,4587,'#fa3718'),(2509,4589,'#8e59f0'),(2510,4590,'#7ace3e'),(2511,4594,'#35d347'),(2512,4595,'#FFFFFF'),(2513,4597,'#FFFFFF'),(2514,4598,'#1926ec'),(2515,4600,'#06ed73'),(2516,4602,'#eada2c'),(2517,4603,'#87dd14'),(2518,4606,'#FFFFFF'),(2519,4607,'#43c0cf'),(2520,4608,'#FFFFFF'),(2521,4609,'#66d9c7'),(2522,4611,'#f4c570'),(2523,4612,'#513bd3'),(2524,4613,'#FFFFFF'),(2525,4614,'#8334cf'),(2526,4615,'#fd04ae'),(2527,4618,'#11f87a'),(2528,4619,'#60cdb0'),(2529,4623,'#68d30b'),(2530,4628,'#FFFFFF'),(2531,4632,'#FFFFFF'),(2532,4635,'#23ea4d'),(2533,4637,'#FFFFFF'),(2534,4640,'#22da2b'),(2535,4641,'#cb21dc'),(2536,4642,'#1af89a'),(2537,4643,'#6148fe'),(2538,4645,'#44ec00'),(2539,4646,'#4b9ce0'),(2540,4647,'#FFFFFF'),(2541,4649,'#cdd033'),(2542,4652,'#4deaa1'),(2543,4654,'#FFFFFF'),(2544,4655,'#79f5ca'),(2545,4656,'#d83e06'),(2546,4658,'#7ad117'),(2547,4659,'#bee40d'),(2548,4660,'#c510ed'),(2549,4662,'#5d11d4'),(2550,4666,'#FFFFFF'),(2551,4667,'#5cf1a4'),(2552,4669,'#fcb90e'),(2553,4670,'#d41de9'),(2554,4671,'#e00903'),(2555,4672,'#34e614'),(2556,4676,'#FFFFFF'),(2557,4677,'#2ddfe9'),(2558,4680,'#f158a3'),(2559,4681,'#8cf24c'),(2560,4682,'#cfa819'),(2561,4683,'#FFFFFF'),(2562,4684,'#92dc66'),(2563,4685,'#FFFFFF'),(2564,4688,'#FFFFFF'),(2565,4693,'#69aafe'),(2566,4695,'#f373e8'),(2567,4696,'#e62f7b'),(2568,4698,'#f9717c'),(2569,4699,'#88cd47'),(2570,4700,'#e53006'),(2571,4701,'#feef37'),(2572,4703,'#99f972'),(2573,4705,'#f69a46'),(2574,4706,'#3ddb75'),(2575,4709,'#FFFFFF'),(2576,4710,'#16e868'),(2577,4712,'#4cd97e'),(2578,4713,'#d61a0a'),(2579,4714,'#7af6d5'),(2580,4715,'#321ae2'),(2581,4717,'#f8047c'),(2582,4722,'#e3ba5b'),(2583,4726,'#d6851c'),(2584,4729,'#e2da18'),(2585,4735,'#FFFFFF'),(2586,4736,'#FFFFFF'),(2587,4737,'#ef5cd7'),(2588,4739,'#FFFFFF'),(2589,4740,'#b01ae6'),(2590,4742,'#4a9cfd'),(2591,4746,'#d71006'),(2592,4747,'#FFFFFF'),(2593,4748,'#FFFFFF'),(2594,4750,'#50acf5'),(2595,4751,'#1c72f4'),(2596,4752,'#ed320d'),(2597,4754,'#df064c'),(2598,4755,'#d56cde'),(2599,4756,'#4ffd13'),(2600,4758,'#fdc150'),(2601,4759,'#bf4ed3'),(2602,4760,'#4cef2d'),(2603,4761,'#ec7214'),(2604,4762,'#e84c60'),(2605,4763,'#bff06c'),(2606,4764,'#17e7a5'),(2607,4765,'#3e84cd'),(2608,4766,'#4973e8'),(2609,4767,'#FFFFFF'),(2610,4772,'#FFFFFF'),(2611,4773,'#FFFFFF'),(2612,4775,'#ee0c26'),(2613,4778,'#6133fe'),(2614,4780,'#FFFFFF'),(2615,4781,'#9cf030'),(2616,4782,'#FFFFFF'),(2617,4784,'#e259e1'),(2618,4786,'#dd0615'),(2619,4789,'#06d50c'),(2620,4790,'#b133f6'),(2621,4791,'#FFFFFF'),(2622,4792,'#c711f2'),(2623,4794,'#5cf5b3'),(2624,4795,'#85ec0f'),(2625,4796,'#5ecec9'),(2626,4797,'#54d7a6'),(2627,4800,'#f84bfa'),(2628,4801,'#FFFFFF'),(2629,4804,'#25e153'),(2630,4805,'#d257e1'),(2631,4806,'#9fdf41'),(2632,4808,'#89d064'),(2633,4811,'#21cfd0'),(2634,4812,'#40ed24'),(2635,4815,'#d2d71f'),(2636,4816,'#FFFFFF'),(2637,4818,'#7ae970'),(2638,4819,'#4bf56d'),(2639,4820,'#dd12b2'),(2640,4821,'#d846af'),(2641,4823,'#52d181'),(2642,4825,'#8ae871'),(2643,4826,'#3ee845'),(2644,4827,'#15d6e6'),(2645,4828,'#0e94d4'),(2646,4830,'#dcc356'),(2647,4832,'#f15b55'),(2648,4833,'#54e357'),(2649,4834,'#3308d9'),(2650,4836,'#2319e7'),(2651,4837,'#e08559'),(2652,4838,'#f13261'),(2653,4840,'#FFFFFF'),(2654,4842,'#FFFFFF'),(2655,4844,'#dc0c5e'),(2656,4845,'#646ecc'),(2657,4846,'#56fdb1'),(2658,4847,'#35ea7f'),(2659,4848,'#d84bcc'),(2660,4849,'#ec5c8a'),(2661,4852,'#cd63fe'),(2662,4853,'#384cce'),(2663,4854,'#2143d3'),(2664,4855,'#ec02de'),(2665,4857,'#4ee3fc'),(2666,4858,'#FFFFFF'),(2667,4859,'#05e887'),(2668,4860,'#aaf463'),(2669,4861,'#FFFFFF'),(2670,4862,'#d9284e'),(2671,4869,'#FFFFFF'),(2672,4875,'#9bf012'),(2673,4878,'#30d268'),(2674,4879,'#9ad45c'),(2675,4880,'#FFFFFF'),(2676,4881,'#51e7fa'),(2677,4882,'#df02ba'),(2678,4887,'#4aa7ce'),(2679,4889,'#1e68db'),(2680,4891,'#6ede4d'),(2681,4892,'#f9c132'),(2682,4893,'#63e940'),(2683,4894,'#cd5965'),(2684,4895,'#f7149d'),(2685,4896,'#8b39dd'),(2686,4897,'#f75527'),(2687,4899,'#f2d752'),(2688,4900,'#cf15b6'),(2689,4901,'#81dc26'),(2690,4902,'#e71f3b'),(2691,4904,'#054efd'),(2692,4905,'#97f070'),(2693,4907,'#e9fd04'),(2694,4909,'#34d220'),(2695,4911,'#e25ed0'),(2696,4912,'#FFFFFF'),(2697,4915,'#592ddc'),(2698,4918,'#d29049'),(2699,4921,'#d3bb01'),(2700,4923,'#fc2f27'),(2701,4925,'#aa6de7'),(2702,4926,'#aad83d'),(2703,4927,'#08e119'),(2704,4931,'#4dfdf4'),(2705,4932,'#42f700'),(2706,4933,'#3588e1'),(2707,4936,'#FFFFFF'),(2708,4937,'#8e55f2'),(2709,4938,'#113af1'),(2710,4939,'#27e3b4'),(2711,4940,'#e31bec'),(2712,4941,'#3cfa42'),(2713,4942,'#3132ee'),(2714,4943,'#2f1df5'),(2715,4944,'#5c56d0'),(2716,4948,'#28eeeb'),(2717,4949,'#68acd1'),(2718,4951,'#52d0ea'),(2719,4958,'#1ef3e8'),(2720,4960,'#6f8df8'),(2721,4961,'#18ddd3'),(2722,4962,'#FFFFFF'),(2723,4965,'#4214cd'),(2724,4968,'#f26705'),(2725,4969,'#4569e3'),(2726,4971,'#9a21cd'),(2727,4972,'#347df3'),(2728,4974,'#46e054'),(2729,4975,'#d91da6'),(2730,4976,'#fd6f0e'),(2731,4977,'#37ea71'),(2732,4979,'#FFFFFF'),(2733,4980,'#60eac0'),(2734,4984,'#ec6c90'),(2735,4986,'#12dc9a'),(2736,4989,'#2dcdaa'),(2737,4990,'#de60d1'),(2738,4991,'#cec908'),(2739,4993,'#68f9bd'),(2740,4994,'#d1452b'),(2741,4995,'#FFFFFF'),(2742,4997,'#03bece'),(2743,4998,'#67d367'),(2744,4999,'#1b59ee'),(2745,5000,'#d61a83'),(2746,5001,'#e39a64'),(2747,5004,'#444eee'),(2748,5005,'#288ee4'),(2749,5006,'#FFFFFF'),(2750,5007,'#ef4552'),(2751,5009,'#0b53ec'),(2752,5012,'#e835b8'),(2753,5013,'#5eda75'),(2754,5015,'#fa6d1e'),(2755,5018,'#2c07d9'),(2756,5019,'#42edc8'),(2757,5020,'#5dea71'),(2758,5021,'#a746ed'),(2759,5022,'#1de7d9'),(2760,5023,'#211eec'),(2761,5028,'#0a6fdc'),(2762,5029,'#c8f26a'),(2763,5032,'#c779f8'),(2764,5036,'#f0f232'),(2765,5039,'#cd65b2'),(2766,5041,'#59cdb3'),(2767,5044,'#6cf978'),(2768,5045,'#26f714'),(2769,5046,'#9ada57'),(2770,5048,'#5bf990'),(2771,5054,'#d640bc'),(2772,5055,'#6cc4f8'),(2773,5056,'#e40fab'),(2774,5059,'#cc05ad'),(2775,5060,'#8d64cf'),(2776,5062,'#b301d9'),(2777,5063,'#53cbd0'),(2778,5066,'#631df1'),(2779,5068,'#6cc9f8'),(2780,5071,'#f5679a'),(2781,5073,'#FFFFFF'),(2782,5074,'#4850e0'),(2783,5075,'#266fd2'),(2784,5077,'#3cf361'),(2785,5078,'#6bfa43'),(2786,5082,'#FFFFFF'),(2787,5083,'#55ce9d'),(2788,5084,'#8569ec'),(2789,5085,'#bffd0d'),(2790,5087,'#5fd868'),(2791,5088,'#e91bb1'),(2792,5091,'#FFFFFF'),(2793,5092,'#a9d05b'),(2794,5094,'#cc7b48'),(2795,5095,'#eb75e1'),(2796,5096,'#9e30e5'),(2797,5097,'#FFFFFF'),(2798,5098,'#49fe3b'),(2799,5102,'#75a7f5'),(2800,5103,'#49baf2'),(2801,5108,'#d04eda'),(2802,5109,'#97ce3b'),(2803,5110,'#d01772'),(2804,5112,'#4fd7a3'),(2805,5113,'#accc3a'),(2806,5114,'#de7840'),(2807,5115,'#11d156'),(2808,5116,'#6213ea'),(2809,5119,'#392bdc'),(2810,5124,'#66fe50'),(2811,5125,'#FFFFFF'),(2812,5126,'#337af1'),(2813,5127,'#ce5b47'),(2814,5128,'#61a4ea'),(2815,5129,'#FFFFFF'),(2816,5130,'#f75eeb'),(2817,5133,'#a7d336'),(2818,5135,'#f859af'),(2819,5136,'#69e99d'),(2820,5140,'#2fde34'),(2821,5142,'#5035f6'),(2822,5145,'#6e3fe8'),(2823,5146,'#e817d8'),(2824,5147,'#41e361'),(2825,5149,'#6299fd'),(2826,5150,'#f0c572'),(2827,5153,'#fe60f8'),(2828,5154,'#d43196'),(2829,5156,'#ce1f2e'),(2830,5157,'#d4c21c'),(2831,5158,'#3807e6'),(2832,5159,'#8fda01'),(2833,5160,'#45d4f9'),(2834,5164,'#d712ce'),(2835,5171,'#8ae75d'),(2836,5172,'#60b1ea'),(2837,5173,'#d9ad23'),(2838,5174,'#FFFFFF'),(2839,5178,'#2449cd'),(2840,5179,'#70c2ea'),(2841,5184,'#69ed27'),(2842,5185,'#f9bb01'),(2843,5186,'#506bd8'),(2844,5191,'#FFFFFF'),(2845,5193,'#7039f5'),(2846,5194,'#b301e4'),(2847,5195,'#FFFFFF'),(2848,5196,'#22f855'),(2849,5197,'#d50b63'),(2850,5202,'#d04f8d'),(2851,5203,'#FFFFFF'),(2852,5205,'#FFFFFF'),(2853,5208,'#FFFFFF'),(2854,5211,'#FFFFFF'),(2855,5213,'#fc3c6e'),(2856,5214,'#db8112'),(2857,5217,'#17f751'),(2858,5218,'#58f503'),(2859,5219,'#97ee51'),(2860,5220,'#FFFFFF'),(2861,5221,'#e44de3'),(2862,5222,'#c444e8'),(2863,5226,'#424ff6'),(2864,5228,'#c023e1'),(2865,5230,'#95fc4b'),(2866,5231,'#7702dd'),(2867,5233,'#5ff845'),(2868,5234,'#e667eb'),(2869,5236,'#7c2ad0'),(2870,5239,'#f77aec'),(2871,5240,'#508fdf'),(2872,5242,'#4dc1e9'),(2873,5243,'#FFFFFF'),(2874,5244,'#ce884a'),(2875,5245,'#2261dc'),(2876,5246,'#3cc5d2'),(2877,5248,'#fb1de9'),(2878,5251,'#5a65db'),(2879,5252,'#d07240'),(2880,5257,'#f9b14e'),(2881,5259,'#FFFFFF'),(2882,5261,'#FFFFFF'),(2883,5262,'#93db00'),(2884,5266,'#0fede6'),(2885,5268,'#FFFFFF'),(2886,5269,'#a0e255'),(2887,5270,'#4fcbe2'),(2888,5271,'#e51582'),(2889,5273,'#FFFFFF'),(2890,5274,'#e1833f'),(2891,5277,'#FFFFFF'),(2892,5278,'#0463cf'),(2893,5279,'#362bea'),(2894,5281,'#68d68c'),(2895,5282,'#FFFFFF'),(2896,5285,'#a762e3'),(2897,5286,'#f72573'),(2898,5287,'#f0a75b'),(2899,5289,'#3ef06e'),(2900,5290,'#6a16ea'),(2901,5291,'#FFFFFF'),(2902,5292,'#d6f361'),(2903,5297,'#fc115f'),(2904,5298,'#245bfa'),(2905,5301,'#d61c55'),(2906,5303,'#df64bb'),(2907,5304,'#543aed'),(2908,5305,'#60fe7c'),(2909,5307,'#7f43f8'),(2910,5308,'#62e3b0'),(2911,5309,'#d9d20c'),(2912,5311,'#ad37d4'),(2913,5312,'#FFFFFF'),(2914,5313,'#1d9ff2'),(2915,5314,'#69d65a'),(2916,5315,'#3de0be'),(2917,5316,'#21ed66'),(2918,5317,'#703fda'),(2919,5318,'#FFFFFF'),(2920,5320,'#FFFFFF'),(2921,5321,'#9bde15'),(2922,5322,'#abf22a'),(2923,5326,'#3cdc90'),(2924,5327,'#f53d5c'),(2925,5330,'#e129c9'),(2926,5331,'#bbcf55'),(2927,5333,'#b642e6'),(2928,5334,'#d98216'),(2929,5335,'#d7e722'),(2930,5336,'#dc0fb3'),(2931,5338,'#9146ea'),(2932,5344,'#72a0e8'),(2933,5345,'#d33ed5'),(2934,5346,'#8ffb6a'),(2935,5348,'#FFFFFF'),(2936,5349,'#597bd1'),(2937,5350,'#45f06d'),(2938,5351,'#5bdb5c'),(2939,5353,'#50a3f1'),(2940,5357,'#e749a3'),(2941,5359,'#71e30f'),(2942,5361,'#25e579'),(2943,5362,'#6612da'),(2944,5365,'#3570f1'),(2945,5366,'#d3402d'),(2946,5367,'#fa16d8'),(2947,5369,'#FFFFFF'),(2948,5372,'#4ed545'),(2949,5374,'#FFFFFF'),(2950,5375,'#7090ea'),(2951,5376,'#c607e4'),(2952,5378,'#66f969'),(2953,5381,'#6f81e5'),(2954,5382,'#dd9f1d'),(2955,5383,'#7d5ed4'),(2956,5385,'#371cd2'),(2957,5386,'#4cce75'),(2958,5388,'#23b9e9'),(2959,5389,'#1837f1'),(2960,5390,'#FFFFFF'),(2961,5392,'#6ec5dc'),(2962,5394,'#FFFFFF'),(2963,5395,'#960ad1'),(2964,5399,'#d0924e'),(2965,5401,'#6b20cc'),(2966,5402,'#e66099'),(2967,5404,'#ab07e6'),(2968,5406,'#5ef4b3'),(2969,5407,'#2921e6'),(2970,5408,'#FFFFFF'),(2971,5410,'#34fc22'),(2972,5411,'#33e8f7'),(2973,5412,'#69d209'),(2974,5415,'#7550d3'),(2975,5417,'#FFFFFF'),(2976,5418,'#1bd5c5'),(2977,5419,'#eb6522'),(2978,5420,'#1df11d'),(2979,5421,'#36f1a0'),(2980,5422,'#FFFFFF'),(2981,5423,'#f664c1'),(2982,5426,'#5bee65'),(2983,5428,'#83db5a'),(2984,5433,'#d0042b'),(2985,5434,'#3b95e2'),(2986,5435,'#340bea'),(2987,5437,'#c3ee1a'),(2988,5443,'#33a1fe'),(2989,5444,'#ea7a64'),(2990,5445,'#c7df3f'),(2991,5449,'#cd5106'),(2992,5453,'#ee5a24'),(2993,5455,'#ece873'),(2994,5457,'#60cd06'),(2995,5458,'#d91a18'),(2996,5459,'#FFFFFF'),(2997,5460,'#c52ed8'),(2998,5461,'#830ffa'),(2999,5468,'#e9379d'),(3000,5473,'#61aae2'),(3001,5478,'#cc65df'),(3002,5480,'#dfea07'),(3003,5481,'#4eead9'),(3004,5482,'#a577f1'),(3005,5486,'#d94615'),(3006,5487,'#5ad061'),(3007,5488,'#4c70cc'),(3008,5489,'#3d52d4'),(3009,5490,'#d7559a'),(3010,5491,'#e96282'),(3011,5492,'#f36a9b'),(3012,5493,'#e408b4'),(3013,5495,'#1273d5'),(3014,5497,'#08e645'),(3015,5498,'#77f990'),(3016,5499,'#fb7c61'),(3017,5502,'#2b61e4'),(3018,5504,'#10a4f9'),(3019,5505,'#53ddde'),(3020,5507,'#FFFFFF'),(3021,5508,'#3cddda'),(3022,5509,'#e7e749'),(3023,5510,'#0234f2'),(3024,5511,'#bd09ee'),(3025,5512,'#5e99e5'),(3026,5513,'#83e736'),(3027,5515,'#b51ff8'),(3028,5516,'#b7dd6b'),(3029,5517,'#f820be'),(3030,5518,'#79da5d'),(3031,5521,'#FFFFFF'),(3032,5522,'#69d571'),(3033,5525,'#a526fd'),(3034,5526,'#17d344'),(3035,5527,'#e7639a'),(3036,5528,'#efc241'),(3037,5529,'#FFFFFF'),(3038,5530,'#0e14e0'),(3039,5531,'#FFFFFF'),(3040,5532,'#4cd545'),(3041,5533,'#950bf4'),(3042,5537,'#27d590'),(3043,5538,'#711fd2'),(3044,5539,'#3ad392'),(3045,5541,'#e5c548'),(3046,5542,'#ce504b'),(3047,5543,'#1baddc'),(3048,5546,'#42d4a4'),(3049,5549,'#FFFFFF'),(3050,5550,'#91e555'),(3051,5553,'#c4d458'),(3052,5556,'#54dead'),(3053,5557,'#19f72d'),(3054,5559,'#fae44b'),(3055,5560,'#1244e7'),(3056,5561,'#cfca5c'),(3057,5563,'#54e29e'),(3058,5565,'#2362dd'),(3059,5567,'#FFFFFF'),(3060,5568,'#1bfba3'),(3061,5571,'#FFFFFF'),(3062,5577,'#dd6bb6'),(3063,5578,'#51e47a'),(3064,5582,'#d0e133'),(3065,5583,'#024be3'),(3066,5584,'#6392e0'),(3067,5585,'#2fd322'),(3068,5587,'#81f64c'),(3069,5588,'#2380df'),(3070,5590,'#47cd1d'),(3071,5591,'#1993e0'),(3072,5592,'#a76ddf'),(3073,5593,'#fb6ad2'),(3074,5594,'#26fc92'),(3075,5596,'#58ed39'),(3076,5599,'#3b4fd0'),(3077,5605,'#cd5622'),(3078,5608,'#FFFFFF'),(3079,5609,'#6e77ee'),(3080,5614,'#4987cd'),(3081,5616,'#54fcf2'),(3082,5617,'#FFFFFF'),(3083,5618,'#a2e616'),(3084,5619,'#cf5b51'),(3085,5620,'#cda854'),(3086,5621,'#5c7edf'),(3087,5622,'#f257f2'),(3088,5623,'#d361a8'),(3089,5624,'#FFFFFF'),(3090,5627,'#78c0f6'),(3091,5629,'#23c4d2'),(3092,5631,'#1f3ae3'),(3093,5632,'#05e773'),(3094,5633,'#c718eb'),(3095,5634,'#39d1b9'),(3096,5635,'#FFFFFF'),(3097,5636,'#5ae385'),(3098,5637,'#baed5b'),(3099,5638,'#cf5ea2'),(3100,5639,'#35f2bd'),(3101,5641,'#e46625'),(3102,5642,'#e031ad'),(3103,5643,'#30ea57'),(3104,5646,'#e81dd1'),(3105,5647,'#e7557a'),(3106,5649,'#ccf81c'),(3107,5650,'#f519cf'),(3108,5651,'#ef863b'),(3109,5653,'#fa2f75'),(3110,5654,'#7077ed'),(3111,5655,'#237ed4'),(3112,5660,'#72cd40'),(3113,5661,'#3cd55a'),(3114,5662,'#1bf8f1'),(3115,5666,'#cd03bd'),(3116,5675,'#81e06a'),(3117,5678,'#4640e8'),(3118,5679,'#fdc52e'),(3119,5680,'#3add29'),(3120,5681,'#ba32e3'),(3121,5686,'#42e481'),(3122,5687,'#5ae47c'),(3123,5688,'#6e07e1'),(3124,5690,'#FFFFFF'),(3125,5691,'#7858d5'),(3126,5697,'#c6de60'),(3127,5698,'#FFFFFF'),(3128,5699,'#d439ad'),(3129,5700,'#36d36f'),(3130,5702,'#694ed4'),(3131,5703,'#fbc446'),(3132,5705,'#fdc27d'),(3133,5707,'#3e89cd'),(3134,5708,'#f39e0b'),(3135,5709,'#9e20d7'),(3136,5711,'#a778f0'),(3137,5712,'#0dcfac'),(3138,5713,'#d331c3'),(3139,5714,'#11dfe8'),(3140,5716,'#dd1798'),(3141,5717,'#6be560'),(3142,5718,'#FFFFFF'),(3143,5720,'#7c39d1'),(3144,5722,'#f17098'),(3145,5723,'#d20387'),(3146,5725,'#5686de'),(3147,5726,'#b8f30a'),(3148,5728,'#FFFFFF'),(3149,5732,'#57ddcb'),(3150,5733,'#d259b9'),(3151,5737,'#FFFFFF'),(3152,5739,'#FFFFFF'),(3153,5749,'#f90fd0'),(3154,5750,'#6fb0e0'),(3155,5753,'#7439f9'),(3156,5754,'#9d3bdf'),(3157,5758,'#89d05c'),(3158,5759,'#2bd695'),(3159,5760,'#05e2e2'),(3160,5761,'#29e130'),(3161,5762,'#c833df'),(3162,5763,'#9269ec'),(3163,5764,'#d5646b'),(3164,5765,'#2bd25d'),(3165,5768,'#ea57bd'),(3166,5769,'#1ac8cc'),(3167,5774,'#dfca3f'),(3168,5775,'#8b01f6'),(3169,5776,'#d6100b'),(3170,5778,'#77dff5'),(3171,5779,'#e9e831'),(3172,5783,'#604ced'),(3173,5784,'#e2e66f'),(3174,5785,'#5f2cd6'),(3175,5789,'#FFFFFF'),(3176,5792,'#47c5cf'),(3177,5793,'#47e9e2'),(3178,5794,'#3dbece'),(3179,5796,'#e7d031'),(3180,5797,'#FFFFFF'),(3181,5798,'#FFFFFF'),(3182,5800,'#4ed8a5'),(3183,5804,'#fe64f5'),(3184,5806,'#08e512'),(3185,5809,'#8f64ce'),(3186,5810,'#d4e052'),(3187,5811,'#fe6abe'),(3188,5815,'#6292f6'),(3189,5816,'#9bfd70'),(3190,5818,'#ed4d7b'),(3191,5819,'#13e7b9'),(3192,5822,'#FFFFFF'),(3193,5823,'#1fd513'),(3194,5824,'#4270d8'),(3195,5826,'#e9451b'),(3196,5827,'#FFFFFF'),(3197,5828,'#50f988'),(3198,5832,'#4ce834'),(3199,5836,'#FFFFFF'),(3200,5838,'#FFFFFF'),(3201,5839,'#7ac2fd'),(3202,5840,'#fb541b'),(3203,5841,'#fccf0d'),(3204,5844,'#db7751'),(3205,5845,'#7dfb79'),(3206,5847,'#350dce'),(3207,5852,'#2099f8'),(3208,5855,'#FFFFFF'),(3209,5857,'#6ce553'),(3210,5860,'#61ee57'),(3211,5863,'#03ce9c'),(3212,5871,'#FFFFFF'),(3213,5874,'#FFFFFF'),(3214,5877,'#FFFFFF'),(3215,5878,'#4afa40'),(3216,5879,'#bde016'),(3217,5880,'#16d1ca'),(3218,5882,'#eb2a0f'),(3219,5884,'#FFFFFF'),(3220,5885,'#5db6d5'),(3221,5889,'#96d762'),(3222,5892,'#3acc54'),(3223,5893,'#f06a90'),(3224,5894,'#f6143a'),(3225,5896,'#f2e15a'),(3226,5897,'#e155f1'),(3227,5898,'#5fd43a'),(3228,5899,'#d1061d'),(3229,5901,'#1ed488'),(3230,5903,'#eb3334'),(3231,5906,'#392ed1'),(3232,5912,'#bf53fc'),(3233,5916,'#FFFFFF'),(3234,5919,'#FFFFFF'),(3235,5921,'#e28e4e'),(3236,5922,'#FFFFFF'),(3237,5928,'#703aee'),(3238,5929,'#dd8723'),(3239,5938,'#d20d81'),(3240,5939,'#714ece'),(3241,5941,'#a939cf'),(3242,5943,'#2593e7'),(3243,5945,'#d15532'),(3244,5946,'#FFFFFF'),(3245,5948,'#1de774'),(3246,5949,'#fd7663'),(3247,5951,'#f4e34b'),(3248,5952,'#fa386a'),(3249,5954,'#ef5df4'),(3250,5957,'#d60d0f'),(3251,5958,'#77e563'),(3252,5961,'#d963fe'),(3253,5964,'#282dd6'),(3254,5965,'#4aaafd'),(3255,5968,'#13b1da'),(3256,5969,'#36dc40'),(3257,5970,'#51c4d4'),(3258,5971,'#39fde2'),(3259,5972,'#8fd433'),(3260,5973,'#FFFFFF'),(3261,5974,'#5cd3ce'),(3262,5975,'#f10780'),(3263,5976,'#3ecd23'),(3264,5978,'#8a0ed2'),(3265,5979,'#6cda46'),(3266,5980,'#d7154c'),(3267,5984,'#b535f5'),(3268,5985,'#bde86e'),(3269,5988,'#fd08b4'),(3270,5989,'#3713cd'),(3271,5991,'#06e26c'),(3272,5994,'#FFFFFF'),(3273,5995,'#7c30eb'),(3274,5996,'#94f938'),(3275,5997,'#2e4ace'),(3276,5998,'#bfd241'),(3277,6000,'#80d02e'),(3278,6005,'#72f2b1'),(3279,6007,'#d7be01'),(3280,6008,'#273fe5'),(3281,6009,'#ec3cf2'),(3282,6014,'#2cd177'),(3283,6015,'#e955a2'),(3284,6019,'#f828b7'),(3285,6020,'#fd662c'),(3286,6021,'#11e325'),(3287,6023,'#eeb662'),(3288,6024,'#eb664f'),(3289,6026,'#5760f2'),(3290,6028,'#8bd167'),(3291,6029,'#3a79ec'),(3292,6030,'#b8ce60'),(3293,6031,'#cf4c4c'),(3294,6032,'#3fcf61'),(3295,6033,'#92f801'),(3296,6034,'#c6f743'),(3297,6035,'#f4df37'),(3298,6037,'#FFFFFF'),(3299,6038,'#b9f171'),(3300,6041,'#FFFFFF'),(3301,6043,'#f79401'),(3302,6045,'#5a6dd6'),(3303,6046,'#fc60e2'),(3304,6047,'#5472d7'),(3305,6048,'#8374f1'),(3306,6049,'#865ef2'),(3307,6051,'#ea55e4'),(3308,6052,'#0877d2'),(3309,6054,'#6de0da'),(3310,6055,'#FFFFFF'),(3311,6057,'#fbaa76'),(3312,6058,'#6c92f5'),(3313,6059,'#6daefc'),(3314,6060,'#f2c143'),(3315,6061,'#8b76fa'),(3316,6062,'#dbd85d'),(3317,6065,'#318ff0'),(3318,6066,'#0ad55a'),(3319,6067,'#FFFFFF'),(3320,6068,'#FFFFFF'),(3321,6070,'#8164f7'),(3322,6074,'#56dd15'),(3323,6081,'#f19d52'),(3324,6082,'#32e9cf'),(3325,6083,'#59f0f6'),(3326,6085,'#27e270'),(3327,6086,'#3cd90a'),(3328,6088,'#3c5cd9'),(3329,6089,'#d8786a'),(3330,6090,'#02fa72'),(3331,6091,'#2e75e9'),(3332,6092,'#f8ec78'),(3333,6095,'#61d9bd'),(3334,6097,'#5845fe'),(3335,6098,'#3fdb6e'),(3336,6099,'#329be2'),(3337,6100,'#FFFFFF'),(3338,6102,'#4000d3'),(3339,6106,'#45a3ef'),(3340,6108,'#e50dcf'),(3341,6109,'#5dbded'),(3342,6114,'#9d09e8'),(3343,6115,'#d6065b'),(3344,6117,'#40db6a'),(3345,6118,'#d51dcd'),(3346,6119,'#FFFFFF'),(3347,6121,'#844dfb'),(3348,6122,'#b362ec'),(3349,6124,'#73ef91'),(3350,6125,'#5dea73'),(3351,6127,'#c02eec'),(3352,6131,'#5be713'),(3353,6132,'#f42453'),(3354,6133,'#de14d8'),(3355,6134,'#ec4ca9'),(3356,6135,'#5a74fc'),(3357,6136,'#1c93f1'),(3358,6139,'#f35e3c'),(3359,6141,'#f32bb0'),(3360,6144,'#4b28cd'),(3361,6147,'#FFFFFF'),(3362,6148,'#41f5ca'),(3363,6149,'#4bda83'),(3364,6153,'#e5bc6f'),(3365,6155,'#51a1e8'),(3366,6156,'#36e56c'),(3367,6159,'#db700b'),(3368,6161,'#e58d44'),(3369,6162,'#87d334'),(3370,6163,'#e8c740'),(3371,6164,'#FFFFFF'),(3372,6165,'#2cefdd'),(3373,6166,'#6628d6'),(3374,6167,'#dd7afc'),(3375,6168,'#15cdcc'),(3376,6169,'#7427f4'),(3377,6170,'#FFFFFF'),(3378,6172,'#f1a80f'),(3379,6173,'#ae41f2'),(3380,6174,'#cc9e51'),(3381,6175,'#4ed84e'),(3382,6176,'#f89416'),(3383,6177,'#7de21e'),(3384,6179,'#7acc2e'),(3385,6181,'#67d9d5'),(3386,6182,'#dfaa4a'),(3387,6183,'#aaf336'),(3388,6185,'#855ed7'),(3389,6186,'#df2003'),(3390,6187,'#5ee3db'),(3391,6192,'#dc35e8'),(3392,6193,'#FFFFFF'),(3393,6194,'#e19f39'),(3394,6196,'#7cea02'),(3395,6198,'#e23bd8'),(3396,6201,'#27dba0'),(3397,6202,'#2937cd'),(3398,6203,'#FFFFFF'),(3399,6205,'#FFFFFF'),(3400,6209,'#FFFFFF'),(3401,6210,'#90fa76'),(3402,6213,'#6fe0b2'),(3403,6214,'#1974db'),(3404,6215,'#d10fcd'),(3405,6216,'#2fd022'),(3406,6219,'#6a75f1'),(3407,6220,'#FFFFFF'),(3408,6221,'#FFFFFF'),(3409,6222,'#d2ca18'),(3410,6224,'#a1d445'),(3411,6226,'#3d72fd'),(3412,6230,'#d90e3e'),(3413,6232,'#FFFFFF'),(3414,6235,'#66d525'),(3415,6237,'#18f1b7'),(3416,6239,'#9be965'),(3417,6240,'#34ec21'),(3418,6242,'#e52e29'),(3419,6243,'#8dd056'),(3420,6244,'#48dc2e'),(3421,6246,'#46e040'),(3422,6247,'#c71ade'),(3423,6250,'#1bdeda'),(3424,6255,'#7d2ee5'),(3425,6257,'#59d199'),(3426,6259,'#cf9c66'),(3427,6260,'#3b2ff8'),(3428,6261,'#5cf54d'),(3429,6263,'#bae855'),(3430,6265,'#e033b4'),(3431,6267,'#cf263d'),(3432,6269,'#f75e5b'),(3433,6271,'#FFFFFF'),(3434,6277,'#2896d7'),(3435,6279,'#6693e6'),(3436,6280,'#FFFFFF'),(3437,6281,'#4a86e2'),(3438,6283,'#6588e5'),(3439,6284,'#FFFFFF'),(3440,6285,'#c668d5'),(3441,6288,'#cc32cc'),(3442,6290,'#1c25fd'),(3443,6291,'#20d366'),(3444,6292,'#52f910'),(3445,6293,'#eb733d'),(3446,6295,'#54d8e9'),(3447,6299,'#FFFFFF'),(3448,6301,'#d8255c'),(3449,6302,'#7e2aea'),(3450,6303,'#99e628'),(3451,6305,'#19da78'),(3452,6307,'#35a1d0'),(3453,6313,'#394beb'),(3454,6314,'#07e3e7'),(3455,6315,'#c24bd5'),(3456,6316,'#4baace'),(3457,6317,'#f474cd'),(3458,6318,'#a854d0'),(3459,6319,'#6a76ed'),(3460,6321,'#df5d59'),(3461,6325,'#8b67dc'),(3462,6327,'#67f5b9'),(3463,6331,'#df6cb1'),(3464,6333,'#5c67d1'),(3465,6335,'#23d853'),(3466,6336,'#4a96ea'),(3467,6337,'#51eef7'),(3468,6338,'#ec726d'),(3469,6341,'#FFFFFF'),(3470,6346,'#e96ac5'),(3471,6347,'#5086f0'),(3472,6349,'#f43a2d'),(3473,6350,'#71cbea'),(3474,6351,'#30f9d0'),(3475,6352,'#7e67f5'),(3476,6355,'#d61b73'),(3477,6356,'#35dbac'),(3478,6360,'#9466ea'),(3479,6361,'#e4e252'),(3480,6362,'#d61da7'),(3481,6363,'#d5654c'),(3482,6364,'#8436de'),(3483,6368,'#5559d1'),(3484,6371,'#FFFFFF'),(3485,6372,'#27e4ed'),(3486,6373,'#4ef838'),(3487,6375,'#f9dd79'),(3488,6377,'#4fc7d2'),(3489,6378,'#6ae177'),(3490,6379,'#4ce1b0'),(3491,6382,'#ee3bfd'),(3492,6383,'#eddd56'),(3493,6384,'#ce67b8'),(3494,6385,'#f6b959'),(3495,6389,'#c2dc37'),(3496,6391,'#52a4e3'),(3497,6392,'#22dfdb'),(3498,6393,'#5dcc12'),(3499,6394,'#FFFFFF'),(3500,6395,'#2238ce'),(3501,6396,'#d19c15'),(3502,6399,'#4f8fd5'),(3503,6401,'#53fd95'),(3504,6403,'#59ebcc'),(3505,6405,'#FFFFFF'),(3506,6406,'#f271f1'),(3507,6408,'#cbf649'),(3508,6409,'#ee01ea'),(3509,6411,'#d529a2'),(3510,6413,'#5ecf4d'),(3511,6417,'#d1cf60'),(3512,6418,'#b6e26a'),(3513,6419,'#04b0d4'),(3514,6421,'#ee3c5c'),(3515,6422,'#9a5bdc'),(3516,6423,'#dc5340'),(3517,6425,'#1d29d2'),(3518,6426,'#da5664'),(3519,6427,'#db43d3'),(3520,6429,'#42fa42'),(3521,6430,'#df545d'),(3522,6434,'#5e82ed'),(3523,6435,'#44dcd3'),(3524,6439,'#0613dc'),(3525,6440,'#58ee94'),(3526,6441,'#beed23'),(3527,6443,'#272ed6'),(3528,6444,'#0a64f3'),(3529,6445,'#f410e3'),(3530,6447,'#f41a14'),(3531,6448,'#7a5fe8'),(3532,6451,'#183fe1'),(3533,6452,'#5af2f4'),(3534,6456,'#29f804'),(3535,6460,'#120ace'),(3536,6462,'#FFFFFF'),(3537,6463,'#e5cf30'),(3538,6465,'#ed277a'),(3539,6468,'#e971c9'),(3540,6469,'#dc7c47'),(3541,6470,'#6ac2fc'),(3542,6471,'#fe4388'),(3543,6472,'#50a9ec'),(3544,6475,'#895cd8'),(3545,6476,'#cf2474'),(3546,6478,'#FFFFFF'),(3547,6479,'#cf2989'),(3548,6480,'#68f1c2'),(3549,6483,'#0e3fe3'),(3550,6484,'#88cf3d'),(3551,6489,'#0bdc38'),(3552,6490,'#FFFFFF'),(3553,6491,'#02d8a9'),(3554,6492,'#b825d7'),(3555,6493,'#23d5df'),(3556,6495,'#950fd5'),(3557,6496,'#e82273'),(3558,6497,'#66eea3'),(3559,6498,'#53eadb'),(3560,6499,'#24e892'),(3561,6505,'#31de6f'),(3562,6506,'#ef3324'),(3563,6508,'#5dd2bb'),(3564,6510,'#df7245'),(3565,6516,'#FFFFFF'),(3566,6517,'#FFFFFF'),(3567,6518,'#08eba5'),(3568,6519,'#4c38df'),(3569,6520,'#1fc8e7'),(3570,6523,'#f84131'),(3571,6524,'#acea3b'),(3572,6525,'#ee1dd3'),(3573,6526,'#d8ad4b'),(3574,6528,'#e35010'),(3575,6530,'#28e36a'),(3576,6531,'#a237d4'),(3577,6532,'#fc5f73'),(3578,6536,'#6fc7e0'),(3579,6537,'#e17164'),(3580,6540,'#FFFFFF'),(3581,6541,'#FFFFFF'),(3582,6542,'#FFFFFF'),(3583,6544,'#FFFFFF'),(3584,6545,'#e48c6c'),(3585,6547,'#60bbdb'),(3586,6549,'#ddcc14'),(3587,6550,'#fa6c5f'),(3588,6551,'#FFFFFF'),(3589,6552,'#db1d56'),(3590,6553,'#a9fc76'),(3591,6554,'#c6ef22'),(3592,6555,'#8355e0'),(3593,6558,'#f375f6'),(3594,6559,'#3ed874'),(3595,6564,'#aeea32'),(3596,6566,'#f02b53'),(3597,6567,'#62e9db'),(3598,6568,'#5fd665'),(3599,6570,'#d76997'),(3600,6571,'#c451ec'),(3601,6572,'#ee4938'),(3602,6573,'#c207fc'),(3603,6583,'#dc365e'),(3604,6585,'#f5cf58'),(3605,6587,'#7b52e0'),(3606,6588,'#4a3fda'),(3607,6589,'#f34703'),(3608,6591,'#5b88f3'),(3609,6593,'#5ce756'),(3610,6594,'#0054f9'),(3611,6596,'#a022d4'),(3612,6598,'#d23a54'),(3613,6601,'#53d05a'),(3614,6602,'#12df57'),(3615,6604,'#ea6fd6'),(3616,6605,'#cc5f59'),(3617,6607,'#ed4f72'),(3618,6608,'#f46ad7'),(3619,6609,'#f1066f'),(3620,6610,'#550ae0'),(3621,6612,'#3c96d7'),(3622,6614,'#b3de18'),(3623,6615,'#d92b2d'),(3624,6616,'#5db1e3'),(3625,6618,'#d5b569'),(3626,6622,'#b1f24a'),(3627,6623,'#e19940'),(3628,6629,'#6736da'),(3629,6633,'#eb418a'),(3630,6637,'#312fcd'),(3631,6638,'#8c77f0'),(3632,6640,'#08f9d2'),(3633,6641,'#f0d50f'),(3634,6642,'#6148e9'),(3635,6644,'#3cd10b'),(3636,6649,'#FFFFFF'),(3637,6652,'#FFFFFF'),(3638,6653,'#c468ea'),(3639,6654,'#2e29d5'),(3640,6656,'#3e53e7'),(3641,6658,'#d9df6f'),(3642,6659,'#43cf55'),(3643,6660,'#62e050'),(3644,6661,'#dc60a3'),(3645,6662,'#2f78cc'),(3646,6670,'#FFFFFF'),(3647,6671,'#FFFFFF'),(3648,6672,'#fcf43b'),(3649,6673,'#e664ef'),(3650,6674,'#53e86d'),(3651,6677,'#FFFFFF'),(3652,6682,'#c9e428'),(3653,6683,'#0185d7'),(3654,6684,'#dfea27'),(3655,6685,'#f479c0'),(3656,6686,'#fc51ba'),(3657,6688,'#3a16d4'),(3658,6689,'#10d746'),(3659,6690,'#ee2463'),(3660,6693,'#35faa4'),(3661,6695,'#13fbbd'),(3662,6697,'#d81aa9'),(3663,6699,'#df9541'),(3664,6706,'#FFFFFF'),(3665,6708,'#b871f7'),(3666,6711,'#33a2ed'),(3667,6712,'#ead465'),(3668,6714,'#de6eed'),(3669,6716,'#FFFFFF'),(3670,6719,'#dfdd42'),(3671,6720,'#cace2e'),(3672,6723,'#0a59cc'),(3673,6724,'#28ed6d'),(3674,6725,'#d39b3e'),(3675,6727,'#7c32da'),(3676,6729,'#5d49d8'),(3677,6731,'#2b1cf0'),(3678,6732,'#d551c0'),(3679,6733,'#f7524f'),(3680,6737,'#f6dd6b'),(3681,6738,'#6b57fa'),(3682,6739,'#22b0d8'),(3683,6740,'#2ccad5'),(3684,6741,'#0406da'),(3685,6742,'#3aee13'),(3686,6743,'#e551dd'),(3687,6745,'#e325a8'),(3688,6746,'#FFFFFF'),(3689,6748,'#54d154'),(3690,6749,'#1be3f3'),(3691,6751,'#FFFFFF'),(3692,6752,'#5980dc'),(3693,6753,'#f5fa6e'),(3694,6755,'#e510be'),(3695,6756,'#1d42d6'),(3696,6760,'#a364d6'),(3697,6761,'#dc36ca'),(3698,6763,'#7ecd0f'),(3699,6764,'#3bd8a9'),(3700,6765,'#FFFFFF'),(3701,6766,'#4bdd38'),(3702,6768,'#1a09d5'),(3703,6771,'#8716d6'),(3704,6773,'#5bd137'),(3705,6774,'#FFFFFF'),(3706,6777,'#fb4586'),(3707,6779,'#d72063'),(3708,6780,'#4df314'),(3709,6782,'#FFFFFF'),(3710,6786,'#1ede42'),(3711,6788,'#FFFFFF'),(3712,6789,'#0ddd6a'),(3713,6790,'#FFFFFF'),(3714,6791,'#cfeb02'),(3715,6792,'#5ff67c'),(3716,6793,'#52a8f7'),(3717,6795,'#59d6a6'),(3718,6796,'#f4d90e'),(3719,6797,'#e962d9'),(3720,6798,'#FFFFFF'),(3721,6799,'#d121a4'),(3722,6800,'#3bf9a5'),(3723,6803,'#e4084a'),(3724,6804,'#e6e14b'),(3725,6805,'#FFFFFF'),(3726,6806,'#d846b3'),(3727,6809,'#f3e361'),(3728,6810,'#83cc47'),(3729,6811,'#acf451'),(3730,6813,'#8de369'),(3731,6814,'#55feaa'),(3732,6816,'#d6e452'),(3733,6818,'#793ad7'),(3734,6819,'#d26054'),(3735,6822,'#08c1f2'),(3736,6823,'#9e6be1'),(3737,6824,'#cc1ade'),(3738,6825,'#ed26d5'),(3739,6828,'#238bfe'),(3740,6829,'#dd0932'),(3741,6830,'#8ee20c'),(3742,6833,'#FFFFFF'),(3743,6836,'#8e54e0'),(3744,6840,'#b531df'),(3745,6842,'#61edf8'),(3746,6845,'#dcf120'),(3747,6847,'#dfa048'),(3748,6848,'#71f628'),(3749,6849,'#a813f2'),(3750,6851,'#e61559'),(3751,6855,'#de874a'),(3752,6858,'#9cf873'),(3753,6859,'#ce5546'),(3754,6860,'#99e96b'),(3755,6861,'#3ae967'),(3756,6862,'#6de852'),(3757,6863,'#4730f2'),(3758,6868,'#4ed83a'),(3759,6871,'#f14063'),(3760,6873,'#b13fe1'),(3761,6874,'#00e1e7'),(3762,6875,'#b261e9'),(3763,6876,'#fc5a60'),(3764,6877,'#eb6161'),(3765,6878,'#df5bf4'),(3766,6879,'#d1fe10'),(3767,6881,'#cc0976'),(3768,6882,'#FFFFFF'),(3769,6885,'#1aef71'),(3770,6888,'#3ae1a4'),(3771,6889,'#e853bf'),(3772,6890,'#7a5ae5'),(3773,6893,'#e24fad'),(3774,6898,'#25f712'),(3775,6899,'#f34c1d'),(3776,6900,'#2e0cfd'),(3777,6902,'#FFFFFF'),(3778,6903,'#FFFFFF'),(3779,6904,'#FFFFFF'),(3780,6905,'#974bdc'),(3781,6906,'#5cefe2'),(3782,6907,'#27e574'),(3783,6911,'#5da1d0'),(3784,6912,'#2251e1'),(3785,6913,'#f6c674'),(3786,6914,'#f3d40a'),(3787,6916,'#00def4'),(3788,6918,'#3a95cc'),(3789,6919,'#FFFFFF'),(3790,6925,'#11d9f6'),(3791,6930,'#1a5de9'),(3792,6933,'#c8cc62'),(3793,6935,'#2ca6d1'),(3794,6936,'#5d80f2'),(3795,6940,'#f1bc5a'),(3796,6944,'#d79749'),(3797,6946,'#2bb8ef'),(3798,6947,'#e8a142'),(3799,6951,'#ce4289'),(3800,6953,'#FFFFFF'),(3801,6954,'#FFFFFF'),(3802,6961,'#FFFFFF'),(3803,6963,'#e73c54'),(3804,6964,'#3cf6d3'),(3805,6968,'#FFFFFF'),(3806,6970,'#35d65a'),(3807,6971,'#db2f96'),(3808,6973,'#6ce2d4'),(3809,6974,'#FFFFFF'),(3810,6975,'#0ce849'),(3811,6979,'#e55bbd'),(3812,6986,'#e1e253'),(3813,6988,'#3dd627'),(3814,6991,'#72d222'),(3815,6992,'#f7d963'),(3816,6993,'#a94bf6'),(3817,6994,'#ced321'),(3818,6995,'#d768a9'),(3819,6996,'#cc6382'),(3820,6997,'#a0e959'),(3821,6999,'#1322e3'),(3822,7000,'#53f442'),(3823,7003,'#76f7fe'),(3824,7004,'#d93604'),(3825,7005,'#2aec6e'),(3826,7006,'#bae258'),(3827,7007,'#e36975'),(3828,7008,'#56d89a'),(3829,7010,'#9ee455'),(3830,7011,'#f8789d'),(3831,7012,'#f87c68'),(3832,7013,'#0359fd'),(3833,7014,'#5ed7a4'),(3834,7015,'#f950ba'),(3835,7016,'#2c47f6'),(3836,7017,'#3656e4'),(3837,7018,'#55e6d7'),(3838,7020,'#d8f755'),(3839,7021,'#e23324'),(3840,7022,'#06dcb4'),(3841,7023,'#f87a9c'),(3842,7024,'#544ff3'),(3843,7025,'#75fef1'),(3844,7026,'#f2ba24'),(3845,7029,'#78f2ae'),(3846,7030,'#ee28d5'),(3847,7031,'#6bc1eb'),(3848,7032,'#038cdc'),(3849,7035,'#8f4be5'),(3850,7038,'#23f527'),(3851,7040,'#eb3bdf'),(3852,7043,'#db4632'),(3853,7045,'#8648e1'),(3854,7049,'#f951ee'),(3855,7050,'#864cd4'),(3856,7051,'#d231c7'),(3857,7052,'#FFFFFF'),(3858,7055,'#baf240'),(3859,7056,'#08d322'),(3860,7057,'#e71492'),(3861,7058,'#9039fa'),(3862,7062,'#FFFFFF'),(3863,7067,'#60fd95'),(3864,7068,'#3ddb03'),(3865,7069,'#6b94f6'),(3866,7070,'#ec8d6a'),(3867,7073,'#9877f5'),(3868,7074,'#72f478'),(3869,7075,'#FFFFFF'),(3870,7078,'#b3f037'),(3871,7080,'#fc79b6'),(3872,7081,'#19d77a'),(3873,7083,'#FFFFFF'),(3874,7084,'#53e7cb'),(3875,7086,'#FFFFFF'),(3876,7088,'#8c0ddc'),(3877,7091,'#50d423'),(3878,7094,'#52ddb1'),(3879,7095,'#53acf0'),(3880,7098,'#5ee5ba'),(3881,7103,'#4a44d1'),(3882,7104,'#11c4e8'),(3883,7105,'#8241cf'),(3884,7106,'#FFFFFF'),(3885,7110,'#5cafe1'),(3886,7113,'#23ce39'),(3887,7117,'#99d81e'),(3888,7119,'#6ad3dd'),(3889,7121,'#52cae9'),(3890,7122,'#a0cd54'),(3891,7125,'#bfe34e'),(3892,7126,'#ea5598'),(3893,7127,'#b328fd'),(3894,7129,'#6aabe7'),(3895,7130,'#67ec77'),(3896,7132,'#5ed8c6'),(3897,7133,'#c2d749'),(3898,7135,'#a261f5'),(3899,7139,'#cc0b0d'),(3900,7140,'#FFFFFF'),(3901,7142,'#a753cf'),(3902,7143,'#5010de'),(3903,7146,'#27ee0a'),(3904,7147,'#7efdf3'),(3905,7148,'#4ca0f6'),(3906,7149,'#532ae5'),(3907,7150,'#aa19f7'),(3908,7152,'#d768e7'),(3909,7154,'#FFFFFF'),(3910,7155,'#22e68b'),(3911,7160,'#1fda52'),(3912,7162,'#FFFFFF'),(3913,7163,'#FFFFFF'),(3914,7164,'#2cd734'),(3915,7166,'#2d40cc'),(3916,7167,'#FFFFFF'),(3917,7171,'#44dbd8'),(3918,7173,'#FFFFFF'),(3919,7174,'#2930fd'),(3920,7177,'#FFFFFF'),(3921,7179,'#0ede0c'),(3922,7180,'#de772e'),(3923,7183,'#d56517'),(3924,7185,'#d0824f'),(3925,7188,'#1cefd6'),(3926,7190,'#FFFFFF'),(3927,7192,'#b6ee10'),(3928,7194,'#c1e73a'),(3929,7195,'#0514dd'),(3930,7196,'#FFFFFF'),(3931,7197,'#b1d762'),(3932,7198,'#5a55e8'),(3933,7199,'#d35afc'),(3934,7203,'#FFFFFF'),(3935,7204,'#0c1af1'),(3936,7207,'#db7e40'),(3937,7208,'#20f643'),(3938,7209,'#76e2f0'),(3939,7212,'#6ea8e6'),(3940,7214,'#FFFFFF'),(3941,7217,'#FFFFFF'),(3942,7218,'#3853fb'),(3943,7219,'#d25e03'),(3944,7220,'#294ad3'),(3945,7222,'#FFFFFF'),(3946,7223,'#dc37de'),(3947,7224,'#5aa2d4'),(3948,7225,'#eb49b3'),(3949,7226,'#51d92a'),(3950,7227,'#e8d253'),(3951,7230,'#32d882'),(3952,7231,'#e873ad'),(3953,7235,'#38ef52'),(3954,7236,'#a1d14f'),(3955,7239,'#bdf157'),(3956,7240,'#e7e516'),(3957,7241,'#7119df'),(3958,7242,'#b5da1e'),(3959,7243,'#614cdf'),(3960,7246,'#FFFFFF'),(3961,7247,'#c1d640'),(3962,7248,'#da8f62'),(3963,7253,'#45f48e'),(3964,7256,'#67d96f'),(3965,7259,'#4edb8f'),(3966,7263,'#b95cf0'),(3967,7266,'#2ddf7a'),(3968,7268,'#08ccaf'),(3969,7269,'#d86754'),(3970,7273,'#FFFFFF'),(3971,7274,'#FFFFFF'),(3972,7276,'#a356dc'),(3973,7277,'#6bd45d'),(3974,7279,'#b40ff3'),(3975,7283,'#FFFFFF'),(3976,7286,'#FFFFFF'),(3977,7290,'#e15256'),(3978,7291,'#8c0ae4'),(3979,7292,'#e27051'),(3980,7293,'#e528c9'),(3981,7294,'#00cc68'),(3982,7295,'#3124ce'),(3983,7299,'#c2da65'),(3984,7300,'#FFFFFF'),(3985,7301,'#94e865'),(3986,7303,'#df294a'),(3987,7304,'#3643d0'),(3988,7306,'#e3ca52'),(3989,7307,'#d324d8'),(3990,7310,'#94d953'),(3991,7312,'#cae656'),(3992,7313,'#ce11cc'),(3993,7314,'#44bee9'),(3994,7316,'#1269df'),(3995,7327,'#4dfb84'),(3996,7330,'#e5e413'),(3997,7332,'#d018d4'),(3998,7333,'#d06847'),(3999,7334,'#0dcced'),(4000,7335,'#bbdf58'),(4001,7336,'#595bf7'),(4002,7337,'#52dc3d'),(4003,7338,'#FFFFFF'),(4004,7340,'#e04811'),(4005,7341,'#6d2ad1'),(4006,7342,'#ec190c'),(4007,7343,'#9fcd54'),(4008,7348,'#546fce'),(4009,7349,'#66f89c'),(4010,7350,'#f1c978'),(4011,7352,'#f7bc0e'),(4012,7353,'#a230dd'),(4013,7355,'#f0d55a'),(4014,7357,'#17e222'),(4015,7358,'#FFFFFF'),(4016,7360,'#a369eb'),(4017,7361,'#24cde1'),(4018,7364,'#09d6d6'),(4019,7365,'#19d6b0'),(4020,7367,'#69dc40'),(4021,7370,'#e59d5d'),(4022,7371,'#e419c9'),(4023,7373,'#287ffc'),(4024,7375,'#8202e3'),(4025,7376,'#48ec05'),(4026,7377,'#e81fbf'),(4027,7379,'#69f918'),(4028,7380,'#47de91'),(4029,7381,'#7744db'),(4030,7382,'#d67967'),(4031,7383,'#298ee0'),(4032,7384,'#16fcc5'),(4033,7385,'#20acf8'),(4034,7386,'#73e255'),(4035,7389,'#32e377'),(4036,7390,'#f42bb8'),(4037,7394,'#e03b56'),(4038,7396,'#fe5ef8'),(4039,7400,'#010ece'),(4040,7401,'#3cce67'),(4041,7404,'#dc9031'),(4042,7406,'#FFFFFF'),(4043,7409,'#cd56d3'),(4044,7411,'#bb5acd'),(4045,7414,'#cfed47'),(4046,7416,'#d81f80'),(4047,7417,'#bf13e1'),(4048,7418,'#973fe9'),(4049,7419,'#3be79d'),(4050,7421,'#d1e13a'),(4051,7422,'#f10db4'),(4052,7424,'#2ae5ee'),(4053,7425,'#71fa41'),(4054,7426,'#a3cf15'),(4055,7428,'#17ebcc'),(4056,7429,'#352ed8'),(4057,7432,'#54a6d7'),(4058,7438,'#e731b7'),(4059,7440,'#dc2654'),(4060,7441,'#484dcf'),(4061,7444,'#b9ed6a'),(4062,7445,'#d6a766'),(4063,7446,'#02f878'),(4064,7448,'#f27924'),(4065,7451,'#95d936'),(4066,7452,'#d753f6'),(4067,7454,'#9722fd'),(4068,7456,'#c0d760'),(4069,7457,'#0328fd'),(4070,7458,'#684adc'),(4071,7459,'#6b7ddd'),(4072,7460,'#b259e7'),(4073,7468,'#19da5d'),(4074,7472,'#afec60'),(4075,7473,'#07e214'),(4076,7476,'#FFFFFF'),(4077,7480,'#9865ce'),(4078,7482,'#1ac4f3'),(4079,7488,'#ca5ffa'),(4080,7489,'#deed59'),(4081,7490,'#FFFFFF'),(4082,7492,'#f14850'),(4083,7493,'#3f88fc'),(4084,7496,'#FFFFFF'),(4085,7498,'#FFFFFF'),(4086,7499,'#e657e7'),(4087,7500,'#700ddd'),(4088,7501,'#6addbc'),(4089,7505,'#FFFFFF'),(4090,7507,'#ef2be6'),(4091,7508,'#08cf8d'),(4092,7509,'#fa22a4'),(4093,7511,'#49d2b2'),(4094,7512,'#9efd2a'),(4095,7515,'#0ee982'),(4096,7516,'#2bb2f6'),(4097,7517,'#69daeb'),(4098,7518,'#f1cd5d'),(4099,7520,'#41cd4f'),(4100,7522,'#21cf27'),(4101,7523,'#34e3ad'),(4102,7524,'#f71018'),(4103,7525,'#d19926'),(4104,7530,'#FFFFFF'),(4105,7531,'#e90840'),(4106,7532,'#65b4f2'),(4107,7534,'#8959f3'),(4108,7535,'#69dee2'),(4109,7537,'#b1e94c'),(4110,7538,'#bfee5f'),(4111,7539,'#1fee2c'),(4112,7540,'#97cc4c'),(4113,7541,'#135dd5'),(4114,7543,'#f026e3'),(4115,7544,'#7018cd'),(4116,7545,'#fc72dd'),(4117,7550,'#FFFFFF'),(4118,7552,'#d48a2c'),(4119,7554,'#23d0a4'),(4120,7555,'#a4d914'),(4121,7556,'#00f72f'),(4122,7558,'#FFFFFF'),(4123,7559,'#ed2c23'),(4124,7563,'#70e210'),(4125,7564,'#ed5e8b'),(4126,7566,'#FFFFFF'),(4127,7567,'#FFFFFF'),(4128,7568,'#34d990'),(4129,7570,'#d44a34'),(4130,7576,'#FFFFFF'),(4131,7578,'#d7ed05'),(4132,7579,'#7e64cd'),(4133,7580,'#fa558f'),(4134,7581,'#b43bd6'),(4135,7584,'#ec202e'),(4136,7586,'#1962e8'),(4137,7587,'#45ec8b'),(4138,7588,'#8ae714'),(4139,7589,'#f5b70b'),(4140,7597,'#6593d3'),(4141,7598,'#FFFFFF'),(4142,7599,'#0238f5'),(4143,7600,'#FFFFFF'),(4144,7601,'#d4d95c'),(4145,7604,'#38dd92'),(4146,7607,'#3a47d6'),(4147,7609,'#17e46a'),(4148,7610,'#2ee3b1'),(4149,7611,'#0024fa'),(4150,7612,'#ce489b'),(4151,7613,'#d3336a'),(4152,7614,'#d0e76e'),(4153,7617,'#49d994'),(4154,7620,'#60d069'),(4155,7621,'#4ad844'),(4156,7622,'#20bee8'),(4157,7624,'#fcb85b'),(4158,7627,'#802fd9'),(4159,7628,'#ee74b3'),(4160,7629,'#FFFFFF'),(4161,7632,'#FFFFFF'),(4162,7634,'#4f0dd9'),(4163,7635,'#e762f4'),(4164,7641,'#ee3230'),(4165,7642,'#53f717'),(4166,7644,'#e76840'),(4167,7645,'#f4ce1a'),(4168,7646,'#FFFFFF'),(4169,7647,'#6ca1f6'),(4170,7648,'#f9eb79'),(4171,7650,'#383df5'),(4172,7651,'#8a0cce'),(4173,7652,'#e31926'),(4174,7654,'#0edaf5'),(4175,7655,'#0f9be6'),(4176,7656,'#ba75ec'),(4177,7657,'#ee4149'),(4178,7659,'#FFFFFF'),(4179,7661,'#e6f132'),(4180,7662,'#adcc3e'),(4181,7663,'#556bcf'),(4182,7664,'#27c3da'),(4183,7666,'#FFFFFF'),(4184,7667,'#f72a68'),(4185,7668,'#d49e53'),(4186,7671,'#cfa10e'),(4187,7672,'#da2d0e'),(4188,7673,'#cc2c02'),(4189,7674,'#4524d4'),(4190,7675,'#50d71c'),(4191,7677,'#4ba1e9'),(4192,7678,'#f3fc4c'),(4193,7679,'#d6558e'),(4194,7681,'#eb331d'),(4195,7685,'#14d802'),(4196,7687,'#e3084e'),(4197,7692,'#FFFFFF'),(4198,7693,'#46a5d6'),(4199,7694,'#cdcb2c'),(4200,7695,'#10e4e0'),(4201,7697,'#e14b50'),(4202,7703,'#ef9666'),(4203,7705,'#6e9ae5'),(4204,7706,'#2aa2cc'),(4205,7708,'#a7d54b'),(4206,7709,'#2859d2'),(4207,7710,'#2fa9e4'),(4208,7711,'#219cfe'),(4209,7713,'#d2c15b'),(4210,7714,'#52d540'),(4211,7716,'#f364fa'),(4212,7719,'#09c9f1'),(4213,7723,'#04a1dd'),(4214,7724,'#FFFFFF'),(4215,7725,'#d36399'),(4216,7726,'#6df2d7'),(4217,7727,'#e70d45'),(4218,7728,'#6889ee'),(4219,7729,'#FFFFFF'),(4220,7730,'#FFFFFF'),(4221,7731,'#5fdf2f'),(4222,7737,'#49c7f0'),(4223,7739,'#76cc4e'),(4224,7740,'#e63c84'),(4225,7742,'#2e1cfe'),(4226,7744,'#6b1df0'),(4227,7745,'#ea301f'),(4228,7746,'#b906d8'),(4229,7754,'#6559e9'),(4230,7755,'#f27a00'),(4231,7756,'#58f0b8'),(4232,7757,'#2918dd'),(4233,7759,'#5ed950'),(4234,7763,'#bb5cfa'),(4235,7764,'#e0049f'),(4236,7765,'#b1f02b'),(4237,7766,'#FFFFFF'),(4238,7767,'#e9171b'),(4239,7768,'#8728e6'),(4240,7769,'#6379ed'),(4241,7770,'#FFFFFF'),(4242,7772,'#63e6f7'),(4243,7773,'#5f4ee4'),(4244,7774,'#25a7ed'),(4245,7776,'#f97d3d'),(4246,7780,'#defc6c'),(4247,7782,'#edf925'),(4248,7783,'#c252e0'),(4249,7784,'#1d65e5'),(4250,7786,'#fc5a4c'),(4251,7789,'#e53320'),(4252,7790,'#e06ad3'),(4253,7794,'#0aedbc'),(4254,7798,'#e92307'),(4255,7800,'#e8425a'),(4256,7801,'#FFFFFF'),(4257,7802,'#cd0b79'),(4258,7804,'#316ee1'),(4259,7805,'#ecb070'),(4260,7807,'#FFFFFF'),(4261,7808,'#6850f5'),(4262,7809,'#d425c9'),(4263,7812,'#0297f5'),(4264,7814,'#e8a756'),(4265,7817,'#ce7e5e'),(4266,7824,'#ee5fb4'),(4267,7825,'#ce3301'),(4268,7828,'#12e023'),(4269,7832,'#4c98e6'),(4270,7833,'#58dbc3'),(4271,7836,'#f87f11'),(4272,7837,'#3bdf24'),(4273,7838,'#47e465'),(4274,7840,'#FFFFFF'),(4275,7841,'#FFFFFF'),(4276,7842,'#d69a3a'),(4277,7845,'#2ee37d'),(4278,7848,'#ae5ace'),(4279,7849,'#acef3c'),(4280,7850,'#1588ee'),(4281,7851,'#77fcc2'),(4282,7852,'#FFFFFF'),(4283,7855,'#47e705'),(4284,7858,'#38fca3'),(4285,7860,'#FFFFFF'),(4286,7864,'#c74ceb'),(4287,7865,'#10bfe0'),(4288,7869,'#cc570e'),(4289,7871,'#FFFFFF'),(4290,7873,'#33d3b8'),(4291,7874,'#c1d00c'),(4292,7875,'#1432df'),(4293,7876,'#FFFFFF'),(4294,7877,'#FFFFFF'),(4295,7878,'#f24e13'),(4296,7879,'#FFFFFF'),(4297,7881,'#dca246'),(4298,7882,'#0c8ddd'),(4299,7883,'#65cde9'),(4300,7884,'#c6d650'),(4301,7885,'#30e139'),(4302,7886,'#FFFFFF'),(4303,7887,'#FFFFFF'),(4304,7890,'#7877fc'),(4305,7891,'#6086df'),(4306,7892,'#FFFFFF'),(4307,7893,'#FFFFFF'),(4308,7894,'#5ee736'),(4309,7898,'#e88d71'),(4310,7906,'#d133af'),(4311,7907,'#f50081'),(4312,7908,'#86d45b'),(4313,7909,'#f09169'),(4314,7910,'#19f278'),(4315,7911,'#f05843'),(4316,7913,'#09dd01'),(4317,7914,'#ddf059'),(4318,7916,'#FFFFFF'),(4319,7917,'#a305e9'),(4320,7918,'#9e63d0'),(4321,7923,'#ac53ee'),(4322,7925,'#6dd500'),(4323,7926,'#5a15cf'),(4324,7927,'#efb85e'),(4325,7928,'#c439fd'),(4326,7929,'#d3d013'),(4327,7935,'#d65f94'),(4328,7938,'#d42f8e'),(4329,7941,'#4f74e0'),(4330,7942,'#fb0644'),(4331,7944,'#fc3270'),(4332,7946,'#fd19a7'),(4333,7947,'#6895ea'),(4334,7951,'#55d8b8'),(4335,7952,'#58c2f5'),(4336,7953,'#30e7ce'),(4337,7954,'#07de0a'),(4338,7956,'#49aee6'),(4339,7961,'#FFFFFF'),(4340,7962,'#eb69e6'),(4341,7963,'#0cecd3'),(4342,7965,'#eb0c37'),(4343,7974,'#FFFFFF'),(4344,7975,'#b9d01a'),(4345,7977,'#51d5fe'),(4346,7978,'#e4e471'),(4347,7979,'#1304e2'),(4348,7983,'#d2af08'),(4349,7984,'#930bf3'),(4350,7986,'#0c72d0'),(4351,7987,'#6991d3'),(4352,7988,'#68fcb3'),(4353,7989,'#ef7457'),(4354,7990,'#f12f26'),(4355,7992,'#85cc0f'),(4356,7993,'#67dff0'),(4357,7997,'#d38061'),(4358,8001,'#c0fb77'),(4359,8003,'#d25bef'),(4360,8007,'#FFFFFF'),(4361,8009,'#5e75f3'),(4362,8010,'#e962c2'),(4363,8012,'#d84acc'),(4364,8016,'#dd0405'),(4365,8017,'#e33f3e'),(4366,8018,'#d3d04c'),(4367,8023,'#ed6bea'),(4368,8024,'#ee6795'),(4369,8026,'#c0fa77'),(4370,8028,'#f05dc7'),(4371,8029,'#FFFFFF'),(4372,8032,'#d61428'),(4373,8033,'#db8363'),(4374,8034,'#cc2a6d'),(4375,8037,'#0a2cea'),(4376,8038,'#f2031b'),(4377,8039,'#ca30e1'),(4378,8042,'#FFFFFF'),(4379,8043,'#e08f6b'),(4380,8045,'#fee81d'),(4381,8046,'#e8ef77'),(4382,8047,'#FFFFFF'),(4383,8050,'#d3a323'),(4384,8051,'#5facec'),(4385,8052,'#adce51'),(4386,8056,'#f820fc'),(4387,8057,'#205afd'),(4388,8058,'#8d6ddc'),(4389,8059,'#d0b95d'),(4390,8060,'#ee0fc0'),(4391,8061,'#e69649'),(4392,8062,'#db29ab'),(4393,8063,'#4436e2'),(4394,8064,'#23f6b8'),(4395,8065,'#139bde'),(4396,8066,'#a453fe'),(4397,8068,'#1b97cf'),(4398,8069,'#FFFFFF'),(4399,8072,'#830bdb'),(4400,8073,'#5d0fce'),(4401,8074,'#6f8be9'),(4402,8077,'#f554ab'),(4403,8082,'#e1e052'),(4404,8084,'#b05fce'),(4405,8086,'#FFFFFF'),(4406,8087,'#a54bcd'),(4407,8088,'#2cd6be'),(4408,8089,'#6c69e0'),(4409,8091,'#6b45fd'),(4410,8092,'#00c9f0'),(4411,8093,'#468df6'),(4412,8094,'#60e2cc'),(4413,8095,'#0cf14f'),(4414,8098,'#da8e14'),(4415,8103,'#cacd45'),(4416,8104,'#FFFFFF'),(4417,8105,'#a066eb'),(4418,8107,'#84d117'),(4419,8108,'#FFFFFF'),(4420,8109,'#6861da'),(4421,8110,'#d2a34c'),(4422,8111,'#fa60ed'),(4423,8112,'#f5d174'),(4424,8116,'#8117d9'),(4425,8118,'#0570d2'),(4426,8120,'#6eedc9'),(4427,8123,'#48d373'),(4428,8126,'#f13b65'),(4429,8127,'#d06d42'),(4430,8129,'#ef4668'),(4431,8130,'#d59155'),(4432,8131,'#c2cd20'),(4433,8133,'#f18351'),(4434,8134,'#4d38eb'),(4435,8137,'#24d438'),(4436,8138,'#2fdd62'),(4437,8139,'#d11ecf'),(4438,8142,'#f21376'),(4439,8145,'#beec60'),(4440,8146,'#dc465b'),(4441,8150,'#cd0a6d'),(4442,8154,'#d48464'),(4443,8157,'#4584e0'),(4444,8159,'#da6abc'),(4445,8160,'#08d98a'),(4446,8162,'#1aeafc'),(4447,8163,'#6f4bd6'),(4448,8164,'#f14143'),(4449,8165,'#FFFFFF'),(4450,8168,'#2c01e7'),(4451,8169,'#32f8a4'),(4452,8170,'#fb7291'),(4453,8172,'#ce4414'),(4454,8173,'#25d3df'),(4455,8175,'#353de4'),(4456,8178,'#f7d535'),(4457,8179,'#f42f74'),(4458,8182,'#88de29'),(4459,8184,'#21eea0'),(4460,8188,'#7447ee'),(4461,8189,'#43f837'),(4462,8190,'#86e14c'),(4463,8191,'#94fe56'),(4464,8192,'#f6945f'),(4465,8195,'#d9e042'),(4466,8196,'#ec1f70'),(4467,8197,'#0469f8'),(4468,8200,'#f55c8c'),(4469,8201,'#fda077'),(4470,8203,'#73eb9d'),(4471,8204,'#5fe10b'),(4472,8210,'#edef32'),(4473,8211,'#50cecc'),(4474,8212,'#c1f373'),(4475,8213,'#f044c2'),(4476,8215,'#7ff555'),(4477,8219,'#2be7d4'),(4478,8220,'#1695fb'),(4479,8222,'#5d37ed'),(4480,8226,'#8b25f8'),(4481,8227,'#8945e3'),(4482,8233,'#dd27a9'),(4483,8236,'#FFFFFF'),(4484,8237,'#f71ddd'),(4485,8238,'#96cc16'),(4486,8240,'#6316e4'),(4487,8241,'#850ee7'),(4488,8242,'#a928d3'),(4489,8243,'#65f986'),(4490,8244,'#d73729'),(4491,8248,'#FFFFFF'),(4492,8249,'#6af2c8'),(4493,8250,'#7b02d0'),(4494,8251,'#FFFFFF'),(4495,8255,'#dad860'),(4496,8256,'#d1cd4e'),(4497,8257,'#22d631'),(4498,8262,'#FFFFFF'),(4499,8263,'#f57974'),(4500,8264,'#f35960'),(4501,8266,'#1b7adb'),(4502,8268,'#c0dd48'),(4503,8269,'#18dc0b'),(4504,8270,'#68f507'),(4505,8272,'#ac0bd8'),(4506,8273,'#ef4961'),(4507,8274,'#f1e812'),(4508,8276,'#dba85a'),(4509,8277,'#FFFFFF'),(4510,8278,'#FFFFFF'),(4511,8280,'#fed252'),(4512,8283,'#FFFFFF'),(4513,8284,'#2364db'),(4514,8286,'#FFFFFF'),(4515,8287,'#FFFFFF'),(4516,8288,'#b9ee2e'),(4517,8290,'#a918fd'),(4518,8291,'#FFFFFF'),(4519,8297,'#e4e370'),(4520,8298,'#60fd63'),(4521,8301,'#56e759'),(4522,8302,'#d775fc'),(4523,8310,'#FFFFFF'),(4524,8312,'#FFFFFF'),(4525,8318,'#FFFFFF'),(4526,8327,'#b3dc26'),(4527,8331,'#4df32c'),(4528,8336,'#FFFFFF'),(4529,8337,'#FFFFFF'),(4530,8344,'#edd060'),(4531,8348,'#FFFFFF'),(4532,8349,'#FFFFFF'),(4533,8353,'#102ad0'),(4534,8356,'#d008c6'),(4535,8359,'#9cf648'),(4536,8360,'#12caf8'),(4537,8362,'#FFFFFF'),(4538,8364,'#6704df'),(4539,8369,'#FFFFFF'),(4540,8370,'#4ed8c5'),(4541,8378,'#def340'),(4542,8380,'#FFFFFF'),(4543,8382,'#FFFFFF'),(4544,8383,'#17ee18'),(4545,8384,'#3b29ce'),(4546,8385,'#FFFFFF'),(4547,8387,'#FFFFFF'),(4548,8390,'#FFFFFF'),(4549,8395,'#FFFFFF'),(4550,8396,'#FFFFFF'),(4551,8397,'#FFFFFF'),(4552,8399,'#23cbcd'),(4553,8404,'#d5129a'),(4554,8405,'#4387ef'),(4555,8406,'#FFFFFF'),(4556,8410,'#FFFFFF'),(4557,8412,'#FFFFFF'),(4558,8413,'#FFFFFF'),(4559,8414,'#FFFFFF'),(4560,8420,'#FFFFFF'),(4561,8422,'#FFFFFF'),(4562,8424,'#FFFFFF'),(4563,8425,'#e527ad'),(4564,8426,'#742bde'),(4565,8427,'#70e8a6'),(4566,8428,'#06d12e'),(4567,8429,'#FFFFFF'),(4568,8430,'#d4b31c'),(4569,8433,'#7be70c'),(4570,8434,'#FFFFFF'),(4571,8438,'#aaea18'),(4572,8439,'#476ad0'),(4573,8440,'#4bd311'),(4574,8441,'#1c12da'),(4575,8442,'#FFFFFF'),(4576,8444,'#FFFFFF'),(4577,8445,'#db20ae'),(4578,8446,'#FFFFFF'),(4579,8447,'#FFFFFF'),(4580,8448,'#eb36ac'),(4581,8449,'#1e23fe'),(4582,8450,'#FFFFFF'),(4583,8452,'#FFFFFF'),(4584,8455,'#FFFFFF'),(4585,8456,'#FFFFFF'),(4586,8458,'#FFFFFF'),(4587,8460,'#FFFFFF'),(4588,8462,'#FFFFFF'),(4589,8463,'#FFFFFF'),(4590,8464,'#FFFFFF'),(4591,8465,'#4ef079'),(4592,8466,'#FFFFFF'),(4593,8467,'#8ada28'),(4594,8472,'#FFFFFF'),(4595,8473,'#FFFFFF'),(4596,8474,'#FFFFFF'),(4597,8477,'#FFFFFF'),(4598,8480,'#FFFFFF'),(4599,8482,'#d16d18'),(4600,8486,'#FFFFFF'),(4601,8487,'#FFFFFF'),(4602,8489,'#FFFFFF'),(4603,8490,'#6bd35b'),(4604,8492,'#e89d00'),(4605,8493,'#518fe1'),(4606,8495,'#FFFFFF'),(4607,8500,'#FFFFFF'),(4608,8501,'#09c1ea'),(4609,8502,'#FFFFFF'),(4610,8505,'#FFFFFF'),(4611,8506,'#3eb1e3'),(4612,8512,'#7385f0'),(4613,8514,'#FFFFFF'),(4614,8516,'#69d218'),(4615,8517,'#2191f7'),(4616,8519,'#FFFFFF'),(4617,8520,'#78fc1b'),(4618,8522,'#FFFFFF'),(4619,8523,'#376fcd'),(4620,8528,'#FFFFFF'),(4621,8532,'#FFFFFF'),(4622,8533,'#50ce31'),(4623,8534,'#cd36c1'),(4624,8542,'#FFFFFF'),(4625,8543,'#e42b75'),(4626,8545,'#3aec7e'),(4627,8546,'#e24afc'),(4628,8548,'#ef6e28'),(4629,8554,'#29facd'),(4630,8557,'#FFFFFF'),(4631,8558,'#FFFFFF'),(4632,8564,'#d8040f'),(4633,8566,'#57e660'),(4634,8569,'#FFFFFF'),(4635,8571,'#FFFFFF'),(4636,8572,'#58d19e'),(4637,8573,'#2310cd'),(4638,8574,'#FFFFFF'),(4639,8575,'#FFFFFF'),(4640,8578,'#f445fc'),(4641,8579,'#FFFFFF'),(4642,8580,'#8430cf'),(4643,8583,'#FFFFFF'),(4644,8584,'#FFFFFF'),(4645,8585,'#e4be40'),(4646,8588,'#FFFFFF'),(4647,8591,'#FFFFFF'),(4648,8592,'#FFFFFF'),(4649,8593,'#FFFFFF'),(4650,8595,'#FFFFFF'),(4651,8596,'#FFFFFF'),(4652,8597,'#3ecfb9'),(4653,8598,'#FFFFFF'),(4654,8599,'#FFFFFF'),(4655,8600,'#FFFFFF'),(4656,8602,'#FFFFFF'),(4657,8603,'#FFFFFF'),(4658,8608,'#28f678'),(4659,8609,'#84f508'),(4660,8611,'#FFFFFF'),(4661,8612,'#68ea42'),(4662,8615,'#FFFFFF'),(4663,8618,'#6392ce'),(4664,8620,'#FFFFFF'),(4665,8621,'#dd2929'),(4666,8623,'#FFFFFF'),(4667,8625,'#6221df'),(4668,8629,'#FFFFFF'),(4669,8630,'#FFFFFF'),(4670,8631,'#FFFFFF'),(4671,8638,'#FFFFFF'),(4672,8639,'#FFFFFF'),(4673,8640,'#FFFFFF'),(4674,8645,'#FFFFFF'),(4675,8647,'#7bddfd'),(4676,8649,'#34c9ce'),(4677,8650,'#FFFFFF'),(4678,8652,'#aeed01'),(4679,8654,'#cc0987'),(4680,8656,'#0ad258'),(4681,8657,'#27f1a2'),(4682,8659,'#fdcb07'),(4683,8664,'#cf3631'),(4684,8668,'#57e581'),(4685,8672,'#FFFFFF'),(4686,8674,'#FFFFFF'),(4687,8676,'#FFFFFF'),(4688,8682,'#FFFFFF'),(4689,8684,'#FFFFFF'),(4690,8685,'#FFFFFF'),(4691,8687,'#FFFFFF'),(4692,8690,'#FFFFFF'),(4693,8691,'#1bdb4c'),(4694,8696,'#FFFFFF'),(4695,8697,'#FFFFFF'),(4696,8699,'#6748dd'),(4697,8703,'#FFFFFF'),(4698,8704,'#FFFFFF'),(4699,8705,'#38d5c4'),(4700,8707,'#FFFFFF'),(4701,8710,'#FFFFFF'),(4702,8711,'#FFFFFF'),(4703,8712,'#FFFFFF'),(4704,8713,'#cb6df3'),(4705,8716,'#FFFFFF'),(4706,8717,'#FFFFFF'),(4707,8718,'#FFFFFF'),(4708,8725,'#e94763'),(4709,8729,'#FFFFFF'),(4710,8730,'#FFFFFF'),(4711,8734,'#594add'),(4712,8737,'#FFFFFF'),(4713,8738,'#FFFFFF'),(4714,8739,'#01f04b'),(4715,8741,'#FFFFFF'),(4716,8742,'#FFFFFF'),(4717,8743,'#FFFFFF'),(4718,8745,'#4647f8'),(4719,8746,'#FFFFFF'),(4720,8747,'#FFFFFF'),(4721,8749,'#45ce4e'),(4722,8752,'#FFFFFF'),(4723,8756,'#FFFFFF'),(4724,8762,'#dd01be'),(4725,8764,'#FFFFFF'),(4726,8765,'#f83988'),(4727,8770,'#FFFFFF'),(4728,8771,'#06ddab'),(4729,8772,'#cb27dd'),(4730,8775,'#FFFFFF'),(4731,8777,'#a7cd1e'),(4732,8778,'#FFFFFF'),(4733,8785,'#5448d5'),(4734,8786,'#ce6aea'),(4735,8787,'#FFFFFF'),(4736,8789,'#FFFFFF'),(4737,8790,'#64f50d'),(4738,8796,'#FFFFFF'),(4739,8797,'#FFFFFF'),(4740,8798,'#FFFFFF'),(4741,8800,'#FFFFFF'),(4742,8801,'#FFFFFF'),(4743,8804,'#FFFFFF'),(4744,8806,'#5c95d0'),(4745,8808,'#FFFFFF'),(4746,8811,'#FFFFFF'),(4747,8813,'#40f702'),(4748,8815,'#FFFFFF'),(4749,8816,'#FFFFFF'),(4750,8817,'#d18a33'),(4751,8818,'#e74953'),(4752,8819,'#FFFFFF'),(4753,8823,'#FFFFFF'),(4754,8824,'#8106df'),(4755,8825,'#6ef0b5'),(4756,8830,'#FFFFFF'),(4757,8832,'#5bf193'),(4758,8833,'#66fb65'),(4759,8837,'#FFFFFF'),(4760,8839,'#e6f61d'),(4761,8848,'#FFFFFF'),(4762,8852,'#FFFFFF'),(4763,8862,'#19e67b'),(4764,8865,'#1761e4'),(4765,8868,'#e0c345'),(4766,8870,'#47d17f'),(4767,8872,'#f84f33'),(4768,8875,'#c205d4'),(4769,8876,'#4a6cee'),(4770,8877,'#FFFFFF'),(4771,8878,'#678cf6'),(4772,8884,'#6e1fd5'),(4773,8885,'#f8895b'),(4774,8888,'#FFFFFF'),(4775,8889,'#7d0aed'),(4776,8890,'#FFFFFF'),(4777,8892,'#FFFFFF'),(4778,8893,'#FFFFFF'),(4779,8897,'#FFFFFF'),(4780,8898,'#23aee6'),(4781,8902,'#FFFFFF'),(4782,8904,'#FFFFFF'),(4783,8905,'#e31730'),(4784,8906,'#FFFFFF'),(4785,8909,'#FFFFFF'),(4786,8910,'#FFFFFF'),(4787,8916,'#89df42'),(4788,8917,'#FFFFFF'),(4789,8918,'#caf365'),(4790,8919,'#FFFFFF'),(4791,8921,'#FFFFFF'),(4792,8924,'#FFFFFF'),(4793,8926,'#FFFFFF'),(4794,8927,'#f252df'),(4795,8928,'#FFFFFF'),(4796,8930,'#FFFFFF'),(4797,8936,'#9606eb'),(4798,8937,'#FFFFFF'),(4799,8943,'#f1d637'),(4800,8945,'#FFFFFF'),(4801,8946,'#18fe97'),(4802,8948,'#a665f5'),(4803,8949,'#FFFFFF'),(4804,8950,'#69e9da'),(4805,8953,'#6431e9'),(4806,8955,'#FFFFFF'),(4807,8956,'#166ace'),(4808,8961,'#68fb8a'),(4809,8962,'#FFFFFF'),(4810,8963,'#e9247a'),(4811,8965,'#d2b703'),(4812,8969,'#5b89f0'),(4813,8973,'#FFFFFF'),(4814,8974,'#FFFFFF'),(4815,8976,'#8de12d'),(4816,8978,'#FFFFFF'),(4817,8979,'#FFFFFF'),(4818,8980,'#06e665'),(4819,8981,'#FFFFFF'),(4820,8982,'#567dfe'),(4821,8985,'#ed53b1'),(4822,8990,'#ea8141'),(4823,8993,'#d9c035'),(4824,8998,'#FFFFFF'),(4825,9000,'#FFFFFF'),(4826,9002,'#4ce4ed'),(4827,9003,'#FFFFFF'),(4828,9009,'#FFFFFF'),(4829,9013,'#FFFFFF'),(4830,9014,'#73e32c'),(4831,9017,'#FFFFFF'),(4832,9018,'#1fd03b'),(4833,9019,'#e18331'),(4834,9020,'#FFFFFF'),(4835,9022,'#e462c9'),(4836,9023,'#731cee'),(4837,9024,'#bbcd0e'),(4838,9025,'#0ee04f'),(4839,9026,'#40b3d8'),(4840,9029,'#FFFFFF'),(4841,9031,'#FFFFFF'),(4842,9033,'#FFFFFF'),(4843,9035,'#FFFFFF'),(4844,9037,'#FFFFFF'),(4845,9038,'#0e2bf7'),(4846,9039,'#FFFFFF'),(4847,9046,'#e6bb12'),(4848,9047,'#254fd5'),(4849,9049,'#FFFFFF'),(4850,9050,'#FFFFFF'),(4851,9051,'#FFFFFF'),(4852,9053,'#FFFFFF'),(4853,9058,'#FFFFFF'),(4854,9059,'#FFFFFF'),(4855,9060,'#d2973e'),(4856,9062,'#FFFFFF'),(4857,9063,'#FFFFFF'),(4858,9064,'#f77464'),(4859,9065,'#FFFFFF'),(4860,9066,'#d4cb47'),(4861,9068,'#FFFFFF'),(4862,9070,'#FFFFFF'),(4863,9071,'#FFFFFF'),(4864,9077,'#ea16a5'),(4865,9079,'#FFFFFF'),(4866,9081,'#FFFFFF'),(4867,9082,'#FFFFFF'),(4868,9083,'#FFFFFF'),(4869,9084,'#FFFFFF'),(4870,9092,'#FFFFFF'),(4871,9098,'#FFFFFF'),(4872,9102,'#34a2fa'),(4873,9104,'#FFFFFF'),(4874,9107,'#31d81e'),(4875,9108,'#6f9ffa'),(4876,9109,'#FFFFFF'),(4877,9111,'#FFFFFF'),(4878,9114,'#61dc11'),(4879,9116,'#4bed99'),(4880,9117,'#FFFFFF'),(4881,9122,'#b614da'),(4882,9125,'#33fd5c'),(4883,9128,'#FFFFFF'),(4884,9130,'#17ef98'),(4885,9135,'#FFFFFF'),(4886,9136,'#FFFFFF'),(4887,9138,'#4bcc16'),(4888,9139,'#d831b3'),(4889,9141,'#FFFFFF'),(4890,9142,'#FFFFFF'),(4891,9143,'#FFFFFF'),(4892,9145,'#FFFFFF'),(4893,9146,'#FFFFFF'),(4894,9148,'#FFFFFF'),(4895,9153,'#FFFFFF'),(4896,9154,'#FFFFFF'),(4897,9157,'#FFFFFF'),(4898,9158,'#FFFFFF'),(4899,9159,'#FFFFFF'),(4900,9160,'#FFFFFF'),(4901,9161,'#FFFFFF'),(4902,9163,'#FFFFFF'),(4903,9165,'#d8f21a'),(4904,9166,'#FFFFFF'),(4905,9167,'#FFFFFF'),(4906,9168,'#FFFFFF'),(4907,9170,'#03b9e8'),(4908,9172,'#FFFFFF'),(4909,9174,'#9aed5b'),(4910,9175,'#62e42e'),(4911,9182,'#FFFFFF'),(4912,9183,'#FFFFFF'),(4913,9185,'#FFFFFF'),(4914,9186,'#e7f331'),(4915,9187,'#FFFFFF'),(4916,9190,'#FFFFFF'),(4917,9192,'#fd083a'),(4918,9194,'#FFFFFF'),(4919,9195,'#FFFFFF'),(4920,9197,'#FFFFFF'),(4921,9200,'#bee348'),(4922,9202,'#92cc33'),(4923,9205,'#FFFFFF'),(4924,9206,'#d1aa61'),(4925,9208,'#0e0cde'),(4926,9211,'#FFFFFF'),(4927,9213,'#FFFFFF'),(4928,9217,'#FFFFFF'),(4929,9219,'#1d9bde'),(4930,9223,'#FFFFFF'),(4931,9226,'#FFFFFF'),(4932,9234,'#e54eca'),(4933,9240,'#FFFFFF'),(4934,9246,'#fe4936'),(4935,9247,'#FFFFFF'),(4936,9250,'#FFFFFF'),(4937,9259,'#FFFFFF'),(4938,9262,'#FFFFFF'),(4939,9264,'#FFFFFF'),(4940,9265,'#FFFFFF'),(4941,9267,'#219eed'),(4942,9277,'#dc46ae'),(4943,9278,'#FFFFFF'),(4944,9279,'#FFFFFF'),(4945,9280,'#FFFFFF'),(4946,9281,'#60d0d8'),(4947,9284,'#013bd1'),(4948,9286,'#43f47d'),(4949,9294,'#FFFFFF'),(4950,9298,'#FFFFFF'),(4951,9299,'#FFFFFF'),(4952,9300,'#11dcac'),(4953,9302,'#FFFFFF'),(4954,9303,'#FFFFFF'),(4955,9304,'#FFFFFF'),(4956,9307,'#FFFFFF'),(4957,9311,'#FFFFFF'),(4958,9315,'#39abd3'),(4959,9316,'#FFFFFF'),(4960,9322,'#52da24'),(4961,9323,'#3ff565'),(4962,9325,'#dcf426'),(4963,9326,'#FFFFFF'),(4964,9327,'#FFFFFF'),(4965,9333,'#FFFFFF'),(4966,9347,'#03e33a'),(4967,9348,'#dd333a'),(4968,9350,'#FFFFFF'),(4969,9351,'#89f93a'),(4970,9353,'#FFFFFF'),(4971,9357,'#925ed4'),(4972,9359,'#FFFFFF'),(4973,9361,'#e2c24d'),(4974,9364,'#52de5f'),(4975,9365,'#FFFFFF'),(4976,9366,'#FFFFFF'),(4977,9370,'#e5ba33'),(4978,9372,'#d9c139'),(4979,9374,'#5cd68e'),(4980,9375,'#FFFFFF'),(4981,9379,'#FFFFFF'),(4982,9381,'#28ce96'),(4983,9391,'#FFFFFF'),(4984,9393,'#3575f3'),(4985,9398,'#e270ed'),(4986,9401,'#FFFFFF'),(4987,9402,'#FFFFFF'),(4988,9403,'#a6da4e'),(4989,9404,'#FFFFFF'),(4990,9405,'#FFFFFF'),(4991,9407,'#3e03f1'),(4992,9408,'#2cef76'),(4993,9409,'#FFFFFF'),(4994,9411,'#FFFFFF'),(4995,9412,'#d87a0d'),(4996,9414,'#FFFFFF'),(4997,9417,'#f4cf1a'),(4998,9422,'#e50731'),(4999,9425,'#FFFFFF'),(5000,9427,'#FFFFFF'),(5001,9428,'#FFFFFF'),(5002,9429,'#FFFFFF'),(5003,9431,'#dd9c06'),(5004,9432,'#e98718'),(5005,9434,'#5656e3'),(5006,9436,'#FFFFFF'),(5007,9438,'#FFFFFF'),(5008,9439,'#FFFFFF'),(5009,9440,'#FFFFFF'),(5010,9444,'#7028d0'),(5011,9448,'#FFFFFF'),(5012,9452,'#cd7e38'),(5013,9457,'#FFFFFF'),(5014,9459,'#FFFFFF'),(5015,9461,'#FFFFFF'),(5016,9462,'#FFFFFF'),(5017,9463,'#5341e6'),(5018,9465,'#FFFFFF'),(5019,9466,'#FFFFFF'),(5020,9467,'#d48343'),(5021,9470,'#FFFFFF'),(5022,9473,'#e6025d'),(5023,9475,'#74e10c'),(5024,9479,'#FFFFFF'),(5025,9480,'#FFFFFF'),(5026,9481,'#FFFFFF'),(5027,9483,'#50adda'),(5028,9485,'#d8b404'),(5029,9488,'#FFFFFF'),(5030,9489,'#FFFFFF'),(5031,9491,'#FFFFFF'),(5032,9494,'#ba59f3'),(5033,9495,'#5c82e1'),(5034,9497,'#FFFFFF'),(5035,9498,'#FFFFFF'),(5036,9499,'#FFFFFF'),(5037,9500,'#cfa150'),(5038,9501,'#6d9cde'),(5039,9504,'#2a96d2'),(5040,9505,'#ec4402'),(5041,9507,'#FFFFFF'),(5042,9508,'#FFFFFF'),(5043,9509,'#d26355'),(5044,9512,'#d9d939'),(5045,9513,'#FFFFFF'),(5046,9514,'#23cd7e'),(5047,9517,'#FFFFFF'),(5048,9518,'#FFFFFF'),(5049,9519,'#ec4b16'),(5050,9521,'#FFFFFF'),(5051,9522,'#FFFFFF'),(5052,9526,'#FFFFFF'),(5053,9530,'#FFFFFF'),(5054,9533,'#FFFFFF'),(5055,9536,'#FFFFFF'),(5056,9538,'#e8df57'),(5057,9539,'#FFFFFF'),(5058,9541,'#FFFFFF'),(5059,9542,'#FFFFFF'),(5060,9543,'#17d5a3'),(5061,9546,'#FFFFFF'),(5062,9549,'#5c69d4'),(5063,9550,'#FFFFFF'),(5064,9551,'#764df4'),(5065,9553,'#FFFFFF'),(5066,9555,'#246efc'),(5067,9558,'#FFFFFF'),(5068,9560,'#FFFFFF'),(5069,9561,'#5bd2ef'),(5070,9563,'#FFFFFF'),(5071,9564,'#33e090'),(5072,9565,'#11f2a3'),(5073,9566,'#1892df'),(5074,9568,'#FFFFFF'),(5075,9575,'#FFFFFF'),(5076,9578,'#FFFFFF'),(5077,9579,'#FFFFFF'),(5078,9584,'#dcd25c'),(5079,9585,'#f5405b'),(5080,9587,'#e06aad'),(5081,9589,'#FFFFFF'),(5082,9591,'#FFFFFF'),(5083,9592,'#cad155'),(5084,9595,'#FFFFFF'),(5085,9599,'#FFFFFF'),(5086,9600,'#b7e455'),(5087,9601,'#FFFFFF'),(5088,9602,'#FFFFFF'),(5089,9604,'#FFFFFF'),(5090,9605,'#FFFFFF'),(5091,9606,'#FFFFFF'),(5092,9607,'#FFFFFF'),(5093,9609,'#FFFFFF'),(5094,9610,'#FFFFFF'),(5095,9611,'#FFFFFF'),(5096,9612,'#b22de3'),(5097,9615,'#e0b34d'),(5098,9618,'#d55c09'),(5099,9622,'#FFFFFF'),(5100,9623,'#6c3fef'),(5101,9624,'#cf2240'),(5102,9626,'#5ecc90'),(5103,9628,'#FFFFFF'),(5104,9631,'#0620fc'),(5105,9633,'#FFFFFF'),(5106,9634,'#FFFFFF'),(5107,9635,'#FFFFFF'),(5108,9636,'#FFFFFF'),(5109,9639,'#FFFFFF'),(5110,9640,'#f431be'),(5111,9641,'#33e160'),(5112,9642,'#6419e8'),(5113,9644,'#FFFFFF'),(5114,9645,'#8f53db'),(5115,9646,'#FFFFFF'),(5116,9647,'#e327bd'),(5117,9648,'#03d7a9'),(5118,9653,'#e369c5'),(5119,9655,'#FFFFFF'),(5120,9656,'#FFFFFF'),(5121,9658,'#FFFFFF'),(5122,9659,'#FFFFFF'),(5123,9660,'#FFFFFF'),(5124,9662,'#FFFFFF'),(5125,9666,'#FFFFFF'),(5126,9668,'#da44b3'),(5127,9669,'#FFFFFF'),(5128,9673,'#FFFFFF'),(5129,9675,'#FFFFFF'),(5130,9678,'#3ff42d'),(5131,9683,'#d461b9'),(5132,9689,'#dfa75b'),(5133,9692,'#4b4fcd'),(5134,9696,'#e3533d'),(5135,9697,'#FFFFFF'),(5136,9699,'#FFFFFF'),(5137,9700,'#FFFFFF'),(5138,9702,'#FFFFFF'),(5139,9703,'#FFFFFF'),(5140,9705,'#FFFFFF'),(5141,9707,'#FFFFFF'),(5142,9711,'#58f11e'),(5143,9713,'#FFFFFF'),(5144,9717,'#2bf743'),(5145,9718,'#FFFFFF'),(5146,9723,'#FFFFFF'),(5147,9724,'#FFFFFF'),(5148,9726,'#FFFFFF'),(5149,9727,'#4f7af5'),(5150,9729,'#FFFFFF'),(5151,9730,'#f12e13'),(5152,9731,'#FFFFFF'),(5153,9732,'#ab19f7'),(5154,9733,'#369fd4'),(5155,9734,'#ee3103'),(5156,9736,'#FFFFFF'),(5157,9739,'#FFFFFF'),(5158,9746,'#dd5de7'),(5159,9748,'#FFFFFF'),(5160,9750,'#FFFFFF'),(5161,9751,'#3cd80b'),(5162,9752,'#FFFFFF'),(5163,9753,'#FFFFFF'),(5164,9754,'#9e42d4'),(5165,9755,'#FFFFFF'),(5166,9758,'#ea5284'),(5167,9762,'#80fc43'),(5168,9763,'#FFFFFF'),(5169,9764,'#61db7e'),(5170,9765,'#4316ce'),(5171,9768,'#7bce56'),(5172,9769,'#f02f92'),(5173,9771,'#4ebed5'),(5174,9772,'#cede5a'),(5175,9773,'#FFFFFF'),(5176,9774,'#00e412'),(5177,9775,'#ac2ed4'),(5178,9776,'#FFFFFF'),(5179,9777,'#ce3e93'),(5180,9779,'#FFFFFF'),(5181,9783,'#FFFFFF'),(5182,9784,'#FFFFFF'),(5183,9785,'#b3ce39'),(5184,9786,'#f0ea70'),(5185,9787,'#FFFFFF'),(5186,9790,'#FFFFFF'),(5187,9793,'#e15dc4'),(5188,9794,'#0226fe'),(5189,9795,'#FFFFFF'),(5190,9796,'#FFFFFF'),(5191,9798,'#9c28f3'),(5192,9800,'#f841f4'),(5193,9801,'#eb4d4f'),(5194,9803,'#FFFFFF'),(5195,9805,'#FFFFFF'),(5196,9808,'#FFFFFF'),(5197,9811,'#03d77a'),(5198,9813,'#FFFFFF'),(5199,9815,'#FFFFFF'),(5200,9817,'#FFFFFF'),(5201,9818,'#FFFFFF'),(5202,9821,'#f5e473'),(5203,9823,'#ee1599'),(5204,9825,'#FFFFFF'),(5205,9827,'#d95111'),(5206,9836,'#FFFFFF'),(5207,9837,'#FFFFFF'),(5208,9838,'#FFFFFF'),(5209,9840,'#FFFFFF'),(5210,9843,'#df5961'),(5211,9847,'#8efd78'),(5212,9848,'#c8d716'),(5213,9852,'#FFFFFF'),(5214,9854,'#FFFFFF'),(5215,9856,'#57f543'),(5216,9858,'#FFFFFF'),(5217,9860,'#FFFFFF'),(5218,9862,'#FFFFFF'),(5219,9863,'#d8b426'),(5220,9865,'#e170f9'),(5221,9868,'#7210d7'),(5222,9869,'#6cdd55'),(5223,9870,'#4171f1'),(5224,9872,'#FFFFFF'),(5225,9874,'#235cde'),(5226,9875,'#FFFFFF'),(5227,9876,'#FFFFFF'),(5228,9878,'#45cbf4'),(5229,9881,'#FFFFFF'),(5230,9882,'#FFFFFF'),(5231,9883,'#d32643'),(5232,9885,'#e5e820'),(5233,9887,'#FFFFFF'),(5234,9888,'#7bdd31'),(5235,9889,'#095ef5'),(5236,9892,'#FFFFFF'),(5237,9895,'#FFFFFF'),(5238,9896,'#75b0fa'),(5239,9897,'#277ceb'),(5240,9898,'#d17958'),(5241,9901,'#FFFFFF'),(5242,9902,'#FFFFFF'),(5243,9903,'#2d5efa'),(5244,9910,'#FFFFFF'),(5245,9913,'#3ee51d'),(5246,9914,'#e08961'),(5247,9916,'#b3e607'),(5248,9918,'#FFFFFF'),(5249,9919,'#FFFFFF'),(5250,9920,'#FFFFFF'),(5251,9921,'#2bf969'),(5252,9922,'#22e488'),(5253,9928,'#FFFFFF'),(5254,9929,'#FFFFFF'),(5255,9930,'#6f4dd2'),(5256,9932,'#FFFFFF'),(5257,9933,'#FFFFFF'),(5258,9941,'#3c8be4'),(5259,9944,'#d51605'),(5260,9947,'#FFFFFF'),(5261,9950,'#FFFFFF'),(5262,9951,'#FFFFFF'),(5263,9953,'#FFFFFF'),(5264,9958,'#FFFFFF'),(5265,9962,'#ea3989'),(5266,9963,'#e9b94b'),(5267,9968,'#FFFFFF'),(5268,9970,'#FFFFFF'),(5269,9971,'#FFFFFF'),(5270,9974,'#FFFFFF'),(5271,9976,'#FFFFFF'),(5272,9977,'#FFFFFF'),(5273,9978,'#e38b37'),(5274,9979,'#FFFFFF'),(5275,9980,'#FFFFFF'),(5276,9981,'#FFFFFF'),(5277,9983,'#FFFFFF'),(5278,9985,'#FFFFFF'),(5279,9987,'#e06578'),(5280,9988,'#FFFFFF'),(5281,9989,'#FFFFFF'),(5282,9991,'#FFFFFF'),(5283,9992,'#d60877'),(5284,9993,'#FFFFFF'),(5285,9994,'#FFFFFF'),(5286,9996,'#6ff3ee'),(5287,9997,'#FFFFFF'),(5288,10000,'#FFFFFF'),(5289,10002,'#81d642'),(5290,10005,'#1bcd3a'),(5291,10006,'#FFFFFF'),(5292,10008,'#ddf517'),(5293,10009,'#d6d950'),(5294,10010,'#6d0ce4'),(5295,10013,'#FFFFFF'),(5296,10017,'#FFFFFF'),(5297,10019,'#FFFFFF'),(5298,10022,'#d002b2'),(5299,10024,'#FFFFFF'),(5300,10027,'#FFFFFF'),(5301,10031,'#e20bed'),(5302,10033,'#8767f4'),(5303,10035,'#FFFFFF'),(5304,10037,'#d30e0c'),(5305,10038,'#FFFFFF'),(5306,10039,'#1738eb'),(5307,10040,'#FFFFFF'),(5308,10045,'#6dd14c'),(5309,10047,'#FFFFFF'),(5310,10048,'#FFFFFF'),(5311,10049,'#FFFFFF'),(5312,10050,'#FFFFFF'),(5313,10053,'#FFFFFF'),(5314,10055,'#FFFFFF'),(5315,10061,'#db4dd7'),(5316,10062,'#FFFFFF'),(5317,10065,'#40eb15'),(5318,10066,'#d4b107'),(5319,10067,'#cf5b6e'),(5320,10069,'#e7575f'),(5321,10070,'#FFFFFF'),(5322,10071,'#9664e2'),(5323,10073,'#90dd63'),(5324,10074,'#6961f1'),(5325,10077,'#32f581'),(5326,10078,'#FFFFFF'),(5327,10079,'#f8c348'),(5328,10080,'#b2ed00'),(5329,10085,'#ed74be'),(5330,10086,'#FFFFFF'),(5331,10089,'#e95286'),(5332,10095,'#8376f3'),(5333,10097,'#FFFFFF'),(5334,10098,'#c12ee9'),(5335,10100,'#FFFFFF'),(5336,10102,'#13d542'),(5337,10104,'#FFFFFF'),(5338,10105,'#d22b6b'),(5339,10106,'#FFFFFF'),(5340,10107,'#FFFFFF'),(5341,10112,'#a458d1'),(5342,10115,'#8a48f4'),(5343,10116,'#FFFFFF'),(5344,10117,'#6ddc2c'),(5345,10118,'#4cd344'),(5346,10119,'#FFFFFF'),(5347,10123,'#FFFFFF'),(5348,10124,'#FFFFFF'),(5349,10126,'#FFFFFF'),(5350,10128,'#FFFFFF'),(5351,10129,'#dd3f45'),(5352,10130,'#6d2cde'),(5353,10139,'#FFFFFF'),(5354,10140,'#9f26e5'),(5355,10141,'#18db7e'),(5356,10143,'#FFFFFF'),(5357,10145,'#f90cd8'),(5358,10148,'#FFFFFF'),(5359,10149,'#68d9b0'),(5360,10151,'#FFFFFF'),(5361,10153,'#FFFFFF'),(5362,10156,'#5ffefa'),(5363,10157,'#FFFFFF'),(5364,10158,'#d65e97'),(5365,10159,'#9deb5a'),(5366,10161,'#FFFFFF'),(5367,10165,'#4a51d0'),(5368,10167,'#eed30a'),(5369,10173,'#3ced82'),(5370,10175,'#FFFFFF'),(5371,10176,'#FFFFFF'),(5372,10177,'#FFFFFF'),(5373,10178,'#dfc943'),(5374,10180,'#72feec'),(5375,10182,'#e804dc'),(5376,10183,'#FFFFFF'),(5377,10185,'#fc6996'),(5378,10195,'#364fd9'),(5379,10196,'#FFFFFF'),(5380,10200,'#9864d0'),(5381,10203,'#31f11d'),(5382,10207,'#FFFFFF'),(5383,10211,'#a0e030'),(5384,10212,'#FFFFFF'),(5385,10213,'#f82365'),(5386,10214,'#d8bf31'),(5387,10216,'#FFFFFF'),(5388,10217,'#096ef0'),(5389,10218,'#FFFFFF'),(5390,10219,'#870bf2'),(5391,10221,'#FFFFFF'),(5392,10223,'#FFFFFF'),(5393,10225,'#d06649'),(5394,10226,'#FFFFFF'),(5395,10227,'#FFFFFF'),(5396,10229,'#35e381'),(5397,10230,'#d3c14e'),(5398,10232,'#FFFFFF'),(5399,10233,'#FFFFFF'),(5400,10234,'#FFFFFF'),(5401,10235,'#56dffc'),(5402,10236,'#8f65db'),(5403,10238,'#FFFFFF'),(5404,10240,'#32cd08'),(5405,10241,'#0742d9'),(5406,10245,'#FFFFFF'),(5407,10246,'#FFFFFF'),(5408,10248,'#FFFFFF'),(5409,10249,'#FFFFFF'),(5410,10253,'#da13d8'),(5411,10254,'#FFFFFF'),(5412,10255,'#f360ec'),(5413,10256,'#eb5ad5'),(5414,10257,'#FFFFFF'),(5415,10258,'#5ea8f1'),(5416,10259,'#c5ce34'),(5417,10260,'#6f9df4'),(5418,10261,'#FFFFFF'),(5419,10262,'#2b9dd2'),(5420,10263,'#FFFFFF'),(5421,10264,'#FFFFFF'),(5422,10265,'#FFFFFF'),(5423,10268,'#cde150'),(5424,10270,'#6be365'),(5425,10273,'#7019f0'),(5426,10275,'#525fce'),(5427,10276,'#FFFFFF'),(5428,10277,'#FFFFFF'),(5429,10279,'#e63749'),(5430,10281,'#FFFFFF'),(5431,10283,'#7924fb'),(5432,10286,'#1cb6cf'),(5433,10287,'#FFFFFF'),(5434,10288,'#FFFFFF'),(5435,10289,'#baeb74'),(5436,10290,'#a8ef4c'),(5437,10295,'#FFFFFF'),(5438,10296,'#43eb16'),(5439,10300,'#e3033d'),(5440,10301,'#0d01d9'),(5441,10304,'#ec5d86'),(5442,10305,'#8c1edd'),(5443,10306,'#FFFFFF'),(5444,10307,'#142dfc'),(5445,10308,'#e859d8'),(5446,10310,'#81f072'),(5447,10311,'#FFFFFF'),(5448,10312,'#edfb18'),(5449,10313,'#19de2b'),(5450,10314,'#aefc24'),(5451,10317,'#e511f1'),(5452,10318,'#FFFFFF'),(5453,10321,'#ec44c6'),(5454,10322,'#5d9bcf'),(5455,10325,'#2762e9'),(5456,10326,'#FFFFFF'),(5457,10327,'#FFFFFF'),(5458,10330,'#FFFFFF'),(5459,10332,'#FFFFFF'),(5460,10336,'#ceb92e'),(5461,10341,'#8fe259'),(5462,10344,'#FFFFFF'),(5463,10346,'#FFFFFF'),(5464,10347,'#FFFFFF'),(5465,10348,'#4567d7'),(5466,10352,'#FFFFFF'),(5467,10355,'#FFFFFF'),(5468,10358,'#FFFFFF'),(5469,10359,'#8d3ae3'),(5470,10360,'#FFFFFF'),(5471,10363,'#f26271'),(5472,10364,'#a2e354'),(5473,10368,'#FFFFFF'),(5474,10372,'#bdf71d'),(5475,10373,'#50d4b9'),(5476,10374,'#d348cb'),(5477,10376,'#FFFFFF'),(5478,10378,'#9adb3f'),(5479,10379,'#c60dd3'),(5480,10380,'#FFFFFF'),(5481,10384,'#64b6d5'),(5482,10385,'#038dd2'),(5483,10386,'#FFFFFF'),(5484,10387,'#9b49dc'),(5485,10390,'#FFFFFF'),(5486,10391,'#ef502b'),(5487,10392,'#FFFFFF'),(5488,10398,'#2397e7'),(5489,10399,'#FFFFFF'),(5490,10400,'#FFFFFF'),(5491,10402,'#afce5f'),(5492,10407,'#FFFFFF'),(5493,10408,'#FFFFFF'),(5494,10410,'#e42a30'),(5495,10411,'#5ed1bd'),(5496,10412,'#FFFFFF'),(5497,10414,'#4df63a'),(5498,10417,'#2b61f8'),(5499,10418,'#4b6df0'),(5500,10419,'#de942b'),(5501,10423,'#b357d5'),(5502,10426,'#FFFFFF'),(5503,10427,'#FFFFFF'),(5504,10428,'#2ecf66'),(5505,10430,'#FFFFFF'),(5506,10431,'#6db4ec'),(5507,10433,'#15f6a4'),(5508,10435,'#67a5f1'),(5509,10436,'#FFFFFF'),(5510,10438,'#FFFFFF'),(5511,10439,'#e35e7f'),(5512,10440,'#0ce94e'),(5513,10441,'#FFFFFF'),(5514,10444,'#cd4439'),(5515,10446,'#FFFFFF'),(5516,10449,'#FFFFFF'),(5517,10450,'#FFFFFF'),(5518,10451,'#c249e4'),(5519,10452,'#FFFFFF'),(5520,10454,'#d4223c'),(5521,10455,'#FFFFFF'),(5522,10456,'#FFFFFF'),(5523,10460,'#26f96d'),(5524,10461,'#e53799'),(5525,10463,'#FFFFFF'),(5526,10465,'#1bdcfc'),(5527,10466,'#FFFFFF'),(5528,10468,'#FFFFFF'),(5529,10470,'#FFFFFF'),(5530,10476,'#d7f36a'),(5531,10477,'#61e919'),(5532,10478,'#FFFFFF'),(5533,10483,'#FFFFFF'),(5534,10485,'#FFFFFF'),(5535,10487,'#67f6e7'),(5536,10488,'#FFFFFF'),(5537,10490,'#51cbd9'),(5538,10491,'#FFFFFF'),(5539,10493,'#FFFFFF'),(5540,10496,'#f541cc'),(5541,10497,'#2cee73'),(5542,10498,'#2c8cf5'),(5543,10505,'#FFFFFF'),(5544,10506,'#FFFFFF'),(5545,10507,'#FFFFFF'),(5546,10508,'#FFFFFF'),(5547,10511,'#FFFFFF'),(5548,10512,'#1c1fe5'),(5549,10514,'#FFFFFF'),(5550,10517,'#FFFFFF'),(5551,10518,'#FFFFFF'),(5552,10523,'#FFFFFF'),(5553,10525,'#a262e9'),(5554,10526,'#53e96c'),(5555,10527,'#FFFFFF'),(5556,10529,'#FFFFFF'),(5557,10531,'#66b7f5'),(5558,10532,'#FFFFFF'),(5559,10535,'#FFFFFF'),(5560,10538,'#35ef29'),(5561,10539,'#90cd0a'),(5562,10541,'#FFFFFF'),(5563,10544,'#FFFFFF'),(5564,10548,'#FFFFFF'),(5565,10550,'#FFFFFF'),(5566,10551,'#ce392c'),(5567,10553,'#e09a56'),(5568,10554,'#f53e2c'),(5569,10555,'#1dd5f6'),(5570,10556,'#3ddb05'),(5571,10557,'#FFFFFF'),(5572,10558,'#FFFFFF'),(5573,10560,'#FFFFFF'),(5574,10561,'#FFFFFF'),(5575,10562,'#d18759'),(5576,10563,'#FFFFFF'),(5577,10565,'#4ea0f5'),(5578,10567,'#FFFFFF'),(5579,10568,'#FFFFFF'),(5580,10569,'#d54c9c'),(5581,10572,'#8c0eea'),(5582,10573,'#FFFFFF'),(5583,10574,'#FFFFFF'),(5584,10575,'#FFFFFF'),(5585,10576,'#7582f2'),(5586,10578,'#FFFFFF'),(5587,10579,'#e53ea5'),(5588,10581,'#03ddb1'),(5589,10582,'#d65786'),(5590,10583,'#0dea60'),(5591,10586,'#e2e90a'),(5592,10587,'#dea030'),(5593,10593,'#703de7'),(5594,10596,'#FFFFFF'),(5595,10598,'#FFFFFF'),(5596,10600,'#abfa23'),(5597,10601,'#FFFFFF'),(5598,10603,'#e6a50a'),(5599,10604,'#4aabe3'),(5600,10606,'#5930f2'),(5601,10610,'#FFFFFF'),(5602,10613,'#FFFFFF'),(5603,10614,'#94f53c'),(5604,10615,'#FFFFFF'),(5605,10616,'#FFFFFF'),(5606,10619,'#FFFFFF'),(5607,10624,'#FFFFFF'),(5608,10627,'#FFFFFF'),(5609,10629,'#FFFFFF'),(5610,10631,'#be43de'),(5611,10633,'#FFFFFF'),(5612,10634,'#FFFFFF'),(5613,10635,'#0385ea'),(5614,10636,'#40ede1'),(5615,10638,'#b94ce1'),(5616,10640,'#5de240'),(5617,10642,'#FFFFFF'),(5618,10643,'#0b81d8'),(5619,10645,'#05db0e'),(5620,10646,'#FFFFFF'),(5621,10647,'#5bf8cc'),(5622,10648,'#FFFFFF'),(5623,10650,'#FFFFFF'),(5624,10651,'#FFFFFF'),(5625,10652,'#FFFFFF'),(5626,10655,'#FFFFFF'),(5627,10658,'#FFFFFF'),(5628,10659,'#FFFFFF'),(5629,10661,'#FFFFFF'),(5630,10662,'#6858d0'),(5631,10665,'#FFFFFF'),(5632,10667,'#FFFFFF'),(5633,10668,'#FFFFFF'),(5634,10673,'#FFFFFF'),(5635,10674,'#18d379'),(5636,10677,'#FFFFFF'),(5637,10678,'#FFFFFF'),(5638,10679,'#0f29e3'),(5639,10680,'#FFFFFF'),(5640,10683,'#FFFFFF'),(5641,10686,'#7aa3fe'),(5642,10688,'#b423e3'),(5643,10690,'#FFFFFF'),(5644,10692,'#FFFFFF'),(5645,10695,'#FFFFFF'),(5646,10696,'#FFFFFF'),(5647,10697,'#FFFFFF'),(5648,10698,'#FFFFFF'),(5649,10699,'#FFFFFF'),(5650,10700,'#c0f459'),(5651,10701,'#FFFFFF'),(5652,10703,'#FFFFFF'),(5653,10706,'#2df385'),(5654,10713,'#FFFFFF'),(5655,10715,'#3647e5'),(5656,10719,'#FFFFFF'),(5657,10731,'#bf71ef'),(5658,10733,'#FFFFFF'),(5659,10734,'#FFFFFF'),(5660,10736,'#FFFFFF'),(5661,10737,'#FFFFFF'),(5662,10741,'#FFFFFF'),(5663,10744,'#d3973b'),(5664,10746,'#40ec40'),(5665,10748,'#FFFFFF'),(5666,10749,'#db57c9'),(5667,10750,'#fe72d7'),(5668,10753,'#3e73e9'),(5669,10754,'#FFFFFF'),(5670,10755,'#25faad'),(5671,10756,'#FFFFFF'),(5672,10757,'#d62227'),(5673,10759,'#8fd75f'),(5674,10760,'#FFFFFF'),(5675,10762,'#FFFFFF'),(5676,10763,'#FFFFFF'),(5677,10764,'#FFFFFF'),(5678,10767,'#FFFFFF'),(5679,10768,'#FFFFFF'),(5680,10771,'#FFFFFF'),(5681,10772,'#FFFFFF'),(5682,10773,'#57d72d'),(5683,10775,'#4298fc'),(5684,10776,'#554be9'),(5685,10777,'#FFFFFF'),(5686,10788,'#d87e3b'),(5687,10790,'#FFFFFF'),(5688,10791,'#FFFFFF'),(5689,10792,'#845fe2'),(5690,10793,'#ed50c9'),(5691,10794,'#FFFFFF'),(5692,10795,'#FFFFFF'),(5693,10796,'#FFFFFF'),(5694,10797,'#FFFFFF'),(5695,10798,'#FFFFFF'),(5696,10802,'#3309f3'),(5697,10803,'#FFFFFF'),(5698,10804,'#fc7e6a'),(5699,10807,'#FFFFFF'),(5700,10811,'#FFFFFF'),(5701,10813,'#FFFFFF'),(5702,10814,'#FFFFFF'),(5703,10817,'#5a6ef3'),(5704,10818,'#4fbfe6'),(5705,10819,'#FFFFFF'),(5706,10821,'#FFFFFF'),(5707,10824,'#FFFFFF'),(5708,10825,'#c86bd7'),(5709,10826,'#f97e10'),(5710,10828,'#FFFFFF'),(5711,10830,'#35d716'),(5712,10831,'#acf068'),(5713,10832,'#FFFFFF'),(5714,10833,'#FFFFFF'),(5715,10836,'#FFFFFF'),(5716,10837,'#FFFFFF'),(5717,10840,'#FFFFFF'),(5718,10841,'#f5ee69'),(5719,10846,'#FFFFFF'),(5720,10850,'#f576a6'),(5721,10854,'#FFFFFF'),(5722,10856,'#6f1af5'),(5723,10858,'#FFFFFF'),(5724,10862,'#2cfa5c'),(5725,10864,'#FFFFFF'),(5726,10865,'#FFFFFF'),(5727,10868,'#c3e438'),(5728,10874,'#FFFFFF'),(5729,10875,'#FFFFFF'),(5730,10876,'#FFFFFF'),(5731,10877,'#FFFFFF'),(5732,10880,'#9f52f7'),(5733,10888,'#3ac6fa'),(5734,10889,'#2812da'),(5735,10893,'#FFFFFF'),(5736,10896,'#fbb501'),(5737,10897,'#76a1fd'),(5738,10898,'#a74fdc'),(5739,10899,'#FFFFFF'),(5740,10902,'#FFFFFF'),(5741,10903,'#e559bf'),(5742,10904,'#FFFFFF'),(5743,10906,'#FFFFFF'),(5744,10909,'#FFFFFF'),(5745,10912,'#9a0cd4'),(5746,10915,'#FFFFFF'),(5747,10916,'#4764f1'),(5748,10917,'#f0608d'),(5749,10919,'#FFFFFF'),(5750,10920,'#FFFFFF'),(5751,10921,'#3735fb'),(5752,10923,'#cfd442'),(5753,10924,'#ce0d0e'),(5754,10927,'#FFFFFF'),(5755,10928,'#01dfd3'),(5756,10929,'#a611f0'),(5757,10930,'#FFFFFF'),(5758,10933,'#FFFFFF'),(5759,10934,'#FFFFFF'),(5760,10935,'#FFFFFF'),(5761,10937,'#f9d866'),(5762,10938,'#d17f5c'),(5763,10939,'#ce2a57'),(5764,10941,'#FFFFFF'),(5765,10943,'#f43b8a'),(5766,10946,'#6bd9bf'),(5767,10949,'#49e98c'),(5768,10951,'#FFFFFF'),(5769,10953,'#FFFFFF'),(5770,10954,'#FFFFFF'),(5771,10956,'#6ee7df'),(5772,10957,'#FFFFFF'),(5773,10958,'#3b58f8'),(5774,10959,'#FFFFFF'),(5775,10962,'#FFFFFF'),(5776,10970,'#d8ec68'),(5777,10971,'#FFFFFF'),(5778,10979,'#FFFFFF'),(5779,10982,'#f10e62'),(5780,10983,'#6a93d6'),(5781,10984,'#FFFFFF'),(5782,10986,'#FFFFFF'),(5783,10988,'#FFFFFF'),(5784,10989,'#FFFFFF'),(5785,10993,'#49ceae'),(5786,10996,'#FFFFFF'),(5787,10999,'#00eb00'),(5788,11001,'#FFFFFF'),(5789,11002,'#FFFFFF'),(5790,11004,'#FFFFFF'),(5791,11005,'#FFFFFF'),(5792,11006,'#57e114'),(5793,11007,'#FFFFFF'),(5794,11009,'#8cda5b'),(5795,11012,'#7fea69'),(5796,11016,'#72f534'),(5797,11017,'#FFFFFF'),(5798,11018,'#1612d8'),(5799,11019,'#FFFFFF'),(5800,11028,'#35d9b0'),(5801,11029,'#6365d8'),(5802,11047,'#d77019'),(5803,11051,'#ea0032'),(5804,11074,'#a5da15'),(5805,11080,'#FFFFFF'),(5806,11094,'#b664e7'),(5807,11096,'#9769ea'),(5808,11098,'#4bcce0'),(5809,11105,'#75d554'),(5810,11132,'#a826f6'),(5811,11135,'#f3f004'),(5812,11139,'#7532e9'),(5813,11148,'#ae08fd'),(5814,11154,'#48d26c'),(5815,11159,'#500cec'),(5816,11161,'#89f15a'),(5817,11162,'#e26e72'),(5818,11164,'#d23cbc'),(5819,11166,'#2c4ae2'),(5820,11167,'#6bdaf7'),(5821,11169,'#e16d60'),(5822,11170,'#f9282d'),(5823,11174,'#0ff742'),(5824,11176,'#FFFFFF'),(5825,11178,'#55b7de'),(5826,11183,'#5fd574'),(5827,11184,'#5871d4'),(5828,11187,'#fd1fbb'),(5829,11191,'#6415e2'),(5830,11194,'#d19852'),(5831,11195,'#FFFFFF'),(5832,11196,'#5ecd34'),(5833,11197,'#3df9f1'),(5834,11204,'#FFFFFF'),(5835,11205,'#f90f50'),(5836,11206,'#f5c221'),(5837,11210,'#2576ed'),(5838,11211,'#eece4e'),(5839,11214,'#68d2e5'),(5840,11215,'#f2b143'),(5841,11217,'#65da9a'),(5842,11219,'#a2e33c'),(5843,11226,'#88e46f'),(5844,11232,'#a665fd'),(5845,11236,'#e06a26'),(5846,11237,'#b211d0'),(5847,11238,'#65ead4'),(5848,11239,'#ebec5f'),(5849,11240,'#56dce7'),(5850,11244,'#a969e6'),(5851,11245,'#FFFFFF'),(5852,11249,'#3a13de'),(5853,11250,'#4550ce'),(5854,11251,'#afde16'),(5855,11254,'#8bde44'),(5856,11256,'#b5f06d'),(5857,11258,'#1402f8'),(5858,11259,'#e5c231'),(5859,11261,'#1ac7d2'),(5860,11262,'#a9dc48'),(5861,11265,'#3de486'),(5862,11266,'#399be0'),(5863,11268,'#2480cf'),(5864,11272,'#6427ef'),(5865,11274,'#1cf81e'),(5866,11277,'#1609df'),(5867,11278,'#61dd6e'),(5868,11280,'#3cfa82'),(5869,11281,'#c609cd'),(5870,11282,'#ee0274'),(5871,11285,'#FFFFFF'),(5872,11291,'#0803f3'),(5873,11292,'#f35add'),(5874,11295,'#d3dc46'),(5875,11297,'#3cdf49'),(5876,11299,'#FFFFFF'),(5877,11303,'#cd72ec'),(5878,11305,'#d9d852'),(5879,11309,'#e06e80'),(5880,11310,'#cd4f83'),(5881,11313,'#26fc37'),(5882,11321,'#bb49ed'),(5883,11322,'#c9e92d'),(5884,11327,'#edfa2c'),(5885,11328,'#12e934'),(5886,11329,'#f27a4a'),(5887,11330,'#01fb9e'),(5888,11333,'#7821e9'),(5889,11338,'#f86994'),(5890,11340,'#5057dd'),(5891,11342,'#b0e133'),(5892,11343,'#e0be4b'),(5893,11345,'#4c36ee'),(5894,11346,'#04ed43'),(5895,11347,'#1e20fe'),(5896,11349,'#654af7'),(5897,11350,'#9ef264'),(5898,11351,'#f34a8e'),(5899,11359,'#9a34dd'),(5900,11360,'#21d34d'),(5901,11361,'#a6cf3a'),(5902,11363,'#cfc61f'),(5903,11367,'#c404d1'),(5904,11372,'#78f84f'),(5905,11377,'#7cfee8'),(5906,11379,'#10e81d'),(5907,11380,'#f62e29'),(5908,11381,'#a5df51'),(5909,11382,'#FFFFFF'),(5910,11385,'#10a3ea'),(5911,11388,'#55dba4'),(5912,11389,'#81fb0e'),(5913,11390,'#e82895'),(5914,11401,'#cd4022'),(5915,11402,'#d662cc'),(5916,11404,'#1083dd'),(5917,11407,'#bdfd6b'),(5918,11408,'#d73e92'),(5919,11409,'#7390e9'),(5920,11410,'#1ad42e'),(5921,11411,'#59ecd1'),(5922,11415,'#eb08d5'),(5923,11417,'#a84bfe'),(5924,11419,'#6df9c5'),(5925,11420,'#fc2916'),(5926,11425,'#62cbe8'),(5927,11426,'#932acc'),(5928,11427,'#eea451'),(5929,11431,'#c03ae4'),(5930,11435,'#9d0bde'),(5931,11438,'#FFFFFF'),(5932,11439,'#FFFFFF'),(5933,11440,'#1bcdfc'),(5934,11442,'#8549d5'),(5935,11443,'#eb66c2'),(5936,11445,'#ea9170'),(5937,11446,'#f475dc'),(5938,11447,'#e053b6'),(5939,11449,'#dc5486'),(5940,11450,'#de096c'),(5941,11452,'#9c4fd1'),(5942,11454,'#FFFFFF'),(5943,11456,'#8f30f7'),(5944,11459,'#1eeaba'),(5945,11460,'#FFFFFF'),(5946,11461,'#b33bde'),(5947,11463,'#b3d706'),(5948,11465,'#fded13'),(5949,11466,'#f34e45'),(5950,11468,'#d26376'),(5951,11469,'#fe549d'),(5952,11477,'#da42ab'),(5953,11479,'#8eeb65'),(5954,11481,'#ce03f3'),(5955,11482,'#db8254'),(5956,11484,'#dd9865'),(5957,11488,'#65a9d3'),(5958,11490,'#896ff4'),(5959,11493,'#31df80'),(5960,11494,'#11e49e'),(5961,11495,'#e3176a'),(5962,11500,'#f32b13'),(5963,11501,'#31dcca'),(5964,11505,'#ca4ae4'),(5965,11506,'#cd4751'),(5966,11507,'#e16bac'),(5967,11508,'#9067d2'),(5968,11509,'#6ce3fc'),(5969,11512,'#34df01'),(5970,11513,'#be3de5'),(5971,11514,'#0f29e8'),(5972,11520,'#07b0cc'),(5973,11521,'#9ffc4f'),(5974,11522,'#03cf87'),(5975,11524,'#be67fd'),(5976,11528,'#7542cd'),(5977,11530,'#f81c81'),(5978,11532,'#e88e4e'),(5979,11533,'#b163cd'),(5980,11535,'#39d67e'),(5981,11538,'#18c7fe'),(5982,11541,'#38fd85'),(5983,11542,'#21db8f'),(5984,11543,'#FFFFFF'),(5985,11546,'#e46270'),(5986,11547,'#1a35fd'),(5987,11548,'#f7d02e'),(5988,11549,'#cfcd3b'),(5989,11550,'#ef775d'),(5990,11554,'#c9fc0c'),(5991,11555,'#6fd320'),(5992,11556,'#e704f7'),(5993,11557,'#20e329'),(5994,11559,'#db0400'),(5995,11561,'#f514c4'),(5996,11563,'#94ec4c'),(5997,11564,'#cc2ede'),(5998,11567,'#52b8db'),(5999,11568,'#b6ef44'),(6000,11569,'#6071fb'),(6001,11570,'#fc2338'),(6002,11575,'#2ad950'),(6003,11576,'#FFFFFF'),(6004,11579,'#a1d607'),(6005,11580,'#f1503e'),(6006,11582,'#dc8132'),(6007,11584,'#fdde65'),(6008,11585,'#33c0f7'),(6009,11587,'#c2e064'),(6010,11588,'#54dcbc'),(6011,11591,'#6e13e6'),(6012,11592,'#66d371'),(6013,11595,'#54fd80'),(6014,11597,'#78f064'),(6015,11599,'#4b1ce5'),(6016,11601,'#f0d42e'),(6017,11602,'#933be0'),(6018,11603,'#f46bdd'),(6019,11604,'#011fd5'),(6020,11606,'#f80b3f'),(6021,11609,'#FFFFFF'),(6022,11611,'#a433e5'),(6023,11613,'#d72677'),(6024,11615,'#b7cc62'),(6025,11618,'#7850ef'),(6026,11621,'#de4d5f'),(6027,11622,'#6bc4e7'),(6028,11626,'#df6df8'),(6029,11628,'#e82d4a'),(6030,11630,'#3691d0'),(6031,11633,'#9665f6'),(6032,11635,'#cc2738'),(6033,11639,'#8cd030'),(6034,11640,'#41decc'),(6035,11641,'#11f926'),(6036,11642,'#18fc61'),(6037,11644,'#ed2f63'),(6038,11645,'#10eaea'),(6039,11651,'#2cf7d0'),(6040,11652,'#7ada4e'),(6041,11658,'#d646fa'),(6042,11659,'#60f7a3'),(6043,11661,'#6af2b7'),(6044,11663,'#0d5bdb'),(6045,11672,'#c1f651'),(6046,11674,'#8fdc6d'),(6047,11675,'#f5d61f'),(6048,11680,'#c5f503'),(6049,11682,'#2d8af3'),(6050,11687,'#ed8a5f'),(6051,11688,'#e73e99'),(6052,11689,'#f84e51'),(6053,11695,'#6e5be4'),(6054,11697,'#e13cb8'),(6055,11698,'#81ed6d'),(6056,11699,'#04e54f'),(6057,11701,'#a529e6'),(6058,11702,'#95d90b'),(6059,11708,'#43fcee'),(6060,11714,'#2d31e6'),(6061,11716,'#e519de'),(6062,11719,'#dd03c9'),(6063,11722,'#e22caa'),(6064,11723,'#ce0b7f'),(6065,11821,'#f24c8a'),(6066,11833,'#FFFFFF'),(6067,11834,'#19f527'),(6068,11868,'#fb53e5'),(6069,11874,'#FFFFFF'),(6070,11875,'#d804d7'),(6071,11880,'#de6dfe'),(6072,11889,'#a8f53c'),(6073,11893,'#FFFFFF'),(6074,11902,'#4dd9ca'),(6075,11927,'#FFFFFF'),(6076,11935,'#deaa6a'),(6077,11961,'#d6cd63'),(6078,11964,'#123cf8'),(6079,11973,'#f53314'),(6080,11993,'#cd3664'),(6081,12027,'#cf4ff0'),(6082,12140,'#a6cf15'),(6083,12149,'#d1352d'),(6084,12216,'#e49e1f'),(6085,12285,'#4ea6f5'),(6086,12294,'#f077af'),(6087,12305,'#3c2ce9'),(6088,12306,'#5f37e7'),(6089,12315,'#d12274'),(6090,12319,'#d848f8'),(6091,12325,'#e9c663'),(6092,12326,'#e2a924'),(6093,12342,'#6bd265'),(6094,12352,'#ed9722'),(6095,12367,'#e17269'),(6096,12369,'#9ef022'),(6097,12405,'#cc4533'),(6098,12434,'#11d09e'),(6099,12437,'#FFFFFF'),(6100,12439,'#FFFFFF'),(6101,12442,'#cf437d'),(6102,12443,'#dc27a5'),(6103,12444,'#0b73da'),(6104,12449,'#4375e0'),(6105,12451,'#93f823'),(6106,12457,'#cdf61d'),(6107,12467,'#f34698'),(6108,12468,'#FFFFFF'),(6109,12470,'#d1e044'),(6110,12471,'#0af6bb'),(6111,12473,'#684bde'),(6112,12474,'#5650da'),(6113,12483,'#e148c1'),(6114,12485,'#d86b62'),(6115,12489,'#43fabe'),(6116,12492,'#FFFFFF'),(6117,12498,'#d743d9'),(6118,12502,'#a7f014'),(6119,12505,'#FFFFFF'),(6120,12506,'#0fd4b2'),(6121,12508,'#1471cf'),(6122,12510,'#2649ed'),(6123,12511,'#fe8100'),(6124,12515,'#FFFFFF'),(6125,12517,'#53ce68'),(6126,12519,'#25d93b'),(6127,12521,'#3eecda'),(6128,12523,'#4663dc'),(6129,12528,'#dc2968'),(6130,12531,'#62d75c'),(6131,12532,'#65e108'),(6132,12533,'#2a95df'),(6133,12539,'#49fd00'),(6134,12543,'#0742ed'),(6135,12547,'#3bfed9'),(6136,12549,'#FFFFFF'),(6137,12552,'#FFFFFF'),(6138,12554,'#e767f3'),(6139,12556,'#f7305f'),(6140,12562,'#db6b89'),(6141,12574,'#FFFFFF'),(6142,12576,'#f114d3'),(6143,12578,'#0ac4dd'),(6144,12580,'#0aec2c'),(6145,12584,'#16e06d'),(6146,12593,'#2113d3'),(6147,12599,'#1ff90d'),(6148,12623,'#d75b7a'),(6149,12636,'#e13db4'),(6150,12641,'#5fe84e'),(6151,12652,'#d81a0a'),(6152,12655,'#59e94d'),(6153,12663,'#d269ae'),(6154,12668,'#da9723'),(6155,12671,'#b0cd4d'),(6156,12675,'#fb5e86'),(6157,12678,'#db28b1'),(6158,12681,'#8b1bf4'),(6159,12686,'#2a40fa'),(6160,12694,'#3b7ddc'),(6161,12695,'#b8df1c'),(6162,12696,'#dc7bf7'),(6163,12701,'#3b5efc'),(6164,12702,'#b2d229'),(6165,12706,'#b9d327'),(6166,12709,'#6066ce'),(6167,12710,'#f8484c'),(6168,12711,'#3199db'),(6169,12715,'#cb40ee'),(6170,12719,'#f7357f'),(6171,12726,'#7bf21a'),(6172,12742,'#90ec6f'),(6173,12744,'#55cebe'),(6174,12747,'#d18f4f'),(6175,12748,'#5eeb9a'),(6176,12754,'#5cf3d5'),(6177,12764,'#50c2cc'),(6178,12767,'#a7e512'),(6179,12770,'#7258de'),(6180,12771,'#08fc88'),(6181,12775,'#fa371f'),(6182,12785,'#f3172f'),(6183,12786,'#ae0cdd'),(6184,12789,'#d7587e'),(6185,12793,'#5b84f5'),(6186,12800,'#d357ad'),(6187,12804,'#6badda'),(6188,12806,'#c733d4'),(6189,12813,'#4bb3df'),(6190,12824,'#289ae2'),(6191,12826,'#2233cf'),(6192,12827,'#2232ee'),(6193,12832,'#e5da2a'),(6194,12833,'#d84c4a'),(6195,12834,'#c74bcd'),(6196,12835,'#1cbaf1'),(6197,12836,'#d30964'),(6198,12837,'#42efe8'),(6199,12838,'#d62f42'),(6200,12845,'#01d0fe'),(6201,12857,'#d3fb2e'),(6202,12859,'#cb3ed8'),(6203,12874,'#a868f7'),(6204,12882,'#43dc8c'),(6205,12883,'#40fd02'),(6206,12884,'#13e3ed'),(6207,12885,'#d18b2d'),(6208,12888,'#1237cd'),(6209,12906,'#cc971a'),(6210,12909,'#e37264'),(6211,12916,'#f5b244'),(6212,12917,'#6b9ef7'),(6213,12923,'#ed6deb'),(6214,12924,'#21cea6'),(6215,12925,'#3897d7'),(6216,12927,'#d1e90c'),(6217,12930,'#60d566'),(6218,12945,'#3f9be4'),(6219,12946,'#e48a3a'),(6220,12951,'#fe704c'),(6221,12952,'#06d2a1'),(6222,12954,'#d9a854'),(6223,12960,'#fefb4b'),(6224,12979,'#1bfee0'),(6225,12984,'#be67ef'),(6226,12985,'#3ce508'),(6227,12986,'#b21ddc'),(6228,12997,'#5ad7b4'),(6229,13004,'#e43e68'),(6230,13007,'#c806f9'),(6231,13008,'#0c1bcd'),(6232,13009,'#17d383'),(6233,13015,'#6a4ed5'),(6234,13022,'#af30df'),(6235,13023,'#3c82f9'),(6236,13029,'#9ff568'),(6237,13032,'#0346ee'),(6238,13047,'#76fee8'),(6239,13050,'#4fccd3'),(6240,13071,'#6479f1'),(6241,13078,'#defc48'),(6242,13080,'#d94d75'),(6243,13081,'#ce8260'),(6244,13084,'#5f53ea'),(6245,13085,'#61dd3d'),(6246,13086,'#a9df25'),(6247,13099,'#e95b53'),(6248,13101,'#8d50f3'),(6249,13102,'#ecf326'),(6250,13113,'#4bcf71'),(6251,13114,'#9fde6b'),(6252,13116,'#5c0ce9'),(6253,13117,'#e14822'),(6254,13119,'#e53d7f'),(6255,13122,'#daa83d'),(6256,13123,'#ec4d93'),(6257,13125,'#24edef'),(6258,13127,'#6cd5e2'),(6259,13128,'#bef82a'),(6260,13133,'#ea29e5'),(6261,13146,'#24f33c'),(6262,13150,'#dcda39'),(6263,13155,'#82da0d'),(6264,13161,'#3ffc38'),(6265,13163,'#a0ee3b'),(6266,13165,'#0027ce'),(6267,13171,'#eb2c84'),(6268,13179,'#1e6cfb'),(6269,13181,'#4f4ce3'),(6270,13185,'#fc5abc'),(6271,13191,'#d39540'),(6272,13194,'#d772fd'),(6273,13197,'#33edb1'),(6274,13198,'#f2fd5e'),(6275,13199,'#38d7fe'),(6276,13204,'#fb11ea'),(6277,13212,'#323ae9'),(6278,13213,'#204fe1'),(6279,13217,'#e1b638'),(6280,13220,'#f4a66c'),(6281,13229,'#4be214'),(6282,13234,'#da1724'),(6283,13236,'#7bfb92'),(6284,13237,'#2df1cc'),(6285,13238,'#92f70b'),(6286,13242,'#e601b0'),(6287,13245,'#28e566'),(6288,13247,'#5bdbdb'),(6289,13250,'#3fcc3d'),(6290,13252,'#2fe125'),(6291,13255,'#FFFFFF'),(6292,13256,'#330ce1'),(6293,13258,'#bff84e'),(6294,13259,'#f565cd'),(6295,13264,'#5bde87'),(6296,13268,'#1be579'),(6297,13273,'#ed638f'),(6298,13276,'#28d550'),(6299,13282,'#e91787'),(6300,13284,'#f13528'),(6301,13287,'#cf8543'),(6302,13290,'#cd6ee8'),(6303,13291,'#d4734a'),(6304,13292,'#15e325'),(6305,13305,'#5ac8eb'),(6306,13317,'#52a0df'),(6307,13326,'#dd3e0c'),(6308,13331,'#56a5e3'),(6309,13341,'#dea743'),(6310,13343,'#4dd6cb'),(6311,13344,'#e4a764'),(6312,13351,'#45f981'),(6313,13354,'#2cbbcc'),(6314,13358,'#f86e9f'),(6315,13360,'#d4687e'),(6316,13362,'#1fcbd0'),(6317,13370,'#e65635'),(6318,13372,'#b2e24c'),(6319,13374,'#6252fa'),(6320,13389,'#b025e6'),(6321,13399,'#5302f4'),(6322,13402,'#e13c27'),(6323,13404,'#fe50d5'),(6324,13419,'#6bdd19'),(6325,13434,'#befc43'),(6326,13435,'#fb6d9a'),(6327,13442,'#98cf1b'),(6328,13449,'#1de239'),(6329,13451,'#d4e827'),(6330,13458,'#e65ee2'),(6331,13459,'#4e2dfe'),(6332,13461,'#3c11d0'),(6333,13467,'#d82580'),(6334,13471,'#7edd67'),(6335,13473,'#aadd4d'),(6336,13476,'#40ea7b'),(6337,13480,'#411cf0'),(6338,13482,'#0580f4'),(6339,13483,'#b51ef2'),(6340,13484,'#f10354'),(6341,13488,'#31b7cd'),(6342,13490,'#4ee17b'),(6343,13492,'#dacf4a'),(6344,13494,'#3a8ece'),(6345,13495,'#f916e5'),(6346,13499,'#68c0fd'),(6347,13500,'#63bdd5'),(6348,13507,'#cc1e51'),(6349,13512,'#9668f8'),(6350,13513,'#3d59da'),(6351,13518,'#f1b523'),(6352,13522,'#deee0f'),(6353,13528,'#6cbcf1'),(6354,13536,'#69cbf1'),(6355,13537,'#8331e9'),(6356,13541,'#65dcac'),(6357,13546,'#d2c831'),(6358,13553,'#1794d5'),(6359,13558,'#ed6fda'),(6360,13562,'#307df7'),(6361,13564,'#e58861'),(6362,13565,'#4920d2'),(6363,13566,'#47efb3'),(6364,13569,'#7af565'),(6365,13579,'#ba4fd4'),(6366,13584,'#ec7d3c'),(6367,13589,'#18a3d8'),(6368,13597,'#9bd149'),(6369,13601,'#e6bf4a'),(6370,13603,'#fa29c5'),(6371,13612,'#e8ec44'),(6372,13613,'#6add68'),(6373,13619,'#bd43ce'),(6374,13624,'#ed894a'),(6375,13630,'#c649fd'),(6376,13631,'#fa75e2'),(6377,13632,'#5705f9'),(6378,13636,'#5e45fd'),(6379,13640,'#f00f64'),(6380,13643,'#cd0d03'),(6381,13645,'#7dcf43'),(6382,13646,'#38cc9c'),(6383,13648,'#50d602'),(6384,13654,'#27eb3a'),(6385,13668,'#fa1236'),(6386,13669,'#dbee71'),(6387,13680,'#FFFFFF'),(6388,13685,'#f539aa'),(6389,13695,'#d63920'),(6390,13710,'#21d5a7'),(6391,13719,'#8c34e7'),(6392,13720,'#e437af'),(6393,13724,'#afd119'),(6394,13729,'#e9c86c'),(6395,13753,'#cc8615'),(6396,13757,'#25f1ab'),(6397,13772,'#28d6f3'),(6398,13776,'#6fe330'),(6399,13777,'#79d1fe'),(6400,13782,'#ee4a9b'),(6401,13785,'#f64342'),(6402,13791,'#f70c41'),(6403,13820,'#f4d75e'),(6404,13830,'#3028e1'),(6405,13839,'#cf04dc'),(6406,13841,'#7b54cf'),(6407,13843,'#5da7d9'),(6408,13850,'#d2470e'),(6409,13876,'#426dd6'),(6410,13957,'#0a12eb'),(6411,13966,'#63d96f'),(6412,13969,'#b676f7'),(6413,13974,'#ce449d'),(6414,13981,'#ce3761'),(6415,13982,'#d27858'),(6416,13992,'#3f0bf3'),(6417,13993,'#87e501'),(6418,13998,'#d750e3'),(6419,14004,'#c540ce'),(6420,14005,'#0acf3e'),(6421,14006,'#731cd2'),(6422,14008,'#e5bf0c'),(6423,14014,'#e1647e'),(6424,14015,'#d0932c'),(6425,14020,'#126fcf'),(6426,14033,'#e95bd5'),(6427,14041,'#8b4ce3'),(6428,14049,'#d27430'),(6429,14051,'#32cf6e'),(6430,14054,'#d7a81f'),(6431,14057,'#eda960'),(6432,14059,'#de3551'),(6433,14061,'#e3ed65'),(6434,14066,'#1f2bdc'),(6435,14067,'#63ddfc'),(6436,14068,'#5b74e0'),(6437,14069,'#dd3e17'),(6438,14074,'#7ff211'),(6439,14075,'#6df8aa'),(6440,14080,'#3983d3'),(6441,14082,'#0f7dcd'),(6442,14084,'#d06b1b'),(6443,14088,'#d87e40'),(6444,14101,'#4dd2e0'),(6445,14105,'#3dec6f'),(6446,14113,'#e117b4'),(6447,14114,'#6e8ef2'),(6448,14122,'#5562e5'),(6449,14131,'#FFFFFF'),(6450,14133,'#FFFFFF'),(6451,14136,'#FFFFFF'),(6452,14137,'#FFFFFF'),(6453,14141,'#35d4fb'),(6454,14145,'#e11846'),(6455,14147,'#37ce8e'),(6456,14150,'#f2bd78'),(6457,14156,'#7a52fc'),(6458,14164,'#FFFFFF'),(6459,14166,'#FFFFFF'),(6460,14169,'#FFFFFF'),(6461,14173,'#FFFFFF'),(6462,14175,'#25e904'),(6463,14178,'#FFFFFF'),(6464,14186,'#d98428'),(6465,14188,'#FFFFFF'),(6466,14197,'#ec12cd'),(6467,14200,'#FFFFFF'),(6468,14203,'#39d58d'),(6469,14205,'#FFFFFF'),(6470,14218,'#d94121'),(6471,14230,'#FFFFFF'),(6472,14243,'#FFFFFF'),(6473,14250,'#fd9453'),(6474,14256,'#ae42f0'),(6475,14261,'#FFFFFF'),(6476,14274,'#4bd051'),(6477,14280,'#FFFFFF'),(6478,14299,'#FFFFFF'),(6479,14303,'#66a3cf'),(6480,14305,'#FFFFFF'),(6481,14318,'#6afb32'),(6482,14332,'#FFFFFF'),(6483,14337,'#FFFFFF'),(6484,14338,'#FFFFFF'),(6485,14340,'#FFFFFF'),(6486,14343,'#FFFFFF'),(6487,14367,'#FFFFFF'),(6488,14369,'#ec15e4'),(6489,14406,'#cc073b'),(6490,14408,'#FFFFFF'),(6491,14421,'#FFFFFF'),(6492,14423,'#f3cf75'),(6493,14424,'#f46513'),(6494,14425,'#78dc6b'),(6495,14430,'#b7e66a'),(6496,14436,'#f97876'),(6497,14446,'#e8cf65'),(6498,14451,'#2a9fd4'),(6499,14463,'#767dfa'),(6500,14467,'#e59120'),(6501,14472,'#456ee4'),(6502,14474,'#3824df'),(6503,14477,'#FFFFFF'),(6504,14483,'#1fd7e8'),(6505,14489,'#de8848'),(6506,14507,'#3555e0'),(6507,14543,'#3f98d6'),(6508,14560,'#aef565'),(6509,14565,'#de55ab'),(6510,14578,'#e95bef'),(6511,14607,'#d6a92e'),(6512,14614,'#4dfd65'),(6513,14624,'#e1b42f'),(6514,14626,'#b9f964'),(6515,14630,'#9aeb49'),(6516,14631,'#df3645'),(6517,14638,'#ce6ffc'),(6518,14642,'#6230d8'),(6519,14646,'#d9f049'),(6520,14649,'#ec0fb4'),(6521,14650,'#3be924'),(6522,14658,'#d02855'),(6523,14660,'#29b9fa'),(6524,14666,'#d739b4'),(6525,14668,'#bb70f5'),(6526,14672,'#7d3bd4'),(6527,14675,'#25e778'),(6528,14683,'#7e2de8'),(6529,14698,'#fd2df9'),(6530,14706,'#FFFFFF'),(6531,14708,'#16cd5b'),(6532,14709,'#acef74'),(6533,14710,'#d53e51'),(6534,14712,'#78b2f9'),(6535,14713,'#d04a4b'),(6536,14714,'#4c7be7'),(6537,14718,'#5438f0'),(6538,14725,'#37cd00'),(6539,14727,'#857dfb'),(6540,14733,'#fe44f6'),(6541,14744,'#30e1e0'),(6542,14746,'#a5e240'),(6543,14749,'#63cf3d'),(6544,14753,'#60e085'),(6545,14757,'#3ebfe3'),(6546,14765,'#fc9861'),(6547,14767,'#758bf1'),(6548,14769,'#6ae832'),(6549,14770,'#dedd55'),(6550,14772,'#f87b67'),(6551,14774,'#ec8d43'),(6552,14775,'#facf0e'),(6553,14779,'#2ed54b'),(6554,14780,'#f87546'),(6555,14784,'#fb0a53'),(6556,14787,'#00e13b'),(6557,14792,'#4805f7'),(6558,14793,'#d07633'),(6559,14798,'#8bdf43'),(6560,14799,'#fb464d'),(6561,14802,'#a159fa'),(6562,14810,'#764beb'),(6563,14811,'#32bfcf'),(6564,14814,'#a26ce5'),(6565,14821,'#d3a327'),(6566,14823,'#e526bb'),(6567,14824,'#4843e0'),(6568,14828,'#3274da'),(6569,14829,'#591def'),(6570,14843,'#bce362'),(6571,14851,'#d33d78'),(6572,14857,'#eafb1f'),(6573,14858,'#d70c8d'),(6574,14861,'#732ef5'),(6575,14864,'#4282d0'),(6576,14865,'#07eafa'),(6577,14866,'#c3ec2e'),(6578,14868,'#46d555'),(6579,14870,'#d044ac'),(6580,14874,'#ce9125'),(6581,14876,'#6fdd0b'),(6582,14877,'#24f426'),(6583,14885,'#dea94a'),(6584,14886,'#ea41af'),(6585,14887,'#f8972d'),(6586,14888,'#d58630'),(6587,14892,'#d39746'),(6588,14893,'#d3e33b'),(6589,14894,'#fdf36d'),(6590,14896,'#595ade'),(6591,14898,'#d58064'),(6592,14899,'#e75d92'),(6593,14901,'#fc28e7'),(6594,14902,'#dd6124'),(6595,14905,'#44dfe3'),(6596,14913,'#6c66e3'),(6597,14917,'#0bcd42'),(6598,14920,'#6cb1e0'),(6599,14921,'#d86bbc'),(6600,14926,'#0bc1e2'),(6601,14927,'#2476f4'),(6602,14929,'#3dcf6f'),(6603,14931,'#FFFFFF'),(6604,14932,'#0629f1'),(6605,14934,'#f8b21d'),(6606,14936,'#598cdf'),(6607,14939,'#f26f4c'),(6608,14942,'#baf220'),(6609,14944,'#c136e2'),(6610,14952,'#f571f5'),(6611,14961,'#2ddbdd'),(6612,14963,'#fbeb0d'),(6613,14965,'#0397d1'),(6614,14968,'#d85a53'),(6615,14969,'#3fdd82'),(6616,14970,'#98e546'),(6617,14980,'#d58561'),(6618,14987,'#5d2acc'),(6619,14988,'#ac39fb'),(6620,14991,'#a6f562'),(6621,14992,'#63cfca'),(6622,14994,'#e260ac'),(6623,14996,'#1ff2c6'),(6624,15005,'#00fd67'),(6625,15006,'#5ce0c8'),(6626,15009,'#51d340'),(6627,15010,'#68e1ef'),(6628,15012,'#e04bc5'),(6629,15013,'#e93ace'),(6630,15019,'#f4670a'),(6631,15020,'#e0da1b'),(6632,15021,'#acd35d'),(6633,15024,'#FFFFFF'),(6634,15029,'#d3260e'),(6635,15031,'#f0195a'),(6636,15037,'#a63fcf'),(6637,15040,'#3e83d7'),(6638,15041,'#de0b43'),(6639,15043,'#eb427a'),(6640,15045,'#5bd022'),(6641,15050,'#32f6f7'),(6642,15051,'#97de5b'),(6643,15062,'#9ae660'),(6644,15063,'#d55014'),(6645,15071,'#ceae4b'),(6646,15075,'#5d97fe'),(6647,15077,'#67d928'),(6648,15085,'#c651d6'),(6649,15090,'#cdf354'),(6650,15095,'#71a2f6'),(6651,15097,'#cfb85a'),(6652,15101,'#4087e8'),(6653,15104,'#ee42af'),(6654,15106,'#76febe'),(6655,15110,'#d431ac'),(6656,15112,'#1e02f9'),(6657,15113,'#2ce073'),(6658,15115,'#d7d90c'),(6659,15121,'#f0557c'),(6660,15125,'#961df8'),(6661,15135,'#f6202a'),(6662,15137,'#3804ed'),(6663,15138,'#4af946'),(6664,15139,'#FFFFFF'),(6665,15143,'#f2f577'),(6666,15148,'#e1fc50'),(6667,15149,'#46a0da'),(6668,15152,'#d5423b'),(6669,15153,'#bad649'),(6670,15157,'#d49937'),(6671,15161,'#23e837'),(6672,15177,'#e7761a'),(6673,15181,'#35dcce'),(6674,15186,'#af01df'),(6675,15187,'#d757a2'),(6676,15189,'#0138ec'),(6677,15190,'#e7c034'),(6678,15193,'#d363c0'),(6679,15194,'#45d280'),(6680,15195,'#FFFFFF'),(6681,15197,'#fd6c75'),(6682,15203,'#a006ef'),(6683,15204,'#48e925'),(6684,15208,'#fea460'),(6685,15210,'#f3ef64'),(6686,15213,'#ca3bd3'),(6687,15217,'#e27073'),(6688,15222,'#e9a349'),(6689,15232,'#f939ac'),(6690,15233,'#e761bc'),(6691,15236,'#fd5648'),(6692,15237,'#a45ffd'),(6693,15238,'#b3f26d'),(6694,15244,'#fc3cd3'),(6695,15245,'#16afd7'),(6696,15246,'#52d33f'),(6697,15252,'#6bf2d5'),(6698,15254,'#4eda0b'),(6699,15258,'#fe42f8'),(6700,15261,'#1dd0ae'),(6701,15272,'#77ec3a'),(6702,15274,'#7546da'),(6703,15279,'#00e779'),(6704,15283,'#49d692'),(6705,15284,'#22df40'),(6706,15285,'#f7d237'),(6707,15286,'#e4ae25'),(6708,15292,'#51dd7c'),(6709,15294,'#6ccf53'),(6710,15298,'#fc8a09'),(6711,15302,'#52d0bd'),(6712,15308,'#da9d5e'),(6713,15312,'#df36a1'),(6714,15318,'#fc6a19'),(6715,15320,'#db1523'),(6716,15324,'#03da4c'),(6717,15325,'#eb3152'),(6718,15327,'#779bf1'),(6719,15329,'#96e25e'),(6720,15332,'#ddd35e'),(6721,15337,'#ea6d3c'),(6722,15340,'#bfcd26'),(6723,15344,'#e270fe'),(6724,15349,'#11ce76'),(6725,15352,'#9a5de4'),(6726,15354,'#dc3830'),(6727,15368,'#22e226'),(6728,15375,'#d2bb39'),(6729,15378,'#3e76d5'),(6730,15379,'#44cef1'),(6731,15382,'#95d237'),(6732,15384,'#1cc9dd'),(6733,15385,'#d57c5a'),(6734,15390,'#e27533'),(6735,15391,'#1b31f7'),(6736,15397,'#39cdeb'),(6737,15398,'#b92cf6'),(6738,15400,'#2382f9'),(6739,15403,'#e16d81'),(6740,15404,'#57df51'),(6741,15408,'#cd3492'),(6742,15412,'#9b2adc'),(6743,15414,'#90e12b'),(6744,15415,'#593feb'),(6745,15417,'#444ecc'),(6746,15423,'#74e344'),(6747,15425,'#d058e4'),(6748,15426,'#90e459'),(6749,15433,'#df176f'),(6750,15436,'#dd68fe'),(6751,15437,'#548cf2'),(6752,15446,'#12fe33'),(6753,15451,'#4fde36'),(6754,15454,'#74cd21'),(6755,15455,'#e040c7'),(6756,15456,'#db557d'),(6757,15458,'#cc4fab'),(6758,15459,'#3a05dc'),(6759,15463,'#37df6b'),(6760,15464,'#a9f42a'),(6761,15466,'#6fe622'),(6762,15467,'#dd2fda'),(6763,15469,'#edd812'),(6764,15472,'#210be9'),(6765,15474,'#6fb8f6'),(6766,15476,'#e87567'),(6767,15478,'#73f238'),(6768,15482,'#ed0a9b'),(6769,15483,'#5cd25f'),(6770,15485,'#ed5e14'),(6771,15486,'#1d66e8'),(6772,15489,'#e548c2'),(6773,15491,'#7224f0'),(6774,15496,'#f9db5d'),(6775,15497,'#92f46e'),(6776,15498,'#60d581'),(6777,15503,'#0b58fd'),(6778,15508,'#e24cef'),(6779,15509,'#0097e7'),(6780,15513,'#b6d05b'),(6781,15514,'#0af403'),(6782,15520,'#e3be4f'),(6783,15521,'#8be301'),(6784,15522,'#26e22c'),(6785,15523,'#40d074'),(6786,15527,'#1822cc'),(6787,15529,'#b4db66'),(6788,15530,'#f84b0c'),(6789,15531,'#00d505'),(6790,15532,'#f71cd5'),(6791,15533,'#1bf9b3'),(6792,15538,'#ce9e50'),(6793,15542,'#cc5985'),(6794,15543,'#fa5348'),(6795,15549,'#0f0bf3'),(6796,15551,'#cefa60'),(6797,15552,'#0de90b'),(6798,15554,'#eb3655'),(6799,15557,'#db467d'),(6800,15558,'#78f876'),(6801,15561,'#d719f1'),(6802,15563,'#5b48d0'),(6803,15565,'#298aeb'),(6804,15569,'#20d024'),(6805,15573,'#11dd92'),(6806,15576,'#e68749'),(6807,15581,'#06d047'),(6808,15582,'#cd4686'),(6809,15584,'#ee4fe1'),(6810,15586,'#34ce92'),(6811,15588,'#4ccad4'),(6812,15590,'#f3d222'),(6813,15591,'#06afd8'),(6814,15594,'#ec715d'),(6815,15597,'#15ee19'),(6816,15600,'#8de535'),(6817,15602,'#6d60ed'),(6818,15603,'#bc4ede'),(6819,15604,'#d530e8'),(6820,15612,'#d4f675'),(6821,15616,'#76e931'),(6822,15617,'#86f24e'),(6823,15620,'#0aea0a'),(6824,15625,'#b5cf5f'),(6825,15628,'#cf7912'),(6826,15629,'#b627f7'),(6827,15635,'#e08615'),(6828,15637,'#f45007'),(6829,15639,'#00e6d7'),(6830,15640,'#2cd853'),(6831,15652,'#5cf4d0'),(6832,15653,'#e911bf'),(6833,15656,'#ef8420'),(6834,15659,'#f02e69'),(6835,15661,'#6ad22e'),(6836,15665,'#d6ef70'),(6837,15668,'#d853ea'),(6838,15669,'#16f48a'),(6839,15670,'#f9760c'),(6840,15671,'#a049de'),(6841,15672,'#1d63cf'),(6842,15674,'#d539bb'),(6843,15675,'#403bfb'),(6844,15677,'#cff872'),(6845,15679,'#f74f46'),(6846,15685,'#ce8c60'),(6847,15689,'#a40ce3'),(6848,15712,'#FFFFFF'),(6849,15713,'#6ed308'),(6850,15734,'#e0831e'),(6851,15751,'#3f83de'),(6852,15806,'#FFFFFF'),(6853,15807,'#FFFFFF'),(6854,15808,'#FFFFFF'),(6855,15809,'#FFFFFF'),(6856,15810,'#e25ba5'),(6857,15811,'#720cf5'),(6858,15812,'#FFFFFF'),(6859,15814,'#FFFFFF'),(6860,15816,'#8546ef'),(6861,15818,'#FFFFFF'),(6862,15819,'#63e1bb'),(6863,15821,'#77ef66'),(6864,15822,'#FFFFFF'),(6865,15824,'#FFFFFF'),(6866,15825,'#ce1bb5'),(6867,15827,'#FFFFFF'),(6868,15828,'#FFFFFF'),(6869,15829,'#31ce16'),(6870,15831,'#FFFFFF'),(6871,15832,'#FFFFFF'),(6872,15834,'#e706bb'),(6873,15837,'#FFFFFF'),(6874,15838,'#FFFFFF'),(6875,15839,'#FFFFFF'),(6876,15844,'#FFFFFF'),(6877,15845,'#fe4d61'),(6878,15848,'#f0a747'),(6879,15849,'#FFFFFF'),(6880,15851,'#d40901'),(6881,15852,'#FFFFFF'),(6882,15854,'#FFFFFF'),(6883,15857,'#FFFFFF'),(6884,15858,'#7ef15b'),(6885,15859,'#3eee73'),(6886,15862,'#FFFFFF'),(6887,15863,'#8dce5f'),(6888,15864,'#1356ea'),(6889,15865,'#e09916'),(6890,15866,'#FFFFFF'),(6891,15867,'#FFFFFF'),(6892,15868,'#fb3d9b'),(6893,15869,'#FFFFFF'),(6894,15871,'#76eee4'),(6895,15872,'#60aad1'),(6896,15873,'#89e552'),(6897,15874,'#f19459'),(6898,15875,'#83d30a'),(6899,15876,'#FFFFFF'),(6900,15878,'#4cd1b8'),(6901,15880,'#71f775'),(6902,15884,'#40ddc3'),(6903,15886,'#37d83b'),(6904,15888,'#4fd5ea'),(6905,15889,'#f48160'),(6906,15890,'#7420e1'),(6907,15891,'#db3e8e'),(6908,15892,'#ce559c'),(6909,15894,'#da394e'),(6910,15895,'#FFFFFF'),(6911,15896,'#FFFFFF'),(6912,15897,'#ed6f0a'),(6913,15900,'#FFFFFF'),(6914,15901,'#edc533'),(6915,15902,'#FFFFFF'),(6916,15907,'#f99140'),(6917,15908,'#f8c542'),(6918,15910,'#9ce65a'),(6919,15911,'#FFFFFF'),(6920,15913,'#69d580'),(6921,15914,'#FFFFFF'),(6922,15915,'#63eee3'),(6923,15917,'#3600d7'),(6924,15918,'#FFFFFF'),(6925,15920,'#fc9830'),(6926,15921,'#e953f1'),(6927,15923,'#FFFFFF'),(6928,15924,'#8a21e0'),(6929,15925,'#e3d858'),(6930,15926,'#6fde65'),(6931,15927,'#5ff1fd'),(6932,15928,'#f5611b'),(6933,15930,'#FFFFFF'),(6934,15931,'#d25181'),(6935,15932,'#bcfb79'),(6936,15935,'#FFFFFF'),(6937,15936,'#57de12'),(6938,15937,'#3a4ce2'),(6939,15938,'#FFFFFF'),(6940,15939,'#6c49f8'),(6941,15940,'#d9e447'),(6942,15941,'#547ffe'),(6943,15942,'#cc5e5a'),(6944,15944,'#FFFFFF'),(6945,15945,'#1c32e3'),(6946,15946,'#ea1cca'),(6947,15948,'#ed9449'),(6948,15949,'#FFFFFF'),(6949,15951,'#FFFFFF'),(6950,15953,'#FFFFFF'),(6951,15954,'#fb27d5'),(6952,15955,'#0d1cfb'),(6953,15957,'#FFFFFF'),(6954,15958,'#FFFFFF'),(6955,15959,'#c3f216'),(6956,15960,'#d85ad3'),(6957,15961,'#0cd288'),(6958,15962,'#f83093'),(6959,15963,'#df9b68'),(6960,15964,'#FFFFFF'),(6961,15965,'#5ed440'),(6962,15966,'#618bd3'),(6963,15970,'#3282dd'),(6964,15971,'#FFFFFF'),(6965,15972,'#ecc54b'),(6966,15973,'#FFFFFF'),(6967,15974,'#FFFFFF'),(6968,15976,'#e28006'),(6969,15977,'#FFFFFF'),(6970,15978,'#FFFFFF'),(6971,15979,'#5f18e0'),(6972,15981,'#FFFFFF'),(6973,15984,'#FFFFFF'),(6974,15987,'#FFFFFF'),(6975,15988,'#FFFFFF'),(6976,15989,'#FFFFFF'),(6977,15990,'#FFFFFF'),(6978,15991,'#FFFFFF'),(6979,15992,'#FFFFFF'),(6980,15993,'#FFFFFF'),(6981,15994,'#FFFFFF'),(6982,15995,'#FFFFFF'),(6983,15998,'#FFFFFF'),(6984,15999,'#933ef6'),(6985,16002,'#d462f8'),(6986,16008,'#FFFFFF'),(6987,16010,'#FFFFFF'),(6988,16013,'#d2dd55'),(6989,16014,'#b9d108'),(6990,16015,'#ca53d6'),(6991,16016,'#da28f2'),(6992,16017,'#8d3bea'),(6993,16019,'#FFFFFF'),(6994,16020,'#61f2e0'),(6995,16021,'#FFFFFF'),(6996,16022,'#3ff337'),(6997,16023,'#FFFFFF'),(6998,16024,'#3fcfb4'),(6999,16025,'#5057d3'),(7000,16026,'#FFFFFF'),(7001,16027,'#FFFFFF'),(7002,16028,'#e24c09'),(7003,16030,'#315cf0'),(7004,16033,'#FFFFFF'),(7005,16034,'#FFFFFF'),(7006,16036,'#072ccc'),(7007,16037,'#ee4d60'),(7008,16038,'#dd8a42'),(7009,16039,'#FFFFFF'),(7010,16040,'#d00a30'),(7011,16042,'#f426b0'),(7012,16043,'#dc01e6'),(7013,16044,'#ac62fb'),(7014,16045,'#FFFFFF'),(7015,16046,'#FFFFFF'),(7016,16047,'#FFFFFF'),(7017,16053,'#d41929'),(7018,16057,'#25f931'),(7019,16058,'#8814d5'),(7020,16059,'#FFFFFF'),(7021,16060,'#FFFFFF'),(7022,16062,'#FFFFFF'),(7023,16063,'#6789ed'),(7024,16065,'#FFFFFF'),(7025,16067,'#be01e4'),(7026,16068,'#2ae11b'),(7027,16069,'#FFFFFF'),(7028,16071,'#5ca8fe'),(7029,16076,'#eebf67'),(7030,16077,'#FFFFFF'),(7031,16079,'#cbd42e'),(7032,16082,'#5055f8'),(7033,16083,'#19bce1'),(7034,16085,'#66d793'),(7035,16086,'#78adf8'),(7036,16087,'#FFFFFF'),(7037,16088,'#FFFFFF'),(7038,16090,'#435ccf'),(7039,16091,'#d0270c'),(7040,16092,'#3a81d6'),(7041,16094,'#FFFFFF'),(7042,16095,'#87fc4f'),(7043,16096,'#f2f054'),(7044,16097,'#FFFFFF'),(7045,16098,'#e33af7'),(7046,16099,'#34f8e2'),(7047,16100,'#344bf8'),(7048,16101,'#FFFFFF'),(7049,16103,'#FFFFFF'),(7050,16104,'#51d2f1'),(7051,16105,'#b259d4'),(7052,16108,'#af43d4'),(7053,16109,'#30dd55'),(7054,16111,'#FFFFFF'),(7055,16112,'#FFFFFF'),(7056,16113,'#FFFFFF'),(7057,16116,'#FFFFFF'),(7058,16119,'#FFFFFF'),(7059,16121,'#4e98d0'),(7060,16122,'#61d8ea'),(7061,16123,'#e2171e'),(7062,16124,'#0842df'),(7063,16127,'#84dc6a'),(7064,16130,'#f59e45'),(7065,16132,'#9916d4'),(7066,16133,'#FFFFFF'),(7067,16136,'#d56e44'),(7068,16138,'#45caf5'),(7069,16139,'#f43da3'),(7070,16140,'#FFFFFF'),(7071,16141,'#63bde9'),(7072,16143,'#ee2fc9'),(7073,16144,'#FFFFFF'),(7074,16145,'#554bfe'),(7075,16146,'#496bf5'),(7076,16148,'#FFFFFF'),(7077,16150,'#2e53d0'),(7078,16151,'#4ee49a'),(7079,16152,'#FFFFFF'),(7080,16153,'#de1b34'),(7081,16154,'#34fadf'),(7082,16156,'#6c55cf'),(7083,16163,'#FFFFFF'),(7084,16164,'#FFFFFF'),(7085,16165,'#c0e36c'),(7086,16166,'#FFFFFF'),(7087,16167,'#fd4324'),(7088,16168,'#FFFFFF'),(7089,16169,'#cd7664'),(7090,16170,'#FFFFFF'),(7091,16171,'#4fb0d7'),(7092,16172,'#0424d3'),(7093,16174,'#FFFFFF'),(7094,16176,'#FFFFFF'),(7095,16178,'#FFFFFF'),(7096,16181,'#2df28a'),(7097,16184,'#FFFFFF'),(7098,16185,'#77ee79'),(7099,16188,'#FFFFFF'),(7100,16189,'#74a3f2'),(7101,16190,'#27fb7c'),(7102,16192,'#FFFFFF'),(7103,16193,'#0ed870'),(7104,16194,'#FFFFFF'),(7105,16196,'#e80ac1'),(7106,16198,'#ced14b'),(7107,16200,'#a04be1'),(7108,16201,'#FFFFFF'),(7109,16202,'#f19459'),(7110,16204,'#eda03d'),(7111,16208,'#fd867a'),(7112,16216,'#d13d37'),(7113,16217,'#c953d0'),(7114,16219,'#d456d5'),(7115,16220,'#FFFFFF'),(7116,16221,'#ee9355'),(7117,16222,'#FFFFFF'),(7118,16223,'#FFFFFF'),(7119,16224,'#FFFFFF'),(7120,16226,'#d12e25'),(7121,16227,'#FFFFFF'),(7122,16228,'#3fb5fb'),(7123,16230,'#19d86f'),(7124,16231,'#ec6773'),(7125,16232,'#a334f3'),(7126,16233,'#0d7bda'),(7127,16235,'#e3e835'),(7128,16237,'#1bc7eb'),(7129,16238,'#FFFFFF'),(7130,16239,'#ed4840'),(7131,16240,'#FFFFFF'),(7132,16241,'#d72082'),(7133,16244,'#FFFFFF'),(7134,16245,'#3fd537'),(7135,16246,'#3696e3'),(7136,16249,'#48fe5a'),(7137,16250,'#f050a3'),(7138,16251,'#FFFFFF'),(7139,16252,'#d749e7'),(7140,16253,'#0bcd57'),(7141,16255,'#FFFFFF'),(7142,16256,'#fb48df'),(7143,16257,'#FFFFFF'),(7144,16259,'#6b79e0'),(7145,16261,'#befa4a'),(7146,16262,'#FFFFFF'),(7147,16263,'#FFFFFF'),(7148,16265,'#FFFFFF'),(7149,16266,'#FFFFFF'),(7150,16270,'#70f797'),(7151,16271,'#6968f6'),(7152,16272,'#FFFFFF'),(7153,16276,'#FFFFFF'),(7154,16277,'#FFFFFF'),(7155,16278,'#69e845'),(7156,16279,'#2af717'),(7157,16280,'#b552e4'),(7158,16281,'#da2194'),(7159,16282,'#FFFFFF'),(7160,16285,'#FFFFFF'),(7161,16287,'#FFFFFF'),(7162,16288,'#FFFFFF'),(7163,16290,'#FFFFFF'),(7164,16291,'#074eee'),(7165,16294,'#9ae139'),(7166,16295,'#FFFFFF'),(7167,16296,'#8ccc49'),(7168,16299,'#FFFFFF'),(7169,16300,'#FFFFFF'),(7170,16301,'#FFFFFF'),(7171,16302,'#FFFFFF'),(7172,16304,'#4a79d1'),(7173,16305,'#bd3bd1'),(7174,16306,'#FFFFFF'),(7175,16309,'#FFFFFF'),(7176,16310,'#FFFFFF'),(7177,16311,'#cf4cd1'),(7178,16312,'#1f9cd5'),(7179,16313,'#FFFFFF'),(7180,16315,'#23e6a7'),(7181,16316,'#479ae4'),(7182,16317,'#194dce'),(7183,16318,'#40fb78'),(7184,16319,'#0fdc20'),(7185,16322,'#FFFFFF'),(7186,16325,'#d53cb0'),(7187,16326,'#dd2f70'),(7188,16327,'#FFFFFF'),(7189,16328,'#cbd120'),(7190,16329,'#FFFFFF'),(7191,16330,'#FFFFFF'),(7192,16334,'#e0241c'),(7193,16343,'#24d6b7'),(7194,16344,'#19e8c8'),(7195,16345,'#FFFFFF'),(7196,16348,'#4ebbd8'),(7197,16349,'#FFFFFF'),(7198,16351,'#e360d7'),(7199,16352,'#e68a58'),(7200,16354,'#23d32b'),(7201,16356,'#FFFFFF'),(7202,16363,'#FFFFFF'),(7203,16364,'#FFFFFF'),(7204,16366,'#FFFFFF'),(7205,16367,'#a73bde'),(7206,16370,'#e636ee'),(7207,16371,'#33deac'),(7208,16372,'#FFFFFF'),(7209,16373,'#FFFFFF'),(7210,16376,'#FFFFFF'),(7211,16377,'#FFFFFF'),(7212,16379,'#ac42f2'),(7213,16380,'#eb13a2'),(7214,16381,'#FFFFFF'),(7215,16382,'#FFFFFF'),(7216,16383,'#56cc09'),(7217,16384,'#FFFFFF'),(7218,16385,'#2ccc1a'),(7219,16387,'#f9cf69'),(7220,16389,'#9cd05d'),(7221,16390,'#FFFFFF'),(7222,16391,'#b40efd'),(7223,16392,'#FFFFFF'),(7224,16393,'#2410f4'),(7225,16394,'#FFFFFF'),(7226,16395,'#e6438a'),(7227,16396,'#cd0cd2'),(7228,16397,'#FFFFFF'),(7229,16402,'#4abade'),(7230,16403,'#403bf2'),(7231,16404,'#d94389'),(7232,16405,'#FFFFFF'),(7233,16406,'#6bef4c'),(7234,16411,'#886ae9'),(7235,16413,'#35b0eb'),(7236,16415,'#4de08b'),(7237,16416,'#FFFFFF'),(7238,16417,'#FFFFFF'),(7239,16419,'#33ef3f'),(7240,16421,'#1c3ad8'),(7241,16422,'#5deba3'),(7242,16423,'#d725b7'),(7243,16424,'#69cc08'),(7244,16427,'#FFFFFF'),(7245,16428,'#FFFFFF'),(7246,16429,'#FFFFFF'),(7247,16430,'#0acee7'),(7248,16433,'#e65e5d'),(7249,16434,'#FFFFFF'),(7250,16435,'#be16d8'),(7251,16436,'#65d2cb'),(7252,16438,'#d76389'),(7253,16439,'#1de121'),(7254,16440,'#3365f5'),(7255,16441,'#1bce1b'),(7256,16442,'#f2ae48'),(7257,16444,'#47f2f1'),(7258,16445,'#FFFFFF'),(7259,16448,'#07ccd0'),(7260,16449,'#fa3ed6'),(7261,16450,'#42fe60'),(7262,16451,'#FFFFFF'),(7263,16453,'#d7e421'),(7264,16457,'#5d9bd3'),(7265,16458,'#FFFFFF'),(7266,16459,'#FFFFFF'),(7267,16461,'#FFFFFF'),(7268,16462,'#6e1dde'),(7269,16463,'#6cda4e'),(7270,16467,'#FFFFFF'),(7271,16469,'#707afe'),(7272,16470,'#FFFFFF'),(7273,16471,'#FFFFFF'),(7274,16476,'#580aef'),(7275,16480,'#262ae7'),(7276,16481,'#FFFFFF'),(7277,16482,'#FFFFFF'),(7278,16483,'#1edf79'),(7279,16484,'#ee2254'),(7280,16486,'#32df4f'),(7281,16487,'#e88c4b'),(7282,16488,'#FFFFFF'),(7283,16492,'#FFFFFF'),(7284,16499,'#f15a3e'),(7285,16500,'#FFFFFF'),(7286,16501,'#f7b72b'),(7287,16503,'#f23eca'),(7288,16504,'#FFFFFF'),(7289,16506,'#FFFFFF'),(7290,16511,'#FFFFFF'),(7291,16519,'#FFFFFF'),(7292,16535,'#43dc79'),(7293,16538,'#FFFFFF'),(7294,16552,'#FFFFFF'),(7295,16556,'#FFFFFF'),(7296,16567,'#FFFFFF'),(7297,16576,'#FFFFFF'),(7298,16588,'#FFFFFF'),(7299,16594,'#52d071'),(7300,16600,'#FFFFFF'),(7301,16601,'#0e0dd1'),(7302,16603,'#cca165'),(7303,16605,'#ee3dd7'),(7304,16607,'#FFFFFF'),(7305,16618,'#381dcf'),(7306,16622,'#f66060'),(7307,16627,'#FFFFFF'),(7308,16628,'#FFFFFF'),(7309,16633,'#5990e7'),(7310,16634,'#fad171'),(7311,16638,'#e7c545'),(7312,16641,'#cd34ba'),(7313,16646,'#FFFFFF'),(7314,16654,'#FFFFFF'),(7315,16659,'#3ae3b9'),(7316,16660,'#4de4b0'),(7317,16663,'#81e671'),(7318,16679,'#FFFFFF'),(7319,16687,'#FFFFFF'),(7320,16692,'#FFFFFF'),(7321,16749,'#b5d165'),(7322,16766,'#26e3ea'),(7323,16792,'#33ecd1'),(7324,16845,'#76c1f4'),(7325,16851,'#28c1d1'),(7326,16858,'#d841f3'),(7327,16859,'#55e045'),(7328,16866,'#d0a43f'),(7329,16869,'#4b71e8'),(7330,16879,'#3952d6'),(7331,16881,'#e53094'),(7332,16917,'#952ffd'),(7333,16924,'#cd1686'),(7334,16926,'#b921e0'),(7335,16945,'#e5367c'),(7336,16947,'#FFFFFF'),(7337,16948,'#ba17fb'),(7338,16953,'#a749d5'),(7339,16970,'#78f5cd'),(7340,16975,'#a4f40b'),(7341,16982,'#f2ec51'),(7342,17027,'#6ded04'),(7343,17042,'#FFFFFF'),(7344,17053,'#82d130'),(7345,17083,'#dc9f6c'),(7346,17085,'#f377ee'),(7347,17086,'#e6ca0b'),(7348,17093,'#f48879'),(7349,17095,'#3299d0'),(7350,17096,'#0033e7'),(7351,17101,'#4b83fb'),(7352,17103,'#FFFFFF'),(7353,17106,'#13d220'),(7354,17107,'#13e63d'),(7355,17108,'#e72d80'),(7356,17111,'#cbe22e'),(7357,17113,'#4037e1'),(7358,17118,'#07b3e3'),(7359,17124,'#2efd6c'),(7360,17125,'#d6d512'),(7361,17126,'#251ace'),(7362,17129,'#582fdf'),(7363,17139,'#c908da'),(7364,17146,'#fc15f1'),(7365,17153,'#f22870'),(7366,17157,'#50f557'),(7367,17159,'#d41827'),(7368,17161,'#26ec5b'),(7369,17162,'#19f04f'),(7370,17167,'#13c9f7'),(7371,17170,'#3fe2e5'),(7372,17176,'#3bee0d'),(7373,17185,'#3c37e5'),(7374,17187,'#73b2f1'),(7375,17188,'#db6951'),(7376,17189,'#75d143'),(7377,17196,'#89f039'),(7378,17214,'#3be462'),(7379,17223,'#a755cf'),(7380,17225,'#e05a09'),(7381,17235,'#5b6cdb'),(7382,17243,'#a6f702'),(7383,17247,'#0cf3bf'),(7384,17248,'#4075d7'),(7385,17250,'#05db58'),(7386,17252,'#3c8ccc'),(7387,17263,'#8476f8'),(7388,17268,'#ceb706'),(7389,17292,'#1ecbf5'),(7390,17293,'#de4f72'),(7391,17294,'#dcb667'),(7392,17298,'#cd1cca'),(7393,17305,'#42a7e1'),(7394,17306,'#9749da'),(7395,17307,'#52f4e1'),(7396,17308,'#54c0cc'),(7397,17314,'#9c3ed1'),(7398,17321,'#ce6720'),(7399,17328,'#FFFFFF'),(7400,17330,'#ab38cf'),(7401,17340,'#FFFFFF'),(7402,17350,'#d06280'),(7403,17363,'#decf6d'),(7404,17365,'#e83f63'),(7405,17381,'#2e7ed8'),(7406,17385,'#79fe70'),(7407,17386,'#08dcf7'),(7408,17415,'#f5147b'),(7409,17416,'#02d76d'),(7410,17418,'#da03a1'),(7411,17422,'#d57662'),(7412,17423,'#84ed0e'),(7413,17427,'#a5d220'),(7414,17433,'#ed46b0'),(7415,17435,'#de0c6f'),(7416,17436,'#eabf35'),(7417,17444,'#cfe104'),(7418,17447,'#52e77a'),(7419,17449,'#fd5ba1'),(7420,17457,'#FFFFFF'),(7421,17459,'#FFFFFF'),(7422,17460,'#FFFFFF'),(7423,17462,'#FFFFFF'),(7424,17468,'#FFFFFF'),(7425,17474,'#1afebf'),(7426,17475,'#FFFFFF'),(7427,17477,'#d324b2'),(7428,17487,'#FFFFFF'),(7429,17489,'#0dd036'),(7430,17497,'#f020da'),(7431,17502,'#FFFFFF'),(7432,17503,'#27dfef'),(7433,17514,'#f551ad'),(7434,17519,'#e4f851'),(7435,17531,'#FFFFFF'),(7436,17536,'#FFFFFF'),(7437,17537,'#FFFFFF'),(7438,17544,'#FFFFFF'),(7439,17545,'#FFFFFF'),(7440,17558,'#FFFFFF'),(7441,17559,'#FFFFFF'),(7442,17566,'#884be8'),(7443,17569,'#2da2fd'),(7444,17573,'#FFFFFF'),(7445,17574,'#1ee9d0'),(7446,17576,'#5cec85'),(7447,17577,'#3ed415'),(7448,17580,'#FFFFFF'),(7449,17582,'#FFFFFF'),(7450,17586,'#5df96e'),(7451,17590,'#dc4fc1'),(7452,17593,'#d6e34c'),(7453,17594,'#ef453a'),(7454,17605,'#FFFFFF'),(7455,17619,'#FFFFFF'),(7456,17626,'#36e71f'),(7457,17629,'#f2f815'),(7458,17631,'#FFFFFF'),(7459,17638,'#c8ec74'),(7460,17639,'#FFFFFF'),(7461,17641,'#ed8920'),(7462,17642,'#FFFFFF'),(7463,17648,'#FFFFFF'),(7464,17657,'#FFFFFF'),(7465,17659,'#FFFFFF'),(7466,17664,'#40f7d2'),(7467,17665,'#FFFFFF'),(7468,17671,'#FFFFFF'),(7469,17672,'#FFFFFF'),(7470,17675,'#FFFFFF'),(7471,17677,'#e12d67'),(7472,17684,'#d3471d'),(7473,17687,'#FFFFFF'),(7474,17691,'#FFFFFF'),(7475,17701,'#ce3065'),(7476,17702,'#FFFFFF'),(7477,17706,'#FFFFFF'),(7478,17710,'#FFFFFF'),(7479,17718,'#d04ac5'),(7480,17719,'#FFFFFF'),(7481,17734,'#FFFFFF'),(7482,17738,'#FFFFFF'),(7483,17739,'#FFFFFF'),(7484,17740,'#f9280b'),(7485,17743,'#2826d4'),(7486,17747,'#cd53ea'),(7487,17773,'#FFFFFF'),(7488,17775,'#FFFFFF'),(7489,17780,'#f4e005'),(7490,17790,'#FFFFFF'),(7491,17793,'#fee612'),(7492,17794,'#FFFFFF'),(7493,17806,'#6beaea'),(7494,17808,'#FFFFFF'),(7495,17809,'#FFFFFF'),(7496,17810,'#0803f0'),(7497,17818,'#21fe89'),(7498,17821,'#f19f21'),(7499,17825,'#FFFFFF'),(7500,17829,'#FFFFFF'),(7501,17835,'#f6611b'),(7502,17837,'#FFFFFF'),(7503,17853,'#fb531e'),(7504,17859,'#e617b3'),(7505,17862,'#9b69d9'),(7506,17872,'#FFFFFF'),(7507,17873,'#cd6c1e'),(7508,17874,'#f725b1'),(7509,17878,'#FFFFFF'),(7510,17881,'#FFFFFF'),(7511,17883,'#58d8ec'),(7512,17884,'#FFFFFF'),(7513,17885,'#FFFFFF'),(7514,17889,'#4ba7eb'),(7515,17902,'#c6da42'),(7516,17903,'#FFFFFF'),(7517,17905,'#3c65e3'),(7518,17911,'#fb28ec'),(7519,17912,'#2241cd'),(7520,17916,'#FFFFFF'),(7521,17918,'#FFFFFF'),(7522,17921,'#6bcfe9'),(7523,17923,'#fdbd19'),(7524,17926,'#94d54b'),(7525,17928,'#2fdbc1'),(7526,17929,'#31d2dd'),(7527,17930,'#61e682'),(7528,17933,'#59f82b'),(7529,17935,'#de2d2a'),(7530,17939,'#4adcd0'),(7531,17940,'#f9da49'),(7532,17943,'#81ce0e'),(7533,17945,'#4db0d7'),(7534,17946,'#070de6'),(7535,17949,'#d3cf64'),(7536,17951,'#943fda'),(7537,17957,'#cc00a7'),(7538,17958,'#3292d5'),(7539,17959,'#aae771'),(7540,17960,'#282bfd'),(7541,17961,'#a6d230'),(7542,17962,'#e4b54d'),(7543,17963,'#088bcd'),(7544,17965,'#ec578e'),(7545,17966,'#4a73d4'),(7546,17968,'#2244e8'),(7547,17969,'#f645f2'),(7548,17977,'#b1f74d'),(7549,17979,'#119ddc'),(7550,17983,'#d2a35c'),(7551,17984,'#d761b1'),(7552,17987,'#ce3663'),(7553,17988,'#e18a18'),(7554,17989,'#3eded6'),(7555,17999,'#6846e2'),(7556,18001,'#d743b7'),(7557,18003,'#525ee8'),(7558,18006,'#06dd25'),(7559,18007,'#e170c9'),(7560,18014,'#e41868'),(7561,18015,'#bee110'),(7562,18016,'#05de69'),(7563,18017,'#17ce9b'),(7564,18021,'#FFFFFF'),(7565,18022,'#cd67de'),(7566,18024,'#e2698b'),(7567,18025,'#92df48'),(7568,18028,'#6ba8df'),(7569,18030,'#7536d8'),(7570,18039,'#d44b8b'),(7571,18040,'#43d601'),(7572,18041,'#e733ed'),(7573,18042,'#2bcd52'),(7574,18044,'#0e6add'),(7575,18045,'#fde407'),(7576,18049,'#c3d756'),(7577,18051,'#ed41c4'),(7578,18052,'#eabe0e'),(7579,18053,'#4ea0d4'),(7580,18055,'#67afd9'),(7581,18056,'#e8b612'),(7582,18059,'#FFFFFF'),(7583,18064,'#cd7553'),(7584,18072,'#dc5044'),(7585,18073,'#64d88b'),(7586,18074,'#5aafda'),(7587,18078,'#f71ae7'),(7588,18079,'#d11e27'),(7589,18081,'#985def'),(7590,18082,'#6ca4e8'),(7591,18085,'#22fc72'),(7592,18086,'#FFFFFF'),(7593,18087,'#5dd1db'),(7594,18088,'#fc8c7a'),(7595,18090,'#f4cd31'),(7596,18093,'#cc58ea'),(7597,18094,'#32ec51'),(7598,18095,'#c2fa66'),(7599,18096,'#FFFFFF'),(7600,18097,'#3559d9'),(7601,18100,'#f96221'),(7602,18101,'#bbfb76'),(7603,18102,'#a40aed'),(7604,18105,'#8e29cc'),(7605,18106,'#91dd17'),(7606,18108,'#ebed4a'),(7607,18109,'#e1f921'),(7608,18110,'#ea5496'),(7609,18111,'#2a6dfc'),(7610,18112,'#0c3be1'),(7611,18115,'#a927d5'),(7612,18117,'#dc5445'),(7613,18118,'#29e0ba'),(7614,18119,'#89eb47'),(7615,18120,'#f34114'),(7616,18123,'#ce5947'),(7617,18124,'#f669d1'),(7618,18125,'#19d031'),(7619,18128,'#FFFFFF'),(7620,18129,'#c438f5'),(7621,18130,'#716fe8'),(7622,18131,'#dc67f8'),(7623,18140,'#edba73'),(7624,18144,'#b6d64e'),(7625,18148,'#5df53e'),(7626,18150,'#4b67e0'),(7627,18151,'#825dcf'),(7628,18152,'#f34cea'),(7629,18153,'#e48d3c'),(7630,18156,'#09f7a8'),(7631,18157,'#2595e6'),(7632,18158,'#f8f97b'),(7633,18159,'#FFFFFF'),(7634,18168,'#70f432'),(7635,18169,'#e450cd'),(7636,18172,'#729df1'),(7637,18173,'#ee786b'),(7638,18179,'#ce53a7'),(7639,18180,'#f4ef02'),(7640,18181,'#3f34d5'),(7641,18182,'#6d2ace'),(7642,18185,'#f775d0'),(7643,18188,'#eb23b4'),(7644,18190,'#8162de'),(7645,18192,'#9832e5'),(7646,18193,'#92fb41'),(7647,18195,'#0d98f7'),(7648,18197,'#cf320d'),(7649,18198,'#49e3b4'),(7650,18199,'#f29d3c'),(7651,18205,'#e4ba6f'),(7652,18207,'#a91cec'),(7653,18212,'#f9154f'),(7654,18213,'#5067e2'),(7655,18215,'#0ec0d4'),(7656,18216,'#914ad6'),(7657,18219,'#dd0280'),(7658,18221,'#cdef75'),(7659,18223,'#1c38df'),(7660,18224,'#10e0b5'),(7661,18226,'#f5ec6e'),(7662,18228,'#37cca3'),(7663,18229,'#8b64fd'),(7664,18235,'#55f254'),(7665,18236,'#e33e7a'),(7666,18238,'#33e422'),(7667,18243,'#e98774'),(7668,18246,'#56e81c'),(7669,18247,'#6891e4'),(7670,18252,'#8c12d9'),(7671,18253,'#0914d6'),(7672,18257,'#FFFFFF'),(7673,18258,'#76fcf2'),(7674,18263,'#55f728'),(7675,18264,'#268ded'),(7676,18268,'#12db68'),(7677,18270,'#68cef9'),(7678,18271,'#FFFFFF'),(7679,18279,'#2ad616'),(7680,18280,'#f37088'),(7681,18282,'#518ee3'),(7682,18284,'#f46382'),(7683,18290,'#03e5ae'),(7684,18291,'#0af888'),(7685,18293,'#e9f63a'),(7686,18295,'#fda31a'),(7687,18300,'#e0fb3e'),(7688,18301,'#1ce040'),(7689,18307,'#d09110'),(7690,18309,'#f1f332'),(7691,18311,'#d83362'),(7692,18312,'#30d429'),(7693,18317,'#41fcb0'),(7694,18318,'#ef5b71'),(7695,18321,'#79e4f2'),(7696,18322,'#cbd240'),(7697,18324,'#6d05e1'),(7698,18328,'#FFFFFF'),(7699,18331,'#6edc64'),(7700,18334,'#46f678'),(7701,18336,'#47cdba'),(7702,18337,'#73d459'),(7703,18340,'#cd9659'),(7704,18342,'#4e74e2'),(7705,18348,'#ea16fb'),(7706,18350,'#cf4dfc'),(7707,18354,'#5fddeb'),(7708,18356,'#50d877'),(7709,18371,'#d79e29'),(7710,18372,'#58e05f'),(7711,18378,'#ea0a6d'),(7712,18380,'#0ba3f4'),(7713,18381,'#8d66f8'),(7714,18382,'#3e29e4'),(7715,18383,'#40cce9'),(7716,18384,'#f58379'),(7717,18391,'#02e867'),(7718,18392,'#0ad1c4'),(7719,18396,'#de4c46'),(7720,18399,'#de594b'),(7721,18400,'#f370c5'),(7722,18401,'#2045ef'),(7723,18403,'#dd4c19'),(7724,18406,'#eeb01e'),(7725,18412,'#747be9'),(7726,18413,'#f372a8'),(7727,18415,'#d835d6'),(7728,18417,'#13d462'),(7729,18418,'#f21881'),(7730,18420,'#fb7c17'),(7731,18421,'#2202fc'),(7732,18422,'#53d65e'),(7733,18424,'#d621d0'),(7734,18425,'#7734e0'),(7735,18426,'#6348f1'),(7736,18427,'#1102e8'),(7737,18428,'#0d7df0'),(7738,18432,'#0096fb'),(7739,18433,'#FFFFFF'),(7740,18439,'#e135da'),(7741,18440,'#f43286'),(7742,18442,'#7d37de'),(7743,18445,'#d6fc35'),(7744,18447,'#da1534'),(7745,18448,'#c432e0'),(7746,18449,'#e062ef'),(7747,18451,'#2bde18'),(7748,18453,'#20bfe3'),(7749,18454,'#0467f1'),(7750,18458,'#cdbf54'),(7751,18459,'#68e50a'),(7752,18461,'#f6407b'),(7753,18463,'#8457da'),(7754,18464,'#02d36e'),(7755,18466,'#f76963'),(7756,18468,'#2023f9'),(7757,18469,'#c53dda'),(7758,18471,'#c76df0'),(7759,18472,'#45e0fa'),(7760,18474,'#29e73d'),(7761,18476,'#53acfc'),(7762,18477,'#d2167e'),(7763,18479,'#cd1a64'),(7764,18480,'#64cdea'),(7765,18483,'#0c67de'),(7766,18489,'#cf5dbe'),(7767,18490,'#0ff2c7'),(7768,18493,'#d6e80f'),(7769,18494,'#5a78d1'),(7770,18495,'#54bce9'),(7771,18498,'#e59d3c'),(7772,18502,'#d7bd37'),(7773,18503,'#7fdd38'),(7774,18504,'#0be6ae'),(7775,18506,'#d65b85'),(7776,18509,'#FFFFFF'),(7777,18510,'#e45d94'),(7778,18514,'#88eb14'),(7779,18515,'#5dd4a7'),(7780,18517,'#da466b'),(7781,18518,'#6fc9e9'),(7782,18524,'#e86205'),(7783,18526,'#71afea'),(7784,18527,'#eb3676'),(7785,18531,'#fe77d8'),(7786,18533,'#29d2ef'),(7787,18536,'#f20afa'),(7788,18539,'#d86b91'),(7789,18540,'#e8ea58'),(7790,18541,'#64b4db'),(7791,18549,'#7554ef'),(7792,18552,'#df8147'),(7793,18554,'#d7146f'),(7794,18555,'#85f818'),(7795,18556,'#6be719'),(7796,18557,'#8b31d9'),(7797,18558,'#f2de35'),(7798,18559,'#c908e1'),(7799,18561,'#dc4a48'),(7800,18564,'#e70c76'),(7801,18568,'#de9761'),(7802,18569,'#e47365'),(7803,18572,'#fb783d'),(7804,18575,'#7259e2'),(7805,18577,'#0e51e0'),(7806,18578,'#66dbb5'),(7807,18581,'#163ae1'),(7808,18583,'#ebcb1c'),(7809,18586,'#cc4f37'),(7810,18587,'#8008f4'),(7811,18590,'#26dd1e'),(7812,18596,'#0a39e9'),(7813,18609,'#7ee260'),(7814,18611,'#840cee'),(7815,18613,'#e1875a'),(7816,18617,'#0bd987'),(7817,18618,'#b9ee10'),(7818,18620,'#8feb64'),(7819,18622,'#3bc4d1'),(7820,18624,'#dcc028'),(7821,18625,'#cf0023'),(7822,18626,'#0b4be0'),(7823,18630,'#dd5a84'),(7824,18631,'#e29569'),(7825,18633,'#f30579'),(7826,18634,'#2792d7'),(7827,18640,'#50fde3'),(7828,18641,'#e52365'),(7829,18642,'#eb1bb5'),(7830,18645,'#3b9bcc'),(7831,18647,'#391edb'),(7832,18648,'#4b61d0'),(7833,18650,'#91cd61'),(7834,18651,'#0cdf53'),(7835,18652,'#26e47e'),(7836,18654,'#49d8d8'),(7837,18661,'#d8db30'),(7838,18662,'#30fcd3'),(7839,18663,'#e4b536'),(7840,18664,'#4acc05'),(7841,18666,'#6254e6'),(7842,18668,'#ccb94e'),(7843,18671,'#013ecf'),(7844,18672,'#d530b3'),(7845,18673,'#ce529e'),(7846,18677,'#db4d53'),(7847,18679,'#0be4dd'),(7848,18680,'#fc8e4b'),(7849,18683,'#31da7b'),(7850,18685,'#46dfdb'),(7851,18686,'#45e306'),(7852,18687,'#FFFFFF'),(7853,18688,'#FFFFFF'),(7854,18690,'#f3c061'),(7855,18694,'#41d960'),(7856,18695,'#f447db'),(7857,18701,'#cd6ce7'),(7858,18702,'#FFFFFF'),(7859,18705,'#83dc52'),(7860,18709,'#0dd407'),(7861,18712,'#59f363'),(7862,18713,'#0406ef'),(7863,18718,'#66d9de'),(7864,18723,'#6aa9de'),(7865,18728,'#f6f86e'),(7866,18729,'#1ce24c'),(7867,18730,'#bb2bf3'),(7868,18731,'#b2e050'),(7869,18746,'#d86327'),(7870,18755,'#09ee55'),(7871,18759,'#dcc74f'),(7872,18761,'#f6388f'),(7873,18790,'#aed567'),(7874,18796,'#37cd00'),(7875,18825,'#d06c58'),(7876,18838,'#66cc59'),(7877,18842,'#33e645'),(7878,18845,'#db1ff8'),(7879,18859,'#dce749'),(7880,18864,'#57acf3'),(7881,18876,'#FFFFFF'),(7882,18898,'#ce5097'),(7883,18901,'#0e9cd2'),(7884,18906,'#1ad51f'),(7885,18908,'#8beb3f'),(7886,18911,'#6d5ae0'),(7887,18913,'#60f497'),(7888,18923,'#a7e31c'),(7889,18924,'#f447c2'),(7890,18927,'#FFFFFF'),(7891,18938,'#2dd977'),(7892,18942,'#36eeb1'),(7893,18952,'#29f643'),(7894,18953,'#d3854d'),(7895,18957,'#9ce543'),(7896,18987,'#52debf'),(7897,18999,'#9635ec'),(7898,19012,'#e82f71'),(7899,19023,'#FFFFFF'),(7900,19032,'#5fd760'),(7901,19035,'#eec65f'),(7902,19040,'#ea3ddc'),(7903,19041,'#e4bd63'),(7904,19053,'#217cd6'),(7905,19056,'#3ef524'),(7906,19057,'#FFFFFF'),(7907,19059,'#45d439'),(7908,19060,'#d1ce56'),(7909,19065,'#fcd314'),(7910,19071,'#48d502'),(7911,19074,'#FFFFFF'),(7912,19075,'#17cd2f'),(7913,19076,'#8323df'),(7914,19077,'#d06c1f'),(7915,19092,'#37e440'),(7916,19094,'#FFFFFF'),(7917,19096,'#3815e8'),(7918,19098,'#fb03c3'),(7919,19102,'#9e40e7'),(7920,19105,'#dada30'),(7921,19109,'#f2da61'),(7922,19110,'#4bcc6a'),(7923,19113,'#85d330'),(7924,19118,'#62eff5'),(7925,19119,'#FFFFFF'),(7926,19120,'#51eff7'),(7927,19121,'#a621cf'),(7928,19124,'#3ef0c2'),(7929,19129,'#61e1e6'),(7930,19134,'#FFFFFF'),(7931,19140,'#df9950'),(7932,19142,'#e86dda'),(7933,19144,'#44f24c'),(7934,19158,'#d46e58'),(7935,19161,'#219dfa'),(7936,19168,'#d40d15'),(7937,19178,'#d200b4'),(7938,19180,'#d620fc'),(7939,19195,'#d98d36'),(7940,19196,'#9305d3'),(7941,19201,'#fc9a75'),(7942,19203,'#21e561'),(7943,19208,'#536ecf'),(7944,19209,'#f40605'),(7945,19215,'#60ccb3'),(7946,19216,'#7043d8'),(7947,19218,'#dae55b'),(7948,19219,'#e66a91'),(7949,19220,'#f73c58'),(7950,19221,'#791efd'),(7951,19224,'#da9e09'),(7952,19225,'#72bbfd'),(7953,19231,'#3845de'),(7954,19232,'#3abad6'),(7955,19245,'#FFFFFF'),(7956,19246,'#FFFFFF'),(7957,19247,'#edae2c'),(7958,19249,'#ce2f6c'),(7959,19253,'#499dda'),(7960,19261,'#dac515'),(7961,19274,'#a9d164'),(7962,19281,'#fc7607'),(7963,19282,'#FFFFFF'),(7964,19283,'#FFFFFF'),(7965,19284,'#2a90d1'),(7966,19285,'#FFFFFF'),(7967,19288,'#7865eb'),(7968,19293,'#e307e2'),(7969,19297,'#3a69e2'),(7970,19299,'#79dffd'),(7971,19305,'#FFFFFF'),(7972,19308,'#FFFFFF'),(7973,19309,'#FFFFFF'),(7974,19310,'#FFFFFF'),(7975,19313,'#FFFFFF'),(7976,19314,'#FFFFFF'),(7977,19321,'#df6ee0'),(7978,19327,'#08ec33'),(7979,19330,'#FFFFFF'),(7980,19335,'#FFFFFF'),(7981,19340,'#FFFFFF'),(7982,19341,'#fbe86c'),(7983,19351,'#FFFFFF'),(7984,19353,'#FFFFFF'),(7985,19361,'#FFFFFF'),(7986,19363,'#FFFFFF'),(7987,19364,'#0bfadd'),(7988,19366,'#f52c68'),(7989,19369,'#d044f3'),(7990,19370,'#FFFFFF'),(7991,19371,'#FFFFFF'),(7992,19372,'#FFFFFF'),(7993,19376,'#FFFFFF'),(7994,19381,'#af47d7'),(7995,19384,'#FFFFFF'),(7996,19386,'#FFFFFF'),(7997,19387,'#FFFFFF'),(7998,19390,'#FFFFFF'),(7999,19393,'#FFFFFF'),(8000,19395,'#a1f56b'),(8001,19398,'#FFFFFF'),(8002,19409,'#FFFFFF'),(8003,19411,'#FFFFFF'),(8004,19412,'#FFFFFF'),(8005,19413,'#FFFFFF'),(8006,19414,'#FFFFFF'),(8007,19423,'#FFFFFF'),(8008,19424,'#FFFFFF'),(8009,19430,'#FFFFFF'),(8010,19431,'#FFFFFF'),(8011,19432,'#64e66f'),(8012,19437,'#FFFFFF'),(8013,19438,'#FFFFFF'),(8014,19444,'#8a3ecc'),(8015,19445,'#75ed51'),(8016,19446,'#29dc67'),(8017,19448,'#FFFFFF'),(8018,19451,'#FFFFFF'),(8019,19469,'#d75c06'),(8020,19470,'#5ce1be'),(8021,19471,'#dd23e5'),(8022,19472,'#FFFFFF'),(8023,19478,'#6ce8d5'),(8024,19480,'#FFFFFF'),(8025,19481,'#FFFFFF'),(8026,19482,'#FFFFFF'),(8027,19483,'#FFFFFF'),(8028,19484,'#FFFFFF'),(8029,19493,'#d2515d'),(8030,19494,'#FFFFFF'),(8031,19495,'#56d90b'),(8032,19496,'#FFFFFF'),(8033,19497,'#FFFFFF'),(8034,19498,'#FFFFFF'),(8035,19500,'#77fca2'),(8036,19501,'#FFFFFF'),(8037,19504,'#9412f4'),(8038,19509,'#fa8e04'),(8039,19512,'#FFFFFF'),(8040,19514,'#FFFFFF'),(8041,19518,'#FFFFFF'),(8042,19521,'#FFFFFF'),(8043,19522,'#FFFFFF'),(8044,19523,'#FFFFFF'),(8045,19524,'#eb8f08'),(8046,19525,'#FFFFFF'),(8047,19526,'#a540cc'),(8048,19530,'#FFFFFF'),(8049,19531,'#FFFFFF'),(8050,19532,'#FFFFFF'),(8051,19533,'#FFFFFF'),(8052,19534,'#4a85dc'),(8053,19546,'#9864f5'),(8054,19551,'#e8b567'),(8055,19553,'#d28d40'),(8056,19558,'#59df3e'),(8057,19565,'#f053f7'),(8058,19572,'#da6498'),(8059,19579,'#24b7d1'),(8060,19581,'#f0b021'),(8061,19593,'#f6589c'),(8062,19597,'#d64635'),(8063,19599,'#f36bfd'),(8064,19600,'#9b59f5'),(8065,19602,'#ec6bc8'),(8066,19603,'#08c3e5'),(8067,19605,'#6417d5'),(8068,19606,'#db72f4'),(8069,19609,'#f570b0'),(8070,19610,'#8b3cde'),(8071,19612,'#da1340'),(8072,19617,'#4b71f9'),(8073,19618,'#39afe8'),(8074,19619,'#2766d5'),(8075,19620,'#9743e1'),(8076,19623,'#06f71f'),(8077,19624,'#f08c2d'),(8078,19627,'#d3525f'),(8079,19630,'#d2c007'),(8080,19635,'#5ee9e9'),(8081,19637,'#13d0e0'),(8082,19639,'#62b8db'),(8083,19644,'#bd55f2'),(8084,19647,'#00cf46'),(8085,19648,'#af5fe5'),(8086,19650,'#faa172'),(8087,19653,'#755cfb'),(8088,19655,'#3cfbe7'),(8089,19656,'#268eee'),(8090,19659,'#3be8ba'),(8091,19661,'#40dbf7'),(8092,19662,'#04fd8d'),(8093,19663,'#3851f8'),(8094,19677,'#23dd04'),(8095,19681,'#73ec4b'),(8096,19685,'#ed9b5c'),(8097,19686,'#71fd7b'),(8098,19690,'#FFFFFF'),(8099,19691,'#4e6ddc'),(8100,19692,'#786eec'),(8101,19695,'#e82e55'),(8102,19699,'#7c5cf0'),(8103,19701,'#4e8bfb'),(8104,19705,'#f4aa4a'),(8105,19706,'#d1159a'),(8106,19708,'#dd12ed'),(8107,19714,'#63c5d7'),(8108,19716,'#db2f44'),(8109,19717,'#79e71c'),(8110,19718,'#d82666'),(8111,19719,'#1fd72b'),(8112,19720,'#685ffc'),(8113,19725,'#d5623d'),(8114,19726,'#f438d6'),(8115,19739,'#07d7ed'),(8116,19750,'#d2b524'),(8117,19754,'#345ddb'),(8118,19756,'#a3f505'),(8119,19757,'#54df4f'),(8120,19760,'#e7aa57'),(8121,19764,'#f4ca1f'),(8122,19765,'#c217e8'),(8123,19766,'#262ad1'),(8124,19768,'#f1556a'),(8125,19769,'#397df2'),(8126,19777,'#4249cc'),(8127,19782,'#df8468'),(8128,19790,'#d15d38'),(8129,19791,'#b8eb5f'),(8130,19794,'#09afe1'),(8131,19797,'#eaab58'),(8132,19798,'#1b18e6'),(8133,19799,'#99fb02'),(8134,19801,'#e0bb48'),(8135,19817,'#b15dd0'),(8136,19818,'#47fdb0'),(8137,19820,'#49cd64'),(8138,19822,'#fcf119'),(8139,19824,'#10d55f'),(8140,19825,'#58f260'),(8141,19830,'#5897d5'),(8142,19831,'#0b43d7'),(8143,19836,'#6ad7c9'),(8144,19837,'#8449d8'),(8145,19839,'#4fea0c'),(8146,19842,'#cc9f23'),(8147,19845,'#32cfe3'),(8148,19850,'#cbe528'),(8149,19864,'#e41385'),(8150,19866,'#ea7590'),(8151,19867,'#51e1e0'),(8152,19872,'#12d926'),(8153,19874,'#dc1a32'),(8154,19875,'#57bcf2'),(8155,19876,'#cc5dc7'),(8156,19878,'#6bfa8e'),(8157,19885,'#ed7776'),(8158,19886,'#db6506'),(8159,19887,'#fa554d'),(8160,19889,'#d73a7d'),(8161,19899,'#d0871e'),(8162,19901,'#e1e065'),(8163,19902,'#eb2909'),(8164,19903,'#cf6c4b'),(8165,19911,'#cd01af'),(8166,19924,'#fc19f2'),(8167,19926,'#2dbbef'),(8168,19928,'#f038c0'),(8169,19929,'#02f850'),(8170,19933,'#557cfe'),(8171,19934,'#7dcd4f'),(8172,19936,'#0773ec'),(8173,19950,'#3968e8'),(8174,19954,'#1318e6'),(8175,19981,'#FFFFFF'),(8176,19992,'#e13143'),(8177,20015,'#42d42e'),(8178,20020,'#4ebdf8'),(8179,20031,'#e88f22'),(8180,20051,'#37da4f'),(8181,20055,'#22dc59'),(8182,20059,'#91f45c'),(8183,20060,'#de089c'),(8184,20061,'#FFFFFF'),(8185,20062,'#f97935'),(8186,20063,'#cdb14a'),(8187,20067,'#58dca0'),(8188,20071,'#14e020'),(8189,20073,'#FFFFFF'),(8190,20074,'#e9446f'),(8191,20076,'#59dff1'),(8192,20077,'#81f118'),(8193,20079,'#64f7fc'),(8194,20082,'#FFFFFF'),(8195,20083,'#3aa3d4'),(8196,20084,'#36d182'),(8197,20086,'#8a22f1'),(8198,20088,'#50d13a'),(8199,20089,'#FFFFFF'),(8200,20092,'#af2adf'),(8201,20093,'#0ffe65'),(8202,20095,'#3cd192'),(8203,20097,'#f1af58'),(8204,20099,'#fbd22b'),(8205,20100,'#FFFFFF'),(8206,20101,'#dcba30'),(8207,20106,'#65afdc'),(8208,20109,'#c458d3'),(8209,20110,'#63fd8b'),(8210,20112,'#ce300a'),(8211,20113,'#0ed024'),(8212,20114,'#fcb079'),(8213,20116,'#732ed4'),(8214,20117,'#d31651'),(8215,20120,'#fe2407'),(8216,20121,'#6d12cf'),(8217,20122,'#1c36fe'),(8218,20125,'#fe36b8'),(8219,20127,'#7ceb0a'),(8220,20130,'#65b5d0'),(8221,20132,'#47f20c'),(8222,20133,'#45f6cc'),(8223,20134,'#0c6ce9'),(8224,20137,'#ce394d'),(8225,20139,'#902cdd'),(8226,20142,'#83dc5a'),(8227,20143,'#de7f05'),(8228,20145,'#8aee29'),(8229,20146,'#15f590'),(8230,20147,'#89ee37'),(8231,20148,'#486fdb'),(8232,20149,'#9fe639'),(8233,20151,'#fa778e'),(8234,20153,'#8ffa66'),(8235,20156,'#4bd176'),(8236,20158,'#FFFFFF'),(8237,20160,'#de0b03'),(8238,20162,'#FFFFFF'),(8239,20166,'#FFFFFF'),(8240,20167,'#89da0c'),(8241,20168,'#3d3ecf'),(8242,20174,'#a8d05a'),(8243,20175,'#dd0842'),(8244,20177,'#33d77f'),(8245,20178,'#7858d0'),(8246,20179,'#54a2d4'),(8247,20183,'#765bd6'),(8248,20185,'#fcd050'),(8249,20186,'#9df243'),(8250,20187,'#a144e8'),(8251,20188,'#f6c32c'),(8252,20189,'#5cd2c2'),(8253,20190,'#e328b1'),(8254,20191,'#d4677b'),(8255,20192,'#d9aa35'),(8256,20194,'#44e833'),(8257,20196,'#1ee495'),(8258,20198,'#f50da4'),(8259,20199,'#f02123'),(8260,20200,'#d0d259'),(8261,20203,'#d86668'),(8262,20204,'#3c84ce'),(8263,20205,'#3f69d5'),(8264,20208,'#d7ab4b'),(8265,20209,'#98da20'),(8266,20210,'#a30eed'),(8267,20212,'#10cd03'),(8268,20213,'#3466cf'),(8269,20214,'#34c4f3'),(8270,20215,'#52e2f0'),(8271,20218,'#e83845'),(8272,20219,'#e26fcc'),(8273,20222,'#56d2f0'),(8274,20225,'#1a31f7'),(8275,20226,'#f7e64a'),(8276,20229,'#d92689'),(8277,20232,'#289ddb'),(8278,20235,'#13d994'),(8279,20236,'#e5412c'),(8280,20238,'#bddb04'),(8281,20239,'#d007ae'),(8282,20240,'#44b7d0'),(8283,20241,'#faaa1a'),(8284,20242,'#FFFFFF'),(8285,20243,'#FFFFFF'),(8286,20245,'#2b74fb'),(8287,20246,'#FFFFFF'),(8288,20249,'#d151f2'),(8289,20250,'#c6d05b'),(8290,20254,'#f00e73'),(8291,20256,'#ce4301'),(8292,20257,'#1ee7e5'),(8293,20262,'#5e24de'),(8294,20263,'#e51794'),(8295,20268,'#d31763'),(8296,20269,'#20ce63'),(8297,20271,'#1a5ccf'),(8298,20274,'#ce1925'),(8299,20275,'#419ccf'),(8300,20276,'#57d365'),(8301,20277,'#ed169e'),(8302,20280,'#09d040'),(8303,20283,'#FFFFFF'),(8304,20284,'#49daa8'),(8305,20285,'#eb72d4'),(8306,20286,'#902ef6'),(8307,20288,'#687bd5'),(8308,20289,'#cd3270'),(8309,20290,'#60cdf3'),(8310,20294,'#05cc05'),(8311,20295,'#d50635'),(8312,20297,'#17d16e'),(8313,20300,'#e813b6'),(8314,20301,'#f7ad36'),(8315,20302,'#FFFFFF'),(8316,20303,'#0a2ce5'),(8317,20305,'#1be2fd'),(8318,20306,'#FFFFFF'),(8319,20307,'#d25a67'),(8320,20308,'#c4ec5a'),(8321,20309,'#e22ad0'),(8322,20311,'#26d344'),(8323,20312,'#1fa3e5'),(8324,20314,'#bd73e8'),(8325,20315,'#d6214d'),(8326,20316,'#64c9e5'),(8327,20320,'#58a8f7'),(8328,20324,'#6ff038'),(8329,20325,'#f5f205'),(8330,20326,'#2d3ad3'),(8331,20327,'#ecd468'),(8332,20328,'#13fd61'),(8333,20329,'#974dec'),(8334,20330,'#e10c3f'),(8335,20332,'#65dae5'),(8336,20333,'#49de84'),(8337,20335,'#ea52e2'),(8338,20336,'#251df2'),(8339,20337,'#2646ef'),(8340,20338,'#23e4f8'),(8341,20339,'#d86feb'),(8342,20340,'#dd4e6a'),(8343,20342,'#3baad0'),(8344,20344,'#740ee2'),(8345,20346,'#b912ed'),(8346,20347,'#cc2bc2'),(8347,20348,'#da7058'),(8348,20350,'#FFFFFF'),(8349,20351,'#f82f3d'),(8350,20352,'#790fcf'),(8351,20356,'#50b1f3'),(8352,20357,'#588bfb'),(8353,20358,'#FFFFFF'),(8354,20359,'#a933ee'),(8355,20360,'#FFFFFF'),(8356,20364,'#b359df'),(8357,20368,'#cfc71e'),(8358,20369,'#d3a029'),(8359,20371,'#FFFFFF'),(8360,20375,'#FFFFFF'),(8361,20376,'#FFFFFF'),(8362,20377,'#2dea7d'),(8363,20380,'#96fe5f'),(8364,20382,'#0723fa'),(8365,20383,'#aa42e8'),(8366,20384,'#7b72fd'),(8367,20385,'#e672f3'),(8368,20387,'#628cfc'),(8369,20388,'#c9ea43'),(8370,20389,'#9ee962'),(8371,20393,'#22e665'),(8372,20396,'#f8403e'),(8373,20397,'#cee335'),(8374,20398,'#518de1'),(8375,20399,'#ce2b01'),(8376,20402,'#FFFFFF'),(8377,20405,'#eaf96f'),(8378,20406,'#2efe57'),(8379,20408,'#32df38'),(8380,20410,'#9b07eb'),(8381,20412,'#e99a43'),(8382,20413,'#4333f3'),(8383,20414,'#410be3'),(8384,20415,'#dc736d'),(8385,20416,'#ac6de9'),(8386,20419,'#548ef4'),(8387,20421,'#6805d1'),(8388,20422,'#cf7d63'),(8389,20425,'#c815e6'),(8390,20426,'#1809d8'),(8391,20427,'#FFFFFF'),(8392,20430,'#de403e'),(8393,20433,'#4ee867'),(8394,20434,'#43e7c7'),(8395,20436,'#65fe24'),(8396,20437,'#fd161c'),(8397,20438,'#79fef2'),(8398,20440,'#b15cdc'),(8399,20442,'#44e0bc'),(8400,20443,'#f014c2'),(8401,20450,'#e1715d'),(8402,20451,'#7b54d0'),(8403,20453,'#73ec65'),(8404,20455,'#3fd142'),(8405,20456,'#3e13e2'),(8406,20458,'#d5801c'),(8407,20459,'#c8ea67'),(8408,20460,'#cc8854'),(8409,20463,'#d65bd5'),(8410,20464,'#5de0c7'),(8411,20465,'#c5d765'),(8412,20466,'#dcef74'),(8413,20469,'#cfa85b'),(8414,20471,'#6449f8'),(8415,20472,'#702fe9'),(8416,20474,'#9d28f5'),(8417,20478,'#FFFFFF'),(8418,20479,'#95f103'),(8419,20480,'#37b7dd'),(8420,20481,'#da8812'),(8421,20482,'#6f31e2'),(8422,20486,'#FFFFFF'),(8423,20489,'#FFFFFF'),(8424,20490,'#FFFFFF'),(8425,20492,'#FFFFFF'),(8426,20496,'#FFFFFF'),(8427,20498,'#FFFFFF'),(8428,20501,'#FFFFFF'),(8429,20502,'#36a8d9'),(8430,20505,'#FFFFFF'),(8431,20507,'#72e70c'),(8432,20508,'#a05fd4'),(8433,20510,'#56a6dc'),(8434,20514,'#b7e349'),(8435,20515,'#6ce93f'),(8436,20517,'#3068ec'),(8437,20520,'#dcc054'),(8438,20521,'#FFFFFF'),(8439,20530,'#ea2950'),(8440,20531,'#d5d32b'),(8441,20535,'#FFFFFF'),(8442,20536,'#FFFFFF'),(8443,20537,'#FFFFFF'),(8444,20543,'#e067d9'),(8445,20546,'#FFFFFF'),(8446,20547,'#FFFFFF'),(8447,20556,'#d93f47'),(8448,20566,'#8e2bfb'),(8449,20567,'#FFFFFF'),(8450,20568,'#b802d6'),(8451,20571,'#fc47df'),(8452,20577,'#9348d4'),(8453,20578,'#3c2ffc'),(8454,20583,'#FFFFFF'),(8455,20586,'#6564f2'),(8456,20588,'#fb2620'),(8457,20591,'#FFFFFF'),(8458,20595,'#e60c8d'),(8459,20596,'#FFFFFF'),(8460,20597,'#ee6cfb'),(8461,20604,'#FFFFFF'),(8462,20605,'#7bfd3f'),(8463,20607,'#FFFFFF'),(8464,20611,'#FFFFFF'),(8465,20618,'#099bd4'),(8466,20621,'#f97097'),(8467,20625,'#d2411b'),(8468,20643,'#7d04dc'),(8469,20648,'#0e90df'),(8470,20650,'#271adc'),(8471,20654,'#FFFFFF'),(8472,20662,'#a40ad8'),(8473,20663,'#e301c3'),(8474,20665,'#42d62b'),(8475,20668,'#14eeaf'),(8476,20670,'#6861dc'),(8477,20673,'#b315d2'),(8478,20674,'#dc3667'),(8479,20678,'#fc1ebb'),(8480,20679,'#FFFFFF'),(8481,20682,'#9635ec'),(8482,20690,'#f93f09'),(8483,20691,'#FFFFFF'),(8484,20696,'#FFFFFF'),(8485,20697,'#FFFFFF'),(8486,20701,'#FFFFFF'),(8487,20706,'#FFFFFF'),(8488,20707,'#c8ec5a'),(8489,20708,'#30d461'),(8490,20710,'#FFFFFF'),(8491,20711,'#FFFFFF'),(8492,20713,'#FFFFFF'),(8493,20722,'#92fc78'),(8494,20723,'#87d93b'),(8495,20724,'#cf4088'),(8496,20726,'#88e76a'),(8497,20731,'#ee921d'),(8498,20732,'#a1d15c'),(8499,20735,'#f048df'),(8500,20742,'#30d509'),(8501,20743,'#a21aed'),(8502,20745,'#d14e95'),(8503,20747,'#2e41ce'),(8504,20749,'#55baf9'),(8505,20750,'#da122d'),(8506,20766,'#e13372'),(8507,20782,'#fd3f85'),(8508,20783,'#d158b2'),(8509,20784,'#68e570'),(8510,20790,'#201fe7'),(8511,20797,'#6cdd30'),(8512,20800,'#33d8b0'),(8513,20804,'#38e8e1'),(8514,20807,'#da8761'),(8515,20811,'#f627ad'),(8516,20812,'#02ef40'),(8517,20814,'#ef272b'),(8518,20815,'#dc61e1'),(8519,20818,'#ed8d76'),(8520,20820,'#cf2b23'),(8521,20821,'#f7ea12'),(8522,20828,'#e862b6'),(8523,20829,'#bb2bfa'),(8524,20830,'#e17af7'),(8525,20832,'#64f702'),(8526,20833,'#4de9d4'),(8527,20835,'#6d5deb'),(8528,20837,'#58ec72'),(8529,20840,'#4559d5'),(8530,20844,'#65f0df'),(8531,20846,'#d0cb5f'),(8532,20847,'#032cd2'),(8533,20850,'#4bf205'),(8534,20851,'#aee64a'),(8535,20852,'#58c4e6'),(8536,20853,'#ad5bd0'),(8537,20854,'#612acf'),(8538,20858,'#f2d733'),(8539,20861,'#e264da'),(8540,20863,'#cc1ac7'),(8541,20868,'#45e7d7'),(8542,20869,'#e85c5c'),(8543,20871,'#6773fe'),(8544,20874,'#d0e64b'),(8545,20876,'#eb5bd3'),(8546,20877,'#2055cd'),(8547,20881,'#FFFFFF'),(8548,20882,'#579bf5'),(8549,20887,'#9d74ec'),(8550,20891,'#04f6e3'),(8551,20895,'#e45792'),(8552,20905,'#4d77ea'),(8553,20912,'#27ee17'),(8554,20913,'#e87af4'),(8555,20921,'#f72b40'),(8556,20922,'#1ed262'),(8557,20923,'#41f7f8'),(8558,20926,'#4850e4'),(8559,20931,'#ef74bb'),(8560,20935,'#ed9056'),(8561,20936,'#f3a96a'),(8562,20940,'#fb00d6'),(8563,20945,'#16de69'),(8564,20948,'#15dbb9'),(8565,20958,'#c4de59'),(8566,20960,'#de45c4'),(8567,20964,'#5954dd'),(8568,20970,'#5c28ef'),(8569,20971,'#a924d4'),(8570,20972,'#fb36fd'),(8571,20975,'#741ffd'),(8572,20976,'#74ee38'),(8573,20978,'#22b5e3'),(8574,20979,'#e143b7'),(8575,20984,'#8a3ecd'),(8576,20985,'#ce3809'),(8577,20986,'#21e234'),(8578,20991,'#a9ee46'),(8579,20992,'#4e5ad6'),(8580,20994,'#FFFFFF'),(8581,20997,'#4a11dc'),(8582,20998,'#fdcc63'),(8583,21003,'#ee1ea6'),(8584,21009,'#3f94d8'),(8585,21014,'#03f6a2'),(8586,21016,'#e4de2d'),(8587,21017,'#9ef930'),(8588,21018,'#084cda'),(8589,21022,'#53a2f9'),(8590,21024,'#fec377'),(8591,21029,'#df5275'),(8592,21036,'#d5c35c'),(8593,21038,'#1525fa'),(8594,21043,'#d5036c'),(8595,21045,'#85d24b'),(8596,21048,'#f85954'),(8597,21049,'#3816e3'),(8598,21052,'#57d776'),(8599,21063,'#FFFFFF'),(8600,21064,'#3c9dd9'),(8601,21066,'#da1e65'),(8602,21068,'#9aef57'),(8603,21071,'#FFFFFF'),(8604,21074,'#00f0f9'),(8605,21076,'#97fa43'),(8606,21079,'#f2118e'),(8607,21080,'#FFFFFF'),(8608,21081,'#21e280'),(8609,21083,'#83d25b'),(8610,21087,'#03ce02'),(8611,21089,'#61df89'),(8612,21090,'#f81570'),(8613,21094,'#84e46c'),(8614,21095,'#8913d3'),(8615,21097,'#3bcfc1'),(8616,21099,'#e676ed'),(8617,21100,'#0042e9'),(8618,21102,'#cf0e35'),(8619,21103,'#9b09cc'),(8620,21105,'#67fae6'),(8621,21107,'#cc4720'),(8622,21109,'#dee966'),(8623,21113,'#de096a'),(8624,21119,'#e43c49'),(8625,21120,'#FFFFFF'),(8626,21121,'#da0341'),(8627,21125,'#61d3aa'),(8628,21128,'#d64796'),(8629,21129,'#FFFFFF'),(8630,21132,'#FFFFFF'),(8631,21140,'#0f69dd'),(8632,21142,'#7cfb6a'),(8633,21143,'#d908d3'),(8634,21144,'#FFFFFF'),(8635,21147,'#df3a2e'),(8636,21148,'#e4ba1f'),(8637,21149,'#FFFFFF'),(8638,21152,'#5cdddb'),(8639,21153,'#79d2f7'),(8640,21154,'#239fef'),(8641,21156,'#ea2eb6'),(8642,21157,'#ee6390'),(8643,21158,'#2742fb'),(8644,21160,'#FFFFFF'),(8645,21161,'#cc0a79'),(8646,21162,'#FFFFFF'),(8647,21163,'#de599b'),(8648,21164,'#14b3de'),(8649,21166,'#fdd70d'),(8650,21167,'#fee337'),(8651,21168,'#FFFFFF'),(8652,21169,'#f326ba'),(8653,21171,'#dc9945'),(8654,21173,'#67e193'),(8655,21174,'#FFFFFF'),(8656,21175,'#f25cad'),(8657,21176,'#e529ec'),(8658,21177,'#91f71f'),(8659,21179,'#40e0c3'),(8660,21184,'#23f486'),(8661,21185,'#dfc956'),(8662,21186,'#413ee3'),(8663,21187,'#FFFFFF'),(8664,21188,'#18ade6'),(8665,21189,'#ee3620'),(8666,21192,'#fa5992'),(8667,21194,'#FFFFFF'),(8668,21196,'#3d35d6'),(8669,21197,'#e164c2'),(8670,21199,'#dc3c51'),(8671,21200,'#5be11d'),(8672,21206,'#c760e1'),(8673,21207,'#FFFFFF'),(8674,21208,'#0560dc'),(8675,21209,'#4597dc'),(8676,21211,'#ce6364'),(8677,21212,'#c642d1'),(8678,21213,'#c3d706'),(8679,21214,'#2389fe'),(8680,21215,'#6de126'),(8681,21216,'#ce7e15'),(8682,21217,'#07d5ef'),(8683,21220,'#a969d4'),(8684,21221,'#faf412'),(8685,21222,'#d13f35'),(8686,21223,'#3dcd87'),(8687,21224,'#15d7c4'),(8688,21225,'#fe773b'),(8689,21226,'#e7719e'),(8690,21231,'#fd823c'),(8691,21232,'#fb6223'),(8692,21233,'#4d09eb'),(8693,21234,'#50e5bd'),(8694,21235,'#47a1d8'),(8695,21239,'#4c8ee3'),(8696,21240,'#9acf08'),(8697,21241,'#43ebc5'),(8698,21242,'#345ed6'),(8699,21243,'#d75d60'),(8700,21245,'#7029d8'),(8701,21246,'#097bef'),(8702,21247,'#FFFFFF'),(8703,21249,'#f8101c'),(8704,21253,'#69cc49'),(8705,21254,'#a04be5'),(8706,21257,'#d841d5'),(8707,21258,'#4731ea'),(8708,21262,'#fc51a5'),(8709,21263,'#532dda'),(8710,21264,'#d1a65a'),(8711,21265,'#4ea4cc'),(8712,21266,'#d847ca'),(8713,21267,'#f37b51'),(8714,21269,'#3cd11e'),(8715,21270,'#6bbbd8'),(8716,21271,'#FFFFFF'),(8717,21272,'#d946f0'),(8718,21273,'#ced41d'),(8719,21278,'#c61bf3'),(8720,21279,'#d53a36'),(8721,21280,'#fea479'),(8722,21281,'#9459d6'),(8723,21284,'#49bbce'),(8724,21285,'#FFFFFF'),(8725,21286,'#669ddd'),(8726,21287,'#47f700'),(8727,21288,'#c5e82d'),(8728,21290,'#ce0e6b'),(8729,21291,'#FFFFFF'),(8730,21292,'#f64f23'),(8731,21294,'#FFFFFF'),(8732,21295,'#e538eb'),(8733,21296,'#48d172'),(8734,21297,'#f8ed48'),(8735,21299,'#2ef299'),(8736,21300,'#b51ef2'),(8737,21301,'#085af8'),(8738,21302,'#b065d2'),(8739,21303,'#dd9c61'),(8740,21307,'#FFFFFF'),(8741,21309,'#32b2dc'),(8742,21312,'#FFFFFF'),(8743,21314,'#eb0471'),(8744,21315,'#52f4da'),(8745,21316,'#b3f879'),(8746,21317,'#eb6285'),(8747,21318,'#f6963b'),(8748,21319,'#75f17d'),(8749,21321,'#f3e942'),(8750,21322,'#d96637'),(8751,21323,'#25d83a'),(8752,21324,'#3666da'),(8753,21327,'#99e83f'),(8754,21328,'#df6ace'),(8755,21330,'#1ed0ce'),(8756,21331,'#f358da'),(8757,21332,'#e02f92'),(8758,21333,'#04f995'),(8759,21334,'#82e80e'),(8760,21335,'#9508ec'),(8761,21338,'#8df309'),(8762,21339,'#69fb72'),(8763,21340,'#edf76e'),(8764,21342,'#ec2654'),(8765,21344,'#FFFFFF'),(8766,21345,'#883bfd'),(8767,21346,'#e57560'),(8768,21347,'#4bd94a'),(8769,21348,'#d65fdf'),(8770,21351,'#294ae3'),(8771,21352,'#da4f9c'),(8772,21353,'#36f930'),(8773,21354,'#2fdc77'),(8774,21355,'#0d65de'),(8775,21356,'#da360b'),(8776,21358,'#3677cd'),(8777,21360,'#5ed7b0'),(8778,21361,'#a0d90a'),(8779,21362,'#44f2b7'),(8780,21363,'#e84884'),(8781,21366,'#eb7587'),(8782,21367,'#d89453'),(8783,21369,'#fb750e'),(8784,21372,'#FFFFFF'),(8785,21373,'#f06514'),(8786,21374,'#59d1ca'),(8787,21378,'#2b72fe'),(8788,21380,'#FFFFFF'),(8789,21381,'#fcb308'),(8790,21382,'#df3947'),(8791,21383,'#d4b942'),(8792,21385,'#d14686'),(8793,21389,'#dc16cf'),(8794,21391,'#7ef323'),(8795,21392,'#4eecea'),(8796,21393,'#55fc5c'),(8797,21394,'#28dbfd'),(8798,21395,'#bcdb21'),(8799,21396,'#df6878'),(8800,21397,'#90f339'),(8801,21400,'#dbd96b'),(8802,21401,'#a124fb'),(8803,21402,'#37a2f5'),(8804,21403,'#6cfc66'),(8805,21404,'#5faed6'),(8806,21405,'#8f50d9'),(8807,21406,'#a1d607'),(8808,21408,'#6bee3f'),(8809,21413,'#f38c42'),(8810,21414,'#dd23a6'),(8811,21415,'#e10d2f'),(8812,21417,'#30cc77'),(8813,21418,'#06d4a0'),(8814,21419,'#31e81d'),(8815,21422,'#ad56ea'),(8816,21423,'#2946f3'),(8817,21424,'#14e7f5'),(8818,21433,'#d55742'),(8819,21435,'#7a28d7'),(8820,21436,'#4d12eb'),(8821,21438,'#e361fb'),(8822,21440,'#56cf42'),(8823,21444,'#6317f1'),(8824,21445,'#dd176a'),(8825,21446,'#2029e1'),(8826,21456,'#d9bf2b'),(8827,21457,'#d4ba36'),(8828,21458,'#96fd05'),(8829,21461,'#66dbf4'),(8830,21465,'#fce674'),(8831,21467,'#edde10'),(8832,21474,'#d6c20b'),(8833,21476,'#eaf05f'),(8834,21481,'#0220ec'),(8835,21482,'#d055bf'),(8836,21486,'#64a2e3'),(8837,21487,'#e01ed2'),(8838,21494,'#09ce5b'),(8839,21497,'#7c47fc'),(8840,21498,'#ce3739'),(8841,21499,'#5ef18b'),(8842,21503,'#af15e4'),(8843,21504,'#4cdc38'),(8844,21513,'#69d98c'),(8845,21514,'#e738fe'),(8846,21515,'#9f49d2'),(8847,21516,'#254fd4'),(8848,21518,'#da6573'),(8849,21524,'#f84697'),(8850,21525,'#f12cee'),(8851,21526,'#78f1ab'),(8852,21527,'#3338ea'),(8853,21529,'#e05bf4'),(8854,21530,'#1942f1'),(8855,21535,'#ae32f1'),(8856,21539,'#cafa7b'),(8857,21540,'#47e22f'),(8858,21541,'#258adb'),(8859,21542,'#d80132'),(8860,21545,'#03d666'),(8861,21546,'#06f0e2'),(8862,21552,'#6a6ed9'),(8863,21557,'#d4cc1a'),(8864,21558,'#f37d73'),(8865,21561,'#3db3e5'),(8866,21563,'#a6f519'),(8867,21569,'#fd0363'),(8868,21570,'#291ddf'),(8869,21573,'#186fd4'),(8870,21580,'#7284f4'),(8871,21584,'#d7795d'),(8872,21591,'#88d536'),(8873,21593,'#f7d10a'),(8874,21596,'#b1e05a'),(8875,21599,'#43cf9b'),(8876,21600,'#f6702b'),(8877,21602,'#4498e9'),(8878,21604,'#8f49d2'),(8879,21606,'#29d1dd'),(8880,21610,'#efcd24'),(8881,21618,'#914bd2'),(8882,21619,'#9dde38'),(8883,21621,'#51e36e'),(8884,21622,'#d10394'),(8885,21624,'#41fb41'),(8886,21625,'#FFFFFF'),(8887,21626,'#c2dc4e'),(8888,21627,'#78ed68'),(8889,21629,'#6d93fd'),(8890,21634,'#4431da'),(8891,21640,'#3df009'),(8892,21648,'#2db7d0'),(8893,21649,'#2592cc'),(8894,21650,'#37e3a1'),(8895,21652,'#ce6c0b'),(8896,21653,'#fcaf64'),(8897,21654,'#d72cdc'),(8898,21655,'#5947e7'),(8899,21656,'#6adcc9'),(8900,21659,'#d649d9'),(8901,21660,'#42f3f4'),(8902,21661,'#43ef9e'),(8903,21662,'#77f2ca'),(8904,21664,'#4e84d3'),(8905,21665,'#17c4f3'),(8906,21669,'#1163fa'),(8907,21671,'#402ee0'),(8908,21672,'#d200cd'),(8909,21673,'#fd5d90'),(8910,21674,'#e5a03e'),(8911,21676,'#e9ba26'),(8912,21678,'#8466d4'),(8913,21679,'#4c82f0'),(8914,21680,'#4eeb25'),(8915,21682,'#f93bb7'),(8916,21684,'#1b9be3'),(8917,21685,'#70e5c7'),(8918,21687,'#f80b1a'),(8919,21688,'#FFFFFF'),(8920,21689,'#2e26fe'),(8921,21690,'#68fdc5'),(8922,21692,'#e2bf6e'),(8923,21694,'#4ef221'),(8924,21695,'#3bde7e'),(8925,21698,'#e668bf'),(8926,21705,'#f3f624'),(8927,21707,'#f1288c'),(8928,21708,'#6838e1'),(8929,21711,'#35ef90'),(8930,21714,'#27d103'),(8931,21715,'#2812e0'),(8932,21717,'#cc306d'),(8933,21718,'#3ee8ae'),(8934,21719,'#e7b057'),(8935,21721,'#cca53e'),(8936,21726,'#70e0b5'),(8937,21727,'#0da3f7'),(8938,21730,'#30cddf'),(8939,21736,'#70eec5'),(8940,21738,'#ce4dd4'),(8941,21742,'#a0f312'),(8942,21743,'#e91ae4'),(8943,21746,'#fe7c70'),(8944,21749,'#ad4ae1'),(8945,21750,'#f86572'),(8946,21751,'#21a2cf'),(8947,21752,'#73a6fb'),(8948,21753,'#dc970b'),(8949,21756,'#fb2032'),(8950,21758,'#33d6fb'),(8951,21759,'#f0d047'),(8952,21760,'#e74081'),(8953,21763,'#1503da'),(8954,21767,'#bfdf01'),(8955,21768,'#d015dd'),(8956,21770,'#d5458a'),(8957,21772,'#cc067f'),(8958,21773,'#2e2eee'),(8959,21774,'#d920eb'),(8960,21784,'#6166eb'),(8961,21785,'#7f50d4'),(8962,21792,'#79e157'),(8963,21793,'#dd64b1'),(8964,21798,'#3c81e6'),(8965,21804,'#46e7b9'),(8966,21806,'#9c4edd'),(8967,21809,'#FFFFFF'),(8968,21810,'#FFFFFF'),(8969,21830,'#ce1528'),(8970,21834,'#7b80fe'),(8971,21846,'#d2600f'),(8972,21853,'#fd6e91'),(8973,21856,'#678dde'),(8974,21861,'#2573d2'),(8975,21863,'#41e173'),(8976,21865,'#d7de29'),(8977,21869,'#1dfa7c'),(8978,21880,'#d4e703'),(8979,21883,'#9e57e0'),(8980,21890,'#fa7189'),(8981,21892,'#FFFFFF'),(8982,21895,'#ce2963'),(8983,21898,'#FFFFFF'),(8984,21900,'#2d6dd0'),(8985,21901,'#67face'),(8986,21902,'#6df89c'),(8987,21906,'#90fd58'),(8988,21909,'#FFFFFF'),(8989,21915,'#ec7521'),(8990,21916,'#fe7bba'),(8991,21917,'#a211e3'),(8992,21918,'#f36b03'),(8993,21919,'#8fe754'),(8994,21928,'#fa1dc9'),(8995,21932,'#f8f073'),(8996,21935,'#FFFFFF'),(8997,21941,'#56a1eb'),(8998,21942,'#FFFFFF'),(8999,21944,'#98de19'),(9000,21945,'#fb802c'),(9001,21946,'#99f658'),(9002,21948,'#15f253'),(9003,21949,'#cff97a'),(9004,21950,'#43e105'),(9005,21953,'#e1124d'),(9006,21955,'#28e779'),(9007,21956,'#d7ac67'),(9008,21967,'#1de4f8'),(9009,21971,'#FFFFFF'),(9010,21976,'#d709cc'),(9011,21985,'#f666f1'),(9012,21989,'#f2cc55'),(9013,21990,'#ce0d7f'),(9014,21991,'#778bf3'),(9015,21993,'#5fb5ed'),(9016,21999,'#3842d0'),(9017,22002,'#96fb2c'),(9018,22005,'#d1ae27'),(9019,22006,'#9f0cf9'),(9020,22009,'#3016e1'),(9021,22010,'#71e066'),(9022,22012,'#FFFFFF'),(9023,22014,'#df106c'),(9024,22017,'#9a00e1'),(9025,22020,'#fa7bdc'),(9026,22022,'#e908a7'),(9027,22025,'#18faaa'),(9028,22026,'#d4ce4c'),(9029,22034,'#dd9f53'),(9030,22036,'#055bf8'),(9031,22041,'#e108ab'),(9032,22046,'#23dff7'),(9033,22047,'#db657a'),(9034,22049,'#db4213'),(9035,22050,'#d9f900'),(9036,22059,'#4f8add'),(9037,22071,'#66e4a2'),(9038,22082,'#1e7bed'),(9039,22085,'#ace921'),(9040,22090,'#c0e704'),(9041,22106,'#90ce0e'),(9042,22107,'#f801df'),(9043,22109,'#0d1ed1'),(9044,22116,'#228adc'),(9045,22122,'#2156ec'),(9046,22133,'#0bcf9b'),(9047,22134,'#dd4a83'),(9048,22142,'#2acc0c'),(9049,22145,'#b50ed5'),(9050,22148,'#5625ec'),(9051,22151,'#b815f5'),(9052,22153,'#c338f9'),(9053,22159,'#ec22f3'),(9054,22161,'#8165db'),(9055,22168,'#e84ad0'),(9056,22178,'#bc22f9'),(9057,22182,'#eb815c'),(9058,22191,'#2e90e5'),(9059,22194,'#489ff2'),(9060,22201,'#38c6cf'),(9061,22204,'#8fdf1d'),(9062,22207,'#fe4ee9'),(9063,22218,'#0afebe'),(9064,22221,'#56da0f'),(9065,22224,'#d16bf4'),(9066,22227,'#0fc1f4'),(9067,22231,'#a75cdc'),(9068,22236,'#8b19e3'),(9069,22244,'#06d71c'),(9070,22249,'#93e32b'),(9071,22254,'#d9bd59'),(9072,22262,'#FFFFFF'),(9073,22265,'#FFFFFF'),(9074,22286,'#0de9da'),(9075,22288,'#1df317'),(9076,22290,'#FFFFFF'),(9077,22294,'#FFFFFF'),(9078,22304,'#f029e0'),(9079,22308,'#b4fd15'),(9080,22311,'#18fb68'),(9081,22322,'#fc7ea8'),(9082,22327,'#ec23a7'),(9083,22328,'#f1b358'),(9084,22346,'#e647a0'),(9085,22350,'#1dd135'),(9086,22351,'#6f62e0'),(9087,22352,'#e451bc'),(9088,22353,'#6be39e'),(9089,22357,'#6cee72'),(9090,22358,'#1fceae'),(9091,22359,'#72d734'),(9092,22368,'#cead04'),(9093,22371,'#a01bed'),(9094,22375,'#FFFFFF'),(9095,22385,'#d11fbb'),(9096,22388,'#e54e81'),(9097,22405,'#f52dd5'),(9098,22406,'#cd8b56'),(9099,22411,'#a129dc'),(9100,22414,'#1dcf80'),(9101,22419,'#fb269d'),(9102,22426,'#bf56f7'),(9103,22433,'#8d3be2'),(9104,22435,'#FFFFFF'),(9105,22450,'#ce080a'),(9106,22458,'#FFFFFF'),(9107,22462,'#e2a859'),(9108,22465,'#d0891c'),(9109,22472,'#89f25f'),(9110,22475,'#607ddf'),(9111,22476,'#a0e149'),(9112,22479,'#FFFFFF'),(9113,22480,'#FFFFFF'),(9114,22492,'#e23240'),(9115,22495,'#FFFFFF'),(9116,22507,'#FFFFFF'),(9117,22511,'#f9ac59'),(9118,22520,'#e50e7b'),(9119,22526,'#ccfd05'),(9120,22539,'#ef9500'),(9121,22552,'#6c2dd2'),(9122,22556,'#ea3082'),(9123,22562,'#e5de46'),(9124,22620,'#265eeb'),(9125,22659,'#d11b91'),(9126,22665,'#f2e956'),(9127,22683,'#e55a0b'),(9128,22684,'#2171e6'),(9129,22687,'#c85ffc'),(9130,22688,'#df67c1'),(9131,22691,'#61f48f'),(9132,22692,'#afe00e'),(9133,22693,'#43abd2'),(9134,22695,'#e838a3'),(9135,22696,'#c725ef'),(9136,22697,'#e70186'),(9137,22698,'#0de7dd'),(9138,22699,'#FFFFFF'),(9139,22700,'#de6433'),(9140,22701,'#bedd36'),(9141,22702,'#dc6f6c'),(9142,22703,'#df764f'),(9143,22704,'#cf6291'),(9144,22706,'#8925f0'),(9145,22708,'#09deb5'),(9146,22709,'#e73473'),(9147,22711,'#ab45e5'),(9148,22712,'#0934d8'),(9149,22713,'#dce769'),(9150,22714,'#5340fd'),(9151,22715,'#01f498'),(9152,22717,'#62dbf9'),(9153,22718,'#42cee8'),(9154,22719,'#cd28d0'),(9155,22720,'#a774ef'),(9156,22726,'#b347d6'),(9157,22727,'#0ae39d'),(9158,22728,'#d144a6'),(9159,22729,'#32e214'),(9160,22731,'#cf667b'),(9161,22733,'#01f20c'),(9162,22734,'#b2de1d'),(9163,22735,'#de0b6e'),(9164,22736,'#54cc80'),(9165,22737,'#dda90b'),(9166,22739,'#28d85a'),(9167,22740,'#fc980c'),(9168,22741,'#021efd'),(9169,22742,'#57daf0'),(9170,22744,'#c126d2'),(9171,22746,'#e056ed'),(9172,22748,'#fbe420'),(9173,22750,'#05f171'),(9174,22753,'#cc1003'),(9175,22754,'#c9f550'),(9176,22758,'#d2d23c'),(9177,22760,'#d53c48'),(9178,22761,'#41a8f0'),(9179,22763,'#3726e2'),(9180,22764,'#f119f6'),(9181,22765,'#f55bd8'),(9182,22766,'#42dafa'),(9183,22767,'#8d6ad7'),(9184,22768,'#f5211e'),(9185,22769,'#6a6fdc'),(9186,22770,'#FFFFFF'),(9187,22774,'#d5ee3a'),(9188,22775,'#24d421'),(9189,22776,'#db68b7'),(9190,22777,'#3da9e5'),(9191,22778,'#d46601'),(9192,22779,'#ef44ab'),(9193,22780,'#9dda1e'),(9194,22781,'#61d262'),(9195,22782,'#d27012'),(9196,22783,'#4da0da'),(9197,22785,'#d01483'),(9198,22788,'#f8a632'),(9199,22790,'#c544da'),(9200,22793,'#602ad4'),(9201,22794,'#f923fc'),(9202,22795,'#d2326a'),(9203,22796,'#51e9e4'),(9204,22800,'#bed54e'),(9205,22803,'#28cbe8'),(9206,22804,'#d6200f'),(9207,22805,'#e5ea46'),(9208,22807,'#d45288'),(9209,22810,'#07cd79'),(9210,22811,'#29e2af'),(9211,22812,'#fd47bc'),(9212,22813,'#dcb855'),(9213,22814,'#3447ef'),(9214,22815,'#14e228'),(9215,22816,'#945af2'),(9216,22818,'#4ad4e3'),(9217,22819,'#ef1413'),(9218,22820,'#2b05e6'),(9219,22821,'#826fe5'),(9220,22823,'#ec3aac'),(9221,22824,'#7d46f4'),(9222,22825,'#a7d955'),(9223,22826,'#becf40'),(9224,22827,'#d0da41'),(9225,22830,'#6ded8b'),(9226,22831,'#2302f7'),(9227,22832,'#33fcd2'),(9228,22833,'#f92b8c'),(9229,22835,'#180ce1'),(9230,22837,'#27ef46'),(9231,22839,'#39e387'),(9232,22840,'#e28347'),(9233,22841,'#59bdcd'),(9234,22843,'#43e721'),(9235,22844,'#da2280'),(9236,22845,'#22d080'),(9237,22846,'#cedd15'),(9238,22847,'#e61513'),(9239,22848,'#d74b01'),(9240,22849,'#d263c5'),(9241,22850,'#62a0d7'),(9242,22851,'#e42699'),(9243,22853,'#fe06f0'),(9244,22854,'#e763e1'),(9245,22855,'#8e58e2'),(9246,22856,'#66ec74'),(9247,22857,'#499edb'),(9248,22859,'#eb3217'),(9249,22861,'#FFFFFF'),(9250,22862,'#d55af3'),(9251,22863,'#db5e9d'),(9252,22864,'#47e9e8'),(9253,22865,'#dac821'),(9254,22866,'#2fee32'),(9255,22868,'#5c03de'),(9256,22869,'#b229e3'),(9257,22870,'#3e1cfc'),(9258,22872,'#17f38e'),(9259,22874,'#6c44d3'),(9260,22876,'#f55244'),(9261,22877,'#dd5c54'),(9262,22878,'#e46b89'),(9263,22881,'#d60415'),(9264,22882,'#63d1be'),(9265,22885,'#4d9bef'),(9266,22886,'#29f04e'),(9267,22887,'#57d3b8'),(9268,22888,'#a1fd56'),(9269,22889,'#aa0de9'),(9270,22891,'#dae26f'),(9271,22893,'#df5045'),(9272,22894,'#6ffc98'),(9273,22895,'#178cec'),(9274,22896,'#32d9c4'),(9275,22897,'#d946d8'),(9276,22898,'#a0d75c'),(9277,22899,'#01ddfd'),(9278,22901,'#d31644'),(9279,22902,'#FFFFFF'),(9280,22903,'#e84882'),(9281,22910,'#cd9c28'),(9282,22911,'#442ce9'),(9283,22912,'#5539e3'),(9284,22913,'#fd7093'),(9285,22914,'#9c30d0'),(9286,22915,'#2021ef'),(9287,22916,'#ceba62'),(9288,22917,'#0c12d6'),(9289,22918,'#fe799b'),(9290,22920,'#fa3d30'),(9291,22921,'#FFFFFF'),(9292,22922,'#FFFFFF'),(9293,22923,'#FFFFFF'),(9294,22924,'#FFFFFF'),(9295,22925,'#FFFFFF'),(9296,22926,'#FFFFFF'),(9297,22927,'#FFFFFF'),(9298,22931,'#FFFFFF'),(9299,22932,'#FFFFFF'),(9300,22933,'#FFFFFF'),(9301,22935,'#FFFFFF'),(9302,22936,'#FFFFFF'),(9303,22937,'#FFFFFF'),(9304,22938,'#FFFFFF'),(9305,22940,'#FFFFFF'),(9306,22941,'#FFFFFF'),(9307,22942,'#FFFFFF'),(9308,22943,'#FFFFFF'),(9309,22944,'#FFFFFF'),(9310,22946,'#FFFFFF'),(9311,22947,'#FFFFFF'),(9312,22948,'#FFFFFF'),(9313,22949,'#FFFFFF'),(9314,22952,'#FFFFFF'),(9315,22953,'#FFFFFF'),(9316,22954,'#FFFFFF'),(9317,22955,'#FFFFFF'),(9318,22956,'#FFFFFF'),(9319,22957,'#FFFFFF'),(9320,22958,'#FFFFFF'),(9321,22959,'#FFFFFF'),(9322,22960,'#FFFFFF'),(9323,22961,'#FFFFFF'),(9324,22962,'#FFFFFF'),(9325,22963,'#FFFFFF'),(9326,22964,'#FFFFFF'),(9327,22965,'#FFFFFF'),(9328,22966,'#FFFFFF'),(9329,22967,'#FFFFFF'),(9330,22969,'#5deb28'),(9331,22970,'#12ce19'),(9332,22972,'#FFFFFF'),(9333,22973,'#FFFFFF'),(9334,22974,'#FFFFFF'),(9335,22975,'#FFFFFF'),(9336,22976,'#FFFFFF'),(9337,22977,'#FFFFFF'),(9338,22978,'#FFFFFF'),(9339,22979,'#FFFFFF'),(9340,22980,'#FFFFFF'),(9341,22981,'#FFFFFF'),(9342,22982,'#FFFFFF'),(9343,22983,'#FFFFFF'),(9344,22984,'#FFFFFF'),(9345,22986,'#FFFFFF'),(9346,22987,'#FFFFFF'),(9347,22989,'#FFFFFF'),(9348,22991,'#FFFFFF'),(9349,22992,'#FFFFFF'),(9350,22993,'#FFFFFF'),(9351,22994,'#7edc4c'),(9352,22996,'#FFFFFF'),(9353,22997,'#FFFFFF'),(9354,22998,'#f66c7b'),(9355,22999,'#FFFFFF'),(9356,23000,'#FFFFFF'),(9357,23001,'#FFFFFF'),(9358,23002,'#FFFFFF'),(9359,23003,'#FFFFFF'),(9360,23004,'#FFFFFF'),(9361,23005,'#FFFFFF'),(9362,23006,'#FFFFFF'),(9363,23007,'#FFFFFF'),(9364,23008,'#FFFFFF'),(9365,23009,'#FFFFFF'),(9366,23010,'#FFFFFF'),(9367,23012,'#FFFFFF'),(9368,23013,'#FFFFFF'),(9369,23014,'#FFFFFF'),(9370,23015,'#fd6d9f'),(9371,23016,'#FFFFFF'),(9372,23017,'#FFFFFF'),(9373,23018,'#FFFFFF'),(9374,23019,'#FFFFFF'),(9375,23020,'#FFFFFF'),(9376,23021,'#FFFFFF'),(9377,23022,'#e3e83f'),(9378,23023,'#FFFFFF'),(9379,23024,'#FFFFFF'),(9380,23025,'#FFFFFF'),(9381,23026,'#FFFFFF'),(9382,23027,'#FFFFFF'),(9383,23028,'#FFFFFF'),(9384,23029,'#FFFFFF'),(9385,23030,'#FFFFFF'),(9386,23031,'#FFFFFF'),(9387,23032,'#FFFFFF'),(9388,23033,'#FFFFFF'),(9389,23034,'#FFFFFF'),(9390,23035,'#FFFFFF'),(9391,23036,'#FFFFFF'),(9392,23037,'#FFFFFF'),(9393,23038,'#FFFFFF'),(9394,23039,'#FFFFFF'),(9395,23040,'#FFFFFF'),(9396,23041,'#FFFFFF'),(9397,23042,'#FFFFFF'),(9398,23043,'#FFFFFF'),(9399,23044,'#FFFFFF'),(9400,23045,'#FFFFFF'),(9401,23046,'#FFFFFF'),(9402,23047,'#FFFFFF'),(9403,23048,'#0cdcc2'),(9404,23049,'#FFFFFF'),(9405,23050,'#FFFFFF'),(9406,23051,'#FFFFFF'),(9407,23052,'#18d12f'),(9408,23053,'#FFFFFF'),(9409,23055,'#FFFFFF'),(9410,23056,'#FFFFFF'),(9411,23057,'#FFFFFF'),(9412,23058,'#FFFFFF'),(9413,23059,'#FFFFFF'),(9414,23060,'#FFFFFF'),(9415,23061,'#FFFFFF'),(9416,23062,'#FFFFFF'),(9417,23063,'#FFFFFF'),(9418,23064,'#FFFFFF'),(9419,23065,'#FFFFFF'),(9420,23066,'#26d09a'),(9421,23068,'#FFFFFF'),(9422,23070,'#FFFFFF'),(9423,23071,'#FFFFFF'),(9424,23072,'#FFFFFF'),(9425,23073,'#FFFFFF'),(9426,23074,'#FFFFFF'),(9427,23075,'#FFFFFF'),(9428,23076,'#FFFFFF'),(9429,23077,'#FFFFFF'),(9430,23078,'#FFFFFF'),(9431,23079,'#FFFFFF'),(9432,23080,'#FFFFFF'),(9433,23081,'#FFFFFF'),(9434,23082,'#FFFFFF'),(9435,23083,'#FFFFFF'),(9436,23084,'#FFFFFF'),(9437,23085,'#FFFFFF'),(9438,23086,'#FFFFFF'),(9439,23087,'#FFFFFF'),(9440,23088,'#FFFFFF'),(9441,23089,'#FFFFFF'),(9442,23090,'#FFFFFF'),(9443,23092,'#FFFFFF'),(9444,23093,'#FFFFFF'),(9445,23094,'#FFFFFF'),(9446,23095,'#FFFFFF'),(9447,23096,'#FFFFFF'),(9448,23097,'#df23cf'),(9449,23098,'#FFFFFF'),(9450,23099,'#FFFFFF'),(9451,23100,'#FFFFFF'),(9452,23101,'#FFFFFF'),(9453,23102,'#FFFFFF'),(9454,23103,'#FFFFFF'),(9455,23104,'#FFFFFF'),(9456,23105,'#FFFFFF'),(9457,23106,'#FFFFFF'),(9458,23107,'#FFFFFF'),(9459,23108,'#FFFFFF'),(9460,23109,'#FFFFFF'),(9461,23110,'#FFFFFF'),(9462,23111,'#FFFFFF'),(9463,23112,'#FFFFFF'),(9464,23113,'#FFFFFF'),(9465,23115,'#FFFFFF'),(9466,23116,'#FFFFFF'),(9467,23117,'#FFFFFF'),(9468,23119,'#FFFFFF'),(9469,23120,'#FFFFFF'),(9470,23122,'#FFFFFF'),(9471,23123,'#FFFFFF'),(9472,23124,'#FFFFFF'),(9473,23125,'#FFFFFF'),(9474,23126,'#FFFFFF'),(9475,23127,'#FFFFFF'),(9476,23128,'#FFFFFF'),(9477,23129,'#FFFFFF'),(9478,23130,'#FFFFFF'),(9479,23133,'#FFFFFF'),(9480,23148,'#836cdd'),(9481,23149,'#1ffea8'),(9482,23160,'#f6c454'),(9483,23165,'#c740e1'),(9484,23170,'#8265f5'),(9485,23218,'#d3a53e'),(9486,23219,'#e55ccc'),(9487,23220,'#427fde'),(9488,23221,'#fc5878'),(9489,23222,'#ea5380'),(9490,23250,'#8f76ee'),(9491,23262,'#5ef5a3'),(9492,23288,'#08ce28'),(9493,23289,'#d6b257'),(9494,23302,'#eee776'),(9495,23324,'#cf6262'),(9496,23338,'#fc406a'),(9497,23341,'#FFFFFF'),(9498,23363,'#22d383'),(9499,23366,'#5d35fb'),(9500,23369,'#e450c4'),(9501,23374,'#66e4c1'),(9502,23393,'#ea6ac1'),(9503,23421,'#5146d1'),(9504,23428,'#f13b35'),(9505,23429,'#e87282'),(9506,23431,'#f89d61'),(9507,23432,'#331ee5'),(9508,23442,'#f4739d'),(9509,23445,'#152ffa'),(9510,23447,'#ab64d1'),(9511,23455,'#e80b12'),(9512,23459,'#4a62d2'),(9513,23462,'#82d04d'),(9514,23483,'#18fb13'),(9515,23487,'#dcf868'),(9516,23490,'#4ea5df'),(9517,23492,'#FFFFFF'),(9518,23493,'#c1f516'),(9519,23495,'#FFFFFF'),(9520,23496,'#bfd013'),(9521,23499,'#0acc89'),(9522,23503,'#4ca2cf'),(9523,23505,'#FFFFFF'),(9524,23506,'#d23b82'),(9525,23510,'#09e3e2'),(9526,23511,'#1be31f'),(9527,23512,'#26d808'),(9528,23513,'#f9bf65'),(9529,23514,'#FFFFFF'),(9530,23516,'#1e17d2'),(9531,23518,'#FFFFFF'),(9532,23522,'#c42dfc'),(9533,23523,'#a7f759'),(9534,23526,'#ce14df'),(9535,23527,'#bd6dee'),(9536,23528,'#6176d9'),(9537,23529,'#082ed4'),(9538,23530,'#3bf060'),(9539,23531,'#28e5b9'),(9540,23533,'#FFFFFF'),(9541,23536,'#b74bf2'),(9542,23540,'#37f0f2'),(9543,23541,'#17e056'),(9544,23542,'#ec3fef'),(9545,23543,'#697ee0'),(9546,23544,'#67dabf'),(9547,23548,'#fc46c1'),(9548,23549,'#FFFFFF'),(9549,23551,'#85eb52'),(9550,23552,'#58ef67'),(9551,23558,'#e31c61'),(9552,23559,'#c3ee64'),(9553,23561,'#d67707'),(9554,23562,'#3af1e1'),(9555,23566,'#b9fb4d'),(9556,23571,'#FFFFFF'),(9557,23572,'#f43cdf'),(9558,23573,'#69e540'),(9559,23575,'#1e06f7'),(9560,23576,'#FFFFFF'),(9561,23579,'#5bea5e'),(9562,23582,'#FFFFFF'),(9563,23583,'#FFFFFF'),(9564,23585,'#FFFFFF'),(9565,23587,'#FFFFFF'),(9566,23588,'#FFFFFF'),(9567,23590,'#e9b131'),(9568,23596,'#dd60ad'),(9569,23597,'#FFFFFF'),(9570,23600,'#FFFFFF'),(9571,23601,'#4ee876'),(9572,23602,'#72f2f0'),(9573,23603,'#e71d26'),(9574,23605,'#5a34dd'),(9575,23608,'#85d356'),(9576,23609,'#FFFFFF'),(9577,23610,'#79d260'),(9578,23611,'#d3a73f'),(9579,23612,'#5ef1e4'),(9580,23613,'#0694fe'),(9581,23615,'#8624d1'),(9582,23617,'#6d50d0'),(9583,23619,'#67e458'),(9584,23620,'#FFFFFF'),(9585,23621,'#a0ce65'),(9586,23622,'#FFFFFF'),(9587,23623,'#f4a211'),(9588,23629,'#f653cf'),(9589,23630,'#02f68c'),(9590,23633,'#FFFFFF'),(9591,23634,'#f522db'),(9592,23635,'#4bdf3e'),(9593,23638,'#646fd1'),(9594,23639,'#dc1282'),(9595,23640,'#1dc2f1'),(9596,23643,'#e80bde'),(9597,23645,'#efb102'),(9598,23646,'#FFFFFF'),(9599,23647,'#FFFFFF'),(9600,23648,'#FFFFFF'),(9601,23649,'#FFFFFF'),(9602,23651,'#32b3df'),(9603,23653,'#1bcc3b'),(9604,23654,'#d43971'),(9605,23655,'#FFFFFF'),(9606,23656,'#FFFFFF'),(9607,23660,'#df8124'),(9608,23662,'#33f34e'),(9609,23664,'#FFFFFF'),(9610,23665,'#FFFFFF'),(9611,23666,'#FFFFFF'),(9612,23668,'#fbc44a'),(9613,23671,'#05edbf'),(9614,23673,'#FFFFFF'),(9615,23674,'#f91f0b'),(9616,23675,'#18c7f3'),(9617,23677,'#cff461'),(9618,23681,'#FFFFFF'),(9619,23686,'#b6ea39'),(9620,23690,'#FFFFFF'),(9621,23691,'#d48c5a'),(9622,23692,'#5b09e8'),(9623,23695,'#FFFFFF'),(9624,23696,'#FFFFFF'),(9625,23698,'#5edabf'),(9626,23699,'#379fcd'),(9627,23700,'#22de8d'),(9628,23702,'#FFFFFF'),(9629,23704,'#f26063'),(9630,23707,'#cc4aec'),(9631,23708,'#f0bc33'),(9632,23712,'#FFFFFF'),(9633,23713,'#3086e7'),(9634,23715,'#28acd9'),(9635,23716,'#fed97d'),(9636,23718,'#FFFFFF'),(9637,23723,'#e97a2b'),(9638,23725,'#FFFFFF'),(9639,23729,'#fc862a'),(9640,23730,'#FFFFFF'),(9641,23732,'#ac26f9'),(9642,23733,'#d39c46'),(9643,23736,'#fa7a9b'),(9644,23737,'#5dd91b'),(9645,23739,'#215efa'),(9646,23741,'#FFFFFF'),(9647,23742,'#d0c855'),(9648,23743,'#FFFFFF'),(9649,23748,'#49e60b'),(9650,23749,'#5cdb9a'),(9651,23752,'#b56cfa'),(9652,23753,'#ea1eaf'),(9653,23754,'#e81365'),(9654,23755,'#0d1ff3'),(9655,23756,'#e51ecd'),(9656,23757,'#3af8db'),(9657,23758,'#FFFFFF'),(9658,23759,'#f92c30'),(9659,23762,'#5567f0'),(9660,23766,'#b852f3'),(9661,23767,'#43ef43'),(9662,23768,'#FFFFFF'),(9663,23769,'#d8bd52'),(9664,23770,'#FFFFFF'),(9665,23771,'#c2da51'),(9666,23772,'#FFFFFF'),(9667,23773,'#d5f463'),(9668,23774,'#d22514'),(9669,23777,'#7fe40c'),(9670,23778,'#b40cde'),(9671,23779,'#b878fe'),(9672,23781,'#f52716'),(9673,23782,'#6eefa5'),(9674,23784,'#ebd81b'),(9675,23785,'#76feb9'),(9676,23786,'#66f35f'),(9677,23787,'#FFFFFF'),(9678,23789,'#e57d2b'),(9679,23790,'#efe301'),(9680,23791,'#dfb323'),(9681,23792,'#22d46e'),(9682,23794,'#ed584d'),(9683,23796,'#31d563'),(9684,23798,'#cd1fb0'),(9685,23799,'#65e8b1'),(9686,23800,'#FFFFFF'),(9687,23801,'#99ee49'),(9688,23803,'#24b6d9'),(9689,23804,'#628bf4'),(9690,23805,'#FFFFFF'),(9691,23807,'#64b5f1'),(9692,23811,'#4732d6'),(9693,23813,'#56de47'),(9694,23814,'#d99b0a'),(9695,23815,'#7e18e6'),(9696,23816,'#c530ce'),(9697,23817,'#e02b4c'),(9698,23818,'#06e2df'),(9699,23820,'#156cd4'),(9700,23821,'#0862fc'),(9701,23824,'#b717df'),(9702,23829,'#6ca8f6'),(9703,23831,'#e76f6a'),(9704,23832,'#765be8'),(9705,23834,'#edb543'),(9706,23835,'#f6aa51'),(9707,23837,'#754bcf'),(9708,23839,'#1ae557'),(9709,23840,'#79d44c'),(9710,23841,'#f879bc'),(9711,23842,'#FFFFFF'),(9712,23843,'#f03ece'),(9713,23845,'#d4f227'),(9714,23846,'#FFFFFF'),(9715,23851,'#cf466f'),(9716,23854,'#fc7973'),(9717,23857,'#96d21d'),(9718,23858,'#eed54a'),(9719,23860,'#9f66e9'),(9720,23861,'#92e14b'),(9721,23863,'#d8b521'),(9722,23866,'#d72dc1'),(9723,23867,'#5a82ef'),(9724,23868,'#12da80'),(9725,23871,'#70e917'),(9726,23874,'#d69c53'),(9727,23875,'#4be330'),(9728,23877,'#FFFFFF'),(9729,23878,'#e32b3d'),(9730,23880,'#3763e4'),(9731,23883,'#FFFFFF'),(9732,23884,'#a6e71d'),(9733,23886,'#bd2de1'),(9734,23888,'#e859c1'),(9735,23891,'#2dfabe'),(9736,23892,'#19d29a'),(9737,23894,'#FFFFFF'),(9738,23896,'#0fd054'),(9739,23897,'#5cce58'),(9740,23899,'#0351cc'),(9741,23901,'#662cd6'),(9742,23904,'#2e38dd'),(9743,23905,'#1776ed'),(9744,23909,'#374df8'),(9745,23911,'#dcb743'),(9746,23912,'#a151d6'),(9747,23917,'#56ade8'),(9748,23919,'#f51ddf'),(9749,23920,'#9b1aef'),(9750,23921,'#FFFFFF'),(9751,23925,'#dd06ba'),(9752,23932,'#a727ef'),(9753,23933,'#cd1a98'),(9754,23935,'#FFFFFF'),(9755,23937,'#FFFFFF'),(9756,23938,'#a4de4d'),(9757,23940,'#cbe541'),(9758,23943,'#a04bf8'),(9759,23944,'#FFFFFF'),(9760,23945,'#e73f5f'),(9761,23951,'#17eee6'),(9762,23955,'#ddf437'),(9763,23956,'#d42a63'),(9764,23958,'#FFFFFF'),(9765,23959,'#FFFFFF'),(9766,23963,'#36d74d'),(9767,23964,'#FFFFFF'),(9768,23965,'#0b5bdc'),(9769,23969,'#FFFFFF'),(9770,23970,'#64f380'),(9771,23971,'#FFFFFF'),(9772,23972,'#03d27a'),(9773,23974,'#c461f0'),(9774,23977,'#FFFFFF'),(9775,23978,'#FFFFFF'),(9776,23979,'#54f81b'),(9777,23982,'#0605d2'),(9778,23985,'#FFFFFF'),(9779,23987,'#d05b50'),(9780,23988,'#5df84d'),(9781,23993,'#5115d3'),(9782,23994,'#f44ef9'),(9783,23995,'#FFFFFF'),(9784,23996,'#26e7eb'),(9785,23999,'#fc9134'),(9786,24000,'#ee27a0'),(9787,24002,'#FFFFFF'),(9788,24003,'#FFFFFF'),(9789,24004,'#77d345'),(9790,24005,'#fd7961'),(9791,24006,'#FFFFFF'),(9792,24007,'#d0da44'),(9793,24010,'#f2ad6c'),(9794,24011,'#7ffe3f'),(9795,24013,'#FFFFFF'),(9796,24014,'#f0e125'),(9797,24015,'#FFFFFF'),(9798,24016,'#FFFFFF'),(9799,24017,'#FFFFFF'),(9800,24018,'#e28e53'),(9801,24019,'#d766a3'),(9802,24021,'#50cc26'),(9803,24023,'#ef0c8f'),(9804,24024,'#f09c66'),(9805,24025,'#74fbb1'),(9806,24026,'#f9a10e'),(9807,24027,'#d98a46'),(9808,24029,'#12f282'),(9809,24030,'#5318ec'),(9810,24031,'#FFFFFF'),(9811,24033,'#bbd02a'),(9812,24035,'#f46ece'),(9813,24036,'#ce199f'),(9814,24038,'#5625d3'),(9815,24039,'#e6fe36'),(9816,24040,'#77f992'),(9817,24042,'#f13cda'),(9818,24045,'#ddf95a'),(9819,24048,'#48d6cb'),(9820,24050,'#48dfbd'),(9821,24051,'#539ef6'),(9822,24052,'#2b57ea'),(9823,24053,'#7be15d'),(9824,24054,'#284ffe'),(9825,24058,'#3b99fc'),(9826,24060,'#edbe45'),(9827,24069,'#b7fd2a'),(9828,24071,'#73f442'),(9829,24076,'#6ed435'),(9830,24077,'#f8df24'),(9831,24078,'#19f936'),(9832,24080,'#55db5d'),(9833,24081,'#17bde0'),(9834,24082,'#03f33f'),(9835,24084,'#ef5586'),(9836,24087,'#e4e15c'),(9837,24088,'#ce198d'),(9838,24089,'#FFFFFF'),(9839,24090,'#f2eb60'),(9840,24092,'#f6513d'),(9841,24095,'#FFFFFF'),(9842,24098,'#bbf32f'),(9843,24100,'#226ff9'),(9844,24101,'#1190e5'),(9845,24102,'#02f923'),(9846,24103,'#f3446b'),(9847,24107,'#fe6876'),(9848,24108,'#bd02de'),(9849,24109,'#b1e204'),(9850,24110,'#FFFFFF'),(9851,24111,'#d62b58'),(9852,24113,'#09f017'),(9853,24115,'#cd4d62'),(9854,24117,'#FFFFFF'),(9855,24118,'#ebfb72'),(9856,24120,'#df6e0b'),(9857,24121,'#56f799'),(9858,24123,'#97fb6a'),(9859,24124,'#e33abf'),(9860,24127,'#f94a10'),(9861,24134,'#2d03fb'),(9862,24135,'#4bf694'),(9863,24136,'#d98c4e'),(9864,24137,'#8936de'),(9865,24141,'#FFFFFF'),(9866,24143,'#0ad3a1'),(9867,24145,'#5af769'),(9868,24146,'#6480f2'),(9869,24147,'#d671ef'),(9870,24154,'#FFFFFF'),(9871,24156,'#eb71ef'),(9872,24160,'#f59c39'),(9873,24161,'#8342cd'),(9874,24162,'#97f519'),(9875,24163,'#e51dbd'),(9876,24164,'#fd57cb'),(9877,24166,'#3983d3'),(9878,24168,'#6cf9fe'),(9879,24171,'#e1314e'),(9880,24179,'#06e280'),(9881,24180,'#76e8fe'),(9882,24182,'#def611'),(9883,24187,'#69e4cc'),(9884,24191,'#66d990'),(9885,24193,'#3cf47b'),(9886,24196,'#d2d438'),(9887,24197,'#bb66f8'),(9888,24199,'#cdca06'),(9889,24203,'#FFFFFF'),(9890,24205,'#3dd18b'),(9891,24206,'#da34d9'),(9892,24207,'#FFFFFF'),(9893,24211,'#0d2fce'),(9894,24213,'#d09960'),(9895,24214,'#e4bb54'),(9896,24215,'#ce3f3e'),(9897,24219,'#5ceaaa'),(9898,24220,'#fd0d84'),(9899,24225,'#ea376d'),(9900,24232,'#2bcf2d'),(9901,24233,'#6665e0'),(9902,24234,'#ea3454'),(9903,24235,'#df2e2f'),(9904,24241,'#8e72fd'),(9905,24245,'#753fd4'),(9906,24249,'#a3de0d'),(9907,24252,'#e33084'),(9908,24258,'#FFFFFF'),(9909,24261,'#3dd0c6'),(9910,24262,'#66a9de'),(9911,24264,'#FFFFFF'),(9912,24266,'#14f8c6'),(9913,24267,'#fc8978'),(9914,24269,'#1c90f2'),(9915,24272,'#fc6c7f'),(9916,24284,'#cd8f66'),(9917,24288,'#f206bd'),(9918,24289,'#d1a041'),(9919,24295,'#f83371'),(9920,24296,'#9e03f2'),(9921,24298,'#f55e7a'),(9922,24301,'#13a1dd'),(9923,24304,'#e15f21'),(9924,24305,'#616dd2'),(9925,24311,'#6beaeb'),(9926,24312,'#FFFFFF'),(9927,24313,'#24acde'),(9928,24315,'#238cf5'),(9929,24319,'#6add70'),(9930,24321,'#1e4efc'),(9931,24322,'#cd4f25'),(9932,24325,'#cd0df3'),(9933,24335,'#fc51fd'),(9934,24336,'#FFFFFF'),(9935,24337,'#03f628'),(9936,24338,'#ef2bc9'),(9937,24341,'#FFFFFF'),(9938,24342,'#4396d2'),(9939,24346,'#319bf7'),(9940,24348,'#e02655'),(9941,24350,'#5cedf4'),(9942,24353,'#b6f55c'),(9943,24354,'#e1aa1e'),(9944,24358,'#FFFFFF'),(9945,24359,'#f9f549'),(9946,24366,'#cf28b5'),(9947,24371,'#12a2e5'),(9948,24376,'#5ace8a'),(9949,24380,'#68f259'),(9950,24381,'#3053f0'),(9951,24382,'#FFFFFF'),(9952,24383,'#5e7ad8'),(9953,24386,'#1730d8'),(9954,24387,'#f25645'),(9955,24390,'#ba36ef'),(9956,24397,'#a8f73c'),(9957,24398,'#2bf2aa'),(9958,24400,'#61eee8'),(9959,24411,'#784eda'),(9960,24415,'#dd843b'),(9961,24416,'#d33d98'),(9962,24419,'#0fd73c'),(9963,24423,'#94ee0f'),(9964,24424,'#27e683'),(9965,24426,'#3eecc1'),(9966,24427,'#5e9bfc'),(9967,24430,'#6026f3'),(9968,24432,'#5c88d2'),(9969,24434,'#ef6555'),(9970,24436,'#66e9b9'),(9971,24437,'#69f0e3'),(9972,24438,'#3882d4'),(9973,24444,'#11ce63'),(9974,24448,'#30dd42'),(9975,24452,'#e7722c'),(9976,24461,'#FFFFFF'),(9977,24462,'#FFFFFF'),(9978,24463,'#b6f721'),(9979,24465,'#256cd7'),(9980,24466,'#fa1f8e'),(9981,24469,'#dc2fb1'),(9982,24473,'#FFFFFF'),(9983,24474,'#ce239d'),(9984,24478,'#6ef8b0'),(9985,24484,'#1783ee'),(9986,24485,'#5454da'),(9987,24487,'#d4da68'),(9988,24490,'#ea704b'),(9989,24491,'#26d7e2'),(9990,24502,'#4e54d0'),(9991,24504,'#58f3e4'),(9992,24507,'#5b73f6'),(9993,24509,'#FFFFFF'),(9994,24511,'#7120f4'),(9995,24512,'#d15fe2'),(9996,24516,'#f0da22'),(9997,24524,'#b6f600'),(9998,24525,'#2c4efa'),(9999,24528,'#0e30f8'),(10000,24530,'#e78e29'),(10001,24536,'#d33445'),(10002,24540,'#3ff05f'),(10003,24542,'#ccc447'),(10004,24544,'#FFFFFF'),(10005,24546,'#e261ae'),(10006,24547,'#2058d5'),(10007,24548,'#52f0ea'),(10008,24553,'#78b5fd'),(10009,24562,'#cd5c38'),(10010,24565,'#d95f47'),(10011,24566,'#db26f2'),(10012,24569,'#cf58e4'),(10013,24571,'#f1611b'),(10014,24577,'#FFFFFF'),(10015,24580,'#3fde69'),(10016,24585,'#90da1e'),(10017,24590,'#e7ad67'),(10018,24591,'#78d350'),(10019,24594,'#ec4921'),(10020,24596,'#22e18b'),(10021,24598,'#e0ac59'),(10022,24599,'#b2df5b'),(10023,24606,'#dddd55'),(10024,24609,'#b0df3d'),(10025,24612,'#d6ca52'),(10026,24615,'#FFFFFF'),(10027,24616,'#fdd228'),(10028,24618,'#1cfbc8'),(10029,24622,'#7214d4'),(10030,24623,'#4fcfd0'),(10031,24626,'#67d5d8'),(10032,24630,'#e732cb'),(10033,24635,'#2379e9'),(10034,24636,'#3020f9'),(10035,24643,'#aa68fb'),(10036,24645,'#d79219'),(10037,24648,'#da4bb6'),(10038,24649,'#ad6cf5'),(10039,24651,'#d6d618'),(10040,24654,'#5b57f2'),(10041,24659,'#FFFFFF'),(10042,24660,'#9161f7'),(10043,24662,'#e3d766'),(10044,24663,'#dd3141'),(10045,24667,'#da984c'),(10046,24671,'#5a00e5'),(10047,24673,'#ee329e'),(10048,24675,'#0c89e6'),(10049,24680,'#c5ee36'),(10050,24683,'#64f7ed'),(10051,24692,'#5fe684'),(10052,24694,'#FFFFFF'),(10053,24696,'#034af4'),(10054,24698,'#e7be5e'),(10055,24699,'#12a5e4'),(10056,24701,'#FFFFFF'),(10057,24702,'#4dcfc0'),(10058,24704,'#bc31d7'),(10059,24709,'#ef8a36'),(10060,24710,'#f50b33'),(10061,24723,'#6b51dd'),(10062,24724,'#467ef3'),(10063,24725,'#36ed44'),(10064,24727,'#0adbc8'),(10065,24731,'#59fd55'),(10066,24733,'#FFFFFF'),(10067,24735,'#ead16c'),(10068,24743,'#a8f017'),(10069,24746,'#d65da2'),(10070,24747,'#5de242'),(10071,24749,'#4c41d2'),(10072,24754,'#715ce7'),(10073,24755,'#4ed0bc'),(10074,24756,'#3ceda6'),(10075,24759,'#215edb'),(10076,24760,'#f764ba'),(10077,24763,'#e1ed53'),(10078,24767,'#abfc2d'),(10079,24769,'#98d062'),(10080,24770,'#3ce993'),(10081,24774,'#f319f6'),(10082,24777,'#4969d0'),(10083,24780,'#1e21fa'),(10084,24785,'#db192e'),(10085,24787,'#335fea'),(10086,24789,'#ea4ceb'),(10087,24790,'#2af15a'),(10088,24792,'#400ad2'),(10089,24795,'#d8be4f'),(10090,24800,'#FFFFFF'),(10091,24804,'#f75263'),(10092,24806,'#ab58d8'),(10093,24812,'#FFFFFF'),(10094,24814,'#f1cc61'),(10095,24815,'#d5451f'),(10096,24816,'#d5aa66'),(10097,24817,'#33d491'),(10098,24824,'#4e68f0'),(10099,24825,'#fcb363'),(10100,24829,'#47cdb8'),(10101,24830,'#af5fd0'),(10102,24832,'#3d53ee'),(10103,24833,'#f96c78'),(10104,24835,'#862dd2'),(10105,24837,'#31d1d8'),(10106,24839,'#66e562'),(10107,24840,'#60ccd0'),(10108,24841,'#ed3632'),(10109,24842,'#d0247d'),(10110,24843,'#db2d19'),(10111,24845,'#FFFFFF'),(10112,24846,'#50c1d8'),(10113,24849,'#70d1ec'),(10114,24851,'#da9d0b'),(10115,24853,'#FFFFFF'),(10116,24854,'#FFFFFF'),(10117,24856,'#09ed31'),(10118,24858,'#fb66d0'),(10119,24859,'#53e6df'),(10120,24863,'#5358e1'),(10121,24865,'#df306f'),(10122,24866,'#FFFFFF'),(10123,24867,'#e63ae3'),(10124,24868,'#FFFFFF'),(10125,24872,'#FFFFFF'),(10126,24873,'#FFFFFF'),(10127,24874,'#ea7956'),(10128,24876,'#FFFFFF'),(10129,24877,'#6378d9'),(10130,24879,'#0ee098'),(10131,24880,'#39cdcd'),(10132,24881,'#cd3d4b'),(10133,24882,'#FFFFFF'),(10134,24885,'#4ce7dc'),(10135,24886,'#FFFFFF'),(10136,24888,'#fe5620'),(10137,24889,'#a440f3'),(10138,24890,'#FFFFFF'),(10139,24892,'#FFFFFF'),(10140,24897,'#FFFFFF'),(10141,24898,'#0ab6f0'),(10142,24899,'#526cef'),(10143,24900,'#FFFFFF'),(10144,24901,'#c160ee'),(10145,24902,'#30aeea'),(10146,24903,'#de526f'),(10147,24908,'#60ddce'),(10148,24910,'#65cd65'),(10149,24913,'#30df3c'),(10150,24914,'#faf972'),(10151,24916,'#3f2dcd'),(10152,24919,'#205ad8'),(10153,24920,'#e638cc'),(10154,24922,'#e84961'),(10155,24924,'#FFFFFF'),(10156,24926,'#54e0b5'),(10157,24928,'#FFFFFF'),(10158,24929,'#de5163'),(10159,24931,'#5413d7'),(10160,24937,'#67fb16'),(10161,24938,'#e332a0'),(10162,24939,'#d01900'),(10163,24942,'#FFFFFF'),(10164,24945,'#FFFFFF'),(10165,24946,'#273af3'),(10166,24947,'#cd660a'),(10167,24949,'#2453eb'),(10168,24950,'#0eb3cf'),(10169,24951,'#FFFFFF'),(10170,24953,'#FFFFFF'),(10171,24954,'#FFFFFF'),(10172,24956,'#FFFFFF'),(10173,24957,'#f941a6'),(10174,24959,'#FFFFFF'),(10175,24961,'#FFFFFF'),(10176,24963,'#e3d365'),(10177,24964,'#2c60da'),(10178,24965,'#5084ef'),(10179,24967,'#FFFFFF'),(10180,24968,'#98ed52'),(10181,24973,'#FFFFFF'),(10182,24974,'#6160e6'),(10183,24975,'#2688e7'),(10184,24976,'#e82014'),(10185,24977,'#d53c07'),(10186,24979,'#29e343'),(10187,24980,'#eb3e8c'),(10188,24981,'#ef4156'),(10189,24985,'#4be150'),(10190,24988,'#5de242'),(10191,24990,'#9af939'),(10192,24992,'#f2ce1c'),(10193,24994,'#3cd546'),(10194,24999,'#e8a203'),(10195,25009,'#FFFFFF'),(10196,25011,'#e7e665'),(10197,25013,'#fbbc5c'),(10198,25015,'#f176eb'),(10199,25016,'#FFFFFF'),(10200,25018,'#3726d7'),(10201,25020,'#d73581'),(10202,25021,'#15f6a2'),(10203,25022,'#e25503'),(10204,25023,'#2515f5'),(10205,25025,'#50dd3a'),(10206,25026,'#e768eb'),(10207,25028,'#4488dd'),(10208,25031,'#3b04f2'),(10209,25033,'#e5e618'),(10210,25036,'#f51c90'),(10211,25037,'#df52ea'),(10212,25038,'#ace569'),(10213,25039,'#f42aaf'),(10214,25040,'#db3ad2'),(10215,25041,'#d8f15e'),(10216,25042,'#d6618c'),(10217,25043,'#b0fe7c'),(10218,25044,'#66d8a8'),(10219,25045,'#FFFFFF'),(10220,25046,'#FFFFFF'),(10221,25049,'#FFFFFF'),(10222,25051,'#6de9a0'),(10223,25053,'#FFFFFF'),(10224,25055,'#02de26'),(10225,25056,'#6e12f0'),(10226,25057,'#820cd3'),(10227,25058,'#9936e1'),(10228,25061,'#c8f214'),(10229,25062,'#6acedc'),(10230,25065,'#e4ef35'),(10231,25066,'#ecb90c'),(10232,25068,'#6fe106'),(10233,25069,'#44d321'),(10234,25070,'#e2082c'),(10235,25071,'#87fd0f'),(10236,25072,'#f74c49'),(10237,25073,'#FFFFFF'),(10238,25075,'#56fa2b'),(10239,25077,'#d41a02'),(10240,25078,'#13c0d4'),(10241,25079,'#FFFFFF'),(10242,25080,'#2047f3'),(10243,25081,'#51d3e6'),(10244,25083,'#FFFFFF'),(10245,25085,'#e15cc8'),(10246,25088,'#c3d226'),(10247,25089,'#cacc31'),(10248,25093,'#06e873'),(10249,25094,'#7fd752'),(10250,25095,'#cf0d16'),(10251,25097,'#58cdad'),(10252,25101,'#FFFFFF'),(10253,25103,'#FFFFFF'),(10254,25104,'#61cfb3'),(10255,25106,'#68d484'),(10256,25109,'#FFFFFF'),(10257,25110,'#40f0c9'),(10258,25113,'#70e193'),(10259,25114,'#5fed95'),(10260,25116,'#e81d3c'),(10261,25118,'#77f600'),(10262,25119,'#FFFFFF'),(10263,25120,'#adf71e'),(10264,25121,'#b7e023'),(10265,25123,'#bce553'),(10266,25126,'#1f6de4'),(10267,25128,'#2c5af6'),(10268,25131,'#9453df'),(10269,25132,'#5ad83c'),(10270,25133,'#FFFFFF'),(10271,25134,'#82cd54'),(10272,25136,'#4621fb'),(10273,25141,'#f9465e'),(10274,25143,'#FFFFFF'),(10275,25144,'#35ddd5'),(10276,25145,'#f659cc'),(10277,25146,'#7e0ee0'),(10278,25147,'#FFFFFF'),(10279,25148,'#3daddb'),(10280,25151,'#1bf921'),(10281,25153,'#3794fc'),(10282,25154,'#c036e0'),(10283,25158,'#71d913'),(10284,25159,'#f6d65f'),(10285,25164,'#17dccd'),(10286,25167,'#FFFFFF'),(10287,25172,'#e7389a'),(10288,25175,'#45d34c'),(10289,25177,'#d7b611'),(10290,25178,'#f40b7d'),(10291,25179,'#62d44f'),(10292,25185,'#63abf9'),(10293,25186,'#d68a50'),(10294,25187,'#FFFFFF'),(10295,25188,'#FFFFFF'),(10296,25189,'#de6391'),(10297,25190,'#f40c64'),(10298,25191,'#4184e5'),(10299,25192,'#4859e3'),(10300,25193,'#95fb41'),(10301,25194,'#FFFFFF'),(10302,25199,'#dba943'),(10303,25203,'#d4123b'),(10304,25204,'#a819d7'),(10305,25210,'#d00799'),(10306,25211,'#e2fe45'),(10307,25214,'#FFFFFF'),(10308,25217,'#f1ad0a'),(10309,25218,'#e74463'),(10310,25219,'#8fdc4c'),(10311,25220,'#FFFFFF'),(10312,25222,'#57c9e3'),(10313,25224,'#3aeaba'),(10314,25229,'#5ce17c'),(10315,25237,'#13d631'),(10316,25239,'#32dc93'),(10317,25240,'#6ad599'),(10318,25243,'#FFFFFF'),(10319,25244,'#686adb'),(10320,25245,'#FFFFFF'),(10321,25248,'#287bda'),(10322,25250,'#46e481'),(10323,25251,'#09cc47'),(10324,25254,'#d1cf59'),(10325,25258,'#cf17a7'),(10326,25262,'#1bf1a2'),(10327,25264,'#FFFFFF'),(10328,25265,'#ace308'),(10329,25266,'#49dc52'),(10330,25268,'#ce530a'),(10331,25271,'#fb20a5'),(10332,25273,'#FFFFFF'),(10333,25274,'#FFFFFF'),(10334,25275,'#d63b3e'),(10335,25276,'#FFFFFF'),(10336,25277,'#FFFFFF'),(10337,25278,'#1d2fcd'),(10338,25282,'#e2c74c'),(10339,25286,'#5e97dc'),(10340,25292,'#c4e22d'),(10341,25296,'#b1dc28'),(10342,25298,'#94de6a'),(10343,25300,'#FFFFFF'),(10344,25309,'#49f87b'),(10345,25310,'#f7dd76'),(10346,25313,'#f0437c'),(10347,25315,'#dc5866'),(10348,25322,'#a2cf44'),(10349,25324,'#74a5ec'),(10350,25325,'#3ccdc0'),(10351,25326,'#96f160'),(10352,25328,'#ce2f0d'),(10353,25329,'#8840ea'),(10354,25334,'#cae652'),(10355,25335,'#FFFFFF'),(10356,25336,'#e857a8'),(10357,25337,'#FFFFFF'),(10358,25338,'#8f3ae1'),(10359,25339,'#FFFFFF'),(10360,25347,'#9502ea'),(10361,25348,'#eeb229'),(10362,25349,'#28d558'),(10363,25351,'#0ceac2'),(10364,25352,'#0ad75e'),(10365,25355,'#d7cb27'),(10366,25356,'#553af9'),(10367,25361,'#60f834'),(10368,25362,'#FFFFFF'),(10369,25363,'#cbd133'),(10370,25368,'#fc55ac'),(10371,25369,'#30d664'),(10372,25370,'#FFFFFF'),(10373,25373,'#6afcde'),(10374,25377,'#76e96d'),(10375,25379,'#FFFFFF'),(10376,25382,'#6ad951'),(10377,25383,'#23b6d4'),(10378,25385,'#fb0373'),(10379,25393,'#e4a96b'),(10380,25394,'#FFFFFF'),(10381,25396,'#88d63c'),(10382,25397,'#6ff13b'),(10383,25398,'#1dc2d8'),(10384,25399,'#5f19f4'),(10385,25400,'#ecca5f'),(10386,25403,'#FFFFFF'),(10387,25408,'#09d7df'),(10388,25410,'#4a9fef'),(10389,25411,'#FFFFFF'),(10390,25414,'#FFFFFF'),(10391,25415,'#68dca3'),(10392,25416,'#d8597b'),(10393,25423,'#FFFFFF'),(10394,25424,'#FFFFFF'),(10395,25425,'#da0405'),(10396,25432,'#2b37fb'),(10397,25433,'#ddab55'),(10398,25434,'#e87466'),(10399,25436,'#FFFFFF'),(10400,25438,'#edda23'),(10401,25440,'#FFFFFF'),(10402,25441,'#fe8065'),(10403,25443,'#d501a5'),(10404,25444,'#5fc3ed'),(10405,25446,'#9ef57a'),(10406,25449,'#FFFFFF'),(10407,25455,'#FFFFFF'),(10408,25456,'#e125b9'),(10409,25457,'#d94434'),(10410,25461,'#2042f6'),(10411,25468,'#3fcf8b'),(10412,25479,'#FFFFFF'),(10413,25484,'#45d598'),(10414,25488,'#659fdd'),(10415,25489,'#1cf935'),(10416,25492,'#9628cd'),(10417,25493,'#cd3238'),(10418,25497,'#2e73e4'),(10419,25499,'#fc8f37'),(10420,25501,'#FFFFFF'),(10421,25502,'#02db9e'),(10422,25503,'#dcd83f'),(10423,25507,'#3fd699'),(10424,25510,'#FFFFFF'),(10425,25516,'#e5ca57'),(10426,25518,'#91fe60'),(10427,25520,'#93d203'),(10428,25522,'#FFFFFF'),(10429,25527,'#FFFFFF'),(10430,25532,'#8f60e1'),(10431,25537,'#FFFFFF'),(10432,25539,'#f8eb39'),(10433,25542,'#d92df6'),(10434,25543,'#FFFFFF'),(10435,25545,'#FFFFFF'),(10436,25546,'#3604e9'),(10437,25548,'#d5a61d'),(10438,25550,'#d424a4'),(10439,25551,'#FFFFFF'),(10440,25554,'#ed24a3'),(10441,25557,'#FFFFFF'),(10442,25558,'#FFFFFF'),(10443,25559,'#ea3f0d'),(10444,25562,'#3fddf3'),(10445,25567,'#e75919'),(10446,25569,'#c762d8'),(10447,25583,'#fad472'),(10448,25584,'#d025ec'),(10449,25588,'#FFFFFF'),(10450,25589,'#ad3bcc'),(10451,25590,'#6ccd00'),(10452,25592,'#f67e51'),(10453,25593,'#cc1dcd'),(10454,25597,'#FFFFFF'),(10455,25599,'#26fcf0'),(10456,25603,'#e15a93'),(10457,25604,'#fb1a43'),(10458,25606,'#4568e6'),(10459,25607,'#71f1c3'),(10460,25608,'#fd0b29'),(10461,25610,'#cf6665'),(10462,25611,'#d0eb35'),(10463,25612,'#e3e06f'),(10464,25619,'#7a22ea'),(10465,25620,'#FFFFFF'),(10466,25621,'#68d272'),(10467,25623,'#6a60e9'),(10468,25625,'#dd926e'),(10469,25626,'#FFFFFF'),(10470,25628,'#d727d6'),(10471,25631,'#dc0ab6'),(10472,25632,'#f83a1c'),(10473,25633,'#FFFFFF'),(10474,25634,'#FFFFFF'),(10475,25636,'#da865e'),(10476,25639,'#FFFFFF'),(10477,25640,'#FFFFFF'),(10478,25642,'#58dfe3'),(10479,25646,'#096bd4'),(10480,25648,'#d65e16'),(10481,25653,'#fd7c5c'),(10482,25654,'#7fea48'),(10483,25662,'#3dc8ec'),(10484,25663,'#f77e04'),(10485,25664,'#05e595'),(10486,25667,'#FFFFFF'),(10487,25673,'#d2c231'),(10488,25676,'#44f39a'),(10489,25679,'#ce5703'),(10490,25680,'#4d01da'),(10491,25683,'#db7224'),(10492,25687,'#bcd126'),(10493,25688,'#6fb7eb'),(10494,25689,'#13ee16'),(10495,25690,'#eefc70'),(10496,25695,'#bbec25'),(10497,25697,'#5dbfd6'),(10498,25698,'#5acee7'),(10499,25706,'#3d5af0'),(10500,25707,'#e25471'),(10501,25708,'#afe75a'),(10502,25711,'#71e424'),(10503,25716,'#b1e16b'),(10504,25719,'#d7b35b'),(10505,25720,'#77d53c'),(10506,25724,'#4b56dd'),(10507,25725,'#b9f23c'),(10508,25726,'#dce41e'),(10509,25728,'#FFFFFF'),(10510,25729,'#dac20a'),(10511,25732,'#f15ee8'),(10512,25734,'#8c0deb'),(10513,25735,'#3fdad8'),(10514,25748,'#e562cc'),(10515,25749,'#FFFFFF'),(10516,25752,'#5fe3d4'),(10517,25756,'#d8c33a'),(10518,25757,'#6438ce'),(10519,25759,'#d406bb'),(10520,25760,'#FFFFFF'),(10521,25761,'#5dfb51'),(10522,25763,'#92e45e'),(10523,25764,'#44cece'),(10524,25769,'#1af64d'),(10525,25771,'#f97b04'),(10526,25772,'#c139e0'),(10527,25774,'#a14af7'),(10528,25776,'#e7029d'),(10529,25781,'#2ce274'),(10530,25782,'#69e77c'),(10531,25784,'#FFFFFF'),(10532,25789,'#e2ef1d'),(10533,25792,'#34aff1'),(10534,25795,'#9d09df'),(10535,25809,'#e2072e'),(10536,25816,'#f503b2'),(10537,25820,'#70dbe8'),(10538,25825,'#f937f0'),(10539,25826,'#ee5315'),(10540,25827,'#34c4f3'),(10541,25828,'#dd2b1b'),(10542,25829,'#d7612a'),(10543,25831,'#20f78d'),(10544,25835,'#FFFFFF'),(10545,25837,'#44eced'),(10546,25838,'#2969e1'),(10547,25839,'#d9856b'),(10548,25840,'#f2f233'),(10549,25842,'#fd5de8'),(10550,25849,'#1ee0bd'),(10551,25852,'#ce4890'),(10552,25854,'#66f946'),(10553,25857,'#424af1'),(10554,25859,'#11db77'),(10555,25861,'#1add55'),(10556,25863,'#8514ed'),(10557,25870,'#FFFFFF'),(10558,25884,'#FFFFFF'),(10559,25886,'#68d57e'),(10560,25887,'#f3c174'),(10561,25891,'#70f8b7'),(10562,25892,'#32acdf'),(10563,25894,'#4fdc62'),(10564,25896,'#2cfb68'),(10565,25897,'#2cfd78'),(10566,25900,'#FFFFFF'),(10567,25903,'#0360da'),(10568,25904,'#fc6ae0'),(10569,25910,'#FFFFFF'),(10570,25911,'#c9e755'),(10571,25914,'#49fc40'),(10572,25917,'#21abe0'),(10573,25921,'#ee8566'),(10574,25925,'#FFFFFF'),(10575,25930,'#1d20f3'),(10576,25931,'#FFFFFF'),(10577,25937,'#FFFFFF'),(10578,25938,'#FFFFFF'),(10579,25941,'#0bd86d'),(10580,25944,'#FFFFFF'),(10581,25948,'#FFFFFF'),(10582,25950,'#FFFFFF'),(10583,25951,'#FFFFFF'),(10584,25954,'#ea7850'),(10585,25956,'#FFFFFF'),(10586,25958,'#FFFFFF'),(10587,25959,'#3ab5dd'),(10588,25962,'#FFFFFF'),(10589,25963,'#25d2b8'),(10590,25965,'#56cf5e'),(10591,25966,'#6edc68'),(10592,25968,'#d944c3'),(10593,25969,'#3c7dd9'),(10594,25970,'#d74535'),(10595,25971,'#FFFFFF'),(10596,25972,'#FFFFFF'),(10597,25976,'#5bced7'),(10598,25977,'#FFFFFF'),(10599,25979,'#6c92e9'),(10600,25980,'#d51b6b'),(10601,25981,'#dd3297'),(10602,25982,'#627be7'),(10603,25985,'#3608e7'),(10604,25988,'#FFFFFF'),(10605,25997,'#FFFFFF'),(10606,26001,'#4961e4'),(10607,26003,'#1ed1a9'),(10608,26005,'#55e3b3'),(10609,26007,'#e49609'),(10610,26013,'#f45e34'),(10611,26015,'#FFFFFF'),(10612,26019,'#FFFFFF'),(10613,26020,'#52d2ea'),(10614,26021,'#f13811'),(10615,26025,'#FFFFFF'),(10616,26027,'#52cc95'),(10617,26032,'#3495e5'),(10618,26036,'#63d798'),(10619,26041,'#FFFFFF'),(10620,26044,'#FFFFFF'),(10621,26045,'#FFFFFF'),(10622,26051,'#FFFFFF'),(10623,26053,'#62f0b0'),(10624,26054,'#FFFFFF'),(10625,26056,'#37f628'),(10626,26057,'#d420be'),(10627,26061,'#FFFFFF'),(10628,26066,'#FFFFFF'),(10629,26067,'#2adc20'),(10630,26068,'#e2770f'),(10631,26069,'#4d6be7'),(10632,26070,'#30b3d8'),(10633,26071,'#e927f4'),(10634,26076,'#d34155'),(10635,26078,'#39d742'),(10636,26079,'#bfd826'),(10637,26081,'#3de49b'),(10638,26085,'#FFFFFF'),(10639,26086,'#e67c0c'),(10640,26087,'#FFFFFF'),(10641,26088,'#FFFFFF'),(10642,26095,'#dc247d'),(10643,26096,'#dc17a6'),(10644,26097,'#FFFFFF'),(10645,26100,'#FFFFFF'),(10646,26103,'#d58146'),(10647,26115,'#3be741'),(10648,26116,'#FFFFFF'),(10649,26117,'#b2d150'),(10650,26125,'#33e269'),(10651,26128,'#c9fc09'),(10652,26132,'#6dec3b'),(10653,26135,'#FFFFFF'),(10654,26139,'#FFFFFF'),(10655,26140,'#FFFFFF'),(10656,26143,'#3ae4b0'),(10657,26144,'#de14d0'),(10658,26149,'#55ed31'),(10659,26150,'#e15e4a'),(10660,26151,'#cdb713'),(10661,26155,'#FFFFFF'),(10662,26156,'#d09915'),(10663,26162,'#cfb960'),(10664,26165,'#1337d3'),(10665,26170,'#50f3f1'),(10666,26172,'#FFFFFF'),(10667,26179,'#FFFFFF'),(10668,26181,'#5adec8'),(10669,26183,'#FFFFFF'),(10670,26191,'#fe5165'),(10671,26193,'#db3837'),(10672,26195,'#FFFFFF'),(10673,26196,'#FFFFFF'),(10674,26200,'#FFFFFF'),(10675,26202,'#72aff3'),(10676,26203,'#FFFFFF'),(10677,26204,'#FFFFFF'),(10678,26205,'#24fa94'),(10679,26206,'#555afc'),(10680,26207,'#f35f3e'),(10681,26209,'#eade4b'),(10682,26210,'#66d65f'),(10683,26212,'#955ad0'),(10684,26216,'#FFFFFF'),(10685,26217,'#5fceaa'),(10686,26218,'#e6580b'),(10687,26219,'#c925e4'),(10688,26220,'#FFFFFF'),(10689,26222,'#64b6ce'),(10690,26225,'#18d366'),(10691,26226,'#FFFFFF'),(10692,26228,'#c36fec'),(10693,26229,'#cc2664'),(10694,26233,'#cc3799'),(10695,26234,'#d35881'),(10696,26235,'#06db50'),(10697,26236,'#3352d7'),(10698,26237,'#d544d0'),(10699,26238,'#60e5c9'),(10700,26239,'#68ee9f'),(10701,26240,'#d0593e'),(10702,26243,'#63e183'),(10703,26244,'#f9dd07'),(10704,26245,'#ef6ef9'),(10705,26246,'#21db2b'),(10706,26253,'#FFFFFF'),(10707,26254,'#16e363'),(10708,26255,'#a666cd'),(10709,26256,'#5e88f6'),(10710,26259,'#20d217'),(10711,26267,'#6011dc'),(10712,26270,'#44f9c9'),(10713,26274,'#e0fc69'),(10714,26276,'#f4755c'),(10715,26278,'#d06ee3'),(10716,26280,'#d67b2d'),(10717,26282,'#FFFFFF'),(10718,26283,'#2fcd17'),(10719,26284,'#0ad467'),(10720,26286,'#43d0bf'),(10721,26292,'#FFFFFF'),(10722,26294,'#eec405'),(10723,26295,'#FFFFFF'),(10724,26296,'#FFFFFF'),(10725,26297,'#2884f6'),(10726,26301,'#FFFFFF'),(10727,26304,'#FFFFFF'),(10728,26306,'#FFFFFF'),(10729,26310,'#FFFFFF'),(10730,26311,'#71c9e6'),(10731,26312,'#e8cf26'),(10732,26313,'#FFFFFF'),(10733,26315,'#30e56b'),(10734,26318,'#4890ce'),(10735,26319,'#fb30e5'),(10736,26321,'#b5fc37'),(10737,26322,'#09fb43'),(10738,26324,'#e4b055'),(10739,26325,'#d4701f'),(10740,26328,'#d43222'),(10741,26331,'#FFFFFF'),(10742,26332,'#FFFFFF'),(10743,26333,'#38fc2a'),(10744,26334,'#dda203'),(10745,26335,'#fd6942'),(10746,26342,'#045fdc'),(10747,26344,'#577ae7'),(10748,26345,'#FFFFFF'),(10749,26346,'#FFFFFF'),(10750,26348,'#da3349'),(10751,26350,'#FFFFFF'),(10752,26352,'#e40f81'),(10753,26353,'#a261fa'),(10754,26354,'#FFFFFF'),(10755,26355,'#FFFFFF'),(10756,26356,'#f7bf21'),(10757,26357,'#FFFFFF'),(10758,26362,'#d31024'),(10759,26363,'#f4fd71'),(10760,26364,'#FFFFFF'),(10761,26365,'#FFFFFF'),(10762,26367,'#FFFFFF'),(10763,26368,'#FFFFFF'),(10764,26370,'#FFFFFF'),(10765,26373,'#b8f32f'),(10766,26374,'#10d7b4'),(10767,26375,'#f20752'),(10768,26378,'#5434e6'),(10769,26383,'#fab177'),(10770,26384,'#e60dc2'),(10771,26386,'#FFFFFF'),(10772,26388,'#FFFFFF'),(10773,26390,'#FFFFFF'),(10774,26393,'#FFFFFF'),(10775,26395,'#FFFFFF'),(10776,26397,'#f5d048'),(10777,26398,'#6ddc53'),(10778,26400,'#2fe4ca'),(10779,26403,'#FFFFFF'),(10780,26406,'#FFFFFF'),(10781,26410,'#5694dd'),(10782,26411,'#e2f532'),(10783,26412,'#7af20a'),(10784,26414,'#63e173'),(10785,26420,'#e98061'),(10786,26421,'#FFFFFF'),(10787,26422,'#aa0ff9'),(10788,26423,'#3c89cc'),(10789,26424,'#FFFFFF'),(10790,26428,'#bd1eec'),(10791,26429,'#df3c99'),(10792,26432,'#FFFFFF'),(10793,26434,'#FFFFFF'),(10794,26441,'#FFFFFF'),(10795,26442,'#f52afe'),(10796,26445,'#d819bf'),(10797,26447,'#FFFFFF'),(10798,26448,'#FFFFFF'),(10799,26449,'#fb7423'),(10800,26454,'#f12253'),(10801,26455,'#FFFFFF'),(10802,26458,'#FFFFFF'),(10803,26459,'#94d817'),(10804,26463,'#59e395'),(10805,26466,'#FFFFFF'),(10806,26471,'#FFFFFF'),(10807,26474,'#d7925f'),(10808,26483,'#FFFFFF'),(10809,26484,'#FFFFFF'),(10810,26492,'#2cd471'),(10811,26494,'#FFFFFF'),(10812,26495,'#ae1dd7'),(10813,26497,'#fc6de7'),(10814,26498,'#cf0da4'),(10815,26499,'#f63a30'),(10816,26500,'#9ee25b'),(10817,26504,'#FFFFFF'),(10818,26505,'#b933e1'),(10819,26508,'#5cfdc2'),(10820,26509,'#FFFFFF'),(10821,26510,'#FFFFFF'),(10822,26512,'#1b8fe9'),(10823,26515,'#94f658'),(10824,26516,'#40fc92'),(10825,26518,'#FFFFFF'),(10826,26522,'#d2ad48'),(10827,26523,'#5562d2'),(10828,26524,'#FFFFFF'),(10829,26525,'#FFFFFF'),(10830,26527,'#FFFFFF'),(10831,26529,'#f64fce'),(10832,26530,'#FFFFFF'),(10833,26531,'#3d56f5'),(10834,26533,'#d1c502'),(10835,26536,'#4ba2f6'),(10836,26538,'#9cda53'),(10837,26545,'#5cee60'),(10838,26547,'#094afe'),(10839,26548,'#FFFFFF'),(10840,26549,'#FFFFFF'),(10841,26555,'#318fe2'),(10842,26556,'#f1ed64'),(10843,26557,'#1ca1ee'),(10844,26559,'#822beb'),(10845,26560,'#6774e1'),(10846,26562,'#921de3'),(10847,26563,'#f1f005'),(10848,26564,'#c433d6'),(10849,26566,'#9233df'),(10850,26569,'#FFFFFF'),(10851,26571,'#f7883a'),(10852,26572,'#a6f445'),(10853,26574,'#43f7da'),(10854,26575,'#ed0282'),(10855,26577,'#ed5a8e'),(10856,26580,'#6a68ed'),(10857,26581,'#ce6773'),(10858,26583,'#67f6c2'),(10859,26586,'#FFFFFF'),(10860,26593,'#FFFFFF'),(10861,26595,'#FFFFFF'),(10862,26596,'#FFFFFF'),(10863,26600,'#FFFFFF'),(10864,26601,'#FFFFFF'),(10865,26608,'#46a6e3'),(10866,26610,'#FFFFFF'),(10867,26611,'#3e7bd0'),(10868,26612,'#425bf9'),(10869,26613,'#FFFFFF'),(10870,26621,'#3437cf'),(10871,26624,'#2360d8'),(10872,26625,'#50facb'),(10873,26629,'#dcad2c'),(10874,26630,'#cd9728'),(10875,26633,'#FFFFFF'),(10876,26635,'#FFFFFF'),(10877,26636,'#FFFFFF'),(10878,26637,'#df3553'),(10879,26643,'#0cdf3e'),(10880,26645,'#49cc3f'),(10881,26647,'#d60164'),(10882,26648,'#FFFFFF'),(10883,26649,'#cb2ef0'),(10884,26652,'#47e496'),(10885,26655,'#567ae6'),(10886,26659,'#FFFFFF'),(10887,26661,'#b7db4f'),(10888,26663,'#70e46f'),(10889,26664,'#eb1a41'),(10890,26672,'#2ecc07'),(10891,26680,'#a72ed0'),(10892,26685,'#21d81f'),(10893,26698,'#FFFFFF'),(10894,26699,'#5ebcd4'),(10895,26732,'#02d216'),(10896,26743,'#fd5c08'),(10897,26750,'#0aa6cf'),(10898,26771,'#FFFFFF'),(10899,26772,'#FFFFFF'),(10900,26773,'#FFFFFF'),(10901,26778,'#5fd0e4'),(10902,26783,'#FFFFFF'),(10903,26784,'#FFFFFF'),(10904,26787,'#cf3b44'),(10905,26792,'#fd53c7'),(10906,26795,'#FFFFFF'),(10907,26796,'#FFFFFF'),(10908,26797,'#FFFFFF'),(10909,26798,'#cade08'),(10910,26799,'#0c61d3'),(10911,26807,'#f0eb77'),(10912,26809,'#f7f13a'),(10913,26810,'#61dcfa'),(10914,26813,'#b2d045'),(10915,26814,'#FFFFFF'),(10916,26815,'#d48a14'),(10917,26816,'#FFFFFF'),(10918,26817,'#FFFFFF'),(10919,26818,'#FFFFFF'),(10920,26819,'#8643ec'),(10921,26820,'#efa629'),(10922,26821,'#FFFFFF'),(10923,26823,'#5bbcd4'),(10924,26824,'#7af1fb'),(10925,26825,'#FFFFFF'),(10926,26826,'#FFFFFF'),(10927,26827,'#FFFFFF'),(10928,26831,'#a5fd21'),(10929,26832,'#FFFFFF'),(10930,26834,'#b052fa'),(10931,26837,'#FFFFFF'),(10932,26838,'#cd30b8'),(10933,26839,'#e6a157'),(10934,26840,'#6fde6b'),(10935,26841,'#3eead0'),(10936,26842,'#af31d6'),(10937,26846,'#FFFFFF'),(10938,26850,'#FFFFFF'),(10939,26853,'#FFFFFF'),(10940,26854,'#72e78a'),(10941,26861,'#FFFFFF'),(10942,26863,'#2f66ee'),(10943,26864,'#66da41'),(10944,26865,'#FFFFFF'),(10945,26869,'#c5e56f'),(10946,26872,'#81fb33'),(10947,26873,'#ece417'),(10948,26880,'#b422de'),(10949,26883,'#5fd50b'),(10950,26889,'#18db8b'),(10951,26893,'#FFFFFF'),(10952,26896,'#FFFFFF'),(10953,26902,'#378bcf'),(10954,26907,'#d6db23'),(10955,26908,'#d35f6e'),(10956,26910,'#6be6f8'),(10957,26918,'#e72fa1'),(10958,26920,'#f65f5c'),(10959,26922,'#44a2d3'),(10960,26925,'#3bf0af'),(10961,26926,'#FFFFFF'),(10962,26930,'#9913ee'),(10963,26934,'#FFFFFF'),(10964,26936,'#FFFFFF'),(10965,26937,'#deaa01'),(10966,26938,'#FFFFFF'),(10967,26943,'#FFFFFF'),(10968,26946,'#5f8ce8'),(10969,26948,'#cc4d4b'),(10970,26951,'#594ee1'),(10971,26952,'#FFFFFF'),(10972,26954,'#64dbde'),(10973,26955,'#f08448'),(10974,26956,'#3be36c'),(10975,26965,'#f5492c'),(10976,26970,'#FFFFFF'),(10977,26974,'#78fbb1'),(10978,26977,'#3ed90b'),(10979,26979,'#7af703'),(10980,26980,'#FFFFFF'),(10981,26982,'#f0656d'),(10982,26985,'#1a6be4'),(10983,26986,'#FFFFFF'),(10984,26987,'#FFFFFF'),(10985,26989,'#FFFFFF'),(10986,26991,'#FFFFFF'),(10987,26992,'#FFFFFF'),(10988,26993,'#62f3ee'),(10989,26995,'#FFFFFF'),(10990,26996,'#FFFFFF'),(10991,26997,'#b7cf44'),(10992,27000,'#3ff417'),(10993,27001,'#FFFFFF'),(10994,27002,'#FFFFFF'),(10995,27006,'#5a03e0'),(10996,27008,'#FFFFFF'),(10997,27009,'#cf4294'),(10998,27013,'#5a2dd8'),(10999,27017,'#3ff056'),(11000,27018,'#45d8aa'),(11001,27019,'#e4588f'),(11002,27020,'#FFFFFF'),(11003,27021,'#FFFFFF'),(11004,27022,'#FFFFFF'),(11005,27023,'#FFFFFF'),(11006,27025,'#f70295'),(11007,27026,'#31cbe9'),(11008,27027,'#FFFFFF'),(11009,27029,'#FFFFFF'),(11010,27031,'#e2ee5a'),(11011,27032,'#aafc61'),(11012,27033,'#7bdc32'),(11013,27036,'#1209f8'),(11014,27040,'#61d2b1'),(11015,27041,'#4065dd'),(11016,27042,'#FFFFFF'),(11017,27046,'#FFFFFF'),(11018,27051,'#3bdc0d'),(11019,27052,'#3dd997'),(11020,27054,'#FFFFFF'),(11021,27055,'#e15aa7'),(11022,27057,'#ac67d1'),(11023,27058,'#1ad11d'),(11024,27060,'#FFFFFF'),(11025,27062,'#e1a503'),(11026,27087,'#FFFFFF'),(11027,27088,'#FFFFFF'),(11028,27089,'#FFFFFF'),(11029,27090,'#FFFFFF'),(11030,27095,'#FFFFFF'),(11031,27096,'#FFFFFF'),(11032,27097,'#FFFFFF'),(11033,27099,'#FFFFFF'),(11034,27101,'#FFFFFF'),(11035,27104,'#62dd8b'),(11036,27105,'#FFFFFF'),(11037,27107,'#FFFFFF'),(11038,27110,'#FFFFFF'),(11039,27114,'#FFFFFF'),(11040,27115,'#FFFFFF'),(11041,27116,'#FFFFFF'),(11042,27118,'#ad6be7'),(11043,27120,'#FFFFFF'),(11044,27121,'#5068e3'),(11045,27122,'#FFFFFF'),(11046,27123,'#FFFFFF'),(11047,27124,'#FFFFFF'),(11048,27126,'#538ff0'),(11049,27128,'#FFFFFF'),(11050,27129,'#35d33a'),(11051,27131,'#a65cdd'),(11052,27134,'#FFFFFF'),(11053,27137,'#FFFFFF'),(11054,27139,'#FFFFFF'),(11055,27141,'#FFFFFF'),(11056,27142,'#FFFFFF'),(11057,27143,'#befa43'),(11058,27144,'#FFFFFF'),(11059,27146,'#FFFFFF'),(11060,27149,'#20d228'),(11061,27152,'#FFFFFF'),(11062,27153,'#FFFFFF'),(11063,27154,'#FFFFFF'),(11064,27155,'#FFFFFF'),(11065,27158,'#FFFFFF'),(11066,27159,'#FFFFFF'),(11067,27160,'#FFFFFF'),(11068,27161,'#FFFFFF'),(11069,27162,'#c36ad6'),(11070,27164,'#FFFFFF'),(11071,27165,'#FFFFFF'),(11072,27168,'#FFFFFF'),(11073,27169,'#50dbf3'),(11074,27172,'#FFFFFF'),(11075,27173,'#FFFFFF'),(11076,27175,'#FFFFFF'),(11077,27177,'#FFFFFF'),(11078,27178,'#FFFFFF'),(11079,27179,'#FFFFFF'),(11080,27181,'#FFFFFF'),(11081,27183,'#FFFFFF'),(11082,27185,'#FFFFFF'),(11083,27192,'#FFFFFF'),(11084,27195,'#a8d509'),(11085,27196,'#FFFFFF'),(11086,27199,'#FFFFFF'),(11087,27201,'#FFFFFF'),(11088,27203,'#2fbcdb'),(11089,27204,'#FFFFFF'),(11090,27205,'#FFFFFF'),(11091,27206,'#FFFFFF'),(11092,27207,'#FFFFFF'),(11093,27208,'#FFFFFF'),(11094,27210,'#3f81db'),(11095,27211,'#FFFFFF'),(11096,27212,'#2ce21f'),(11097,27213,'#FFFFFF'),(11098,27216,'#FFFFFF'),(11099,27217,'#186bda'),(11100,27219,'#df0f57'),(11101,27222,'#FFFFFF'),(11102,27223,'#fb1054'),(11103,27224,'#FFFFFF'),(11104,27225,'#FFFFFF'),(11105,27227,'#9146d5'),(11106,27229,'#FFFFFF'),(11107,27232,'#FFFFFF'),(11108,27234,'#FFFFFF'),(11109,27235,'#FFFFFF'),(11110,27237,'#FFFFFF'),(11111,27241,'#FFFFFF'),(11112,27242,'#FFFFFF'),(11113,27243,'#c0d11e'),(11114,27244,'#0beb46'),(11115,27245,'#FFFFFF'),(11116,27246,'#482fd4'),(11117,27247,'#FFFFFF'),(11118,27248,'#FFFFFF'),(11119,27252,'#FFFFFF'),(11120,27253,'#240ef0'),(11121,27254,'#FFFFFF'),(11122,27255,'#00f8f5'),(11123,27258,'#FFFFFF'),(11124,27259,'#d40a14'),(11125,27266,'#38ea5b'),(11126,27267,'#FFFFFF'),(11127,27268,'#c04fe9'),(11128,27269,'#FFFFFF'),(11129,27270,'#FFFFFF'),(11130,27271,'#FFFFFF'),(11131,27273,'#d81acc'),(11132,27274,'#FFFFFF'),(11133,27280,'#327ee4'),(11134,27282,'#FFFFFF'),(11135,27289,'#4980eb'),(11136,27290,'#b6f467'),(11137,27292,'#df6e7d'),(11138,27307,'#e2f04b'),(11139,27313,'#FFFFFF'),(11140,27315,'#f6f516'),(11141,27320,'#f829f8'),(11142,27326,'#f544ee'),(11143,27327,'#ec6196'),(11144,27334,'#FFFFFF'),(11145,27344,'#6526f9'),(11146,27347,'#0497f6'),(11147,27359,'#79f853'),(11148,27363,'#0ccdfe'),(11149,27366,'#4eeb74'),(11150,27367,'#FFFFFF'),(11151,27368,'#946be1'),(11152,27370,'#7bd00e'),(11153,27379,'#FFFFFF'),(11154,27382,'#43fea3'),(11155,27386,'#f71aef'),(11156,27390,'#FFFFFF'),(11157,27391,'#37dab5'),(11158,27398,'#ec6d69'),(11159,27402,'#0fd677'),(11160,27416,'#6ad6a2'),(11161,27417,'#FFFFFF'),(11162,27420,'#f1572e'),(11163,27423,'#e4bb21'),(11164,27424,'#d37d62'),(11165,27425,'#d75179'),(11166,27426,'#FFFFFF'),(11167,27428,'#FFFFFF'),(11168,27431,'#FFFFFF'),(11169,27433,'#1dd6d6'),(11170,27434,'#1fa4e1'),(11171,27435,'#FFFFFF'),(11172,27439,'#dcc403'),(11173,27440,'#1a08d5'),(11174,27441,'#fa0f81'),(11175,27448,'#FFFFFF'),(11176,27450,'#bacc23'),(11177,27455,'#2c68fe'),(11178,27457,'#ebb62b'),(11179,27458,'#eb5612'),(11180,27464,'#6a97ec'),(11181,27470,'#fc9050'),(11182,27472,'#327fd7'),(11183,27476,'#FFFFFF'),(11184,27480,'#6868d1'),(11185,27483,'#52f894'),(11186,27485,'#ea5750'),(11187,27488,'#0ae8b8'),(11188,27489,'#de22fb'),(11189,27492,'#1ddf29'),(11190,27493,'#2f59e3'),(11191,27494,'#ee28aa'),(11192,27497,'#3c39f2'),(11193,27500,'#FFFFFF'),(11194,27503,'#5733e5'),(11195,27511,'#FFFFFF'),(11196,27513,'#d54e40'),(11197,27514,'#30b4da'),(11198,27515,'#FFFFFF'),(11199,27517,'#4929e5'),(11200,27524,'#6395d2'),(11201,27528,'#b551cd'),(11202,27529,'#e63ae1'),(11203,27549,'#f3b726'),(11204,27557,'#fc0827'),(11205,27559,'#55d1c8'),(11206,27561,'#d864eb'),(11207,27562,'#f5c673'),(11208,27563,'#18f4ea'),(11209,27565,'#d12149'),(11210,27566,'#85fc63'),(11211,27567,'#ceb257'),(11212,27568,'#1eda2d'),(11213,27569,'#be5eda'),(11214,27570,'#e9bd33'),(11215,27573,'#f377a8'),(11216,27574,'#f5d71c'),(11217,27575,'#6463d9'),(11218,27576,'#6f0bcf'),(11219,27578,'#2a65fe'),(11220,27579,'#2a05f2'),(11221,27580,'#3390d3'),(11222,27581,'#cefa15'),(11223,27582,'#1cc0eb'),(11224,27583,'#f86448'),(11225,27585,'#f6a20c'),(11226,27586,'#26dfa2'),(11227,27587,'#1deefd'),(11228,27589,'#fd9974'),(11229,27590,'#9412d5'),(11230,27591,'#e90112'),(11231,27592,'#d4cb5c'),(11232,27593,'#e24e58'),(11233,27594,'#fa528d'),(11234,27595,'#22e32b'),(11235,27597,'#46d220'),(11236,27598,'#6fee7f'),(11237,27599,'#5d97f7'),(11238,27600,'#84f93c'),(11239,27601,'#38d3f8'),(11240,27602,'#5de543'),(11241,27603,'#5c34f9'),(11242,27604,'#2935db'),(11243,27605,'#d6258f'),(11244,27607,'#f83115'),(11245,27610,'#572edb'),(11246,27614,'#e1d953'),(11247,27615,'#0ccd21'),(11248,27616,'#4449d6'),(11249,27617,'#c956d0'),(11250,27619,'#d4731a'),(11251,27620,'#3f87e1'),(11252,27621,'#f22b33'),(11253,27622,'#2ee6e7'),(11254,27624,'#5990f5'),(11255,27625,'#10e95b'),(11256,27626,'#5f5bee'),(11257,27627,'#5c59e9'),(11258,27628,'#4fdc97'),(11259,27629,'#1081dd'),(11260,27630,'#4cdab8'),(11261,27632,'#2fd911'),(11262,27634,'#fd6075'),(11263,27638,'#f55037'),(11264,27639,'#65e09b'),(11265,27640,'#71c8f7'),(11266,27642,'#1ce1e0'),(11267,27643,'#33f6c2'),(11268,27644,'#d4ca12'),(11269,27646,'#FFFFFF'),(11270,27647,'#e91a5b'),(11271,27648,'#6c99f3'),(11272,27650,'#4b07dc'),(11273,27651,'#1af0fa'),(11274,27652,'#c8fb7b'),(11275,27653,'#db2855'),(11276,27654,'#aaf842'),(11277,27655,'#c359d1'),(11278,27656,'#3383f7'),(11279,27657,'#3223fb'),(11280,27659,'#c9dc15'),(11281,27660,'#7bd427'),(11282,27661,'#ec0bdf'),(11283,27662,'#d19120'),(11284,27664,'#4ee2dd'),(11285,27665,'#5de9bd'),(11286,27667,'#e7394d'),(11287,27668,'#f5872b'),(11288,27669,'#7539fb'),(11289,27670,'#a9e549'),(11290,27671,'#bb48d1'),(11291,27674,'#bd51dd'),(11292,27675,'#2bebcc'),(11293,27676,'#4d1bee'),(11294,27678,'#daf44f'),(11295,27680,'#9bf317'),(11296,27681,'#42e9aa'),(11297,27682,'#FFFFFF'),(11298,27684,'#57facb'),(11299,27686,'#7446ef'),(11300,27687,'#f477b4'),(11301,27689,'#FFFFFF'),(11302,27691,'#d1011e'),(11303,27692,'#66ecc4'),(11304,27694,'#f0027d'),(11305,27696,'#0afb45'),(11306,27697,'#ae07d3'),(11307,27699,'#34ec07'),(11308,27700,'#48dc5c'),(11309,27702,'#a642d7'),(11310,27703,'#67da5f'),(11311,27704,'#21ce9e'),(11312,27710,'#3efac9'),(11313,27711,'#f3e814'),(11314,27712,'#3cf495'),(11315,27713,'#6efe7d'),(11316,27714,'#2204dc'),(11317,27716,'#dfaf2c'),(11318,27717,'#fa3941'),(11319,27720,'#4f99df'),(11320,27722,'#cd3767'),(11321,27723,'#053aee'),(11322,27724,'#c346f7'),(11323,27725,'#64fb36'),(11324,27727,'#ecd832'),(11325,27728,'#f3ad33'),(11326,27730,'#ea3ba4'),(11327,27732,'#573dfd'),(11328,27733,'#4827ef'),(11329,27734,'#39e65a'),(11330,27736,'#9cef17'),(11331,27737,'#dc4702'),(11332,27740,'#f24147'),(11333,27741,'#e536a5'),(11334,27742,'#ab1fde'),(11335,27745,'#FFFFFF'),(11336,27746,'#73fb15'),(11337,27748,'#eda260'),(11338,27749,'#42e24f'),(11339,27751,'#3138cf'),(11340,27752,'#48b9eb'),(11341,27753,'#1ef92b'),(11342,27760,'#e48a36'),(11343,27761,'#6488d0'),(11344,27763,'#db14fd'),(11345,27764,'#16edd9'),(11346,27769,'#f37d6c'),(11347,27771,'#3067d7'),(11348,27772,'#b6d114'),(11349,27775,'#d9ab54'),(11350,27776,'#cee822'),(11351,27777,'#012cd9'),(11352,27778,'#2106fb'),(11353,27780,'#29f37f'),(11354,27782,'#4826df'),(11355,27783,'#a34be7'),(11356,27784,'#4719d9'),(11357,27785,'#5e16d8'),(11358,27787,'#5e2ecc'),(11359,27788,'#72f487'),(11360,27789,'#f410f1'),(11361,27790,'#d69048'),(11362,27791,'#51eb9f'),(11363,27792,'#cbef13'),(11364,27794,'#f55b24'),(11365,27795,'#2ccc71'),(11366,27796,'#4ff5ca'),(11367,27797,'#8508f4'),(11368,27799,'#29e61a'),(11369,27801,'#df3d30'),(11370,27804,'#2df7e7'),(11371,27805,'#5081d3'),(11372,27808,'#0bc2f2'),(11373,27809,'#da9869'),(11374,27810,'#dfd43b'),(11375,27812,'#4e4bf7'),(11376,27813,'#6be227'),(11377,27815,'#e05c6a'),(11378,27816,'#69dbdb'),(11379,27817,'#20d209'),(11380,27818,'#54e16c'),(11381,27819,'#403dd6'),(11382,27821,'#e12525'),(11383,27822,'#56d301'),(11384,27824,'#71ecfa'),(11385,27826,'#5bda79'),(11386,27828,'#edaa72'),(11387,27829,'#d65134'),(11388,27831,'#1be8df'),(11389,27834,'#cf0555'),(11390,27835,'#0237e5'),(11391,27836,'#6bf1aa'),(11392,27838,'#ef563f'),(11393,27840,'#fe6bb0'),(11394,27841,'#77f97b'),(11395,27843,'#d78c2a'),(11396,27844,'#d6f105'),(11397,27845,'#df42ef'),(11398,27846,'#9ee012'),(11399,27847,'#ad6dde'),(11400,27848,'#7725cf'),(11401,27851,'#18fea2'),(11402,27852,'#cc5e07'),(11403,27856,'#20d7ba'),(11404,27860,'#FFFFFF'),(11405,27861,'#37f672'),(11406,27862,'#ed694f'),(11407,27864,'#cfee2d'),(11408,27865,'#f0397d'),(11409,27866,'#df25b7'),(11410,27869,'#dd269d'),(11411,27875,'#db68a5'),(11412,27876,'#41eda8'),(11413,27878,'#42ed4e'),(11414,27882,'#223edc'),(11415,27883,'#0705d2'),(11416,27887,'#bfcf18'),(11417,27889,'#9a5df0'),(11418,27890,'#2d58de'),(11419,27891,'#f93cdd'),(11420,27892,'#f63773'),(11421,27893,'#cbcd0b'),(11422,27895,'#1ee3bf'),(11423,27896,'#19f60e'),(11424,27897,'#02b7ce'),(11425,27898,'#fa4700'),(11426,27899,'#b0f135'),(11427,27901,'#fbcd45'),(11428,27902,'#eb2eeb'),(11429,27903,'#1cde75'),(11430,27905,'#e465ba'),(11431,27906,'#f153a6'),(11432,27907,'#e8618c'),(11433,27909,'#4ed85c'),(11434,27910,'#cc4b1a'),(11435,27911,'#d77afe'),(11436,27912,'#1ef570'),(11437,27913,'#FFFFFF'),(11438,27915,'#d6f50a'),(11439,27916,'#92dc25'),(11440,27917,'#4963dc'),(11441,27921,'#6d06ea'),(11442,27925,'#fcb270'),(11443,27926,'#2df103'),(11444,27927,'#456dd4'),(11445,27931,'#521cd4'),(11446,27932,'#7ecd26'),(11447,27933,'#f57019'),(11448,27934,'#2ae710'),(11449,27935,'#cfc054'),(11450,27936,'#e80ab1'),(11451,27937,'#d37f46'),(11452,27939,'#3263cc'),(11453,27940,'#46d602'),(11454,27943,'#2dc3cd'),(11455,27945,'#cc8154'),(11456,27946,'#d24850'),(11457,27948,'#FFFFFF'),(11458,27949,'#f61d6d'),(11459,27950,'#d96172'),(11460,27951,'#fde86d'),(11461,27952,'#b1d650'),(11462,27954,'#dca92b'),(11463,27955,'#e31fb4'),(11464,27957,'#47f0eb'),(11465,27958,'#ea2d7a'),(11466,27959,'#da0268'),(11467,27961,'#9edd39'),(11468,27962,'#FFFFFF'),(11469,27963,'#FFFFFF'),(11470,27966,'#FFFFFF'),(11471,27967,'#fa52b4'),(11472,27968,'#5ef258'),(11473,27969,'#fc584c'),(11474,27970,'#5dcc8e'),(11475,27971,'#d934a3'),(11476,27972,'#cda604'),(11477,27973,'#f372ad'),(11478,27974,'#32e84a'),(11479,27975,'#FFFFFF'),(11480,27977,'#e7251e'),(11481,27978,'#aed809'),(11482,27979,'#46d1ab'),(11483,27981,'#d92af6'),(11484,27982,'#ce555b'),(11485,27984,'#62e4cb'),(11486,27985,'#2639e3'),(11487,27987,'#0bcce7'),(11488,27988,'#e9599f'),(11489,27989,'#eadf2f'),(11490,27990,'#df454b'),(11491,27992,'#da1672'),(11492,27994,'#0d3ad1'),(11493,27995,'#e33c33'),(11494,27996,'#f52d78'),(11495,27997,'#38fe1d'),(11496,27999,'#1359e5'),(11497,28000,'#3de559'),(11498,28005,'#5d3dd9'),(11499,28007,'#8701ec'),(11500,28008,'#6c4fe7'),(11501,28009,'#2c84de'),(11502,28010,'#d237e1'),(11503,28011,'#6bf457'),(11504,28012,'#e52b5e'),(11505,28013,'#68ecc1'),(11506,28015,'#FFFFFF'),(11507,28016,'#3cf305'),(11508,28017,'#0d0bd3'),(11509,28018,'#d33c96'),(11510,28019,'#6dfcc2'),(11511,28020,'#f75b59'),(11512,28024,'#e8ef58'),(11513,28026,'#5062ce'),(11514,28027,'#d4423b'),(11515,28028,'#76d338'),(11516,28029,'#FFFFFF'),(11517,28030,'#51d92a'),(11518,28031,'#33e8a2'),(11519,28033,'#37d5bc'),(11520,28036,'#d02779'),(11521,28040,'#4642ec'),(11522,28041,'#20f18a'),(11523,28042,'#5872d1'),(11524,28043,'#d121c1'),(11525,28044,'#FFFFFF'),(11526,28047,'#d8c202'),(11527,28050,'#e9a361'),(11528,28051,'#6ceb6e'),(11529,28054,'#d24489'),(11530,28055,'#6bdc31'),(11531,28058,'#19ccfb'),(11532,28061,'#1dea40'),(11533,28062,'#5af1b0'),(11534,28066,'#e7b915'),(11535,28067,'#76e337'),(11536,28068,'#57b8db'),(11537,28069,'#0b24e7'),(11538,28070,'#f762c4'),(11539,28071,'#b7fb75'),(11540,28072,'#FFFFFF'),(11541,28074,'#e9d712'),(11542,28076,'#3640cc'),(11543,28077,'#715fe9'),(11544,28078,'#7dd80c'),(11545,28081,'#14adcf'),(11546,28082,'#FFFFFF'),(11547,28083,'#26f35e'),(11548,28084,'#cdde22'),(11549,28087,'#FFFFFF'),(11550,28115,'#36e4c3'),(11551,28134,'#cc0351'),(11552,28142,'#fd3e0a'),(11553,28150,'#4be67b'),(11554,28168,'#369ed0'),(11555,28178,'#FFFFFF'),(11556,28181,'#3ded1c'),(11557,28185,'#FFFFFF'),(11558,28193,'#2748d1'),(11559,28200,'#f41a55'),(11560,28201,'#4a19e9'),(11561,28202,'#88ea09'),(11562,28207,'#FFFFFF'),(11563,28210,'#f26942'),(11564,28217,'#29f618'),(11565,28219,'#FFFFFF'),(11566,28225,'#FFFFFF'),(11567,28227,'#0408d8'),(11568,28229,'#FFFFFF'),(11569,28230,'#5ae4eb'),(11570,28231,'#ebd852'),(11571,28236,'#e86869'),(11572,28237,'#FFFFFF'),(11573,28241,'#db4af2'),(11574,28242,'#da784d'),(11575,28247,'#3b33cf'),(11576,28249,'#66a2f0'),(11577,28253,'#e71130'),(11578,28257,'#FFFFFF'),(11579,28260,'#9e37f3'),(11580,28262,'#FFFFFF'),(11581,28265,'#FFFFFF'),(11582,28267,'#FFFFFF'),(11583,28269,'#36f028'),(11584,28270,'#FFFFFF'),(11585,28272,'#FFFFFF'),(11586,28276,'#e13b7e'),(11587,28277,'#FFFFFF'),(11588,28280,'#6313ef'),(11589,28283,'#b54afc'),(11590,28288,'#FFFFFF'),(11591,28289,'#5175f1'),(11592,28290,'#1092d6'),(11593,28296,'#de4998'),(11594,28297,'#FFFFFF'),(11595,28299,'#FFFFFF'),(11596,28306,'#a9fb0d'),(11597,28312,'#FFFFFF'),(11598,28313,'#FFFFFF'),(11599,28315,'#444ad2'),(11600,28318,'#FFFFFF'),(11601,28323,'#0746fd'),(11602,28324,'#39e58a'),(11603,28333,'#FFFFFF'),(11604,28335,'#FFFFFF'),(11605,28336,'#eb7e1a'),(11606,28337,'#FFFFFF'),(11607,28338,'#e14954'),(11608,28339,'#FFFFFF'),(11609,28346,'#1bd94d'),(11610,28348,'#419edf'),(11611,28350,'#b55cde'),(11612,28351,'#b4ed2e'),(11613,28353,'#FFFFFF'),(11614,28359,'#FFFFFF'),(11615,28362,'#0cfa98'),(11616,28363,'#d44832'),(11617,28364,'#15f75e'),(11618,28365,'#df1615'),(11619,28366,'#547dcc'),(11620,28367,'#e6c559'),(11621,28368,'#287fec'),(11622,28373,'#ee687d'),(11623,28378,'#ce380d'),(11624,28379,'#55ecde'),(11625,28380,'#84e513'),(11626,28381,'#FFFFFF'),(11627,28383,'#FFFFFF'),(11628,28386,'#28ede9'),(11629,28390,'#a762dc'),(11630,28392,'#FFFFFF'),(11631,28394,'#FFFFFF'),(11632,28398,'#e4b962'),(11633,28400,'#FFFFFF'),(11634,28401,'#36fb67'),(11635,28403,'#3494f9'),(11636,28404,'#FFFFFF'),(11637,28407,'#ba3bee'),(11638,28408,'#FFFFFF'),(11639,28409,'#ce8a0d'),(11640,28410,'#68b2fd'),(11641,28413,'#fd0b97'),(11642,28417,'#39d404'),(11643,28418,'#e6a023'),(11644,28420,'#e8150d'),(11645,28427,'#FFFFFF'),(11646,28429,'#FFFFFF'),(11647,28434,'#85d439'),(11648,28435,'#FFFFFF'),(11649,28436,'#4cb5d1'),(11650,28437,'#FFFFFF'),(11651,28441,'#3b75e0'),(11652,28442,'#73e965'),(11653,28445,'#FFFFFF'),(11654,28447,'#89fc29'),(11655,28452,'#e808ad'),(11656,28453,'#FFFFFF'),(11657,28462,'#FFFFFF'),(11658,28463,'#FFFFFF'),(11659,28464,'#FFFFFF'),(11660,28472,'#d0d75f'),(11661,28473,'#d4a219'),(11662,28474,'#fe7928'),(11663,28476,'#ac21d9'),(11664,28478,'#FFFFFF'),(11665,28479,'#FFFFFF'),(11666,28482,'#fbeb13'),(11667,28484,'#4ffddd'),(11668,28487,'#FFFFFF'),(11669,28490,'#c233dd'),(11670,28496,'#FFFFFF'),(11671,28497,'#4813fe'),(11672,28504,'#38d19b'),(11673,28506,'#c0e163'),(11674,28507,'#912fd6'),(11675,28511,'#eb5cc8'),(11676,28512,'#36c1d5'),(11677,28515,'#FFFFFF'),(11678,28518,'#f5581a'),(11679,28520,'#FFFFFF'),(11680,28521,'#34c8da'),(11681,28525,'#e47b19'),(11682,28526,'#FFFFFF'),(11683,28528,'#27cd7f'),(11684,28529,'#aaee6f'),(11685,28530,'#dbf426'),(11686,28532,'#FFFFFF'),(11687,28535,'#32eb5e'),(11688,28536,'#cc06fd'),(11689,28538,'#FFFFFF'),(11690,28541,'#2af9a4'),(11691,28544,'#177ef5'),(11692,28545,'#8f19fe'),(11693,28546,'#2a87ec'),(11694,28547,'#db15f3'),(11695,28549,'#fa1ba0'),(11696,28550,'#FFFFFF'),(11697,28553,'#01cae1'),(11698,28554,'#61f24d'),(11699,28555,'#FFFFFF'),(11700,28557,'#9a1cf0'),(11701,28559,'#f5f621'),(11702,28564,'#9854e4'),(11703,28565,'#FFFFFF'),(11704,28567,'#0cd9d9'),(11705,28568,'#FFFFFF'),(11706,28577,'#d35669'),(11707,28579,'#cc0d74'),(11708,28581,'#58f995'),(11709,28585,'#FFFFFF'),(11710,28587,'#FFFFFF'),(11711,28588,'#8bed65'),(11712,28617,'#3cb1d2'),(11713,28641,'#15d113'),(11714,28645,'#6075cf'),(11715,28646,'#17dadd'),(11716,28649,'#0bdb7d'),(11717,28651,'#0816e7'),(11718,28661,'#79ee3c'),(11719,28672,'#45f250'),(11720,28673,'#bf60d9'),(11721,28676,'#46cbcd'),(11722,28682,'#ccc812'),(11723,28687,'#2cf0b5'),(11724,28697,'#f6c42e'),(11725,28699,'#30de3d'),(11726,28711,'#43e53f'),(11727,28713,'#deb10b'),(11728,28719,'#67d536'),(11729,28721,'#da578f'),(11730,28724,'#5b70ea'),(11731,28725,'#ab26e2'),(11732,28726,'#4bd340'),(11733,28740,'#23fde1'),(11734,28746,'#1edc1d'),(11735,28747,'#3fe785'),(11736,28756,'#a84aee'),(11737,28764,'#3fd41a'),(11738,28777,'#00da6e'),(11739,28783,'#FFFFFF'),(11740,28805,'#d9cd2f'),(11741,28806,'#13cdae'),(11742,28818,'#FFFFFF'),(11743,28819,'#FFFFFF'),(11744,28824,'#17d0c1'),(11745,28828,'#FFFFFF'),(11746,28831,'#FFFFFF'),(11747,28833,'#f78c5b'),(11748,28837,'#FFFFFF'),(11749,28839,'#FFFFFF'),(11750,28840,'#FFFFFF'),(11751,28841,'#30d560'),(11752,28848,'#FFFFFF'),(11753,28850,'#5545f5'),(11754,28854,'#FFFFFF'),(11755,28858,'#FFFFFF'),(11756,28859,'#FFFFFF'),(11757,28862,'#FFFFFF'),(11758,28863,'#76aafe'),(11759,28864,'#3445ef'),(11760,28868,'#FFFFFF'),(11761,28869,'#FFFFFF'),(11762,28870,'#FFFFFF'),(11763,28871,'#e6e86e'),(11764,28883,'#6efb27'),(11765,28889,'#2713f8'),(11766,28898,'#0a61de'),(11767,28903,'#3c13f4'),(11768,28904,'#e1c653'),(11769,28907,'#75b5ed'),(11770,28908,'#7dfa2d'),(11771,28911,'#60a6f0'),(11772,28912,'#6e22d0'),(11773,28913,'#5a3cf5'),(11774,28915,'#cff535'),(11775,28916,'#d1bd67'),(11776,28917,'#d96ae3'),(11777,28921,'#36f4d6'),(11778,28923,'#cd713e'),(11779,28928,'#0fd970'),(11780,28931,'#FFFFFF'),(11781,28932,'#e21cc4'),(11782,28935,'#e0f433'),(11783,28939,'#FFFFFF'),(11784,28942,'#e443b7'),(11785,28945,'#cfc65f'),(11786,28946,'#FFFFFF'),(11787,28948,'#47a5eb'),(11788,28950,'#d60e26'),(11789,28953,'#63e5c3'),(11790,28956,'#FFFFFF'),(11791,28958,'#FFFFFF'),(11792,28959,'#FFFFFF'),(11793,28960,'#FFFFFF'),(11794,28964,'#47ec74'),(11795,28966,'#952ce4'),(11796,28968,'#6965f0'),(11797,28969,'#FFFFFF'),(11798,28972,'#d3770f'),(11799,28973,'#50acf0'),(11800,28976,'#9af01c'),(11801,28978,'#4090cf'),(11802,28979,'#7c46f9'),(11803,28980,'#e82140'),(11804,28984,'#5ab4d6'),(11805,28985,'#8e2af6'),(11806,28987,'#f27269'),(11807,28988,'#528cd1'),(11808,28989,'#0ada28'),(11809,28991,'#bb2ad1'),(11810,28992,'#f76436'),(11811,28993,'#FFFFFF'),(11812,29002,'#FFFFFF'),(11813,29013,'#FFFFFF'),(11814,29015,'#e2573d'),(11815,29018,'#c9ea74'),(11816,29019,'#d68658'),(11817,29031,'#80f36b'),(11818,29038,'#FFFFFF'),(11819,29042,'#FFFFFF'),(11820,29049,'#db6aa7'),(11821,29052,'#FFFFFF'),(11822,29054,'#FFFFFF'),(11823,29063,'#39e1b7'),(11824,29065,'#a827f8'),(11825,29068,'#ef20ce'),(11826,29071,'#FFFFFF'),(11827,29073,'#b5eb0c'),(11828,29076,'#FFFFFF'),(11829,29084,'#FFFFFF'),(11830,29093,'#a6d515'),(11831,29094,'#FFFFFF'),(11832,29099,'#4cd733'),(11833,29108,'#bacd50'),(11834,29117,'#6aa1ea'),(11835,29121,'#3677cc'),(11836,29122,'#FFFFFF'),(11837,29137,'#4fd3d4'),(11838,29140,'#4794f4'),(11839,29144,'#FFFFFF'),(11840,29161,'#32d083'),(11841,29163,'#2853f4'),(11842,29169,'#e4614d'),(11843,29172,'#93ed6a'),(11844,29176,'#8fe865'),(11845,29177,'#6cf909'),(11846,29178,'#FFFFFF'),(11847,29180,'#FFFFFF'),(11848,29182,'#cd7e5a'),(11849,29185,'#FFFFFF'),(11850,29187,'#9ecc36'),(11851,29188,'#FFFFFF'),(11852,29193,'#16e189'),(11853,29198,'#FFFFFF'),(11854,29202,'#FFFFFF'),(11855,29203,'#dc6cf5'),(11856,29205,'#85e919'),(11857,29209,'#63f741'),(11858,29212,'#8745e0'),(11859,29213,'#90ce54'),(11860,29215,'#0ec8ea'),(11861,29220,'#36d2d7'),(11862,29224,'#FFFFFF'),(11863,29225,'#647ed9'),(11864,29227,'#7de418'),(11865,29229,'#FFFFFF'),(11866,29234,'#b017ee'),(11867,29236,'#e79d0a'),(11868,29251,'#d4571a'),(11869,29254,'#FFFFFF'),(11870,29256,'#d71019'),(11871,29258,'#FFFFFF'),(11872,29259,'#e5f861'),(11873,29265,'#FFFFFF'),(11874,29266,'#FFFFFF'),(11875,29270,'#54f5ab'),(11876,29275,'#FFFFFF'),(11877,29278,'#4cb3ec'),(11878,29279,'#FFFFFF'),(11879,29287,'#18ce0b'),(11880,29289,'#4893d3'),(11881,29291,'#17d402'),(11882,29292,'#f7248f'),(11883,29293,'#d64fad'),(11884,29300,'#6df290'),(11885,29306,'#00e553'),(11886,29307,'#4afe21'),(11887,29311,'#FFFFFF'),(11888,29313,'#2ed619'),(11889,29333,'#FFFFFF'),(11890,29334,'#FFFFFF'),(11891,29347,'#966bda'),(11892,29349,'#2efa16'),(11893,29354,'#54e592'),(11894,29359,'#d3b25e'),(11895,29366,'#4e63de'),(11896,29373,'#147cf1'),(11897,29376,'#51e449'),(11898,29389,'#a736eb'),(11899,29397,'#b5cf3b'),(11900,29401,'#cf4cba'),(11901,29407,'#61d089'),(11902,29408,'#d453c3'),(11903,29411,'#f922c6'),(11904,29418,'#34eef6'),(11905,29433,'#3df057'),(11906,29446,'#e68c1d'),(11907,29449,'#ce57bd'),(11908,29450,'#33e365'),(11909,29454,'#cf9c1e'),(11910,29462,'#FFFFFF'),(11911,29475,'#FFFFFF'),(11912,29482,'#FFFFFF'),(11913,29486,'#164ffe'),(11914,29491,'#de4b7f'),(11915,29492,'#1cf431'),(11916,29493,'#44d5bb'),(11917,29513,'#FFFFFF'),(11918,29527,'#33f381'),(11919,29528,'#36dcfd'),(11920,29536,'#72fb56'),(11921,29543,'#b8eb4d'),(11922,29549,'#3647e4'),(11923,29552,'#f0135a'),(11924,29556,'#29f113'),(11925,29570,'#FFFFFF'),(11926,29573,'#dbc86a'),(11927,29575,'#6bf0ad'),(11928,29585,'#5fe72d'),(11929,29588,'#46e5ea'),(11930,29591,'#4cf842'),(11931,29595,'#dc0d7b'),(11932,29606,'#4e1ee0'),(11933,29619,'#13dbad'),(11934,29630,'#d45ce9'),(11935,29632,'#79e95f'),(11936,29636,'#db601b'),(11937,29647,'#cc4b9e'),(11938,29669,'#FFFFFF'),(11939,29748,'#2aedf0'),(11940,29755,'#8fdd67'),(11941,29772,'#d7db63'),(11942,29796,'#d8b40d'),(11943,29806,'#5dc3d5'),(11944,29807,'#e94e83'),(11945,29824,'#638ddc'),(11946,29836,'#74fc77'),(11947,29847,'#6cd8e8'),(11948,29848,'#e84145'),(11949,29909,'#cdd546'),(11950,29926,'#b9e535'),(11951,30005,'#FFFFFF'),(11952,30036,'#c3de16'),(11953,30043,'#2bd851'),(11954,30351,'#fa55e9'),(11955,30388,'#7a6ff2'),(11956,30394,'#FFFFFF'),(11957,30404,'#e13371'),(11958,30407,'#59baf0'),(11959,30412,'#FFFFFF'),(11960,30433,'#c865fd'),(11961,30437,'#cd64b6'),(11962,30447,'#27e06a'),(11963,30470,'#3cf35f'),(11964,30510,'#da299d'),(11965,30516,'#5fd06c'),(11966,30530,'#e18143'),(11967,30536,'#ed113c'),(11968,30560,'#77daf2'),(11969,30564,'#b12be2'),(11970,30565,'#24e13f'),(11971,30566,'#2cbede'),(11972,30583,'#FFFFFF'),(11973,30587,'#b623d2'),(11974,30588,'#57c9d7'),(11975,30594,'#FFFFFF'),(11976,30624,'#fc6159'),(11977,30628,'#feb575'),(11978,30629,'#6dc1e7'),(11979,30630,'#8454d4'),(11980,30639,'#25c2e7'),(11981,30649,'#FFFFFF'),(11982,30657,'#e85612'),(11983,30659,'#FFFFFF'),(11984,30660,'#f68e55'),(11985,30661,'#FFFFFF'),(11986,30664,'#7ebbfd'),(11987,30674,'#FFFFFF'),(11988,30679,'#20fdd0'),(11989,30680,'#ea201c'),(11990,30682,'#65d9fe'),(11991,30685,'#68cbe5'),(11992,30690,'#2bb8d6'),(11993,30693,'#22f0cd'),(11994,30694,'#f35070'),(11995,30696,'#FFFFFF'),(11996,30700,'#69bbfc'),(11997,30702,'#88d350'),(11998,30703,'#1ad7a2'),(11999,30706,'#86d30e'),(12000,30710,'#61fdc2'),(12001,30711,'#f65885'),(12002,30713,'#e35bf4'),(12003,30714,'#57fe85'),(12004,30722,'#6d71e1'),(12005,30734,'#a56bdd'),(12006,30749,'#56d691'),(12007,30754,'#6caff1'),(12008,30790,'#d161b2'),(12009,30867,'#fb9a14'),(12010,30874,'#cd3bc1'),(12011,30882,'#f04e24'),(12012,30900,'#e8e354'),(12013,30907,'#dae202'),(12014,30911,'#e2884f'),(12015,30915,'#fdfa66'),(12016,30916,'#d28118'),(12017,30918,'#4076dc'),(12018,30920,'#e96cc2'),(12019,30933,'#d71627'),(12020,30934,'#4fa5cc'),(12021,30935,'#06f32d'),(12022,30936,'#4ef9a3'),(12023,30940,'#71a6ef'),(12024,30965,'#cdfa72'),(12025,30967,'#5913e7'),(12026,30976,'#6873d9'),(12027,30978,'#2b8fd0'),(12028,30983,'#63d7e5'),(12029,30990,'#587ece'),(12030,30998,'#54e01e'),(12031,31005,'#35d5e9'),(12032,31018,'#54d907'),(12033,31026,'#fc2663'),(12034,31028,'#e84fd9'),(12035,31034,'#6c3df2'),(12036,31035,'#4febf3'),(12037,31037,'#3af20f'),(12038,31038,'#78eb12'),(12039,31041,'#87d855'),(12040,31042,'#ea03e6'),(12041,31050,'#ceeb43'),(12042,31068,'#d55a33'),(12043,31082,'#66daf1'),(12044,31087,'#6de29e'),(12045,31092,'#d1ae4a'),(12046,31108,'#3424df'),(12047,31133,'#6f67dc'),(12048,31137,'#a1ec34'),(12049,31143,'#1d50d4'),(12050,31180,'#e02dfc'),(12051,31210,'#a5fc14'),(12052,31216,'#5adbc1'),(12053,31217,'#ed11c0'),(12054,31219,'#4c75e4'),(12055,31232,'#51e946'),(12056,31237,'#FFFFFF'),(12057,31245,'#FFFFFF'),(12058,31247,'#d51cd5'),(12059,31255,'#d1db65'),(12060,31258,'#d64652'),(12061,31267,'#FFFFFF'),(12062,31272,'#49da5e'),(12063,31275,'#ea03a4'),(12064,31280,'#22a2d1'),(12065,31286,'#05bbe1'),(12066,31287,'#dcd969'),(12067,31290,'#ec42d0'),(12068,31291,'#eb9970'),(12069,31295,'#3ce081'),(12070,31306,'#6627dd'),(12071,31316,'#d60fd6'),(12072,31318,'#FFFFFF'),(12073,31322,'#06d8e6'),(12074,31330,'#4378e8'),(12075,31338,'#e05d8c'),(12076,31346,'#edcb19'),(12077,31362,'#6757e0'),(12078,31363,'#265ede'),(12079,31370,'#FFFFFF'),(12080,31378,'#e2836c'),(12081,31396,'#b9f252'),(12082,31401,'#58f092'),(12083,31409,'#2c6bda'),(12084,31422,'#dae61b'),(12085,31424,'#8f3dfe'),(12086,31427,'#e6527a'),(12087,31433,'#619bf7'),(12088,31434,'#FFFFFF'),(12089,31437,'#48e2be'),(12090,31440,'#cf5ab0'),(12091,31442,'#64eea4'),(12092,31449,'#09c4d3'),(12093,31455,'#7561f6'),(12094,31461,'#4d69e4'),(12095,31471,'#d91c1a'),(12096,31472,'#284de5'),(12097,31473,'#da5133'),(12098,31482,'#FFFFFF'),(12099,31490,'#ea3060'),(12100,31497,'#36e839'),(12101,31518,'#e76ddd'),(12102,31525,'#ab3ee8'),(12103,31527,'#74ee5a'),(12104,31531,'#59d8d4'),(12105,31537,'#c709dc'),(12106,31540,'#df73e9'),(12107,31546,'#26e85f'),(12108,31550,'#e72de3'),(12109,31552,'#d363ec'),(12110,31557,'#34c5de'),(12111,31567,'#4bb4f4'),(12112,31569,'#2439d7'),(12113,31578,'#7d16df'),(12114,31581,'#FFFFFF'),(12115,31592,'#28d514'),(12116,31601,'#9b6fdf'),(12117,31606,'#FFFFFF'),(12118,31612,'#6745d3'),(12119,31619,'#ec52b2'),(12120,31620,'#d75933'),(12121,31623,'#2380d9'),(12122,31630,'#78d83b'),(12123,31650,'#0ddc4e'),(12124,31657,'#53d057'),(12125,31658,'#53d07b'),(12126,31659,'#e62b8a'),(12127,31661,'#9e36f6'),(12128,31662,'#fc7354'),(12129,31672,'#3dfe58'),(12130,31680,'#44c0dc'),(12131,31681,'#8ce965'),(12132,31690,'#68d8dc'),(12133,31691,'#1738e3'),(12134,31692,'#b6e123'),(12135,31693,'#1b06cc'),(12136,31696,'#01e736'),(12137,31698,'#dad70f'),(12138,31701,'#FFFFFF'),(12139,31708,'#e8366e'),(12140,31719,'#ddac44'),(12141,31729,'#a801f3'),(12142,31732,'#771dcd'),(12143,31733,'#d43439'),(12144,31735,'#fb77d4'),(12145,31736,'#79e866'),(12146,31753,'#93e35f'),(12147,31760,'#FFFFFF'),(12148,31764,'#FFFFFF'),(12149,31767,'#4df966'),(12150,31768,'#FFFFFF'),(12151,31772,'#8def75'),(12152,31773,'#66cc81'),(12153,31781,'#1ed54d'),(12154,31790,'#fbee4c'),(12155,31793,'#1801ea'),(12156,31800,'#f47d4c'),(12157,31807,'#45f684'),(12158,31808,'#8325ee'),(12159,31810,'#f828ab'),(12160,31815,'#36d9b9'),(12161,31816,'#d8c063'),(12162,31817,'#4feffc'),(12163,31825,'#f36df3'),(12164,31826,'#ea3f85'),(12165,31827,'#f41193'),(12166,31836,'#d769bd'),(12167,31843,'#71efe6'),(12168,31849,'#fe4bf3'),(12169,31859,'#d75ffa'),(12170,31862,'#db16ad'),(12171,31865,'#FFFFFF'),(12172,31876,'#7dde5a'),(12173,31879,'#e87c2b'),(12174,31880,'#485feb'),(12175,31881,'#2ee698'),(12176,31885,'#e39c20'),(12177,31886,'#fccd01'),(12178,31889,'#db6da0'),(12179,31894,'#509ee8'),(12180,31905,'#03e8c2'),(12181,31909,'#90f125'),(12182,31911,'#fab922'),(12183,31913,'#3ae6a3'),(12184,31919,'#75f9a7'),(12185,31921,'#86d148'),(12186,31922,'#e39239'),(12187,31923,'#88e64a'),(12188,31935,'#5239d4'),(12189,31936,'#e0b66a'),(12190,31938,'#692ccd'),(12191,31948,'#d35fa2'),(12192,31953,'#cfbd47'),(12193,31954,'#30e1f3'),(12194,31955,'#416fe7'),(12195,31957,'#d8de11'),(12196,31960,'#389df5'),(12197,31970,'#fef17c'),(12198,31996,'#4bc4e8'),(12199,32001,'#5ce495'),(12200,32002,'#FFFFFF'),(12201,32004,'#8655f8'),(12202,32009,'#d05c5f'),(12203,32013,'#9750d8'),(12204,32020,'#19b2d6'),(12205,32024,'#FFFFFF'),(12206,32029,'#68d990'),(12207,32034,'#FFFFFF'),(12208,32036,'#d8ad1c'),(12209,32040,'#106dfd'),(12210,32050,'#942fd4'),(12211,32068,'#9871e7'),(12212,32071,'#21c4e2'),(12213,32079,'#42fe22'),(12214,32094,'#2ee965'),(12215,32118,'#d641de'),(12216,32124,'#5cb5da'),(12217,32128,'#698dea'),(12218,32152,'#10d0d1'),(12219,32153,'#7ecc34'),(12220,32155,'#2ce067'),(12221,32157,'#68fb06'),(12222,32158,'#b433cf'),(12223,32159,'#d33786'),(12224,32162,'#a156e6'),(12225,32163,'#2cec5e'),(12226,32171,'#51cebd'),(12227,32174,'#d3db1c'),(12228,32175,'#ccae56'),(12229,32178,'#e98042'),(12230,32184,'#d1cb63'),(12231,32196,'#f4a820'),(12232,32197,'#d6f97c'),(12233,32198,'#d446c8'),(12234,32199,'#33d04d'),(12235,32215,'#27d885'),(12236,32216,'#14ec97'),(12237,32219,'#5ed4c3'),(12238,32229,'#de6b6a'),(12239,32230,'#8bd628'),(12240,32246,'#f77b36'),(12241,32248,'#97ee64'),(12242,32252,'#6ee195'),(12243,32255,'#5cd35c'),(12244,32259,'#8365f5'),(12245,32264,'#d93194'),(12246,32265,'#6dbbe1'),(12247,32266,'#1fe32d'),(12248,32270,'#379cdc'),(12249,32271,'#dc1911'),(12250,32272,'#8eeb14'),(12251,32273,'#71d313'),(12252,32281,'#1546e2'),(12253,32282,'#8b59e1'),(12254,32286,'#57d5e2'),(12255,32288,'#1c86f1'),(12256,32289,'#28d02a'),(12257,32291,'#f94b04'),(12258,32292,'#67f9c1'),(12259,32293,'#d97826'),(12260,32295,'#d85b18'),(12261,32298,'#5de739'),(12262,32300,'#3edd38'),(12263,32301,'#fc0d75'),(12264,32302,'#3d91e7'),(12265,32304,'#6ccd2e'),(12266,32305,'#7649db'),(12267,32317,'#95cc4e'),(12268,32324,'#e8a10b'),(12269,32325,'#ced917'),(12270,32339,'#13d948'),(12271,32341,'#fce824'),(12272,32343,'#1907ce'),(12273,32360,'#07a3fc'),(12274,32364,'#7af6d6'),(12275,32366,'#dcc162'),(12276,32370,'#5bbee5'),(12277,32397,'#a3d636'),(12278,32405,'#07aece'),(12279,32409,'#1cec4f'),(12280,32414,'#dec14d'),(12281,32422,'#fa56d6'),(12282,32424,'#e7d628'),(12283,32441,'#67ee31'),(12284,32450,'#4df55b'),(12285,32452,'#d15c81'),(12286,32463,'#277acc'),(12287,32467,'#4eea78'),(12288,32474,'#9df731'),(12289,32483,'#53dc4b'),(12290,32489,'#0fece3'),(12291,32491,'#40d37b'),(12292,32493,'#da2194'),(12293,32502,'#f2bd5c'),(12294,32510,'#9b4fd7'),(12295,32523,'#6732cd'),(12296,32527,'#FFFFFF'),(12297,32535,'#f57017'),(12298,32553,'#6cd122'),(12299,32566,'#d64c81'),(12300,32605,'#69d73b'),(12301,32626,'#7a6aec'),(12302,32648,'#ef6884'),(12303,32653,'#40dde1'),(12304,32658,'#25d618'),(12305,32661,'#7296e6'),(12306,32760,'#FFFFFF'),(12307,32784,'#94db67'),(12308,32786,'#9739d2'),(12309,32792,'#dbaf2a'),(12310,32802,'#55f577'),(12311,32813,'#bc51fd'),(12312,32818,'#e119b1'),(12313,32820,'#56fe75'),(12314,32830,'#26cc21'),(12315,32835,'#fac278'),(12316,32842,'#3c39f8'),(12317,32855,'#32e940'),(12318,32856,'#FFFFFF'),(12319,32881,'#6126cc'),(12320,32882,'#802df8'),(12321,32885,'#5df5a3'),(12322,32899,'#09f58c'),(12323,32900,'#5d74eb'),(12324,32902,'#FFFFFF'),(12325,32908,'#FFFFFF'),(12326,32909,'#f64941'),(12327,32937,'#e9b603'),(12328,32954,'#e0e059'),(12329,32963,'#fe9871'),(12330,32969,'#55ea37'),(12331,32983,'#FFFFFF'),(12332,32984,'#FFFFFF'),(12333,32985,'#FFFFFF'),(12334,32988,'#FFFFFF'),(12335,32989,'#3f44e2'),(12336,32990,'#FFFFFF'),(12337,32993,'#FFFFFF'),(12338,32996,'#b9d04c'),(12339,32997,'#FFFFFF'),(12340,33001,'#FFFFFF'),(12341,33003,'#ef771a'),(12342,33004,'#f80f05'),(12343,33006,'#FFFFFF'),(12344,33010,'#FFFFFF'),(12345,33012,'#f49f3a'),(12346,33013,'#FFFFFF'),(12347,33014,'#698ad7'),(12348,33018,'#FFFFFF'),(12349,33019,'#FFFFFF'),(12350,33022,'#443fe0'),(12351,33023,'#FFFFFF'),(12352,33024,'#FFFFFF'),(12353,33027,'#FFFFFF'),(12354,33032,'#FFFFFF'),(12355,33034,'#FFFFFF'),(12356,33039,'#FFFFFF'),(12357,33042,'#FFFFFF'),(12358,33045,'#c2e545'),(12359,33048,'#FFFFFF'),(12360,33049,'#FFFFFF'),(12361,33050,'#FFFFFF'),(12362,33057,'#FFFFFF'),(12363,33060,'#FFFFFF'),(12364,33061,'#497ee3'),(12365,33064,'#FFFFFF'),(12366,33065,'#d0429a'),(12367,33066,'#FFFFFF'),(12368,33069,'#FFFFFF'),(12369,33072,'#FFFFFF'),(12370,33074,'#FFFFFF'),(12371,33075,'#FFFFFF'),(12372,33076,'#25f3ee'),(12373,33077,'#FFFFFF'),(12374,33078,'#FFFFFF'),(12375,33079,'#FFFFFF'),(12376,33081,'#FFFFFF'),(12377,33083,'#FFFFFF'),(12378,33087,'#FFFFFF'),(12379,33090,'#FFFFFF'),(12380,33091,'#FFFFFF'),(12381,33092,'#FFFFFF'),(12382,33096,'#de5ad5'),(12383,33100,'#FFFFFF'),(12384,33103,'#FFFFFF'),(12385,33104,'#e707a8'),(12386,33105,'#FFFFFF'),(12387,33107,'#FFFFFF'),(12388,33111,'#FFFFFF'),(12389,33113,'#FFFFFF'),(12390,33114,'#FFFFFF'),(12391,33118,'#e0272a'),(12392,33119,'#FFFFFF'),(12393,33121,'#f7673e'),(12394,33122,'#FFFFFF'),(12395,33133,'#55e303'),(12396,33139,'#FFFFFF'),(12397,33142,'#FFFFFF'),(12398,33143,'#FFFFFF'),(12399,33148,'#e9e542'),(12400,33150,'#FFFFFF'),(12401,33151,'#FFFFFF'),(12402,33152,'#239fcc'),(12403,33154,'#FFFFFF'),(12404,33155,'#FFFFFF'),(12405,33161,'#FFFFFF'),(12406,33162,'#FFFFFF'),(12407,33163,'#FFFFFF'),(12408,33164,'#FFFFFF'),(12409,33167,'#FFFFFF'),(12410,33168,'#d751c4'),(12411,33171,'#FFFFFF'),(12412,33172,'#FFFFFF'),(12413,33173,'#FFFFFF'),(12414,33175,'#FFFFFF'),(12415,33177,'#FFFFFF'),(12416,33180,'#FFFFFF'),(12417,33188,'#FFFFFF'),(12418,33193,'#FFFFFF'),(12419,33194,'#FFFFFF'),(12420,33197,'#FFFFFF'),(12421,33199,'#FFFFFF'),(12422,33200,'#FFFFFF'),(12423,33201,'#FFFFFF'),(12424,33203,'#FFFFFF'),(12425,33204,'#FFFFFF'),(12426,33207,'#FFFFFF'),(12427,33208,'#f0c807'),(12428,33210,'#FFFFFF'),(12429,33219,'#eb7470'),(12430,33222,'#8320e0'),(12431,33228,'#ca01e7'),(12432,33229,'#4906f3'),(12433,33231,'#7ff204'),(12434,33234,'#FFFFFF'),(12435,33235,'#3bdf9d'),(12436,33242,'#62d6ab'),(12437,33247,'#21f03b'),(12438,33252,'#74d31a'),(12439,33258,'#d36bdb'),(12440,33260,'#7d68f0'),(12441,33263,'#0ace7b'),(12442,33267,'#27e11f'),(12443,33268,'#ef5674'),(12444,33275,'#6385f6'),(12445,33277,'#FFFFFF'),(12446,33278,'#a5d94a'),(12447,33281,'#FFFFFF'),(12448,33282,'#d97f3b'),(12449,33283,'#cd2c2f'),(12450,33286,'#2f9cf4'),(12451,33288,'#ea4ac4'),(12452,33289,'#f9e61c'),(12453,33293,'#d6794f'),(12454,33297,'#e058d9'),(12455,33298,'#44f573'),(12456,33300,'#24f23a'),(12457,33316,'#a951e0'),(12458,33320,'#e32bae'),(12459,33321,'#FFFFFF'),(12460,33326,'#dc4c0f'),(12461,33330,'#cccc53'),(12462,33331,'#9af928'),(12463,33332,'#0dce9a'),(12464,33335,'#FFFFFF'),(12465,33337,'#75c5fd'),(12466,33339,'#0be527'),(12467,33343,'#FFFFFF'),(12468,33344,'#9bf432'),(12469,33346,'#FFFFFF'),(12470,33349,'#5bcc1c'),(12471,33350,'#ec657d'),(12472,33351,'#de5548'),(12473,33355,'#e343f9'),(12474,33359,'#8ae362'),(12475,33360,'#cc4c4f'),(12476,33363,'#1ca8eb'),(12477,33368,'#50fbde'),(12478,33371,'#084fe0'),(12479,33373,'#ce2a9c'),(12480,33374,'#fc1134'),(12481,33377,'#3ded08'),(12482,33378,'#a7f13f'),(12483,33380,'#cf0cd7'),(12484,33384,'#f13bc2'),(12485,33385,'#625bf6'),(12486,33386,'#f33a92'),(12487,33394,'#FFFFFF'),(12488,33399,'#e4462f'),(12489,33400,'#10cef9'),(12490,33406,'#FFFFFF'),(12491,33421,'#FFFFFF'),(12492,33428,'#f81f83'),(12493,33445,'#29e070'),(12494,33446,'#a706d0'),(12495,33452,'#4beb43'),(12496,33457,'#cc6558'),(12497,33469,'#f99f1e'),(12498,33477,'#df5ab7'),(12499,33485,'#e1855a'),(12500,33501,'#e73ddf'),(12501,33504,'#f98865'),(12502,33513,'#07f435'),(12503,33519,'#4090d4'),(12504,33520,'#a127f5'),(12505,33526,'#1bf670'),(12506,33530,'#669cdf'),(12507,33532,'#e66014'),(12508,33553,'#FFFFFF'),(12509,33567,'#d9080d'),(12510,33591,'#9461f5'),(12511,33592,'#f423c4'),(12512,33598,'#ce5fda'),(12513,33607,'#66fc79'),(12514,33612,'#e56568'),(12515,33624,'#e87326'),(12516,33643,'#FFFFFF'),(12517,33644,'#ab5ed7'),(12518,33651,'#d4b259'),(12519,33684,'#21e891'),(12520,33692,'#5428d0'),(12521,33702,'#b3fa68'),(12522,33719,'#434ce5'),(12523,33723,'#f057fe'),(12524,33724,'#FFFFFF'),(12525,33733,'#e0d638'),(12526,33738,'#e6258a'),(12527,33750,'#43e698'),(12528,33751,'#58d20a'),(12529,33755,'#0949ec'),(12530,33764,'#cd2bce'),(12531,33766,'#d8f04a'),(12532,33784,'#06f49e'),(12533,33795,'#e37910'),(12534,33796,'#549dd7'),(12535,33800,'#12dcad'),(12536,33805,'#FFFFFF'),(12537,33809,'#bce75b'),(12538,33828,'#3eedb9'),(12539,33829,'#FFFFFF'),(12540,33833,'#8f59e9'),(12541,33842,'#5626d2'),(12542,33844,'#6640d0'),(12543,33857,'#6b0eee'),(12544,33861,'#79fc76'),(12545,33865,'#08e065'),(12546,33874,'#3ff389'),(12547,33883,'#231cf3'),(12548,33884,'#74f1bf'),(12549,33890,'#29ed00'),(12550,33895,'#6403ce'),(12551,33897,'#2813ce'),(12552,33899,'#e15e1e'),(12553,33911,'#780bf7'),(12554,33918,'#fe967c'),(12555,33924,'#342cf1'),(12556,33932,'#e26fb9'),(12557,33933,'#1fd2f8'),(12558,33935,'#b4f800'),(12559,33942,'#43bffe'),(12560,33946,'#458acd'),(12561,33954,'#dbda3d'),(12562,33962,'#db589d'),(12563,33964,'#b816d4'),(12564,33981,'#c9ce4d'),(12565,34007,'#7943cf'),(12566,34018,'#28fdb5'),(12567,34025,'#1cbccc'),(12568,34026,'#ef6383'),(12569,34027,'#4bd3f0'),(12570,34049,'#9844ea'),(12571,34050,'#8eec1f'),(12572,34058,'#eb400d'),(12573,34066,'#5d41eb'),(12574,34072,'#5af530'),(12575,34073,'#20d783'),(12576,34074,'#db3fd4'),(12577,34079,'#23d621'),(12578,34091,'#f2c35b'),(12579,34103,'#f65673'),(12580,34108,'#d6ac5d'),(12581,34110,'#68ebca'),(12582,34115,'#FFFFFF'),(12583,34136,'#5ea0ef'),(12584,34138,'#ea9b14'),(12585,34140,'#a9df2f'),(12586,34150,'#da14ae'),(12587,34167,'#fb77d6'),(12588,34168,'#8eef09'),(12589,34173,'#c12dd5'),(12590,34191,'#4be612'),(12591,34197,'#03e4e6'),(12592,34198,'#ef2d0a'),(12593,34214,'#3403f5'),(12594,34216,'#dc64a5'),(12595,34218,'#cf17d8'),(12596,34223,'#497af1'),(12597,34225,'#FFFFFF'),(12598,34232,'#cebe18'),(12599,34248,'#c1e548'),(12600,34250,'#FFFFFF'),(12601,34251,'#FFFFFF'),(12602,34253,'#e7f77a'),(12603,34256,'#5af11f'),(12604,34264,'#73dd4e'),(12605,34267,'#FFFFFF'),(12606,34270,'#edf468'),(12607,34275,'#8210fd'),(12608,34276,'#ee3352'),(12609,34286,'#FFFFFF'),(12610,34293,'#FFFFFF'),(12611,34299,'#41bddb'),(12612,34305,'#dcca26'),(12613,34308,'#ddd757'),(12614,34309,'#2378f9'),(12615,34321,'#8ae062'),(12616,34327,'#FFFFFF'),(12617,34330,'#ef5952'),(12618,34343,'#5bb0f4'),(12619,34356,'#5e6bd5'),(12620,34362,'#FFFFFF'),(12621,34363,'#FFFFFF'),(12622,34366,'#b1d562'),(12623,34382,'#FFFFFF'),(12624,34385,'#5bf907'),(12625,34391,'#FFFFFF'),(12626,34397,'#32da3c'),(12627,34402,'#92fc66'),(12628,34403,'#1535ce'),(12629,34421,'#a4f800'),(12630,34423,'#6dfcea'),(12631,34435,'#FFFFFF'),(12632,34439,'#d0e361'),(12633,34452,'#FFFFFF'),(12634,34457,'#FFFFFF'),(12635,34459,'#FFFFFF'),(12636,34460,'#FFFFFF'),(12637,34466,'#FFFFFF'),(12638,34475,'#d9f459'),(12639,34483,'#429aea'),(12640,34499,'#FFFFFF'),(12641,34502,'#19fba2'),(12642,34503,'#6cddca'),(12643,34504,'#e96979'),(12644,34506,'#FFFFFF'),(12645,34507,'#FFFFFF'),(12646,34508,'#99f737'),(12647,34511,'#FFFFFF'),(12648,34512,'#FFFFFF'),(12649,34515,'#61df5f'),(12650,34517,'#FFFFFF'),(12651,34518,'#90e35a'),(12652,34522,'#FFFFFF'),(12653,34523,'#FFFFFF'),(12654,34526,'#FFFFFF'),(12655,34527,'#FFFFFF'),(12656,34529,'#FFFFFF'),(12657,34530,'#d303a2'),(12658,34537,'#f2f95f'),(12659,34543,'#FFFFFF'),(12660,34546,'#dec50b'),(12661,34551,'#e3ba44'),(12662,34555,'#FFFFFF'),(12663,34559,'#FFFFFF'),(12664,34564,'#FFFFFF'),(12665,34565,'#FFFFFF'),(12666,34569,'#73f596'),(12667,34573,'#11fc9f'),(12668,34574,'#FFFFFF'),(12669,34577,'#FFFFFF'),(12670,34587,'#f3f426'),(12671,34603,'#6d12e4'),(12672,34604,'#e54e91'),(12673,34605,'#24d937'),(12674,34608,'#71f677'),(12675,34610,'#1cbde7'),(12676,34612,'#b256cd'),(12677,34627,'#fb1632'),(12678,34629,'#f5b500'),(12679,34636,'#09ce99'),(12680,34637,'#67fadd'),(12681,34647,'#d7d419'),(12682,34661,'#e56514'),(12683,34662,'#eb616c'),(12684,34663,'#cf1033'),(12685,34665,'#e56b6e'),(12686,34669,'#e60b64'),(12687,34681,'#61b1d2'),(12688,34696,'#eeb42d'),(12689,34702,'#FFFFFF'),(12690,34705,'#FFFFFF'),(12691,34707,'#FFFFFF'),(12692,34710,'#FFFFFF'),(12693,34724,'#FFFFFF'),(12694,34730,'#FFFFFF'),(12695,34732,'#FFFFFF'),(12696,34737,'#FFFFFF'),(12697,34738,'#d70be9'),(12698,34739,'#FFFFFF'),(12699,34740,'#FFFFFF'),(12700,34741,'#FFFFFF'),(12701,34742,'#FFFFFF'),(12702,34743,'#FFFFFF'),(12703,34745,'#FFFFFF'),(12704,34746,'#FFFFFF'),(12705,34751,'#FFFFFF'),(12706,34752,'#FFFFFF'),(12707,34756,'#e7186d'),(12708,34758,'#FFFFFF'),(12709,34760,'#FFFFFF'),(12710,34773,'#FFFFFF'),(12711,34782,'#e637d2'),(12712,34794,'#FFFFFF'),(12713,34798,'#FFFFFF'),(12714,34805,'#FFFFFF'),(12715,34807,'#10e9ab'),(12716,34813,'#98ec4d'),(12717,34820,'#FFFFFF'),(12718,34824,'#FFFFFF'),(12719,34835,'#FFFFFF'),(12720,34839,'#FFFFFF'),(12721,34843,'#FFFFFF'),(12722,34844,'#FFFFFF'),(12723,34851,'#FFFFFF'),(12724,34853,'#f75c6e'),(12725,34858,'#FFFFFF'),(12726,34865,'#FFFFFF'),(12727,34872,'#FFFFFF'),(12728,34873,'#FFFFFF'),(12729,34883,'#eddf73'),(12730,34890,'#FFFFFF'),(12731,34893,'#0fe2ec'),(12732,34897,'#FFFFFF'),(12733,34922,'#52d619'),(12734,34955,'#af36ee'),(12735,34956,'#e0b802'),(12736,34970,'#d4139d'),(12737,34998,'#f53ec0'),(12738,35028,'#1fed54'),(12739,35055,'#6c68f0'),(12740,35100,'#cc520d'),(12741,35111,'#a769d4'),(12742,35129,'#8245cf'),(12743,35133,'#96e31c'),(12744,35136,'#9355de'),(12745,35138,'#1e41fb'),(12746,35139,'#18e055'),(12747,35159,'#27b0f5'),(12748,35201,'#67f025'),(12749,35226,'#FFFFFF'),(12750,35233,'#fac719'),(12751,35238,'#dc7b29'),(12752,35259,'#FFFFFF'),(12753,35263,'#FFFFFF'),(12754,35273,'#1ad0e8'),(12755,35275,'#FFFFFF'),(12756,35277,'#FFFFFF'),(12757,35280,'#42f392'),(12758,35289,'#FFFFFF'),(12759,35314,'#15dff7'),(12760,35316,'#fe806e'),(12761,35334,'#f1fb62'),(12762,35344,'#17e965'),(12763,35362,'#c2e228'),(12764,35364,'#7bf964'),(12765,35365,'#ec3e7d'),(12766,35368,'#f19f53'),(12767,35370,'#1ab2ec'),(12768,35372,'#6ee90a'),(12769,35380,'#0fe69b'),(12770,35383,'#fd0248'),(12771,35396,'#e73a64'),(12772,35427,'#f3781d'),(12773,35434,'#04f2c7'),(12774,35460,'#69f4bd'),(12775,35488,'#FFFFFF'),(12776,35565,'#d42b58'),(12777,35600,'#1627f3'),(12778,35601,'#e45205'),(12779,35602,'#b94cd1'),(12780,35603,'#4eaed4'),(12781,35604,'#FFFFFF'),(12782,35606,'#FFFFFF'),(12783,35607,'#FFFFFF'),(12784,35609,'#FFFFFF'),(12785,35612,'#68f072'),(12786,35615,'#6bcc33'),(12787,35618,'#ed5470'),(12788,35620,'#65f83d'),(12789,35621,'#0a42d3'),(12790,35624,'#d1e464'),(12791,35625,'#da9b34'),(12792,35630,'#FFFFFF'),(12793,35636,'#82d048'),(12794,35638,'#FFFFFF'),(12795,35639,'#f2fc0a'),(12796,35640,'#FFFFFF'),(12797,35644,'#c4d143'),(12798,35645,'#176bd0'),(12799,35650,'#df6bf9'),(12800,35653,'#adce44'),(12801,35655,'#69f49b'),(12802,35662,'#57d99d'),(12803,35665,'#7358cc'),(12804,35667,'#76e109'),(12805,35668,'#42ec90'),(12806,35673,'#19f793'),(12807,35675,'#FFFFFF'),(12808,35677,'#a81dd3'),(12809,35681,'#471dd8'),(12810,35682,'#ef9e6a'),(12811,35683,'#ec8415'),(12812,35684,'#70e56a'),(12813,35688,'#4e04ce'),(12814,35690,'#FFFFFF'),(12815,35699,'#e270ba'),(12816,35700,'#5233cd'),(12817,35701,'#FFFFFF'),(12818,35702,'#FFFFFF'),(12819,35705,'#d8ec08'),(12820,35710,'#FFFFFF'),(12821,35711,'#29cc83'),(12822,35713,'#336cdd'),(12823,35717,'#1edc28'),(12824,35718,'#FFFFFF'),(12825,35719,'#FFFFFF'),(12826,35724,'#b0e855'),(12827,35727,'#FFFFFF'),(12828,35733,'#FFFFFF'),(12829,35735,'#1ee2ed'),(12830,35743,'#f3de39'),(12831,35753,'#6dfe66'),(12832,35755,'#27d7eb'),(12833,35756,'#1562fc'),(12834,35758,'#d8ec55'),(12835,35761,'#4167f0'),(12836,35763,'#2145e9'),(12837,35769,'#05cd03'),(12838,35774,'#f6b001'),(12839,35778,'#e2654c'),(12840,35779,'#3947dc'),(12841,35788,'#15e96c'),(12842,35789,'#cc5b39'),(12843,35795,'#FFFFFF'),(12844,35797,'#6180fb'),(12845,35800,'#FFFFFF'),(12846,35801,'#FFFFFF'),(12847,35802,'#FFFFFF'),(12848,35806,'#FFFFFF'),(12849,35810,'#27cbf6'),(12850,35816,'#FFFFFF'),(12851,35818,'#2744f0'),(12852,35827,'#58f5ab'),(12853,35829,'#22faba'),(12854,35831,'#d2ea58'),(12855,35844,'#b7de16'),(12856,35852,'#d62cc4'),(12857,35853,'#3288fb'),(12858,35859,'#11cec9'),(12859,35860,'#e858dc'),(12860,35862,'#FFFFFF'),(12861,35865,'#37bbd7'),(12862,35867,'#FFFFFF'),(12863,35870,'#FFFFFF'),(12864,35872,'#dc5d87'),(12865,35874,'#f72e1c'),(12866,35875,'#25cca5'),(12867,35878,'#FFFFFF'),(12868,35879,'#a9d064'),(12869,35882,'#FFFFFF'),(12870,35883,'#ea57bb'),(12871,35885,'#91d460'),(12872,35887,'#7767d2'),(12873,35895,'#FFFFFF'),(12874,35896,'#FFFFFF'),(12875,35899,'#db4432'),(12876,35911,'#7273e7'),(12877,35915,'#FFFFFF'),(12878,35916,'#fa66d4'),(12879,35917,'#d61264'),(12880,35920,'#2c55f4'),(12881,35922,'#FFFFFF'),(12882,35923,'#371acf'),(12883,35930,'#FFFFFF'),(12884,35932,'#eef778'),(12885,35933,'#4878f7'),(12886,35934,'#6bd44e'),(12887,35936,'#FFFFFF'),(12888,35937,'#e9df52'),(12889,35939,'#e57d46'),(12890,35943,'#d58723'),(12891,35949,'#77d4f3'),(12892,35950,'#b7df6a'),(12893,35952,'#b6db2c'),(12894,35955,'#facf3e'),(12895,35956,'#6fcf11'),(12896,35963,'#ec2fcb'),(12897,35965,'#5ed183'),(12898,35968,'#6284cf'),(12899,35969,'#de6ce5'),(12900,35973,'#e052a2'),(12901,35974,'#d212c1'),(12902,35975,'#6d7ef7'),(12903,35979,'#fd3651'),(12904,35982,'#fba047'),(12905,35983,'#813ce6'),(12906,35984,'#FFFFFF'),(12907,35986,'#de8f27'),(12908,35988,'#e5bf68'),(12909,35990,'#1de1db'),(12910,35992,'#b2e736'),(12911,35995,'#0acfb7'),(12912,35996,'#78da64'),(12913,36002,'#2137f1'),(12914,36006,'#e7c75f'),(12915,36011,'#44dc7e'),(12916,36014,'#e6b84e'),(12917,36016,'#7063cf'),(12918,36018,'#d42a3b'),(12919,36019,'#3f0fd4'),(12920,36021,'#70d414'),(12921,36022,'#16f3c7'),(12922,36023,'#d5ec3f'),(12923,36024,'#da7445'),(12924,36025,'#ce21d2'),(12925,36026,'#2e1dfc'),(12926,36027,'#054ecd'),(12927,36030,'#de497d'),(12928,36031,'#d330fe'),(12929,36032,'#6e0afa'),(12930,36034,'#4acc9e'),(12931,36036,'#d97628'),(12932,36037,'#e98009'),(12933,36038,'#df3512'),(12934,36041,'#dcd90a'),(12935,36042,'#42fdcb'),(12936,36044,'#d45090'),(12937,36046,'#5ddcb3'),(12938,36048,'#afd51e'),(12939,36050,'#9d69ed'),(12940,36052,'#d83ea0'),(12941,36054,'#a439d7'),(12942,36056,'#e2f13c'),(12943,36057,'#b3df0e'),(12944,36058,'#4ef7ba'),(12945,36060,'#3db0e5'),(12946,36061,'#5f1bdf'),(12947,36065,'#e015d7'),(12948,36067,'#50e575'),(12949,36068,'#d8b810'),(12950,36083,'#7488f1'),(12951,36085,'#ddc519'),(12952,36087,'#4209f5'),(12953,36088,'#56aad0'),(12954,36089,'#b7ef04'),(12955,36091,'#f7f25b'),(12956,36092,'#50ea8a'),(12957,36094,'#fd594d'),(12958,36097,'#45a5ea'),(12959,36098,'#1551ef'),(12960,36099,'#74ee5d'),(12961,36100,'#ceaf5e'),(12962,36104,'#56c5d4'),(12963,36111,'#ef774d'),(12964,36113,'#e00cc6'),(12965,36118,'#04f484'),(12966,36120,'#8cf477'),(12967,36121,'#d8ca18'),(12968,36122,'#bc4fde'),(12969,36123,'#78e80f'),(12970,36131,'#ae63ea'),(12971,36133,'#e446af'),(12972,36138,'#1232fa'),(12973,36141,'#f24eb1'),(12974,36142,'#e5b159'),(12975,36145,'#1b7ed6'),(12976,36154,'#cd1123'),(12977,36155,'#d6467c'),(12978,36159,'#33e4ba'),(12979,36161,'#21c3e1'),(12980,36165,'#f75a24'),(12981,36168,'#ae4df3'),(12982,36169,'#09d98b'),(12983,36172,'#f9a10d'),(12984,36173,'#FFFFFF'),(12985,36175,'#f49c03'),(12986,36181,'#8af920'),(12987,36185,'#40e1b6'),(12988,36186,'#FFFFFF'),(12989,36188,'#31d08d'),(12990,36191,'#9fe438'),(12991,36192,'#3063da'),(12992,36208,'#94e638'),(12993,36216,'#d65442'),(12994,36218,'#c0fd6c'),(12995,36223,'#da9d29'),(12996,36233,'#08cc96'),(12997,36235,'#FFFFFF'),(12998,36241,'#2e94dd'),(12999,36243,'#FFFFFF'),(13000,36253,'#d84b70'),(13001,36257,'#cdca43'),(13002,36261,'#d6885d'),(13003,36262,'#b5f13d'),(13004,36264,'#e64eec'),(13005,36267,'#7ae14d'),(13006,36271,'#30c6dd'),(13007,36277,'#f2ab42'),(13008,36279,'#f2df73'),(13009,36280,'#26e416'),(13010,36282,'#db0039'),(13011,36283,'#da6e68'),(13012,36285,'#d0cc3f'),(13013,36289,'#e65603'),(13014,36291,'#FFFFFF'),(13015,36296,'#f90378'),(13016,36304,'#8be24d'),(13017,36307,'#9618e7'),(13018,36315,'#eb026f'),(13019,36318,'#0bd81d'),(13020,36320,'#2196df'),(13021,36322,'#7be302'),(13022,36333,'#f99e58'),(13023,36336,'#a2fe56'),(13024,36338,'#1bd947'),(13025,36344,'#5195fd'),(13026,36353,'#FFFFFF'),(13027,36360,'#0af30d'),(13028,36361,'#23f978'),(13029,36367,'#89f745'),(13030,36374,'#75e413'),(13031,36376,'#5cd341'),(13032,36384,'#b5fc69'),(13033,36385,'#d3df31'),(13034,36388,'#06cebf'),(13035,36403,'#2de133'),(13036,36405,'#FFFFFF'),(13037,36406,'#FFFFFF'),(13038,36412,'#FFFFFF'),(13039,36414,'#3627f9'),(13040,36419,'#db5d85'),(13041,36420,'#ed1913'),(13042,36426,'#FFFFFF'),(13043,36428,'#32f752'),(13044,36429,'#985ffc'),(13045,36430,'#2fd414'),(13046,36432,'#e9128f'),(13047,36433,'#f2bf2c'),(13048,36434,'#e16e7f'),(13049,36435,'#FFFFFF'),(13050,36438,'#b0f844'),(13051,36442,'#4fcf83'),(13052,36446,'#de1b06'),(13053,36450,'#FFFFFF'),(13054,36453,'#2cee79'),(13055,36454,'#47c2e3'),(13056,36455,'#f26ebf'),(13057,36459,'#e41cf9'),(13058,36464,'#FFFFFF'),(13059,36471,'#f921ef'),(13060,36474,'#cf36e7'),(13061,36476,'#FFFFFF'),(13062,36482,'#502dd4'),(13063,36483,'#dc7214'),(13064,36486,'#4d96e5'),(13065,36488,'#6715eb'),(13066,36489,'#ebf571'),(13067,36491,'#71e7d1'),(13068,36497,'#FFFFFF'),(13069,36503,'#FFFFFF'),(13070,36507,'#745dcf'),(13071,36514,'#cef859'),(13072,36527,'#78f59f'),(13073,36531,'#0531f1'),(13074,36532,'#f56d42'),(13075,36533,'#FFFFFF'),(13076,36541,'#5e6ecd'),(13077,36543,'#01d9ca'),(13078,36546,'#e03287'),(13079,36550,'#8829e4'),(13080,36554,'#2af546'),(13081,36557,'#fa8658'),(13082,36558,'#f52bbf'),(13083,36559,'#FFFFFF'),(13084,36561,'#FFFFFF'),(13085,36572,'#da090c'),(13086,36573,'#d39e2e'),(13087,36576,'#fd15fb'),(13088,36584,'#803cdc'),(13089,36597,'#e56f65'),(13090,36607,'#dccc6b'),(13091,36608,'#5d8ef3'),(13092,36610,'#FFFFFF'),(13093,36615,'#FFFFFF'),(13094,36621,'#38cfef'),(13095,36622,'#FFFFFF'),(13096,36627,'#36cffc'),(13097,36629,'#f34020'),(13098,36630,'#5ff15b'),(13099,36631,'#a9f831'),(13100,36632,'#a0e550'),(13101,36637,'#FFFFFF'),(13102,36639,'#e3de38'),(13103,36641,'#a132f7'),(13104,36642,'#a8d424'),(13105,36651,'#44e49c'),(13106,36653,'#FFFFFF'),(13107,36661,'#FFFFFF'),(13108,36663,'#535bf8'),(13109,36667,'#f66126'),(13110,36670,'#dd5f15'),(13111,36672,'#6eb1e0'),(13112,36679,'#9e46d7'),(13113,36684,'#d55272'),(13114,36687,'#c347e3'),(13115,36690,'#d3649d'),(13116,36692,'#FFFFFF'),(13117,36694,'#50b4e9'),(13118,36695,'#FFFFFF'),(13119,36698,'#FFFFFF'),(13120,36702,'#e41f75'),(13121,36711,'#7551e8'),(13122,36721,'#bc27e6'),(13123,36730,'#5ccd63'),(13124,36733,'#22e830'),(13125,36736,'#aa19cf'),(13126,36738,'#aa33f2'),(13127,36742,'#287be8'),(13128,36745,'#4518d1'),(13129,36748,'#72fcf3'),(13130,36754,'#00d3c1'),(13131,36756,'#1457e7'),(13132,36767,'#4699da'),(13133,36775,'#adcf21'),(13134,36777,'#16d326'),(13135,36778,'#0fd6e9'),(13136,36779,'#494fdf'),(13137,36782,'#6ffae2'),(13138,36786,'#6a56e6'),(13139,36792,'#FFFFFF'),(13140,36798,'#3078dc'),(13141,36799,'#FFFFFF'),(13142,36801,'#05df31'),(13143,36810,'#1a6eed'),(13144,36815,'#5bd248'),(13145,36816,'#0672d0'),(13146,36817,'#aecc62'),(13147,36818,'#64de36'),(13148,36819,'#9f22e4'),(13149,36822,'#8530d9'),(13150,36832,'#e6765e'),(13151,36857,'#8bf228'),(13152,36902,'#91df6a'),(13153,36909,'#7b56d1'),(13154,36910,'#75fe5c'),(13155,36929,'#38b1cd'),(13156,36943,'#d82f6e'),(13157,36945,'#FFFFFF'),(13158,36947,'#623cee'),(13159,36950,'#ea1cb1'),(13160,36955,'#FFFFFF'),(13161,36957,'#a855cd'),(13162,36962,'#FFFFFF'),(13163,36964,'#FFFFFF'),(13164,36980,'#7a55d3'),(13165,36983,'#35d065'),(13166,36984,'#FFFFFF'),(13167,36985,'#FFFFFF'),(13168,36989,'#FFFFFF'),(13169,36996,'#55e57b'),(13170,37001,'#FFFFFF'),(13171,37005,'#dd4b5b'),(13172,37006,'#0d25d1'),(13173,37008,'#76b1ef'),(13174,37011,'#FFFFFF'),(13175,37020,'#FFFFFF'),(13176,37027,'#FFFFFF'),(13177,37031,'#FFFFFF'),(13178,37036,'#2455f5'),(13179,37051,'#FFFFFF'),(13180,37057,'#5218dd'),(13181,37058,'#b82dea'),(13182,37059,'#669def'),(13183,37060,'#00d1d4'),(13184,37063,'#61de27'),(13185,37069,'#89d40e'),(13186,37081,'#FFFFFF'),(13187,37082,'#FFFFFF'),(13188,37088,'#9ff230'),(13189,37092,'#28d974'),(13190,37094,'#FFFFFF'),(13191,37095,'#FFFFFF'),(13192,37103,'#FFFFFF'),(13193,37119,'#FFFFFF'),(13194,37123,'#d6e44d'),(13195,37124,'#64f539'),(13196,37125,'#a140ea'),(13197,37140,'#dde65e'),(13198,37141,'#FFFFFF'),(13199,37142,'#FFFFFF'),(13200,37146,'#d2e506'),(13201,37147,'#82fc42'),(13202,37149,'#65ce88'),(13203,37150,'#6c24d3'),(13204,37151,'#FFFFFF'),(13205,37152,'#FFFFFF'),(13206,37161,'#FFFFFF'),(13207,37163,'#f428dd'),(13208,37164,'#FFFFFF'),(13209,37168,'#FFFFFF'),(13210,37173,'#d160a8'),(13211,37175,'#cbf501'),(13212,37185,'#66d793'),(13213,37187,'#db630c'),(13214,37196,'#FFFFFF'),(13215,37200,'#3e3cef'),(13216,37205,'#d1c853'),(13217,37207,'#53fc07'),(13218,37209,'#fbd14f'),(13219,37211,'#FFFFFF'),(13220,37214,'#FFFFFF'),(13221,37215,'#cf2f68'),(13222,37220,'#FFFFFF'),(13223,37223,'#FFFFFF'),(13224,37224,'#f03ad5'),(13225,37225,'#3ae0cf'),(13226,37227,'#aa41d8'),(13227,37230,'#3eecde'),(13228,37232,'#FFFFFF'),(13229,37235,'#f1c546'),(13230,37236,'#5afd47'),(13231,37237,'#FFFFFF'),(13232,37238,'#FFFFFF'),(13233,37242,'#660be3'),(13234,37243,'#FFFFFF'),(13235,37244,'#3cbae2'),(13236,37249,'#4b8ee9'),(13237,37252,'#4ed6b0'),(13238,37256,'#ec73fb'),(13239,37263,'#b628e7'),(13240,37265,'#FFFFFF'),(13241,37271,'#03e6fd'),(13242,37276,'#4c43e3'),(13243,37279,'#ddfa65'),(13244,37283,'#4ce92a'),(13245,37285,'#e26cb1'),(13246,37286,'#52ee67'),(13247,37290,'#50e3b1'),(13248,37291,'#4df86a'),(13249,37292,'#eedf6c'),(13250,37294,'#5ad797'),(13251,37298,'#b00ffd'),(13252,37302,'#08cced'),(13253,37306,'#FFFFFF'),(13254,37309,'#424ae7'),(13255,37310,'#e2456e'),(13256,37314,'#9d35d0'),(13257,37317,'#64d493'),(13258,37318,'#FFFFFF'),(13259,37320,'#FFFFFF'),(13260,37323,'#3190d1'),(13261,37330,'#FFFFFF'),(13262,37333,'#cd9b2f'),(13263,37337,'#FFFFFF'),(13264,37340,'#FFFFFF'),(13265,37342,'#9864ee'),(13266,37344,'#FFFFFF'),(13267,37345,'#27e907'),(13268,37347,'#fd5815'),(13269,37349,'#FFFFFF'),(13270,37354,'#FFFFFF'),(13271,37356,'#42bbf1'),(13272,37368,'#FFFFFF'),(13273,37372,'#d9c21e'),(13274,37376,'#06f114'),(13275,37378,'#f566b5'),(13276,37383,'#FFFFFF'),(13277,37385,'#FFFFFF'),(13278,37389,'#ab0be1'),(13279,37393,'#c7f82c'),(13280,37395,'#4980de'),(13281,37397,'#FFFFFF'),(13282,37401,'#5dee85'),(13283,37402,'#73b1eb'),(13284,37403,'#FFFFFF'),(13285,37409,'#02d920'),(13286,37410,'#ce5967'),(13287,37414,'#b257fb'),(13288,37423,'#d748ac'),(13289,37427,'#db63c9'),(13290,37431,'#FFFFFF'),(13291,37432,'#35feac'),(13292,37453,'#0261de'),(13293,37466,'#df5e97'),(13294,37467,'#be1fe1'),(13295,37473,'#6c30ee'),(13296,37476,'#6ef7e9'),(13297,37479,'#b7fe3d'),(13298,37481,'#785dfa'),(13299,37485,'#bcfc33'),(13300,37493,'#d8632a'),(13301,37502,'#FFFFFF'),(13302,37508,'#7ad008'),(13303,37516,'#33b5d1'),(13304,37528,'#e50ae5'),(13305,37537,'#e116b4'),(13306,37540,'#13a4f4'),(13307,37543,'#d06787'),(13308,37551,'#6b53d4'),(13309,37552,'#8d61fe'),(13310,37569,'#FFFFFF'),(13311,37578,'#d0596c'),(13312,37581,'#e98212'),(13313,37586,'#0fd747'),(13314,37590,'#36c3e0'),(13315,37593,'#13fea9'),(13316,37598,'#a2e10a'),(13317,37603,'#d7c760'),(13318,37607,'#d8f10e'),(13319,37624,'#b438e9'),(13320,37633,'#32caf6'),(13321,37635,'#dd1f65'),(13322,37639,'#26e389'),(13323,37665,'#fe54a7'),(13324,37675,'#678cdd'),(13325,37685,'#d86002'),(13326,37689,'#0c8cce'),(13327,37694,'#FFFFFF'),(13328,37697,'#eadc3c'),(13329,37701,'#31ee62'),(13330,37703,'#c45bcf'),(13331,37704,'#64d570'),(13332,37707,'#7c2efd'),(13333,37709,'#FFFFFF'),(13334,37710,'#FFFFFF'),(13335,37711,'#FFFFFF'),(13336,37712,'#FFFFFF'),(13337,37715,'#FFFFFF'),(13338,37716,'#c768d9'),(13339,37718,'#f024f3'),(13340,37720,'#d9b15c'),(13341,37723,'#92dd0f'),(13342,37725,'#528ada'),(13343,37726,'#3a40e6'),(13344,37727,'#FFFFFF'),(13345,37730,'#b5fe4f'),(13346,37731,'#24c7e4'),(13347,37732,'#dd3412'),(13348,37734,'#066de3'),(13349,37745,'#FFFFFF'),(13350,37748,'#27e982'),(13351,37754,'#8aec3f'),(13352,37757,'#f2e01e'),(13353,37763,'#FFFFFF'),(13354,37767,'#aa34fc'),(13355,37768,'#d39914'),(13356,37769,'#1ad59a'),(13357,37770,'#FFFFFF'),(13358,37775,'#FFFFFF'),(13359,37776,'#fa4542'),(13360,37778,'#ecdd52'),(13361,37779,'#f961f1'),(13362,37781,'#fb3d20'),(13363,37782,'#414ef1'),(13364,37784,'#FFFFFF'),(13365,37789,'#37e92a'),(13366,37790,'#f6b76c'),(13367,37802,'#FFFFFF'),(13368,37809,'#FFFFFF'),(13369,37813,'#FFFFFF'),(13370,37815,'#FFFFFF'),(13371,37816,'#FFFFFF'),(13372,37818,'#FFFFFF'),(13373,37821,'#00eca6'),(13374,37824,'#3da1fd'),(13375,37827,'#FFFFFF'),(13376,37838,'#FFFFFF'),(13377,37839,'#b958e7'),(13378,37841,'#7808d9'),(13379,37844,'#1ad730'),(13380,37845,'#FFFFFF'),(13381,37850,'#87fa6a'),(13382,37851,'#FFFFFF'),(13383,37860,'#ece951'),(13384,37861,'#FFFFFF'),(13385,37862,'#3cf8da'),(13386,37873,'#FFFFFF'),(13387,37878,'#06d5ee'),(13388,37879,'#cf8c47'),(13389,37880,'#FFFFFF'),(13390,37881,'#FFFFFF'),(13391,37882,'#FFFFFF'),(13392,37884,'#e42edd'),(13393,37886,'#FFFFFF'),(13394,37887,'#FFFFFF'),(13395,37888,'#FFFFFF'),(13396,37894,'#FFFFFF'),(13397,37909,'#3df0a5'),(13398,37910,'#48cc4a'),(13399,37912,'#06ddd1'),(13400,37926,'#41e6dc'),(13401,37928,'#a06bfc'),(13402,37930,'#126aeb'),(13403,37934,'#e28c0d'),(13404,37936,'#FFFFFF'),(13405,37939,'#FFFFFF'),(13406,37940,'#ed42a3'),(13407,37948,'#FFFFFF'),(13408,37954,'#79ec47'),(13409,37958,'#FFFFFF'),(13410,37959,'#FFFFFF'),(13411,37960,'#e9f73c'),(13412,37961,'#4c21de'),(13413,37964,'#FFFFFF'),(13414,37965,'#61d5d1'),(13415,37966,'#e0124f'),(13416,37968,'#cc33e3'),(13417,37969,'#fe8644'),(13418,37971,'#8967eb'),(13419,37972,'#3c78fa'),(13420,37973,'#FFFFFF'),(13421,37974,'#FFFFFF'),(13422,37975,'#FFFFFF'),(13423,37977,'#afde4b'),(13424,37979,'#FFFFFF'),(13425,37980,'#17bae1'),(13426,37991,'#c33cce'),(13427,37993,'#e61a6d'),(13428,38002,'#f65f76'),(13429,38003,'#087df4'),(13430,38004,'#FFFFFF'),(13431,38006,'#FFFFFF'),(13432,38007,'#FFFFFF'),(13433,38013,'#FFFFFF'),(13434,38014,'#ed04aa'),(13435,38015,'#400df2'),(13436,38016,'#FFFFFF'),(13437,38020,'#FFFFFF'),(13438,38022,'#FFFFFF'),(13439,38023,'#FFFFFF'),(13440,38025,'#d4174f'),(13441,38028,'#FFFFFF'),(13442,38030,'#6e8cec'),(13443,38035,'#6df99a'),(13444,38040,'#6289cc'),(13445,38044,'#ea46a4'),(13446,38051,'#02ebee'),(13447,38054,'#c92ff4'),(13448,38055,'#1d9bf5'),(13449,38061,'#0a29e9'),(13450,38063,'#FFFFFF'),(13451,38064,'#FFFFFF'),(13452,38067,'#e973b1'),(13453,38068,'#cbda39'),(13454,38071,'#d96bec'),(13455,38074,'#FFFFFF'),(13456,38076,'#FFFFFF'),(13457,38078,'#FFFFFF'),(13458,38085,'#e28a16'),(13459,38086,'#124ad0'),(13460,38087,'#FFFFFF'),(13461,38098,'#b4d75c'),(13462,38105,'#230fdd'),(13463,38106,'#6fd7e2'),(13464,38107,'#2b42dc'),(13465,38108,'#58fa4d'),(13466,38115,'#ee3238'),(13467,38121,'#37dfe0'),(13468,38127,'#1fa5e1'),(13469,38133,'#07ea47'),(13470,38137,'#3ea5ed'),(13471,38145,'#db6347'),(13472,38158,'#16e1cc'),(13473,38162,'#5cce0d'),(13474,38197,'#75f8d8'),(13475,38199,'#FFFFFF'),(13476,38202,'#FFFFFF'),(13477,38203,'#FFFFFF'),(13478,38210,'#FFFFFF'),(13479,38211,'#c4dd4a'),(13480,38216,'#FFFFFF'),(13481,38227,'#c566ea'),(13482,38232,'#FFFFFF'),(13483,38245,'#FFFFFF'),(13484,38247,'#e7d212'),(13485,38267,'#da5384'),(13486,38271,'#FFFFFF'),(13487,38275,'#f05b56'),(13488,38287,'#d49f19'),(13489,38294,'#FFFFFF'),(13490,38296,'#3cebe9'),(13491,38302,'#FFFFFF'),(13492,38304,'#42efad'),(13493,38311,'#FFFFFF'),(13494,38317,'#dd343b'),(13495,38318,'#dafe64'),(13496,38322,'#30f8f9'),(13497,38332,'#e7d80f'),(13498,38338,'#FFFFFF'),(13499,38345,'#FFFFFF'),(13500,38356,'#8cf05d'),(13501,38357,'#ae1bdd'),(13502,38362,'#FFFFFF'),(13503,38364,'#43d52d'),(13504,38377,'#FFFFFF'),(13505,38381,'#e27082'),(13506,38401,'#FFFFFF'),(13507,38406,'#FFFFFF'),(13508,38411,'#754ecc'),(13509,38423,'#FFFFFF'),(13510,38424,'#FFFFFF'),(13511,38426,'#a911fd'),(13512,38427,'#FFFFFF'),(13513,38429,'#13ee9e'),(13514,38445,'#FFFFFF'),(13515,38446,'#FFFFFF'),(13516,38447,'#f7fa79'),(13517,38451,'#eb764d'),(13518,38455,'#69e2a5'),(13519,38460,'#17f3e4'),(13520,38467,'#f25af2'),(13521,38477,'#60d68a'),(13522,38481,'#FFFFFF'),(13523,38482,'#FFFFFF'),(13524,38485,'#d1d138'),(13525,38486,'#FFFFFF'),(13526,38487,'#FFFFFF'),(13527,38495,'#FFFFFF'),(13528,38502,'#FFFFFF'),(13529,38503,'#FFFFFF'),(13530,38510,'#FFFFFF'),(13531,38516,'#FFFFFF'),(13532,38517,'#254df0'),(13533,38521,'#b8f766'),(13534,38534,'#FFFFFF'),(13535,38540,'#FFFFFF'),(13536,38541,'#FFFFFF'),(13537,38551,'#207ece'),(13538,38556,'#4edf28'),(13539,38557,'#e08555'),(13540,38565,'#FFFFFF'),(13541,38567,'#FFFFFF'),(13542,38573,'#fa3143'),(13543,38579,'#FFFFFF'),(13544,38582,'#25d2a2'),(13545,38584,'#9f48fc'),(13546,38601,'#FFFFFF'),(13547,38608,'#FFFFFF'),(13548,38609,'#FFFFFF'),(13549,38610,'#ce4677'),(13550,38611,'#FFFFFF'),(13551,38612,'#6d66dd'),(13552,38613,'#d22440'),(13553,38614,'#FFFFFF'),(13554,38639,'#e36586'),(13555,38647,'#d71b0a'),(13556,38650,'#FFFFFF'),(13557,38651,'#FFFFFF'),(13558,38668,'#FFFFFF'),(13559,38673,'#FFFFFF'),(13560,38678,'#cce643'),(13561,38679,'#f337c9'),(13562,38682,'#3ffe4c'),(13563,38684,'#8dcf20'),(13564,38687,'#FFFFFF'),(13565,38689,'#FFFFFF'),(13566,38698,'#FFFFFF'),(13567,38704,'#8dd74d'),(13568,38705,'#FFFFFF'),(13569,38711,'#f846c2'),(13570,38714,'#FFFFFF'),(13571,38715,'#FFFFFF'),(13572,38719,'#cff967'),(13573,38721,'#FFFFFF'),(13574,38723,'#FFFFFF'),(13575,38734,'#FFFFFF'),(13576,38757,'#f03f90'),(13577,38764,'#61e1db'),(13578,38767,'#d6545c'),(13579,38771,'#e054f4'),(13580,38774,'#FFFFFF'),(13581,38783,'#f476ea'),(13582,38789,'#FFFFFF'),(13583,38798,'#FFFFFF'),(13584,38803,'#55b8da'),(13585,38807,'#2249f7'),(13586,38808,'#e35982'),(13587,38813,'#FFFFFF'),(13588,38814,'#FFFFFF'),(13589,38816,'#FFFFFF'),(13590,38817,'#04bee1'),(13591,38818,'#FFFFFF'),(13592,38819,'#FFFFFF'),(13593,38825,'#FFFFFF'),(13594,38826,'#FFFFFF'),(13595,38833,'#FFFFFF'),(13596,38843,'#FFFFFF'),(13597,38860,'#FFFFFF'),(13598,38865,'#d15d3c'),(13599,38882,'#df00bc'),(13600,38886,'#b6f02c'),(13601,38887,'#4adbd7'),(13602,38889,'#FFFFFF'),(13603,38890,'#FFFFFF'),(13604,38895,'#FFFFFF'),(13605,38896,'#FFFFFF'),(13606,38899,'#4ab9eb'),(13607,38904,'#FFFFFF'),(13608,38905,'#df172d'),(13609,38912,'#FFFFFF'),(13610,38925,'#5ae838'),(13611,38926,'#ddf15e'),(13612,38928,'#FFFFFF'),(13613,38931,'#FFFFFF'),(13614,38932,'#FFFFFF'),(13615,38936,'#FFFFFF'),(13616,38937,'#d28704'),(13617,38944,'#3be2a6'),(13618,38950,'#FFFFFF'),(13619,38955,'#FFFFFF'),(13620,38956,'#db9329'),(13621,38963,'#FFFFFF'),(13622,38979,'#0ccfbe'),(13623,38985,'#e1bb29'),(13624,38986,'#FFFFFF'),(13625,38990,'#FFFFFF'),(13626,38996,'#FFFFFF'),(13627,38997,'#FFFFFF'),(13628,39020,'#FFFFFF'),(13629,39021,'#FFFFFF'),(13630,39022,'#71a0e4'),(13631,39025,'#a029d7'),(13632,39026,'#9c10ef'),(13633,39028,'#e54769'),(13634,39032,'#dc9616'),(13635,39034,'#d81a8d'),(13636,39035,'#FFFFFF'),(13637,39036,'#FFFFFF'),(13638,39037,'#FFFFFF'),(13639,39039,'#72d45f'),(13640,39048,'#FFFFFF'),(13641,39053,'#9cd84d'),(13642,39062,'#e24d7b'),(13643,39066,'#FFFFFF'),(13644,39067,'#c542f2'),(13645,39071,'#FFFFFF'),(13646,39074,'#73e7b1'),(13647,39085,'#FFFFFF'),(13648,39088,'#e78466'),(13649,39096,'#dd9a3c'),(13650,39100,'#d6475a'),(13651,39103,'#50db6b'),(13652,39118,'#f2755a'),(13653,39119,'#57d0f0'),(13654,39120,'#e840f8'),(13655,39131,'#FFFFFF'),(13656,39133,'#FFFFFF'),(13657,39150,'#FFFFFF'),(13658,39153,'#08e52b'),(13659,39155,'#FFFFFF'),(13660,39163,'#FFFFFF'),(13661,39164,'#FFFFFF'),(13662,39165,'#ce8b55'),(13663,39187,'#FFFFFF'),(13664,39191,'#fe9916'),(13665,39199,'#d9725d'),(13666,39203,'#44d510'),(13667,39204,'#FFFFFF'),(13668,39212,'#3ee02a'),(13669,39215,'#FFFFFF'),(13670,39222,'#FFFFFF'),(13671,39223,'#FFFFFF'),(13672,39232,'#d413e3'),(13673,39235,'#cf0c9d'),(13674,39242,'#c4f968'),(13675,39245,'#f2b151'),(13676,39246,'#926dfe'),(13677,39248,'#6d54f0'),(13678,39252,'#54c9e5'),(13679,39255,'#FFFFFF'),(13680,39276,'#FFFFFF'),(13681,39280,'#FFFFFF'),(13682,39284,'#FFFFFF'),(13683,39287,'#96ed4f'),(13684,39288,'#FFFFFF'),(13685,39291,'#FFFFFF'),(13686,39294,'#FFFFFF'),(13687,39295,'#FFFFFF'),(13688,39299,'#31e17e'),(13689,39305,'#FFFFFF'),(13690,39312,'#a471f4'),(13691,39319,'#FFFFFF'),(13692,39322,'#b904e1'),(13693,39323,'#FFFFFF'),(13694,39326,'#FFFFFF'),(13695,39331,'#62ecbb'),(13696,39335,'#813dd5'),(13697,39339,'#5342e6'),(13698,39340,'#FFFFFF'),(13699,39344,'#ecef75'),(13700,39345,'#FFFFFF'),(13701,39347,'#FFFFFF'),(13702,39349,'#FFFFFF'),(13703,39363,'#c9cf48'),(13704,39367,'#FFFFFF'),(13705,39372,'#45deae'),(13706,39375,'#695ad4'),(13707,39381,'#d0139d'),(13708,39390,'#de65ab'),(13709,39395,'#26d51b'),(13710,39406,'#dc515f'),(13711,39409,'#fed63c'),(13712,39410,'#affc38'),(13713,39414,'#75cf5b'),(13714,39417,'#16dbd6'),(13715,39418,'#ec6c65'),(13716,39422,'#7e61d9'),(13717,39423,'#FFFFFF'),(13718,39424,'#00d5af'),(13719,39429,'#33b1ef'),(13720,39432,'#4f47fd'),(13721,39433,'#FFFFFF'),(13722,39440,'#d86395'),(13723,39449,'#72d6ec'),(13724,39461,'#fda377'),(13725,39468,'#a461f3'),(13726,39474,'#FFFFFF'),(13727,39476,'#fc394d'),(13728,39480,'#07e081'),(13729,39481,'#fd789a'),(13730,39490,'#6ce9d5'),(13731,39492,'#81f070'),(13732,39493,'#ba3ed7'),(13733,39498,'#527cdc'),(13734,39499,'#1dea12'),(13735,39501,'#FFFFFF'),(13736,39503,'#87e63a'),(13737,39508,'#FFFFFF'),(13738,39509,'#FFFFFF'),(13739,39512,'#FFFFFF'),(13740,39513,'#750acf'),(13741,39523,'#d15967'),(13742,39524,'#FFFFFF'),(13743,39525,'#64f3f2'),(13744,39532,'#51e6c7'),(13745,39533,'#FFFFFF'),(13746,39536,'#FFFFFF'),(13747,39538,'#d407ca'),(13748,39541,'#FFFFFF'),(13749,39545,'#FFFFFF'),(13750,39546,'#e014db'),(13751,39547,'#FFFFFF'),(13752,39554,'#70f5db'),(13753,39556,'#d90d6e'),(13754,39557,'#FFFFFF'),(13755,39560,'#FFFFFF'),(13756,39561,'#FFFFFF'),(13757,39562,'#FFFFFF'),(13758,39564,'#FFFFFF'),(13759,39574,'#FFFFFF'),(13760,39575,'#FFFFFF'),(13761,39579,'#4ac9f7'),(13762,39582,'#3f14e4'),(13763,39588,'#e65d26'),(13764,39590,'#FFFFFF'),(13765,39601,'#FFFFFF'),(13766,39608,'#FFFFFF'),(13767,39616,'#FFFFFF'),(13768,39617,'#e3473b'),(13769,39618,'#1309d4'),(13770,39619,'#FFFFFF'),(13771,39623,'#927afb'),(13772,39626,'#6bf548'),(13773,39628,'#f39970'),(13774,39633,'#d71724'),(13775,39635,'#FFFFFF'),(13776,39644,'#40d8b5'),(13777,39646,'#FFFFFF'),(13778,39648,'#1842f8'),(13779,39659,'#5becaf'),(13780,39688,'#54db06'),(13781,39691,'#d29030'),(13782,39695,'#d24f6b'),(13783,39708,'#a634e7'),(13784,39723,'#fa6cb8'),(13785,39729,'#43ea85'),(13786,39733,'#e3c16d'),(13787,39734,'#5450e4'),(13788,39735,'#262de6'),(13789,39737,'#b125d9'),(13790,39739,'#34fc6f'),(13791,39740,'#1d6dfb'),(13792,39747,'#8e26ef'),(13793,39749,'#FFFFFF'),(13794,39755,'#0fd1e0'),(13795,39757,'#69cefc'),(13796,39763,'#38dba7'),(13797,39765,'#82f13b'),(13798,39778,'#260ad7'),(13799,39784,'#e2906b'),(13800,39785,'#7130e2'),(13801,39786,'#c4f45b'),(13802,39788,'#a1e06c'),(13803,39794,'#eda45b'),(13804,39795,'#d13cd4'),(13805,39802,'#FFFFFF'),(13806,39804,'#1c11d0'),(13807,39811,'#b4f579'),(13808,39812,'#19f60f'),(13809,39816,'#f04294'),(13810,39825,'#a14cf8'),(13811,39828,'#2de382'),(13812,39831,'#FFFFFF'),(13813,39835,'#3b18cf'),(13814,39854,'#8963d1'),(13815,39857,'#757ded'),(13816,39858,'#68f273'),(13817,39868,'#54f0b2'),(13818,39873,'#f1d23f'),(13819,39885,'#4accdc'),(13820,39888,'#0a18dc'),(13821,39900,'#1abffe'),(13822,39908,'#df9f4c'),(13823,39939,'#d849e6'),(13824,39940,'#7744dc'),(13825,39947,'#45d3cf'),(13826,39948,'#e57017'),(13827,39960,'#de8a0f'),(13828,39967,'#FFFFFF'),(13829,39974,'#5985d8'),(13830,39992,'#fcef6f'),(13831,39993,'#6416ef'),(13832,40000,'#0fda5c'),(13833,40001,'#edbb37'),(13834,40004,'#c1f965'),(13835,40013,'#d8ea30'),(13836,40033,'#FFFFFF'),(13837,40040,'#d243ab'),(13838,40044,'#78edfd'),(13839,40046,'#2983f4'),(13840,40052,'#d1315d'),(13841,40057,'#2605f7'),(13842,40069,'#68e46f'),(13843,40071,'#a3e34e'),(13844,40076,'#5765dc'),(13845,40080,'#FFFFFF'),(13846,40088,'#daa823'),(13847,40139,'#FFFFFF'),(13848,40142,'#65badd'),(13849,40151,'#06ee6d'),(13850,40153,'#baee28'),(13851,40159,'#5e33e8'),(13852,40165,'#e23d6b'),(13853,40168,'#38ce2f'),(13854,40171,'#45d665'),(13855,40172,'#d81821'),(13856,40191,'#f277c0'),(13857,40206,'#cde237'),(13858,40225,'#52d28a'),(13859,40280,'#51cffc'),(13860,40281,'#66e1e1'),(13861,40317,'#fd7447'),(13862,40325,'#9df013'),(13863,40327,'#FFFFFF'),(13864,40329,'#FFFFFF'),(13865,40335,'#e55900'),(13866,40339,'#16ee09'),(13867,40346,'#c03cd9'),(13868,40356,'#39ade8'),(13869,40377,'#31daae'),(13870,40381,'#FFFFFF'),(13871,40384,'#4ac6e6'),(13872,40387,'#d22fb9'),(13873,40394,'#de6135'),(13874,40395,'#FFFFFF'),(13875,40396,'#6141ec'),(13876,40405,'#FFFFFF'),(13877,40429,'#FFFFFF'),(13878,40433,'#01eac9'),(13879,40436,'#2cf3d1'),(13880,40438,'#8a4ed7'),(13881,40445,'#ebc26f'),(13882,40452,'#4ce09f'),(13883,40460,'#784fd4'),(13884,40461,'#FFFFFF'),(13885,40465,'#e8a705'),(13886,40467,'#FFFFFF'),(13887,40468,'#e1575d'),(13888,40469,'#de455a'),(13889,40471,'#2aa8e6'),(13890,40481,'#19fe6d'),(13891,40484,'#458ffd'),(13892,40489,'#FFFFFF'),(13893,40492,'#d6314b'),(13894,40495,'#FFFFFF'),(13895,40509,'#FFFFFF'),(13896,40513,'#5be0c7'),(13897,40517,'#cf4fa4'),(13898,40519,'#456ecc'),(13899,40521,'#f01ddc'),(13900,40524,'#FFFFFF'),(13901,40526,'#e066dc'),(13902,40528,'#69d830'),(13903,40540,'#cee439'),(13904,40550,'#d44278'),(13905,40552,'#eeae00'),(13906,40553,'#d35876'),(13907,40554,'#FFFFFF'),(13908,40570,'#d7b108'),(13909,40571,'#FFFFFF'),(13910,40573,'#dd2a3e'),(13911,40581,'#FFFFFF'),(13912,40582,'#4ff148'),(13913,40585,'#eb2bc4'),(13914,40589,'#41da46'),(13915,40599,'#1b21e8'),(13916,40601,'#2dcc3d'),(13917,40605,'#67bdfe'),(13918,40619,'#637ce5'),(13919,40624,'#cd4730'),(13920,40625,'#FFFFFF'),(13921,40636,'#FFFFFF'),(13922,40640,'#FFFFFF'),(13923,40641,'#FFFFFF'),(13924,40643,'#FFFFFF'),(13925,40661,'#38d568'),(13926,40665,'#52d54f'),(13927,40667,'#f3490b'),(13928,40670,'#FFFFFF'),(13929,40672,'#e88e24'),(13930,40673,'#FFFFFF'),(13931,40674,'#f01757'),(13932,40680,'#03ddc0'),(13933,40681,'#4ad16f'),(13934,40686,'#d34fde'),(13935,40696,'#FFFFFF'),(13936,40700,'#FFFFFF'),(13937,40703,'#68e78a'),(13938,40714,'#4ff746'),(13939,40716,'#a0da06'),(13940,40722,'#57ea85'),(13941,40723,'#3ceca9'),(13942,40727,'#06e36b'),(13943,40778,'#72de2a'),(13944,40811,'#d18e44'),(13945,40870,'#0bcafc'),(13946,40875,'#da0d1b'),(13947,40883,'#f08d44'),(13948,40885,'#608df7'),(13949,40887,'#FFFFFF'),(13950,40888,'#f5a369'),(13951,40889,'#9cf318'),(13952,40892,'#FFFFFF'),(13953,40894,'#FFFFFF'),(13954,40895,'#26cbd3'),(13955,40897,'#53dd5a'),(13956,40904,'#8cf26a'),(13957,40913,'#FFFFFF'),(13958,40925,'#ce8a62'),(13959,40928,'#3de437'),(13960,40936,'#63d067'),(13961,40939,'#4060cc'),(13962,40941,'#f36c08'),(13963,40942,'#f34af3'),(13964,40947,'#0e3fd5'),(13965,40950,'#ec9848'),(13966,40952,'#c46ae9'),(13967,40963,'#e305d9'),(13968,40967,'#f225b8'),(13969,40970,'#FFFFFF'),(13970,40971,'#FFFFFF'),(13971,40972,'#dd379d'),(13972,40974,'#dadd5e'),(13973,40977,'#4b80d6'),(13974,40983,'#FFFFFF'),(13975,40994,'#e16b1b'),(13976,40997,'#fb3f53'),(13977,40998,'#8aec61'),(13978,41006,'#FFFFFF'),(13979,41008,'#dafd00'),(13980,41010,'#FFFFFF'),(13981,41021,'#FFFFFF'),(13982,41022,'#ab6fe2'),(13983,41029,'#413bdf'),(13984,41030,'#4049d5'),(13985,41031,'#0efb3b'),(13986,41032,'#60d79c'),(13987,41035,'#f44d9c'),(13988,41036,'#ead45a'),(13989,41037,'#b2d820'),(13990,41038,'#FFFFFF'),(13991,41039,'#0eeed8'),(13992,41040,'#f25ec2'),(13993,41041,'#e9f032'),(13994,41046,'#57b8ee'),(13995,41049,'#9ecf3c'),(13996,41050,'#2d13d0'),(13997,41051,'#8f31f4'),(13998,41052,'#c4ce0a'),(13999,41056,'#66e767'),(14000,41057,'#71a0e2'),(14001,41058,'#7b5bce'),(14002,41059,'#1425e3'),(14003,41060,'#a7e96e'),(14004,41063,'#a1ee1a'),(14005,41064,'#e03a1c'),(14006,41065,'#8cd042'),(14007,41069,'#b86deb'),(14008,41070,'#408fd8'),(14009,41071,'#52df1c'),(14010,41074,'#e129db'),(14011,41076,'#FFFFFF'),(14012,41077,'#FFFFFF'),(14013,41080,'#5ea1d6'),(14014,41081,'#7d59cc'),(14015,41082,'#c2e05f'),(14016,41083,'#9232e8'),(14017,41084,'#49edad'),(14018,41085,'#FFFFFF'),(14019,41087,'#31c8df'),(14020,41090,'#3f4eec'),(14021,41091,'#FFFFFF'),(14022,41094,'#1fdb9f'),(14023,41095,'#42bff8'),(14024,41097,'#FFFFFF'),(14025,41099,'#67f3a7'),(14026,41100,'#fc443d'),(14027,41101,'#666ae4'),(14028,41102,'#19e820'),(14029,41103,'#db6c97'),(14030,41105,'#FFFFFF'),(14031,41106,'#dd0f40'),(14032,41109,'#f6eb69'),(14033,41110,'#ed821b'),(14034,41111,'#FFFFFF'),(14035,41113,'#d21470'),(14036,41114,'#2ecd50'),(14037,41117,'#e756d2'),(14038,41118,'#2fe22d'),(14039,41123,'#303dd9'),(14040,41124,'#02aef5'),(14041,41127,'#e42123'),(14042,41128,'#d56cf2'),(14043,41130,'#eeaa59'),(14044,41132,'#caf512'),(14045,41133,'#FFFFFF'),(14046,41136,'#3b5bdb'),(14047,41137,'#6df7b9'),(14048,41139,'#fcbf29'),(14049,41140,'#FFFFFF'),(14050,41145,'#cc9043'),(14051,41148,'#eb5f25'),(14052,41151,'#4a5bfc'),(14053,41152,'#FFFFFF'),(14054,41153,'#282feb'),(14055,41154,'#1cfc82'),(14056,41155,'#db61ae'),(14057,41159,'#ea6a2a'),(14058,41160,'#e47203'),(14059,41162,'#7043e4'),(14060,41163,'#536bd0'),(14061,41164,'#0e80ef'),(14062,41165,'#24da3b'),(14063,41166,'#FFFFFF'),(14064,41168,'#2e40db'),(14065,41170,'#5722d2'),(14066,41171,'#e86745'),(14067,41173,'#5ececb'),(14068,41174,'#de3104'),(14069,41176,'#13e226'),(14070,41178,'#FFFFFF'),(14071,41179,'#86cd60'),(14072,41188,'#e8294d'),(14073,41190,'#f08655'),(14074,41191,'#f97226'),(14075,41192,'#e82950'),(14076,41197,'#d3ee61'),(14077,41198,'#335bfa'),(14078,41200,'#26d11b'),(14079,41201,'#FFFFFF'),(14080,41203,'#1fdca0'),(14081,41206,'#6af58b'),(14082,41207,'#6cda5a'),(14083,41217,'#5883fd'),(14084,41218,'#FFFFFF'),(14085,41219,'#FFFFFF'),(14086,41221,'#eb1b6f'),(14087,41222,'#e46ca1'),(14088,41224,'#4ff2bf'),(14089,41225,'#d9b13d'),(14090,41226,'#fc2939'),(14091,41227,'#65dd7a'),(14092,41228,'#8816fa'),(14093,41229,'#05d4e9'),(14094,41230,'#d94254'),(14095,41231,'#eccf04'),(14096,41232,'#ed8469'),(14097,41235,'#0765da'),(14098,41237,'#f95b1f'),(14099,41238,'#e38f1a'),(14100,41239,'#1e35e6'),(14101,41240,'#1ccc50'),(14102,41243,'#b455ce'),(14103,41245,'#2bcf60'),(14104,41249,'#e75586'),(14105,41251,'#e37b61'),(14106,41254,'#fd06cb'),(14107,41258,'#f913bd'),(14108,41260,'#6fed0c'),(14109,41261,'#df8307'),(14110,41265,'#40f8b4'),(14111,41267,'#FFFFFF'),(14112,41268,'#29b2fb'),(14113,41269,'#719ff5'),(14114,41270,'#e5ef41'),(14115,41272,'#FFFFFF'),(14116,41273,'#f24085'),(14117,41277,'#5181f9'),(14118,41278,'#eac071'),(14119,41280,'#9f46de'),(14120,41281,'#7d57e9'),(14121,41283,'#d55822'),(14122,41285,'#cf0792'),(14123,41286,'#e3d60b'),(14124,41288,'#5edc11'),(14125,41289,'#FFFFFF'),(14126,41290,'#0b2cec'),(14127,41291,'#4069cc'),(14128,41293,'#6cd970'),(14129,41297,'#4cda03'),(14130,41299,'#cee338'),(14131,41300,'#2aea0e'),(14132,41301,'#FFFFFF'),(14133,41302,'#d21261'),(14134,41303,'#5e6ef0'),(14135,41304,'#36c1d4'),(14136,41305,'#745ce1'),(14137,41307,'#fd6a11'),(14138,41308,'#e0fc7a'),(14139,41313,'#fd6d5f'),(14140,41317,'#dacd41'),(14141,41318,'#FFFFFF'),(14142,41319,'#6971e4'),(14143,41320,'#1bd126'),(14144,41321,'#34d536'),(14145,41323,'#0315ea'),(14146,41324,'#fa1ee5'),(14147,41325,'#594ecd'),(14148,41326,'#569de3'),(14149,41328,'#6aec75'),(14150,41329,'#7d67e0'),(14151,41330,'#d75f7d'),(14152,41331,'#eaaf37'),(14153,41334,'#68e03c'),(14154,41339,'#eb2eb1'),(14155,41341,'#6aee6a'),(14156,41343,'#b3ed51'),(14157,41344,'#73eeb7'),(14158,41345,'#f7f216'),(14159,41346,'#b7d142'),(14160,41347,'#f74e9a'),(14161,41349,'#4fd75b'),(14162,41350,'#65ef28'),(14163,41352,'#4f52ed'),(14164,41354,'#dac217'),(14165,41355,'#FFFFFF'),(14166,41356,'#1ed15c'),(14167,41359,'#FFFFFF'),(14168,41364,'#FFFFFF'),(14169,41365,'#d8f478'),(14170,41366,'#ebf74d'),(14171,41368,'#c5fb1b'),(14172,41369,'#FFFFFF'),(14173,41370,'#fe3121'),(14174,41372,'#e76425'),(14175,41373,'#1acd6c'),(14176,41374,'#f2c351'),(14177,41375,'#41dea9'),(14178,41376,'#b75eef'),(14179,41377,'#449eed'),(14180,41382,'#eb8c02'),(14181,41383,'#cf8d3c'),(14182,41388,'#42e488'),(14183,41389,'#d34d70'),(14184,41390,'#fd0894'),(14185,41393,'#1e4cd1'),(14186,41394,'#dc66d8'),(14187,41396,'#33f48f'),(14188,41397,'#bb02f9'),(14189,41398,'#55e46a'),(14190,41399,'#e94751'),(14191,41400,'#75eddd'),(14192,41401,'#d24713'),(14193,41402,'#44d263'),(14194,41403,'#FFFFFF'),(14195,41405,'#FFFFFF'),(14196,41406,'#fdb271'),(14197,41407,'#FFFFFF'),(14198,41411,'#2e55e3'),(14199,41416,'#cc0d4f'),(14200,41417,'#0824e9'),(14201,41418,'#c2d62d'),(14202,41420,'#0d72e6'),(14203,41425,'#FFFFFF'),(14204,41428,'#1c45d1'),(14205,41429,'#ed9102'),(14206,41430,'#f0871b'),(14207,41431,'#43f3ef'),(14208,41432,'#df9368'),(14209,41433,'#49ec00'),(14210,41435,'#4c70e1'),(14211,41439,'#4496d2'),(14212,41440,'#2476df'),(14213,41441,'#ab3ed1'),(14214,41442,'#74e333'),(14215,41443,'#200cd2'),(14216,41444,'#e36ad5'),(14217,41445,'#7085f5'),(14218,41446,'#ccee15'),(14219,41448,'#45e55b'),(14220,41449,'#dcc163'),(14221,41454,'#fb0389'),(14222,41455,'#a907d1'),(14223,41456,'#add80e'),(14224,41457,'#FFFFFF'),(14225,41458,'#FFFFFF'),(14226,41460,'#f20437'),(14227,41461,'#FFFFFF'),(14228,41463,'#62f5a7'),(14229,41464,'#FFFFFF'),(14230,41465,'#cd40c0'),(14231,41466,'#42f7ce'),(14232,41467,'#864ae7'),(14233,41468,'#FFFFFF'),(14234,41470,'#06e5d5'),(14235,41471,'#FFFFFF'),(14236,41473,'#de47b0'),(14237,41475,'#FFFFFF'),(14238,41480,'#2fd461'),(14239,41481,'#FFFFFF'),(14240,41484,'#FFFFFF'),(14241,41485,'#FFFFFF'),(14242,41486,'#FFFFFF'),(14243,41487,'#FFFFFF'),(14244,41489,'#43edac'),(14245,41490,'#cda943'),(14246,41491,'#FFFFFF'),(14247,41492,'#FFFFFF'),(14248,41493,'#FFFFFF'),(14249,41494,'#FFFFFF'),(14250,41495,'#FFFFFF'),(14251,41497,'#FFFFFF'),(14252,41498,'#FFFFFF'),(14253,41500,'#69ecf8'),(14254,41501,'#FFFFFF'),(14255,41502,'#FFFFFF'),(14256,41504,'#f26e4c'),(14257,41505,'#FFFFFF'),(14258,41510,'#e43a59'),(14259,41511,'#FFFFFF'),(14260,41513,'#FFFFFF'),(14261,41514,'#FFFFFF'),(14262,41515,'#FFFFFF'),(14263,41516,'#46ce5a'),(14264,41517,'#FFFFFF'),(14265,41518,'#FFFFFF'),(14266,41519,'#FFFFFF'),(14267,41520,'#FFFFFF'),(14268,41521,'#f5d31f'),(14269,41522,'#FFFFFF'),(14270,41523,'#FFFFFF'),(14271,41524,'#FFFFFF'),(14272,41525,'#FFFFFF'),(14273,41527,'#26ce15'),(14274,41531,'#FFFFFF'),(14275,41533,'#FFFFFF'),(14276,41534,'#FFFFFF'),(14277,41535,'#FFFFFF'),(14278,41536,'#FFFFFF'),(14279,41537,'#FFFFFF'),(14280,41538,'#FFFFFF'),(14281,41542,'#FFFFFF'),(14282,41544,'#cc43a0'),(14283,41547,'#FFFFFF'),(14284,41548,'#FFFFFF'),(14285,41549,'#FFFFFF'),(14286,41552,'#FFFFFF'),(14287,41553,'#FFFFFF'),(14288,41554,'#FFFFFF'),(14289,41555,'#FFFFFF'),(14290,41556,'#FFFFFF'),(14291,41558,'#FFFFFF'),(14292,41559,'#FFFFFF'),(14293,41561,'#e1f174'),(14294,41562,'#FFFFFF'),(14295,41565,'#b838cf'),(14296,41568,'#FFFFFF'),(14297,41573,'#FFFFFF'),(14298,41574,'#FFFFFF'),(14299,41575,'#FFFFFF'),(14300,41576,'#56d7b3'),(14301,41577,'#FFFFFF'),(14302,41578,'#FFFFFF'),(14303,41580,'#FFFFFF'),(14304,41581,'#FFFFFF'),(14305,41582,'#FFFFFF'),(14306,41583,'#FFFFFF'),(14307,41585,'#FFFFFF'),(14308,41587,'#FFFFFF'),(14309,41588,'#FFFFFF'),(14310,41589,'#a22acc'),(14311,41592,'#FFFFFF'),(14312,41594,'#FFFFFF'),(14313,41596,'#FFFFFF'),(14314,41597,'#FFFFFF'),(14315,41600,'#fb58d1'),(14316,41601,'#FFFFFF'),(14317,41603,'#FFFFFF'),(14318,41604,'#FFFFFF'),(14319,41607,'#FFFFFF'),(14320,41608,'#de4063'),(14321,41610,'#FFFFFF'),(14322,41611,'#FFFFFF'),(14323,41612,'#d05974'),(14324,41614,'#FFFFFF'),(14325,41617,'#FFFFFF'),(14326,41618,'#FFFFFF'),(14327,41619,'#FFFFFF'),(14328,41620,'#FFFFFF'),(14329,41621,'#FFFFFF'),(14330,41622,'#FFFFFF'),(14331,41623,'#FFFFFF'),(14332,41625,'#34f2da'),(14333,41626,'#FFFFFF'),(14334,41628,'#FFFFFF'),(14335,41629,'#FFFFFF'),(14336,41632,'#3dd454'),(14337,41633,'#FFFFFF'),(14338,41634,'#FFFFFF'),(14339,41635,'#FFFFFF'),(14340,41638,'#FFFFFF'),(14341,41639,'#FFFFFF'),(14342,41641,'#FFFFFF'),(14343,41642,'#FFFFFF'),(14344,41643,'#FFFFFF'),(14345,41644,'#FFFFFF'),(14346,41645,'#FFFFFF'),(14347,41646,'#FFFFFF'),(14348,41647,'#FFFFFF'),(14349,41649,'#FFFFFF'),(14350,41651,'#FFFFFF'),(14351,41652,'#FFFFFF'),(14352,41653,'#FFFFFF'),(14353,41654,'#FFFFFF'),(14354,41655,'#FFFFFF'),(14355,41656,'#FFFFFF'),(14356,41657,'#FFFFFF'),(14357,41658,'#FFFFFF'),(14358,41659,'#FFFFFF'),(14359,41664,'#825fe6'),(14360,41665,'#FFFFFF'),(14361,41666,'#FFFFFF'),(14362,41667,'#FFFFFF'),(14363,41668,'#FFFFFF'),(14364,41671,'#FFFFFF'),(14365,41672,'#FFFFFF'),(14366,41673,'#FFFFFF'),(14367,41678,'#FFFFFF'),(14368,41679,'#cf9854'),(14369,41682,'#FFFFFF'),(14370,41683,'#FFFFFF'),(14371,41684,'#56f22e'),(14372,41685,'#FFFFFF'),(14373,41686,'#55fbf5'),(14374,41689,'#FFFFFF'),(14375,41692,'#FFFFFF'),(14376,41693,'#FFFFFF'),(14377,41694,'#FFFFFF'),(14378,41695,'#FFFFFF'),(14379,41696,'#FFFFFF'),(14380,41697,'#FFFFFF'),(14381,41698,'#FFFFFF'),(14382,41699,'#e01051'),(14383,41700,'#FFFFFF'),(14384,41703,'#FFFFFF'),(14385,41704,'#ebb149'),(14386,41707,'#f44732'),(14387,41708,'#8b52de'),(14388,41709,'#2f25ce'),(14389,41710,'#605de3'),(14390,41711,'#5c5ce0'),(14391,41714,'#FFFFFF'),(14392,41715,'#FFFFFF'),(14393,41716,'#FFFFFF'),(14394,41717,'#f3ba6b'),(14395,41719,'#FFFFFF'),(14396,41720,'#13d4d9'),(14397,41723,'#FFFFFF'),(14398,41724,'#FFFFFF'),(14399,41726,'#FFFFFF'),(14400,41727,'#FFFFFF'),(14401,41729,'#FFFFFF'),(14402,41730,'#cd4b7b'),(14403,41731,'#FFFFFF'),(14404,41733,'#FFFFFF'),(14405,41734,'#FFFFFF'),(14406,41735,'#e27b6d'),(14407,41737,'#FFFFFF'),(14408,41739,'#FFFFFF'),(14409,41740,'#FFFFFF'),(14410,41742,'#FFFFFF'),(14411,41743,'#FFFFFF'),(14412,41744,'#d96669'),(14413,41745,'#FFFFFF'),(14414,41747,'#FFFFFF'),(14415,41748,'#FFFFFF'),(14416,41749,'#30d644'),(14417,41750,'#FFFFFF'),(14418,41752,'#FFFFFF'),(14419,41753,'#FFFFFF'),(14420,41754,'#FFFFFF'),(14421,41755,'#d52c07'),(14422,41757,'#FFFFFF'),(14423,41758,'#FFFFFF'),(14424,41760,'#FFFFFF'),(14425,41763,'#FFFFFF'),(14426,41764,'#FFFFFF'),(14427,41765,'#FFFFFF'),(14428,41766,'#ae1df9'),(14429,41767,'#FFFFFF'),(14430,41768,'#FFFFFF'),(14431,41769,'#FFFFFF'),(14432,41770,'#FFFFFF'),(14433,41771,'#FFFFFF'),(14434,41773,'#13e200'),(14435,41774,'#FFFFFF'),(14436,41775,'#FFFFFF'),(14437,41776,'#FFFFFF'),(14438,41778,'#FFFFFF'),(14439,41779,'#FFFFFF'),(14440,41780,'#FFFFFF'),(14441,41781,'#FFFFFF'),(14442,41782,'#d6d91b'),(14443,41784,'#FFFFFF'),(14444,41789,'#FFFFFF'),(14445,41790,'#FFFFFF'),(14446,41791,'#FFFFFF'),(14447,41792,'#FFFFFF'),(14448,41793,'#FFFFFF'),(14449,41794,'#FFFFFF'),(14450,41795,'#FFFFFF'),(14451,41796,'#6555e2'),(14452,41797,'#FFFFFF'),(14453,41798,'#FFFFFF'),(14454,41799,'#FFFFFF'),(14455,41801,'#FFFFFF'),(14456,41807,'#FFFFFF'),(14457,41809,'#14fb43'),(14458,41810,'#FFFFFF'),(14459,41811,'#FFFFFF'),(14460,41812,'#FFFFFF'),(14461,41814,'#FFFFFF'),(14462,41816,'#36e1bd'),(14463,41819,'#FFFFFF'),(14464,41820,'#c2ef33'),(14465,41822,'#e354ca'),(14466,41823,'#FFFFFF'),(14467,41824,'#FFFFFF'),(14468,41826,'#FFFFFF'),(14469,41827,'#FFFFFF'),(14470,41828,'#FFFFFF'),(14471,41829,'#FFFFFF'),(14472,41830,'#FFFFFF'),(14473,41833,'#FFFFFF'),(14474,41835,'#FFFFFF'),(14475,41846,'#2604f0'),(14476,41903,'#1d01fa'),(14477,41951,'#5805d8'),(14478,41991,'#da55c9'),(14479,42012,'#0f6ffa'),(14480,42017,'#9cfe54'),(14481,42033,'#43dc6a'),(14482,42055,'#47d04d'),(14483,42065,'#ccae22'),(14484,42114,'#0927e1'),(14485,42133,'#e6d441'),(14486,42139,'#5bef94'),(14487,42152,'#40f151'),(14488,42160,'#16d725'),(14489,42163,'#FFFFFF'),(14490,42173,'#FFFFFF'),(14491,42177,'#FFFFFF'),(14492,42187,'#FFFFFF'),(14493,42197,'#7889f4'),(14494,42207,'#d8547b'),(14495,42211,'#FFFFFF'),(14496,42214,'#FFFFFF'),(14497,42218,'#5bce39'),(14498,42233,'#a3d167'),(14499,42237,'#FFFFFF'),(14500,42252,'#FFFFFF'),(14501,42254,'#FFFFFF'),(14502,42261,'#FFFFFF'),(14503,42269,'#FFFFFF'),(14504,42271,'#FFFFFF'),(14505,42276,'#FFFFFF'),(14506,42292,'#FFFFFF'),(14507,42296,'#FFFFFF'),(14508,42304,'#1012ef'),(14509,42305,'#FFFFFF'),(14510,42307,'#FFFFFF'),(14511,42310,'#FFFFFF'),(14512,42316,'#FFFFFF'),(14513,42344,'#ddaa4f'),(14514,42345,'#e08a29'),(14515,42353,'#abeb2e'),(14516,42357,'#FFFFFF'),(14517,42373,'#59ce7e'),(14518,42375,'#FFFFFF'),(14519,42376,'#6cf781'),(14520,42385,'#FFFFFF'),(14521,42395,'#FFFFFF'),(14522,42396,'#bbe742'),(14523,42403,'#FFFFFF'),(14524,42411,'#FFFFFF'),(14525,42416,'#FFFFFF'),(14526,42420,'#FFFFFF'),(14527,42421,'#FFFFFF'),(14528,42427,'#FFFFFF'),(14529,42431,'#108dcc'),(14530,42437,'#FFFFFF'),(14531,42440,'#FFFFFF'),(14532,42441,'#FFFFFF'),(14533,42447,'#2251ed'),(14534,42454,'#FFFFFF'),(14535,42460,'#61c9d5'),(14536,42474,'#e973aa'),(14537,42487,'#FFFFFF'),(14538,42491,'#FFFFFF'),(14539,42493,'#FFFFFF'),(14540,42494,'#FFFFFF'),(14541,42499,'#FFFFFF'),(14542,42504,'#e38e09'),(14543,42505,'#FFFFFF'),(14544,42511,'#FFFFFF'),(14545,42515,'#FFFFFF'),(14546,42520,'#FFFFFF'),(14547,42522,'#FFFFFF'),(14548,42524,'#FFFFFF'),(14549,42530,'#FFFFFF'),(14550,42534,'#43de37'),(14551,42538,'#FFFFFF'),(14552,42545,'#FFFFFF'),(14553,42551,'#FFFFFF'),(14554,42556,'#e6cf23'),(14555,42562,'#FFFFFF'),(14556,42567,'#dd303b'),(14557,42587,'#06a5fa'),(14558,42589,'#FFFFFF'),(14559,42598,'#ed5ae5'),(14560,42604,'#FFFFFF'),(14561,42616,'#FFFFFF'),(14562,42619,'#FFFFFF'),(14563,42620,'#FFFFFF'),(14564,42621,'#FFFFFF'),(14565,42624,'#a7ef60'),(14566,42625,'#FFFFFF'),(14567,42636,'#FFFFFF'),(14568,42638,'#FFFFFF'),(14569,42639,'#FFFFFF'),(14570,42640,'#68d780'),(14571,42641,'#e839f4'),(14572,42644,'#5024e7'),(14573,42645,'#FFFFFF'),(14574,42646,'#55e9e5'),(14575,42647,'#FFFFFF'),(14576,42648,'#FFFFFF'),(14577,42652,'#FFFFFF'),(14578,42653,'#8d5ce5'),(14579,42656,'#dc8e5f'),(14580,42660,'#FFFFFF'),(14581,42662,'#1050de'),(14582,42663,'#FFFFFF'),(14583,42665,'#FFFFFF'),(14584,42667,'#d75846'),(14585,42672,'#2e31dd'),(14586,42673,'#FFFFFF'),(14587,42674,'#FFFFFF'),(14588,42675,'#FFFFFF'),(14589,42677,'#FFFFFF'),(14590,42678,'#f557ae'),(14591,42679,'#FFFFFF'),(14592,42684,'#FFFFFF'),(14593,42686,'#FFFFFF'),(14594,42687,'#FFFFFF'),(14595,42688,'#FFFFFF'),(14596,42695,'#86cc04'),(14597,42697,'#2dd8cf'),(14598,42698,'#FFFFFF'),(14599,42699,'#FFFFFF'),(14600,42702,'#FFFFFF'),(14601,42703,'#FFFFFF'),(14602,42707,'#FFFFFF'),(14603,42709,'#FFFFFF'),(14604,42712,'#FFFFFF'),(14605,42713,'#d73dcf'),(14606,42714,'#3efea2'),(14607,42716,'#FFFFFF'),(14608,42717,'#FFFFFF'),(14609,42722,'#FFFFFF'),(14610,42725,'#FFFFFF'),(14611,42728,'#FFFFFF'),(14612,42729,'#FFFFFF'),(14613,42730,'#FFFFFF'),(14614,42731,'#FFFFFF'),(14615,42733,'#FFFFFF'),(14616,42739,'#FFFFFF'),(14617,42740,'#FFFFFF'),(14618,42743,'#edc86f'),(14619,42744,'#FFFFFF'),(14620,42747,'#FFFFFF'),(14621,42748,'#FFFFFF'),(14622,42750,'#FFFFFF'),(14623,42753,'#FFFFFF'),(14624,42756,'#FFFFFF'),(14625,42758,'#1ffaef'),(14626,42759,'#FFFFFF'),(14627,42761,'#4244f0'),(14628,42764,'#FFFFFF'),(14629,42765,'#32fe40'),(14630,42770,'#FFFFFF'),(14631,42771,'#0340f1'),(14632,42773,'#FFFFFF'),(14633,42774,'#FFFFFF'),(14634,42778,'#FFFFFF'),(14635,42779,'#FFFFFF'),(14636,42780,'#FFFFFF'),(14637,42781,'#efa353'),(14638,42782,'#d709d1'),(14639,42783,'#FFFFFF'),(14640,42785,'#FFFFFF'),(14641,42787,'#ef9d33'),(14642,42788,'#55d512'),(14643,42790,'#d80659'),(14644,42794,'#FFFFFF'),(14645,42795,'#FFFFFF'),(14646,42796,'#FFFFFF'),(14647,42797,'#FFFFFF'),(14648,42799,'#FFFFFF'),(14649,42801,'#FFFFFF'),(14650,42802,'#FFFFFF'),(14651,42803,'#FFFFFF'),(14652,42805,'#FFFFFF'),(14653,42814,'#72e9a7'),(14654,42815,'#FFFFFF'),(14655,42816,'#FFFFFF'),(14656,42817,'#13f025'),(14657,42818,'#FFFFFF'),(14658,42819,'#FFFFFF'),(14659,42820,'#FFFFFF'),(14660,42821,'#FFFFFF'),(14661,42822,'#FFFFFF'),(14662,42823,'#FFFFFF'),(14663,42824,'#b155dd'),(14664,42825,'#bbd537'),(14665,42827,'#d6a216'),(14666,42828,'#FFFFFF'),(14667,42829,'#FFFFFF'),(14668,42830,'#cfa947'),(14669,42831,'#54e0ea'),(14670,42832,'#FFFFFF'),(14671,42833,'#FFFFFF'),(14672,42834,'#FFFFFF'),(14673,42835,'#FFFFFF'),(14674,42836,'#FFFFFF'),(14675,42837,'#FFFFFF'),(14676,42838,'#ae3ad2'),(14677,42839,'#f2b116'),(14678,42840,'#FFFFFF'),(14679,42842,'#FFFFFF'),(14680,42843,'#FFFFFF'),(14681,42844,'#FFFFFF'),(14682,42845,'#FFFFFF'),(14683,42846,'#FFFFFF'),(14684,42847,'#FFFFFF'),(14685,42848,'#67f2a3'),(14686,42849,'#FFFFFF'),(14687,42850,'#FFFFFF'),(14688,42851,'#FFFFFF'),(14689,42852,'#FFFFFF'),(14690,42853,'#fcc41c'),(14691,42854,'#e10921'),(14692,42855,'#FFFFFF'),(14693,42856,'#5058da'),(14694,42857,'#ecde0c'),(14695,42858,'#FFFFFF'),(14696,42859,'#FFFFFF'),(14697,42860,'#c551f4'),(14698,42862,'#FFFFFF'),(14699,42863,'#FFFFFF'),(14700,42864,'#FFFFFF'),(14701,42865,'#FFFFFF'),(14702,42866,'#5ff84e'),(14703,42867,'#FFFFFF'),(14704,42868,'#FFFFFF'),(14705,42869,'#FFFFFF'),(14706,42870,'#FFFFFF'),(14707,42871,'#23c1e7'),(14708,42874,'#FFFFFF'),(14709,42875,'#FFFFFF'),(14710,42876,'#51a9df'),(14711,42877,'#00d1d1'),(14712,42878,'#FFFFFF'),(14713,42879,'#429fcd'),(14714,42880,'#0289e5'),(14715,42881,'#FFFFFF'),(14716,42882,'#afdd32'),(14717,42883,'#FFFFFF'),(14718,42884,'#FFFFFF'),(14719,42885,'#FFFFFF'),(14720,42886,'#FFFFFF'),(14721,42887,'#d6dd37'),(14722,42888,'#15ed70'),(14723,42889,'#0adb33'),(14724,42890,'#FFFFFF'),(14725,42891,'#FFFFFF'),(14726,42893,'#FFFFFF'),(14727,42894,'#FFFFFF'),(14728,42895,'#FFFFFF'),(14729,42896,'#eba615'),(14730,42897,'#FFFFFF'),(14731,42898,'#FFFFFF'),(14732,42899,'#FFFFFF'),(14733,42900,'#43f8e5'),(14734,42901,'#fac603'),(14735,42902,'#FFFFFF'),(14736,42903,'#7817fa'),(14737,42904,'#FFFFFF'),(14738,42905,'#FFFFFF'),(14739,42906,'#09a0ec'),(14740,42907,'#fe411d'),(14741,42908,'#f25070'),(14742,42909,'#FFFFFF'),(14743,42910,'#b713de'),(14744,42911,'#ee0cb0'),(14745,42912,'#FFFFFF'),(14746,42913,'#ee0e54'),(14747,42914,'#d8d464'),(14748,42915,'#FFFFFF'),(14749,42917,'#FFFFFF'),(14750,42918,'#FFFFFF'),(14751,42919,'#FFFFFF'),(14752,42920,'#8ae527'),(14753,42921,'#cdda3e'),(14754,42922,'#b6f849'),(14755,42923,'#e91c3b'),(14756,42924,'#d767c9'),(14757,42925,'#FFFFFF'),(14758,42926,'#FFFFFF'),(14759,42927,'#FFFFFF'),(14760,42928,'#FFFFFF'),(14761,42929,'#FFFFFF'),(14762,42930,'#FFFFFF'),(14763,42931,'#FFFFFF'),(14764,42932,'#e87135'),(14765,42933,'#FFFFFF'),(14766,42934,'#FFFFFF'),(14767,42935,'#FFFFFF'),(14768,42938,'#fefb7e'),(14769,42939,'#FFFFFF'),(14770,42941,'#d1aa38'),(14771,42942,'#FFFFFF'),(14772,42943,'#2808ce'),(14773,42944,'#FFFFFF'),(14774,42947,'#FFFFFF'),(14775,42949,'#FFFFFF'),(14776,42950,'#FFFFFF'),(14777,42951,'#e02c4c'),(14778,42952,'#FFFFFF'),(14779,42953,'#FFFFFF'),(14780,42954,'#FFFFFF'),(14781,42955,'#FFFFFF'),(14782,42956,'#FFFFFF'),(14783,42957,'#FFFFFF'),(14784,42958,'#FFFFFF'),(14785,42959,'#9f18dd'),(14786,42960,'#3edab8'),(14787,42961,'#a0eb69'),(14788,42962,'#1ed300'),(14789,42963,'#FFFFFF'),(14790,42964,'#3de855'),(14791,42965,'#FFFFFF'),(14792,42966,'#fd4381'),(14793,42967,'#dabb41'),(14794,42968,'#FFFFFF'),(14795,42969,'#e11a9a'),(14796,42970,'#FFFFFF'),(14797,42972,'#FFFFFF'),(14798,42973,'#7201d4'),(14799,42974,'#FFFFFF'),(14800,42975,'#FFFFFF'),(14801,42976,'#cd41a2'),(14802,42977,'#FFFFFF'),(14803,42978,'#FFFFFF'),(14804,42979,'#FFFFFF'),(14805,42980,'#FFFFFF'),(14806,42981,'#FFFFFF'),(14807,42982,'#FFFFFF'),(14808,42983,'#FFFFFF'),(14809,42984,'#a0e10c'),(14810,42986,'#3bf72f'),(14811,42987,'#507dcd'),(14812,42988,'#FFFFFF'),(14813,42989,'#FFFFFF'),(14814,42990,'#FFFFFF'),(14815,42991,'#5b7dd8'),(14816,42992,'#FFFFFF'),(14817,42993,'#FFFFFF'),(14818,42994,'#FFFFFF'),(14819,42995,'#FFFFFF'),(14820,42997,'#60d004'),(14821,42998,'#b43aea'),(14822,42999,'#0087fa'),(14823,43000,'#FFFFFF'),(14824,43001,'#FFFFFF'),(14825,43002,'#FFFFFF'),(14826,43003,'#FFFFFF'),(14827,43004,'#FFFFFF'),(14828,43005,'#143be6'),(14829,43006,'#FFFFFF'),(14830,43007,'#FFFFFF'),(14831,43008,'#FFFFFF'),(14832,43009,'#FFFFFF'),(14833,43010,'#5da3d5'),(14834,43011,'#FFFFFF'),(14835,43012,'#FFFFFF'),(14836,43013,'#FFFFFF'),(14837,43014,'#22e668'),(14838,43015,'#36f9b1'),(14839,43016,'#FFFFFF'),(14840,43017,'#FFFFFF'),(14841,43018,'#da02e5'),(14842,43019,'#f169d9'),(14843,43020,'#FFFFFF'),(14844,43021,'#FFFFFF'),(14845,43023,'#a6d667'),(14846,43024,'#FFFFFF'),(14847,43025,'#c0d85c'),(14848,43026,'#FFFFFF'),(14849,43027,'#FFFFFF'),(14850,43028,'#8df306'),(14851,43029,'#61ace0'),(14852,43030,'#FFFFFF'),(14853,43031,'#FFFFFF'),(14854,43032,'#FFFFFF'),(14855,43034,'#FFFFFF'),(14856,43036,'#FFFFFF'),(14857,43038,'#FFFFFF'),(14858,43039,'#0224dc'),(14859,43040,'#FFFFFF'),(14860,43041,'#FFFFFF'),(14861,43042,'#FFFFFF'),(14862,43043,'#5b10fb'),(14863,43044,'#FFFFFF'),(14864,43045,'#FFFFFF'),(14865,43046,'#56f772'),(14866,43047,'#5693d5'),(14867,43048,'#f7915e'),(14868,43049,'#51cca7'),(14869,43050,'#cfd043'),(14870,43051,'#e23594'),(14871,43052,'#FFFFFF'),(14872,43053,'#FFFFFF'),(14873,43054,'#FFFFFF'),(14874,43055,'#FFFFFF'),(14875,43056,'#FFFFFF'),(14876,43057,'#11f8df'),(14877,43058,'#FFFFFF'),(14878,43059,'#FFFFFF'),(14879,43060,'#FFFFFF'),(14880,43061,'#FFFFFF'),(14881,43062,'#ee3524'),(14882,43063,'#FFFFFF'),(14883,43064,'#FFFFFF'),(14884,43065,'#FFFFFF'),(14885,43066,'#FFFFFF'),(14886,43067,'#691fea'),(14887,43068,'#6cfb74'),(14888,43069,'#FFFFFF'),(14889,43070,'#FFFFFF'),(14890,43071,'#FFFFFF'),(14891,43072,'#FFFFFF'),(14892,43073,'#da761a'),(14893,43074,'#3420cf'),(14894,43075,'#c7e56c'),(14895,43076,'#24d32e'),(14896,43077,'#f82a6e'),(14897,43078,'#646dfa'),(14898,43080,'#FFFFFF'),(14899,43081,'#FFFFFF'),(14900,43082,'#21d1ea'),(14901,43084,'#6cf15a'),(14902,43085,'#FFFFFF'),(14903,43086,'#d01da7'),(14904,43087,'#FFFFFF'),(14905,43088,'#FFFFFF'),(14906,43089,'#FFFFFF'),(14907,43090,'#29faf7'),(14908,43091,'#15e59d'),(14909,43092,'#FFFFFF'),(14910,43093,'#FFFFFF'),(14911,43094,'#FFFFFF'),(14912,43095,'#FFFFFF'),(14913,43096,'#dc531a'),(14914,43097,'#FFFFFF'),(14915,43098,'#FFFFFF'),(14916,43099,'#d166fe'),(14917,43100,'#FFFFFF'),(14918,43101,'#FFFFFF'),(14919,43102,'#FFFFFF'),(14920,43103,'#064cfc'),(14921,43104,'#FFFFFF'),(14922,43105,'#2dfd2d'),(14923,43106,'#d70a65'),(14924,43107,'#FFFFFF'),(14925,43108,'#FFFFFF'),(14926,43109,'#FFFFFF'),(14927,43110,'#c6da33'),(14928,43111,'#14d823'),(14929,43112,'#8bf95d'),(14930,43114,'#FFFFFF'),(14931,43117,'#cc5ee5'),(14932,43118,'#fa7b48'),(14933,43119,'#FFFFFF'),(14934,43121,'#e46f2d'),(14935,43122,'#8d24cd'),(14936,43123,'#856fe8'),(14937,43124,'#FFFFFF'),(14938,43127,'#FFFFFF'),(14939,43128,'#FFFFFF'),(14940,43130,'#FFFFFF'),(14941,43131,'#FFFFFF'),(14942,43132,'#FFFFFF'),(14943,43133,'#0c55ec'),(14944,43134,'#c773f2'),(14945,43135,'#46d2d7'),(14946,43136,'#FFFFFF'),(14947,43137,'#11d35b'),(14948,43138,'#8361f0'),(14949,43140,'#75e758'),(14950,43142,'#e00298'),(14951,43144,'#FFFFFF'),(14952,43145,'#2efe19'),(14953,43146,'#FFFFFF'),(14954,43147,'#FFFFFF'),(14955,43148,'#4cd249'),(14956,43150,'#FFFFFF'),(14957,43152,'#FFFFFF'),(14958,43155,'#FFFFFF'),(14959,43156,'#FFFFFF'),(14960,43157,'#FFFFFF'),(14961,43158,'#FFFFFF'),(14962,43159,'#FFFFFF'),(14963,43160,'#eb34c0'),(14964,43161,'#FFFFFF'),(14965,43163,'#da5fe2'),(14966,43164,'#fae862'),(14967,43165,'#FFFFFF'),(14968,43166,'#FFFFFF'),(14969,43167,'#FFFFFF'),(14970,43168,'#FFFFFF'),(14971,43170,'#9441f9'),(14972,43171,'#FFFFFF'),(14973,43172,'#dbb713'),(14974,43173,'#FFFFFF'),(14975,43174,'#15b4ea'),(14976,43175,'#FFFFFF'),(14977,43176,'#FFFFFF'),(14978,43177,'#FFFFFF'),(14979,43178,'#FFFFFF'),(14980,43179,'#FFFFFF'),(14981,43180,'#FFFFFF'),(14982,43181,'#865ddc'),(14983,43182,'#FFFFFF'),(14984,43183,'#FFFFFF'),(14985,43184,'#FFFFFF'),(14986,43185,'#e6069c'),(14987,43186,'#FFFFFF'),(14988,43187,'#FFFFFF'),(14989,43188,'#1ef194'),(14990,43189,'#e80832'),(14991,43191,'#d8407e'),(14992,43192,'#FFFFFF'),(14993,43193,'#5737f4'),(14994,43194,'#FFFFFF'),(14995,43195,'#FFFFFF'),(14996,43196,'#FFFFFF'),(14997,43197,'#FFFFFF'),(14998,43198,'#f7590e'),(14999,43199,'#FFFFFF'),(15000,43201,'#FFFFFF'),(15001,43202,'#FFFFFF'),(15002,43203,'#ce6386'),(15003,43204,'#4a9be7'),(15004,43205,'#FFFFFF'),(15005,43206,'#8313d0'),(15006,43207,'#FFFFFF'),(15007,43208,'#68ebd6'),(15008,43209,'#FFFFFF'),(15009,43210,'#195ed4'),(15010,43211,'#b6f36c'),(15011,43212,'#FFFFFF'),(15012,43213,'#71f6b6'),(15013,43214,'#FFFFFF'),(15014,43215,'#FFFFFF'),(15015,43216,'#FFFFFF'),(15016,43217,'#FFFFFF'),(15017,43218,'#FFFFFF'),(15018,43219,'#FFFFFF'),(15019,43220,'#FFFFFF'),(15020,43221,'#FFFFFF'),(15021,43222,'#d715d9'),(15022,43223,'#4a3acc'),(15023,43225,'#f43dcd'),(15024,43227,'#b464cd'),(15025,43229,'#FFFFFF'),(15026,43230,'#FFFFFF'),(15027,43231,'#FFFFFF'),(15028,43232,'#FFFFFF'),(15029,43234,'#65d44e'),(15030,43235,'#FFFFFF'),(15031,43236,'#FFFFFF'),(15032,43237,'#FFFFFF'),(15033,43238,'#FFFFFF'),(15034,43239,'#FFFFFF'),(15035,43240,'#1c5ccf'),(15036,43241,'#FFFFFF'),(15037,43242,'#FFFFFF'),(15038,43245,'#fb8934'),(15039,43246,'#0083de'),(15040,43247,'#13f2ea'),(15041,43248,'#95d216'),(15042,43249,'#d87621'),(15043,43250,'#FFFFFF'),(15044,43251,'#FFFFFF'),(15045,43252,'#FFFFFF'),(15046,43253,'#FFFFFF'),(15047,43254,'#4fea95'),(15048,43255,'#FFFFFF'),(15049,43256,'#FFFFFF'),(15050,43257,'#FFFFFF'),(15051,43258,'#c13bf4'),(15052,43260,'#22f890'),(15053,43261,'#bf65de'),(15054,43262,'#FFFFFF'),(15055,43263,'#FFFFFF'),(15056,43264,'#FFFFFF'),(15057,43265,'#FFFFFF'),(15058,43267,'#fe6725'),(15059,43268,'#FFFFFF'),(15060,43269,'#FFFFFF'),(15061,43270,'#FFFFFF'),(15062,43271,'#FFFFFF'),(15063,43272,'#3ee92f'),(15064,43274,'#FFFFFF'),(15065,43276,'#3f4dde'),(15066,43278,'#FFFFFF'),(15067,43279,'#FFFFFF'),(15068,43280,'#FFFFFF'),(15069,43281,'#FFFFFF'),(15070,43282,'#e65690'),(15071,43283,'#FFFFFF'),(15072,43284,'#FFFFFF'),(15073,43285,'#FFFFFF'),(15074,43286,'#FFFFFF'),(15075,43287,'#FFFFFF'),(15076,43288,'#FFFFFF'),(15077,43290,'#0d79d0'),(15078,43291,'#FFFFFF'),(15079,43292,'#FFFFFF'),(15080,43293,'#FFFFFF'),(15081,43294,'#4ccff2'),(15082,43295,'#FFFFFF'),(15083,43296,'#FFFFFF'),(15084,43297,'#4a59d9'),(15085,43298,'#FFFFFF'),(15086,43299,'#FFFFFF'),(15087,43300,'#f4db37'),(15088,43301,'#da45c3'),(15089,43302,'#b421e2'),(15090,43303,'#FFFFFF'),(15091,43304,'#7a0aef'),(15092,43305,'#e02c9a'),(15093,43306,'#FFFFFF'),(15094,43308,'#FFFFFF'),(15095,43309,'#6ef287'),(15096,43310,'#ce258f'),(15097,43311,'#3cdbea'),(15098,43312,'#FFFFFF'),(15099,43313,'#e36f26'),(15100,43315,'#ea2aeb'),(15101,43316,'#FFFFFF'),(15102,43317,'#7629e4'),(15103,43318,'#FFFFFF'),(15104,43319,'#FFFFFF'),(15105,43321,'#FFFFFF'),(15106,43322,'#FFFFFF'),(15107,43324,'#FFFFFF'),(15108,43326,'#FFFFFF'),(15109,43327,'#f06c2d'),(15110,43328,'#FFFFFF'),(15111,43329,'#FFFFFF'),(15112,43330,'#a328ee'),(15113,43331,'#FFFFFF'),(15114,43333,'#86f220'),(15115,43334,'#e368e7'),(15116,43335,'#3762ed'),(15117,43338,'#FFFFFF'),(15118,43339,'#e2319f'),(15119,43340,'#2682fa'),(15120,43341,'#aad42c'),(15121,43342,'#FFFFFF'),(15122,43343,'#FFFFFF'),(15123,43344,'#6264f5'),(15124,43345,'#FFFFFF'),(15125,43346,'#FFFFFF'),(15126,43347,'#FFFFFF'),(15127,43349,'#FFFFFF'),(15128,43350,'#FFFFFF'),(15129,43351,'#FFFFFF'),(15130,43352,'#3276f9'),(15131,43353,'#00df40'),(15132,43354,'#FFFFFF'),(15133,43355,'#FFFFFF'),(15134,43356,'#FFFFFF'),(15135,43357,'#FFFFFF'),(15136,43358,'#d80432'),(15137,43359,'#65e64d'),(15138,43361,'#FFFFFF'),(15139,43362,'#FFFFFF'),(15140,43363,'#20cd94'),(15141,43364,'#FFFFFF'),(15142,43365,'#FFFFFF'),(15143,43366,'#FFFFFF'),(15144,43367,'#FFFFFF'),(15145,43368,'#f9002b'),(15146,43370,'#FFFFFF'),(15147,43371,'#FFFFFF'),(15148,43372,'#FFFFFF'),(15149,43373,'#FFFFFF'),(15150,43375,'#FFFFFF'),(15151,43376,'#FFFFFF'),(15152,43377,'#FFFFFF'),(15153,43378,'#FFFFFF'),(15154,43380,'#5c48de'),(15155,43381,'#FFFFFF'),(15156,43382,'#FFFFFF'),(15157,43383,'#d44488'),(15158,43385,'#FFFFFF'),(15159,43386,'#a143fa'),(15160,43387,'#d85d9f'),(15161,43388,'#FFFFFF'),(15162,43390,'#FFFFFF'),(15163,43391,'#FFFFFF'),(15164,43392,'#FFFFFF'),(15165,43393,'#FFFFFF'),(15166,43394,'#FFFFFF'),(15167,43395,'#FFFFFF'),(15168,43397,'#FFFFFF'),(15169,43398,'#FFFFFF'),(15170,43399,'#8c64f0'),(15171,43400,'#FFFFFF'),(15172,43401,'#cebb01'),(15173,43402,'#d6e04c'),(15174,43403,'#360ad2'),(15175,43404,'#FFFFFF'),(15176,43405,'#FFFFFF'),(15177,43407,'#FFFFFF'),(15178,43408,'#FFFFFF'),(15179,43409,'#FFFFFF'),(15180,43410,'#bbf14f'),(15181,43411,'#FFFFFF'),(15182,43412,'#3ff5bb'),(15183,43413,'#FFFFFF'),(15184,43415,'#2be7d0'),(15185,43416,'#FFFFFF'),(15186,43417,'#f668e1'),(15187,43418,'#3febe1'),(15188,43419,'#FFFFFF'),(15189,43420,'#3d97d2'),(15190,43422,'#FFFFFF'),(15191,43423,'#870ad3'),(15192,43424,'#79e95a'),(15193,43425,'#FFFFFF'),(15194,43426,'#FFFFFF'),(15195,43427,'#FFFFFF'),(15196,43428,'#FFFFFF'),(15197,43429,'#FFFFFF'),(15198,43430,'#FFFFFF'),(15199,43431,'#FFFFFF'),(15200,43432,'#daf507'),(15201,43433,'#FFFFFF'),(15202,43435,'#FFFFFF'),(15203,43436,'#FFFFFF'),(15204,43437,'#FFFFFF'),(15205,43438,'#FFFFFF'),(15206,43439,'#FFFFFF'),(15207,43440,'#3ef85e'),(15208,43443,'#FFFFFF'),(15209,43444,'#02a6d6'),(15210,43445,'#a1d83b'),(15211,43446,'#FFFFFF'),(15212,43447,'#FFFFFF'),(15213,43448,'#FFFFFF'),(15214,43449,'#FFFFFF'),(15215,43450,'#FFFFFF'),(15216,43451,'#77fb1d'),(15217,43453,'#FFFFFF'),(15218,43454,'#FFFFFF'),(15219,43455,'#27d762'),(15220,43456,'#64d688'),(15221,43457,'#FFFFFF'),(15222,43458,'#FFFFFF'),(15223,43459,'#FFFFFF'),(15224,43460,'#c20ddc'),(15225,43461,'#FFFFFF'),(15226,43463,'#FFFFFF'),(15227,43464,'#FFFFFF'),(15228,43465,'#FFFFFF'),(15229,43466,'#FFFFFF'),(15230,43467,'#2dcc1e'),(15231,43468,'#FFFFFF'),(15232,43469,'#4b69f6'),(15233,43470,'#FFFFFF'),(15234,43471,'#FFFFFF'),(15235,43473,'#FFFFFF'),(15236,43474,'#FFFFFF'),(15237,43475,'#FFFFFF'),(15238,43476,'#cf05ab'),(15239,43477,'#FFFFFF'),(15240,43479,'#24dd9b'),(15241,43480,'#FFFFFF'),(15242,43481,'#FFFFFF'),(15243,43482,'#FFFFFF'),(15244,43483,'#FFFFFF'),(15245,43484,'#FFFFFF'),(15246,43485,'#FFFFFF'),(15247,43486,'#FFFFFF'),(15248,43487,'#FFFFFF'),(15249,43488,'#FFFFFF'),(15250,43489,'#FFFFFF'),(15251,43490,'#FFFFFF'),(15252,43491,'#7845d3'),(15253,43492,'#FFFFFF'),(15254,43493,'#b0e914'),(15255,43495,'#FFFFFF'),(15256,43496,'#d762ce'),(15257,43497,'#FFFFFF'),(15258,43498,'#4c43f2'),(15259,43499,'#FFFFFF'),(15260,43500,'#FFFFFF'),(15261,43501,'#e34631'),(15262,43503,'#FFFFFF'),(15263,43504,'#FFFFFF'),(15264,43505,'#04acf6'),(15265,43507,'#FFFFFF'),(15266,43508,'#FFFFFF'),(15267,43509,'#FFFFFF'),(15268,43510,'#FFFFFF'),(15269,43511,'#FFFFFF'),(15270,43512,'#FFFFFF'),(15271,43514,'#FFFFFF'),(15272,43515,'#FFFFFF'),(15273,43516,'#5ee797'),(15274,43517,'#d229dc'),(15275,43518,'#FFFFFF'),(15276,43519,'#FFFFFF'),(15277,43520,'#7aea50'),(15278,43521,'#FFFFFF'),(15279,43522,'#FFFFFF'),(15280,43523,'#387be7'),(15281,43524,'#FFFFFF'),(15282,43525,'#FFFFFF'),(15283,43526,'#FFFFFF'),(15284,43527,'#7ef020'),(15285,43530,'#FFFFFF'),(15286,43531,'#FFFFFF'),(15287,43532,'#FFFFFF'),(15288,43533,'#FFFFFF'),(15289,43534,'#FFFFFF'),(15290,43535,'#FFFFFF'),(15291,43536,'#FFFFFF'),(15292,43537,'#FFFFFF'),(15293,43538,'#f8c018'),(15294,43539,'#FFFFFF'),(15295,43540,'#74c5f3'),(15296,43541,'#FFFFFF'),(15297,43542,'#FFFFFF'),(15298,43543,'#FFFFFF'),(15299,43544,'#FFFFFF'),(15300,43545,'#FFFFFF'),(15301,43546,'#FFFFFF'),(15302,43548,'#FFFFFF'),(15303,43549,'#FFFFFF'),(15304,43550,'#FFFFFF'),(15305,43551,'#FFFFFF'),(15306,43552,'#FFFFFF'),(15307,43553,'#FFFFFF'),(15308,43554,'#FFFFFF'),(15309,43555,'#FFFFFF'),(15310,43556,'#FFFFFF'),(15311,43557,'#9a31d9'),(15312,43558,'#ccb445'),(15313,43559,'#FFFFFF'),(15314,43564,'#FFFFFF'),(15315,43565,'#FFFFFF'),(15316,43566,'#477fea'),(15317,43567,'#FFFFFF'),(15318,43568,'#f41dc3'),(15319,43569,'#FFFFFF'),(15320,43570,'#FFFFFF'),(15321,43571,'#FFFFFF'),(15322,43572,'#cad927'),(15323,43573,'#FFFFFF'),(15324,43574,'#FFFFFF'),(15325,43575,'#41f913'),(15326,43576,'#FFFFFF'),(15327,43578,'#FFFFFF'),(15328,43579,'#d82b94'),(15329,43580,'#16cd8d'),(15330,43581,'#FFFFFF'),(15331,43582,'#FFFFFF'),(15332,43583,'#FFFFFF'),(15333,43585,'#FFFFFF'),(15334,43586,'#FFFFFF'),(15335,43587,'#FFFFFF'),(15336,43588,'#FFFFFF'),(15337,43589,'#FFFFFF'),(15338,43590,'#FFFFFF'),(15339,43591,'#d0b50b'),(15340,43592,'#FFFFFF'),(15341,43594,'#97f95f'),(15342,43596,'#FFFFFF'),(15343,43597,'#FFFFFF'),(15344,43598,'#FFFFFF'),(15345,43599,'#d20b46'),(15346,43600,'#FFFFFF'),(15347,43601,'#2f46df'),(15348,43602,'#FFFFFF'),(15349,43603,'#FFFFFF'),(15350,43604,'#FFFFFF'),(15351,43605,'#FFFFFF'),(15352,43606,'#FFFFFF'),(15353,43607,'#31fed8'),(15354,43608,'#6388d2'),(15355,43609,'#70d33b'),(15356,43610,'#e30c54'),(15357,43611,'#cd7a43'),(15358,43612,'#b1d124'),(15359,43613,'#43efec'),(15360,43614,'#FFFFFF'),(15361,43615,'#FFFFFF'),(15362,43616,'#FFFFFF'),(15363,43617,'#e1f622'),(15364,43618,'#c22dee'),(15365,43623,'#9bfa4a'),(15366,43624,'#1d64e8'),(15367,43625,'#FFFFFF'),(15368,43626,'#FFFFFF'),(15369,43627,'#FFFFFF'),(15370,43628,'#FFFFFF'),(15371,43629,'#FFFFFF'),(15372,43630,'#FFFFFF'),(15373,43631,'#FFFFFF'),(15374,43632,'#FFFFFF'),(15375,43633,'#FFFFFF'),(15376,43634,'#ca2be9'),(15377,43635,'#FFFFFF'),(15378,43636,'#d88556'),(15379,43638,'#FFFFFF'),(15380,43639,'#FFFFFF'),(15381,43640,'#89f767'),(15382,43641,'#FFFFFF'),(15383,43642,'#FFFFFF'),(15384,43643,'#FFFFFF'),(15385,43644,'#FFFFFF'),(15386,43645,'#43f21f'),(15387,43646,'#FFFFFF'),(15388,43647,'#FFFFFF'),(15389,43648,'#FFFFFF'),(15390,43649,'#58da6f'),(15391,43650,'#FFFFFF'),(15392,43651,'#FFFFFF'),(15393,43652,'#FFFFFF'),(15394,43653,'#FFFFFF'),(15395,43655,'#974aec'),(15396,43656,'#caf700'),(15397,43657,'#ec4784'),(15398,43659,'#FFFFFF'),(15399,43660,'#FFFFFF'),(15400,43661,'#d04579'),(15401,43662,'#FFFFFF'),(15402,43664,'#FFFFFF'),(15403,43665,'#cbef4f'),(15404,43666,'#FFFFFF'),(15405,43667,'#FFFFFF'),(15406,43669,'#0880d6'),(15407,43670,'#FFFFFF'),(15408,43671,'#FFFFFF'),(15409,43672,'#90e571'),(15410,43673,'#FFFFFF'),(15411,43675,'#d68705'),(15412,43676,'#FFFFFF'),(15413,43678,'#FFFFFF'),(15414,43679,'#FFFFFF'),(15415,43680,'#FFFFFF'),(15416,43681,'#FFFFFF'),(15417,43682,'#FFFFFF'),(15418,43683,'#FFFFFF'),(15419,43684,'#f88c3c'),(15420,43685,'#cff972'),(15421,43686,'#FFFFFF'),(15422,43687,'#FFFFFF'),(15423,43688,'#FFFFFF'),(15424,43689,'#FFFFFF'),(15425,43690,'#FFFFFF'),(15426,43691,'#FFFFFF'),(15427,43692,'#4268cc'),(15428,43693,'#FFFFFF'),(15429,43694,'#FFFFFF'),(15430,43695,'#FFFFFF'),(15431,43696,'#66fd6b'),(15432,43697,'#FFFFFF'),(15433,43698,'#FFFFFF'),(15434,43699,'#FFFFFF'),(15435,43700,'#FFFFFF'),(15436,43701,'#66da20'),(15437,43702,'#5a9dcc'),(15438,43703,'#FFFFFF'),(15439,43704,'#db40db'),(15440,43705,'#41a2f5'),(15441,43706,'#da0325'),(15442,43707,'#FFFFFF'),(15443,43709,'#FFFFFF'),(15444,43710,'#FFFFFF'),(15445,43711,'#65e075'),(15446,43712,'#1ede2f'),(15447,43713,'#e21119'),(15448,43715,'#badc34'),(15449,43716,'#FFFFFF'),(15450,43717,'#8441e1'),(15451,43718,'#FFFFFF'),(15452,43719,'#FFFFFF'),(15453,43720,'#FFFFFF'),(15454,43721,'#FFFFFF'),(15455,43722,'#FFFFFF'),(15456,43723,'#b476ed'),(15457,43724,'#9ae93e'),(15458,43725,'#19cdb7'),(15459,43726,'#FFFFFF'),(15460,43727,'#FFFFFF'),(15461,43728,'#bfcf65'),(15462,43729,'#FFFFFF'),(15463,43730,'#d8983a'),(15464,43731,'#FFFFFF'),(15465,43732,'#57ce5e'),(15466,43733,'#FFFFFF'),(15467,43734,'#ecde6e'),(15468,43735,'#e4091a'),(15469,43736,'#FFFFFF'),(15470,43738,'#6d87f5'),(15471,43739,'#1de237'),(15472,43740,'#FFFFFF'),(15473,43741,'#FFFFFF'),(15474,43743,'#f6294b'),(15475,43744,'#FFFFFF'),(15476,43745,'#FFFFFF'),(15477,43746,'#FFFFFF'),(15478,43747,'#FFFFFF'),(15479,43748,'#FFFFFF'),(15480,43749,'#FFFFFF'),(15481,43750,'#FFFFFF'),(15482,43751,'#FFFFFF'),(15483,43752,'#FFFFFF'),(15484,43753,'#FFFFFF'),(15485,43754,'#FFFFFF'),(15486,43755,'#FFFFFF'),(15487,43756,'#FFFFFF'),(15488,43757,'#f45f94'),(15489,43758,'#785ff9'),(15490,43759,'#FFFFFF'),(15491,43760,'#FFFFFF'),(15492,43761,'#FFFFFF'),(15493,43762,'#FFFFFF'),(15494,43763,'#e760cd'),(15495,43764,'#FFFFFF'),(15496,43765,'#FFFFFF'),(15497,43766,'#ce0058'),(15498,43767,'#FFFFFF'),(15499,43769,'#6b62df'),(15500,43770,'#FFFFFF'),(15501,43771,'#FFFFFF'),(15502,43773,'#FFFFFF'),(15503,43775,'#e046cd'),(15504,43777,'#FFFFFF'),(15505,43778,'#d88920'),(15506,43779,'#f0f223'),(15507,43781,'#eab368'),(15508,43782,'#FFFFFF'),(15509,43783,'#238ed5'),(15510,43784,'#FFFFFF'),(15511,43785,'#FFFFFF'),(15512,43786,'#FFFFFF'),(15513,43787,'#59b3f5'),(15514,43788,'#FFFFFF'),(15515,43789,'#FFFFFF'),(15516,43790,'#FFFFFF'),(15517,43791,'#FFFFFF'),(15518,43792,'#FFFFFF'),(15519,43793,'#767af1'),(15520,43794,'#FFFFFF'),(15521,43795,'#FFFFFF'),(15522,43796,'#FFFFFF'),(15523,43798,'#FFFFFF'),(15524,43799,'#FFFFFF'),(15525,43800,'#FFFFFF'),(15526,43801,'#FFFFFF'),(15527,43802,'#FFFFFF'),(15528,43803,'#FFFFFF'),(15529,43804,'#FFFFFF'),(15530,43805,'#FFFFFF'),(15531,43806,'#FFFFFF'),(15532,43807,'#48d408'),(15533,43808,'#37d270'),(15534,43809,'#FFFFFF'),(15535,43811,'#FFFFFF'),(15536,43812,'#FFFFFF'),(15537,43813,'#c42fee'),(15538,43815,'#FFFFFF'),(15539,43816,'#c340dc'),(15540,43817,'#FFFFFF'),(15541,43818,'#FFFFFF'),(15542,43819,'#FFFFFF'),(15543,43820,'#FFFFFF'),(15544,43822,'#e93418'),(15545,43823,'#f2f77a'),(15546,43824,'#FFFFFF'),(15547,43825,'#FFFFFF'),(15548,43826,'#66cc42'),(15549,43827,'#FFFFFF'),(15550,43828,'#0f8fe9'),(15551,43829,'#FFFFFF'),(15552,43830,'#FFFFFF'),(15553,43831,'#FFFFFF'),(15554,43832,'#FFFFFF'),(15555,43834,'#FFFFFF'),(15556,43835,'#FFFFFF'),(15557,43837,'#72f234'),(15558,43838,'#FFFFFF'),(15559,43840,'#FFFFFF'),(15560,43843,'#FFFFFF'),(15561,43844,'#FFFFFF'),(15562,43846,'#ef1e88'),(15563,43847,'#FFFFFF'),(15564,43848,'#FFFFFF'),(15565,43849,'#FFFFFF'),(15566,43850,'#FFFFFF'),(15567,43851,'#d0329b'),(15568,43852,'#FFFFFF'),(15569,43853,'#FFFFFF'),(15570,43854,'#FFFFFF'),(15571,43855,'#FFFFFF'),(15572,43857,'#FFFFFF'),(15573,43858,'#FFFFFF'),(15574,43859,'#FFFFFF'),(15575,43860,'#FFFFFF'),(15576,43862,'#d0cc20'),(15577,43863,'#FFFFFF'),(15578,43864,'#da6d44'),(15579,43865,'#cf2dc0'),(15580,43866,'#FFFFFF'),(15581,43868,'#FFFFFF'),(15582,43869,'#FFFFFF'),(15583,43870,'#FFFFFF'),(15584,43871,'#FFFFFF'),(15585,43872,'#FFFFFF'),(15586,43874,'#FFFFFF'),(15587,43875,'#FFFFFF'),(15588,43876,'#FFFFFF'),(15589,43877,'#FFFFFF'),(15590,43878,'#FFFFFF'),(15591,43879,'#FFFFFF'),(15592,43880,'#d55b66'),(15593,43881,'#FFFFFF'),(15594,43882,'#FFFFFF'),(15595,43883,'#FFFFFF'),(15596,43884,'#25fb2b'),(15597,43885,'#FFFFFF'),(15598,43886,'#FFFFFF'),(15599,43887,'#FFFFFF'),(15600,43888,'#FFFFFF'),(15601,43889,'#FFFFFF'),(15602,43890,'#FFFFFF'),(15603,43891,'#d35ebd'),(15604,43892,'#cd4d9b'),(15605,43893,'#18a8e7'),(15606,43894,'#FFFFFF'),(15607,43896,'#FFFFFF'),(15608,43897,'#FFFFFF'),(15609,43898,'#FFFFFF'),(15610,43899,'#FFFFFF'),(15611,43900,'#58d010'),(15612,43901,'#FFFFFF'),(15613,43902,'#451edf'),(15614,43904,'#FFFFFF'),(15615,43906,'#FFFFFF'),(15616,43907,'#5ed6e5'),(15617,43908,'#bfd014'),(15618,43909,'#FFFFFF'),(15619,43910,'#d1651b'),(15620,43911,'#FFFFFF'),(15621,43912,'#b909e3'),(15622,43913,'#FFFFFF'),(15623,43914,'#FFFFFF'),(15624,43915,'#FFFFFF'),(15625,43916,'#FFFFFF'),(15626,43917,'#FFFFFF'),(15627,43918,'#FFFFFF'),(15628,43919,'#FFFFFF'),(15629,43920,'#0be018'),(15630,43921,'#FFFFFF'),(15631,43922,'#58de1b'),(15632,43923,'#FFFFFF'),(15633,43924,'#f2882a'),(15634,43925,'#e64240'),(15635,43926,'#5fdee8'),(15636,43927,'#FFFFFF'),(15637,43928,'#3ae0aa'),(15638,43929,'#FFFFFF'),(15639,43930,'#29fbc8'),(15640,43931,'#c5e667'),(15641,43932,'#FFFFFF'),(15642,43933,'#FFFFFF'),(15643,43934,'#FFFFFF'),(15644,43935,'#FFFFFF'),(15645,43936,'#e6632f'),(15646,43937,'#FFFFFF'),(15647,43938,'#FFFFFF'),(15648,43939,'#ceda4c'),(15649,43940,'#b740fb'),(15650,43941,'#2ad2eb'),(15651,43942,'#FFFFFF'),(15652,43944,'#2ef12d'),(15653,43945,'#c3f850'),(15654,43946,'#ea66cf'),(15655,43947,'#FFFFFF'),(15656,43948,'#666de1'),(15657,43950,'#FFFFFF'),(15658,43951,'#dffa72'),(15659,43952,'#dc61b4'),(15660,43953,'#FFFFFF'),(15661,43954,'#e949f3'),(15662,43955,'#FFFFFF'),(15663,43956,'#FFFFFF'),(15664,43957,'#fe5cc5'),(15665,43958,'#FFFFFF'),(15666,43959,'#FFFFFF'),(15667,43960,'#a917f6'),(15668,43961,'#0eaaed'),(15669,43962,'#FFFFFF'),(15670,43963,'#FFFFFF'),(15671,43964,'#FFFFFF'),(15672,43965,'#FFFFFF'),(15673,43966,'#FFFFFF'),(15674,43967,'#FFFFFF'),(15675,43968,'#f4f21e'),(15676,43969,'#FFFFFF'),(15677,43971,'#FFFFFF'),(15678,43972,'#FFFFFF'),(15679,43973,'#FFFFFF'),(15680,43974,'#FFFFFF'),(15681,43975,'#FFFFFF'),(15682,43976,'#FFFFFF'),(15683,43978,'#b434d0'),(15684,43979,'#e34a81'),(15685,43980,'#7542cd'),(15686,43981,'#FFFFFF'),(15687,43982,'#FFFFFF'),(15688,43983,'#1aed49'),(15689,43984,'#FFFFFF'),(15690,43985,'#62d298'),(15691,43986,'#FFFFFF'),(15692,43987,'#FFFFFF'),(15693,43988,'#e0f40b'),(15694,43989,'#FFFFFF'),(15695,43990,'#FFFFFF'),(15696,43991,'#FFFFFF'),(15697,43992,'#FFFFFF'),(15698,43994,'#FFFFFF'),(15699,43995,'#FFFFFF'),(15700,43996,'#FFFFFF'),(15701,43998,'#49e987'),(15702,44000,'#FFFFFF'),(15703,44001,'#FFFFFF'),(15704,44002,'#FFFFFF'),(15705,44003,'#FFFFFF'),(15706,44004,'#6955ef'),(15707,44005,'#d78e4b'),(15708,44006,'#FFFFFF'),(15709,44007,'#d7c22f'),(15710,44008,'#55cdab'),(15711,44009,'#FFFFFF'),(15712,44010,'#19f7a0'),(15713,44011,'#FFFFFF'),(15714,44012,'#fe50cc'),(15715,44013,'#FFFFFF'),(15716,44014,'#FFFFFF'),(15717,44015,'#FFFFFF'),(15718,44016,'#FFFFFF'),(15719,44017,'#FFFFFF'),(15720,44019,'#3e37db'),(15721,44020,'#4de04e'),(15722,44021,'#FFFFFF'),(15723,44022,'#FFFFFF'),(15724,44023,'#FFFFFF'),(15725,44024,'#FFFFFF'),(15726,44026,'#FFFFFF'),(15727,44027,'#895dfa'),(15728,44028,'#FFFFFF'),(15729,44029,'#FFFFFF'),(15730,44030,'#FFFFFF'),(15731,44031,'#debc41'),(15732,44032,'#e04365'),(15733,44033,'#FFFFFF'),(15734,44034,'#FFFFFF'),(15735,44036,'#FFFFFF'),(15736,44037,'#FFFFFF'),(15737,44039,'#FFFFFF'),(15738,44040,'#FFFFFF'),(15739,44041,'#4a9df9'),(15740,44042,'#FFFFFF'),(15741,44043,'#FFFFFF'),(15742,44044,'#FFFFFF'),(15743,44045,'#3d40eb'),(15744,44046,'#FFFFFF'),(15745,44047,'#ee1a7a'),(15746,44049,'#FFFFFF'),(15747,44050,'#ea146d'),(15748,44051,'#FFFFFF'),(15749,44052,'#FFFFFF'),(15750,44053,'#FFFFFF'),(15751,44054,'#FFFFFF'),(15752,44055,'#FFFFFF'),(15753,44056,'#e44e24'),(15754,44057,'#FFFFFF'),(15755,44059,'#c44fe7'),(15756,44060,'#FFFFFF'),(15757,44061,'#FFFFFF'),(15758,44062,'#FFFFFF'),(15759,44063,'#cca80b'),(15760,44064,'#db3569'),(15761,44065,'#FFFFFF'),(15762,44066,'#e1246a'),(15763,44067,'#db1aa7'),(15764,44069,'#FFFFFF'),(15765,44070,'#FFFFFF'),(15766,44071,'#FFFFFF'),(15767,44072,'#FFFFFF'),(15768,44073,'#FFFFFF'),(15769,44074,'#FFFFFF'),(15770,44075,'#FFFFFF'),(15771,44076,'#37dcda'),(15772,44077,'#7360cf'),(15773,44078,'#FFFFFF'),(15774,44079,'#eada08'),(15775,44080,'#FFFFFF'),(15776,44081,'#86fa20'),(15777,44082,'#FFFFFF'),(15778,44083,'#FFFFFF'),(15779,44085,'#1fec03'),(15780,44086,'#FFFFFF'),(15781,44087,'#FFFFFF'),(15782,44088,'#FFFFFF'),(15783,44089,'#FFFFFF'),(15784,44090,'#FFFFFF'),(15785,44091,'#FFFFFF'),(15786,44092,'#FFFFFF'),(15787,44093,'#FFFFFF'),(15788,44095,'#FFFFFF'),(15789,44096,'#FFFFFF'),(15790,44097,'#FFFFFF'),(15791,44098,'#FFFFFF'),(15792,44099,'#cc0b3b'),(15793,44100,'#361eef'),(15794,44101,'#FFFFFF'),(15795,44102,'#e6c163'),(15796,44103,'#5989f8'),(15797,44104,'#8008f2'),(15798,44105,'#FFFFFF'),(15799,44106,'#FFFFFF'),(15800,44107,'#FFFFFF'),(15801,44108,'#d4dc47'),(15802,44109,'#ce57c2'),(15803,44110,'#FFFFFF'),(15804,44111,'#FFFFFF'),(15805,44112,'#FFFFFF'),(15806,44114,'#70e886'),(15807,44116,'#d03570'),(15808,44117,'#FFFFFF'),(15809,44118,'#FFFFFF'),(15810,44119,'#FFFFFF'),(15811,44120,'#FFFFFF'),(15812,44121,'#28deac'),(15813,44122,'#FFFFFF'),(15814,44123,'#FFFFFF'),(15815,44124,'#FFFFFF'),(15816,44125,'#FFFFFF'),(15817,44126,'#FFFFFF'),(15818,44127,'#a421f4'),(15819,44128,'#FFFFFF'),(15820,44129,'#fb927d'),(15821,44130,'#86f726'),(15822,44131,'#FFFFFF'),(15823,44132,'#0419dd'),(15824,44133,'#c2fb6c'),(15825,44134,'#29e5e0'),(15826,44135,'#FFFFFF'),(15827,44136,'#FFFFFF'),(15828,44137,'#FFFFFF'),(15829,44138,'#adec25'),(15830,44139,'#04ce1f'),(15831,44140,'#FFFFFF'),(15832,44141,'#FFFFFF'),(15833,44142,'#df5ed1'),(15834,44143,'#6b95ea'),(15835,44144,'#f3b96d'),(15836,44145,'#fdb15b'),(15837,44146,'#FFFFFF'),(15838,44148,'#FFFFFF'),(15839,44149,'#8752f6'),(15840,44150,'#FFFFFF'),(15841,44151,'#e63107'),(15842,44152,'#FFFFFF'),(15843,44153,'#75f8e8'),(15844,44154,'#2bfc4f'),(15845,44155,'#FFFFFF'),(15846,44156,'#FFFFFF'),(15847,44157,'#FFFFFF'),(15848,44158,'#FFFFFF'),(15849,44159,'#FFFFFF'),(15850,44160,'#FFFFFF'),(15851,44161,'#FFFFFF'),(15852,44162,'#FFFFFF'),(15853,44163,'#FFFFFF'),(15854,44164,'#FFFFFF'),(15855,44165,'#FFFFFF'),(15856,44166,'#0cdf85'),(15857,44167,'#f622b6'),(15858,44169,'#b444cc'),(15859,44170,'#FFFFFF'),(15860,44171,'#114ed9'),(15861,44172,'#c82ee7'),(15862,44173,'#FFFFFF'),(15863,44174,'#FFFFFF'),(15864,44175,'#FFFFFF'),(15865,44176,'#FFFFFF'),(15866,44177,'#FFFFFF'),(15867,44178,'#FFFFFF'),(15868,44179,'#FFFFFF'),(15869,44180,'#FFFFFF'),(15870,44181,'#FFFFFF'),(15871,44182,'#23e7b3'),(15872,44183,'#a605df'),(15873,44184,'#13cf70'),(15874,44185,'#e08029'),(15875,44186,'#3bdcce'),(15876,44187,'#FFFFFF'),(15877,44188,'#FFFFFF'),(15878,44189,'#FFFFFF'),(15879,44190,'#FFFFFF'),(15880,44191,'#FFFFFF'),(15881,44192,'#f2044c'),(15882,44193,'#FFFFFF'),(15883,44194,'#FFFFFF'),(15884,44195,'#FFFFFF'),(15885,44196,'#FFFFFF'),(15886,44197,'#FFFFFF'),(15887,44198,'#7b63f7'),(15888,44199,'#f355d1'),(15889,44200,'#FFFFFF'),(15890,44201,'#4c7ff3'),(15891,44204,'#FFFFFF'),(15892,44205,'#FFFFFF'),(15893,44206,'#38cf80'),(15894,44207,'#2d49ec'),(15895,44210,'#FFFFFF'),(15896,44211,'#FFFFFF'),(15897,44212,'#FFFFFF'),(15898,44213,'#FFFFFF'),(15899,44214,'#10a6ef'),(15900,44215,'#FFFFFF'),(15901,44216,'#FFFFFF'),(15902,44217,'#f367cc'),(15903,44218,'#FFFFFF'),(15904,44219,'#FFFFFF'),(15905,44220,'#FFFFFF'),(15906,44221,'#d89254'),(15907,44222,'#FFFFFF'),(15908,44223,'#FFFFFF'),(15909,44224,'#FFFFFF'),(15910,44225,'#4145f4'),(15911,44226,'#FFFFFF'),(15912,44227,'#f56ade'),(15913,44228,'#FFFFFF'),(15914,44229,'#FFFFFF'),(15915,44230,'#FFFFFF'),(15916,44231,'#bed804'),(15917,44232,'#FFFFFF'),(15918,44233,'#FFFFFF'),(15919,44234,'#FFFFFF'),(15920,44235,'#FFFFFF'),(15921,44236,'#FFFFFF'),(15922,44238,'#55fc91'),(15923,44240,'#3cd724'),(15924,44241,'#9edf32'),(15925,44242,'#FFFFFF'),(15926,44243,'#6bec7f'),(15927,44244,'#FFFFFF'),(15928,44245,'#ae49ea'),(15929,44246,'#FFFFFF'),(15930,44247,'#FFFFFF'),(15931,44248,'#FFFFFF'),(15932,44249,'#fc464b'),(15933,44250,'#FFFFFF'),(15934,44251,'#FFFFFF'),(15935,44252,'#10f8f7'),(15936,44253,'#f11050'),(15937,44254,'#e7ea6c'),(15938,44255,'#FFFFFF'),(15939,44257,'#FFFFFF'),(15940,44258,'#FFFFFF'),(15941,44259,'#252eec'),(15942,44260,'#FFFFFF'),(15943,44261,'#FFFFFF'),(15944,44262,'#FFFFFF'),(15945,44263,'#FFFFFF'),(15946,44264,'#02d401'),(15947,44265,'#516ae6'),(15948,44266,'#FFFFFF'),(15949,44267,'#FFFFFF'),(15950,44268,'#FFFFFF'),(15951,44269,'#FFFFFF'),(15952,44270,'#FFFFFF'),(15953,44272,'#d59f37'),(15954,44273,'#FFFFFF'),(15955,44274,'#FFFFFF'),(15956,44275,'#FFFFFF'),(15957,44276,'#FFFFFF'),(15958,44277,'#e9866c'),(15959,44279,'#FFFFFF'),(15960,44280,'#FFFFFF'),(15961,44281,'#FFFFFF'),(15962,44282,'#FFFFFF'),(15963,44283,'#FFFFFF'),(15964,44284,'#6989fc'),(15965,44285,'#FFFFFF'),(15966,44286,'#FFFFFF'),(15967,44287,'#FFFFFF'),(15968,44288,'#3aa1ef'),(15969,44289,'#FFFFFF'),(15970,44290,'#FFFFFF'),(15971,44291,'#15b6f6'),(15972,44292,'#90d252'),(15973,44293,'#55e13f'),(15974,44294,'#FFFFFF'),(15975,44296,'#FFFFFF'),(15976,44297,'#FFFFFF'),(15977,44298,'#FFFFFF'),(15978,44299,'#64e833'),(15979,44300,'#d6e952'),(15980,44301,'#1628e9'),(15981,44302,'#FFFFFF'),(15982,44303,'#FFFFFF'),(15983,44304,'#FFFFFF'),(15984,44305,'#cd6b2d'),(15985,44306,'#FFFFFF'),(15986,44307,'#fe0993'),(15987,44308,'#85e10b'),(15988,44309,'#52cf5c'),(15989,44310,'#a134db'),(15990,44311,'#FFFFFF'),(15991,44312,'#FFFFFF'),(15992,44313,'#FFFFFF'),(15993,44314,'#FFFFFF'),(15994,44316,'#FFFFFF'),(15995,44317,'#cd1b2b'),(15996,44318,'#FFFFFF'),(15997,44319,'#FFFFFF'),(15998,44320,'#FFFFFF'),(15999,44321,'#FFFFFF'),(16000,44322,'#eefb5a'),(16001,44323,'#f457a9'),(16002,44324,'#FFFFFF'),(16003,44325,'#FFFFFF'),(16004,44326,'#FFFFFF'),(16005,44327,'#FFFFFF'),(16006,44328,'#FFFFFF'),(16007,44329,'#1519df'),(16008,44330,'#FFFFFF'),(16009,44331,'#f14e4a'),(16010,44332,'#FFFFFF'),(16011,44333,'#FFFFFF'),(16012,44334,'#5d2cfc'),(16013,44336,'#ed58e6'),(16014,44337,'#FFFFFF'),(16015,44339,'#dcdc4e'),(16016,44340,'#cd61c1'),(16017,44341,'#FFFFFF'),(16018,44342,'#5bde11'),(16019,44343,'#FFFFFF'),(16020,44344,'#5c89e6'),(16021,44345,'#FFFFFF'),(16022,44346,'#cc481b'),(16023,44347,'#FFFFFF'),(16024,44348,'#FFFFFF'),(16025,44349,'#FFFFFF'),(16026,44350,'#FFFFFF'),(16027,44351,'#FFFFFF'),(16028,44352,'#FFFFFF'),(16029,44353,'#FFFFFF'),(16030,44354,'#FFFFFF'),(16031,44355,'#FFFFFF'),(16032,44356,'#3b0dfd'),(16033,44357,'#FFFFFF'),(16034,44358,'#56c9d2'),(16035,44359,'#FFFFFF'),(16036,44360,'#d94673'),(16037,44361,'#FFFFFF'),(16038,44362,'#FFFFFF'),(16039,44363,'#FFFFFF'),(16040,44364,'#FFFFFF'),(16041,44365,'#FFFFFF'),(16042,44366,'#FFFFFF'),(16043,44367,'#FFFFFF'),(16044,44368,'#FFFFFF'),(16045,44369,'#FFFFFF'),(16046,44370,'#FFFFFF'),(16047,44371,'#FFFFFF'),(16048,44372,'#11f5d0'),(16049,44373,'#FFFFFF'),(16050,44374,'#FFFFFF'),(16051,44377,'#d46254'),(16052,44378,'#FFFFFF'),(16053,44379,'#d05f5f'),(16054,44380,'#FFFFFF'),(16055,44381,'#4d13d8'),(16056,44382,'#46d821'),(16057,44383,'#FFFFFF'),(16058,44384,'#FFFFFF'),(16059,44386,'#FFFFFF'),(16060,44387,'#FFFFFF'),(16061,44388,'#FFFFFF'),(16062,44389,'#FFFFFF'),(16063,44390,'#FFFFFF'),(16064,44391,'#FFFFFF'),(16065,44392,'#d09165'),(16066,44393,'#FFFFFF'),(16067,44394,'#1b2bdc'),(16068,44395,'#FFFFFF'),(16069,44396,'#f9080e'),(16070,44397,'#FFFFFF'),(16071,44398,'#FFFFFF'),(16072,44399,'#FFFFFF'),(16073,44400,'#FFFFFF'),(16074,44401,'#4717cc'),(16075,44402,'#bcd064'),(16076,44403,'#FFFFFF'),(16077,44405,'#9a0ddd'),(16078,44407,'#18e324'),(16079,44408,'#FFFFFF'),(16080,44409,'#FFFFFF'),(16081,44410,'#FFFFFF'),(16082,44411,'#FFFFFF'),(16083,44412,'#FFFFFF'),(16084,44413,'#FFFFFF'),(16085,44414,'#FFFFFF'),(16086,44415,'#FFFFFF'),(16087,44417,'#FFFFFF'),(16088,44418,'#FFFFFF'),(16089,44419,'#FFFFFF'),(16090,44420,'#fbc74c'),(16091,44421,'#FFFFFF'),(16092,44422,'#FFFFFF'),(16093,44423,'#e3921e'),(16094,44424,'#FFFFFF'),(16095,44425,'#2af06e'),(16096,44426,'#FFFFFF'),(16097,44427,'#FFFFFF'),(16098,44428,'#FFFFFF'),(16099,44429,'#aed610'),(16100,44431,'#FFFFFF'),(16101,44432,'#5f0ffb'),(16102,44433,'#FFFFFF'),(16103,44435,'#03d2d6'),(16104,44436,'#FFFFFF'),(16105,44437,'#e82b85'),(16106,44438,'#FFFFFF'),(16107,44439,'#FFFFFF'),(16108,44441,'#FFFFFF'),(16109,44442,'#FFFFFF'),(16110,44443,'#24edc4'),(16111,44444,'#FFFFFF'),(16112,44445,'#b003e3'),(16113,44446,'#6afcf4'),(16114,44447,'#FFFFFF'),(16115,44448,'#a217ec'),(16116,44449,'#FFFFFF'),(16117,44450,'#FFFFFF'),(16118,44452,'#FFFFFF'),(16119,44453,'#FFFFFF'),(16120,44455,'#FFFFFF'),(16121,44456,'#FFFFFF'),(16122,44457,'#FFFFFF'),(16123,44459,'#FFFFFF'),(16124,44460,'#FFFFFF'),(16125,44461,'#dde736'),(16126,44462,'#FFFFFF'),(16127,44463,'#FFFFFF'),(16128,44465,'#FFFFFF'),(16129,44466,'#34f9a7'),(16130,44468,'#FFFFFF'),(16131,44469,'#b849d4'),(16132,44470,'#FFFFFF'),(16133,44471,'#FFFFFF'),(16134,44474,'#FFFFFF'),(16135,44475,'#FFFFFF'),(16136,44476,'#FFFFFF'),(16137,44477,'#FFFFFF'),(16138,44479,'#FFFFFF'),(16139,44480,'#FFFFFF'),(16140,44481,'#FFFFFF'),(16141,44482,'#FFFFFF'),(16142,44483,'#FFFFFF'),(16143,44484,'#FFFFFF'),(16144,44485,'#FFFFFF'),(16145,44487,'#FFFFFF'),(16146,44488,'#FFFFFF'),(16147,44489,'#FFFFFF'),(16148,44490,'#FFFFFF'),(16149,44493,'#FFFFFF'),(16150,44494,'#25d87c'),(16151,44495,'#FFFFFF'),(16152,44496,'#FFFFFF'),(16153,44497,'#FFFFFF'),(16154,44498,'#e05e2c'),(16155,44499,'#FFFFFF'),(16156,44500,'#FFFFFF'),(16157,44501,'#cfd61b'),(16158,44503,'#04d2cf'),(16159,44505,'#d76019'),(16160,44506,'#FFFFFF'),(16161,44509,'#31ddda'),(16162,44510,'#FFFFFF'),(16163,44511,'#ed548a'),(16164,44512,'#eb6c24'),(16165,44514,'#FFFFFF'),(16166,44515,'#FFFFFF'),(16167,44518,'#FFFFFF'),(16168,44520,'#03eae2'),(16169,44521,'#FFFFFF'),(16170,44522,'#FFFFFF'),(16171,44524,'#FFFFFF'),(16172,44525,'#FFFFFF'),(16173,44526,'#FFFFFF'),(16174,44527,'#FFFFFF'),(16175,44528,'#af51e2'),(16176,44529,'#FFFFFF'),(16177,44530,'#d6d239'),(16178,44531,'#FFFFFF'),(16179,44533,'#d8aa5f'),(16180,44534,'#d83d87'),(16181,44535,'#FFFFFF'),(16182,44537,'#FFFFFF'),(16183,44539,'#FFFFFF'),(16184,44540,'#FFFFFF'),(16185,44542,'#FFFFFF'),(16186,44543,'#da45b1'),(16187,44544,'#db629e'),(16188,44546,'#FFFFFF'),(16189,44549,'#FFFFFF'),(16190,44552,'#FFFFFF'),(16191,44555,'#FFFFFF'),(16192,44556,'#FFFFFF'),(16193,44557,'#FFFFFF'),(16194,44560,'#cea567'),(16195,44561,'#FFFFFF'),(16196,44562,'#FFFFFF'),(16197,44564,'#a6d106'),(16198,44566,'#1ee98d'),(16199,44567,'#783cd5'),(16200,44568,'#6af788'),(16201,44569,'#8056cd'),(16202,44571,'#3a6deb'),(16203,44572,'#FFFFFF'),(16204,44575,'#FFFFFF'),(16205,44576,'#6f4ccd'),(16206,44577,'#FFFFFF'),(16207,44579,'#FFFFFF'),(16208,44580,'#FFFFFF'),(16209,44581,'#FFFFFF'),(16210,44583,'#3547e7'),(16211,44586,'#FFFFFF'),(16212,44588,'#FFFFFF'),(16213,44589,'#FFFFFF'),(16214,44591,'#FFFFFF'),(16215,44592,'#4039dc'),(16216,44593,'#6beee2'),(16217,44596,'#FFFFFF'),(16218,44597,'#FFFFFF'),(16219,44598,'#5c41ea'),(16220,44599,'#FFFFFF'),(16221,44600,'#FFFFFF'),(16222,44601,'#FFFFFF'),(16223,44602,'#FFFFFF'),(16224,44603,'#f947f7'),(16225,44605,'#FFFFFF'),(16226,44608,'#FFFFFF'),(16227,44610,'#FFFFFF'),(16228,44611,'#FFFFFF'),(16229,44613,'#FFFFFF'),(16230,44615,'#FFFFFF'),(16231,44617,'#FFFFFF'),(16232,44618,'#cb31f4'),(16233,44620,'#bb45f8'),(16234,44621,'#FFFFFF'),(16235,44622,'#FFFFFF'),(16236,44623,'#FFFFFF'),(16237,44624,'#FFFFFF'),(16238,44626,'#5580d5'),(16239,44628,'#FFFFFF'),(16240,44630,'#FFFFFF'),(16241,44632,'#FFFFFF'),(16242,44634,'#FFFFFF'),(16243,44635,'#FFFFFF'),(16244,44731,'#e2d93e'),(16245,44852,'#4ee05d'),(16246,44857,'#e8de4e'),(16247,44858,'#6962f3'),(16248,44886,'#6396d9'),(16249,44890,'#ef4394'),(16250,44894,'#824fec'),(16251,44900,'#1cb7fc'),(16252,44932,'#cfed58'),(16253,44938,'#FFFFFF'),(16254,44939,'#FFFFFF'),(16255,44945,'#FFFFFF'),(16256,44946,'#28a6ce'),(16257,44949,'#FFFFFF'),(16258,44954,'#FFFFFF'),(16259,44976,'#FFFFFF'),(16260,44978,'#FFFFFF'),(16261,44986,'#FFFFFF'),(16262,44996,'#FFFFFF'),(16263,44997,'#FFFFFF'),(16264,45001,'#FFFFFF'),(16265,45010,'#FFFFFF'),(16266,45015,'#2bf11e'),(16267,45020,'#08d430'),(16268,45025,'#FFFFFF'),(16269,45033,'#FFFFFF'),(16270,45034,'#FFFFFF'),(16271,45037,'#FFFFFF'),(16272,45042,'#FFFFFF'),(16273,45052,'#FFFFFF'),(16274,45059,'#FFFFFF'),(16275,45060,'#FFFFFF'),(16276,45062,'#FFFFFF'),(16277,45071,'#FFFFFF'),(16278,45082,'#FFFFFF'),(16279,45087,'#8a44d2'),(16280,45091,'#FFFFFF'),(16281,45094,'#76fd0a'),(16282,45098,'#4b2df6'),(16283,45099,'#69fe53'),(16284,45110,'#eb35df'),(16285,45128,'#c256df'),(16286,45132,'#18ceb5'),(16287,45139,'#2a8be2'),(16288,45145,'#43e615'),(16289,45169,'#44caee'),(16290,45171,'#77fa81'),(16291,45177,'#6ddbe9'),(16292,45187,'#54e5c3'),(16293,45210,'#FFFFFF'),(16294,45214,'#d96667'),(16295,45216,'#fae00e'),(16296,45223,'#5699ef'),(16297,45239,'#6289f9'),(16298,45246,'#f46ad6'),(16299,45247,'#dc09bf'),(16300,45258,'#58f7b6'),(16301,45260,'#9561ee'),(16302,45267,'#795dce'),(16303,45273,'#f979bd'),(16304,45275,'#0cf9f3'),(16305,45280,'#FFFFFF'),(16306,45282,'#e3da29'),(16307,45287,'#54d31f'),(16308,45288,'#FFFFFF'),(16309,45289,'#f8734a'),(16310,45302,'#64daab'),(16311,45307,'#FFFFFF'),(16312,45336,'#FFFFFF'),(16313,45369,'#e10520'),(16314,45377,'#49e4a7'),(16315,45389,'#50f8fc'),(16316,45390,'#10d666'),(16317,45417,'#dd0b99'),(16318,45421,'#b6ec67'),(16319,45429,'#0af0a4'),(16320,45434,'#48e079'),(16321,45443,'#FFFFFF'),(16322,45445,'#3f0ced'),(16323,45474,'#5ccd8f'),(16324,45478,'#90dc5c'),(16325,45503,'#a028db'),(16326,45521,'#edb70d'),(16327,45527,'#662bcc'),(16328,45537,'#e95a4c'),(16329,45540,'#1cb7fa'),(16330,45541,'#ed47af'),(16331,45548,'#6c64e9'),(16332,45576,'#de6bef'),(16333,45583,'#f09b36'),(16334,45592,'#47cf10'),(16335,45609,'#fc3c47'),(16336,45619,'#d0358e'),(16337,45624,'#d169d6'),(16338,45634,'#FFFFFF'),(16339,45635,'#9534de'),(16340,45637,'#786ee4'),(16341,45643,'#FFFFFF'),(16342,45657,'#b9ee36'),(16343,45670,'#5cde0b'),(16344,45703,'#9163e0'),(16345,45708,'#77e841'),(16346,45723,'#cd3292'),(16347,45732,'#f10b59'),(16348,45758,'#f69d3a'),(16349,45763,'#dfa43a'),(16350,45768,'#FFFFFF'),(16351,45773,'#FFFFFF'),(16352,45777,'#d6995e'),(16353,45778,'#FFFFFF'),(16354,45780,'#FFFFFF'),(16355,45785,'#FFFFFF'),(16356,45787,'#e61f82'),(16357,45790,'#FFFFFF'),(16358,45794,'#71ea40'),(16359,45796,'#FFFFFF'),(16360,45803,'#FFFFFF'),(16361,45806,'#36c8e2'),(16362,45809,'#40f2f2'),(16363,45817,'#FFFFFF'),(16364,45824,'#FFFFFF'),(16365,45825,'#fd78df'),(16366,45838,'#FFFFFF'),(16367,45848,'#d59563'),(16368,45857,'#d7932b'),(16369,45858,'#FFFFFF'),(16370,45860,'#FFFFFF'),(16371,45865,'#FFFFFF'),(16372,45868,'#ec2e34'),(16373,45873,'#820de0'),(16374,45880,'#FFFFFF'),(16375,45884,'#65b6fc'),(16376,45892,'#76df5a'),(16377,45910,'#1bf9d8'),(16378,45927,'#FFFFFF'),(16379,45943,'#07f886'),(16380,45953,'#FFFFFF'),(16381,45958,'#594af0'),(16382,45984,'#d00c95'),(16383,45990,'#e95d5a'),(16384,45995,'#4cd0aa'),(16385,45996,'#46c1f8'),(16386,46000,'#ee8b71'),(16387,46007,'#07d938'),(16388,46057,'#7bee1d'),(16389,46065,'#b7da6b'),(16390,46083,'#dc12f0'),(16391,46133,'#f2ed3e'),(16392,46134,'#5de567'),(16393,46208,'#38de27'),(16394,46321,'#6ee0c7'),(16395,46352,'#6b25fe'),(16396,46374,'#8042de'),(16397,46450,'#fb8007'),(16398,46452,'#e84322'),(16399,46465,'#FFFFFF'),(16400,46466,'#FFFFFF'),(16401,46467,'#fa5a72'),(16402,46468,'#fdd12f'),(16403,46469,'#FFFFFF'),(16404,46471,'#cd096b'),(16405,46472,'#4e4cd4'),(16406,46473,'#cd2ac8'),(16407,46474,'#51e732'),(16408,46475,'#FFFFFF'),(16409,46478,'#fb25db'),(16410,46479,'#FFFFFF'),(16411,46481,'#8b71f2'),(16412,46482,'#FFFFFF'),(16413,46483,'#ec4303'),(16414,46484,'#3aaaf4'),(16415,46485,'#FFFFFF'),(16416,46487,'#FFFFFF'),(16417,46488,'#1562f0'),(16418,46489,'#FFFFFF'),(16419,46490,'#FFFFFF'),(16420,46491,'#cf1a7b'),(16421,46492,'#FFFFFF'),(16422,46493,'#4ae080'),(16423,46494,'#FFFFFF'),(16424,46495,'#f0a452'),(16425,46496,'#FFFFFF'),(16426,46497,'#d26791'),(16427,46498,'#FFFFFF'),(16428,46499,'#fe3cd4'),(16429,46500,'#a4e759'),(16430,46502,'#4cd967'),(16431,46503,'#FFFFFF'),(16432,46504,'#FFFFFF'),(16433,46505,'#FFFFFF'),(16434,46506,'#FFFFFF'),(16435,46507,'#4ed344'),(16436,46508,'#FFFFFF'),(16437,46510,'#9548df'),(16438,46511,'#FFFFFF'),(16439,46512,'#FFFFFF'),(16440,46513,'#FFFFFF'),(16441,46514,'#e13606'),(16442,46515,'#FFFFFF'),(16443,46516,'#FFFFFF'),(16444,46517,'#FFFFFF'),(16445,46519,'#FFFFFF'),(16446,46520,'#FFFFFF'),(16447,46524,'#FFFFFF'),(16448,46525,'#6349e2'),(16449,46526,'#FFFFFF'),(16450,46527,'#FFFFFF'),(16451,46528,'#59ccf9'),(16452,46529,'#FFFFFF'),(16453,46531,'#FFFFFF'),(16454,46532,'#54d28f'),(16455,46533,'#30c0e5'),(16456,46534,'#FFFFFF'),(16457,46535,'#FFFFFF'),(16458,46536,'#FFFFFF'),(16459,46537,'#FFFFFF'),(16460,46538,'#e73b7c'),(16461,46539,'#ef5dd5'),(16462,46540,'#FFFFFF'),(16463,46541,'#FFFFFF'),(16464,46542,'#FFFFFF'),(16465,46543,'#FFFFFF'),(16466,46544,'#FFFFFF'),(16467,46545,'#db3b71'),(16468,46546,'#FFFFFF'),(16469,46547,'#FFFFFF'),(16470,46548,'#44cee6'),(16471,46549,'#3404d0'),(16472,46551,'#20f12e'),(16473,46554,'#FFFFFF'),(16474,46555,'#FFFFFF'),(16475,46556,'#f30cd8'),(16476,46557,'#FFFFFF'),(16477,46558,'#FFFFFF'),(16478,46559,'#FFFFFF'),(16479,46560,'#FFFFFF'),(16480,46561,'#FFFFFF'),(16481,46562,'#f717a4'),(16482,46563,'#FFFFFF'),(16483,46564,'#38df50'),(16484,46565,'#FFFFFF'),(16485,46566,'#FFFFFF'),(16486,46568,'#FFFFFF'),(16487,46569,'#FFFFFF'),(16488,46570,'#FFFFFF'),(16489,46571,'#1ceee8'),(16490,46572,'#FFFFFF'),(16491,46573,'#FFFFFF'),(16492,46574,'#FFFFFF'),(16493,46575,'#fd3df3'),(16494,46576,'#FFFFFF'),(16495,46577,'#FFFFFF'),(16496,46578,'#FFFFFF'),(16497,46579,'#FFFFFF'),(16498,46580,'#FFFFFF'),(16499,46581,'#FFFFFF'),(16500,46583,'#FFFFFF'),(16501,46584,'#FFFFFF'),(16502,46585,'#f5f664'),(16503,46586,'#FFFFFF'),(16504,46587,'#49d3d6'),(16505,46588,'#f8c90f'),(16506,46589,'#d938f4'),(16507,46590,'#FFFFFF'),(16508,46591,'#FFFFFF'),(16509,46593,'#FFFFFF'),(16510,46594,'#FFFFFF'),(16511,46595,'#e65269'),(16512,46596,'#e3ac32'),(16513,46597,'#FFFFFF'),(16514,46598,'#e3b263'),(16515,46599,'#3ccf7c'),(16516,46600,'#FFFFFF'),(16517,46601,'#d163ec'),(16518,46602,'#36fc3b'),(16519,46603,'#d0664f'),(16520,46604,'#FFFFFF'),(16521,46605,'#FFFFFF'),(16522,46606,'#FFFFFF'),(16523,46607,'#FFFFFF'),(16524,46608,'#FFFFFF'),(16525,46609,'#2ff4e1'),(16526,46610,'#FFFFFF'),(16527,46611,'#FFFFFF'),(16528,46612,'#decd6c'),(16529,46614,'#f18614'),(16530,46615,'#FFFFFF'),(16531,46616,'#97fb28'),(16532,46617,'#d922af'),(16533,46618,'#FFFFFF'),(16534,46619,'#eb7a04'),(16535,46620,'#b2e804'),(16536,46621,'#FFFFFF'),(16537,46622,'#FFFFFF'),(16538,46623,'#f77571'),(16539,46624,'#FFFFFF'),(16540,46625,'#FFFFFF'),(16541,46626,'#FFFFFF'),(16542,46627,'#FFFFFF'),(16543,46628,'#FFFFFF'),(16544,46629,'#FFFFFF'),(16545,46630,'#FFFFFF'),(16546,46631,'#FFFFFF'),(16547,46632,'#FFFFFF'),(16548,46633,'#FFFFFF'),(16549,46635,'#cbfb48'),(16550,46638,'#FFFFFF'),(16551,46639,'#FFFFFF'),(16552,46640,'#15cf9b'),(16553,46641,'#d066b1'),(16554,46642,'#FFFFFF'),(16555,46643,'#5bf4c7'),(16556,46644,'#FFFFFF'),(16557,46645,'#b064f9'),(16558,46646,'#FFFFFF'),(16559,46647,'#FFFFFF'),(16560,46648,'#e5cf38'),(16561,46649,'#fbe953'),(16562,46650,'#FFFFFF'),(16563,46651,'#FFFFFF'),(16564,46652,'#b012ea'),(16565,46653,'#FFFFFF'),(16566,46655,'#FFFFFF'),(16567,46656,'#FFFFFF'),(16568,46657,'#FFFFFF'),(16569,46658,'#efa929'),(16570,46659,'#d1d53b'),(16571,46660,'#FFFFFF'),(16572,46661,'#c2ea6b'),(16573,46662,'#d00cb4'),(16574,46663,'#1da4d0'),(16575,46664,'#57d941'),(16576,46665,'#45e548'),(16577,46666,'#FFFFFF'),(16578,46667,'#FFFFFF'),(16579,46668,'#FFFFFF'),(16580,46669,'#6352e1'),(16581,46670,'#FFFFFF'),(16582,46671,'#7ed160'),(16583,46672,'#05aed8'),(16584,46673,'#FFFFFF'),(16585,46675,'#dc448c'),(16586,46677,'#FFFFFF'),(16587,46678,'#FFFFFF'),(16588,46679,'#6ec2de'),(16589,46680,'#FFFFFF'),(16590,46681,'#FFFFFF'),(16591,46682,'#FFFFFF'),(16592,46683,'#FFFFFF'),(16593,46684,'#FFFFFF'),(16594,46685,'#FFFFFF'),(16595,46686,'#FFFFFF'),(16596,46687,'#FFFFFF'),(16597,46688,'#FFFFFF'),(16598,46689,'#FFFFFF'),(16599,46691,'#FFFFFF'),(16600,46692,'#4bd26c'),(16601,46694,'#FFFFFF'),(16602,46695,'#FFFFFF'),(16603,46696,'#FFFFFF'),(16604,46697,'#57de96'),(16605,46698,'#43e196'),(16606,46700,'#FFFFFF'),(16607,46701,'#ef5959'),(16608,46702,'#FFFFFF'),(16609,46703,'#4ef837'),(16610,46704,'#c41ecf'),(16611,46705,'#FFFFFF'),(16612,46706,'#FFFFFF'),(16613,46708,'#FFFFFF'),(16614,46709,'#FFFFFF'),(16615,46710,'#FFFFFF'),(16616,46711,'#FFFFFF'),(16617,46712,'#FFFFFF'),(16618,46713,'#FFFFFF'),(16619,46714,'#FFFFFF'),(16620,46715,'#FFFFFF'),(16621,46717,'#FFFFFF'),(16622,46719,'#FFFFFF'),(16623,46720,'#5cd7ef'),(16624,46721,'#FFFFFF'),(16625,46722,'#FFFFFF'),(16626,46723,'#dc40ba'),(16627,46724,'#FFFFFF'),(16628,46725,'#FFFFFF'),(16629,46726,'#db9c49'),(16630,46727,'#FFFFFF'),(16631,46728,'#fc9020'),(16632,46729,'#a60fe4'),(16633,46730,'#FFFFFF'),(16634,46731,'#FFFFFF'),(16635,46732,'#FFFFFF'),(16636,46733,'#db9266'),(16637,46734,'#FFFFFF'),(16638,46735,'#FFFFFF'),(16639,46736,'#85d30b'),(16640,46737,'#FFFFFF'),(16641,46738,'#FFFFFF'),(16642,46739,'#FFFFFF'),(16643,46740,'#FFFFFF'),(16644,46741,'#FFFFFF'),(16645,46742,'#FFFFFF'),(16646,46744,'#e76498'),(16647,46745,'#FFFFFF'),(16648,46746,'#FFFFFF'),(16649,46747,'#e4ba56'),(16650,46748,'#FFFFFF'),(16651,46751,'#FFFFFF'),(16652,46752,'#3df254'),(16653,46753,'#FFFFFF'),(16654,46755,'#FFFFFF'),(16655,46756,'#FFFFFF'),(16656,46757,'#FFFFFF'),(16657,46758,'#FFFFFF'),(16658,46759,'#d51c54'),(16659,46760,'#FFFFFF'),(16660,46761,'#FFFFFF'),(16661,46762,'#FFFFFF'),(16662,46763,'#FFFFFF'),(16663,46764,'#ec2ff1'),(16664,46765,'#FFFFFF'),(16665,46766,'#41aafe'),(16666,46767,'#42e8fb'),(16667,46768,'#FFFFFF'),(16668,46769,'#FFFFFF'),(16669,46770,'#FFFFFF'),(16670,46771,'#FFFFFF'),(16671,46772,'#FFFFFF'),(16672,46773,'#FFFFFF'),(16673,46774,'#14a3f0'),(16674,46775,'#0ad868'),(16675,46776,'#FFFFFF'),(16676,46778,'#FFFFFF'),(16677,46779,'#FFFFFF'),(16678,46780,'#FFFFFF'),(16679,46781,'#4ea1da'),(16680,46782,'#e09113'),(16681,46784,'#FFFFFF'),(16682,46785,'#FFFFFF'),(16683,46786,'#de3078'),(16684,46787,'#FFFFFF'),(16685,46788,'#07eb30'),(16686,46789,'#FFFFFF'),(16687,46790,'#FFFFFF'),(16688,46791,'#FFFFFF'),(16689,46792,'#FFFFFF'),(16690,46793,'#e6a440'),(16691,46794,'#FFFFFF'),(16692,46795,'#FFFFFF'),(16693,46796,'#d42bcc'),(16694,46797,'#63e8d9'),(16695,46798,'#FFFFFF'),(16696,46799,'#f34c21'),(16697,46800,'#FFFFFF'),(16698,46801,'#d0c554'),(16699,46802,'#fc6129'),(16700,46803,'#FFFFFF'),(16701,46804,'#4ced03'),(16702,46805,'#c1d955'),(16703,46806,'#FFFFFF'),(16704,46808,'#FFFFFF'),(16705,46809,'#ec8012'),(16706,46810,'#e459d8'),(16707,46811,'#85d162'),(16708,46812,'#4a9be9'),(16709,46813,'#FFFFFF'),(16710,46814,'#8b7ffe'),(16711,46815,'#5ad218'),(16712,46816,'#FFFFFF'),(16713,46817,'#FFFFFF'),(16714,46818,'#FFFFFF'),(16715,46819,'#FFFFFF'),(16716,46820,'#FFFFFF'),(16717,46821,'#0e4de3'),(16718,46822,'#FFFFFF'),(16719,46824,'#3425ce'),(16720,46825,'#FFFFFF'),(16721,46827,'#d75dee'),(16722,46828,'#FFFFFF'),(16723,46829,'#2c52db'),(16724,46830,'#FFFFFF'),(16725,46831,'#dff91f'),(16726,46832,'#FFFFFF'),(16727,46833,'#FFFFFF'),(16728,46834,'#FFFFFF'),(16729,46835,'#62a4ec'),(16730,46836,'#FFFFFF'),(16731,46837,'#1936d8'),(16732,46838,'#FFFFFF'),(16733,46839,'#FFFFFF'),(16734,46840,'#FFFFFF'),(16735,46841,'#9aeb1e'),(16736,46842,'#94d860'),(16737,46843,'#FFFFFF'),(16738,46844,'#da5620'),(16739,46845,'#FFFFFF'),(16740,46846,'#FFFFFF'),(16741,46847,'#9bf302'),(16742,46848,'#FFFFFF'),(16743,46849,'#FFFFFF'),(16744,46850,'#FFFFFF'),(16745,46851,'#cca83c'),(16746,46852,'#FFFFFF'),(16747,46853,'#64e64e'),(16748,46854,'#FFFFFF'),(16749,46855,'#FFFFFF'),(16750,46856,'#FFFFFF'),(16751,46857,'#d74ee7'),(16752,46858,'#FFFFFF'),(16753,46859,'#82d855'),(16754,46860,'#FFFFFF'),(16755,46861,'#43ea07'),(16756,46862,'#FFFFFF'),(16757,46863,'#FFFFFF'),(16758,46864,'#FFFFFF'),(16759,46865,'#FFFFFF'),(16760,46867,'#FFFFFF'),(16761,46868,'#3adeeb'),(16762,46869,'#FFFFFF'),(16763,46870,'#FFFFFF'),(16764,46871,'#FFFFFF'),(16765,46872,'#FFFFFF'),(16766,46873,'#FFFFFF'),(16767,46874,'#69fd66'),(16768,46875,'#FFFFFF'),(16769,46876,'#FFFFFF'),(16770,46878,'#FFFFFF'),(16771,46879,'#FFFFFF'),(16772,46880,'#FFFFFF'),(16773,46881,'#dd2faf'),(16774,46882,'#6af4b4'),(16775,46883,'#FFFFFF'),(16776,46884,'#FFFFFF'),(16777,46885,'#FFFFFF'),(16778,46887,'#FFFFFF'),(16779,46888,'#FFFFFF'),(16780,46889,'#FFFFFF'),(16781,46890,'#52f555'),(16782,46891,'#FFFFFF'),(16783,46892,'#e244e1'),(16784,46893,'#b9f704'),(16785,46894,'#FFFFFF'),(16786,46895,'#FFFFFF'),(16787,46896,'#FFFFFF'),(16788,46897,'#f550b3'),(16789,46898,'#FFFFFF'),(16790,46899,'#FFFFFF'),(16791,46900,'#35db84'),(16792,46902,'#0ae065'),(16793,46903,'#FFFFFF'),(16794,46904,'#26e165'),(16795,46905,'#31d17a'),(16796,46907,'#FFFFFF'),(16797,46908,'#FFFFFF'),(16798,46909,'#FFFFFF'),(16799,46910,'#FFFFFF'),(16800,46911,'#FFFFFF'),(16801,46912,'#d6b753'),(16802,46913,'#FFFFFF'),(16803,46914,'#FFFFFF'),(16804,46915,'#d79157'),(16805,46916,'#5f6dd4'),(16806,46917,'#FFFFFF'),(16807,46918,'#FFFFFF'),(16808,46919,'#FFFFFF'),(16809,46920,'#FFFFFF'),(16810,46921,'#FFFFFF'),(16811,46922,'#FFFFFF'),(16812,46923,'#FFFFFF'),(16813,46924,'#FFFFFF'),(16814,46925,'#FFFFFF'),(16815,46926,'#b0ec30'),(16816,46927,'#FFFFFF'),(16817,46928,'#e26467'),(16818,46929,'#26ce80'),(16819,46930,'#2cd79a'),(16820,46931,'#e7c269'),(16821,46932,'#FFFFFF'),(16822,46933,'#FFFFFF'),(16823,46934,'#FFFFFF'),(16824,46936,'#dbb662'),(16825,46937,'#FFFFFF'),(16826,46938,'#FFFFFF'),(16827,46939,'#FFFFFF'),(16828,46940,'#FFFFFF'),(16829,46942,'#FFFFFF'),(16830,46943,'#2b59db'),(16831,46944,'#FFFFFF'),(16832,46945,'#FFFFFF'),(16833,46946,'#FFFFFF'),(16834,46947,'#FFFFFF'),(16835,46948,'#3c4bf5'),(16836,46950,'#FFFFFF'),(16837,46951,'#FFFFFF'),(16838,46952,'#ca5bd6'),(16839,46953,'#FFFFFF'),(16840,46954,'#FFFFFF'),(16841,46955,'#4debac'),(16842,46956,'#FFFFFF'),(16843,46957,'#FFFFFF'),(16844,46959,'#51d2f1'),(16845,46960,'#FFFFFF'),(16846,46961,'#e4d932'),(16847,46962,'#cdb747'),(16848,46964,'#e0713d'),(16849,46965,'#FFFFFF'),(16850,46966,'#FFFFFF'),(16851,46967,'#FFFFFF'),(16852,46968,'#a872ef'),(16853,46969,'#f735f4'),(16854,46970,'#FFFFFF'),(16855,46971,'#408fd5'),(16856,46972,'#5d6cf4'),(16857,46974,'#FFFFFF'),(16858,46975,'#FFFFFF'),(16859,46976,'#a9d13e'),(16860,46978,'#8936f2'),(16861,46979,'#FFFFFF'),(16862,46980,'#FFFFFF'),(16863,46981,'#FFFFFF'),(16864,46982,'#f55873'),(16865,46983,'#FFFFFF'),(16866,46984,'#2c8bd0'),(16867,46985,'#FFFFFF'),(16868,46986,'#3ad921'),(16869,46987,'#FFFFFF'),(16870,46991,'#bc1fd7'),(16871,46992,'#FFFFFF'),(16872,46993,'#FFFFFF'),(16873,46995,'#FFFFFF'),(16874,46996,'#29d4c4'),(16875,46997,'#FFFFFF'),(16876,46998,'#FFFFFF'),(16877,46999,'#d129d1'),(16878,47000,'#FFFFFF'),(16879,47001,'#FFFFFF'),(16880,47002,'#FFFFFF'),(16881,47003,'#fdea63'),(16882,47004,'#f93d3d'),(16883,47005,'#FFFFFF'),(16884,47006,'#11f7b6'),(16885,47007,'#FFFFFF'),(16886,47008,'#FFFFFF'),(16887,47009,'#e7c94a'),(16888,47010,'#3785d2'),(16889,47011,'#e57a3d'),(16890,47012,'#FFFFFF'),(16891,47013,'#FFFFFF'),(16892,47014,'#FFFFFF'),(16893,47017,'#3a79fc'),(16894,47019,'#FFFFFF'),(16895,47020,'#50f064'),(16896,47021,'#FFFFFF'),(16897,47022,'#FFFFFF'),(16898,47023,'#FFFFFF'),(16899,47024,'#7234e2'),(16900,47025,'#cdc616'),(16901,47026,'#FFFFFF'),(16902,47027,'#3cf891'),(16903,47028,'#cf7429'),(16904,47031,'#e0bc54'),(16905,47032,'#dd0e9a'),(16906,47033,'#FFFFFF'),(16907,47034,'#5b1fd8'),(16908,47036,'#54f071'),(16909,47037,'#fc65cf'),(16910,47038,'#4cd34e'),(16911,47039,'#8008e9'),(16912,47040,'#FFFFFF'),(16913,47041,'#FFFFFF'),(16914,47042,'#FFFFFF'),(16915,47043,'#48f437'),(16916,47044,'#FFFFFF'),(16917,47045,'#FFFFFF'),(16918,47046,'#FFFFFF'),(16919,47047,'#30d6da'),(16920,47048,'#FFFFFF'),(16921,47049,'#FFFFFF'),(16922,47050,'#FFFFFF'),(16923,47051,'#FFFFFF'),(16924,47052,'#57d8e4'),(16925,47053,'#cc121b'),(16926,47055,'#8261f8'),(16927,47056,'#88d64d'),(16928,47057,'#FFFFFF'),(16929,47059,'#FFFFFF'),(16930,47060,'#FFFFFF'),(16931,47061,'#4579f8'),(16932,47062,'#FFFFFF'),(16933,47063,'#FFFFFF'),(16934,47064,'#FFFFFF'),(16935,47065,'#cf2b06'),(16936,47066,'#FFFFFF'),(16937,47067,'#FFFFFF'),(16938,47068,'#2956d9'),(16939,47069,'#d60aca'),(16940,47070,'#FFFFFF'),(16941,47071,'#FFFFFF'),(16942,47072,'#FFFFFF'),(16943,47074,'#e71f20'),(16944,47075,'#FFFFFF'),(16945,47076,'#8d54d2'),(16946,47078,'#d23f61'),(16947,47079,'#FFFFFF'),(16948,47083,'#721aeb'),(16949,47084,'#FFFFFF'),(16950,47085,'#FFFFFF'),(16951,47086,'#FFFFFF'),(16952,47088,'#FFFFFF'),(16953,47089,'#3feec1'),(16954,47090,'#703ce2'),(16955,47091,'#FFFFFF'),(16956,47092,'#FFFFFF'),(16957,47093,'#FFFFFF'),(16958,47094,'#FFFFFF'),(16959,47095,'#FFFFFF'),(16960,47096,'#FFFFFF'),(16961,47097,'#FFFFFF'),(16962,47098,'#FFFFFF'),(16963,47099,'#5b0bcf'),(16964,47100,'#d4549e'),(16965,47102,'#d16d4c'),(16966,47103,'#2fe3b5'),(16967,47104,'#FFFFFF'),(16968,47105,'#FFFFFF'),(16969,47106,'#FFFFFF'),(16970,47107,'#FFFFFF'),(16971,47108,'#FFFFFF'),(16972,47109,'#28fd09'),(16973,47110,'#FFFFFF'),(16974,47111,'#f23e3a'),(16975,47113,'#FFFFFF'),(16976,47114,'#f23e55'),(16977,47115,'#FFFFFF'),(16978,47116,'#ceeb10'),(16979,47117,'#e5d851'),(16980,47118,'#FFFFFF'),(16981,47120,'#FFFFFF'),(16982,47121,'#c5da02'),(16983,47122,'#FFFFFF'),(16984,47123,'#0958e7'),(16985,47124,'#d34b01'),(16986,47125,'#FFFFFF'),(16987,47127,'#FFFFFF'),(16988,47128,'#FFFFFF'),(16989,47130,'#FFFFFF'),(16990,47131,'#FFFFFF'),(16991,47132,'#a114dc'),(16992,47133,'#1ae1d1'),(16993,47134,'#FFFFFF'),(16994,47135,'#FFFFFF'),(16995,47136,'#FFFFFF'),(16996,47137,'#FFFFFF'),(16997,47138,'#f72c65'),(16998,47139,'#e6a544'),(16999,47140,'#FFFFFF'),(17000,47142,'#FFFFFF'),(17001,47143,'#FFFFFF'),(17002,47144,'#3f56ea'),(17003,47145,'#FFFFFF'),(17004,47146,'#FFFFFF'),(17005,47147,'#FFFFFF'),(17006,47148,'#FFFFFF'),(17007,47149,'#FFFFFF'),(17008,47150,'#935ee4'),(17009,47151,'#43d379'),(17010,47152,'#FFFFFF'),(17011,47156,'#FFFFFF'),(17012,47157,'#64cf5b'),(17013,47158,'#FFFFFF'),(17014,47159,'#FFFFFF'),(17015,47160,'#FFFFFF'),(17016,47161,'#FFFFFF'),(17017,47162,'#FFFFFF'),(17018,47163,'#FFFFFF'),(17019,47164,'#ee993e'),(17020,47165,'#74c5ef'),(17021,47166,'#c127d5'),(17022,47167,'#85fa67'),(17023,47168,'#cf20ce'),(17024,47169,'#FFFFFF'),(17025,47170,'#FFFFFF'),(17026,47171,'#FFFFFF'),(17027,47172,'#FFFFFF'),(17028,47173,'#FFFFFF'),(17029,47174,'#FFFFFF'),(17030,47175,'#FFFFFF'),(17031,47176,'#FFFFFF'),(17032,47177,'#FFFFFF'),(17033,47178,'#FFFFFF'),(17034,47179,'#FFFFFF'),(17035,47180,'#FFFFFF'),(17036,47181,'#FFFFFF'),(17037,47182,'#e8548a'),(17038,47183,'#FFFFFF'),(17039,47184,'#e8dd4b'),(17040,47185,'#FFFFFF'),(17041,47186,'#FFFFFF'),(17042,47187,'#FFFFFF'),(17043,47188,'#dc5487'),(17044,47189,'#e2ea09'),(17045,47190,'#FFFFFF'),(17046,47191,'#19fcfa'),(17047,47192,'#FFFFFF'),(17048,47193,'#8ce23b'),(17049,47196,'#8121d9'),(17050,47197,'#d1b406'),(17051,47205,'#61a9cd'),(17052,47207,'#726afe'),(17053,47208,'#2bdcea'),(17054,47236,'#d05e2c'),(17055,47246,'#ef803b'),(17056,47280,'#57adee'),(17057,47285,'#ef1e43'),(17058,47293,'#5e44da'),(17059,47294,'#6e7ade'),(17060,47303,'#d33c54'),(17061,47324,'#6693ea'),(17062,47332,'#31aad7'),(17063,47337,'#8d2ed4'),(17064,47346,'#d89e02'),(17065,47391,'#64e8bc'),(17066,47393,'#da63aa'),(17067,47409,'#dc0187'),(17068,47415,'#10eed3'),(17069,47423,'#31e2dc'),(17070,47424,'#FFFFFF'),(17071,47425,'#347eeb'),(17072,47426,'#FFFFFF'),(17073,47427,'#FFFFFF'),(17074,47429,'#FFFFFF'),(17075,47430,'#FFFFFF'),(17076,47431,'#FFFFFF'),(17077,47432,'#FFFFFF'),(17078,47433,'#2e13d1'),(17079,47434,'#FFFFFF'),(17080,47435,'#FFFFFF'),(17081,47436,'#f32f2a'),(17082,47437,'#FFFFFF'),(17083,47438,'#FFFFFF'),(17084,47439,'#FFFFFF'),(17085,47440,'#FFFFFF'),(17086,47441,'#FFFFFF'),(17087,47443,'#FFFFFF'),(17088,47444,'#FFFFFF'),(17089,47445,'#FFFFFF'),(17090,47446,'#56dae3'),(17091,47447,'#2dd7fe'),(17092,47448,'#FFFFFF'),(17093,47449,'#FFFFFF'),(17094,47450,'#FFFFFF'),(17095,47452,'#FFFFFF'),(17096,47453,'#FFFFFF'),(17097,47454,'#FFFFFF'),(17098,47456,'#FFFFFF'),(17099,47458,'#FFFFFF'),(17100,47459,'#FFFFFF'),(17101,47460,'#FFFFFF'),(17102,47462,'#FFFFFF'),(17103,47464,'#FFFFFF'),(17104,47465,'#FFFFFF'),(17105,47466,'#FFFFFF'),(17106,47468,'#FFFFFF'),(17107,47469,'#77b0f7'),(17108,47470,'#FFFFFF'),(17109,47472,'#FFFFFF'),(17110,47473,'#FFFFFF'),(17111,47474,'#FFFFFF'),(17112,47475,'#FFFFFF'),(17113,47476,'#FFFFFF'),(17114,47477,'#FFFFFF'),(17115,47479,'#FFFFFF'),(17116,47480,'#FFFFFF'),(17117,47481,'#FFFFFF'),(17118,47483,'#FFFFFF'),(17119,47485,'#FFFFFF'),(17120,47486,'#FFFFFF'),(17121,47487,'#FFFFFF'),(17122,47489,'#FFFFFF'),(17123,47490,'#FFFFFF'),(17124,47492,'#FFFFFF'),(17125,47493,'#FFFFFF'),(17126,47494,'#FFFFFF'),(17127,47495,'#FFFFFF'),(17128,47496,'#4ce123'),(17129,47497,'#FFFFFF'),(17130,47498,'#FFFFFF'),(17131,47499,'#FFFFFF'),(17132,47500,'#FFFFFF'),(17133,47501,'#FFFFFF'),(17134,47502,'#FFFFFF'),(17135,47503,'#FFFFFF'),(17136,47504,'#FFFFFF'),(17137,47506,'#FFFFFF'),(17138,47507,'#FFFFFF'),(17139,47508,'#FFFFFF'),(17140,47509,'#FFFFFF'),(17141,47510,'#38f30c'),(17142,47511,'#FFFFFF'),(17143,47512,'#FFFFFF'),(17144,47513,'#FFFFFF'),(17145,47514,'#FFFFFF'),(17146,47515,'#FFFFFF'),(17147,47516,'#FFFFFF'),(17148,47517,'#FFFFFF'),(17149,47518,'#FFFFFF'),(17150,47519,'#FFFFFF'),(17151,47520,'#d6f90e'),(17152,47523,'#FFFFFF'),(17153,47524,'#FFFFFF'),(17154,47525,'#FFFFFF'),(17155,47526,'#FFFFFF'),(17156,47527,'#FFFFFF'),(17157,47528,'#FFFFFF'),(17158,47529,'#FFFFFF'),(17159,47530,'#d2537c'),(17160,47531,'#cf1faf'),(17161,47532,'#FFFFFF'),(17162,47533,'#FFFFFF'),(17163,47534,'#FFFFFF'),(17164,47535,'#FFFFFF'),(17165,47536,'#FFFFFF'),(17166,47537,'#FFFFFF'),(17167,47538,'#FFFFFF'),(17168,47539,'#FFFFFF'),(17169,47540,'#FFFFFF'),(17170,47542,'#e69b1d'),(17171,47543,'#a601db'),(17172,47544,'#29fc3a'),(17173,47545,'#FFFFFF'),(17174,47546,'#d78b69'),(17175,47547,'#FFFFFF'),(17176,47548,'#FFFFFF'),(17177,47550,'#FFFFFF'),(17178,47551,'#FFFFFF'),(17179,47553,'#FFFFFF'),(17180,47554,'#FFFFFF'),(17181,47557,'#FFFFFF'),(17182,47558,'#FFFFFF'),(17183,47559,'#FFFFFF'),(17184,47561,'#FFFFFF'),(17185,47562,'#f74b26'),(17186,47563,'#FFFFFF'),(17187,47564,'#FFFFFF'),(17188,47565,'#FFFFFF'),(17189,47566,'#80e35f'),(17190,47567,'#FFFFFF'),(17191,47569,'#FFFFFF'),(17192,47570,'#FFFFFF'),(17193,47571,'#203ffa'),(17194,47574,'#FFFFFF'),(17195,47576,'#FFFFFF'),(17196,47577,'#FFFFFF'),(17197,47579,'#0631cc'),(17198,47580,'#FFFFFF'),(17199,47581,'#FFFFFF'),(17200,47582,'#FFFFFF'),(17201,47583,'#FFFFFF'),(17202,47584,'#FFFFFF'),(17203,47585,'#FFFFFF'),(17204,47586,'#FFFFFF'),(17205,47587,'#FFFFFF'),(17206,47588,'#FFFFFF'),(17207,47589,'#FFFFFF'),(17208,47593,'#f19144'),(17209,47595,'#0cf8da'),(17210,47596,'#ce496d'),(17211,47597,'#FFFFFF'),(17212,47598,'#FFFFFF'),(17213,47599,'#FFFFFF'),(17214,47600,'#FFFFFF'),(17215,47601,'#FFFFFF'),(17216,47602,'#FFFFFF'),(17217,47603,'#FFFFFF'),(17218,47604,'#FFFFFF'),(17219,47605,'#bff449'),(17220,47607,'#FFFFFF'),(17221,47608,'#FFFFFF'),(17222,47610,'#FFFFFF'),(17223,47611,'#FFFFFF'),(17224,47612,'#FFFFFF'),(17225,47613,'#FFFFFF'),(17226,47614,'#FFFFFF'),(17227,47615,'#FFFFFF'),(17228,47616,'#FFFFFF'),(17229,47618,'#FFFFFF'),(17230,47619,'#FFFFFF'),(17231,47620,'#FFFFFF'),(17232,47621,'#FFFFFF'),(17233,47622,'#FFFFFF'),(17234,47623,'#FFFFFF'),(17235,47624,'#FFFFFF'),(17236,47625,'#FFFFFF'),(17237,47626,'#cc236f'),(17238,47627,'#FFFFFF'),(17239,47629,'#FFFFFF'),(17240,47630,'#FFFFFF'),(17241,47631,'#FFFFFF'),(17242,47633,'#FFFFFF'),(17243,47634,'#FFFFFF'),(17244,47635,'#FFFFFF'),(17245,47636,'#FFFFFF'),(17246,47637,'#FFFFFF'),(17247,47638,'#f62b33'),(17248,47639,'#08f76d'),(17249,47640,'#FFFFFF'),(17250,47641,'#FFFFFF'),(17251,47642,'#FFFFFF'),(17252,47643,'#099be1'),(17253,47644,'#FFFFFF'),(17254,47645,'#FFFFFF'),(17255,47646,'#FFFFFF'),(17256,47647,'#FFFFFF'),(17257,47648,'#f85bb4'),(17258,47649,'#d38f54'),(17259,47651,'#FFFFFF'),(17260,47653,'#FFFFFF'),(17261,47654,'#FFFFFF'),(17262,47655,'#2be8e4'),(17263,47656,'#FFFFFF'),(17264,47659,'#FFFFFF'),(17265,47660,'#FFFFFF'),(17266,47661,'#FFFFFF'),(17267,47662,'#FFFFFF'),(17268,47664,'#FFFFFF'),(17269,47666,'#FFFFFF'),(17270,47667,'#FFFFFF'),(17271,47668,'#FFFFFF'),(17272,47669,'#FFFFFF'),(17273,47670,'#FFFFFF'),(17274,47671,'#FFFFFF'),(17275,47673,'#FFFFFF'),(17276,47674,'#FFFFFF'),(17277,47675,'#FFFFFF'),(17278,47676,'#FFFFFF'),(17279,47678,'#FFFFFF'),(17280,47680,'#FFFFFF'),(17281,47681,'#FFFFFF'),(17282,47682,'#FFFFFF'),(17283,47683,'#FFFFFF'),(17284,47684,'#FFFFFF'),(17285,47685,'#FFFFFF'),(17286,47686,'#FFFFFF'),(17287,47687,'#FFFFFF'),(17288,47688,'#97f246'),(17289,47689,'#4bfdc2'),(17290,47690,'#FFFFFF'),(17291,47691,'#FFFFFF'),(17292,47692,'#FFFFFF'),(17293,47693,'#FFFFFF'),(17294,47694,'#FFFFFF'),(17295,47695,'#FFFFFF'),(17296,47696,'#FFFFFF'),(17297,47698,'#befe72'),(17298,47699,'#FFFFFF'),(17299,47700,'#FFFFFF'),(17300,47701,'#FFFFFF'),(17301,47704,'#FFFFFF'),(17302,47705,'#FFFFFF'),(17303,47706,'#FFFFFF'),(17304,47707,'#FFFFFF'),(17305,47708,'#FFFFFF'),(17306,47709,'#FFFFFF'),(17307,47711,'#FFFFFF'),(17308,47712,'#FFFFFF'),(17309,47713,'#FFFFFF'),(17310,47714,'#FFFFFF'),(17311,47716,'#FFFFFF'),(17312,47717,'#FFFFFF'),(17313,47718,'#FFFFFF'),(17314,47719,'#FFFFFF'),(17315,47720,'#FFFFFF'),(17316,47722,'#342ae7'),(17317,47724,'#FFFFFF'),(17318,47725,'#FFFFFF'),(17319,47726,'#FFFFFF'),(17320,47727,'#FFFFFF'),(17321,47728,'#FFFFFF'),(17322,47729,'#FFFFFF'),(17323,47730,'#FFFFFF'),(17324,47731,'#FFFFFF'),(17325,47732,'#FFFFFF'),(17326,47733,'#FFFFFF'),(17327,47735,'#FFFFFF'),(17328,47736,'#28d1c3'),(17329,47738,'#FFFFFF'),(17330,47739,'#FFFFFF'),(17331,47741,'#FFFFFF'),(17332,47742,'#FFFFFF'),(17333,47743,'#FFFFFF'),(17334,47745,'#FFFFFF'),(17335,47746,'#FFFFFF'),(17336,47747,'#FFFFFF'),(17337,47748,'#FFFFFF'),(17338,47749,'#b3f416'),(17339,47751,'#FFFFFF'),(17340,47752,'#FFFFFF'),(17341,47753,'#0ee4bd'),(17342,47754,'#27fa01'),(17343,47755,'#FFFFFF'),(17344,47757,'#FFFFFF'),(17345,47758,'#FFFFFF'),(17346,47759,'#24ce49'),(17347,47760,'#FFFFFF'),(17348,47761,'#FFFFFF'),(17349,47762,'#5577f3'),(17350,47764,'#FFFFFF'),(17351,47765,'#FFFFFF'),(17352,47767,'#FFFFFF'),(17353,47769,'#FFFFFF'),(17354,47770,'#FFFFFF'),(17355,47771,'#FFFFFF'),(17356,47772,'#5d36cd'),(17357,47773,'#1af1aa'),(17358,47774,'#FFFFFF'),(17359,47775,'#FFFFFF'),(17360,47777,'#FFFFFF'),(17361,47778,'#FFFFFF'),(17362,47780,'#FFFFFF'),(17363,47781,'#FFFFFF'),(17364,47782,'#FFFFFF'),(17365,47783,'#FFFFFF'),(17366,47785,'#FFFFFF'),(17367,47787,'#FFFFFF'),(17368,47788,'#FFFFFF'),(17369,47789,'#FFFFFF'),(17370,47790,'#6725d0'),(17371,47791,'#bb2cef'),(17372,47793,'#de30c8'),(17373,47795,'#FFFFFF'),(17374,47796,'#FFFFFF'),(17375,47797,'#11bbce'),(17376,47798,'#FFFFFF'),(17377,47799,'#FFFFFF'),(17378,47800,'#FFFFFF'),(17379,47801,'#5334e3'),(17380,47802,'#FFFFFF'),(17381,47803,'#FFFFFF'),(17382,47804,'#FFFFFF'),(17383,47805,'#FFFFFF'),(17384,47806,'#FFFFFF'),(17385,47807,'#FFFFFF'),(17386,47808,'#FFFFFF'),(17387,47809,'#FFFFFF'),(17388,47810,'#FFFFFF'),(17389,47811,'#FFFFFF'),(17390,47812,'#FFFFFF'),(17391,47813,'#e1e520'),(17392,47814,'#FFFFFF'),(17393,47815,'#a338e3'),(17394,47816,'#FFFFFF'),(17395,47821,'#FFFFFF'),(17396,47823,'#38bbd6'),(17397,47825,'#FFFFFF'),(17398,47826,'#FFFFFF'),(17399,47828,'#FFFFFF'),(17400,47829,'#FFFFFF'),(17401,47833,'#FFFFFF'),(17402,47834,'#FFFFFF'),(17403,47835,'#FFFFFF'),(17404,47836,'#FFFFFF'),(17405,47837,'#FFFFFF'),(17406,47839,'#b1d63c'),(17407,47840,'#FFFFFF'),(17408,47841,'#FFFFFF'),(17409,47842,'#FFFFFF'),(17410,47843,'#e1be4a'),(17411,47844,'#eaa824'),(17412,47845,'#0fa4fd'),(17413,47847,'#FFFFFF'),(17414,47848,'#FFFFFF'),(17415,47849,'#FFFFFF'),(17416,47851,'#FFFFFF'),(17417,47852,'#FFFFFF'),(17418,47853,'#f9d546'),(17419,47855,'#FFFFFF'),(17420,47856,'#FFFFFF'),(17421,47857,'#FFFFFF'),(17422,47858,'#FFFFFF'),(17423,47859,'#FFFFFF'),(17424,47860,'#FFFFFF'),(17425,47861,'#FFFFFF'),(17426,47862,'#3d62f5'),(17427,47863,'#FFFFFF'),(17428,47864,'#FFFFFF'),(17429,47865,'#FFFFFF'),(17430,47866,'#FFFFFF'),(17431,47867,'#FFFFFF'),(17432,47868,'#FFFFFF'),(17433,47869,'#69fb88'),(17434,47870,'#14d11f'),(17435,47871,'#FFFFFF'),(17436,47873,'#FFFFFF'),(17437,47874,'#FFFFFF'),(17438,47875,'#FFFFFF'),(17439,47876,'#FFFFFF'),(17440,47877,'#FFFFFF'),(17441,47879,'#FFFFFF'),(17442,47880,'#FFFFFF'),(17443,47881,'#FFFFFF'),(17444,47882,'#FFFFFF'),(17445,47883,'#cf3c3b'),(17446,47884,'#FFFFFF'),(17447,47885,'#FFFFFF'),(17448,47887,'#fa6ff3'),(17449,47888,'#5adcc2'),(17450,47889,'#dab555'),(17451,47890,'#FFFFFF'),(17452,47891,'#FFFFFF'),(17453,47892,'#FFFFFF'),(17454,47894,'#f90b21'),(17455,47897,'#FFFFFF'),(17456,47899,'#FFFFFF'),(17457,47901,'#FFFFFF'),(17458,47902,'#FFFFFF'),(17459,47903,'#FFFFFF'),(17460,47904,'#FFFFFF'),(17461,47905,'#FFFFFF'),(17462,47906,'#FFFFFF'),(17463,47907,'#FFFFFF'),(17464,47909,'#FFFFFF'),(17465,47910,'#FFFFFF'),(17466,47911,'#FFFFFF'),(17467,47912,'#FFFFFF'),(17468,47913,'#FFFFFF'),(17469,47914,'#FFFFFF'),(17470,47915,'#FFFFFF'),(17471,47916,'#7964cd'),(17472,47917,'#FFFFFF'),(17473,47918,'#FFFFFF'),(17474,47919,'#FFFFFF'),(17475,47920,'#FFFFFF'),(17476,47921,'#FFFFFF'),(17477,47922,'#FFFFFF'),(17478,47923,'#FFFFFF'),(17479,47924,'#FFFFFF'),(17480,47925,'#FFFFFF'),(17481,47926,'#FFFFFF'),(17482,47927,'#FFFFFF'),(17483,47928,'#FFFFFF'),(17484,47929,'#FFFFFF'),(17485,47930,'#FFFFFF'),(17486,47931,'#FFFFFF'),(17487,47932,'#FFFFFF'),(17488,47933,'#66eb9d'),(17489,47934,'#FFFFFF'),(17490,47935,'#f3ec35'),(17491,47936,'#FFFFFF'),(17492,47937,'#FFFFFF'),(17493,47938,'#5d2ad3'),(17494,47939,'#FFFFFF'),(17495,47940,'#FFFFFF'),(17496,47941,'#FFFFFF'),(17497,47942,'#FFFFFF'),(17498,47943,'#FFFFFF'),(17499,47944,'#FFFFFF'),(17500,47945,'#FFFFFF'),(17501,47946,'#FFFFFF'),(17502,47947,'#FFFFFF'),(17503,47948,'#FFFFFF'),(17504,47949,'#FFFFFF'),(17505,47950,'#FFFFFF'),(17506,47951,'#e2b328'),(17507,47952,'#FFFFFF'),(17508,47953,'#FFFFFF'),(17509,47954,'#FFFFFF'),(17510,47955,'#FFFFFF'),(17511,47956,'#FFFFFF'),(17512,47957,'#FFFFFF'),(17513,47958,'#FFFFFF'),(17514,47959,'#FFFFFF'),(17515,47961,'#FFFFFF'),(17516,47962,'#FFFFFF'),(17517,47963,'#FFFFFF'),(17518,47964,'#FFFFFF'),(17519,47965,'#FFFFFF'),(17520,47966,'#FFFFFF'),(17521,47967,'#FFFFFF'),(17522,47969,'#FFFFFF'),(17523,47970,'#FFFFFF'),(17524,47971,'#FFFFFF'),(17525,47972,'#25dc7d'),(17526,47973,'#d5dd52'),(17527,47974,'#FFFFFF'),(17528,47975,'#FFFFFF'),(17529,47976,'#FFFFFF'),(17530,47978,'#FFFFFF'),(17531,47979,'#04db81'),(17532,47981,'#cb11f1'),(17533,47982,'#FFFFFF'),(17534,47984,'#FFFFFF'),(17535,47985,'#FFFFFF'),(17536,47986,'#FFFFFF'),(17537,47987,'#FFFFFF'),(17538,47988,'#FFFFFF'),(17539,47989,'#FFFFFF'),(17540,47990,'#FFFFFF'),(17541,47991,'#ba0bcd'),(17542,47992,'#FFFFFF'),(17543,47994,'#FFFFFF'),(17544,47995,'#9fe560'),(17545,47996,'#FFFFFF'),(17546,47997,'#FFFFFF'),(17547,47998,'#FFFFFF'),(17548,47999,'#FFFFFF'),(17549,48000,'#FFFFFF'),(17550,48001,'#FFFFFF'),(17551,48002,'#FFFFFF'),(17552,48003,'#19dfa6'),(17553,48004,'#FFFFFF'),(17554,48006,'#FFFFFF'),(17555,48008,'#FFFFFF'),(17556,48009,'#FFFFFF'),(17557,48010,'#FFFFFF'),(17558,48012,'#FFFFFF'),(17559,48013,'#FFFFFF'),(17560,48014,'#FFFFFF'),(17561,48015,'#FFFFFF'),(17562,48016,'#FFFFFF'),(17563,48018,'#FFFFFF'),(17564,48019,'#FFFFFF'),(17565,48020,'#FFFFFF'),(17566,48021,'#FFFFFF'),(17567,48022,'#FFFFFF'),(17568,48023,'#9a4ff5'),(17569,48024,'#FFFFFF'),(17570,48026,'#55e4a1'),(17571,48027,'#FFFFFF'),(17572,48028,'#FFFFFF'),(17573,48029,'#FFFFFF'),(17574,48030,'#FFFFFF'),(17575,48033,'#FFFFFF'),(17576,48034,'#FFFFFF'),(17577,48035,'#FFFFFF'),(17578,48036,'#FFFFFF'),(17579,48037,'#FFFFFF'),(17580,48038,'#FFFFFF'),(17581,48039,'#FFFFFF'),(17582,48040,'#FFFFFF'),(17583,48042,'#b7f855'),(17584,48043,'#FFFFFF'),(17585,48044,'#3a06e5'),(17586,48045,'#FFFFFF'),(17587,48046,'#cd6763'),(17588,48048,'#FFFFFF'),(17589,48049,'#FFFFFF'),(17590,48050,'#9936df'),(17591,48051,'#FFFFFF'),(17592,48052,'#FFFFFF'),(17593,48053,'#FFFFFF'),(17594,48054,'#FFFFFF'),(17595,48056,'#FFFFFF'),(17596,48057,'#FFFFFF'),(17597,48058,'#FFFFFF'),(17598,48059,'#FFFFFF'),(17599,48060,'#FFFFFF'),(17600,48063,'#FFFFFF'),(17601,48064,'#FFFFFF'),(17602,48066,'#FFFFFF'),(17603,48067,'#e88173'),(17604,48068,'#FFFFFF'),(17605,48069,'#FFFFFF'),(17606,48072,'#8bd703'),(17607,48073,'#FFFFFF'),(17608,48074,'#FFFFFF'),(17609,48075,'#12dd1c'),(17610,48077,'#FFFFFF'),(17611,48079,'#FFFFFF'),(17612,48080,'#2931f2'),(17613,48081,'#FFFFFF'),(17614,48082,'#FFFFFF'),(17615,48083,'#FFFFFF'),(17616,48084,'#FFFFFF'),(17617,48085,'#FFFFFF'),(17618,48086,'#FFFFFF'),(17619,48087,'#FFFFFF'),(17620,48088,'#FFFFFF'),(17621,48091,'#FFFFFF'),(17622,48092,'#FFFFFF'),(17623,48093,'#FFFFFF'),(17624,48094,'#FFFFFF'),(17625,48095,'#FFFFFF'),(17626,48096,'#FFFFFF'),(17627,48097,'#FFFFFF'),(17628,48098,'#FFFFFF'),(17629,48099,'#FFFFFF'),(17630,48100,'#FFFFFF'),(17631,48101,'#FFFFFF'),(17632,48102,'#FFFFFF'),(17633,48103,'#FFFFFF'),(17634,48104,'#f86cf9'),(17635,48105,'#FFFFFF'),(17636,48106,'#FFFFFF'),(17637,48107,'#FFFFFF'),(17638,48108,'#FFFFFF'),(17639,48109,'#FFFFFF'),(17640,48110,'#FFFFFF'),(17641,48111,'#FFFFFF'),(17642,48112,'#FFFFFF'),(17643,48113,'#FFFFFF'),(17644,48115,'#FFFFFF'),(17645,48116,'#FFFFFF'),(17646,48117,'#FFFFFF'),(17647,48118,'#FFFFFF'),(17648,48120,'#e59b0f'),(17649,48121,'#FFFFFF'),(17650,48122,'#FFFFFF'),(17651,48123,'#FFFFFF'),(17652,48124,'#FFFFFF'),(17653,48125,'#2fadfa'),(17654,48126,'#FFFFFF'),(17655,48127,'#FFFFFF'),(17656,48128,'#16d5a2'),(17657,48129,'#FFFFFF'),(17658,48130,'#6726f6'),(17659,48131,'#FFFFFF'),(17660,48133,'#FFFFFF'),(17661,48134,'#FFFFFF'),(17662,48135,'#FFFFFF'),(17663,48136,'#FFFFFF'),(17664,48137,'#FFFFFF'),(17665,48138,'#FFFFFF'),(17666,48139,'#FFFFFF'),(17667,48140,'#FFFFFF'),(17668,48142,'#FFFFFF'),(17669,48143,'#FFFFFF'),(17670,48144,'#FFFFFF'),(17671,48145,'#FFFFFF'),(17672,48146,'#FFFFFF'),(17673,48147,'#FFFFFF'),(17674,48148,'#FFFFFF'),(17675,48150,'#FFFFFF'),(17676,48151,'#FFFFFF'),(17677,48152,'#FFFFFF'),(17678,48153,'#FFFFFF'),(17679,48154,'#FFFFFF'),(17680,48156,'#FFFFFF'),(17681,48157,'#FFFFFF'),(17682,48158,'#FFFFFF'),(17683,48159,'#FFFFFF'),(17684,48160,'#FFFFFF'),(17685,48161,'#fc3b74'),(17686,48162,'#3c66ef'),(17687,48163,'#FFFFFF'),(17688,48164,'#f5931a'),(17689,48165,'#FFFFFF'),(17690,48166,'#FFFFFF'),(17691,48167,'#FFFFFF'),(17692,48168,'#FFFFFF'),(17693,48170,'#FFFFFF'),(17694,48171,'#FFFFFF'),(17695,48172,'#FFFFFF'),(17696,48174,'#FFFFFF'),(17697,48175,'#FFFFFF'),(17698,48176,'#FFFFFF'),(17699,48177,'#FFFFFF'),(17700,48178,'#FFFFFF'),(17701,48179,'#FFFFFF'),(17702,48180,'#FFFFFF'),(17703,48181,'#FFFFFF'),(17704,48182,'#FFFFFF'),(17705,48183,'#FFFFFF'),(17706,48184,'#FFFFFF'),(17707,48185,'#FFFFFF'),(17708,48186,'#FFFFFF'),(17709,48187,'#FFFFFF'),(17710,48188,'#FFFFFF'),(17711,48189,'#FFFFFF'),(17712,48192,'#FFFFFF'),(17713,48193,'#FFFFFF'),(17714,48194,'#FFFFFF'),(17715,48196,'#FFFFFF'),(17716,48197,'#FFFFFF'),(17717,48198,'#FFFFFF'),(17718,48199,'#FFFFFF'),(17719,48200,'#a2db2f'),(17720,48201,'#FFFFFF'),(17721,48202,'#FFFFFF'),(17722,48204,'#FFFFFF'),(17723,48205,'#FFFFFF'),(17724,48206,'#FFFFFF'),(17725,48207,'#FFFFFF'),(17726,48208,'#FFFFFF'),(17727,48209,'#FFFFFF'),(17728,48210,'#FFFFFF'),(17729,48211,'#FFFFFF'),(17730,48212,'#FFFFFF'),(17731,48213,'#54f1dc'),(17732,48214,'#FFFFFF'),(17733,48216,'#FFFFFF'),(17734,48217,'#FFFFFF'),(17735,48218,'#FFFFFF'),(17736,48219,'#FFFFFF'),(17737,48220,'#FFFFFF'),(17738,48221,'#1e04d7'),(17739,48222,'#FFFFFF'),(17740,48223,'#FFFFFF'),(17741,48224,'#FFFFFF'),(17742,48227,'#b6e15e'),(17743,48228,'#FFFFFF'),(17744,48231,'#FFFFFF'),(17745,48232,'#1ce1d4'),(17746,48233,'#FFFFFF'),(17747,48234,'#FFFFFF'),(17748,48235,'#FFFFFF'),(17749,48236,'#FFFFFF'),(17750,48238,'#FFFFFF'),(17751,48239,'#FFFFFF'),(17752,48240,'#FFFFFF'),(17753,48241,'#FFFFFF'),(17754,48242,'#FFFFFF'),(17755,48243,'#afd056'),(17756,48247,'#FFFFFF'),(17757,48248,'#FFFFFF'),(17758,48249,'#FFFFFF'),(17759,48250,'#FFFFFF'),(17760,48251,'#FFFFFF'),(17761,48252,'#61d0c4'),(17762,48253,'#eb3517'),(17763,48254,'#FFFFFF'),(17764,48256,'#FFFFFF'),(17765,48257,'#FFFFFF'),(17766,48258,'#FFFFFF'),(17767,48259,'#FFFFFF'),(17768,48260,'#FFFFFF'),(17769,48261,'#FFFFFF'),(17770,48262,'#FFFFFF'),(17771,48263,'#FFFFFF'),(17772,48266,'#FFFFFF'),(17773,48267,'#FFFFFF'),(17774,48270,'#fe76d0'),(17775,48271,'#FFFFFF'),(17776,48272,'#FFFFFF'),(17777,48273,'#FFFFFF'),(17778,48274,'#FFFFFF'),(17779,48275,'#FFFFFF'),(17780,48276,'#FFFFFF'),(17781,48277,'#fd1368'),(17782,48278,'#FFFFFF'),(17783,48279,'#FFFFFF'),(17784,48280,'#FFFFFF'),(17785,48281,'#FFFFFF'),(17786,48282,'#a625ed'),(17787,48285,'#ee4947'),(17788,48287,'#FFFFFF'),(17789,48288,'#49cd16'),(17790,48290,'#66e859'),(17791,48291,'#FFFFFF'),(17792,48292,'#FFFFFF'),(17793,48293,'#FFFFFF'),(17794,48295,'#f4419f'),(17795,48296,'#FFFFFF'),(17796,48297,'#FFFFFF'),(17797,48298,'#FFFFFF'),(17798,48299,'#FFFFFF'),(17799,48300,'#FFFFFF'),(17800,48301,'#FFFFFF'),(17801,48302,'#FFFFFF'),(17802,48303,'#FFFFFF'),(17803,48304,'#FFFFFF'),(17804,48305,'#FFFFFF'),(17805,48306,'#FFFFFF'),(17806,48307,'#FFFFFF'),(17807,48308,'#FFFFFF'),(17808,48309,'#FFFFFF'),(17809,48311,'#FFFFFF'),(17810,48312,'#51d217'),(17811,48314,'#FFFFFF'),(17812,48315,'#FFFFFF'),(17813,48316,'#FFFFFF'),(17814,48317,'#FFFFFF'),(17815,48318,'#FFFFFF'),(17816,48319,'#FFFFFF'),(17817,48320,'#FFFFFF'),(17818,48321,'#FFFFFF'),(17819,48323,'#FFFFFF'),(17820,48324,'#FFFFFF'),(17821,48325,'#FFFFFF'),(17822,48326,'#FFFFFF'),(17823,48327,'#FFFFFF'),(17824,48329,'#fc12a9'),(17825,48332,'#89f909'),(17826,48333,'#6cec9b'),(17827,48334,'#FFFFFF'),(17828,48335,'#FFFFFF'),(17829,48336,'#FFFFFF'),(17830,48337,'#db6c9c'),(17831,48338,'#FFFFFF'),(17832,48340,'#FFFFFF'),(17833,48341,'#FFFFFF'),(17834,48342,'#FFFFFF'),(17835,48343,'#FFFFFF'),(17836,48344,'#FFFFFF'),(17837,48345,'#FFFFFF'),(17838,48346,'#e5ab5b'),(17839,48347,'#FFFFFF'),(17840,48348,'#FFFFFF'),(17841,48349,'#dc1b66'),(17842,48350,'#FFFFFF'),(17843,48351,'#FFFFFF'),(17844,48352,'#FFFFFF'),(17845,48353,'#FFFFFF'),(17846,48354,'#FFFFFF'),(17847,48355,'#FFFFFF'),(17848,48356,'#FFFFFF'),(17849,48357,'#FFFFFF'),(17850,48358,'#FFFFFF'),(17851,48359,'#FFFFFF'),(17852,48360,'#FFFFFF'),(17853,48361,'#FFFFFF'),(17854,48362,'#FFFFFF'),(17855,48363,'#ec4d92'),(17856,48364,'#4b2fd0'),(17857,48365,'#FFFFFF'),(17858,48366,'#FFFFFF'),(17859,48367,'#FFFFFF'),(17860,48371,'#FFFFFF'),(17861,48373,'#FFFFFF'),(17862,48374,'#FFFFFF'),(17863,48375,'#d5b349'),(17864,48377,'#FFFFFF'),(17865,48378,'#FFFFFF'),(17866,48379,'#FFFFFF'),(17867,48381,'#FFFFFF'),(17868,48382,'#2720e6'),(17869,48383,'#FFFFFF'),(17870,48384,'#FFFFFF'),(17871,48386,'#bf6edf'),(17872,48387,'#FFFFFF'),(17873,48388,'#FFFFFF'),(17874,48389,'#FFFFFF'),(17875,48390,'#FFFFFF'),(17876,48391,'#FFFFFF'),(17877,48392,'#FFFFFF'),(17878,48393,'#FFFFFF'),(17879,48394,'#FFFFFF'),(17880,48395,'#d0a445'),(17881,48396,'#FFFFFF'),(17882,48397,'#FFFFFF'),(17883,48398,'#FFFFFF'),(17884,48399,'#FFFFFF'),(17885,48400,'#68a5d7'),(17886,48401,'#FFFFFF'),(17887,48402,'#FFFFFF'),(17888,48403,'#FFFFFF'),(17889,48406,'#FFFFFF'),(17890,48407,'#FFFFFF'),(17891,48408,'#FFFFFF'),(17892,48410,'#50d42f'),(17893,48411,'#FFFFFF'),(17894,48415,'#FFFFFF'),(17895,48416,'#FFFFFF'),(17896,48431,'#2cbdf5'),(17897,48432,'#76e22a'),(17898,48434,'#69de93'),(17899,48447,'#7df660'),(17900,48467,'#55a6de'),(17901,48478,'#73fe87'),(17902,48522,'#FFFFFF'),(17903,48535,'#5deb88'),(17904,48543,'#e33d34'),(17905,48544,'#5d07f2'),(17906,48545,'#3564f1'),(17907,48548,'#ce14b8'),(17908,48551,'#2a53d8'),(17909,48569,'#d69510'),(17910,48572,'#af29fb'),(17911,48587,'#e451ad'),(17912,48594,'#260eed'),(17913,48600,'#bc44e7'),(17914,48624,'#150cda'),(17915,48626,'#fbae5a'),(17916,48718,'#0efa5f'),(17917,48741,'#22e94c'),(17918,48772,'#4cccbf'),(17919,48794,'#d81a57'),(17920,48795,'#e3c45f'),(17921,48801,'#FFFFFF'),(17922,48814,'#420acf'),(17923,48843,'#45dbc5'),(17924,48844,'#5e48d1'),(17925,48847,'#e77464'),(17926,48851,'#5ba8ce'),(17927,48854,'#31d883'),(17928,48865,'#661eef'),(17929,48871,'#FFFFFF'),(17930,48872,'#a8e244'),(17931,48894,'#16e7bd'),(17932,48900,'#a3d66a'),(17933,48905,'#15abd1'),(17934,48909,'#91cf44'),(17935,48910,'#47f246'),(17936,48912,'#5eeaf2'),(17937,48913,'#d36268'),(17938,48918,'#ec7547'),(17939,48938,'#45d894'),(17940,48944,'#2197e0'),(17941,48952,'#4a55e8'),(17942,48957,'#fd3741'),(17943,48959,'#84d132'),(17944,48961,'#FFFFFF'),(17945,48971,'#FFFFFF'),(17946,48972,'#372ee6'),(17947,48975,'#e444b9'),(17948,48991,'#4da6f1'),(17949,48993,'#405ffe'),(17950,49001,'#dd5d7d'),(17951,49014,'#ec5d79'),(17952,49020,'#e81950'),(17953,49024,'#4ae72d'),(17954,49025,'#8ffc79'),(17955,49038,'#20e7a9'),(17956,49042,'#689cf0'),(17957,49048,'#1113da'),(17958,49056,'#fe68b4'),(17959,49064,'#d6163c'),(17960,49086,'#219ce5'),(17961,49094,'#ca1ed1'),(17962,49100,'#64b3ec'),(17963,49112,'#06e545'),(17964,49120,'#fd7dbe'),(17965,49124,'#2039d6'),(17966,49127,'#41cfd9'),(17967,49129,'#12cfd7'),(17968,49137,'#ab5bd3'),(17969,49138,'#472ede'),(17970,49179,'#ce29d7'),(17971,49182,'#6ce3cf'),(17972,49196,'#FFFFFF'),(17973,49204,'#f133f2'),(17974,49213,'#FFFFFF'),(17975,49214,'#FFFFFF'),(17976,49215,'#FFFFFF'),(17977,49216,'#FFFFFF'),(17978,49217,'#FFFFFF'),(17979,49218,'#FFFFFF'),(17980,49219,'#FFFFFF'),(17981,49220,'#e15b49'),(17982,49221,'#25f5b4'),(17983,49222,'#FFFFFF'),(17984,49223,'#d85da4'),(17985,49225,'#0882ed'),(17986,49226,'#FFFFFF'),(17987,49227,'#FFFFFF'),(17988,49228,'#FFFFFF'),(17989,49229,'#f1c82e'),(17990,49234,'#FFFFFF'),(17991,49236,'#FFFFFF'),(17992,49240,'#FFFFFF'),(17993,49241,'#71b1e8'),(17994,49242,'#e56ddd'),(17995,49243,'#d936d2'),(17996,49244,'#268ddf'),(17997,49246,'#FFFFFF'),(17998,49247,'#FFFFFF'),(17999,49248,'#FFFFFF'),(18000,49252,'#21d96d'),(18001,49253,'#FFFFFF'),(18002,49255,'#be3ef1'),(18003,49259,'#FFFFFF'),(18004,49261,'#FFFFFF'),(18005,49265,'#ccde66'),(18006,49266,'#0ee9a2'),(18007,49272,'#d6c839'),(18008,49274,'#d368d6'),(18009,49276,'#FFFFFF'),(18010,49277,'#FFFFFF'),(18011,49278,'#FFFFFF'),(18012,49279,'#FFFFFF'),(18013,49282,'#FFFFFF'),(18014,49283,'#FFFFFF'),(18015,49284,'#FFFFFF'),(18016,49287,'#cd00bf'),(18017,49288,'#FFFFFF'),(18018,49291,'#6cd85c'),(18019,49295,'#FFFFFF'),(18020,49296,'#FFFFFF'),(18021,49297,'#FFFFFF'),(18022,49298,'#fdc470'),(18023,49302,'#7f6fe2'),(18024,49304,'#FFFFFF'),(18025,49305,'#FFFFFF'),(18026,49306,'#57d27a'),(18027,49307,'#cf13a1'),(18028,49310,'#FFFFFF'),(18029,49312,'#FFFFFF'),(18030,49313,'#FFFFFF'),(18031,49314,'#FFFFFF'),(18032,49316,'#ea4d53'),(18033,49317,'#f36596'),(18034,49318,'#FFFFFF'),(18035,49320,'#3860e5'),(18036,49322,'#da541c'),(18037,49323,'#FFFFFF'),(18038,49324,'#FFFFFF'),(18039,49325,'#2ba1f3'),(18040,49327,'#4c00d4'),(18041,49329,'#ec6cc9'),(18042,49330,'#FFFFFF'),(18043,49331,'#e4d52c'),(18044,49336,'#FFFFFF'),(18045,49337,'#00d1f5'),(18046,49339,'#FFFFFF'),(18047,49343,'#FFFFFF'),(18048,49345,'#FFFFFF'),(18049,49346,'#dda70a'),(18050,49353,'#FFFFFF'),(18051,49355,'#FFFFFF'),(18052,49356,'#FFFFFF'),(18053,49358,'#FFFFFF'),(18054,49359,'#FFFFFF'),(18055,49366,'#FFFFFF'),(18056,49369,'#e5816a'),(18057,49370,'#6ce88d'),(18058,49371,'#0077ef'),(18059,49373,'#FFFFFF'),(18060,49374,'#FFFFFF'),(18061,49378,'#FFFFFF'),(18062,49379,'#FFFFFF'),(18063,49380,'#f1f676'),(18064,49383,'#FFFFFF'),(18065,49384,'#FFFFFF'),(18066,49385,'#FFFFFF'),(18067,49389,'#4d5ff8'),(18068,49391,'#f0ae6f'),(18069,49395,'#5911e2'),(18070,49397,'#FFFFFF'),(18071,49401,'#FFFFFF'),(18072,49403,'#b668d4'),(18073,49404,'#FFFFFF'),(18074,49405,'#e14ab8'),(18075,49407,'#d4b316'),(18076,49408,'#7a75ec'),(18077,49409,'#FFFFFF'),(18078,49412,'#FFFFFF'),(18079,49416,'#75eaef'),(18080,49419,'#FFFFFF'),(18081,49420,'#FFFFFF'),(18082,49421,'#FFFFFF'),(18083,49422,'#dc0db0'),(18084,49423,'#FFFFFF'),(18085,49424,'#FFFFFF'),(18086,49427,'#FFFFFF'),(18087,49429,'#66df92'),(18088,49430,'#97d233'),(18089,49431,'#4455dd'),(18090,49434,'#FFFFFF'),(18091,49435,'#FFFFFF'),(18092,49439,'#FFFFFF'),(18093,49440,'#FFFFFF'),(18094,49441,'#733ecf'),(18095,49444,'#f9ee23'),(18096,49445,'#FFFFFF'),(18097,49446,'#18a0d4'),(18098,49447,'#FFFFFF'),(18099,49450,'#FFFFFF'),(18100,49453,'#27d490'),(18101,49454,'#9e28d4'),(18102,49455,'#0ff7c6'),(18103,49457,'#FFFFFF'),(18104,49458,'#FFFFFF'),(18105,49459,'#3191e8'),(18106,49462,'#882bdf'),(18107,49463,'#d62af8'),(18108,49468,'#FFFFFF'),(18109,49469,'#FFFFFF'),(18110,49470,'#FFFFFF'),(18111,49472,'#FFFFFF'),(18112,49473,'#f034b0'),(18113,49474,'#b4ce62'),(18114,49477,'#FFFFFF'),(18115,49478,'#FFFFFF'),(18116,49479,'#FFFFFF'),(18117,49481,'#FFFFFF'),(18118,49484,'#FFFFFF'),(18119,49486,'#FFFFFF'),(18120,49487,'#FFFFFF'),(18121,49491,'#FFFFFF'),(18122,49492,'#FFFFFF'),(18123,49494,'#35d9b5'),(18124,49495,'#FFFFFF'),(18125,49500,'#FFFFFF'),(18126,49501,'#d71b4f'),(18127,49503,'#FFFFFF'),(18128,49504,'#FFFFFF'),(18129,49505,'#FFFFFF'),(18130,49506,'#FFFFFF'),(18131,49508,'#FFFFFF'),(18132,49509,'#FFFFFF'),(18133,49511,'#FFFFFF'),(18134,49514,'#FFFFFF'),(18135,49519,'#FFFFFF'),(18136,49521,'#2dd340'),(18137,49522,'#FFFFFF'),(18138,49523,'#FFFFFF'),(18139,49525,'#FFFFFF'),(18140,49527,'#f2ef59'),(18141,49530,'#ebcd27'),(18142,49531,'#9a5eea'),(18143,49532,'#FFFFFF'),(18144,49533,'#dd6dfc'),(18145,49534,'#FFFFFF'),(18146,49535,'#FFFFFF'),(18147,49537,'#FFFFFF'),(18148,49539,'#FFFFFF'),(18149,49544,'#FFFFFF'),(18150,49545,'#d41418'),(18151,49546,'#FFFFFF'),(18152,49547,'#FFFFFF'),(18153,49548,'#FFFFFF'),(18154,49551,'#d4834e'),(18155,49553,'#FFFFFF'),(18156,49554,'#FFFFFF'),(18157,49555,'#FFFFFF'),(18158,49557,'#FFFFFF'),(18159,49558,'#951eef'),(18160,49559,'#FFFFFF'),(18161,49563,'#FFFFFF'),(18162,49564,'#FFFFFF'),(18163,49565,'#FFFFFF'),(18164,49568,'#FFFFFF'),(18165,49569,'#FFFFFF'),(18166,49571,'#FFFFFF'),(18167,49572,'#70ed5b'),(18168,49573,'#FFFFFF'),(18169,49577,'#FFFFFF'),(18170,49578,'#ec648e'),(18171,49580,'#FFFFFF'),(18172,49581,'#FFFFFF'),(18173,49582,'#FFFFFF'),(18174,49583,'#FFFFFF'),(18175,49584,'#11cbd7'),(18176,49585,'#FFFFFF'),(18177,49587,'#FFFFFF'),(18178,49588,'#01d54e'),(18179,49590,'#FFFFFF'),(18180,49591,'#FFFFFF'),(18181,49592,'#FFFFFF'),(18182,49594,'#FFFFFF'),(18183,49596,'#4e77f6'),(18184,49598,'#e8fc63'),(18185,49602,'#8018e4'),(18186,49604,'#3c6aef'),(18187,49605,'#FFFFFF'),(18188,49608,'#FFFFFF'),(18189,49609,'#FFFFFF'),(18190,49611,'#FFFFFF'),(18191,49612,'#FFFFFF'),(18192,49613,'#FFFFFF'),(18193,49614,'#FFFFFF'),(18194,49615,'#FFFFFF'),(18195,49616,'#FFFFFF'),(18196,49617,'#6898e1'),(18197,49618,'#FFFFFF'),(18198,49619,'#e53e96'),(18199,49620,'#FFFFFF'),(18200,49622,'#915bf2'),(18201,49623,'#FFFFFF'),(18202,49627,'#d25149'),(18203,49628,'#6d3cde'),(18204,49630,'#FFFFFF'),(18205,49631,'#FFFFFF'),(18206,49632,'#38f392'),(18207,49633,'#f6806d'),(18208,49634,'#FFFFFF'),(18209,49637,'#db5850'),(18210,49638,'#27ef59'),(18211,49639,'#e16948'),(18212,49643,'#4085d2'),(18213,49644,'#42d3f7'),(18214,49646,'#060aef'),(18215,49650,'#FFFFFF'),(18216,49654,'#3a51cd'),(18217,49659,'#e6490b'),(18218,49660,'#d231f2'),(18219,49661,'#e98914'),(18220,49663,'#FFFFFF'),(18221,49665,'#FFFFFF'),(18222,49666,'#FFFFFF'),(18223,49669,'#05feeb'),(18224,49671,'#ce1dbd'),(18225,49673,'#de5fa3'),(18226,49674,'#FFFFFF'),(18227,49678,'#FFFFFF'),(18228,49680,'#e3220c'),(18229,49681,'#FFFFFF'),(18230,49682,'#49edf2'),(18231,49683,'#FFFFFF'),(18232,49688,'#FFFFFF'),(18233,49690,'#faad49'),(18234,49692,'#FFFFFF'),(18235,49695,'#40d29a'),(18236,49699,'#0099de'),(18237,49702,'#FFFFFF'),(18238,49703,'#FFFFFF'),(18239,49706,'#FFFFFF'),(18240,49709,'#FFFFFF'),(18241,49711,'#b5e001'),(18242,49712,'#FFFFFF'),(18243,49714,'#a275ef'),(18244,49716,'#FFFFFF'),(18245,49717,'#FFFFFF'),(18246,49720,'#FFFFFF'),(18247,49721,'#1e90cd'),(18248,49723,'#FFFFFF'),(18249,49724,'#589ef8'),(18250,49736,'#f7b24a'),(18251,49739,'#FFFFFF'),(18252,49741,'#d09444'),(18253,49743,'#2ff415'),(18254,49745,'#dc515c'),(18255,49746,'#ef9373'),(18256,49750,'#FFFFFF'),(18257,49762,'#FFFFFF'),(18258,49766,'#61e58b'),(18259,49768,'#FFFFFF'),(18260,49771,'#FFFFFF'),(18261,49773,'#ea6de9'),(18262,49775,'#51e61c'),(18263,49776,'#FFFFFF'),(18264,49780,'#FFFFFF'),(18265,49782,'#8ee46b'),(18266,49785,'#FFFFFF'),(18267,49786,'#a8e32b'),(18268,49789,'#FFFFFF'),(18269,49791,'#FFFFFF'),(18270,49792,'#FFFFFF'),(18271,49795,'#3eefaa'),(18272,49796,'#d18e32'),(18273,49798,'#FFFFFF'),(18274,49799,'#FFFFFF'),(18275,49801,'#6c34fe'),(18276,49803,'#aaf570'),(18277,49804,'#f600d2'),(18278,49807,'#FFFFFF'),(18279,49809,'#FFFFFF'),(18280,49810,'#d625a7'),(18281,49812,'#19e5be'),(18282,49815,'#FFFFFF'),(18283,49816,'#7c3cd7'),(18284,49820,'#ae61fd'),(18285,49823,'#FFFFFF'),(18286,49825,'#FFFFFF'),(18287,49832,'#FFFFFF'),(18288,49833,'#abef0c'),(18289,49834,'#FFFFFF'),(18290,49836,'#FFFFFF'),(18291,49839,'#FFFFFF'),(18292,49840,'#FFFFFF'),(18293,49841,'#FFFFFF'),(18294,49842,'#FFFFFF'),(18295,49843,'#e22eea'),(18296,49857,'#FFFFFF'),(18297,49864,'#e1e257'),(18298,49865,'#d04df9'),(18299,49866,'#4e23e5'),(18300,49870,'#a84ae3'),(18301,49878,'#87e11d'),(18302,49879,'#ea181d'),(18303,49881,'#cf29d1'),(18304,49886,'#9ade29'),(18305,49890,'#e69d70'),(18306,49891,'#FFFFFF'),(18307,49892,'#ee9b70'),(18308,49893,'#FFFFFF'),(18309,49894,'#fcf72a'),(18310,49895,'#8e39db'),(18311,49898,'#FFFFFF'),(18312,49902,'#d4036d'),(18313,49906,'#33d09f'),(18314,49907,'#FFFFFF'),(18315,49908,'#c8cf05'),(18316,49911,'#23e553'),(18317,49912,'#FFFFFF'),(18318,49913,'#FFFFFF'),(18319,49916,'#FFFFFF'),(18320,49919,'#FFFFFF'),(18321,49921,'#ce425c'),(18322,49922,'#e03322'),(18323,49923,'#7a94fe'),(18324,49924,'#1030d1'),(18325,49925,'#62d389'),(18326,49931,'#FFFFFF'),(18327,49932,'#235ad7'),(18328,49937,'#1dfd44'),(18329,49939,'#FFFFFF'),(18330,49941,'#cffa1a'),(18331,49942,'#bb65d8'),(18332,49943,'#fbba6e'),(18333,49946,'#FFFFFF'),(18334,49948,'#FFFFFF'),(18335,49951,'#1c01f4'),(18336,49954,'#e550d5'),(18337,49956,'#FFFFFF'),(18338,49958,'#a745fe'),(18339,49963,'#FFFFFF'),(18340,49966,'#fd5330'),(18341,49968,'#e5200d'),(18342,49971,'#8f52fb'),(18343,49972,'#28f73a'),(18344,49979,'#FFFFFF'),(18345,49982,'#FFFFFF'),(18346,49986,'#FFFFFF'),(18347,49987,'#FFFFFF'),(18348,49988,'#FFFFFF'),(18349,49989,'#FFFFFF'),(18350,49992,'#b972ec'),(18351,49993,'#FFFFFF'),(18352,49994,'#FFFFFF'),(18353,49997,'#14eb6e'),(18354,50002,'#FFFFFF'),(18355,50003,'#e614f1'),(18356,50007,'#FFFFFF'),(18357,50010,'#4069fc'),(18358,50011,'#63fa20'),(18359,50012,'#e7a25d'),(18360,50013,'#FFFFFF'),(18361,50016,'#FFFFFF'),(18362,50017,'#FFFFFF'),(18363,50020,'#FFFFFF'),(18364,50024,'#2face0'),(18365,50025,'#FFFFFF'),(18366,50027,'#41fda4'),(18367,50028,'#FFFFFF'),(18368,50029,'#bbf86f'),(18369,50030,'#FFFFFF'),(18370,50035,'#38a4e3'),(18371,50036,'#FFFFFF'),(18372,50037,'#3883f3'),(18373,50040,'#fbc50f'),(18374,50042,'#FFFFFF'),(18375,50043,'#FFFFFF'),(18376,50044,'#18d15c'),(18377,50045,'#FFFFFF'),(18378,50046,'#b4e03a'),(18379,50049,'#FFFFFF'),(18380,50054,'#FFFFFF'),(18381,50055,'#10f9a1'),(18382,50056,'#FFFFFF'),(18383,50060,'#52cd7d'),(18384,50063,'#d5586b'),(18385,50064,'#FFFFFF'),(18386,50065,'#c6fe20'),(18387,50066,'#FFFFFF'),(18388,50067,'#e04469'),(18389,50070,'#FFFFFF'),(18390,50071,'#177cfd'),(18391,50072,'#FFFFFF'),(18392,50073,'#14e413'),(18393,50074,'#fb2875'),(18394,50075,'#FFFFFF'),(18395,50076,'#FFFFFF'),(18396,50077,'#c370f8'),(18397,50078,'#4a2df1'),(18398,50080,'#fa59d1'),(18399,50082,'#6c65f4'),(18400,50083,'#4308dd'),(18401,50088,'#e53ac3'),(18402,50091,'#FFFFFF'),(18403,50094,'#dec61f'),(18404,50095,'#5eeb77'),(18405,50096,'#f32242'),(18406,50099,'#FFFFFF'),(18407,50101,'#d69029'),(18408,50104,'#4d73d8'),(18409,50105,'#e6ac47'),(18410,50106,'#fd5e3f'),(18411,50107,'#FFFFFF'),(18412,50108,'#ddac54'),(18413,50120,'#FFFFFF'),(18414,50121,'#f64c7a'),(18415,50122,'#FFFFFF'),(18416,50124,'#9527f4'),(18417,50126,'#c161d0'),(18418,50127,'#FFFFFF'),(18419,50128,'#a2e763'),(18420,50129,'#61d482'),(18421,50131,'#d50c69'),(18422,50133,'#FFFFFF'),(18423,50134,'#9c28ea'),(18424,50136,'#FFFFFF'),(18425,50139,'#0589e2'),(18426,50140,'#FFFFFF'),(18427,50145,'#FFFFFF'),(18428,50147,'#FFFFFF'),(18429,50150,'#FFFFFF'),(18430,50151,'#c5d832'),(18431,50153,'#9941ef'),(18432,50156,'#e96597'),(18433,50160,'#FFFFFF'),(18434,50162,'#479ada'),(18435,50166,'#0b31d3'),(18436,50167,'#FFFFFF'),(18437,50168,'#52d934'),(18438,50170,'#10fb41'),(18439,50172,'#dc6753'),(18440,50173,'#ee5200'),(18441,50174,'#4b77d9'),(18442,50175,'#FFFFFF'),(18443,50176,'#7d39e4'),(18444,50186,'#FFFFFF'),(18445,50187,'#fb3778'),(18446,50189,'#0ad225'),(18447,50192,'#FFFFFF'),(18448,50193,'#0f97df'),(18449,50195,'#2a57df'),(18450,50196,'#e5b215'),(18451,50197,'#FFFFFF'),(18452,50198,'#FFFFFF'),(18453,50200,'#17f282'),(18454,50203,'#5dc3e4'),(18455,50212,'#448ad3'),(18456,50213,'#d111e3'),(18457,50214,'#3f8dd3'),(18458,50215,'#FFFFFF'),(18459,50216,'#6efd60'),(18460,50219,'#FFFFFF'),(18461,50220,'#71fb9f'),(18462,50222,'#42d710'),(18463,50224,'#FFFFFF'),(18464,50225,'#FFFFFF'),(18465,50226,'#5e4cdc'),(18466,50227,'#FFFFFF'),(18467,50228,'#c827f6'),(18468,50231,'#d430e6'),(18469,50232,'#FFFFFF'),(18470,50234,'#4be3b7'),(18471,50235,'#FFFFFF'),(18472,50236,'#d875fb'),(18473,50237,'#FFFFFF'),(18474,50238,'#6d77fe'),(18475,50243,'#FFFFFF'),(18476,50244,'#a9d844'),(18477,50245,'#FFFFFF'),(18478,50249,'#d56341'),(18479,50250,'#1e0ad1'),(18480,50256,'#b843e1'),(18481,50259,'#4f5ee4'),(18482,50260,'#FFFFFF'),(18483,50261,'#ce58cd'),(18484,50263,'#FFFFFF'),(18485,50266,'#e4d65b'),(18486,50268,'#769afb'),(18487,50269,'#2ee4d3'),(18488,50271,'#eb9b28'),(18489,50272,'#da0298'),(18490,50273,'#764ee6'),(18491,50274,'#e23a6f'),(18492,50275,'#43d36e'),(18493,50282,'#1ae01b'),(18494,50288,'#dc4797'),(18495,50291,'#3081d1'),(18496,50292,'#FFFFFF'),(18497,50293,'#FFFFFF'),(18498,50296,'#FFFFFF'),(18499,50297,'#5af3d7'),(18500,50298,'#FFFFFF'),(18501,50299,'#e4242e'),(18502,50300,'#6ad3e0'),(18503,50303,'#d62758'),(18504,50307,'#FFFFFF'),(18505,50309,'#FFFFFF'),(18506,50314,'#db6d56'),(18507,50315,'#FFFFFF'),(18508,50316,'#4beabd'),(18509,50320,'#83d409'),(18510,50321,'#FFFFFF'),(18511,50322,'#e7448b'),(18512,50323,'#FFFFFF'),(18513,50324,'#FFFFFF'),(18514,50327,'#f912e0'),(18515,50329,'#e44de5'),(18516,50330,'#dcdb1f'),(18517,50332,'#FFFFFF'),(18518,50333,'#f1e775'),(18519,50334,'#FFFFFF'),(18520,50336,'#FFFFFF'),(18521,50338,'#7bfc9b'),(18522,50339,'#9d4cf8'),(18523,50340,'#ec297f'),(18524,50341,'#02fe30'),(18525,50346,'#39aed7'),(18526,50347,'#FFFFFF'),(18527,50352,'#FFFFFF'),(18528,50353,'#aed433'),(18529,50358,'#FFFFFF'),(18530,50359,'#FFFFFF'),(18531,50360,'#a7f44b'),(18532,50364,'#de00be'),(18533,50365,'#FFFFFF'),(18534,50366,'#0d01cf'),(18535,50368,'#1047f1'),(18536,50371,'#FFFFFF'),(18537,50374,'#419ad2'),(18538,50375,'#FFFFFF'),(18539,50379,'#2e8ee7'),(18540,50386,'#b70fdf'),(18541,50388,'#b0fd39'),(18542,50389,'#5681e4'),(18543,50391,'#FFFFFF'),(18544,50392,'#dc54d2'),(18545,50393,'#ce1502'),(18546,50394,'#0479e0'),(18547,50395,'#ebd800'),(18548,50397,'#4aa1f2'),(18549,50399,'#9c4eda'),(18550,50400,'#76d914'),(18551,50404,'#71e4a2'),(18552,50405,'#e9f636'),(18553,50406,'#ef0336'),(18554,50408,'#19fdab'),(18555,50410,'#09e6e2'),(18556,50412,'#FFFFFF'),(18557,50414,'#6885d1'),(18558,50416,'#1fe0dd'),(18559,50419,'#FFFFFF'),(18560,50421,'#FFFFFF'),(18561,50424,'#FFFFFF'),(18562,50429,'#FFFFFF'),(18563,50431,'#8748f7'),(18564,50432,'#e1ec26'),(18565,50436,'#50fb0a'),(18566,50449,'#9c50e1'),(18567,50452,'#b1ed41'),(18568,50456,'#50e18b'),(18569,50457,'#b7f228'),(18570,50465,'#cfee02'),(18571,50472,'#FFFFFF'),(18572,50481,'#dd028b'),(18573,50483,'#FFFFFF'),(18574,50489,'#b8f51f'),(18575,50490,'#FFFFFF'),(18576,50491,'#074cea'),(18577,50493,'#6643cf'),(18578,50498,'#29e5fe'),(18579,50503,'#ebce22'),(18580,50505,'#7dd864'),(18581,50512,'#FFFFFF'),(18582,50515,'#f96fe2'),(18583,50516,'#FFFFFF'),(18584,50517,'#FFFFFF'),(18585,50523,'#6b02ec'),(18586,50528,'#FFFFFF'),(18587,50530,'#478fe8'),(18588,50531,'#f84478'),(18589,50532,'#FFFFFF'),(18590,50533,'#196ece'),(18591,50536,'#FFFFFF'),(18592,50541,'#f2846a'),(18593,50542,'#2b5acf'),(18594,50550,'#3ff53f'),(18595,50551,'#FFFFFF'),(18596,50559,'#dd3cbf'),(18597,50560,'#f1ac69'),(18598,50568,'#15daf2'),(18599,50571,'#d3a75d'),(18600,50572,'#f9f769'),(18601,50573,'#f0474e'),(18602,50579,'#c623e7'),(18603,50585,'#4e1dd3'),(18604,50591,'#d0ce55'),(18605,50592,'#2902e3'),(18606,50594,'#04f89d'),(18607,50595,'#45ed4f'),(18608,50599,'#46e9cc'),(18609,50604,'#FFFFFF'),(18610,50609,'#33da21'),(18611,50610,'#1eedf6'),(18612,50612,'#686beb'),(18613,50617,'#FFFFFF'),(18614,50618,'#FFFFFF'),(18615,50622,'#d28900'),(18616,50625,'#1c40ec'),(18617,50626,'#2505cd'),(18618,50632,'#346cd2'),(18619,50633,'#e30c4f'),(18620,50634,'#67dc67'),(18621,50636,'#FFFFFF'),(18622,50638,'#6ecf33'),(18623,50639,'#FFFFFF'),(18624,50644,'#d3e447'),(18625,50678,'#3ed80f'),(18626,50689,'#ea653a'),(18627,50692,'#ce7f65'),(18628,50701,'#14cdb8'),(18629,50707,'#FFFFFF'),(18630,50709,'#cd1a31'),(18631,50710,'#e6d76e'),(18632,50711,'#ecf50e'),(18633,50713,'#3ff9e6'),(18634,50716,'#cf545c'),(18635,50720,'#cf9039'),(18636,50721,'#02d566'),(18637,50730,'#FFFFFF'),(18638,50731,'#e0d81a'),(18639,50743,'#f48f74'),(18640,50744,'#4239e1'),(18641,50750,'#FFFFFF'),(18642,50771,'#4adccd'),(18643,50796,'#3271ee'),(18644,50798,'#f23b5c'),(18645,50810,'#49d296'),(18646,50811,'#984feb'),(18647,50816,'#c92cec'),(18648,50829,'#0ce6af'),(18649,50831,'#eb5be8'),(18650,50834,'#18f9a9'),(18651,50855,'#0fe0a2'),(18652,50874,'#794bcd'),(18653,50879,'#FFFFFF'),(18654,50880,'#FFFFFF'),(18655,50881,'#f9ae4f'),(18656,50884,'#e3a62f'),(18657,50894,'#c6d93b'),(18658,50901,'#4dfcd5'),(18659,50902,'#FFFFFF'),(18660,50905,'#1ee1f9'),(18661,50909,'#5c51f1'),(18662,50922,'#92e861'),(18663,50926,'#9de951'),(18664,50927,'#73e979'),(18665,50928,'#FFFFFF'),(18666,50929,'#FFFFFF'),(18667,50931,'#FFFFFF'),(18668,50933,'#FFFFFF'),(18669,50934,'#6ca9df'),(18670,50937,'#FFFFFF'),(18671,50938,'#FFFFFF'),(18672,50941,'#FFFFFF'),(18673,50942,'#FFFFFF'),(18674,50943,'#FFFFFF'),(18675,50945,'#FFFFFF'),(18676,50946,'#FFFFFF'),(18677,50947,'#FFFFFF'),(18678,50948,'#d4154b'),(18679,50949,'#FFFFFF'),(18680,50950,'#FFFFFF'),(18681,50951,'#FFFFFF'),(18682,50954,'#FFFFFF'),(18683,50955,'#FFFFFF'),(18684,50956,'#FFFFFF'),(18685,50958,'#FFFFFF'),(18686,50959,'#FFFFFF'),(18687,50960,'#FFFFFF'),(18688,50961,'#e3dd26'),(18689,50962,'#d1e933'),(18690,50964,'#FFFFFF'),(18691,50965,'#FFFFFF'),(18692,50966,'#FFFFFF'),(18693,50967,'#FFFFFF'),(18694,50968,'#FFFFFF'),(18695,50972,'#FFFFFF'),(18696,50973,'#5499d1'),(18697,50974,'#FFFFFF'),(18698,50975,'#FFFFFF'),(18699,50982,'#e6ac41'),(18700,50983,'#45c7de'),(18701,50984,'#14fe79'),(18702,50985,'#FFFFFF'),(18703,50986,'#FFFFFF'),(18704,50987,'#FFFFFF'),(18705,50989,'#FFFFFF'),(18706,50991,'#FFFFFF'),(18707,50992,'#FFFFFF'),(18708,50993,'#FFFFFF'),(18709,50994,'#FFFFFF'),(18710,50998,'#FFFFFF'),(18711,50999,'#FFFFFF'),(18712,51000,'#FFFFFF'),(18713,51004,'#ddb96d'),(18714,51010,'#FFFFFF'),(18715,51012,'#FFFFFF'),(18716,51016,'#FFFFFF'),(18717,51017,'#FFFFFF'),(18718,51018,'#FFFFFF'),(18719,51019,'#FFFFFF'),(18720,51023,'#FFFFFF'),(18721,51024,'#FFFFFF'),(18722,51025,'#FFFFFF'),(18723,51026,'#FFFFFF'),(18724,51031,'#FFFFFF'),(18725,51032,'#FFFFFF'),(18726,51033,'#FFFFFF'),(18727,51034,'#FFFFFF'),(18728,51041,'#FFFFFF'),(18729,51042,'#FFFFFF'),(18730,51043,'#FFFFFF'),(18731,51044,'#FFFFFF'),(18732,51045,'#FFFFFF'),(18733,51047,'#FFFFFF'),(18734,51048,'#FFFFFF'),(18735,51049,'#FFFFFF'),(18736,51050,'#FFFFFF'),(18737,51051,'#FFFFFF'),(18738,51053,'#FFFFFF'),(18739,51054,'#FFFFFF'),(18740,51055,'#FFFFFF'),(18741,51059,'#FFFFFF'),(18742,51060,'#FFFFFF'),(18743,51064,'#FFFFFF'),(18744,51066,'#FFFFFF'),(18745,51067,'#b7d128'),(18746,51068,'#FFFFFF'),(18747,51070,'#FFFFFF'),(18748,51071,'#FFFFFF'),(18749,51072,'#FFFFFF'),(18750,51073,'#b8fa0b'),(18751,51075,'#FFFFFF'),(18752,51077,'#FFFFFF'),(18753,51079,'#FFFFFF'),(18754,51081,'#FFFFFF'),(18755,51082,'#FFFFFF'),(18756,51083,'#FFFFFF'),(18757,51086,'#FFFFFF'),(18758,51090,'#FFFFFF'),(18759,51093,'#FFFFFF'),(18760,51094,'#FFFFFF'),(18761,51095,'#FFFFFF'),(18762,51096,'#FFFFFF'),(18763,51098,'#FFFFFF'),(18764,51099,'#FFFFFF'),(18765,51100,'#FFFFFF'),(18766,51101,'#FFFFFF'),(18767,51102,'#FFFFFF'),(18768,51104,'#FFFFFF'),(18769,51105,'#FFFFFF'),(18770,51108,'#FFFFFF'),(18771,51109,'#FFFFFF'),(18772,51111,'#FFFFFF'),(18773,51112,'#FFFFFF'),(18774,51113,'#FFFFFF'),(18775,51114,'#cd44cc'),(18776,51115,'#53e75a'),(18777,51117,'#FFFFFF'),(18778,51118,'#FFFFFF'),(18779,51120,'#FFFFFF'),(18780,51121,'#FFFFFF'),(18781,51124,'#FFFFFF'),(18782,51125,'#FFFFFF'),(18783,51126,'#e845a0'),(18784,51127,'#FFFFFF'),(18785,51130,'#FFFFFF'),(18786,51131,'#FFFFFF'),(18787,51133,'#FFFFFF'),(18788,51134,'#FFFFFF'),(18789,51135,'#FFFFFF'),(18790,51136,'#d9f963'),(18791,51137,'#FFFFFF'),(18792,51138,'#FFFFFF'),(18793,51139,'#FFFFFF'),(18794,51141,'#FFFFFF'),(18795,51142,'#FFFFFF'),(18796,51143,'#FFFFFF'),(18797,51145,'#FFFFFF'),(18798,51147,'#FFFFFF'),(18799,51148,'#FFFFFF'),(18800,51150,'#FFFFFF'),(18801,51152,'#FFFFFF'),(18802,51154,'#FFFFFF'),(18803,51155,'#FFFFFF'),(18804,51156,'#FFFFFF'),(18805,51157,'#FFFFFF'),(18806,51158,'#FFFFFF'),(18807,51160,'#FFFFFF'),(18808,51161,'#FFFFFF'),(18809,51162,'#FFFFFF'),(18810,51163,'#FFFFFF'),(18811,51164,'#FFFFFF'),(18812,51166,'#FFFFFF'),(18813,51167,'#FFFFFF'),(18814,51169,'#FFFFFF'),(18815,51170,'#FFFFFF'),(18816,51171,'#FFFFFF'),(18817,51175,'#FFFFFF'),(18818,51178,'#FFFFFF'),(18819,51179,'#FFFFFF'),(18820,51180,'#FFFFFF'),(18821,51181,'#FFFFFF'),(18822,51182,'#FFFFFF'),(18823,51186,'#FFFFFF'),(18824,51188,'#FFFFFF'),(18825,51189,'#FFFFFF'),(18826,51191,'#FFFFFF'),(18827,51195,'#FFFFFF'),(18828,51197,'#FFFFFF'),(18829,51198,'#FFFFFF'),(18830,51199,'#FFFFFF'),(18831,51200,'#FFFFFF'),(18832,51201,'#FFFFFF'),(18833,51202,'#FFFFFF'),(18834,51205,'#FFFFFF'),(18835,51206,'#FFFFFF'),(18836,51207,'#FFFFFF'),(18837,51210,'#b062d3'),(18838,51212,'#fa7c0a'),(18839,51214,'#FFFFFF'),(18840,51215,'#FFFFFF'),(18841,51216,'#FFFFFF'),(18842,51218,'#FFFFFF'),(18843,51219,'#FFFFFF'),(18844,51220,'#9a24d9'),(18845,51221,'#4b9ecc'),(18846,51224,'#FFFFFF'),(18847,51226,'#FFFFFF'),(18848,51229,'#ce6e48'),(18849,51230,'#FFFFFF'),(18850,51231,'#FFFFFF'),(18851,51232,'#FFFFFF'),(18852,51234,'#FFFFFF'),(18853,51235,'#46ef6c'),(18854,51238,'#07add9'),(18855,51239,'#FFFFFF'),(18856,51242,'#FFFFFF'),(18857,51244,'#3ce687'),(18858,51245,'#ef9e71'),(18859,51246,'#FFFFFF'),(18860,51247,'#FFFFFF'),(18861,51248,'#FFFFFF'),(18862,51250,'#FFFFFF'),(18863,51260,'#FFFFFF'),(18864,51261,'#FFFFFF'),(18865,51262,'#FFFFFF'),(18866,51263,'#FFFFFF'),(18867,51264,'#26e206'),(18868,51266,'#FFFFFF'),(18869,51267,'#FFFFFF'),(18870,51268,'#FFFFFF'),(18871,51269,'#FFFFFF'),(18872,51271,'#FFFFFF'),(18873,51273,'#FFFFFF'),(18874,51274,'#FFFFFF'),(18875,51275,'#FFFFFF'),(18876,51278,'#f89f65'),(18877,51280,'#FFFFFF'),(18878,51281,'#FFFFFF'),(18879,51282,'#FFFFFF'),(18880,51284,'#cd08bb'),(18881,51285,'#FFFFFF'),(18882,51286,'#FFFFFF'),(18883,51288,'#58d771'),(18884,51289,'#FFFFFF'),(18885,51290,'#FFFFFF'),(18886,51291,'#FFFFFF'),(18887,51292,'#FFFFFF'),(18888,51293,'#FFFFFF'),(18889,51295,'#FFFFFF'),(18890,51297,'#FFFFFF'),(18891,51300,'#24eae9'),(18892,51303,'#24d056'),(18893,51304,'#FFFFFF'),(18894,51306,'#FFFFFF'),(18895,51307,'#FFFFFF'),(18896,51311,'#FFFFFF'),(18897,51313,'#FFFFFF'),(18898,51315,'#FFFFFF'),(18899,51316,'#FFFFFF'),(18900,51318,'#FFFFFF'),(18901,51319,'#FFFFFF'),(18902,51321,'#FFFFFF'),(18903,51322,'#56a3db'),(18904,51323,'#FFFFFF'),(18905,51324,'#FFFFFF'),(18906,51325,'#FFFFFF'),(18907,51326,'#FFFFFF'),(18908,51328,'#FFFFFF'),(18909,51334,'#e1c957'),(18910,51336,'#26cfad'),(18911,51337,'#7029d0'),(18912,51340,'#FFFFFF'),(18913,51341,'#FFFFFF'),(18914,51342,'#74d913'),(18915,51343,'#FFFFFF'),(18916,51345,'#FFFFFF'),(18917,51346,'#f81ffa'),(18918,51348,'#FFFFFF'),(18919,51350,'#a9d113'),(18920,51354,'#FFFFFF'),(18921,51358,'#FFFFFF'),(18922,51359,'#FFFFFF'),(18923,51360,'#FFFFFF'),(18924,51362,'#42de5e'),(18925,51364,'#FFFFFF'),(18926,51366,'#FFFFFF'),(18927,51368,'#dd364e'),(18928,51370,'#FFFFFF'),(18929,51372,'#f44ac4'),(18930,51374,'#FFFFFF'),(18931,51375,'#FFFFFF'),(18932,51376,'#FFFFFF'),(18933,51377,'#FFFFFF'),(18934,51378,'#FFFFFF'),(18935,51379,'#FFFFFF'),(18936,51380,'#FFFFFF'),(18937,51381,'#FFFFFF'),(18938,51382,'#FFFFFF'),(18939,51383,'#FFFFFF'),(18940,51385,'#FFFFFF'),(18941,51388,'#63ce60'),(18942,51389,'#FFFFFF'),(18943,51390,'#a64ad5'),(18944,51393,'#FFFFFF'),(18945,51394,'#FFFFFF'),(18946,51396,'#FFFFFF'),(18947,51401,'#FFFFFF'),(18948,51402,'#FFFFFF'),(18949,51403,'#FFFFFF'),(18950,51404,'#FFFFFF'),(18951,51406,'#FFFFFF'),(18952,51407,'#FFFFFF'),(18953,51408,'#b249d0'),(18954,51412,'#FFFFFF'),(18955,51413,'#FFFFFF'),(18956,51414,'#f721ec'),(18957,51415,'#FFFFFF'),(18958,51417,'#FFFFFF'),(18959,51418,'#FFFFFF'),(18960,51419,'#FFFFFF'),(18961,51421,'#FFFFFF'),(18962,51424,'#FFFFFF'),(18963,51425,'#FFFFFF'),(18964,51427,'#FFFFFF'),(18965,51428,'#fa5617'),(18966,51429,'#FFFFFF'),(18967,51430,'#449bdc'),(18968,51431,'#6de54c'),(18969,51432,'#FFFFFF'),(18970,51433,'#eaac3f'),(18971,51434,'#b40cdc'),(18972,51437,'#ed4be9'),(18973,51438,'#5f93dc'),(18974,51441,'#FFFFFF'),(18975,51444,'#e03a40'),(18976,51445,'#FFFFFF'),(18977,51446,'#cd50d8'),(18978,51447,'#FFFFFF'),(18979,51450,'#FFFFFF'),(18980,51451,'#FFFFFF'),(18981,51454,'#FFFFFF'),(18982,51455,'#FFFFFF'),(18983,51456,'#FFFFFF'),(18984,51457,'#FFFFFF'),(18985,51458,'#FFFFFF'),(18986,51460,'#FFFFFF'),(18987,51462,'#be6fe8'),(18988,51463,'#FFFFFF'),(18989,51464,'#FFFFFF'),(18990,51465,'#FFFFFF'),(18991,51467,'#FFFFFF'),(18992,51470,'#d34643'),(18993,51471,'#FFFFFF'),(18994,51472,'#FFFFFF'),(18995,51473,'#FFFFFF'),(18996,51474,'#FFFFFF'),(18997,51475,'#FFFFFF'),(18998,51479,'#FFFFFF'),(18999,51482,'#65e29e'),(19000,51483,'#FFFFFF'),(19001,51485,'#FFFFFF'),(19002,51489,'#FFFFFF'),(19003,51490,'#09e4eb'),(19004,51491,'#FFFFFF'),(19005,51492,'#FFFFFF'),(19006,51493,'#FFFFFF'),(19007,51495,'#7cee26'),(19008,51496,'#FFFFFF'),(19009,51497,'#FFFFFF'),(19010,51498,'#FFFFFF'),(19011,51499,'#028af2'),(19012,51500,'#f2f210'),(19013,51501,'#41f6bb'),(19014,51502,'#FFFFFF'),(19015,51503,'#FFFFFF'),(19016,51504,'#FFFFFF'),(19017,51505,'#FFFFFF'),(19018,51506,'#d05766'),(19019,51507,'#abf467'),(19020,51509,'#FFFFFF'),(19021,51510,'#1a33d2'),(19022,51511,'#0be677'),(19023,51512,'#FFFFFF'),(19024,51514,'#FFFFFF'),(19025,51517,'#FFFFFF'),(19026,51518,'#FFFFFF'),(19027,51520,'#32acec'),(19028,51521,'#FFFFFF'),(19029,51523,'#FFFFFF'),(19030,51526,'#b427d6'),(19031,51528,'#FFFFFF'),(19032,51530,'#FFFFFF'),(19033,51532,'#acda32'),(19034,51533,'#FFFFFF'),(19035,51534,'#FFFFFF'),(19036,51535,'#FFFFFF'),(19037,51539,'#FFFFFF'),(19038,51540,'#FFFFFF'),(19039,51543,'#FFFFFF'),(19040,51544,'#FFFFFF'),(19041,51545,'#25a3cf'),(19042,51546,'#FFFFFF'),(19043,51547,'#FFFFFF'),(19044,51548,'#FFFFFF'),(19045,51549,'#FFFFFF'),(19046,51551,'#db7f34'),(19047,51552,'#7318d9'),(19048,51553,'#FFFFFF'),(19049,51556,'#7bf7ad'),(19050,51560,'#5fd400'),(19051,51563,'#fb249d'),(19052,51565,'#FFFFFF'),(19053,51567,'#d11263'),(19054,51568,'#0fd84e'),(19055,51569,'#20eec1'),(19056,51570,'#FFFFFF'),(19057,51573,'#47ddbb'),(19058,51575,'#FFFFFF'),(19059,51576,'#FFFFFF'),(19060,51579,'#FFFFFF'),(19061,51581,'#FFFFFF'),(19062,51582,'#cb56f7'),(19063,51583,'#FFFFFF'),(19064,51584,'#cfcd56'),(19065,51585,'#FFFFFF'),(19066,51590,'#FFFFFF'),(19067,51593,'#FFFFFF'),(19068,51597,'#dcf97b'),(19069,51598,'#e651da'),(19070,51601,'#FFFFFF'),(19071,51602,'#FFFFFF'),(19072,51603,'#FFFFFF'),(19073,51604,'#18e424'),(19074,51606,'#FFFFFF'),(19075,51608,'#FFFFFF'),(19076,51609,'#FFFFFF'),(19077,51611,'#cc52d8'),(19078,51612,'#FFFFFF'),(19079,51613,'#d10046'),(19080,51614,'#FFFFFF'),(19081,51615,'#FFFFFF'),(19082,51617,'#FFFFFF'),(19083,51618,'#FFFFFF'),(19084,51621,'#FFFFFF'),(19085,51623,'#ef8a6f'),(19086,51626,'#d638db'),(19087,51627,'#FFFFFF'),(19088,51628,'#FFFFFF'),(19089,51629,'#FFFFFF'),(19090,51631,'#e863a6'),(19091,51633,'#ee379f'),(19092,51634,'#FFFFFF'),(19093,51636,'#FFFFFF'),(19094,51638,'#FFFFFF'),(19095,51640,'#FFFFFF'),(19096,51641,'#FFFFFF'),(19097,51643,'#25d783'),(19098,51644,'#FFFFFF'),(19099,51645,'#FFFFFF'),(19100,51646,'#0a2ddd'),(19101,51647,'#FFFFFF'),(19102,51648,'#FFFFFF'),(19103,51650,'#FFFFFF'),(19104,51652,'#FFFFFF'),(19105,51653,'#38f921'),(19106,51654,'#64cd6e'),(19107,51655,'#FFFFFF'),(19108,51656,'#52e2bd'),(19109,51657,'#13d29a'),(19110,51658,'#FFFFFF'),(19111,51661,'#cbed22'),(19112,51663,'#d06364'),(19113,51664,'#FFFFFF'),(19114,51667,'#FFFFFF'),(19115,51672,'#FFFFFF'),(19116,51674,'#FFFFFF'),(19117,51675,'#FFFFFF'),(19118,51676,'#FFFFFF'),(19119,51677,'#FFFFFF'),(19120,51679,'#FFFFFF'),(19121,51680,'#fa4a49'),(19122,51681,'#FFFFFF'),(19123,51682,'#cd488a'),(19124,51685,'#FFFFFF'),(19125,51688,'#eaee45'),(19126,51689,'#15dabb'),(19127,51692,'#FFFFFF'),(19128,51694,'#d7581e'),(19129,51696,'#FFFFFF'),(19130,51697,'#FFFFFF'),(19131,51698,'#FFFFFF'),(19132,51700,'#FFFFFF'),(19133,51702,'#2651f2'),(19134,51703,'#2411dc'),(19135,51704,'#FFFFFF'),(19136,51706,'#0a53df'),(19137,51712,'#73fca5'),(19138,51713,'#FFFFFF'),(19139,51714,'#14d79b'),(19140,51715,'#FFFFFF'),(19141,51716,'#FFFFFF'),(19142,51717,'#FFFFFF'),(19143,51718,'#FFFFFF'),(19144,51719,'#FFFFFF'),(19145,51720,'#FFFFFF'),(19146,51721,'#06d853'),(19147,51723,'#f13830'),(19148,51724,'#54d47f'),(19149,51725,'#FFFFFF'),(19150,51727,'#FFFFFF'),(19151,51729,'#55d643'),(19152,51730,'#d52113'),(19153,51731,'#FFFFFF'),(19154,51733,'#2247fe'),(19155,51735,'#07dbd4'),(19156,51736,'#FFFFFF'),(19157,51737,'#FFFFFF'),(19158,51738,'#FFFFFF'),(19159,51739,'#d77069'),(19160,51740,'#FFFFFF'),(19161,51741,'#FFFFFF'),(19162,51743,'#FFFFFF'),(19163,51744,'#4736f2'),(19164,51745,'#FFFFFF'),(19165,51746,'#0aa2e4'),(19166,51748,'#2f26f6'),(19167,51749,'#613bec'),(19168,51750,'#FFFFFF'),(19169,51753,'#FFFFFF'),(19170,51755,'#6cf834'),(19171,51757,'#d9055d'),(19172,51758,'#FFFFFF'),(19173,51759,'#FFFFFF'),(19174,51760,'#FFFFFF'),(19175,51762,'#FFFFFF'),(19176,51763,'#FFFFFF'),(19177,51767,'#db144d'),(19178,51768,'#3091e2'),(19179,51771,'#22c6fe'),(19180,51772,'#2d40dd'),(19181,51773,'#FFFFFF'),(19182,51774,'#FFFFFF'),(19183,51778,'#FFFFFF'),(19184,51781,'#FFFFFF'),(19185,51782,'#FFFFFF'),(19186,51783,'#1a3def'),(19187,51784,'#30b1e9'),(19188,51787,'#FFFFFF'),(19189,51788,'#FFFFFF'),(19190,51791,'#FFFFFF'),(19191,51794,'#FFFFFF'),(19192,51795,'#FFFFFF'),(19193,51796,'#60d3bf'),(19194,51801,'#FFFFFF'),(19195,51802,'#FFFFFF'),(19196,51803,'#374bf8'),(19197,51805,'#FFFFFF'),(19198,51806,'#FFFFFF'),(19199,51807,'#FFFFFF'),(19200,51808,'#5bbfd3'),(19201,51809,'#FFFFFF'),(19202,51813,'#FFFFFF'),(19203,51815,'#e470a2'),(19204,51852,'#3fded5'),(19205,51859,'#335fee'),(19206,51870,'#5659df'),(19207,51886,'#6beb8b'),(19208,51904,'#27dea9'),(19209,51917,'#d5158d'),(19210,51934,'#ea8f32'),(19211,51946,'#f62b00'),(19212,51948,'#6cec45'),(19213,51952,'#FFFFFF'),(19214,51953,'#FFFFFF'),(19215,51955,'#FFFFFF'),(19216,51959,'#368ce4'),(19217,51960,'#ec2477'),(19218,51962,'#3fdeb6'),(19219,51963,'#5eb9eb'),(19220,51964,'#FFFFFF'),(19221,51966,'#7373fa'),(19222,51968,'#72f7ac'),(19223,51971,'#FFFFFF'),(19224,51974,'#FFFFFF'),(19225,51975,'#6262d2'),(19226,51979,'#FFFFFF'),(19227,51981,'#FFFFFF'),(19228,51982,'#FFFFFF'),(19229,51984,'#b641e4'),(19230,51986,'#FFFFFF'),(19231,51992,'#71f795'),(19232,51993,'#f06d7e'),(19233,51994,'#b6df3a'),(19234,51995,'#FFFFFF'),(19235,51996,'#7ad12c'),(19236,51997,'#ed744a'),(19237,51998,'#FFFFFF'),(19238,51999,'#FFFFFF'),(19239,52001,'#03eba1'),(19240,52003,'#FFFFFF'),(19241,52004,'#FFFFFF'),(19242,52006,'#FFFFFF'),(19243,52007,'#FFFFFF'),(19244,52010,'#9201f2'),(19245,52015,'#ed1ea8'),(19246,52016,'#9544e8'),(19247,52018,'#5dd2eb'),(19248,52019,'#FFFFFF'),(19249,52021,'#6fd41b'),(19250,52022,'#ee49e6'),(19251,52023,'#2ef9b9'),(19252,52025,'#FFFFFF'),(19253,52028,'#FFFFFF'),(19254,52029,'#2ea0db'),(19255,52031,'#fc4ed9'),(19256,52032,'#d7f55c'),(19257,52036,'#FFFFFF'),(19258,52037,'#FFFFFF'),(19259,52038,'#FFFFFF'),(19260,52039,'#d2168f'),(19261,52043,'#f830ce'),(19262,52046,'#25c5f3'),(19263,52050,'#FFFFFF'),(19264,52051,'#FFFFFF'),(19265,52052,'#FFFFFF'),(19266,52056,'#cf00c7'),(19267,52058,'#FFFFFF'),(19268,52060,'#8539da'),(19269,52061,'#FFFFFF'),(19270,52063,'#e8f025'),(19271,52077,'#f6ac11'),(19272,52086,'#FFFFFF'),(19273,52087,'#aade34'),(19274,52091,'#FFFFFF'),(19275,52092,'#FFFFFF'),(19276,52093,'#49a4ec'),(19277,52096,'#FFFFFF'),(19278,52097,'#FFFFFF'),(19279,52100,'#1463f2'),(19280,52102,'#FFFFFF'),(19281,52104,'#FFFFFF'),(19282,52108,'#91d65e'),(19283,52110,'#FFFFFF'),(19284,52111,'#e0518c'),(19285,52113,'#f5b279'),(19286,52116,'#FFFFFF'),(19287,52119,'#c8e245'),(19288,52120,'#ea4c6c'),(19289,52124,'#FFFFFF'),(19290,52126,'#FFFFFF'),(19291,52127,'#37e621'),(19292,52128,'#bfea31'),(19293,52130,'#FFFFFF'),(19294,52132,'#66b9f5'),(19295,52133,'#FFFFFF'),(19296,52134,'#FFFFFF'),(19297,52137,'#FFFFFF'),(19298,52138,'#63da7d'),(19299,52143,'#4af86f'),(19300,52146,'#FFFFFF'),(19301,52147,'#7de46d'),(19302,52149,'#FFFFFF'),(19303,52152,'#FFFFFF'),(19304,52154,'#9c1bd7'),(19305,52157,'#FFFFFF'),(19306,52158,'#FFFFFF'),(19307,52159,'#FFFFFF'),(19308,52161,'#faa203'),(19309,52162,'#FFFFFF'),(19310,52163,'#FFFFFF'),(19311,52166,'#4a23d8'),(19312,52173,'#FFFFFF'),(19313,52174,'#FFFFFF'),(19314,52175,'#FFFFFF'),(19315,52180,'#cd3f3b'),(19316,52183,'#FFFFFF'),(19317,52184,'#114cd0'),(19318,52186,'#9323ce'),(19319,52188,'#FFFFFF'),(19320,52191,'#FFFFFF'),(19321,52202,'#FFFFFF'),(19322,52205,'#FFFFFF'),(19323,52208,'#cde662'),(19324,52210,'#e23e37'),(19325,52217,'#ccb436'),(19326,52228,'#0cd6c1'),(19327,52234,'#FFFFFF'),(19328,52236,'#FFFFFF'),(19329,52245,'#75f0db'),(19330,52253,'#9928e2'),(19331,52254,'#fed35c'),(19332,52256,'#FFFFFF'),(19333,52258,'#FFFFFF'),(19334,52260,'#FFFFFF'),(19335,52262,'#FFFFFF'),(19336,52266,'#34e0c8'),(19337,52270,'#d29019'),(19338,52274,'#FFFFFF'),(19339,52276,'#FFFFFF'),(19340,52277,'#f053e9'),(19341,52282,'#71e6d4'),(19342,52295,'#ef0bf5'),(19343,52297,'#4d08cc'),(19344,52302,'#2fcee5'),(19345,52305,'#ed5dc1'),(19346,52312,'#40fde6'),(19347,52317,'#e74ce2'),(19348,52321,'#3eef38'),(19349,52332,'#e457a5'),(19350,52333,'#4ce98b'),(19351,52336,'#FFFFFF'),(19352,52337,'#FFFFFF'),(19353,52340,'#a3d25a'),(19354,52346,'#8bfb2b'),(19355,52351,'#FFFFFF'),(19356,52354,'#FFFFFF'),(19357,52357,'#FFFFFF'),(19358,52359,'#65f6ef'),(19359,52362,'#dbb65e'),(19360,52367,'#59ee24'),(19361,52368,'#fa5f27'),(19362,52377,'#e6806a'),(19363,52381,'#3560db'),(19364,52387,'#cf6023'),(19365,52396,'#f6d567'),(19366,52407,'#fdeb78'),(19367,52409,'#10d2f0'),(19368,52414,'#FFFFFF'),(19369,52416,'#e69903'),(19370,52432,'#FFFFFF'),(19371,52435,'#9c09f9'),(19372,52439,'#32d270'),(19373,52447,'#98f412'),(19374,52448,'#48e777'),(19375,52450,'#FFFFFF'),(19376,52451,'#FFFFFF'),(19377,52459,'#ec5773'),(19378,52462,'#FFFFFF'),(19379,52464,'#FFFFFF'),(19380,52465,'#49d077'),(19381,52469,'#FFFFFF'),(19382,52471,'#FFFFFF'),(19383,52472,'#FFFFFF'),(19384,52475,'#FFFFFF'),(19385,52477,'#d5a662'),(19386,52478,'#FFFFFF'),(19387,52480,'#FFFFFF'),(19388,52481,'#FFFFFF'),(19389,52488,'#FFFFFF'),(19390,52489,'#FFFFFF'),(19391,52490,'#FFFFFF'),(19392,52493,'#FFFFFF'),(19393,52495,'#FFFFFF'),(19394,52496,'#FFFFFF'),(19395,52497,'#FFFFFF'),(19396,52498,'#84eb6d'),(19397,52500,'#f7fa4e'),(19398,52501,'#88f20c'),(19399,52508,'#FFFFFF'),(19400,52511,'#FFFFFF'),(19401,52512,'#FFFFFF'),(19402,52513,'#f9c803'),(19403,52518,'#a9de6e'),(19404,52520,'#FFFFFF'),(19405,52521,'#FFFFFF'),(19406,52524,'#FFFFFF'),(19407,52525,'#FFFFFF'),(19408,52526,'#FFFFFF'),(19409,52527,'#d257d3'),(19410,52528,'#de7a3b'),(19411,52531,'#36de7c'),(19412,52533,'#FFFFFF'),(19413,52535,'#cfbc3c'),(19414,52536,'#25ec88'),(19415,52537,'#f50a5f'),(19416,52539,'#FFFFFF'),(19417,52541,'#FFFFFF'),(19418,52542,'#FFFFFF'),(19419,52543,'#FFFFFF'),(19420,52545,'#FFFFFF'),(19421,52548,'#FFFFFF'),(19422,52550,'#FFFFFF'),(19423,52555,'#FFFFFF'),(19424,52557,'#FFFFFF'),(19425,52564,'#FFFFFF'),(19426,52565,'#e0358e'),(19427,52568,'#FFFFFF'),(19428,52571,'#FFFFFF'),(19429,52575,'#cc5b74'),(19430,52576,'#befb6f'),(19431,52579,'#FFFFFF'),(19432,52580,'#FFFFFF'),(19433,52581,'#FFFFFF'),(19434,52583,'#FFFFFF'),(19435,52584,'#FFFFFF'),(19436,52595,'#FFFFFF'),(19437,52596,'#FFFFFF'),(19438,52597,'#FFFFFF'),(19439,52599,'#FFFFFF'),(19440,52602,'#FFFFFF'),(19441,52606,'#268aeb'),(19442,52608,'#FFFFFF'),(19443,52612,'#FFFFFF'),(19444,52614,'#FFFFFF'),(19445,52618,'#FFFFFF'),(19446,52619,'#FFFFFF'),(19447,52621,'#FFFFFF'),(19448,52623,'#caef61'),(19449,52625,'#FFFFFF'),(19450,52627,'#FFFFFF'),(19451,52634,'#30a1d1'),(19452,52635,'#FFFFFF'),(19453,52636,'#d9061e'),(19454,52639,'#FFFFFF'),(19455,52640,'#39b2fc'),(19456,52646,'#FFFFFF'),(19457,52649,'#8cd963'),(19458,52652,'#FFFFFF'),(19459,52653,'#FFFFFF'),(19460,52655,'#FFFFFF'),(19461,52656,'#FFFFFF'),(19462,52658,'#FFFFFF'),(19463,52663,'#FFFFFF'),(19464,52664,'#FFFFFF'),(19465,52665,'#FFFFFF'),(19466,52666,'#dc5ee2'),(19467,52670,'#5b75e7'),(19468,52672,'#3456f8'),(19469,52673,'#FFFFFF'),(19470,52674,'#FFFFFF'),(19471,52675,'#ce1d92'),(19472,52676,'#73eafb'),(19473,52677,'#d35554'),(19474,52682,'#FFFFFF'),(19475,52686,'#FFFFFF'),(19476,52687,'#ed5608'),(19477,52689,'#FFFFFF'),(19478,52692,'#FFFFFF'),(19479,52695,'#FFFFFF'),(19480,52696,'#FFFFFF'),(19481,52697,'#FFFFFF'),(19482,52699,'#FFFFFF'),(19483,52700,'#FFFFFF'),(19484,52701,'#9ceb0b'),(19485,52702,'#FFFFFF'),(19486,52703,'#FFFFFF'),(19487,52704,'#FFFFFF'),(19488,52706,'#FFFFFF'),(19489,52708,'#FFFFFF'),(19490,52709,'#FFFFFF'),(19491,52711,'#FFFFFF'),(19492,52712,'#FFFFFF'),(19493,52714,'#FFFFFF'),(19494,52715,'#FFFFFF'),(19495,52717,'#FFFFFF'),(19496,52719,'#edfb24'),(19497,52720,'#FFFFFF'),(19498,52722,'#FFFFFF'),(19499,52725,'#FFFFFF'),(19500,52726,'#FFFFFF'),(19501,52728,'#ed70c9'),(19502,52731,'#4d52e9'),(19503,52732,'#df7534'),(19504,52733,'#FFFFFF'),(19505,52735,'#FFFFFF'),(19506,52736,'#FFFFFF'),(19507,52739,'#FFFFFF'),(19508,52740,'#FFFFFF'),(19509,52741,'#62d4e0'),(19510,52742,'#FFFFFF'),(19511,52747,'#da5c3c'),(19512,52749,'#33d89b'),(19513,52754,'#61d5b9'),(19514,52757,'#f06f78'),(19515,52758,'#FFFFFF'),(19516,52759,'#FFFFFF'),(19517,52760,'#FFFFFF'),(19518,52761,'#FFFFFF'),(19519,52762,'#FFFFFF'),(19520,52763,'#FFFFFF'),(19521,52765,'#644cce'),(19522,52768,'#FFFFFF'),(19523,52769,'#f4df4d'),(19524,52771,'#fd7456'),(19525,52772,'#d644db'),(19526,52774,'#FFFFFF'),(19527,52777,'#FFFFFF'),(19528,52778,'#ac5ed2'),(19529,52779,'#62c4f8'),(19530,52780,'#FFFFFF'),(19531,52783,'#6f1dd2'),(19532,52785,'#FFFFFF'),(19533,52788,'#FFFFFF'),(19534,52792,'#FFFFFF'),(19535,52794,'#FFFFFF'),(19536,52795,'#FFFFFF'),(19537,52796,'#FFFFFF'),(19538,52797,'#FFFFFF'),(19539,52798,'#FFFFFF'),(19540,52800,'#FFFFFF'),(19541,52804,'#FFFFFF'),(19542,52805,'#FFFFFF'),(19543,52809,'#FFFFFF'),(19544,52817,'#e6873f'),(19545,52820,'#1a91e9'),(19546,52823,'#7dfc64'),(19547,52824,'#FFFFFF'),(19548,52825,'#db21ab'),(19549,52827,'#FFFFFF'),(19550,52830,'#FFFFFF'),(19551,52831,'#FFFFFF'),(19552,52832,'#FFFFFF'),(19553,52835,'#FFFFFF'),(19554,52836,'#c3fa69'),(19555,52837,'#FFFFFF'),(19556,52838,'#8dcf59'),(19557,52841,'#9326db'),(19558,52842,'#d06711'),(19559,52845,'#FFFFFF'),(19560,52847,'#FFFFFF'),(19561,52848,'#29e1df'),(19562,52849,'#FFFFFF'),(19563,52854,'#FFFFFF'),(19564,52855,'#af31d3'),(19565,52856,'#b6ec41'),(19566,52858,'#FFFFFF'),(19567,52860,'#91dd16'),(19568,52862,'#d3794c'),(19569,52863,'#FFFFFF'),(19570,52866,'#FFFFFF'),(19571,52867,'#67e55e'),(19572,52869,'#FFFFFF'),(19573,52870,'#FFFFFF'),(19574,52871,'#6330f3'),(19575,52872,'#FFFFFF'),(19576,52874,'#FFFFFF'),(19577,52875,'#FFFFFF'),(19578,52881,'#FFFFFF'),(19579,52884,'#FFFFFF'),(19580,52885,'#e6bc62'),(19581,52891,'#FFFFFF'),(19582,52892,'#0fdadb'),(19583,52893,'#FFFFFF'),(19584,52894,'#0045d6'),(19585,52895,'#f270f1'),(19586,52907,'#FFFFFF'),(19587,52908,'#FFFFFF'),(19588,52913,'#305ed1'),(19589,52916,'#FFFFFF'),(19590,52918,'#6be52c'),(19591,52920,'#f0a134'),(19592,52921,'#4525e1'),(19593,52922,'#fd410b'),(19594,52924,'#FFFFFF'),(19595,52926,'#FFFFFF'),(19596,52927,'#3ae248'),(19597,52929,'#ef871d'),(19598,52935,'#d01a42'),(19599,52941,'#d72e62'),(19600,52943,'#FFFFFF'),(19601,52944,'#d8cf0d'),(19602,52947,'#FFFFFF'),(19603,52948,'#FFFFFF'),(19604,52949,'#7af3fb'),(19605,52952,'#ce14ec'),(19606,52959,'#FFFFFF'),(19607,52960,'#FFFFFF'),(19608,52964,'#FFFFFF'),(19609,52966,'#47e80c'),(19610,52967,'#b05de7'),(19611,52968,'#FFFFFF'),(19612,52969,'#d5d406'),(19613,52974,'#FFFFFF'),(19614,52977,'#d403b6'),(19615,52980,'#83cf3f'),(19616,52981,'#FFFFFF'),(19617,52984,'#FFFFFF'),(19618,52985,'#FFFFFF'),(19619,52986,'#FFFFFF'),(19620,52987,'#FFFFFF'),(19621,52988,'#FFFFFF'),(19622,52989,'#FFFFFF'),(19623,52990,'#FFFFFF'),(19624,52991,'#fa0d14'),(19625,52992,'#FFFFFF'),(19626,52998,'#FFFFFF'),(19627,52999,'#27dc14'),(19628,53000,'#da50de'),(19629,53002,'#d94abe'),(19630,53005,'#fa0ed6'),(19631,53007,'#FFFFFF'),(19632,53011,'#FFFFFF'),(19633,53016,'#af2ae1'),(19634,53017,'#FFFFFF'),(19635,53020,'#dbc667'),(19636,53021,'#FFFFFF'),(19637,53024,'#FFFFFF'),(19638,53025,'#FFFFFF'),(19639,53026,'#FFFFFF'),(19640,53027,'#FFFFFF'),(19641,53028,'#FFFFFF'),(19642,53034,'#dc5ab6'),(19643,53035,'#FFFFFF'),(19644,53037,'#fd04fc'),(19645,53045,'#FFFFFF'),(19646,53048,'#06fc4b'),(19647,53049,'#35d845'),(19648,53054,'#08fcf6'),(19649,53056,'#41cf3f'),(19650,53062,'#FFFFFF'),(19651,53064,'#FFFFFF'),(19652,53072,'#fe32e0'),(19653,53074,'#42fa59'),(19654,53076,'#FFFFFF'),(19655,53078,'#445bf1'),(19656,53081,'#cf44d6'),(19657,53085,'#f78046'),(19658,53086,'#51e635'),(19659,53088,'#e9f442'),(19660,53095,'#FFFFFF'),(19661,53103,'#4910cc'),(19662,53105,'#439fd7'),(19663,53106,'#5082fa'),(19664,53107,'#FFFFFF'),(19665,53115,'#0ad021'),(19666,53116,'#fbb37a'),(19667,53117,'#FFFFFF'),(19668,53135,'#cc35a5'),(19669,53137,'#10f4dd'),(19670,53142,'#FFFFFF'),(19671,53143,'#0e26cd'),(19672,53145,'#FFFFFF'),(19673,53148,'#FFFFFF'),(19674,53149,'#FFFFFF'),(19675,53150,'#4b23fe'),(19676,53159,'#d90284'),(19677,53160,'#d24776'),(19678,53162,'#e6a953'),(19679,53171,'#2af043'),(19680,53172,'#f64d9f'),(19681,53177,'#8b52fd'),(19682,53181,'#fc094f'),(19683,53187,'#25a8e0'),(19684,53188,'#33dace'),(19685,53192,'#da6d69'),(19686,53201,'#9a6cda'),(19687,53214,'#576cf4'),(19688,53234,'#FFFFFF'),(19689,53238,'#72eff7'),(19690,53243,'#98fd1f'),(19691,53245,'#FFFFFF'),(19692,53260,'#6a05d4'),(19693,53282,'#e16481'),(19694,53286,'#df74ea'),(19695,53287,'#f98a21'),(19696,53290,'#14de05'),(19697,53291,'#14ddfa'),(19698,53297,'#FFFFFF'),(19699,53299,'#3ad287'),(19700,53309,'#cbd05b'),(19701,53310,'#3acd4c'),(19702,53312,'#dcc93f'),(19703,53313,'#1206e6'),(19704,53331,'#1ea8fc'),(19705,53337,'#db6c80'),(19706,53346,'#72f0d7'),(19707,53354,'#6a3fd4'),(19708,53360,'#0cf087'),(19709,53363,'#6f04cf'),(19710,53366,'#f915d4'),(19711,53367,'#28d375'),(19712,53368,'#5c50d3'),(19713,53369,'#49f85b'),(19714,53371,'#d53ef2'),(19715,53375,'#49fb3f'),(19716,53379,'#f7903e'),(19717,53386,'#da3979'),(19718,53387,'#d3260b'),(19719,53390,'#2a28e8'),(19720,53398,'#644dd3'),(19721,53399,'#55f67e'),(19722,53420,'#02e61e'),(19723,53432,'#FFFFFF'),(19724,53437,'#f8a65e'),(19725,53442,'#18de1d'),(19726,53443,'#4805e8'),(19727,53460,'#34d600'),(19728,53463,'#fe2743'),(19729,53470,'#FFFFFF'),(19730,53472,'#f44f56'),(19731,53483,'#20f8b0'),(19732,53484,'#d01b85'),(19733,53487,'#71f66f'),(19734,53499,'#71f263'),(19735,53501,'#e373e8'),(19736,53510,'#1ae0b1'),(19737,53523,'#e23404'),(19738,53524,'#5af1a5'),(19739,53527,'#d1be64'),(19740,53533,'#dda637'),(19741,53543,'#3b59d6'),(19742,53545,'#FFFFFF'),(19743,53546,'#FFFFFF'),(19744,53551,'#59d1c8'),(19745,53552,'#e2956c'),(19746,53558,'#FFFFFF'),(19747,53567,'#f29122'),(19748,53580,'#FFFFFF'),(19749,53586,'#5be6bd'),(19750,53588,'#FFFFFF'),(19751,53594,'#e760ba'),(19752,53595,'#0492f2'),(19753,53597,'#FFFFFF'),(19754,53599,'#FFFFFF'),(19755,53602,'#33ef3b'),(19756,53610,'#1bcd6a'),(19757,53611,'#87ea3d'),(19758,53615,'#680ecd'),(19759,53616,'#9749d4'),(19760,53617,'#7e69f5'),(19761,53623,'#FFFFFF'),(19762,53625,'#d0692f'),(19763,53627,'#65d88e'),(19764,53628,'#6067f9'),(19765,53633,'#c9d843'),(19766,53635,'#FFFFFF'),(19767,53646,'#cb49f2'),(19768,53661,'#4dd459'),(19769,53662,'#9ccf5a'),(19770,53668,'#FFFFFF'),(19771,53673,'#FFFFFF'),(19772,53690,'#fa6861'),(19773,53692,'#1e67ea'),(19774,53699,'#0b30de'),(19775,53701,'#f631b6'),(19776,53703,'#1821d0'),(19777,53707,'#FFFFFF'),(19778,53715,'#FFFFFF'),(19779,53722,'#bb5dd0'),(19780,53723,'#d95bbf'),(19781,53731,'#e2615f'),(19782,53736,'#03d66f'),(19783,53743,'#FFFFFF'),(19784,53744,'#9917f2'),(19785,53748,'#82ec57'),(19786,53751,'#66d100'),(19787,53763,'#df3b67'),(19788,53765,'#f066f3'),(19789,53767,'#ae25f8'),(19790,53772,'#FFFFFF'),(19791,53774,'#0bd8f3'),(19792,53779,'#7bf91d'),(19793,53797,'#3fcfda'),(19794,53810,'#3445cd'),(19795,53816,'#ebfb6d'),(19796,53825,'#cf5798'),(19797,53835,'#7468dd'),(19798,53838,'#6ae380'),(19799,53847,'#f057a5'),(19800,53848,'#4ad8b7'),(19801,53860,'#f0fb78'),(19802,53883,'#a142dc'),(19803,53899,'#e0de4a'),(19804,53908,'#38c0d4'),(19805,53917,'#FFFFFF'),(19806,53918,'#FFFFFF'),(19807,53924,'#19cfde'),(19808,53941,'#FFFFFF'),(19809,53952,'#d46f56'),(19810,53957,'#FFFFFF'),(19811,53958,'#FFFFFF'),(19812,53966,'#ce4822'),(19813,53967,'#FFFFFF'),(19814,53968,'#fb1116'),(19815,53973,'#8ccc5c'),(19816,53976,'#3fe529'),(19817,53996,'#e2d83c'),(19818,53997,'#44fdd3'),(19819,54000,'#c909dc'),(19820,54005,'#FFFFFF'),(19821,54012,'#FFFFFF'),(19822,54013,'#c82ff7'),(19823,54019,'#f08d21'),(19824,54025,'#71f313'),(19825,54037,'#49d327'),(19826,54043,'#4d6dfa'),(19827,54047,'#FFFFFF'),(19828,54052,'#6adeb6'),(19829,54062,'#f1545f'),(19830,54068,'#5ce247'),(19831,54076,'#cfd424'),(19832,54080,'#5eec5a'),(19833,54081,'#5aeb8d'),(19834,54083,'#FFFFFF'),(19835,54084,'#FFFFFF'),(19836,54088,'#FFFFFF'),(19837,54089,'#41cd44'),(19838,54091,'#e1b96a'),(19839,54093,'#FFFFFF'),(19840,54095,'#9fde48'),(19841,54097,'#fee43d'),(19842,54098,'#FFFFFF'),(19843,54099,'#FFFFFF'),(19844,54100,'#3bcf48'),(19845,54102,'#bfea5c'),(19846,54104,'#51f68a'),(19847,54105,'#e06ddc'),(19848,54106,'#FFFFFF'),(19849,54107,'#5064dc'),(19850,54108,'#FFFFFF'),(19851,54109,'#FFFFFF'),(19852,54111,'#FFFFFF'),(19853,54112,'#d121d3'),(19854,54113,'#318bf5'),(19855,54114,'#FFFFFF'),(19856,54116,'#d13cee'),(19857,54117,'#FFFFFF'),(19858,54118,'#74b8f3'),(19859,54120,'#d84519'),(19860,54121,'#FFFFFF'),(19861,54123,'#46add2'),(19862,54124,'#1c49dd'),(19863,54126,'#FFFFFF'),(19864,54128,'#c3ec0b'),(19865,54129,'#6302d6'),(19866,54130,'#29e08a'),(19867,54131,'#e6cf63'),(19868,54132,'#fd66d9'),(19869,54133,'#FFFFFF'),(19870,54134,'#FFFFFF'),(19871,54135,'#6de573'),(19872,54136,'#FFFFFF'),(19873,54137,'#6fe45f'),(19874,54138,'#ef43d7'),(19875,54139,'#FFFFFF'),(19876,54140,'#FFFFFF'),(19877,54143,'#FFFFFF'),(19878,54145,'#FFFFFF'),(19879,54146,'#FFFFFF'),(19880,54147,'#FFFFFF'),(19881,54148,'#FFFFFF'),(19882,54150,'#e560c7'),(19883,54151,'#FFFFFF'),(19884,54153,'#172def'),(19885,54154,'#623eec'),(19886,54156,'#FFFFFF'),(19887,54157,'#3a20f0'),(19888,54158,'#43cdd4'),(19889,54159,'#FFFFFF'),(19890,54160,'#d60730'),(19891,54162,'#cc559f'),(19892,54163,'#1139cd'),(19893,54164,'#62d036'),(19894,54165,'#FFFFFF'),(19895,54166,'#07f7e2'),(19896,54167,'#f06b99'),(19897,54168,'#FFFFFF'),(19898,54169,'#6d5fcc'),(19899,54170,'#FFFFFF'),(19900,54171,'#e92d38'),(19901,54172,'#FFFFFF'),(19902,54175,'#bf6df0'),(19903,54176,'#FFFFFF'),(19904,54178,'#8e54fe'),(19905,54181,'#FFFFFF'),(19906,54182,'#0fce81'),(19907,54183,'#FFFFFF'),(19908,54185,'#FFFFFF'),(19909,54187,'#FFFFFF'),(19910,54188,'#e302be'),(19911,54189,'#FFFFFF'),(19912,54190,'#2fd9f3'),(19913,54191,'#cc1cae'),(19914,54193,'#fd5270'),(19915,54194,'#801eed'),(19916,54195,'#FFFFFF'),(19917,54197,'#FFFFFF'),(19918,54199,'#a641dc'),(19919,54201,'#FFFFFF'),(19920,54202,'#e5ac42'),(19921,54204,'#0732ec'),(19922,54205,'#83fa3f'),(19923,54206,'#e72921'),(19924,54207,'#d04a6e'),(19925,54208,'#FFFFFF'),(19926,54209,'#FFFFFF'),(19927,54211,'#c369f2'),(19928,54212,'#FFFFFF'),(19929,54213,'#ca25f5'),(19930,54214,'#FFFFFF'),(19931,54216,'#FFFFFF'),(19932,54217,'#FFFFFF'),(19933,54218,'#FFFFFF'),(19934,54219,'#f86ded'),(19935,54222,'#00e42a'),(19936,54223,'#40dcd0'),(19937,54224,'#62deb1'),(19938,54225,'#FFFFFF'),(19939,54226,'#e760e4'),(19940,54228,'#FFFFFF'),(19941,54229,'#2474d3'),(19942,54230,'#04d3dc'),(19943,54231,'#0a6ed4'),(19944,54232,'#c1f345'),(19945,54233,'#FFFFFF'),(19946,54234,'#9959e5'),(19947,54236,'#FFFFFF'),(19948,54237,'#FFFFFF'),(19949,54238,'#0ffc9e'),(19950,54239,'#FFFFFF'),(19951,54240,'#df2b02'),(19952,54243,'#0ffa27'),(19953,54244,'#FFFFFF'),(19954,54246,'#94f534'),(19955,54247,'#FFFFFF'),(19956,54248,'#FFFFFF'),(19957,54250,'#FFFFFF'),(19958,54253,'#FFFFFF'),(19959,54254,'#FFFFFF'),(19960,54256,'#FFFFFF'),(19961,54257,'#FFFFFF'),(19962,54258,'#FFFFFF'),(19963,54259,'#FFFFFF'),(19964,54261,'#c8fe51'),(19965,54264,'#193bfe'),(19966,54267,'#bbdd4e'),(19967,54268,'#8e36ce'),(19968,54270,'#FFFFFF'),(19969,54271,'#FFFFFF'),(19970,54272,'#FFFFFF'),(19971,54273,'#FFFFFF'),(19972,54274,'#FFFFFF'),(19973,54275,'#38cf2a'),(19974,54276,'#FFFFFF'),(19975,54278,'#db1968'),(19976,54279,'#FFFFFF'),(19977,54280,'#FFFFFF'),(19978,54281,'#1248d9'),(19979,54282,'#FFFFFF'),(19980,54283,'#cd3166'),(19981,54284,'#FFFFFF'),(19982,54285,'#e0ba60'),(19983,54287,'#FFFFFF'),(19984,54290,'#d30d0f'),(19985,54291,'#11c7d1'),(19986,54292,'#FFFFFF'),(19987,54293,'#ebf022'),(19988,54294,'#7085f3'),(19989,54295,'#FFFFFF'),(19990,54299,'#FFFFFF'),(19991,54300,'#5af7fa'),(19992,54302,'#FFFFFF'),(19993,54303,'#FFFFFF'),(19994,54304,'#f94995'),(19995,54307,'#1213d7'),(19996,54308,'#FFFFFF'),(19997,54309,'#edce5d'),(19998,54311,'#8e5be7'),(19999,54312,'#27dcdb'),(20000,54313,'#FFFFFF'),(20001,54314,'#FFFFFF'),(20002,54315,'#FFFFFF'),(20003,54316,'#FFFFFF'),(20004,54317,'#FFFFFF'),(20005,54319,'#FFFFFF'),(20006,54320,'#FFFFFF'),(20007,54321,'#FFFFFF'),(20008,54323,'#FFFFFF'),(20009,54328,'#FFFFFF'),(20010,54331,'#FFFFFF'),(20011,54332,'#4c61df'),(20012,54333,'#FFFFFF'),(20013,54334,'#fa7628'),(20014,54335,'#58e844'),(20015,54336,'#FFFFFF'),(20016,54337,'#f33709'),(20017,54338,'#f9f86c'),(20018,54340,'#21a4ec'),(20019,54341,'#FFFFFF'),(20020,54342,'#6b3df9'),(20021,54343,'#cd7b17'),(20022,54344,'#FFFFFF'),(20023,54345,'#ae65d1'),(20024,54346,'#FFFFFF'),(20025,54347,'#FFFFFF'),(20026,54348,'#FFFFFF'),(20027,54349,'#FFFFFF'),(20028,54350,'#FFFFFF'),(20029,54352,'#FFFFFF'),(20030,54354,'#FFFFFF'),(20031,54358,'#718df4'),(20032,54359,'#FFFFFF'),(20033,54360,'#2e58d4'),(20034,54361,'#cffe5c'),(20035,54364,'#ee4d73'),(20036,54365,'#FFFFFF'),(20037,54368,'#FFFFFF'),(20038,54369,'#01fcf4'),(20039,54371,'#FFFFFF'),(20040,54373,'#FFFFFF'),(20041,54374,'#FFFFFF'),(20042,54375,'#e4c837'),(20043,54376,'#5dbde4'),(20044,54377,'#91d70f'),(20045,54378,'#3bf371'),(20046,54380,'#FFFFFF'),(20047,54381,'#FFFFFF'),(20048,54382,'#46d31c'),(20049,54384,'#d37166'),(20050,54387,'#44cd6f'),(20051,54388,'#FFFFFF'),(20052,54389,'#FFFFFF'),(20053,54390,'#12e27e'),(20054,54391,'#FFFFFF'),(20055,54392,'#17db02'),(20056,54393,'#FFFFFF'),(20057,54394,'#5dd1eb'),(20058,54396,'#FFFFFF'),(20059,54398,'#FFFFFF'),(20060,54399,'#FFFFFF'),(20061,54400,'#FFFFFF'),(20062,54401,'#d1482d'),(20063,54402,'#12de97'),(20064,54403,'#FFFFFF'),(20065,54404,'#FFFFFF'),(20066,54406,'#6ace5d'),(20067,54411,'#FFFFFF'),(20068,54412,'#e3843c'),(20069,54413,'#FFFFFF'),(20070,54417,'#6584f3'),(20071,54418,'#FFFFFF'),(20072,54419,'#105ffb'),(20073,54420,'#FFFFFF'),(20074,54422,'#52b6fa'),(20075,54423,'#FFFFFF'),(20076,54424,'#f25475'),(20077,54425,'#FFFFFF'),(20078,54434,'#FFFFFF'),(20079,54436,'#e8a827'),(20080,54437,'#c80cdf'),(20081,54439,'#f75ddc'),(20082,54440,'#ced048'),(20083,54441,'#FFFFFF'),(20084,54442,'#d5d908'),(20085,54446,'#FFFFFF'),(20086,54447,'#FFFFFF'),(20087,54449,'#FFFFFF'),(20088,54450,'#51a7ea'),(20089,54451,'#FFFFFF'),(20090,54452,'#FFFFFF'),(20091,54453,'#FFFFFF'),(20092,54454,'#FFFFFF'),(20093,54455,'#FFFFFF'),(20094,54456,'#FFFFFF'),(20095,54457,'#FFFFFF'),(20096,54458,'#FFFFFF'),(20097,54459,'#c4d754'),(20098,54461,'#dddd2b'),(20099,54464,'#FFFFFF'),(20100,54465,'#FFFFFF'),(20101,54470,'#FFFFFF'),(20102,54471,'#FFFFFF'),(20103,54472,'#FFFFFF'),(20104,54474,'#FFFFFF'),(20105,54475,'#FFFFFF'),(20106,54476,'#FFFFFF'),(20107,54477,'#FFFFFF'),(20108,54478,'#de3fe5'),(20109,54479,'#ea088e'),(20110,54483,'#dea45a'),(20111,54486,'#FFFFFF'),(20112,54487,'#0aebc7'),(20113,54492,'#FFFFFF'),(20114,54493,'#d0963b'),(20115,54495,'#FFFFFF'),(20116,54497,'#FFFFFF'),(20117,54502,'#63dfc6'),(20118,54506,'#2adc98'),(20119,54508,'#FFFFFF'),(20120,54510,'#FFFFFF'),(20121,54512,'#FFFFFF'),(20122,54514,'#e55782'),(20123,54517,'#FFFFFF'),(20124,54518,'#FFFFFF'),(20125,54520,'#0259da'),(20126,54522,'#9b1ae1'),(20127,54526,'#FFFFFF'),(20128,54527,'#f776be'),(20129,54529,'#89fc23'),(20130,54530,'#52d008'),(20131,54535,'#FFFFFF'),(20132,54538,'#FFFFFF'),(20133,54542,'#e4c75f'),(20134,54544,'#fa39a2'),(20135,54545,'#79e606'),(20136,54547,'#2ab1e3'),(20137,54548,'#4a93e7'),(20138,54549,'#d99a11'),(20139,54553,'#FFFFFF'),(20140,54554,'#FFFFFF'),(20141,54555,'#629de3'),(20142,54556,'#FFFFFF'),(20143,54557,'#FFFFFF'),(20144,54558,'#FFFFFF'),(20145,54559,'#e21bed'),(20146,54560,'#FFFFFF'),(20147,54561,'#FFFFFF'),(20148,54562,'#d9eb5c'),(20149,54566,'#d9bb64'),(20150,54567,'#FFFFFF'),(20151,54569,'#1ef34b'),(20152,54570,'#FFFFFF'),(20153,54571,'#FFFFFF'),(20154,54572,'#FFFFFF'),(20155,54574,'#FFFFFF'),(20156,54576,'#FFFFFF'),(20157,54578,'#6afd36'),(20158,54579,'#5371d6'),(20159,54581,'#FFFFFF'),(20160,54582,'#a053df'),(20161,54583,'#FFFFFF'),(20162,54588,'#FFFFFF'),(20163,54589,'#FFFFFF'),(20164,54591,'#FFFFFF'),(20165,54592,'#6ff7cf'),(20166,54593,'#FFFFFF'),(20167,54594,'#FFFFFF'),(20168,54595,'#c733f5'),(20169,54597,'#FFFFFF'),(20170,54598,'#FFFFFF'),(20171,54602,'#1434cf'),(20172,54607,'#d7a456'),(20173,54608,'#4294f7'),(20174,54609,'#FFFFFF'),(20175,54615,'#FFFFFF'),(20176,54617,'#3879d8'),(20177,54620,'#2fd5e5'),(20178,54621,'#6bd7fd'),(20179,54622,'#FFFFFF'),(20180,54623,'#0a87d2'),(20181,54625,'#52e4b6'),(20182,54626,'#FFFFFF'),(20183,54627,'#f6f417'),(20184,54631,'#cc2de7'),(20185,54632,'#FFFFFF'),(20186,54633,'#6dfb7c'),(20187,54634,'#d86aa4'),(20188,54635,'#FFFFFF'),(20189,54637,'#FFFFFF'),(20190,54640,'#d625ad'),(20191,54642,'#FFFFFF'),(20192,54643,'#ed0ce8'),(20193,54645,'#FFFFFF'),(20194,54648,'#cb52ef'),(20195,54650,'#FFFFFF'),(20196,54652,'#FFFFFF'),(20197,54653,'#FFFFFF'),(20198,54657,'#FFFFFF'),(20199,54658,'#FFFFFF'),(20200,54659,'#9af501'),(20201,54660,'#FFFFFF'),(20202,54662,'#6fe787'),(20203,54663,'#3ff955'),(20204,54664,'#f431a3'),(20205,54665,'#38f5ad'),(20206,54666,'#c3ef63'),(20207,54667,'#b645fd'),(20208,54670,'#fd4d59'),(20209,54671,'#d01957'),(20210,54673,'#FFFFFF'),(20211,54675,'#FFFFFF'),(20212,54676,'#FFFFFF'),(20213,54677,'#FFFFFF'),(20214,54678,'#FFFFFF'),(20215,54680,'#FFFFFF'),(20216,54683,'#FFFFFF'),(20217,54684,'#FFFFFF'),(20218,54685,'#FFFFFF'),(20219,54686,'#FFFFFF'),(20220,54687,'#FFFFFF'),(20221,54688,'#FFFFFF'),(20222,54690,'#FFFFFF'),(20223,54691,'#FFFFFF'),(20224,54692,'#FFFFFF'),(20225,54693,'#FFFFFF'),(20226,54694,'#FFFFFF'),(20227,54695,'#FFFFFF'),(20228,54696,'#FFFFFF'),(20229,54697,'#FFFFFF'),(20230,54698,'#FFFFFF'),(20231,54699,'#FFFFFF'),(20232,54700,'#FFFFFF'),(20233,54702,'#FFFFFF'),(20234,54703,'#FFFFFF'),(20235,54705,'#FFFFFF'),(20236,54706,'#FFFFFF'),(20237,54707,'#FFFFFF'),(20238,54708,'#FFFFFF'),(20239,54709,'#FFFFFF'),(20240,54710,'#FFFFFF'),(20241,54711,'#FFFFFF'),(20242,54715,'#FFFFFF'),(20243,54716,'#FFFFFF'),(20244,54721,'#FFFFFF'),(20245,54722,'#FFFFFF'),(20246,54723,'#FFFFFF'),(20247,54727,'#FFFFFF'),(20248,54728,'#FFFFFF'),(20249,54729,'#FFFFFF'),(20250,54730,'#FFFFFF'),(20251,54731,'#FFFFFF'),(20252,54733,'#FFFFFF'),(20253,54734,'#FFFFFF'),(20254,54735,'#FFFFFF'),(20255,54736,'#FFFFFF'),(20256,54738,'#FFFFFF'),(20257,54740,'#FFFFFF'),(20258,54741,'#FFFFFF'),(20259,54742,'#FFFFFF'),(20260,54743,'#FFFFFF'),(20261,54745,'#FFFFFF'),(20262,54746,'#FFFFFF'),(20263,54747,'#FFFFFF'),(20264,54748,'#FFFFFF'),(20265,54750,'#FFFFFF'),(20266,54755,'#FFFFFF'),(20267,54756,'#FFFFFF'),(20268,54757,'#FFFFFF'),(20269,54758,'#FFFFFF'),(20270,54760,'#FFFFFF'),(20271,54761,'#FFFFFF'),(20272,54762,'#FFFFFF'),(20273,54763,'#FFFFFF'),(20274,54764,'#FFFFFF'),(20275,54765,'#FFFFFF'),(20276,54766,'#FFFFFF'),(20277,54767,'#FFFFFF'),(20278,54768,'#FFFFFF'),(20279,54769,'#FFFFFF'),(20280,54770,'#FFFFFF'),(20281,54771,'#FFFFFF'),(20282,54772,'#FFFFFF'),(20283,54775,'#FFFFFF'),(20284,54777,'#FFFFFF'),(20285,54778,'#FFFFFF'),(20286,54779,'#FFFFFF'),(20287,54780,'#FFFFFF'),(20288,54781,'#FFFFFF'),(20289,54782,'#FFFFFF'),(20290,54783,'#FFFFFF'),(20291,54784,'#FFFFFF'),(20292,54785,'#FFFFFF'),(20293,54786,'#FFFFFF'),(20294,54787,'#FFFFFF'),(20295,54789,'#FFFFFF'),(20296,54790,'#FFFFFF'),(20297,54791,'#FFFFFF'),(20298,54792,'#FFFFFF'),(20299,54793,'#FFFFFF'),(20300,54794,'#FFFFFF'),(20301,54795,'#FFFFFF'),(20302,54796,'#FFFFFF'),(20303,54797,'#FFFFFF'),(20304,54798,'#FFFFFF'),(20305,54799,'#FFFFFF'),(20306,54802,'#FFFFFF'),(20307,54803,'#FFFFFF'),(20308,54804,'#FFFFFF'),(20309,54807,'#FFFFFF'),(20310,54808,'#FFFFFF'),(20311,54809,'#FFFFFF'),(20312,54810,'#FFFFFF'),(20313,54811,'#c0eb5d'),(20314,54812,'#FFFFFF'),(20315,54813,'#FFFFFF'),(20316,54814,'#FFFFFF'),(20317,54816,'#FFFFFF'),(20318,54818,'#FFFFFF'),(20319,54820,'#FFFFFF'),(20320,54821,'#FFFFFF'),(20321,54822,'#FFFFFF'),(20322,54823,'#FFFFFF'),(20323,54824,'#FFFFFF'),(20324,54825,'#FFFFFF'),(20325,54826,'#FFFFFF'),(20326,54827,'#FFFFFF'),(20327,54828,'#FFFFFF'),(20328,54829,'#FFFFFF'),(20329,54830,'#FFFFFF'),(20330,54831,'#FFFFFF'),(20331,54833,'#FFFFFF'),(20332,54834,'#FFFFFF'),(20333,54835,'#FFFFFF'),(20334,54836,'#FFFFFF'),(20335,54837,'#FFFFFF'),(20336,54841,'#FFFFFF'),(20337,54842,'#FFFFFF'),(20338,54843,'#FFFFFF'),(20339,54844,'#FFFFFF'),(20340,54845,'#FFFFFF'),(20341,54846,'#FFFFFF'),(20342,54850,'#FFFFFF'),(20343,54852,'#FFFFFF'),(20344,54854,'#FFFFFF'),(20345,54855,'#FFFFFF'),(20346,54856,'#FFFFFF'),(20347,54857,'#FFFFFF'),(20348,54858,'#FFFFFF'),(20349,54859,'#FFFFFF'),(20350,54860,'#FFFFFF'),(20351,54863,'#FFFFFF'),(20352,54864,'#FFFFFF'),(20353,54865,'#FFFFFF'),(20354,54866,'#77d73c'),(20355,54868,'#FFFFFF'),(20356,54870,'#FFFFFF'),(20357,54871,'#FFFFFF'),(20358,54872,'#FFFFFF'),(20359,54873,'#FFFFFF'),(20360,54874,'#FFFFFF'),(20361,54876,'#FFFFFF'),(20362,54878,'#FFFFFF'),(20363,54879,'#FFFFFF'),(20364,54880,'#FFFFFF'),(20365,54881,'#FFFFFF'),(20366,54886,'#FFFFFF'),(20367,54888,'#FFFFFF'),(20368,54889,'#FFFFFF'),(20369,54890,'#FFFFFF'),(20370,54891,'#FFFFFF'),(20371,54892,'#FFFFFF'),(20372,54893,'#FFFFFF'),(20373,54895,'#FFFFFF'),(20374,54896,'#FFFFFF'),(20375,54898,'#FFFFFF'),(20376,54899,'#FFFFFF'),(20377,54900,'#FFFFFF'),(20378,54901,'#FFFFFF'),(20379,54902,'#FFFFFF'),(20380,54904,'#FFFFFF'),(20381,54908,'#FFFFFF'),(20382,54909,'#FFFFFF'),(20383,54910,'#FFFFFF'),(20384,54911,'#FFFFFF'),(20385,54912,'#FFFFFF'),(20386,54913,'#FFFFFF'),(20387,54914,'#FFFFFF'),(20388,54915,'#FFFFFF'),(20389,54916,'#FFFFFF'),(20390,54918,'#FFFFFF'),(20391,54919,'#FFFFFF'),(20392,54920,'#FFFFFF'),(20393,54921,'#FFFFFF'),(20394,54923,'#FFFFFF'),(20395,54924,'#FFFFFF'),(20396,54925,'#FFFFFF'),(20397,54926,'#FFFFFF'),(20398,54927,'#FFFFFF'),(20399,54928,'#FFFFFF'),(20400,54931,'#FFFFFF'),(20401,54932,'#FFFFFF'),(20402,54933,'#FFFFFF'),(20403,54935,'#FFFFFF'),(20404,54937,'#FFFFFF'),(20405,54940,'#FFFFFF'),(20406,54942,'#FFFFFF'),(20407,54943,'#FFFFFF'),(20408,54944,'#FFFFFF'),(20409,54946,'#FFFFFF'),(20410,54947,'#FFFFFF'),(20411,54948,'#FFFFFF'),(20412,54951,'#FFFFFF'),(20413,54952,'#FFFFFF'),(20414,54955,'#FFFFFF'),(20415,54957,'#e55469'),(20416,54958,'#FFFFFF'),(20417,54959,'#d16612'),(20418,54961,'#FFFFFF'),(20419,54962,'#f11ca3'),(20420,54963,'#FFFFFF'),(20421,54964,'#FFFFFF'),(20422,54965,'#FFFFFF'),(20423,54966,'#FFFFFF'),(20424,54968,'#FFFFFF'),(20425,54970,'#FFFFFF'),(20426,54972,'#FFFFFF'),(20427,54973,'#FFFFFF'),(20428,54974,'#FFFFFF'),(20429,54975,'#FFFFFF'),(20430,54977,'#FFFFFF'),(20431,54978,'#FFFFFF'),(20432,54979,'#FFFFFF'),(20433,54980,'#FFFFFF'),(20434,54981,'#FFFFFF'),(20435,54982,'#FFFFFF'),(20436,54983,'#FFFFFF'),(20437,54984,'#FFFFFF'),(20438,54985,'#FFFFFF'),(20439,54986,'#FFFFFF'),(20440,54987,'#FFFFFF'),(20441,54988,'#FFFFFF'),(20442,54989,'#FFFFFF'),(20443,54992,'#FFFFFF'),(20444,54993,'#FFFFFF'),(20445,54994,'#FFFFFF'),(20446,54996,'#FFFFFF'),(20447,55000,'#FFFFFF'),(20448,55001,'#FFFFFF'),(20449,55002,'#FFFFFF'),(20450,55003,'#FFFFFF'),(20451,55004,'#FFFFFF'),(20452,55005,'#FFFFFF'),(20453,55006,'#FFFFFF'),(20454,55007,'#FFFFFF'),(20455,55008,'#FFFFFF'),(20456,55010,'#FFFFFF'),(20457,55012,'#FFFFFF'),(20458,55013,'#FFFFFF'),(20459,55014,'#FFFFFF'),(20460,55015,'#FFFFFF'),(20461,55016,'#FFFFFF'),(20462,55019,'#FFFFFF'),(20463,55020,'#FFFFFF'),(20464,55022,'#FFFFFF'),(20465,55023,'#FFFFFF'),(20466,55024,'#FFFFFF'),(20467,55025,'#FFFFFF'),(20468,55027,'#FFFFFF'),(20469,55030,'#FFFFFF'),(20470,55031,'#FFFFFF'),(20471,55032,'#FFFFFF'),(20472,55033,'#FFFFFF'),(20473,55034,'#FFFFFF'),(20474,55035,'#FFFFFF'),(20475,55037,'#FFFFFF'),(20476,55039,'#FFFFFF'),(20477,55040,'#FFFFFF'),(20478,55043,'#FFFFFF'),(20479,55045,'#FFFFFF'),(20480,55046,'#FFFFFF'),(20481,55047,'#FFFFFF'),(20482,55048,'#FFFFFF'),(20483,55049,'#FFFFFF'),(20484,55050,'#FFFFFF'),(20485,55051,'#FFFFFF'),(20486,55052,'#FFFFFF'),(20487,55053,'#FFFFFF'),(20488,55054,'#FFFFFF'),(20489,55055,'#FFFFFF'),(20490,55056,'#FFFFFF'),(20491,55057,'#FFFFFF'),(20492,55058,'#FFFFFF'),(20493,55059,'#FFFFFF'),(20494,55063,'#FFFFFF'),(20495,55064,'#FFFFFF'),(20496,55072,'#d14176'),(20497,55084,'#62ef8a'),(20498,55093,'#00d411'),(20499,55095,'#FFFFFF'),(20500,55108,'#1f97f9'),(20501,55115,'#4fdef7'),(20502,55116,'#9d4cd9'),(20503,55132,'#05f617'),(20504,55133,'#FFFFFF'),(20505,55140,'#464bde'),(20506,55158,'#d6f13f'),(20507,55160,'#FFFFFF'),(20508,55171,'#1dcea8'),(20509,55172,'#FFFFFF'),(20510,55175,'#dd5692'),(20511,55181,'#FFFFFF'),(20512,55184,'#beee74'),(20513,55207,'#c8df4f'),(20514,55209,'#ad02ef'),(20515,55214,'#1eeb30'),(20516,55219,'#e31b81'),(20517,55221,'#0cbbdd'),(20518,55226,'#c8e322'),(20519,55229,'#425fed'),(20520,55230,'#61de68'),(20521,55231,'#f76b81'),(20522,55233,'#fb9c15'),(20523,55240,'#3ce951'),(20524,55242,'#364fcc'),(20525,55247,'#1bd808'),(20526,55249,'#37edaf'),(20527,55251,'#803cdc'),(20528,55254,'#c563de'),(20529,55256,'#504af2'),(20530,55261,'#273fe1'),(20531,55266,'#43ced1'),(20532,55267,'#d94f5e'),(20533,55268,'#9108dc'),(20534,55269,'#d7aa6b'),(20535,55279,'#FFFFFF'),(20536,55285,'#74a5f8'),(20537,55308,'#65d9f0'),(20538,55314,'#a34fd2'),(20539,55317,'#d15c6a'),(20540,55323,'#93ec38'),(20541,55333,'#dcf71c'),(20542,55337,'#FFFFFF'),(20543,55370,'#73afee'),(20544,55378,'#560add'),(20545,55394,'#604cd4'),(20546,55395,'#FFFFFF'),(20547,55397,'#0a71de'),(20548,55398,'#FFFFFF'),(20549,55400,'#FFFFFF'),(20550,55401,'#FFFFFF'),(20551,55404,'#FFFFFF'),(20552,55406,'#FFFFFF'),(20553,55407,'#FFFFFF'),(20554,55408,'#47c4eb'),(20555,55409,'#FFFFFF'),(20556,55410,'#FFFFFF'),(20557,55411,'#49fe6e'),(20558,55413,'#FFFFFF'),(20559,55414,'#e90f98'),(20560,55415,'#e33294'),(20561,55416,'#75fcad'),(20562,55417,'#FFFFFF'),(20563,55419,'#1ad547'),(20564,55420,'#13a4e3'),(20565,55421,'#10f430'),(20566,55422,'#FFFFFF'),(20567,55423,'#dd3822'),(20568,55424,'#33a0e9'),(20569,55426,'#43cc58'),(20570,55428,'#FFFFFF'),(20571,55430,'#ed1881'),(20572,55431,'#FFFFFF'),(20573,55435,'#FFFFFF'),(20574,55436,'#e96aec'),(20575,55437,'#6eeb65'),(20576,55439,'#2370f1'),(20577,55440,'#FFFFFF'),(20578,55441,'#FFFFFF'),(20579,55442,'#44e57e'),(20580,55443,'#FFFFFF'),(20581,55444,'#FFFFFF'),(20582,55448,'#FFFFFF'),(20583,55450,'#fc659b'),(20584,55453,'#542ed9'),(20585,55454,'#d44ee3'),(20586,55455,'#f2cd5e'),(20587,55456,'#FFFFFF'),(20588,55459,'#3eecb4'),(20589,55460,'#FFFFFF'),(20590,55461,'#FFFFFF'),(20591,55462,'#ee4371'),(20592,55463,'#FFFFFF'),(20593,55464,'#FFFFFF'),(20594,55466,'#FFFFFF'),(20595,55468,'#FFFFFF'),(20596,55470,'#FFFFFF'),(20597,55471,'#FFFFFF'),(20598,55473,'#FFFFFF'),(20599,55474,'#FFFFFF'),(20600,55475,'#44d499'),(20601,55476,'#FFFFFF'),(20602,55477,'#d90028'),(20603,55478,'#10d50c'),(20604,55479,'#61afd6'),(20605,55480,'#2ee333'),(20606,55481,'#51d6d2'),(20607,55483,'#49de70'),(20608,55484,'#FFFFFF'),(20609,55485,'#FFFFFF'),(20610,55486,'#49b1e7'),(20611,55487,'#FFFFFF'),(20612,55488,'#FFFFFF'),(20613,55490,'#FFFFFF'),(20614,55491,'#1ae675'),(20615,55492,'#FFFFFF'),(20616,55493,'#f62f30'),(20617,55494,'#FFFFFF'),(20618,55495,'#FFFFFF'),(20619,55496,'#FFFFFF'),(20620,55497,'#FFFFFF'),(20621,55498,'#FFFFFF'),(20622,55500,'#FFFFFF'),(20623,55501,'#697cd6'),(20624,55502,'#FFFFFF'),(20625,55503,'#FFFFFF'),(20626,55504,'#FFFFFF'),(20627,55505,'#d342ef'),(20628,55506,'#FFFFFF'),(20629,55507,'#FFFFFF'),(20630,55508,'#FFFFFF'),(20631,55509,'#FFFFFF'),(20632,55511,'#8bf316'),(20633,55512,'#FFFFFF'),(20634,55513,'#1d94d9'),(20635,55514,'#FFFFFF'),(20636,55515,'#15ce8e'),(20637,55516,'#FFFFFF'),(20638,55517,'#FFFFFF'),(20639,55518,'#FFFFFF'),(20640,55520,'#53cce3'),(20641,55521,'#FFFFFF'),(20642,55522,'#FFFFFF'),(20643,55523,'#FFFFFF'),(20644,55524,'#FFFFFF'),(20645,55526,'#FFFFFF'),(20646,55527,'#FFFFFF'),(20647,55529,'#5dd164'),(20648,55530,'#63dd9a'),(20649,55531,'#FFFFFF'),(20650,55532,'#FFFFFF'),(20651,55533,'#FFFFFF'),(20652,55534,'#FFFFFF'),(20653,55535,'#FFFFFF'),(20654,55536,'#FFFFFF'),(20655,55537,'#e4c954'),(20656,55538,'#FFFFFF'),(20657,55539,'#cc14f3'),(20658,55542,'#FFFFFF'),(20659,55543,'#FFFFFF'),(20660,55544,'#FFFFFF'),(20661,55545,'#9257d3'),(20662,55546,'#FFFFFF'),(20663,55547,'#FFFFFF'),(20664,55548,'#8a56ee'),(20665,55549,'#deba65'),(20666,55550,'#FFFFFF'),(20667,55551,'#FFFFFF'),(20668,55553,'#de3e7c'),(20669,55554,'#FFFFFF'),(20670,55555,'#8355ee'),(20671,55557,'#7456cc'),(20672,55558,'#a157f9'),(20673,55559,'#FFFFFF'),(20674,55561,'#f261fc'),(20675,55563,'#FFFFFF'),(20676,55565,'#FFFFFF'),(20677,55567,'#FFFFFF'),(20678,55568,'#f9089b'),(20679,55570,'#FFFFFF'),(20680,55571,'#FFFFFF'),(20681,55574,'#FFFFFF'),(20682,55575,'#FFFFFF'),(20683,55576,'#FFFFFF'),(20684,55577,'#08d3bb'),(20685,55578,'#535bd0'),(20686,55579,'#FFFFFF'),(20687,55580,'#FFFFFF'),(20688,55582,'#FFFFFF'),(20689,55583,'#5cf666'),(20690,55584,'#FFFFFF'),(20691,55585,'#FFFFFF'),(20692,55586,'#FFFFFF'),(20693,55588,'#FFFFFF'),(20694,55589,'#d630b5'),(20695,55590,'#FFFFFF'),(20696,55592,'#cc1e87'),(20697,55593,'#e2be53'),(20698,55595,'#932ad8'),(20699,55597,'#FFFFFF'),(20700,55598,'#a46cd9'),(20701,55599,'#FFFFFF'),(20702,55600,'#ecfb3c'),(20703,55601,'#FFFFFF'),(20704,55602,'#3e3dd9'),(20705,55603,'#FFFFFF'),(20706,55604,'#e068e8'),(20707,55605,'#91df6d'),(20708,55606,'#FFFFFF'),(20709,55607,'#f979d6'),(20710,55609,'#FFFFFF'),(20711,55610,'#c80dfd'),(20712,55611,'#FFFFFF'),(20713,55612,'#FFFFFF'),(20714,55614,'#f670b4'),(20715,55616,'#FFFFFF'),(20716,55617,'#FFFFFF'),(20717,55618,'#FFFFFF'),(20718,55619,'#e1a541'),(20719,55620,'#FFFFFF'),(20720,55621,'#FFFFFF'),(20721,55622,'#b763de'),(20722,55623,'#eb6138'),(20723,55624,'#d74bfa'),(20724,55626,'#da9569'),(20725,55627,'#d0cc09'),(20726,55628,'#FFFFFF'),(20727,55629,'#FFFFFF'),(20728,55631,'#FFFFFF'),(20729,55632,'#FFFFFF'),(20730,55633,'#FFFFFF'),(20731,55634,'#e7e86b'),(20732,55636,'#9fe762'),(20733,55638,'#f93040'),(20734,55639,'#FFFFFF'),(20735,55640,'#FFFFFF'),(20736,55641,'#cf278b'),(20737,55642,'#FFFFFF'),(20738,55643,'#FFFFFF'),(20739,55644,'#FFFFFF'),(20740,55645,'#21dcf2'),(20741,55646,'#FFFFFF'),(20742,55647,'#FFFFFF'),(20743,55649,'#FFFFFF'),(20744,55651,'#e84d49'),(20745,55652,'#005ddd'),(20746,55654,'#37ea3a'),(20747,55655,'#FFFFFF'),(20748,55657,'#FFFFFF'),(20749,55658,'#4c63f6'),(20750,55659,'#FFFFFF'),(20751,55661,'#FFFFFF'),(20752,55662,'#d04ac9'),(20753,55664,'#FFFFFF'),(20754,55665,'#FFFFFF'),(20755,55666,'#FFFFFF'),(20756,55667,'#FFFFFF'),(20757,55668,'#FFFFFF'),(20758,55669,'#FFFFFF'),(20759,55671,'#e07b51'),(20760,55672,'#FFFFFF'),(20761,55673,'#f56282'),(20762,55674,'#e35a92'),(20763,55675,'#86ef4e'),(20764,55677,'#49cd38'),(20765,55679,'#ddf82d'),(20766,55680,'#23dddd'),(20767,55681,'#a92de0'),(20768,55682,'#FFFFFF'),(20769,55684,'#FFFFFF'),(20770,55685,'#FFFFFF'),(20771,55686,'#FFFFFF'),(20772,55687,'#FFFFFF'),(20773,55688,'#FFFFFF'),(20774,55690,'#FFFFFF'),(20775,55691,'#19bfe6'),(20776,55692,'#FFFFFF'),(20777,55693,'#FFFFFF'),(20778,55743,'#0ce6c9'),(20779,55748,'#3dce79'),(20780,55847,'#FFFFFF'),(20781,55860,'#6efa53'),(20782,55866,'#fe4b74'),(20783,55905,'#e8f24d'),(20784,55911,'#e34bf6'),(20785,55919,'#78cf64'),(20786,55923,'#FFFFFF'),(20787,55924,'#49fb6f'),(20788,55925,'#2acbd7'),(20789,55926,'#FFFFFF'),(20790,55928,'#FFFFFF'),(20791,55929,'#4770fc'),(20792,55931,'#3bdbf2'),(20793,55932,'#FFFFFF'),(20794,55933,'#FFFFFF'),(20795,55934,'#53d59b'),(20796,55935,'#7296e7'),(20797,55936,'#FFFFFF'),(20798,55937,'#5adeb8'),(20799,55938,'#FFFFFF'),(20800,55939,'#ea5148'),(20801,55942,'#82f107'),(20802,55943,'#FFFFFF'),(20803,55944,'#FFFFFF'),(20804,55945,'#FFFFFF'),(20805,55946,'#FFFFFF'),(20806,55949,'#FFFFFF'),(20807,55950,'#FFFFFF'),(20808,55951,'#FFFFFF'),(20809,55952,'#FFFFFF'),(20810,55956,'#FFFFFF'),(20811,55957,'#FFFFFF'),(20812,55958,'#FFFFFF'),(20813,55960,'#FFFFFF'),(20814,55961,'#FFFFFF'),(20815,55962,'#FFFFFF'),(20816,55963,'#FFFFFF'),(20817,55965,'#6a3bea'),(20818,55966,'#0fde76'),(20819,55967,'#62e604'),(20820,55968,'#FFFFFF'),(20821,55969,'#FFFFFF'),(20822,55971,'#FFFFFF'),(20823,55972,'#8860cc'),(20824,55974,'#a217cd'),(20825,55975,'#9842eb'),(20826,55976,'#FFFFFF'),(20827,55977,'#FFFFFF'),(20828,55979,'#FFFFFF'),(20829,55980,'#65efad'),(20830,55981,'#FFFFFF'),(20831,55982,'#FFFFFF'),(20832,55983,'#f93c68'),(20833,55984,'#FFFFFF'),(20834,55985,'#0bbce5'),(20835,55986,'#73ebd8'),(20836,55988,'#55dfef'),(20837,55989,'#de06b2'),(20838,55990,'#FFFFFF'),(20839,55991,'#e969eb'),(20840,55992,'#8adb4f'),(20841,55993,'#56d7a5'),(20842,55994,'#FFFFFF'),(20843,55995,'#FFFFFF'),(20844,55997,'#45ddd2'),(20845,55998,'#62d8a1'),(20846,55999,'#FFFFFF'),(20847,56000,'#FFFFFF'),(20848,56001,'#FFFFFF'),(20849,56002,'#FFFFFF'),(20850,56003,'#FFFFFF'),(20851,56004,'#480cd0'),(20852,56006,'#FFFFFF'),(20853,56007,'#FFFFFF'),(20854,56008,'#FFFFFF'),(20855,56009,'#FFFFFF'),(20856,56010,'#FFFFFF'),(20857,56011,'#db11c4'),(20858,56012,'#FFFFFF'),(20859,56015,'#FFFFFF'),(20860,56017,'#FFFFFF'),(20861,56018,'#FFFFFF'),(20862,56019,'#FFFFFF'),(20863,56020,'#FFFFFF'),(20864,56021,'#FFFFFF'),(20865,56022,'#c904de'),(20866,56024,'#3e5beb'),(20867,56025,'#24f987'),(20868,56027,'#e00dd8'),(20869,56028,'#FFFFFF'),(20870,56029,'#FFFFFF'),(20871,56030,'#FFFFFF'),(20872,56031,'#43c7f1'),(20873,56032,'#48dae5'),(20874,56034,'#FFFFFF'),(20875,56035,'#FFFFFF'),(20876,56037,'#FFFFFF'),(20877,56038,'#f324b1'),(20878,56039,'#FFFFFF'),(20879,56040,'#FFFFFF'),(20880,56041,'#8f62e7'),(20881,56042,'#FFFFFF'),(20882,56043,'#f0044a'),(20883,56044,'#FFFFFF'),(20884,56045,'#FFFFFF'),(20885,56048,'#16a2fe'),(20886,56049,'#FFFFFF'),(20887,56050,'#11fded'),(20888,56052,'#2ecdc0'),(20889,56054,'#FFFFFF'),(20890,56055,'#FFFFFF'),(20891,56056,'#FFFFFF'),(20892,56057,'#FFFFFF'),(20893,56059,'#FFFFFF'),(20894,56060,'#FFFFFF'),(20895,56061,'#FFFFFF'),(20896,56062,'#45e8e6'),(20897,56064,'#FFFFFF'),(20898,56065,'#24fe83'),(20899,56066,'#9d50f4'),(20900,56067,'#307be0'),(20901,56068,'#ae09fd'),(20902,56070,'#e13794'),(20903,56072,'#4beac6'),(20904,56073,'#51e941'),(20905,56074,'#276ced'),(20906,56075,'#FFFFFF'),(20907,56078,'#FFFFFF'),(20908,56079,'#f76ddd'),(20909,56080,'#60cd38'),(20910,56081,'#FFFFFF'),(20911,56082,'#bd78fa'),(20912,56084,'#FFFFFF'),(20913,56085,'#79faf5'),(20914,56086,'#FFFFFF'),(20915,56087,'#FFFFFF'),(20916,56088,'#FFFFFF'),(20917,56089,'#FFFFFF'),(20918,56090,'#FFFFFF'),(20919,56091,'#FFFFFF'),(20920,56092,'#FFFFFF'),(20921,56093,'#FFFFFF'),(20922,56094,'#FFFFFF'),(20923,56095,'#FFFFFF'),(20924,56096,'#75fef3'),(20925,56097,'#FFFFFF'),(20926,56098,'#FFFFFF'),(20927,56099,'#FFFFFF'),(20928,56100,'#FFFFFF'),(20929,56101,'#FFFFFF'),(20930,56102,'#4baef2'),(20931,56103,'#FFFFFF'),(20932,56105,'#FFFFFF'),(20933,56106,'#FFFFFF'),(20934,56107,'#FFFFFF'),(20935,56108,'#FFFFFF'),(20936,56109,'#FFFFFF'),(20937,56110,'#FFFFFF'),(20938,56112,'#FFFFFF'),(20939,56113,'#26d232'),(20940,56115,'#FFFFFF'),(20941,56116,'#5450de'),(20942,56117,'#10a6fc'),(20943,56118,'#d64ee5'),(20944,56119,'#32abcc'),(20945,56121,'#FFFFFF'),(20946,56122,'#FFFFFF'),(20947,56123,'#FFFFFF'),(20948,56124,'#FFFFFF'),(20949,56125,'#FFFFFF'),(20950,56126,'#5df930'),(20951,56127,'#FFFFFF'),(20952,56128,'#FFFFFF'),(20953,56129,'#FFFFFF'),(20954,56131,'#FFFFFF'),(20955,56132,'#FFFFFF'),(20956,56133,'#FFFFFF'),(20957,56134,'#FFFFFF'),(20958,56135,'#FFFFFF'),(20959,56138,'#FFFFFF'),(20960,56139,'#FFFFFF'),(20961,56140,'#39c8fe'),(20962,56141,'#FFFFFF'),(20963,56144,'#FFFFFF'),(20964,56145,'#FFFFFF'),(20965,56146,'#c4fe66'),(20966,56147,'#FFFFFF'),(20967,56148,'#FFFFFF'),(20968,56150,'#FFFFFF'),(20969,56151,'#FFFFFF'),(20970,56154,'#FFFFFF'),(20971,56155,'#FFFFFF'),(20972,56156,'#FFFFFF'),(20973,56157,'#2bdcf6'),(20974,56158,'#953cfe'),(20975,56160,'#d20685'),(20976,56161,'#FFFFFF'),(20977,56162,'#FFFFFF'),(20978,56163,'#4e4ceb'),(20979,56164,'#FFFFFF'),(20980,56165,'#FFFFFF'),(20981,56166,'#FFFFFF'),(20982,56168,'#c1dc6b'),(20983,56169,'#FFFFFF'),(20984,56170,'#FFFFFF'),(20985,56171,'#0be7a2'),(20986,56172,'#FFFFFF'),(20987,56173,'#FFFFFF'),(20988,56174,'#27ef27'),(20989,56175,'#FFFFFF'),(20990,56176,'#FFFFFF'),(20991,56178,'#FFFFFF'),(20992,56179,'#cd6e1e'),(20993,56180,'#FFFFFF'),(20994,56181,'#FFFFFF'),(20995,56182,'#FFFFFF'),(20996,56183,'#FFFFFF'),(20997,56184,'#6dfe22'),(20998,56185,'#ec7117'),(20999,56186,'#FFFFFF'),(21000,56187,'#3fea11'),(21001,56190,'#FFFFFF'),(21002,56191,'#FFFFFF'),(21003,56192,'#FFFFFF'),(21004,56193,'#f06dc7'),(21005,56194,'#3dc3e3'),(21006,56196,'#FFFFFF'),(21007,56199,'#FFFFFF'),(21008,56200,'#ea7b36'),(21009,56201,'#FFFFFF'),(21010,56202,'#FFFFFF'),(21011,56203,'#c7f776'),(21012,56204,'#FFFFFF'),(21013,56205,'#FFFFFF'),(21014,56207,'#c2cc59'),(21015,56208,'#FFFFFF'),(21016,56209,'#FFFFFF'),(21017,56211,'#FFFFFF'),(21018,56212,'#73f1be'),(21019,56213,'#FFFFFF'),(21020,56214,'#497efc'),(21021,56215,'#ea64de'),(21022,56216,'#5dacde'),(21023,56217,'#4546d3'),(21024,56218,'#FFFFFF'),(21025,56219,'#3fcf97'),(21026,56220,'#FFFFFF'),(21027,56221,'#f4e64f'),(21028,56223,'#FFFFFF'),(21029,56225,'#33ce72'),(21030,56226,'#57cf9f'),(21031,56228,'#FFFFFF'),(21032,56232,'#FFFFFF'),(21033,56233,'#FFFFFF'),(21034,56234,'#f978e7'),(21035,56237,'#FFFFFF'),(21036,56238,'#FFFFFF'),(21037,56239,'#FFFFFF'),(21038,56241,'#FFFFFF'),(21039,56242,'#FFFFFF'),(21040,56243,'#a3db59'),(21041,56246,'#FFFFFF'),(21042,56249,'#FFFFFF'),(21043,56250,'#FFFFFF'),(21044,56251,'#FFFFFF'),(21045,56252,'#FFFFFF'),(21046,56253,'#98df02'),(21047,56254,'#FFFFFF'),(21048,56255,'#FFFFFF'),(21049,56256,'#FFFFFF'),(21050,56257,'#FFFFFF'),(21051,56258,'#FFFFFF'),(21052,56259,'#FFFFFF'),(21053,56260,'#FFFFFF'),(21054,56261,'#FFFFFF'),(21055,56262,'#cad003'),(21056,56264,'#FFFFFF'),(21057,56265,'#c5d52a'),(21058,56266,'#FFFFFF'),(21059,56267,'#FFFFFF'),(21060,56268,'#FFFFFF'),(21061,56269,'#FFFFFF'),(21062,56270,'#FFFFFF'),(21063,56271,'#FFFFFF'),(21064,56272,'#FFFFFF'),(21065,56273,'#FFFFFF'),(21066,56274,'#d3f91e'),(21067,56275,'#FFFFFF'),(21068,56276,'#FFFFFF'),(21069,56277,'#FFFFFF'),(21070,56278,'#FFFFFF'),(21071,56279,'#FFFFFF'),(21072,56280,'#FFFFFF'),(21073,56281,'#FFFFFF'),(21074,56282,'#FFFFFF'),(21075,56283,'#FFFFFF'),(21076,56284,'#FFFFFF'),(21077,56285,'#FFFFFF'),(21078,56286,'#3f8cd3'),(21079,56287,'#cd7921'),(21080,56288,'#FFFFFF'),(21081,56289,'#FFFFFF'),(21082,56290,'#FFFFFF'),(21083,56291,'#e111f4'),(21084,56292,'#FFFFFF'),(21085,56294,'#48e3e9'),(21086,56296,'#39e24d'),(21087,56297,'#FFFFFF'),(21088,56298,'#abdb0d'),(21089,56300,'#e9a967'),(21090,56303,'#FFFFFF'),(21091,56305,'#f5db6f'),(21092,56306,'#3767ed'),(21093,56307,'#f90081'),(21094,56308,'#d35ec6'),(21095,56311,'#88df4f'),(21096,56317,'#d53397'),(21097,56318,'#e424fe'),(21098,56321,'#e0167e'),(21099,56325,'#f040e2'),(21100,56328,'#cbd515'),(21101,56330,'#dc7455'),(21102,56331,'#FFFFFF'),(21103,56333,'#4f8bd9'),(21104,56334,'#f51955'),(21105,56335,'#8e57da'),(21106,56336,'#6ae0a5'),(21107,56341,'#f4b827'),(21108,56342,'#9d74fb'),(21109,56349,'#d9b22e'),(21110,56350,'#9e5ef6'),(21111,56353,'#5ab9f5'),(21112,56356,'#c866da'),(21113,56357,'#33c8ea'),(21114,56358,'#e9ab40'),(21115,56359,'#fa50d3'),(21116,56366,'#f75052'),(21117,56367,'#FFFFFF'),(21118,56368,'#c6f601'),(21119,56370,'#FFFFFF'),(21120,56373,'#FFFFFF'),(21121,56378,'#FFFFFF'),(21122,56390,'#2910f6'),(21123,56393,'#013bea'),(21124,56395,'#1973e0'),(21125,56400,'#f75755'),(21126,56401,'#6d07f8'),(21127,56402,'#62d254'),(21128,56407,'#ea7049'),(21129,56409,'#e45ba4'),(21130,56413,'#872ae7'),(21131,56414,'#06df6c'),(21132,56416,'#df69a1'),(21133,56419,'#62aed5'),(21134,56431,'#d03b42'),(21135,56433,'#ea6fc6'),(21136,56437,'#FFFFFF'),(21137,56438,'#ed08be'),(21138,56443,'#eb7272'),(21139,56445,'#db651e'),(21140,56446,'#4cd68d'),(21141,56447,'#be4afe'),(21142,56449,'#FFFFFF'),(21143,56451,'#f9be3f'),(21144,56455,'#FFFFFF'),(21145,56459,'#ef1b35'),(21146,56461,'#cc7645'),(21147,56465,'#4dee74'),(21148,56466,'#49f091'),(21149,56467,'#2fe83c'),(21150,56469,'#e74f48'),(21151,56470,'#f66c0d'),(21152,56471,'#FFFFFF'),(21153,56472,'#eebd08'),(21154,56473,'#FFFFFF'),(21155,56477,'#FFFFFF'),(21156,56478,'#FFFFFF'),(21157,56481,'#FFFFFF'),(21158,56483,'#FFFFFF'),(21159,56489,'#9ccc41'),(21160,56490,'#FFFFFF'),(21161,56491,'#e340de'),(21162,56492,'#4dcc72'),(21163,56498,'#18ee28'),(21164,56499,'#f15ca9'),(21165,56506,'#d5ea6d'),(21166,56507,'#FFFFFF'),(21167,56509,'#2064d6'),(21168,56510,'#2ad4ee'),(21169,56515,'#FFFFFF'),(21170,56522,'#ef7b0b'),(21171,56523,'#FFFFFF'),(21172,56524,'#FFFFFF'),(21173,56525,'#FFFFFF'),(21174,56526,'#4586d1'),(21175,56530,'#FFFFFF'),(21176,56531,'#FFFFFF'),(21177,56532,'#FFFFFF'),(21178,56533,'#77f156'),(21179,56534,'#FFFFFF'),(21180,56539,'#38c3e9'),(21181,56540,'#6db0fa'),(21182,56541,'#5b84d4'),(21183,56542,'#FFFFFF'),(21184,56543,'#64e5f5'),(21185,56544,'#FFFFFF'),(21186,56545,'#26fa9b'),(21187,56546,'#e25de0'),(21188,56548,'#71fbde'),(21189,56549,'#FFFFFF'),(21190,56550,'#f23faa'),(21191,56551,'#FFFFFF'),(21192,56552,'#FFFFFF'),(21193,56553,'#0ceb81'),(21194,56554,'#FFFFFF'),(21195,56555,'#FFFFFF'),(21196,56556,'#FFFFFF'),(21197,56557,'#FFFFFF'),(21198,56558,'#FFFFFF'),(21199,56560,'#cc0400'),(21200,56562,'#FFFFFF'),(21201,56565,'#9e3bf5'),(21202,56567,'#af48d7'),(21203,56568,'#e53ed2'),(21204,56569,'#FFFFFF'),(21205,56570,'#FFFFFF'),(21206,56571,'#7bf87c'),(21207,56573,'#FFFFFF'),(21208,56574,'#2ed67b'),(21209,56575,'#1ad1e1'),(21210,56577,'#ceb830'),(21211,56578,'#6182f1'),(21212,56581,'#ebe874'),(21213,56582,'#45f671'),(21214,56586,'#FFFFFF'),(21215,56587,'#FFFFFF'),(21216,56588,'#44f07f'),(21217,56589,'#eb6b56'),(21218,56595,'#cd37cf'),(21219,56597,'#FFFFFF'),(21220,56603,'#8cfe76'),(21221,56604,'#b01ccd'),(21222,56605,'#FFFFFF'),(21223,56606,'#FFFFFF'),(21224,56609,'#7459f1'),(21225,56611,'#FFFFFF'),(21226,56615,'#0070d6'),(21227,56620,'#FFFFFF'),(21228,56621,'#FFFFFF'),(21229,56623,'#4ae2a9'),(21230,56624,'#FFFFFF'),(21231,56625,'#FFFFFF'),(21232,56626,'#FFFFFF'),(21233,56629,'#FFFFFF'),(21234,56630,'#b5d23c'),(21235,56631,'#FFFFFF'),(21236,56632,'#FFFFFF'),(21237,56638,'#0f71db'),(21238,56639,'#73efe6'),(21239,56642,'#d157c2'),(21240,56644,'#FFFFFF'),(21241,56645,'#FFFFFF'),(21242,56646,'#FFFFFF'),(21243,56648,'#f0be74'),(21244,56655,'#FFFFFF'),(21245,56657,'#FFFFFF'),(21246,56661,'#FFFFFF'),(21247,56662,'#f80654'),(21248,56663,'#FFFFFF'),(21249,56664,'#FFFFFF'),(21250,56666,'#deaf69'),(21251,56668,'#FFFFFF'),(21252,56671,'#FFFFFF'),(21253,56672,'#e109ee'),(21254,56675,'#700dda'),(21255,56677,'#FFFFFF'),(21256,56678,'#FFFFFF'),(21257,56679,'#5b64d2'),(21258,56680,'#FFFFFF'),(21259,56681,'#FFFFFF'),(21260,56684,'#FFFFFF'),(21261,56685,'#cf647b'),(21262,56686,'#ccf044'),(21263,56689,'#FFFFFF'),(21264,56691,'#FFFFFF'),(21265,56692,'#ca51d4'),(21266,56694,'#FFFFFF'),(21267,56698,'#FFFFFF'),(21268,56699,'#b268f5'),(21269,56702,'#FFFFFF'),(21270,56704,'#e73f52'),(21271,56706,'#FFFFFF'),(21272,56707,'#FFFFFF'),(21273,56711,'#FFFFFF'),(21274,56717,'#FFFFFF'),(21275,56718,'#FFFFFF'),(21276,56720,'#9cfa57'),(21277,56721,'#39d837'),(21278,56723,'#de2b23'),(21279,56725,'#FFFFFF'),(21280,56730,'#FFFFFF'),(21281,56733,'#1d1de2'),(21282,56737,'#2aa2fc'),(21283,56738,'#bbe265'),(21284,56740,'#6b9fde'),(21285,56741,'#c510cd'),(21286,56742,'#2931fa'),(21287,56743,'#85d260'),(21288,56745,'#FFFFFF'),(21289,56749,'#0eed47'),(21290,56750,'#f19e62'),(21291,56754,'#d47205'),(21292,56756,'#FFFFFF'),(21293,56757,'#FFFFFF'),(21294,56765,'#5b14dc'),(21295,56766,'#3bd903'),(21296,56773,'#4eef15'),(21297,56778,'#50e2c8'),(21298,56782,'#FFFFFF'),(21299,56784,'#2b63f9'),(21300,56785,'#FFFFFF'),(21301,56790,'#FFFFFF'),(21302,56793,'#a8d119'),(21303,56794,'#FFFFFF'),(21304,56795,'#3b4de1'),(21305,56800,'#FFFFFF'),(21306,56802,'#ee2550'),(21307,56810,'#FFFFFF'),(21308,56813,'#76d463'),(21309,56814,'#51dd54'),(21310,56815,'#07de28'),(21311,56816,'#FFFFFF'),(21312,56817,'#FFFFFF'),(21313,56818,'#2cd5d9'),(21314,56819,'#FFFFFF'),(21315,56820,'#12d8ca'),(21316,56825,'#FFFFFF'),(21317,56827,'#FFFFFF'),(21318,56831,'#FFFFFF'),(21319,56833,'#FFFFFF'),(21320,56838,'#fcb077'),(21321,56839,'#FFFFFF'),(21322,56840,'#e47e3c'),(21323,56848,'#df6ab9'),(21324,56850,'#FFFFFF'),(21325,56851,'#FFFFFF'),(21326,56854,'#FFFFFF'),(21327,56858,'#3c35f1'),(21328,56859,'#FFFFFF'),(21329,56863,'#f654bc'),(21330,56867,'#FFFFFF'),(21331,56870,'#1ee8a6'),(21332,56875,'#5277f4'),(21333,56876,'#3b20dc'),(21334,56877,'#d571e8'),(21335,56879,'#d5c249'),(21336,56880,'#FFFFFF'),(21337,56881,'#FFFFFF'),(21338,56884,'#4d88cf'),(21339,56885,'#3e37ec'),(21340,56886,'#d05b35'),(21341,56890,'#FFFFFF'),(21342,56893,'#e0aa58'),(21343,56895,'#ed3797'),(21344,56900,'#FFFFFF'),(21345,56906,'#FFFFFF'),(21346,56907,'#7cfad6'),(21347,56909,'#db1a92'),(21348,56914,'#ec6be0'),(21349,56915,'#cd54b5'),(21350,56916,'#da1812'),(21351,56917,'#0c9fea'),(21352,56919,'#9b6ad6'),(21353,56922,'#FFFFFF'),(21354,56924,'#df9d13'),(21355,56928,'#7fcd0b'),(21356,56929,'#d6af50'),(21357,56930,'#FFFFFF'),(21358,56932,'#FFFFFF'),(21359,56933,'#35d4c5'),(21360,56937,'#FFFFFF'),(21361,56938,'#ea1ac7'),(21362,56939,'#FFFFFF'),(21363,56940,'#FFFFFF'),(21364,56941,'#FFFFFF'),(21365,56942,'#FFFFFF'),(21366,56946,'#FFFFFF'),(21367,56948,'#e7cb28'),(21368,56949,'#FFFFFF'),(21369,56952,'#fc7a89'),(21370,56954,'#a753d1'),(21371,56955,'#FFFFFF'),(21372,56957,'#ef6f44'),(21373,56959,'#FFFFFF'),(21374,56962,'#FFFFFF'),(21375,56963,'#FFFFFF'),(21376,56965,'#FFFFFF'),(21377,56966,'#6247fa'),(21378,56967,'#FFFFFF'),(21379,56968,'#FFFFFF'),(21380,56969,'#FFFFFF'),(21381,56972,'#4eb9e0'),(21382,56974,'#FFFFFF'),(21383,56975,'#FFFFFF'),(21384,56976,'#FFFFFF'),(21385,56977,'#e7ad6c'),(21386,56978,'#FFFFFF'),(21387,56979,'#3e64dd'),(21388,56980,'#FFFFFF'),(21389,56981,'#43b0cc'),(21390,56983,'#279ce3'),(21391,56985,'#dc0a80'),(21392,56987,'#0efb8b'),(21393,56989,'#6a30d2'),(21394,56993,'#27f9d5'),(21395,56994,'#FFFFFF'),(21396,56995,'#FFFFFF'),(21397,56996,'#FFFFFF'),(21398,56997,'#FFFFFF'),(21399,56998,'#FFFFFF'),(21400,57000,'#6fe9fc'),(21401,57004,'#97f84d'),(21402,57006,'#6732d4'),(21403,57007,'#d3843e'),(21404,57015,'#1a23d6'),(21405,57024,'#FFFFFF'),(21406,57025,'#b1fd1c'),(21407,57026,'#FFFFFF'),(21408,57027,'#FFFFFF'),(21409,57028,'#1b94de'),(21410,57030,'#bdda16'),(21411,57031,'#FFFFFF'),(21412,57034,'#4b20ec'),(21413,57036,'#e90b5b'),(21414,57044,'#FFFFFF'),(21415,57047,'#ce36fc'),(21416,57050,'#2ee78a'),(21417,57052,'#FFFFFF'),(21418,57056,'#d8a51b'),(21419,57057,'#FFFFFF'),(21420,57058,'#FFFFFF'),(21421,57061,'#FFFFFF'),(21422,57062,'#FFFFFF'),(21423,57063,'#FFFFFF'),(21424,57064,'#0496da'),(21425,57067,'#9cf56d'),(21426,57068,'#d08653'),(21427,57069,'#FFFFFF'),(21428,57070,'#cf701b'),(21429,57072,'#f2821d'),(21430,57077,'#de2fee'),(21431,57079,'#FFFFFF'),(21432,57080,'#FFFFFF'),(21433,57085,'#6576d8'),(21434,57091,'#FFFFFF'),(21435,57093,'#FFFFFF'),(21436,57094,'#cbd81e'),(21437,57095,'#d8054b'),(21438,57097,'#FFFFFF'),(21439,57100,'#2dea37'),(21440,57102,'#6ff963'),(21441,57104,'#ac2eda'),(21442,57109,'#245dcd'),(21443,57110,'#FFFFFF'),(21444,57111,'#FFFFFF'),(21445,57115,'#6b54fc'),(21446,57119,'#587ce0'),(21447,57121,'#afd31e'),(21448,57123,'#FFFFFF'),(21449,57124,'#b237d0'),(21450,57125,'#FFFFFF'),(21451,57126,'#FFFFFF'),(21452,57127,'#5eeaee'),(21453,57130,'#FFFFFF'),(21454,57132,'#0ae96c'),(21455,57133,'#FFFFFF'),(21456,57136,'#FFFFFF'),(21457,57137,'#FFFFFF'),(21458,57140,'#8b2dec'),(21459,57142,'#7be53e'),(21460,57143,'#FFFFFF'),(21461,57144,'#03d79c'),(21462,57145,'#FFFFFF'),(21463,57149,'#cc5aa1'),(21464,57150,'#095ccf'),(21465,57152,'#FFFFFF'),(21466,57153,'#FFFFFF'),(21467,57154,'#0553cf'),(21468,57155,'#05e5ba'),(21469,57156,'#d9651f'),(21470,57158,'#d4520c'),(21471,57159,'#FFFFFF'),(21472,57161,'#FFFFFF'),(21473,57163,'#FFFFFF'),(21474,57164,'#FFFFFF'),(21475,57165,'#FFFFFF'),(21476,57166,'#ed758d'),(21477,57167,'#FFFFFF'),(21478,57170,'#4523d7'),(21479,57171,'#FFFFFF'),(21480,57172,'#d13930'),(21481,57173,'#46e876'),(21482,57175,'#7acd25'),(21483,57176,'#FFFFFF'),(21484,57177,'#FFFFFF'),(21485,57179,'#6efd7b'),(21486,57182,'#afe55c'),(21487,57183,'#FFFFFF'),(21488,57184,'#6544cf'),(21489,57185,'#e8775b'),(21490,57187,'#FFFFFF'),(21491,57188,'#FFFFFF'),(21492,57189,'#a5fb05'),(21493,57192,'#FFFFFF'),(21494,57194,'#FFFFFF'),(21495,57195,'#FFFFFF'),(21496,57196,'#e205f8'),(21497,57197,'#4720d5'),(21498,57198,'#f91dcd'),(21499,57200,'#14e248'),(21500,57201,'#FFFFFF'),(21501,57202,'#FFFFFF'),(21502,57203,'#FFFFFF'),(21503,57204,'#cc39cb'),(21504,57205,'#6466e5'),(21505,57206,'#FFFFFF'),(21506,57207,'#FFFFFF'),(21507,57208,'#FFFFFF'),(21508,57210,'#FFFFFF'),(21509,57211,'#c3ee2f'),(21510,57212,'#FFFFFF'),(21511,57213,'#FFFFFF'),(21512,57214,'#FFFFFF'),(21513,57216,'#FFFFFF'),(21514,57217,'#6483f0'),(21515,57218,'#d48918'),(21516,57220,'#47f702'),(21517,57222,'#FFFFFF'),(21518,57224,'#FFFFFF'),(21519,57225,'#FFFFFF'),(21520,57226,'#FFFFFF'),(21521,57228,'#6eecf3'),(21522,57230,'#FFFFFF'),(21523,57231,'#2ff31f'),(21524,57232,'#0d53f2'),(21525,57233,'#2aeee5'),(21526,57234,'#FFFFFF'),(21527,57235,'#FFFFFF'),(21528,57238,'#FFFFFF'),(21529,57239,'#FFFFFF'),(21530,57241,'#721ae1'),(21531,57244,'#FFFFFF'),(21532,57245,'#b1d755'),(21533,57246,'#d85c7a'),(21534,57247,'#FFFFFF'),(21535,57248,'#e9346f'),(21536,57250,'#FFFFFF'),(21537,57251,'#98fd46'),(21538,57252,'#FFFFFF'),(21539,57253,'#FFFFFF'),(21540,57254,'#63f69e'),(21541,57256,'#FFFFFF'),(21542,57257,'#FFFFFF'),(21543,57259,'#f84946'),(21544,57260,'#ca59d1'),(21545,57261,'#68ec7c'),(21546,57262,'#FFFFFF'),(21547,57264,'#FFFFFF'),(21548,57265,'#614df8'),(21549,57266,'#FFFFFF'),(21550,57267,'#d5d453'),(21551,57268,'#FFFFFF'),(21552,57270,'#c550f5'),(21553,57272,'#FFFFFF'),(21554,57274,'#FFFFFF'),(21555,57275,'#FFFFFF'),(21556,57277,'#16a9d1'),(21557,57278,'#FFFFFF'),(21558,57279,'#FFFFFF'),(21559,57280,'#59e8ed'),(21560,57281,'#FFFFFF'),(21561,57283,'#22cc5f'),(21562,57284,'#4bb5d3'),(21563,57286,'#FFFFFF'),(21564,57288,'#98f724'),(21565,57290,'#d5bc28'),(21566,57291,'#FFFFFF'),(21567,57292,'#FFFFFF'),(21568,57294,'#FFFFFF'),(21569,57299,'#f21c01'),(21570,57300,'#FFFFFF'),(21571,57301,'#17e880'),(21572,57302,'#FFFFFF'),(21573,57303,'#FFFFFF'),(21574,57304,'#FFFFFF'),(21575,57305,'#FFFFFF'),(21576,57306,'#FFFFFF'),(21577,57307,'#FFFFFF'),(21578,57308,'#FFFFFF'),(21579,57310,'#0accae'),(21580,57312,'#FFFFFF'),(21581,57314,'#FFFFFF'),(21582,57316,'#b322d9'),(21583,57317,'#39a7e4'),(21584,57318,'#FFFFFF'),(21585,57319,'#FFFFFF'),(21586,57323,'#3fe668'),(21587,57324,'#34de1e'),(21588,57325,'#f169fc'),(21589,57326,'#FFFFFF'),(21590,57327,'#FFFFFF'),(21591,57328,'#57d0bc'),(21592,57329,'#FFFFFF'),(21593,57330,'#FFFFFF'),(21594,57331,'#FFFFFF'),(21595,57332,'#d6b20f'),(21596,57333,'#FFFFFF'),(21597,57334,'#FFFFFF'),(21598,57335,'#FFFFFF'),(21599,57337,'#FFFFFF'),(21600,57340,'#42e4cf'),(21601,57341,'#FFFFFF'),(21602,57342,'#FFFFFF'),(21603,57343,'#7762cd'),(21604,57344,'#fe4272'),(21605,57348,'#FFFFFF'),(21606,57349,'#39d282'),(21607,57351,'#fa4b6a'),(21608,57352,'#ef286b'),(21609,57354,'#FFFFFF'),(21610,57356,'#FFFFFF'),(21611,57360,'#9247f8'),(21612,57361,'#4fe46d'),(21613,57362,'#FFFFFF'),(21614,57368,'#2bf933'),(21615,57370,'#4cef62'),(21616,57380,'#26cfab'),(21617,57381,'#FFFFFF'),(21618,57387,'#d5c05c'),(21619,57389,'#FFFFFF'),(21620,57396,'#abd008'),(21621,57410,'#5cf389'),(21622,57420,'#FFFFFF'),(21623,57426,'#FFFFFF'),(21624,57429,'#71fd18'),(21625,57437,'#046ef5'),(21626,57438,'#FFFFFF'),(21627,57440,'#FFFFFF'),(21628,57441,'#f3fe32'),(21629,57442,'#24dd8d'),(21630,57444,'#f8910c'),(21631,57454,'#e73fd2'),(21632,57457,'#4663ea'),(21633,57459,'#FFFFFF'),(21634,57461,'#0ed20d'),(21635,57462,'#e72a52'),(21636,57464,'#4e7de0'),(21637,57468,'#af3ed9'),(21638,57469,'#64d778'),(21639,57470,'#FFFFFF'),(21640,57472,'#aae508'),(21641,57473,'#73ebc3'),(21642,57474,'#FFFFFF'),(21643,57475,'#5bd65f'),(21644,57477,'#FFFFFF'),(21645,57480,'#c41bdf'),(21646,57481,'#FFFFFF'),(21647,57482,'#FFFFFF'),(21648,57485,'#FFFFFF'),(21649,57486,'#ea40a7'),(21650,57489,'#5262cd'),(21651,57493,'#1a37f1'),(21652,57495,'#6490cf'),(21653,57497,'#e566c0'),(21654,57498,'#FFFFFF'),(21655,57502,'#FFFFFF'),(21656,57506,'#b20fd8'),(21657,57510,'#FFFFFF'),(21658,57513,'#0be7da'),(21659,57518,'#72edf2'),(21660,57519,'#469dde'),(21661,57521,'#d856b3'),(21662,57522,'#FFFFFF'),(21663,57523,'#2101d8'),(21664,57524,'#FFFFFF'),(21665,57527,'#e3cd20'),(21666,57528,'#ab65cc'),(21667,57533,'#FFFFFF'),(21668,57537,'#7c17e2'),(21669,57538,'#b969ea'),(21670,57540,'#FFFFFF'),(21671,57543,'#00df93'),(21672,57544,'#cea33a'),(21673,57545,'#FFFFFF'),(21674,57547,'#55fd96'),(21675,57549,'#e69b22'),(21676,57550,'#f1a660'),(21677,57552,'#d8896b'),(21678,57558,'#FFFFFF'),(21679,57561,'#a039f7'),(21680,57562,'#22fe5c'),(21681,57569,'#3ad8b2'),(21682,57572,'#d7f161'),(21683,57578,'#038be2'),(21684,57581,'#FFFFFF'),(21685,57582,'#e57b6d'),(21686,57583,'#ec3055'),(21687,57584,'#3b77d5'),(21688,57585,'#01d441'),(21689,57586,'#a7eb49'),(21690,57589,'#e60b07'),(21691,57594,'#69ec97'),(21692,57595,'#FFFFFF'),(21693,57596,'#e2a21b'),(21694,57599,'#458ce9'),(21695,57601,'#FFFFFF'),(21696,57605,'#e3559f'),(21697,57607,'#6ac4d8'),(21698,57608,'#FFFFFF'),(21699,57609,'#FFFFFF'),(21700,57611,'#36d10b'),(21701,57612,'#e82089'),(21702,57614,'#FFFFFF'),(21703,57617,'#FFFFFF'),(21704,57619,'#20cc39'),(21705,57625,'#ec102e'),(21706,57626,'#06e55a'),(21707,57629,'#da3550'),(21708,57631,'#71fcdc'),(21709,57632,'#ef7523'),(21710,57635,'#46fbe8'),(21711,57636,'#84ed4b'),(21712,57637,'#FFFFFF'),(21713,57638,'#9a32d3'),(21714,57639,'#b560f7'),(21715,57641,'#69d89e'),(21716,57643,'#d11b78'),(21717,57645,'#df116c'),(21718,57646,'#FFFFFF'),(21719,57647,'#FFFFFF'),(21720,57650,'#03f487'),(21721,57654,'#29d328'),(21722,57655,'#5e89ce'),(21723,57658,'#FFFFFF'),(21724,57659,'#e84870'),(21725,57660,'#f23b8d'),(21726,57661,'#f0135a'),(21727,57663,'#f8276a'),(21728,57665,'#FFFFFF'),(21729,57667,'#43e053'),(21730,57670,'#56e426'),(21731,57671,'#d405af'),(21732,57676,'#652dda'),(21733,57677,'#FFFFFF'),(21734,57680,'#3039ed'),(21735,57681,'#fa7bfa'),(21736,57684,'#FFFFFF'),(21737,57685,'#FFFFFF'),(21738,57686,'#FFFFFF'),(21739,57687,'#7cd310'),(21740,57688,'#e6bb38'),(21741,57689,'#3c1fd3'),(21742,57692,'#5441ee'),(21743,57694,'#9a27ed'),(21744,57695,'#51affd'),(21745,57696,'#40dc92'),(21746,57698,'#de0a1e'),(21747,57699,'#FFFFFF'),(21748,57700,'#489fcf'),(21749,57701,'#f7a10b'),(21750,57702,'#FFFFFF'),(21751,57706,'#FFFFFF'),(21752,57707,'#c9e059'),(21753,57708,'#65f081'),(21754,57709,'#52d92a'),(21755,57710,'#69db69'),(21756,57711,'#FFFFFF'),(21757,57712,'#a245d1'),(21758,57713,'#833dd8'),(21759,57715,'#FFFFFF'),(21760,57716,'#FFFFFF'),(21761,57717,'#FFFFFF'),(21762,57718,'#45e15b'),(21763,57719,'#ddb16c'),(21764,57720,'#FFFFFF'),(21765,57721,'#FFFFFF'),(21766,57723,'#FFFFFF'),(21767,57725,'#FFFFFF'),(21768,57726,'#b85dd1'),(21769,57727,'#28abd3'),(21770,57728,'#FFFFFF'),(21771,57729,'#3e80d3'),(21772,57731,'#c11dcd'),(21773,57733,'#d11a8c'),(21774,57734,'#FFFFFF'),(21775,57735,'#dfc556'),(21776,57736,'#FFFFFF'),(21777,57737,'#FFFFFF'),(21778,57738,'#FFFFFF'),(21779,57740,'#FFFFFF'),(21780,57741,'#f27077'),(21781,57742,'#FFFFFF'),(21782,57743,'#FFFFFF'),(21783,57744,'#FFFFFF'),(21784,57745,'#eb5e17'),(21785,57746,'#FFFFFF'),(21786,57747,'#FFFFFF'),(21787,57748,'#FFFFFF'),(21788,57749,'#FFFFFF'),(21789,57750,'#af52db'),(21790,57751,'#FFFFFF'),(21791,57752,'#FFFFFF'),(21792,57753,'#32d5de'),(21793,57754,'#14fe01'),(21794,57755,'#904ce9'),(21795,57756,'#8435d4'),(21796,57757,'#FFFFFF'),(21797,57758,'#FFFFFF'),(21798,57759,'#FFFFFF'),(21799,57760,'#28cc75'),(21800,57761,'#FFFFFF'),(21801,57762,'#FFFFFF'),(21802,57763,'#FFFFFF'),(21803,57765,'#e4452f'),(21804,57766,'#FFFFFF'),(21805,57767,'#FFFFFF'),(21806,57770,'#e33c63'),(21807,57771,'#FFFFFF'),(21808,57772,'#FFFFFF'),(21809,57773,'#5ef094'),(21810,57774,'#db0731'),(21811,57775,'#FFFFFF'),(21812,57776,'#56e386'),(21813,57779,'#FFFFFF'),(21814,57780,'#FFFFFF'),(21815,57781,'#FFFFFF'),(21816,57782,'#FFFFFF'),(21817,57783,'#d40e77'),(21818,57784,'#e6eb42'),(21819,57785,'#FFFFFF'),(21820,57786,'#46de61'),(21821,57787,'#FFFFFF'),(21822,57788,'#74d50f'),(21823,57789,'#3f1fcd'),(21824,57791,'#FFFFFF'),(21825,57792,'#FFFFFF'),(21826,57793,'#FFFFFF'),(21827,57795,'#25d7c1'),(21828,57797,'#FFFFFF'),(21829,57798,'#de8b6e'),(21830,57799,'#FFFFFF'),(21831,57800,'#9be43e'),(21832,57802,'#fa1d95'),(21833,57803,'#FFFFFF'),(21834,57804,'#dedf5f'),(21835,57805,'#3efbcd'),(21836,57806,'#FFFFFF'),(21837,57807,'#FFFFFF'),(21838,57808,'#0bf73b'),(21839,57810,'#FFFFFF'),(21840,57811,'#FFFFFF'),(21841,57812,'#FFFFFF'),(21842,57813,'#d5e218'),(21843,57814,'#FFFFFF'),(21844,57815,'#79f82a'),(21845,57816,'#3dfd95'),(21846,57817,'#eb124d'),(21847,57818,'#FFFFFF'),(21848,57820,'#FFFFFF'),(21849,57821,'#FFFFFF'),(21850,57824,'#FFFFFF'),(21851,57825,'#FFFFFF'),(21852,57826,'#ea8e2b'),(21853,57827,'#FFFFFF'),(21854,57828,'#f5936c'),(21855,57829,'#c0e259'),(21856,57830,'#c24add'),(21857,57831,'#d52b29'),(21858,57832,'#FFFFFF'),(21859,57833,'#2c5fd7'),(21860,57834,'#FFFFFF'),(21861,57835,'#FFFFFF'),(21862,57836,'#FFFFFF'),(21863,57837,'#FFFFFF'),(21864,57838,'#26c0e6'),(21865,57840,'#76efc1'),(21866,57841,'#d4c707'),(21867,57842,'#FFFFFF'),(21868,57843,'#6afe20'),(21869,57844,'#FFFFFF'),(21870,57845,'#FFFFFF'),(21871,57846,'#7a34f9'),(21872,57847,'#FFFFFF'),(21873,57848,'#FFFFFF'),(21874,57849,'#FFFFFF'),(21875,57850,'#4ed669'),(21876,57851,'#FFFFFF'),(21877,57852,'#68f02e'),(21878,57853,'#FFFFFF'),(21879,57854,'#4c81d8'),(21880,57855,'#6b59db'),(21881,57856,'#FFFFFF'),(21882,57857,'#ed5d0f'),(21883,57858,'#FFFFFF'),(21884,57859,'#FFFFFF'),(21885,57861,'#FFFFFF'),(21886,57862,'#03d395'),(21887,57863,'#FFFFFF'),(21888,57864,'#e44183'),(21889,57865,'#4470da'),(21890,57866,'#e4f957'),(21891,57868,'#61eb2c'),(21892,57869,'#FFFFFF'),(21893,57871,'#3752d1'),(21894,57872,'#6486dc'),(21895,57874,'#672ee7'),(21896,57876,'#FFFFFF'),(21897,57877,'#FFFFFF'),(21898,57878,'#FFFFFF'),(21899,57879,'#fd6dbd'),(21900,57881,'#FFFFFF'),(21901,57882,'#40ec93'),(21902,57883,'#FFFFFF'),(21903,57884,'#dc4981'),(21904,57885,'#FFFFFF'),(21905,57886,'#FFFFFF'),(21906,57887,'#FFFFFF'),(21907,57888,'#ea68d1'),(21908,57889,'#66f7c0'),(21909,57890,'#d5dc6d'),(21910,57891,'#333deb'),(21911,57892,'#FFFFFF'),(21912,57893,'#3227f6'),(21913,57895,'#FFFFFF'),(21914,57896,'#FFFFFF'),(21915,57897,'#FFFFFF'),(21916,57898,'#FFFFFF'),(21917,57899,'#5b37ed'),(21918,57900,'#FFFFFF'),(21919,57901,'#FFFFFF'),(21920,57902,'#FFFFFF'),(21921,57903,'#cf2ac2'),(21922,57904,'#FFFFFF'),(21923,57905,'#FFFFFF'),(21924,57907,'#FFFFFF'),(21925,57908,'#166ae6'),(21926,57912,'#FFFFFF'),(21927,57913,'#63bdd5'),(21928,57914,'#77e304'),(21929,57915,'#8137e4'),(21930,57916,'#FFFFFF'),(21931,57917,'#FFFFFF'),(21932,57919,'#fd9d3d'),(21933,57920,'#f5e316'),(21934,57921,'#37b4da'),(21935,57922,'#dc2535'),(21936,57924,'#0ce6ca'),(21937,57925,'#e83672'),(21938,57926,'#FFFFFF'),(21939,57927,'#FFFFFF'),(21940,57928,'#FFFFFF'),(21941,57929,'#16d22a'),(21942,57931,'#bbf462'),(21943,57932,'#17ee49'),(21944,57933,'#236ad3'),(21945,57934,'#e75a48'),(21946,57935,'#FFFFFF'),(21947,57936,'#5fefe5'),(21948,57937,'#8c48d7'),(21949,57938,'#cc39b2'),(21950,57939,'#ebe554'),(21951,57941,'#16fbca'),(21952,57942,'#FFFFFF'),(21953,57945,'#FFFFFF'),(21954,57946,'#FFFFFF'),(21955,57947,'#FFFFFF'),(21956,57950,'#d2388f'),(21957,57951,'#fe18f5'),(21958,57952,'#FFFFFF'),(21959,57953,'#edbc48'),(21960,57954,'#FFFFFF'),(21961,57955,'#FFFFFF'),(21962,57957,'#51e312'),(21963,57959,'#FFFFFF'),(21964,57960,'#FFFFFF'),(21965,57961,'#0208ee'),(21966,57962,'#04e9c5'),(21967,57964,'#FFFFFF'),(21968,57965,'#FFFFFF'),(21969,57966,'#FFFFFF'),(21970,57967,'#40fa81'),(21971,57968,'#54d0a1'),(21972,57969,'#FFFFFF'),(21973,57970,'#FFFFFF'),(21974,57971,'#bc2dd9'),(21975,57972,'#8ae772'),(21976,57973,'#FFFFFF'),(21977,57974,'#8524e8'),(21978,57975,'#8705e3'),(21979,57976,'#df7746'),(21980,57977,'#FFFFFF'),(21981,57979,'#FFFFFF'),(21982,57980,'#d774ed'),(21983,57981,'#a73bcc'),(21984,57982,'#FFFFFF'),(21985,57983,'#FFFFFF'),(21986,57984,'#befe74'),(21987,57985,'#a45fdf'),(21988,57986,'#3feb50'),(21989,57987,'#70dd25'),(21990,57988,'#FFFFFF'),(21991,57989,'#FFFFFF'),(21992,57990,'#FFFFFF'),(21993,57991,'#280ae6'),(21994,57993,'#fe67ae'),(21995,57994,'#f72561'),(21996,57995,'#FFFFFF'),(21997,57997,'#27add0'),(21998,57998,'#FFFFFF'),(21999,57999,'#FFFFFF'),(22000,58000,'#FFFFFF'),(22001,58001,'#FFFFFF'),(22002,58002,'#FFFFFF'),(22003,58003,'#FFFFFF'),(22004,58004,'#FFFFFF'),(22005,58005,'#46baf9'),(22006,58006,'#FFFFFF'),(22007,58007,'#5ff7b2'),(22008,58008,'#FFFFFF'),(22009,58009,'#42d9f3'),(22010,58011,'#FFFFFF'),(22011,58012,'#FFFFFF'),(22012,58014,'#e76474'),(22013,58015,'#FFFFFF'),(22014,58016,'#17b0df'),(22015,58017,'#fcfd63'),(22016,58018,'#FFFFFF'),(22017,58019,'#eef94a'),(22018,58020,'#7157d0'),(22019,58021,'#2be988'),(22020,58022,'#FFFFFF'),(22021,58023,'#416ce0'),(22022,58024,'#ecec35'),(22023,58025,'#FFFFFF'),(22024,58027,'#FFFFFF'),(22025,58028,'#FFFFFF'),(22026,58029,'#5bd414'),(22027,58030,'#FFFFFF'),(22028,58031,'#69d4ae'),(22029,58032,'#286bfb'),(22030,58033,'#ebcc0e'),(22031,58034,'#FFFFFF'),(22032,58035,'#3ee8ca'),(22033,58036,'#FFFFFF'),(22034,58037,'#eb3b19'),(22035,58038,'#FFFFFF'),(22036,58039,'#FFFFFF'),(22037,58040,'#4cf4f4'),(22038,58042,'#10f3b5'),(22039,58043,'#12f16f'),(22040,58044,'#32d20c'),(22041,58045,'#FFFFFF'),(22042,58046,'#FFFFFF'),(22043,58047,'#43cf28'),(22044,58048,'#d86658'),(22045,58049,'#8df83c'),(22046,58050,'#FFFFFF'),(22047,58051,'#FFFFFF'),(22048,58052,'#0f61fd'),(22049,58053,'#FFFFFF'),(22050,58055,'#FFFFFF'),(22051,58057,'#72e47d'),(22052,58058,'#13e789'),(22053,58059,'#FFFFFF'),(22054,58060,'#a3df0e'),(22055,58061,'#FFFFFF'),(22056,58062,'#FFFFFF'),(22057,58063,'#FFFFFF'),(22058,58064,'#FFFFFF'),(22059,58065,'#FFFFFF'),(22060,58066,'#e650cf'),(22061,58067,'#e407be'),(22062,58068,'#8b1cf0'),(22063,58069,'#06e4fd'),(22064,58070,'#ddaf5f'),(22065,58071,'#FFFFFF'),(22066,58072,'#FFFFFF'),(22067,58073,'#e99007'),(22068,58074,'#21daa4'),(22069,58075,'#e55228'),(22070,58076,'#5cd0ac'),(22071,58077,'#FFFFFF'),(22072,58078,'#FFFFFF'),(22073,58079,'#d3e409'),(22074,58080,'#FFFFFF'),(22075,58081,'#FFFFFF'),(22076,58082,'#FFFFFF'),(22077,58083,'#df5577'),(22078,58084,'#FFFFFF'),(22079,58085,'#FFFFFF'),(22080,58086,'#6bcdfa'),(22081,58087,'#FFFFFF'),(22082,58088,'#FFFFFF'),(22083,58089,'#FFFFFF'),(22084,58090,'#0d4dd0'),(22085,58092,'#FFFFFF'),(22086,58093,'#9804ce'),(22087,58094,'#18e90b'),(22088,58095,'#da00da'),(22089,58096,'#FFFFFF'),(22090,58097,'#FFFFFF'),(22091,58098,'#FFFFFF'),(22092,58099,'#9af677'),(22093,58100,'#FFFFFF'),(22094,58101,'#9be540'),(22095,58102,'#2ae343'),(22096,58103,'#FFFFFF'),(22097,58104,'#df6ec6'),(22098,58106,'#51abe9'),(22099,58107,'#FFFFFF'),(22100,58108,'#FFFFFF'),(22101,58109,'#FFFFFF'),(22102,58110,'#FFFFFF'),(22103,58111,'#FFFFFF'),(22104,58112,'#cd584d'),(22105,58113,'#ef6e90'),(22106,58114,'#84cc41'),(22107,58115,'#FFFFFF'),(22108,58116,'#cc307a'),(22109,58117,'#da2215'),(22110,58118,'#FFFFFF'),(22111,58119,'#2de951'),(22112,58120,'#ed1cba'),(22113,58121,'#FFFFFF'),(22114,58122,'#fbb427'),(22115,58123,'#FFFFFF'),(22116,58124,'#FFFFFF'),(22117,58125,'#dd3b06'),(22118,58126,'#FFFFFF'),(22119,58127,'#FFFFFF'),(22120,58128,'#8f29cf'),(22121,58129,'#0ccc5a'),(22122,58130,'#ea198f'),(22123,58132,'#59e20c'),(22124,58133,'#f43978'),(22125,58134,'#db224c'),(22126,58135,'#FFFFFF'),(22127,58136,'#f05b3a'),(22128,58140,'#3e8ecd'),(22129,58141,'#4845ef'),(22130,58142,'#730be2'),(22131,58143,'#c84dd2'),(22132,58145,'#FFFFFF'),(22133,58146,'#FFFFFF'),(22134,58147,'#d3d00c'),(22135,58149,'#4acd57'),(22136,58150,'#2560f9'),(22137,58151,'#1edd31'),(22138,58152,'#FFFFFF'),(22139,58153,'#d7dd46'),(22140,58154,'#FFFFFF'),(22141,58155,'#68e543'),(22142,58156,'#e3a14a'),(22143,58157,'#FFFFFF'),(22144,58158,'#FFFFFF'),(22145,58159,'#8fe713'),(22146,58160,'#03de13'),(22147,58161,'#bc39fa'),(22148,58162,'#f17851'),(22149,58164,'#ed9762'),(22150,58165,'#FFFFFF'),(22151,58166,'#FFFFFF'),(22152,58167,'#78f827'),(22153,58168,'#e9b61c'),(22154,58169,'#FFFFFF'),(22155,58170,'#FFFFFF'),(22156,58171,'#FFFFFF'),(22157,58172,'#e7fe1c'),(22158,58173,'#e35ff1'),(22159,58174,'#f9db35'),(22160,58175,'#e88e52'),(22161,58177,'#FFFFFF'),(22162,58178,'#FFFFFF'),(22163,58179,'#3260db'),(22164,58180,'#d8582e'),(22165,58181,'#FFFFFF'),(22166,58182,'#8052f6'),(22167,58183,'#1f06e1'),(22168,58184,'#FFFFFF'),(22169,58186,'#FFFFFF'),(22170,58187,'#ecf811'),(22171,58188,'#FFFFFF'),(22172,58189,'#74f581'),(22173,58191,'#81e229'),(22174,58192,'#FFFFFF'),(22175,58193,'#FFFFFF'),(22176,58194,'#e72d4c'),(22177,58195,'#6afcb1'),(22178,58196,'#FFFFFF'),(22179,58197,'#641ecf'),(22180,58198,'#FFFFFF'),(22181,58199,'#FFFFFF'),(22182,58200,'#3749e4'),(22183,58201,'#188be0'),(22184,58202,'#FFFFFF'),(22185,58203,'#2274dc'),(22186,58204,'#41e46d'),(22187,58205,'#FFFFFF'),(22188,58206,'#FFFFFF'),(22189,58207,'#63deb4'),(22190,58208,'#42e84c'),(22191,58209,'#FFFFFF'),(22192,58211,'#FFFFFF'),(22193,58212,'#55b2cc'),(22194,58213,'#FFFFFF'),(22195,58215,'#FFFFFF'),(22196,58216,'#0028f2'),(22197,58218,'#FFFFFF'),(22198,58219,'#da48f2'),(22199,58220,'#35f691'),(22200,58221,'#FFFFFF'),(22201,58222,'#FFFFFF'),(22202,58223,'#FFFFFF'),(22203,58224,'#FFFFFF'),(22204,58225,'#FFFFFF'),(22205,58226,'#FFFFFF'),(22206,58227,'#efda34'),(22207,58228,'#FFFFFF'),(22208,58230,'#FFFFFF'),(22209,58231,'#1dd30a'),(22210,58233,'#FFFFFF'),(22211,58234,'#FFFFFF'),(22212,58235,'#FFFFFF'),(22213,58236,'#FFFFFF'),(22214,58237,'#FFFFFF'),(22215,58238,'#13f838'),(22216,58239,'#fa1db7'),(22217,58240,'#b1eb06'),(22218,58241,'#FFFFFF'),(22219,58242,'#FFFFFF'),(22220,58243,'#FFFFFF'),(22221,58244,'#FFFFFF'),(22222,58245,'#FFFFFF'),(22223,58246,'#FFFFFF'),(22224,58247,'#FFFFFF'),(22225,58248,'#81d305'),(22226,58249,'#FFFFFF'),(22227,58250,'#FFFFFF'),(22228,58251,'#FFFFFF'),(22229,58252,'#68fee8'),(22230,58253,'#3bd586'),(22231,58254,'#FFFFFF'),(22232,58255,'#FFFFFF'),(22233,58256,'#f71be4'),(22234,58257,'#FFFFFF'),(22235,58258,'#e3d530'),(22236,58259,'#51d27f'),(22237,58260,'#32c2d8'),(22238,58261,'#FFFFFF'),(22239,58262,'#d762ec'),(22240,58263,'#FFFFFF'),(22241,58264,'#FFFFFF'),(22242,58265,'#fdc471'),(22243,58266,'#FFFFFF'),(22244,58267,'#FFFFFF'),(22245,58268,'#FFFFFF'),(22246,58270,'#FFFFFF'),(22247,58271,'#dc7768'),(22248,58272,'#FFFFFF'),(22249,58273,'#FFFFFF'),(22250,58274,'#dd659e'),(22251,58275,'#fa6d7f'),(22252,58277,'#df2f15'),(22253,58278,'#FFFFFF'),(22254,58279,'#FFFFFF'),(22255,58280,'#d8ee74'),(22256,58281,'#FFFFFF'),(22257,58283,'#FFFFFF'),(22258,58284,'#FFFFFF'),(22259,58285,'#FFFFFF'),(22260,58286,'#FFFFFF'),(22261,58287,'#FFFFFF'),(22262,58289,'#FFFFFF'),(22263,58290,'#e859df'),(22264,58291,'#e73407'),(22265,58292,'#FFFFFF'),(22266,58294,'#1116f2'),(22267,58295,'#9727fc'),(22268,58296,'#71e8dd'),(22269,58297,'#FFFFFF'),(22270,58298,'#e70f32'),(22271,58299,'#FFFFFF'),(22272,58300,'#e3239b'),(22273,58301,'#FFFFFF'),(22274,58303,'#8df452'),(22275,58304,'#FFFFFF'),(22276,58305,'#FFFFFF'),(22277,58306,'#FFFFFF'),(22278,58307,'#67cbdc'),(22279,58308,'#5f63d4'),(22280,58311,'#FFFFFF'),(22281,58312,'#efc154'),(22282,58313,'#FFFFFF'),(22283,58314,'#FFFFFF'),(22284,58315,'#1843e2'),(22285,58317,'#fb3ec5'),(22286,58318,'#dd7e2a'),(22287,58319,'#FFFFFF'),(22288,58321,'#FFFFFF'),(22289,58322,'#579dce'),(22290,58323,'#FFFFFF'),(22291,58324,'#FFFFFF'),(22292,58325,'#FFFFFF'),(22293,58326,'#FFFFFF'),(22294,58327,'#FFFFFF'),(22295,58328,'#FFFFFF'),(22296,58329,'#FFFFFF'),(22297,58330,'#0b40f1'),(22298,58332,'#cc25da'),(22299,58333,'#d226a7'),(22300,58334,'#FFFFFF'),(22301,58335,'#e72bd7'),(22302,58337,'#FFFFFF'),(22303,58338,'#2d91dd'),(22304,58339,'#FFFFFF'),(22305,58340,'#8e5de2'),(22306,58341,'#cd2aa3'),(22307,58342,'#65dc1f'),(22308,58343,'#db5c79'),(22309,58344,'#2733ee'),(22310,58345,'#FFFFFF'),(22311,58346,'#FFFFFF'),(22312,58347,'#FFFFFF'),(22313,58349,'#FFFFFF'),(22314,58350,'#FFFFFF'),(22315,58351,'#FFFFFF'),(22316,58352,'#e73a7b'),(22317,58353,'#FFFFFF'),(22318,58354,'#FFFFFF'),(22319,58355,'#FFFFFF'),(22320,58356,'#93d435'),(22321,58357,'#cc2e09'),(22322,58358,'#f4330a'),(22323,58359,'#b360e3'),(22324,58360,'#FFFFFF'),(22325,58361,'#FFFFFF'),(22326,58362,'#d5a55b'),(22327,58363,'#FFFFFF'),(22328,58365,'#fecc1f'),(22329,58366,'#FFFFFF'),(22330,58367,'#FFFFFF'),(22331,58368,'#091fe9'),(22332,58369,'#FFFFFF'),(22333,58370,'#FFFFFF'),(22334,58371,'#b964de'),(22335,58372,'#FFFFFF'),(22336,58373,'#FFFFFF'),(22337,58374,'#FFFFFF'),(22338,58375,'#FFFFFF'),(22339,58376,'#FFFFFF'),(22340,58377,'#8135df'),(22341,58378,'#a1e968'),(22342,58379,'#FFFFFF'),(22343,58380,'#FFFFFF'),(22344,58381,'#05d715'),(22345,58382,'#FFFFFF'),(22346,58383,'#FFFFFF'),(22347,58384,'#FFFFFF'),(22348,58385,'#FFFFFF'),(22349,58386,'#48e498'),(22350,58387,'#FFFFFF'),(22351,58388,'#FFFFFF'),(22352,58389,'#FFFFFF'),(22353,58390,'#FFFFFF'),(22354,58391,'#40e9ea'),(22355,58392,'#FFFFFF'),(22356,58393,'#FFFFFF'),(22357,58394,'#39d778'),(22358,58395,'#FFFFFF'),(22359,58396,'#58b2d2'),(22360,58397,'#e25479'),(22361,58399,'#FFFFFF'),(22362,58400,'#7c33eb'),(22363,58401,'#FFFFFF'),(22364,58402,'#52db67'),(22365,58404,'#FFFFFF'),(22366,58405,'#FFFFFF'),(22367,58406,'#FFFFFF'),(22368,58407,'#FFFFFF'),(22369,58408,'#FFFFFF'),(22370,58409,'#FFFFFF'),(22371,58410,'#FFFFFF'),(22372,58411,'#FFFFFF'),(22373,58412,'#d07c1b'),(22374,58413,'#ce545d'),(22375,58415,'#FFFFFF'),(22376,58416,'#53ecc0'),(22377,58417,'#FFFFFF'),(22378,58418,'#06b4e2'),(22379,58419,'#70f27b'),(22380,58420,'#FFFFFF'),(22381,58421,'#FFFFFF'),(22382,58422,'#FFFFFF'),(22383,58423,'#4a4fd3'),(22384,58424,'#41f32e'),(22385,58425,'#FFFFFF'),(22386,58426,'#FFFFFF'),(22387,58427,'#FFFFFF'),(22388,58428,'#e64b66'),(22389,58429,'#eef90f'),(22390,58430,'#FFFFFF'),(22391,58431,'#FFFFFF'),(22392,58432,'#FFFFFF'),(22393,58433,'#FFFFFF'),(22394,58434,'#5eafd2'),(22395,58435,'#ea42c3'),(22396,58436,'#FFFFFF'),(22397,58437,'#FFFFFF'),(22398,58438,'#FFFFFF'),(22399,58439,'#f468f6'),(22400,58440,'#ce167d'),(22401,58441,'#df67dc'),(22402,58442,'#0fe609'),(22403,58443,'#5dea22'),(22404,58445,'#FFFFFF'),(22405,58446,'#fa5060'),(22406,58447,'#FFFFFF'),(22407,58448,'#FFFFFF'),(22408,58449,'#d334d5'),(22409,58450,'#FFFFFF'),(22410,58451,'#abd551'),(22411,58453,'#FFFFFF'),(22412,58454,'#69f2d5'),(22413,58455,'#4dd0e9'),(22414,58456,'#f818da'),(22415,58457,'#d7aa39'),(22416,58458,'#486afc'),(22417,58459,'#9ee318'),(22418,58460,'#1acdef'),(22419,58461,'#FFFFFF'),(22420,58462,'#FFFFFF'),(22421,58463,'#b8e920'),(22422,58464,'#a05eea'),(22423,58465,'#FFFFFF'),(22424,58466,'#60f014'),(22425,58467,'#FFFFFF'),(22426,58468,'#FFFFFF'),(22427,58469,'#FFFFFF'),(22428,58470,'#4ce8bf'),(22429,58471,'#5b6ddb'),(22430,58472,'#dad127'),(22431,58473,'#FFFFFF'),(22432,58474,'#2244fa'),(22433,58475,'#FFFFFF'),(22434,58476,'#FFFFFF'),(22435,58477,'#e46a0e'),(22436,58478,'#FFFFFF'),(22437,58479,'#b459df'),(22438,58480,'#FFFFFF'),(22439,58481,'#FFFFFF'),(22440,58482,'#FFFFFF'),(22441,58483,'#82e707'),(22442,58484,'#0e57e8'),(22443,58485,'#FFFFFF'),(22444,58486,'#FFFFFF'),(22445,58487,'#19ecc5'),(22446,58488,'#FFFFFF'),(22447,58489,'#ed09c8'),(22448,58490,'#FFFFFF'),(22449,58491,'#0ebbe5'),(22450,58492,'#FFFFFF'),(22451,58493,'#5ee1fa'),(22452,58494,'#7e6ff0'),(22453,58495,'#FFFFFF'),(22454,58496,'#FFFFFF'),(22455,58497,'#6ee204'),(22456,58498,'#dd21c2'),(22457,58499,'#de3a3c'),(22458,58500,'#FFFFFF'),(22459,58501,'#FFFFFF'),(22460,58502,'#FFFFFF'),(22461,58503,'#5330e6'),(22462,58504,'#d215a0'),(22463,58505,'#FFFFFF'),(22464,58506,'#FFFFFF'),(22465,58507,'#e9fd54'),(22466,58508,'#FFFFFF'),(22467,58509,'#FFFFFF'),(22468,58510,'#eb9a5a'),(22469,58511,'#e16884'),(22470,58512,'#FFFFFF'),(22471,58513,'#0c4ad6'),(22472,58514,'#50acd4'),(22473,58515,'#FFFFFF'),(22474,58516,'#FFFFFF'),(22475,58517,'#FFFFFF'),(22476,58518,'#FFFFFF'),(22477,58520,'#a0f830'),(22478,58521,'#2fd06d'),(22479,58522,'#FFFFFF'),(22480,58523,'#80ea19'),(22481,58524,'#FFFFFF'),(22482,58525,'#09bdea'),(22483,58526,'#FFFFFF'),(22484,58527,'#67f8ce'),(22485,58528,'#FFFFFF'),(22486,58529,'#FFFFFF'),(22487,58530,'#e46116'),(22488,58531,'#FFFFFF'),(22489,58532,'#327deb'),(22490,58533,'#FFFFFF'),(22491,58534,'#FFFFFF'),(22492,58535,'#FFFFFF'),(22493,58536,'#FFFFFF'),(22494,58537,'#FFFFFF'),(22495,58538,'#f6935c'),(22496,58539,'#FFFFFF'),(22497,58540,'#FFFFFF'),(22498,58541,'#FFFFFF'),(22499,58542,'#a94acc'),(22500,58543,'#FFFFFF'),(22501,58544,'#4549e6'),(22502,58545,'#FFFFFF'),(22503,58546,'#52e1e6'),(22504,58547,'#FFFFFF'),(22505,58548,'#545dd2'),(22506,58549,'#d05bca'),(22507,58550,'#FFFFFF'),(22508,58551,'#FFFFFF'),(22509,58552,'#e86a56'),(22510,58553,'#FFFFFF'),(22511,58556,'#FFFFFF'),(22512,58557,'#FFFFFF'),(22513,58558,'#FFFFFF'),(22514,58559,'#be09f1'),(22515,58560,'#9e5acd'),(22516,58561,'#05e1fc'),(22517,58563,'#3f83f2'),(22518,58564,'#FFFFFF'),(22519,58565,'#FFFFFF'),(22520,58566,'#4bf2c6'),(22521,58567,'#e4a672'),(22522,58568,'#cca842'),(22523,58569,'#0ee669'),(22524,58570,'#FFFFFF'),(22525,58571,'#FFFFFF'),(22526,58572,'#FFFFFF'),(22527,58573,'#FFFFFF'),(22528,58574,'#FFFFFF'),(22529,58575,'#FFFFFF'),(22530,58576,'#FFFFFF'),(22531,58577,'#ad36ee'),(22532,58578,'#52dddf'),(22533,58579,'#4fbecc'),(22534,58580,'#FFFFFF'),(22535,58581,'#6fee50'),(22536,58582,'#2fd307'),(22537,58583,'#363ee7'),(22538,58584,'#fc023d'),(22539,58585,'#FFFFFF'),(22540,58586,'#5fd203'),(22541,58587,'#FFFFFF'),(22542,58588,'#FFFFFF'),(22543,58589,'#FFFFFF'),(22544,58590,'#4be089'),(22545,58591,'#FFFFFF'),(22546,58592,'#FFFFFF'),(22547,58593,'#FFFFFF'),(22548,58594,'#e93157'),(22549,58595,'#FFFFFF'),(22550,58596,'#FFFFFF'),(22551,58597,'#FFFFFF'),(22552,58598,'#FFFFFF'),(22553,58599,'#2435d8'),(22554,58600,'#8960fb'),(22555,58601,'#418ae0'),(22556,58602,'#FFFFFF'),(22557,58603,'#FFFFFF'),(22558,58604,'#FFFFFF'),(22559,58605,'#9d70f5'),(22560,58606,'#FFFFFF'),(22561,58607,'#58d4c7'),(22562,58609,'#FFFFFF'),(22563,58610,'#FFFFFF'),(22564,58611,'#FFFFFF'),(22565,58613,'#FFFFFF'),(22566,58614,'#FFFFFF'),(22567,58615,'#ae15ef'),(22568,58616,'#FFFFFF'),(22569,58617,'#FFFFFF'),(22570,58618,'#FFFFFF'),(22571,58619,'#4286ed'),(22572,58620,'#23eb17'),(22573,58621,'#daa360'),(22574,58622,'#e04d3b'),(22575,58623,'#FFFFFF'),(22576,58624,'#FFFFFF'),(22577,58625,'#a1f865'),(22578,58626,'#85dd1e'),(22579,58627,'#FFFFFF'),(22580,58628,'#FFFFFF'),(22581,58629,'#07e17d'),(22582,58630,'#d325ab'),(22583,58631,'#c8f245'),(22584,58632,'#66dace'),(22585,58633,'#FFFFFF'),(22586,58634,'#3ecfd2'),(22587,58635,'#FFFFFF'),(22588,58636,'#fc498c'),(22589,58637,'#f54f97'),(22590,58638,'#FFFFFF'),(22591,58639,'#32f79d'),(22592,58640,'#FFFFFF'),(22593,58642,'#21e1e3'),(22594,58643,'#FFFFFF'),(22595,58644,'#6084e0'),(22596,58645,'#FFFFFF'),(22597,58646,'#f3a473'),(22598,58647,'#FFFFFF'),(22599,58648,'#ee3b14'),(22600,58649,'#3504dd'),(22601,58650,'#67f5fc'),(22602,58651,'#ce4d63'),(22603,58652,'#3f45da'),(22604,58653,'#FFFFFF'),(22605,58654,'#dbc654'),(22606,58655,'#FFFFFF'),(22607,58656,'#8b29dd'),(22608,58657,'#cf9344'),(22609,58658,'#FFFFFF'),(22610,58659,'#58fc49'),(22611,58661,'#1ef469'),(22612,58662,'#FFFFFF'),(22613,58663,'#FFFFFF'),(22614,58664,'#FFFFFF'),(22615,58665,'#FFFFFF'),(22616,58666,'#e56d52'),(22617,58667,'#FFFFFF'),(22618,58668,'#FFFFFF'),(22619,58669,'#FFFFFF'),(22620,58670,'#29cf08'),(22621,58671,'#FFFFFF'),(22622,58672,'#FFFFFF'),(22623,58673,'#FFFFFF'),(22624,58674,'#FFFFFF'),(22625,58675,'#3e41dc'),(22626,58676,'#15dbec'),(22627,58678,'#FFFFFF'),(22628,58679,'#fe60c9'),(22629,58680,'#40f981'),(22630,58681,'#8f13e4'),(22631,58682,'#bb60e0'),(22632,58683,'#FFFFFF'),(22633,58684,'#1a9fcd'),(22634,58685,'#f93a41'),(22635,58686,'#FFFFFF'),(22636,58687,'#1ef891'),(22637,58688,'#FFFFFF'),(22638,58689,'#FFFFFF'),(22639,58690,'#b4f178'),(22640,58691,'#FFFFFF'),(22641,58693,'#FFFFFF'),(22642,58694,'#FFFFFF'),(22643,58695,'#fef93c'),(22644,58696,'#FFFFFF'),(22645,58697,'#3420cd'),(22646,58698,'#FFFFFF'),(22647,58699,'#ec5dde'),(22648,58700,'#FFFFFF'),(22649,58701,'#2a99f1'),(22650,58702,'#5bf078'),(22651,58703,'#b7ed28'),(22652,58704,'#FFFFFF'),(22653,58705,'#FFFFFF'),(22654,58706,'#FFFFFF'),(22655,58707,'#86d059'),(22656,58708,'#FFFFFF'),(22657,58709,'#fd7754'),(22658,58710,'#FFFFFF'),(22659,58711,'#ad29d0'),(22660,58712,'#d2667c'),(22661,58713,'#e02294'),(22662,58714,'#FFFFFF'),(22663,58716,'#FFFFFF'),(22664,58717,'#19d679'),(22665,58718,'#FFFFFF'),(22666,58719,'#FFFFFF'),(22667,58720,'#FFFFFF'),(22668,58721,'#FFFFFF'),(22669,58722,'#f4a227'),(22670,58723,'#FFFFFF'),(22671,58724,'#FFFFFF'),(22672,58725,'#FFFFFF'),(22673,58726,'#FFFFFF'),(22674,58727,'#7adf6c'),(22675,58728,'#9038eb'),(22676,58729,'#FFFFFF'),(22677,58730,'#FFFFFF'),(22678,58731,'#e5be55'),(22679,58733,'#FFFFFF'),(22680,58734,'#FFFFFF'),(22681,58735,'#FFFFFF'),(22682,58736,'#FFFFFF'),(22683,58737,'#5df1c4'),(22684,58738,'#FFFFFF'),(22685,58739,'#e9a147'),(22686,58740,'#FFFFFF'),(22687,58741,'#FFFFFF'),(22688,58742,'#FFFFFF'),(22689,58743,'#FFFFFF'),(22690,58744,'#FFFFFF'),(22691,58745,'#4034e4'),(22692,58746,'#FFFFFF'),(22693,58747,'#FFFFFF'),(22694,58748,'#c3fa45'),(22695,58750,'#62f536'),(22696,58751,'#FFFFFF'),(22697,58752,'#FFFFFF'),(22698,58754,'#FFFFFF'),(22699,58755,'#FFFFFF'),(22700,58756,'#FFFFFF'),(22701,58757,'#FFFFFF'),(22702,58758,'#FFFFFF'),(22703,58759,'#FFFFFF'),(22704,58760,'#a056f0'),(22705,58761,'#FFFFFF'),(22706,58762,'#FFFFFF'),(22707,58763,'#FFFFFF'),(22708,58764,'#FFFFFF'),(22709,58765,'#FFFFFF'),(22710,58766,'#FFFFFF'),(22711,58767,'#FFFFFF'),(22712,58768,'#d2100b'),(22713,58769,'#4430d4'),(22714,58770,'#e838f6'),(22715,58771,'#FFFFFF'),(22716,58772,'#FFFFFF'),(22717,58773,'#FFFFFF'),(22718,58774,'#FFFFFF'),(22719,58775,'#13d26f'),(22720,58776,'#FFFFFF'),(22721,58777,'#21cf0e'),(22722,58779,'#FFFFFF'),(22723,58780,'#d00b45'),(22724,58781,'#3edb40'),(22725,58782,'#FFFFFF'),(22726,58783,'#3432f9'),(22727,58784,'#69e4c9'),(22728,58785,'#a9fc3c'),(22729,58786,'#4bf41e'),(22730,58787,'#FFFFFF'),(22731,58788,'#34cf89'),(22732,58789,'#FFFFFF'),(22733,58790,'#FFFFFF'),(22734,58791,'#FFFFFF'),(22735,58793,'#FFFFFF'),(22736,58794,'#FFFFFF'),(22737,58795,'#FFFFFF'),(22738,58796,'#FFFFFF'),(22739,58797,'#FFFFFF'),(22740,58798,'#FFFFFF'),(22741,58800,'#FFFFFF'),(22742,58801,'#1bcf27'),(22743,58802,'#8e6ae1'),(22744,58803,'#6addef'),(22745,58804,'#FFFFFF'),(22746,58805,'#FFFFFF'),(22747,58806,'#FFFFFF'),(22748,58807,'#e513f2'),(22749,58808,'#f259dd'),(22750,58809,'#FFFFFF'),(22751,58810,'#FFFFFF'),(22752,58811,'#FFFFFF'),(22753,58812,'#d101f5'),(22754,58813,'#FFFFFF'),(22755,58814,'#d35b20'),(22756,58815,'#FFFFFF'),(22757,58816,'#2bcd02'),(22758,58817,'#FFFFFF'),(22759,58818,'#FFFFFF'),(22760,58819,'#5e57cd'),(22761,58820,'#FFFFFF'),(22762,58821,'#FFFFFF'),(22763,58822,'#ec3ef7'),(22764,58823,'#FFFFFF'),(22765,58824,'#FFFFFF'),(22766,58825,'#6460cf'),(22767,58826,'#FFFFFF'),(22768,58827,'#FFFFFF'),(22769,58828,'#FFFFFF'),(22770,58829,'#FFFFFF'),(22771,58831,'#FFFFFF'),(22772,58832,'#FFFFFF'),(22773,58833,'#FFFFFF'),(22774,58834,'#FFFFFF'),(22775,58835,'#1ecc40'),(22776,58836,'#FFFFFF'),(22777,58837,'#FFFFFF'),(22778,58838,'#FFFFFF'),(22779,58839,'#f75539'),(22780,58840,'#fdbe0a'),(22781,58841,'#FFFFFF'),(22782,58842,'#FFFFFF'),(22783,58843,'#f01ffc'),(22784,58844,'#FFFFFF'),(22785,58845,'#FFFFFF'),(22786,58846,'#97d617'),(22787,58847,'#2725cf'),(22788,58848,'#dad45c'),(22789,58849,'#FFFFFF'),(22790,58850,'#FFFFFF'),(22791,58851,'#FFFFFF'),(22792,58852,'#d7842a'),(22793,58853,'#2ee2c0'),(22794,58854,'#d4020b'),(22795,58855,'#FFFFFF'),(22796,58856,'#FFFFFF'),(22797,58857,'#FFFFFF'),(22798,58858,'#FFFFFF'),(22799,58859,'#FFFFFF'),(22800,58860,'#FFFFFF'),(22801,58861,'#FFFFFF'),(22802,58863,'#FFFFFF'),(22803,58864,'#FFFFFF'),(22804,58865,'#e0ca3f'),(22805,58866,'#FFFFFF'),(22806,58868,'#FFFFFF'),(22807,58869,'#c66ae9'),(22808,58870,'#ece76a'),(22809,58871,'#FFFFFF'),(22810,58872,'#364de4'),(22811,58873,'#d24afd'),(22812,58874,'#db3d43'),(22813,58876,'#e69464'),(22814,58878,'#f98b46'),(22815,58879,'#14e109'),(22816,58880,'#d84d6b'),(22817,58881,'#6daedb'),(22818,58882,'#FFFFFF'),(22819,58884,'#b4cc3c'),(22820,58885,'#ec2201'),(22821,58886,'#FFFFFF'),(22822,58887,'#FFFFFF'),(22823,58888,'#cfd84d'),(22824,58889,'#f0d376'),(22825,58890,'#c55bd6'),(22826,58891,'#FFFFFF'),(22827,58892,'#FFFFFF'),(22828,58893,'#FFFFFF'),(22829,58894,'#FFFFFF'),(22830,58895,'#FFFFFF'),(22831,58896,'#646dcf'),(22832,58897,'#59e2ee'),(22833,58898,'#1b2ddd'),(22834,58899,'#FFFFFF'),(22835,58900,'#74cc5a'),(22836,58902,'#d006d0'),(22837,58903,'#FFFFFF'),(22838,58904,'#FFFFFF'),(22839,58905,'#ec4b89'),(22840,58906,'#FFFFFF'),(22841,58908,'#ec637c'),(22842,58909,'#606bcd'),(22843,58910,'#FFFFFF'),(22844,58911,'#FFFFFF'),(22845,58912,'#FFFFFF'),(22846,58913,'#17d2d4'),(22847,58914,'#e6dd69'),(22848,58915,'#FFFFFF'),(22849,58916,'#FFFFFF'),(22850,58917,'#FFFFFF'),(22851,58918,'#2799dc'),(22852,58921,'#FFFFFF'),(22853,58922,'#FFFFFF'),(22854,58923,'#f3393b'),(22855,58924,'#69b4e8'),(22856,58925,'#FFFFFF'),(22857,58926,'#FFFFFF'),(22858,58928,'#FFFFFF'),(22859,58929,'#1a4eda'),(22860,58930,'#FFFFFF'),(22861,58931,'#FFFFFF'),(22862,58932,'#d7c739'),(22863,58933,'#df8a27'),(22864,58934,'#f565c1'),(22865,58935,'#FFFFFF'),(22866,58936,'#4d8dd9'),(22867,58937,'#afd24e'),(22868,58938,'#bee511'),(22869,58939,'#295de8'),(22870,58940,'#FFFFFF'),(22871,58941,'#FFFFFF'),(22872,58942,'#fdcf39'),(22873,58943,'#abf908'),(22874,58944,'#FFFFFF'),(22875,58945,'#FFFFFF'),(22876,58946,'#FFFFFF'),(22877,58947,'#d15c6e'),(22878,58948,'#FFFFFF'),(22879,58949,'#FFFFFF'),(22880,58950,'#FFFFFF'),(22881,58951,'#FFFFFF'),(22882,58952,'#FFFFFF'),(22883,58953,'#d6b663'),(22884,58954,'#db395e'),(22885,58956,'#f2bf3a'),(22886,58957,'#FFFFFF'),(22887,58958,'#f626c2'),(22888,58959,'#FFFFFF'),(22889,58960,'#FFFFFF'),(22890,58961,'#7ef026'),(22891,58962,'#FFFFFF'),(22892,58963,'#FFFFFF'),(22893,58964,'#FFFFFF'),(22894,58965,'#FFFFFF'),(22895,58966,'#FFFFFF'),(22896,58967,'#FFFFFF'),(22897,58968,'#50fb72'),(22898,58970,'#3ccd34'),(22899,58971,'#dc3fa5'),(22900,58972,'#8a05cf'),(22901,58973,'#FFFFFF'),(22902,58974,'#FFFFFF'),(22903,58975,'#ef6aa4'),(22904,58976,'#FFFFFF'),(22905,58977,'#FFFFFF'),(22906,58978,'#FFFFFF'),(22907,58979,'#43d8f2'),(22908,58980,'#FFFFFF'),(22909,58981,'#4cdd59'),(22910,58982,'#FFFFFF'),(22911,58983,'#FFFFFF'),(22912,58984,'#d6e145'),(22913,58985,'#6175e7'),(22914,58986,'#FFFFFF'),(22915,58987,'#5979e9'),(22916,58988,'#70eba1'),(22917,58989,'#FFFFFF'),(22918,58990,'#FFFFFF'),(22919,58991,'#72d03f'),(22920,58992,'#FFFFFF'),(22921,58993,'#FFFFFF'),(22922,58994,'#FFFFFF'),(22923,58995,'#1198f5'),(22924,58996,'#FFFFFF'),(22925,58997,'#FFFFFF'),(22926,58998,'#67de54'),(22927,58999,'#d2e163'),(22928,59000,'#FFFFFF'),(22929,59001,'#FFFFFF'),(22930,59002,'#FFFFFF'),(22931,59003,'#FFFFFF'),(22932,59004,'#c83fd3'),(22933,59006,'#cd2a25'),(22934,59007,'#f62276'),(22935,59008,'#FFFFFF'),(22936,59009,'#FFFFFF'),(22937,59010,'#5e53f5'),(22938,59011,'#8354e9'),(22939,59012,'#3bcd68'),(22940,59013,'#FFFFFF'),(22941,59014,'#FFFFFF'),(22942,59015,'#02f3e5'),(22943,59016,'#FFFFFF'),(22944,59017,'#e642cb'),(22945,59018,'#fb5b57'),(22946,59019,'#FFFFFF'),(22947,59020,'#FFFFFF'),(22948,59021,'#d4900c'),(22949,59022,'#fd9309'),(22950,59023,'#afcc4f'),(22951,59024,'#FFFFFF'),(22952,59025,'#FFFFFF'),(22953,59026,'#6d15e2'),(22954,59027,'#FFFFFF'),(22955,59028,'#10d6e7'),(22956,59029,'#FFFFFF'),(22957,59030,'#4845d9'),(22958,59031,'#FFFFFF'),(22959,59032,'#FFFFFF'),(22960,59033,'#FFFFFF'),(22961,59034,'#FFFFFF'),(22962,59035,'#FFFFFF'),(22963,59036,'#FFFFFF'),(22964,59037,'#FFFFFF'),(22965,59038,'#29cfab'),(22966,59039,'#dba056'),(22967,59040,'#FFFFFF'),(22968,59041,'#FFFFFF'),(22969,59042,'#5461e8'),(22970,59043,'#f04923'),(22971,59044,'#019ee3'),(22972,59046,'#FFFFFF'),(22973,59047,'#FFFFFF'),(22974,59048,'#FFFFFF'),(22975,59049,'#80d229'),(22976,59050,'#4720d6'),(22977,59051,'#e0d140'),(22978,59052,'#0ee93d'),(22979,59053,'#FFFFFF'),(22980,59054,'#FFFFFF'),(22981,59055,'#FFFFFF'),(22982,59056,'#72d042'),(22983,59057,'#d52999'),(22984,59058,'#2cf9da'),(22985,59059,'#4c72ec'),(22986,59060,'#FFFFFF'),(22987,59061,'#FFFFFF'),(22988,59063,'#05ec97'),(22989,59064,'#FFFFFF'),(22990,59065,'#1a78fd'),(22991,59066,'#FFFFFF'),(22992,59067,'#FFFFFF'),(22993,59068,'#FFFFFF'),(22994,59069,'#eee504'),(22995,59071,'#FFFFFF'),(22996,59072,'#6b4de9'),(22997,59073,'#FFFFFF'),(22998,59074,'#8848fa'),(22999,59075,'#7e11e0'),(23000,59076,'#FFFFFF'),(23001,59077,'#e951d3'),(23002,59078,'#ed752f'),(23003,59079,'#ed72c0'),(23004,59080,'#cc4745'),(23005,59081,'#00ccd7'),(23006,59082,'#FFFFFF'),(23007,59083,'#FFFFFF'),(23008,59084,'#FFFFFF'),(23009,59085,'#FFFFFF'),(23010,59086,'#FFFFFF'),(23011,59087,'#5ef658'),(23012,59088,'#FFFFFF'),(23013,59089,'#2afb40'),(23014,59090,'#d963f7'),(23015,59091,'#FFFFFF'),(23016,59092,'#a6dc4c'),(23017,59093,'#FFFFFF'),(23018,59094,'#FFFFFF'),(23019,59095,'#db36f4'),(23020,59096,'#FFFFFF'),(23021,59097,'#b272e6'),(23022,59098,'#6ddedf'),(23023,59100,'#FFFFFF'),(23024,59101,'#FFFFFF'),(23025,59102,'#FFFFFF'),(23026,59103,'#FFFFFF'),(23027,59104,'#FFFFFF'),(23028,59105,'#FFFFFF'),(23029,59106,'#FFFFFF'),(23030,59107,'#FFFFFF'),(23031,59108,'#64fa78'),(23032,59109,'#FFFFFF'),(23033,59110,'#FFFFFF'),(23034,59111,'#FFFFFF'),(23035,59112,'#FFFFFF'),(23036,59113,'#FFFFFF'),(23037,59114,'#FFFFFF'),(23038,59115,'#FFFFFF'),(23039,59116,'#FFFFFF'),(23040,59117,'#FFFFFF'),(23041,59118,'#FFFFFF'),(23042,59119,'#FFFFFF'),(23043,59120,'#2fd7de'),(23044,59121,'#e6d22c'),(23045,59122,'#FFFFFF'),(23046,59123,'#6ddd99'),(23047,59124,'#0bfa4a'),(23048,59125,'#FFFFFF'),(23049,59126,'#FFFFFF'),(23050,59127,'#44c0d0'),(23051,59128,'#FFFFFF'),(23052,59129,'#7ff85e'),(23053,59130,'#e5f42f'),(23054,59131,'#FFFFFF'),(23055,59132,'#FFFFFF'),(23056,59133,'#FFFFFF'),(23057,59134,'#d62e39'),(23058,59135,'#FFFFFF'),(23059,59136,'#dfab11'),(23060,59137,'#FFFFFF'),(23061,59138,'#f07381'),(23062,59139,'#FFFFFF'),(23063,59140,'#FFFFFF'),(23064,59141,'#FFFFFF'),(23065,59142,'#FFFFFF'),(23066,59143,'#eb2e45'),(23067,59144,'#fe6ca5'),(23068,59145,'#FFFFFF'),(23069,59146,'#55f071'),(23070,59147,'#fb6455'),(23071,59148,'#FFFFFF'),(23072,59149,'#FFFFFF'),(23073,59150,'#FFFFFF'),(23074,59151,'#FFFFFF'),(23075,59152,'#FFFFFF'),(23076,59153,'#FFFFFF'),(23077,59154,'#d866ed'),(23078,59155,'#FFFFFF'),(23079,59156,'#FFFFFF'),(23080,59157,'#FFFFFF'),(23081,59158,'#6789cf'),(23082,59159,'#f5b132'),(23083,59160,'#254aea'),(23084,59161,'#eb40b1'),(23085,59162,'#36f30e'),(23086,59163,'#e731bf'),(23087,59164,'#FFFFFF'),(23088,59165,'#FFFFFF'),(23089,59166,'#FFFFFF'),(23090,59167,'#FFFFFF'),(23091,59168,'#3757e8'),(23092,59169,'#136ff7'),(23093,59170,'#FFFFFF'),(23094,59171,'#FFFFFF'),(23095,59172,'#FFFFFF'),(23096,59173,'#e75546'),(23097,59174,'#37faaa'),(23098,59175,'#49df7b'),(23099,59176,'#FFFFFF'),(23100,59177,'#FFFFFF'),(23101,59178,'#FFFFFF'),(23102,59179,'#FFFFFF'),(23103,59180,'#FFFFFF'),(23104,59181,'#FFFFFF'),(23105,59182,'#0edff5'),(23106,59183,'#0284e7'),(23107,59184,'#FFFFFF'),(23108,59185,'#3593d1'),(23109,59186,'#fa3f75'),(23110,59187,'#FFFFFF'),(23111,59188,'#2da2d7'),(23112,59189,'#ec8e4b'),(23113,59190,'#FFFFFF'),(23114,59191,'#6269d0'),(23115,59192,'#98ee60'),(23116,59193,'#FFFFFF'),(23117,59194,'#FFFFFF'),(23118,59195,'#FFFFFF'),(23119,59196,'#FFFFFF'),(23120,59197,'#FFFFFF'),(23121,59198,'#FFFFFF'),(23122,59199,'#FFFFFF'),(23123,59200,'#FFFFFF'),(23124,59201,'#FFFFFF'),(23125,59202,'#FFFFFF'),(23126,59203,'#FFFFFF'),(23127,59204,'#FFFFFF'),(23128,59205,'#69bdd4'),(23129,59206,'#FFFFFF'),(23130,59207,'#FFFFFF'),(23131,59208,'#FFFFFF'),(23132,59209,'#FFFFFF'),(23133,59210,'#FFFFFF'),(23134,59211,'#FFFFFF'),(23135,59212,'#FFFFFF'),(23136,59213,'#eb59e2'),(23137,59214,'#66dd91'),(23138,59215,'#FFFFFF'),(23139,59216,'#FFFFFF'),(23140,59217,'#06e792'),(23141,59218,'#FFFFFF'),(23142,59219,'#f237ba'),(23143,59220,'#FFFFFF'),(23144,59221,'#48cde2'),(23145,59222,'#3fce35'),(23146,59223,'#fe3693'),(23147,59224,'#df0840'),(23148,59225,'#FFFFFF'),(23149,59226,'#FFFFFF'),(23150,59227,'#FFFFFF'),(23151,59228,'#FFFFFF'),(23152,59229,'#49d2a7'),(23153,59230,'#FFFFFF'),(23154,59231,'#FFFFFF'),(23155,59232,'#a93be3'),(23156,59233,'#FFFFFF'),(23157,59234,'#eb3247'),(23158,59235,'#FFFFFF'),(23159,59236,'#FFFFFF'),(23160,59237,'#46d724'),(23161,59238,'#FFFFFF'),(23162,59239,'#eb3c8e'),(23163,59240,'#e404fc'),(23164,59241,'#04ec75'),(23165,59242,'#FFFFFF'),(23166,59243,'#FFFFFF'),(23167,59244,'#FFFFFF'),(23168,59245,'#9bdc3a'),(23169,59246,'#d11b96'),(23170,59247,'#FFFFFF'),(23171,59248,'#FFFFFF'),(23172,59249,'#FFFFFF'),(23173,59250,'#FFFFFF'),(23174,59251,'#3359f6'),(23175,59252,'#cd4d5b'),(23176,59253,'#FFFFFF'),(23177,59254,'#FFFFFF'),(23178,59255,'#FFFFFF'),(23179,59256,'#fe2131'),(23180,59257,'#FFFFFF'),(23181,59258,'#FFFFFF'),(23182,59259,'#5ff8cc'),(23183,59260,'#d5e828'),(23184,59261,'#FFFFFF'),(23185,59262,'#FFFFFF'),(23186,59263,'#FFFFFF'),(23187,59264,'#e87849'),(23188,59265,'#FFFFFF'),(23189,59266,'#FFFFFF'),(23190,59267,'#3b40ed'),(23191,59268,'#FFFFFF'),(23192,59269,'#FFFFFF'),(23193,59270,'#FFFFFF'),(23194,59271,'#FFFFFF'),(23195,59272,'#FFFFFF'),(23196,59273,'#FFFFFF'),(23197,59274,'#e9ec42'),(23198,59275,'#4feb81'),(23199,59276,'#FFFFFF'),(23200,59277,'#FFFFFF'),(23201,59278,'#94d537'),(23202,59279,'#FFFFFF'),(23203,59280,'#FFFFFF'),(23204,59281,'#FFFFFF'),(23205,59282,'#4984de'),(23206,59283,'#FFFFFF'),(23207,59284,'#FFFFFF'),(23208,59285,'#FFFFFF'),(23209,59286,'#FFFFFF'),(23210,59287,'#FFFFFF'),(23211,59288,'#FFFFFF'),(23212,59289,'#FFFFFF'),(23213,59290,'#54dc3c'),(23214,59291,'#ef2c3e'),(23215,59292,'#FFFFFF'),(23216,59293,'#FFFFFF'),(23217,59294,'#ab0cdc'),(23218,59295,'#FFFFFF'),(23219,59296,'#FFFFFF'),(23220,59297,'#FFFFFF'),(23221,59298,'#50d005'),(23222,59299,'#de1944'),(23223,59300,'#FFFFFF'),(23224,59301,'#FFFFFF'),(23225,59302,'#FFFFFF'),(23226,59303,'#FFFFFF'),(23227,59304,'#FFFFFF'),(23228,59305,'#FFFFFF'),(23229,59306,'#FFFFFF'),(23230,59307,'#FFFFFF'),(23231,59308,'#FFFFFF'),(23232,59309,'#FFFFFF'),(23233,59310,'#58d89b'),(23234,59311,'#d94c6d'),(23235,59312,'#FFFFFF'),(23236,59313,'#FFFFFF'),(23237,59314,'#FFFFFF'),(23238,59315,'#FFFFFF'),(23239,59316,'#130cd0'),(23240,59317,'#FFFFFF'),(23241,59318,'#69fcde'),(23242,59319,'#49cc4c'),(23243,59320,'#720de8'),(23244,59321,'#FFFFFF'),(23245,59322,'#88f10e'),(23246,59323,'#d7da3d'),(23247,59324,'#FFFFFF'),(23248,59325,'#52b0f2'),(23249,59326,'#FFFFFF'),(23250,59327,'#FFFFFF'),(23251,59328,'#FFFFFF'),(23252,59329,'#a158e5'),(23253,59330,'#FFFFFF'),(23254,59331,'#FFFFFF'),(23255,59332,'#FFFFFF'),(23256,59333,'#38a8cf'),(23257,59334,'#FFFFFF'),(23258,59335,'#FFFFFF'),(23259,59336,'#6df549'),(23260,59337,'#FFFFFF'),(23261,59338,'#0cc2eb'),(23262,59339,'#0c02e4'),(23263,59340,'#FFFFFF'),(23264,59341,'#51fc49'),(23265,59342,'#1dec09'),(23266,59343,'#f74452'),(23267,59344,'#3076f0'),(23268,59345,'#FFFFFF'),(23269,59346,'#FFFFFF'),(23270,59347,'#FFFFFF'),(23271,59348,'#FFFFFF'),(23272,59349,'#c4d109'),(23273,59350,'#FFFFFF'),(23274,59351,'#FFFFFF'),(23275,59352,'#FFFFFF'),(23276,59353,'#FFFFFF'),(23277,59354,'#5bf392'),(23278,59355,'#FFFFFF'),(23279,59356,'#FFFFFF'),(23280,59357,'#FFFFFF'),(23281,59358,'#865bd9'),(23282,59359,'#FFFFFF'),(23283,59360,'#FFFFFF'),(23284,59361,'#FFFFFF'),(23285,59362,'#FFFFFF'),(23286,59363,'#FFFFFF'),(23287,59364,'#FFFFFF'),(23288,59365,'#FFFFFF'),(23289,59366,'#FFFFFF'),(23290,59367,'#FFFFFF'),(23291,59368,'#FFFFFF'),(23292,59369,'#FFFFFF'),(23293,59370,'#19f1d4'),(23294,59371,'#9114d1'),(23295,59372,'#7dd02d'),(23296,59373,'#FFFFFF'),(23297,59374,'#FFFFFF'),(23298,59375,'#8f2ce8'),(23299,59376,'#FFFFFF'),(23300,59377,'#0ee4c8'),(23301,59378,'#FFFFFF'),(23302,59379,'#FFFFFF'),(23303,59380,'#FFFFFF'),(23304,59381,'#4dda7f'),(23305,59382,'#4208ef'),(23306,59383,'#FFFFFF'),(23307,59384,'#FFFFFF'),(23308,59385,'#41efc3'),(23309,59386,'#8dcc5d'),(23310,59387,'#3a76e7'),(23311,59388,'#66eed2'),(23312,59389,'#FFFFFF'),(23313,59390,'#d3341b'),(23314,59391,'#FFFFFF'),(23315,59392,'#FFFFFF'),(23316,59393,'#FFFFFF'),(23317,59394,'#FFFFFF'),(23318,59395,'#FFFFFF'),(23319,59396,'#FFFFFF'),(23320,59397,'#02d8cd'),(23321,59398,'#FFFFFF'),(23322,59399,'#FFFFFF'),(23323,59400,'#874fe6'),(23324,59401,'#FFFFFF'),(23325,59402,'#bf1fd5'),(23326,59403,'#FFFFFF'),(23327,59404,'#c7fa01'),(23328,59405,'#f5fe66'),(23329,59406,'#eb9738'),(23330,59407,'#FFFFFF'),(23331,59408,'#FFFFFF'),(23332,59409,'#be3cf5'),(23333,59410,'#5ffd19'),(23334,59411,'#FFFFFF'),(23335,59412,'#FFFFFF'),(23336,59413,'#ec8045'),(23337,59414,'#FFFFFF'),(23338,59415,'#a4da46'),(23339,59416,'#8be726'),(23340,59417,'#73f39f'),(23341,59418,'#FFFFFF'),(23342,59419,'#FFFFFF'),(23343,59420,'#FFFFFF'),(23344,59421,'#FFFFFF'),(23345,59422,'#FFFFFF'),(23346,59423,'#FFFFFF'),(23347,59424,'#9e66f4'),(23348,59425,'#98d44f'),(23349,59426,'#FFFFFF'),(23350,59427,'#FFFFFF'),(23351,59428,'#FFFFFF'),(23352,59429,'#FFFFFF'),(23353,59430,'#FFFFFF'),(23354,59431,'#ee5696'),(23355,59432,'#FFFFFF'),(23356,59433,'#FFFFFF'),(23357,59434,'#99e62d'),(23358,59435,'#FFFFFF'),(23359,59436,'#FFFFFF'),(23360,59437,'#FFFFFF'),(23361,59438,'#FFFFFF'),(23362,59439,'#eb756b'),(23363,59440,'#64f9a8'),(23364,59441,'#FFFFFF'),(23365,59442,'#FFFFFF'),(23366,59443,'#FFFFFF'),(23367,59444,'#FFFFFF'),(23368,59445,'#FFFFFF'),(23369,59446,'#FFFFFF'),(23370,59447,'#db1ff4'),(23371,59448,'#FFFFFF'),(23372,59449,'#83f824'),(23373,59450,'#FFFFFF'),(23374,59451,'#7352d6'),(23375,59452,'#b2d03c'),(23376,59453,'#FFFFFF'),(23377,59454,'#FFFFFF'),(23378,59455,'#FFFFFF'),(23379,59456,'#FFFFFF'),(23380,59457,'#FFFFFF'),(23381,59458,'#FFFFFF'),(23382,59459,'#d77414'),(23383,59460,'#60d74b'),(23384,59461,'#FFFFFF'),(23385,59462,'#1277e9'),(23386,59463,'#7849f2'),(23387,59464,'#a902f1'),(23388,59465,'#5edb8b'),(23389,59466,'#FFFFFF'),(23390,59467,'#4843e4'),(23391,59468,'#FFFFFF'),(23392,59469,'#e40557'),(23393,59470,'#FFFFFF'),(23394,59471,'#25a1db'),(23395,59472,'#FFFFFF'),(23396,59473,'#fb1ecb'),(23397,59474,'#e51481'),(23398,59475,'#FFFFFF'),(23399,59476,'#FFFFFF'),(23400,59477,'#FFFFFF'),(23401,59478,'#e818a8'),(23402,59479,'#6cdaf3'),(23403,59480,'#FFFFFF'),(23404,59481,'#e1634e'),(23405,59482,'#e4ae3e'),(23406,59483,'#edfd3a');
/*!40000 ALTER TABLE `pham_color` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `pham_history`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `pham_history` (
  `name` int(10) unsigned NOT NULL,
  `parent` int(10) unsigned NOT NULL,
  `action` enum('join','split') NOT NULL,
  `datetime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (`name`,`parent`,`action`),
  KEY `parent_index` (`parent`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `pham_history` WRITE;
/*!40000 ALTER TABLE `pham_history` DISABLE KEYS */;
/*!40000 ALTER TABLE `pham_history` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `pham_old`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `pham_old` (
  `GeneID` varchar(35) DEFAULT NULL,
  `name` int(10) unsigned DEFAULT NULL,
  `orderAdded` int(5) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`orderAdded`),
  KEY `orderAdded_index` (`orderAdded`),
  KEY `GeneID_index` (`GeneID`),
  CONSTRAINT `pham_old_ibfk_1` FOREIGN KEY (`GeneID`) REFERENCES `gene` (`GeneID`) ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `pham_old` WRITE;
/*!40000 ALTER TABLE `pham_old` DISABLE KEYS */;
/*!40000 ALTER TABLE `pham_old` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `scores_summary`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `scores_summary` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `query` varchar(35) DEFAULT NULL,
  `subject` varchar(35) DEFAULT NULL,
  `blast_score` double unsigned DEFAULT NULL,
  `clustalw_score` decimal(5,4) unsigned DEFAULT NULL,
  `blast_bit_score` double unsigned DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `scores_summary_ibfk_1` (`query`),
  KEY `scores_summary_ibfk_2` (`subject`),
  CONSTRAINT `scores_summary_ibfk_1` FOREIGN KEY (`query`) REFERENCES `gene` (`GeneID`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `scores_summary_ibfk_2` FOREIGN KEY (`subject`) REFERENCES `gene` (`GeneID`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `scores_summary` WRITE;
/*!40000 ALTER TABLE `scores_summary` DISABLE KEYS */;
/*!40000 ALTER TABLE `scores_summary` ENABLE KEYS */;
UNLOCK TABLES;
DROP TABLE IF EXISTS `version`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `version` (
  `version` int(11) unsigned NOT NULL,
  `schema_version` int(11) unsigned NOT NULL,
  PRIMARY KEY (`version`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
/*!40101 SET character_set_client = @saved_cs_client */;

LOCK TABLES `version` WRITE;
/*!40000 ALTER TABLE `version` DISABLE KEYS */;
INSERT INTO `version` VALUES (247,3);
/*!40000 ALTER TABLE `version` ENABLE KEYS */;
UNLOCK TABLES;
/*!40103 SET TIME_ZONE=@OLD_TIME_ZONE */;

/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;

