Create phamilies
================

After phage genomes are added to PhameratorDB, the new gene products need to be grouped with the other genes into phamilies (“phams”) [Figure 1]. This is accomplished using the k_phamerate.py script (not to be confused with the name of the entire script repository, k_phamerate), which requires only one argument [Table 11].

In general, the script groups genes into phamilies using a kmer-based strategy implemented with kClust [ref Hauser 2013]. First, it keeps track of genes and phams present in the database prior to a new round of clustering. Using all gene products in the database, it runs kClust with sensitivity of ~ 70% identity covering 25% of the two proteins (using the following options: -s 3.53 -c 0.25). New phams are inserted into the database. For each pham, gene products are aligned using kalign. The alignments are used to create markov models of the phams using hhmake, a consensus sequence is created using hhconsensus from the HH-suite software package [ref Remmert 2011], and the output sequences from hhconsensus are re-clustered with kClust (using the following options: -s 0 -c 0.5). The new phams after the second iteration are inserted into the Pham table of the database, and pham colors that reflect unique phams are computed and inserted into the Pham_Color table of the database. The script attempts to maintain consistency of pham designations and colors between rounds of clustering.
